From 4668772cd026b847b6ea6c1b5e389ab91f3b1bbe Mon Sep 17 00:00:00 2001
From: Peter O'Connor <peter@solus-project.com>
Date: Sat, 14 Apr 2018 20:07:25 +1000
Subject: [PATCH] Disable building unshipped code

Signed-off-by: Peter O'Connor <peter@solus-project.com>
---
 CMakeLists.txt | 2 --
 1 file changed, 2 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 71f6d1a..2005cfd 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -80,10 +80,8 @@ if(epoxy_FOUND)
     set(HAVE_EPOXY 1)
 endif()
 
-add_subdirectory(autotests)
 add_subdirectory(src)
 if(KF5Config_FOUND AND KF5I18n_FOUND AND KF5IconThemes_FOUND AND KF5KIO_FOUND)
-    add_subdirectory(tests)
 endif()
 
 # Create a Config.cmake and a ConfigVersion.cmake file and install them
-- 
2.16.3

