diff -ruNb a/docs/faq/Makefile.in b/docs/faq/Makefile.in
--- a/docs/faq/Makefile.in	2013-03-04 13:51:06.000000000 +0000
+++ b/docs/faq/Makefile.in	2013-04-08 12:54:23.504477473 +0100
@@ -602,7 +602,7 @@
 @HAVE_DOCBOOK_TRUE@html:
 @HAVE_DOCBOOK_TRUE@	if test -w $(srcdir); then \
 @HAVE_DOCBOOK_TRUE@	  (cd $(srcdir); \
-@HAVE_DOCBOOK_TRUE@	   db2html gtk-faq.sgml; \
+@HAVE_DOCBOOK_TRUE@	   db2html gtk-faq.sgml -o gtk-faq; \
 @HAVE_DOCBOOK_TRUE@	   test -d html && rm -r html; \
 @HAVE_DOCBOOK_TRUE@	   mv gtk-faq html); \
 @HAVE_DOCBOOK_TRUE@	fi
diff -ruNb a/docs/reference/gtk/Makefile.in b/docs/reference/gtk/Makefile.in
--- a/docs/reference/gtk/Makefile.in	2013-03-04 13:51:06.000000000 +0000
+++ b/docs/reference/gtk/Makefile.in	2013-04-08 12:54:23.521477738 +0100
@@ -787,7 +787,7 @@
 @ENABLE_GTK_DOC_TRUE@@GTK_DOC_BUILD_PDF_TRUE@PDF_BUILD_STAMP = pdf-build.stamp
 
 ########################################################################
-@ENABLE_MAN_TRUE@man_MANS = gtk-query-immodules-2.0.1 gtk-update-icon-cache.1 gtk-builder-convert.1
+man_MANS = gtk-query-immodules-2.0.1 gtk-update-icon-cache.1 gtk-builder-convert.1
 @ENABLE_MAN_TRUE@BUILT_EXTRA_DIST = $(man_MANS)
 all: all-am
 
diff -ruNb a/docs/tutorial/Makefile.in b/docs/tutorial/Makefile.in
--- a/docs/tutorial/Makefile.in	2013-03-04 13:51:06.000000000 +0000
+++ b/docs/tutorial/Makefile.in	2013-04-08 12:54:23.508477536 +0100
@@ -615,7 +615,7 @@
 @HAVE_DOCBOOK_TRUE@html: gtk-tut.sgml
 @HAVE_DOCBOOK_TRUE@	if test -w $(srcdir); then \
 @HAVE_DOCBOOK_TRUE@	  (cd $(srcdir); \
-@HAVE_DOCBOOK_TRUE@	   db2html gtk-tut.sgml; \
+@HAVE_DOCBOOK_TRUE@	   db2html gtk-tut.sgml -o gtk-tut; \
 @HAVE_DOCBOOK_TRUE@	   test -d html && rm -r html; \
 @HAVE_DOCBOOK_TRUE@	   mv gtk-tut html; \
 @HAVE_DOCBOOK_TRUE@	   mkdir html/images; \
