From ecc3b872c291039e5cb7d670cf2026890aa8adb9 Mon Sep 17 00:00:00 2001
From: Joshua Strobl <joshua@streambits.io>
Date: Sat, 26 Dec 2020 22:35:41 +0200
Subject: [PATCH 1/1] Support stateless path.

---
 src/if-options.c | 12 ++++++++----
 1 file changed, 8 insertions(+), 4 deletions(-)

diff --git a/src/if-options.c b/src/if-options.c
index 509db378..3339bdf5 100644
--- a/src/if-options.c
+++ b/src/if-options.c
@@ -2532,10 +2532,14 @@ read_config(struct dhcpcd_ctx *ctx,
 	/* Parse our options file */
 	buflen = dhcp_readfile(ctx, ctx->cffile, buf, sizeof(buf));
 	if (buflen == -1) {
-		/* dhcpcd can continue without it, but no DNS options
-		 * would be requested ... */
-		logerr("%s: %s", __func__, ctx->cffile);
-		return ifo;
+		buflen = dhcp_readfile(ctx, "/usr/share/defaults/dhcpcd/dhcpcd.conf", buf, sizeof(buf));
+
+		if (buflen == -1) {
+			/* dhcpcd can continue without it, but no DNS options
+			* would be requested ... */
+			logerr("%s: %s", __func__, ctx->cffile);
+			return ifo;
+		}
 	}
 	if (buf[buflen - 1] != '\0') {
 		if ((size_t)buflen < sizeof(buf) - 1)
-- 
2.29.2

