From bb1c2757ce48e4fc508bdebe672d8826494f73b5 Mon Sep 17 00:00:00 2001
From: Ikey Doherty <ikey@solus-project.com>
Date: Mon, 29 Jun 2015 09:30:48 +0100
Subject: [PATCH] Do not enforce a hostname change

Signed-off-by: Ikey Doherty <ikey@solus-project.com>
---
 dhcpcd.conf | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/dhcpcd.conf b/dhcpcd.conf
index eb625a7..632d848 100644
--- a/dhcpcd.conf
+++ b/dhcpcd.conf
@@ -9,7 +9,7 @@ hostname
 #clientid
 
 # A list of options to request from the DHCP server.
-option domain_name_servers, domain_name, domain_search, host_name
+option domain_name_servers, domain_name, domain_search
 option classless_static_routes
 # Most distributions have NTP support.
 option ntp_servers
-- 
2.4.5

