From 9a77e36440d71e6c701bfcf19f392109a4dcec37 Mon Sep 17 00:00:00 2001
From: Joey Riches <josephriches@gmail.com>
Date: Mon, 11 Jan 2021 18:22:17 +0000
Subject: [PATCH 1/1] fix path

Signed-off-by: Joey Riches <josephriches@gmail.com>
---
 docs/manpages/meson.build | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/docs/manpages/meson.build b/docs/manpages/meson.build
index f90c757..101cbca 100644
--- a/docs/manpages/meson.build
+++ b/docs/manpages/meson.build
@@ -32,7 +32,7 @@ foreach m : manpages
     command: [xsltproc,
               '--output', '@OUTPUT@',
               '--stringparam', 'man.authors.section.enabled', '0',
-              '/etc/asciidoc/docbook-xsl/manpage.xsl', '@INPUT@'],
+              '/usr/share/asciidoc/docbook-xsl/manpage.xsl', '@INPUT@'],
     input: xml,
     output: manpage,
     install: true,
-- 
2.30.0

