From 3bcfd58951ae30d35f27da9aed210926fad648db Mon Sep 17 00:00:00 2001
From: Joey Riches <josephriches@gmail.com>
Date: Mon, 10 Jan 2022 14:44:10 +0000
Subject: [PATCH 1/1] Print out failing lstat's

---
 abi/report.go | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/abi/report.go b/abi/report.go
index 78f25a8..6e33eec 100644
--- a/abi/report.go
+++ b/abi/report.go
@@ -122,6 +122,7 @@ func (r Report) Save(path string) error {
 func (r Report) Add(root, path string) error {
 	info, err := os.Lstat(path)
 	if err != nil {
+		fmt.Fprintf(os.Stderr, "error: %s\n", path)
 		return nil
 	}
 	return r.walkPivot(root, path, info)
@@ -130,6 +131,7 @@ func (r Report) Add(root, path string) error {
 func (r Report) walkDir(root, path string) error {
 	infos, err := ioutil.ReadDir(path)
 	if err != nil {
+        fmt.Fprintln(os.Stderr, err.Error())
 		return err
 	}
 	for _, info := range infos {
@@ -166,6 +168,7 @@ func (r Report) walkSym(root, path string) error {
 	} // implicit else: abs path outside root
 	info, err := os.Lstat(link)
 	if err != nil {
+		fmt.Fprintf(os.Stderr, "error: %s\n", link)
 		return nil
 	}
 	return r.walkPivot(root, link, info)
-- 
2.34.0

