From fad7672189425230d1f17b075e0411c277a8da7d Mon Sep 17 00:00:00 2001
From: Peter O'Connor <peter@solus-project.com>
Date: Wed, 4 Apr 2018 10:57:38 +1000
Subject: [PATCH] Disable building unshipped code

Signed-off-by: Peter O'Connor <peter@solus-project.com>
---
 CMakeLists.txt | 2 --
 1 file changed, 2 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 47c4687..08a903d 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -719,8 +719,6 @@ ecm_install_icons(
 )
 
 add_subdirectory(qml)
-add_subdirectory(autotests)
-add_subdirectory(tests)
 add_subdirectory(packageplugins)
 
 if (KF5DocTools_FOUND)
-- 
2.16.3

