From 29308ee50d211f1209258061b25976bc36e14820 Mon Sep 17 00:00:00 2001
From: Troy Harvey <harveydevel@gmail.com>
Date: Sun, 23 Jan 2022 10:53:26 +1100
Subject: [PATCH] Fix zsh and fish

---
 data/scripts/Xsession | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/data/scripts/Xsession b/data/scripts/Xsession
index 5449810..27f8887 100755
--- a/data/scripts/Xsession
+++ b/data/scripts/Xsession
@@ -21,7 +21,7 @@ case $SHELL in
     ;;
 */zsh)
     [ -z "$ZSH_NAME" ] && exec $SHELL $0 "$@"
-    [ -d /etc/zsh ] && zdir=/etc/zsh || zdir=/etc
+    [ -d /usr/share/defaults/etc/zsh ] && zdir=/usr/share/defaults/etc/zsh
     zhome=${ZDOTDIR:-$HOME}
     # zshenv is always sourced automatically.
     [ -f $zdir/zprofile ] && . $zdir/zprofile
@@ -39,6 +39,7 @@ case $SHELL in
     rm -f $xsess_tmp
     ;;
   */fish)
+    [ -f /usr/share/defaults/etc/profile ] && . /usr/share/defaults/etc/profile
     xsess_tmp=`mktemp /tmp/xsess-env-XXXXXX`
     $SHELL --login -c "/bin/sh -c 'export -p' > $xsess_tmp"
     . $xsess_tmp
-- 
2.34.0

