From e45d53d7d8a152dcdfa43e8e902ff59c993f812d Mon Sep 17 00:00:00 2001
From: Joshua Strobl <joshua@streambits.io>
Date: Fri, 17 Jul 2020 14:12:04 +0300
Subject: [PATCH 1/1] Source /usr/share/defaults/etc/xprofile

---
 data/scripts/Xsession | 1 +
 1 file changed, 1 insertion(+)

diff --git a/data/scripts/Xsession b/data/scripts/Xsession
index db32dab..311c816 100755
--- a/data/scripts/Xsession
+++ b/data/scripts/Xsession
@@ -51,6 +51,7 @@ case $SHELL in
     ;;
 esac
 
+[ -f /usr/share/defaults/etc/xprofile ] && . /usr/share/defaults/etc/xprofile
 [ -f /etc/xprofile ] && . /etc/xprofile
 [ -f $HOME/.xprofile ] && . $HOME/.xprofile
 
-- 
2.27.0

