From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Reilly Brogan <reilly@reillybrogan.com>
Date: Mon, 26 Feb 2024 11:38:38 -0600
Subject: [PATCH] Support stateless dir for xinitrc.d files

---
 data/scripts/Xsession | 9 +++++++++
 1 file changed, 9 insertions(+)

diff --git a/data/scripts/Xsession b/data/scripts/Xsession
index d5139eb..fa07877 100755
--- a/data/scripts/Xsession
+++ b/data/scripts/Xsession
@@ -44,6 +44,15 @@ esac
 [ -f /usr/local/etc/xprofile ] && . /usr/local/etc/xprofile
 [ -f $HOME/.xprofile ] && . $HOME/.xprofile
 
+# run all system xinitrc shell scripts that are in the vendordir
+if [ -d /usr/share/defaults/etc/X11/xinit/xinitrc.d ]; then
+  for i in /usr/share/defaults/etc/X11/xinit/xinitrc.d/* ; do
+  if [ -x "$i" ]; then
+    . "$i"
+  fi
+  done
+fi
+
 # run all system xinitrc shell scripts.
 if [ -d /etc/X11/xinit/xinitrc.d ]; then
   for i in /etc/X11/xinit/xinitrc.d/* ; do
