From 991c9f733e654376616f8907017db8d7d713ecb1 Mon Sep 17 00:00:00 2001
From: "F. von Gellhorn" <flinux@vongellhorn.ch>
Date: Wed, 12 Feb 2020 18:57:09 +0100
Subject: [PATCH 1/1] Fix zsh for Solus

---
 data/scripts/Xsession | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/data/scripts/Xsession b/data/scripts/Xsession
index c7fa214..072d783 100755
--- a/data/scripts/Xsession
+++ b/data/scripts/Xsession
@@ -21,7 +21,7 @@ case $SHELL in
     ;;
 */zsh)
     [ -z "$ZSH_NAME" ] && exec $SHELL $0 "$@"
-    [ -d /etc/zsh ] && zdir=/etc/zsh || zdir=/etc
+    [ -d /etc/zsh ] || [ -d /usr/share/zsh ] && zdir=/etc/zsh || zdir=/etc || zdir=/usr/share/zsh
     zhome=${ZDOTDIR:-$HOME}
     # zshenv is always sourced automatically.
     [ -f $zdir/zprofile ] && . $zdir/zprofile
-- 
2.25.0

