From b0d48cbbea8cc4259402b3e38d1b44abae30d3de Mon Sep 17 00:00:00 2001
From: "F. von Gellhorn" <flinux@vongellhorn.ch>
Date: Tue, 20 Apr 2021 17:27:12 +0200
Subject: [PATCH 2/2] Fix session started as the wrong type on autologin

---
 src/daemon/Display.cpp | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/src/daemon/Display.cpp b/src/daemon/Display.cpp
index b95f6e5..9f1fabc 100644
--- a/src/daemon/Display.cpp
+++ b/src/daemon/Display.cpp
@@ -245,6 +245,11 @@ namespace SDDM {
     }
 
     bool Display::findSessionEntry(const QDir &dir, const QString &name) const {
+        // Given an absolute path: Check that it matches dir
+        const QFileInfo fileInfo(name);
+        if (fileInfo.isAbsolute() && fileInfo.absolutePath() != dir.absolutePath())
+            return false;
+
         QString fileName = name;
 
         // append extension
-- 
2.30.2

