From dceea3f8b880c4eaefdb9647fe7755ac5a658ee5 Mon Sep 17 00:00:00 2001
From: Troy Harvey <harveydev@protonmail.com>
Date: Thu, 20 Feb 2020 22:26:39 +1100
Subject: [PATCH] Fix zsh for solus

---
 data/scripts/Xsession | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/data/scripts/Xsession b/data/scripts/Xsession
index c7fa214..658c3a9 100755
--- a/data/scripts/Xsession
+++ b/data/scripts/Xsession
@@ -21,7 +21,7 @@ case $SHELL in
     ;;
 */zsh)
     [ -z "$ZSH_NAME" ] && exec $SHELL $0 "$@"
-    [ -d /etc/zsh ] && zdir=/etc/zsh || zdir=/etc
+    [ -d /usr/share/defaults/etc/zsh ] && zdir=/usr/share/defaults/etc/zsh
     zhome=${ZDOTDIR:-$HOME}
     # zshenv is always sourced automatically.
     [ -f $zdir/zprofile ] && . $zdir/zprofile
-- 
2.25.1

