From 8597f66e69c3716edee7a95dd9f5a7a538524430 Mon Sep 17 00:00:00 2001
From: "F. von Gellhorn" <flinux@vongellhorn.ch>
Date: Fri, 22 Feb 2019 14:14:44 +0100
Subject: [PATCH 1/1] Uncomment the zsh part, somehow this brakes with KF5
 5.55.0

---
 data/scripts/Xsession | 22 +++++++++++-----------
 1 file changed, 11 insertions(+), 11 deletions(-)

diff --git a/data/scripts/Xsession b/data/scripts/Xsession
index c7fa214..1f7eeae 100755
--- a/data/scripts/Xsession
+++ b/data/scripts/Xsession
@@ -19,17 +19,17 @@ case $SHELL in
       . $HOME/.profile
     fi
     ;;
-*/zsh)
-    [ -z "$ZSH_NAME" ] && exec $SHELL $0 "$@"
-    [ -d /etc/zsh ] && zdir=/etc/zsh || zdir=/etc
-    zhome=${ZDOTDIR:-$HOME}
-    # zshenv is always sourced automatically.
-    [ -f $zdir/zprofile ] && . $zdir/zprofile
-    [ -f $zhome/.zprofile ] && . $zhome/.zprofile
-    [ -f $zdir/zlogin ] && . $zdir/zlogin
-    [ -f $zhome/.zlogin ] && . $zhome/.zlogin
-    emulate -R sh
-    ;;
+#*/zsh)
+#    [ -z "$ZSH_NAME" ] && exec $SHELL $0 "$@"
+#    [ -d /etc/zsh ] && zdir=/etc/zsh || zdir=/etc
+#    zhome=${ZDOTDIR:-$HOME}
+#    # zshenv is always sourced automatically.
+#    [ -f $zdir/zprofile ] && . $zdir/zprofile
+#    [ -f $zhome/.zprofile ] && . $zhome/.zprofile
+#    [ -f $zdir/zlogin ] && . $zdir/zlogin
+#    [ -f $zhome/.zlogin ] && . $zhome/.zlogin
+#    emulate -R sh
+#    ;;
   */csh|*/tcsh)
     # [t]cshrc is always sourced automatically.
     # Note that sourcing csh.login after .cshrc is non-standard.
-- 
2.20.1

