From 6d395d02180e483bf57248948c13529cac39e8d2 Mon Sep 17 00:00:00 2001
From: Jocelyn Falempe <jfalempe@redhat.com>
Date: Thu, 26 Mar 2026 19:43:51 +0100
Subject: [PATCH] sddm.service: conflicts with kmsconvt@ttyX

To prevent kmscon to start on the same tty as sddm, add kmsconvt
to the conflicts list, like getty.

Signed-off-by: Jocelyn Falempe <jfalempe@redhat.com>
---
 services/sddm.service.in | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/services/sddm.service.in b/services/sddm.service.in
index 0b464439c..a84874fae 100644
--- a/services/sddm.service.in
+++ b/services/sddm.service.in
@@ -1,8 +1,8 @@
 [Unit]
 Description=Simple Desktop Display Manager
 Documentation=man:sddm(1) man:sddm.conf(5)
-Conflicts=getty@tty${SDDM_INITIAL_VT}.service
-After=systemd-user-sessions.service getty@tty${SDDM_INITIAL_VT}.service plymouth-quit.service systemd-logind.service
+Conflicts=getty@tty${SDDM_INITIAL_VT}.service kmsconvt@tty${SDDM_INITIAL_VT}.service
+After=systemd-user-sessions.service getty@tty${SDDM_INITIAL_VT}.service kmsconvt@tty${SDDM_INITIAL_VT}.service plymouth-quit.service systemd-logind.service
 PartOf=graphical.target
 StartLimitIntervalSec=30
 StartLimitBurst=2
