From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Reilly Brogan <reilly@reillybrogan.com>
Date: Wed, 26 Jun 2024 23:05:14 -0500
Subject: [PATCH] Fix compat with MINIUPNPC API Version 18

---
 src/network/C4Network2UPnPLinux.cpp | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/src/network/C4Network2UPnPLinux.cpp b/src/network/C4Network2UPnPLinux.cpp
index 8cad5b707..385af83bd 100644
--- a/src/network/C4Network2UPnPLinux.cpp
+++ b/src/network/C4Network2UPnPLinux.cpp
@@ -77,7 +77,11 @@ void C4Network2UPnPP::Init()
 	if ((devlist = upnpDiscover(2000, nullptr, nullptr, UPNP_LOCAL_PORT_ANY, 0, 2, &error)))
 #endif
 	{
+#if MINIUPNPC_API_VERSION >= 18
+		if ((status = UPNP_GetValidIGD(devlist, &upnp_urls, &igd_data, lanaddr, sizeof(lanaddr), nullptr, 0)))
+#else
 		if ((status = UPNP_GetValidIGD(devlist, &upnp_urls, &igd_data, lanaddr, sizeof(lanaddr))))
+#endif
 		{
 			ThreadLogS("UPnP: Found IGD %s (status %d)", upnp_urls.controlURL, status);
 			initialized = true;
