From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Reilly Brogan <reilly@reillybrogan.com>
Date: Sat, 7 Sep 2024 22:03:24 -0500
Subject: [PATCH] ftbfs gcc-14

---
 src/zlib/gzio.c  | 1 +
 src/zlib/zutil.h | 1 +
 2 files changed, 2 insertions(+)

diff --git a/src/zlib/gzio.c b/src/zlib/gzio.c
index b9985423a..2b0a62833 100644
--- a/src/zlib/gzio.c
+++ b/src/zlib/gzio.c
@@ -15,6 +15,7 @@
 #define _BSD_SOURCE       /* for vsnprintf */
 
 #include <stdio.h>
+#include <string.h>
 
 #include "zutil.h"
 
diff --git a/src/zlib/zutil.h b/src/zlib/zutil.h
index 0f916be0a..720976350 100644
--- a/src/zlib/zutil.h
+++ b/src/zlib/zutil.h
@@ -231,6 +231,7 @@ extern const char * const z_errmsg[10]; /* indexed by 2-zlib_error */
 #    define zmemcmp _fmemcmp
 #    define zmemzero(dest, len) _fmemset(dest, 0, len)
 #  else
+#    include <string.h>
 #    define zmemcpy memcpy
 #    define zmemcmp memcmp
 #    define zmemzero(dest, len) memset(dest, 0, len)
