diff --git a/CMakeLists.txt b/CMakeLists.txt
index ccd2973..22d6238 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -167,7 +167,7 @@ if(WIN32 AND MINGW)
 endif()
 
 if(UNIX)
-	add_definitions("-DOC_SYSTEM_DATA_DIR=\"${CMAKE_INSTALL_PREFIX}/share/games/openclonk\"")
+	add_definitions("-DOC_SYSTEM_DATA_DIR=\"${CMAKE_INSTALL_PREFIX}/share/openclonk\"")
 endif()
 
 if(APPLE)
@@ -1537,13 +1537,13 @@ foreach(group ${OC_C4GROUPS})
 			DEPENDS "${native_c4group}"
 			VERBATIM
 		)
-		install(FILES ${CMAKE_CURRENT_BINARY_DIR}/${group} DESTINATION share/games/openclonk)
+		install(FILES ${CMAKE_CURRENT_BINARY_DIR}/${group} DESTINATION share/openclonk)
 	endif()
 endforeach()
 
 if(NOT APPLE)
 	if(NOT HEADLESS_ONLY)
-		install(DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR}/planet/Music.ocg DESTINATION share/games/openclonk)
+		install(DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR}/planet/Music.ocg DESTINATION share/openclonk)
 	endif()
 	# group files (game data)
 	add_custom_target(groups DEPENDS ${OC_C4GROUPS})
@@ -1563,7 +1563,7 @@ if(NOT HEADLESS_ONLY)
 		install(FILES ${CMAKE_CURRENT_SOURCE_DIR}/openclonk.appdata.xml DESTINATION share/appdata)
 
 		# Install binaries
-		install(TARGETS openclonk DESTINATION games)
+		install(TARGETS openclonk DESTINATION /usr/bin)
 	else()
 		install(TARGETS openclonk
 			BUNDLE DESTINATION .
