diff --git a/tecmake.mak b/tecmake.mak
index 48991a4..c359896 100644
--- a/tecmake.mak
+++ b/tecmake.mak
@@ -297,6 +297,9 @@ endif
 
 ifdef GTK_DEFAULT
   ifndef USE_GTK2
+    ifneq ($(findstring Linux6, $(TEC_UNAME)), )
+      USE_GTK3 = Yes
+    endif
     ifneq ($(findstring Linux5, $(TEC_UNAME)), )
       USE_GTK3 = Yes
     endif
@@ -1210,6 +1213,9 @@ ifdef USE_CD
   ifneq ($(findstring Linux5, $(TEC_UNAME)), )
     LIBS += fontconfig
   endif
+  ifneq ($(findstring Linux6, $(TEC_UNAME)), )
+    LIBS += fontconfig
+  endif
   ifneq ($(findstring cygw, $(TEC_UNAME)), )
     LIBS += fontconfig
   endif
@@ -1245,22 +1251,26 @@ ifdef USE_IM
 endif
 
 ifdef LINK_WEBKIT
-  ifneq ($(findstring Linux5, $(TEC_UNAME)), )
+  ifneq ($(findstring Linux6, $(TEC_UNAME)), )
     LIBS += webkit2gtk-4.0 gio-2.0
-  else 
-    ifneq ($(findstring Linux4, $(TEC_UNAME)), )
+  else
+    ifneq ($(findstring Linux5, $(TEC_UNAME)), )
       LIBS += webkit2gtk-4.0 gio-2.0
     else 
-      ifneq ($(findstring Linux3, $(TEC_UNAME)), )
-        ifdef USE_GTK3
-          LIBS += webkitgtk-3.0
+      ifneq ($(findstring Linux4, $(TEC_UNAME)), )
+        LIBS += webkit2gtk-4.0 gio-2.0
+      else 
+        ifneq ($(findstring Linux3, $(TEC_UNAME)), )
+          ifdef USE_GTK3
+            LIBS += webkitgtk-3.0
+          else
+            LIBS += webkitgtk-1.0
+          endif
         else
-          LIBS += webkitgtk-1.0
+          LIBS += webkit-1.0
         endif
-      else
-        LIBS += webkit-1.0
-      endif
-    endif 
+      endif 
+    endif
   endif
 endif
 
@@ -1479,6 +1489,9 @@ ifdef USE_GTK
     ifneq ($(findstring Linux5, $(TEC_UNAME)), )
       STDINCS += /usr/include/harfbuzz
     endif
+    ifneq ($(findstring Linux6, $(TEC_UNAME)), )
+      STDINCS += /usr/include/harfbuzz
+    endif
   endif
 endif
 
