diff --git a/src/cdlua5.mak b/src/cdlua5.mak
index e6631ef..0c2ba4e 100755
--- a/src/cdlua5.mak
+++ b/src/cdlua5.mak
@@ -31,7 +31,7 @@ ifdef USE_LUA54
   LIBNAME := $(LIBNAME)54
 else
 ifdef USE_LUA53
-  LIBNAME := $(LIBNAME)53
+  LIBNAME := $(LIBNAME)
 else
 ifdef USE_LUA52
   LIBNAME := $(LIBNAME)52
diff --git a/src/cdluacairo5.mak b/src/cdluacairo5.mak
index dea11d2..23bc79e 100755
--- a/src/cdluacairo5.mak
+++ b/src/cdluacairo5.mak
@@ -33,7 +33,7 @@ ifdef USE_LUA54
   LIBNAME := $(LIBNAME)54
 else
 ifdef USE_LUA53
-  LIBNAME := $(LIBNAME)53
+  LIBNAME := $(LIBNAME)
 else
 ifdef USE_LUA52
   LIBNAME := $(LIBNAME)52
diff --git a/src/cdluacontextplus5.mak b/src/cdluacontextplus5.mak
index bdb05d8..16b6974 100755
--- a/src/cdluacontextplus5.mak
+++ b/src/cdluacontextplus5.mak
@@ -43,7 +43,7 @@ ifdef USE_LUA54
   LIBNAME := $(LIBNAME)54
 else
 ifdef USE_LUA53
-  LIBNAME := $(LIBNAME)53
+  LIBNAME := $(LIBNAME)
 else
 ifdef USE_LUA52
   LIBNAME := $(LIBNAME)52
diff --git a/src/cdluadirect2d5.mak b/src/cdluadirect2d5.mak
index 244a52e..9252d53 100755
--- a/src/cdluadirect2d5.mak
+++ b/src/cdluadirect2d5.mak
@@ -33,7 +33,7 @@ ifdef USE_LUA54
   LIBNAME := $(LIBNAME)54
 else
 ifdef USE_LUA53
-  LIBNAME := $(LIBNAME)53
+  LIBNAME := $(LIBNAME)
 else
 ifdef USE_LUA52
   LIBNAME := $(LIBNAME)52
diff --git a/src/cdluagl5.mak b/src/cdluagl5.mak
index 61ed576..fa4f76c 100755
--- a/src/cdluagl5.mak
+++ b/src/cdluagl5.mak
@@ -33,7 +33,7 @@ ifdef USE_LUA54
   LIBNAME := $(LIBNAME)54
 else
 ifdef USE_LUA53
-  LIBNAME := $(LIBNAME)53
+  LIBNAME := $(LIBNAME)
 else
 ifdef USE_LUA52
   LIBNAME := $(LIBNAME)52
diff --git a/src/cdluaim5.mak b/src/cdluaim5.mak
index 4858162..8fa930d 100755
--- a/src/cdluaim5.mak
+++ b/src/cdluaim5.mak
@@ -32,7 +32,7 @@ ifdef USE_LUA54
   LIBNAME := $(LIBNAME)54
 else
 ifdef USE_LUA53
-  LIBNAME := $(LIBNAME)53
+  LIBNAME := $(LIBNAME)
 else
 ifdef USE_LUA52
   LIBNAME := $(LIBNAME)52
diff --git a/src/cdluapdf5.mak b/src/cdluapdf5.mak
index fae16fb..f7f0efd 100755
--- a/src/cdluapdf5.mak
+++ b/src/cdluapdf5.mak
@@ -33,7 +33,7 @@ ifdef USE_LUA54
   LIBNAME := $(LIBNAME)54
 else
 ifdef USE_LUA53
-  LIBNAME := $(LIBNAME)53
+  LIBNAME := $(LIBNAME)
 else
 ifdef USE_LUA52
   LIBNAME := $(LIBNAME)52
diff --git a/tecmake.mak b/tecmake.mak
index 48991a4..b712a36 100755
--- a/tecmake.mak
+++ b/tecmake.mak
@@ -478,12 +478,12 @@ endif
 ifdef USE_OLDLIBLUA
   TEC_UNAME_LIBLUA_DIR ?= $(TEC_UNAME_LIB_DIR)
 else
-  TEC_UNAME_LIBLUA_DIR ?= $(TEC_UNAME_LIB_DIR)/Lua$(LIBLUA_SFX)
+  TEC_UNAME_LIBLUA_DIR ?= $(TEC_UNAME_LIB_DIR)/Lua
 endif
 
 # Subfolder for Lua Modules
 ifdef LUAMOD_DIR
-  TEC_UNAME_DIR := $(TEC_UNAME_DIR)/Lua$(LIBLUA_SFX)
+  TEC_UNAME_DIR := $(TEC_UNAME_DIR)/Lua
 endif
 
 OBJDIR := $(OBJROOT)/$(TEC_UNAME_DIR)
@@ -941,9 +941,9 @@ ifdef USE_IUPCONTROLS
   
   ifdef USE_IUPLUA
     ifdef USE_STATIC
-      SLIB += $(IUPLUA_LIB)/libiupluacontrols$(LIBLUA_SFX).a
+      SLIB += $(IUPLUA_LIB)/libiupluacontrols.a
     else
-      LIBS += iupluacontrols$(LIBLUA_SFX)
+      LIBS += iupluacontrols
     endif
     override USE_CDLUA = Yes
   endif
@@ -964,9 +964,9 @@ ifdef USE_IUPGLCONTROLS
   
   ifdef USE_IUPLUA
     ifdef USE_STATIC
-      SLIB += $(IUPLUA_LIB)/libiupluaglcontrols$(LIBLUA_SFX).a
+      SLIB += $(IUPLUA_LIB)/libiupluaglcontrols.a
     else
-      LIBS += iupluaglcontrols$(LIBLUA_SFX)
+      LIBS += iupluaglcontrols
     endif
   endif
   
@@ -983,9 +983,9 @@ ifdef USE_IUPWEB
   
   ifdef USE_IUPLUA
     ifdef USE_STATIC
-      SLIB += $(IUPLUA_LIB)/libiupluaweb$(LIBLUA_SFX).a
+      SLIB += $(IUPLUA_LIB)/libiupluaweb.a
     else
-      LIBS += iupluaweb$(LIBLUA_SFX)
+      LIBS += iupluaweb
     endif
   endif
   
@@ -1000,9 +1000,9 @@ ifdef USE_IMLUA
   override USE_IM = Yes
   IMLUA_LIB ?= $(IM)/lib/$(TEC_UNAME_LIBLUA_DIR)
   ifdef USE_STATIC
-    SLIB += $(IMLUA_LIB)/libimlua$(LIBLUA_SFX).a
+    SLIB += $(IMLUA_LIB)/libimlua.a
   else
-    LIBS += imlua$(LIBLUA_SFX)
+    LIBS += imlua
     LDIR += $(IMLUA_LIB)
   endif
 endif
@@ -1011,9 +1011,9 @@ ifdef USE_CDLUA
   override USE_CD = Yes
   CDLUA_LIB ?= $(CD)/lib/$(TEC_UNAME_LIBLUA_DIR)
   ifdef USE_STATIC
-    SLIB += $(CDLUA_LIB)/libcdlua$(LIBLUA_SFX).a
+    SLIB += $(CDLUA_LIB)/libcdlua.a
   else
-    LIBS += cdlua$(LIBLUA_SFX)
+    LIBS += cdlua
     LDIR += $(CDLUA_LIB)
   endif
 endif
@@ -1025,27 +1025,27 @@ ifdef USE_IUPLUA
   ifdef USE_STATIC
     ifdef USE_CD
       ifeq ($(findstring iupluacd, $(LIBNAME)), )
-        SLIB += $(IUPLUA_LIB)/libiupluacd$(LIBLUA_SFX).a
+        SLIB += $(IUPLUA_LIB)/libiupluacd.a
       endif
     endif
     ifdef USE_OPENGL
       ifeq ($(findstring iupluagl, $(LIBNAME)), )
-        SLIB += $(IUPLUA_LIB)/libiupluagl$(LIBLUA_SFX).a
+        SLIB += $(IUPLUA_LIB)/libiupluagl.a
       endif
     endif
-    SLIB += $(IUPLUA_LIB)/libiuplua$(LIBLUA_SFX).a
+    SLIB += $(IUPLUA_LIB)/libiuplua.a
   else
     ifdef USE_CD
       ifeq ($(findstring iupluacd, $(LIBNAME)), )
-        LIBS += iupluacd$(LIBLUA_SFX)
+        LIBS += iupluacd
       endif
     endif
     ifdef USE_OPENGL
       ifeq ($(findstring iupluagl, $(LIBNAME)), )
-        LIBS += iupluagl$(LIBLUA_SFX)
+        LIBS += iupluagl
       endif
     endif
-    LIBS += iuplua$(LIBLUA_SFX)
+    LIBS += iuplua
     LDIR += $(IUPLUA_LIB)
   endif
 endif
@@ -1054,19 +1054,19 @@ ifdef USE_LUA
   LUA_LIB ?= $(LUA)/lib/$(TEC_UNAME)
   ifdef USE_STATIC
     ifndef NO_LUALIB
-      SLIB += $(LUA_LIB)/liblualib$(LUA_SFX).a
+      SLIB += $(LUA_LIB)/liblualib.a
     endif
-    SLIB += $(LUA_LIB)/liblua$(LUA_SFX).a
+    SLIB += $(LUA_LIB)/liblua.a
   else
     ifndef NO_LUALIB
-      LIBS += lualib$(LUA_SFX)
+      LIBS += lualib
     endif
     ifndef NO_LUALINK
-        LIBS += lua$(LUA_SFX)
+        LIBS += lua
         LDIR += $(LUA_LIB)
     else
       ifneq ($(findstring cygw, $(TEC_UNAME)), )
-        LIBS += lua$(LUA_SFX)
+        LIBS += lua
         LDIR += $(LUA_LIB)
       endif
     endif
@@ -1077,12 +1077,12 @@ ifdef USE_LUA
 
   LUA_BIN ?= $(LUA)/bin/$(TEC_UNAME)
   ifdef USE_BIN2C_LUA
-    BIN2C := $(LUA_BIN)/lua$(LUA_SFX) $(BIN2C_PATH)bin2c.lua
+    BIN2C := $(LUA_BIN)/lua $(BIN2C_PATH)bin2c.lua
   else
-    BIN2C := $(LUA_BIN)/bin2c$(LUA_SFX)
+    BIN2C := $(LUA_BIN)/bin2c
   endif
-  LUAC   := $(LUA_BIN)/luac$(LUA_SFX)
-  LUABIN := $(LUA_BIN)/lua$(LUA_SFX)
+  LUAC   := $(LUA_BIN)/luac
+  LUABIN := $(LUA_BIN)/lua
 endif
 
 ifdef USE_IUP
