From 62cf7443acb8ab9caaf00688802be50ed9caee10 Mon Sep 17 00:00:00 2001
From: Ikey Doherty <ikey@solus-project.com>
Date: Fri, 20 Apr 2018 08:47:14 +0100
Subject: [PATCH] intercept: Support new sonames in Feral's Tomb Raider port

Signed-off-by: Ikey Doherty <ikey@solus-project.com>
---
 src/intercept/main.c | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/src/intercept/main.c b/src/intercept/main.c
index 9f6e4c0..aa59a99 100644
--- a/src/intercept/main.c
+++ b/src/intercept/main.c
@@ -262,6 +262,8 @@ static const char *vendor_transmute_source[] = {
 #ifdef LSI_OVERRIDE_LIBRESSL
         "libcrypto.so.36",
         "libssl.so.37",
+        "libcrypto.so.42",
+        "libssl.so.44",
 #endif
 
         /* old name for openal */
@@ -297,9 +299,13 @@ static const char *vendor_transmute_target[] = {
 #if defined(LSI_LIBRESSL_MODE_SHIM)
         "libcrypto" LSI_LIBRESSL_SUFFIX ".so",
         "libssl" LSI_LIBRESSL_SUFFIX ".so",
+        "libcrypto" LSI_LIBRESSL_SUFFIX ".so",
+        "libssl" LSI_LIBRESSL_SUFFIX ".so",
 #elif defined(LSI_LIBRESSL_MODE_NATIVE)
         "libcrypto.so.1.0.0",
         "libssl.so.1.0.0",
+        "libcrypto.so.1.0.0",
+        "libssl.so.1.0.0",
 #endif
 
         /* new name for openal */
-- 
2.16.3

