From baa421a186e4d8156f4b0a0bfc0fd9187f5f98f6 Mon Sep 17 00:00:00 2001
From: "F. von Gellhorn" <flinux@vongellhorn.ch>
Date: Tue, 22 Jan 2019 23:26:13 +0100
Subject: [PATCH 1/1] play nice with solus

---
 src/shim/shim.c | 16 ++++++++--------
 1 file changed, 8 insertions(+), 8 deletions(-)

diff --git a/src/shim/shim.c b/src/shim/shim.c
index ff4d882..b6fec8d 100644
--- a/src/shim/shim.c
+++ b/src/shim/shim.c
@@ -183,6 +183,8 @@ static void shim_export_extra(const char *prefix)
          * This allows us to handle the special-case multiarch mounts.
          */
         static const char *ld_library_dirs[] = {
+                "/usr/lib/glx-provider/default",   /**<Solus mesa, 64-bit */
+                "/usr/lib32/glx-provider/default", /**<Solus mesa, 32bit */
                 "/var/lib/snapd/lib/gl/vdpau",   /**<64-bit vdpau */
                 "/var/lib/snapd/lib/gl32/vdpau", /**<32bit vdpau */
                 "/var/lib/snapd/lib/gl",         /**<Potential host NVIDIA libraries */
@@ -338,23 +340,21 @@ bool shim_bootstrap()
 static int shim_execute_internal(const char *command, int argc, char **argv, bool use_path)
 {
         bool is_x86_64;
-        const char *n_argv[argc + 5];
+        const char *n_argv[argc + 3];
         const char *exec_command = NULL;
         int i = 1;
         int8_t off = 1;
         int (*vfunc)(const char *, char *const argv[]) = NULL;
 
         is_x86_64 = lsi_system_is_64bit();
-        memset(&n_argv, 0, sizeof(char *) * (argc + 5));
+        memset(&n_argv, 0, sizeof(char *) * (argc + 3));
 
         /* If we're 64-bit and 32-bit is forced, proxy via setarch linux32 */
         if (lsi_config.force_32 && is_x86_64) {
-                exec_command = "setarch";
-                n_argv[0] = "setarch";
-                n_argv[1] = "linux32";
-                n_argv[2] = "--";
-                n_argv[3] = command;
-                off = 4;
+                exec_command = "linux32";
+                n_argv[0] = "linux32";
+                n_argv[1] = command;
+                off = 2;
                 /* Use setarch in the path */
                 vfunc = execvp;
         } else {
-- 
2.20.1

