From 63496527401d476223847a4880aebc4213aef43b Mon Sep 17 00:00:00 2001
From: Ikey Doherty <ikey@solus-project.com>
Date: Fri, 13 Oct 2017 02:48:52 +0100
Subject: [PATCH 2/9] intercept: Clamp down on libGL{,ES}, libGLU*, and libGLEW
 offenders

Signed-off-by: Ikey Doherty <ikey@solus-project.com>
---
 src/intercept/main.c | 10 ++++++++++
 1 file changed, 10 insertions(+)

diff --git a/src/intercept/main.c b/src/intercept/main.c
index 2886e86..5d38983 100644
--- a/src/intercept/main.c
+++ b/src/intercept/main.c
@@ -107,6 +107,16 @@ static const char *vendor_blacklist[] = {
 
         /* general problem causer. */
         "libopenal.so.",
+
+        /* thou shalt not replace drivers (swiftshader) ! */
+        "libGLESv2.so",
+        "libGL.so",
+
+        /* glews (provide glew + glew110 in your distro for full compat) */
+        "libGLEW.so.",
+
+        /* libglu has stable soname */
+        "libGLU.so.",
 };
 
 /**
-- 
2.14.2

