From fa41b2c5726fe5c8b32af07fdd95f0ff7fa4e325 Mon Sep 17 00:00:00 2001
From: Ikey Doherty <ikey@solus-project.com>
Date: Fri, 13 Oct 2017 15:43:27 +0100
Subject: [PATCH 5/8] intercept: Only emit soname change when LSI_DEBUG is set

Signed-off-by: Ikey Doherty <ikey@solus-project.com>
---
 src/intercept/main.c | 7 ++++++-
 1 file changed, 6 insertions(+), 1 deletion(-)

diff --git a/src/intercept/main.c b/src/intercept/main.c
index c097221..b8a01ae 100644
--- a/src/intercept/main.c
+++ b/src/intercept/main.c
@@ -296,7 +296,12 @@ static bool lsi_override_soname(unsigned int flag, const char *orig_name, const
                         continue;
                 }
                 *soname = vendor_transmute_target[i];
-                emit_replaced_name(orig_name, *soname);
+
+                /* If LSI_DEBUG is set, spam it. */
+                if (getenv("LSI_DEBUG")) {
+                        emit_replaced_name(orig_name, *soname);
+                }
+
                 return true;
         }
 
-- 
2.14.2

