From 071304d8bd4df318b66bec26cde2839b48813e72 Mon Sep 17 00:00:00 2001
From: Ikey Doherty <ikey@solus-project.com>
Date: Sun, 15 Oct 2017 20:39:06 +0100
Subject: [PATCH 10/12] Point blame to right place in comment

Signed-off-by: Ikey Doherty <ikey@solus-project.com>
---
 src/intercept/main.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/intercept/main.c b/src/intercept/main.c
index 43e0958..b00b3df 100644
--- a/src/intercept/main.c
+++ b/src/intercept/main.c
@@ -122,7 +122,7 @@ static const char *vendor_blacklist[] = {
         /* Nuh uh. Sorry, Feral games. libcurl MUST be up to date for security! */
         "libcurl.so.",
 
-        /* Oh hell no. Seen in Hyper Light Drifer */
+        /* Oh hell no. Seen in Game Maker bundling */
         "libcrypto.so.1.0.0",
         "libssl.so.1.0.0",
 };
-- 
2.14.2

