From 312b7a888aaf04af7352bb75d668db91937d13ca Mon Sep 17 00:00:00 2001
From: Ikey Doherty <ikey@solus-project.com>
Date: Sun, 15 Oct 2017 20:39:28 +0100
Subject: [PATCH 11/12] Just group the security ones.

Signed-off-by: Ikey Doherty <ikey@solus-project.com>
---
 src/intercept/main.c | 4 +---
 1 file changed, 1 insertion(+), 3 deletions(-)

diff --git a/src/intercept/main.c b/src/intercept/main.c
index b00b3df..aed386f 100644
--- a/src/intercept/main.c
+++ b/src/intercept/main.c
@@ -119,10 +119,8 @@ static const char *vendor_blacklist[] = {
         /* libglu has stable soname */
         "libGLU.so.",
 
-        /* Nuh uh. Sorry, Feral games. libcurl MUST be up to date for security! */
+        /* Security sensitive libraries should not be replaced */
         "libcurl.so.",
-
-        /* Oh hell no. Seen in Game Maker bundling */
         "libcrypto.so.1.0.0",
         "libssl.so.1.0.0",
 };
-- 
2.14.2

