diff --git a/src/shim/shim.c b/src/shim/shim.c
index ff4d882..505ba89 100644
--- a/src/shim/shim.c
+++ b/src/shim/shim.c
@@ -183,6 +183,8 @@ static void shim_export_extra(const char *prefix)
          * This allows us to handle the special-case multiarch mounts.
          */
         static const char *ld_library_dirs[] = {
+                "/usr/lib/glx-provider/default",   /**<Solus mesa, 64-bit */
+                "/usr/lib32/glx-provider/default", /**<Solus mesa, 32bit */
                 "/var/lib/snapd/lib/gl/vdpau",   /**<64-bit vdpau */
                 "/var/lib/snapd/lib/gl32/vdpau", /**<32bit vdpau */
                 "/var/lib/snapd/lib/gl",         /**<Potential host NVIDIA libraries */
