diff --git a/src/shim/shim.c b/src/shim/shim.c
index ff4d882..efc77e8 100644
--- a/src/shim/shim.c
+++ b/src/shim/shim.c
@@ -338,7 +340,7 @@ bool shim_bootstrap()
 static int shim_execute_internal(const char *command, int argc, char **argv, bool use_path)
 {
         bool is_x86_64;
-        const char *n_argv[argc + 5];
+        const char *n_argv[argc + 3];
         const char *exec_command = NULL;
         int i = 1;
         int8_t off = 1;
@@ -349,13 +351,10 @@ static int shim_execute_internal(const char *command, int argc, char **argv, boo
 
         /* If we're 64-bit and 32-bit is forced, proxy via setarch linux32 */
         if (lsi_config.force_32 && is_x86_64) {
-                exec_command = "setarch";
-                n_argv[0] = "setarch";
-                n_argv[1] = "linux32";
-                n_argv[2] = "--";
-                n_argv[3] = command;
-                off = 4;
-                /* Use setarch in the path */
+                exec_command = "linux32";
+                n_argv[0] = "linux32";
+                n_argv[1] = command;
+                off = 2;
                 vfunc = execvp;
         } else {
                 /* Directly call lsi_exec_bin */
