diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
index e3eaa697fe..9a10fd7ba4 100644
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -353,7 +353,10 @@ if(USE_WEBP)
 endif(USE_WEBP)
 
 if (USE_AVIF)
-    find_package(libavif 0.8.2 CONFIG)
+    find_package(libavif 0.8.2 CONFIG QUIET)
+    if(NOT libavif_FOUND)
+      find_package(libavif 1 CONFIG)
+    endif()
     if (TARGET avif)
         list(APPEND LIBS avif)
         add_definitions(-DHAVE_LIBAVIF=1)
