--- a/cmake/modules/FindOpenEXR.cmake
+++ b/cmake/modules/FindOpenEXR.cmake
@@ -15,7 +15,11 @@ find_path(_OPENEXR_INCLUDE_DIR  OpenEXR
   PATHS /usr/include
   /usr/include/OpenEXR
   HINTS ENV OPENEXR_INCLUDE_DIR)
-set(OpenEXR_INCLUDE_DIR  ${_OPENEXR_INCLUDE_DIR} ${_OPENEXR_INCLUDE_DIR}/OpenEXR)
+find_path(_IMATH_INCLUDE_DIR Imath
+  PATHS /usr/include
+  /usr/include/Imath
+  HINTS ENV IMATH_INCLUDE_DIR)
+set(OpenEXR_INCLUDE_DIR  ${_OPENEXR_INCLUDE_DIR} ${_OPENEXR_INCLUDE_DIR}/OpenEXR ${_IMATH_INCLUDE_DIR} ${_IMATH_INCLUDE_DIR}/Imath)
 mark_as_advanced(OpenEXR_INCLUDE_DIR)
 
 # Find libraries and add them to OpenEXR_LIBRARY
@@ -24,9 +28,9 @@ find_path(_OPENEXR_LIB_DIR OpenEXR
 set(OpenEXR_LIB_DIR ${_OPENEXR_LIB_DIR})
 mark_as_advanced(OpenEXR_LIB_DIR)
 
-find_library(_OPENEXR_LIBRARY_IMIMF IlmImf
+find_library(_OPENEXR_LIBRARY_OPENEXR OpenEXR
   HINTS ENV OPENEXR_LIB_DIR)
-set(OpenEXR_LIBRARY ${OpenEXR_LIBRARY} ${_OPENEXR_LIBRARY_IMIMF})
+set(OpenEXR_LIBRARY ${OpenEXR_LIBRARY} ${_OPENEXR_LIBRARY_OPENEXR})
 find_library(_OPENEXR_LIBRARY_IMATH Imath
   HINTS ENV OPENEXR_LIB_DIR)
 set(OpenEXR_LIBRARY ${OpenEXR_LIBRARY} ${_OPENEXR_LIBRARY_IMATH})
