From 348f5327ab60380fe4eb230b014ffbc4077fed07 Mon Sep 17 00:00:00 2001
From: Nobuhiro Iwamatsu <iwamatsu@debian.org>
Date: Sun, 9 Mar 2025 23:12:30 +0900
Subject: [PATCH] data/guvcview.appdata.xml.in: Fix format

Signed-off-by: Nobuhiro Iwamatsu <iwamatsu@debian.org>
---
 data/guvcview.appdata.xml.in | 14 +++++++-------
 1 file changed, 7 insertions(+), 7 deletions(-)

diff --git a/data/guvcview.appdata.xml.in b/data/guvcview.appdata.xml.in
index de934f7..37bf7cd 100644
--- a/data/guvcview.appdata.xml.in
+++ b/data/guvcview.appdata.xml.in
@@ -6,15 +6,15 @@
  <metadata_license>CC0-1.0</metadata_license>
  <project_license>GPL-2.0+</project_license>
  <name>Simple UVC Viewer</name>
- <_summary>Capture video from v4l2 devices</_summary>
+ <summary>Capture video from v4l2 devices</summary>
  <description>
-  <_p>Video capture tool for video4linux2 compliant cameras.</_p>
-  <_p>With a special emphasis on the linux uvc driver,
+  <p>Video capture tool for video4linux2 compliant cameras.</p>
+  <p>With a special emphasis on the linux uvc driver,
      it provides Image (jpg, png, bmp) and
      Video (mjpeg, flv1, wmv1, mpg2, mpg4,...) capture
-     with sound in several formats( currently: avi and matroska).</_p>
-  <_p>It also supports a control panel option (--control_only)
-     that is compatible with any other v4l2 app.</_p>
+     with sound in several formats( currently: avi and matroska).</p>
+  <p>It also supports a control panel option (--control_only)
+     that is compatible with any other v4l2 app.</p>
  </description>
  <screenshots>
   <screenshot>
@@ -33,6 +33,6 @@
    <image>https://a.fsdn.com/con/app/proj/guvcview/screenshots/guvcview_video_preview.png</image>
   </screenshot>
  </screenshots>
- <url type="homepage">http://guvcview.sourceforge.net/</url>
+ <url type="homepage">https://guvcview.sourceforge.net/</url>
  <update_contact>assisp@gmail.com</update_contact>
 </component>
-- 
2.47.2

