From 3801b7d9ab2e460cb009302cb589519aa3b7fcc9 Mon Sep 17 00:00:00 2001
From: Joey Riches <josephriches@gmail.com>
Date: Wed, 2 May 2018 21:47:01 +0100
Subject: [PATCH 1/1] Bump the release number you silleh

Signed-off-by: Joey Riches <josephriches@gmail.com>
---
 src/ui/include/notepadqq.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/ui/include/notepadqq.h b/src/ui/include/notepadqq.h
index decde8f..444ef89 100644
--- a/src/ui/include/notepadqq.h
+++ b/src/ui/include/notepadqq.h
@@ -33,7 +33,7 @@
 #include <QList>
 #include "include/mainwindow.h"
 
-#define POINTVERSION "1.4.2" // major.minor.revision
+#define POINTVERSION "1.4.4" // major.minor.revision
 
 #define COPYRIGHT_YEAR "2018"
 
-- 
2.16.3

