From c340237e3f795c1bbf53259ed812b97f6dfee737 Mon Sep 17 00:00:00 2001
From: choryuidentify <yous87uu2@naver.com>
Date: Sun, 17 Dec 2017 03:23:57 +0900
Subject: [PATCH] Fix Qt5 version detection. Fixes #516 (#518)

---
 src/ui/main.cpp | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/src/ui/main.cpp b/src/ui/main.cpp
index e5e150fb9..1ad132752 100644
--- a/src/ui/main.cpp
+++ b/src/ui/main.cpp
@@ -141,9 +141,9 @@ int main(int argc, char *argv[])
 void checkQtVersion()
 {
     QString runtimeVersion = qVersion();
-    if (runtimeVersion.startsWith("5.0") ||
-            runtimeVersion.startsWith("5.1") ||
-            runtimeVersion.startsWith("5.2")) {
+    int minorVersion = runtimeVersion.split(".")[1].toInt();
+    if (runtimeVersion.startsWith("5") &&
+            minorVersion <= 3) {
 
         Notepadqq::setOldQt(true);
     }
