From 8f586fe834a8f485fcf96e36656ce02f8b05a2ff Mon Sep 17 00:00:00 2001
From: Nicolas Chauvet <kwizart@gmail.com>
Date: Wed, 21 Aug 2019 22:47:54 +0200
Subject: [PATCH] Add support for VP9 in vdpauinfo

Signed-off-by: Nicolas Chauvet <kwizart@gmail.com>
---
 vdpauinfo.cpp | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/vdpauinfo.cpp b/vdpauinfo.cpp
index 25229fa..9db705f 100644
--- a/vdpauinfo.cpp
+++ b/vdpauinfo.cpp
@@ -359,6 +359,10 @@ Desc decoder_profiles[] = {
 {"H264_PROGRESSIVE_HIGH",       VDP_DECODER_PROFILE_H264_PROGRESSIVE_HIGH},
 {"H264_CONSTRAINED_HIGH",       VDP_DECODER_PROFILE_H264_CONSTRAINED_HIGH},
 {"H264_HIGH_444_PREDICTIVE",    VDP_DECODER_PROFILE_H264_HIGH_444_PREDICTIVE},
+{"VP9_PROFILE_0",               VDP_DECODER_PROFILE_VP9_PROFILE_0},
+{"VP9_PROFILE_1",               VDP_DECODER_PROFILE_VP9_PROFILE_1},
+{"VP9_PROFILE_2",               VDP_DECODER_PROFILE_VP9_PROFILE_2},
+{"VP9_PROFILE_3",               VDP_DECODER_PROFILE_VP9_PROFILE_3},
 {"HEVC_MAIN",                   VDP_DECODER_PROFILE_HEVC_MAIN},
 {"HEVC_MAIN_10",                VDP_DECODER_PROFILE_HEVC_MAIN_10},
 {"HEVC_MAIN_STILL",             VDP_DECODER_PROFILE_HEVC_MAIN_STILL},
-- 
2.22.0

