From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Muhammad Alfi Syahrin <malfisya.dev@hotmail.com>
Date: Mon, 26 May 2025 10:37:54 +0700
Subject: [PATCH] Add missing include

---
 src/cartridge.cpp | 1 +
 1 file changed, 1 insertion(+)

diff --git a/src/cartridge.cpp b/src/cartridge.cpp
index 9711c07..cb4d336 100644
--- a/src/cartridge.cpp
+++ b/src/cartridge.cpp
@@ -7,6 +7,7 @@
 #include <cstring>
 #include <algorithm>
 #include <memory>
+#include <cstdint>
 #include <string>
 
 const uint32_t CARTRIDGE_NB_PAGES = 32;
