From 46133d52a2392d60a69094c5f638094588d9d70b Mon Sep 17 00:00:00 2001
From: Colin Pitrat <colin.pitrat@gmail.com>
Date: Mon, 10 May 2021 12:00:40 +0100
Subject: [PATCH] Fix some warnings

---
 src/configuration.cpp | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/configuration.cpp b/src/configuration.cpp
index eb3debe..774e025 100644
--- a/src/configuration.cpp
+++ b/src/configuration.cpp
@@ -67,10 +67,10 @@ namespace config
 
   std::ostream& Config::toStream(std::ostream& out) const
   {
-    for(auto section : config_)
+    for(const auto& section : config_)
     {
       out << "[" << section.first << "]" << std::endl;
-      for(auto keyval : section.second)
+      for(const auto& keyval : section.second)
       {
         out << keyval.first << "=" << keyval.second << std::endl;
       }
