From 0c97835d1f0528f1e46936016fc73bb7ada82dab Mon Sep 17 00:00:00 2001
From: Peter O'Connor <peter@solus-project.com>
Date: Sat, 14 Apr 2018 20:06:33 +1000
Subject: [PATCH] Disable building unshipped code

Signed-off-by: Peter O'Connor <peter@solus-project.com>
---
 CMakeLists.txt | 1 -
 1 file changed, 1 deletion(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 3048498..42c6189 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -84,7 +84,6 @@ if(NOT WIN32)
     endif()
 endif()
 
-add_subdirectory(autotests)
 add_subdirectory(icons)
 add_subdirectory(icons-dark)
 
-- 
2.16.3

