diff --git a/src/imlua5.mak b/src/imlua5.mak
index f137b89..dd0be3c 100755
--- a/src/imlua5.mak
+++ b/src/imlua5.mak
@@ -57,7 +57,7 @@ endif
 endif
 endif
 
-LIBNAME := $(LIBNAME)$(LUASFX)
+LIBNAME := $(LIBNAME)
 ifdef NO_LUAOBJECT
   DEFINES += IMLUA_USELH
   USE_LH_SUBDIR = Yes
diff --git a/src/imlua_avi5.mak b/src/imlua_avi5.mak
index ff7cb22..77bebab 100755
--- a/src/imlua_avi5.mak
+++ b/src/imlua_avi5.mak
@@ -47,7 +47,7 @@ endif
 endif
 endif
 
-LIBNAME := $(LIBNAME)$(LUASFX)
+LIBNAME := $(LIBNAME)
 
 USE_IMLUA = Yes
 # To not link with the Lua dynamic library in UNIX
diff --git a/src/imlua_capture5.mak b/src/imlua_capture5.mak
index 5e11cb4..a550a0e 100755
--- a/src/imlua_capture5.mak
+++ b/src/imlua_capture5.mak
@@ -47,7 +47,7 @@ endif
 endif
 endif
 
-LIBNAME := $(LIBNAME)$(LUASFX)
+LIBNAME := $(LIBNAME)
 
 USE_IMLUA = Yes
 # To not link with the Lua dynamic library in UNIX
diff --git a/src/imlua_fftw35.mak b/src/imlua_fftw35.mak
index 5c46764..3007e02 100755
--- a/src/imlua_fftw35.mak
+++ b/src/imlua_fftw35.mak
@@ -57,7 +57,7 @@ endif
 endif
 endif
 
-LIBNAME := $(LIBNAME)$(LUASFX)
+LIBNAME := $(LIBNAME)
 ifdef NO_LUAOBJECT
   DEFINES += IMLUA_USELH
   USE_LH_SUBDIR = Yes
diff --git a/src/imlua_jp25.mak b/src/imlua_jp25.mak
index d811c32..56bd763 100755
--- a/src/imlua_jp25.mak
+++ b/src/imlua_jp25.mak
@@ -47,7 +47,7 @@ endif
 endif
 endif
 
-LIBNAME := $(LIBNAME)$(LUASFX)
+LIBNAME := $(LIBNAME)
 
 USE_IMLUA = Yes
 # To not link with the Lua dynamic library in UNIX
diff --git a/src/imlua_process5.mak b/src/imlua_process5.mak
index 5ec9ec7..1ea6eed 100755
--- a/src/imlua_process5.mak
+++ b/src/imlua_process5.mak
@@ -62,7 +62,7 @@ ifdef USE_OPENMP
   LIBNAME := $(LIBNAME)_omp
   LIBS := im_process_omp
 endif
-LIBNAME := $(LIBNAME)$(LUASFX)
+LIBNAME := $(LIBNAME)
 
 ifdef NO_LUAOBJECT
   DEFINES += IMLUA_USELH
diff --git a/src/imlua_wmv5.mak b/src/imlua_wmv5.mak
index 38c3572..0e547f1 100755
--- a/src/imlua_wmv5.mak
+++ b/src/imlua_wmv5.mak
@@ -47,7 +47,7 @@ endif
 endif
 endif
 
-LIBNAME := $(LIBNAME)$(LUASFX)
+LIBNAME := $(LIBNAME)
 
 IM = ..
 USE_IMLUA = Yes
diff --git a/tecmake.mak b/tecmake.mak
index 48991a4..03df592 100755
--- a/tecmake.mak
+++ b/tecmake.mak
@@ -478,12 +478,12 @@ endif
 ifdef USE_OLDLIBLUA
   TEC_UNAME_LIBLUA_DIR ?= $(TEC_UNAME_LIB_DIR)
 else
-  TEC_UNAME_LIBLUA_DIR ?= $(TEC_UNAME_LIB_DIR)/Lua$(LIBLUA_SFX)
+  TEC_UNAME_LIBLUA_DIR ?= $(TEC_UNAME_LIB_DIR)/Lua
 endif
 
 # Subfolder for Lua Modules
 ifdef LUAMOD_DIR
-  TEC_UNAME_DIR := $(TEC_UNAME_DIR)/Lua$(LIBLUA_SFX)
+  TEC_UNAME_DIR := $(TEC_UNAME_DIR)/Lua
 endif
 
 OBJDIR := $(OBJROOT)/$(TEC_UNAME_DIR)
@@ -941,9 +941,9 @@ ifdef USE_IUPCONTROLS
   
   ifdef USE_IUPLUA
     ifdef USE_STATIC
-      SLIB += $(IUPLUA_LIB)/libiupluacontrols$(LIBLUA_SFX).a
+      SLIB += $(IUPLUA_LIB)/libiupluacontrols.a
     else
-      LIBS += iupluacontrols$(LIBLUA_SFX)
+      LIBS += iupluacontrols
     endif
     override USE_CDLUA = Yes
   endif
@@ -964,9 +964,9 @@ ifdef USE_IUPGLCONTROLS
   
   ifdef USE_IUPLUA
     ifdef USE_STATIC
-      SLIB += $(IUPLUA_LIB)/libiupluaglcontrols$(LIBLUA_SFX).a
+      SLIB += $(IUPLUA_LIB)/libiupluaglcontrols.a
     else
-      LIBS += iupluaglcontrols$(LIBLUA_SFX)
+      LIBS += iupluaglcontrols
     endif
   endif
   
@@ -983,9 +983,9 @@ ifdef USE_IUPWEB
   
   ifdef USE_IUPLUA
     ifdef USE_STATIC
-      SLIB += $(IUPLUA_LIB)/libiupluaweb$(LIBLUA_SFX).a
+      SLIB += $(IUPLUA_LIB)/libiupluaweb.a
     else
-      LIBS += iupluaweb$(LIBLUA_SFX)
+      LIBS += iupluaweb
     endif
   endif
   
@@ -1000,9 +1000,9 @@ ifdef USE_IMLUA
   override USE_IM = Yes
   IMLUA_LIB ?= $(IM)/lib/$(TEC_UNAME_LIBLUA_DIR)
   ifdef USE_STATIC
-    SLIB += $(IMLUA_LIB)/libimlua$(LIBLUA_SFX).a
+    SLIB += $(IMLUA_LIB)/libimlua.a
   else
-    LIBS += imlua$(LIBLUA_SFX)
+    LIBS += imlua
     LDIR += $(IMLUA_LIB)
   endif
 endif
@@ -1011,9 +1011,9 @@ ifdef USE_CDLUA
   override USE_CD = Yes
   CDLUA_LIB ?= $(CD)/lib/$(TEC_UNAME_LIBLUA_DIR)
   ifdef USE_STATIC
-    SLIB += $(CDLUA_LIB)/libcdlua$(LIBLUA_SFX).a
+    SLIB += $(CDLUA_LIB)/libcdlua.a
   else
-    LIBS += cdlua$(LIBLUA_SFX)
+    LIBS += cdlua
     LDIR += $(CDLUA_LIB)
   endif
 endif
@@ -1025,27 +1025,27 @@ ifdef USE_IUPLUA
   ifdef USE_STATIC
     ifdef USE_CD
       ifeq ($(findstring iupluacd, $(LIBNAME)), )
-        SLIB += $(IUPLUA_LIB)/libiupluacd$(LIBLUA_SFX).a
+        SLIB += $(IUPLUA_LIB)/libiupluacd.a
       endif
     endif
     ifdef USE_OPENGL
       ifeq ($(findstring iupluagl, $(LIBNAME)), )
-        SLIB += $(IUPLUA_LIB)/libiupluagl$(LIBLUA_SFX).a
+        SLIB += $(IUPLUA_LIB)/libiupluagl.a
       endif
     endif
-    SLIB += $(IUPLUA_LIB)/libiuplua$(LIBLUA_SFX).a
+    SLIB += $(IUPLUA_LIB)/libiuplua.a
   else
     ifdef USE_CD
       ifeq ($(findstring iupluacd, $(LIBNAME)), )
-        LIBS += iupluacd$(LIBLUA_SFX)
+        LIBS += iupluacd
       endif
     endif
     ifdef USE_OPENGL
       ifeq ($(findstring iupluagl, $(LIBNAME)), )
-        LIBS += iupluagl$(LIBLUA_SFX)
+        LIBS += iupluagl
       endif
     endif
-    LIBS += iuplua$(LIBLUA_SFX)
+    LIBS += iuplua
     LDIR += $(IUPLUA_LIB)
   endif
 endif
