From ede649bcbdfcb8f4a25be90d0a9109e109aea10e Mon Sep 17 00:00:00 2001
From: Muhammad Alfi Syahrin <malfisya.dev@hotmail.com>
Date: Tue, 14 May 2024 08:30:27 +0700
Subject: [PATCH] fix build with gcc13

---
 src/core/model3d/model3d.cpp | 1 +
 1 file changed, 1 insertion(+)

diff --git a/src/core/model3d/model3d.cpp b/src/core/model3d/model3d.cpp
index 29b302d..c2c4dd2 100644
--- a/src/core/model3d/model3d.cpp
+++ b/src/core/model3d/model3d.cpp
@@ -31,6 +31,7 @@
 #include <cmath>
 #include <cstring>
 #include <fstream>
+#include <cstdint>
 
 namespace viewizard {
 
-- 
2.45.0

