From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Reilly Brogan <reilly@reillybrogan.com>
Date: Thu, 8 Aug 2024 16:50:53 -0500
Subject: [PATCH] Don't build with soname

---
 src/libkcups/CMakeLists.txt | 4 ----
 1 file changed, 4 deletions(-)

diff --git a/src/libkcups/CMakeLists.txt b/src/libkcups/CMakeLists.txt
index 03f4ddc..c12da04 100644
--- a/src/libkcups/CMakeLists.txt
+++ b/src/libkcups/CMakeLists.txt
@@ -41,10 +41,6 @@ ki18n_wrap_ui(libkcups_SRCS
 )
 
 add_library(kcupslib SHARED ${libkcups_SRCS})
-set_target_properties(kcupslib PROPERTIES
-    SOVERSION ${PROJECT_VERSION}
-    VERSION ${PROJECT_VERSION}
-)
 
 generate_export_header(kcupslib)
 
