From 42e30748262ab271dfa1ec76bb3bd4274b58857d Mon Sep 17 00:00:00 2001
From: Peter O'Connor <peter@solus-project.com>
Date: Fri, 20 Apr 2018 10:18:09 +1000
Subject: [PATCH] Disable building unshipped code

Signed-off-by: Peter O'Connor <peter@solus-project.com>
---
 CMakeLists.txt | 2 --
 1 file changed, 2 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 7578cab..f77219a 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -60,9 +60,7 @@ add_definitions(-DQT_DISABLE_DEPRECATED_BEFORE=0x000000)
 
 ############## Targets ##############
 add_subdirectory(src)
-add_subdirectory(examples)
 if (BUILD_TESTING)
-    add_subdirectory(autotests)
 endif()
 
 ############## CMake Config Files ##############
-- 
2.16.3

