From 428fabc32012523428fdfff977f2e55d130232b0 Mon Sep 17 00:00:00 2001
From: Joshua Strobl <joshua@streambits.io>
Date: Fri, 22 Oct 2021 19:06:40 +0300
Subject: [PATCH 1/1] Don't install all the things

---
 Makefile | 9 +++------
 1 file changed, 3 insertions(+), 6 deletions(-)

diff --git a/Makefile b/Makefile
index 987708f..a03d482 100644
--- a/Makefile
+++ b/Makefile
@@ -53,9 +53,6 @@ all: CC_LIBRARY($(MINIGBM_FILENAME))
 clean: CLEAN($(MINIGBM_FILENAME))
 
 install: all
-	mkdir -p $(DESTDIR)/$(LIBDIR)
-	install -D -m 755 $(OUT)/$(MINIGBM_FILENAME) $(DESTDIR)/$(LIBDIR)
-	ln -sf $(MINIGBM_FILENAME) $(DESTDIR)/$(LIBDIR)/libgbm.so
-	ln -sf $(MINIGBM_FILENAME) $(DESTDIR)/$(LIBDIR)/libgbm.so.$(GBM_VERSION_MAJOR)
-	install -D -m 0644 $(SRC)/gbm.pc $(DESTDIR)$(LIBDIR)/pkgconfig/gbm.pc
-	install -D -m 0644 $(SRC)/gbm.h $(DESTDIR)/usr/include/gbm.h
+	install -dm00644 $(DESTDIR)/$(LIBDIR)
+	install -Dm00755 $(OUT)/$(MINIGBM_FILENAME) $(DESTDIR)/$(LIBDIR)
+	ln -sf $(MINIGBM_FILENAME) $(DESTDIR)/$(LIBDIR)/libminigbm.so
-- 
2.32.0

