From 4ba73f8a3e91c7bfb93c588f5e4c65869b8e6821 Mon Sep 17 00:00:00 2001
From: Ikey Doherty <ikey@solus-project.com>
Date: Thu, 26 May 2016 04:00:03 +0100
Subject: [PATCH] Use the CC variable, don't force GCC. Required for multilib

Signed-off-by: Ikey Doherty <ikey@solus-project.com>
---
 Makefile | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/Makefile b/Makefile
index 79364f8..edd6896 100644
--- a/Makefile
+++ b/Makefile
@@ -740,7 +740,7 @@ $(LIB):	$(LIB_OBJ)
 	-$(RANLIB) $(LIB)
 
 $(SHLIB): $(LIB_OBJ)
-	gcc -shared -fPIC -Wl,-z,relro -Wl,-soname -Wl,$(SHLIB).$(MAJOR) \
+	$(CC) -shared -fPIC -Wl,-z,relro -Wl,-soname -Wl,$(SHLIB).$(MAJOR) \
 		-o $(SHLIB).$(MAJOR).$(MINOR).$(REL) $^ $(LIBS)
 	ln -s $(SHLIB).$(MAJOR).$(MINOR).$(REL) $(SHLIB).$(MAJOR)
 	ln -s $(SHLIB).$(MAJOR).$(MINOR).$(REL) $(SHLIB)
-- 
2.8.3

