From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Reilly Brogan <reilly@reillybrogan.com>
Date: Fri, 13 Sep 2024 14:53:31 -0500
Subject: [PATCH] Fix Wayland appId

---
 gui/main.cpp | 1 +
 1 file changed, 1 insertion(+)

diff --git a/gui/main.cpp b/gui/main.cpp
index 7d9d83d..69de6b5 100644
--- a/gui/main.cpp
+++ b/gui/main.cpp
@@ -39,6 +39,7 @@ int main(int argc, char **argv)
 {
     QApplication app(argc, argv);
     app.setWindowIcon(QIcon(":/resources/qapitrace.png"));
+    app.setDesktopFileName("io.github.apitrace.ApiTrace");
 
     QCoreApplication::setOrganizationName("ApiTrace");
     QCoreApplication::setApplicationName("QApiTrace");
