From 654fa90af3ab0daabe796609205ce957f7bd722f Mon Sep 17 00:00:00 2001
From: Pierre-Yves <pyu@riseup.net>
Date: Mon, 16 Mar 2020 16:57:59 +0100
Subject: [PATCH] Fix python3 include dir in bootstrap

Signed-off-by: Pierre-Yves <pyu@riseup.net>
---
 bootstrap.sh | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/bootstrap.sh b/bootstrap.sh
index ca0b08d58..30fc013fe 100755
--- a/bootstrap.sh
+++ b/bootstrap.sh
@@ -359,7 +359,7 @@ if test "x$flag_no_python" = x; then
 import python ;
 if ! [ python.configured ]
 {
-    using python : $PYTHON_VERSION : "$PYTHON_ROOT" ;
+    using python : $PYTHON_VERSION : "$PYTHON_ROOT" : /usr/include/python${PYTHON_VERSION} ;
 }
 EOF
 fi
-- 
2.25.1

