From 1a5d11be3c98de24fa74bcf6f97679cca6d38197 Mon Sep 17 00:00:00 2001
From: Pierre-Yves <pyu@riseup.net>
Date: Thu, 12 Oct 2017 23:22:24 +0200
Subject: [PATCH] Fix python3 include dir in bootstrap

Signed-off-by: Pierre-Yves <pyu@riseup.net>
---
 bootstrap.sh | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/bootstrap.sh b/bootstrap.sh
index 9e61569..454101e 100755
--- a/bootstrap.sh
+++ b/bootstrap.sh
@@ -356,7 +356,7 @@ if test "x$flag_no_python" = x; then
 import python ;
 if ! [ python.configured ]
 {
-    using python : $PYTHON_VERSION : $PYTHON_ROOT ;
+    using python : $PYTHON_VERSION : $PYTHON_ROOT : /usr/include/python${PYTHON_VERSION}m ;
 }
 EOF
 fi
-- 
2.14.2

