--- a/panels/info-overview/cc-info-overview-panel.c	2023-07-18 11:43:24.892974397 -0400
+++ b/panels/info-overview/cc-info-overview-panel.c	2023-07-18 12:02:45.312666469 -0400
@@ -780,31 +780,9 @@
 }
 
 static gboolean
-does_gnome_software_allow_updates (void)
+does_solus_sc_exist (void)
 {
-  const gchar *schema_id  = "org.gnome.software";
-  GSettingsSchemaSource *source;
-  g_autoptr(GSettingsSchema) schema = NULL;
-  g_autoptr(GSettings) settings = NULL;
-
-  source = g_settings_schema_source_get_default ();
-
-  if (source == NULL)
-    return FALSE;
-
-  schema = g_settings_schema_source_lookup (source, schema_id, FALSE);
-
-  if (schema == NULL)
-    return FALSE;
-
-  settings = g_settings_new (schema_id);
-  return g_settings_get_boolean (settings, "allow-updates");
-}
-
-static gboolean
-does_gnome_software_exist (void)
-{
-  g_autofree gchar *path = g_find_program_in_path ("gnome-software");
+  g_autofree gchar *path = g_find_program_in_path ("solus-sc");
   return path != NULL;
 }
 
@@ -822,10 +800,10 @@
   gboolean ret;
   char *argv[3];
 
-  if (does_gnome_software_exist ())
+  if (does_solus_sc_exist ())
     {
-      argv[0] = "gnome-software";
-      argv[1] = "--mode=updates";
+      argv[0] = "solus-sc";
+      argv[1] = "--update-view";
       argv[2] = NULL;
     }
   else
@@ -931,7 +909,7 @@
 
   g_resources_register (cc_info_overview_get_resource ());
 
-  if ((!does_gnome_software_exist () || !does_gnome_software_allow_updates ()) && !does_gpk_update_viewer_exist ())
+  if (!does_solus_sc_exist () && !does_gpk_update_viewer_exist ())
     gtk_widget_set_visible (GTK_WIDGET (self->software_updates_row), FALSE);
 
   info_overview_panel_setup_overview (self);
