From c7068bcf17dbaced5e0ab8b7dd787bf9302c7e8b Mon Sep 17 00:00:00 2001
From: Joshua Strobl <joshua@stroblindustries.com>
Date: Sun, 25 Jun 2017 19:26:16 +0300
Subject: [PATCH 1/1] Fix binary location of passwd.

---
 panels/user-accounts/run-passwd.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/panels/user-accounts/run-passwd.c b/panels/user-accounts/run-passwd.c
index 00239ce0f..f39f41b61 100644
--- a/panels/user-accounts/run-passwd.c
+++ b/panels/user-accounts/run-passwd.c
@@ -150,7 +150,7 @@ spawn_passwd (PasswdHandler *passwd_handler, GError **error)
         gchar  **envp;
         gint    my_stdin, my_stdout, my_stderr;
 
-        argv[0] = "/usr/bin/passwd";    /* Is it safe to rely on a hard-coded path? */
+        argv[0] = "/bin/passwd";    /* Is it safe to rely on a hard-coded path? */
         argv[1] = NULL;
 
         envp = g_get_environ ();
-- 
2.13.1

