From 0ed47d4635c2e448ab12379b1cafdb574b6482d0 Mon Sep 17 00:00:00 2001
From: Zach Bacon <zachbacon@vba-m.com>
Date: Thu, 12 Oct 2023 14:13:11 -0400
Subject: [PATCH] use solus-sc instead of gnome-software

Signed-off-by: Zach Bacon <zachbacon@vba-m.com>

---
 panels/system/cc-system-panel.c | 34 ++++++---------------------------
 1 file changed, 6 insertions(+), 28 deletions(-)

diff --git a/panels/system/cc-system-panel.c b/panels/system/cc-system-panel.c
index 357410074..cba2f6a16 100644
--- a/panels/system/cc-system-panel.c
+++ b/panels/system/cc-system-panel.c
@@ -47,31 +47,9 @@ struct _CcSystemPanel
 CC_PANEL_REGISTER (CcSystemPanel, cc_system_panel)

 static gboolean
-gnome_software_allows_updates (void)
+solus_sc_exists (void)
 {
-  const gchar *schema_id  = "org.gnome.software";
-  GSettingsSchemaSource *source;
-  g_autoptr(GSettingsSchema) schema = NULL;
-  g_autoptr(GSettings) settings = NULL;
-
-  source = g_settings_schema_source_get_default ();
-
-  if (source == NULL)
-    return FALSE;
-
-  schema = g_settings_schema_source_lookup (source, schema_id, FALSE);
-
-  if (schema == NULL)
-    return FALSE;
-
-  settings = g_settings_new (schema_id);
-  return g_settings_get_boolean (settings, "allow-updates");
-}
-
-static gboolean
-gnome_software_exists (void)
-{
-  g_autofree gchar *path = g_find_program_in_path ("gnome-software");
+  g_autofree gchar *path = g_find_program_in_path ("solus-sc");
   return path != NULL;
 }

@@ -85,7 +63,7 @@ gpk_update_viewer_exists (void)
 static gboolean
 show_software_updates_group (CcSystemPanel *self)
 {
-  return (gnome_software_exists () && gnome_software_allows_updates ()) ||
+  return (solus_sc_exists ()) ||
          gpk_update_viewer_exists ();
 }

@@ -96,10 +74,10 @@ cc_system_page_open_software_update (CcSystemPanel *self)
   gboolean ret;
   char *argv[3];

-  if (gnome_software_exists ())
+  if (solus_sc_exists ())
     {
-      argv[0] = "gnome-software";
-      argv[1] = "--mode=updates";
+      argv[0] = "solus-sc";
+      argv[1] = "--update-view";
       argv[2] = NULL;
     }
   else
--
2.44.0

