From 8a979ae771f7f83be9858be7f9280b2977b0cb2a Mon Sep 17 00:00:00 2001
From: Joshua Strobl <joshua@stroblindustries.com>
Date: Sun, 22 Apr 2018 23:24:45 +0300
Subject: [PATCH 1/1] Ensure Cheese isn't assumed.

---
 panels/user-accounts/um-photo-dialog.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/panels/user-accounts/um-photo-dialog.c b/panels/user-accounts/um-photo-dialog.c
index 66b236057..1219ef21f 100644
--- a/panels/user-accounts/um-photo-dialog.c
+++ b/panels/user-accounts/um-photo-dialog.c
@@ -564,7 +564,9 @@ um_photo_dialog_class_init (UmPhotoDialogClass *klass)
         gtk_widget_class_bind_template_child (wclass, UmPhotoDialog, take_picture_button);
 
         gtk_widget_class_bind_template_callback (wclass, um_photo_dialog_select_file);
+#ifdef HAVE_CHEESE
         gtk_widget_class_bind_template_callback (wclass, webcam_icon_selected);
+#endif
 
         oclass->dispose = um_photo_dialog_dispose;
 }
-- 
2.16.3

