From b8dddc258ee914f4b8ddbe7a09dccd078c39de34 Mon Sep 17 00:00:00 2001
From: Joshua Strobl <joshua@stroblindustries.com>
Date: Sun, 8 Oct 2017 04:06:25 +0300
Subject: [PATCH 1/1] Fix binary location of passwd.

---
 panels/user-accounts/run-passwd.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/panels/user-accounts/run-passwd.c b/panels/user-accounts/run-passwd.c
index 00239ce..52154f1 100644
--- a/panels/user-accounts/run-passwd.c
+++ b/panels/user-accounts/run-passwd.c
@@ -150,7 +150,7 @@ spawn_passwd (PasswdHandler *passwd_handler, GError **error)
         gchar  **envp;
         gint    my_stdin, my_stdout, my_stderr;
 
-        argv[0] = "/usr/bin/passwd";    /* Is it safe to rely on a hard-coded path? */
+        argv[0] = "/bin/passwd";    /* Is it safe to rely on a hard-coded path? Josh: No, no it isn't. */
         argv[1] = NULL;
 
         envp = g_get_environ ();
-- 
2.14.2

