From 62867527490731ce8d85260bad38e8e5418f1a1d Mon Sep 17 00:00:00 2001
From: Joshua Strobl <joshua@streambits.io>
Date: Fri, 6 Nov 2020 17:27:32 +0200
Subject: [PATCH 1/1] What do we say to libhandy as a shared lib?

Hell no.
---
 meson.build | 1 +
 1 file changed, 1 insertion(+)

diff --git a/meson.build b/meson.build
index 48044cc98..32650b227 100644
--- a/meson.build
+++ b/meson.build
@@ -107,6 +107,7 @@ if not libhandy_dep.found()
       'examples=false',
       'glade_catalog=disabled',
       'introspection=disabled',
+      'static=true',
       'tests=false',
       'vapi=false',
     ]
-- 
2.29.2

