From 5c449ac4a9a15fd9d14069e56a90e1cf122e48dd Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Abdulkadir=20Furkan=20=C5=9Eanl=C4=B1?= <me@abdulocra.cy>
Date: Sun, 3 Jan 2021 15:54:41 +0100
Subject: [PATCH] Fix binary location of passwd
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

Signed-off-by: Abdulkadir Furkan Şanlı <me@abdulocra.cy>
---
 panels/user-accounts/run-passwd.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/panels/user-accounts/run-passwd.c b/panels/user-accounts/run-passwd.c
index 56eea9f29..ac8743272 100644
--- a/panels/user-accounts/run-passwd.c
+++ b/panels/user-accounts/run-passwd.c
@@ -150,7 +150,7 @@ spawn_passwd (PasswdHandler *passwd_handler, GError **error)
         gchar  **envp;
         gint    my_stdin, my_stdout, my_stderr;
 
-        argv[0] = "/usr/bin/passwd";    /* Is it safe to rely on a hard-coded path? */
+        argv[0] = "/bin/passwd";    /* Is it safe to rely on a hard-coded path? Josh: No, no it isn't. */
         argv[1] = NULL;
 
         envp = g_get_environ ();
-- 
2.30.0

