From d3455acbcf183467ce30d3c6d5d93657b8e01bfc Mon Sep 17 00:00:00 2001
From: Joshua Strobl <joshua@streambits.io>
Date: Wed, 11 Mar 2020 22:58:07 +0200
Subject: [PATCH 1/1] Add black as a color option.

---
 panels/background/bg-colors-source.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/panels/background/bg-colors-source.c b/panels/background/bg-colors-source.c
index fbf116af9..58ae2ecac 100644
--- a/panels/background/bg-colors-source.c
+++ b/panels/background/bg-colors-source.c
@@ -55,6 +55,7 @@ struct {
   { G_DESKTOP_BACKGROUND_SHADING_SOLID, -1, "#475b52" },
   { G_DESKTOP_BACKGROUND_SHADING_SOLID, -1, "#425265" },
   { G_DESKTOP_BACKGROUND_SHADING_SOLID, -1, "#7a634b" },
+  { G_DESKTOP_BACKGROUND_SHADING_SOLID, -1, "#000000" },
 };
 
 static gchar *
-- 
2.25.1

