From 36b4582128ee012dc7b1f4148b7f1dcd5b4a9865 Mon Sep 17 00:00:00 2001
From: Joshua Strobl <joshua@stroblindustries.com>
Date: Sun, 22 Apr 2018 23:01:54 +0300
Subject: [PATCH 2/4] Add black as a color option.

---
 panels/background/bg-colors-source.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/panels/background/bg-colors-source.c b/panels/background/bg-colors-source.c
index 7c837b809..aff4d9566 100644
--- a/panels/background/bg-colors-source.c
+++ b/panels/background/bg-colors-source.c
@@ -55,6 +55,7 @@ struct {
   { G_DESKTOP_BACKGROUND_SHADING_SOLID, -1, "#475b52" },
   { G_DESKTOP_BACKGROUND_SHADING_SOLID, -1, "#425265" },
   { G_DESKTOP_BACKGROUND_SHADING_SOLID, -1, "#7a634b" },
+  { G_DESKTOP_BACKGROUND_SHADING_SOLID, -1, "#000000" },
 };
 
 static gchar *
-- 
2.16.3

