diff --git a/panels/info-overview/cc-info-overview-panel.c b/panels/info-overview/cc-info-overview-panel.c
index b348c0f..ea3ebce 100644
--- a/panels/info-overview/cc-info-overview-panel.c
+++ b/panels/info-overview/cc-info-overview-panel.c
@@ -802,31 +802,9 @@ info_overview_panel_setup_overview (CcInfoOverviewPanel *self)
 }
 
 static gboolean
-does_gnome_software_allow_updates (void)
+does_solus_sc_exist (void)
 {
-  const gchar *schema_id  = "org.gnome.software";
-  GSettingsSchemaSource *source;
-  g_autoptr(GSettingsSchema) schema = NULL;
-  g_autoptr(GSettings) settings = NULL;
-
-  source = g_settings_schema_source_get_default ();
-
-  if (source == NULL)
-    return FALSE;
-
-  schema = g_settings_schema_source_lookup (source, schema_id, FALSE);
-
-  if (schema == NULL)
-    return FALSE;
-
-  settings = g_settings_new (schema_id);
-  return g_settings_get_boolean (settings, "allow-updates");
-}
-
-static gboolean
-does_gnome_software_exist (void)
-{
-  g_autofree gchar *path = g_find_program_in_path ("gnome-software");
+  g_autofree gchar *path = g_find_program_in_path ("solus-sc");
   return path != NULL;
 }
 
@@ -844,10 +822,10 @@ open_software_update (CcInfoOverviewPanel *self)
   gboolean ret;
   char *argv[3];
 
-  if (does_gnome_software_exist ())
+  if (does_solus_sc_exist ())
     {
-      argv[0] = "gnome-software";
-      argv[1] = "--mode=updates";
+      argv[0] = "solus-sc";
+      argv[1] = "--update-view";
       argv[2] = NULL;
     }
   else
@@ -948,9 +926,9 @@ use_dark_theme (CcInfoOverviewPanel *panel)
 static void
 setup_os_logo (CcInfoOverviewPanel *panel)
 {
-  GtkIconTheme *icon_theme;
+  /* GtkIconTheme *icon_theme; */
   g_autofree char *logo_name = g_get_os_info ("LOGO");
-  g_autoptr(GtkIconPaintable) icon_paintable = NULL;
+  /* g_autoptr(GtkIconPaintable) icon_paintable = NULL;
   g_autoptr(GPtrArray) array = NULL;
   g_autoptr(GIcon) icon = NULL;
   gboolean dark;
@@ -974,7 +952,9 @@ setup_os_logo (CcInfoOverviewPanel *panel)
                                                    gtk_widget_get_scale_factor (GTK_WIDGET (panel)),
                                                    gtk_widget_get_direction (GTK_WIDGET (panel)),
                                                    0);
-  gtk_picture_set_paintable (panel->os_logo, GDK_PAINTABLE (icon_paintable));
+  gtk_picture_set_paintable (panel->os_logo, GDK_PAINTABLE (icon_paintable)); */
+  gtk_image_set_from_resource (panel->os_logo, "/org/gnome/control-center/info-overview/GnomeLogoVerticalMedium.svg");
+  gtk_image_set_pixel_size (panel->os_logo, 256);
 }
 
 static void
@@ -1018,7 +998,7 @@ cc_info_overview_panel_init (CcInfoOverviewPanel *self)
 
   g_resources_register (cc_info_overview_get_resource ());
 
-  if ((!does_gnome_software_exist () || !does_gnome_software_allow_updates ()) && !does_gpk_update_viewer_exist ())
+  if (!does_solus_sc_exist () && !does_gpk_update_viewer_exist ())
     gtk_widget_hide (GTK_WIDGET (self->software_updates_row));
 
   info_overview_panel_setup_overview (self);
