diff --git a/desktop/packages/mullvad-vpn/tasks/distribution.js b/desktop/packages/mullvad-vpn/tasks/distribution.js
index fe52ea4c2..ef0ad5300 100644
--- a/desktop/packages/mullvad-vpn/tasks/distribution.js
+++ b/desktop/packages/mullvad-vpn/tasks/distribution.js
@@ -190,7 +190,8 @@ function newConfig() {
         },
       ],
       executableName: 'mullvad-vpn',
-      artifactName: 'MullvadVPN-${version}_${arch}.${ext}',
+      target: ['dir'],
+      //artifactName: 'MullvadVPN-${version}_${arch}.${ext}',
       category: 'Network',
       icon: distAssets('icon.icns'),
       extraFiles: [{ from: distAssets('linux/mullvad-gui-launcher.sh'), to: '.' }],
