diff --git a/desktop/packages/mullvad-vpn/tasks/distribution.cjs b/desktop/packages/mullvad-vpn/tasks/distribution.cjs
--- a/desktop/packages/mullvad-vpn/tasks/distribution.cjs
+++ b/desktop/packages/mullvad-vpn/tasks/distribution.cjs
@@ -189,7 +189,8 @@ function newConfig() {
         },
       ],
       executableName: 'mullvad-vpn',
-      artifactName: 'MullvadVPN-${version}_${arch}.${ext}',
+      target: ['dir'],
+      //artifactName: 'MullvadVPN-${version}_${arch}.${ext}',
       category: 'Network',
       icon: distAssets('icon.icns'),
       extraFiles: [{ from: distAssets('linux/mullvad-gui-launcher.sh'), to: '.' }],
