From 5c89fbfb6e92d88380964197969ac67b350b7e47 Mon Sep 17 00:00:00 2001
From: Cory Forsstrom <cforsstrom18@gmail.com>
Date: Fri, 14 Jul 2023 09:45:13 -0700
Subject: [PATCH 3/3] update-build.sh

---
 build.sh | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/build.sh b/build.sh
index 2539d6120..e34473e88 100755
--- a/build.sh
+++ b/build.sh
@@ -52,7 +52,7 @@ done
 # sign the artifacts, AND we are currently building on a release git tag.
 # Everything that is not a release build is called a "dev build" and has "-dev-{commit hash}"
 # appended to the version name.
-IS_RELEASE="false"
+IS_RELEASE="true"
 if [[ "$SIGN" == "true" && "$OPTIMIZE" == "true" && "$PRODUCT_VERSION" != *"-dev-"* ]]; then
     IS_RELEASE="true"
 fi
-- 
2.40.1

