From 2e13888b8071497201e1f88d010d3cefed49cc1d Mon Sep 17 00:00:00 2001
From: Cory Forsstrom <cforsstrom18@gmail.com>
Date: Sat, 23 Apr 2022 10:54:41 -0700
Subject: [PATCH] update build.sh

---
 build.sh | 24 ++++++++++++------------
 1 file changed, 12 insertions(+), 12 deletions(-)

diff --git a/build.sh b/build.sh
index 3c01b8373..93654a91c 100755
--- a/build.sh
+++ b/build.sh
@@ -78,7 +78,7 @@ done
 # sign the artifacts, AND we are currently building on a release git tag.
 # Everything that is not a release build is called a "dev build" and has "-dev-{commit hash}"
 # appended to the version name.
-IS_RELEASE="false"
+IS_RELEASE="true"
 product_version_commit_hash=$(git rev-parse "$PRODUCT_VERSION^{commit}" || echo "")
 current_head_commit_hash=$(git rev-parse "HEAD^{commit}")
 if [[ "$SIGN" == "true" && "$OPTIMIZE" == "true" && \
@@ -380,17 +380,17 @@ esac
 
 popd
 
-SEMVER_VERSION=$(echo "$PRODUCT_VERSION" | sed -Ee 's/($|-.*)/.0\1/g')
-for semver_path in dist/*"$SEMVER_VERSION"*; do
-    product_path=$(echo "$semver_path" | sed -Ee "s/$SEMVER_VERSION/$PRODUCT_VERSION/g")
-    log_info "Moving $semver_path -> $product_path"
-    mv "$semver_path" "$product_path"
-
-    if [[ "$SIGN" == "true" && "$(uname -s)" == "MINGW"* && "$product_path" == *.exe ]]; then
-        # sign installer
-        sign_win "$product_path"
-    fi
-done
+#SEMVER_VERSION=$(echo "$PRODUCT_VERSION" | sed -Ee 's/($|-.*)/.0\1/g')
+#for semver_path in dist/*"$SEMVER_VERSION"*; do
+#    product_path=$(echo "$semver_path" | sed -Ee "s/$SEMVER_VERSION/$PRODUCT_VERSION/g")
+#    log_info "Moving $semver_path -> $product_path"
+#    mv "$semver_path" "$product_path"
+#
+#    if [[ "$SIGN" == "true" && "$(uname -s)" == "MINGW"* && "$product_path" == *.exe ]]; then
+#        # sign installer
+#        sign_win "$product_path"
+#    fi
+#done
 
 log_success "**********************************"
 log_success ""
-- 
2.35.1

