From 1275f7b22ef917a8c354770d52d93341e7c17f9a Mon Sep 17 00:00:00 2001
From: Cory Forsstrom <cforsstrom18@gmail.com>
Date: Wed, 1 Sep 2021 08:28:02 -0700
Subject: [PATCH] Add env var to disable version check in gui

---
 gui/src/main/index.ts | 10 +++++++++-
 1 file changed, 9 insertions(+), 1 deletion(-)

diff --git a/gui/src/main/index.ts b/gui/src/main/index.ts
index ac5e9f87f..c24959007 100644
--- a/gui/src/main/index.ts
+++ b/gui/src/main/index.ts
@@ -89,6 +89,8 @@ const GUI_VERSION = app.getVersion().replace('.0', '');
 /// Mirrors the beta check regex in the daemon. Matches only well formed beta versions
 const IS_BETA = /^(\d{4})\.(\d+)-beta(\d+)$/;
 
+const VERSION_CHECK_DISABLED = process.env.DISABLE_VERSION_CHECK === '1';
+
 const SANDBOX_DISABLED = app.commandLine.hasSwitch('no-sandbox');
 
 enum AppQuitStage {
@@ -583,7 +585,9 @@ class ApplicationMain {
     }
 
     // fetch the latest version info in background
-    consumePromise(this.fetchLatestVersion());
+    if (!VERSION_CHECK_DISABLED) {
+      consumePromise(this.fetchLatestVersion());
+    }
 
     // reset the reconnect backoff when connection established.
     this.reconnectBackoff.reset();
@@ -920,6 +924,10 @@ class ApplicationMain {
   }
 
   private setLatestVersion(latestVersionInfo: IAppVersionInfo) {
+    if (VERSION_CHECK_DISABLED) {
+      return;
+    }
+
     const suggestedIsBeta =
       latestVersionInfo.suggestedUpgrade !== undefined &&
       IS_BETA.test(latestVersionInfo.suggestedUpgrade);
-- 
2.32.0

