From c5b244a0975f5bec87a324377ea1148b0627ea62 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Aurimas=20=C4=8Cernius?= <aurisc4@gmail.com>
Date: Tue, 17 Mar 2020 23:25:16 +0200
Subject: [PATCH] Fix condition for checking if note exists

---
 src/synchronization/syncmanager.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/synchronization/syncmanager.cpp b/src/synchronization/syncmanager.cpp
index 2c7c59f6..5d03f82f 100644
--- a/src/synchronization/syncmanager.cpp
+++ b/src/synchronization/syncmanager.cpp
@@ -221,7 +221,7 @@ namespace sync {
       //       Want this stuff to happen all at once first, but
       //       maybe there's a way to store this info and pass it on?
       for(auto & iter : noteUpdates) {
-        if(!find_note_by_uuid(iter.second.m_uuid) != 0) {
+        if(find_note_by_uuid(iter.second.m_uuid) != 0) {
           NoteBase::Ptr existingNote = note_mgr().find(iter.second.m_title);
           if(existingNote != 0 && !iter.second.basically_equal_to(std::static_pointer_cast<Note>(existingNote))) {
             DBG_OUT("Sync: Early conflict detection for '%s'", iter.second.m_title.c_str());
-- 
2.24.1

