From 84e62921a9b20a9a094a4df1d36af03195d685d3 Mon Sep 17 00:00:00 2001
From: Ikey Doherty <ikey@solus-project.com>
Date: Sat, 24 Sep 2016 17:50:42 +0100
Subject: [PATCH] Stop butchering flags

Signed-off-by: Ikey Doherty <ikey@solus-project.com>
---
 GNUmakefile | 16 ----------------
 1 file changed, 16 deletions(-)

diff --git a/GNUmakefile b/GNUmakefile
index 4de9d10..4a3ea74 100755
--- a/GNUmakefile
+++ b/GNUmakefile
@@ -125,22 +125,6 @@ ifeq ($(IS_X86)$(IS_X32)$(IS_CYGWIN)$(IS_MINGW)$(SUN_COMPILER),00000)
  endif
 endif
 
-# Guard use of -march=native
-ifeq ($(GCC42_OR_LATER)$(IS_NETBSD),10)
-   CXXFLAGS += -march=native
-else ifneq ($(CLANG_COMPILER)$(INTEL_COMPILER),00)
-   CXXFLAGS += -march=native
-else
-  # GCC 3.3 and "unknown option -march="
-  # Ubuntu GCC 4.1 compiler crash with -march=native
-  # NetBSD GCC 4.8 compiler and "bad value (native) for -march= switch"
-  # Sun compiler is handled below
-  ifeq ($(SUN_COMPILER)$(IS_X64),01)
-    CXXFLAGS += -m64
-  else ifeq ($(SUN_COMPILER)$(IS_X86),01)
-    CXXFLAGS += -m32
-  endif # X86/X32/X64
-endif
 
 # Aligned access required for -O3 and above due to vectorization
 UNALIGNED_ACCESS := $(shell $(EGREP) -c "^[[:space:]]*//[[:space:]]*\#[[:space:]]*define[[:space:]]*CRYPTOPP_NO_UNALIGNED_DATA_ACCESS" config.h)
-- 
2.10.0

