From 3f9ed47cce4e4c40887601d3c5ac3b4a35fe7ca2 Mon Sep 17 00:00:00 2001
From: Pierre-Yves <pyu@riseup.net>
Date: Mon, 18 Dec 2017 21:34:31 +0100
Subject: [PATCH 2/2] Stop building everywhere

Signed-off-by: Pierre-Yves <pyu@riseup.net>
---
 Makefile | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/Makefile b/Makefile
index acc96e0..a75c372 100644
--- a/Makefile
+++ b/Makefile
@@ -22,7 +22,7 @@ release:
 grip:
 	@gawk -f build.awk readme && grip
 
-test: build
+test:
 	@gawk -f test.awk
 
 check: test
@@ -30,7 +30,7 @@ check: test
 	[ "`$(BUILDDIR)/$(COMMAND) -no-init -D -b 忍者`" = 'Ninja' ] &&\
 	[ "`$(BUILDDIR)/$(COMMAND) -no-init -D -b 'hello world'`" = 'hello world' ]
 
-install: build
+install:
 	@mkdir -p $(DESTDIR)$(PREFIX)/bin &&\
 	install $(BUILDDIR)/$(COMMAND) $(DESTDIR)$(PREFIX)/bin/$(COMMAND) &&\
 	mkdir -p $(DESTDIR)$(PREFIX)/share/man/man1 &&\
-- 
2.15.1

