From d9e8f5376ed0873d3c6f18a10003701f96440f74 Mon Sep 17 00:00:00 2001
From: Ikey Doherty <ikey@solus-project.com>
Date: Sat, 12 Mar 2016 12:48:01 +0000
Subject: [PATCH] Don't use nonexistent user

Signed-off-by: Ikey Doherty <ikey@solus-project.com>
---
 init/systemd/man-db.conf | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/init/systemd/man-db.conf b/init/systemd/man-db.conf
index 10b27b4..e31438a 100644
--- a/init/systemd/man-db.conf
+++ b/init/systemd/man-db.conf
@@ -1 +1 @@
-d /var/cache/man 2755 man root 1w
+d /var/cache/man 2755 root root 1w
-- 
2.7.1

