From bc404b910d16e21ae833f9774d702dd879b17a8e Mon Sep 17 00:00:00 2001
From: Algent Albrahimi <algent@protonmail.com>
Date: Fri, 25 Oct 2019 23:40:27 +0200
Subject: [PATCH] fix

---
 Makefile.am | 7 +------
 1 file changed, 1 insertion(+), 6 deletions(-)

diff --git a/Makefile.am b/Makefile.am
index 03a21ab..0f840cb 100644
--- a/Makefile.am
+++ b/Makefile.am
@@ -10,12 +10,7 @@ dist_doc_DATA = README.txt COPYING LICENSES
 # WHEN ADDING TRANSLATIONS: Please append all new gettext file names
 # here (both the .po file and the .mo file).  This insures the
 # translation is both distributed and installed properly.
-nobase_dist_pkgdata_DATA = data/* gfx/* music/* sound/* \
-	locale/build.py \
-	locale/genpot.sh \
-	locale/pr-starfighter.pot \
-	locale/en.po \
-	locale/en/LC_MESSAGES/pr-starfighter.mo
+nobase_dist_pkgdata_DATA = data/* gfx/* music/* sound/* locale/*
 
 EXTRA_DIST = starfighter.sh win32build.sh misc
 
-- 
2.23.0

