From 546765dfd396a4bb482b98a7b6ca22f1dca809af Mon Sep 17 00:00:00 2001
From: "Rasyuqa A. H." <qampidh@gmail.com>
Date: Mon, 8 Jan 2024 17:48:34 +0700
Subject: [PATCH] JPEG-XL: Disable export bug workaround for libjxl >= 0.9.0

---
 plugins/impex/jxl/JPEGXLExport.cpp | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/plugins/impex/jxl/JPEGXLExport.cpp b/plugins/impex/jxl/JPEGXLExport.cpp
index 4b4b21dabf9..7a7f96e52bb 100644
--- a/plugins/impex/jxl/JPEGXLExport.cpp
+++ b/plugins/impex/jxl/JPEGXLExport.cpp
@@ -832,14 +832,16 @@ KisImportExportErrorCode JPEGXLExport::convert(KisDocument *document, QIODevice
         }();
 
         // XXX: Workaround for a buggy lossless patches. Set to disable instead.
-        // TODO Kampidh: revisit this when upstream got fixed.
+        // Patch only for libjxl under v0.9.0
         //
         // See: https://github.com/libjxl/libjxl/issues/2463
         const int setPatches = [&]() -> int {
+#if JPEGXL_NUMERIC_VERSION < JPEGXL_COMPUTE_NUMERIC_VERSION(0, 9, 0)
             if ((cfg->getInt("effort", 7) > 4) && !cfg->getBool("flattenLayers", true)) {
                 warnFile << "Using workaround for layer exports, disabling patches option on effort > 4";
                 return 0;
             }
+#endif
             return cfg->getInt("patches", -1);
         }();
 
-- 
GitLab

