From 7c414ddc0bd2fae590cc09944579e90d822d9e0b Mon Sep 17 00:00:00 2001
From: Gwyn Ciesla <gwync@protonmail.com>
Date: Tue, 28 Jan 2025 13:33:22 -0600
Subject: [PATCH] Fix build with most recent gcc/glibc

---
 include/mdds/trie_map.hpp | 1 +
 1 file changed, 1 insertion(+)

diff --git a/include/mdds/trie_map.hpp b/include/mdds/trie_map.hpp
index 96f6cb6a..ac4c9d70 100644
--- a/include/mdds/trie_map.hpp
+++ b/include/mdds/trie_map.hpp
@@ -37,6 +37,7 @@
 #include <map>
 #include <memory>
 #include <limits>
+#include <cstdint>
 
 namespace mdds {
 
-- 
GitLab

