From d4888a23a2704f0738a46774140357b767ae348e Mon Sep 17 00:00:00 2001
From: Alexander Vorobyev <avorobyev@protonmail.com>
Date: Mon, 29 Mar 2021 19:49:54 +0300
Subject: Set minimum version for Python interpreter

---
 CMakeLists.txt | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 51e08b59..44b12d42 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -64,7 +64,7 @@ endif()
 find_package(${QT_PREFIX} REQUIRED COMPONENTS Core Widgets Gui Svg Network)
 
 if(CUTTER_ENABLE_PYTHON)
-    find_package(PythonInterp REQUIRED)
+    find_package(PythonInterp ${CUTTER_PYTHON_MIN} REQUIRED)
     find_package(PythonLibs ${CUTTER_PYTHON_MIN} REQUIRED)
 
     include_directories(${PYTHON_INCLUDE_DIRS})
-- 
2.30.2

