From 4a446b5e63b4c49b7bf61e06a9450693676c0253 Mon Sep 17 00:00:00 2001
From: Alexander Vorobyev <avorobyev@protonmail.com>
Date: Fri, 4 Sep 2020 09:46:49 +0300
Subject: Set minimum version for Python interpreter

---
 src/CMakeLists.txt | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
index ed084533..22bf580a 100644
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -66,7 +66,7 @@ else()
 endif()
 
 if(CUTTER_ENABLE_PYTHON)
-    find_package(PythonInterp REQUIRED)
+    find_package(PythonInterp ${CUTTER_PYTHON_MIN} REQUIRED)
     find_package(PythonLibs ${CUTTER_PYTHON_MIN} REQUIRED)
 
     include_directories(${PYTHON_INCLUDE_DIRS})
-- 
2.27.0

