From 07fea9c772dc573588dc2e5771f0740ee1883738 Mon Sep 17 00:00:00 2001
From: Premade <128969155+PremadeS@users.noreply.github.com>
Date: Fri, 1 May 2026 15:43:36 +0500
Subject: [PATCH] PluginsManager: handle type index case for shiboken6 (#3606)

---
 src/plugins/PluginManager.cpp | 6 ++++--
 1 file changed, 4 insertions(+), 2 deletions(-)

diff --git a/src/plugins/PluginManager.cpp b/src/plugins/PluginManager.cpp
index d1c5465da5..a4a4791fe4 100644
--- a/src/plugins/PluginManager.cpp
+++ b/src/plugins/PluginManager.cpp
@@ -215,10 +215,12 @@ CutterPlugin *PluginManager::loadPythonPlugin(const char *moduleName)
     }
 
     PythonToCppFunc pythonToCpp = Shiboken::Conversions::isPythonToCppPointerConvertible(
-#    if QT_VERSION < QT_VERSION_CHECK(6, 2, 0)
+#    if QT_VERSION < QT_VERSION_CHECK(6, 0, 0)
             reinterpret_cast<SbkObjectType *>(SbkCutterBindingsTypes[SBK_CUTTERPLUGIN_IDX]),
+#    elif QT_VERSION < QT_VERSION_CHECK(6, 2, 0)
+            reinterpret_cast<SbkObjectType *>(SbkCutterBindingsTypes[SBK_CutterPlugin_IDX]),
 #    else
-            reinterpret_cast<PyTypeObject **>(SbkCutterBindingsTypeStructs)[SBK_CUTTERPLUGIN_IDX],
+            reinterpret_cast<PyTypeObject **>(SbkCutterBindingsTypeStructs)[SBK_CutterPlugin_IDX],
 #    endif
             pluginObject);
     if (!pythonToCpp) {
