From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: "Jan Alexander Steffens (heftig)" <heftig@archlinux.org>
Date: Mon, 12 Dec 2022 00:03:03 +0100
Subject: [PATCH 05/13] ZEN: INTERACTIVE: Use Kyber as the elevator for MQ
 devices

---
 block/elevator.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/block/elevator.c b/block/elevator.c
index 85410403f346..a9d58fcd8a6d 100644
--- a/block/elevator.c
+++ b/block/elevator.c
@@ -558,7 +558,11 @@ static struct elevator_type *elevator_get_default(struct request_queue *q)
 
 	if (q->nr_hw_queues != 1 &&
 	    !blk_mq_is_shared_tags(q->tag_set->flags))
+#if defined(CONFIG_MQ_IOSCHED_KYBER)
+		return elevator_find_get("kyber");
+#else
 		return NULL;
+#endif
 
 	return elevator_find_get("bfq");
 }
