From 91a726c9add02c9571702356605e8b37bc80c549 Mon Sep 17 00:00:00 2001
From: "Jan Alexander Steffens (heftig)" <jan.steffens@gmail.com>
Date: Mon, 27 Jan 2020 18:11:05 +0100
Subject: [PATCH 20/29] ZEN: INTERACTIVE: Use BFQ as the elevator for SQ
 devices

---
 block/elevator.c | 4 ++++
 init/Kconfig     | 4 ++++
 2 files changed, 8 insertions(+)

diff --git a/block/elevator.c b/block/elevator.c
index e2ebfbf107b3..178c89b3bc9c 100644
--- a/block/elevator.c
+++ b/block/elevator.c
@@ -741,7 +741,11 @@ void elv_update_nr_hw_queues(struct request_queue *q, struct elevator_type *e,
 void elevator_set_default(struct request_queue *q)
 {
 	struct elv_change_ctx ctx = {
+#if defined(CONFIG_ZEN_INTERACTIVE) && defined(CONFIG_IOSCHED_BFQ)
+		.name = "bfq",
+#else
 		.name = "mq-deadline",
+#endif
 		.no_uevent = true,
 	};
 	int err;
diff --git a/init/Kconfig b/init/Kconfig
index 11d1bb4ea28f..6866e4f815cd 100644
--- a/init/Kconfig
+++ b/init/Kconfig
@@ -193,6 +193,10 @@ config ZEN_INTERACTIVE
 	help
 	  Tunes the kernel for responsiveness at the cost of throughput and power usage.
 
+	  --- Block Layer ----------------------------------------
+
+	    Default scheduler for SQ..: mq-deadline ->   bfq
+
 config BROKEN
 	bool
 	help
-- 
2.52.0

