From 669f0096c96eda5618f4e42766c665f3ed341e92 Mon Sep 17 00:00:00 2001
From: Steven Barrett <steven@liquorix.net>
Date: Sun, 19 Sep 2021 16:03:36 -0500
Subject: [PATCH 29/29] ZEN: INTERACTIVE: Document PDS/BMQ configuration

---
 init/Kconfig | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/init/Kconfig b/init/Kconfig
index 6f4378c26f36..55217cc7e416 100644
--- a/init/Kconfig
+++ b/init/Kconfig
@@ -212,6 +212,11 @@ config ZEN_INTERACTIVE
 	    Bandwidth slice size...........:   5    ->   3    ms
 	    Task rebalancing threshold.....:  32    ->   8
 
+	  --- PDS/BMQ CPU Scheduler ------------------------------
+
+	    Scheduling timeslice...........:   4    ->   2    ms
+	    Yield type.....................:   1    ->   0
+
 	  --- CPUFreq Settings -----------------------------------
 
 	    Ondemand sampling down factor..:   1    ->   5
-- 
2.52.0

