From 7cec44962c55032978b9c43327fbf96d9a65e717 Mon Sep 17 00:00:00 2001
From: Joey Riches <josephriches@gmail.com>
Date: Wed, 24 Mar 2021 08:25:04 +0000
Subject: [PATCH 1/1] strip out legacy libtinfo

---
 CMakeLists.txt | 5 ++---
 1 file changed, 2 insertions(+), 3 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 4410b229..3b3c44e6 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -492,12 +492,11 @@ else()
         target_link_libraries(${PROJECT_NAME} pthread z curses)
     else()
         if (ISPC_STATIC_LINK)
-            target_link_libraries(${PROJECT_NAME} pthread z.a tinfo.a curses.a)
+            target_link_libraries(${PROJECT_NAME} pthread z.a curses.a)
         else()
             find_package(Curses REQUIRED)
             find_package(ZLIB REQUIRED)
-            find_library(NCURSES_TINFO_LIBRARY tinfo)
-            target_link_libraries(${PROJECT_NAME} pthread ${ZLIB_LIBRARIES} ${NCURSES_TINFO_LIBRARY} ${CURSES_LIBRARIES})
+            target_link_libraries(${PROJECT_NAME} pthread ${ZLIB_LIBRARIES} ${CURSES_LIBRARIES})
         endif()
     endif()
 endif()
-- 
2.30.2

