From 907aa458cabbce52fd22a4372d7c9689f021a449 Mon Sep 17 00:00:00 2001
From: Joey Riches <josephriches@gmail.com>
Date: Sat, 9 Jan 2021 10:40:11 +0000
Subject: [PATCH 1/1] strip out legacy libtinfo

---
 CMakeLists.txt | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 2bb4a7e6..899fe4ba 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -473,7 +473,7 @@ else()
         if (ISPC_STATIC_LINK)
             target_link_libraries(${PROJECT_NAME} pthread z.a tinfo.a curses.a)
         else()
-            target_link_libraries(${PROJECT_NAME} pthread z tinfo curses)
+            target_link_libraries(${PROJECT_NAME} pthread z curses)
         endif()
     endif()
 endif()
-- 
2.30.0

