From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Reilly Brogan <reilly@reillybrogan.com>
Date: Mon, 9 Dec 2024 18:05:44 -0600
Subject: [PATCH] llvm-19 fixes

---
 src/opt/XeReplaceLLVMIntrinsics.cpp | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/opt/XeReplaceLLVMIntrinsics.cpp b/src/opt/XeReplaceLLVMIntrinsics.cpp
index d0e399ad..4dcf2480 100644
--- a/src/opt/XeReplaceLLVMIntrinsics.cpp
+++ b/src/opt/XeReplaceLLVMIntrinsics.cpp
@@ -25,7 +25,7 @@ restart:
             if (func == nullptr || !func->isIntrinsic())
                 continue;
 
-            if (func->getName().equals("llvm.trap")) {
+            if (func->getName() == "llvm.trap") {
                 llvm::Type *argTypes[] = {LLVMTypes::Int1VectorType, LLVMTypes::Int16VectorType};
                 // Description of parameters for genx_raw_send_noresult can be found in target-genx.ll
                 auto Fn =
@@ -49,7 +49,7 @@ restart:
                     modifiedAny = true;
                     goto restart;
                 }
-            } else if (func->getName().equals("llvm.experimental.noalias.scope.decl")) {
+            } else if (func->getName() == "llvm.experimental.noalias.scope.decl") {
                 // These intrinsics are not supported by backend so remove them.
                 ci->eraseFromParent();
                 modifiedAny = true;
