From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Thomas Staudinger <Staudi.Kaos@gmail.com>
Date: Fri, 10 May 2024 22:55:04 +0200
Subject: [PATCH] Add support for zstd-compressed charmaps

Signed-off-by: Thomas Staudinger <Staudi.Kaos@gmail.com>
---
 configure.ac | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/configure.ac b/configure.ac
index 5abde50..a3fc526 100644
--- a/configure.ac
+++ b/configure.ac
@@ -65,7 +65,10 @@ else :
 fi
 
 if test "x$CHARMAPS" != "x"; then
-  if test -r "$CHARMAPS/EUC-JP.gz"; then
+  if test -r "$CHARMAPS/EUC-JP.zst"; then
+    CHARMAP_EXT=".zst"
+    CHARMAP_CAT="zstdcat"
+  elif test -r "$CHARMAPS/EUC-JP.gz"; then
     CHARMAP_EXT=".gz"
     CHARMAP_CAT="zcat"
   elif test -r "$CHARMAPS/EUC-JP"; then
