From 560e2dbd955002b44519086d88d077d8dfa1dde2 Mon Sep 17 00:00:00 2001
From: Bastian Blank <waldi@debian.org>
Date: Tue, 30 Jun 2015 11:46:07 +0200
Subject: [PATCH 1/3] Fix cache dirs.

---
 lib/config/config.c | 10 ++++------
 1 file changed, 4 insertions(+), 6 deletions(-)

diff --git a/lib/config/config.c b/lib/config/config.c
index ad816c2..17ca01e 100644
--- a/lib/config/config.c
+++ b/lib/config/config.c
@@ -2365,10 +2365,8 @@ const char *get_default_devices_cache_CFG(struct cmd_context *cmd, struct profil
 
 	if (cache_dir || cache_file_prefix) {
 		if (dm_snprintf(buf, sizeof(buf),
-				"%s%s%s/%s.cache",
-				cache_dir ? "" : cmd->system_dir,
-				cache_dir ? "" : "/",
-				cache_dir ? : DEFAULT_CACHE_SUBDIR,
+				"%s/%s.cache",
+				cache_dir ? : DEFAULT_RUN_DIR,
 				cache_file_prefix ? : DEFAULT_CACHE_FILE_PREFIX) < 0) {
 			log_error("Persistent cache filename too long.");
 			return NULL;
@@ -2376,8 +2374,8 @@ const char *get_default_devices_cache_CFG(struct cmd_context *cmd, struct profil
 		return dm_pool_strdup(cmd->mem, buf);
 	}
 
-	if (dm_snprintf(buf, sizeof(buf), "%s/%s/%s.cache", cmd->system_dir,
-			DEFAULT_CACHE_SUBDIR, DEFAULT_CACHE_FILE_PREFIX) < 0) {
+	if (dm_snprintf(buf, sizeof(buf), "%s/%s.cache",
+			DEFAULT_RUN_DIR, DEFAULT_CACHE_FILE_PREFIX) < 0) {
 		log_error("Persistent cache filename too long.");
 		return NULL;
 	}
-- 
2.23.0
