From 9d944e682b7b83ecfe3cb07a49648010c750a690 Mon Sep 17 00:00:00 2001
From: Pierre-Yves <pyu@riseup.net>
Date: Mon, 3 Dec 2018 09:29:37 +0100
Subject: [PATCH] stunnel service privatetmp

Signed-off-by: Pierre-Yves <pyu@riseup.net>
---
 tools/stunnel.service.in | 1 +
 1 file changed, 1 insertion(+)

diff --git a/tools/stunnel.service.in b/tools/stunnel.service.in
index 64ba7fa..620a0e7 100644
--- a/tools/stunnel.service.in
+++ b/tools/stunnel.service.in
@@ -5,6 +5,7 @@ After=syslog.target network.target
 [Service]
 ExecStart=@bindir@/stunnel
 Type=forking
+PrivateTmp=true
 
 [Install]
 WantedBy=multi-user.target
-- 
2.19.2

