From 1bf887e51517aeb75dd9ffefc0f42b8413e61666 Mon Sep 17 00:00:00 2001
From: Ikey Doherty <ikey@solus-project.com>
Date: Fri, 28 Aug 2015 10:49:33 +0100
Subject: [PATCH] Do not lie about includedir location

Signed-off-by: Ikey Doherty <ikey@solus-project.com>
---
 libmodplug.pc.in | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/libmodplug.pc.in b/libmodplug.pc.in
index bbf05f9..1c6feb7 100644
--- a/libmodplug.pc.in
+++ b/libmodplug.pc.in
@@ -1,7 +1,7 @@
 prefix=@prefix@
 exec_prefix=@exec_prefix@
 libdir=@libdir@
-includedir=${prefix}/include
+includedir=${prefix}/include/libmodplug
 
 Name: libmodplug
 Description: The ModPlug mod file playing library.
-- 
2.5.0

