From 11d37df3bdc80a972c0359db3e1adcdd2c9c9fb3 Mon Sep 17 00:00:00 2001
From: "Bryan T. Meyers" <bmeyers@datadrake.com>
Date: Wed, 22 May 2019 21:33:52 -0400
Subject: [PATCH] Remove xterm as a dependency

---
 src/gsc_main_window.cpp | 4 ----
 1 file changed, 4 deletions(-)

diff --git a/src/gsc_main_window.cpp b/src/gsc_main_window.cpp
index d6423fa..d79c93d 100644
--- a/src/gsc_main_window.cpp
+++ b/src/gsc_main_window.cpp
@@ -1107,10 +1107,6 @@ void GscMainWindow::run_update_drivedb()
 	update_binary += "update-smart-drivedb";
 	update_binary = Glib::shell_quote(update_binary);
 
-#ifndef _WIN32
-	update_binary = "xterm -hold -e " + update_binary;
-#endif
-
 	hz::scoped_ptr<GError> spawn_error(0, g_error_free);
 	if (!g_spawn_command_line_async(update_binary.c_str(), &spawn_error.get_ref())) {
 		gui_show_error_dialog("Error Updating Drive Database", spawn_error->message, this);
-- 
2.21.0

