From 029b1c755057343c9e095bfb4e5c1d41fede70ae Mon Sep 17 00:00:00 2001
From: "Vladimir Golovnev (Glassez)" <glassez@yandex.ru>
Date: Wed, 27 Mar 2024 17:51:24 +0300
Subject: [PATCH] Prevent app from being closed when disabling system tray icon

---
 src/app/application.cpp | 1 +
 1 file changed, 1 insertion(+)

diff --git a/src/app/application.cpp b/src/app/application.cpp
index feaf2d245dc9..1d10731ed44a 100644
--- a/src/app/application.cpp
+++ b/src/app/application.cpp
@@ -251,6 +251,7 @@ Application::Application(int &argc, char **argv)
 #if !defined(DISABLE_GUI)
     setDesktopFileName(u"org.qbittorrent.qBittorrent"_s);
     setQuitOnLastWindowClosed(false);
+    setQuitLockEnabled(false);
     QPixmapCache::setCacheLimit(PIXMAP_CACHE_SIZE);
 #endif
 
