From 90af8d1805e63354683d18756bc7a3fd679bf2ee Mon Sep 17 00:00:00 2001
From: Ikey Doherty <ikey@solus-project.com>
Date: Thu, 10 Dec 2015 03:55:00 +0000
Subject: [PATCH 1/2] Fix Makefile flags

Signed-off-by: Ikey Doherty <ikey@solus-project.com>
---
 Makefile | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/Makefile b/Makefile
index 5ffb952..34a850b 100644
--- a/Makefile
+++ b/Makefile
@@ -15,7 +15,7 @@ PROFILE        := NO
 
 # TINYXML_USE_STL can be used to turn on STL support. NO, then STL
 # will not be used. YES will include the STL files.
-TINYXML_USE_STL := NO
+TINYXML_USE_STL := YES
 
 #****************************************************************************
 
@@ -26,7 +26,7 @@ AR     := ar rc
 RANLIB := ranlib
 
 DEBUG_CFLAGS     := -Wall -Wno-format -g -DDEBUG
-RELEASE_CFLAGS   := -Wall -Wno-unknown-pragmas -Wno-format -O3
+RELEASE_CFLAGS   := $(CFLAGS) -Wall -Wno-unknown-pragmas -Wno-format -O3
 
 LIBS		 :=
 
-- 
2.6.4

