From bbc6f991fe72164db5d3d8730fcf24a738ce744a Mon Sep 17 00:00:00 2001
From: Ikey Doherty <ikey@solus-project.com>
Date: Thu, 10 Dec 2015 03:57:50 +0000
Subject: [PATCH 2/2] Force define STL usage

Signed-off-by: Ikey Doherty <ikey@solus-project.com>
---
 tinyxml.h | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/tinyxml.h b/tinyxml.h
index a3589e5..a6201b6 100644
--- a/tinyxml.h
+++ b/tinyxml.h
@@ -26,6 +26,8 @@ distribution.
 #ifndef TINYXML_INCLUDED
 #define TINYXML_INCLUDED
 
+#define TIXML_USE_STL 1
+
 #ifdef _MSC_VER
 #pragma warning( push )
 #pragma warning( disable : 4530 )
-- 
2.6.4

