From 620826cf1c44faa553d9272385544fd547df6e20 Mon Sep 17 00:00:00 2001
From: "flinux@vongellhorn.ch" <flinux@vongellhorn.ch>
Date: Sat, 27 Jul 2019 19:06:32 +0200
Subject: [PATCH 1/1] Fix desktop

---
 gui/resources/solarus.desktop | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/gui/resources/solarus.desktop b/gui/resources/solarus.desktop
index 65b456c..1ef10c7 100644
--- a/gui/resources/solarus.desktop
+++ b/gui/resources/solarus.desktop
@@ -4,7 +4,7 @@ Type=Application
 Name=Solarus Quest Launcher
 GenericName=Topdown 2D adventure game
 Comment=A graphical user interface to organize and play quests for the Solarus engine
-Exec=solarus
+Exec=solarus-launcher
 Icon=solarus
 Terminal=false
 StartupNotify=true
-- 
2.22.0

