From 63804e8adcf2e6f07e00cf414c7ac24075fb2a77 Mon Sep 17 00:00:00 2001
From: pancake <pancake@nopcode.org>
Date: Wed, 22 Jul 2020 12:39:47 +0200
Subject: [PATCH] Fix missing grep glitch in r2pm

---
 binr/r2pm/r2pm | 6 ++----
 1 file changed, 2 insertions(+), 4 deletions(-)

diff --git a/binr/r2pm/r2pm b/binr/r2pm/r2pm
index e1b4887fd6c..7856ee685a1 100755
--- a/binr/r2pm/r2pm
+++ b/binr/r2pm/r2pm
@@ -333,12 +333,10 @@ R2PM_GIT() {
 		git clone --depth 1 --recursive "${URL}" "${DIR}" || R2PM_FAIL "Oops"
 		cd "${DIR}" || R2PM_FAIL "Oops"
 	fi
-	# r2>=4.5 : R2_VERSION=$(r2 -H R2_VERSION)
-	R2_VERSION=$(r2 -qv)
-	if [ -z "`echo $R2_VERSION | grep git`" ]; then
+	r2 -qv | grep -q git || { # r2 >= 4.5
 		git checkout "r2-${R2_VERSION}"
 		git pull
-	fi
+	}
 }
 
 R2PM_SVN() {
