From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: liushuyu <liushuyu011@gmail.com>
Date: Sat, 2 Nov 2024 13:40:28 +0100
Subject: [PATCH 04/11] gen/dibuilder.cpp: use the newer debug record API

---
 gen/dibuilder.cpp | 11 +++++------
 1 file changed, 5 insertions(+), 6 deletions(-)

diff --git a/gen/dibuilder.cpp b/gen/dibuilder.cpp
index f93656a11..1ae565896 100644
--- a/gen/dibuilder.cpp
+++ b/gen/dibuilder.cpp
@@ -1137,14 +1137,13 @@ void DIBuilder::EmitValue(llvm::Value *val, VarDeclaration *vd) {
   if (!mustEmitFullDebugInfo() || !debugVariable)
     return;
 
-  llvm::Instruction *instr = DBuilder.insertDbgValueIntrinsic(
+  auto instr = DBuilder.insertDbgValueIntrinsic(
       val, debugVariable, DBuilder.createExpression(),
-      IR->ir->getCurrentDebugLocation(), IR->scopebb())
-#if LDC_LLVM_VER >= 1901
-  .dyn_cast<llvm::Instruction *>()
+      IR->ir->getCurrentDebugLocation(), IR->scopebb());
+#if LDC_LLVM_VER >= 1900
+  llvm::cast<llvm::DbgRecord *>
 #endif
-    ;
-  instr->setDebugLoc(IR->ir->getCurrentDebugLocation());
+  (instr)->setDebugLoc(IR->ir->getCurrentDebugLocation());
 }
 
 void DIBuilder::EmitLocalVariable(llvm::Value *ll, VarDeclaration *vd,
