From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: liushuyu <liushuyu011@gmail.com>
Date: Tue, 5 Nov 2024 17:54:16 -0700
Subject: [PATCH 10/11] gen/dcompute/targetOCL.cpp: update target data layout
 for LLVM 19

---
 gen/dcompute/targetOCL.cpp | 11 +++++++++++
 1 file changed, 11 insertions(+)

diff --git a/gen/dcompute/targetOCL.cpp b/gen/dcompute/targetOCL.cpp
index 0bc69a494..ce2c6c82a 100644
--- a/gen/dcompute/targetOCL.cpp
+++ b/gen/dcompute/targetOCL.cpp
@@ -24,6 +24,7 @@
 #include <string>
 
 // from SPIRVInternal.h
+#if LDC_LLVM_VER < 1900
 #define SPIR_TARGETTRIPLE32 "spir-unknown-unknown"
 #define SPIR_TARGETTRIPLE64 "spir64-unknown-unknown"
 #define SPIR_DATALAYOUT32                                                      \
@@ -38,6 +39,16 @@
   "-v32:32:32-v48:64:64-v64:64:64-v96:128:128"                                 \
   "-v128:128:128-v192:256:256-v256:256:256"                                    \
   "-v512:512:512-v1024:1024:1024"
+#else
+#define SPIR_TARGETTRIPLE32 "spirv-unknown-unknown"
+#define SPIR_TARGETTRIPLE64 "spirv64-unknown-unknown"
+#define SPIR_DATALAYOUT32                                                      \
+  "e-p:32:32-i64:64-v16:16-v24:32-v32:32-v48:64"                               \
+  "-v96:128-v192:256-v256:256-v512:512-v1024:1024-G1"
+#define SPIR_DATALAYOUT64                                                      \
+  "e-i64:64-v16:16-v24:32-v32:32-v48:64-v96:128"                               \
+  "-v192:256-v256:256-v512:512-v1024:1024-G1"
+#endif
 
 namespace {
 class TargetOCL : public DComputeTarget {
