--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -140,17 +140,6 @@ include (pythonutils)
 # Dependency finding utilities and all dependency-related options
 include (externalpackages)
 
-# Include all our testing apparatus and utils, but not if it's a subproject
-if (PROJECT_IS_TOP_LEVEL)
-    include (testing)
-else ()
-    macro (oiio_add_tests)
-    endmacro ()
-    macro (oiio_add_all_tests)
-    endmacro ()
-endif ()
-
-
 include_directories (
     BEFORE
     "${CMAKE_SOURCE_DIR}/src/include"
@@ -283,10 +272,5 @@ install (EXPORT OIIO_EXPORTED_TARGETS
 
 
 if (PROJECT_IS_TOP_LEVEL)
-    oiio_setup_test_data()
-    oiio_add_all_tests()
-endif ()
-
-if (PROJECT_IS_TOP_LEVEL)
     include (packaging)
 endif ()
