--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -161,17 +161,6 @@ include (pythonutils)
 # Dependency finding utilities and all dependency-related options
 include (externalpackages)
 
-# Include all our testing apparatus and utils, but not if it's a subproject
-if (PROJECT_IS_TOP_LEVEL)
-    include (testing)
-else ()
-    macro (oiio_add_tests)
-    endmacro ()
-    macro (oiio_add_all_tests)
-    endmacro ()
-endif ()
-
-
 include_directories (
     BEFORE
     "${CMAKE_SOURCE_DIR}/src/include"
@@ -301,11 +290,6 @@ install (EXPORT OIIO_EXPORTED_TARGETS
         NAMESPACE ${PROJECT_NAME}::)
 
 
-if (PROJECT_IS_TOP_LEVEL AND BUILD_TESTING)
-    oiio_setup_test_data()
-    oiio_add_all_tests()
-endif ()
-
 if (PROJECT_IS_TOP_LEVEL)
     include (packaging)
 endif ()
-- 
2.39.0
