--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -120,15 +120,10 @@ include (pythonutils)
 # Dependency finding utilities and all dependency-related options
 include (externalpackages)
 
-# Include all our testing apparatus and utils, but not if it's a subproject
-if (NOT ${PROJECT_NAME}_IS_SUBPROJECT)
-    include (testing)
-else ()
-    macro (oiio_add_tests)
-    endmacro ()
-    macro (oiio_add_all_tests)
-    endmacro ()
-endif ()
+macro (oiio_add_tests)
+endmacro ()
+macro (oiio_add_all_tests)
+endmacro ()
 
 
 include_directories (
