--- a/zsh_tealdeer
+++ b/zsh_tealdeer
@@ -1,7 +1,9 @@
 #compdef tldr
 
 _applications() {
-  _values 'applications' "${(uonzf)$(tldr --list | sed -e 's/, /\n/g')}"
+    local -a commands
+    commands=(${(uonzf)"$(tldr --list 2>/dev/null)"//:/\\:})
+    _describe -t commands 'command' commands
 }
 
 _tealdeer() {
@@ -31,8 +33,7 @@ _tealdeer() {
             never
         ))"
         '(- *)'{-h,--help}'[Display help]'
-            '(- *)'{-v,--version}'[Show version information]'
-            '*:file:_files'
+        '(- *)'{-v,--version}'[Show version information]'
         '1: :_applications'
     )
 
