From 0ee672a392e251b54091fcb125e07b0def410b47 Mon Sep 17 00:00:00 2001
From: Algent Albrahimi <algent@protonmail.com>
Date: Fri, 25 Dec 2020 20:53:34 +0100
Subject: [PATCH] Fix version tag

---
 setup.py | 10 +++++++++-
 1 file changed, 9 insertions(+), 1 deletion(-)

diff --git a/setup.py b/setup.py
index 6068493..3e51f8b 100644
--- a/setup.py
+++ b/setup.py
@@ -1,3 +1,11 @@
 from setuptools import setup
 
-setup()
+setup(
+    use_scm_version={
+        'version_scheme': 'post-release',
+        'local_scheme': 'dirty-tag'
+    },
+    setup_requires=[
+        'setuptools_scm >= 1.7.0'
+    ]
+)
-- 
2.29.2

