From ea4dfd70ce4ae3d62b513b07764859d5281927bc Mon Sep 17 00:00:00 2001
From: Pierre-Yves <pyu@riseup.net>
Date: Fri, 30 Nov 2018 15:03:13 +0100
Subject: [PATCH] test-exported is not allowed in our sandbox

Signed-off-by: Pierre-Yves <pyu@riseup.net>
---
 meson.build | 2 --
 1 file changed, 2 deletions(-)

diff --git a/meson.build b/meson.build
index 0bc111b..76893c9 100644
--- a/meson.build
+++ b/meson.build
@@ -372,8 +372,6 @@ tests = [
   ['test-auth', [libdaemon]],
   ['test-common', [], test_enums],
   ['test-device', [libdaemon]],
-  ['test-exported', [libdaemon], [test_resources,
-                                  test_enums]],
   ['test-logging', [libdaemon]],
   ['test-store', [libdaemon]],
   ['test-journal', [libdaemon]]
-- 
2.19.2

