From 15d35148158ce096d752f49eab12fff2fdb64bc2 Mon Sep 17 00:00:00 2001
From: Pierre-Yves <pyu@riseup.net>
Date: Mon, 30 Jul 2018 22:03:06 +0200
Subject: [PATCH] test-exported is not allowed in our sandbox

Signed-off-by: Pierre-Yves <pyu@riseup.net>
---
 meson.build | 1 -
 1 file changed, 1 deletion(-)

diff --git a/meson.build b/meson.build
index a4ab5dc..5231999 100644
--- a/meson.build
+++ b/meson.build
@@ -337,7 +337,6 @@ test_enums = gnome.mkenums_simple(
 
 tests = [
   ['test-common', [], test_enums],
-  ['test-exported', [libdaemon], [test_resources]],
   ['test-logging', [libdaemon]],
   ['test-store', [libdaemon]]
 ]
-- 
2.18.0

