From 2f9ea37c0e5ffd7f2dcb36f343a2e9d8cd7110e9 Mon Sep 17 00:00:00 2001
From: Algent Albrahimi <algent@protonmail.com>
Date: Mon, 14 Feb 2022 20:22:58 +0100
Subject: [PATCH] use system data dir

---
 src/platform/tiles.c | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/src/platform/tiles.c b/src/platform/tiles.c
index 4ff2c8b..ee5cd71 100644
--- a/src/platform/tiles.c
+++ b/src/platform/tiles.c
@@ -445,7 +445,7 @@ static void optimizeTiles() {
 /// Loads the PNG and analyses it.
 void initTiles() {
     char filename[BROGUE_FILENAME_MAX];
-    sprintf(filename, "%s/assets/tiles.png", dataDirectory);
+    sprintf(filename, "/usr/share/brogue/assets/tiles.png", dataDirectory);
 
     // are we running Brogue from the correct folder to begin with?
     if (!fileExists(filename)) {
@@ -469,7 +469,7 @@ void initTiles() {
     }
 
     // load shifts
-    sprintf(filename, "%s/assets/tiles.bin", dataDirectory);
+    sprintf(filename, "/usr/share/brogue/assets/tiles.bin", dataDirectory);
     FILE *file = fopen(filename, "rb");
     if (file) {
         fread(tileShifts, 1, sizeof(tileShifts), file);
@@ -759,7 +759,7 @@ void resizeWindow(int width, int height) {
 
         // set its icon
         char filename[BROGUE_FILENAME_MAX];
-        sprintf(filename, "%s/assets/icon.png", dataDirectory);
+        sprintf(filename, "/usr/share/brogue/assets/icon.png", dataDirectory);
         SDL_Surface *icon = IMG_Load(filename);
         if (!icon) imgfatal(__FILE__, __LINE__);
         SDL_SetWindowIcon(Win, icon);
-- 
2.35.1

