--- a/brogue.desktop.orig	2016-11-12 23:23:09.448514852 -0800
+++ b/brogue.desktop	2016-11-12 23:23:39.618362622 -0800
@@ -1,11 +1,10 @@
 [Desktop Entry]
 Version=1.0
 Type=Application
-Name=Brogue 1.7.2
+Name=Brogue
 GenericName=Roguelike
 Comment=Brave the Dungeons of Doom!
-Exec=/home/o/brogue-linux-1.7.2/brogue-1.7.2/brogue
-Path=/home/o/brogue-linux-1.7.2/brogue-1.7.2
-Icon=/home/o/brogue-linux-1.7.2/brogue-1.7.2/bin/brogue-icon.png
+Exec=/usr/bin/brogue
+Icon=/usr/share/pixmaps/brogue-icon.png
 Terminal=false
 
