--- a/Makefile.orig	2016-11-12 22:59:37.126194905 -0800
+++ b/Makefile	2016-11-12 23:02:53.503616223 -0800
@@ -41,9 +41,9 @@
 	src/platform/tcod-platform.o \
 	src/platform/term.o
 
-TCOD_DEF = -DBROGUE_TCOD -I$(LIBTCODDIR)/include
-TCOD_DEP = ${LIBTCODDIR}
-TCOD_LIB = -L. -L${LIBTCODDIR} ${SDL_FLAGS} -ltcod -Wl,-rpath,.
+TCOD_DEF = -DBROGUE_TCOD -I/usr/include/libtcod
+TCOD_DEP = 
+TCOD_LIB = -L. ${SDL_FLAGS} -ltcod -lSDL -lm -Wl,-rpath,.
 
 CURSES_DEF = -DBROGUE_CURSES
 CURSES_LIB = -lncurses -lm
