--- a/src/platform/tcod-platform.c.orig	2016-11-12 23:19:43.719267526 -0800
+++ a/src/platform/tcod-platform.c	2016-11-12 23:20:08.137213006 -0800
@@ -51,7 +51,7 @@
 
 	}
 
-	sprintf(font, "fonts/font-%i.png", brogueFontSize);
+	sprintf(font, "/usr/share/brogue/fonts/font-%i.png", brogueFontSize);
 	
 	TCOD_console_set_custom_font(font, (TCOD_FONT_TYPE_GREYSCALE | TCOD_FONT_LAYOUT_ASCII_INROW), 0, 0);
 	TCOD_console_init_root(COLS, ROWS, "Brogue", false, renderer);
