From 240a0827cbb74cb46890230f38fab0c344cb3847 Mon Sep 17 00:00:00 2001
From: Pierre-Yves <pyu@riseup.net>
Date: Sun, 1 Aug 2021 13:00:29 +0200
Subject: [PATCH] QGLWidget: No such file or directory

Signed-off-by: Pierre-Yves <pyu@riseup.net>
---
 configure | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/configure b/configure
index 91bc0c0..cf5772d 100755
--- a/configure
+++ b/configure
@@ -1561,7 +1561,7 @@ EOF
         qt5_ver=`pkg-config Qt5Core --modversion 2>> $LOG`
         if [ $? -eq 0 ]; then
           echo "(Qt5 from pkg-config)" >> $LOG
-          FLGQT5=`pkg-config Qt5Core Qt5Gui --cflags`
+          FLGQT5=`pkg-config Qt5Core --cflags`
           # gcc 4.8 is able to compile with C++11 (see also VBOX_GCC_std in Config.kmk)
           [ $(($cc_maj * 100 + $cc_min)) -ge 408 ] && FLGQT5="$FLGQT5 -std=c++11"
           INCQT5=`strip_I "$FLGQT5"`
-- 
2.32.0

