diff -u -r VBox/src/VBox/Additions/linux/drm/vbox_ttm.c VBox-new/src/VBox/Additions/linux/drm/vbox_ttm.c
--- a/src/VBox/Additions/linux/drm/vbox_ttm.c
+++ b/src/VBox/Additions/linux/drm/vbox_ttm.c
@@ -209,8 +209,13 @@
 };
 
+#if LINUX_VERSION_CODE < KERNEL_VERSION(4, 17, 0)
 static struct ttm_tt *vbox_ttm_tt_create(struct ttm_bo_device *bdev,
 					 unsigned long size,
 					 u32 page_flags,
 					 struct page *dummy_read_page)
+#else
+static struct ttm_tt *vbox_ttm_tt_create(struct ttm_buffer_object *bo,
+					 u32 page_flags)
+#endif
 {
 	struct ttm_tt *tt;
@@ -221,5 +226,9 @@
 
 	tt->func = &vbox_tt_backend_func;
+#if LINUX_VERSION_CODE < KERNEL_VERSION(4, 17, 0)
 	if (ttm_tt_init(tt, bdev, size, page_flags, dummy_read_page)) {
+#else
+	if (ttm_tt_init(tt, bo, page_flags)) {
+#endif
 		kfree(tt);
 		return NULL;
@@ -402,5 +411,9 @@
 	ret = ttm_bo_init(&vbox->ttm.bdev, &vboxbo->bo, size,
 			  ttm_bo_type_device, &vboxbo->placement,
+#if LINUX_VERSION_CODE < KERNEL_VERSION(4, 17, 0)
 			  align >> PAGE_SHIFT, false, NULL, acc_size,
+#else
+			  align >> PAGE_SHIFT, false, acc_size,
+#endif
 #if LINUX_VERSION_CODE >= KERNEL_VERSION(3, 18, 0) || defined(RHEL_73)
 			  NULL,

