diff -up VirtualBox-5.2.6/configure~ VirtualBox-5.2.6/configure
--- VirtualBox-5.2.6/configure~	2018-01-15 15:49:37.000000000 +0100
+++ VirtualBox-5.2.6/configure	2018-02-14 08:53:25.602712747 +0100
@@ -440,8 +440,8 @@ check_gcc()
       elif [ $cc_maj -lt 4 \
              -o \( $cc_maj -eq 4 -a $cc_min -lt 4 -a "$OS" != "darwin" \) \
              -o \( $cc_maj -eq 4 -a $cc_min -lt 2 -a "$OS"  = "darwin" \) \
-             -o $cc_maj -gt 7 ]; then
-        log_failure "gcc version $cc_maj.$cc_min found, expected gcc 4.x...7.x"
+             -o $cc_maj -gt 8 ]; then
+        log_failure "gcc version $cc_maj.$cc_min found, expected gcc 4.x...8.x"
         fail really
       else
         log_success "found version $cc_ver"
