Index: VirtualBox-6.1.0_BETA1/src/VBox/Frontends/VirtualBox/src/manager/tools/UITools.cpp
===================================================================
--- VirtualBox-6.1.0_BETA1.orig/src/VBox/Frontends/VirtualBox/src/manager/tools/UITools.cpp
+++ VirtualBox-6.1.0_BETA1/src/VBox/Frontends/VirtualBox/src/manager/tools/UITools.cpp
@@ -114,10 +114,6 @@ void UITools::preparePalette()
 {
     /* Setup palette: */
     setAutoFillBackground(true);
-    QPalette pal = palette();
-    QColor bodyColor = pal.color(QPalette::Active, QPalette::Midlight).darker(110);
-    pal.setColor(QPalette::Window, bodyColor);
-    setPalette(pal);
 }
 
 void UITools::prepareLayout()
Index: VirtualBox-6.1.0_BETA1/src/VBox/Frontends/VirtualBox/src/manager/UIVirtualBoxManager.cpp
===================================================================
--- VirtualBox-6.1.0_BETA1.orig/src/VBox/Frontends/VirtualBox/src/manager/UIVirtualBoxManager.cpp
+++ VirtualBox-6.1.0_BETA1/src/VBox/Frontends/VirtualBox/src/manager/UIVirtualBoxManager.cpp
@@ -1328,7 +1328,7 @@ void UIVirtualBoxManager::prepareMenuBar
         /* Make sure menu-bar fills own solid background: */
         menuBar()->setAutoFillBackground(true);
         QPalette pal = menuBar()->palette();
-        const QColor color = pal.color(QPalette::Active, QPalette::Mid).lighter(160);
+        const QColor color = pal.color(QPalette::Active, QPalette::Window);
         pal.setColor(QPalette::Active, QPalette::Button, color);
         menuBar()->setPalette(pal);
     }
Index: VirtualBox-6.1.0_BETA1/src/VBox/Frontends/VirtualBox/src/manager/chooser/UIChooser.cpp
===================================================================
--- VirtualBox-6.1.0_BETA1.orig/src/VBox/Frontends/VirtualBox/src/manager/chooser/UIChooser.cpp
+++ VirtualBox-6.1.0_BETA1/src/VBox/Frontends/VirtualBox/src/manager/chooser/UIChooser.cpp
@@ -121,10 +121,6 @@ void UIChooser::preparePalette()
 {
     /* Setup palette: */
     setAutoFillBackground(true);
-    QPalette pal = palette();
-    QColor bodyColor = pal.color(QPalette::Active, QPalette::Midlight).darker(110);
-    pal.setColor(QPalette::Window, bodyColor);
-    setPalette(pal);
 }
 
 void UIChooser::prepareLayout()
Index: VirtualBox-6.1.0_BETA1/src/VBox/Frontends/VirtualBox/src/manager/chooser/UIChooserItemMachine.cpp
===================================================================
--- VirtualBox-6.1.0_BETA1.orig/src/VBox/Frontends/VirtualBox/src/manager/chooser/UIChooserItemMachine.cpp
+++ VirtualBox-6.1.0_BETA1/src/VBox/Frontends/VirtualBox/src/manager/chooser/UIChooserItemMachine.cpp
@@ -902,7 +902,7 @@ void UIChooserItemMachine::paintBackgrou
     else
     {
         /* Prepare color: */
-        QColor backgroundColor = pal.color(QPalette::Active, QPalette::Mid);
+        QColor backgroundColor = pal.color(QPalette::Active, QPalette::Button);
         /* Draw gradient: */
         QLinearGradient bgGrad(rectangle.topLeft(), rectangle.bottomLeft());
         bgGrad.setColorAt(0, backgroundColor.lighter(m_iDefaultLightnessMax));
@@ -931,7 +931,7 @@ void UIChooserItemMachine::paintBackgrou
             /* Default background: */
             else
             {
-                QColor backgroundColor = pal.color(QPalette::Active, QPalette::Mid);
+                QColor backgroundColor = pal.color(QPalette::Active, QPalette::Button);
                 color1 = backgroundColor.lighter(m_iDefaultLightnessMax);
                 color2 = backgroundColor.lighter(m_iDefaultLightnessMax - 70);
             }
@@ -952,7 +952,7 @@ void UIChooserItemMachine::paintBackgrou
             /* Default background: */
             else
             {
-                QColor backgroundColor = pal.color(QPalette::Active, QPalette::Mid);
+                QColor backgroundColor = pal.color(QPalette::Active, QPalette::Button);
                 color1 = backgroundColor.lighter(m_iDefaultLightnessMin);
                 color2 = backgroundColor.lighter(m_iDefaultLightnessMin - 40);
             }
@@ -987,7 +987,7 @@ void UIChooserItemMachine::paintFrame(QP
         strokeColor = pal.color(QPalette::Active, QPalette::Highlight).lighter(m_iHoverLightnessMin - 50);
     /* Default frame: */
     else
-        strokeColor = pal.color(QPalette::Active, QPalette::Mid).lighter(m_iDefaultLightnessMin);
+        strokeColor = pal.color(QPalette::Active, QPalette::Button).lighter(m_iDefaultLightnessMin);
 
     /* Create/assign pen: */
     QPen pen(strokeColor);
Index: VirtualBox-6.1.0_BETA1/src/VBox/Frontends/VirtualBox/src/manager/tools/UIToolsItem.cpp
===================================================================
--- VirtualBox-6.1.0_BETA1.orig/src/VBox/Frontends/VirtualBox/src/manager/tools/UIToolsItem.cpp
+++ VirtualBox-6.1.0_BETA1/src/VBox/Frontends/VirtualBox/src/manager/tools/UIToolsItem.cpp
@@ -710,7 +710,7 @@ void UIToolsItem::paintBackground(QPaint
         /* Prepare color: */
         const QColor backgroundColor = isEnabled()
                                      ? pal.color(QPalette::Active, QPalette::Highlight)
-                                     : pal.color(QPalette::Disabled, QPalette::Midlight);
+                                     : pal.color(QPalette::Disabled, QPalette::Highlight);
         /* Draw gradient: */
         QLinearGradient bgGrad(rectangle.topLeft(), rectangle.bottomLeft());
         bgGrad.setColorAt(0, backgroundColor.lighter(m_iHighlightLightnessMax));
@@ -749,7 +749,7 @@ void UIToolsItem::paintBackground(QPaint
         /* Prepare color: */
         const QColor backgroundColor = isEnabled()
                                      ? pal.color(QPalette::Active, QPalette::Highlight)
-                                     : pal.color(QPalette::Disabled, QPalette::Midlight);
+                                     : pal.color(QPalette::Disabled, QPalette::Highlight);
         /* Draw gradient: */
         QLinearGradient bgGrad(rectangle.topLeft(), rectangle.bottomLeft());
         bgGrad.setColorAt(0, backgroundColor.lighter(m_iHoverLightnessMax));
@@ -787,8 +787,8 @@ void UIToolsItem::paintBackground(QPaint
     {
         /* Prepare color: */
         const QColor backgroundColor = isEnabled()
-                                     ? pal.color(QPalette::Active, QPalette::Mid)
-                                     : pal.color(QPalette::Disabled, QPalette::Midlight);
+                                     ? pal.color(QPalette::Active, QPalette::Window)
+                                     : pal.color(QPalette::Disabled, QPalette::Window);
         /* Draw gradient: */
         QLinearGradient bgGrad(rectangle.topLeft(), rectangle.bottomLeft());
         bgGrad.setColorAt(0, backgroundColor.lighter(m_iDefaultLightnessMax));
@@ -821,7 +821,7 @@ void UIToolsItem::paintFrame(QPainter *p
         strokeColor = pal.color(QPalette::Active, QPalette::Highlight).lighter(m_iHoverLightnessMin - 50);
     /* Default frame: */
     else
-        strokeColor = pal.color(QPalette::Active, QPalette::Mid).lighter(m_iDefaultLightnessMin);
+        strokeColor = pal.color(QPalette::Active, QPalette::Button).lighter(m_iDefaultLightnessMin);
 
     /* Create/assign pen: */
     QPen pen(strokeColor);
Index: VirtualBox-6.1.0_BETA1/src/VBox/Frontends/VirtualBox/src/manager/UIVirtualBoxManagerWidget.cpp
===================================================================
--- VirtualBox-6.1.0_BETA1.orig/src/VBox/Frontends/VirtualBox/src/manager/UIVirtualBoxManagerWidget.cpp
+++ VirtualBox-6.1.0_BETA1/src/VBox/Frontends/VirtualBox/src/manager/UIVirtualBoxManagerWidget.cpp
@@ -314,14 +314,6 @@ void UIVirtualBoxManagerWidget::prepare(
 {
     /* Configure palette: */
     setAutoFillBackground(true);
-    QPalette pal = palette();
-#ifdef VBOX_WS_MAC
-    const QColor color = pal.color(QPalette::Active, QPalette::Mid).lighter(145);
-#else
-    const QColor color = pal.color(QPalette::Active, QPalette::Mid).lighter(160);
-#endif
-    pal.setColor(QPalette::Window, color);
-    setPalette(pal);
 
     /* Prepare: */
     prepareWidgets();
@@ -449,7 +441,7 @@ void UIVirtualBoxManagerWidget::prepareW
             }
 
             /* Adjust splitter colors according to main widgets it splits: */
-            m_pSplitter->configureColor(palette().color(QPalette::Active, QPalette::Midlight).darker(110));
+            m_pSplitter->configureColor(palette().color(QPalette::Active, QPalette::Button).darker(110));
             /* Set the initial distribution. The right site is bigger. */
             m_pSplitter->setStretchFactor(0, 2);
             m_pSplitter->setStretchFactor(1, 3);
Index: VirtualBox-6.1.0_BETA1/src/VBox/Frontends/VirtualBox/src/manager/chooser/UIChooserItem.cpp
===================================================================
--- VirtualBox-6.1.0_BETA1.orig/src/VBox/Frontends/VirtualBox/src/manager/chooser/UIChooserItem.cpp
+++ VirtualBox-6.1.0_BETA1/src/VBox/Frontends/VirtualBox/src/manager/chooser/UIChooserItem.cpp
@@ -643,7 +643,7 @@ void UIChooserItem::paintFlatButton(QPai
 
     /* Prepare colors: */
     const QPalette pal = QApplication::palette();
-    const QColor color = pal.color(QPalette::Active, QPalette::Mid);
+    const QColor color = pal.color(QPalette::Active, QPalette::Button);
 
     /* Prepare pen: */
     QPen pen;
Index: VirtualBox-6.1.0_BETA1/src/VBox/Frontends/VirtualBox/src/manager/chooser/UIChooserView.cpp
===================================================================
--- VirtualBox-6.1.0_BETA1.orig/src/VBox/Frontends/VirtualBox/src/manager/chooser/UIChooserView.cpp
+++ VirtualBox-6.1.0_BETA1/src/VBox/Frontends/VirtualBox/src/manager/chooser/UIChooserView.cpp
@@ -245,10 +245,6 @@ void UIChooserView::prepare()
 void UIChooserView::preparePalette()
 {
     /* Setup palette: */
-    QPalette pal = qApp->palette();
-    const QColor bodyColor = pal.color(QPalette::Active, QPalette::Midlight).darker(110);
-    pal.setColor(QPalette::Base, bodyColor);
-    setPalette(pal);
 }
 
 void UIChooserView::resizeEvent(QResizeEvent *pEvent)
Index: VirtualBox-6.1.0_BETA1/src/VBox/Frontends/VirtualBox/src/manager/tools/UIToolsView.cpp
===================================================================
--- VirtualBox-6.1.0_BETA1.orig/src/VBox/Frontends/VirtualBox/src/manager/tools/UIToolsView.cpp
+++ VirtualBox-6.1.0_BETA1/src/VBox/Frontends/VirtualBox/src/manager/tools/UIToolsView.cpp
@@ -182,10 +182,6 @@ void UIToolsView::prepare()
 void UIToolsView::preparePalette()
 {
     /* Setup palette: */
-    QPalette pal = qApp->palette();
-    const QColor bodyColor = pal.color(QPalette::Active, QPalette::Midlight).darker(110);
-    pal.setColor(QPalette::Base, bodyColor);
-    setPalette(pal);
 }
 
 void UIToolsView::resizeEvent(QResizeEvent *pEvent)
Index: VirtualBox-6.1.0_BETA1/src/VBox/Frontends/VirtualBox/src/manager/details/UIDetailsElement.cpp
===================================================================
--- VirtualBox-6.1.0_BETA1.orig/src/VBox/Frontends/VirtualBox/src/manager/details/UIDetailsElement.cpp
+++ VirtualBox-6.1.0_BETA1/src/VBox/Frontends/VirtualBox/src/manager/details/UIDetailsElement.cpp
@@ -1183,7 +1183,7 @@ void UIDetailsElement::paintBackground(Q
     const QPalette pal = palette();
 
     /* Paint default background: */
-    const QColor defaultColor = pal.color(QPalette::Active, QPalette::Mid);
+    const QColor defaultColor = pal.color(QPalette::Active, QPalette::Window);
     const QColor dcTone1 = defaultColor.lighter(m_iDefaultToneFinal);
     const QColor dcTone2 = defaultColor.lighter(m_iDefaultToneStart);
     QLinearGradient gradientDefault(fullRect.topLeft(), fullRect.bottomLeft());
@@ -1224,7 +1224,7 @@ void UIDetailsElement::paintFrame(QPaint
                           : optionRect;
 
     /* Paint frame: */
-    const QColor strokeColor = palette().color(QPalette::Active, QPalette::Mid).lighter(m_iDefaultToneStart);
+    const QColor strokeColor = palette().color(QPalette::Active, QPalette::Window).lighter(m_iDefaultToneStart);
     QPen pen(strokeColor);
     pen.setWidth(0);
     pPainter->setPen(pen);
Index: VirtualBox-6.1.0_BETA1/src/VBox/Frontends/VirtualBox/src/widgets/UIToolBar.cpp
===================================================================
--- VirtualBox-6.1.0_BETA1.orig/src/VBox/Frontends/VirtualBox/src/widgets/UIToolBar.cpp
+++ VirtualBox-6.1.0_BETA1/src/VBox/Frontends/VirtualBox/src/widgets/UIToolBar.cpp
@@ -110,7 +110,7 @@ void UIToolBar::paintEvent(QPaintEvent *
         const QRect rectangle = pEvent->rect();
 
         /* Prepare gradient: */
-        const QColor backgroundColor = palette().color(QPalette::Active, QPalette::Mid);
+        const QColor backgroundColor = palette().color(QPalette::Active, QPalette::Window);
         QLinearGradient gradient(rectangle.topLeft(), rectangle.bottomLeft());
         gradient.setColorAt(0,   backgroundColor.lighter(130));
         gradient.setColorAt(1,   backgroundColor.lighter(125));
Index: VirtualBox-6.1.0_BETA1/src/VBox/Frontends/VirtualBox/src/manager/details/UIDetailsSet.cpp
===================================================================
--- VirtualBox-6.1.0_BETA1.orig/src/VBox/Frontends/VirtualBox/src/manager/details/UIDetailsSet.cpp
+++ VirtualBox-6.1.0_BETA1/src/VBox/Frontends/VirtualBox/src/manager/details/UIDetailsSet.cpp
@@ -700,7 +700,7 @@ void UIDetailsSet::paintBackground(QPain
     const QRect optionRect = pOptions->rect;
 
     /* Paint default background: */
-    const QColor defaultColor = palette().color(QPalette::Active, QPalette::Midlight).darker(110);
+    const QColor defaultColor = palette().color(QPalette::Active, QPalette::Window).darker(110);
     pPainter->fillRect(optionRect, defaultColor);
 
     /* Restore painter: */
Index: VirtualBox-6.1.0_BETA1/src/VBox/Frontends/VirtualBox/src/manager/chooser/UIChooserItemGlobal.cpp
===================================================================
--- VirtualBox-6.1.0_BETA1.orig/src/VBox/Frontends/VirtualBox/src/manager/chooser/UIChooserItemGlobal.cpp
+++ VirtualBox-6.1.0_BETA1/src/VBox/Frontends/VirtualBox/src/manager/chooser/UIChooserItemGlobal.cpp
@@ -639,7 +639,7 @@ void UIChooserItemGlobal::paintBackgroun
         pPainter->fillRect(rectangle, bgGrad);
 #else
         /* Prepare color: */
-        QColor backgroundColor = pal.color(QPalette::Active, QPalette::Mid).lighter(160);
+        QColor backgroundColor = pal.color(QPalette::Active, QPalette::Window).lighter(160);
         /* Draw gradient: */
         pPainter->fillRect(rectangle, backgroundColor);
 #endif
@@ -670,7 +670,7 @@ void UIChooserItemGlobal::paintFrame(QPa
         strokeColor = pal.color(QPalette::Active, QPalette::Highlight).lighter(m_iHoverLightnessMin - 50);
     /* Default frame: */
     else
-        strokeColor = pal.color(QPalette::Active, QPalette::Mid).lighter(m_iDefaultLightnessMin);
+        strokeColor = pal.color(QPalette::Active, QPalette::Window).lighter(m_iDefaultLightnessMin);
 
     /* Create/assign pen: */
     QPen pen(strokeColor);
Index: VirtualBox-6.1.0_BETA1/src/VBox/Frontends/VirtualBox/src/manager/details/UIDetailsGroup.cpp
===================================================================
--- VirtualBox-6.1.0_BETA1.orig/src/VBox/Frontends/VirtualBox/src/manager/details/UIDetailsGroup.cpp
+++ VirtualBox-6.1.0_BETA1/src/VBox/Frontends/VirtualBox/src/manager/details/UIDetailsGroup.cpp
@@ -277,7 +277,7 @@ void UIDetailsGroup::paintBackground(QPa
 #ifdef VBOX_WS_MAC
     const QColor defaultColor = palette().color(QPalette::Active, QPalette::Mid).lighter(145);
 #else
-    const QColor defaultColor = palette().color(QPalette::Active, QPalette::Mid).lighter(160);
+    const QColor defaultColor = palette().color(QPalette::Active, QPalette::Window).lighter(160);
 #endif
     pPainter->fillRect(optionRect, defaultColor);
 
Index: VirtualBox-6.1.0_BETA1/src/VBox/Frontends/VirtualBox/src/manager/chooser/UIChooserItemGroup.cpp
===================================================================
--- VirtualBox-6.1.0_BETA1.orig/src/VBox/Frontends/VirtualBox/src/manager/chooser/UIChooserItemGroup.cpp
+++ VirtualBox-6.1.0_BETA1/src/VBox/Frontends/VirtualBox/src/manager/chooser/UIChooserItemGroup.cpp
@@ -1499,7 +1499,7 @@ void UIChooserItemGroup::paintBackground
     const QPalette pal = palette();
     const QColor headerColor = pal.color(QPalette::Active,
                                          model()->selectedItems().contains(this) ?
-                                         QPalette::Highlight : QPalette::Midlight);
+                                         QPalette::Highlight : QPalette::Button);
 
     /* Root-item: */
     if (isRoot())
@@ -1596,7 +1596,7 @@ void UIChooserItemGroup::paintFrame(QPai
     const QPalette pal = palette();
     const QColor strokeColor = pal.color(QPalette::Active,
                                          model()->selectedItems().contains(this) ?
-                                         QPalette::Highlight : QPalette::Midlight).darker(headerDarkness() + 10);
+                                         QPalette::Highlight : QPalette::Button).darker(headerDarkness() + 10);
 
     /* Create/assign pen: */
     QPen pen(strokeColor);

