From 0297cb9ba5a33eab15173b217e48756008a8791f Mon Sep 17 00:00:00 2001
From: "F. von Gellhorn" <flinux@vongellhorn.ch>
Date: Sat, 16 Jan 2021 15:19:02 +0100
Subject: [PATCH 1/1] Adjust udev rules locations

---
 CMakeLists.txt | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 7325f78..8c3ec1f 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -101,7 +101,7 @@ install(FILES
 
 if(${CMAKE_SYSTEM_NAME} MATCHES "Linux")
     set(INSTALL_UDEV_RULE ON CACHE BOOL "Whether to install udev rule")
-    set(UDEV_RULES_INSTALL_DIR /lib/udev/rules.d CACHE PATH "The location of udev rules")
+    set(UDEV_RULES_INSTALL_DIR /usr/lib64/udev/rules.d CACHE PATH "The location of udev rules")
 
     if(${INSTALL_UDEV_RULE})
         install(FILES 61-kde-bluetooth-rfkill.rules DESTINATION ${UDEV_RULES_INSTALL_DIR})
-- 
2.30.0

