From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: "Luke D. Jones" <luke@ljones.dev>
Date: Wed, 7 Aug 2024 17:16:57 +1200
Subject: [PATCH 17/17] Ally led report ID change

---
 drivers/hid/hid-asus-ally.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/drivers/hid/hid-asus-ally.c b/drivers/hid/hid-asus-ally.c
index 9e8007fc3c19c..fd3d23b2c2841 100644
--- a/drivers/hid/hid-asus-ally.c
+++ b/drivers/hid/hid-asus-ally.c
@@ -1907,6 +1907,7 @@ static void ally_schedule_work(struct ally_rgb_leds *led)
 static void ally_led_do_brightness(struct work_struct *work)
 {
 	struct ally_rgb_leds *led = container_of(work, struct ally_rgb_leds, work);
+	u8 buf1[] = { FEATURE_ROG_ALLY_REPORT_ID, 0xb4, 0x00, 0x00, 0x00 };
 	u8 buf[] = { FEATURE_ROG_ALLY_REPORT_ID, 0xba, 0xc5, 0xc4, 0x00 };
 	unsigned long flags;
 
@@ -1921,6 +1922,8 @@ static void ally_led_do_brightness(struct work_struct *work)
 
 	if (asus_dev_set_report(led->hdev, buf, sizeof(buf)) < 0)
 		hid_err(led->hdev, "Ally failed to set backlight\n");
+	if (asus_dev_set_report(led->hdev, buf1, sizeof(buf1)) < 0)
+		hid_err(led->hdev, "Ally failed to set backlight\n");
 }
 
 static void ally_led_do_rgb(struct work_struct *work)
