From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Muhammad Alfi Syahrin <malfisya.dev@hotmail.com>
Date: Mon, 12 May 2025 10:14:35 +0700
Subject: [PATCH] Add missing include

---
 src/tools.h | 1 +
 1 file changed, 1 insertion(+)

diff --git a/src/tools.h b/src/tools.h
index 26af0e6..df4dc02 100644
--- a/src/tools.h
+++ b/src/tools.h
@@ -40,6 +40,7 @@ using namespace std;
 #include "../config.h"
 #endif
 #include "../gettext.h"
+#include <cstdint>
 #if ENABLE_NLS
 #define _(str) gettext (str)
 #else
