From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Muhammad Alfi Syahrin <malfisya.dev@hotmail.com>
Date: Sat, 13 Sep 2025 06:11:02 +0700
Subject: [PATCH] Use Standard theme instead of Classic

---
 src/config.cpp | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/config.cpp b/src/config.cpp
index 6038f73..02a4b92 100644
--- a/src/config.cpp
+++ b/src/config.cpp
@@ -78,8 +78,8 @@ Config::Config()
 	/* various */
 	use_hints = 1;
 	return_on_click = 0;
-	theme_id = 0;
-	theme_count = 1;
+	theme_id = 1;
+	theme_count = 2;
 	edit_setname ="newset";
 	bcc_type = 0; /* trajectory */
 
