# Fedora patch
From: Peter Lemenkov <lemenkov@gmail.com>
Date: Mon, 24 Mar 2025 09:17:05 +0100
Subject: [PATCH] Adjust malloc types

Signed-off-by: Peter Lemenkov <lemenkov@gmail.com>

diff --git a/src/malloc.c b/src/malloc.c
index 1bca97a..c506e69 100644
--- a/src/malloc.c
+++ b/src/malloc.c
@@ -165,7 +165,7 @@ my_strdup(char *src)
 #include <sys/types.h>
 #undef malloc
 
-void *malloc();
+void *malloc(size_t);
 
 /* Allocate an N-byte block of memory from the heap.
 If N is zero, allocate a 1-byte block.  */
@@ -179,4 +179,4 @@ rpl_malloc(size_t n)
 	}
 	return malloc(n);
 }
-#endif
\ No newline at end of file
+#endif
