diff -ruNb a/data/org.gnome.settings-daemon.plugins.xsettings.gschema.xml.in.in b/data/org.gnome.settings-daemon.plugins.xsettings.gschema.xml.in.in
--- a/data/org.gnome.settings-daemon.plugins.xsettings.gschema.xml.in.in	2012-04-07 17:58:07.000000000 +0100
+++ b/data/org.gnome.settings-daemon.plugins.xsettings.gschema.xml.in.in	2013-06-11 20:50:01.519588358 +0100
@@ -6,12 +6,12 @@
       <_description>Whether this plugin would be activated by gnome-settings-daemon or not</_description>
     </key>
     <key name="antialiasing" enum="org.gnome.settings-daemon.GsdFontAntialiasingMode">
-      <default>'grayscale'</default>
+      <default>'rgba'</default>
       <_summary>Antialiasing</_summary>
       <_description>The type of antialiasing to use when rendering fonts. Possible values are: "none" for no antialiasing, "grayscale" for standard grayscale antialiasing, and "rgba" for subpixel antialiasing (LCD screens only).</_description>
     </key>
     <key name="hinting" enum="org.gnome.settings-daemon.GsdFontHinting">
-      <default>'medium'</default>
+      <default>'full'</default>
       <_summary>Hinting</_summary>
       <_description>The type of hinting to use when rendering fonts. Possible values are: "none" for no hinting, "slight" for basic, "medium" for moderate, and "full" for maximum hinting (may cause distortion of letter forms).</_description>
     </key>
