diff --git a/data/org.gnome.settings-daemon.plugins.xsettings.gschema.xml.in.in b/data/org.gnome.settings-daemon.plugins.xsettings.gschema.xml.in.in
index caccf56..8eeb79b 100644
--- a/data/org.gnome.settings-daemon.plugins.xsettings.gschema.xml.in.in
+++ b/data/org.gnome.settings-daemon.plugins.xsettings.gschema.xml.in.in
@@ -6,7 +6,7 @@
       <_description>Whether this plugin would be activated by gnome-settings-daemon or not</_description>
     </key>
     <key name="antialiasing" enum="org.gnome.settings-daemon.GsdFontAntialiasingMode">
-      <default>'grayscale'</default>
+      <default>'rgba'</default>
       <_summary>Antialiasing</_summary>
       <_description>The type of antialiasing to use when rendering fonts. Possible values are: "none" for no antialiasing, "grayscale" for standard grayscale antialiasing, and "rgba" for subpixel antialiasing (LCD screens only).</_description>
     </key>
