From 98c08dfe16a73f2fccd4c22e3e6dc9019fb90087 Mon Sep 17 00:00:00 2001
From: Peter Hutterer <peter.hutterer@who-t.net>
Date: Thu, 17 Dec 2020 10:18:12 +1000
Subject: [PATCH] meson.build: fix typo causing the data file test to be
 installed (#319)

Signed-off-by: Peter Hutterer <peter.hutterer@who-t.net>
---
 meson.build | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/meson.build b/meson.build
index 6a38b36..0d4e346 100644
--- a/meson.build
+++ b/meson.build
@@ -107,7 +107,7 @@ install_subdir('data',
 	       exclude_files: ['wacom.example',
 			       'check-files-in-git.sh',
 			       'check-svg-exists.sh',
-			       'test-data_files.py',
+			       'test_data_files.py',
 			       'Makefile.am',
 			       'Makefile.in',
 			       'layouts/Makefile.am',
