From bb9ed3c1fce8d64bdaac1ef2dbafbbeca8260134 Mon Sep 17 00:00:00 2001
From: Ikey Doherty <ikey@solus-project.com>
Date: Fri, 21 Aug 2015 10:10:07 +0100
Subject: [PATCH] Ensure we link correctly

Signed-off-by: Ikey Doherty <ikey@solus-project.com>
---
 Makefile | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/Makefile b/Makefile
index 6524ef3..1bf8d7f 100644
--- a/Makefile
+++ b/Makefile
@@ -5,7 +5,7 @@ ZOPFLI=zopfli/src/zopfli/
 # use gcc and gmake on Solaris
 
 pigz: pigz.o yarn.o try.o ${ZOPFLI}deflate.o ${ZOPFLI}blocksplitter.o ${ZOPFLI}tree.o ${ZOPFLI}lz77.o ${ZOPFLI}cache.o ${ZOPFLI}hash.o ${ZOPFLI}util.o ${ZOPFLI}squeeze.o ${ZOPFLI}katajainen.o
-	$(CC) $(LDFLAGS) -o pigz $^ -lpthread -lm
+	$(CC) $(CFLAGS) $(LDFLAGS) -o pigz $^ -lpthread -lm -lz
 	ln -f pigz unpigz
 
 pigz.o: pigz.c yarn.h try.h ${ZOPFLI}deflate.h ${ZOPFLI}util.h
-- 
2.5.0

