From 78aa7a883384e85242ff6a10f157c4bd26c5f7d4 Mon Sep 17 00:00:00 2001
From: Ikey Doherty <ikey@evolve-os.com>
Date: Sat, 13 Sep 2014 15:26:33 +0100
Subject: [PATCH] Fix autotools per LFS wiki

Signed-off-by: Ikey Doherty <ikey@evolve-os.com>
---
 configure.in         | 2 +-
 frontend/Makefile.am | 2 +-
 2 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/configure.in b/configure.in
index 6315e01..151f375 100644
--- a/configure.in
+++ b/configure.in
@@ -25,7 +25,7 @@ AC_PROG_LN_S
 AC_PROG_MAKE_SET
 AC_CHECK_PROGS(RPMBUILD, rpmbuild, rpm)
 
-AM_CONFIG_HEADER(config.h)
+AC_CONFIG_HEADERS(config.h)
 
 AC_ARG_WITH(xmms,[  --with-xmms             compile XMMS-1 plugin],
 	     WITHXMMS=$withval, WITHXMMS=no)
diff --git a/frontend/Makefile.am b/frontend/Makefile.am
index 74af733..532fa13 100644
--- a/frontend/Makefile.am
+++ b/frontend/Makefile.am
@@ -1,5 +1,5 @@
 bin_PROGRAMS = faad
-man_MANS = faad.man
+man1_MANS = faad.man
 
 INCLUDES = -I$(top_srcdir)/include -I$(top_srcdir)/common/faad \
 	   -I$(top_srcdir)/common/mp4ff
-- 
1.8.4

