From 7cf2050e26a38ab47a2636c99d8ebbbdc01b46bf Mon Sep 17 00:00:00 2001
From: Zach Bacon <zachbacon@vba-m.com>
Date: Thu, 27 Jul 2023 18:04:49 -0400
Subject: [PATCH] Fixes build of Freefilesync against newer gcc

Signed-off-by: Zach Bacon <zachbacon@vba-m.com>
---
 FreeFileSync/Source/base/structures.cpp | 1 +
 1 file changed, 1 insertion(+)

diff --git a/FreeFileSync/Source/base/structures.cpp b/FreeFileSync/Source/base/structures.cpp
index 4a89d6e..8b02cab 100755
--- a/FreeFileSync/Source/base/structures.cpp
+++ b/FreeFileSync/Source/base/structures.cpp
@@ -8,6 +8,7 @@
 #include <iterator>
 #include <stdexcept>
 #include <ctime>
+#include <limits>
 #include <zen/i18n.h>
 #include <zen/time.h>
 #include "hard_filter.h"
-- 
2.41.0

