From e4f0cd13e12184bfbe8b88cd29cc45a483a1ae87 Mon Sep 17 00:00:00 2001
From: Zach Bacon <zachbacon@vba-m.com>
Date: Sat, 4 Nov 2023 11:18:33 -0400
Subject: [PATCH] ffs_path: Change zstr to point to /usr/share/freefilesync

Signed-off-by: Zach Bacon <zachbacon@vba-m.com>
---
 FreeFileSync/Source/ffs_paths.cpp | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/FreeFileSync/Source/ffs_paths.cpp b/FreeFileSync/Source/ffs_paths.cpp
index f1f6cd0..fd2e156 100644
--- a/FreeFileSync/Source/ffs_paths.cpp
+++ b/FreeFileSync/Source/ffs_paths.cpp
@@ -55,7 +55,8 @@ Zstring fff::getInstallDirPath()
 
 Zstring fff::getResourceDirPath()
 {
-    return appendPath(getProcessParentFolderPath(), Zstr("Resources"));
+	//We like to ensure data is found in a specific folder in Solus, and that's not beside the executable
+    return Zstr("/usr/share/FreeFileSync/");
 }
 
 
-- 
2.42.0

