diff --git a/FreeFileSync/Source/Makefile b/FreeFileSync/Source/Makefile
index 3319726..7537d45 100755
--- a/FreeFileSync/Source/Makefile
+++ b/FreeFileSync/Source/Makefile
@@ -90,6 +90,7 @@ CPP_LIST+=../../wx+/image_resources.cpp
 CPP_LIST+=../../wx+/popup_dlg.cpp
 CPP_LIST+=../../wx+/popup_dlg_generated.cpp
 CPP_LIST+=../../wx+/zlib_wrap.cpp
+CPP_LIST+=../../xBRZ/src/xbrz.cpp
 
 OBJECT_LIST = $(CPP_LIST:%.cpp=../Obj/FFS_GCC_Make_Release/ffs/src/%.o)
 
diff --git a/FreeFileSync/Source/RealTimeSync/Makefile b/FreeFileSync/Source/RealTimeSync/Makefile
index baf4ef6..cc57936 100755
--- a/FreeFileSync/Source/RealTimeSync/Makefile
+++ b/FreeFileSync/Source/RealTimeSync/Makefile
@@ -36,6 +36,7 @@ CPP_LIST+=../../../wx+/image_tools.cpp
 CPP_LIST+=../../../wx+/image_resources.cpp
 CPP_LIST+=../../../wx+/popup_dlg.cpp
 CPP_LIST+=../../../wx+/popup_dlg_generated.cpp
+CPP_LIST+=../../../xBRZ/src/xbrz.cpp
 
 OBJECT_LIST=$(CPP_LIST:%.cpp=../../Obj/RTS_GCC_Make_Release/ffs/src/rts/%.o)
 
