diff -ru a/FreeFileSync/Source/ui/gui_generated.cpp b/FreeFileSync/Source/ui/gui_generated.cpp
--- a/FreeFileSync/Source/ui/gui_generated.cpp	2023-09-13 12:16:12.000000000 +0200
+++ b/FreeFileSync/Source/ui/gui_generated.cpp	2023-09-21 10:05:00.360102541 +0200
@@ -108,10 +108,10 @@
 	m_menuItemHelp = new wxMenuItem( m_menuHelp, wxID_HELP, wxString( _("&View help") ) + wxT('\t') + wxT("F1"), wxEmptyString, wxITEM_NORMAL );
 	m_menuHelp->Append( m_menuItemHelp );
 
-	m_menuHelp->AppendSeparator();
+	//m_menuHelp->AppendSeparator();
 
 	m_menuItemCheckVersionNow = new wxMenuItem( m_menuHelp, wxID_ANY, wxString( _("&Check for updates now") ) , wxEmptyString, wxITEM_NORMAL );
-	m_menuHelp->Append( m_menuItemCheckVersionNow );
+	//m_menuHelp->Append( m_menuItemCheckVersionNow );
 
 	m_menuHelp->AppendSeparator();
 
diff -ru a/FreeFileSync/Source/ui/main_dlg.cpp b/FreeFileSync/Source/ui/main_dlg.cpp
--- a/FreeFileSync/Source/ui/main_dlg.cpp	2023-09-13 12:16:12.000000000 +0200
+++ b/FreeFileSync/Source/ui/main_dlg.cpp	2023-09-21 10:05:32.806975589 +0200
@@ -962,7 +962,7 @@
     updateGui();
 
     //register regular check for update on next idle event
-    Bind(wxEVT_IDLE, &MainDialog::onStartupUpdateCheck, this);
+    //Bind(wxEVT_IDLE, &MainDialog::onStartupUpdateCheck, this);
 
     //asynchronous call to wxWindow::Dimensions(): fix superfluous frame on right and bottom when FFS is started in fullscreen mode
     Bind(wxEVT_IDLE, &MainDialog::onLayoutWindowAsync, this);
diff -ru a/FreeFileSync/Source/ui/version_check.cpp b/FreeFileSync/Source/ui/version_check.cpp
--- a/FreeFileSync/Source/ui/version_check.cpp	2023-09-13 12:16:13.000000000 +0200
+++ b/FreeFileSync/Source/ui/version_check.cpp	2023-09-21 10:06:09.283874265 +0200
@@ -58,7 +58,8 @@
 bool fff::automaticUpdateCheckDue(time_t lastUpdateCheck)
 {
     const time_t now = std::time(nullptr);
-    return std::abs(now - lastUpdateCheck) >= 7 * 24 * 3600; //check weekly
+    //return std::abs(now - lastUpdateCheck) >= 7 * 24 * 3600; //check weekly
+    return false;
 }