--- tests/Makefile.common	2026-09-06 21:14:29.965888170 +0100
+++ tests/Makefile.common	2026-09-06 21:14:42.973295541 +0100
@@ -57,7 +57,7 @@
   classes-methods.R \
   reg-tests-1a.R reg-tests-1b.R reg-tests-1c.R reg-tests-1d.R \
   reg-tests-1e.R reg-tests-2.R \
-  reg-examples1.R reg-examples2.R reg-packages.R \
+  reg-examples1.R reg-examples2.R \
   p-qbeta-strict-tst.R d-p-q-r-tst-2.R \
   r-strict-tst.R \
   reg-IO.R reg-IO2.R reg-plot.R reg-S4.R reg-S4-examples.R reg-BLAS.R
