--- src/main/Makefile.in.old	2026-10-01 19:05:57.659032084 +0100
+++ src/main/Makefile.in	2026-10-01 19:07:02.215846178 +0100
@@ -100,11 +100,13 @@
 
 EXTRA_LIBS = $(BLAS_LIBS) $(FLIBS) @R_XTRA_LIBS@ @LIBINTL@ $(READLINE_LIBS) $(LIBS) @BUILD_NEW_ACCELERATE_TRUE@ -framework Accelerate
 
+MIMALLOC_LIBS = -Wl,--push-state,--no-as-needed -lmimalloc -Wl,--pop-state
+
 R_binary = R.bin
 R_bin_OBJECTS = Rmain.o @WANT_R_SHLIB_FALSE@$(OBJECTS)
-@WANT_R_SHLIB_FALSE@R_bin_LDADD = $(MAIN_OBJS) $(EXTRA_STATIC_LIBS) $(EXTRA_LIBS)
+@WANT_R_SHLIB_FALSE@R_bin_LDADD = $(MAIN_OBJS) $(EXTRA_STATIC_LIBS) $(EXTRA_LIBS) $(MIMALLOC_LIBS)
 ## Linked against -lRblas because -lR is and otherwise ld complains.
-@WANT_R_SHLIB_TRUE@R_bin_LDADD = -lR @BLAS_SHLIB_TRUE@-lRblas
+@WANT_R_SHLIB_TRUE@R_bin_LDADD = -lR $(MIMALLOC_LIBS) @BLAS_SHLIB_TRUE@-lRblas
 ## This should depend on MAIN_OBJS not MAIN_LIBS, but we can't use that.
 ## There is also a dependence on libRblas when that is internal and static.
 @WANT_R_SHLIB_FALSE@R_bin_DEPENDENCIES = $(MAIN_LIBS) $(EXTRA_STATIC_LIBS)@USE_EXPORTFILES_TRUE@ $(top_builddir)/etc/R.exp
