diff --git a/etc/systemd/wsdd.service b/etc/systemd/wsdd.service
index c9cfce0..3c52e7b 100644
--- a/etc/systemd/wsdd.service
+++ b/etc/systemd/wsdd.service
@@ -4,20 +4,22 @@ Description=Web Services Dynamic Discovery host daemon
 After=network-online.target
 Wants=network-online.target
 ; It makes sense to have Samba running when wsdd starts, but is not required
-;Wants=smb.service
+Wants=smb.service
+After=smb.service
 
 [Service]
 Type=simple
-ExecStart=/usr/bin/wsdd --shortlog
-; Replace those with an unprivledged user/group that matches your environment,
+ExecStart=/usr/bin/wsdd --shortlog --chroot /run/wsdd/chroot
+; Replace these with an unprivileged user/group that matches your environment,
 ; like nobody/nogroup or daemon:daemon or a dedicated user for wsdd
-User=nobody
-Group=nobody
+User=wsdd
+Group=wsdd
 ; The following lines can be used for a chroot execution of wsdd.
 ; Also append '--chroot /run/wsdd/chroot' to ExecStart to enable chrooting
-;AmbientCapabilities=CAP_SYS_CHROOT
-;ExecStartPre=/usr/bin/install -d -o nobody -g nobody -m 0700 /run/wsdd/chroot
+AmbientCapabilities=CAP_SYS_CHROOT
+;ExecStartPre=/usr/bin/install -d -o wsdd -g wsdd -m 0700 /run/wsdd/chroot
 ;ExecStopPost=rmdir /run/wsdd/chroot
 
 [Install]
 WantedBy=multi-user.target
+
