diff --git a/putu8.c b/putu8.c
index 212f623..b4f2402 100644
--- a/putu8.c
+++ b/putu8.c
@@ -3,6 +3,8 @@
  * Based on  Roman Czyborra:  http://czyborra.com/utf/
  */
 
+#include <stdio.h>
+
 void
 putu8(unsigned long c)
 {
diff --git a/unifuzz.c b/unifuzz.c
index 37dedc0..ee0eec9 100644
--- a/unifuzz.c
+++ b/unifuzz.c
@@ -21,6 +21,7 @@
 #include "config.h"
 #include <stdio.h>
 #include <stdlib.h>
+#include <time.h>
 #ifdef HAVE_LOCALE_H
 #include <locale.h>
 #endif
@@ -68,12 +69,14 @@ ShowUsage(void){
   putc('\n',stderr);
 }
 
+void
 EmitLineOfX (unsigned long len, int c) {
   unsigned long i;
   for (i = 0; i < len; i++) putchar(c);
   putchar('\n');
 }
 
+void
 EmitEmbeddedNull() {
   putchar('a');
   putchar('b');
@@ -83,6 +86,7 @@ EmitEmbeddedNull() {
   putchar('\n');
 }
 
+void
 EmitBadUTF8 () {
   /* invalid first byte */
   putchar(0x80);
@@ -97,6 +101,7 @@ EmitBadUTF8 () {
 }
 
 /* Emit the middle character from each range */
+void
 EmitAllRanges(short AboveBMPP) {
   int i;
   UTF32 scp;
@@ -112,6 +117,7 @@ EmitAllRanges(short AboveBMPP) {
   }
 }
 
+void
 EmitRandomCharacters(unsigned long n,short AboveBMPP) {
   UTF32 c;
   unsigned long k = 0;
@@ -131,6 +137,7 @@ EmitRandomCharacters(unsigned long n,short AboveBMPP) {
 }
 
 
+void
 EmitSpecificStrings(short AboveBMPP) {
   extern void putu8(UTF32);
 
diff --git a/uniname.c b/uniname.c
index 23813ef..6d3b4c2 100644
--- a/uniname.c
+++ b/uniname.c
@@ -23,6 +23,7 @@
 #include "config.h"
 #include <stdio.h>
 #include <stdlib.h>
+#include <ctype.h>
 #ifdef HAVE_LOCALE_H
 #include <locale.h>
 #endif
diff --git a/unisurrogate.c b/unisurrogate.c
index 544d2e7..5f056a8 100644
--- a/unisurrogate.c
+++ b/unisurrogate.c
@@ -26,6 +26,7 @@ ShowUsage(void){
   putc('\n',stderr);
 }
 
+void
 main(int ac, char **av) {
   u_int32_t cp, cpi, high, low;
   char *istr;
