From 4805dfe7481cf6e3fd037a901673bc0146ef3071 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Abdulkadir=20Furkan=20=C5=9Eanl=C4=B1?=
 <abdulkadirfsanli@protonmail.com>
Date: Tue, 1 Jan 2019 17:02:12 +0100
Subject: [PATCH] Fix shader install directory

---
 Makefile | 3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

diff --git a/Makefile b/Makefile
index 5c640ad..379e6f4 100644
--- a/Makefile
+++ b/Makefile
@@ -40,7 +40,7 @@ ifeq ($(INSTALL),unix)
         ifdef XDG_CONFIG_DIRS
             SHADERDIR = /$(firstword $(subst :, ,$(XDG_CONFIG_DIRS)))/glava/
         else
-            SHADERDIR = /etc/xdg/glava/
+            SHADERDIR = /usr/share/glava/
         endif
     endif
 endif
@@ -124,4 +124,3 @@ install:
 uninstall:
 	rm $(EXECTARGET)
 	rm -rf $(SHADERTARGET)
-
-- 
2.20.1

