From b6bda83d106945531ee7dffd7866dcb288edb871 Mon Sep 17 00:00:00 2001
From: Mathew Hodson <mathew.hodson@gmail.com>
Date: Sun, 2 Feb 2020 21:48:22 +0000
Subject: [PATCH] Fix mapping of GStreamer album volume count

---
 metadata/rb-metadata-gst-common.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/metadata/rb-metadata-gst-common.c b/metadata/rb-metadata-gst-common.c
index 32e4c87eb..1386d2e98 100644
--- a/metadata/rb-metadata-gst-common.c
+++ b/metadata/rb-metadata-gst-common.c
@@ -59,7 +59,7 @@ rb_metadata_gst_tag_to_field (const char *tag)
 	else if (!strcmp (tag, GST_TAG_ALBUM_VOLUME_NUMBER))
 		return RB_METADATA_FIELD_DISC_NUMBER;
 	else if (!strcmp (tag, GST_TAG_ALBUM_VOLUME_COUNT))
-		return RB_METADATA_FIELD_MAX_TRACK_NUMBER;
+		return RB_METADATA_FIELD_MAX_DISC_NUMBER;
 	else if (!strcmp (tag, GST_TAG_DESCRIPTION))
 		return RB_METADATA_FIELD_DESCRIPTION;
 	else if (!strcmp (tag, GST_TAG_VERSION))
-- 
2.24.1

