From 89800f893315ce0bb673bd91d34a14415cc3b033 Mon Sep 17 00:00:00 2001
From: Thomas Staudinger <Staudi.Kaos@gmail.com>
Date: Sun, 5 May 2019 00:33:15 +0200
Subject: [PATCH] Remove obsolote ITMS_HANDLER reference in desktop file

---
 data/rhythmbox.desktop.in.in | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/data/rhythmbox.desktop.in.in b/data/rhythmbox.desktop.in.in
index 85fb659c6..edc4042cc 100644
--- a/data/rhythmbox.desktop.in.in
+++ b/data/rhythmbox.desktop.in.in
@@ -10,7 +10,7 @@ Type=Application
 Icon=rhythmbox
 X-GNOME-DocPath=rhythmbox/rhythmbox.xml
 Categories=GNOME;GTK;AudioVideo;Audio;Player;
-MimeType=application/x-ogg;application/ogg;audio/x-vorbis+ogg;audio/vorbis;audio/x-vorbis;audio/x-scpls;audio/x-mp3;audio/x-mpeg;audio/mpeg;audio/x-mpegurl;audio/x-flac;audio/mp4;audio/x-it;audio/x-mod;audio/x-s3m;audio/x-stm;audio/x-xm;@ITMS_HANDLER@
+MimeType=application/x-ogg;application/ogg;audio/x-vorbis+ogg;audio/vorbis;audio/x-vorbis;audio/x-scpls;audio/x-mp3;audio/x-mpeg;audio/mpeg;audio/x-mpegurl;audio/x-flac;audio/mp4;audio/x-it;audio/x-mod;audio/x-s3m;audio/x-stm;audio/x-xm;
 StartupNotify=true
 X-GNOME-Bugzilla-Bugzilla=GNOME
 X-GNOME-Bugzilla-Product=rhythmbox
-- 
2.21.0

