From 5ac5d27b33b4a37af3413aadeb83b88eb6ca5f1e Mon Sep 17 00:00:00 2001
From: Thomas Staudinger <Staudi.Kaos@gmail.com>
Date: Sun, 7 Aug 2022 03:12:15 +0200
Subject: [PATCH] Fix git(commit) colours for dark terminals

---
 git.nanorc | 14 +++++++-------
 1 file changed, 7 insertions(+), 7 deletions(-)

diff --git a/git.nanorc b/git.nanorc
index 0aaf475..711ee5f 100644
--- a/git.nanorc
+++ b/git.nanorc
@@ -4,7 +4,7 @@ color brightcyan "\<(true|false)\>"
 color cyan "^[[:space:]]*[^=]*="
 color brightmagenta "^[[:space:]]*\[.*\]$"
 color yellow ""(\\.|[^"])*"|'(\\.|[^'])*'"
-color brightblack "(^|[[:space:]])#([^{].*)?$"
+color brightblue "(^|[[:space:]])#([^{].*)?$"
 color ,green "[[:space:]]+$"
 color ,red "	+"
 
@@ -22,7 +22,7 @@ syntax "git-commit" "COMMIT_EDITMSG|TAG_EDITMSG"
 color yellow ".*"
 
 # Comments
-color brightblack "^#.*"
+color brightblue "^#.*"
 
 # Files changes
 color white       "#[[:space:]](deleted|modified|new file|renamed):[[:space:]].*"
@@ -36,14 +36,14 @@ color black "^#	[^/?*:;{}\\]+\.[^/?*:;{}\\]+$"
 
 color brightmagenta "^#[[:space:]]Changes.*[:]"
 color brightred "^#[[:space:]]Your branch and '[^']+"
-color brightblack "^#[[:space:]]Your branch and '"
+color brightblue "^#[[:space:]]Your branch and '"
 color brightwhite "^#[[:space:]]On branch [^ ]+"
-color brightblack "^#[[:space:]]On branch"
+color brightblue "^#[[:space:]]On branch"
 
 # Recolor hash symbols
 
 # Recolor hash symbols
-color brightblack "#"
+color brightblue "#"
 
 # Trailing spaces (+LINT is not ok, git uses tabs)
 color ,green "[[:space:]]+$"
@@ -56,7 +56,7 @@ syntax "git-rebase-todo" "git-rebase-todo"
 color yellow ".*"
 
 # Comments
-color brightblack "^#.*"
+color brightblue "^#.*"
 
 # Rebase commands
 color green       "^(e|edit) [0-9a-f]{7,40}"
@@ -73,7 +73,7 @@ color yellow      "^(x|exec) [^ ]+ [0-9a-f]{7,40}"
 color yellow      "^#  (x, exec)"
 
 # Recolor hash symbols
-color brightblack "#"
+color brightblue "#"
 
 # Commit IDs
 color brightblue "[0-9a-f]{7,40}"
-- 
2.35.4

