diff -rNu splix-2.0.0/src/ppdfile.cpp splix-2.0.0.new/src/ppdfile.cpp
- --- splix-2.0.0/src/ppdfile.cpp       2009-02-06 12:49:10.000000000 +0100
+++ splix-2.0.0.new/src/ppdfile.cpp     2010-03-17 17:55:38.585903986 +0100
@@ -282,7 +282,7 @@
  * Opérateur d'assignation
  * Assignment operator
  */
- -void PPDFile::Value::operator = (const PPDFile::Value::Value &val)
+void PPDFile::Value::operator = (const PPDFile::Value &val)
 {
     if (_preformatted)
         delete[] _preformatted;