diff --git a/.github/workflows/scripts/linux/pcsx2-qt.desktop b/.github/workflows/scripts/linux/pcsx2-qt.desktop
index 30ae545..f4d73e9 100644
--- a/.github/workflows/scripts/linux/pcsx2-qt.desktop
+++ b/.github/workflows/scripts/linux/pcsx2-qt.desktop
@@ -6,7 +6,7 @@ Name=PCSX2
 StartupWMClass=PCSX2
 GenericName=PlayStation 2 Emulator
 Comment=Sony PlayStation 2 emulator
-Exec=pcsx2-qt
+Exec=/usr/share/pcsx2/bin/pcsx2-qt
 Icon=PCSX2
 Keywords=game;emulator;
 Categories=Game;Emulator;
