From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Reilly Brogan <reilly@reillybrogan.com>
Date: Sun, 26 May 2024 22:33:18 -0500
Subject: [PATCH] Change default sddm background

---
 sddm-theme/theme.conf.cmake | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/sddm-theme/theme.conf.cmake b/sddm-theme/theme.conf.cmake
index a560da377..ded046aa5 100644
--- a/sddm-theme/theme.conf.cmake
+++ b/sddm-theme/theme.conf.cmake
@@ -4,5 +4,5 @@ logo=${KDE_INSTALL_FULL_DATADIR}/sddm/themes/breeze/default-logo.svg
 type=image
 color=#1d99f3
 fontSize=10
-background=${KDE_INSTALL_FULL_WALLPAPERDIR}/Next/contents/images/5120x2880.png
+background=${KDE_INSTALL_FULL_WALLPAPERDIR}/SolusPlasma/contents/images/3840x2160.jxl
 needsFullUserModel=false
