From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Justin Zobel <justin.zobel@gmail.com>
Date: Tue, 16 Jan 2024 11:06:49 +1030
Subject: [PATCH 4/6] Use Breeze Dark by default

---
 kcms/landingpage/landingpage_kdeglobalssettings.kcfg | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/kcms/landingpage/landingpage_kdeglobalssettings.kcfg b/kcms/landingpage/landingpage_kdeglobalssettings.kcfg
index 0831d45dc..d0e7d585f 100644
--- a/kcms/landingpage/landingpage_kdeglobalssettings.kcfg
+++ b/kcms/landingpage/landingpage_kdeglobalssettings.kcfg
@@ -7,7 +7,7 @@
   <group name="General">
     <entry name="colorScheme" key="ColorScheme" type="String">
       <label>Color scheme name</label>
-      <default>BreezeLight</default>
+      <default>BreezeDark</default>
     </entry>
   </group>
   <group name="KDE">
@@ -17,7 +17,7 @@
     </entry>
     <entry name="lookAndFeelPackage" key="LookAndFeelPackage" type="String">
       <label>Global Look and Feel package</label>
-      <default>org.kde.breeze.desktop</default>
+      <default>org.kde.breezedark.desktop</default>
     </entry>
     <entry name="defaultLightLookAndFeel" key="DefaultLightLookAndFeel" type="String">
       <label>Global Look and Feel package, alternate</label>
