From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Justin Zobel <justin.zobel@gmail.com>
Date: Tue, 16 Jan 2024 11:14:47 +1030
Subject: [PATCH 5/6] - Switch to Solus default panel

---
 desktoppackage/contents/main.js | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/desktoppackage/contents/main.js b/desktoppackage/contents/main.js
index e62f46770..9d547f910 100644
--- a/desktoppackage/contents/main.js
+++ b/desktoppackage/contents/main.js
@@ -1,5 +1,5 @@
 
-loadTemplate("org.kde.plasma.desktop.defaultPanel")
+loadTemplate("solus.desktop.defaultPanel")
 
 var desktopsArray = desktopsForActivity(currentActivity());
 for( var j = 0; j < desktopsArray.length; j++) {
