From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Reilly Brogan <reilly@reillybrogan.com>
Date: Thu, 1 Jun 2023 18:42:25 -0500
Subject: [PATCH 6/6] Solus tweak to keyboard settings

---
 kcms/keyboard/keyboardmiscsettings.kcfg | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/kcms/keyboard/keyboardmiscsettings.kcfg b/kcms/keyboard/keyboardmiscsettings.kcfg
index 3d511a503..b4497b2cd 100644
--- a/kcms/keyboard/keyboardmiscsettings.kcfg
+++ b/kcms/keyboard/keyboardmiscsettings.kcfg
@@ -15,8 +15,8 @@ SPDX-License-Identifier: GPL-2.0-or-later
       <default code="true">qgetenv("QT_IM_MODULE") == "plasmaim" ? QStringLiteral("accent") : QStringLiteral("repeat")</default>
     </entry>
     <entry name="numLock" key="NumLock" type="Int">
-    <!-- STATE_UNCHANGED -->
-      <default>2</default>
+    <!-- STATE_ON -->
+      <default>0</default>
     </entry>
     <entry name="repeatDelay" key="RepeatDelay" type="Int">
       <tooltip>If supported, this option allows you to set the delay after which a pressed key will start generating keycodes. The 'Repeat rate' option controls the frequency of these keycodes.If supported, this option allows you to set the delay after which a pressed key will start generating keycodes. The 'Repeat rate' option controls the frequency of these keycodes.</tooltip>
