From 7f54c7b8ebf9624e22deed17e936d2e85c337161 Mon Sep 17 00:00:00 2001
From: Joshua Strobl <joshua@streambits.io>
Date: Tue, 22 Oct 2019 20:00:13 +0300
Subject: [PATCH 1/1] Disable running gyp on shared deps

---
 Makefile | 3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

diff --git a/Makefile b/Makefile
index 0a5177d1..843a86ff 100644
--- a/Makefile
+++ b/Makefile
@@ -140,8 +140,7 @@ with-code-cache:
 test-code-cache: with-code-cache
 	echo "'test-code-cache' target is a noop"
 
-out/Makefile: config.gypi common.gypi node.gyp \
-	deps/uv/uv.gyp deps/http_parser/http_parser.gyp deps/zlib/zlib.gyp \
+out/Makefile: deps/http_parser/http_parser.gyp \
 	tools/v8_gypfiles/toolchain.gypi tools/v8_gypfiles/features.gypi \
 	tools/v8_gypfiles/inspector.gypi tools/v8_gypfiles/v8.gyp
 	$(PYTHON) tools/gyp_node.py -f make
-- 
2.23.0

