From 0461fabf259b36a156c55dc402b1d66d09eb90ca Mon Sep 17 00:00:00 2001
From: Stephen Gallagher <sgallagh@redhat.com>
Date: Tue, 1 May 2018 08:05:30 -0400
Subject: [PATCH 2/3] Suppress NPM message to run global update

Signed-off-by: Stephen Gallagher <sgallagh@redhat.com>
---
 deps/npm/bin/npm-cli.js | 54 -----------------------------------------
 1 file changed, 54 deletions(-)

diff --git a/deps/npm/bin/npm-cli.js b/deps/npm/bin/npm-cli.js
index 93eddc7a3c892b0f5d35c477c042a8e5087d7b01..ceb12b5d9a33ab7b98df05d7ef082c42f03eedb9 100755
--- a/deps/npm/bin/npm-cli.js
+++ b/deps/npm/bin/npm-cli.js
@@ -67,69 +67,15 @@
   if (conf.usage && npm.command !== 'help') {
     npm.argv.unshift(npm.command)
     npm.command = 'help'
   }
 
-  var isGlobalNpmUpdate = conf.global && ['install', 'update'].includes(npm.command) && npm.argv.includes('npm')
-
   // now actually fire up npm and run the command.
   // this is how to use npm programmatically:
   conf._exit = true
   npm.load(conf, function (er) {
     if (er) return errorHandler(er)
-    if (
-      !isGlobalNpmUpdate &&
-      npm.config.get('update-notifier') &&
-      !unsupported.checkVersion(process.version).unsupported
-    ) {
-      const pkg = require('../package.json')
-      let notifier = require('update-notifier')({pkg})
-      const isCI = require('ci-info').isCI
-      if (
-        notifier.update &&
-        notifier.update.latest !== pkg.version &&
-        !isCI
-      ) {
-        const color = require('ansicolors')
-        const useColor = npm.config.get('color')
-        const useUnicode = npm.config.get('unicode')
-        const old = notifier.update.current
-        const latest = notifier.update.latest
-        let type = notifier.update.type
-        if (useColor) {
-          switch (type) {
-            case 'major':
-              type = color.red(type)
-              break
-            case 'minor':
-              type = color.yellow(type)
-              break
-            case 'patch':
-              type = color.green(type)
-              break
-          }
-        }
-        const changelog = `https://github.com/npm/cli/releases/tag/v${latest}`
-        notifier.notify({
-          message: `New ${type} version of ${pkg.name} available! ${
-            useColor ? color.red(old) : old
-          } ${useUnicode ? '→' : '->'} ${
-            useColor ? color.green(latest) : latest
-          }\n` +
-          `${
-            useColor ? color.yellow('Changelog:') : 'Changelog:'
-          } ${
-            useColor ? color.cyan(changelog) : changelog
-          }\n` +
-          `Run ${
-            useColor
-              ? color.green(`npm install -g ${pkg.name}`)
-              : `npm i -g ${pkg.name}`
-          } to update!`
-        })
-      }
-    }
     npm.commands[npm.command](npm.argv, function (err) {
       // https://genius.com/Lin-manuel-miranda-your-obedient-servant-lyrics
       if (
         !err &&
         npm.config.get('ham-it-up') &&
-- 
2.21.0

