From 4ca97f40988ae5d45fbd59ed9b9253af23f51b7a Mon Sep 17 00:00:00 2001
From: Joshua Strobl <joshua@streambits.io>
Date: Thu, 28 Oct 2021 18:17:45 +0300
Subject: [PATCH 1/1] Suppress NPM message to run global update

---
 deps/npm/lib/cli.js | 4 ----
 1 file changed, 4 deletions(-)

diff --git a/deps/npm/lib/cli.js b/deps/npm/lib/cli.js
index e33ac91fa5..69670e71ef 100644
--- a/deps/npm/lib/cli.js
+++ b/deps/npm/lib/cli.js
@@ -36,8 +36,6 @@ module.exports = async (process) => {
   process.on('uncaughtException', exitHandler)
   process.on('unhandledRejection', exitHandler)
 
-  const updateNotifier = require('../lib/utils/update-notifier.js')
-
   // now actually fire up npm and run the command.
   // this is how to use npm programmatically:
   try {
@@ -53,8 +51,6 @@ module.exports = async (process) => {
       npm.config.set('usage', false, 'cli')
     }
 
-    updateNotifier(npm)
-
     const cmd = npm.argv.shift()
     if (!cmd) {
       npm.output(npm.usage)
-- 
2.33.1

