From bbaaed90ca7eaf248434ffa435b5fcd89b7a44e5 Mon Sep 17 00:00:00 2001
From: Pierre-Yves <pyu@riseup.net>
Date: Fri, 21 Dec 2018 09:44:50 +0100
Subject: [PATCH] testsuite disable panic-tests.sh

Signed-off-by: Pierre-Yves <pyu@riseup.net>
---
 Makefile.in        | 2 +-
 testsuite/local.mk | 1 -
 2 files changed, 1 insertion(+), 2 deletions(-)

diff --git a/Makefile.in b/Makefile.in
index ab1e6f3..c6e205c 100644
--- a/Makefile.in
+++ b/Makefile.in
@@ -2072,7 +2072,7 @@ T = testsuite/misc.pl testsuite/bug32082.sh testsuite/bug32271-1.sh \
 	testsuite/mb-match-slash.sh testsuite/mb-y-translate.sh \
 	testsuite/missing-filename.sh testsuite/newline-dfa-bug.sh \
 	testsuite/normalize-text.sh testsuite/nulldata.sh \
-	testsuite/obinary.sh testsuite/panic-tests.sh \
+	testsuite/obinary.sh \
 	testsuite/posix-char-class.sh testsuite/posix-mode-addr.sh \
 	testsuite/posix-mode-bad-ref.sh testsuite/posix-mode-ERE.sh \
 	testsuite/posix-mode-s.sh testsuite/posix-mode-N.sh \
diff --git a/testsuite/local.mk b/testsuite/local.mk
index 43623bf..49f390b 100644
--- a/testsuite/local.mk
+++ b/testsuite/local.mk
@@ -70,7 +70,6 @@ T =					\
   testsuite/normalize-text.sh		\
   testsuite/nulldata.sh			\
   testsuite/obinary.sh			\
-  testsuite/panic-tests.sh		\
   testsuite/posix-char-class.sh		\
   testsuite/posix-mode-addr.sh		\
   testsuite/posix-mode-bad-ref.sh	\
-- 
2.20.0

