From 565e3d832943f9e53eba29dc862cbc7a373f00b8 Mon Sep 17 00:00:00 2001
From: Silke Hofstra <silke@slxh.eu>
Date: Sat, 12 Jun 2021 11:53:31 +0200
Subject: [PATCH] Install udev rules in /usr/lib instead of /etc

---
 CMakeLists.txt | 6 ++++--
 1 file changed, 4 insertions(+), 2 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index ed3167a..62bb459 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -115,9 +115,9 @@ install(TARGETS headsetcontrol DESTINATION bin)
 
 # install udev files on linux
 if(UNIX AND NOT APPLE)
-    set (program_cmd headsetcontrol)
+    set (program_cmd "${CMAKE_CURRENT_BINARY_DIR}/headsetcontrol")
     set (program_arg "-u")
-    set (program_output "/etc/udev/rules.d/70-headsets.rules")
+    set (program_output "${CMAKE_CURRENT_BINARY_DIR}/70-headsets.rules")
     install( CODE
 	            "
     execute_process(COMMAND ${program_cmd} ${program_arg}
@@ -126,6 +126,8 @@ if(UNIX AND NOT APPLE)
     message(STATUS \"Installed udev rules to ${program_output}\")
 	    "
     )
+
+    install(FILES ${program_output} DESTINATION "lib/udev/rules.d")
 endif()
 
 
-- 
2.30.2

