From 886700da4e495b9466f25002580596b77b63ac20 Mon Sep 17 00:00:00 2001
From: Silke Hofstra <silke@slxh.eu>
Date: Tue, 17 Nov 2020 11:49:17 +0100
Subject: [PATCH] Install udev rules in /usr/lib instead of /etc

---
 CMakeLists.txt | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 6bfbae6..4a086f3 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -115,7 +115,7 @@ install(TARGETS headsetcontrol DESTINATION bin)
 
 # install udev files on linux
 if(UNIX AND NOT APPLE)
-    install(DIRECTORY udev/ DESTINATION /etc/udev/rules.d)
+    install(DIRECTORY udev/ DESTINATION /usr/lib/udev/rules.d)
 endif()
 
 
-- 
2.29.2

