From 6213c1c3388a94654bb24ac844c7f982492ac238 Mon Sep 17 00:00:00 2001
From: Ikey Doherty <ikey@solus-project.com>
Date: Wed, 9 Aug 2017 14:15:17 +0100
Subject: [PATCH] Use merged usr tree layout

Signed-off-by: Ikey Doherty <ikey@solus-project.com>
---
 include/builddefs.in | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/include/builddefs.in b/include/builddefs.in
index ec630bd..21ea437 100644
--- a/include/builddefs.in
+++ b/include/builddefs.in
@@ -55,7 +55,7 @@ top_builddir	= @top_builddir@
 
 PKG_SBIN_DIR	= @sbindir@
 PKG_ROOT_SBIN_DIR = @root_sbindir@
-PKG_ROOT_LIB_DIR= @root_libdir@@libdirsuffix@
+PKG_ROOT_LIB_DIR= @libdir@@libdirsuffix@
 PKG_LIB_DIR	= @libdir@@libdirsuffix@
 PKG_INC_DIR	= @includedir@/xfs
 DK_INC_DIR	= @includedir@/disk
-- 
2.13.4

