From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Reilly Brogan <reilly@reillybrogan.com>
Date: Tue, 14 Jul 2026 11:55:09 -0500
Subject: [PATCH] Maybe fix

---
 healer/Makefile | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/healer/Makefile b/healer/Makefile
index e3c41e66..434777d2 100644
--- a/healer/Makefile
+++ b/healer/Makefile
@@ -69,7 +69,9 @@ install: $(INSTALL_HEALER)
 
 install-healer: default
 	$(INSTALL) -m 755 -d $(PKG_LIBEXEC_DIR)
-	$(LTINSTALL) -m 755 $(BUILD_TARGETS) $(PKG_LIBEXEC_DIR)
+	@for t in $(BUILD_TARGETS); do \
+		$(LTINSTALL) -m 755 $$t $(PKG_LIBEXEC_DIR); \
+	done
 
 install-systemd: default
 	$(INSTALL) -m 755 -d $(SYSTEMD_SYSTEM_UNIT_DIR)
