From 2213db92549d8e26500873a33ebe7c8615bbf8fc Mon Sep 17 00:00:00 2001
From: Joshua Strobl <joshua@streambits.io>
Date: Mon, 3 May 2021 17:34:30 +0300
Subject: [PATCH 1/1] Ensure headerbar is set as the titlebar

---
 src/interface.cpp | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/interface.cpp b/src/interface.cpp
index 272b5f8a..918b7040 100644
--- a/src/interface.cpp
+++ b/src/interface.cpp
@@ -719,9 +719,9 @@ create_main_window (GsmApplication *app)
     mainbox = GTK_BOX (gtk_builder_get_object (builder, "main_box"));
     headerbar = GTK_HEADER_BAR (gtk_builder_get_object (builder, "header_bar"));
     gtk_style_context_remove_class (gtk_widget_get_style_context (GTK_WIDGET (headerbar)), "titlebar");
-    gtk_window_set_titlebar (GTK_WINDOW (main_window), NULL);
-    gtk_header_bar_set_show_close_button (headerbar, FALSE);
+    gtk_header_bar_set_show_close_button (headerbar, TRUE);
     gtk_box_pack_start (mainbox, GTK_WIDGET (headerbar), FALSE, FALSE, 0);
+    gtk_window_set_titlebar (GTK_WINDOW (main_window), GTK_WIDGET(headerbar));
 
     g_settings_get (app->settings->gobj(), GSM_SETTING_WINDOW_STATE, "(iiii)",
                     &width, &height, &xpos, &ypos);
-- 
2.30.2

