diff -ruNb a/setup.py b/setup.py
--- a/setup.py	2008-09-09 18:40:34.000000000 +0100
+++ b/setup.py	2013-05-26 12:53:15.602558544 +0100
@@ -113,6 +113,7 @@
     for e in split_quoted(os.popen("'%s' --features" % CURL_CONFIG).read()):
         if e == 'SSL':
             define_macros.append(('HAVE_CURL_SSL', 1))
+    define_macros.append (('HAVE_CURL_OPENSSL', 1)) # Force use of OpenSSL
     if not libraries:
         libraries.append("curl")
     # Add extra compile flag for MacOS X
