From 994e618833f085077be23caf13765b26dc4a8779 Mon Sep 17 00:00:00 2001
From: Joshua Strobl <joshua@streambits.io>
Date: Mon, 28 Dec 2020 18:24:20 +0200
Subject: [PATCH 1/1] Fall back to stateless directory.

---
 avahi-daemon/main.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/avahi-daemon/main.c b/avahi-daemon/main.c
index 10cb41e..ca69c2c 100644
--- a/avahi-daemon/main.c
+++ b/avahi-daemon/main.c
@@ -614,7 +614,8 @@ static int load_config_file(DaemonConfig *c) {
     assert(c);
 
     if (!(f = avahi_ini_file_load(c->config_file ? c->config_file : AVAHI_CONFIG_FILE)))
-        goto finish;
+        if (!(f = avahi_ini_file_load("/usr/share/defaults/avahi/avahi-daemon.conf")))
+            goto finish;
 
     for (g = f->groups; g; g = g->groups_next) {
 
-- 
2.29.2

