diff --git a/Modules/_bsddb.c b/Modules/_bsddb.c
index 203bba2..372ab57 100644
--- a/Modules/_bsddb.c
+++ b/Modules/_bsddb.c
@@ -668,8 +668,6 @@ static int makeDBError(int err)
         case DB_VERIFY_BAD:         errObj = DBVerifyBadError;      break;
         case DB_NOSERVER:           errObj = DBNoServerError;       break;
 #if (DBVER < 52)
-        case DB_NOSERVER_HOME:      errObj = DBNoServerHomeError;   break;
-        case DB_NOSERVER_ID:        errObj = DBNoServerIDError;     break;
 #endif
         case DB_PAGE_NOTFOUND:      errObj = DBPageNotFoundError;   break;
         case DB_SECONDARY_BAD:      errObj = DBSecondaryBadError;   break;
@@ -7880,6 +7878,7 @@ DBEnv_repmgr_set_local_site(DBEnvObject* self, PyObject* args, PyObject*
     char *host;
     int port;
     int flags = 0;
+    DB_SITE *dbSite;
     static char* kwnames[] = {"host", "port", "flags", NULL};
 
     if (!PyArg_ParseTupleAndKeywords(args, kwargs,
@@ -7889,7 +7888,9 @@ DBEnv_repmgr_set_local_site(DBEnvObject* self, PyObject* args, PyObject*
     }
     CHECK_ENV_NOT_CLOSED(self);
     MYDB_BEGIN_ALLOW_THREADS;
-    err = self->db_env->repmgr_set_local_site(self->db_env, host, port, flags);
+    err = self->db_env->repmgr_site(self->db_env,host, port, &dbSite, flags);
+    dbSite->set_config(dbSite, DB_LOCAL_SITE, 1);
+    dbSite->close(dbSite);
     MYDB_END_ALLOW_THREADS;
     RETURN_IF_ERR();
     RETURN_NONE();
@@ -7904,6 +7905,7 @@ DBEnv_repmgr_add_remote_site(DBEnvObject* self, PyObject* args, PyObject*
     int port;
     int flags = 0;
     int eidp;
+    DB_SITE *dbSite;
     static char* kwnames[] = {"host", "port", "flags", NULL};
 
     if (!PyArg_ParseTupleAndKeywords(args, kwargs,
@@ -7913,7 +7915,9 @@ DBEnv_repmgr_add_remote_site(DBEnvObject* self, PyObject* args, PyObject*
     }
     CHECK_ENV_NOT_CLOSED(self);
     MYDB_BEGIN_ALLOW_THREADS;
-    err = self->db_env->repmgr_add_remote_site(self->db_env, host, port, &eidp, flags);
+    err = self->db_env->repmgr_site(self->db_env, host, port, &dbSite, flags);
+    dbSite->get_eid(dbSite, &eidp);
+    dbSite->close(dbSite);
     MYDB_END_ALLOW_THREADS;
     RETURN_IF_ERR();
     return NUMBER_FromLong(eidp);
@@ -9903,8 +9907,6 @@ PyMODINIT_FUNC  PyInit__bsddb(void)    /* Note the two underscores */
     ADD_INT(d, DB_LOCK_NOTGRANTED);
     ADD_INT(d, DB_NOSERVER);
 #if (DBVER < 52)
-    ADD_INT(d, DB_NOSERVER_HOME);
-    ADD_INT(d, DB_NOSERVER_ID);
 #endif
     ADD_INT(d, DB_NOTFOUND);
     ADD_INT(d, DB_OLD_VERSION);
