From 668fdfa604e97c8a6099e7043c4382905377e376 Mon Sep 17 00:00:00 2001
From: Ikey Doherty <ikey@solusos.com>
Date: Tue, 13 Aug 2013 17:46:03 +0100
Subject: [PATCH] Fix building with X.Org Server 1.14.0

---
 src/tdfx_driver.c |    2 --
 1 file changed, 2 deletions(-)

diff --git a/src/tdfx_driver.c b/src/tdfx_driver.c
index ddd5431..5c58a30 100644
--- a/src/tdfx_driver.c
+++ b/src/tdfx_driver.c
@@ -64,7 +64,6 @@ SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
 
 /* Drivers using the mi implementation of backing store need: */
 
-#include "mibstore.h"
 
 /* All drivers using the vgahw module need this */
 /* This driver needs to be modified to not use vgaHW for multihead operation */
@@ -2373,7 +2372,6 @@ TDFXScreenInit(SCREEN_INIT_ARGS_DECL) {
     }
   }
 
-  miInitializeBackingStore(pScreen);
   xf86SetBackingStore(pScreen);
   xf86SetSilkenMouse(pScreen);
 
-- 
1.7.10.4

