From 06fa8ad4bad2892144ca5a9e2d3a18a92749f671 Mon Sep 17 00:00:00 2001
From: Peter O'Connor <peter@solus-project.com>
Date: Fri, 16 Mar 2018 13:26:57 +1100
Subject: [PATCH] Fix for glibc-2.26

Signed-off-by: Peter O'Connor <peter@solus-project.com>
---
 libs/std/sys.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/libs/std/sys.c b/libs/std/sys.c
index 8003d41..3df6a43 100644
--- a/libs/std/sys.c
+++ b/libs/std/sys.c
@@ -41,7 +41,7 @@
 #	include <sys/time.h>
 #	include <sys/times.h>
 #	include <sys/wait.h>
-#	include <xlocale.h>
+#	include <locale.h>
 #endif
 
 #ifdef NEKO_MAC
-- 
2.16.2

