From 5600daaa6ec9cd6ac113d1f59f26b4241fcb0d5b Mon Sep 17 00:00:00 2001
From: Joey Riches <josephriches@gmail.com>
Date: Fri, 1 Dec 2017 21:22:02 +0000
Subject: [PATCH 1/1] Add StartWMClass to desktop file

Signed-off-by: Joey Riches <josephriches@gmail.com>
---
 anoise.desktop.in | 1 +
 1 file changed, 1 insertion(+)

diff --git a/anoise.desktop.in b/anoise.desktop.in
index d19726b..4696d30 100644
--- a/anoise.desktop.in
+++ b/anoise.desktop.in
@@ -8,4 +8,5 @@ Terminal=false
 Type=Application
 Categories=AudioVideo;Player;
 Keywords=Player;Audio;
+StartupWMClass=Anoise.py
 X-Ubuntu-Gettext-Domain=anoise
-- 
2.15.1

