From 336447257b700597cc5bef95887e4ec734849a88 Mon Sep 17 00:00:00 2001
From: Fernando Mussel <fernandomussel91@gmail.com>
Date: Tue, 27 Sep 2016 22:39:44 -0300
Subject: [PATCH 2/3] fix: Adding identity string to be displayed by raven

---
 anoise/sound_menu.py | 1 +
 1 file changed, 1 insertion(+)

diff --git a/anoise/sound_menu.py b/anoise/sound_menu.py
index 392cba6..360dc79 100644
--- a/anoise/sound_menu.py
+++ b/anoise/sound_menu.py
@@ -224,6 +224,7 @@ class SoundMenuControls(dbus.service.Object):
             'CanGoPrevious':True,
             'HasTrackList': False,
             'DesktopEntry': self.desktop_name,
+            'Identity':     'ANoise',
         } # Fixed #1440061
         #return [DesktopEntry, PlaybackStatus, MetaData]
 
-- 
2.10.0

