From 1f48f98d472cb316adcab4f3b5c18793e4e03957 Mon Sep 17 00:00:00 2001
From: Fernando Mussel <fernandomussel91@gmail.com>
Date: Wed, 28 Sep 2016 21:25:14 -0300
Subject: [PATCH 3/3] fix: Create autostart dir if it does not exist

---
 anoise/preferences.py | 10 +++++++---
 1 file changed, 7 insertions(+), 3 deletions(-)

diff --git a/anoise/preferences.py b/anoise/preferences.py
index ebfce81..943fed9 100644
--- a/anoise/preferences.py
+++ b/anoise/preferences.py
@@ -28,7 +28,8 @@ _ = gettext.gettext
 class Preferences:
     """This will be for DE as MATE 14.10+ which hasn't sound indicator with Gtk3"""
     def __init__(self, player):
-        self.AUTOSTART = os.path.join(os.getenv('HOME'), '.config', 'autostart', 'anoise.desktop')
+        self.AUTOSTART_DIR = os.path.join(os.getenv('HOME'), '.config', 'autostart')
+        self.AUTOSTART = os.path.join(self.AUTOSTART_DIR, 'anoise.desktop')
         self.DESKTOP = '/usr/share/applications/anoise.desktop'
         
         self.player = player
@@ -44,8 +45,11 @@ class Preferences:
         self.btn_noises   = builder.get_object('btn_show_noises')
         self.web          = builder.get_object('boxWeb')
         
-        # Autostart
-        if os.path.isfile(self.AUTOSTART):
+        if not os.path.isdir(self.AUTOSTART_DIR):
+            # autostart dir does not exits. Create it
+            os.makedirs(self.AUTOSTART_DIR)
+        elif os.path.isfile(self.AUTOSTART):
+            # autostart dir exists and autostart file is present
             self.cb_autostart.set_active(True)
         else:
             self.cb_autostart.set_active(False)
-- 
2.10.0

