From 7943d6c9d4999a58c4552077c47f123150724ad7 Mon Sep 17 00:00:00 2001
From: Peter O'Connor <peter@solus-project.com>
Date: Fri, 16 Mar 2018 13:52:34 +1100
Subject: [PATCH] Fix for glibc-226

---
 src/Standard/Standard_CLocaleSentry.hxx | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/Standard/Standard_CLocaleSentry.hxx b/src/Standard/Standard_CLocaleSentry.hxx
index 2b226e7..0ef69e9 100644
--- a/src/Standard/Standard_CLocaleSentry.hxx
+++ b/src/Standard/Standard_CLocaleSentry.hxx
@@ -36,7 +36,7 @@
 #endif // ifndef HAVE_LOCALE_H
 
 #ifdef HAVE_XLOCALE_H
-  #include <xlocale.h>
+  #include <locale.h>
 #endif
 
 #if !defined(__ANDROID__)
-- 
2.16.2

