--- a/adm/cmake/vtk.cmake
+++ b/adm/cmake/vtk.cmake
@@ -6,7 +6,7 @@
 
 # vtk directory
 if (NOT DEFINED 3RDPARTY_VTK_DIR)
-  set (3RDPARTY_VTK_DIR "" CACHE PATH "The directory containing VTK")
+  set (3RDPARTY_VTK_DIR "/usr" CACHE PATH "The directory containing VTK")
 endif()
 
 # include occt macros. compiler_bitness, os_wiht_bit, compiler
@@ -28,13 +28,13 @@
 
 # vtk include directory
 if (NOT DEFINED 3RDPARTY_VTK_INCLUDE_DIR)
-  set (3RDPARTY_VTK_INCLUDE_DIR "" CACHE PATH "The directory containing headers of VTK")
+  set (3RDPARTY_VTK_INCLUDE_DIR "/usr/include/vtk-9.1" CACHE PATH "The directory containing headers of VTK")
 endif()
 
 #if (BUILD_SHARED_LIBS)
   # vtk library directory
   if (NOT DEFINED 3RDPARTY_VTK_LIBRARY_DIR)
-    set (3RDPARTY_VTK_LIBRARY_DIR "" CACHE PATH "The directory containing VTK libraries")
+    set (3RDPARTY_VTK_LIBRARY_DIR "/usr/lib64" CACHE PATH "The directory containing VTK libraries")
   endif()
 
   # vtk dll directory
