--- a/adm/cmake/rapidjson.cmake
+++ b/adm/cmake/rapidjson.cmake
@@ -6,7 +6,7 @@ endif()
 
 # RapidJSON directory
 if (NOT DEFINED 3RDPARTY_RAPIDJSON_DIR)
-  set (3RDPARTY_RAPIDJSON_DIR "" CACHE PATH "The directory containing RapidJSON")
+  set (3RDPARTY_RAPIDJSON_DIR "/usr" CACHE PATH "The directory containing RapidJSON")
 endif()
 
 # search for RapidJSON in user defined directory
@@ -20,7 +20,7 @@ if (3RDPARTY_DIR AND EXISTS "${3RDPARTY_
 endif()
 
 if (NOT DEFINED 3RDPARTY_RAPIDJSON_INCLUDE_DIR)
-  set (3RDPARTY_RAPIDJSON_INCLUDE_DIR  "" CACHE FILEPATH "The directory containing headers of the RAPIDJSON")
+  set (3RDPARTY_RAPIDJSON_INCLUDE_DIR  "/usr/include" CACHE FILEPATH "The directory containing headers of the RAPIDJSON")
 endif()
 
 if (NOT 3RDPARTY_RAPIDJSON_INCLUDE_DIR OR NOT EXISTS "${3RDPARTY_RAPIDJSON_INCLUDE_DIR}")
