Index: occt-V7_8_0/adm/templates/OpenCASCADEConfig.cmake.in
===================================================================
--- occt-V7_8_0.orig/adm/templates/OpenCASCADEConfig.cmake.in
+++ occt-V7_8_0/adm/templates/OpenCASCADEConfig.cmake.in
@@ -27,7 +27,7 @@ get_filename_component (OpenCASCADE_INST
 if (OpenCASCADE_INSTALL_PREFIX MATCHES "/cmake$")
   get_filename_component (OpenCASCADE_INSTALL_PREFIX "${OpenCASCADE_INSTALL_PREFIX}" PATH)
 endif()
-if (OpenCASCADE_INSTALL_PREFIX MATCHES "/lib$")
+if (OpenCASCADE_INSTALL_PREFIX MATCHES "/lib$" OR OpenCASCADE_INSTALL_PREFIX MATCHES "/lib64$")
   get_filename_component (OpenCASCADE_INSTALL_PREFIX "${OpenCASCADE_INSTALL_PREFIX}" PATH)
 endif()
 if (OpenCASCADE_INSTALL_PREFIX MATCHES "/libs/${CMAKE_ANDROID_ARCH_ABI}$")
Index: occt-V7_8_0/adm/cmake/tbb.cmake
===================================================================
--- occt-V7_8_0.orig/adm/cmake/tbb.cmake
+++ occt-V7_8_0/adm/cmake/tbb.cmake
@@ -153,21 +153,21 @@ else ()
     if ((NOT "${3RDPARTY_TBB_DIR}" STREQUAL "") AND (EXISTS "${3RDPARTY_TBB_DIR}"))
       # Find TBB 2021.5 in existing directory.
       find_package (
-      TBB 2021.5
+      TBB
       PATHS "${3RDPARTY_TBB_DIR}" NO_DEFAULT_PATH
       REQUIRED
       CONFIG)
     else()
-      # Find TBB 2021.5 in system directory.
+      # Find TBB 2021 in system directory.
       find_package (
-      TBB 2021.5
+      TBB
       REQUIRED
       CONFIG)
     endif()
   else()
-    # Find TBB 2021.5 in system directory.
+    # Find TBB 2021 in system directory.
     find_package (
-    TBB 2021.5
+    TBB
     REQUIRED
     CONFIG)
   endif()
