Index: occt-V7_7_2/src/IVtkDraw/IVtkDraw.cxx
===================================================================
--- occt-V7_7_2.orig/src/IVtkDraw/IVtkDraw.cxx
+++ occt-V7_7_2/src/IVtkDraw/IVtkDraw.cxx
@@ -31,10 +31,8 @@
 #include <NCollection_List.hxx>
 #include <NCollection_DataMap.hxx>
 #include <TopTools_DataMapOfIntegerShape.hxx>
-#include <OpenGl_GraphicDriver.hxx>
 #include <V3d.hxx>
 #include <V3d_TypeOfOrientation.hxx>
-#include <Aspect_DisplayConnection.hxx>
 
 #include <IVtk_Types.hxx>
 #include <IVtkVTK_ShapeData.hxx>
@@ -75,6 +73,9 @@
 #include <vtkSmartPointer.h>
 #include <vtkTIFFWriter.h>
 #include <vtkWindowToImageFilter.h>
+// Move X11 including files after vtk
+#include <OpenGl_GraphicDriver.hxx>
+#include <Aspect_DisplayConnection.hxx>
 #ifndef _WIN32
   #include <X11/X.h>
   #include <X11/Shell.h>
Index: occt-V7_7_2/src/IVtkDraw/IVtkDraw_Interactor.hxx
===================================================================
--- occt-V7_7_2.orig/src/IVtkDraw/IVtkDraw_Interactor.hxx
+++ occt-V7_7_2/src/IVtkDraw/IVtkDraw_Interactor.hxx
@@ -40,6 +30,17 @@
 #include <IVtkDraw_HighlightAndSelectionPipeline.hxx>
 #include <Aspect_Window.hxx>
 
+// Include X11 headers after VTK for 9.2
+#ifdef _WIN32
+#include <windows.h>
+#else
+#include <X11/X.h>
+#include <X11/Xlib.h>
+#include <X11/Shell.h>
+#include <X11/Xutil.h>
+#include <tk.h>
+#endif
+
 class vtkWin32RenderWindowInteractor;
 typedef vtkSmartPointer<IVtkTools_ShapePicker> PSelector;
 
Index: occt-V7_7_2/src/IVtkDraw/IVtkDraw_Interactor.cxx
===================================================================
--- occt-V7_7_2.orig/src/IVtkDraw/IVtkDraw_Interactor.cxx
+++ occt-V7_7_2/src/IVtkDraw/IVtkDraw_Interactor.cxx
@@ -54,6 +40,16 @@
 #include <IVtkTools_ShapeObject.hxx>
 #include <IVtkTools_ShapeDataSource.hxx>
 
+// Include X11 headers after VTK for VTK 9.2.5
+#ifdef _WIN32
+#include <vtkWin32RenderWindowInteractor.h>
+#include <vtkWin32OpenGLRenderWindow.h>
+#else
+#include <GL/glx.h>
+#include <vtkXRenderWindowInteractor.h>
+#include <vtkXOpenGLRenderWindow.h>
+#endif
+
 #include <Message.hxx>
 #include <Message_Messenger.hxx>
 
