diff -ruNb a/lib/Makefile.elf-lib b/lib/Makefile.elf-lib
--- a/lib/Makefile.elf-lib	2012-08-15 21:43:55.000000000 +0100
+++ b/lib/Makefile.elf-lib	2013-04-04 00:15:46.830015154 +0100
@@ -50,8 +50,6 @@
 	$(E) "	SYMLINK $(libdir)/$(ELF_IMAGE).so"
 	$(Q) $(INSTALL_SYMLINK) $(ELF_INSTALL_DIR)/$(ELF_SONAME) \
 		$(libdir)/$(ELF_IMAGE).so $(DESTDIR)
-	$(E) "	LDCONFIG"
-	$(Q) -$(LDCONFIG)
 
 install-strip: install
 	$(E) "	STRIP-LIB $(ELF_INSTALL_DIR)/$(ELF_LIB)"
@@ -67,7 +65,6 @@
 	$(RM) -f $(DESTDIR)$(ELF_INSTALL_DIR)/$(ELF_LIB) \
 		$(DESTDIR)$(ELF_INSTALL_DIR)/$(ELF_SONAME) \
 		$(DESTDIR)$(libdir)/$(ELF_IMAGE).so
-	-$(LDCONFIG)
 
 clean::
 	$(RM) -rf elfshared
