From e4df66126490c5b0793991db64f4ce47df9377de Mon Sep 17 00:00:00 2001
From: Fabio Forni <livingsilver94.solus@redaril.me>
Date: Thu, 14 Apr 2022 23:37:52 +0200
Subject: [PATCH] Set XDG variables in tests

---
 tests/conftest.py | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/tests/conftest.py b/tests/conftest.py
index 964458b..82218de 100644
--- a/tests/conftest.py
+++ b/tests/conftest.py
@@ -69,5 +69,7 @@ def set_shell(monkeypatch):
 @pytest.fixture(autouse=True)
 def os_environ(monkeypatch):
     env = {'PATH': os.environ['PATH']}
+    env['XDG_CONFIG_HOME'] = str(Path('.').resolve())
+    env['XDG_CACHE_HOME']  = str(Path('.').resolve())
     monkeypatch.setattr('os.environ', env)
     return env
-- 
2.35.1

