From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Thomas Staudinger <Staudi.Kaos@gmail.com>
Date: Tue, 27 Feb 2024 04:00:03 +0100
Subject: [PATCH] Account for permission errors during tests in build env

Signed-off-by: Thomas Staudinger <Staudi.Kaos@gmail.com>
---
 thefuck/conf.py | 14 +++++++++-----
 1 file changed, 9 insertions(+), 5 deletions(-)

diff --git a/thefuck/conf.py b/thefuck/conf.py
index 611ec84..8ab8506 100644
--- a/thefuck/conf.py
+++ b/thefuck/conf.py
@@ -58,11 +58,15 @@ class Settings(dict):
         legacy_user_dir = Path('~', '.thefuck').expanduser()
 
         # For backward compatibility use legacy '~/.thefuck' if it exists:
-        if legacy_user_dir.is_dir():
-            warn(u'Config path {} is deprecated. Please move to {}'.format(
-                legacy_user_dir, user_dir))
-            return legacy_user_dir
-        else:
+        try:
+            if legacy_user_dir.is_dir():
+                warn(u'Config path {} is deprecated. Please move to {}'.format(
+                    legacy_user_dir, user_dir))
+                return legacy_user_dir
+            else:
+                return user_dir
+        except PermissionError:
+            warn(u'Config path {} not accesible'.format(legacy_user_dir))
             return user_dir
 
     def _setup_user_dir(self):
