From dd26fb91a0fdec42fc1990bb91eab21e2c44a0a8 Mon Sep 17 00:00:00 2001
From: Branch Vincent <branchevincent@gmail.com>
Date: Sun, 8 Oct 2023 09:54:07 -0700
Subject: [PATCH] replace distutils for python 3.12

---
 thefuck/system/unix.py | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/thefuck/system/unix.py b/thefuck/system/unix.py
index 9ae01cf4c..c3385983f 100644
--- a/thefuck/system/unix.py
+++ b/thefuck/system/unix.py
@@ -3,7 +3,7 @@
 import tty
 import termios
 import colorama
-from distutils.spawn import find_executable
+from shutil import which
 from .. import const
 
 init_output = colorama.init
@@ -38,7 +38,7 @@ def get_key():
 
 
 def open_command(arg):
-    if find_executable('xdg-open'):
+    if which('xdg-open'):
         return 'xdg-open ' + arg
     return 'open ' + arg
 
