diff --git i/tests/entrypoints/test_alias.py w/tests/entrypoints/test_alias.py
index ddb13e1..26e74bf 100644
--- i/tests/entrypoints/test_alias.py
+++ w/tests/entrypoints/test_alias.py
@@ -1,4 +1,4 @@
-from mock import Mock
+from unittest.mock import Mock
 import pytest
 from thefuck.entrypoints.alias import _get_alias, print_alias
 
diff --git i/tests/entrypoints/test_fix_command.py w/tests/entrypoints/test_fix_command.py
index 3012bb4..e1ad2a8 100644
--- i/tests/entrypoints/test_fix_command.py
+++ w/tests/entrypoints/test_fix_command.py
@@ -1,5 +1,5 @@
 import pytest
-from mock import Mock
+from unittest.mock import Mock
 from thefuck.entrypoints.fix_command import _get_raw_command
 
 
diff --git i/tests/entrypoints/test_not_configured.py w/tests/entrypoints/test_not_configured.py
index 79ed29f..f4c225a 100644
--- i/tests/entrypoints/test_not_configured.py
+++ w/tests/entrypoints/test_not_configured.py
@@ -1,7 +1,7 @@
 import pytest
 import json
 from six import StringIO
-from mock import MagicMock
+from unittest.mock import MagicMock
 from thefuck.shells.generic import ShellConfiguration
 from thefuck.entrypoints.not_configured import main
 
diff --git i/tests/output_readers/test_rerun.py w/tests/output_readers/test_rerun.py
index 1d3a2c8..3f0f1d2 100644
--- i/tests/output_readers/test_rerun.py
+++ w/tests/output_readers/test_rerun.py
@@ -1,6 +1,6 @@
 # -*- encoding: utf-8 -*-
 
-from mock import Mock, patch
+from unittest.mock import Mock, patch
 from psutil import AccessDenied, TimeoutExpired
 
 from thefuck.output_readers import rerun
diff --git i/tests/rules/test_has_exists_script.py w/tests/rules/test_has_exists_script.py
index 7814a75..9f292e4 100644
--- i/tests/rules/test_has_exists_script.py
+++ w/tests/rules/test_has_exists_script.py
@@ -1,4 +1,4 @@
-from mock import patch
+from unittest.mock import patch
 from thefuck.rules.has_exists_script import match, get_new_command
 from thefuck.types import Command
 
diff --git i/tests/rules/test_pacman.py w/tests/rules/test_pacman.py
index 985b8b8..ffbf526 100644
--- i/tests/rules/test_pacman.py
+++ w/tests/rules/test_pacman.py
@@ -1,5 +1,5 @@
 import pytest
-from mock import patch
+from unittest.mock import patch
 from thefuck.rules import pacman
 from thefuck.rules.pacman import match, get_new_command
 from thefuck.types import Command
diff --git i/tests/rules/test_pacman_not_found.py w/tests/rules/test_pacman_not_found.py
index a1b6532..ead71c9 100644
--- i/tests/rules/test_pacman_not_found.py
+++ w/tests/rules/test_pacman_not_found.py
@@ -1,5 +1,5 @@
 import pytest
-from mock import patch
+from unittest.mock import patch
 from thefuck.rules import pacman_not_found
 from thefuck.rules.pacman_not_found import match, get_new_command
 from thefuck.types import Command
diff --git i/tests/test_conf.py w/tests/test_conf.py
index e03473a..72d7d7e 100644
--- i/tests/test_conf.py
+++ w/tests/test_conf.py
@@ -1,7 +1,7 @@
 import pytest
 import six
 import os
-from mock import Mock
+from unittest.mock import Mock
 from thefuck import const
 
 
diff --git i/tests/test_types.py w/tests/test_types.py
index a322d21..5d4455e 100644
--- i/tests/test_types.py
+++ w/tests/test_types.py
@@ -2,7 +2,7 @@
 
 import os
 from subprocess import PIPE, STDOUT
-from mock import Mock
+from unittest.mock import Mock
 import pytest
 from tests.utils import CorrectedCommand, Rule
 from thefuck import const
diff --git i/tests/test_utils.py w/tests/test_utils.py
index eae743f..a56a293 100644
--- i/tests/test_utils.py
+++ w/tests/test_utils.py
@@ -2,7 +2,7 @@
 
 import pytest
 import warnings
-from mock import Mock, call, patch
+from unittest.mock import Mock, call, patch
 from thefuck.utils import default_settings, \
     memoize, get_closest, get_all_executables, replace_argument, \
     get_all_matched_commands, is_app, for_app, cache, \
