diff '--color=auto' -pruN a/setup.py b/setup.py
--- a/setup.py	2022-01-02 16:45:09.000000000 -0500
+++ b/setup.py	2026-06-08 17:30:06.375266926 -0400
@@ -1,17 +1,17 @@
 #!/usr/bin/env python
 from setuptools import setup, find_packages
-import pkg_resources
+from importlib.metadata import version as get_version, PackageNotFoundError
 import sys
 import os
 import fastentrypoints
 
 
 try:
-    if int(pkg_resources.get_distribution("pip").version.split('.')[0]) < 6:
+    if int(get_version("pip").split('.')[0]) < 6:
         print('pip older than 6.0 not supported, please upgrade pip with:\n\n'
               '    pip install -U pip')
         sys.exit(-1)
-except pkg_resources.DistributionNotFound:
+except PackageNotFoundError:
     pass
 
 if os.environ.get('CONVERT_README'):
