From 232e2773af7d51cb99cf794dc033164ac273323f Mon Sep 17 00:00:00 2001
From: Oded Arbel <oded@geek.co.il>
Date: Tue, 2 Jan 2024 01:29:02 +0200
Subject: [PATCH 1/2] add support for Plasma 6

---
 data/scripts/set_wallpaper | 14 ++++++++++----
 1 file changed, 10 insertions(+), 4 deletions(-)

diff --git a/data/scripts/set_wallpaper b/data/scripts/set_wallpaper
index 188dcee9..ccda46ee 100755
--- a/data/scripts/set_wallpaper
+++ b/data/scripts/set_wallpaper
@@ -128,16 +128,22 @@ fi
 if [ "${KDE_FULL_SESSION}" == "true" ]; then
     # Plasma 5.7 introduced a new feature to set the wallpaper via a dbus script:
     # https://github.com/KDE/plasma-workspace/commit/903cbfd7e267a4812a6ec222eb7e1b5dd775686f
-    if [[ -n "${KDE_SESSION_VERSION}" && "${KDE_SESSION_VERSION}" == '5' ]]; then
-        dbus-send --type=method_call --dest=org.kde.plasmashell /PlasmaShell org.kde.PlasmaShell.evaluateScript string:"
+    plasma_qdbus_script="
         let allDesktops = desktops();
         for (let d of allDesktops) {
             if (d.wallpaperPlugin == 'org.kde.image') {
                 d.currentConfigGroup = Array('Wallpaper', 'org.kde.image', 'General');
-                d.writeConfig('Image', 'file://""$WP""')
+                d.writeConfig('Image', 'file://""$WP""');
             }
         }
-        "
+    "
+    if [[ -n "${KDE_SESSION_VERSION}" && "${KDE_SESSION_VERSION}" == '6' ]]; then
+        dbus-send --type=method_call --dest=org.kde.plasmashell /PlasmaShell org.kde.PlasmaShell.evaluateScript string:"$plasma_qdbus_script"
+        # Reuse the exit code from dbus
+        exit "$?"
+
+    elif [[ -n "${KDE_SESSION_VERSION}" && "${KDE_SESSION_VERSION}" == '5' ]]; then
+        dbus-send --type=method_call --dest=org.kde.plasmashell /PlasmaShell org.kde.PlasmaShell.evaluateScript string:"$plasma_qdbus_script"
         # Reuse the exit code from dbus
         dbus_exitcode="$?"
 

From 718bbba2a712b8c83be7184f22825f98fe885c9d Mon Sep 17 00:00:00 2001
From: Oded Arbel <oded@geek.co.il>
Date: Tue, 2 Jan 2024 22:43:51 +0200
Subject: [PATCH 2/2] use the same block for Plasma 5 and 6

---
 data/scripts/set_wallpaper | 9 ++-------
 1 file changed, 2 insertions(+), 7 deletions(-)

diff --git a/data/scripts/set_wallpaper b/data/scripts/set_wallpaper
index ccda46ee..8bfe2391 100755
--- a/data/scripts/set_wallpaper
+++ b/data/scripts/set_wallpaper
@@ -137,17 +137,12 @@ if [ "${KDE_FULL_SESSION}" == "true" ]; then
             }
         }
     "
-    if [[ -n "${KDE_SESSION_VERSION}" && "${KDE_SESSION_VERSION}" == '6' ]]; then
-        dbus-send --type=method_call --dest=org.kde.plasmashell /PlasmaShell org.kde.PlasmaShell.evaluateScript string:"$plasma_qdbus_script"
-        # Reuse the exit code from dbus
-        exit "$?"
-
-    elif [[ -n "${KDE_SESSION_VERSION}" && "${KDE_SESSION_VERSION}" == '5' ]]; then
+    if [[ -n "${KDE_SESSION_VERSION}" && "${KDE_SESSION_VERSION}" -ge '5' ]]; then
         dbus-send --type=method_call --dest=org.kde.plasmashell /PlasmaShell org.kde.PlasmaShell.evaluateScript string:"$plasma_qdbus_script"
         # Reuse the exit code from dbus
         dbus_exitcode="$?"
 
-        if [[ "$dbus_exitcode" -ne 0 ]]; then
+        if [[ "$dbus_exitcode" -ne 0 && "${KDE_SESSION_VERSION}" -eq '5' ]]; then
             # If the script fails, show a notification.
             kdialog --title "Variety: cannot change Plasma wallpaper" --passivepopup "Could not change the Plasma 5 wallpaper; \
                 make sure that you're using Plasma 5.7+ and have widgets unlocked.\n----\n \
