diff -up fcitx-4.2.9.9/src/frontend/qt/qtkey.h.orig fcitx-4.2.9.9/src/frontend/qt/qtkey.h
--- fcitx-4.2.9.9/src/frontend/qt/qtkey.h.orig	2023-01-25 15:16:47.485468140 -0500
+++ fcitx-4.2.9.9/src/frontend/qt/qtkey.h	2023-01-25 15:16:59.368500430 -0500
@@ -21,6 +21,7 @@
 #define _PLATFORMINPUTCONTEXT_QTKEY_H_
 
 #include <QString>
+#include <cstdint>
 
 int keysymToQtKey(uint32_t keysym, const QString &text);
 
