From 0704eb79385910e9029caf062c09b6743b579a96 Mon Sep 17 00:00:00 2001
From: Peter O'Connor <peter@solus-project.com>
Date: Fri, 20 Apr 2018 10:17:40 +1000
Subject: [PATCH] Disable building unshipped code

Signed-off-by: Peter O'Connor <peter@solus-project.com>
---
 CMakeLists.txt | 1 -
 1 file changed, 1 deletion(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index c0fe421..7c691fa 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -124,7 +124,6 @@ add_subdirectory(extensions)
 add_subdirectory(sounds)
 add_subdirectory(desktop)
 #add_subdirectory(plasma) TODO PORT TO KF5/QML
-add_subdirectory(tests)
 add_subdirectory(doc)
 
 find_package(LibKWorkspace CONFIG)
-- 
2.16.3

