---
 0.5.1/Lector-0.5.1/lector/toolbars.py | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/lector/toolbars.py b/lector/toolbars.py
index 3020e6c..05e526d 100644
--- a/lector/toolbars.py
+++ b/lector/toolbars.py
@@ -498,7 +498,7 @@ class FixedComboBox(QtWidgets.QComboBox):
 
     def sizeHint(self):
         # This and the one below should adjust to screen size
-        return QtCore.QSize(self.adjusted_size, 22)
+        return QtCore.QSize(int(self.adjusted_size), 32)
 
     def wheelEvent(self, QWheelEvent):
         # Disable mouse wheel scrolling in the ComboBox
@@ -511,7 +511,7 @@ class FixedLineEdit(QtWidgets.QLineEdit):
         self.adjusted_size = screen_width // 4.5
 
     def sizeHint(self):
-        return QtCore.QSize(self.adjusted_size, 22)
+        return QtCore.QSize(int(self.adjusted_size), 32)
 
     def keyReleaseEvent(self, event):
         if event.key() == QtCore.Qt.Key_Escape:
-- 
2.35.3

