From d3f9cf387a21e0f8d3bbff980245ba8fcab124c1 Mon Sep 17 00:00:00 2001
From: Gavin Zhao <git@gzgz.dev>
Date: Sun, 28 Sep 2025 22:13:43 -0400
Subject: [PATCH] systemd: use /usr/bin for exectuable path

---
 resources/xwayland-satellite.service | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/resources/xwayland-satellite.service b/resources/xwayland-satellite.service
index 060ab6b..f68cefd 100644
--- a/resources/xwayland-satellite.service
+++ b/resources/xwayland-satellite.service
@@ -8,7 +8,7 @@ Requisite=graphical-session.target
 [Service]
 Type=notify
 NotifyAccess=all
-ExecStart=/usr/local/bin/xwayland-satellite
+ExecStart=/usr/bin/xwayland-satellite
 StandardOutput=journal
 
 [Install]
-- 
2.51.0

