From 7c14ef015f2f95f264f53404dfcea68b1214d6e9 Mon Sep 17 00:00:00 2001
From: Julien Schueller <schueller@phimeca.com>
Date: Sun, 13 Oct 2019 09:56:05 +0200
Subject: [PATCH] Allow llvm>=9.0

---
 ffi/build.py | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/ffi/build.py b/ffi/build.py
index 9efb1895..778e028b 100755
--- a/ffi/build.py
+++ b/ffi/build.py
@@ -12,6 +12,7 @@
 import shutil
 import sys
 import tempfile
+from distutils.version import LooseVersion
 
 
 here_dir = os.path.abspath(os.path.dirname(__file__))
@@ -109,8 +110,7 @@ def main_posix(kind, library_ext):
 
     out = out.decode('latin1')
     print(out)
-    if not (out.startswith('8.0.') or out.startswith('7.0.')
-            or out.startswith('7.1.')):
+    if LooseVersion(out) < '7.0':
         msg = (
             "Building llvmlite requires LLVM 7.0+ Be sure to "
             "set LLVM_CONFIG to the right executable path.\n"
