diff -ur a/ffi/Makefile.linux b/ffi/Makefile.linux
--- a/ffi/Makefile.linux	2019-05-29 23:43:46.000000000 +0800
+++ b/ffi/Makefile.linux	2019-09-27 23:40:04.196721223 +0800
@@ -2,7 +2,7 @@
 CXX ?= g++
 
 # -flto and --exclude-libs allow us to remove those parts of LLVM we don't use
-CXX_FLTO_FLAGS ?= -flto
+CXX_FLTO_FLAGS ?= -flto -fPIC
 LD_FLTO_FLAGS ?= -flto -Wl,--exclude-libs=ALL
 
 CXXFLAGS = $(LLVM_CXXFLAGS) $(CXX_FLTO_FLAGS)
