From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Evan Maddock <maddock.evan@vivaldi.net>
Date: Fri, 31 Oct 2025 20:04:35 -0400
Subject: [PATCH] build: Port to libpeas-2 and budgie-1.0

Signed-off-by: Evan Maddock <maddock.evan@vivaldi.net>
---
 meson.build | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/meson.build b/meson.build
index 269afd8f..a396836a 100644
--- a/meson.build
+++ b/meson.build
@@ -67,11 +67,11 @@ posix_dep = meson.get_compiler('vala').find_library('posix')
 for_wayland = get_option('for-wayland')
 
 if for_wayland == false
-    budgie_dep = dependency('budgie-1.0')
+    budgie_dep = dependency('budgie-2.0')
 else
     budgie_dep = dependency('budgie-2.0')
 endif
-peas_dep = dependency('libpeas-gtk-1.0')
+peas_dep = dependency('libpeas-2')
 
 plank_dep = dependency('plank', required: false)
 if plank_dep.version().version_compare('>=0.10.9')
