From 4bd619715823cc84a4f56deecc2b7ecc56fd3fda Mon Sep 17 00:00:00 2001
From: Joshua Strobl <joshua@stroblindustries.com>
Date: Sun, 9 Apr 2017 05:03:44 +0300
Subject: [PATCH 1/1] Fix build.

---
 src/gnome-chess.vala | 48 ++++++++++++++++++++++++++----------------------
 1 file changed, 26 insertions(+), 22 deletions(-)

diff --git a/src/gnome-chess.vala b/src/gnome-chess.vala
index 35d0c3f..3a1f7ea 100644
--- a/src/gnome-chess.vala
+++ b/src/gnome-chess.vala
@@ -1143,7 +1143,11 @@ Copyright © 2015–2016 Sahil Sareen""";
         var model = (Gtk.ListStore) history_combo.model;
         Gtk.TreeIter iter;
         model.iter_nth_child (out iter, null, model.iter_n_children (null) - 1);
-        model.remove (ref iter);
+        #if VALA_0_36
+                model.remove (ref iter);
+        #else
+                model.remove (iter);
+        #endif
 
         /* Always undo from the most recent move */
         scene.move_number = -1;
@@ -1382,7 +1386,7 @@ Copyright © 2015–2016 Sahil Sareen""";
         black_time_label.queue_draw ();
     }
 
-    [CCode (cname = "G_MODULE_EXPORT gnome_chess_app_delete_event_cb", instance_pos = -1)]
+    [CCode (cname = "gnome_chess_app_delete_event_cb", instance_pos = -1)]
     public bool gnome_chess_app_delete_event_cb (Gtk.Widget widget, Gdk.Event event)
     {
         quit_game ();
@@ -1551,7 +1555,7 @@ Copyright © 2015–2016 Sahil Sareen""";
         quit_game ();
     }
 
-    [CCode (cname = "G_MODULE_EXPORT white_time_draw_cb", instance_pos = -1)]
+    [CCode (cname = "white_time_draw_cb", instance_pos = -1)]
     public bool white_time_draw_cb (Gtk.Widget widget, Cairo.Context c)
     {
         double fg[3] = { 0.0, 0.0, 0.0 };
@@ -1561,7 +1565,7 @@ Copyright © 2015–2016 Sahil Sareen""";
         return false;
     }
 
-    [CCode (cname = "G_MODULE_EXPORT black_time_draw_cb", instance_pos = -1)]
+    [CCode (cname = "black_time_draw_cb", instance_pos = -1)]
     public bool black_time_draw_cb (Gtk.Widget widget, Cairo.Context c)
     {
         double fg[3] = { 1.0, 1.0, 1.0 };
@@ -1648,7 +1652,7 @@ Copyright © 2015–2016 Sahil Sareen""";
             widget.set_size_request (compute_time_label_width_request (c), -1);
     }
 
-    [CCode (cname = "G_MODULE_EXPORT history_combo_changed_cb", instance_pos = -1)]
+    [CCode (cname = "history_combo_changed_cb", instance_pos = -1)]
     public void history_combo_changed_cb (Gtk.ComboBox combo)
     {
         Gtk.TreeIter iter;
@@ -1661,13 +1665,13 @@ Copyright © 2015–2016 Sahil Sareen""";
         scene.move_number = move_number;
     }
 
-    [CCode (cname = "G_MODULE_EXPORT history_latest_clicked_cb", instance_pos = -1)]
+    [CCode (cname = "history_latest_clicked_cb", instance_pos = -1)]
     public void history_latest_clicked_cb (Gtk.Widget widget)
     {
         scene.move_number = -1;
     }
 
-    [CCode (cname = "G_MODULE_EXPORT history_next_clicked_cb", instance_pos = -1)]
+    [CCode (cname = "history_next_clicked_cb", instance_pos = -1)]
     public void history_next_clicked_cb (Gtk.Widget widget)
     {
         if (scene.move_number == -1)
@@ -1680,7 +1684,7 @@ Copyright © 2015–2016 Sahil Sareen""";
             scene.move_number = move_number;
     }
 
-    [CCode (cname = "G_MODULE_EXPORT history_previous_clicked_cb", instance_pos = -1)]
+    [CCode (cname = "history_previous_clicked_cb", instance_pos = -1)]
     public void history_previous_clicked_cb (Gtk.Widget widget)
     {
         if (scene.move_number == 0)
@@ -1692,7 +1696,7 @@ Copyright © 2015–2016 Sahil Sareen""";
             scene.move_number = scene.move_number - 1;
     }
 
-    [CCode (cname = "G_MODULE_EXPORT history_start_clicked_cb", instance_pos = -1)]
+    [CCode (cname = "history_start_clicked_cb", instance_pos = -1)]
     public void history_start_clicked_cb (Gtk.Widget widget)
     {
         scene.move_number = 0;
@@ -1811,7 +1815,7 @@ Copyright © 2015–2016 Sahil Sareen""";
         return value;
     }
 
-    [CCode (cname = "G_MODULE_EXPORT side_combo_changed_cb", instance_pos = -1)]
+    [CCode (cname = "side_combo_changed_cb", instance_pos = -1)]
     public void side_combo_changed_cb (Gtk.ComboBox combo)
     {
         Gtk.TreeIter iter;
@@ -1823,7 +1827,7 @@ Copyright © 2015–2016 Sahil Sareen""";
         settings.set_enum ("play-as", player);
     }
 
-    [CCode (cname = "G_MODULE_EXPORT opponent_combo_changed_cb", instance_pos = -1)]
+    [CCode (cname = "opponent_combo_changed_cb", instance_pos = -1)]
     public void opponent_combo_changed_cb (Gtk.ComboBox combo)
     {
         Gtk.TreeIter iter;
@@ -1837,7 +1841,7 @@ Copyright © 2015–2016 Sahil Sareen""";
         difficulty_combo.sensitive = !vs_human;
     }
 
-    [CCode (cname = "G_MODULE_EXPORT difficulty_combo_changed_cb", instance_pos = -1)]
+    [CCode (cname = "difficulty_combo_changed_cb", instance_pos = -1)]
     public void difficulty_combo_changed_cb (Gtk.ComboBox combo)
     {
         Gtk.TreeIter iter;
@@ -2000,7 +2004,7 @@ Copyright © 2015–2016 Sahil Sareen""";
         return Source.REMOVE;
     }
 
-    [CCode (cname = "G_MODULE_EXPORT duration_changed_cb", instance_pos = -1)]
+    [CCode (cname = "duration_changed_cb", instance_pos = -1)]
     public void duration_changed_cb (Gtk.Adjustment adjustment)
     {
         var model = (Gtk.ListStore) custom_duration_units_combo.model;
@@ -2031,13 +2035,13 @@ Copyright © 2015–2016 Sahil Sareen""";
         save_duration ();
     }
 
-    [CCode (cname = "G_MODULE_EXPORT duration_units_changed_cb", instance_pos = -1)]
+    [CCode (cname = "duration_units_changed_cb", instance_pos = -1)]
     public void duration_units_changed_cb (Gtk.Widget widget)
     {
         save_duration ();
     }
 
-    [CCode (cname = "G_MODULE_EXPORT timer_increment_units_changed_cb", instance_pos = -1)]
+    [CCode (cname = "timer_increment_units_changed_cb", instance_pos = -1)]
     public void timer_increment_units_changed_cb (Gtk.Widget widget)
     {
         var model = (Gtk.ListStore) timer_increment_units_combo.model;
@@ -2096,7 +2100,7 @@ Copyright © 2015–2016 Sahil Sareen""";
         save_duration_timeout = Timeout.add (100, save_duration_cb);
     }
 
-    [CCode (cname = "G_MODULE_EXPORT duration_combo_changed_cb", instance_pos = -1)]
+    [CCode (cname = "duration_combo_changed_cb", instance_pos = -1)]
     public void duration_combo_changed_cb (Gtk.ComboBox combo)
     {
         Gtk.TreeIter iter;
@@ -2119,7 +2123,7 @@ Copyright © 2015–2016 Sahil Sareen""";
         save_duration ();
     }
 
-    [CCode (cname = "G_MODULE_EXPORT clock_type_changed_cb", instance_pos = -1)]
+    [CCode (cname = "clock_type_changed_cb", instance_pos = -1)]
     public void clock_type_changed_cb (Gtk.ComboBox combo)
     {
         Gtk.TreeIter iter;
@@ -2133,32 +2137,32 @@ Copyright © 2015–2016 Sahil Sareen""";
         settings.set_string ("clock-type", clock_type.to_string ());
     }
 
-    [CCode (cname = "G_MODULE_EXPORT preferences_response_cb", instance_pos = -1)]
+    [CCode (cname = "preferences_response_cb", instance_pos = -1)]
     public void preferences_response_cb (Gtk.Widget widget, int response_id)
     {
         preferences_dialog.hide ();
     }
 
-    [CCode (cname = "G_MODULE_EXPORT preferences_delete_event_cb", instance_pos = -1)]
+    [CCode (cname = "preferences_delete_event_cb", instance_pos = -1)]
     public bool preferences_delete_event_cb (Gtk.Widget widget, Gdk.Event event)
     {
         preferences_response_cb (widget, Gtk.ResponseType.CANCEL);
         return true;
     }
 
-    [CCode (cname = "G_MODULE_EXPORT piece_style_combo_changed_cb", instance_pos = -1)]
+    [CCode (cname = "piece_style_combo_changed_cb", instance_pos = -1)]
     public void piece_style_combo_changed_cb (Gtk.ComboBox combo)
     {
         settings.set_string ("piece-theme", get_combo (combo, 1));
     }
 
-    [CCode (cname = "G_MODULE_EXPORT move_format_combo_changed_cb", instance_pos = -1)]
+    [CCode (cname = "move_format_combo_changed_cb", instance_pos = -1)]
     public void move_format_combo_changed_cb (Gtk.ComboBox combo)
     {
         settings.set_string ("move-format", get_combo (combo, 1));
     }
 
-    [CCode (cname = "G_MODULE_EXPORT orientation_combo_changed_cb", instance_pos = -1)]
+    [CCode (cname = "orientation_combo_changed_cb", instance_pos = -1)]
     public void orientation_combo_changed_cb (Gtk.ComboBox combo)
     {
         settings.set_string ("board-side", get_combo (combo, 1));
-- 
2.12.2

