--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -9,17 +9,17 @@
 
 include(gd)
 
-OPTION(ENABLE_GD_FORMATS "Enable GD image formats" 0)
-OPTION(ENABLE_PNG "Enable PNG support" 0)
-OPTION(ENABLE_LIQ "Enable libimagequant support" 0)
-OPTION(ENABLE_JPEG "Enable JPEG support" 0)
-OPTION(ENABLE_TIFF "Enable TIFF support" 0)
-OPTION(ENABLE_ICONV "Enable ICONV support" 0)
-OPTION(ENABLE_XPM "Enable XPM support" 0)
-OPTION(ENABLE_FREETYPE "Enable Freetype2 support" 0)
-OPTION(ENABLE_FONTCONFIG "Enable FontConfig support" 0)
-OPTION(ENABLE_WEBP "Enable WebP support" 0)
-OPTION(ENABLE_RAQM "Enable RAQM support" 0)
+OPTION(ENABLE_GD_FORMATS "Enable GD image formats" 1)
+OPTION(ENABLE_PNG "Enable PNG support" 1)
+OPTION(ENABLE_LIQ "Enable libimagequant support" 1)
+OPTION(ENABLE_JPEG "Enable JPEG support" 1)
+OPTION(ENABLE_TIFF "Enable TIFF support" 1)
+OPTION(ENABLE_ICONV "Enable ICONV support" 1)
+OPTION(ENABLE_XPM "Enable XPM support" 1)
+OPTION(ENABLE_FREETYPE "Enable Freetype2 support" 1)
+OPTION(ENABLE_FONTCONFIG "Enable FontConfig support" 1)
+OPTION(ENABLE_WEBP "Enable WebP support" 1)
+OPTION(ENABLE_RAQM "Enable RAQM support" 1)
 
 if (BUILD_TEST)
 	ENABLE_TESTING()
