--- a/src/gd_gd.c
+++ b/src/gd_gd.c
@@ -60,7 +60,7 @@
 #include "gd_errors.h"
 
 /* 2.3: gd is no longer mandatory */
-#if ENABLE_GD_FORMATS
+#ifdef ENABLE_GD_FORMATS
 
 #define TRUE 1
 #define FALSE 0
--- a/src/gd_gd2.c
+++ b/src/gd_gd2.c
@@ -85,7 +85,7 @@
 /* 2.03: gd2 is no longer mandatory */
 /* JCE - test after including gd.h so that HAVE_LIBZ can be set in
  * a config.h file included by gd.h */
-#if defined(HAVE_LIBZ) && ENABLE_GD_FORMATS
+#if defined(HAVE_LIBZ) && defined(ENABLE_GD_FORMATS)
 #include <zlib.h>
 
 #define TRUE 1
