From 967eabab6dd55f5e14c26c67707bee53c0976598 Mon Sep 17 00:00:00 2001
From: Ikey Doherty <ikey@solus-project.com>
Date: Thu, 3 May 2018 11:06:25 +0100
Subject: [PATCH] util: Emit compatible rules for newer udev + kernels using
 BIND

Signed-off-by: Ikey Doherty <ikey@solus-project.com>
---
 util/mtp-hotplug.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/util/mtp-hotplug.c b/util/mtp-hotplug.c
index 8e118ae..d534e15 100644
--- a/util/mtp-hotplug.c
+++ b/util/mtp-hotplug.c
@@ -140,7 +140,7 @@ int main (int argc, char **argv)
     case style_udev:
       printf("# UDEV-style hotplug map for libmtp\n");
       printf("# Put this file in /etc/udev/rules.d\n\n");
-      printf("ACTION!=\"add\", GOTO=\"libmtp_rules_end\"\n");
+      printf("ACTION!=\"bind\", ACTION!=\"add\", GOTO=\"libmtp_rules_end\"\n");
       printf("ENV{MAJOR}!=\"?*\", GOTO=\"libmtp_rules_end\"\n");
       printf("SUBSYSTEM==\"usb\", GOTO=\"libmtp_usb_rules\"\n"
 	     "GOTO=\"libmtp_rules_end\"\n\n"
@@ -165,7 +165,7 @@ int main (int argc, char **argv)
     case style_udev_old:
       printf("# UDEV-style hotplug map for libmtp\n");
       printf("# Put this file in /etc/udev/rules.d\n\n");
-      printf("ACTION!=\"add\", GOTO=\"libmtp_rules_end\"\n");
+      printf("ACTION!=\"bind\", ACTION!=\"add\", GOTO=\"libmtp_rules_end\"\n");
       printf("ENV{MAJOR}!=\"?*\", GOTO=\"libmtp_rules_end\"\n");
       printf("SUBSYSTEM==\"usb_device\", GOTO=\"libmtp_usb_device_rules\"\n"
 	     "GOTO=\"libmtp_rules_end\"\n\n"
-- 
2.16.3

