--- glslang-715c353a15836e5ae192a64a4cf54e2ce7e8d66a/Test/runtests
+++ glslang-715c353a15836e5ae192a64a4cf54e2ce7e8d66a-new/Test/runtests
@@ -2,8 +2,8 @@
 
 TARGETDIR=localResults
 BASEDIR=baseResults
-EXE=../build/install/bin/glslangValidator
-REMAPEXE=../build/install/bin/spirv-remap
+EXE=../solusBuildDir/StandAlone/glslangValidator
+REMAPEXE=../solusBuildDir/StandAlone/spirv-remap
 HASERROR=0
 mkdir -p localResults
 

--- glslang-715c353a15836e5ae192a64a4cf54e2ce7e8d66a/Test/glslangValidator
+++ glslang-715c353a15836e5ae192a64a4cf54e2ce7e8d66a-new/Test/glslangValidator
@@ -1,2 +1,2 @@
 #!/usr/bin/env bash
-../build/install/bin/glslangValidator $*
+../solusBuildDir/StandAlone/glslangValidator $*

