From 9c9ef49fdc790e78d285e4627484eaa5cc0394ed Mon Sep 17 00:00:00 2001
From: Reilly Brogan <reilly@reillybrogan.com>
Date: Wed, 2 Aug 2023 23:43:14 -0500
Subject: [PATCH] Fix status icon on Solus

---
 syncthing_gtk/statusicon.py | 6 +++++-
 1 file changed, 5 insertions(+), 1 deletion(-)

diff --git a/syncthing_gtk/statusicon.py b/syncthing_gtk/statusicon.py
index 89b2438..95a9068 100644
--- a/syncthing_gtk/statusicon.py
+++ b/syncthing_gtk/statusicon.py
@@ -218,6 +218,10 @@ class StatusIconGTK3(StatusIcon):
                 # Unity fakes SysTray support but actually hides all icons...
                 raise NotImplementedError
 
+            if IS_KDE:
+                # GTK3 icon never seems to load on Solus KDE
+                raise NotImplementedError
+
         self._tray = Gtk.StatusIcon()
 
         self._tray.connect("activate", self._on_click)
@@ -487,7 +491,7 @@ class StatusIconProxy(StatusIcon):
 
     def _load_fallback(self):
         if IS_KDE:
-            status_icon_backends = [StatusIconAppIndicator, StatusIconQt5, StatusIconDummy]
+            status_icon_backends = [StatusIconAppIndicator, StatusIconQt, StatusIconDummy]
         else:
             status_icon_backends = [StatusIconAppIndicator, StatusIconDummy]
 
-- 
2.41.0

