From f245d73bd44645540418caaafbe5a098a6445052 Mon Sep 17 00:00:00 2001
From: Reilly Brogan <reilly@reillybrogan.com>
Date: Wed, 2 Aug 2023 23:55:21 -0500
Subject: [PATCH] solus: Set version variable

---
 setup.py | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/setup.py b/setup.py
index 8e19561..14edd89 100755
--- a/setup.py
+++ b/setup.py
@@ -17,8 +17,8 @@ def get_version():
     path. If both methods fails, returns 'unknown'.
     """
     try:
-        version = subprocess.check_output(["git", "describe", "--tags"])
-        version = version.decode("utf-8").strip("\n\r \t")
+        # On Solus just pull the version from the $version variable which is set at build time
+        version = os.environ['version']
     except subprocess.CalledProcessError:
         # Git-describe method failed, try to guess from working directory name
         project_dir_name = Path(__file__).parent.name
-- 
2.41.0

