From a5539cc04e975c3636425a54345e3c5818cd9aad Mon Sep 17 00:00:00 2001
From: Tracey Clark <traceyc.dev@tlcnet.info>
Date: Fri, 29 Aug 2025 17:54:59 -0500
Subject: [PATCH] Update to work with syncthing 2.0

---
 syncthing_gtk/app.py | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/syncthing_gtk/app.py b/syncthing_gtk/app.py
index bb5909b..e38657e 100644
--- a/syncthing_gtk/app.py
+++ b/syncthing_gtk/app.py
@@ -508,11 +508,11 @@ class App(Gtk.Application, TimerManager):
         Sets self.process, adds related handlers and starts daemon.
         Just so I don't have to write same code all over the place.
         """
-        cmdline = [self.config["syncthing_binary"], "-no-browser"]
+        cmdline = [self.config["syncthing_binary"], "--no-browser"]
         vars, preargs, args = parse_config_arguments(self.config["syncthing_arguments"])
         cmdline = preargs + cmdline + args
         if self.home_dir_override:
-            cmdline += ["-home", self.home_dir_override]
+            cmdline += ["--home", self.home_dir_override]
 
         self.process = DaemonProcess(cmdline, self.config["daemon_priority"], self.config["max_cpus"], env=vars)
         self.process.connect("failed", self.cb_daemon_startup_failed)
-- 
2.49.1

