From 747adc56b782c99541969ddaadf723af666a9d84 Mon Sep 17 00:00:00 2001
From: Joey Riches <josephriches@gmail.com>
Date: Thu, 22 May 2025 17:39:15 +0100
Subject: [PATCH 1/1] appstream: Pass `appstreamcli validate`

---
 data/brasero.appdata.xml.in | 25 +++++++++++++++++--------
 1 file changed, 17 insertions(+), 8 deletions(-)

diff --git a/data/brasero.appdata.xml.in b/data/brasero.appdata.xml.in
index e7caf98f..877d4ba9 100644
--- a/data/brasero.appdata.xml.in
+++ b/data/brasero.appdata.xml.in
@@ -1,7 +1,10 @@
 <?xml version="1.0" encoding="UTF-8"?>
-<application>
-  <id type="desktop">brasero.desktop</id>
-  <licence>CC0</licence>
+<component type="desktop-application">
+  <id>org.gnome.brasero</id>
+  <name>Brasero</name>
+  <metadata_license>CC0</metadata_license>
+  <project_license>GPL-2.0-or-later</project_license>
+  <_summary>Brasero is a CD/DVD mastering tool, it is designed to be simple and easy to use</_summary>
   <description>
     <_p>
       Brasero is a application to burn CD/DVD for the GNOME Desktop.
@@ -16,12 +19,18 @@
     </_p>
   </description>
   <screenshots>
-    <screenshot type="default">https://wiki.gnome.org/Apps/Brasero?action=AttachFile&amp;do=get&amp;target=copy-disc.png</screenshot>
-    <screenshot>https://wiki.gnome.org/Apps/Brasero?action=AttachFile&amp;do=get&amp;target=audio-disc.png</screenshot>
-    <screenshot>https://wiki.gnome.org/Apps/Brasero?action=AttachFile&amp;do=get&amp;target=data-project.png</screenshot>
+    <screenshot type="default">
+      <caption>Image of the main window of brasero when the software is first started</caption>
+      <image type="source">https://help.gnome.org/users/brasero/stable/figures/brasero-main-window.png</image>
+    </screenshot>
   </screenshots>
+  <launchable type="desktop-id">brasero.desktop</launchable>
   <url type="homepage">https://wiki.gnome.org/Apps/Brasero</url>
-  <updatecontact>brasero-list@gnome.org</updatecontact>
+  <update_contact>brasero-list@gnome.org</update_contact>
   <project_group>GNOME</project_group>
+  <developer id="org.gnome">
+    <name>The GNOME Project</name>
+  </developer>
   <translation type="gettext">brasero</translation>
-</application>
+  <content_rating type="oars-1.0"/>
+</component>
-- 
2.49.0

