From d20d3f6046d9b6cf647d0f1c2f226fd5077cd98e Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Ball=C3=B3=20Gy=C3=B6rgy?= <ballogyor@gmail.com>
Date: Thu, 6 May 2021 15:52:34 +0000
Subject: [PATCH] Update metainfo file to follow the latest specification

---
 data/brasero.appdata.xml.in | 82 +++++++++++++++++++++++++++++++++----
 1 file changed, 74 insertions(+), 8 deletions(-)

diff --git a/data/brasero.appdata.xml.in b/data/brasero.appdata.xml.in
index e7caf98fe..ecc37df01 100644
--- a/data/brasero.appdata.xml.in
+++ b/data/brasero.appdata.xml.in
@@ -1,7 +1,10 @@
 <?xml version="1.0" encoding="UTF-8"?>
-<application>
-  <id type="desktop">brasero.desktop</id>
-  <licence>CC0</licence>
+<component type="desktop-application">
+  <id>org.gnome.Brasero</id>
+  <metadata_license>CC0-1.0</metadata_license>
+  <project_license>GPL-2.0-or-later</project_license>
+  <_name>Brasero</_name>
+  <_summary>Create and copy CDs and DVDs</_summary>
   <description>
     <_p>
       Brasero is a application to burn CD/DVD for the GNOME Desktop.
@@ -16,12 +19,75 @@
     </_p>
   </description>
   <screenshots>
-    <screenshot type="default">https://wiki.gnome.org/Apps/Brasero?action=AttachFile&amp;do=get&amp;target=copy-disc.png</screenshot>
-    <screenshot>https://wiki.gnome.org/Apps/Brasero?action=AttachFile&amp;do=get&amp;target=audio-disc.png</screenshot>
-    <screenshot>https://wiki.gnome.org/Apps/Brasero?action=AttachFile&amp;do=get&amp;target=data-project.png</screenshot>
+    <screenshot type="default">
+      <image>https://wiki.gnome.org/Apps/Brasero?action=AttachFile&amp;do=get&amp;target=highlight3-figure.png</image>
+    </screenshot>
+    <screenshot>
+      <image>https://wiki.gnome.org/Apps/Brasero?action=AttachFile&amp;do=get&amp;target=copy-disc.png</image>
+    </screenshot>
+    <screenshot>
+      <image>https://wiki.gnome.org/Apps/Brasero?action=AttachFile&amp;do=get&amp;target=audio-disc.png</image>
+    </screenshot>
+    <screenshot>
+      <image>https://wiki.gnome.org/Apps/Brasero?action=AttachFile&amp;do=get&amp;target=data-project.png</image>
+    </screenshot>
   </screenshots>
   <url type="homepage">https://wiki.gnome.org/Apps/Brasero</url>
-  <updatecontact>brasero-list@gnome.org</updatecontact>
+  <url type="bugtracker">https://gitlab.gnome.org/GNOME/brasero/issues</url>
+  <url type="donation">https://www.gnome.org/donate/</url>
+  <url type="help">https://help.gnome.org/users/brasero/stable/</url>
+  <url type="translate">https://l10n.gnome.org/module/brasero/</url>
+  <url type="vcs-browser">https://gitlab.gnome.org/GNOME/brasero</url>
+  <launchable type="desktop-id">brasero.desktop</launchable>
+  <_developer_name>The Brasero developers</_developer_name>
   <project_group>GNOME</project_group>
   <translation type="gettext">brasero</translation>
-</application>
+  <releases>
+    <release type="stable" version="3.12.3" date="2021-09-20T00:00:00Z">
+      <description>
+        <p>This release includes the following changes:</p>
+        <ul>
+          <li>Blu-ray fixes (Bastien Nocera)</li>
+          <li>Support Tracker 3.0 (Carlos Garnacho)</li>
+          <li>Add CI (Bastien Nocera)</li>
+          <li>Transcode plugin gstreamer plugin improvement (Robert McQueen)</li>
+          <li>Fix crash with malformed URI (Geyslan)</li>
+          <li>Fix default project label (Pimm "de Chinchilla" Hogeling)</li>
+          <li>Man page fixes (Ville Skyttä)</li>
+          <li>Header guard fix (Ting-Wei Lan)</li>
+          <li>Help improvements (scootergrisen, Andre Klapper)</li>
+          <li>Translation updates</li>
+        </ul>
+      </description>
+    </release>
+    <release type="stable" version="3.12.2" date="2017-07-31T00:00:00Z">
+      <description>
+        <p>This release includes the following changes:</p>
+        <ul>
+          <li>Build fixes</li>
+          <li>Support Tracker 2.0</li>
+          <li>Translation updates</li>
+        </ul>
+      </description>
+    </release>
+    <release type="stable" version="3.12.1" date="2015-04-28T00:00:00Z">
+      <description>
+        <p>This release includes the following changes:</p>
+        <ul>
+          <li>Add a symbolic app icon</li>
+          <li>Translation updates</li>
+        </ul>
+      </description>
+    </release>
+    <release type="stable" version="3.12.0" date="2014-11-13T00:00:00Z">
+      <description>
+        <p>This release includes the following changes:</p>
+        <ul>
+          <li>Fix crash in plugin dialog</li>
+          <li>Translation updates</li>
+        </ul>
+      </description>
+    </release>
+  </releases>
+  <content_rating type="oars-1.1" />
+</component>
-- 
GitLab

