From e9b9b5a86b1a2824608d2db91c075cc44ff36554 Mon Sep 17 00:00:00 2001
From: Reilly Brogan <reilly@reillybrogan.com>
Date: Thu, 21 Sep 2023 16:26:51 -0500
Subject: [PATCH 3/3] BORE: Disable by default for testing

Signed-off-by: Reilly Brogan <reilly@reillybrogan.com>
---
 kernel/sched/fair.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/kernel/sched/fair.c b/kernel/sched/fair.c
index 88700ca2a47d..18e16c862df0 100644
--- a/kernel/sched/fair.c
+++ b/kernel/sched/fair.c
@@ -153,7 +153,7 @@ static unsigned int normalized_sysctl_sched_wakeup_granularity	= 1000000UL;
 const_debug unsigned int sysctl_sched_migration_cost	= 500000UL;
 
 #ifdef CONFIG_SCHED_BORE
-unsigned int __read_mostly sched_bore                  = 1;
+unsigned int __read_mostly sched_bore                  = 0;
 unsigned int __read_mostly sched_bore_extra_flags      = 0;
 unsigned int __read_mostly sched_burst_cache_lifetime  = 60000000;
 unsigned int __read_mostly sched_burst_penalty_offset  = 22;
-- 
2.42.0

