diff --git a/src/basedir.c b/src/basedir.c
index 1a2a8c4..dd6c2b9 100644
--- a/src/basedir.c
+++ b/src/basedir.c
@@ -160,6 +160,11 @@ static void xdgFreeData(xdgCachedData *cache)
 		free(cache->cacheHome);
 		cache->cacheHome = 0;
 	}
+	if (cache->runtimeDirectory)
+	{
+		free(cache->runtimeDirectory);
+		cache->runtimeDirectory = 0;
+	}
 	xdgFreeStringList(cache->searchableDataDirectories);
 	cache->searchableDataDirectories = 0;
 	xdgFreeStringList(cache->searchableConfigDirectories);
