From 56aea721e8cb0e16b20dedae3ff6e6d07a7b7c74 Mon Sep 17 00:00:00 2001
From: Thomas Staudinger <Staudi.Kaos@gmail.com>
Date: Thu, 27 Apr 2023 18:35:11 +0200
Subject: [PATCH] Use internal qm-dsp

---
 libs/qm-dsp/wscript | 29 ++---------------------------
 wscript             |  2 --
 2 files changed, 2 insertions(+), 29 deletions(-)

diff --git a/libs/qm-dsp/wscript b/libs/qm-dsp/wscript
index a59801bf36..a9973e8d96 100644
--- a/libs/qm-dsp/wscript
+++ b/libs/qm-dsp/wscript
@@ -24,35 +24,10 @@ def options(opt):
 
 def configure(conf):
     if conf.is_defined('USE_EXTERNAL_LIBS'):
-        conf.check_cxx(header_name="base/Pitch.h", mandatory=True, msg="Checking for qm-dsp headers")
-        conf.check_cxx(lib="qm-dsp", uselib_store="QMDSP", mandatory=True)
-
-        # Verify that qm-dsp is self-contained and that FFT transforms actually works
-        if not conf.check_cxx(msg = 'Checking qm-dsp without explicit kiss linking',
-                              features = 'cxx cxxprogram',
-                              fragment = """#include "dsp/transforms/FFT.h"\nint main(void) {\nFFT(4).process(false, nullptr, 0, nullptr, nullptr);\nFFTReal(4).forward(nullptr, nullptr, nullptr);\nreturn 0;\n}""",
-                              uselib = 'QMDSP',
-                              okmsg = 'ok',
-                              errmsg = "-lqm-dsp by itself failed",
-                              mandatory = False,
-                              ):
-            # The external qm-dsp might be built without embedded kiss - try to link with it explicitly and check again
-            conf.check_cxx(lib="kiss_fft_double", uselib_store="KISSFFT", mandatory=True)
-            conf.check_cxx(lib="kiss_fftr_double", uselib_store="KISSFFTR", mandatory=True)
-            conf.env.append_value('LIB_QMDSP', conf.env['LIB_KISSFFT'] + conf.env['LIB_KISSFFTR'])
-
-            conf.check_cxx(msg = 'Checking qm-dsp with explicit kiss linking',
-                           features = 'cxx cxxprogram',
-                           fragment = """#include "dsp/transforms/FFT.h"\nint main(void) {\nFFT(4).process(false, nullptr, 0, nullptr, nullptr);\nFFTReal(4).forward(nullptr, nullptr, nullptr);\nreturn 0;\n}""",
-                           uselib = 'QMDSP',
-                           okmsg = '-lqm-dsp needs explicit kiss linking',
-                           errmsg = "qm-dsp doesn't work, even with kiss libraries",
-                           mandatory = True,
-                           )
+        conf.load('compiler_cxx')
+        autowaf.configure(conf)
 
 def build(bld):
-    if bld.is_defined('USE_EXTERNAL_LIBS'):
-        return
 
     # Host Library
     obj = bld(features = 'c cxx cxxstlib')
diff --git a/wscript b/wscript
index c0a8c80dc3..b461c49c9d 100644
--- a/wscript
+++ b/wscript
@@ -1140,8 +1140,6 @@ def configure(conf):
 
     if Options.options.use_external_libs:
         conf.define('USE_EXTERNAL_LIBS', 1)
-        conf.env.append_value(
-            'CXXFLAGS', '-I' + Options.options.qm_dsp_include)
 
     if Options.options.boost_include != '':
         conf.env.append_value('CXXFLAGS', '-I' + Options.options.boost_include)
-- 
2.40.1

