From 4e207c63bcb66807bb8b4cf2954acc45ba262f9e Mon Sep 17 00:00:00 2001
From: cryptonaut <cryptonaut64@gmail.com>
Date: Wed, 2 Nov 2016 18:31:50 -0700
Subject: [PATCH 1/1] UI: Fix crash when switching encoders in advanced mode

---
 UI/window-basic-settings.cpp | 18 ++++++++++--------
 1 file changed, 10 insertions(+), 8 deletions(-)

diff --git a/UI/window-basic-settings.cpp b/UI/window-basic-settings.cpp
index 9d9444b..ed29180 100644
--- a/UI/window-basic-settings.cpp
+++ b/UI/window-basic-settings.cpp
@@ -1290,14 +1290,16 @@ OBSPropertiesView *OBSBasicSettings::CreateEncoderPropertyView(
 	obs_data_t *settings = obs_encoder_defaults(encoder);
 	OBSPropertiesView *view;
 
-	char encoderJsonPath[512];
-	int ret = GetProfilePath(encoderJsonPath, sizeof(encoderJsonPath),
-			path);
-	if (ret > 0) {
-		obs_data_t *data = obs_data_create_from_json_file_safe(
-				encoderJsonPath, "bak");
-		obs_data_apply(settings, data);
-		obs_data_release(data);
+	if (path) {
+		char encoderJsonPath[512];
+		int ret = GetProfilePath(encoderJsonPath, sizeof(encoderJsonPath),
+				path);
+		if (ret > 0) {
+			obs_data_t *data = obs_data_create_from_json_file_safe(
+					encoderJsonPath, "bak");
+			obs_data_apply(settings, data);
+			obs_data_release(data);
+		}
 	}
 
 	view = new OBSPropertiesView(settings, encoder,
-- 
2.10.2

