From aa20278cc9c2f4f32901875e80518c6d84da7601 Mon Sep 17 00:00:00 2001
From: Thomas Staudinger <Staudi.Kaos@gmail.com>
Date: Wed, 3 May 2023 23:19:28 +0200
Subject: [PATCH] Disable Werror

Signed-off-by: Thomas Staudinger <Staudi.Kaos@gmail.com>
---
 cmake/Modules/CompilerConfig.cmake | 1 -
 1 file changed, 1 deletion(-)

diff --git a/cmake/Modules/CompilerConfig.cmake b/cmake/Modules/CompilerConfig.cmake
index 41c56da8b..b8f8bf330 100644
--- a/cmake/Modules/CompilerConfig.cmake
+++ b/cmake/Modules/CompilerConfig.cmake
@@ -101,7 +101,6 @@ else()
 
   option(CALM_DEPRECATION "Keep deprecated-declarations as warnings" OFF)
   add_compile_options(
-    -Werror
     -Wextra
     -Wvla
     -Wswitch
-- 
2.40.1

