diff --git a/frontend/cmake/ui-qt.cmake b/frontend/cmake/ui-qt.cmake
index 4995cb5fa..d1a241a3e 100644
--- a/frontend/cmake/ui-qt.cmake
+++ b/frontend/cmake/ui-qt.cmake
@@ -1,7 +1,7 @@
 find_package(Qt6 REQUIRED Widgets Network Svg Xml)
 
 if(OS_LINUX OR OS_FREEBSD OR OS_OPENBSD)
-  find_package(Qt6 REQUIRED Gui DBus)
+  find_package(Qt6 REQUIRED Gui GuiPrivate DBus)
 endif()
 
 if(NOT TARGET OBS::qt-wrappers)
diff --git a/frontend/plugins/aja-output-ui/CMakeLists.txt b/frontend/plugins/aja-output-ui/CMakeLists.txt
index 37e4ba04d..1be2f06e9 100644
--- a/frontend/plugins/aja-output-ui/CMakeLists.txt
+++ b/frontend/plugins/aja-output-ui/CMakeLists.txt
@@ -10,7 +10,7 @@ find_package(LibAJANTV2 REQUIRED)
 find_package(Qt6 REQUIRED Widgets)
 
 if(OS_LINUX OR OS_FREEBSD OR OS_OPENBSD)
-  find_package(Qt6 REQUIRED Gui)
+  find_package(Qt6 REQUIRED Gui GuiPrivate)
 
   find_package(X11 REQUIRED)
 endif()
diff --git a/frontend/plugins/decklink-output-ui/CMakeLists.txt b/frontend/plugins/decklink-output-ui/CMakeLists.txt
index d72dac269..eeb5e7ab7 100644
--- a/frontend/plugins/decklink-output-ui/CMakeLists.txt
+++ b/frontend/plugins/decklink-output-ui/CMakeLists.txt
@@ -8,7 +8,7 @@ endif()
 find_package(Qt6 REQUIRED Widgets)
 
 if(OS_LINUX OR OS_FREEBSD OR OS_OPENBSD)
-  find_package(Qt6 REQUIRED Gui)
+  find_package(Qt6 REQUIRED Gui GuiPrivate)
 
   find_package(X11 REQUIRED)
 endif()
diff --git a/frontend/plugins/frontend-tools/CMakeLists.txt b/frontend/plugins/frontend-tools/CMakeLists.txt
index f7ffa4ae3..1e102bc13 100644
--- a/frontend/plugins/frontend-tools/CMakeLists.txt
+++ b/frontend/plugins/frontend-tools/CMakeLists.txt
@@ -3,7 +3,7 @@ cmake_minimum_required(VERSION 3.28...3.30)
 find_package(Qt6 REQUIRED Widgets)
 
 if(OS_LINUX OR OS_FREEBSD OR OS_OPENBSD)
-  find_package(Qt6 REQUIRED Gui)
+  find_package(Qt6 REQUIRED Gui GuiPrivate)
   find_package(X11 REQUIRED)
 endif()
 
