From bc177cd02b7d83d43a2624ac3ca5dd8b71d333f5 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Bla=C5=BE=20Me=C5=BEnar=C5=A1i=C4=8D?=
 <blazingeek@gmail.com>
Date: Mon, 10 Oct 2016 13:43:19 +0200
Subject: [PATCH] Removed exo from deps

---
 configure.ac | 5 -----
 1 file changed, 5 deletions(-)

diff --git a/configure.ac b/configure.ac
index df5bc0b..c26aba6 100644
--- a/configure.ac
+++ b/configure.ac
@@ -31,11 +31,6 @@ PKG_CHECK_MODULES([LIGHTDMGOBJECT], [liblightdm-gobject-1 >= 1.19.2],
 )
 PKG_CHECK_MODULES([LIBX11], [x11])
 
-AC_CHECK_PROG([HAVE_EXO_CSOURCE], [exo-csource], [yes])
-if test x"$HAVE_EXO_CSOURCE" != xyes; then
-  AC_MSG_FAILURE([exo-csource not installed])
-fi
-
 dnl ###########################################################################
 dnl Optional dependencies
 dnl ###########################################################################
-- 
2.10.1

