From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Evan Maddock <maddock.evan@vivaldi.net>
Date: Sat, 1 Nov 2025 21:39:06 -0400
Subject: [PATCH] build: Port to libpeas-2 and Budgie-2.0

Signed-off-by: Evan Maddock <maddock.evan@vivaldi.net>
---
 meson.build | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/meson.build b/meson.build
index 78ce63f..90ed770 100644
--- a/meson.build
+++ b/meson.build
@@ -70,7 +70,7 @@ xp = dependency('libxfce4panel-2.0', required: get_option('xfce'))
 xc = dependency('libxfconf-0', required: get_option('xfce'))
 xfce_found = xp.found() and xc.found()
 
-bp = dependency('budgie-1.0', required: get_option('budgie'))
+bp = dependency('budgie-2.0', required: get_option('budgie'))
 budgie_found = bp.found()
 
 
