From 646216e6325cf218b56f1f63d7099dfb20877072 Mon Sep 17 00:00:00 2001
From: Joshua Strobl <joshua@stroblindustries.com>
Date: Sun, 21 Apr 2019 19:22:57 +0300
Subject: [PATCH 1/1] Tweak CMake for bamf check

---
 CMakeLists.txt | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 6c49943..a2846b6 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -63,7 +63,7 @@ find_package (PkgConfig)
 find_package(GLIB2 2.50 REQUIRED COMPONENTS GOBJECT GIO GIO_UNIX GMODULE GTHREAD CODEGEN MKENUMS GENMARSHAL COMPILE_SCHEMAS COMPILE_RESOURCES)
 include(GLibProgramHandlers)
 find_package(GTK 3.22 REQUIRED COMPONENTS GDK GTK X11 WAYLAND)
-find_package(BAMF 0.5.0 REQUIRED COMPONENTS LIB DAEMON)
+find_package(BAMF 0.5.4 REQUIRED COMPONENTS LIB)
 if(${GLIB2_FOUND} AND ${GTK_FOUND})
     set(CORE_FOUND TRUE)
 endif()
-- 
2.21.0

