From 91ab7d538ce8b540a731271461e838cabc99b7cd Mon Sep 17 00:00:00 2001
From: Konstantin <ria.freelander@gmail.com>
Date: Fri, 12 Jan 2018 22:03:16 +0300
Subject: [PATCH 1/1] MATE: Fix expand. Fixes #179

---
 lib/mate-plugin-appmenu.vala | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/lib/mate-plugin-appmenu.vala b/lib/mate-plugin-appmenu.vala
index 783c121..ef13a28 100644
--- a/lib/mate-plugin-appmenu.vala
+++ b/lib/mate-plugin-appmenu.vala
@@ -25,7 +25,7 @@ private bool factory_callback(MatePanel.Applet applet, string iid)
         return false;
     }
 
-    applet.flags = MatePanel.AppletFlags.HAS_HANDLE | MatePanel.AppletFlags.EXPAND_MINOR;
+    applet.flags = MatePanel.AppletFlags.HAS_HANDLE | MatePanel.AppletFlags.EXPAND_MAJOR;
 
     var layout = new Appmenu.AppMenuBar();
     applet.add(layout);
-- 
2.16.1

