From f1b848ae69128eb82db4a9b180b4e88acd1d4824 Mon Sep 17 00:00:00 2001
From: Sam Lane <samlane00@gmail.com>
Date: Tue, 22 Feb 2022 18:48:59 +0000
Subject: [PATCH] Add support for Budgie Control Center

---
 lib/helper-desktop.vala | 8 ++++++++
 1 file changed, 8 insertions(+)

diff --git a/lib/helper-desktop.vala b/lib/helper-desktop.vala
index 694042d..96b8a8a 100644
--- a/lib/helper-desktop.vala
+++ b/lib/helper-desktop.vala
@@ -134,6 +134,14 @@ namespace Appmenu
                     case "LXDE":
                         info = new DesktopAppInfo("lxappearance.desktop");
                         break;
+                    case "Budgie:GNOME":
+                        string control_center = "gnome-control-center";
+                        if (Environment.find_program_in_path("budgie-control-center") != null) {
+                            control_center = "budgie-control-center";
+                        }
+                        info = AppInfo.create_from_commandline(control_center,null,
+                        AppInfoCreateFlags.SUPPORTS_STARTUP_NOTIFICATION) as DesktopAppInfo;
+                        break;
                     default:
                         warning("Unknown desktop environment\n");
                         info = AppInfo.create_from_commandline("gnome-control-center",null,
-- 
GitLab

