From 5d39a18febbc365639f8c80e619facf57fef3e21 Mon Sep 17 00:00:00 2001
From: Konstantin <ria.freelander@gmail.com>
Date: Thu, 2 Jan 2025 16:31:23 +0300
Subject: [PATCH] xfce: expand now use plugin prop instead of widget one

---
 applets/xfce4-plugin-appmenu.vala | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/applets/xfce4-plugin-appmenu.vala b/applets/xfce4-plugin-appmenu.vala
index 24b4fc3..3005461 100644
--- a/applets/xfce4-plugin-appmenu.vala
+++ b/applets/xfce4-plugin-appmenu.vala
@@ -37,7 +37,7 @@ public class AppmenuPlugin : Xfce.PanelPlugin {
             channel = this.get_channel();
             Xfconf.Property.bind(channel,this.get_property_base()+"/"+Key.COMPACT_MODE,typeof(bool),widget,Key.COMPACT_MODE);
             Xfconf.Property.bind(channel,this.get_property_base()+"/"+Key.BOLD_APPLICATION_NAME,typeof(bool),widget,Key.BOLD_APPLICATION_NAME);
-            Xfconf.Property.bind(channel,this.get_property_base()+"/"+"expand",typeof(bool),widget,"hexpand");
+            Xfconf.Property.bind(channel,this.get_property_base()+"/"+"expand",typeof(bool),this,"expand");
             this.menu_show_configure();
         } catch (Xfconf.Error e) {
             stderr.printf("Xfconf init failed. Configuration will not be saved.\n");
@@ -61,7 +61,7 @@ public class AppmenuPlugin : Xfce.PanelPlugin {
         dlg_vbox.pack_start(entry,false,false,2);
         entry.show();
         entry = new CheckButton.with_label(_("Expand plugin on panel"));
-        widget.bind_property("hexpand", entry, "active", BindingFlags.BIDIRECTIONAL | BindingFlags.SYNC_CREATE);
+        this.bind_property("expand", entry, "active", BindingFlags.BIDIRECTIONAL | BindingFlags.SYNC_CREATE);
         dlg_vbox.pack_start(entry,false,false,2);
         entry.show();
         dlg.show();
-- 
GitLab

