From cc97d45a93a0c87521e1f34348df2f38e81e1a1a Mon Sep 17 00:00:00 2001
From: Ikey Doherty <ikey@solus-project.com>
Date: Sun, 8 Nov 2015 16:41:25 +0000
Subject: [PATCH] Fix linking of web_extension with gnome-desktop-3.0

Signed-off-by: Ikey Doherty <ikey@solus-project.com>
---
 configure.ac | 1 +
 1 file changed, 1 insertion(+)

diff --git a/configure.ac b/configure.ac
index c73b56f..74ccfc0 100644
--- a/configure.ac
+++ b/configure.ac
@@ -121,6 +121,7 @@ PKG_CHECK_MODULES([DEPENDENCIES], [
 PKG_CHECK_MODULES([WEB_EXTENSION], [
                   webkit2gtk-web-extension-4.0 >= $WEBKIT_GTK_REQUIRED
                   libsecret-1 >= $LIBSECRET_REQUIRED
+                  gnome-desktop-3.0 >= $GNOME_DESKTOP_REQUIRED
                   ])
 AC_SUBST([WEB_EXTENSION_CFLAGS])
 AC_SUBST([WEB_EXTENSION_LIBS])
-- 
2.6.3

