Index: b/SConstruct
===================================================================
--- a/SConstruct	2016-07-28 13:52:42.000000000 -0700
+++ b/SConstruct	2016-07-30 16:11:04.989334472 -0700
@@ -19,8 +19,8 @@
   BoolVariable('FRAMESKIP', 'Enable frameskipping', 1),
   BoolVariable('OPENGL',    'Enable OpenGL support', 1),
   BoolVariable('LUA',       'Enable Lua support', 1),
-  BoolVariable('GTK', 'Enable GTK2 GUI (SDL only)', 1),
-  BoolVariable('GTK3', 'Enable GTK3 GUI (SDL only)', 0),
+  BoolVariable('GTK', 'Enable GTK2 GUI (SDL only)', 0),
+  BoolVariable('GTK3', 'Enable GTK3 GUI (SDL only)', 1),
   BoolVariable('NEWPPU',    'Enable new PPU core', 1),
   BoolVariable('CREATE_AVI', 'Enable avi creation support (SDL only)', 1),
   BoolVariable('LOGO', 'Enable a logoscreen when creating avis (SDL only)', 1),
