From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Evan Maddock <maddock.evan@vivaldi.net>
Date: Thu, 9 Jul 2026 19:18:23 -0400
Subject: [PATCH] fonts: Fix paths for fontconfig directories

Signed-off-by: Evan Maddock <maddock.evan@vivaldi.net>
---
 src/handlers/xdg/fonts.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/handlers/xdg/fonts.c b/src/handlers/xdg/fonts.c
index e4d2ccd..195a9e8 100644
--- a/src/handlers/xdg/fonts.c
+++ b/src/handlers/xdg/fonts.c
@@ -18,8 +18,8 @@
 static const char *font_paths[] = {
         "/usr/share/fonts/*/*",
         "/usr/share/fonts/*",
-        "/usr/share/fontconfig/conf.avail/*",
-        "/usr/share/fontconfig/conf.default/*",
+        "/usr/share/fontconfig/conf.avail",
+        "/usr/share/fontconfig/conf.default",
 };
 
 /**
