From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Reilly Brogan <reilly@reillybrogan.com>
Date: Sun, 7 Jul 2024 03:34:40 -0500
Subject: [PATCH] Update LDM for new Nvidia location

Signed-off-by: Reilly Brogan <reilly@reillybrogan.com>
---
 src/handlers/linux-driver-management.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/handlers/linux-driver-management.c b/src/handlers/linux-driver-management.c
index 291a977..fc5ef58 100644
--- a/src/handlers/linux-driver-management.c
+++ b/src/handlers/linux-driver-management.c
@@ -21,7 +21,7 @@
 static const char *driver_paths[] = {
         "/usr/lib/glx-provider",   "/usr/lib/glx-provider/*",       "/usr/lib/nvidia",
         "/usr/lib32/glx-provider", "/usr/lib32/glx-provider/*",     "/usr/lib32/nvidia",
-        "/usr/lib/nvidia/modules", "/usr/lib/xorg/modules/drivers", "/usr/share/glvnd/egl_vendor.d",
+        "/usr/lib64/nvidia/modules", "/usr/lib/xorg/modules/drivers", "/usr/share/glvnd/egl_vendor.d",
 };
 
 /**
