From 230a1d4086f23dcefc0c4591e408bb9da2537024 Mon Sep 17 00:00:00 2001
From: Ikey Doherty <ikey@solus-project.com>
Date: Wed, 29 Nov 2017 14:54:01 +0000
Subject: [PATCH 1/5] handlers/cbm: Only support CBM invocation from non chroot

Signed-off-by: Ikey Doherty <ikey@solus-project.com>
---
 src/handlers/kernel/clr-boot-manager.c | 9 +++++++++
 1 file changed, 9 insertions(+)

diff --git a/src/handlers/kernel/clr-boot-manager.c b/src/handlers/kernel/clr-boot-manager.c
index d0341e8..f666f52 100644
--- a/src/handlers/kernel/clr-boot-manager.c
+++ b/src/handlers/kernel/clr-boot-manager.c
@@ -45,7 +45,16 @@ static UscHandlerStatus usc_handler_cbm_exec(UscContext *ctx, const char *path)
                 return USC_HANDLER_SKIP;
         }
 
+        /* Only support this in non chroot environments. Users will need to
+         * perform CBM recovery in chroots using CBM directly to ensure that
+         * CBM is never accidentally invoked
+         */
         usc_context_emit_task_start(ctx, "Updating clr-boot-manager");
+        if (usc_context_has_flag(ctx, USC_FLAGS_CHROOTED)) {
+                usc_context_emit_task_finish(ctx, USC_HANDLER_SKIP);
+                return USC_HANDLER_SKIP | USC_HANDLER_BREAK;
+        }
+
         int ret = usc_exec_command(command);
         if (ret != 0) {
                 usc_context_emit_task_finish(ctx, USC_HANDLER_FAIL);
-- 
2.15.1

