From 34c64c848268e429bfb56903148198cd03305210 Mon Sep 17 00:00:00 2001
From: Reilly Brogan <reilly@reillybrogan.com>
Date: Tue, 5 Dec 2023 22:47:08 -0600
Subject: [PATCH] Recurse further looking for kernel modules

---
 src/handlers/kernel/depmod.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/handlers/kernel/depmod.c b/src/handlers/kernel/depmod.c
index d831c4c..e5334da 100644
--- a/src/handlers/kernel/depmod.c
+++ b/src/handlers/kernel/depmod.c
@@ -22,7 +22,7 @@ static const char *module_paths[] = {
         /* Glob all module directories and track individually */
         KERNEL_MODULES_DIR "/*",
         KERNEL_MODULES_DIR "/*/*",                /* ./extra/ */
-        KERNEL_MODULES_DIR "/*/kernel/drivers/*", /* i.e. nvidia in video dir */
+        KERNEL_MODULES_DIR "/*/kernel/drivers/*/*", /* i.e. nvidia in video dir */
 };
 
 /**
-- 
2.43.0

