From 38195ee138ddb57d53626f740bc27659be45b2eb Mon Sep 17 00:00:00 2001
From: Ikey Doherty <ikey@solus-project.com>
Date: Sat, 16 Dec 2017 21:49:32 +0000
Subject: [PATCH] state: Stop spamming me pls.

Signed-off-by: Ikey Doherty <ikey@solus-project.com>
---
 src/state.c | 1 -
 1 file changed, 1 deletion(-)

diff --git a/src/state.c b/src/state.c
index 5185d48..2aabd9c 100644
--- a/src/state.c
+++ b/src/state.c
@@ -245,7 +245,6 @@ bool usc_state_tracker_load(UscStateTracker *self)
 
                 /* Drop old cache entries */
                 if (!usc_file_exists(snd)) {
-                        fprintf(stderr, "Dropping old cache item '%s'\n", snd);
                         errno = 0;
                         goto next;
                 }
-- 
2.15.1

