From 04ad4a136155405a12a0dedae8ce6e504fc270f3 Mon Sep 17 00:00:00 2001
From: Ikey Doherty <ikey@solus-project.com>
Date: Sun, 3 Dec 2017 11:32:48 +0000
Subject: [PATCH 2/2] handlers: Catch all qol-assist paths

Signed-off-by: Ikey Doherty <ikey@solus-project.com>
---
 src/handlers/qol-assist.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/src/handlers/qol-assist.c b/src/handlers/qol-assist.c
index 62ebce3..f71de1c 100644
--- a/src/handlers/qol-assist.c
+++ b/src/handlers/qol-assist.c
@@ -24,6 +24,7 @@
  */
 static const char *qol_paths[] = {
         "/usr/sbin/qol-assist",
+        "/usr/lib/systemd/system/qol-assist-migration.service",
 };
 
 /**
-- 
2.15.1

