From 9f26bd25f9af5155e81c1185eb0b8419d61d881e Mon Sep 17 00:00:00 2001
From: Ikey Doherty <ikey@solus-project.com>
Date: Wed, 18 Apr 2018 20:56:55 +0100
Subject: [PATCH 2/2] Ensure we mark the operation as finished

Signed-off-by: Ikey Doherty <ikey@solus-project.com>
---
 src/handlers/kernel/apparmor.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/src/handlers/kernel/apparmor.c b/src/handlers/kernel/apparmor.c
index 0118127..487e6f2 100644
--- a/src/handlers/kernel/apparmor.c
+++ b/src/handlers/kernel/apparmor.c
@@ -59,6 +59,7 @@ static UscHandlerStatus usc_handler_apparmor_exec(UscContext *ctx, __usc_unused_
                 usc_context_emit_task_finish(ctx, USC_HANDLER_FAIL);
                 return USC_HANDLER_FAIL;
         }
+        usc_context_emit_task_finish(ctx, USC_HANDLER_SUCCESS);
 
         /* Only run once */
         return USC_HANDLER_SUCCESS | USC_HANDLER_BREAK;
-- 
2.16.3

