From 4706c2ec2d60746556bc5a2d91d0cce00cc0a4ea Mon Sep 17 00:00:00 2001
From: Ikey Doherty <ikey@solus-project.com>
Date: Sun, 21 Jan 2018 22:57:32 +0000
Subject: [PATCH 3/3] Ensure we make nvidia modules known to usysconf

Signed-off-by: Ikey Doherty <ikey@solus-project.com>
---
 src/handlers/linux-driver-management.c | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/src/handlers/linux-driver-management.c b/src/handlers/linux-driver-management.c
index d3a6c6c..4d109c2 100644
--- a/src/handlers/linux-driver-management.c
+++ b/src/handlers/linux-driver-management.c
@@ -24,9 +24,9 @@
  * Paths that trigger an LDM update
  */
 static const char *driver_paths[] = {
-        "/usr/lib/glx-provider",         "/usr/lib/glx-provider/*",       "/usr/lib/nvidia",
-        "/usr/lib32/glx-provider",       "/usr/lib32/glx-provider/*",     "/usr/lib32/nvidia",
-        "/usr/lib/xorg/modules/drivers", "/usr/share/glvnd/egl_vendor.d",
+        "/usr/lib/glx-provider",   "/usr/lib/glx-provider/*",       "/usr/lib/nvidia",
+        "/usr/lib32/glx-provider", "/usr/lib32/glx-provider/*",     "/usr/lib32/nvidia",
+        "/usr/lib/nvidia/modules", "/usr/lib/xorg/modules/drivers", "/usr/share/glvnd/egl_vendor.d",
 };
 
 /**
-- 
2.15.1

