From 6ccee4412fb981fa1a0090a6587beffb3ba84d46 Mon Sep 17 00:00:00 2001
From: Ikey Doherty <ikey@solus-project.com>
Date: Wed, 29 Nov 2017 19:09:34 +0000
Subject: [PATCH 3/6] handlers: Handle ALL LDM cases

Signed-off-by: Ikey Doherty <ikey@solus-project.com>
---
 src/handlers/linux-driver-management.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/src/handlers/linux-driver-management.c b/src/handlers/linux-driver-management.c
index 677e36c..ab4fd8b 100644
--- a/src/handlers/linux-driver-management.c
+++ b/src/handlers/linux-driver-management.c
@@ -25,7 +25,9 @@
  */
 static const char *driver_paths[] = {
         "/usr/lib/glx-provider",
+        "/usr/lib/glx-provider/*",
         "/usr/lib32/glx-provider",
+        "/usr/lib32/glx-provider/*",
 };
 
 /**
-- 
2.15.1

