From 0996df9c43fc7ae5b806721667702d27a3a4bb4c Mon Sep 17 00:00:00 2001
From: Ikey Doherty <ikey@solus-project.com>
Date: Wed, 29 Nov 2017 22:06:45 +0000
Subject: [PATCH 5/6] Absolutely require that /proc is mounted before running

Signed-off-by: Ikey Doherty <ikey@solus-project.com>
---
 src/context.c |  5 +++++
 src/files.c   | 27 +++++++++++++++++++++++++++
 src/files.h   |  9 +++++++++
 3 files changed, 41 insertions(+)

diff --git a/src/context.c b/src/context.c
index c4b80e3..6f24b06 100644
--- a/src/context.c
+++ b/src/context.c
@@ -393,6 +393,11 @@ bool usc_context_run_triggers(UscContext *context, const char *name, bool force_
         bool ran_trigger = false;
         bool did_sync = false;
 
+        if (!usc_is_proc_mounted()) {
+                fputs("Refusing to run without a mounted /proc filesystem\n", stderr);
+                return false;
+        }
+
         tracker = usc_state_tracker_new();
         if (!tracker) {
                 fputs("Cannot continue without valid UscStateTracker\n", stderr);
diff --git a/src/files.c b/src/files.c
index 4369613..2cbee51 100644
--- a/src/files.c
+++ b/src/files.c
@@ -11,7 +11,10 @@
 
 #define _GNU_SOURCE
 
+#include <limits.h>
+#include <mntent.h>
 #include <stdio.h>
+#include <string.h>
 #include <sys/stat.h>
 
 #include "files.h"
@@ -40,6 +43,30 @@ bool usc_is_chrooted()
         return false;
 }
 
+bool usc_is_proc_mounted()
+{
+        struct mntent *ent = NULL;
+        struct mntent mnt = { 0 };
+        FILE *mnts = NULL;
+        char buf[PATH_MAX];
+        bool ret = false;
+
+        mnts = setmntent("/proc/self/mounts", "r");
+        if (!mnts) {
+                return false;
+        }
+
+        while ((ent = getmntent_r(mnts, &mnt, buf, sizeof(buf)))) {
+                if (mnt.mnt_dir && strcmp("/proc", mnt.mnt_dir) == 0) {
+                        ret = true;
+                        break;
+                }
+        }
+
+        endmntent(mnts);
+        return ret;
+}
+
 bool usc_file_mtime(const char *path, time_t *time)
 {
         struct stat st = { 0 };
diff --git a/src/files.h b/src/files.h
index a7e3a31..984da12 100644
--- a/src/files.h
+++ b/src/files.h
@@ -22,6 +22,15 @@
  */
 bool usc_is_chrooted(void);
 
+/**
+ * Sanity test, determine if the /proc filesystem is mounted and available.
+ * To be useful and function correctly, usysconf absolutely requires that
+ * the procfs is mounted.
+ *
+ * @returns True if proc was successfully detected as mounted.
+ */
+bool usc_is_proc_mounted(void);
+
 /**
  * Grab the mtime for a given path
  *
-- 
2.15.1

