From 56b93df3fce3178edc2ddec08bd58066cc757cdd Mon Sep 17 00:00:00 2001
From: Ikey Doherty <ikey@solus-project.com>
Date: Wed, 29 Nov 2017 22:21:36 +0000
Subject: [PATCH 6/6] handlers/qol: Only run QoL migration for real systems

Signed-off-by: Ikey Doherty <ikey@solus-project.com>
---
 src/handlers/qol-assist.c | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/src/handlers/qol-assist.c b/src/handlers/qol-assist.c
index 5db51be..62ebce3 100644
--- a/src/handlers/qol-assist.c
+++ b/src/handlers/qol-assist.c
@@ -48,7 +48,12 @@ static UscHandlerStatus usc_handler_qol_assist_exec(UscContext *ctx, const char
                 return USC_HANDLER_SKIP;
         }
 
+        /* QoL migrations only make sense for real live systems */
         usc_context_emit_task_start(ctx, "Registering QoL migration on next boot");
+        if (usc_context_has_flag(ctx, USC_FLAGS_CHROOTED)) {
+                usc_context_emit_task_finish(ctx, USC_HANDLER_SKIP);
+                return USC_HANDLER_SKIP | USC_HANDLER_BREAK;
+        }
 
         int ret = usc_exec_command(command);
         if (ret != 0) {
-- 
2.15.1

