From 2641207c9d9b21d7724f5f4601f710d64ae46359 Mon Sep 17 00:00:00 2001
From: Joey Riches <josephriches@gmail.com>
Date: Fri, 22 Mar 2024 20:36:01 +0000
Subject: [PATCH] handlers/kernel: Trigger clr-boot-manager on shim changes

---
 src/handlers/kernel/clr-boot-manager.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/src/handlers/kernel/clr-boot-manager.c b/src/handlers/kernel/clr-boot-manager.c
index 19ee09c..b9a539e 100644
--- a/src/handlers/kernel/clr-boot-manager.c
+++ b/src/handlers/kernel/clr-boot-manager.c
@@ -23,6 +23,7 @@ static const char *boot_paths[] = {
         KERNEL_DIR,
         "/usr/lib/goofiboot",
         "/usr/lib/systemd/boot/efi",
+        "/usr/lib/shim/",
 };
 
 /**
