--- libmypaint-1.3.0-orig/gegl/mypaint-gegl-surface.c   2019-06-18 17:03:46.025060319 +0800
+++ libmypaint-1.3.0/gegl/mypaint-gegl-surface.c        2019-06-18 17:05:08.693421652 +0800
@@ -78,7 +78,7 @@

     if (buffer_is_native(self)) {
         GeglBufferIterator *iterator = gegl_buffer_iterator_new(self->buffer, &tile_bbox, 0, self->format,
-                                      read_write_flags, GEGL_ABYSS_NONE);
+                                      read_write_flags, GEGL_ABYSS_NONE,100);

         // Read out
         gboolean completed = gegl_buffer_iterator_next(iterator);
@@ -88,7 +88,7 @@
             g_critical("Unable to get tile aligned access to GeglBuffer");
             request->buffer = NULL;
         } else {
-            request->buffer = (uint16_t *)(iterator->data[0]);
+            request->buffer = (uint16_t *)(iterator->items[0].data);
         }

         // So we can finish the iterator in tile_request_end()

