From d29a2231c59e43cd98ff0372f01b67ba48f26a52 Mon Sep 17 00:00:00 2001
From: Ikey Doherty <ikey@solus-project.com>
Date: Wed, 21 Feb 2018 14:59:18 +0000
Subject: [PATCH] Link with proper systemd

Signed-off-by: Ikey Doherty <ikey@solus-project.com>
---
 libmultipath/Makefile | 2 +-
 multipathd/Makefile   | 3 +--
 2 files changed, 2 insertions(+), 3 deletions(-)

diff --git a/libmultipath/Makefile b/libmultipath/Makefile
index 6364364..b3f87fe 100644
--- a/libmultipath/Makefile
+++ b/libmultipath/Makefile
@@ -9,7 +9,7 @@ DEVLIB = libmultipath.so
 LIBS = $(DEVLIB).$(SONAME)
 LIBDEPS = -lpthread -ldl -ldevmapper -ludev
 ifdef SYSTEMD
-	LIBDEPS += -lsystemd-daemon
+	LIBDEPS += -lsystemd
 endif
 
 OBJS = memory.o parser.o vector.o devmapper.o callout.o \
diff --git a/multipathd/Makefile b/multipathd/Makefile
index 781122a..9ea8fa6 100644
--- a/multipathd/Makefile
+++ b/multipathd/Makefile
@@ -11,7 +11,7 @@ ifdef SYSTEMD
 endif
 LDFLAGS += -lpthread -ldevmapper -lreadline
 ifdef SYSTEMD
-	LDFLAGS += -lsystemd-daemon
+	LDFLAGS += -lsystemd
 endif
 LDFLAGS += -ludev -ldl \
 	-L$(multipathdir) -lmultipath -L$(mpathpersistdir) -lmpathpersist
@@ -59,4 +59,3 @@ uninstall:
 
 clean:
 	rm -f core *.o $(EXEC) *.gz
-
-- 
2.16.1

