--- a/cmake/pcl_utils.cmake
+++ b/cmake/pcl_utils.cmake
@@ -107,7 +107,7 @@ macro(SET_INSTALL_DIRS)
     if(WIN32 AND NOT MINGW)
         set(PCLCONFIG_INSTALL_DIR "cmake")
       else()
-        set(PCLCONFIG_INSTALL_DIR "share/${PROJECT_NAME_LOWER}-${PCL_VERSION_MAJOR}.${PCL_VERSION_MINOR}")
+        set(PCLCONFIG_INSTALL_DIR "${LIB_INSTALL_DIR}/cmake/${PROJECT_NAME_LOWER}")
       endif()
 endmacro()
 
--- a/PCLConfig.cmake.in
+++ b/PCLConfig.cmake.in
@@ -388,8 +388,8 @@ if(WIN32 AND NOT MINGW)
     set(PCL_ALL_IN_ONE_INSTALLER ON)
   endif()
 else()
-# PCLConfig.cmake is installed to PCL_ROOT/share/pcl-x.y
-  get_filename_component(PCL_ROOT "${CMAKE_CURRENT_LIST_DIR}/../.." ABSOLUTE)
+# PCLConfig.cmake is installed to PCL_ROOT/lib/cmake/pcl
+  get_filename_component(PCL_ROOT "${CMAKE_CURRENT_LIST_DIR}/../../.." ABSOLUTE)
 endif()
 
 # check whether PCLConfig.cmake is found into a PCL installation or in a build tree
