From c73b40de11f376f4515a0edfe97fb3721a264b35 Mon Sep 17 00:00:00 2001
From: Jerry Casiano <JerryCasiano@gmail.com>
Date: Wed, 7 Aug 2019 20:56:54 -0400
Subject: [PATCH] - Fix build failure with newer Vala versions

---
 src/font-manager/FontList.vala | 6 ++++--
 1 files changed, 4 insertions(+), 2 deletions(-)

diff --git a/src/font-manager/FontList.vala b/src/font-manager/FontList.vala
index 78a47f0..fe492bf 100644
--- a/src/font-manager/FontList.vala
+++ b/src/font-manager/FontList.vala
@@ -104,8 +104,10 @@ namespace FontManager {
         string default_sample;
         string local_sample;
 
-        public BaseFontList () {
-            Object(name: "BaseFontList", headers_visible: false, expand: true);
+        construct {
+            name = "BaseFontList";
+            headers_visible = false;
+            expand = true;
             toggle = new Gtk.CellRendererToggle();
             var text = new Gtk.CellRendererText();
             var count = new CellRendererStyleCount();
