From 9a062d61d1ef897beb0d93e2e05fb4344b26ee6e Mon Sep 17 00:00:00 2001
From: Paolo Pisati <paolo.pisati@canonical.com>
Date: Tue, 27 Oct 2020 09:45:23 +0000
Subject: [PATCH 15/16] UBUNTU: SAUCE: apparmor: rename kzfree() to
 kfree_sensitive()

Signed-off-by: Paolo Pisati <paolo.pisati@canonical.com>
---
 security/apparmor/policy.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/security/apparmor/policy.c b/security/apparmor/policy.c
index a00e39b5c7b2..f03726d962cc 100644
--- a/security/apparmor/policy.c
+++ b/security/apparmor/policy.c
@@ -222,7 +222,7 @@ void aa_free_profile(struct aa_profile *profile)
 	aa_free_file_rules(&profile->file);
 	aa_free_cap_rules(&profile->caps);
 	aa_free_rlimit_rules(&profile->rlimits);
-	kzfree(profile->net_compat);
+	kfree_sensitive(profile->net_compat);
 
 	for (i = 0; i < profile->xattr_count; i++)
 		kfree_sensitive(profile->xattrs[i]);
-- 
2.32.0

