From 13246d3c619c61c1f1aac172e3b74482ddb8fe49 Mon Sep 17 00:00:00 2001
From: Paolo Pisati <paolo.pisati@canonical.com>
Date: Tue, 27 Oct 2020 09:45:23 +0000
Subject: [PATCH 31/36] UBUNTU: SAUCE: apparmor: rename kzfree() to
 kfree_sensitive()

Signed-off-by: Paolo Pisati <paolo.pisati@canonical.com>
---
 security/apparmor/policy.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/security/apparmor/policy.c b/security/apparmor/policy.c
index 0ba9fed391d4..2c97d076ad5c 100644
--- a/security/apparmor/policy.c
+++ b/security/apparmor/policy.c
@@ -257,7 +257,7 @@ void aa_free_profile(struct aa_profile *profile)
 	kfree_sensitive(profile->rename);
 
 	free_attachment(&profile->attach);
-	kzfree(profile->net_compat);
+	kfree_sensitive(profile->net_compat);
 
 	/*
 	 * at this point there are no tasks that can have a reference
-- 
2.40.1

