From a80f5ad1ad5b0185d7503b0bbe62167b0eea936e Mon Sep 17 00:00:00 2001
From: Paolo Pisati <paolo.pisati@canonical.com>
Date: Tue, 27 Oct 2020 11:04:12 +0000
Subject: [PATCH 33/36] UBUNTU: SAUCE: LSM: Use lsmblob in smk_netlbl_mls()

Signed-off-by: Paolo Pisati <paolo.pisati@canonical.com>
---
 security/smack/smack_access.c | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/security/smack/smack_access.c b/security/smack/smack_access.c
index 585e5e35710b..70700f2320a9 100644
--- a/security/smack/smack_access.c
+++ b/security/smack/smack_access.c
@@ -520,8 +520,11 @@ int smk_netlbl_mls(int level, char *catset, struct netlbl_lsm_secattr *sap,
 int smack_populate_secattr(struct smack_known *skp)
 {
 	int slen;
+	struct lsmblob *blob;
 
-	skp->smk_netlabel.attr.secid = skp->smk_secid;
+	blob = &skp->smk_netlabel.attr.lsmblob;
+	lsmblob_init(blob, 0);
+	blob->secid[smack_lsmid.slot] = skp->smk_secid;
 	skp->smk_netlabel.domain = skp->smk_known;
 	skp->smk_netlabel.cache = netlbl_secattr_cache_alloc(GFP_ATOMIC);
 	if (skp->smk_netlabel.cache != NULL) {
-- 
2.40.1

