From b476947c11025b18ac72c03434e7522aa5e2ac8d Mon Sep 17 00:00:00 2001
From: Andrea Righi <andrea.righi@canonical.com>
Date: Wed, 19 Jul 2023 19:16:17 +0200
Subject: [PATCH 57/57] UBUNTU: SAUCE: apparmor3.2.0: fix const build warning

Signed-off-by: Andrea Righi <andrea.righi@canonical.com>
Signed-off-by: John Johansen <john.johansen@canonical.com>
---
 security/apparmor/net.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/security/apparmor/net.c b/security/apparmor/net.c
index 9d6921ad88f4..6552dcfd4af5 100644
--- a/security/apparmor/net.c
+++ b/security/apparmor/net.c
@@ -97,7 +97,7 @@ static void audit_unix_addr(struct audit_buffer *ab, const char *str,
 static void audit_unix_sk_addr(struct audit_buffer *ab, const char *str,
 			       const struct sock *sk)
 {
-	struct unix_sock *u = unix_sk(sk);
+	const struct unix_sock *u = unix_sk(sk);
 
 	if (u && u->addr)
 		audit_unix_addr(ab, str, u->addr->name, u->addr->len);
-- 
2.41.0

