--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -135,5 +135,5 @@ configure_file(
   ${CMAKE_BINARY_DIR}/sz.pc
   @ONLY
 )
-install(FILES ${CMAKE_BINARY_DIR}/sz.pc DESTINATION ${CMAKE_INSTALL_PREFIX}/share/pkgconfig)
+install(FILES ${CMAKE_BINARY_DIR}/sz.pc DESTINATION ${CMAKE_INSTALL_LIBDIR}/pkgconfig)
 
--- a/sz/CMakeLists.txt
+++ b/sz/CMakeLists.txt
@@ -100,6 +100,6 @@ install (TARGETS SZ ${thirdparty_export}
   RUNTIME DESTINATION ${CMAKE_INSTALL_BINDIR}
   )
 
-install (EXPORT SZConfig DESTINATION share/SZ/cmake)
+install (EXPORT SZConfig DESTINATION ${CMAKE_INSTALL_LIBDIR}/cmake/SZ)
 install (DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR}/include/ DESTINATION ${CMAKE_INSTALL_INCLUDEDIR}/sz)
-export(TARGETS SZ FILE SZ.cmake)
+export(TARGETS SZ FILE "${CMAKE_CURRENT_BINARY_DIR}/SZ.cmake")
+install (FILES "${CMAKE_CURRENT_BINARY_DIR}/SZ.cmake" DESTINATION ${CMAKE_INSTALL_LIBDIR}/cmake/SZ)
