diff -up openscad-2019.05/features/cgal.prf.bak openscad-2019.05/features/cgal.prf
--- openscad-2019.05/features/cgal.prf.bak	2019-10-02 15:35:12.951314278 +0200
+++ openscad-2019.05/features/cgal.prf	2019-10-02 15:44:04.034063578 +0200
@@ -1,4 +1,6 @@
-DEFINES += ENABLE_CGAL
+DEFINES += ENABLE_CGAL CGAL_HEADER_ONLY
+
+QMAKE_CXXFLAGS += -std=c++14
 
 # Optionally specify location of CGAL using the 
 # CGALDIR env. variable
@@ -9,7 +11,7 @@ CGAL_DIR = $$(CGALDIR)
   message("CGAL location: $$CGAL_DIR")
 }
 
-LIBS += -lCGAL -lmpfr -lgmp
+LIBS += -lmpfr -lgmp
 
 *g++* {
   QMAKE_CXXFLAGS += -frounding-math
