From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Reilly Brogan <reilly@reillybrogan.com>
Date: Mon, 25 Nov 2024 16:51:02 -0600
Subject: [PATCH] Update Wayland appId to match desktop rename

---
 cockatrice/src/main.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/cockatrice/src/main.cpp b/cockatrice/src/main.cpp
index b036b240..d38d14d0 100644
--- a/cockatrice/src/main.cpp
+++ b/cockatrice/src/main.cpp
@@ -170,7 +170,7 @@ int main(int argc, char *argv[])
     ui.setWindowIcon(QPixmap("theme:cockatrice"));
 #if QT_VERSION >= QT_VERSION_CHECK(5, 7, 0)
     // set name of the app desktop file; used by wayland to load the window icon
-    QGuiApplication::setDesktopFileName("cockatrice");
+    QGuiApplication::setDesktopFileName("io.github.Cockatrice.cockatrice");
 #endif
 
     SettingsCache::instance().setClientID(generateClientID());
