From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Reilly Brogan <reilly@reillybrogan.com>
Date: Mon, 25 Nov 2024 16:51:02 -0600
Subject: [PATCH] Update Wayland appId to match desktop rename

---
 cockatrice/src/main.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/cockatrice/src/main.cpp b/cockatrice/src/main.cpp
--- a/cockatrice/src/main.cpp
+++ b/cockatrice/src/main.cpp
@@ -262,5 +262,5 @@
     ui.setWindowIcon(QPixmap("theme:cockatrice"));
     // set name of the app desktop file; used by wayland to load the window icon
-    QGuiApplication::setDesktopFileName("cockatrice");
+    QGuiApplication::setDesktopFileName("io.github.Cockatrice.cockatrice");
 
     SettingsCache::instance().setClientID(generateClientID());
