From 40ecd73c4e07be23a7ef6ade735fe9d3ad6736fd Mon Sep 17 00:00:00 2001
From: ebbit1q <ebbit1q@gmail.com>
Date: Sat, 19 Oct 2024 12:34:18 +0200
Subject: [PATCH] don't delete ""

---
 cockatrice/src/tab_replays.cpp | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/cockatrice/src/tab_replays.cpp b/cockatrice/src/tab_replays.cpp
index fc70cd4a86..c284a531e1 100644
--- a/cockatrice/src/tab_replays.cpp
+++ b/cockatrice/src/tab_replays.cpp
@@ -146,6 +146,9 @@ void TabReplays::actOpenLocalReplay()
 void TabReplays::actDeleteLocalReplay()
 {
     QModelIndex curLeft = localDirView->selectionModel()->currentIndex();
+    if (!curLeft.isValid())
+        return;
+
     if (QMessageBox::warning(this, tr("Delete local file"),
                              tr("Are you sure you want to delete \"%1\"?").arg(localDirModel->fileName(curLeft)),
                              QMessageBox::Yes | QMessageBox::No) != QMessageBox::Yes)
