From 25e28174821a9d7d3fae5894bb3040f5dbf8cb08 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Ga=C3=ABl=20Bonithon?= <gael@xfce.org>
Date: Wed, 14 Aug 2024 19:19:32 +0200
Subject: [PATCH] wayland: wrapper: Prevent plugin allocation blocking

Fixes: #849
---
 wrapper/wrapper-plug-wayland.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/wrapper/wrapper-plug-wayland.c b/wrapper/wrapper-plug-wayland.c
index f385d03a7..23f65821c 100644
--- a/wrapper/wrapper-plug-wayland.c
+++ b/wrapper/wrapper-plug-wayland.c
@@ -166,6 +166,10 @@ wrapper_plug_wayland_init (WrapperPlugWayland *plug)
   plug->geometry.width = 1;
   plug->geometry.height = 1;
 
+  /* set a minimum size to start with so as not to block plugin allocation (e.g. for systray,
+   * see #849); this will then correct itself through geometry exchanges between socket and plug */
+  gtk_widget_set_size_request (GTK_WIDGET (plug), 16, 16);
+
   /* add panel css classes so they apply to external plugins as they do to internal ones */
   context = gtk_widget_get_style_context (GTK_WIDGET (plug));
   gtk_style_context_add_class (context, "panel");
-- 
GitLab

