From 8e90506ee4a761029c21a60dd9f9ade42000632b Mon Sep 17 00:00:00 2001
From: Ikey Doherty <ikey@solus-project.com>
Date: Sun, 16 Oct 2016 00:46:36 +0100
Subject: [PATCH 8/8] schemas: Unbreak livecd configuration

Signed-off-by: Ikey Doherty <ikey@solus-project.com>
---
 schemas/livecd/Makefile.am | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/schemas/livecd/Makefile.am b/schemas/livecd/Makefile.am
index be82895..a9be665 100644
--- a/schemas/livecd/Makefile.am
+++ b/schemas/livecd/Makefile.am
@@ -1,5 +1,5 @@
 LIVECD_SCHEMAS = \
-	org.gnome.desktop.screensaver.gschema.override
+	org.gnome.desktop.screensaver.gschema.override \
 	com.solus-project.icon-tasklist.gschema.override
 
 %.gschema.override:
-- 
2.10.1

