From 83d0bf4365b09ddd17dddb400ba5d262ddf16fb8 Mon Sep 17 00:00:00 2001
From: Daniel Scharrer <daniel@constexpr.org>
Date: Wed, 20 Jan 2021 08:39:19 +0100
Subject: [PATCH] cmake: Fix warning with newer cmake versions

---
 cmake/VersionScript.cmake | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/cmake/VersionScript.cmake b/cmake/VersionScript.cmake
index be04b3644e5525f91844df49e9575622c70c1864..47c8542e41cb0c17904301ed0a417766d203fcd4 100644
--- a/cmake/VersionScript.cmake
+++ b/cmake/VersionScript.cmake
@@ -17,7 +17,7 @@
 #    misrepresented as being the original software.
 # 3. This notice may not be removed or altered from any source distribution.
 
-cmake_minimum_required(VERSION 2.8)
+cmake_minimum_required(VERSION 2.8...3.19)
 
 # CMake script that reads a VERSION file and the current git history and the calls configure_file().
 # This is used by version_file() in VersionString.cmake
