From 76df4975ce563daddaa77dfdc5fca5196957c5f6 Mon Sep 17 00:00:00 2001
From: Pierre-Yves <pyu@riseup.net>
Date: Sat, 14 Apr 2018 20:02:49 +0200
Subject: [PATCH] Fix bash completion path

Signed-off-by: Pierre-Yves <pyu@riseup.net>
---
 src/Makefile | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/src/Makefile b/src/Makefile
index 2e13ffb..1caa240 100644
--- a/src/Makefile
+++ b/src/Makefile
@@ -823,9 +823,9 @@ endif
 	  cp -f ../man/gmic.1.gz $(DESTDIR)$(USR)/share/man/fr/man1/gmic.1.gz; \
 	fi
 	@if [ -f ../resources/gmic_bashcompletion.sh ]; then \
-	  if [ -d /etc/bash_completion.d/ ]; then \
-	    mkdir -p $(DESTDIR)/etc/bash_completion.d/; \
-            cp -f ../resources/gmic_bashcompletion.sh $(DESTDIR)/etc/bash_completion.d/gmic; \
+	  if [ -d /usr/share/bash-completion/completions/ ]; then \
+	    mkdir -p $(DESTDIR)/usr/share/bash-completion/completions/; \
+            cp -f ../resources/gmic_bashcompletion.sh $(DESTDIR)/usr/share/bash-completion/completions/gmic; \
 	  fi; \
 	  if [ -d /opt/local/etc/bash_completion.d/ ]; then \
             mkdir -p $(DESTDIR)/opt/local/etc/bash_completion.d/; \
-- 
2.16.3

