From a910c0daa0f49094b92e18047a4b0530b8f77576 Mon Sep 17 00:00:00 2001
From: Pierre-Yves <pyu@riseup.net>
Date: Thu, 21 Jun 2018 21:51:33 +0200
Subject: [PATCH] Zart: Fix build with Qt 5.11

Signed-off-by: Pierre-Yves <pyu@riseup.net>
---
 zart/src/MainWindow.cpp | 1 +
 1 file changed, 1 insertion(+)

diff --git a/zart/src/MainWindow.cpp b/zart/src/MainWindow.cpp
index c4e6590..638b887 100644
--- a/zart/src/MainWindow.cpp
+++ b/zart/src/MainWindow.cpp
@@ -47,6 +47,7 @@
 
 #include <QAction>
 #include <QActionGroup>
+#include <QButtonGroup>
 #include <QComboBox>
 #include <QDesktopServices>
 #include <QDir>
-- 
2.17.1

