From 2288040e37879f08098b800b429028952fe7304b Mon Sep 17 00:00:00 2001
From: Ikey Doherty <ikey@solus-project.com>
Date: Mon, 23 Oct 2017 17:33:12 +0100
Subject: [PATCH] limits: Disable core dump behaviour

Signed-off-by: Ikey Doherty <ikey@solus-project.com>
---
 modules/pam_limits/limits.conf | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/modules/pam_limits/limits.conf b/modules/pam_limits/limits.conf
index 5d5c3f7..0cd471c 100644
--- a/modules/pam_limits/limits.conf
+++ b/modules/pam_limits/limits.conf
@@ -47,4 +47,6 @@
 #ftp             hard    nproc           0
 #@student        -       maxlogins       4
 
+*               hard    core            0
+
 # End of file
-- 
2.14.2

