From 09995ea699aa1c46d48ace2e79132665f26b16a8 Mon Sep 17 00:00:00 2001
From: Reilly Brogan <reilly@reillybrogan.com>
Date: Sun, 12 Nov 2023 22:20:04 -0600
Subject: [PATCH] Solus defaults

---
 modules/pam_limits/limits.conf | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/modules/pam_limits/limits.conf b/modules/pam_limits/limits.conf
index e8a746cc..343e7320 100644
--- a/modules/pam_limits/limits.conf
+++ b/modules/pam_limits/limits.conf
@@ -59,3 +59,7 @@
 #@student        -       maxlogins       4
 
 # End of file
+
+# Solus defaults:
+*                soft    nofile          524288
+*                hard    nofile          524288
-- 
2.42.0

