diff -rub Linux-PAM-1.1.5/modules/pam_unix/pam_unix_passwd.c Linux-PAM-1.1.5-patched/modules/pam_unix/pam_unix_passwd.c
--- Linux-PAM-1.1.5/modules/pam_unix/pam_unix_passwd.c	2011-06-21 12:04:56.000000000 +0300
+++ Linux-PAM-1.1.5-patched/modules/pam_unix/pam_unix_passwd.c	2012-02-22 14:57:25.000000000 +0300
@@ -736,9 +736,6 @@
 			 * password is acceptable.
 			 */
 
-			if (*(const char *)pass_new == '\0') {	/* "\0" password = NULL */
-				pass_new = NULL;
-			}
 			retval = _pam_unix_approve_pass(pamh, ctrl, pass_old,
 			                                pass_new, pass_min_len);
 
