diff --git a/qt.css b/qt.css
index 5d330683..82871ca0 100644
--- a/qt.css
+++ b/qt.css
@@ -713,11 +713,12 @@ AccountView QPushButton#mSettingsPushButton {
     border: 0;
     border-radius: 0;
     icon-size: 20px;
+    image: url(:/images/toolbar/settings.png);
 }
 
 AccountView QPushButton:hover#mSettingsPushButton {
     border: 0;
     border-radius: 0;
     icon-size: 20px;
-    icon: url(:/images/toolbar/settings-orange.png);
+    image: url(:/images/toolbar/settings-orange.png);
 }
\ No newline at end of file
diff --git a/src/ui/account-view.cpp b/src/ui/account-view.cpp
index 3dc27250..38d7fd45 100644
--- a/src/ui/account-view.cpp
+++ b/src/ui/account-view.cpp
@@ -66,7 +66,6 @@ AccountView::AccountView(QWidget *parent)
     // automatically.
     mRefreshLabel->setPixmap(QIcon(":/images/toolbar/refresh-new.png").pixmap(20));
     mRefreshLabel->installEventFilter(this);
-    mSettingsPushButton->setIcon(QIcon(":/images/toolbar/settings.png"));
     connect(mSettingsPushButton, &QPushButton::clicked, this, &AccountView::slotShowSettingsDialog);
 }
 
