From cb273ff4e0a622b5149ef48c04ed131840c2c111 Mon Sep 17 00:00:00 2001
From: James Lee <jamesl33info@gmail.com>
Date: Sat, 27 Oct 2018 14:56:39 +0100
Subject: [PATCH] Don't include external ldflags. It doesn't like them.

---
 Makefile | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/Makefile b/Makefile
index 8ca872f..b707b37 100644
--- a/Makefile
+++ b/Makefile
@@ -6,7 +6,7 @@ ROOT_DIR       := $(shell dirname $(MAKEFILE))
 SOURCES        := $(wildcard *.go src/*.go src/*/*.go) $(MAKEFILE)
 
 REVISION       := $(shell git log -n 1 --pretty=format:%h -- $(SOURCES))
-BUILD_FLAGS    := -a -ldflags "-X main.revision=$(REVISION) -w '-extldflags=$(LDFLAGS)'" -tags "$(TAGS)"
+BUILD_FLAGS    := -a -ldflags "-X main.revision=$(REVISION) -w" -tags "$(TAGS)"
 
 BINARY32       := fzf-$(GOOS)_386
 BINARY64       := fzf-$(GOOS)_amd64
