From 486097bbb4083b83a1fdbde6d2c1b71827091e72 Mon Sep 17 00:00:00 2001
From: Joshua Strobl <joshua@stroblindustries.com>
Date: Thu, 26 Oct 2017 06:59:01 +0300
Subject: [PATCH 1/1] Don't include external ldflags. It doesn't like them.

---
 Makefile | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/Makefile b/Makefile
index 5c15f69..993108b 100644
--- a/Makefile
+++ b/Makefile
@@ -21,7 +21,7 @@ GLIDE_LOCK  := glide.lock
 SOURCES     := $(wildcard *.go src/*.go src/*/*.go) $(SRC_LINK) $(VENDOR_LINK) $(GLIDE_LOCK) $(MAKEFILE)
 
 REVISION    := $(shell git log -n 1 --pretty=format:%h -- $(SOURCES))
-BUILD_FLAGS := -a -ldflags "-X main.revision=$(REVISION) -w -extldflags=$(LDFLAGS)" -tags "$(TAGS)"
+BUILD_FLAGS := -a -ldflags "-X main.revision=$(REVISION) -w" -tags "$(TAGS)"
 
 BINARY32    := fzf-$(GOOS)_386
 BINARY64    := fzf-$(GOOS)_amd64
-- 
2.14.3

