From fd079de243d2f778435aa7e134e31eef7eed0621 Mon Sep 17 00:00:00 2001
From: Joshua Strobl <joshua@streambits.io>
Date: Tue, 24 Sep 2019 13:05:34 +0300
Subject: [PATCH 1/1] Fix use of weak return type in get_providers.

---
 src/Backend/Uploader.vala | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/Backend/Uploader.vala b/src/Backend/Uploader.vala
index ce6efe2..8583651 100755
--- a/src/Backend/Uploader.vala
+++ b/src/Backend/Uploader.vala
@@ -88,7 +88,7 @@ private class Uploader : GLib.Object
         return upload_in_progress;
     }
 
-    public unowned GLib.HashTable<string, weak Providers.IProvider> get_providers() {
+    public unowned GLib.HashTable<string, Providers.IProvider> get_providers() {
         return upload_providers;
     }
 }
-- 
2.23.0

