From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Evan Maddock <maddock.evan@vivaldi.net>
Date: Fri, 31 Oct 2025 19:22:02 -0400
Subject: [PATCH] build: Port to libpeas-2 and budgie-2.0

Signed-off-by: Evan Maddock <maddock.evan@vivaldi.net>
---
 src/meson.build | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/meson.build b/src/meson.build
index a837be9..5c466b0 100644
--- a/src/meson.build
+++ b/src/meson.build
@@ -32,7 +32,7 @@ bha_sources = ['HasteApplet.vala',
                res]
 
 bha_deps = [dependency('gtk+-3.0', version: '>=3.18'),
-            dependency('budgie-1.0', version: '>=2'),
+            dependency('budgie-2.0', version: '>=3'),
             dependency('json-glib-1.0'),
             dependency('libsoup-2.4')]
 
@@ -49,4 +49,4 @@ shared_library('hasteapplet', bha_sources,
                vala_args: bha_vala_args,
                c_args: bha_c_args,
                install: true,
-               install_dir: LIB_INSTALL_DIR)
\ No newline at end of file
+               install_dir: LIB_INSTALL_DIR)
