From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Reilly Brogan <reilly@reillybrogan.com>
Date: Mon, 25 Nov 2024 15:52:33 -0600
Subject: [PATCH] wayland: Fix appId

Signed-off-by: Reilly Brogan <reilly@reillybrogan.com>
---
 src/main.cpp | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/src/main.cpp b/src/main.cpp
index e61aff9..1efc440 100644
--- a/src/main.cpp
+++ b/src/main.cpp
@@ -415,6 +415,9 @@ int main(int argc, char *argv[])
     QTranslator qtTranslator;
 
 #if defined(Q_OS_UNIX)
+    // Ensure that the Wayland appId matches the .desktop file name
+    QGuiApplication::setDesktopFileName("io.github.antimicrox.antimicrox");
+
     installSignalHandlers();
 
     QString transPath = QLibraryInfo::location(QLibraryInfo::TranslationsPath);
