From 45e443887a640dc8dc21d59423204ac34a4832a5 Mon Sep 17 00:00:00 2001
From: Joey Riches <josephriches@gmail.com>
Date: Thu, 16 Jul 2026 20:44:49 +0100
Subject: [PATCH 2/3] backends/eopkg: Mark security updates with critical
 severity

---
 backends/eopkg/eopkgBackend.py | 3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

diff --git a/backends/eopkg/eopkgBackend.py b/backends/eopkg/eopkgBackend.py
index fd4730a5f..6936d6bc8 100644
--- a/backends/eopkg/eopkgBackend.py
+++ b/backends/eopkg/eopkgBackend.py
@@ -654,9 +654,8 @@ class PackageKitEopkgBackend(PackageKitBaseBackend, PackagekitPackage):
             histories = self._get_history_between(oldRelease, pkg)
 
             securities = [x for x in histories if x.type == "security"]
-            # FIXME: INFO_BUGFIX Support? We would have to match against #123 Github issues
             if len(securities) > 0:
-                self.package(id, INFO_SECURITY, pkg.summary)
+                self.package(id, INFO_SECURITY, pkg.summary, INFO_CRITICAL)
             else:
                 self.package(id, INFO_NORMAL, pkg.summary)
 
-- 
2.54.0

