--- gcc-12.1.0/gcc/opts.cc~	2022-05-06 07:30:57.000000000 +0000
+++ gcc-12.1.0/gcc/opts.cc	2022-05-12 22:59:07.654517795 +0000
@@ -636,7 +636,7 @@
     { OPT_LEVELS_2_PLUS, OPT_ftree_tail_merge, NULL, 1 },
     { OPT_LEVELS_2_PLUS, OPT_ftree_vrp, NULL, 1 },
     { OPT_LEVELS_2_PLUS, OPT_fvect_cost_model_, NULL,
-      VECT_COST_MODEL_VERY_CHEAP },
+      VECT_COST_MODEL_CHEAP },
     { OPT_LEVELS_2_PLUS, OPT_finline_functions, NULL, 1 },
     { OPT_LEVELS_2_PLUS, OPT_ftree_loop_distribute_patterns, NULL, 1 },
 
