From 05c9913a3ced7273a378bad1d9597eab29d6aa1e Mon Sep 17 00:00:00 2001
From: Philipp Trulson <philipp@trulson.de>
Date: Sun, 4 Jun 2017 13:36:45 +0200
Subject: [PATCH] Fix Budgie entry

---
 umbrello/umbrello.desktop | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/umbrello/umbrello.desktop b/umbrello/umbrello.desktop
index d9d1c87..be144b3 100644
--- a/umbrello/umbrello.desktop
+++ b/umbrello/umbrello.desktop
@@ -1,7 +1,7 @@
 # KDE Config File
 [Desktop Entry]
 Type=Application
-Exec=umbrello %u -caption %c
+Exec=umbrello5 %u -caption %c
 Icon=umbrello
 X-DocPath=umbrello/index.html
 Terminal=false
-- 
2.13.0

