From ae81b53bacfb352789087d53efb789ad952ab41d Mon Sep 17 00:00:00 2001
From: Peter O'Connor <peter@solus-project.com>
Date: Fri, 20 Apr 2018 14:46:44 +1000
Subject: [PATCH] Disable building unshipped code

Signed-off-by: Peter O'Connor <peter@solus-project.com>
---
 CMakeLists.txt | 5 -----
 1 file changed, 5 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index d9c2194..e638457 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -342,9 +342,4 @@ endif(LIBXSLT_FOUND AND LIBXML2_FOUND)
 
 add_subdirectory(maintainer)
 
-# for automatic unit tests
-if(BUILD_UNITTESTS)
-    ecm_optional_add_subdirectory(unittests)
-endif()
-
 feature_summary(WHAT ALL FATAL_ON_MISSING_REQUIRED_PACKAGES)
-- 
2.16.3

