From e6834fd3613247d0014040b5996b2c5540123952 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Alfonso=20Rodr=C3=ADguez?= <arc.profesional@gmail.com>
Date: Thu, 6 Apr 2017 00:47:52 +0200
Subject: [PATCH 1/1] Fix icon

---
 flow.desktop | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/flow.desktop b/flow.desktop
index aec4e59..82c7ac3 100755
--- a/flow.desktop
+++ b/flow.desktop
@@ -1,6 +1,6 @@
 [Desktop Entry]
 Name=flow
-Icon=kmines
+Icon=flow
 Exec=flow
 Type=Application
 GenericName=flow pomodoro
-- 
2.12.2

