diff --git a/tecmake.mak b/tecmake.mak
index 48991a4..b56e14b 100755
--- a/tecmake.mak
+++ b/tecmake.mak
@@ -1667,7 +1667,9 @@ dynamic-lib: $(TARGETDIR)/$(TARGETDLIBNAME)
 
 $(TARGETDIR)/$(TARGETDLIBNAME) : $(LUAS) $(OBJS) $(EXTRADEPS)
 	@echo ''; echo Tecmake: linking $(@F) ...
-	$(ECHO)$(LD) $(STDLDFLAGS) -o $@ $(OBJS) $(SLIB) $(LFLAGS)
+	$(ECHO)$(LD) $(STDLDFLAGS) -Wl,-soname,$(@F).2 -o $@.2.1.5 $(OBJS) $(SLIB) $(LFLAGS)
+	@echo ''; ln -sv $(@F).2.1.5 $@.2
+	@echo ''; ln -sv $(@F).2.1.5 $@
 	@echo ''; echo 'Tecmake: Dynamic Library ($@) Done.'; echo ''
 
 
