From fd0d12162ffb24fb1048467d265e942d54ca482d Mon Sep 17 00:00:00 2001
From: Andreas Beckmann <anbe@debian.org>
Date: Sun, 13 Oct 2024 08:52:51 +0200
Subject: [PATCH] let the virt_addr_valid() macro use nv_pfn_valid() on ppc64el

---
 common/inc/nv-linux.h | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/common/inc/nv-linux.h b/common/inc/nv-linux.h
index 7f93fd0..2b685c9 100644
--- a/common/inc/nv-linux.h
+++ b/common/inc/nv-linux.h
@@ -2016,6 +2016,11 @@ static inline int nv_pfn_valid(unsigned long pfn)
         return ret;
 }
 
+#if defined(NVCPU_PPC64LE)
+// let the virt_addr_valid() macro use nv_pfn_valid()
+#define pfn_valid nv_pfn_valid
+#endif
+
 #else
 #  define nv_pfn_valid pfn_valid
 #endif
-- 
2.39.5

