From 7e21bf18dce793785e32e159b410356c9bbabcd5 Mon Sep 17 00:00:00 2001
From: Andreas Beckmann <anbe@debian.org>
Date: Sun, 8 Dec 2024 03:20:30 +0100
Subject: [PATCH] backport uvm warning fixes from 510.39.01

---
 nvidia-uvm/uvm_pascal_mmu.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/nvidia-uvm/uvm_pascal_mmu.c b/nvidia-uvm/uvm_pascal_mmu.c
index cecfc5a4..d5e577fd 100644
--- a/nvidia-uvm/uvm_pascal_mmu.c
+++ b/nvidia-uvm/uvm_pascal_mmu.c
@@ -171,7 +171,7 @@ static NvLength entry_size_pascal(NvU32 depth)
 
 static NvU32 index_bits_pascal(NvU32 depth, NvU32 page_size)
 {
-    const static NvU32 bit_widths[] = {2, 9, 9, 8};
+    static const NvU32 bit_widths[] = {2, 9, 9, 8};
     // some code paths keep on querying this until they get a 0, meaning only the page offset remains.
     UVM_ASSERT(depth < 5);
     if (depth < 4) {
-- 
2.39.5

