From 66dde5c63945a9f788593a354fbda76d10244723 Mon Sep 17 00:00:00 2001
From: Andreas Beckmann <anbe@debian.org>
Date: Wed, 28 May 2025 17:15:22 +0200
Subject: [PATCH] backport drm_gem_object_vmap_has_map_arg changes from
 535.171.04

---
 conftest.sh | 8 ++++++--
 1 file changed, 6 insertions(+), 2 deletions(-)

diff --git a/conftest.sh b/conftest.sh
index ebb5aa07..eabd0f87 100755
--- a/conftest.sh
+++ b/conftest.sh
@@ -4609,11 +4609,15 @@ compile_test() {
             # vmap ops and convert GEM backends") update
             # drm_gem_object_funcs::vmap to take 'map' argument.
             #
+            # Note that the 'map' argument type is changed from 'struct dma_buf_map'
+            # to 'struct iosys_map' by commit 7938f4218168 ("dma-buf-map: Rename
+            # to iosys-map) in v5.18.
+            #
             CODE="
             #include <drm/drm_gem.h>
             int conftest_drm_gem_object_vmap_has_map_arg(
-                    struct drm_gem_object *obj, struct dma_buf_map *map) {
-                return obj->funcs->vmap(obj, map);
+                    struct drm_gem_object *obj) {
+                return obj->funcs->vmap(obj, NULL);
             }"
 
             compile_check_conftest "$CODE" "NV_DRM_GEM_OBJECT_VMAP_HAS_MAP_ARG" "" "types"
-- 
2.39.5

