diff --git a/src/sdk/src/Common/CompileOptions.cs b/src/sdk/src/Common/CompileOptions.cs
index bd17db1200..44fd9b681f 100644
--- a/src/sdk/src/Common/CompileOptions.cs
+++ b/src/sdk/src/Common/CompileOptions.cs
@@ -10,17 +10,7 @@ namespace Microsoft.DotNet.Cli
 {
     static class CompileOptions
     {
-        public const bool TelemetryOptOutDefault =
-#if MICROSOFT_ENABLE_TELEMETRY
-        false;
-#else
-        true;
-#endif
-        public const string TelemetryOptOutDefaultString =
-#if MICROSOFT_ENABLE_TELEMETRY
-        "False";
-#else
-        "True";
-#endif
+        public const bool TelemetryOptOutDefault = true;
+        public const string TelemetryOptOutDefaultString = "True";
     }
 }
