diff --git a/src/diagnostics/src/SOS/extensions/hostcoreclr.cpp b/src/diagnostics/src/SOS/extensions/hostcoreclr.cpp
index db940bbf5b..32a689c7b0 100644
--- a/src/diagnostics/src/SOS/extensions/hostcoreclr.cpp
+++ b/src/diagnostics/src/SOS/extensions/hostcoreclr.cpp
@@ -117,6 +117,7 @@ namespace RuntimeHostingConstants
 #endif
         "/usr/local/share/dotnet"
 #else
+        "/usr/lib64/dotnet",
         "/rh-dotnet60/root/usr/bin/dotnet",
         "/rh-dotnet70/root/usr/bin/dotnet",
         "/rh-dotnet80/root/usr/bin/dotnet",
diff --git a/src/runtime/src/native/corehost/hostmisc/pal.unix.cpp b/src/runtime/src/native/corehost/hostmisc/pal.unix.cpp
index 34520aefd7..ec39f4b060 100644
--- a/src/runtime/src/native/corehost/hostmisc/pal.unix.cpp
+++ b/src/runtime/src/native/corehost/hostmisc/pal.unix.cpp
@@ -592,7 +592,7 @@ bool pal::get_default_installation_dir_for_arch(pal::architecture arch, pal::str
     }
 #endif
 #else
-    recv->assign(_X("/usr/share/dotnet"));
+    recv->assign(_X("/usr/lib64/dotnet"));
 #endif
     return true;
 }
