From 88f8ee20e6de50f8f05771b9c668b11f3f38f84c Mon Sep 17 00:00:00 2001
From: Joshua Strobl <joshua@stroblindustries.com>
Date: Fri, 26 May 2017 16:54:52 +0300
Subject: [PATCH 1/1] Fix names of executables

---
 ding | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/ding b/ding
index 328e759..c0c223a 100755
--- a/ding
+++ b/ding
@@ -1,6 +1,6 @@
 #!/bin/sh
 # the next line restarts using wish \
-LC_CTYPE=de_DE.UTF-8; export LC_CTYPE; exec wish "$0" "$@"
+LC_CTYPE=de_DE.UTF-8; export LC_CTYPE; exec wish8.6 "$0" "$@"
 
 # This is "ding", 
 #  * A dictionary lookup program,
@@ -95,14 +95,14 @@ set default_searchmeth(0,foldedresult) 1
 set default_searchmeth(1,name) {Spell check}
 set default_searchmeth(1,grepcmds) {hunspell ispell aspell}
 set default_searchmeth(1,language1)  "English spell check"
-set default_searchmeth(1,grepopts)  "-B -d english -a"
+set default_searchmeth(1,grepopts)  "-B -d en_GB -a"
 # --encoding utf-8
 
 # spell checker
 set default_searchmeth(2,name) {Rechtschreibung}
 set default_searchmeth(2,grepcmds) {hunspell ispell aspell}
 set default_searchmeth(2,language1)  "Deutsche Rechtschreibprüfung"
-set default_searchmeth(2,grepopts)  "-B -d german -a"
+set default_searchmeth(2,grepopts)  "-B -d de_DE -a"
 
 # dict - english ditionary
 set default_searchmeth(3,name) {English}
-- 
2.13.0

