--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -32,7 +32,7 @@
 # OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGES.
 
 cmake_minimum_required(VERSION 3.15.0)
-project(partio LANGUAGES CXX)
+project(partio LANGUAGES CXX VERSION 1.14.0)
 
 option(PARTIO_GTEST_ENABLED "Enable GTest for tests" OFF)
 
@@ -114,8 +114,7 @@ set(OpenGL_GL_PREFERENCE GLVND)
 find_package(Threads REQUIRED)
 find_package(GLUT REQUIRED)
 find_package(OpenGL REQUIRED)
-find_package(PythonInterp REQUIRED)
-find_package(PythonLibs REQUIRED)
+find_package(Python COMPONENTS Interpreter Development REQUIRED)
 find_package(ZLIB)
 if (ZLIB_FOUND)
     add_definitions(-DPARTIO_USE_ZLIB)
--- a/src/py/CMakeLists.txt
+++ b/src/py/CMakeLists.txt
@@ -47,12 +47,12 @@ get_property(swig_files TARGET partio_mo
 target_include_directories(
     partio_mod
     PRIVATE
-        ${PYTHON_INCLUDE_DIRS}
+        ${Python_INCLUDE_DIRS}
         ${CMAKE_SOURCE_DIR}/src/lib
 )
-target_link_libraries(partio_mod PRIVATE ${PYTHON_LIBRARIES} partio)
+target_link_libraries(partio_mod PRIVATE ${Python_LIBRARIES} partio)
 
-set(PYTHON_DEST "${CMAKE_INSTALL_LIBDIR}/python${PYTHON_VERSION_MAJOR}.${PYTHON_VERSION_MINOR}/site-packages" )
+set(PYTHON_DEST "${CMAKE_INSTALL_LIBDIR}/python${Python_VERSION_MAJOR}.${Python_VERSION_MINOR}/site-packages/partio" )
 
 install(TARGETS partio_mod DESTINATION ${PYTHON_DEST})
 install(FILES partio.i DESTINATION ${CMAKE_INSTALL_DATADIR}/swig)
--- a/src/tools/CMakeLists.txt
+++ b/src/tools/CMakeLists.txt
@@ -67,5 +67,5 @@ install(PROGRAMS partedit.py DESTINATION
 install(PROGRAMS partjson.py DESTINATION ${CMAKE_INSTALL_BINDIR} RENAME partjson)
 install(PROGRAMS partinspect.py DESTINATION ${CMAKE_INSTALL_BINDIR} RENAME partinspect)
 
-set(PYTHON_DEST "${CMAKE_INSTALL_LIBDIR}/python${PYTHON_VERSION_MAJOR}.${PYTHON_VERSION_MINOR}/site-packages" )
+set(PYTHON_DEST "${CMAKE_INSTALL_LIBDIR}/python${Python_VERSION_MAJOR}.${Python_VERSION_MINOR}/site-packages/partio" )
 install(FILES partedit.py partjson.py partinspect.py DESTINATION ${PYTHON_DEST})
