--- a/src/lib/CMakeLists.txt
+++ b/src/lib/CMakeLists.txt
@@ -41,6 +41,8 @@ else()
 endif()
 add_library(partio ${PARTIO_LIBRARY_TYPE} ${io_cpp} ${core_cpp})
 set_target_properties(partio PROPERTIES OUTPUT_NAME partio POSITION_INDEPENDENT_CODE ON)
+set_target_properties(partio PROPERTIES VERSION "${PROJECT_VERSION}")
+set_target_properties(partio PROPERTIES SOVERSION 1)
 
 target_include_directories(partio
     PUBLIC
