From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Reilly Brogan <reilly@reillybrogan.com>
Date: Mon, 15 Apr 2024 16:58:29 -0500
Subject: [PATCH] Stub out pip check

---
 python/mach/mach/site.py | 14 --------------
 1 file changed, 14 deletions(-)

diff --git a/python/mach/mach/site.py b/python/mach/mach/site.py
index 7041437bf8..150976ec0d 100644
--- a/python/mach/mach/site.py
+++ b/python/mach/mach/site.py
@@ -1213,20 +1213,6 @@ def _assert_pip_check(pthfile_lines, virtualenv_name, requirements):
                     "system Python packages."
                 )
 
-        check_result = subprocess.run(
-            pip + ["check"],
-            stdout=subprocess.PIPE,
-            stderr=subprocess.STDOUT,
-            universal_newlines=True,
-        )
-        if check_result.returncode:
-            subprocess.check_call(pip + ["list", "-v"], stdout=sys.stderr)
-            print(check_result.stdout, file=sys.stderr)
-            raise Exception(
-                'According to "pip check", the current Python '
-                "environment has package-compatibility issues."
-            )
-
         os.environ[
             f"MACH_SYSTEM_ASSERTED_COMPATIBLE_WITH_{virtualenv_name.upper()}_SITE"
         ] = "1"
