diff --git a/third_party/libwebrtc/modules/video_coding/webrtc_libvpx_interface_gn/moz.build b/third_party/libwebrtc/modules/video_coding/webrtc_libvpx_interface_gn/moz.build
--- a/third_party/libwebrtc/modules/video_coding/webrtc_libvpx_interface_gn/moz.build
+++ b/third_party/libwebrtc/modules/video_coding/webrtc_libvpx_interface_gn/moz.build
@@ -3,10 +3,13 @@
 # file, You can obtain one at http://mozilla.org/MPL/2.0/.
 
 
   ### This moz.build was AUTOMATICALLY GENERATED from a GN config,  ###
   ### DO NOT edit it by hand.                                       ###
+if not CONFIG["MOZ_SYSTEM_LIBVPX"]:
+    LOCAL_INCLUDES += [ "/media/libvpx/libvpx/" ]
+    CXXFLAGS += CONFIG["MOZ_LIBVPX_CFLAGS"]
 
 COMPILE_FLAGS["OS_INCLUDES"] = []
 AllowCompilerWarnings()
 
 DEFINES["ABSL_ALLOCATOR_NOTHROW"] = "1"
@@ -23,11 +26,10 @@
 
 LOCAL_INCLUDES += [
     "!/ipc/ipdl/_ipdlheaders",
     "!/third_party/libwebrtc/gen",
     "/ipc/chromium/src",
-    "/media/libvpx/libvpx/",
     "/third_party/libwebrtc/",
     "/third_party/libwebrtc/third_party/abseil-cpp/",
     "/tools/profiler/public"
 ]
 
diff --git a/third_party/libwebrtc/modules/video_coding/webrtc_vp8_gn/moz.build b/third_party/libwebrtc/modules/video_coding/webrtc_vp8_gn/moz.build
--- a/third_party/libwebrtc/modules/video_coding/webrtc_vp8_gn/moz.build
+++ b/third_party/libwebrtc/modules/video_coding/webrtc_vp8_gn/moz.build
@@ -3,10 +3,13 @@
 # file, You can obtain one at http://mozilla.org/MPL/2.0/.
 
 
   ### This moz.build was AUTOMATICALLY GENERATED from a GN config,  ###
   ### DO NOT edit it by hand.                                       ###
+if not CONFIG["MOZ_SYSTEM_LIBVPX"]:
+    LOCAL_INCLUDES += [ "/media/libvpx/libvpx/" ]
+    CXXFLAGS += CONFIG["MOZ_LIBVPX_CFLAGS"]
 
 COMPILE_FLAGS["OS_INCLUDES"] = []
 AllowCompilerWarnings()
 
 DEFINES["ABSL_ALLOCATOR_NOTHROW"] = "1"
@@ -23,11 +26,10 @@
 
 LOCAL_INCLUDES += [
     "!/ipc/ipdl/_ipdlheaders",
     "!/third_party/libwebrtc/gen",
     "/ipc/chromium/src",
-    "/media/libvpx/libvpx/",
     "/media/libyuv/",
     "/media/libyuv/libyuv/include/",
     "/third_party/libwebrtc/",
     "/third_party/libwebrtc/third_party/abseil-cpp/",
     "/tools/profiler/public"
diff --git a/third_party/libwebrtc/modules/video_coding/webrtc_vp9_gn/moz.build b/third_party/libwebrtc/modules/video_coding/webrtc_vp9_gn/moz.build
--- a/third_party/libwebrtc/modules/video_coding/webrtc_vp9_gn/moz.build
+++ b/third_party/libwebrtc/modules/video_coding/webrtc_vp9_gn/moz.build
@@ -3,10 +3,13 @@
 # file, You can obtain one at http://mozilla.org/MPL/2.0/.
 
 
   ### This moz.build was AUTOMATICALLY GENERATED from a GN config,  ###
   ### DO NOT edit it by hand.                                       ###
+if not CONFIG["MOZ_SYSTEM_LIBVPX"]:
+    LOCAL_INCLUDES += [ "/media/libvpx/libvpx/" ]
+    CXXFLAGS += CONFIG["MOZ_LIBVPX_CFLAGS"]
 
 COMPILE_FLAGS["OS_INCLUDES"] = []
 AllowCompilerWarnings()
 
 DEFINES["ABSL_ALLOCATOR_NOTHROW"] = "1"
@@ -23,11 +26,10 @@
 
 LOCAL_INCLUDES += [
     "!/ipc/ipdl/_ipdlheaders",
     "!/third_party/libwebrtc/gen",
     "/ipc/chromium/src",
-    "/media/libvpx/libvpx/",
     "/media/libyuv/",
     "/media/libyuv/libyuv/include/",
     "/third_party/libwebrtc/",
     "/third_party/libwebrtc/third_party/abseil-cpp/",
     "/tools/profiler/public"
diff --git a/third_party/libwebrtc/moz.build b/third_party/libwebrtc/moz.build
--- a/third_party/libwebrtc/moz.build
+++ b/third_party/libwebrtc/moz.build
@@ -437,11 +437,10 @@
     "/third_party/libwebrtc/third_party/abseil-cpp/absl/types/bad_variant_access_gn",
     "/third_party/libwebrtc/third_party/abseil-cpp/absl/types/optional_gn",
     "/third_party/libwebrtc/third_party/abseil-cpp/absl/types/span_gn",
     "/third_party/libwebrtc/third_party/abseil-cpp/absl/types/variant_gn",
     "/third_party/libwebrtc/third_party/abseil-cpp/absl/utility/utility_gn",
-    "/third_party/libwebrtc/third_party/libvpx/libvpx_gn",
     "/third_party/libwebrtc/third_party/libyuv/libyuv_gn",
     "/third_party/libwebrtc/third_party/pffft/pffft_gn",
     "/third_party/libwebrtc/third_party/rnnoise/rnn_vad_gn",
     "/third_party/libwebrtc/video/adaptation/video_adaptation_gn",
     "/third_party/libwebrtc/video/config/encoder_config_gn",

