diff -up firefox-56.0/media/libyuv/libyuv/tools_libyuv/autoroller/unittests/testdata/DEPS.chromium.old firefox-56.0/media/libyuv/libyuv/tools_libyuv/autoroller/unittests/testdata/DEPS.chromium
diff -up firefox-56.0/media/webrtc/trunk/Makefile.old firefox-56.0/media/webrtc/trunk/Makefile
diff -up firefox-56.0/widget/gtk/mozgtk/mozgtk.c.old firefox-56.0/widget/gtk/mozgtk/mozgtk.c
--- firefox-56.0/widget/gtk/mozgtk/mozgtk.c.old	2017-10-04 09:21:56.155553585 +0200
+++ firefox-56.0/widget/gtk/mozgtk/mozgtk.c	2017-10-04 09:22:35.562427061 +0200
@@ -61,7 +61,6 @@ STUB(gdk_screen_get_height_mm)
 STUB(gdk_screen_get_n_monitors)
 STUB(gdk_screen_get_monitor_at_window)
 STUB(gdk_screen_get_monitor_geometry)
-STUB(gdk_screen_get_monitor_workarea)
 STUB(gdk_screen_get_monitor_height_mm)
 STUB(gdk_screen_get_number)
 STUB(gdk_screen_get_resolution)
@@ -550,6 +549,7 @@ STUB(gtk_render_line)
 STUB(gtk_render_option)
 STUB(gtk_render_slider)
 STUB(gtk_scale_new)
+STUB(gdk_screen_get_monitor_workarea)
 STUB(gtk_scrollbar_new)
 STUB(gtk_style_context_add_class)
 STUB(gtk_style_context_add_region)
