From eeccb989196baf072ab2e9ebfbb96b12b9b71b76 Mon Sep 17 00:00:00 2001
From: Algent Albrahimi <algent@protonmail.com>
Date: Wed, 6 Apr 2022 23:10:13 +0200
Subject: [PATCH] fix actions icons

---
 data/pixmaps/CMakeLists.txt | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/data/pixmaps/CMakeLists.txt b/data/pixmaps/CMakeLists.txt
index e44e9b4..dc495c5 100644
--- a/data/pixmaps/CMakeLists.txt
+++ b/data/pixmaps/CMakeLists.txt
@@ -15,7 +15,7 @@ file(COPY ${CMAKE_CURRENT_SOURCE_DIR}
     PATTERN hicolor EXCLUDE
 )
 install(
-    DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR}/hicolor DESTINATION  ${CMAKE_INSTALL_DATADIR}/gnucash/icons
+    DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR}/hicolor DESTINATION  ${CMAKE_INSTALL_DATADIR}/icons
     REGEX "hicolor/.*/apps/.*" EXCLUDE
 )
 file(
-- 
2.35.1

