diff --git a/data/pixmaps/CMakeLists.txt b/data/pixmaps/CMakeLists.txt
index db331fb..f8efa60 100644
--- a/data/pixmaps/CMakeLists.txt
+++ b/data/pixmaps/CMakeLists.txt
@@ -56,8 +56,8 @@ gnucash-icon-48x48.bmp
 gnucash_splash.png
 )
 
-install(FILES ${gnc_other_pixmaps} DESTINATION ${CMAKE_INSTALL_DATADIR}/gnucash/pixmaps)
-file(COPY ${gnc_other_pixmaps} DESTINATION ${DATADIR_BUILD}/gnucash/pixmaps)
+install(FILES ${gnc_other_pixmaps} DESTINATION ${CMAKE_INSTALL_DATADIR}/pixmaps)
+file(COPY ${gnc_other_pixmaps} DESTINATION ${DATADIR_BUILD}/pixmaps)
 
 set(dest_base_dir "gnucash/icons")
 
@@ -70,7 +70,7 @@ function (copy_iconpaths_to_dest iconpaths dest_base_dir)
     endforeach()
 endfunction()
 
-copy_iconpaths_to_dest ("${gnc_action_icons}" "gnucash/icons")
+copy_iconpaths_to_dest ("${gnc_action_icons}" "icons")
 copy_iconpaths_to_dest ("${gnc_app_icons}" "icons")
 
 #install(FILES ${gnc_app_icons} DESTINATION ${CMAKE_INSTALL_DATADIR}/icons)

