From 467c7f61c7a81fafee50e2c3661dceb60938273c Mon Sep 17 00:00:00 2001
From: Ikey Doherty <ikey@solus-project.com>
Date: Sun, 10 Dec 2017 19:43:56 +0000
Subject: [PATCH] Use stateless XDG autostart dir

Signed-off-by: Ikey Doherty <ikey@solus-project.com>
---
 bus/Makefile.am | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/bus/Makefile.am b/bus/Makefile.am
index 688f82a..3950cac 100644
--- a/bus/Makefile.am
+++ b/bus/Makefile.am
@@ -13,7 +13,7 @@ at_spi_bus_launcher_LDADD = $(GIO_LIBS) $(X11_LIBS)
 
 substitutions = "s,@libexecdir[@],$(libexecdir),"
 
-default_sessiondir = $(sysconfdir)/xdg/autostart
+default_sessiondir = $(datadir)/xdg/autostart
 default_session_DATA = at-spi-dbus-bus.desktop
 at-spi-dbus-bus.desktop: at-spi-dbus-bus.desktop.in
 	$(AM_V_GEN) $(SED) -e$ $(substitutions)  $< > $@.tmp && mv $@.tmp $@
-- 
2.15.1

