From d2aea31bbeabb58005445537b3377e374aea916d Mon Sep 17 00:00:00 2001
From: Joshua Strobl <joshua@streambits.io>
Date: Wed, 26 May 2021 18:34:07 +0300
Subject: [PATCH 1/1] Default to #Solus

---
 src/fe-gtk/joind.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/fe-gtk/joind.c b/src/fe-gtk/joind.c
index ce3cbcae..4d278ee8 100644
--- a/src/fe-gtk/joind.c
+++ b/src/fe-gtk/joind.c
@@ -249,7 +249,7 @@ joind_show_dialog (server *serv)
 	if (serv->network)
 		if (g_ascii_strcasecmp(((ircnet*)serv->network)->name, "Libera.Chat") == 0)
 		{
-			gtk_entry_set_text (GTK_ENTRY (entry1), "#hexchat");
+			gtk_entry_set_text (GTK_ENTRY (entry1), "#solus");
 		}
 
 	gtk_widget_grab_focus (okbutton1);
-- 
2.30.2

