From c0aa47c7a18dbcdf1a2900f251466fdd9c9c4f18 Mon Sep 17 00:00:00 2001
From: Igor <f2404@yandex.ru>
Date: Tue, 11 Oct 2016 12:59:34 +0300
Subject: [PATCH 5/7] fishlim: Fix MEMZERO macro using element number not bytes

Closes #1837
---
 plugins/fishlim/dh1080.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/plugins/fishlim/dh1080.c b/plugins/fishlim/dh1080.c
index 41b6584..5afb1c6 100644
--- a/plugins/fishlim/dh1080.c
+++ b/plugins/fishlim/dh1080.c
@@ -39,7 +39,7 @@
 #define DH1080_PRIME_BYTES 135
 #define SHA256_DIGEST_LENGTH 32
 #define B64ABC "ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789+/"
-#define MEMZERO(x) memset(x, 0x00, (sizeof(x)/sizeof(*x)) )
+#define MEMZERO(x) memset(x, 0x00, sizeof(x))
 
 /* All clients must use the same prime number to be able to keyx */
 static const guchar prime1080[DH1080_PRIME_BYTES] =
-- 
2.10.1

