From 116a394ef6d76c914c3d684784a2a46f1fb6fa22 Mon Sep 17 00:00:00 2001
From: Igor <f2404@yandex.ru>
Date: Tue, 11 Oct 2016 18:19:14 +0400
Subject: [PATCH 7/7] perl: Remove dead assignment

---
 plugins/perl/perl.c | 2 --
 1 file changed, 2 deletions(-)

diff --git a/plugins/perl/perl.c b/plugins/perl/perl.c
index 12cae78..ab06dc9 100644
--- a/plugins/perl/perl.c
+++ b/plugins/perl/perl.c
@@ -208,8 +208,6 @@ get_filename (char *word[], char *word_eol[])
 	int len;
 	char *file;
 
-	len = strlen (word[2]);
-
 	/* if called as /load "filename.pl" the only difference between word and
 	 * word_eol will be the two quotes
 	 */
-- 
2.10.1

