From 5993580d0454401f69c8bae57cd7ac8c6bc040d1 Mon Sep 17 00:00:00 2001
From: Campbell Jones <git@serebit.com>
Date: Tue, 7 Dec 2021 22:07:14 -0500
Subject: [PATCH] Add includes to SWSB_G4IR.cpp

---
 visa/LocalScheduler/SWSB_G4IR.h | 1 +
 1 file changed, 1 insertion(+)

diff --git a/visa/LocalScheduler/SWSB_G4IR.h b/visa/LocalScheduler/SWSB_G4IR.h
index ce44136..0e1fbb9 100644
--- a/visa/LocalScheduler/SWSB_G4IR.h
+++ b/visa/LocalScheduler/SWSB_G4IR.h
@@ -20,6 +20,7 @@ SPDX-License-Identifier: MIT
 #include "../BitSet.h"
 #include "LocalScheduler_G4IR.h"
 #include <utility>
+#include <limits>
 
 namespace vISA
 {
-- 
2.34.0

