From 4bebc524d7fed4ee16a4767a33cbbc2d2dbae462 Mon Sep 17 00:00:00 2001
From: Campbell Jones <git@serebit.com>
Date: Tue, 7 Dec 2021 21:42:48 -0500
Subject: [PATCH] Remove default value for SPIRV-Tools dir

---
 external/SPIRV-Tools/CMakeLists.txt | 2 --
 1 file changed, 2 deletions(-)

diff --git a/external/SPIRV-Tools/CMakeLists.txt b/external/SPIRV-Tools/CMakeLists.txt
index 9e2f083..5d2a450 100644
--- a/external/SPIRV-Tools/CMakeLists.txt
+++ b/external/SPIRV-Tools/CMakeLists.txt
@@ -10,8 +10,6 @@ cmake_minimum_required(VERSION 2.8.12)
 
 message(STATUS "============================ SPIRV-Tools project ============================")
 
-set(SPIRV-Headers_SOURCE_DIR "${CMAKE_CURRENT_SOURCE_DIR}/../../../SPIRV-Headers") # used in subdirectory
-set(SPIRV-Tools_SOURCE_DIR "${CMAKE_CURRENT_SOURCE_DIR}/../../../SPIRV-Tools")
 set(SPIRV-Tools_OUTPUT_DIR "${IGC_OPTION__OUTPUT_DIR}/external/SPIRV-Tools/build")
 set(IGC_BUILD__SPIRV-Headers_DIR "${SPIRV-Headers_SOURCE_DIR}" PARENT_SCOPE)
 set(IGC_BUILD__SPIRV-Tools_DIR "${SPIRV-Tools_SOURCE_DIR}" PARENT_SCOPE)
-- 
2.34.0

