From 91ab320957b15188c63a280133a71ae3fdb2e758 Mon Sep 17 00:00:00 2001
From: Kushal Das <mail@kushaldas.in>
Date: Sun, 7 Jun 2020 13:11:55 +0530
Subject: [PATCH] Use double-conversion from the system

---
 setup.py | 10 ++++------
 1 file changed, 4 insertions(+), 6 deletions(-)

diff --git a/setup.py b/setup.py
index 87b96fa..47c409c 100644
--- a/setup.py
+++ b/setup.py
@@ -17,22 +17,20 @@ Programming Language :: Python :: 3.8
 Programming Language :: Python :: 3 :: Only
 """
 
-dconv_source_files = glob("./deps/double-conversion/double-conversion/*.cc")
-dconv_source_files.append("./lib/dconv_wrapper.cc")
 
 module1 = Extension(
     "ujson",
-    sources=dconv_source_files
-    + [
+    sources=[
+        "./lib/dconv_wrapper.cc",
         "./python/ujson.c",
         "./python/objToJSON.c",
         "./python/JSONtoObj.c",
         "./lib/ultrajsonenc.c",
         "./lib/ultrajsondec.c",
     ],
-    include_dirs=["./python", "./lib", "./deps/double-conversion/double-conversion"],
+    include_dirs=["./python", "./lib", "/usr/include/double-conversion"],
     extra_compile_args=["-D_GNU_SOURCE"],
-    extra_link_args=["-lstdc++", "-lm"],
+    extra_link_args=["-lstdc++", "-lm", "-ldouble-conversion"],
 )
 
 
-- 
2.26.2


