From 9123d05eea6cd66ec76fe30be26eeed6b17f9789 Mon Sep 17 00:00:00 2001
From: Pierre-Yves <pyu@riseup.net>
Date: Fri, 9 Jun 2017 22:56:48 +0200
Subject: [PATCH] Support stateless configuration and use it by default

Signed-off-by: Pierre-Yves <pyu@riseup.net>
---
 libinput-gestures       | 3 ++-
 libinput-gestures-setup | 5 +++--
 2 files changed, 5 insertions(+), 3 deletions(-)

diff --git a/libinput-gestures b/libinput-gestures
index d4c8979..8bb7112 100755
--- a/libinput-gestures
+++ b/libinput-gestures
@@ -10,7 +10,8 @@ PROG = Path(sys.argv[0]).name
 # Search first for user file then system file.
 CONFNAME = '{}.conf'.format(PROG)
 USERDIR = os.getenv('XDG_CONFIG_DIR', os.path.expanduser('~/.config'))
-CONFDIRS = (USERDIR, '/etc')
+VENDORDIR = '/usr/share/{}'.format(PROG)
+CONFDIRS = (USERDIR, '/etc', VENDORDIR)
 
 # Ratio of X/Y (or Y/X) at which we consider an oblique swipe gesture.
 # The number is the trigger angle in degrees.
diff --git a/libinput-gestures-setup b/libinput-gestures-setup
index 55802f0..8a179bc 100755
--- a/libinput-gestures-setup
+++ b/libinput-gestures-setup
@@ -12,6 +12,7 @@ ICODIR="$ICOBAS/128x128/apps"
 OCODIR="/usr/share/pixmaps"
 DOCDIR="/usr/share/doc/$NAME"
 CNFDIR="/etc"
+VNDDIR="/usr/share/$NAME"
 HCFDIR="${XDG_CONFIG_HOME:-$HOME/.config}"
 AUTDIR="$HCFDIR/autostart"
 
@@ -150,13 +151,13 @@ if [ $cmd = "install" -o $cmd = "uninstall" ]; then
 	install -CDv -m 755 -t $DESTDIR$BINDIR $NAME
 	install -CDv -m 644 -t $DESTDIR$APPDIR $NAME.desktop
 	install -CDv -m 644 -t $DESTDIR$ICODIR $NAME.png
-	install -CDv -m 644 -t $DESTDIR$CNFDIR $NAME.conf
+	install -CDv -m 644 -t $DESTDIR$VNDDIR $NAME.conf
 	install -CDv -m 644 -t $DESTDIR$DOCDIR README.md
     else
 	rm -rfv $DESTDIR$BINDIR/$NAME
 	rm -rfv $DESTDIR$APPDIR/$NAME.desktop
 	rm -rfv $DESTDIR$ICODIR/$NAME.png
-	rm -rfv $DESTDIR$CNFDIR/$NAME.conf
+	rm -rfv $DESTDIR$VNDDIR/$NAME.conf
 	rm -rfv $DESTDIR$DOCDIR
 	rm -rfv $DESTDIR$BINDIR/$NAME-setup
     fi
-- 
2.13.1

