From 8e62b454f7ab7827f022669bb7f586d7e1f9a91a Mon Sep 17 00:00:00 2001
From: taaem <taaem@mailbox.org>
Date: Wed, 4 Jan 2017 14:54:23 +0100
Subject: [PATCH] Support stateless configuration and use it by default

---
 libinput-gestures       | 3 ++-
 libinput-gestures-setup | 5 +++--
 2 files changed, 5 insertions(+), 3 deletions(-)

diff --git a/libinput-gestures b/libinput-gestures
index 97b16d8..f0dd589 100755
--- a/libinput-gestures
+++ b/libinput-gestures
@@ -10,7 +10,8 @@ PROG = Path(sys.argv[0]).name
 # Search first for user file then system file.
 CONFNAME = '{}.conf'.format(PROG)
 USERDIR = os.getenv('XDG_CONFIG_DIR', os.path.expanduser('~/.config'))
-CONFDIRS = (USERDIR, '/etc')
+VENDORDIR = '/usr/share/{}'.format(PROG)
+CONFDIRS = (USERDIR, '/etc', VENDORDIR)
 
 # Set up command line arguments
 opt = argparse.ArgumentParser(description=__doc__)
diff --git a/libinput-gestures-setup b/libinput-gestures-setup
index 173169a..d9e9cac 100755
--- a/libinput-gestures-setup
+++ b/libinput-gestures-setup
@@ -12,6 +12,7 @@ ICODIR="$ICOBAS/128x128/apps"
 OCODIR="/usr/share/pixmaps"
 DOCDIR="/usr/share/doc/$NAME"
 CNFDIR="/etc"
+VNDDIR="/usr/share/$NAME"
 AUTDIR="${XDG_CONFIG_HOME:-$HOME/.config}/autostart"
 
 usage() {
@@ -143,13 +144,13 @@ if [ $cmd = "install" -o $cmd = "uninstall" ]; then
 	install -CDv -m 755 -t $DESTDIR$BINDIR $NAME
 	install -CDv -m 644 -t $DESTDIR$APPDIR $NAME.desktop
 	install -CDv -m 644 -t $DESTDIR$ICODIR $NAME.png
-	install -CDv -m 644 -t $DESTDIR$CNFDIR $NAME.conf
+	install -CDv -m 644 -t $DESTDIR$VNDDIR $NAME.conf
 	install -CDv -m 644 -t $DESTDIR$DOCDIR README.md
     else
 	rm -rfv $DESTDIR$BINDIR/$NAME
 	rm -rfv $DESTDIR$APPDIR/$NAME.desktop
 	rm -rfv $DESTDIR$ICODIR/$NAME.png
-	rm -rfv $DESTDIR$CNFDIR/$NAME.conf
+	rm -rfv $DESTDIR$VNDDIR/$NAME.conf
 	rm -rfv $DESTDIR$DOCDIR
 	rm -rfv $DESTDIR$BINDIR/$NAME-setup
     fi
-- 
2.11.0

