From cd16b3bf3b1cae802cc8729826b5deaeb55f7216 Mon Sep 17 00:00:00 2001
From: Pierre-Yves <pyu@riseup.net>
Date: Sun, 27 Jan 2019 22:18:47 +0100
Subject: [PATCH] Support stateless configuration and use it by default

Signed-off-by: Pierre-Yves <pyu@riseup.net>
---
 libinput-gestures       | 3 ++-
 libinput-gestures-setup | 5 +++--
 2 files changed, 5 insertions(+), 3 deletions(-)

diff --git a/libinput-gestures b/libinput-gestures
index df923ad..cb5e789 100755
--- a/libinput-gestures
+++ b/libinput-gestures
@@ -13,7 +13,8 @@ PROGNAME = PROGPATH.stem
 # Search first for user file then system file.
 CONFNAME = '{}.conf'.format(PROGNAME)
 USERDIR = os.getenv('XDG_CONFIG_HOME', os.path.expanduser('~/.config'))
-CONFDIRS = (USERDIR, '/etc')
+VENDORDIR = '/usr/share/{}'.format(PROGNAME)
+CONFDIRS = (USERDIR, '/etc', VENDORDIR)
 
 # Ratio of X/Y (or Y/X) at which we consider an oblique swipe gesture.
 # The number is the trigger angle in degrees and set to 360/8/2.
diff --git a/libinput-gestures-setup b/libinput-gestures-setup
index db68018..b0cfd97 100755
--- a/libinput-gestures-setup
+++ b/libinput-gestures-setup
@@ -12,6 +12,7 @@ ICODIR="$ICOBAS/128x128/apps"
 OCODIR="/usr/share/pixmaps"
 DOCDIR="/usr/share/doc/$NAME"
 CNFDIR="/etc"
+VNDDIR="/usr/share/$NAME"
 HCFDIR="${XDG_CONFIG_HOME:-$HOME/.config}"
 AUTDIR="$HCFDIR/autostart"
 
@@ -158,7 +159,7 @@ if [[ $cmd == install || $cmd == uninstall ]]; then
 	install -CDv -m 755 -t $DESTDIR$BINDIR $NAME
 	install -CDv -m 644 -t $DESTDIR$APPDIR $NAME.desktop
 	install -CDv -m 644 -t $DESTDIR$ICODIR $NAME.png
-	install -CDv -m 644 -t $DESTDIR$CNFDIR $NAME.conf
+	install -CDv -m 644 -t $DESTDIR$VNDDIR $NAME.conf
 	install -CDv -m 644 -t $DESTDIR$DOCDIR README.md
 
 	# Also install HTML file if markdown is available
@@ -169,7 +170,7 @@ if [[ $cmd == install || $cmd == uninstall ]]; then
 	rm -rfv $DESTDIR$BINDIR/$NAME
 	rm -rfv $DESTDIR$APPDIR/$NAME.desktop
 	rm -rfv $DESTDIR$ICODIR/$NAME.png
-	rm -rfv $DESTDIR$CNFDIR/$NAME.conf
+	rm -rfv $DESTDIR$VNDDIR/$NAME.conf
 	rm -rfv $DESTDIR$DOCDIR
 	rm -rfv $DESTDIR$BINDIR/$NAME-setup
     fi
-- 
2.20.1

