From 3726badcb2b591cf0efb17d158a014ad849778de Mon Sep 17 00:00:00 2001
From: Peter O'Connor <peter@solus-project.com>
Date: Sat, 14 Apr 2018 20:08:46 +1000
Subject: [PATCH] Disable building unshipped code

Signed-off-by: Peter O'Connor <peter@solus-project.com>
---
 CMakeLists.txt | 3 ---
 1 file changed, 3 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index b0fad07..2012efd 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -203,11 +203,8 @@ endif()
 add_subdirectory(src)
 
 if (BUILD_EXAMPLES)
-    add_subdirectory(examples)
 endif()
 
-add_subdirectory(autotests)
-add_subdirectory(tests)
 add_subdirectory(templates)
 
 ################ create PlasmaConfig.cmake and install it ###########################
-- 
2.16.3

