From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Reilly Brogan <reilly@reillybrogan.com>
Date: Sat, 17 Feb 2024 15:01:35 -0600
Subject: [PATCH] Missing libs glslc

---
 libshaderc_util/CMakeLists.txt | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/libshaderc_util/CMakeLists.txt b/libshaderc_util/CMakeLists.txt
index 69ba519..5218688 100644
--- a/libshaderc_util/CMakeLists.txt
+++ b/libshaderc_util/CMakeLists.txt
@@ -45,8 +45,9 @@ target_include_directories(shaderc_util
 add_definitions(-DENABLE_HLSL)
 
 find_package(Threads)
+find_package(glslang)
 target_link_libraries(shaderc_util PRIVATE
-  glslang SPIRV
+  glslang glslang::MachineIndependent SPIRV
   SPIRV-Tools-opt ${CMAKE_THREAD_LIBS_INIT})
 
 shaderc_add_tests(
