From 9f33ec0be7bfb8a7265b54035d8cf2c8a83fd1d2 Mon Sep 17 00:00:00 2001
From: Algent Albrahimi <algent@protonmail.com>
Date: Tue, 5 Dec 2023 21:41:20 +0100
Subject: [PATCH] fix

Signed-off-by: Algent Albrahimi <algent@protonmail.com>
---
 PKG-INFO | 2 +-
 setup.py | 2 +-
 2 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/PKG-INFO b/PKG-INFO
index 0ee4db3..d3e8163 100644
--- a/PKG-INFO
+++ b/PKG-INFO
@@ -16,5 +16,5 @@ Classifier: Programming Language :: Python :: 3.6
 Classifier: Programming Language :: Python :: 3.7
 Classifier: Programming Language :: Python :: 3.8
 Classifier: Programming Language :: Python :: 3.9
-Requires-Dist: Pillow (>=9.2.0,<10.0.0)
+Requires-Dist: Pillow>=9.2.0
 Requires-Dist: regex
diff --git a/setup.py b/setup.py
index a7c1e38..bd84ef5 100644
--- a/setup.py
+++ b/setup.py
@@ -15,7 +15,7 @@ package_data = \
 {'': ['*']}
 
 install_requires = \
-['Pillow>=9.2.0,<10.0.0', 'regex']
+['Pillow>=9.2.0', 'regex']
 
 setup_kwargs = {
     'name': 'material-color-utilities-python',
-- 
2.43.0
