From eda931abbaa2bb8d6f5b827d8141214b56ee6047 Mon Sep 17 00:00:00 2001
From: Pierre-Yves <pyu@riseup.net>
Date: Sun, 15 Aug 2021 16:06:56 +0200
Subject: [PATCH] Fix .desktop file path in autostart.py

Signed-off-by: Pierre-Yves <pyu@riseup.net>
---
 src/vorta/autostart.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/vorta/autostart.py b/src/vorta/autostart.py
index d87dffb..0fc5572 100644
--- a/src/vorta/autostart.py
+++ b/src/vorta/autostart.py
@@ -36,7 +36,7 @@ def open_app_at_startup(enabled=True):
 
         is_flatpak = Path('/.flatpak-info').exists()
 
-        with open(Path(__file__).parent / "assets/metadata/com.borgbase.Vorta.desktop") as desktop_file:
+        with open("/usr/share/applications/com.borgbase.Vorta.desktop") as desktop_file:
             desktop_file_text = desktop_file.read()
 
         # Find XDG_CONFIG_HOME unless when running in flatpak
-- 
2.32.0

