From f09393c44ce3f4efaf3ceb5bc4decc7aa55302d7 Mon Sep 17 00:00:00 2001
From: Ikey Doherty <ikey@solus-project.com>
Date: Wed, 10 Aug 2016 03:16:07 +0100
Subject: [PATCH] Default to using overlayfs

Signed-off-by: Ikey Doherty <ikey@solus-project.com>
---
 contrib/init/systemd/docker.service | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/contrib/init/systemd/docker.service b/contrib/init/systemd/docker.service
index c3f3472..e0502eb 100644
--- a/contrib/init/systemd/docker.service
+++ b/contrib/init/systemd/docker.service
@@ -9,7 +9,7 @@ Type=notify
 # the default is not to use systemd for cgroups because the delegate issues still
 # exists and systemd currently does not support the cgroup feature set required
 # for containers run by docker
-ExecStart=/usr/bin/dockerd -H fd://
+ExecStart=/usr/bin/dockerd --storage-driver=overlay -H fd://
 ExecReload=/bin/kill -s HUP $MAINPID
 # Having non-zero Limit*s causes performance problems due to accounting overhead
 # in the kernel. We recommend using cgroups to do container-local accounting.
-- 
2.9.2

