--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -51,8 +51,10 @@ if(WITH_OPENMP)
         )
 endif()
 
+include(GNUInstallDirs)
+
 add_library(Library-C++
-    STATIC
+    SHARED
     src/cluster.cpp
     src/direct.cpp
     src/direct_tree.cpp
@@ -120,7 +122,7 @@ install(FILES
     DESTINATION include
     )
 install(TARGETS Library-C++
-    DESTINATION lib
+    DESTINATION ${CMAKE_INSTALL_LIBDIR}
     EXPORT fgt-targets
     )
 
@@ -162,9 +164,9 @@ write_basic_package_version_file(
 install(FILES
     ${PROJECT_BINARY_DIR}/fgt-config.cmake
     ${PROJECT_BINARY_DIR}/fgt-config-version.cmake
-    DESTINATION lib/cmake/fgt
+    DESTINATION ${CMAKE_INSTALL_LIBDIR}/cmake/fgt
     )
 install(EXPORT fgt-targets
-    DESTINATION lib/cmake/fgt
+    DESTINATION ${CMAKE_INSTALL_LIBDIR}/cmake/fgt
     NAMESPACE Fgt::
     )
