From f170b546d4579a5aa9c4195effa7051fe4cf4a9b Mon Sep 17 00:00:00 2001
From: Muhammad Alfi Syahrin <malfisya.dev@hotmail.com>
Date: Mon, 12 Feb 2024 07:04:22 +0700
Subject: [PATCH] build shared lib

---
 CMakeLists.txt | 8 +++++---
 1 file changed, 5 insertions(+), 3 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 42d6fcf..a01919b 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -65,7 +65,9 @@ if(WIN32)
         ${PROJECT_BINARY_DIR}/version.cpp
         )
 else()
+    include(GNUInstallDirs)
     add_library(Library-C++
+        SHARED
         src/cluster.cpp
         src/direct.cpp
         src/direct_tree.cpp
@@ -131,7 +133,7 @@ install(FILES
     DESTINATION include
     )
 install(TARGETS Library-C++
-    DESTINATION lib
+    DESTINATION ${CMAKE_INSTALL_LIBDIR}
     EXPORT fgt-targets
     )
 
@@ -173,9 +175,9 @@ write_basic_package_version_file(
 install(FILES
     ${PROJECT_BINARY_DIR}/fgt-config.cmake
     ${PROJECT_BINARY_DIR}/fgt-config-version.cmake
-    DESTINATION lib/cmake/fgt
+    DESTINATION ${CMAKE_INSTALL_LIBDIR}/cmake/fgt
     )
 install(EXPORT fgt-targets
-    DESTINATION lib/cmake/fgt
+    DESTINATION ${CMAKE_INSTALL_LIBDIR}/cmake/fgt
     NAMESPACE Fgt::
     )
-- 
2.43.0

