--- support-files/mariadb.service	2016-10-03 22:07:26.177912262 -0700
+++ support-files/mariadb.service.new	2016-11-21 19:00:49.688998123 -0800
@@ -79,7 +79,8 @@
 # Use the [service] section and Environment="MYSQLD_OPTS=...".
 # This isn't a replacement for my.cnf.
 # _WSREP_NEW_CLUSTER is for the exclusive use of the script galera_new_cluster
-ExecStart=/usr/sbin/mysqld $MYSQLD_OPTS $_WSREP_NEW_CLUSTER $_WSREP_START_POSITION
+ExecStart=/usr/sbin/mysqld $MYSQLD_OPTS $_WSREP_NEW_CLUSTER $_WSREP_START_POSITION --datadir=/var/db/mysql/data --plugin-dir=/usr/lib/mysql/plugin --basedir=/usr
+
 
 
 # Unset _WSREP_START_POSITION environment variable.
