From 9fb4636edf9bbc29374ec9b55bcd14a5e21b789e Mon Sep 17 00:00:00 2001
From: Hans K <hans@communitycomputing.net>
Date: Sun, 14 Jan 2024 15:59:23 -0500
Subject: [PATCH] Use different database

---
 pisi/constants.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/pisi/constants.py b/pisi/constants.py
index 98a68e5..f4c3f49 100644
--- a/pisi/constants.py
+++ b/pisi/constants.py
@@ -89,7 +89,7 @@ class Constants(metaclass=Singleton):
         self.__c.needs_restart = "needsrestart"
         self.__c.needs_reboot = "needsreboot"
         self.__c.auto_installed = "autoinstalled"
-        self.__c.files_db = "files.db"
+        self.__c.files_db = "files4.db"
         self.__c.repos = "repos"
         self.__c.devel_package_end = "-devel"
         self.__c.doc_package_end = "-docs?$"
-- 
2.43.0

