From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Reilly Brogan <reilly@reillybrogan.com>
Date: Mon, 15 Jul 2024 16:21:44 -0500
Subject: [PATCH] Solus: Add initrd folders for firmware detection

---
 drivers/base/firmware_loader/main.c | 6 +++++-
 1 file changed, 5 insertions(+), 1 deletion(-)

diff --git a/drivers/base/firmware_loader/main.c b/drivers/base/firmware_loader/main.c
index da8ca01d011c3..f93940f277e0e 100644
--- a/drivers/base/firmware_loader/main.c
+++ b/drivers/base/firmware_loader/main.c
@@ -474,7 +474,11 @@ static const char * const fw_path[] = {
 	"/lib/firmware/updates/" UTS_RELEASE,
 	"/lib/firmware/updates",
 	"/lib/firmware/" UTS_RELEASE,
-	"/lib/firmware"
+	"/lib/firmware",
+	"/kernel-firmware",
+	"/dracut-firmware-main",
+	"/dracut-firmware-nvidia-open",
+	"/dracut-firmware-nvidia-proprietary"
 };
 
 /*
