From 3b660f1af707c4ca5446d75934a9fa5286a2fbcd Mon Sep 17 00:00:00 2001
From: Joshua Strobl <joshua@streambits.io>
Date: Sat, 26 Dec 2020 23:00:33 +0200
Subject: [PATCH 1/1] Install to /usr/share/xdg instead.

---
 src/notify/meson.build | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/notify/meson.build b/src/notify/meson.build
index 8ef29ff7..f4fe0a3a 100644
--- a/src/notify/meson.build
+++ b/src/notify/meson.build
@@ -25,5 +25,5 @@ configure_file(
   output: '@BASENAME@',
   configuration: {'libexecdir': gdu_prefix / gdu_libexecdir},
   install: true,
-  install_dir: gdu_sysconfdir / 'xdg/autostart',
+  install_dir: '/usr/share/xdg/autostart',
 )
-- 
2.29.2

