From 89d3cd8a0e945e5669cf8d090900acbde92b90ef Mon Sep 17 00:00:00 2001
From: Bastien Dejean <nihilhill@gmail.com>
Date: Mon, 19 Aug 2019 09:42:27 +0200
Subject: [PATCH] bspc: Write failure messages to standard error

Fixes #999.
---
 src/bspc.c | 7 ++++---
 1 file changed, 4 insertions(+), 3 deletions(-)

diff --git a/src/bspc.c b/src/bspc.c
index ca91d076..c6e49f7a 100644
--- a/src/bspc.c
+++ b/src/bspc.c
@@ -94,11 +94,12 @@ int main(int argc, char *argv[])
 				rsp[nb] = '\0';
 				if (rsp[0] == FAILURE_MESSAGE[0]) {
 					ret = EXIT_FAILURE;
-					printf("%s", rsp + 1);
+					fprintf(stderr, "%s", rsp + 1);
+					fflush(stderr);
 				} else {
-					printf("%s", rsp);
+					fprintf(stdout, "%s", rsp);
+					fflush(stdout);
 				}
-				fflush(stdout);
 			} else {
 				break;
 			}
