From 5298a9121134103a81b9a47dfafd949e6350c55c Mon Sep 17 00:00:00 2001
From: Bastien Dejean <nihilhill@gmail.com>
Date: Mon, 3 Aug 2020 14:06:57 +0200
Subject: [PATCH] Arrange across all desktops when handling struts

Fixes #1165.
---
 src/window.c | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/src/window.c b/src/window.c
index a22b5e78..cd2340df 100644
--- a/src/window.c
+++ b/src/window.c
@@ -81,7 +81,9 @@ bool manage_window(xcb_window_t win, rule_consequence_t *csq, int fd)
 
 	if (!ignore_ewmh_struts && ewmh_handle_struts(win)) {
 		for (monitor_t *m = mon_head; m != NULL; m = m->next) {
-			arrange(m, m->desk);
+			for (desktop_t *d = m->desk_head; d != NULL; d = d->next) {
+				arrange(m, d);
+			}
 		}
 	}
 
