From e36a68f3a8d10715fd470b32a2cd0e85fc5f745b Mon Sep 17 00:00:00 2001
From: Bastien Dejean <nihilhill@gmail.com>
Date: Fri, 2 Aug 2019 12:18:22 +0200
Subject: [PATCH] Honor `single_monocle` in `set_hidden`

Fixes #990.
---
 src/tree.c | 8 ++++++++
 1 file changed, 8 insertions(+)

diff --git a/src/tree.c b/src/tree.c
index 6b522d14..2aa42ebe 100644
--- a/src/tree.c
+++ b/src/tree.c
@@ -1931,6 +1931,14 @@ void set_hidden(monitor_t *m, desktop_t *d, node_t *n, bool value)
 			activate_node(m, d, d->focus);
 		}
 	}
+
+	if (single_monocle) {
+		if (value && d->layout != LAYOUT_MONOCLE && tiled_count(d->root, true) <= 1) {
+			set_layout(m, d, LAYOUT_MONOCLE, false);
+		} else if (!value && d->layout == LAYOUT_MONOCLE && tiled_count(d->root, true) > 1) {
+			set_layout(m, d, d->user_layout, false);
+		}
+	}
 }
 
 void set_hidden_local(monitor_t *m, desktop_t *d, node_t *n, bool value)
