From f0558346b2f63373a8805e87b8bbf8c3b360e79a Mon Sep 17 00:00:00 2001
From: Bastien Dejean <nihilhill@gmail.com>
Date: Mon, 19 Aug 2019 16:08:08 +0200
Subject: [PATCH] Fix GCC warnings

---
 src/bspwm.c | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/src/bspwm.c b/src/bspwm.c
index dcdfdd5c..cda9482e 100644
--- a/src/bspwm.c
+++ b/src/bspwm.c
@@ -128,7 +128,9 @@ int main(int argc, char *argv[])
 		}
 
 		sock_address.sun_family = AF_UNIX;
-		snprintf(sock_address.sun_path, sizeof(sock_address.sun_path), "%s", socket_path);
+		if (snprintf(sock_address.sun_path, sizeof(sock_address.sun_path), "%s", socket_path) < 0) {
+			err("Couldn't write the socket path.\n");
+		}
 
 		sock_fd = socket(AF_UNIX, SOCK_STREAM, 0);
 
