From df7c6cc7813deec854922a5c2cc56a6b2e7cc8d2 Mon Sep 17 00:00:00 2001
From: Bastien Dejean <nihilhill@gmail.com>
Date: Mon, 26 Aug 2019 17:07:49 +0200
Subject: [PATCH] Pass `--tags` to `git describe`

Fixes #1003.
---
 Makefile | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/Makefile b/Makefile
index 69839391..9cbedc22 100644
--- a/Makefile
+++ b/Makefile
@@ -1,4 +1,4 @@
-VERCMD  ?= git describe 2> /dev/null
+VERCMD  ?= git describe --tags 2> /dev/null
 VERSION := $(shell $(VERCMD) || cat VERSION)
 
 CPPFLAGS += -D_POSIX_C_SOURCE=200809L -DVERSION=\"$(VERSION)\"
