From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Reilly Brogan <reilly@reillybrogan.com>
Date: Thu, 13 Jun 2024 01:34:56 -0500
Subject: [PATCH] Unpack package.json

---
 packages/insomnia/electron-builder.config.js | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/packages/insomnia/electron-builder.config.js b/packages/insomnia/electron-builder.config.js
index 39c421a8..02f032b9 100644
--- a/packages/insomnia/electron-builder.config.js
+++ b/packages/insomnia/electron-builder.config.js
@@ -10,6 +10,9 @@ const BINARY_PREFIX = 'Insomnia.Core';
 const config = {
   npmRebuild: false,
   appId: 'com.insomnia.app',
+  asarUnpack: [
+    'package.json',
+  ],
   protocols: [
     {
       name: 'Insomnia',
