From 0a4041a91a0d3d86587451293dae80d626c92d60 Mon Sep 17 00:00:00 2001
From: Joshua Strobl <joshua@streambits.io>
Date: Wed, 21 Oct 2020 14:56:20 +0300
Subject: [PATCH 1/1] Skip Node version check

---
 packages/insomnia-app/scripts/build.js | 5 -----
 1 file changed, 5 deletions(-)

diff --git a/packages/insomnia-app/scripts/build.js b/packages/insomnia-app/scripts/build.js
index dae2894..3fb2568 100644
--- a/packages/insomnia-app/scripts/build.js
+++ b/packages/insomnia-app/scripts/build.js
@@ -56,11 +56,6 @@ module.exports.start = async function(forceFromGitRef) {
   console.log(`[build] npm: ${childProcess.spawnSync('npm', ['--version']).stdout}`.trim());
   console.log(`[build] node: ${childProcess.spawnSync('node', ['--version']).stdout}`.trim());
 
-  if (process.version.indexOf('v12.') !== 0) {
-    console.log('[build] Node v12.x.x is required to build');
-    process.exit(1);
-  }
-
   // Remove folders first
   console.log('[build] Removing existing directories');
   await emptyDir(buildFolder);
-- 
2.28.0

