From 5e30d1a7fee3110292956359d3451c93504ee98a Mon Sep 17 00:00:00 2001
From: Julien Guillot <julienguillot@protonmail.com>
Date: Wed, 30 Jun 2021 18:48:29 +0200
Subject: [PATCH 1/1] Skip Node version check

---
 packages/insomnia-app/scripts/build.ts | 5 -----
 1 file changed, 5 deletions(-)

diff --git a/packages/insomnia-app/scripts/build.ts b/packages/insomnia-app/scripts/build.ts
index 91004448..f8bd4001 100644
--- a/packages/insomnia-app/scripts/build.ts
+++ b/packages/insomnia-app/scripts/build.ts
@@ -212,11 +212,6 @@ export const start = async ({ forceFromGitRef }: { forceFromGitRef: boolean }) =
   console.log(`[build] npm: ${childProcess.spawnSync('npm', ['--version']).stdout}`.trim());
   console.log(`[build] node: ${childProcess.spawnSync('node', ['--version']).stdout}`.trim());
 
-  if (process.version.indexOf('v12.') !== 0) {
-    console.log('[build] Node v12.x.x is required to build');
-    process.exit(1);
-  }
-
   const buildFolder = path.join('../build');
 
   // Remove folders first
-- 
2.32.0
