From 9ee9c9ea35571625cecdaa582d781aad380651f9 Mon Sep 17 00:00:00 2001
From: Fabio Forni <livingsilver94.solus@redaril.me>
Date: Fri, 31 Jul 2020 23:00:44 +0200
Subject: [PATCH] Skip Node version check

---
 packages/insomnia-app/scripts/build.js | 5 -----
 1 file changed, 5 deletions(-)

diff --git a/packages/insomnia-app/scripts/build.js b/packages/insomnia-app/scripts/build.js
index a57f2156..c07ec9a3 100644
--- a/packages/insomnia-app/scripts/build.js
+++ b/packages/insomnia-app/scripts/build.js
@@ -50,11 +50,6 @@ module.exports.start = async function(forcedVersion = null) {
   console.log(`[build] npm: ${childProcess.spawnSync('npm', ['--version']).stdout}`.trim());
   console.log(`[build] node: ${childProcess.spawnSync('node', ['--version']).stdout}`.trim());
 
-  if (process.version.indexOf('v10.') !== 0) {
-    console.log('[build] Node v10.x.x is required to build');
-    process.exit(1);
-  }
-
   // Remove folders first
   console.log('[build] Removing existing directories');
   await emptyDir('../build');
-- 
2.27.0

