From fe0d7c30946c3bf934ffbd9921a9ea83a727b176 Mon Sep 17 00:00:00 2001
From: Silke Hofstra <silke@slxh.eu>
Date: Mon, 28 Aug 2017 13:32:11 +0200
Subject: [PATCH] Add dependency to network-online to rpc-gssd

---
 systemd/rpc-gssd.service.in | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/systemd/rpc-gssd.service.in b/systemd/rpc-gssd.service.in
index b353027..37c834f 100644
--- a/systemd/rpc-gssd.service.in
+++ b/systemd/rpc-gssd.service.in
@@ -2,8 +2,8 @@
 Description=RPC security service for NFS client and server
 DefaultDependencies=no
 Conflicts=umount.target
-Requires=var-lib-nfs-rpc_pipefs.mount
-After=var-lib-nfs-rpc_pipefs.mount
+Requires=var-lib-nfs-rpc_pipefs.mount network-online.target
+After=var-lib-nfs-rpc_pipefs.mount network-online.target
 
 ConditionPathExists=@_sysconfdir@/krb5.keytab
 
-- 
2.14.1

