From 616cceb8b2502aeb6d9e79e725b8261dd4e95967 Mon Sep 17 00:00:00 2001
From: Fabio Forni <livingsilver94.solus@redaril.me>
Date: Tue, 11 Aug 2020 18:36:36 +0200
Subject: [PATCH] Use system theme

---
 picard/ui/theme.py | 5 -----
 1 file changed, 5 deletions(-)

diff --git a/picard/ui/theme.py b/picard/ui/theme.py
index 00d2efb..0890e9f 100644
--- a/picard/ui/theme.py
+++ b/picard/ui/theme.py
@@ -53,11 +53,6 @@ dark_syntax_theme = SyntaxTheme(
 
 class BaseTheme:
     def setup(self, app):
-        # Use the new fusion style from PyQt5 for a modern and consistent look
-        # across all OSes.
-        if not IS_MACOS and not IS_HAIKU:
-            app.setStyle('Fusion')
-
         app.setStyleSheet(
             'QGroupBox::title { /* PICARD-1206, Qt bug workaround */ }'
         )
-- 
2.27.0

