From 11f2fa1fbefeb7ea750758dd191ccc5f62ac04b6 Mon Sep 17 00:00:00 2001
From: Fabio Forni <livingsilver94.solus@redaril.me>
Date: Wed, 20 May 2020 22:30:27 +0200
Subject: [PATCH] Use system theme

---
 picard/tagger.py | 5 -----
 1 file changed, 5 deletions(-)

diff --git a/picard/tagger.py b/picard/tagger.py
index 0d28264..b6b045e 100644
--- a/picard/tagger.py
+++ b/picard/tagger.py
@@ -161,11 +161,6 @@ class Tagger(QtWidgets.QApplication):
 
     def __init__(self, picard_args, unparsed_args, localedir, autoupdate):
 
-        # Use the new fusion style from PyQt5 for a modern and consistent look
-        # across all OSes.
-        if not IS_MACOS and not IS_HAIKU:
-            self.setStyle('Fusion')
-
         super().__init__(sys.argv)
         self.__class__.__instance = self
         config._setup(self, picard_args.config_file)
-- 
2.26.2

