From 0721ec78603080a3aa5a45642d70af9329f7f05f Mon Sep 17 00:00:00 2001
From: Fabio Forni <livingsilver94.solus@redaril.me>
Date: Sat, 24 Oct 2020 13:39:54 +0200
Subject: [PATCH] Use system theme

---
 picard/ui/options/interface.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/picard/ui/options/interface.py b/picard/ui/options/interface.py
index d9401d0..8b2437c 100644
--- a/picard/ui/options/interface.py
+++ b/picard/ui/options/interface.py
@@ -135,7 +135,7 @@ class InterfaceOptionsPage(OptionsPage):
         config.BoolOption("setting", "use_adv_search_syntax", False),
         config.BoolOption("setting", "quit_confirmation", True),
         config.TextOption("setting", "ui_language", ""),
-        config.BoolOption("setting", "use_system_theme", False),
+        config.BoolOption("setting", "use_system_theme", True),
         config.BoolOption("setting", "starting_directory", False),
         config.TextOption("setting", "starting_directory_path", _default_starting_dir),
         config.TextOption("setting", "load_image_behavior", "append"),
-- 
2.29.0

