From 07268f85acd6f840ecbf33d94316d1f69d294203 Mon Sep 17 00:00:00 2001
From: Fabio Forni <livingsilver94.solus@redaril.me>
Date: Thu, 17 Dec 2020 23:54:31 +0100
Subject: [PATCH] Use system theme

---
 picard/ui/options/interface.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/picard/ui/options/interface.py b/picard/ui/options/interface.py
index 1a1a8fe..1e3abfe 100644
--- a/picard/ui/options/interface.py
+++ b/picard/ui/options/interface.py
@@ -135,7 +135,7 @@ class InterfaceOptionsPage(OptionsPage):
         config.BoolOption("setting", "use_adv_search_syntax", False),
         config.BoolOption("setting", "quit_confirmation", True),
         config.TextOption("setting", "ui_language", ""),
-        config.BoolOption("setting", "use_system_theme", False),
+        config.BoolOption("setting", "use_system_theme", True),
         config.BoolOption("setting", "filebrowser_horizontal_autoscroll", True),
         config.BoolOption("setting", "starting_directory", False),
         config.TextOption("setting", "starting_directory_path", _default_starting_dir),
-- 
2.29.2

