From e08574e13e26f6994d06de42c1efffc978d500a3 Mon Sep 17 00:00:00 2001
From: Fabio Forni <livingsilver94.solus@redaril.me>
Date: Tue, 20 Apr 2021 21:51:27 +0200
Subject: [PATCH] Use system theme

---
 picard/ui/options/interface.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/picard/ui/options/interface.py b/picard/ui/options/interface.py
index 81fad103..07272dab 100644
--- a/picard/ui/options/interface.py
+++ b/picard/ui/options/interface.py
@@ -141,7 +141,7 @@ class InterfaceOptionsPage(OptionsPage):
         BoolOption("setting", "use_adv_search_syntax", False),
         BoolOption("setting", "quit_confirmation", True),
         TextOption("setting", "ui_language", ""),
-        TextOption("setting", "ui_theme", str(UiTheme.DEFAULT)),
+        TextOption("setting", "ui_theme", str(UiTheme.SYSTEM)),
         BoolOption("setting", "filebrowser_horizontal_autoscroll", True),
         BoolOption("setting", "starting_directory", False),
         TextOption("setting", "starting_directory_path", _default_starting_dir),
-- 
2.30.2

