diff --git a/kitty/desktop.c b/kitty/desktop.c
index 5084ad279..1a788ed51 100644
--- a/kitty/desktop.c
+++ b/kitty/desktop.c
@@ -113,8 +113,8 @@ load_libcanberra(void) {
     if (done) return;
     done = true;
     libcanberra_handle = dlopen(libname, RTLD_LAZY);
-    if (libcanberra_handle == NULL) libsn_handle = dlopen(libname2, RTLD_LAZY);
-    if (libcanberra_handle == NULL) libsn_handle = dlopen(libname3, RTLD_LAZY);
+    if (libcanberra_handle == NULL) libcanberra_handle = dlopen(libname2, RTLD_LAZY);
+    if (libcanberra_handle == NULL) libcanberra_handle = dlopen(libname3, RTLD_LAZY);
     if (libcanberra_handle == NULL) {
         fprintf(stderr, "Failed to load %s, cannot play beep sound, with error: %s\n", libname, dlerror());
         return;