diff --git a/src/util/gobjectptr.h b/src/util/gobjectptr.h
index 1f4f37136de055c20b6bec367305583f5a5a1601..aa142e765bbabc6671398bbf2da1a3fbe3bcfe9d 100644
--- a/src/util/gobjectptr.h
+++ b/src/util/gobjectptr.h
@@ -20,7 +20,7 @@ public:
     GObjectPtr() = default;
     explicit GObjectPtr(T *p, bool add_ref = false) : _p(p) { if (add_ref) _ref(); }
     GObjectPtr(GObjectPtr const &other) : _p(other._p) { _ref(); }
-    GObjectPtr &operator=(GObjectPtr const &other) { if (&other != this) { _unref(); _p = other.p; _ref(); } return *this; }
+    GObjectPtr &operator=(GObjectPtr const &other) { if (&other != this) { _unref(); _p = other._p; _ref(); } return *this; }
     GObjectPtr(GObjectPtr &&other) noexcept : _p(other._p) { other._p = nullptr; }
     GObjectPtr &operator=(GObjectPtr &&other) { if (&other != this) { _unref(); _p = other._p; other._p = nullptr; } return *this; }
     ~GObjectPtr() { _unref(); }
