diff -up inkscape-1.0.1_2020-09-07_3bc2e813f5/src/3rdparty/autotrace/autotrace.h.glib-extern inkscape-1.0.1_2020-09-07_3bc2e813f5/src/3rdparty/autotrace/autotrace.h
--- inkscape-1.0.1_2020-09-07_3bc2e813f5/src/3rdparty/autotrace/autotrace.h.glib-extern	2021-02-16 12:59:00.917852141 +0100
+++ inkscape-1.0.1_2020-09-07_3bc2e813f5/src/3rdparty/autotrace/autotrace.h	2021-02-16 12:59:06.675894640 +0100
@@ -22,6 +22,7 @@
 #define AUTOTRACE_H
 
 #include <stdio.h>
+#include "types.h"
 
 #ifdef __cplusplus
 extern "C" {
@@ -35,7 +36,6 @@ extern "C" {
  * Typedefs
  * ===================================================================== */
 
-#include "types.h"
 #include "color.h"
 
 /* Third degree is the highest we deal with.  */
