diff --git a/src/extension/internal/pdfinput/pdf-parser.cpp b/src/extension/internal/pdfinput/pdf-parser.cpp
index 28000a87b0c8550ada31dd7cb576f87c522f2160..9ea30b90a481198d49793e41474cc41ecda24bee 100644
--- a/src/extension/internal/pdfinput/pdf-parser.cpp
+++ b/src/extension/internal/pdfinput/pdf-parser.cpp
@@ -2403,6 +2403,7 @@ void PdfParser::doImage(Object * /*ref*/, Stream *str, GBool inlineImg)
     int bits;
     GBool interpolate;
     StreamColorSpaceMode csMode;
+    GBool hasAlpha;
     GBool mask;
     GBool invert;
     Object maskObj, smaskObj;
@@ -2414,7 +2415,8 @@ void PdfParser::doImage(Object * /*ref*/, Stream *str, GBool inlineImg)
     // get info from the stream
     bits = 0;
     csMode = streamCSNone;
-    str->getImageParams(&bits, &csMode);
+    hasAlpha = false;
+    str->_POPPLER_GET_IMAGE_PARAMS(&bits, &csMode, &hasAlpha);
     
     // get stream dict
     dict = str->getDict();
diff --git a/src/extension/internal/pdfinput/poppler-transition-api.h b/src/extension/internal/pdfinput/poppler-transition-api.h
index 8f03aa17779700c47989e3bb3122ccfe9d0a4d04..b7a54828e749d3db00eaffebc56ddb79ac45e00a 100644
--- a/src/extension/internal/pdfinput/poppler-transition-api.h
+++ b/src/extension/internal/pdfinput/poppler-transition-api.h
@@ -39,6 +39,12 @@
 #define _POPPLER_FUNCTION_TYPE_STITCHING 3
 #endif
 
+#if POPPLER_CHECK_VERSION(24,12,0)
+#define _POPPLER_GET_IMAGE_PARAMS(bits, csMode, hasAlpha) getImageParams(bits, csMode, hasAlpha)
+#else
+#define _POPPLER_GET_IMAGE_PARAMS(bits, csMode, hasAlpha) getImageParams(bits, csMode)
+#endif
+
 #if POPPLER_CHECK_VERSION(22, 4, 0)
 #define _POPPLER_FONTPTR_TO_GFX8(font_ptr) ((Gfx8BitFont *)font_ptr.get())
 #else
