diff --git a/src/extension/internal/pdfinput/pdf-parser.h b/src/extension/internal/pdfinput/pdf-parser.h
index 8325ea24364fc08edcffd891e4f01ac41fa6c79a..7d9189dfc0a0845b27cc5aa599dd5c9cc0d8c4b4 100644
--- a/src/extension/internal/pdfinput/pdf-parser.h
+++ b/src/extension/internal/pdfinput/pdf-parser.h
@@ -24,13 +24,12 @@
 #pragma interface
 #endif
 
-#include "glib/poppler-features.h"
-#include "Object.h"
-
+#include <2geom/affine.h>
+#include <glib/poppler-features.h>
 #include <map>
 #include <memory>
+#include <poppler/Object.h>
 #include <string>
-#include <2geom/affine.h>
 
 #define Operator Operator_Gfx
 #include <poppler/Gfx.h>
