From af5cf22793ecea7725f8da502ec36edcfb34e761 Mon Sep 17 00:00:00 2001
From: Peter O'Connor <peter@solus-project.com>
Date: Tue, 3 Oct 2017 11:44:06 +1100
Subject: [PATCH] Disable ICMP echo address checking

Signed-off-by: Peter O'Connor <peter@solus-project.com>
---
 src/option.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/src/option.c b/src/option.c
index f4d22a2..660b791 100644
--- a/src/option.c
+++ b/src/option.c
@@ -4658,6 +4658,8 @@ void read_opts(int argc, char **argv, char *compile_opts)
   add_txt("servers.bind", NULL, TXT_STAT_SERVERS);
 #endif
 
+  set_option_bool(OPT_NO_PING);
+
   while (1) 
     {
 #ifdef HAVE_GETOPT_LONG
-- 
2.14.2

