From d7aa8222e4e51dcc7cc44666d0893ac236b13828 Mon Sep 17 00:00:00 2001
From: Peter O'Connor <peter@solus-project.com>
Date: Tue, 3 Oct 2017 11:44:06 +1100
Subject: [PATCH] Disable ICMP echo address checking

Signed-off-by: Peter O'Connor <peter@solus-project.com>
---
 src/option.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/src/option.c b/src/option.c
index e4810fd..b5ec8b2 100644
--- a/src/option.c
+++ b/src/option.c
@@ -5876,6 +5876,8 @@ void read_opts(int argc, char **argv, char *compile_opts)
     }
 #endif
 
+  set_option_bool(OPT_NO_PING);
+
   /* port might not be known when the address is parsed - fill in here */
   if (daemon->servers)
     {
-- 
2.43.0

