From c195d3bf24a69cf2591f170089bda94bb9b060a0 Mon Sep 17 00:00:00 2001
From: Fabio Forni <livingsilver94.solus@redaril.me>
Date: Tue, 11 Aug 2020 15:43:29 +0200
Subject: [PATCH] Make it stateless

---
 src/config.h |  8 ++++++++
 src/option.c | 15 ++++++++++++++-
 2 files changed, 22 insertions(+), 1 deletion(-)

diff --git a/src/config.h b/src/config.h
index 1e7b30f..46d2bb0 100644
--- a/src/config.h
+++ b/src/config.h
@@ -165,6 +165,12 @@ RESOLVFILE
    the default locations of these files are determined below, but may be overridden 
    in a build command line using COPTS.
 
+DEFAULTCONFFILE
+   define a fallback config file path in case CONFFILE does not exist
+   or is not readable. This default config file will make dnsmasq die
+   if, in turn, it is not readable.
+   This options is suitable for so-called stateless installations.
+
 */
 
 /* Defining this builds a binary which handles time differently and works better on a system without a 
@@ -239,6 +245,8 @@ RESOLVFILE
 #    endif
 #endif
 
+#define DEFAULTCONFFILE "/usr/share/defaults/dnsmasq/dnsmasq.conf"
+
 /* platform dependent options: these are determined automatically below
 
 HAVE_LINUX_NETWORK
diff --git a/src/option.c b/src/option.c
index b5ec8b2..bf7430a 100644
--- a/src/option.c
+++ b/src/option.c
@@ -397,6 +397,11 @@ static struct {
   { 'b', OPT_BOGUSPRIV, NULL, gettext_noop("Fake reverse lookups for RFC1918 private address ranges."), NULL },
   { 'B', ARG_DUP, "<ipaddr>", gettext_noop("Treat ipaddr as NXDOMAIN (defeats Verisign wildcard)."), NULL }, 
   { 'c', ARG_ONE, "<integer>", gettext_noop("Specify the size of the cache in entries (defaults to %s)."), "$" },
+  #ifdef DEFAULTCONFFILE
+  { 'C', ARG_DUP, "<path>", gettext_noop("Specify config file (defaults to %s, then "DEFAULTCONFFILE")."), CONFFILE },
+  #else
+  { 'C', ARG_DUP, "<path>", gettext_noop("Specify configuration file (defaults to %s)."), CONFFILE },
+  #endif
   { 'C', ARG_DUP, "<path>", gettext_noop("Specify configuration file (defaults to %s)."), CONFFILE },
   { 'd', OPT_DEBUG, NULL, gettext_noop("Do NOT fork into the background: run in debug mode."), NULL },
   { 'D', OPT_NODOTS_LOCAL, NULL, gettext_noop("Do NOT forward queries with no domain part."), NULL }, 
@@ -5855,7 +5860,15 @@ void read_opts(int argc, char **argv, char *compile_opts)
       free(conffile);
     }
   else
-    one_file(CONFFILE, LOPT_CONF_OPT);
+    {
+      one_file(CONFFILE, LOPT_CONF_OPT);
+      #ifdef DEFAULTCONFFILE
+      if (errno == ENOENT)
+        {
+          one_file(DEFAULTCONFFILE, 0);
+        }
+      #endif
+    }
 
   /* Add TXT records if wanted */
 #ifndef NO_ID
-- 
2.43.0

