diff --git a/src/dhcp.c b/src/dhcp.c
index e6fceb1..3189acf 100644
--- a/src/dhcp.c
+++ b/src/dhcp.c
@@ -120,7 +120,7 @@ void dhcp_init(void)
      we drop root. Also, set buffer size small, to avoid wasting
      kernel buffers */
   
-  if (option_bool(OPT_NO_PING))
+  if (/*option_bool(OPT_NO_PING)*/ 1)
     daemon->dhcp_icmp_fd = -1;
   else if ((daemon->dhcp_icmp_fd = make_icmp_sock()) == -1 ||
 	   setsockopt(daemon->dhcp_icmp_fd, SOL_SOCKET, SO_RCVBUF, &oneopt, sizeof(oneopt)) == -1 )
@@ -715,7 +715,7 @@ int address_allocate(struct dhcp_context *context,
 
 		if (!r) 
 		  {
-		    if ((count < max) && !option_bool(OPT_NO_PING) && icmp_ping(addr))
+		    if ((count < max) && ! /*option_bool(OPT_NO_PING)*/ 1 && icmp_ping(addr))
 		      {
 			/* address in use: perturb address selection so that we are
 			   less likely to try this address again. */
