From b319432897fc1ca6bf5395a5292a667617837325 Mon Sep 17 00:00:00 2001
From: Fabio Forni <livingsilver94.solus@redaril.me>
Date: Sun, 22 Aug 2021 11:25:32 +0200
Subject: [PATCH] Use AyatanaAppIndicator

---
 configure.ac | 10 +++++-----
 1 file changed, 5 insertions(+), 5 deletions(-)

diff --git a/configure.ac b/configure.ac
index 38347f6..d399a7f 100644
--- a/configure.ac
+++ b/configure.ac
@@ -110,16 +110,16 @@ fi
 AM_CONDITIONAL(HAVE_ZEITGEIST, test "x$enable_zeitgeist" = "xyes")
 
 if test "x$enable_indicator" = "xauto" ; then
-	PKG_CHECK_EXISTS([appindicator3-0.1],
+	PKG_CHECK_EXISTS([ayatana-appindicator3-0.1],
 	                 enable_indicator="yes",
 	                 enable_indicator="no")
 fi
 
 if test "x$enable_indicator" = "xyes" ; then
-	PKG_CHECK_EXISTS([appindicator3-0.1],,
-	                 AC_MSG_ERROR([appindicator3-0.1 is not installed]))
-	PKG_CHECK_MODULES(APP_INDICATOR, appindicator3-0.1)
-	APP_INDICATOR_VALAFLAGS=" --pkg appindicator3-0.1 -D HAVE_INDICATOR"
+	PKG_CHECK_EXISTS([ayatana-appindicator3-0.1],,
+	                 AC_MSG_ERROR([ayatana-appindicator3-0.1 is not installed]))
+	PKG_CHECK_MODULES(APP_INDICATOR, ayatana-appindicator3-0.1)
+	APP_INDICATOR_VALAFLAGS=" --pkg ayatana-appindicator3-0.1 -D HAVE_INDICATOR"
 	AC_SUBST(APP_INDICATOR_VALAFLAGS)
 	AC_DEFINE(HAVE_INDICATOR, 1, [Have AppIndicator])
 fi
-- 
2.32.0

