From 69fa6439eebff9bfda05c35a836ec89915f21a31 Mon Sep 17 00:00:00 2001
From: "F. von Gellhorn" <flinux@vongellhorn.ch>
Date: Sun, 16 Feb 2020 14:04:24 +0100
Subject: [PATCH 1/1] Set Breeze-Dark as default gtk theme

---
 kconf_update/gtktheme.cpp | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/kconf_update/gtktheme.cpp b/kconf_update/gtktheme.cpp
index bf9993c..8552c4f 100644
--- a/kconf_update/gtktheme.cpp
+++ b/kconf_update/gtktheme.cpp
@@ -38,7 +38,7 @@ void upgradeGtk2Theme() {
      || currentGtk2Theme == QStringLiteral("BreezyGTK")
      || currentGtk2Theme == QStringLiteral("Orion")
     ) {
-        ConfigEditor::setGtk2ConfigValue(QStringLiteral("gtk-theme-name"), QStringLiteral("Breeze"));
+        ConfigEditor::setGtk2ConfigValue(QStringLiteral("gtk-theme-name"), QStringLiteral("Breeze-Dark"));
     }
 }
 
@@ -50,7 +50,7 @@ void upgradeGtk3Theme() {
      || currentGtk3Theme == QStringLiteral("Orion")
     ) {
         ConfigEditor::setGtk3ConfigValueDconf(QStringLiteral("gtk-theme"), QStringLiteral("Breeze"));
-        ConfigEditor::setGtk3ConfigValueSettingsIni(QStringLiteral("gtk-theme-name"), QStringLiteral("Breeze"));
+        ConfigEditor::setGtk3ConfigValueSettingsIni(QStringLiteral("gtk-theme-name"), QStringLiteral("Breeze-Dark"));
         ConfigEditor::setGtk3ConfigValueXSettingsd(QStringLiteral("Net/ThemeName"),  QStringLiteral("Breeze"));
     }
 }
-- 
2.25.0

