From 00e9749f96c03ae441adea6e9e1b1262204f9a6f Mon Sep 17 00:00:00 2001
From: Joey Riches <josephriches@gmail.com>
Date: Tue, 18 Jul 2023 22:39:52 +0100
Subject: [PATCH 1/1] Fix wxwidgets GTK3 build issue

---
 src/viacamcontroller.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/viacamcontroller.cpp b/src/viacamcontroller.cpp
index 27c80a3..e6ee497 100755
--- a/src/viacamcontroller.cpp
+++ b/src/viacamcontroller.cpp
@@ -229,7 +229,7 @@ CCamera* CViacamController::SetUpCamera()
 
 			wxSingleChoiceDialog choiceDlg(
 				NULL, _("Choose the camera to use"), _T("Enable Viacam"), strArray, 
-				(char**)NULL, wxDEFAULT_DIALOG_STYLE | wxOK | wxCANCEL | wxCENTRE);
+				(void**)NULL, wxDEFAULT_DIALOG_STYLE | wxOK | wxCANCEL | wxCENTRE);
 
 			if (choiceDlg.ShowModal ()!= wxID_OK) return NULL;
 
-- 
2.41.0

