From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Reilly Brogan <reilly@reillybrogan.com>
Date: Wed, 25 Jun 2025 23:08:59 -0500
Subject: [PATCH] ftbfs gettext

---
 configure.ac | 1 +
 1 file changed, 1 insertion(+)

diff --git a/configure.ac b/configure.ac
index d4ca1aa..16558ab 100644
--- a/configure.ac
+++ b/configure.ac
@@ -31,6 +31,7 @@ AC_PROG_MKDIR_P
 ALL_LINGUAS="ca es gl it tr ast de oc fr ar he ru pt_BR ja el lt zh_TW zh_CN"
 
 AM_GNU_GETTEXT([external])
+AM_GNU_GETTEXT_VERSION([0.24])
 
 # ----------------- Debug mode switch --------------------
 AC_ARG_ENABLE(debug,
