From d9c92d3f542f578682bbd3bea9b6897cedebada1 Mon Sep 17 00:00:00 2001
From: Joey Riches <josephriches@gmail.com>
Date: Tue, 25 Jul 2023 11:33:29 +0100
Subject: [PATCH 1/1] Run on x11

---
 src/eviacam.desktop | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/eviacam.desktop b/src/eviacam.desktop
index 46a2a93..2fba3a1 100755
--- a/src/eviacam.desktop
+++ b/src/eviacam.desktop
@@ -22,5 +22,5 @@ Keywords[ca]=Accessibilitat;Ratolí de cap;webcam;Ratolí Facial;ratolí
 Type=Application
 Terminal=false
 Icon=eviacam
-Exec=eviacamloader
+Exec=env GDK_BACKEND=x11 eviacamloader
 Categories=GTK;Utility;Accessibility;
-- 
2.41.0

