From 3983dfdf2db1a86adca79f37fc85fd2506dafe1e Mon Sep 17 00:00:00 2001
From: Joey Riches <josephriches@gmail.com>
Date: Tue, 18 Jul 2023 22:26:30 +0100
Subject: [PATCH 1/1] wviacam: Fix lang redeclaration with wx gtk3

---
 src/wviacam.cpp | 1 -
 1 file changed, 1 deletion(-)

diff --git a/src/wviacam.cpp b/src/wviacam.cpp
index f6e45bf..b3afd30 100644
--- a/src/wviacam.cpp
+++ b/src/wviacam.cpp
@@ -598,7 +598,6 @@ void WViacam::OnToolHelpClick( wxCommandEvent& event )
 		case (wxLANGUAGE_SPANISH_GUATEMALA):
 		case (wxLANGUAGE_SPANISH_HONDURAS):
 		case (wxLANGUAGE_SPANISH_MEXICAN):
-		case (wxLANGUAGE_SPANISH_MODERN):
 		case (wxLANGUAGE_SPANISH_NICARAGUA):
 		case (wxLANGUAGE_SPANISH_PANAMA):
 		case (wxLANGUAGE_SPANISH_PARAGUAY):
-- 
2.41.0

