diff --git a/ImageLounge/xgd-data/org.nomacs.ImageLounge.appdata.xml.in b/ImageLounge/xgd-data/org.nomacs.ImageLounge.appdata.xml.in
index 13449d7..deacd9c 100644
--- a/ImageLounge/xgd-data/org.nomacs.ImageLounge.appdata.xml.in
+++ b/ImageLounge/xgd-data/org.nomacs.ImageLounge.appdata.xml.in
@@ -33,11 +33,11 @@
     </content_rating>
 
     <screenshots>
-        <screenshot type="default">https://nomacs.org/wp-content/uploads/2020/03/img-2020-03-18-11.49.26-1.png</screenshot>
-        <screenshot type="default">https://nomacs.org/wp-content/uploads/2020/03/img-2020-03-18-11.59.10.png</screenshot>
+        <screenshot type="default">https://github.com/flathub/org.nomacs.ImageLounge/raw/4521f054b41049aca4cbaf7d1b1f051d3d53f77a/screenshots/img-2020-03-18-11.49.26-1.png</screenshot>
+        <screenshot type="default">https://github.com/flathub/org.nomacs.ImageLounge/raw/4521f054b41049aca4cbaf7d1b1f051d3d53f77a/screenshots/img-2020-03-18-11.59.10.png</screenshot>
     </screenshots>
 
     <releases>
         <release version="@NOMACS_VERSION@" date="@TODAY@"/>
     </releases>
-</component>
\ No newline at end of file
+</component>
