From bc2f8c194f1bf8486ba60d5c7495954c81d2652d Mon Sep 17 00:00:00 2001
From: Ikey Doherty <ikey@solus-project.com>
Date: Thu, 26 May 2016 05:28:35 +0100
Subject: [PATCH] Support a stateless configuration

Signed-off-by: Ikey Doherty <ikey@solus-project.com>
---
 src/conf/alsa.conf | 1 +
 1 file changed, 1 insertion(+)

diff --git a/src/conf/alsa.conf b/src/conf/alsa.conf
index f22918f..c6c3673 100644
--- a/src/conf/alsa.conf
+++ b/src/conf/alsa.conf
@@ -15,6 +15,7 @@
 					"/alsa.conf.d/"
 				]
 			}
+			"/usr/share/defaults/alsa/asound.conf"
 			"/etc/asound.conf"
 			"~/.asoundrc"
 		]
-- 
2.8.3

