From d9820e8bf1d942744a25295bc7425b7dffe1e1d8 Mon Sep 17 00:00:00 2001
From: Gavin Zhao <git@gzgz.dev>
Date: Thu, 5 Dec 2024 00:25:42 -0500
Subject: [PATCH] Extend MIOpen ISA compatibility

Signed-off-by: Gavin Zhao <git@gzgz.dev>
---
 src/hip/handlehip.cpp | 26 +++++++++++++++++++++++++-
 1 file changed, 25 insertions(+), 1 deletion(-)

diff --git a/src/hip/handlehip.cpp b/src/hip/handlehip.cpp
index cf4e4eccb..e70039c0f 100644
--- a/src/hip/handlehip.cpp
+++ b/src/hip/handlehip.cpp
@@ -218,8 +218,32 @@ struct HandleImpl
     {
         hipDeviceProp_t props{};
         hipGetDeviceProperties(&props, device);
-        const std::string name(props.gcnArchName);
+        std::string name(props.gcnArchName);
         MIOPEN_LOG_NQI("Raw device name: " << name);
+
+        // coerce device name
+        size_t loc = std::string::npos;
+        if((loc = name.find("gfx103")) != std::string::npos)
+        {
+            size_t offset = std::char_traits<char>::length("gfx103");
+            name.at(loc + offset) = '0';
+        }
+        else if((loc = name.find("gfx101")) != std::string::npos)
+        {
+            size_t offset = std::char_traits<char>::length("gfx101");
+            name.at(loc + offset) = '0';
+        }
+        else if((loc = name.find("gfx90")) != std::string::npos)
+        {
+            size_t offset = std::char_traits<char>::length("gfx90");
+            char ver = name.at(loc + offset);
+            if (ver == '2' || ver == '9' || ver == 'c')
+            {
+                name.at(loc + offset) = '0';
+            }
+        }
+
+        MIOPEN_LOG_NQI("Coerced device name: " << name);
         return name; // NOLINT (performance-no-automatic-move)
     }
 
-- 
2.47.0

