diff --git a/cmake/ClangTidy.cmake b/cmake/ClangTidy.cmake
index fc3eb75..15bf8a5 100644
--- a/cmake/ClangTidy.cmake
+++ b/cmake/ClangTidy.cmake
@@ -140,25 +140,5 @@ macro(enable_clang_tidy)
 endmacro()
 
 function(clang_tidy_check TARGET)
-    get_target_property(SOURCES ${TARGET} SOURCES)
-    # TODO: Use generator expressions instead
-    # COMMAND ${CLANG_TIDY_COMMAND} $<TARGET_PROPERTY:${TARGET},SOURCES>
-    # COMMAND ${CLANG_TIDY_COMMAND} $<JOIN:$<TARGET_PROPERTY:${TARGET},SOURCES>, >
-    foreach(SOURCE ${SOURCES})
-        if((NOT "${SOURCE}" MATCHES "(h|hpp|hxx)$") AND (NOT "${SOURCE}" MATCHES "TARGET_OBJECTS"))
-            string(MAKE_C_IDENTIFIER "${SOURCE}" tidy_file)
-            set(tidy_target tidy-target-${TARGET}-${tidy_file})
-            add_custom_target(${tidy_target}
-                # for some targets clang-tidy not able to get information from .clang-tidy
-                DEPENDS ${SOURCE}
-                COMMAND ${CLANG_TIDY_COMMAND} "-config=\{CheckOptions: \[\{key: bugprone-reserved-identifier.AllowedIdentifiers,value: __HIP_PLATFORM_HCC__\; __HIP_ROCclr__\}\]\}" ${SOURCE} "-export-fixes=${CLANG_TIDY_FIXIT_DIR}/${TARGET}-${tidy_file}.yaml"
-                WORKING_DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR}
-                COMMENT "clang-tidy: Running clang-tidy on target ${SOURCE}..."
-            )
-            add_dependencies(${tidy_target} ${TARGET})
-            add_dependencies(${tidy_target} tidy-base)
-            add_dependencies(tidy ${tidy_target})
-        endif()
-    endforeach()
 endfunction()
 
