From b21bd3bb858a7cd2fd22214414020bf3ac9034a3 Mon Sep 17 00:00:00 2001
From: Pierre-Yves <pyu@riseup.net>
Date: Mon, 10 Feb 2020 22:03:58 +0100
Subject: [PATCH] Use stateless directories

Signed-off-by: Pierre-Yves <pyu@riseup.net>
---
 data/Makefile.am | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/data/Makefile.am b/data/Makefile.am
index 7c6f03f..2bb74ea 100644
--- a/data/Makefile.am
+++ b/data/Makefile.am
@@ -32,7 +32,7 @@ gsettings_SCHEMAS = $(msd_gschemas_in:.xml.in=.xml)
 
 @GSETTINGS_RULES@
 
-desktopdir = $(sysconfdir)/xdg/autostart
+desktopdir = $(datadir)/xdg/autostart
 desktop_in_files = mate-settings-daemon.desktop.in
 desktop_DATA = $(desktop_in_files:.desktop.in=.desktop)
 
-- 
2.25.0

