From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Reilly Brogan <reilly@reillybrogan.com>
Date: Fri, 2 Aug 2024 17:24:39 -0500
Subject: [PATCH] Build polly libraries shared

---
 polly/cmake/polly_macros.cmake | 4 +---
 1 file changed, 1 insertion(+), 3 deletions(-)

diff --git a/polly/cmake/polly_macros.cmake b/polly/cmake/polly_macros.cmake
index a791be4fa5bc..19d91d636261 100644
--- a/polly/cmake/polly_macros.cmake
+++ b/polly/cmake/polly_macros.cmake
@@ -15,10 +15,8 @@ macro(add_polly_library name)
   endif(MSVC_IDE OR XCODE)
   if (MODULE)
     set(libkind MODULE)
-  elseif (SHARED_LIBRARY)
-    set(libkind SHARED)
   else()
-    set(libkind)
+    set(libkind SHARED)
   endif()
   add_library( ${name} ${libkind} ${srcs} )
   set_target_properties(${name} PROPERTIES FOLDER "Polly/Libraries")
