From 9ec1a236f7b0beb6e66a5977e8730c8bb7ea91b7 Mon Sep 17 00:00:00 2001
From: "Bryan T. Meyers" <bmeyers@datadrake.com>
Date: Sun, 24 Jun 2018 13:42:04 -0400
Subject: [PATCH 1/1] Fix network entry point for apparmor

---
 security/apparmor/apparmorfs.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/security/apparmor/apparmorfs.c b/security/apparmor/apparmorfs.c
index e28398bbbfc3..9b854228697c 100644
--- a/security/apparmor/apparmorfs.c
+++ b/security/apparmor/apparmorfs.c
@@ -2369,6 +2369,7 @@ static struct aa_sfs_entry aa_sfs_entry_features[] = {
 	AA_SFS_DIR("domain",			aa_sfs_entry_domain),
 	AA_SFS_DIR("file",			aa_sfs_entry_file),
 	AA_SFS_DIR("network_v8",		aa_sfs_entry_network),
+	AA_SFS_DIR("network",		aa_sfs_entry_network),
 	AA_SFS_DIR("mount",			aa_sfs_entry_mount),
 	AA_SFS_DIR("namespaces",		aa_sfs_entry_ns),
 	AA_SFS_FILE_U64("capability",		VFS_CAP_FLAGS_MASK),
-- 
2.29.1

