From 41e532c427f43a5eed9081766963d6e29a9975a1 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?S=C3=A9bastien=20Wilmet?= <swilmet@mailfence.com>
Date: Thu, 25 Apr 2024 21:29:19 +0200
Subject: [PATCH] Adapt for Tepl -> libgedit-tepl rename

---
 configure.ac | 3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

diff --git a/configure.ac b/configure.ac
index a5eadef2..5045e80a 100644
--- a/configure.ac
+++ b/configure.ac
@@ -46,8 +46,7 @@ PKG_CHECK_MODULES([DEP], [
 	gio-2.0 >= 2.56
 	gtk+-3.0 >= 3.22
 	gspell-1 >= 1.8
-	libgedit-amtk-5
-	tepl-6 >= 6.8
+	libgedit-tepl-6
 	gee-0.8 >= 0.10
 	gsettings-desktop-schemas
 ])
-- 
GitLab

