--- a/android-tools/src/adb/usb_linux.c
+++ b/android-tools/src/adb/usb_linux.c
@@ -274,7 +274,7 @@ static void find_usb_device(const char *base,
                             char *slash;
                             ssize_t link_len;
                             snprintf(pathbuf, sizeof(pathbuf), "/sys/dev/char/%d:%d",
-                                     major(st.st_rdev), minor(st.st_rdev));
+                                     gnu_dev_major(st.st_rdev), gnu_dev_minor(st.st_rdev));
                             link_len = readlink(pathbuf, link, sizeof(link) - 1);
                             if (link_len > 0) {
                                 link[link_len] = '\0';
--- a/android-tools/src/fastboot/usb_linux.c
+++ b/android-tools/src/fastboot/usb_linux.c
@@ -187,7 +187,7 @@ static int filter_usb_device(int fd, char *ptr, int len, int writable,
         char *slash;
         ssize_t link_len;
         snprintf(cdev, sizeof(cdev), "/sys/dev/char/%d:%d",
-                 major(st.st_rdev), minor(st.st_rdev));
+                 gnu_dev_major(st.st_rdev), gnu_dev_minor(st.st_rdev));
         link_len = readlink(cdev, link, sizeof(link) - 1);
         if (link_len > 0) {
             link[link_len] = '\0';
