From 313a8a06c64a3417e040b8a3a031cd14ac6d12c4 Mon Sep 17 00:00:00 2001
From: Ikey Doherty <ikey@evolve-os.com>
Date: Sat, 17 Jan 2015 05:08:07 +0000
Subject: [PATCH 2/2] Use correct location for documentation files

Signed-off-by: Ikey Doherty <ikey@evolve-os.com>
---
 Makefile.am | 2 +-
 Makefile.in | 2 +-
 2 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/Makefile.am b/Makefile.am
index e7df7fb..606b5b2 100644
--- a/Makefile.am
+++ b/Makefile.am
@@ -11,7 +11,7 @@ SUBDIRS = src po test doc
 pkgconfigdir = $(libdir)/pkgconfig
 pkgconfig_DATA = gtkhotkey-1.0.pc
 
-gtkhotkeydocdir = ${prefix}/doc/gtkhotkey
+gtkhotkeydocdir = ${datadir}/doc/gtkhotkey
 gtkhotkeydoc_DATA = \
 	README			\
 	COPYING			\
diff --git a/Makefile.in b/Makefile.in
index 161a4a6..8daad06 100644
--- a/Makefile.in
+++ b/Makefile.in
@@ -246,7 +246,7 @@ TEST_PROGS =
 SUBDIRS = src po test doc
 pkgconfigdir = $(libdir)/pkgconfig
 pkgconfig_DATA = gtkhotkey-1.0.pc
-gtkhotkeydocdir = ${prefix}/doc/gtkhotkey
+gtkhotkeydocdir = ${datadir}/doc/gtkhotkey
 gtkhotkeydoc_DATA = \
 	README			\
 	COPYING			\
-- 
1.8.4

