From fa6226e0380cbce79c118f9d7c603b6044aaada7 Mon Sep 17 00:00:00 2001
From: Ikey Doherty <ikey@solus-project.com>
Date: Thu, 17 Mar 2016 15:22:01 +0000
Subject: [PATCH] fusermount: Support a stateless configuration

Signed-off-by: Ikey Doherty <ikey@solus-project.com>
---
 util/fusermount.c | 23 ++++++++++++++++-------
 1 file changed, 16 insertions(+), 7 deletions(-)

diff --git a/util/fusermount.c b/util/fusermount.c
index 26a0b75..438e405 100644
--- a/util/fusermount.c
+++ b/util/fusermount.c
@@ -36,6 +36,7 @@
 #define FUSE_DEV_NEW "/dev/fuse"
 #define FUSE_VERSION_FILE_OLD "/proc/fs/fuse/version"
 #define FUSE_CONF "/etc/fuse.conf"
+#define SYSTEM_FUSE_CONF "/usr/share/defaults/fuse/fuse.conf"
 
 #ifndef MS_DIRSYNC
 #define MS_DIRSYNC 128
@@ -525,7 +526,7 @@ static void strip_line(char *line)
 		memmove(line, s, strlen(s)+1);
 }
 
-static void parse_line(char *line, int linenum)
+static void parse_line(char *conf_file, char *line, int linenum)
 {
 	int tmp;
 	if (strcmp(line, "user_allow_other") == 0)
@@ -535,12 +536,20 @@ static void parse_line(char *line, int linenum)
 	else if(line[0])
 		fprintf(stderr,
 			"%s: unknown parameter in %s at line %i: '%s'\n",
-			progname, FUSE_CONF, linenum, line);
+			progname, conf_file, linenum, line);
 }
 
 static void read_conf(void)
 {
-	FILE *fp = fopen(FUSE_CONF, "r");
+	const char *fuse_conf = NULL;
+
+	if (access(FUSE_CONF, F_OK) == 0) {
+		fuse_conf = FUSE_CONF;
+	} else {
+		fuse_conf = SYSTEM_FUSE_CONF;
+	}
+
+	FILE *fp = fopen(fuse_conf, "r");
 	if (fp != NULL) {
 		int linenum = 1;
 		char line[256];
@@ -549,12 +558,12 @@ static void read_conf(void)
 			if (isnewline) {
 				if (line[strlen(line)-1] == '\n') {
 					strip_line(line);
-					parse_line(line, linenum);
+					parse_line(fuse_conf, line, linenum);
 				} else {
 					isnewline = 0;
 				}
 			} else if(line[strlen(line)-1] == '\n') {
-				fprintf(stderr, "%s: reading %s: line %i too long\n", progname, FUSE_CONF, linenum);
+				fprintf(stderr, "%s: reading %s: line %i too long\n", progname, fuse_conf, linenum);
 
 				isnewline = 1;
 			}
@@ -562,13 +571,13 @@ static void read_conf(void)
 				linenum ++;
 		}
 		if (!isnewline) {
-			fprintf(stderr, "%s: reading %s: missing newline at end of file\n", progname, FUSE_CONF);
+			fprintf(stderr, "%s: reading %s: missing newline at end of file\n", progname, fuse_conf);
 
 		}
 		fclose(fp);
 	} else if (errno != ENOENT) {
 		fprintf(stderr, "%s: failed to open %s: %s\n",
-			progname, FUSE_CONF, strerror(errno));
+			progname, fuse_conf, strerror(errno));
 	}
 }
 
-- 
2.7.3

