From 4e8a306a3e21a10447a41c87891c7293a3248f2b Mon Sep 17 00:00:00 2001
From: Ikey Doherty <ikey@solus-project.com>
Date: Mon, 25 Dec 2017 21:37:14 +0000
Subject: [PATCH] Don't be stupid.

Signed-off-by: Ikey Doherty <ikey@solus-project.com>
---
 configure.ac | 20 +-------------------
 1 file changed, 1 insertion(+), 19 deletions(-)

diff --git a/configure.ac b/configure.ac
index d15eaca..833c567 100644
--- a/configure.ac
+++ b/configure.ac
@@ -460,25 +460,7 @@ dnl correctly is complex and lengthy (about as long as the entire configure.ac
 dnl file).
 
 if test x$have_x86cpu = xtrue -o x$have_ppccpu = xtrue; then
-  if test x$host_alias != x; then
-    dnl This is to allow cross compiling
-    AC_MSG_CHECKING(sub-architecture settings)
-    if test x$have_x86cpu = xtrue; then
-       host_mod_cpu=`echo $host_cpu | tr _ -`
-       ARCHFLAGS="-march=$host_mod_cpu -mcpu=$host_mod_cpu"
-    fi
-  else
-    if test ! -r $srcdir/cpuinfo.sh; then
-       AC_MSG_ERROR([cpuinfo.sh script not found - do a cvs update])
-    fi
-    AC_MSG_CHECKING(sub-architecture settings)
-    chmod +x $srcdir/cpuinfo.sh
-    if test x$have_x86cpu = xtrue; then
-       ARCHFLAGS=`$srcdir/cpuinfo.sh x86`
-    elif test x$have_ppccpu = xtrue; then
-       ARCHFLAGS=`$srcdir/cpuinfo.sh ppc`
-    fi
-  fi
+  ARCHFLAGS=-"mtune=generic -march=x86-64"
   AC_MSG_RESULT($ARCHFLAGS)
 fi
 
-- 
2.15.1

