--- a/configure.ac
+++ b/configure.ac
@@ -153,7 +153,7 @@ AC_ARG_WITH([v4l], AC_HELP_STRING([--wit
 AS_IF([test "x$with_v4l" != "xno"], [
 case $host in
   *-*-linux*)
-  AC_CHECK_HEADER(linux/videodev.h,
+  AC_CHECK_HEADER(libv4l1-videodev.h,
                   [have_video4linux=true
                    AC_DEFINE(HAVE_V4L, 1,
 		             [Building for Linux - using the video4linux API])], [])
--- a/lavtools/lavvideo.c
+++ b/lavtools/lavvideo.c
@@ -63,7 +63,9 @@ Copyright by Gernot Ziegler.
  * errors here, check your linux/time.h && sys/time.h header setup.
  */
 #define _LINUX_TIME_H
-#include <linux/videodev.h>
+#include <linux/videodev2.h>
+#include <libv4l1-videodev.h>
+#define BASE_VIDIOCPRIVATE BASE_VIDIOC_PRIVATE
 
 #include <videodev_mjpeg.h>
 #include <frequencies.h>
--- a/lavtools/liblavplay.c
+++ b/lavtools/liblavplay.c
@@ -68,7 +68,9 @@
  * errors here, check your linux/time.h && sys/time.h header setup.
  */
 #define _LINUX_TIME_H
-#include <linux/videodev.h>
+#include <linux/videodev2.h>
+#include <libv4l1-videodev.h>
+#define BASE_VIDIOCPRIVATE BASE_VIDIOC_PRIVATE
 #else
 #define VIDEO_MODE_PAL		0
 #define VIDEO_MODE_NTSC		1
--- a/lavtools/liblavrec.c
+++ b/lavtools/liblavrec.c
@@ -63,7 +63,9 @@
  * errors here, check your linux/time.h && sys/time.h header setup.
  */
 #define _LINUX_TIME_H
-#include <linux/videodev.h>
+#include <linux/videodev2.h>
+#include <libv4l1-videodev.h>
+#define BASE_VIDIOCPRIVATE BASE_VIDIOC_PRIVATE
 #ifdef HAVE_SYS_SOUNDCARD_H
 #include <sys/soundcard.h>
 #endif
--- a/lavtools/testrec.c
+++ b/lavtools/testrec.c
@@ -89,7 +89,9 @@
  * errors here, check your linux/time.h && sys/time.h header setup.
  */
 #define _LINUX_TIME_H
-#include <linux/videodev.h>
+#include <linux/videodev2.h>
+#include <libv4l1-videodev.h>
+#define BASE_VIDIOCPRIVATE BASE_VIDIOC_PRIVATE
 #include <linux/soundcard.h>
 
 /* These are explicit prototypes for the compiler, to prepare separation of audiolib.c */
