From 87922e5b090e3deb1e7129d0a5efb6d9c170c1c9 Mon Sep 17 00:00:00 2001
From: Troy Harvey <harveydevel@gmail.com>
Date: Fri, 16 Apr 2021 12:30:56 +1000
Subject: [PATCH] Disable docs

---
 Makefile.am | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/Makefile.am b/Makefile.am
index 87c3412..2818e6b 100644
--- a/Makefile.am
+++ b/Makefile.am
@@ -3,7 +3,7 @@
 pkgconfigdir = $(libdir)/pkgconfig
 pkgconfig_DATA = gtkdatabox.pc
 
-SUBDIRS = gtk glade examples doc
+SUBDIRS = gtk glade examples
 
 EXTRA_DIST = m4 depcomp scripts gtkdatabox.pc.in gtkdatabox.spec.in glade/gtkdatabox.xml
 
-- 
2.30.2

