From bc4896dc11152196c12fcc3f8fe92fda43e920d3 Mon Sep 17 00:00:00 2001
From: Joshua Strobl <joshua@streambits.io>
Date: Wed, 14 Jul 2021 18:06:28 +0300
Subject: [PATCH 1/1] Do not use CMAKE_INSTALL_PREFIX for config path.

---
 CMakeLists.txt   | 2 +-
 src/rsbuild.h.in | 2 +-
 2 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 3ac6b60..aa69c81 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -224,7 +224,7 @@ endif()
 install(TARGETS ${binary} DESTINATION bin PERMISSIONS ${permissions})
 
 # Install config
-install(FILES sys/replay-sorcery.conf DESTINATION etc)
+install(FILES sys/replay-sorcery.conf DESTINATION "/etc")
 
 # Install services
 set(RS_SYSTEMD_DIR /usr/lib/systemd CACHE STRING "Where to install the systemd services")
diff --git a/src/rsbuild.h.in b/src/rsbuild.h.in
index 70d9dc2..0809aae 100644
--- a/src/rsbuild.h.in
+++ b/src/rsbuild.h.in
@@ -20,7 +20,7 @@
 #ifndef RS_BUILD_H
 #define RS_BUILD_H
 
-#define RS_BUILD_GLOBAL_CONFIG "@CMAKE_INSTALL_PREFIX@/etc/replay-sorcery.conf"
+#define RS_BUILD_GLOBAL_CONFIG "/etc/replay-sorcery.conf"
 #define RS_BUILD_LOCAL_CONFIG "%s/.config/replay-sorcery.conf"
 
 #cmakedefine RS_BUILD_UNISTD_FOUND
-- 
2.32.0

