--- a/build_files/cmake/Modules/FindOpenEXR.cmake
+++ b/build_files/cmake/Modules/FindOpenEXR.cmake
@@ -9,9 +9,8 @@
 #  OPENEXR_FOUND, If false, do not try to use OpenEXR.
 #
 # For individual library access these advanced settings are available
-#  OPENEXR_HALF_LIBRARY, Path to Half library
+#  OPENEXR_OPENEXR_LIBRARY, Path to OpenEXR library
 #  OPENEXR_IEX_LIBRARY, Path to Half library
-#  OPENEXR_ILMIMF_LIBRARY, Path to Ilmimf library
 #  OPENEXR_ILMTHREAD_LIBRARY, Path to IlmThread library
 #  OPENEXR_IMATH_LIBRARY, Path to Imath library
 #
@@ -34,9 +33,8 @@ ENDIF()
 SET(_openexr_libs_ver_init "2.0")
 
 SET(_openexr_FIND_COMPONENTS
-  Half
+  OpenEXR
   Iex
-  IlmImf
   IlmThread
   Imath
 )
@@ -120,7 +118,7 @@ FIND_PACKAGE_HANDLE_STANDARD_ARGS(OpenEX
 IF(OPENEXR_FOUND)
   SET(OPENEXR_LIBRARIES ${_openexr_LIBRARIES})
   # Both include paths are needed because of dummy OSL headers mixing #include <OpenEXR/foo.h> and #include <foo.h> :(
-  SET(OPENEXR_INCLUDE_DIRS ${OPENEXR_INCLUDE_DIR} ${OPENEXR_INCLUDE_DIR}/OpenEXR)
+  SET(OPENEXR_INCLUDE_DIRS ${OPENEXR_INCLUDE_DIR} ${OPENEXR_INCLUDE_DIR}/OpenEXR ${OPENEXR_INCLUDE_DIR}/Imath)
 ENDIF()
 
 MARK_AS_ADVANCED(
--- a/source/blender/imbuf/intern/openexr/openexr_api.cpp
+++ b/source/blender/imbuf/intern/openexr/openexr_api.cpp
@@ -38,8 +38,10 @@
 #include <ImfChannelList.h>
 #include <ImfCompression.h>
 #include <ImfCompressionAttribute.h>
+#include <ImfFrameBuffer.h>
 #include <ImfIO.h>
 #include <ImfInputFile.h>
+#include <ImfInt64.h>
 #include <ImfOutputFile.h>
 #include <ImfPixelType.h>
 #include <ImfStandardAttributes.h>
