From c240f9c96860ea8c9546b949e95d1a46cac3c8b0 Mon Sep 17 00:00:00 2001
From: Joshua Strobl <joshua@streambits.io>
Date: Tue, 24 Sep 2019 10:36:23 +0300
Subject: [PATCH 1/1] Fix compilation against Vala 0.46

---
 src/utils/animation/GtkAnimation.vala | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/utils/animation/GtkAnimation.vala b/src/utils/animation/GtkAnimation.vala
index a1cd56c..6f77d6a 100644
--- a/src/utils/animation/GtkAnimation.vala
+++ b/src/utils/animation/GtkAnimation.vala
@@ -51,7 +51,7 @@ namespace DesktopFolder.UtilGtkAnimation  {
          * @param {double} duration the duration of the animation
          * @param {AnimateFn} animation_fn the animation function
          */
-        public WidgetAnimation (Gtk.Widget widget, double duration, AnimateFn animation_fn) {
+        protected WidgetAnimation (Gtk.Widget widget, double duration, AnimateFn animation_fn) {
             this._widget       = widget;
             this._animation_fn = animation_fn;
             this._duration     = duration;
-- 
2.23.0

