From f2a32215b623bb1dd77376b74e88cfb8db7e2456 Mon Sep 17 00:00:00 2001
From: Reilly Brogan <reilly@reillybrogan.com>
Date: Sun, 1 Oct 2023 22:39:50 -0500
Subject: [PATCH] Use /run instead of /var/run for runstatedir

---
 meson.build | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/meson.build b/meson.build
index ffc6586ecd..38473eb567 100644
--- a/meson.build
+++ b/meson.build
@@ -42,7 +42,7 @@ nm_libexecdir = join_paths(nm_prefix, get_option('libexecdir'))
 nm_localedir = join_paths(nm_prefix, get_option('localedir'))
 nm_localstatedir = join_paths(nm_prefix, get_option('localstatedir'))
 nm_mandir = join_paths(nm_prefix, get_option('mandir'))
-nm_runstatedir = join_paths(nm_localstatedir, 'run')
+nm_runstatedir = join_paths('/', 'run')
 nm_sbindir = join_paths(nm_prefix, get_option('sbindir'))
 nm_sysconfdir = join_paths(nm_prefix, get_option('sysconfdir'))
 
-- 
2.42.0

