Description: Fix compilation with Qt6
Author: xunil-cloud <freed602om@gmail.com>
Origin: upstream
Bug: https://github.com/supercollider/supercollider/issues/7279
Applied-Upstream: 0ea656f4c2d3845b2b0e0f953dfac759b5447f10
Last-Update: 2026-04-29
---
--- a/editors/sc-ide/widgets/code_editor/editor.cpp
+++ b/editors/sc-ide/widgets/code_editor/editor.cpp
@@ -698,7 +698,7 @@ void CodeEditor::keyPressEvent(QKeyEvent* e) {
                                          .arg(e->position().x())
                                          .arg(e->position().y())
 #endif
-                                         .arg(e->modifiers())
+                                         .arg(static_cast<Qt::KeyboardModifiers::Int>(e->modifiers()))
                                          .arg(button),
                                      true);
     }
@@ -732,7 +732,7 @@ void CodeEditor::keyReleaseEvent(QKeyEvent* e) {
                                          .arg(e->position().x())
                                          .arg(e->position().y())
 #endif
-                                         .arg(e->modifiers())
+                                         .arg(static_cast<Qt::KeyboardModifiers::Int>(e->modifiers()))
                                          .arg(button),
                                      true);
     }
@@ -766,7 +766,7 @@ void CodeEditor::inputMethodEvent(QInputMethodEvent* e) {
                                          .arg(e->position().x())
                                          .arg(e->position().y())
 #endif
-                                         .arg(e->modifiers())
+                                         .arg(static_cast<Qt::KeyboardModifiers::Int>(e->modifiers()))
                                          .arg(button),
                                      true);
     }
