From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Reilly Brogan <reilly@reillybrogan.com>
Date: Mon, 10 Jun 2024 15:25:16 -0500
Subject: [PATCH] solus: Add /usr/share/xdg when XDG_CONFIG_DIRS is unset

Signed-off-by: Reilly Brogan <reilly@reillybrogan.com>
---
 obt/paths.c | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/obt/paths.c b/obt/paths.c
index 25cb6b0e..3731efa4 100644
--- a/obt/paths.c
+++ b/obt/paths.c
@@ -173,6 +173,11 @@ ObtPaths* obt_paths_new(void)
                                         (G_DIR_SEPARATOR_S,
                                          "etc", "xdg", NULL),
                                         (GSListFunc) g_slist_append);
+        p->config_dirs = slist_path_add(p->config_dirs,
+                                        g_build_filename
+                                        (G_DIR_SEPARATOR_S,
+                                         "usr", "share", "xdg", NULL),
+                                        (GSListFunc) g_slist_append);
     }
     p->config_dirs = slist_path_add(p->config_dirs,
                                     g_strdup(p->config_home),
