From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Muhammad Alfi Syahrin <malfisya.dev@hotmail.com>
Date: Fri, 28 Feb 2025 10:17:50 +0700
Subject: [PATCH] Don not use path on desktop file

---
 gscriptor.desktop.in | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/gscriptor.desktop.in b/gscriptor.desktop.in
index 42564ae..937ae6b 100644
--- a/gscriptor.desktop.in
+++ b/gscriptor.desktop.in
@@ -8,4 +8,4 @@ Terminal=false
 Type=Application
 Categories=Utility;GTK;
 Keywords=smartcard;apdu;
-Icon=@pcsc_dir@/gscriptor.png
+Icon=gscriptor
-- 
2.47.2

