From a59893527e6f7cfc826fac5b484badee72f11b99 Mon Sep 17 00:00:00 2001
From: Thomas Staudinger <Staudi.Kaos@gmail.com>
Date: Thu, 19 May 2022 20:53:03 +0200
Subject: [PATCH] Update Downloads checksums

---
 .../md5                                                         | 2 +-
 .../sha512                                                      | 2 +-
 2 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/deps/checksums/Downloads-26d79afcde9cf837a331fce023b31f1d3699700c.tar.gz/md5 b/deps/checksums/Downloads-26d79afcde9cf837a331fce023b31f1d3699700c.tar.gz/md5
index 02c7e0f..4fae323 100644
--- a/deps/checksums/Downloads-26d79afcde9cf837a331fce023b31f1d3699700c.tar.gz/md5
+++ b/deps/checksums/Downloads-26d79afcde9cf837a331fce023b31f1d3699700c.tar.gz/md5
@@ -1 +1 @@
-2265854c26d1ff5a061ef9033eb3b9d4
+1122bbb538fe12d45ca86a952a2b3863
diff --git a/deps/checksums/Downloads-26d79afcde9cf837a331fce023b31f1d3699700c.tar.gz/sha512 b/deps/checksums/Downloads-26d79afcde9cf837a331fce023b31f1d3699700c.tar.gz/sha512
index 19d71e7..4d3948b 100644
--- a/deps/checksums/Downloads-26d79afcde9cf837a331fce023b31f1d3699700c.tar.gz/sha512
+++ b/deps/checksums/Downloads-26d79afcde9cf837a331fce023b31f1d3699700c.tar.gz/sha512
@@ -1 +1 @@
-7817f2ab0ccd3224420e65ae7525c2a801f0dbd7e5986e3a0eeac29e3842ea94933e494b895a78e0495ab1188a7375cd98033f507d270aea38eeeb5532226873
+34ca266274f3c28b00ca291ef264067b2eed8558e11c493c3603e2332117c66ce20e6fb64e996b1384002bba7e49c2e8fb06149a6c3028d1027a45a2c3fd73f2
-- 
2.35.3

