From 2a216386f7145b12b2827e5b3ebad22dfb71175b Mon Sep 17 00:00:00 2001
From: Fabio Forni <livingsilver94.solus@redaril.me>
Date: Sat, 14 Nov 2020 01:09:56 +0100
Subject: [PATCH] Do not rebuild on install

---
 Makefile | 7 +------
 1 file changed, 1 insertion(+), 6 deletions(-)

diff --git a/Makefile b/Makefile
index f3100ea..e53226e 100644
--- a/Makefile
+++ b/Makefile
@@ -283,12 +283,7 @@ JL_PRIVATE_LIBS-0 += libgfortran libgcc_s libquadmath
 endif
 
 
-install: $(build_depsbindir)/stringreplace $(BUILDROOT)/doc/_build/html/en/index.html
-ifeq ($(BUNDLE_DEBUG_LIBS),1)
-	@$(MAKE) $(QUIET_MAKE) all
-else
-	@$(MAKE) $(QUIET_MAKE) release
-endif
+install: $(build_depsbindir)/stringreplace
 	@for subdir in $(bindir) $(datarootdir)/julia/stdlib/$(VERSDIR) $(docdir) $(man1dir) $(includedir)/julia $(libdir) $(private_libdir) $(sysconfdir) $(libexecdir); do \
 		mkdir -p $(DESTDIR)$$subdir; \
 	done
-- 
2.29.2

