From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Reilly Brogan <reilly@reillybrogan.com>
Date: Fri, 26 Jul 2024 14:56:56 -0500
Subject: [PATCH] Fix pkgconfig

---
 src/libmaxminddb.pc.in | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/libmaxminddb.pc.in b/src/libmaxminddb.pc.in
index c695cf9..8c8e9d9 100644
--- a/src/libmaxminddb.pc.in
+++ b/src/libmaxminddb.pc.in
@@ -1,7 +1,7 @@
 prefix=@prefix@
 exec_prefix=@prefix@
-libdir=@libdir@
-includedir=@includedir@
+libdir=${exec_prefix}/@libdir@
+includedir=${prefix}/@includedir@
 
 Name: libmaxminddb
 Description: C library for the MaxMind DB file format
