From 40026c50a4133c5989ebc0b17f5bb44df3af715a Mon Sep 17 00:00:00 2001
From: Pierre-Yves <pyu@riseup.net>
Date: Sat, 27 May 2017 23:34:29 +0200
Subject: [PATCH] do not install sample sudoers /etc/sudoers.dist

---
 plugins/sudoers/Makefile.in | 3 ---
 1 file changed, 3 deletions(-)

diff --git a/plugins/sudoers/Makefile.in b/plugins/sudoers/Makefile.in
index 3c09173..82c6d2f 100644
--- a/plugins/sudoers/Makefile.in
+++ b/plugins/sudoers/Makefile.in
@@ -349,9 +349,6 @@ install-plugin: sudoers.la install-dirs
 
 install-sudoers: install-dirs
 	$(INSTALL) -d $(INSTALL_OWNER) -m 0750 $(DESTDIR)$(sudoersdir)/sudoers.d
-	$(INSTALL) $(INSTALL_OWNER) -m $(sudoers_mode) sudoers $(DESTDIR)$(sudoersdir)/sudoers.dist
-	test -r $(DESTDIR)$(sudoersdir)/sudoers || \
-	    cp -p $(DESTDIR)$(sudoersdir)/sudoers.dist $(DESTDIR)$(sudoersdir)/sudoers
 
 uninstall:
 	-$(LIBTOOL) $(LTFLAGS) --mode=uninstall rm -f $(DESTDIR)$(plugindir)/sudoers.la
-- 
2.13.0

