From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Ikey Doherty <michael.i.doherty@intel.com>
Date: Mon, 18 May 2015 16:52:05 +0100
Subject: [PATCH] visudo: Use sane default file

Signed-off-by: Patrick McCarty <patrick.mccarty@intel.com>
---
 plugins/sudoers/visudo.c | 8 ++++++++
 1 file changed, 8 insertions(+)

diff --git a/plugins/sudoers/visudo.c b/plugins/sudoers/visudo.c
index 0713229..1443c4f 100644
--- a/plugins/sudoers/visudo.c
+++ b/plugins/sudoers/visudo.c
@@ -130,6 +130,7 @@ main(int argc, char *argv[])
     int ch, oldlocale, editor_argc, exitcode = 0;
     bool quiet, strict, fflag;
     debug_decl(main, SUDOERS_DEBUG_MAIN);
+    __attribute__ ((unused)) struct stat st = {0};
 
 #if defined(SUDO_DEVEL) && defined(__OpenBSD__)
     {
@@ -148,6 +149,10 @@ main(int argc, char *argv[])
     if (argc < 1)
 	usage(1);
 
+    if (stat("/etc/sudoers.d", &st) < 0) {
+        (void)mkdir("/etc/sudoers.d", 00755);
+    }
+
     /* Register fatal/fatalx callback. */
     sudo_fatal_callback_register(visudo_cleanup);
 
@@ -165,6 +170,9 @@ main(int argc, char *argv[])
     /* Parse sudoers plugin options, if any. */
     parse_sudoers_options();
 
+    /* Set default sudoers file, can be overriden with -f, --file */
+    sudoers_file = "/etc/sudoers.d/visudo";
+
     /*
      * Arg handling.
      */
