From 090be2a1e81f43440f117b4b936d3899f80217e6 Mon Sep 17 00:00:00 2001
From: Ikey Doherty <michael.i.doherty@intel.com>
Date: Mon, 18 May 2015 16:52:05 +0100
Subject: [PATCH] visudo: Use sane default file

Signed-off-by: Patrick McCarty <patrick.mccarty@intel.com>
---
 plugins/sudoers/visudo.c | 7 ++++++-
 1 file changed, 6 insertions(+), 1 deletion(-)

diff --git a/plugins/sudoers/visudo.c b/plugins/sudoers/visudo.c
index 088117cec..2fd50fff1 100644
--- a/plugins/sudoers/visudo.c
+++ b/plugins/sudoers/visudo.c
@@ -109,7 +109,7 @@ extern void get_hostname(void);
  */
 struct sudo_user sudo_user;
 struct passwd *list_pw;
-static const char *path_sudoers = _PATH_SUDOERS;
+static const char *path_sudoers = "/etc/sudoers.d/visudo";
 static struct sudoersfile_list sudoerslist = TAILQ_HEAD_INITIALIZER(sudoerslist);
 static struct sudoers_parser_config sudoers_conf = SUDOERS_PARSER_CONFIG_INITIALIZER;
 static bool checkonly;
@@ -141,6 +141,7 @@ main(int argc, char *argv[])
     int ch, oldlocale, editor_argc, exitcode = 0;
     bool use_perms, use_owner, quiet, strict, fflag;
     debug_decl(main, SUDOERS_DEBUG_MAIN);
+	__attribute__ ((unused)) struct stat st = {0};
 
 #if defined(SUDO_DEVEL) && defined(__OpenBSD__)
     {
@@ -158,6 +159,10 @@ main(int argc, char *argv[])
 
     if (argc < 1)
 	usage();
+ 
+    if (stat("/etc/sudoers.d", &st) < 0) {
+        (void)mkdir("/etc/sudoers.d", 00755);
+    }
 
     /* Register fatal/fatalx callback. */
     sudo_fatal_callback_register(visudo_cleanup);
-- 
2.41.0

