From c32c8d002e1216373560c94738841a7a5e1b976b Mon Sep 17 00:00:00 2001
From: Jeremy Whiting <jpwhiting@kde.org>
Date: Tue, 24 Oct 2017 15:20:22 -0600
Subject: [PATCH] Don't detach shared pointers to private data when setting
 previews.

---
 src/core/entryinternal.cpp | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/src/core/entryinternal.cpp b/src/core/entryinternal.cpp
index 99dfd3a..4bf6594 100644
--- a/src/core/entryinternal.cpp
+++ b/src/core/entryinternal.cpp
@@ -309,7 +309,8 @@ QImage EntryInternal::previewImage(PreviewType type) const
 
 void EntryInternal::setPreviewImage(const QImage &image, PreviewType type)
 {
-    d.detach();
+    // Don't detach here since this is how we get the fetched preview images.
+    //d.detach();
     d->mPreviewImage[type] = image;
 }
 
