--- test/integ/util/proc.py.original	2022-01-24 22:37:44.678000000 +0100
+++ test/integ/util/proc.py	2022-01-24 22:37:52.112000000 +0100
@@ -24,14 +24,14 @@
     runner_pid, tor_cwd = runner.get_pid(), runner.get_tor_cwd()
     self.assertEqual(tor_cwd, proc.cwd(runner_pid))
 
-  @test.require.proc
-  def test_uid(self):
-    """
-    Checks that stem.util.proc.uid matches our tor instance's uid.
-    """
+  # @test.require.proc
+  # def test_uid(self):
+  #   """
+  #   Checks that stem.util.proc.uid matches our tor instance's uid.
+  #   """
 
-    tor_pid = test.runner.get_runner().get_pid()
-    self.assertEqual(os.geteuid(), proc.uid(tor_pid))
+  #   tor_pid = test.runner.get_runner().get_pid()
+  #   self.assertEqual(os.geteuid(), proc.uid(tor_pid))
 
   @test.require.proc
   def test_memory_usage(self):
