From 16997dcbebb82799dc04e5ed44dae85fc9083438 Mon Sep 17 00:00:00 2001
From: Troy Harvey <harveydevel@gmail.com>
Date: Sat, 3 May 2025 13:28:50 +1000
Subject: [PATCH] revert remove unnecessary lamda

Signed-off-by: Troy Harvey <harveydevel@gmail.com>
---
 src/btop_tools.hpp | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/src/btop_tools.hpp b/src/btop_tools.hpp
index 6a93aad..d1c6561 100644
--- a/src/btop_tools.hpp
+++ b/src/btop_tools.hpp
@@ -237,14 +237,14 @@ namespace Tools {
 	}
 
 	//* Return <str> with only uppercase characters
-	inline auto str_to_upper(string str) {
-		std::ranges::for_each(str, ::toupper);
+	inline string str_to_upper(string str) {
+		std::ranges::for_each(str, [](auto& c) { c = ::toupper(c); } );
 		return str;
 	}
 
 	//* Return <str> with only lowercase characters
-	inline auto str_to_lower(string str) {
-		std::ranges::for_each(str, ::tolower);
+	inline string str_to_lower(string str) {
+		std::ranges::for_each(str, [](char& c) { c = ::tolower(c); } );
 		return str;
 	}
 
-- 
2.49.0

