From 3651d109b069689583e3718a7c6d955e0b567c9a Mon Sep 17 00:00:00 2001
From: Ikey Doherty <ikey@solus-project.com>
Date: Fri, 10 Jul 2015 01:28:11 +0100
Subject: [PATCH] Undelete wallpaper that shouldn't be deleted

---
 backgrounds/solus-wallpapers.xml.in | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/backgrounds/solus-wallpapers.xml.in b/backgrounds/solus-wallpapers.xml.in
index d60a55b..666152c 100644
--- a/backgrounds/solus-wallpapers.xml.in
+++ b/backgrounds/solus-wallpapers.xml.in
@@ -41,7 +41,7 @@
     <scolor>#000000</scolor>
     <shade_type>solid</shade_type>
   </wallpaper>
-  <wallpaper deleted="true">
+  <wallpaper deleted="false">
     <name>Book</name>
     <filename>@prefix@/share/backgrounds/solus/book.png</filename>
     <options>zoom</options>
-- 
2.4.5

