From a31744738bf488419849889fd596de3480f4365e Mon Sep 17 00:00:00 2001
From: solbuild user <build@z77-ud3h.lan>
Date: Mon, 16 Apr 2018 16:03:12 +0000
Subject: [PATCH 1/1] etcher: disable update checking

---
 lib/shared/store.js | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/lib/shared/store.js b/lib/shared/store.js
index f044d25..451af2e 100644
--- a/lib/shared/store.js
+++ b/lib/shared/store.js
@@ -101,7 +101,7 @@ const DEFAULT_STATE = Immutable.fromJS({
     errorReporting: true,
     unmountOnSuccess: true,
     validateWriteOnSuccess: true,
-    updatesEnabled: packageJSON.updates.enabled && !_.includes([ 'rpm', 'deb' ], packageJSON.packageType),
+    updatesEnabled: packageJSON.updates.disabled && !_.includes([ 'rpm', 'deb' ], packageJSON.packageType),
     includeUnstableUpdateChannel: !release.isStableRelease(packageJSON.version),
     lastSleptUpdateNotifier: null,
     lastSleptUpdateNotifierVersion: null
-- 
2.16.3

