From dc4c1e5dc7ee8116a3d4cb5105a7bfe31615b05d Mon Sep 17 00:00:00 2001
From: Peter O'Connor <peter@solus-project.com>
Date: Sat, 14 Apr 2018 14:33:31 +1000
Subject: [PATCH] Install binaries always

Signed-off-by: Peter O'Connor <peter@solus-project.com>
---
 CMakeLists.txt | 10 ++++------
 1 file changed, 4 insertions(+), 6 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index ecfbb83..29c45b9 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -260,12 +260,10 @@ generate_pkg_config ("${CMAKE_CURRENT_BINARY_DIR}/libwoff2enc.pc"
   LIBRARIES woff2enc)
 
 # Installation
-if (NOT BUILD_SHARED_LIBS)
-  install(
-    TARGETS woff2_decompress woff2_compress woff2_info
-    RUNTIME DESTINATION "${CMAKE_INSTALL_BINDIR}"
-  )
-endif()
+install(
+  TARGETS woff2_decompress woff2_compress woff2_info
+  RUNTIME DESTINATION "${CMAKE_INSTALL_BINDIR}"
+)
 
 install(
   TARGETS woff2common woff2dec woff2enc
-- 
2.16.3

