From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Evan Maddock <maddock.evan@vivaldi.net>
Date: Fri, 17 Oct 2025 15:20:23 -0400
Subject: [PATCH 1/5] Implement stateless config

Signed-off-by: Evan Maddock <maddock.evan@vivaldi.net>
---
 src/suricata.c | 21 +++++++++++++++++++--
 1 file changed, 19 insertions(+), 2 deletions(-)

diff --git a/src/suricata.c b/src/suricata.c
index f7b62c9dd..732f2160d 100644
--- a/src/suricata.c
+++ b/src/suricata.c
@@ -150,6 +150,8 @@
 #include "win32-syscall.h"
 #endif
 
+#include <unistd.h>
+
 /*
  * we put this here, because we only use it here in main.
  */
@@ -1018,6 +1020,9 @@ TmEcode SCLoadYamlConfig(void)
     if (suri->conf_filename == NULL)
         suri->conf_filename = DEFAULT_CONF_FILE;
 
+    if (access("/etc/suricata/suricata.yaml", F_OK) == 0)
+        suri->conf_filename = "/etc/suricata/suricata.yaml";
+
     if (SCConfYamlLoadFile(suri->conf_filename) != 0) {
         /* Error already displayed. */
         SCReturnInt(TM_ECODE_FAILED);
@@ -2397,13 +2402,25 @@ int SCStartInternalRunMode(int argc, char **argv)
             if (suri->conf_filename != NULL) {
                 return ListAppLayerProtocols(suri->conf_filename);
             } else {
-                return ListAppLayerProtocols(DEFAULT_CONF_FILE);
+                const char *conf_file = NULL;
+                if (access("/etc/suricata/suricata.yaml", F_OK) == 0)
+                    conf_file = "/etc/suricata/suricata.yaml";
+                else
+                    conf_file = DEFAULT_CONF_FILE;
+
+                return ListAppLayerProtocols(conf_file);
             }
         case RUNMODE_LIST_APP_LAYER_HOOKS:
             if (suri->conf_filename != NULL) {
                 return ListAppLayerHooks(suri->conf_filename);
             } else {
-                return ListAppLayerHooks(DEFAULT_CONF_FILE);
+                const char *conf_file = NULL;
+                if (access("/etc/suricata/suricata.yaml", F_OK) == 0)
+                    conf_file = "/etc/suricata/suricata.yaml";
+                else
+                    conf_file = DEFAULT_CONF_FILE;
+
+                return ListAppLayerHooks(conf_file);
             }
         case RUNMODE_PRINT_VERSION:
             PrintVersion();
