From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Evan Maddock <maddock.evan@vivaldi.net>
Date: Fri, 17 Oct 2025 15:32:33 -0400
Subject: [PATCH 2/5] Implement stateless classification config

Signed-off-by: Evan Maddock <maddock.evan@vivaldi.net>
---
 src/util-classification-config.c | 12 ++++++++++--
 1 file changed, 10 insertions(+), 2 deletions(-)

diff --git a/src/util-classification-config.c b/src/util-classification-config.c
index d7f4e9cc8..154fa997f 100644
--- a/src/util-classification-config.c
+++ b/src/util-classification-config.c
@@ -36,6 +36,8 @@
 #include "util-fmemopen.h"
 #include "util-byte.h"
 
+#include <unistd.h>
+
 /* Regex to parse the classtype argument from a Signature.  The first substring
  * holds the classtype name, the second substring holds the classtype the
  * classtype description, and the third argument holds the priority */
@@ -162,12 +164,18 @@ static const char *SCClassConfGetConfFilename(const DetectEngineCtx *de_ctx)
          * fails. */
         if (SCConfGet(config_value, &log_filename) != 1) {
             if (SCConfGet("classification-file", &log_filename) != 1) {
-                log_filename = (char *)SC_CLASS_CONF_DEF_CONF_FILEPATH;
+                if (access("/etc/suricata/classification.config", F_OK) == 0)
+                    log_filename = "/etc/suricata/classification.config";
+                else
+                    log_filename = (char *)SC_CLASS_CONF_DEF_CONF_FILEPATH;
             }
         }
     } else {
         if (SCConfGet("classification-file", &log_filename) != 1) {
-            log_filename = (char *)SC_CLASS_CONF_DEF_CONF_FILEPATH;
+            if (access("/etc/suricata/classification.config", F_OK) == 0)
+                log_filename = "/etc/suricata/classification.config";
+            else
+                log_filename = (char *)SC_CLASS_CONF_DEF_CONF_FILEPATH;
         }
     }
 
