From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Evan Maddock <maddock.evan@vivaldi.net>
Date: Fri, 17 Oct 2025 15:36:38 -0400
Subject: [PATCH 3/5] Implement stateless reference config

Signed-off-by: Evan Maddock <maddock.evan@vivaldi.net>
---
 src/util-reference-config.c | 12 ++++++++++--
 1 file changed, 10 insertions(+), 2 deletions(-)

diff --git a/src/util-reference-config.c b/src/util-reference-config.c
index 677a891c3..625b1cb80 100644
--- a/src/util-reference-config.c
+++ b/src/util-reference-config.c
@@ -32,6 +32,8 @@
 #include "util-debug.h"
 #include "util-fmemopen.h"
 
+#include <unistd.h>
+
 /* Regex to parse each line from reference.config file.  The first substring
  * is for the system name and the second for the url */
 /*-----------------------------------------------------------system-------------------url----*/
@@ -152,12 +154,18 @@ static const char *SCRConfGetConfFilename(const DetectEngineCtx *de_ctx)
          * fails. */
         if (SCConfGet(config_value, &path) != 1) {
             if (SCConfGet("reference-config-file", &path) != 1) {
-                return (char *)SC_RCONF_DEFAULT_FILE_PATH;
+                if (access("/etc/suricata/reference.config", F_OK) == 0)
+                    return "/etc/suricata/reference.config";
+                else
+                    return (char *)SC_RCONF_DEFAULT_FILE_PATH;
             }
         }
     } else {
         if (SCConfGet("reference-config-file", &path) != 1) {
-            return (char *)SC_RCONF_DEFAULT_FILE_PATH;
+            if (access("/etc/suricata/reference.config", F_OK) == 0)
+                return "/etc/suricata/reference.config";
+            else
+                return (char *)SC_RCONF_DEFAULT_FILE_PATH;
         }
     }
     return path;
