From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Evan Maddock <maddock.evan@vivaldi.net>
Date: Fri, 17 Oct 2025 15:40:03 -0400
Subject: [PATCH 4/5] Implement stateless threshold config

Signed-off-by: Evan Maddock <maddock.evan@vivaldi.net>
---
 src/util-threshold-config.c | 12 ++++++++++--
 1 file changed, 10 insertions(+), 2 deletions(-)

diff --git a/src/util-threshold-config.c b/src/util-threshold-config.c
index e665d1ca9..63a8a2843 100644
--- a/src/util-threshold-config.c
+++ b/src/util-threshold-config.c
@@ -51,6 +51,8 @@
 #include "util-debug.h"
 #include "util-fmemopen.h"
 
+#include <unistd.h>
+
 typedef enum ThresholdRuleType {
     THRESHOLD_TYPE_EVENT_FILTER,
     THRESHOLD_TYPE_THRESHOLD,
@@ -141,12 +143,18 @@ static const char *SCThresholdConfGetConfFilename(const DetectEngineCtx *de_ctx)
          * fails. */
         if (SCConfGet(config_value, &log_filename) != 1) {
             if (SCConfGet("threshold-file", &log_filename) != 1) {
-                log_filename = (char *)THRESHOLD_CONF_DEF_CONF_FILEPATH;
+                if (access("/etc/suricata/threshold.config", F_OK) == 0)
+                    log_filename = "/etc/suricata/threshold.config";
+                else
+                    log_filename = (char *)THRESHOLD_CONF_DEF_CONF_FILEPATH;
             }
         }
     } else {
         if (SCConfGet("threshold-file", &log_filename) != 1) {
-            log_filename = (char *)THRESHOLD_CONF_DEF_CONF_FILEPATH;
+            if (access("/etc/suricata/threshold.config", F_OK) == 0)
+                log_filename = "/etc/suricata/threshold.config";
+            else
+                log_filename = (char *)THRESHOLD_CONF_DEF_CONF_FILEPATH;
         }
     }
     return log_filename;
