From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Evan Maddock <maddock.evan@vivaldi.net>
Date: Fri, 17 Oct 2025 20:05:02 -0400
Subject: [PATCH 5/5] Fix service file for Solus

Signed-off-by: Evan Maddock <maddock.evan@vivaldi.net>
---
 etc/suricata.service.in | 17 ++++++++++++-----
 1 file changed, 12 insertions(+), 5 deletions(-)

diff --git a/etc/suricata.service.in b/etc/suricata.service.in
index cce1baad6..46a521049 100644
--- a/etc/suricata.service.in
+++ b/etc/suricata.service.in
@@ -1,16 +1,23 @@
 # Sample Suricata systemd unit file.
 [Unit]
 Description=Suricata Intrusion Detection Service
-After=syslog.target network-online.target
+After=syslog.target network-online.target systemd-tmpfiles-setup.service
+Documentation=man:suricata(1)
 
 [Service]
 # Environment file to pick up $OPTIONS. On Fedora/EL this would be
 # /etc/sysconfig/suricata, or on Debian/Ubuntu, /etc/default/suricata.
-#EnvironmentFile=-/etc/sysconfig/suricata
+EnvironmentFile=-/etc/sysconfig/suricata
 #EnvironmentFile=-/etc/default/suricata
-ExecStartPre=/bin/rm -f @e_rundir@suricata.pid
-ExecStart=/sbin/suricata -c @e_sysconfdir@suricata.yaml --pidfile @e_rundir@suricata.pid $OPTIONS
-ExecReload=/bin/kill -USR2 $MAINPID
+ExecStartPre=/usr/bin/rm -f @e_rundir@suricata.pid
+ExecStart=/usr/bin/suricata -c @e_sysconfdir@suricata.yaml --pidfile @e_rundir@suricata.pid $OPTIONS
+ExecReload=/usr/bin/kill -USR2 $MAINPID
+
+# Security settings
+MemoryDenyWriteExecute=true
+LockPersonality=true
+ProtectControlGroups=true
+ProtectKernelModules=true
 
 [Install]
 WantedBy=multi-user.target
