diff -urp suricata-5.0.4.orig/suricata.yaml.in suricata-5.0.4/suricata.yaml.in
--- suricata-5.0.4.orig/suricata.yaml.in	2020-10-07 17:01:44.000000000 -0400
+++ suricata-5.0.4/suricata.yaml.in	2020-10-15 12:07:08.731747692 -0400
@@ -1062,7 +1062,7 @@ unix-command:
 
 # GeoIP2 database file. Specify path and filename of GeoIP2 database
 # if using rules with "geoip" rule option.
-#geoip-database: /usr/local/share/GeoLite2/GeoLite2-Country.mmdb
+#geoip-database: /usr/share/GeoIP/GeoLite2-Country.mmdb
 
 legacy:
   uricontent: enabled
