From bf4fda1421b757a0416c396277287c0c6fb10109 Mon Sep 17 00:00:00 2001
From: Leon Lombar <leon@makase.ro>
Date: Mon, 1 Sep 2025 11:50:13 +0200
Subject: [PATCH] link(ELF): add ZSTD decompression support for compressed
 sections

---
 src/link/Elf/Object.zig | 11 +++++++++++
 1 file changed, 11 insertions(+)

diff --git a/src/link/Elf/Object.zig b/src/link/Elf/Object.zig
index eddb5c90dd85..4ea4f663bf3a 100644
--- a/src/link/Elf/Object.zig
+++ b/src/link/Elf/Object.zig
@@ -1207,6 +1207,17 @@ pub fn codeDecompressAlloc(self: *Object, elf_file: *Elf, atom_index: Atom.Index
                 _ = try zlib_stream.reader.streamRemaining(&aw.writer);
                 return aw.toOwnedSlice();
             },
+            .ZSTD => {
+                var input: std.Io.Reader = .fixed(data[@sizeOf(elf.Elf64_Chdr)..]);
+                var stream: std.compress.zstd.Decompress = .init(&input, &.{}, .{});
+                const size = std.math.cast(usize, chdr.ch_size) orelse return error.Overflow;
+                var aw: std.Io.Writer.Allocating = .init(gpa);
+                defer aw.deinit();
+                try aw.ensureUnusedCapacity(size);
+                _ = try stream.reader.streamRemaining(&aw.writer);
+
+                return aw.toOwnedSlice();
+            },
             else => @panic("TODO unhandled compression scheme"),
         }
     }