From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Reilly Brogan <reilly@reillybrogan.com>
Date: Wed, 22 Apr 2026 15:52:14 -0500
Subject: [PATCH] Try versioned zig libdir

---
 src/introspect.zig | 13 +++++++++++++
 1 file changed, 13 insertions(+)

diff --git a/src/introspect.zig b/src/introspect.zig
index 13d0052093..69fb3dfc36 100644
--- a/src/introspect.zig
+++ b/src/introspect.zig
@@ -19,6 +19,19 @@ const Package = @import("Package.zig");
 fn testZigInstallPrefix(io: Io, base_dir: Io.Dir) ?Cache.Directory {
     const test_index_file = "std" ++ Dir.path.sep_str ++ "std.zig";
 
+    // Try the versioned libdir first
+    zig_dir_versioned: {
+        // Try lib/zig-0.16/std/std.zig
+        const lib_zig = "lib" ++ Dir.path.sep_str ++ "zig-0.16";
+        var test_zig_dir = base_dir.openDir(io, lib_zig, .{}) catch break :zig_dir_versioned;
+        const file = test_zig_dir.openFile(io, test_index_file, .{}) catch {
+            test_zig_dir.close(io);
+            break :zig_dir_versioned;
+        };
+        file.close(io);
+        return .{ .handle = test_zig_dir, .path = lib_zig };
+    }
+
     zig_dir: {
         // Try lib/zig/std/std.zig
         const lib_zig = "lib" ++ Dir.path.sep_str ++ "zig";
