diff --git a/src/config.c b/src/config.c
index 31b1f7a5..33c9986f 100644
--- a/src/config.c
+++ b/src/config.c
@@ -437,7 +437,6 @@ int read_config(const char * file)
 
   f = fopen(cfg_infile, "r");
   if (f == NULL) {
-    pmsg_ext_error("cannot open config file %s: %s\n", cfg_infile, strerror(errno));
     free(cfg_infile);
     cfg_infile = NULL;
     return -1;
diff --git a/src/main.c b/src/main.c
index 9d5db46f..a4f08134 100644
--- a/src/main.c
+++ b/src/main.c
@@ -950,15 +950,24 @@ int main(int argc, char * argv [])
 
   if(*sys_config) {
     char *real_sys_config = realpath(sys_config, NULL);
-    if(real_sys_config) {
-     imsg_notice("System wide configuration file is %s\n", real_sys_config);
-    } else
-      pmsg_warning("cannot determine realpath() of config file %s: %s\n", sys_config, strerror(errno));
 
     rc = read_config(real_sys_config);
     if (rc) {
-      pmsg_error("unable to process system wide configuration file %s\n", real_sys_config);
-      exit(1);
+      rc = read_config("/usr/share/defaults/avrdude/avrdude.conf");
+
+      if (rc) {
+        pmsg_error("%s: error reading vendor configuration file \"%s\"\n",
+                   progname, "/usr/share/defaults/avrdude/avrdude.conf");
+        exit(1);
+      } else {
+        imsg_notice("Using vendor provided configuration file in \"%s\"\n",
+                    "/usr/share/defaults/avrdude/avrdude.conf");
+      }
+    } else {
+      if(real_sys_config) {
+        imsg_notice("System wide configuration file is %s\n", real_sys_config);
+      } else
+        pmsg_warning("cannot determine realpath() of config file %s: %s\n", sys_config, strerror(errno));
     }
     free(real_sys_config);
   }
