From 9ba0f44542b457a12a60596279dc43eeb093e709 Mon Sep 17 00:00:00 2001
From: Joshua Strobl <joshua@streambits.io>
Date: Mon, 28 Dec 2020 18:59:18 +0200
Subject: [PATCH 1/1] Make avrdude stateless

---
 config.c |  2 --
 main.c   | 18 +++++++++++++-----
 2 files changed, 13 insertions(+), 7 deletions(-)

diff --git a/config.c b/config.c
index a62823b..a3df061 100644
--- a/config.c
+++ b/config.c
@@ -327,8 +327,6 @@ int read_config(const char * file)
 
   f = fopen(file, "r");
   if (f == NULL) {
-    avrdude_message(MSG_INFO, "%s: can't open config file \"%s\": %s\n",
-            progname, file, strerror(errno));
     return -1;
   }
 
diff --git a/main.c b/main.c
index ad7c05e..37a9781 100644
--- a/main.c
+++ b/main.c
@@ -683,14 +683,22 @@ int main(int argc, char * argv [])
                     "%sCopyright (c) 2000-2005 Brian Dean, http://www.bdmicro.com/\n"
                     "%sCopyright (c) 2007-2014 Joerg Wunsch\n\n",
                     progname, version, __DATE__, __TIME__, progbuf, progbuf);
-  avrdude_message(MSG_NOTICE, "%sSystem wide configuration file is \"%s\"\n",
-            progbuf, sys_config);
 
   rc = read_config(sys_config);
   if (rc) {
-    avrdude_message(MSG_INFO, "%s: error reading system wide configuration file \"%s\"\n",
-                    progname, sys_config);
-    exit(1);
+    rc = read_config("/usr/share/defaults/avrdude/avrdude.conf");
+
+    if (rc) {
+      avrdude_message(MSG_INFO, "%s: error reading vendor configuration file \"%s\"\n",
+                      progname, "/usr/share/defaults/avrdude/avrdude.conf");
+      exit(1);
+    } else {
+      avrdude_message(MSG_NOTICE, "%sUsing vendor provided configuration file in \"%s\"\n",
+                progbuf, "/usr/share/defaults/avrdude/avrdude.conf");
+    }
+  } else {
+    avrdude_message(MSG_NOTICE, "%sSystem wide configuration file is \"%s\"\n",
+              progbuf, sys_config);
   }
 
   if (usr_config[0] != 0) {
-- 
2.29.2

