From 77983c1ac38efa87a3c644b3918dd1648040aee0 Mon Sep 17 00:00:00 2001
From: wodim <neikokz@gmail.com>
Date: Mon, 1 Apr 2019 15:02:39 +0200
Subject: [PATCH] This is not necessary anymore

---
 src/modules/python/libkvipython.cpp | 26 --------------------------
 1 file changed, 26 deletions(-)

diff --git a/src/modules/python/libkvipython.cpp b/src/modules/python/libkvipython.cpp
index 700e8939b..6bdd56a80 100644
--- a/src/modules/python/libkvipython.cpp
+++ b/src/modules/python/libkvipython.cpp
@@ -502,31 +502,6 @@ static bool python_kvs_fnc_isAvailable(KviKvsModuleFunctionCall * c)
 	return true;
 }
 
-/*
-	@doc: python.version
-	@type:
-		function
-	@title:
-		$python.version
-	@short:
-		Check which version of Python is supported in this build of KVIrc
-	@syntax:
-		$python.version
-	@description:
-		Returns which major version of Python is KVIrc linked to ([b]2[/b] or [b]3[/b])
-		or [b]0[/b] if Python is not supported at all.
-*/
-
-static bool python_kvs_fnc_version(KviKvsModuleFunctionCall * c)
-{
-#ifdef COMPILE_PYTHON_SUPPORT
-	c->returnValue()->setInteger(PY_MAJOR_VERSION);
-#else
-	c->returnValue()->setBoolean(false);
-#endif
-	return true;
-}
-
 static bool python_module_init(KviModule * m)
 {
 	// register the command anyway
@@ -534,7 +509,6 @@ static bool python_module_init(KviModule * m)
 	KVSM_REGISTER_SIMPLE_COMMAND(m, "destroy", python_kvs_cmd_destroy);
 
 	KVSM_REGISTER_FUNCTION(m, "isAvailable", python_kvs_fnc_isAvailable);
-	KVSM_REGISTER_FUNCTION(m, "version", python_kvs_fnc_version);
 #ifdef COMPILE_PYTHON_SUPPORT
 	g_pPythonCoreModule = g_pModuleManager->getModule("pythoncore");
 #endif
-- 
2.25.0
