From 40b632bc846e8641f2df021c20cf736967051a8c Mon Sep 17 00:00:00 2001
From: Ikey Doherty <ikey@solus-project.com>
Date: Thu, 17 Mar 2016 15:29:13 +0000
Subject: [PATCH] Set default configuration in absence of config file

Signed-off-by: Ikey Doherty <ikey@solus-project.com>
---
 src/vdpau_wrapper.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/src/vdpau_wrapper.c b/src/vdpau_wrapper.c
index 79dcb94..2b1fc9a 100644
--- a/src/vdpau_wrapper.c
+++ b/src/vdpau_wrapper.c
@@ -362,6 +362,8 @@ static void init_config(void)
 
     fp = fopen(VDPAU_SYSCONFDIR "/vdpau_wrapper.cfg", "r");
     if (!fp) {
+        _enable_flash_uv_swap = 1;
+        _disable_flash_pq_bg_color = 1;
         return;
     }
 
-- 
2.7.3

