--- a/src/XDGData/org.freecadweb.FreeCAD.desktop
+++ b/src/XDGData/org.freecadweb.FreeCAD.desktop
@@ -21,3 +21,4 @@ Icon=org.freecadweb.FreeCAD
 Categories=Graphics;Science;Education;Engineering;
 StartupNotify=true
 MimeType=application/x-extension-fcstd;model/obj;model/iges;image/vnd.dwg;image/vnd.dxf;model/vnd.collada+xml;application/iges;model/iges;model/step;model/step+zip;model/stl;application/vnd.shp;model/vrml;
+StartupWMClass=org.freecad.FreeCAD
