Index: FreeCAD-0.19.4/src/Mod/Part/App/BRepOffsetAPI_MakeFillingPyImp.cpp
===================================================================
--- FreeCAD-0.19.4.orig/src/Mod/Part/App/BRepOffsetAPI_MakeFillingPyImp.cpp
+++ FreeCAD-0.19.4/src/Mod/Part/App/BRepOffsetAPI_MakeFillingPyImp.cpp
@@ -30,6 +30,9 @@
 # include <TopoDS_Face.hxx>
 #endif
 
+// For std::unique_ptr
+#include <memory>
+
 #include "BRepOffsetAPI_MakeFillingPy.h"
 #include "BRepOffsetAPI_MakeFillingPy.cpp"
 #include "TopoShapeEdgePy.h"
Index: FreeCAD-0.19.4/src/Mod/Part/App/ShapeUpgrade/UnifySameDomainPyImp.cpp
===================================================================
--- FreeCAD-0.19.4.orig/src/Mod/Part/App/ShapeUpgrade/UnifySameDomainPyImp.cpp
+++ FreeCAD-0.19.4/src/Mod/Part/App/ShapeUpgrade/UnifySameDomainPyImp.cpp
@@ -29,6 +29,8 @@
 
 // Needed for OCCT 7.5.2
 #include <TopoDS_Edge.hxx>
+// Needed for std::unique_ptr
+#include <memory>
 
 #include "ShapeUpgrade/UnifySameDomainPy.h"
 #include "ShapeUpgrade/UnifySameDomainPy.cpp"
