From 540c36413afcdc6ea5ad9efc146feda1bef92673 Mon Sep 17 00:00:00 2001
From: "chris jones @ipatch" <chris.r.jones.1983@gmail.com>
Date: Mon, 30 Mar 2026 12:41:56 -0500
Subject: [PATCH] core: cmake: fix build errors with qt v6.11

---
 cMake/FreeCAD_Helpers/SetupQt.cmake | 12 ++++++++++--
 1 file changed, 10 insertions(+), 2 deletions(-)

diff --git a/cMake/FreeCAD_Helpers/SetupQt.cmake b/cMake/FreeCAD_Helpers/SetupQt.cmake
index 8a7ba9bdeb37..7654ef39e9fc 100644
--- a/cMake/FreeCAD_Helpers/SetupQt.cmake
+++ b/cMake/FreeCAD_Helpers/SetupQt.cmake
@@ -29,8 +29,16 @@ endif ()
 
 foreach(COMPONENT IN LISTS FREECAD_QT_COMPONENTS)
     find_package(Qt${FREECAD_QT_MAJOR_VERSION} REQUIRED COMPONENTS ${COMPONENT})
-    set(Qt${COMPONENT}_LIBRARIES ${Qt${FREECAD_QT_MAJOR_VERSION}${COMPONENT}_LIBRARIES})
-    set(Qt${COMPONENT}_INCLUDE_DIRS ${Qt${FREECAD_QT_MAJOR_VERSION}${COMPONENT}_INCLUDE_DIRS})
+    if(TARGET Qt${FREECAD_QT_MAJOR_VERSION}::${COMPONENT})
+        set(Qt${COMPONENT}_LIBRARIES Qt${FREECAD_QT_MAJOR_VERSION}::${COMPONENT})
+        get_target_property(Qt${COMPONENT}_INCLUDE_DIRS Qt${FREECAD_QT_MAJOR_VERSION}::${COMPONENT} INTERFACE_INCLUDE_DIRECTORIES)
+        if(NOT Qt${COMPONENT}_INCLUDE_DIRS)
+            set(Qt${COMPONENT}_INCLUDE_DIRS "")
+        endif()
+    else()
+        set(Qt${COMPONENT}_LIBRARIES ${Qt${FREECAD_QT_MAJOR_VERSION}${COMPONENT}_LIBRARIES})
+        set(Qt${COMPONENT}_INCLUDE_DIRS ${Qt${FREECAD_QT_MAJOR_VERSION}${COMPONENT}_INCLUDE_DIRS})
+    endif()
     set(Qt${COMPONENT}_FOUND ${Qt${FREECAD_QT_MAJOR_VERSION}${COMPONENT}_FOUND})
     set(Qt${COMPONENT}_VERSION ${Qt${FREECAD_QT_MAJOR_VERSION}${COMPONENT}_VERSION})
 endforeach()
