--- FreeCAD-1.0.0/src/3rdParty/OndselSolver/OndselSolver/Constraint.h.old	2025-05-21 12:08:41.587716542 +0000
+++ FreeCAD-1.0.0/src/3rdParty/OndselSolver/OndselSolver/Constraint.h	2025-05-21 12:09:02.626042098 +0000
@@ -8,6 +8,7 @@
 
 #pragma once
 
+#include <cstdint>
 #include <memory>
 
 #include "enum.h"
--- FreeCAD-1.0.0/src/3rdParty/OndselSolver/OndselSolver/FullMatrix.h.old	2025-05-21 12:19:27.306968173 +0000
+++ FreeCAD-1.0.0/src/3rdParty/OndselSolver/OndselSolver/FullMatrix.h	2025-05-21 12:19:52.037420050 +0000
@@ -9,6 +9,7 @@
 #pragma once
 
 #include <cmath>
+#include <cstdint>
 #include <memory>
 
 #include "RowTypeMatrix.h"
--- FreeCAD-1.0.0/src/3rdParty/OndselSolver/OndselSolver/NewtonRaphson.h.old	2025-05-21 12:24:06.845976961 +0000
+++ FreeCAD-1.0.0/src/3rdParty/OndselSolver/OndselSolver/NewtonRaphson.h	2025-05-21 12:24:29.971558826 +0000
@@ -5,9 +5,10 @@
  *                                                                         *
  *   See LICENSE file for details about copyright.                         *
  ***************************************************************************/
- 
+
 #pragma once
 
+#include <cstdint>
 #include <memory>
 #include <vector>

