--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -46,6 +46,7 @@ if(NOT FREECAD_LIBPACK_USE OR FREECAD_LI
     SetupPCL()
     SetupPybind11()
     SetupBoost()
+    set(Boost_LIBRARIES_FreeCADApp ${Boost_LIBRARIES})
     SetupXercesC()
     find_package(ZLIB REQUIRED)
     find_package(PyCXX REQUIRED)
--- a/src/App/CMakeLists.txt
+++ b/src/App/CMakeLists.txt
@@ -59,7 +59,7 @@ include_directories(
 
 set(FreeCADApp_LIBS
     FreeCADBase
-    ${Boost_LIBRARIES}
+    ${Boost_LIBRARIES_FreeCADApp}
 )
 
 if (BUILD_QT5)
--- a/src/3rdParty/salomesmesh/CMakeLists.txt
+++ b/src/3rdParty/salomesmesh/CMakeLists.txt
@@ -232,7 +232,7 @@ endif(WIN32)
 #############
 FILE(GLOB DriverSTL_source_files src/DriverSTL/*.cpp inc/DriverSTL*.h*)
 ADD_LIBRARY(DriverSTL SHARED ${DriverSTL_source_files})
-TARGET_LINK_LIBRARIES(DriverSTL ${SMESH_LIBS} Driver SMDS ${Boost_LIBRARIES})
+TARGET_LINK_LIBRARIES(DriverSTL ${SMESH_LIBS} Driver SMDS ${Boost_LIBRARIES_FreeCADApp})
 
 SET_BIN_DIR(DriverSTL DriverSTL)
 if(WIN32)
@@ -293,7 +293,7 @@ endif(WIN32)
 ###########
 FILE(GLOB SMESHDS_source_files src/SMESHDS/*.cpp inc/SMESHDS_*.h*)
 ADD_LIBRARY(SMESHDS SHARED ${SMESHDS_source_files})
-TARGET_LINK_LIBRARIES(SMESHDS ${SMESH_LIBS} SMDS ${Boost_LIBRARIES} )
+TARGET_LINK_LIBRARIES(SMESHDS ${SMESH_LIBS} SMDS ${Boost_LIBRARIES_FreeCADApp})
 
 SET_BIN_DIR(SMESHDS SMESHDS)
 
