--- a/src/Mod/Fem/femtaskpanels/task_result_mechanical.py
+++ b/src/Mod/Fem/femtaskpanels/task_result_mechanical.py
@@ -449,8 +449,8 @@ class _TaskPanel:
 
         # https://forum.freecadweb.org/viewtopic.php?f=18&t=42425&start=10#p368774 ff
         # https://github.com/FreeCAD/FreeCAD/pull/3020
-        from ply import lex
-        from ply import yacc
+        from pycparser.ply import lex
+        from pycparser.ply import yacc
         import femtools.tokrules as tokrules
         identifiers = [
             "x", "y", "z", "T", "vM", "Peeq", "P1", "P2", "P3",
--- a/src/Mod/OpenSCAD/importCSG.py
+++ b/src/Mod/OpenSCAD/importCSG.py
@@ -41,8 +41,8 @@ else:
     if printverbose: print("FreeCAD Gui not present.")
     gui = False
 
-import ply.lex as lex
-import ply.yacc as yacc
+import pycparser.ply.lex as lex
+import pycparser.ply.yacc as yacc
 import Part
 
 from OpenSCADFeatures import *
--- a/src/Mod/Fem/femtools/tokrules.py
+++ b/src/Mod/Fem/femtools/tokrules.py
@@ -83,7 +83,7 @@ def t_error(t):
 
 
 # Build the lexer
-import ply.lex as lex
+import pycparser.ply.lex as lex
 lex.lex()
 
 
@@ -152,5 +152,5 @@ def p_error(p):
     print("Syntax error at '%s'" % p.value)
 
 
-import ply.yacc as yacc
+import pycparser.ply.yacc as yacc
 yacc.yacc()
