--- a/src/Tools/freecad-thumbnailer
+++ b/src/Tools/freecad-thumbnailer
@@ -1,4 +1,4 @@
-#!/usr/bin/python
+#!/usr/bin/python3
 """Support file to show FreeCAD thumbnails on Free Desktop Environments (like GNOME or KDE)
 
 Installation:
@@ -24,7 +24,7 @@ Installation:
     Exec=freecad-thumbnailer -s %s %i %o
     MimeType=application/x-extension-fcstd;
 
-HINT: Make sure that the symlink /usr/bin/python exists and points to the Python executable
+HINT: Make sure that the symlink /usr/bin/python3 exists and points to the Python executable
 
 NOTE: To make sure FreeCAD saves thumbnail information:
 
