From 3ebde5d1cc37007366037720af7e1111aa4aa3b2 Mon Sep 17 00:00:00 2001
From: Ikey Doherty <ikey@solus-project.com>
Date: Wed, 25 May 2016 04:15:01 +0100
Subject: [PATCH] Disable Python build, required for multilib build

Signed-off-by: Ikey Doherty <ikey@solus-project.com>
---
 bindings/Makefile.am |  1 -
 configure.ac         | 28 ----------------------------
 2 files changed, 29 deletions(-)

diff --git a/bindings/Makefile.am b/bindings/Makefile.am
index d1f6d73..dccca6c 100644
--- a/bindings/Makefile.am
+++ b/bindings/Makefile.am
@@ -3,7 +3,6 @@ SUBDIRS =	\
 	vala
 else
 SUBDIRS =		\
-	python		\
 	vala
 endif
 
diff --git a/configure.ac b/configure.ac
index ee03390..9694899 100644
--- a/configure.ac
+++ b/configure.ac
@@ -181,32 +181,6 @@ AM_CONDITIONAL(BUILD_MONO_TEST, test x${have_nunit} = xyes)
 with_localinstall="no"
 AC_ARG_ENABLE(localinstall, AS_HELP_STRING([--enable-localinstall], [install all of the files localy instead of system directories (for distcheck)]), with_localinstall=$enableval, with_localinstall=no)
 
-###########################
-# Python
-###########################
-
-PYGTK_REQUIRED=2.14.0
-PYGOBJECT_REQUIRED=0.22
-
-AM_PATH_PYTHON(2.3.5)
-AM_CHECK_PYTHON_HEADERS(,[AC_MSG_ERROR(could not find Python headers)])
-
-PKG_CHECK_MODULES(APPINDICATOR_PYTHON,
-                  [
-					pygtk-2.0 >= $PYGTK_REQUIRED
-					gtk+-2.0 >= $GTK_REQUIRED_VERSION
-					pygobject-2.0 >= $PYGOBJECT_REQUIRED
-                  ])
-
-AC_MSG_CHECKING(for pygtk defs)
-PYGTK_DEFSDIR=`$PKG_CONFIG --variable=defsdir pygtk-2.0`
-AC_SUBST(PYGTK_DEFSDIR)
-AC_MSG_RESULT($PYGTK_DEFSDIR)
-
-AC_MSG_CHECKING(for pygtk codegen)
-PYGTK_CODEGEN="$PYTHON `$PKG_CONFIG --variable=codegendir pygtk-2.0`/codegen.py"
-AC_SUBST(PYGTK_CODEGEN)
-AC_MSG_RESULT($PYGTK_CODEGEN)
 
 #########################
 # Check if build tests
@@ -239,8 +213,6 @@ src/Makefile
 src/appindicator-0.1.pc
 src/appindicator3-0.1.pc
 bindings/Makefile
-bindings/python/Makefile
-bindings/python/appindicator.override
 bindings/vala/Makefile
 bindings/vala/examples/Makefile
 tests/Makefile
-- 
2.8.3

