From 448b236d893170edc8fc4dff029044948516352b Mon Sep 17 00:00:00 2001
From: Rudi Heitbaum <rudi@heitbaum.com>
Date: Sun, 27 Apr 2025 15:03:59 +1000
Subject: [PATCH] Allow build with gcc-15 by setting -std=gnu17

---
 kernel-open/Kbuild | 1 +
 1 file changed, 1 insertion(+)

diff --git a/kernel-open/Kbuild b/kernel-open/Kbuild
index 44f6e55d5..09373cf31 100644
--- a/kernel-open/Kbuild
+++ b/kernel-open/Kbuild
@@ -87,6 +87,7 @@ EXTRA_CFLAGS += -I$(src)
 EXTRA_CFLAGS += $(call cc-option,-Werror=undef,)
 EXTRA_CFLAGS += -DNV_SPECTRE_V2=$(NV_SPECTRE_V2)
 EXTRA_CFLAGS += -DNV_KERNEL_INTERFACE_LAYER
+EXTRA_CFLAGS += -std=gnu17
 
 #
 # Detect SGI UV systems and apply system-specific optimizations.
