--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -55,14 +55,14 @@ endif()
 #######################################
 
 # Include these two modules without enable/disable options
-add_subdirectory(IlmBase)
+# add_subdirectory(IlmBase)
 
 # Tell CMake where to find the IlmBaseConfig.cmake file. Makes it posible to call 
 # find_package(IlmBase) in downstream projects
-set(IlmBase_DIR "${CMAKE_CURRENT_BINARY_DIR}/IlmBase/config" CACHE PATH "" FORCE)
+# set(IlmBase_DIR "${CMAKE_CURRENT_BINARY_DIR}/IlmBase/config" CACHE PATH "" FORCE)
 # Add an empty IlmBaseTargets.cmake file for the config to use. 
 # Can be empty since we already defined the targets in add_subdirectory
-file(WRITE "${CMAKE_CURRENT_BINARY_DIR}/IlmBase/config/IlmBaseTargets.cmake" "# Dummy file")
+# file(WRITE "${CMAKE_CURRENT_BINARY_DIR}/IlmBase/config/IlmBaseTargets.cmake" "# Dummy file")
 
 add_subdirectory(OpenEXR)
 
