From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Reilly Brogan <reilly@reillybrogan.com>
Date: Wed, 24 Apr 2024 16:25:14 -0500
Subject: [PATCH 1/2] Unpack the package.json so we can fix the Wayland AppID

---
 electron-builder.ts | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/electron-builder.ts b/electron-builder.ts
index 6e48290..3160869 100644
--- a/electron-builder.ts
+++ b/electron-builder.ts
@@ -59,7 +59,7 @@ const config: Omit<Writable<Configuration>, "electronFuses"> & {
     electronFuses: Required<Configuration["electronFuses"]>;
 } = {
     appId: DEFAULT_APP_ID,
-    asarUnpack: "**/*.node",
+    asarUnpack: ["**/*.node", "package.json"],
     electronFuses: {
         enableCookieEncryption: true,
         onlyLoadAppFromAsar: true,
