From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Reilly Brogan <reilly@reillybrogan.com>
Date: Wed, 24 Apr 2024 16:25:14 -0500
Subject: [PATCH] Unpack package.json

---
 electron-builder.ts | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/electron-builder.ts b/electron-builder.ts
index dedb5f7..8cc0570 100644
--- a/electron-builder.ts
+++ b/electron-builder.ts
@@ -55,7 +55,7 @@ interface Configuration extends BaseConfiguration {
  */
 const config: Writable<Configuration> = {
     appId: "im.riot.app",
-    asarUnpack: "**/*.node",
+    asarUnpack: [ "**/*.node", "package.json"],
     afterPack: async (context: AfterPackContext) => {
         if (context.electronPlatformName !== "darwin" || context.arch === Arch.universal) {
             // Burn in electron fuses for proactive security hardening.
