From 3a42f14bafb7b7b3839737c2d275a9e600334a2f Mon Sep 17 00:00:00 2001
From: Reilly Brogan <reilly@reillybrogan.com>
Date: Sat, 6 Jan 2024 18:49:26 -0600
Subject: [PATCH] Migrate data from im.riot.Riot directory

---
 src/electron-main.ts | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/electron-main.ts b/src/electron-main.ts
index b5d13ac..f05d026 100644
--- a/src/electron-main.ts
+++ b/src/electron-main.ts
@@ -79,7 +79,7 @@ if (userDataPathInProtocol) {
         newUserDataPath += "-" + argv["profile"];
     }
     const newUserDataPathExists = isRealUserDataDir(newUserDataPath);
-    let oldUserDataPath = path.join(app.getPath("appData"), app.getName().replace("Element", "Riot"));
+    let oldUserDataPath = path.join(app.getPath("appData"), "Element");
     if (argv["profile"]) {
         oldUserDataPath += "-" + argv["profile"];
     }
-- 
2.43.0

