From bc18487067e998f1a66a9d863ad7444d54662296 Mon Sep 17 00:00:00 2001
From: Reilly Brogan <reilly@reillybrogan.com>
Date: Wed, 24 Apr 2024 16:25:14 -0500
Subject: [PATCH 1/2] Unpack the package.json so we can fix the Wayland AppID

---
 electron-builder.ts | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/electron-builder.ts b/electron-builder.ts
index abf0ca6..383a489 100644
--- a/electron-builder.ts
+++ b/electron-builder.ts
@@ -75,7 +75,7 @@ const config: Omit<Writable<Configuration>, "electronFuses"> & {
     electronFuses: Required<Configuration["electronFuses"]>;
 } = {
     appId: "im.riot.app",
-    asarUnpack: "**/*.node",
+    asarUnpack: [ "**/*.node", "package.json"],
     electronFuses: {
         enableCookieEncryption: true,
         onlyLoadAppFromAsar: true,
-- 
2.47.2

