From 5c49bc8d18c10ba0490148fcf1a636d85e28c0b5 Mon Sep 17 00:00:00 2001
From: Ikey Doherty <ikey@solus-project.com>
Date: Sat, 15 Oct 2016 20:47:10 +0100
Subject: [PATCH 1/4] Fix the broken launcher on LiveCD (incorrect desktop
 name)

Signed-off-by: Ikey Doherty <ikey@solus-project.com>
---
 panel/solus-live.layout | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/panel/solus-live.layout b/panel/solus-live.layout
index 8a8603c..e5dd87c 100644
--- a/panel/solus-live.layout
+++ b/panel/solus-live.layout
@@ -18,7 +18,7 @@ locked=true
 
 [Object install]
 object-type=launcher
-launcher-location=/usr/share/applications/os-installer.desktop
+launcher-location=/usr/share/applications/os-installer-gtk.desktop
 toplevel-id=bottom
 position=20
 locked=true
-- 
2.10.1

