From 1a1882e7731dfdb25a572b8ccf65fc72b0296939 Mon Sep 17 00:00:00 2001
From: Campbell Jones <git@serebit.com>
Date: Mon, 8 Jun 2020 19:58:01 -0400
Subject: [PATCH] fix jdk search and use java 11

---
 projectlibre_build/resources/projectlibre | 20 +++++++++++++-------
 1 file changed, 13 insertions(+), 7 deletions(-)

diff --git a/projectlibre_build/resources/projectlibre b/projectlibre_build/resources/projectlibre
index 9f38901..c460cc1 100755
--- a/projectlibre_build/resources/projectlibre
+++ b/projectlibre_build/resources/projectlibre
@@ -56,15 +56,21 @@
 # logo it must direct them back to http://www.projectlibre.com. 
 #*******************************************************************************
 
-CONF_DIR="$HOME/.projectlibre"
+if [ -d "$XDG_CONFIG_HOME" ]; then
+    CONF_DIR="$XDG_CONFIG_HOME/projectlibre"
+else
+    CONF_DIR="$HOME/.config/projectlibre"
+fi
+
+# Migrate the old conf dir
+if [ -d "$HOME/.projectlibre" ]; then
+    mv "$HOME/.projectlibre" "$CONF_DIR"
+fi
+
 RUN_CONF="$CONF_DIR/run.conf"
 VERSION_FILE="$CONF_DIR/java-version"
-JAVA_INSTALL_DIR="/usr/java"
+JAVA_INSTALL_DIR="/usr/lib64/openjdk-11"
 JAVA_INSTALL_DIR_EXE="bin/java"
-JAVA_INSTALL_DIR2="/usr/lib/jvm"
-JAVA_INSTALL_DIR_EXE2="bin/java"
-JAVA_INSTALL_DIR3="/opt/ibm"
-JAVA_INSTALL_DIR_EXE3="jre/bin/java"
 MIN_JAVA_VERSION="6"
 
 #Default conf values
@@ -239,7 +245,7 @@ if [ "$JAVA_OK" -eq "0" ]; then
 			check_java "$J"
 		fi
 		if [ "$JAVA_OK" -eq "0" ]; then
-			J="$JAVA_INSTALL_DIR/default/$JAVA_INSTALL_DIR_EXE"
+			J="$JAVA_INSTALL_DIR/$JAVA_INSTALL_DIR_EXE"
 			if [ -x "$J" ]; then
 				check_java "$J"
 			fi
-- 
2.27.0

