From 01b61a2f50e92d266b5c833ba9f790fbe190b780 Mon Sep 17 00:00:00 2001
From: Campbell Jones <git@serebit.com>
Date: Fri, 3 Jan 2020 19:15:16 -0500
Subject: [PATCH] remove usages of git as we use a tarball

---
 Makefile | 4 +---
 1 file changed, 1 insertion(+), 3 deletions(-)

diff --git a/Makefile b/Makefile
index 76da995..33c94a3 100644
--- a/Makefile
+++ b/Makefile
@@ -2,13 +2,12 @@
 
 VERSION = $(shell GOOS=$(shell go env GOHOSTOS) GOARCH=$(shell go env GOHOSTARCH) \
 	go run tools/build-version.go)
-HASH = $(shell git rev-parse --short HEAD)
 DATE = $(shell GOOS=$(shell go env GOHOSTOS) GOARCH=$(shell go env GOHOSTARCH) \
 	go run tools/build-date.go)
 ADDITIONAL_GO_LINKER_FLAGS = $(shell GOOS=$(shell go env GOHOSTOS) \
 	GOARCH=$(shell go env GOHOSTARCH))
 GOBIN ?= $(shell go env GOPATH)/bin
-GOVARS = -X github.com/zyedidia/micro/internal/util.Version=$(VERSION) -X github.com/zyedidia/micro/internal/util.CommitHash=$(HASH) -X 'github.com/zyedidia/micro/internal/util.CompileDate=$(DATE)' -X github.com/zyedidia/micro/internal/util.Debug=OFF
+GOVARS = -X github.com/zyedidia/micro/internal/util.Version=$(VERSION) -X 'github.com/zyedidia/micro/internal/util.CompileDate=$(DATE)' -X github.com/zyedidia/micro/internal/util.Debug=OFF
 
 # Builds micro after checking dependencies but without updating the runtime
 build:
@@ -37,7 +36,6 @@ install-quick:
 
 # Builds the runtime
 runtime:
-	git submodule update --init
 	go run runtime/syntax/make_headers.go runtime/syntax
 	go build -o tools/bindata ./tools/go-bindata
 	tools/bindata -pkg config -nomemcopy -nometadata -o runtime.go runtime/...
-- 
2.24.1

