From 3cb22838865f77b552ca094085ccb73bd7ae91cc Mon Sep 17 00:00:00 2001
From: Joshua Strobl <joshua@stroblindustries.com>
Date: Sat, 11 Aug 2018 13:53:13 +0300
Subject: [PATCH 1/1] Drop git pull on update since we're using a tarball.

---
 Makefile | 1 -
 1 file changed, 1 deletion(-)

diff --git a/Makefile b/Makefile
index e03ee2f..c68d2f7 100644
--- a/Makefile
+++ b/Makefile
@@ -33,7 +33,6 @@ install-quick:
 	go install -ldflags "-s -w -X main.Version=$(VERSION) -X main.CommitHash=$(HASH) -X 'main.CompileDate=$(DATE)' $(ADDITIONAL_GO_LINKER_FLAGS)"  ./cmd/micro
 
 update:
-	git pull
 	git submodule update --init
 
 # Builds the runtime
-- 
2.18.0

