From 89ca6513651d0afd31af1f84608c47a52669bf84 Mon Sep 17 00:00:00 2001
From: Gavin Zhao <git@gzgz.dev>
Date: Sat, 25 Jul 2026 20:21:26 -0700
Subject: [PATCH] amdsmi: Don't install libamdsminic.a

---
 projects/amdsmi/src/CMakeLists.txt | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/projects/amdsmi/src/CMakeLists.txt b/projects/amdsmi/src/CMakeLists.txt
index dc2c762d35..33d39b7393 100644
--- a/projects/amdsmi/src/CMakeLists.txt
+++ b/projects/amdsmi/src/CMakeLists.txt
@@ -130,7 +130,7 @@ if(BUILD_BOTH_LIBS)
 else()
     add_library(${AMD_SMI} ${SRC_LIST} ${INC_LIST})
     set(AMD_SMI_TARGET ${AMD_SMI} PARENT_SCOPE)
-    set(AMD_SMI_EXPORT_TARGETS ${AMD_SMI} amdsminic PARENT_SCOPE)
+    set(AMD_SMI_EXPORT_TARGETS ${AMD_SMI} PARENT_SCOPE)
     set(_AMD_SMI_ALL_TARGETS ${AMD_SMI})
 endif()
 
@@ -209,7 +209,7 @@ set_property(TARGET ${AMD_SMI} PROPERTY SOVERSION "${MAJOR}")
 set_property(TARGET ${AMD_SMI} PROPERTY VERSION "${SO_VERSION_STRING}")
 
 ## Add the install directives for the runtime library.
-set(_INSTALL_TARGETS ${AMD_SMI} amdsminic)
+set(_INSTALL_TARGETS ${AMD_SMI})
 if(BUILD_BOTH_LIBS)
     list(APPEND _INSTALL_TARGETS ${AMD_SMI}_static)
 endif()
-- 
2.55.0

