From a9ad6452f72ee73674a2cad901ca72e33dbc3fa3 Mon Sep 17 00:00:00 2001
From: Johannes Marbach <JohannesMarbach@gmail.com>
Date: Sat, 29 Dec 2018 08:29:31 +0100
Subject: [PATCH 1/1] Customise theme and icons

---
 etc/gimprc.in | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/etc/gimprc.in b/etc/gimprc.in
index c923d39159..b7b542f8c0 100644
--- a/etc/gimprc.in
+++ b/etc/gimprc.in
@@ -729,7 +729,7 @@
 
 # The name of the theme to use.  This is a string value.
 # 
-# (theme "Dark")
+(theme "System")
 
 # Sets the icon theme search path.  This is a colon-separated list of folders
 # to search.
@@ -743,7 +743,7 @@
 # The size of the icons to use.  Possible values are auto, theme, small,
 # medium, large and huge.
 # 
-# (icon-size auto)
+(icon-size theme)
 
 # When enabled, pressing F1 will open the help browser.  Possible values are
 # yes and no.
-- 
2.20.1

