diff '--color=auto' -pruN a/meson.build b/meson.build
--- a/meson.build	2025-10-05 13:14:02.000000000 -0400
+++ b/meson.build	2025-11-05 17:26:52.772129229 -0500
@@ -456,7 +456,7 @@ add_project_arguments('-DGLIB_VERSION_MI
 # minimum requirement.
 add_project_arguments('-DGLIB_VERSION_MAX_ALLOWED=GLIB_VERSION_2_76', language: 'c')
 
-gi                = dependency('gobject-introspection-1.0')
+gi                = dependency('girepository-2.0')
 
 gobject           = dependency('gobject-2.0',        version: '>='+glib_minver)
 gmodule           = dependency('gmodule-no-export-2.0')
diff '--color=auto' -pruN a/plug-ins/script-fu/libscriptfu/scheme-wrapper.c b/plug-ins/script-fu/libscriptfu/scheme-wrapper.c
--- a/plug-ins/script-fu/libscriptfu/scheme-wrapper.c	2025-10-05 13:14:02.000000000 -0400
+++ b/plug-ins/script-fu/libscriptfu/scheme-wrapper.c	2025-11-05 17:22:02.146713961 -0500
@@ -24,7 +24,7 @@
 
 #include <glib/gstdio.h>
 
-#include <girepository.h>
+#include <girepository/girepository.h>
 
 #include <gtk/gtk.h>
 
@@ -249,8 +249,8 @@ tinyscheme_init (GList    *path,
   script_fu_regex_init (&sc);
 
   /* Fetch the typelib */
-  repo = g_irepository_get_default ();
-  typelib = g_irepository_require (repo, "Gimp", NULL, 0, &error);
+  repo = gi_repository_dup_default ();
+  typelib = gi_repository_require (repo, "Gimp", NULL, 0, &error);
   if (!typelib)
     {
       g_warning ("%s", error->message);
@@ -441,7 +441,7 @@ ts_init_enum (scheme     *sc,
 {
   int n_values;
 
-  n_values = g_enum_info_get_n_values ((GIEnumInfo *) info);
+  n_values = gi_enum_info_get_n_values ((GIEnumInfo *) info);
   for (int j = 0; j < n_values; j++)
     {
       GIValueInfo *value_info;
@@ -450,15 +450,15 @@ ts_init_enum (scheme     *sc,
       int         int_value;
       pointer     symbol;
 
-      value_info = g_enum_info_get_value (info, j);
+      value_info = gi_enum_info_get_value (info, j);
 
       /* Get name & value. Normally, we would use the actual name of the
        * GIBaseInfo here, but that would break bw-compatibility */
-      c_identifier = g_base_info_get_attribute ((GIBaseInfo *) value_info, "c:identifier");
+      c_identifier = gi_base_info_get_attribute ((GIBaseInfo *) value_info, "c:identifier");
       if (c_identifier == NULL)
         {
           g_warning ("Problem in the GIR file: enum value without \"c:identifier\"!");
-          g_base_info_unref ((GIBaseInfo *) value_info);
+          gi_base_info_unref ((GIBaseInfo *) value_info);
           continue;
         }
 
@@ -474,12 +474,12 @@ ts_init_enum (scheme     *sc,
           /* Other namespaces: skip non-prefixed symbols, to prevent clashes */
           if (g_ascii_strncasecmp (c_identifier, namespace, strlen (namespace)) != 0)
             {
-              g_base_info_unref ((GIBaseInfo *) value_info);
+              gi_base_info_unref ((GIBaseInfo *) value_info);
               continue;
             }
         }
       scheme_name = g_strdelimit (g_strdup (c_identifier), "_", '-');
-      int_value = g_value_info_get_value (value_info);
+      int_value = gi_value_info_get_value (value_info);
 
       /* Register the symbol */
       symbol = sc->vptr->mk_symbol (sc, scheme_name);
@@ -488,7 +488,7 @@ ts_init_enum (scheme     *sc,
       sc->vptr->setimmutable (symbol);
 
       g_free (scheme_name);
-      g_base_info_unref ((GIBaseInfo *) value_info);
+      gi_base_info_unref ((GIBaseInfo *) value_info);
     }
 }
 
@@ -499,18 +499,18 @@ ts_init_enums (scheme       *sc,
 {
   int i, n_infos;
 
-  n_infos = g_irepository_get_n_infos (repo, namespace);
+  n_infos = gi_repository_get_n_infos (repo, namespace);
   for (i = 0; i < n_infos; i++)
     {
       GIBaseInfo *info;
 
-      info = g_irepository_get_info (repo, namespace, i);
+      info = gi_repository_get_info (repo, namespace, i);
       if (GI_IS_ENUM_INFO (info))
         {
           ts_init_enum (sc, (GIEnumInfo *) info, namespace);
         }
 
-      g_base_info_unref (info);
+      gi_base_info_unref (info);
     }
 }
 
