From b9f2d8d26b54a5e2b82b0e9f6efdfa72d1a39c76 Mon Sep 17 00:00:00 2001
From: Fabio Forni <fabio@redaril.me>
Date: Mon, 23 Aug 2021 21:17:58 +0200
Subject: [PATCH] Use AyatanaAppIndicator

---
 polychromatic-tray-applet | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/polychromatic-tray-applet b/polychromatic-tray-applet
index f66c0cb..cb3fa1f 100755
--- a/polychromatic-tray-applet
+++ b/polychromatic-tray-applet
@@ -21,8 +21,8 @@
 import gi
 gi.require_version('Gtk', '3.0')
 gi.require_version('Gdk', '3.0')
-gi.require_version('AppIndicator3', '0.1')
-from gi.repository import Gtk, Gdk, AppIndicator3 as appindicator
+gi.require_version('AyatanaAppIndicator3', '0.1')
+from gi.repository import Gtk, Gdk, AyatanaAppIndicator3 as appindicator
 import argparse
 import collections
 import os
-- 
2.32.0

