From 385014e344c7029b56e772f2401e13c7f11e550b Mon Sep 17 00:00:00 2001
From: "Bryan T. Meyers" <root@datadrake.com>
Date: Thu, 26 Aug 2021 21:41:31 -0400
Subject: [PATCH] Use AyatanaAppIndicator

---
 polychromatic-tray-applet | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/polychromatic-tray-applet b/polychromatic-tray-applet
index 17b5971..cb37e38 100755
--- a/polychromatic-tray-applet
+++ b/polychromatic-tray-applet
@@ -37,8 +37,8 @@ gi.require_version("Gdk", "3.0")
 from gi.repository import Gtk, Gdk
 
 try:
-    gi.require_version("AppIndicator3", "0.1")
-    from gi.repository import AppIndicator3 as appindicator
+    gi.require_version("AyatanaAppIndicator3", "0.1")
+    from gi.repository import AyatanaAppIndicator3 as appindicator
     APPINDICATOR = True
 except:
     APPINDICATOR = False
-- 
2.32.0

