diff --git a/tensorflow/BUILD b/tensorflow/BUILD
index f72931217b3..b45b279ff69 100644
--- a/tensorflow/BUILD
+++ b/tensorflow/BUILD
@@ -1178,7 +1178,7 @@ tf_cc_shared_library(
             "-lexecinfo",
         ],
         "//conditions:default": [
-            "-Wl,--version-script,$(location //tensorflow:tf_framework_version_script.lds)",
+            # "-Wl,--version-script,$(location //tensorflow:tf_framework_version_script.lds)",
             # copybara:uncomment_begin(google-only)
             # "-Wl,--undefined-version",
             # copybara:uncomment_end(google-only)
diff --git a/tensorflow/tensorflow.bzl b/tensorflow/tensorflow.bzl
index 6762ccd8f9b..5911035094a 100644
--- a/tensorflow/tensorflow.bzl
+++ b/tensorflow/tensorflow.bzl
@@ -2333,7 +2333,8 @@ def _append_init_to_versionscript_impl(ctx):
             template = ctx.file.template_file,
             output = ctx.outputs.versionscript,
             substitutions = {
-                "global:": "global:\n     init_%s;\n     _init_%s;\n     PyInit_*;\n     _PyInit_*;" % (mod_name, mod_name),
+                # "global:": "global:\n     init_%s;\n     _init_%s;\n     PyInit_*;\n     _PyInit_*;" % (mod_name, mod_name),
+                "global:": "global:\n     PyInit_*;\n     _PyInit_*;",
             },
             is_executable = False,
         )
@@ -3037,8 +3038,8 @@ def pybind_extension_opensource(
     filegroup_name = "%s_filegroup" % name
     pyd_file = "%s%s.pyd" % (prefix, sname)
     exported_symbols = [
-        "init%s" % sname,
-        "init_%s" % sname,
+        # "init%s" % sname,
+        # "init_%s" % sname,
         "PyInit_%s" % sname,
     ] + additional_exported_symbols
 
diff --git a/third_party/xla/third_party/tsl/tsl/tsl.bzl b/third_party/xla/third_party/tsl/tsl/tsl.bzl
index e1c2b364fbc..bc9922f67e8 100644
--- a/third_party/xla/third_party/tsl/tsl/tsl.bzl
+++ b/third_party/xla/third_party/tsl/tsl/tsl.bzl
@@ -584,8 +584,8 @@ def tsl_pybind_extension_opensource(
     filegroup_name = "%s_filegroup" % name
     pyd_file = "%s%s.pyd" % (prefix, sname)
     exported_symbols = [
-        "init%s" % sname,
-        "init_%s" % sname,
+        # "init%s" % sname,
+        # "init_%s" % sname,
         "PyInit_%s" % sname,
     ] + additional_exported_symbols
 
