--- tensorflow-2.19.0/third_party/tf_runtime/workspace.bzl.old	2025-06-05 10:20:31.329424254 +0000
+++ tensorflow-2.19.0/third_party/tf_runtime/workspace.bzl	2025-06-05 10:21:30.553156829 +0000
@@ -20,5 +20,5 @@
         },
         # A patch file can be provided for atomic commits to both TF and TFRT.
         # The job that bumps the TFRT_COMMIT also resets patch_file to 'None'.
-        patch_file = None,
+        patch_file = ["//third_party/tf_runtime:0001-Add-missing-cstdint-includes-for-gcc-stdc-15.patch"],
     )
