From b0b99c8571c361e2281c12e0a91b810df37d5acf Mon Sep 17 00:00:00 2001
From: Peter O'Connor <peter@solus-project.com>
Date: Sun, 11 Jun 2017 15:04:06 +1000
Subject: [PATCH] Use Solus defaults for build

---
 Makeconf     | 8 ++++----
 Makeconf.g95 | 8 ++++----
 2 files changed, 8 insertions(+), 8 deletions(-)

diff --git a/Makeconf b/Makeconf
index 72572f1..4257df0 100644
--- a/Makeconf
+++ b/Makeconf
@@ -6,21 +6,21 @@ FFLAGS=-fimplicit-none -O3 -funroll-loops
 FPICFLAGS=-fPIC

 # BLAS library (only required for tests)
-BLAS=-lblas
+BLAS=-lopenblas
 # LAPACK library (only required for tests)
 LAPACK=-llapack

 # Library version
 VERSION=1.1
 MAJOR=1

 # The default library dir
-LIBDIR=lib
+LIBDIR=lib64

 # Destination installation offset
 DESTDIR=

 # set default prefix to /usr/local
 ifeq ($(strip $(PREFIX)),)
-  PREFIX=/usr/local
+  PREFIX=/usr
 endif
diff --git a/Makeconf.g95 b/Makeconf.g95
index b148ad6..16620cf 100644
--- a/Makeconf.g95
+++ b/Makeconf.g95
@@ -6,21 +6,21 @@ FFLAGS=-O3 -funroll-loops
 FPICFLAGS=-fPIC

 # BLAS library (only required for tests)
-BLAS=-lblas
+BLAS=-lopenblas
 # LAPACK library (only required for tests)
 LAPACK=-llapack

 # Library version
 VERSION=1.1
 MAJOR=1

 # The default library dir
-LIBDIR=lib
+LIBDIR=lib64

 # Destination installation offset
 DESTDIR=

 # set default prefix to /usr/local
 ifeq ($(strip $(PREFIX)),)
-  PREFIX=/usr/local
+  PREFIX=/usr
 endif
--
2.13.1
