From a1e47d4ae605cf447354b90f18b58e1bf1ec9451 Mon Sep 17 00:00:00 2001
From: Ikey Doherty <michael.i.doherty@intel.com>
Date: Sat, 12 Mar 2016 10:01:43 +0000
Subject: [PATCH] Use vendor config files as fallback for a stateless
 implementation

Signed-off-by: Ikey Doherty <michael.i.doherty@intel.com>
---
 src/getnetconfig.c | 12 ++++++++++--
 tirpc/netconfig.h  |  5 +++++
 2 files changed, 15 insertions(+), 2 deletions(-)

diff --git a/src/getnetconfig.c b/src/getnetconfig.c
index 92e7c43..d0af5c0 100644
--- a/src/getnetconfig.c
+++ b/src/getnetconfig.c
@@ -123,6 +123,14 @@ extern pthread_mutex_t nc_db_lock;
 
 #define MAXNETCONFIGLINE    1000
 
+static inline const char *__nc_get_config_file(void)
+{
+	if (access(NETCONFIG, F_OK) == 0) {
+		return NETCONFIG;
+	}
+	return SYSTEM_NETCONFIG;
+}
+
 static int *
 __nc_error()
 {
@@ -194,7 +202,7 @@ setnetconfig()
      */
     mutex_lock(&nc_db_lock);
     ni.ref++;
-    if ((nc_file != NULL) || (nc_file = fopen(NETCONFIG, "r")) != NULL) {
+    if ((nc_file != NULL) || (nc_file = fopen(__nc_get_config_file(), "r")) != NULL) {
 	nc_vars->valid = NC_VALID;
 	nc_vars->flag = 0;
 	nc_vars->nc_configs = ni.head;
@@ -468,7 +476,7 @@ getnetconfigent(netid)
     }
     mutex_unlock(&nc_db_lock);
 
-    if ((file = fopen(NETCONFIG, "r")) == NULL) {
+    if ((file = fopen(__nc_get_config_file(), "r")) == NULL) {
 	nc_error = NC_NONETCONFIG;
 	return (NULL);
     }
diff --git a/tirpc/netconfig.h b/tirpc/netconfig.h
index 3f1a685..83fd42e 100644
--- a/tirpc/netconfig.h
+++ b/tirpc/netconfig.h
@@ -5,7 +5,12 @@
 #include <features.h>
 #endif
 
+/* Local admin netconfig file */
 #define NETCONFIG "/etc/netconfig"
+
+/* Vendor netconfig file */
+#define SYSTEM_NETCONFIG    "/usr/share/defaults/etc/netconfig"
+
 #define NETPATH	  "NETPATH"
 
 struct netconfig {
-- 
2.7.1

