From fad90ec83c497abeeeb292bb5acbd1273d9459f4 Mon Sep 17 00:00:00 2001
From: Joshua Strobl <joshua@stroblindustries.com>
Date: Tue, 11 Apr 2017 14:30:57 +0300
Subject: [PATCH 1/1] Tweak configure.ac

---
 configure.ac | 10 +++-------
 1 file changed, 3 insertions(+), 7 deletions(-)

diff --git a/configure.ac b/configure.ac
index 542d706b5..7c373f2b9 100644
--- a/configure.ac
+++ b/configure.ac
@@ -28,7 +28,9 @@ AC_DEFINE_UNQUOTED(GETTEXT_PACKAGE,"$GETTEXT_PACKAGE", [gettext package])
 # User Documentation
 YELP_HELP_INIT
 
-PKG_CHECK_MODULES([PYGOBJECT], [pygobject-3.0 >= pygobject_required_version])
+AM_PATH_PYTHON(3.5)
+PKG_CHECK_MODULES(PYTHON, [pygobject-3.0 >= pygobject_required_version])
+
 PKG_CHECK_MODULES([ATSPI2], [atspi-2 >= atspi_required_version])
 PKG_CHECK_MODULES([ATKBRIDGE], [atk-bridge-2.0 >= atkbridge_required_version])
 PKG_CHECK_MODULES([GSTREAMER], [gstreamer-1.0], [gstreamer="yes"], [gstreamer="no"])
@@ -36,16 +38,10 @@ PKG_CHECK_MODULES([GSTREAMER], [gstreamer-1.0], [gstreamer="yes"], [gstreamer="n
 dnl Needed programs
 AC_PROG_INSTALL
 
-dnl Check for python and python modules needed by Orca
-dnl TODO: check for incompatibilities between Orca and liblouis
-dnl per http://live.gnome.org/Orca/Braille
-AM_PATH_PYTHON(3.3)
 
 # Don't require the DISPLAY to be set - the gtk checking will fail 
 # if DISPLAY is not set, and we don't like that.
 
-AM_CHECK_PYMOD(gi,,,[AC_MSG_ERROR(Could not find python module: gi)])
-AM_CHECK_PYMOD(json,,,[AC_MSG_ERROR(Could not find python module: json)])
 AM_CHECK_PYMOD(brlapi,,[brlapi_available="yes"],[brlapi_available="no"])
 AM_CHECK_PYMOD(speechd,,[speechd_available="yes"],[speechd_available="no"])
 AC_ARG_WITH([liblouis],
-- 
2.12.2

