From 137e9591c56a8985b2331634c8e268533ffd36a1 Mon Sep 17 00:00:00 2001
From: Joshua Strobl <joshua@streambits.io>
Date: Sat, 26 Dec 2020 20:31:28 +0200
Subject: [PATCH 1/1] Support stateless path.

---
 libattr/attr_copy_action.c | 11 +++++++----
 1 file changed, 7 insertions(+), 4 deletions(-)

diff --git a/libattr/attr_copy_action.c b/libattr/attr_copy_action.c
index 279f2df..a6c7178 100644
--- a/libattr/attr_copy_action.c
+++ b/libattr/attr_copy_action.c
@@ -29,6 +29,7 @@
 #include "error_context.h"
 
 #define ATTR_CONF SYSCONFDIR "/xattr.conf"
+#define STATELESS_CONF "/usr/share/defaults/attr/xattr.conf"
 
 struct attr_action {
 	struct attr_action *next;
@@ -66,9 +67,11 @@ attr_parse_attr_conf(struct error_context *ctx)
 
 repeat:
 	if ((file = fopen(ATTR_CONF, "r")) == NULL) {
-		if (errno == ENOENT)
-			return 0;
-		goto fail;
+		if ((file = fopen(STATELESS_CONF, "r") == NULL)) {
+			if (errno == ENOENT)
+				return 0;
+			goto fail;
+		}
 	}
 
 	text = malloc(size_guess + 1);
@@ -139,7 +142,7 @@ parse_error:
 
 fail:
 	{
-		const char *q = quote (ctx, ATTR_CONF);
+		const char *q = quote (ctx, STATELESS_CONF);
 		error (ctx, "%s", q);
 		quote_free (ctx, q);
 	}
-- 
2.29.2

