From 3ee53ee5a672ab6b483cb6f001d1d17051e9dbed Mon Sep 17 00:00:00 2001
From: "F. von Gellhorn" <flinux@vongellhorn.ch>
Date: Wed, 20 Apr 2022 15:29:27 +0200
Subject: [PATCH] Disable spellchecker

---
 examples/webenginewidgets/CMakeLists.txt | 4 ----
 1 file changed, 4 deletions(-)

diff --git a/examples/webenginewidgets/CMakeLists.txt b/examples/webenginewidgets/CMakeLists.txt
index cc4a4d6c0..36591ca18 100644
--- a/examples/webenginewidgets/CMakeLists.txt
+++ b/examples/webenginewidgets/CMakeLists.txt
@@ -17,10 +17,6 @@ if(QT_FEATURE_webengine_printing_and_pdf)
     qt_internal_add_example(printme)
     qt_internal_add_example(html2pdf)
 endif()
-if(QT_FEATURE_webengine_spellchecker AND NOT CMAKE_CROSSCOMPILING
-    AND NOT QT_FEATURE_webengine_native_spellchecker AND NOT WIN32)
-    qt_internal_add_example(spellchecker)
-endif()
 if(QT_FEATURE_ssl)
     qt_internal_add_example(clientcertificate)
 endif()
-- 
2.41.0

