diff --git a/Makefile b/Makefile
index fb8863f6..c379d098 100644
--- a/Makefile
+++ b/Makefile
@@ -63,7 +63,7 @@ all: run
 pyenv:
 # 	https://github.com/PyO3/maturin/issues/283 - Expected `python` to be a python interpreter inside a virtualenv
 	set -eu -o pipefail ${SHELLFLAGS}; \
-	"${PYTHON_BIN}" -m venv pyenv; \
+	"${PYTHON_BIN}" -m venv --system-site-packages pyenv; \
 	case "$$(uname -s)" in CYGWIN*|MINGW*|MSYS*) \
 		dos2unix "${ACTIVATE_SCRIPT}"; \
 		VIRTUAL_ENV="$$(pwd)"; \
@@ -72,12 +72,7 @@ pyenv:
 		sed -i -- "s@export PATH@export PATH; VIRTUAL_ENV=\"$${VIRTUAL_ENV}/pyenv\";@g" "${ACTIVATE_SCRIPT}"; \
 		;; esac; \
 	. "${ACTIVATE_SCRIPT}"; \
-	python --version; \
-	python -m pip install --upgrade pip setuptools; \
-	${ANKI_EXTRA_PIP}; \
-	if ! python -c 'import PyQt5' 2>/dev/null; then \
-		python -m pip install -r qt/requirements.qt; \
-	fi;
+	python --version;
 
 # update build hash
 .PHONY: buildhash
@@ -115,7 +110,7 @@ qt/po/repo:
 	$(MAKE) pull-i18n
 
 .PHONY: build
-build: clean-dist build-ts build-rspy build-pylib build-qt add-buildhash
+build: prepare build-rspy build-pylib build-ts build-qt add-buildhash
 	@echo
 	@echo "Build complete."
 
@@ -127,7 +122,8 @@ build-ts:
 build-rspy: pyenv buildhash
 	@set -eu -o pipefail ${SHELLFLAGS}; \
 	. "${ACTIVATE_SCRIPT}"; \
-	$(SUBMAKE) -C rspy build BUILDFLAGS="$(BUILDFLAGS)"
+	$(SUBMAKE) -C rspy build BUILDFLAGS="$(BUILDFLAGS)"; \
+	python -m pip install ./dist/ankirspy-2.1.35-cp39-cp39-linux_x86_64.whl
 
 .PHONY: build-pylib
 build-pylib:
