diff --git a/Makefile b/Makefile
index e8fa05b6..2da77da1 100644
--- a/Makefile
+++ b/Makefile
@@ -63,8 +63,7 @@ pyenv:
 	# https://github.com/PyO3/maturin/issues/283
 	# Expected `python` to be a python interpreter inside a virtualenv
 	set -eu -o pipefail ${SHELLFLAGS}; \
-	"${PYTHON_BIN}" -m pip install virtualenv; \
-	"${PYTHON_BIN}" -m venv pyenv; \
+	"${PYTHON_BIN}" -m venv --system-site-packages pyenv; \
 	case "$$(uname -s)" in CYGWIN*|MINGW*|MSYS*) \
 		dos2unix "${ACTIVATE_SCRIPT}"; \
 		VIRTUAL_ENV="$$(pwd)"; \
@@ -74,11 +73,7 @@ pyenv:
 		;; esac; \
 	. "${ACTIVATE_SCRIPT}"; \
 	python --version; \
-	python -m pip install --upgrade pip setuptools; \
-	${ANKI_EXTRA_PIP}; \
-	if ! python -c 'import PyQt5' 2>/dev/null; then \
-		python -m pip install -r qt/requirements.qt; \
-	fi;
+	${ANKI_EXTRA_PIP};
 
 # update build hash
 .PHONY: buildhash
@@ -117,7 +112,7 @@ qt/po/repo:
 	$(MAKE) pull-i18n
 
 .PHONY: build
-build: clean-dist build-rspy build-pylib build-qt add-buildhash
+build:  buildhash prepare build-rspy build-pylib build-qt add-buildhash
 	@echo
 	@echo "Build complete."
 
