diff --git a/rslib/build.rs b/rslib/build.rs
index acc1ed09..797ddcec 100644
--- a/rslib/build.rs
+++ b/rslib/build.rs
@@ -210,11 +210,11 @@ fn main() -> std::io::Result<()> {
         .compile_protos(&["../proto/backend.proto"], &["../proto"])
         .unwrap();
     // rustfmt the protobuf code
-    let rustfmt = Command::new("rustfmt")
-        .arg(Path::new("src/backend_proto.rs"))
-        .status()
-        .unwrap();
-    assert!(rustfmt.success(), "rustfmt backend_proto.rs failed");
+    //let rustfmt = Command::new("rustfmt")
+    //    .arg(Path::new("src/backend_proto.rs"))
+    //    .status()
+    //    .unwrap();
+    //assert!(rustfmt.success(), "rustfmt backend_proto.rs failed");
 
     // write the other language ftl files
     let mut ftl_lang_dirs = vec!["./ftl/repo/core".to_string()];
diff --git a/rspy/Makefile b/rspy/Makefile
index 5eb9ea8f..e3336edb 100644
--- a/rspy/Makefile
+++ b/rspy/Makefile
@@ -88,9 +88,9 @@ RUST_TOOLCHAIN := $(shell cat rust-toolchain)
 
 .build/tools: requirements.txt rust-toolchain
 	python -m pip install -r requirements.txt
-	rustup toolchain install $(RUST_TOOLCHAIN)
-	rustup component add rustfmt-preview --toolchain $(RUST_TOOLCHAIN)
-	rustup component add clippy-preview --toolchain $(RUST_TOOLCHAIN)
+	#rustup toolchain install $(RUST_TOOLCHAIN)
+	#rustup component add rustfmt-preview --toolchain $(RUST_TOOLCHAIN)
+	#rustup component add clippy-preview --toolchain $(RUST_TOOLCHAIN)
 	@touch $@
 
 # we should not call clippy because it break things when running make check Mac OS
diff --git a/rslib/Makefile b/rslib/Makefile
index 7a18b53e..c0becde7 100644
--- a/rslib/Makefile
+++ b/rslib/Makefile
@@ -36,9 +36,9 @@ ALL_SOURCE := $(shell "${FIND}" src -type f | egrep -v "i18n/autogen|i18n/ftl|_p
 RUST_TOOLCHAIN := $(shell cat rust-toolchain)
 
 .build/rs-tools: rust-toolchain
-	rustup toolchain install $(RUST_TOOLCHAIN)
-	rustup component add rustfmt-preview --toolchain $(RUST_TOOLCHAIN)
-	rustup component add clippy-preview --toolchain $(RUST_TOOLCHAIN)
+	#rustup toolchain install $(RUST_TOOLCHAIN)
+	#rustup component add rustfmt-preview --toolchain $(RUST_TOOLCHAIN)
+	#rustup component add clippy-preview --toolchain $(RUST_TOOLCHAIN)
 	@touch $@
 
 .build/check: ftl/repo .build/rs-tools $(ALL_SOURCE)
