diff --git a/rslib/build.rs b/rslib/build.rs
index acc1ed09..797ddcec 100644
--- a/rslib/build.rs
+++ b/rslib/build.rs
@@ -210,11 +210,11 @@ fn main() -> std::io::Result<()> {
         .compile_protos(&["../proto/backend.proto"], &["../proto"])
         .unwrap();
     // rustfmt the protobuf code
-    let rustfmt = Command::new("rustfmt")
-        .arg(Path::new("src/backend_proto.rs"))
-        .status()
-        .unwrap();
-    assert!(rustfmt.success(), "rustfmt backend_proto.rs failed");
+    //let rustfmt = Command::new("rustfmt")
+    //    .arg(Path::new("src/backend_proto.rs"))
+    //    .status()
+    //    .unwrap();
+    //assert!(rustfmt.success(), "rustfmt backend_proto.rs failed");
 
     // write the other language ftl files
     let mut ftl_lang_dirs = vec!["./ftl/repo/core".to_string()];
diff --git a/rspy/rust-toolchain b/rspy/rust-toolchain
index 2bf5ad04..22bde816 100644
--- a/rspy/rust-toolchain
+++ b/rspy/rust-toolchain
@@ -1 +1 @@
-stable
+nightly-2020-12-27
