diff --git a/src/droidcam.c b/src/droidcam.c
index 7637e7e..7ed5a4d 100644
--- a/src/droidcam.c
+++ b/src/droidcam.c
@@ -50,7 +50,7 @@ struct settings g_settings = {0};
 extern const char *thread_cmd_val_str;
 extern char snd_device[32];
 extern char v4l2_device[32];
-const char *APP_ICON_FILE = "/opt/droidcam-icon.png";
+const char *APP_ICON_NAME = "droidcam";
 
 void * AudioThreadProc(void * args);
 void * VideoThreadProc(void * args);
@@ -421,7 +421,7 @@ ERROR:
 }
 
 static void add_indicator(GtkWidget *window) {
-	AppIndicator *indicator = app_indicator_new("droidcam", APP_ICON_FILE, APP_INDICATOR_CATEGORY_APPLICATION_STATUS);
+	AppIndicator *indicator = app_indicator_new("droidcam", APP_ICON_NAME, APP_INDICATOR_CATEGORY_APPLICATION_STATUS);
 	GtkWidget *menu = gtk_menu_new();
 	GtkWidget *name_menu_item = gtk_menu_item_new_with_label("Droidcam");
 	GtkWidget *show_menu_item = gtk_menu_item_new_with_label("Show");
@@ -468,7 +468,7 @@ int main(int argc, char *argv[])
 	gtk_window_set_resizable(GTK_WINDOW(window), FALSE);
 	gtk_window_set_position(GTK_WINDOW(window), GTK_WIN_POS_NONE);
 	gtk_container_set_border_width(GTK_CONTAINER(window), 4);
-	gtk_window_set_icon_from_file(GTK_WINDOW(window), APP_ICON_FILE, NULL);
+	gtk_window_set_icon_name(GTK_WINDOW(window), APP_ICON_NAME);
 
 	// keyboard shortcuts
 	gtk_accel = gtk_accel_group_new ();
