From 0cbb72fd33063e954928a33e594b6d3aef6dad44 Mon Sep 17 00:00:00 2001
From: Joshua Strobl <joshua@stroblindustries.com>
Date: Sat, 16 Sep 2017 22:59:54 +0300
Subject: [PATCH 1/1] Fix reference to lua 5.1

---
 include/tolua++.h | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/include/tolua++.h b/include/tolua++.h
index ed53449..e3ddfb9 100644
--- a/include/tolua++.h
+++ b/include/tolua++.h
@@ -43,8 +43,8 @@ extern "C" {
 
 typedef int lua_Object;
 
-#include "lua.h"
-#include "lauxlib.h"
+#include "lua5.1/lua.h"
+#include "lua5.1/lauxlib.h"
 
 struct tolua_Error
 {
-- 
2.14.1

