From c9698e59db3f5777d36688048bcb60fc1d9b908d Mon Sep 17 00:00:00 2001
From: Thomas Staudinger <Staudi.Kaos@gmail.com>
Date: Wed, 2 Nov 2022 00:50:59 +0100
Subject: [PATCH] Don't install angelscript devel files

---
 lib/angelscript/projects/cmake/CMakeLists.txt | 48 -------------------
 1 file changed, 48 deletions(-)

diff --git a/lib/angelscript/projects/cmake/CMakeLists.txt b/lib/angelscript/projects/cmake/CMakeLists.txt
index 42883d5e3..8bdcfc2fe 100644
--- a/lib/angelscript/projects/cmake/CMakeLists.txt
+++ b/lib/angelscript/projects/cmake/CMakeLists.txt
@@ -210,51 +210,3 @@ endif()
 if(MSVC AND MSVC_COMPILE_FLAGS)
 	target_compile_options(${ANGELSCRIPT_LIBRARY_NAME} PRIVATE "${MSVC_COMPILE_FLAGS}")
 endif()
-
-# Don't override the default runtime output path to avoid conflicts when building for multiple target platforms
-#set(RUNTIME_OUTPUT_DIRECTORY ${PROJECT_SOURCE_DIR}/../../bin)
-
-#See https://cmake.org/cmake/help/latest/manual/cmake-packages.7.html#creating-packages for a detailed explanation about this part
-install(TARGETS ${ANGELSCRIPT_LIBRARY_NAME} EXPORT AngelscriptTargets
-	RUNTIME DESTINATION bin
-	LIBRARY DESTINATION lib
-	ARCHIVE DESTINATION lib
-	INCLUDES DESTINATION include
-)
-
-install(FILES
-	${CMAKE_CURRENT_SOURCE_DIR}/../../include/angelscript.h
-	DESTINATION include
-	COMPONENT Devel
-)
-
-include(CMakePackageConfigHelpers)
-write_basic_package_version_file(
-	"${CMAKE_CURRENT_BINARY_DIR}/Angelscript/AngelscriptConfigVersion.cmake"
-	VERSION ${PROJECT_VERSION}
-	COMPATIBILITY AnyNewerVersion
-)
-
-export(EXPORT AngelscriptTargets
-	FILE "${CMAKE_CURRENT_BINARY_DIR}/Angelscript/AngelscriptTargets.cmake"
-	NAMESPACE Angelscript::
-)
-configure_file(cmake/AngelscriptConfig.cmake
-	"${CMAKE_CURRENT_BINARY_DIR}/Angelscript/AngelscriptConfig.cmake"
-	COPYONLY
-)
-
-set(ConfigPackageLocation lib/cmake/Angelscript)
-install(EXPORT AngelscriptTargets
-	FILE AngelscriptTargets.cmake
-	NAMESPACE Angelscript::
-	DESTINATION ${ConfigPackageLocation}
-)
-install(
-	FILES
-		cmake/AngelscriptConfig.cmake
-		"${CMAKE_CURRENT_BINARY_DIR}/Angelscript/AngelscriptConfigVersion.cmake"
-	DESTINATION ${ConfigPackageLocation}
-	COMPONENT Devel
-)
-
-- 
2.35.5

