From 13f3666152e1291515c67d5f6c49be62717f6f81 Mon Sep 17 00:00:00 2001
From: Silke Hofstra <silke@slxh.eu>
Date: Fri, 20 Sep 2024 20:39:19 +0200
Subject: [PATCH] Disable test that hangs on Solus build server

---
 tests/data/DISABLED | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/tests/data/DISABLED b/tests/data/DISABLED
index 80f835d4b..5785041b6 100644
--- a/tests/data/DISABLED
+++ b/tests/data/DISABLED
@@ -110,3 +110,5 @@
 2047
 %endif
 %endif
+# Hangs on Solus build server
+477
-- 
2.46.1

