From 8e9e9d341f7cf5a17dfaa0a7ec8881c39cd08bb6 Mon Sep 17 00:00:00 2001
From: Peter O'Connor <peter@solus-project.com>
Date: Sat, 24 Mar 2018 18:52:35 +1100
Subject: [PATCH] Don't compile icon cache...

Signed-off-by: Peter O'Connor <peter@solus-project.com>
---
 CMakeLists.txt | 3 ---
 1 file changed, 3 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 31264f6..8bf2e74 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -134,9 +134,6 @@ install (DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR}/data/styles DESTINATION share/qal
 # install (CODE
 #     "execute_process (COMMAND rm /usr/share/qalculate/datasetsNasc.xml)"
 # )
-install (CODE
-    "execute_process (COMMAND gtk-update-icon-cache /usr/share/icons/hicolor)"
-)
 
 # uninstall target
  configure_file(
-- 
2.16.3

