From 01bbd2c0dcfedc062efcbed1ec82eabb64af7c5f Mon Sep 17 00:00:00 2001
From: Joshua Strobl <joshua@stroblindustries.com>
Date: Fri, 3 May 2019 18:27:50 +0300
Subject: [PATCH 1/1] Fix missing MimeType handler for Calendar.

---
 data/org.gnome.Calendar.desktop.in | 1 +
 1 file changed, 1 insertion(+)

diff --git a/data/org.gnome.Calendar.desktop.in b/data/org.gnome.Calendar.desktop.in
index 376d31c..5399e7c 100644
--- a/data/org.gnome.Calendar.desktop.in
+++ b/data/org.gnome.Calendar.desktop.in
@@ -11,3 +11,4 @@ Categories=GNOME;GTK;Office;Calendar;Core;
 # Translators: Search terms to find this application. Do NOT translate or localize the semicolons! The list MUST also end with a semicolon!
 Keywords=Calendar;Event;Reminder;
 DBusActivatable=true
+MimeType=text/calendar;x-scheme-handler/webcal;x-scheme-handler/calendar;
-- 
2.21.0

