From 07338446907992409b3163bb787ce834ffd149ff Mon Sep 17 00:00:00 2001
From: Michael Catanzaro <mcatanzaro@gnome.org>
Date: Tue, 12 Jan 2021 15:08:50 -0600
Subject: [PATCH] Remove autocleanup for GWeatherLocation with new gweather

---
 src/utils/gcal-utils.h | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/src/utils/gcal-utils.h b/src/utils/gcal-utils.h
index c2edc72d..78e9f9cb 100644
--- a/src/utils/gcal-utils.h
+++ b/src/utils/gcal-utils.h
@@ -38,7 +38,11 @@ G_DEFINE_AUTOPTR_CLEANUP_FUNC (ESource, g_object_unref)
 #endif
 
 G_DEFINE_AUTOPTR_CLEANUP_FUNC (ECalComponent, g_object_unref)
+
+#if !GWEATHER_CHECK_VERSION(3, 39, 0)
 G_DEFINE_AUTOPTR_CLEANUP_FUNC (GWeatherLocation, gweather_location_unref)
+#endif
+
 G_DEFINE_AUTOPTR_CLEANUP_FUNC (ICalTime, g_object_unref)
 
 gchar*               gcal_get_weekday                            (gint                i);
-- 
GitLab

