From 174038296371d0b817b9fe262e9ba4c410c17e17 Mon Sep 17 00:00:00 2001
From: Michael Catanzaro <mcatanzaro@gnome.org>
Date: Tue, 12 Jan 2021 16:39:47 -0600
Subject: [PATCH] Fix typo because I am hopeless

---
 src/weather/gcal-weather-service.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/weather/gcal-weather-service.c b/src/weather/gcal-weather-service.c
index 29e3276c..4cf47f02 100644
--- a/src/weather/gcal-weather-service.c
+++ b/src/weather/gcal-weather-service.c
@@ -642,7 +642,7 @@ update_location (GcalWeatherService  *self,
        * https://gitlab.gnome.org/GNOME/gnome-calendar/-/issues/682
        */
       gweather_info_set_enabled_providers (self->gweather_info, GWEATHER_PROVIDER_METAR | GWEATHER_PROVIDER_OWM);
-      g_signal_connect_object (self->gweather_info, "updated", (GCallback) on_gqweather_update_cb, self, 0);
+      g_signal_connect_object (self->gweather_info, "updated", (GCallback) on_gweather_update_cb, self, 0);
 
       /*
        * gweather_info_update might or might not trigger a
-- 
GitLab

