From 86ab2b23d5668cba1d262e73ab031645d00bacc9 Mon Sep 17 00:00:00 2001
From: Bobby Rong <rjl931189261@126.com>
Date: Wed, 14 Jan 2026 22:53:11 +0800
Subject: [PATCH] build: Support Budgie 10.10

Similar to 4c15deb.
---
 meson.build | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/meson.build b/meson.build
index f8ad29e..5e24995 100644
--- a/meson.build
+++ b/meson.build
@@ -30,7 +30,7 @@ endif
 # Dependencies
 
 dep_accountsservice = dependency('accountsservice')
-dep_budgie = dependency('budgie-2.0', version: '>= 3')
+dep_budgie = dependency('budgie-3.0', version: '>= 4')
 dep_gee = dependency('gee-0.8', version: '>= 0.20.0')
 dep_gtk3 = dependency('gtk+-3.0', version: '>= 3.24.0')
 dep_glib = dependency('glib-2.0', version: '>= 2.64.0')
