From 4c15debe9ba63a4e9515b8f98b4227daf0cb02d8 Mon Sep 17 00:00:00 2001
From: Evan Maddock <maddock.evan@vivaldi.net>
Date: Sun, 11 May 2025 10:46:28 -0400
Subject: [PATCH] build: Update to libpeas-2 and new Budgie

Signed-off-by: Evan Maddock <maddock.evan@vivaldi.net>
---
 meson.build | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/meson.build b/meson.build
index 74af2f9..b35cd8f 100644
--- a/meson.build
+++ b/meson.build
@@ -30,12 +30,12 @@ endif
 # Dependencies
 
 dep_accountsservice = dependency('accountsservice')
-dep_budgie = dependency('budgie-1.0', version: '>= 2')
+dep_budgie = dependency('budgie-2.0', version: '>= 3')
 dep_gee = dependency('gee-0.8', version: '>= 0.20.0')
 dep_gtk3 = dependency('gtk+-3.0', version: '>= 3.24.0')
 dep_glib = dependency('glib-2.0', version: '>= 2.64.0')
 dep_giounix = dependency('gio-unix-2.0', version: '>= 2.64.0')
-dep_peas = dependency('libpeas-1.0', version: '>= 1.26.0')
+dep_peas = dependency('libpeas-2', version: '>= 2.0.0')
 dep_vala = dependency('vapigen', version: '>= 0.52.5')
 
 # Directories
