From 11e4d49d7e6ca30d1ffd81d4521ed9679d1d3b5e Mon Sep 17 00:00:00 2001
From: Volker Krause <vkrause@kde.org>
Date: Thu, 16 Apr 2026 18:37:30 +0200
Subject: [PATCH] Add forward compatibility for KMime from KDE Frameworks

---
 CMakeLists.txt     | 6 +++++-
 src/CMakeLists.txt | 2 +-
 2 files changed, 6 insertions(+), 2 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index ddf29c7..dee0334 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -145,7 +145,11 @@ find_package(KF6KIO ${KF_MIN_VERSION} REQUIRED)
 find_package(KPim6Akonadi ${AKONADI_VERSION} CONFIG REQUIRED)
 find_package(KF6Contacts ${KF_MIN_VERSION} CONFIG REQUIRED)
 find_package(KF6CalendarCore ${KF_MIN_VERSION} CONFIG REQUIRED)
-find_package(KPim6Mime ${KMIME_LIB_VERSION} CONFIG REQUIRED)
+find_package(KF6Mime ${KF_MIN_VERSION} CONFIG)
+if(NOT TARGET KF6::Mime)
+    find_package(KPim6Mime ${KMIME_LIB_VERSION} CONFIG REQUIRED)
+    add_library(KF6::Mime ALIAS KPim6::Mime)
+endif()
 find_package(KPim6AkonadiMime ${AKONADIMIME_LIB_VERSION} CONFIG REQUIRED)
 find_package(KPim6Libkdepim ${LIBKDEPIM_LIB_VERSION} CONFIG REQUIRED) # Because of KPIM::KCheckComboBox
 
diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
index be4a850..78fa3ce 100644
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -157,7 +157,7 @@ target_link_libraries(
     KPim6::AkonadiPrivate
     KPim6::AkonadiWidgets
     KPim6::AkonadiXml
-    KPim6::Mime
+    KF6::Mime
     KF6::Contacts
     KF6::CalendarCore
     Qt::Sql
-- 
GitLab

