--- ./justfile	2024-03-15 13:59:51.256329082 -0500
+++ ./justfile.new	2024-03-15 15:37:57.363809570 -0500
@@ -29,7 +29,7 @@
 build-vendored *args: vendor-extract (build-release '--frozen --offline' args)
 
 _link_applet name:
-    ln -sf {{cosmic-applets-bin}} {{bindir}}/{{name}}
+    ln -sf cosmic-applets {{bindir}}/{{name}}
 
 _install_icons name:
     find {{name}}/'data'/'icons' -type f -exec echo {} \; | rev | cut -d'/' -f-3 | rev | xargs -d '\n' -I {} install -Dm0644 {{name}}/'data'/'icons'/{} {{iconsdir}}/{}
