From 219acd41b85dbe3193934dcd231c0471c92c98c2 Mon Sep 17 00:00:00 2001
From: Almarhoon Ibraheem <ibeem@outlook.com>
Date: Sun, 22 Sep 2019 18:33:24 +0300
Subject: [PATCH] Fix compatibility with GNOME 3.34

---
 impatience/extension.js | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/impatience/extension.js b/impatience/extension.js
index 33b6e62..01e3598 100644
--- a/impatience/extension.js
+++ b/impatience/extension.js
@@ -17,7 +17,7 @@ var noop = function() {};
 Ext.prototype = {};
 Ext.prototype._init = function() {
 	this.enabled = false;
-	this.original_speed = St.get_slow_down_factor();
+	this.original_speed = St.Settings.get().slow_down_factor;
 	this.modified_speed = DEFAULT_SPEED;
 	this.unbind = noop;
 };
@@ -39,7 +39,7 @@ Ext.prototype.enable = function() {
 Ext.prototype.disable = function() {
 	this.enabled = false;
 	this.unbind();
-	St.set_slow_down_factor(this.original_speed);
+	St.Settings.get().slow_down_factor = this.original_speed;
 };
 
 Ext.prototype.set_speed = function(new_speed) {
@@ -51,7 +51,7 @@ Ext.prototype.set_speed = function(new_speed) {
 		this.modified_speed = new_speed;
 	}
 	LOG("setting new speed: " + this.modified_speed);
-	St.set_slow_down_factor(this.modified_speed);
+	St.Settings.get().slow_down_factor = this.modified_speed;
 };
 
 function init() {
