--- a/packages/build/src/packaging/run-package.ts
+++ b/packages/build/src/packaging/run-package.ts
@@ -3,7 +3,6 @@ import path from 'path';
 import os from 'os';
 import { Config, Platform, validateBuildVariant } from '../config';
 import { downloadMongocrypt } from './download-mongocryptd';
-import { macOSSignAndNotarize } from './macos-sign';
 import { notarizeMsi } from './msi-sign';
 import { createPackage, PackageFile } from './package';
 
@@ -36,17 +35,6 @@ export async function runPackage(
     );
   };
 
-  if (os.platform() === Platform.MacOs) {
-    return await macOSSignAndNotarize(
-      [
-        config.executablePath,
-        config.mongocryptdPath
-      ],
-      config,
-      runCreatePackage
-    );
-  }
-
   const packaged = await runCreatePackage();
 
   if (distributionBuildVariant === 'win32msi-x64') {
