From ccf1d3027ed7c2d618e058400e106cefe8c9d606 Mon Sep 17 00:00:00 2001
From: Joshua Strobl <joshua@stroblindustries.com>
Date: Sun, 17 Sep 2017 00:04:17 +0300
Subject: [PATCH 1/1] Ensure we use correct lua 5.1 header location

---
 source/clua.h     | 6 +++---
 source/cluautil.h | 6 +++---
 2 files changed, 6 insertions(+), 6 deletions(-)

diff --git a/source/clua.h b/source/clua.h
index 0718e84..b79c3d8 100644
--- a/source/clua.h
+++ b/source/clua.h
@@ -1,9 +1,9 @@
 #pragma once
 
 extern "C" {
-#include <lua.h>
-#include <lauxlib.h>
-#include <lualib.h>
+#include <lua5.1/lua.h>
+#include <lua5.1/lauxlib.h>
+#include <lua5.1/lualib.h>
 }
 
 #include <cstdarg>
diff --git a/source/cluautil.h b/source/cluautil.h
index cef68d5..0280261 100644
--- a/source/cluautil.h
+++ b/source/cluautil.h
@@ -6,9 +6,9 @@
 #pragma once
 
 extern "C" {
-#include <lua.h>
-#include <lauxlib.h>
-#include <lualib.h>
+#include <lua5.1/lua.h>
+#include <lua5.1/lauxlib.h>
+#include <lua5.1/lualib.h>
 }
 
 /*
-- 
2.14.1

