--- a/lashd/client.c	2026-08-28 18:09:54.048155849 +0200
+++ b/lashd/client.c	2026-08-28 18:10:04.234437192 +0200
@@ -26,7 +26,7 @@
 #include "alsa_patch.h"
 
 client_t *
-client_new(lash_connect_params_t * params)
+client_new(void)
 {
 	client_t *client;
 
--- a/lashd/client.h	2026-08-28 18:09:58.334923686 +0200
+++ b/lashd/client.h	2026-08-28 18:10:08.433760165 +0200
@@ -63,7 +63,7 @@
 #define CLIENT_SAVED(x)             (((x)->flags) & LASH_Saved)
 
 
-client_t * client_new ();
+client_t * client_new (void);
 void       client_destroy (client_t * client);
 
 const char * client_get_id_str   (client_t * client);
--- a/lashd/server_event.h	2026-08-28 18:10:01.240421653 +0200
+++ b/lashd/server_event.h	2026-08-28 18:10:15.004570384 +0200
@@ -51,7 +51,7 @@
 };
 
 server_event_t * server_event_new ();
-void             server_event_destroy ();
+void             server_event_destroy (server_event_t *event);
 
 void server_event_set_type           (server_event_t * event, enum Server_Event_Type type);
 void server_event_set_conn_id        (server_event_t * event, unsigned long id);
