diff -Naur lash-0.5.4-org/clients/synth/lash.c lash-0.5.4/clients/synth/lash.c
--- lash-0.5.4-org/clients/synth/lash.c	2005-09-15 10:23:29.000000000 +0200
+++ lash-0.5.4/clients/synth/lash.c	2009-11-04 17:35:56.000000000 +0100
@@ -23,6 +23,9 @@
 #include "config.h"
 
 #include <unistd.h>
+#include <string.h>
+#include <stdio.h>
+#include <sys/socket.h>
 
 #include <lash/lash.h>
 
diff -Naur lash-0.5.4-org/liblash/socket.c lash-0.5.4/liblash/socket.c
--- lash-0.5.4-org/liblash/socket.c	2006-09-16 21:25:31.000000000 +0200
+++ lash-0.5.4/liblash/socket.c	2009-11-04 17:25:02.000000000 +0100
@@ -24,6 +24,9 @@
 #include <sys/types.h>
 #include <sys/socket.h>
 #include <netdb.h>
+#define NI_MAXHOST      1025
+#define NI_MAXSERV      32
+
 #include <string.h>
 #include <stdlib.h>
 #include <errno.h>

