--- configure.ac.orig
+++ configure.ac
@@ -6,7 +6,9 @@ AM_ACLOCAL_INCLUDE([m4])
 
 ### Check for programs ###
 AC_LANG([C])
-AC_PROG_CC
+AC_PROG_CC_STDC
+AC_USE_SYSTEM_EXTENSIONS
+AC_SYS_LARGEFILE
 AC_PROG_LIBTOOL
 
 #################
@@ -31,6 +33,7 @@ AC_SUBST(JACK_LIBS)
 AC_DEFINE_UNQUOTED(JACK_VERSION, "$JACK_VERSION", [The version of jack we're compiling against])
 PC_REQUIRES="jack"
 
+AC_SEARCH_LIBS(sin, m)
 
 ############
 ### ALSA ###

