Index: lash-0.5.4/liblash/socket.c
===================================================================
--- lash-0.5.4.orig/liblash/socket.c
+++ lash-0.5.4/liblash/socket.c
@@ -18,7 +18,7 @@
  *   Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
  */
 
-#define _POSIX_SOURCE /* addrinfo */
+#define _POSIX_C_SOURCE 200112L /* addrinfo */
 
 #include <stdint.h>
 #include <sys/types.h>

