From af11b5678a0ea86cf190e3e1f7ada41226cba641 Mon Sep 17 00:00:00 2001
From: Ivan Trepakov <liontiger23@gmail.com>
Date: Sat, 16 May 2026 20:04:49 +0700
Subject: [PATCH] Gradle 9.x.x fixes

---
 build.gradle    | 22 +++++++++++-----------
 settings.gradle |  6 +++---
 2 files changed, 14 insertions(+), 14 deletions(-)

diff --git a/build.gradle b/build.gradle
index 48d7265..cc544cf 100644
--- a/build.gradle
+++ b/build.gradle
@@ -11,7 +11,7 @@ plugins {
     id 'me.champeau.jmh' version '0.7.2'
 
     // This is https://github.com/java9-modularity/gradle-modules-plugin/pull/282
-    id 'com.github.koppor.gradle-modules-plugin' version 'jitpack-SNAPSHOT'
+    id 'com.github.koppor.gradle-modules-plugin' version 'v1.8.15-cmd-1'
 
     id 'org.openjfx.javafxplugin' version '0.1.0'
 
@@ -221,7 +221,7 @@ dependencies {
     implementation 'org.kordamp.ikonli:ikonli-materialdesign2-pack:12.3.1'
     implementation 'com.github.sialcasa.mvvmFX:mvvmfx-validation:f195849ca9' //jitpack
     implementation 'de.saxsys:mvvmfx:1.8.0'
-    implementation('com.tobiasdiez:easybind:2.2.1-SNAPSHOT') {
+    implementation('org.jabref:easybind:2.3.0') {
         exclude group: 'org.openjfx'
     }
     implementation 'org.fxmisc.flowless:flowless:0.7.3'
@@ -384,7 +384,7 @@ tasks.register('generateSource') {
             "generateSearchGrammarSource",
             "generateCitaviSource")
     group = 'JabRef'
-    description 'Generates all necessary (Java) source files.'
+    description = 'Generates all necessary (Java) source files.'
 }
 
 tasks.register("generateBstGrammarSource", JavaExec) {
@@ -533,17 +533,17 @@ test {
 testlogger {
     // See https://github.com/radarsh/gradle-test-logger-plugin#configuration for configuration options
 
-    theme 'standard'
+    theme = 'standard'
 
-    showPassed false
-    showSkipped false
+    showPassed = false
+    showSkipped = false
 
-    showCauses false
-    showStackTraces false
+    showCauses = false
+    showStackTraces = false
 }
 
 tasks.withType(Test).configureEach {
-    reports.html.outputLocation.set(file("${reporting.baseDir}/${name}"))
+    reports.html.outputLocation.set(file("${reporting.baseDirectory}/${name}"))
     // Enable parallel tests (on desktop).
     // See https://docs.gradle.org/8.1/userguide/performance.html#execute_tests_in_parallel for details.
     if (!providers.environmentVariable("CI").isPresent()) {
@@ -559,7 +559,7 @@ tasks.register('databaseTest', Test) {
     testLogging {
         // set options for log level LIFECYCLE
         events = ["FAILED"]
-        exceptionFormat "full"
+        exceptionFormat = "full"
     }
 
     maxParallelForks = 1
@@ -581,7 +581,7 @@ tasks.register('guiTest', Test) {
     testLogging {
         // set options for log level LIFECYCLE
         events = ["FAILED"]
-        exceptionFormat "full"
+        exceptionFormat = "full"
     }
 
     maxParallelForks = 1
diff --git a/settings.gradle b/settings.gradle
index 7908374..d820ff0 100644
--- a/settings.gradle
+++ b/settings.gradle
@@ -5,21 +5,21 @@ pluginManagement {
             // Updated solution at https://github.com/foodiestudio/convention-plugins?tab=readme-ov-file#convention-plugins
             if (requested.id.id.startsWith("com.github.koppor")) {
                 // This is https://github.com/java9-modularity/gradle-modules-plugin/pull/282
-                useModule("com.github.koppor:gradle-modules-plugin:jitpack-SNAPSHOT")
+                useModule("com.github.koppor:gradle-modules-plugin:1.8.15-cmd-1")
             }
         }
     }
 
     repositories {
         maven {
-            url 'https://jitpack.io'
+            url = 'https://jitpack.io'
         }
         gradlePluginPortal()
     }
 }
 
 plugins {
-  id("org.gradle.toolchains.foojay-resolver-convention") version "0.8.0"
+    id("org.gradle.toolchains.foojay-resolver-convention") version "0.10.0"
 }
 
 rootProject.name = "JabRef"
-- 
2.54.0

