From 568e96faf0429a992d100333571967d137784551 Mon Sep 17 00:00:00 2001
From: Ikey Doherty <ikey@solusos.com>
Date: Tue, 13 Aug 2013 23:25:01 +0100
Subject: [PATCH] Fix building with readline

---
 wpa_supplicant/Makefile |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/wpa_supplicant/Makefile b/wpa_supplicant/Makefile
index 65fef41..924a481 100644
--- a/wpa_supplicant/Makefile
+++ b/wpa_supplicant/Makefile
@@ -1265,7 +1265,7 @@ LIBS += $(DBUS_LIBS)
 
 ifdef CONFIG_READLINE
 OBJS_c += ../src/utils/edit_readline.o
-LIBS_c += -lncurses -lreadline
+LIBS_c += -lreadline -lncurses -ldl
 else
 ifdef CONFIG_WPA_CLI_EDIT
 OBJS_c += ../src/utils/edit.o
-- 
1.7.10.4

