From 052df1530fe8ad77aceae95aaa3ba150425b124f Mon Sep 17 00:00:00 2001
From: Ikey Doherty <ikey@solusos.com>
Date: Tue, 13 Aug 2013 23:33:36 +0100
Subject: [PATCH 2/2] Change prefix to SolusOS sane values

---
 wpa_supplicant/Makefile |    4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/wpa_supplicant/Makefile b/wpa_supplicant/Makefile
index 924a481..2f6526c 100644
--- a/wpa_supplicant/Makefile
+++ b/wpa_supplicant/Makefile
@@ -6,8 +6,8 @@ ifndef CFLAGS
 CFLAGS = -MMD -O2 -Wall -g
 endif
 
-export LIBDIR ?= /usr/local/lib/
-export BINDIR ?= /usr/local/sbin/
+export LIBDIR ?= /lib/
+export BINDIR ?= /sbin/
 PKG_CONFIG ?= pkg-config
 
 CFLAGS += -I../src
-- 
1.7.10.4

