From 2ec9cf8b10fe52c8b0e48ccefa5a466025283881 Mon Sep 17 00:00:00 2001
From: Pierre-Yves <pyu@riseup.net>
Date: Tue, 25 Dec 2018 12:35:04 +0100
Subject: [PATCH] remove requires dbus-python

Signed-off-by: Pierre-Yves <pyu@riseup.net>
---
 setup.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/setup.py b/setup.py
index e7c515d..dfcd709 100644
--- a/setup.py
+++ b/setup.py
@@ -166,7 +166,7 @@ setup(
         ]
     },
     scripts=['autokey-run', 'autokey-shell'],
-    install_requires=['dbus-python', 'pyinotify', 'python-xlib'],
+    install_requires=['pyinotify', 'python-xlib'],
     classifiers=[
         'Development Status :: 4 - Beta',
         'Intended Audience :: Developers',
-- 
2.20.1

