From 96c5f5683e095c4fde451a52f401cf2a09036ab0 Mon Sep 17 00:00:00 2001
From: Ikey Doherty <ikey@solus-project.com>
Date: Tue, 26 Dec 2017 15:10:25 +0000
Subject: [PATCH] Remove broken requirement on python3-xlib

Signed-off-by: Ikey Doherty <ikey@solus-project.com>
---
 setup.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/setup.py b/setup.py
index a3bfb8b..8fc725a 100644
--- a/setup.py
+++ b/setup.py
@@ -75,7 +75,7 @@ setup(
         'console_scripts': ['autokey-gtk=autokey.gtkui.__main__:main']
     },
     scripts=['autokey-qt', 'autokey-run', 'autokey-shell'],
-    install_requires=['pyinotify', 'python3-xlib'],
+    install_requires=['pyinotify'],
     classifiers=[
         'Development Status :: 4 - Beta',
         'Intended Audience :: Developers',
-- 
2.15.1

