From 9789c789c2a9a6fce0d38faefbee152fe1c6cc2f Mon Sep 17 00:00:00 2001
From: Peter O'Connor <peter@solus-project.com>
Date: Wed, 4 Apr 2018 10:57:41 +1000
Subject: [PATCH] Disable building unshipped code

Signed-off-by: Peter O'Connor <peter@solus-project.com>
---
 CMakeLists.txt | 2 --
 1 file changed, 2 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index f259e25..a9d317e 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -53,8 +53,6 @@ if (IS_DIRECTORY "${CMAKE_CURRENT_SOURCE_DIR}/po")
     ki18n_install(po)
 endif()
 add_subdirectory(src)
-add_subdirectory(autotests)
-add_subdirectory(tests)
 
 find_package(FontNotoSans)
 set_package_properties(FontNotoSans PROPERTIES
-- 
2.16.3

