diff --git a/CMakeLists.txt b/CMakeLists.txt
index 9d52082..5f9736c 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -175,9 +175,9 @@ endif()
 
 enable_testing()
 
-# Temporarily for CI to work
-set(ROCPROFILER_BUILD_TESTS ON)
-set(ROCPROFILER_BUILD_CI ON)
+# # Temporarily for CI to work
+# set(ROCPROFILER_BUILD_TESTS ON)
+# set(ROCPROFILER_BUILD_CI ON)
 
 if(ROCPROFILER_BUILD_CI)
     include(CTest)
@@ -187,7 +187,7 @@ endif()
 add_subdirectory(src)
 
 # Build Plugins
-add_subdirectory(plugin)
+# add_subdirectory(plugin)
 
 if(ROCPROFILER_BUILD_SAMPLES)
     # Build samples
diff --git a/src/api/CMakeLists.txt b/src/api/CMakeLists.txt
index 8cd41c5..e5947e8 100644
--- a/src/api/CMakeLists.txt
+++ b/src/api/CMakeLists.txt
@@ -254,14 +254,13 @@ get_target_property(ROCPROFILER_LIBRARY_V1_SOVERSION ${ROCPROFILER_TARGET} SOVER
 # installed using symlinks
 install(
     TARGETS ${ROCPROFILER_TARGET}
-    LIBRARY NAMELINK_SKIP
-            DESTINATION ${CMAKE_INSTALL_LIBDIR}
-            COMPONENT runtime)
-install(
-    TARGETS ${ROCPROFILER_TARGET}
-    LIBRARY NAMELINK_SKIP
-            DESTINATION ${CMAKE_INSTALL_LIBDIR}
-            COMPONENT asan)
+    DESTINATION ${CMAKE_INSTALL_LIBDIR}
+    COMPONENT runtime)
+# install(
+#     TARGETS ${ROCPROFILER_TARGET}
+#     LIBRARY NAMELINK_SKIP
+#             DESTINATION ${CMAKE_INSTALL_LIBDIR}
+#             COMPONENT asan)
 
 # V2 Library Compiling/Installing ROCProfiler API
 add_library(
@@ -295,7 +294,6 @@ set_target_properties(
     PROPERTIES CXX_VISIBILITY_PRESET hidden
                DEFINE_SYMBOL "ROCPROFILER_EXPORTS"
                LINK_DEPENDS ${CMAKE_CURRENT_SOURCE_DIR}/exportmap
-               OUTPUT_NAME rocprofiler64
                LIBRARY_OUTPUT_DIRECTORY ${CMAKE_BINARY_DIR}/lib
                VERSION ${PROJECT_VERSION}
                SOVERSION ${PROJECT_VERSION_MAJOR})
@@ -339,55 +337,54 @@ get_target_property(ROCPROFILER_LIBRARY_V2_NAME rocprofiler-v2 OUTPUT_NAME)
 get_target_property(ROCPROFILER_LIBRARY_V2_VERSION rocprofiler-v2 VERSION)
 get_target_property(ROCPROFILER_LIBRARY_V2_SOVERSION rocprofiler-v2 SOVERSION)
 
-# Prepare Name Link SO files for V1 & V2 Libraries
-add_custom_command(
-    TARGET rocprofiler-v2
-    POST_BUILD
-    COMMAND ${CMAKE_COMMAND} -E rm -f
-            ${CMAKE_BINARY_DIR}/lib/lib${ROCPROFILER_LIBRARY_V1_NAME}.so
-    COMMAND
-        ${CMAKE_COMMAND} -E create_symlink
-        lib${ROCPROFILER_LIBRARY_V1_NAME}.so.${ROCPROFILER_LIBRARY_V1_SOVERSION}
-        ${CMAKE_BINARY_DIR}/lib/lib${ROCPROFILER_LIBRARY_V1_NAME}.so
-    COMMAND
-        ${CMAKE_COMMAND} -E create_symlink
-        lib${ROCPROFILER_LIBRARY_V2_NAME}.so.${ROCPROFILER_LIBRARY_V2_SOVERSION}
-        ${CMAKE_BINARY_DIR}/lib/lib${ROCPROFILER_LIBRARY_V2_NAME}v2.so
-    # Temporarily up till Jenkins side is fixed
-    COMMAND
-        ${CMAKE_COMMAND} -E create_symlink
-        lib/lib${ROCPROFILER_LIBRARY_V1_NAME}.so
-        ${CMAKE_BINARY_DIR}/lib${ROCPROFILER_LIBRARY_V1_NAME}.so)
-# Add custom target to trigger the create_symlink command
-add_custom_target(create_rocprofiler_lib DEPENDS rocprofiler-v2 ${ROCPROFILER_TARGET})
+# # Prepare Name Link SO files for V1 & V2 Libraries
+# add_custom_command(
+#     TARGET rocprofiler-v2
+#     POST_BUILD
+#     COMMAND ${CMAKE_COMMAND} -E rm -f
+#             ${CMAKE_BINARY_DIR}/lib/lib${ROCPROFILER_LIBRARY_V1_NAME}.so
+#     COMMAND
+#         ${CMAKE_COMMAND} -E create_symlink
+#         lib${ROCPROFILER_LIBRARY_V1_NAME}.so.${ROCPROFILER_LIBRARY_V1_SOVERSION}
+#         ${CMAKE_BINARY_DIR}/lib/lib${ROCPROFILER_LIBRARY_V1_NAME}.so
+#     COMMAND
+#         ${CMAKE_COMMAND} -E create_symlink
+#         lib${ROCPROFILER_LIBRARY_V2_NAME}.so.${ROCPROFILER_LIBRARY_V2_SOVERSION}
+#         ${CMAKE_BINARY_DIR}/lib/lib${ROCPROFILER_LIBRARY_V2_NAME}v2.so
+#     # Temporarily up till Jenkins side is fixed
+#     COMMAND
+#         ${CMAKE_COMMAND} -E create_symlink
+#         lib/lib${ROCPROFILER_LIBRARY_V1_NAME}.so
+#         ${CMAKE_BINARY_DIR}/lib${ROCPROFILER_LIBRARY_V1_NAME}.so)
+# # Add custom target to trigger the create_symlink command
+# add_custom_target(create_rocprofiler_lib DEPENDS rocprofiler-v2 ${ROCPROFILER_TARGET})
 
 # Install libraries: Non versioned lib file in dev package Skipping NameLink as it will be
 # installed using symlinks
 install(
     TARGETS rocprofiler-v2
-    LIBRARY NAMELINK_SKIP
-            DESTINATION ${CMAKE_INSTALL_LIBDIR}
-            COMPONENT runtime)
-install(
-    TARGETS rocprofiler-v2
-    LIBRARY NAMELINK_SKIP
-            DESTINATION ${CMAKE_INSTALL_LIBDIR}
-            COMPONENT asan)
-
-# Installing NameLinks for V1 & V2 librocprofiler64.so links to V1 library
-# librocprofiler64v2.so links to V2 library
-install(
-    PROGRAMS ${CMAKE_BINARY_DIR}/lib/lib${ROCPROFILER_LIBRARY_V2_NAME}v2.so
-    TYPE LIB COMPONENT dev)
-install(
-    PROGRAMS ${CMAKE_BINARY_DIR}/lib/lib${ROCPROFILER_LIBRARY_V1_NAME}.so
-    TYPE LIB COMPONENT dev)
-install(
-    PROGRAMS ${CMAKE_BINARY_DIR}/lib/lib${ROCPROFILER_LIBRARY_V2_NAME}v2.so
-    TYPE LIB COMPONENT asan)
-install(
-    PROGRAMS ${CMAKE_BINARY_DIR}/lib/lib${ROCPROFILER_LIBRARY_V1_NAME}.so
-    TYPE LIB COMPONENT asan)
+    DESTINATION ${CMAKE_INSTALL_LIBDIR}
+    COMPONENT runtime)
+# install(
+#     TARGETS rocprofiler-v2
+#     LIBRARY NAMELINK_SKIP
+#             DESTINATION ${CMAKE_INSTALL_LIBDIR}
+#             COMPONENT asan)
+
+# # Installing NameLinks for V1 & V2 librocprofiler64.so links to V1 library
+# # librocprofiler64v2.so links to V2 library
+# install(
+#     PROGRAMS ${CMAKE_BINARY_DIR}/lib/lib${ROCPROFILER_LIBRARY_V2_NAME}v2.so
+#     TYPE LIB COMPONENT dev)
+# install(
+#     PROGRAMS ${CMAKE_BINARY_DIR}/lib/lib${ROCPROFILER_LIBRARY_V1_NAME}.so
+#     TYPE LIB COMPONENT dev)
+# install(
+#     PROGRAMS ${CMAKE_BINARY_DIR}/lib/lib${ROCPROFILER_LIBRARY_V2_NAME}v2.so
+#     TYPE LIB COMPONENT asan)
+# install(
+#     PROGRAMS ${CMAKE_BINARY_DIR}/lib/lib${ROCPROFILER_LIBRARY_V1_NAME}.so
+#     TYPE LIB COMPONENT asan)
 
 configure_file(
     ${PROJECT_SOURCE_DIR}/src/core/counters/metrics/basic_counters.xml
