From cda354e44c1f491f93e19cc246308560a672e23c Mon Sep 17 00:00:00 2001
From: Thomas Staudinger <Staudi.Kaos@gmail.com>
Date: Fri, 18 Mar 2022 19:40:38 +0100
Subject: [PATCH] Only start vdagent under virtualisation

---
 data/spice-vdagentd.service | 1 +
 1 file changed, 1 insertion(+)

diff --git a/data/spice-vdagentd.service b/data/spice-vdagentd.service
index 9c70b30..36a2fc8 100644
--- a/data/spice-vdagentd.service
+++ b/data/spice-vdagentd.service
@@ -1,6 +1,7 @@
 [Unit]
 Description=Agent daemon for Spice guests
 After=dbus.target
+ConditionVirtualization=yes
 Requires=spice-vdagentd.socket
 
 [Service]
-- 
2.35.1

