From 685d32125a26124d1e176e9d9ef075ea82ce5451 Mon Sep 17 00:00:00 2001
From: Ikey Doherty <ikey@solus-project.com>
Date: Sun, 28 Jan 2018 12:26:01 +0000
Subject: [PATCH] Only start vdagent under virtualisation

Signed-off-by: Ikey Doherty <ikey@solus-project.com>
---
 data/spice-vdagentd.service | 1 +
 1 file changed, 1 insertion(+)

diff --git a/data/spice-vdagentd.service b/data/spice-vdagentd.service
index 8c5effe..7bb6005 100644
--- a/data/spice-vdagentd.service
+++ b/data/spice-vdagentd.service
@@ -1,6 +1,7 @@
 [Unit]
 Description=Agent daemon for Spice guests
 After=dbus.target
+ConditionVirtualization=yes
 
 # TODO we should use:
 #Requires=spice-vdagentd.socket
-- 
2.16.1

