From 110ce5b0a55a3f7c5917b75b2ab12a4dc5c14fae Mon Sep 17 00:00:00 2001
From: Peter O'Connor <peter@solus-project.com>
Date: Wed, 23 May 2018 13:07:05 +1000
Subject: [PATCH] Disable static libs

Signed-off-by: Peter O'Connor <peter@solus-project.com>
---
 lib/SignOn/SignOn.pro | 3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

diff --git a/lib/SignOn/SignOn.pro b/lib/SignOn/SignOn.pro
index f6a1d61..180056f 100644
--- a/lib/SignOn/SignOn.pro
+++ b/lib/SignOn/SignOn.pro
@@ -1,7 +1,6 @@
 TEMPLATE = subdirs
 CONFIG   += ordered
 SUBDIRS = \
-    libsignon-qt-shared.pro \
-    libsignon-qt-static.pro
+    libsignon-qt-shared.pro
 
 include(doc/doc.pri)
--
2.17.0
