From 2bb941cbe9cf9d94410b257c26cab1f8504ed4a1 Mon Sep 17 00:00:00 2001
From: Aleix Pol <aleixpol@kde.org>
Date: Tue, 17 Mar 2020 02:00:54 +0100
Subject: [PATCH] Port away from QHash::unite, which is deprecated now

---
 lib/SignOn/sessiondata.h | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/lib/SignOn/sessiondata.h b/lib/SignOn/sessiondata.h
index 390e374a..3578a8d4 100644
--- a/lib/SignOn/sessiondata.h
+++ b/lib/SignOn/sessiondata.h
@@ -121,7 +121,9 @@ public:
      * @return reference to this object
      */
     SessionData &operator+=(const SessionData &other) {
-        m_data.unite(other.m_data);
+        for (auto it = other.m_data.constBegin(), itEnd = other.m_data.constEnd(); it != itEnd; ++it) {
+            m_data[it.key()] = it.value();
+        }
         return *this;
     }
 
-- 
GitLab