From fd8daa179a05d1ba4adde64cc6fa804302b61500 Mon Sep 17 00:00:00 2001
From: Shan Kang <shan.kang@intel.com>
Date: Tue, 4 Jun 2019 07:38:21 +0000
Subject: [PATCH] Force ffsll to be 64 bytes aligned

Signed-off-by: Shan Kang <shan.kang@intel.com>
---
 sysdeps/x86_64/ffsll.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/sysdeps/x86_64/ffsll.c b/sysdeps/x86_64/ffsll.c
index 1caf6ac1..72a04a8f 100644
--- a/sysdeps/x86_64/ffsll.c
+++ b/sysdeps/x86_64/ffsll.c
@@ -24,6 +24,7 @@
 #undef	ffsll
 
 int
+__attribute__((aligned(64)))
 ffsll (long long int x)
 {
   long long int cnt;
-- 
2.21.0

--- glibc-2.30/sysdeps/x86_64/ffs.c~	2019-08-01 04:29:31.000000000 +0000
+++ glibc-2.30/sysdeps/x86_64/ffs.c	2020-02-27 02:05:49.505629357 +0000
@@ -23,6 +23,7 @@
 #undef	ffs
 
 int
+__attribute__((aligned(64)))
 __ffs (int x)
 {
   int cnt;
