From 41eef0b9c5d70cb4f38fbc16dfe7f94b43508a97 Mon Sep 17 00:00:00 2001
From: Ikey Doherty <ikey@solus-project.com>
Date: Tue, 20 Sep 2016 05:28:23 +0100
Subject: [PATCH] elf: Currently skip untested hardware to ensure no issues in
 repos

Signed-off-by: Ikey Doherty <ikey@solus-project.com>
---
 elf/dl-hwcaps.c | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/elf/dl-hwcaps.c b/elf/dl-hwcaps.c
index 99d0495..2cace32 100644
--- a/elf/dl-hwcaps.c
+++ b/elf/dl-hwcaps.c
@@ -58,6 +58,11 @@ int _avx2_available(void)
     return 0;
   }
 
+  if (!__builtin_cpu_is("intel")) {
+      _dl_debug_printf (" Skipping AVX2 on untested hardware\n");
+      return 0;
+  }
+
   if ((ecx & bit_OSXSAVE) == (bit_OSXSAVE))
   {
     if (__glibc_unlikely (GLRO(dl_debug_mask) & DL_DEBUG_LIBS))
-- 
2.10.0

