From eee892115db20610c08a3f0ef948b71805c041b5 Mon Sep 17 00:00:00 2001
From: Reilly Brogan <reilly@reillybrogan.com>
Date: Fri, 2 Feb 2024 21:18:45 -0600
Subject: [PATCH 2/3] tune adaptive spin count

---
 sysdeps/generic/adaptive_spin_count.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/sysdeps/generic/adaptive_spin_count.h b/sysdeps/generic/adaptive_spin_count.h
index 06c47b5c48..6df5440fb9 100644
--- a/sysdeps/generic/adaptive_spin_count.h
+++ b/sysdeps/generic/adaptive_spin_count.h
@@ -19,4 +19,4 @@
 /* The choice of 100 spins for the default spin count for an adaptive spin
    is a completely arbitrary choice that has not been evaluated thoroughly
    using modern hardware.  */
-#define DEFAULT_ADAPTIVE_COUNT 100
+#define DEFAULT_ADAPTIVE_COUNT 5000
-- 
2.43.0

