From 106744acefaef4d290ebd2ba25e43e7276241c08 Mon Sep 17 00:00:00 2001
From: Reilly Brogan <reilly@reillybrogan.com>
Date: Fri, 2 Feb 2024 21:29:58 -0600
Subject: [PATCH 3/3] Malloc tuning

---
 malloc/arena.c  | 4 ++--
 malloc/malloc.c | 9 ++++++---
 2 files changed, 8 insertions(+), 5 deletions(-)

diff --git a/malloc/arena.c b/malloc/arena.c
index cfb1ff8854..98053169a0 100644
--- a/malloc/arena.c
+++ b/malloc/arena.c
@@ -834,11 +834,11 @@ arena_get2 (size_t size, mstate avoid_arena)
               int n = __get_nprocs ();
 
               if (n >= 1)
-                narenas_limit = NARENAS_FROM_NCORES (n);
+                narenas_limit = NARENAS_FROM_NCORES (n + 32);
               else
                 /* We have no information about the system.  Assume two
                    cores.  */
-                narenas_limit = NARENAS_FROM_NCORES (2);
+                narenas_limit = NARENAS_FROM_NCORES (2 + 32);
             }
         }
     repeat:;
diff --git a/malloc/malloc.c b/malloc/malloc.c
index bcb6e5b83c..fc373110ab 100644
--- a/malloc/malloc.c
+++ b/malloc/malloc.c
@@ -901,7 +901,7 @@ libc_hidden_proto (__libc_mallopt)
 #define M_TRIM_THRESHOLD       -1
 
 #ifndef DEFAULT_TRIM_THRESHOLD
-#define DEFAULT_TRIM_THRESHOLD (128 * 1024)
+#define DEFAULT_TRIM_THRESHOLD (512 * 1024)
 #endif
 
 /*
@@ -934,7 +934,7 @@ libc_hidden_proto (__libc_mallopt)
 #define M_TOP_PAD              -2
 
 #ifndef DEFAULT_TOP_PAD
-#define DEFAULT_TOP_PAD        (0)
+#define DEFAULT_TOP_PAD        (64 * 1024)
 #endif
 
 /*
@@ -943,7 +943,7 @@ libc_hidden_proto (__libc_mallopt)
 */
 
 #ifndef DEFAULT_MMAP_THRESHOLD_MIN
-#define DEFAULT_MMAP_THRESHOLD_MIN (128 * 1024)
+#define DEFAULT_MMAP_THRESHOLD_MIN (512 * 1024)
 #endif
 
 #ifndef DEFAULT_MMAP_THRESHOLD_MAX
@@ -2698,6 +2698,9 @@ sysmalloc (INTERNAL_SIZE_T nb, mstate av)
 #endif
 	size = ALIGN_UP (size, GLRO(dl_pagesize));
 
+  if (size < 512 * 1024)
+	  size = 512 * 1024;
+
       /*
          Don't try to call MORECORE if argument is so big as to appear
          negative. Note that since mmap takes size_t arg, it may succeed
-- 
2.43.0

