From 3f1f3616f9b938fd0d240ccdd3e213f7b2aee855 Mon Sep 17 00:00:00 2001
From: Pierre-Yves <pyu@riseup.net>
Date: Sat, 25 Sep 2021 21:09:47 +0200
Subject: [PATCH] Do not hijack file associations

Signed-off-by: Pierre-Yves <pyu@riseup.net>
---
 programs/winecfg/theme.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/programs/winecfg/theme.c b/programs/winecfg/theme.c
index 9c8737c..1c39f44 100644
--- a/programs/winecfg/theme.c
+++ b/programs/winecfg/theme.c
@@ -1077,7 +1077,7 @@ static void on_select_font(HWND hDlg)
 
 static void init_mime_types(HWND hDlg)
 {
-    WCHAR *buf = get_reg_key(config_key, keypath(L"FileOpenAssociations"), L"Enable", L"Y");
+    WCHAR *buf = get_reg_key(config_key, keypath(L"FileOpenAssociations"), L"Enable", L"N");
     int state = IS_OPTION_TRUE(*buf) ? BST_CHECKED : BST_UNCHECKED;
 
     CheckDlgButton(hDlg, IDC_ENABLE_FILE_ASSOCIATIONS, state);
-- 
2.32.0

