From 882f1caad52c6291c260c503f4696932624158d6 Mon Sep 17 00:00:00 2001
From: Fabio Forni <livingsilver94.solus@redaril.me>
Date: Fri, 29 Nov 2019 00:15:03 +0100
Subject: [PATCH] Do not hijack file associations

---
 programs/winecfg/theme.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/programs/winecfg/theme.c b/programs/winecfg/theme.c
index 66d2344..c47c9ed 100644
--- a/programs/winecfg/theme.c
+++ b/programs/winecfg/theme.c
@@ -1165,7 +1165,7 @@ static void on_select_font(HWND hDlg)
 
 static void init_mime_types(HWND hDlg)
 {
-    char *buf = get_reg_key(config_key, keypath("FileOpenAssociations"), "Enable", "Y");
+    char *buf = get_reg_key(config_key, keypath("FileOpenAssociations"), "Enable", "N");
     int state = IS_OPTION_TRUE(*buf) ? BST_CHECKED : BST_UNCHECKED;
 
     CheckDlgButton(hDlg, IDC_ENABLE_FILE_ASSOCIATIONS, state);
-- 
2.24.0

