From 65fc7e936735b4a8f4e400b7e4735b7c0850bc11 Mon Sep 17 00:00:00 2001
From: Ikey Doherty <ikey@solus-project.com>
Date: Mon, 7 Dec 2015 04:33:30 +0000
Subject: [PATCH] Fix critical issue with runtime deps

Signed-off-by: Ikey Doherty <ikey@solus-project.com>
---
 sanity.py | 9 +++------
 1 file changed, 3 insertions(+), 6 deletions(-)

diff --git a/sanity.py b/sanity.py
index 1022828..0987ada 100644
--- a/sanity.py
+++ b/sanity.py
@@ -134,9 +134,7 @@ def add_runtime_dep(pkg, dep):
     init_mutations()
 
     maybe_mutate(pkg)
-    pkgname = pkg
-    if pkgname in mutations:
-        pkgname = mutations[pkgname]
+    pkgname = pkg if pkg == name else mutations[pkg]
 
     if not rundeps:
         rundeps = dict()
@@ -159,9 +157,8 @@ def add_replaces(pkg, pkg2):
     init_mutations()
 
     maybe_mutate(pkg)
-    pkgname = pkg
-    if pkgname in mutations:
-        pkgname = mutations[pkgname]
+    maybe_mutate(pkg)
+    pkgname = pkg if pkg == name else mutations[pkg]
 
     if not pkg_replaces:
         pkg_replaces = dict()
-- 
2.6.3

