From 77f493e7181f34ad49a3f2b2e07951737d9228c0 Mon Sep 17 00:00:00 2001
From: Ikey Doherty <ikey@solus-project.com>
Date: Sun, 17 Apr 2016 08:26:19 +0100
Subject: [PATCH] Ensure -32bit-devel depends on -devel too

Signed-off-by: Ikey Doherty <ikey@solus-project.com>
---
 ypkg2/metadata.py | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/ypkg2/metadata.py b/ypkg2/metadata.py
index 3355464..f50eed9 100644
--- a/ypkg2/metadata.py
+++ b/ypkg2/metadata.py
@@ -288,8 +288,8 @@ def handle_dependencies(context, gene, metadata, package, files):
     elif package.name == "32bit-devel":
         if "32bit" in gene.packages:
             dependencies.add(context.spec.get_package_name("32bit"))
-        if "main" in gene.packages:
-            dependencies.add(context.spec.get_package_name("main"))
+        if "devel" in gene.packages:
+            dependencies.add(context.spec.get_package_name("devel"))
     elif package.name == "devel":
         if "main" in gene.packages:
             dependencies.add(context.spec.get_package_name("main"))
-- 
2.8.1

