From 591a9cefb893a459433ee603e1c8c0ba524b55b5 Mon Sep 17 00:00:00 2001
From: Ikey Doherty <ikey@solus-project.com>
Date: Sat, 16 Apr 2016 23:44:51 +0100
Subject: [PATCH] Disable parallel inspection as deadline scheduler is being a
 douche

Signed-off-by: Ikey Doherty <ikey@solus-project.com>
---
 ypkg2/examine.py | 17 +++++++++--------
 1 file changed, 9 insertions(+), 8 deletions(-)

diff --git a/ypkg2/examine.py b/ypkg2/examine.py
index 8946d3a..b10dadd 100644
--- a/ypkg2/examine.py
+++ b/ypkg2/examine.py
@@ -348,7 +348,7 @@ class PackageExaminer:
         # Right now we actually only care about magic matching
         removed = set()
 
-        pool = multiprocessing.Pool()
+        # pool = multiprocessing.Pool()
         results = list()
 
         for file in package.emit_files():
@@ -377,18 +377,19 @@ class PackageExaminer:
             if not self.file_is_of_interest("/" + file, fpath, mgs):
                 continue
             # Handle this asynchronously
-            results.append(pool.apply_async(examine_file, [
-                           package, "/" + file, fpath, mgs],
-                           callback=None))
+            results.append(examine_file(package, "/" + file, fpath, mgs))
+            # results.append(pool.apply_async(examine_file, [
+            #               package, "/" + file, fpath, mgs],
+            #               callback=None))
 
-        pool.close()
-        pool.join()
+        # pool.close()
+        # pool.join()
 
-        infos = [x.get() for x in results]
+        # infos = [x.get() for x in results]
 
         for r in removed:
             package.remove_file(r)
-        return infos
+        return results
 
     def examine_packages(self, context, packages):
         """ Examine all packages, in order to update dependencies, etc """
-- 
2.8.1

