From 27f36725ff4baf188cd2af091c3eee2bd84c5096 Mon Sep 17 00:00:00 2001
From: Ikey Doherty <ikey@solus-project.com>
Date: Sat, 18 Jul 2015 16:16:23 +0100
Subject: [PATCH] Add per-package support to disable ccache (i.e. atom goes
 boom)

---
 build.py  | 2 +-
 sanity.py | 6 ++++++
 2 files changed, 7 insertions(+), 1 deletion(-)

diff --git a/build.py b/build.py
index facf81e..d68acff 100644
--- a/build.py
+++ b/build.py
@@ -64,7 +64,7 @@ global pkgfile
 
 def get_path():
     path = "/usr/bin:/bin"
-    if ccache:
+    if ccache and sanity.can_ccache:
         cpath = ""
         if os.path.exists("/usr/lib/ccache/bin"):
             cpath = "/usr/lib/ccache/bin"
diff --git a/sanity.py b/sanity.py
index cb1ab8b..36d8a5c 100644
--- a/sanity.py
+++ b/sanity.py
@@ -31,6 +31,9 @@ emul32 = False
 global autodep
 autodep = True
 
+global can_ccache
+can_ccache = True
+
 global rundeps
 rundeps = None
 
@@ -338,6 +341,9 @@ def sane(fpath):
     if "extract" in y:
         global pkg_extract
         pkg_extract = bool(y['extract'])
+    if "ccache" in y:
+        global can_ccache
+        can_ccache = bool(y['ccache'])
 
     if not "source" in y:
         print "Required list '%s' is missing" % "source"
-- 
2.4.6

