From 2021b884e1df643285149f33757f15160cd47c47 Mon Sep 17 00:00:00 2001
From: Ikey Doherty <ikey@solus-project.com>
Date: Fri, 17 Jul 2015 00:49:45 +0100
Subject: [PATCH] Enforce LD_AS_NEEDED by default

---
 build.py | 1 +
 1 file changed, 1 insertion(+)

diff --git a/build.py b/build.py
index 6f83ef7..facf81e 100644
--- a/build.py
+++ b/build.py
@@ -170,6 +170,7 @@ export package="%%package%%"
 export release="%%release%%"
 export version="%%version%%"
 export sources="%%sources%%"
+export LD_AS_NEEDED=1
 """ % get_path()
     if emul32:
         header += "\nexport EMUL32BUILD=1\n"
-- 
2.4.6

