From ab7765fb25aac8207882da09392decde66a17056 Mon Sep 17 00:00:00 2001
From: Ikey Doherty <ikey@solus-project.com>
Date: Sat, 2 Jul 2016 17:46:15 +0100
Subject: [PATCH 1/6] Fix automatic deps for git/emul32 builds

Signed-off-by: Ikey Doherty <ikey@solus-project.com>
---
 ypkg2/ypkgspec.py | 18 ++++++++++++++++++
 1 file changed, 18 insertions(+)

diff --git a/ypkg2/ypkgspec.py b/ypkg2/ypkgspec.py
index a7accc9..40b2a98 100644
--- a/ypkg2/ypkgspec.py
+++ b/ypkg2/ypkgspec.py
@@ -15,6 +15,7 @@ from . import console_ui
 from . import yamlhelper
 
 from .yamlhelper import OneOrMoreString, MultimapFormat
+from .sources import SourceManager, GitSource
 
 import os
 from collections import OrderedDict
@@ -250,6 +251,23 @@ class YpkgSpec:
         if "docs" not in self.summaries:
             self.add_summary("docs", "Documentation for {}".format(name))
 
+        extras = []
+        if self.pkg_emul32:
+            extras.extend(["glibc-32bit-devel", "libgcc-32bit",
+                           "libstdc++-32bit"])
+        man = SourceManager()
+        try:
+            man.identify_sources(self)
+            for x in man.sources:
+                if isinstance(x, GitSource):
+                    extras.append("git")
+                    break
+        except:
+            pass
+        for x in extras:
+            if x not in self.pkg_builddeps:
+                self.pkg_builddeps.append(x)
+
     def load_from_path(self, path):
         self.path = path
         if not os.path.exists(path):
-- 
2.10.0

