From c5807e12ca99f657f0391cb40d6b19b8849c2f00 Mon Sep 17 00:00:00 2001
From: Ikey Doherty <ikey@solus-project.com>
Date: Sat, 22 Aug 2015 01:02:59 +0100
Subject: [PATCH] Disable clang by default from now on

---
 build.py | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/build.py b/build.py
index 0a4d4b8..6993b0e 100644
--- a/build.py
+++ b/build.py
@@ -55,7 +55,7 @@ def which(p):
     return False
 
 global Clang
-Clang = which("clang")
+Clang = False
 
 global emul32
 emul32 = False
@@ -291,8 +291,8 @@ def build(fpath):
         wdir = get_work_dir()
         if "clang" in d:
             cl = bool(d['clang'])
-            if not cl:
-                print "Disabling clang"
+            if cl:
+                print "Enabling clang"
             global Clang
             Clang = cl
 
-- 
2.5.0

