From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Reilly Brogan <reilly@reillybrogan.com>
Date: Mon, 2 Dec 2024 14:16:47 -0600
Subject: [PATCH] ypkg: Add Qt6 SBOM files to devel

Signed-off-by: Reilly Brogan <reilly@reillybrogan.com>
---
 ypkg2/packages.py | 1 +
 1 file changed, 1 insertion(+)

diff --git a/ypkg2/packages.py b/ypkg2/packages.py
index 6a035a9..4c101ea 100644
--- a/ypkg2/packages.py
+++ b/ypkg2/packages.py
@@ -252,6 +252,7 @@ class PackageGenerator:
         self.add_pattern("/usr/lib64/qt6/metatypes/qt*metatypes.json", "devel")
         self.add_pattern("/usr/lib/qt6/modules/*.json", "devel")
         self.add_pattern("/usr/lib64/qt6/modules/*.json", "devel")
+        self.add_pattern("/usr/lib64/qt6/sbom", "devel")
 
     def add_file(self, path):
         """ Add a file path to the owned list and place it into the correct
