From e844c7f6d7b6fcd8bd80439a987828c668271836 Mon Sep 17 00:00:00 2001
From: Ikey Doherty <ikey@solus-project.com>
Date: Sun, 24 Apr 2016 17:43:16 +0100
Subject: [PATCH] Sync with Clear flags and set function alignment

Signed-off-by: Ikey Doherty <ikey@solus-project.com>
---
 ypkg2/ypkgcontext.py | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/ypkg2/ypkgcontext.py b/ypkg2/ypkgcontext.py
index e5cfa2e..680e8bd 100644
--- a/ypkg2/ypkgcontext.py
+++ b/ypkg2/ypkgcontext.py
@@ -19,7 +19,8 @@ import shutil
 
 # These flag sets are courtesy of autospec in
 # Clear Linux Project For Intel Architecture
-SPEED_FLAGS = "-flto -ffunction-sections -fno-semantic-interposition -O3"
+SPEED_FLAGS = "-flto -ffunction-sections -fno-semantic-interposition -O3 " \
+              "-falign-functions=32"
 SIZE_FLAGS = "-Os -ffunction-sections"
 PGO_GEN_FLAGS = "-fprofile-generate -fprofile-dir=\"{}\""
 PGO_USE_FLAGS = "-fprofile-use -fprofile-dir=\"{}\" -fprofile-correction"
-- 
2.8.1

