From d2616e4ac6c69eed4a9e4b991a817ed1465b76f4 Mon Sep 17 00:00:00 2001
From: Ikey Doherty <ikey@solus-project.com>
Date: Wed, 29 Nov 2017 14:45:09 +0000
Subject: [PATCH] context: Ensure we REALLY prevent symbolic

Signed-off-by: Ikey Doherty <ikey@solus-project.com>
---
 ypkg2/ypkgcontext.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/ypkg2/ypkgcontext.py b/ypkg2/ypkgcontext.py
index 9b1fe14..969efab 100644
--- a/ypkg2/ypkgcontext.py
+++ b/ypkg2/ypkgcontext.py
@@ -339,7 +339,7 @@ class YpkgContext:
             self.build.cxxflags = Flags.optimize_flags(self.build.cxxflags,
                                                        opt,
                                                        self.spec.pkg_clang)
-            if opt == "no-bind-now":
+            if opt == "no-bind-now" or opt == "no-symbolic":
                 self.build.ldflags = Flags.optimize_flags(self.build.ldflags,
                                                           opt,
                                                           self.spec.pkg_clang)
-- 
2.15.1

