From e3fdcbec7d6c42366cbdd539d89b2ddcc55ab007 Mon Sep 17 00:00:00 2001
From: Ikey Doherty <ikey@solus-project.com>
Date: Fri, 15 May 2015 17:17:26 +0100
Subject: [PATCH 1/2] ypkg: Preserve file permissions/times by using move, not
 copy

---
 ypkg.py | 5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

diff --git a/ypkg.py b/ypkg.py
index fea51e6..6c21d97 100755
--- a/ypkg.py
+++ b/ypkg.py
@@ -98,10 +98,11 @@ def main():
     packageit.packageit(fpath, build.InstallDir, pspec)
     f = open(actions, "w")
     tmpl = """
-from pisi.actionsapi import pisitools
+import shutil
+from pisi.actionsapi import get
 
 def install():
-    pisitools.insinto("/", "%s/*")
+    shutil.move("%s/", get.installDIR())
 """ % build.InstallDir
 
     if not sanity.autodep:
-- 
2.3.2

