From 53b16554af2b9d3ab7f9fe078161484999755daa Mon Sep 17 00:00:00 2001
From: Ikey Doherty <ikey@solus-project.com>
Date: Sun, 10 Apr 2016 06:26:45 +0100
Subject: [PATCH 1/7] Fix history generation slightly

Signed-off-by: Ikey Doherty <ikey@solus-project.com>
---
 ypkg-gen-history | 5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

diff --git a/ypkg-gen-history b/ypkg-gen-history
index 31bc82b..d9b9129 100755
--- a/ypkg-gen-history
+++ b/ypkg-gen-history
@@ -139,8 +139,6 @@ def main():
     outputDir = os.path.abspath(outputDir)
 
     tags = get_git_tags(wdir)
-    if len(tags) > MAX_HISTORY_LEN:
-        tags = tags[:MAX_HISTORY_LEN]
 
     history = list()
 
@@ -158,6 +156,9 @@ def main():
 
     history = sorted(history, key=lambda x: x[0].pkg_release, reverse=True)
 
+    if len(history) > MAX_HISTORY_LEN:
+        history = history[0:MAX_HISTORY_LEN]
+
     hist = os.path.join(outputDir, "history.xml")
 
     hist_obj = PackageHistory()
-- 
2.8.1

