From 26b25b024de422e81d0872e56033dd9335046ce3 Mon Sep 17 00:00:00 2001
From: Ikey Doherty <ikey@solus-project.com>
Date: Wed, 21 Oct 2015 23:43:31 +0100
Subject: [PATCH] Force-purge the evil .la's

Signed-off-by: Ikey Doherty <ikey@solus-project.com>
---
 build.py | 1 +
 1 file changed, 1 insertion(+)

diff --git a/build.py b/build.py
index 6993b0e..020f842 100644
--- a/build.py
+++ b/build.py
@@ -199,6 +199,7 @@ fi"""
 if [ -f "%installroot%/usr/share/man/dir" ]; then
     rm -v "%installroot%/usr/share/man/dir"
 fi
+find "%installroot%/%libdir%" -type f -name *.la|xargs rm -rf
 """
 
     while (True):
-- 
2.6.2

