From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Reilly Brogan <reilly@reillybrogan.com>
Date: Fri, 22 Nov 2024 21:23:14 -0600
Subject: [PATCH] ypkg: Adapt to clang rename

Signed-off-by: Reilly Brogan <reilly@reillybrogan.com>
---
 ypkg2/ypkgspec.py | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/ypkg2/ypkgspec.py b/ypkg2/ypkgspec.py
index 21bb1a2..be43979 100644
--- a/ypkg2/ypkgspec.py
+++ b/ypkg2/ypkgspec.py
@@ -283,7 +283,10 @@ class YpkgSpec:
             extras.extend(["glibc-32bit-devel", "libgcc-32bit",
                            "libstdc++-32bit", "fakeroot-32bit"])
         if self.pkg_clang:
-            extras.extend(["llvm-clang", "llvm-clang-devel"])
+            if self.pkg_emul32:
+                extras.extend(["clang-32bit"])
+            else:
+                extras.extend(["clang"])
         if self.pkg_optimize:
             if "thin-lto" or "icf-safe" or "icf-all" in self.pkg_optimize and not self.pkg_clang:
                 extras.extend(["binutils-gold"])
