From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Reilly Brogan <reilly@reillybrogan.com>
Date: Tue, 17 Dec 2024 22:24:37 -0600
Subject: [PATCH] examine: Fix runpath detection

Signed-off-by: Reilly Brogan <reilly@reillybrogan.com>
---
 ypkg2/examine.py | 9 +++++++++
 1 file changed, 9 insertions(+)

diff --git a/ypkg2/examine.py b/ypkg2/examine.py
index f383aed..d67abc8 100644
--- a/ypkg2/examine.py
+++ b/ypkg2/examine.py
@@ -32,6 +32,7 @@ v_pie = re.compile(r".*ELF (64|32)\-bit LSB pie executable,")
 v_rel = re.compile(r".*ELF (64|32)\-bit LSB relocatable,")
 shared_lib = re.compile(r".*Shared library: \[(.*)\].*")
 r_path = re.compile(r".*Library rpath: \[(.*)\].*")
+run_path = re.compile(r".*Library runpath: \[(.*)\].*")
 r_soname = re.compile(r".*Library soname: \[(.*)\].*")
 
 global_xattrs = dict()
@@ -141,6 +142,14 @@ class FileReport:
                 self.rpaths.update(r.group(1).split(":"))
                 continue
 
+            # Also check runpath
+            r2 = run_path.match(line)
+            if r2:
+                if self.rpaths is None:
+                    self.rpaths = set()
+                self.rpaths.update(r2.group(1).split(":"))
+                continue
+
             # Match direct needed dependency
             m = shared_lib.match(line)
             if m:
