From 72cedbc74fa77695fadad6699dd923be14c8ab15 Mon Sep 17 00:00:00 2001
From: Joey Riches <josephriches@gmail.com>
Date: Thu, 1 Dec 2022 09:55:24 +0000
Subject: [PATCH 1/1] ypkg2: Add paths containing /usr/share/doc to -docs
 subpackage

---
 ypkg2/packages.py | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/ypkg2/packages.py b/ypkg2/packages.py
index 73def86..2c07dc9 100644
--- a/ypkg2/packages.py
+++ b/ypkg2/packages.py
@@ -198,7 +198,9 @@ class PackageGenerator:
         self.add_pattern("/usr/lib/debug/", "dbginfo", priority=DBG)
         self.add_pattern("/usr/lib32/debug/", "32bit-dbginfo", priority=DBG)
 
-        self.add_pattern("/usr/share/gtk-doc/html/", "docs")
+        # Docs subpackage
+        self.add_pattern("/usr/share/gtk-doc/html/", "docs") # gtk-doc
+        self.add_pattern("/usr/share/doc", "docs") # gi-docgen and others
 
         # cmake stuffs
         self.add_pattern("/usr/share/cmake/", "devel")
-- 
2.35.5

