From 1b80abce6c5eda7440fb32a4ba3426e126710478 Mon Sep 17 00:00:00 2001
From: Ikey Doherty <ikey@solus-project.com>
Date: Mon, 14 Dec 2015 14:52:06 +0000
Subject: [PATCH] Reset --build host on native build

Signed-off-by: Ikey Doherty <ikey@solus-project.com>
---
 build.py | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/build.py b/build.py
index e21f80c..8cd8602 100644
--- a/build.py
+++ b/build.py
@@ -113,6 +113,8 @@ def escape(inp, wdir, name):
         if "-march=%s" % arch in cxxflags:
             cxxflags = cxxflags.replace("-march=%s" % arch, "-march=i686")
         host = "i686-pc-linux-gnu"
+    else:
+        host = conf.values.build.host
 
     prefix = "/usr" if not emul32 else "/emul32"
 
-- 
2.6.4

