From 4878405ef85128007c55ea423091cf47cc668ae2 Mon Sep 17 00:00:00 2001
From: Ikey Doherty <ikey@solus-project.com>
Date: Sun, 3 Apr 2016 21:48:32 +0100
Subject: [PATCH] Only override component from component.xml if not set in yaml

Signed-off-by: Ikey Doherty <ikey@solus-project.com>
---
 ypkg.py | 11 ++++++-----
 1 file changed, 6 insertions(+), 5 deletions(-)

diff --git a/ypkg.py b/ypkg.py
index c827f75..4e68f90 100755
--- a/ypkg.py
+++ b/ypkg.py
@@ -68,13 +68,14 @@ def main():
     if not sane(fpath, True): # then y made for linoox
         return 1
 
-    comp = os.path.join(os.path.dirname(os.path.abspath(fpath)), "..", "component.xml")
-    if os.path.exists(comp):
-        packageit.component = load_component(comp)
-    else:
-        comp = os.path.join(os.path.dirname(os.path.abspath(fpath)), "..", "..", "component.xml")
+    if packageit.component is None:
+        comp = os.path.join(os.path.dirname(os.path.abspath(fpath)), "..", "component.xml")
         if os.path.exists(comp):
             packageit.component = load_component(comp)
+        else:
+            comp = os.path.join(os.path.dirname(os.path.abspath(fpath)), "..", "..", "component.xml")
+            if os.path.exists(comp):
+                packageit.component = load_component(comp)
 
 
     pspec = build.BuildPrefix + "/%s.xml" % sanity.name
-- 
2.7.3

