From c4829f063670ba41d460f99319b6ae8f5164db7c Mon Sep 17 00:00:00 2001
From: Ikey Doherty <ikey@solus-project.com>
Date: Sun, 24 Apr 2016 12:55:34 +0100
Subject: [PATCH] Also export FFLAGS/FCFLAGS

Signed-off-by: Ikey Doherty <ikey@solus-project.com>
---
 ypkg2/scripts.py | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/ypkg2/scripts.py b/ypkg2/scripts.py
index 61f2239..0c2c0d3 100644
--- a/ypkg2/scripts.py
+++ b/ypkg2/scripts.py
@@ -139,6 +139,8 @@ class ScriptGenerator:
         self.define_export("CFLAGS", " ".join(self.context.build.cflags))
         self.define_export("CXXFLAGS", " ".join(self.context.build.cxxflags))
         self.define_export("LDFLAGS", " ".join(self.context.build.ldflags))
+        self.define_export("FFLAGS", " ".join(self.context.build.cflags))
+        self.define_export("FCFLAGS", " ".join(self.context.build.cflags))
         self.define_export("PATH", self.context.get_path())
         self.define_export("workdir", "%workdir%")
         self.define_export("package", "%package%")
-- 
2.8.1

