From 1649aeea913a242df4604e9c4c5cd9d52c062e9b Mon Sep 17 00:00:00 2001
From: Ikey Doherty <ikey@solus-project.com>
Date: Thu, 30 Nov 2017 17:32:53 +0000
Subject: [PATCH 1/6] meta: Don't autodep 32-bit onto a kernel module

Signed-off-by: Ikey Doherty <ikey@solus-project.com>
---
 ypkg2/metadata.py | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/ypkg2/metadata.py b/ypkg2/metadata.py
index 41cb537..b453eee 100644
--- a/ypkg2/metadata.py
+++ b/ypkg2/metadata.py
@@ -299,7 +299,8 @@ def handle_dependencies(context, gene, metadata, package, files):
 
     # Ensure some sane defaults are in place
     if package.name == "32bit" and "main" in gene.packages:
-        dependencies.add(context.spec.get_package_name("main"))
+        if context.spec.get_component("main") != "kernel.drivers":
+            dependencies.add(context.spec.get_package_name("main"))
     elif package.name == "32bit-devel":
         if "32bit" in gene.packages:
             dependencies.add(context.spec.get_package_name("32bit"))
-- 
2.16.1

