From 9a079ebb0f97ad44b24f25b2e70707b164764497 Mon Sep 17 00:00:00 2001
From: Ikey Doherty <ikey@solus-project.com>
Date: Thu, 20 Aug 2015 10:02:37 +0100
Subject: [PATCH] No longer set docdir, causes many issues..

Signed-off-by: Ikey Doherty <ikey@solus-project.com>
---
 build.py | 1 -
 1 file changed, 1 deletion(-)

diff --git a/build.py b/build.py
index d68acff..0a4d4b8 100644
--- a/build.py
+++ b/build.py
@@ -122,7 +122,6 @@ def escape(inp, wdir, name):
                            --mandir=/usr/share/man \
                            --infodir=/usr/share/man \
                            --datadir=/usr/share/ \
-                           --docdir=/usr/share/doc \
                            --sysconfdir=/etc \
                            --localstatedir=/var \
                            --libexecdir=/usr/%s/%s" % (prefix, host, libdir, libdir, name)
-- 
2.5.0

