From 7288e708029818e3c141b4a5dbb30776d2d57b61 Mon Sep 17 00:00:00 2001
From: Ikey Doherty <ikey@solus-project.com>
Date: Sun, 17 Apr 2016 06:08:51 +0100
Subject: [PATCH] Ensure we nuke /emul32 paths

Signed-off-by: Ikey Doherty <ikey@solus-project.com>
---
 ypkg2/examine.py | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/ypkg2/examine.py b/ypkg2/examine.py
index b10dadd..2791d74 100644
--- a/ypkg2/examine.py
+++ b/ypkg2/examine.py
@@ -325,6 +325,8 @@ class PackageExaminer:
             return True
         if pretty == "/usr/share/info/dir":
             return True
+        if pretty.startswith("/emul32"):
+            return True
         return False
 
     def file_is_of_interest(self, pretty, file, mgs):
-- 
2.8.1

