From e98836da232c9c8b25c8b901bac840bed47f1556 Mon Sep 17 00:00:00 2001
From: Ikey Doherty <ikey@solus-project.com>
Date: Sun, 30 Oct 2016 11:54:38 +0000
Subject: [PATCH 2/4] Drop potentially harmful -msse4 due to building for all
 vendors

Signed-off-by: Ikey Doherty <ikey@solus-project.com>
---
 ypkg2/ypkgcontext.py | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/ypkg2/ypkgcontext.py b/ypkg2/ypkgcontext.py
index 2d87f15..0d4128c 100644
--- a/ypkg2/ypkgcontext.py
+++ b/ypkg2/ypkgcontext.py
@@ -26,8 +26,8 @@ PGO_GEN_FLAGS = "-fprofile-generate -fprofile-dir=\"{}\""
 PGO_USE_FLAGS = "-fprofile-use -fprofile-dir=\"{}\" -fprofile-correction"
 BIND_NOW_FLAGS = ["-Wl,-z,now"]
 
-# AVX2/Haswell optimisations
-AVX2_FLAGS = "-msse3 -mssse3 -msse4 -msse4.1 -msse4.2 -mavx -mavx2"
+# AVX2
+AVX2_FLAGS = "-msse3 -mssse3 -msse4.1 -msse4.2 -mavx -mavx2"
 
 
 class Flags:
-- 
2.10.1

