From e63238ae563463de9ed0c1f48f1645cc1f757536 Mon Sep 17 00:00:00 2001
From: Ikey Doherty <ikey@solus-project.com>
Date: Sat, 28 May 2016 22:29:02 +0100
Subject: [PATCH 2/3] Use the correct context when cleaning, so that 32-bit PGO
 is clean

Signed-off-by: Ikey Doherty <ikey@solus-project.com>
---
 ypkg2/main.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/ypkg2/main.py b/ypkg2/main.py
index 79da911..6b0867d 100644
--- a/ypkg2/main.py
+++ b/ypkg2/main.py
@@ -276,7 +276,7 @@ def build_package(filename, outputDir):
             # existing build directories for the current context and then
             # re-extracting sources
             if step == "setup":
-                if not clean_build_dirs(ctx):
+                if not clean_build_dirs(context):
                     sys.exit(1)
 
                 # Only ever extract the primary source ourselves
-- 
2.8.3

