From 0a427a10fff91696f16f6b1fe846164f1d4c5061 Mon Sep 17 00:00:00 2001
From: Ikey Doherty <ikey@solus-project.com>
Date: Sun, 22 Jan 2017 04:13:04 +0000
Subject: [PATCH 2/2] metadata: Force file uid to 0 when matching the current
 UID

Signed-off-by: Ikey Doherty <ikey@solus-project.com>
---
 ypkg2/metadata.py | 7 +++++++
 1 file changed, 7 insertions(+)

diff --git a/ypkg2/metadata.py b/ypkg2/metadata.py
index f39e511..aba4ddf 100644
--- a/ypkg2/metadata.py
+++ b/ypkg2/metadata.py
@@ -413,6 +413,13 @@ def create_eopkg(context, gene, package, outputDir):
         orgname = os.path.join(context.get_install_dir(), finfo.path)
         orgname = orgname.encode('utf-8').decode('utf-8').encode("latin1")
 
+        try:
+            st = os.stat(orgname)
+            if st.st_uid >= 1000:
+                os.lchown(orgname, 0, 0)
+        except Exception as e:
+            console_ui.emit_warning("Fail to chown {}: {}".format(orgname, e))
+
         if os.path.islink(orgname) and not os.path.isdir(orgname):
             t = history_timestamp
             cmd = "touch -d \"@{}\" -h \"{}\"".format(t, orgname)
-- 
2.11.0

