From 8eb11007ee0015a3816b4a2306f148445e560f11 Mon Sep 17 00:00:00 2001
From: Ikey Doherty <ikey@solus-project.com>
Date: Sun, 17 Apr 2016 05:20:17 +0100
Subject: [PATCH 2/2] Only perform *filter* of no-bind-now on ldflags for xorg
 modules

Signed-off-by: Ikey Doherty <ikey@solus-project.com>
---
 ypkg2/ypkgcontext.py | 1 +
 1 file changed, 1 insertion(+)

diff --git a/ypkg2/ypkgcontext.py b/ypkg2/ypkgcontext.py
index bf36324..e5cfa2e 100644
--- a/ypkg2/ypkgcontext.py
+++ b/ypkg2/ypkgcontext.py
@@ -227,6 +227,7 @@ class YpkgContext:
                                                      self.spec.pkg_optimize)
             self.build.cxxflags = Flags.optimize_flags(self.build.cxxflags,
                                                        self.spec.pkg_optimize)
+        if self.spec.pkg_optimize in ["no-bind-now"]:
             self.build.ldflags = Flags.optimize_flags(self.build.ldflags,
                                                       self.spec.pkg_optimize)
 
-- 
2.8.1

