From fb4f116c4789a411b706990b83d2357cc8897fac Mon Sep 17 00:00:00 2001
From: Ikey Doherty <ikey@solus-project.com>
Date: Fri, 2 Jun 2017 02:30:22 +0100
Subject: [PATCH 2/4] spec: Ignore duplication when key isn't the same as the
 value

Signed-off-by: Ikey Doherty <ikey@solus-project.com>
---
 ypkg2/ypkgspec.py | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/ypkg2/ypkgspec.py b/ypkg2/ypkgspec.py
index 2c65895..c4adc5a 100644
--- a/ypkg2/ypkgspec.py
+++ b/ypkg2/ypkgspec.py
@@ -161,7 +161,8 @@ class YpkgSpec:
         """ Add a 'replaces:' to the package """
         if key not in self.replaces:
             self.replaces[key] = list()
-        if val in self.replaces:
+        # i.e. llvm-clang-devel replaces clang-devel
+        if val in self.replaces and val != key:
             console_ui.emit_warning("YAML",
                                     "Duplicate replace: {}".format(val))
             return
-- 
2.13.1

