From 861693f37ccc3ae0ab4cae67cadf1fdb9a1ed51d Mon Sep 17 00:00:00 2001
From: Ikey Doherty <ikey@solus-project.com>
Date: Sat, 2 Jul 2016 17:50:36 +0100
Subject: [PATCH 2/8] Fix borkage with empty builddep set

Signed-off-by: Ikey Doherty <ikey@solus-project.com>
---
 ypkg2/ypkgspec.py | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/ypkg2/ypkgspec.py b/ypkg2/ypkgspec.py
index 40b2a98..3f00974 100644
--- a/ypkg2/ypkgspec.py
+++ b/ypkg2/ypkgspec.py
@@ -265,6 +265,8 @@ class YpkgSpec:
         except:
             pass
         for x in extras:
+            if not self.pkg_builddeps:
+                self.pkg_builddeps = list()
             if x not in self.pkg_builddeps:
                 self.pkg_builddeps.append(x)
 
-- 
2.10.1

