From e4116a00b663de79af507a0d832c17748c733c28 Mon Sep 17 00:00:00 2001
From: Ikey Doherty <ikey@solus-project.com>
Date: Sun, 10 Apr 2016 06:30:44 +0100
Subject: [PATCH 2/2] Only use history if its complete

Signed-off-by: Ikey Doherty <ikey@solus-project.com>
---
 ypkg2/metadata.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/ypkg2/metadata.py b/ypkg2/metadata.py
index 3f9e10e..9824f74 100644
--- a/ypkg2/metadata.py
+++ b/ypkg2/metadata.py
@@ -190,7 +190,7 @@ def metadata_from_package(context, package, files):
             update = topup
             packager.name = topup.name
             packager.email = topup.email
-        meta.package.history = context.spec.history.history
+            meta.package.history = context.spec.history.history
 
     if not update:
         update = pisi.specfile.Update()
-- 
2.7.3

