From 200d4f9c740cbb8ffa207d0e5c003e30273fbbe0 Mon Sep 17 00:00:00 2001
From: Ikey Doherty <ikey@solus-project.com>
Date: Tue, 5 Jul 2016 18:17:53 +0100
Subject: [PATCH 3/6] Allow '+' as a valid character in package names

Signed-off-by: Ikey Doherty <ikey@solus-project.com>
---
 ypkg2/ypkgspec.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/ypkg2/ypkgspec.py b/ypkg2/ypkgspec.py
index 3f00974..ceb0c34 100644
--- a/ypkg2/ypkgspec.py
+++ b/ypkg2/ypkgspec.py
@@ -43,7 +43,7 @@ class PackageSanity:
                                   "Package names cannot contain whitespace")
             return False
         illegal = set()
-        permitted = ['-', '_']
+        permitted = ['-', '_', '+']
         for char in name[0:]:
             if char in permitted:
                 continue
-- 
2.10.0

