From 120457396ee9f97add10b1bb02df98e2beb7da4a Mon Sep 17 00:00:00 2001
From: Ikey Doherty <ikey@solus-project.com>
Date: Mon, 21 Nov 2016 22:01:04 +0000
Subject: [PATCH 3/5] scripts: Export YPKG_NJOBS for use in other macros

Signed-off-by: Ikey Doherty <ikey@solus-project.com>
---
 ypkg2/scripts.py | 1 +
 1 file changed, 1 insertion(+)

diff --git a/ypkg2/scripts.py b/ypkg2/scripts.py
index 425a730..638ccdd 100644
--- a/ypkg2/scripts.py
+++ b/ypkg2/scripts.py
@@ -123,6 +123,7 @@ class ScriptGenerator:
         self.define_macro("installroot", self.context.get_install_dir())
         self.define_macro("workdir", self.work_dir)
         self.define_macro("JOBS", "-j{}".format(self.context.build.jobcount))
+        self.define_macro("YPKG_NJOBS", "{}".format(self.context.build.jobcount))
 
         # Consider moving this somewhere else
         self.define_macro("CFLAGS", " ".join(self.context.build.cflags))
-- 
2.10.2

