From e04c68337c21221753977f27b3caa34eba1d325b Mon Sep 17 00:00:00 2001
From: Ikey Doherty <ikey@solus-project.com>
Date: Fri, 14 Apr 2017 22:35:21 +0100
Subject: [PATCH 3/3] context: Update BIND_NOW_FLAGS to be useful

Signed-off-by: Ikey Doherty <ikey@solus-project.com>
---
 ypkg2/ypkgcontext.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/ypkg2/ypkgcontext.py b/ypkg2/ypkgcontext.py
index 9d214cd..2fdae5c 100644
--- a/ypkg2/ypkgcontext.py
+++ b/ypkg2/ypkgcontext.py
@@ -24,7 +24,7 @@ SPEED_FLAGS = "-ffunction-sections -fno-semantic-interposition -O3 " \
 SIZE_FLAGS = "-Os -ffunction-sections"
 PGO_GEN_FLAGS = "-fprofile-generate -fprofile-dir=\"{}\""
 PGO_USE_FLAGS = "-fprofile-use -fprofile-dir=\"{}\" -fprofile-correction"
-BIND_NOW_FLAGS = ["-Wl,-z,now"]
+BIND_NOW_FLAGS = ["-Wl,-z,now", "-Wl,-z -Wl,relro", "-Wl,-z -Wl,now"]
 
 # AVX2
 AVX2_FLAGS = "-mavx2"
-- 
2.12.2

