From 6544fe0850439b504880909b46b509535688a5c3 Mon Sep 17 00:00:00 2001
From: Ikey Doherty <ikey@solus-project.com>
Date: Sun, 29 May 2016 07:50:12 +0100
Subject: [PATCH 3/3] Don't check mimetype of .pc

Signed-off-by: Ikey Doherty <ikey@solus-project.com>
---
 ypkg2/examine.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/ypkg2/examine.py b/ypkg2/examine.py
index d955b55..546ceee 100644
--- a/ypkg2/examine.py
+++ b/ypkg2/examine.py
@@ -37,7 +37,7 @@ def is_pkgconfig_file(pretty, mgs):
     """ Simple as it sounds, work out if this is a pkgconfig file """
     if pretty.endswith(".pc"):
         pname = os.path.basename(os.path.dirname(pretty))
-        if pname == "pkgconfig" and "ASCII" in mgs:
+        if pname == "pkgconfig":
             return True
     return False
 
-- 
2.8.3

