From e8f89ac7a45cdbac62ff8713f969c67a3df6ba77 Mon Sep 17 00:00:00 2001
From: Ikey Doherty <ikey@solus-project.com>
Date: Wed, 6 Jul 2016 22:18:52 +0100
Subject: [PATCH 4/4] "." is also valid numnuts

Signed-off-by: Ikey Doherty <ikey@solus-project.com>
---
 ypkg2/ypkgspec.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/ypkg2/ypkgspec.py b/ypkg2/ypkgspec.py
index ceb0c34..a773789 100644
--- a/ypkg2/ypkgspec.py
+++ b/ypkg2/ypkgspec.py
@@ -43,7 +43,7 @@ class PackageSanity:
                                   "Package names cannot contain whitespace")
             return False
         illegal = set()
-        permitted = ['-', '_', '+']
+        permitted = ['-', '_', '+', '.']
         for char in name[0:]:
             if char in permitted:
                 continue
-- 
2.9.0

