From ed711e63e8d5d382b6b98e53f2562afa4ac9a26a Mon Sep 17 00:00:00 2001
From: Ikey Doherty <ikey@solus-project.com>
Date: Sun, 21 Jan 2018 19:55:07 +0000
Subject: [PATCH 5/6] dependencies: Prepare for glvnd switch

Signed-off-by: Ikey Doherty <ikey@solus-project.com>
---
 ypkg2/dependencies.py | 11 ++++++-----
 1 file changed, 6 insertions(+), 5 deletions(-)

diff --git a/ypkg2/dependencies.py b/ypkg2/dependencies.py
index 8d743e5..f380c65 100644
--- a/ypkg2/dependencies.py
+++ b/ypkg2/dependencies.py
@@ -17,9 +17,9 @@ from pisi.db.packagedb import PackageDB
 from pisi.db.filesdb import FilesDB
 import os
 
-
-# On Solus, these are provided only by links. Ensure that it only ever
-# depends on mesalib, and *not* the resolved nvidia, etc, possibilities
+# Provided historically for our pre-glvnd architecture.
+# Technically speaking this isn't required anymore, but lets just
+# play it safe for those directly using ypkg on old NVIDIA drivers.
 ExceptionRules = [
     "libEGL.so",
     "libEGL.so.1",
@@ -32,6 +32,7 @@ ExceptionRules = [
     "libGLESv2.so.2.0.0",
     "libGL.so",
     "libGL.so.1",
+    "libGL.so.1.0.0",
     "libGL.so.1.2.0",
     "libglx.so",
     "libglx.so.1",
@@ -129,9 +130,9 @@ class DependencyResolver:
 
         if symbol in ExceptionRules:
             if info.emul32:
-                return "mesalib-32bit"
+                return "libglvnd-32bit"
             else:
-                return "mesalib"
+                return "libglvnd"
 
         if not paths:
             paths = ["/usr/lib64", "/usr/lib"]
-- 
2.16.1

