From 6a385ebb597fc3d3edc87840fdfd651f1d0b0b68 Mon Sep 17 00:00:00 2001
From: Ikey Doherty <ikey@solus-project.com>
Date: Sat, 16 Jul 2016 23:57:29 +0100
Subject: [PATCH 5/5] scripts: Digits are valid too

Signed-off-by: Ikey Doherty <ikey@solus-project.com>
---
 ypkg2/scripts.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/ypkg2/scripts.py b/ypkg2/scripts.py
index 0c2c0d3..4126cb2 100644
--- a/ypkg2/scripts.py
+++ b/ypkg2/scripts.py
@@ -184,7 +184,7 @@ class ScriptGenerator:
         return ret
 
     def is_valid_macro_char(self, char):
-        if char.isalpha():
+        if char.isalpha() or char.isdigit():
             return True
         if char == "_":
             return True
-- 
2.9.1

