From e6280d153eb2f67cf8b4b77c1db09c1f7751811a Mon Sep 17 00:00:00 2001
From: Ikey Doherty <ikey@solus-project.com>
Date: Thu, 2 Feb 2017 01:55:57 +0000
Subject: [PATCH 5/5] metada: Also fallback the date object itself

Signed-off-by: Ikey Doherty <ikey@solus-project.com>
---
 ypkg2/metadata.py | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/ypkg2/metadata.py b/ypkg2/metadata.py
index acdb9b2..27fcb81 100644
--- a/ypkg2/metadata.py
+++ b/ypkg2/metadata.py
@@ -50,6 +50,7 @@ FileTypes = OrderedDict([
 history_timestamp = None
 history_date = None
 fallback_timestamp = None
+fallback_date = None
 
 accum_packages = dict()
 
@@ -74,10 +75,12 @@ def initialize_timestamp(spec):
     global history_date
     global history_timestamp
     global fallback_timestamp
+    global fallback_date
 
     dt = datetime.datetime.utcnow()
     s = dt.strftime("%Y-%m-%d")
     fallback_timestamp = utc_date_for_date_only(s)
+    fallback_date = s
 
     # If it was already set, just skip it.
     if history_timestamp is not None:
@@ -166,6 +169,7 @@ def metadata_from_package(context, package, files):
     global history_date
     global history_timestamp
     global fallback_timestamp
+    global fallback_date
     global accum_packages
 
     meta = pisi.metadata.MetaData()
@@ -195,6 +199,7 @@ def metadata_from_package(context, package, files):
         if l_release != release or l_version != version:
             console_ui.emit_info("History", "Constructing new history entry")
             history_timestamp = fallback_timestamp
+            history_date = fallback_date
         else:
             # Last updater is listed as maintainer in eopkg blame
             update = topup
-- 
2.11.0

