From 98b0897eb3541e18ffe9a8b64caa5d636179246a Mon Sep 17 00:00:00 2001
From: Ikey Doherty <ikey@solus-project.com>
Date: Sat, 4 Jun 2016 08:07:11 +0100
Subject: [PATCH 5/5] Fix error message

Signed-off-by: Ikey Doherty <ikey@solus-project.com>
---
 ypkg-install-deps | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/ypkg-install-deps b/ypkg-install-deps
index 0ae6066..29da1fb 100755
--- a/ypkg-install-deps
+++ b/ypkg-install-deps
@@ -95,7 +95,7 @@ def main():
                                     "This dependency is not in any repo")
         if not pkg:
             console_ui.emit_error("BuildDep", "pkgconfig32({}) build dep"
-                                  " does not exist in the repository.")
+                                  " does't exist in the repository.".format(i))
             sys.exit(1)
         if not idb.has_package(pkg.name):
             ndeps.add(pkg.name)
-- 
2.8.3

