From 41d806b7b59250ade20106e85e2ad806ccd31590 Mon Sep 17 00:00:00 2001
From: Ikey Doherty <ikey@solus-project.com>
Date: Sun, 10 Apr 2016 09:51:41 +0100
Subject: [PATCH 6/9] Correctly add dbginfo deps

Signed-off-by: Ikey Doherty <ikey@solus-project.com>
---
 ypkg2/metadata.py | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/ypkg2/metadata.py b/ypkg2/metadata.py
index 9824f74..6926778 100644
--- a/ypkg2/metadata.py
+++ b/ypkg2/metadata.py
@@ -293,6 +293,12 @@ def handle_dependencies(context, gene, metadata, package, files):
     elif package.name == "devel":
         if "main" in gene.packages:
             dependencies.add(context.spec.get_package_name("main"))
+    elif package.name == "dbginfo":
+        if "main" in gene.packages:
+            dependencies.add(context.spec.get_package_name("main"))
+    elif package.name == "32bit-dbginfo":
+        if "32bit" in gene.packages:
+            dependencies.add(context.spec.get_package_name("32bit"))
 
     for dependency in dependencies:
         release = context.spec.pkg_release
-- 
2.8.1

