From 32d3951459744cc576637bef183fec81af904a6f Mon Sep 17 00:00:00 2001
From: Ikey Doherty <ikey@solus-project.com>
Date: Sun, 18 Jun 2017 13:11:44 +0100
Subject: [PATCH 6/7] dependencies: Forbid circular pkgconfig dependencies

Signed-off-by: Ikey Doherty <ikey@solus-project.com>
---
 ypkg2/dependencies.py | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/ypkg2/dependencies.py b/ypkg2/dependencies.py
index 734d245..ffd1778 100644
--- a/ypkg2/dependencies.py
+++ b/ypkg2/dependencies.py
@@ -229,6 +229,9 @@ class DependencyResolver:
             # Yes, it's a set, but i  dont want the ui emission spam
             if prov in tgtPkg.depend_packages:
                 continue
+            # Forbid circular dependencies
+            if pkgName == prov:
+                continue
             tgtPkg.depend_packages.add(prov)
 
             console_ui.emit_info("PKGCONFIG", "{} adds depend on {}".
-- 
2.13.2

