From f220eb0bede222e7f829057016c8c3fdb7b0de85 Mon Sep 17 00:00:00 2001
From: Ikey Doherty <ikey@solus-project.com>
Date: Mon, 11 Apr 2016 23:18:42 +0100
Subject: [PATCH 09/11] Correctly support mesalib/mesalib-32bit dependencies

Signed-off-by: Ikey Doherty <ikey@solus-project.com>
---
 ypkg2/dependencies.py | 26 ++++++++++++++++++++++++++
 1 file changed, 26 insertions(+)

diff --git a/ypkg2/dependencies.py b/ypkg2/dependencies.py
index ca2aa34..77f1eac 100644
--- a/ypkg2/dependencies.py
+++ b/ypkg2/dependencies.py
@@ -18,6 +18,26 @@ from pisi.db.filesdb import FilesDB
 import os
 
 
+# On Solus, these are provided only by links. Ensure that it only ever
+# depends on mesalib, and *not* the resolved nvidia, etc, possibilities
+ExceptionRules = [
+    "libEGL.so",
+    "libEGL.so.1",
+    "libEGL.so.1.0.0",
+    "libGLESv1_CM.so",
+    "libGLESv1_CM.so.1",
+    "libGLESv1_CM.so.1.1.0",
+    "libGLESv2.so",
+    "libGLESv2.so.2",
+    "libGLESv2.so.2.0.0",
+    "libGL.so",
+    "libGL.so.1",
+    "libGL.so.1.2.0",
+    "libglx.so",
+    "libglx.so.1",
+]
+
+
 class DependencyResolver:
 
     idb = None
@@ -74,6 +94,12 @@ class DependencyResolver:
             if symbol in self.bindeps_cache:
                 return self.bindeps_cache[symbol]
 
+        if symbol in ExceptionRules:
+            if info.emul32:
+                return "mesalib-32bit"
+            else:
+                return "mesalib"
+
         if not paths:
             paths = ["/usr/lib64", "/usr/lib"]
             if info.emul32:
-- 
2.8.1

