From e823fb9ce5161ba2a1c675dcf92b6582ad46c13b Mon Sep 17 00:00:00 2001
From: Ikey Doherty <ikey@solus-project.com>
Date: Thu, 20 Jul 2017 12:05:56 +0100
Subject: [PATCH 09/10] spec: Automatically add llvm-clang when using clang
 build

Signed-off-by: Ikey Doherty <ikey@solus-project.com>
---
 ypkg2/ypkgspec.py | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/ypkg2/ypkgspec.py b/ypkg2/ypkgspec.py
index c4adc5a..c40fe02 100644
--- a/ypkg2/ypkgspec.py
+++ b/ypkg2/ypkgspec.py
@@ -268,6 +268,11 @@ class YpkgSpec:
         if self.pkg_emul32:
             extras.extend(["glibc-32bit-devel", "libgcc-32bit",
                            "libstdc++-32bit", "fakeroot-32bit"])
+            if self.pkg_clang:
+                extras.extend(["llvm-clang-32bit", "llvm-clang-32bit-devel"])
+        if self.pkg_clang:
+            extras.extend(["llvm-clang", "llvm-clang-devel"])
+
         man = SourceManager()
         try:
             man.identify_sources(self)
-- 
2.13.3

