diff --git a/ypkg2/dependencies.py b/ypkg2/dependencies.py
index f380c65..79b1d39 100644
--- a/ypkg2/dependencies.py
+++ b/ypkg2/dependencies.py
@@ -197,6 +197,9 @@ class DependencyResolver:
             if nom:
                 pkg = self.idb.get_package(nom[0])
             if not pkg:
+                console_ui.emit_warning(
+                    "PKGCONFIG32",
+                    "Cannot find pkgconfig32({})".format(name))
                 nom = self.fdb.get_pkgconfig_provider(name)
                 if nom:
                     pkg = self.idb.get_package(nom[0])
@@ -204,6 +207,10 @@ class DependencyResolver:
             # PackageDB set
             if not pkg:
                 if name in self.pkgConfigs32:
+                    console_ui.emit_warning(
+                        "PKGCONFIG32",
+                        "Falling back to pdb for {}: {}".format(
+                            name, self.pkgConfigs32[name]))
                     pkg = self.pdb.get_package(self.pkgConfigs32[name])
             if not pkg:
                 if name in self.pkgConfigs:
