diff --git a/ypkg2/dependencies.py b/ypkg2/dependencies.py
index f380c65..eae4eb3 100644
--- a/ypkg2/dependencies.py
+++ b/ypkg2/dependencies.py
@@ -197,6 +197,9 @@ class DependencyResolver:
             if nom:
                 pkg = self.idb.get_package(nom[0])
             if not pkg:
+                console_ui.emit_warning(
+                    "PKGCONFIG32",
+                    "Cannot find pkgconfig32({})".format(name))
                 nom = self.fdb.get_pkgconfig_provider(name)
                 if nom:
                     pkg = self.idb.get_package(nom[0])
@@ -204,6 +207,10 @@ class DependencyResolver:
             # PackageDB set
             if not pkg:
                 if name in self.pkgConfigs32:
+                    console_ui.emit_warning(
+                        "PKGCONFIG32",
+                        "Falling back to pdb for {}: {}".format(
+                            name, self.pkgConfigs32[name]))
                     pkg = self.pdb.get_package(self.pkgConfigs32[name])
             if not pkg:
                 if name in self.pkgConfigs:
@@ -246,9 +253,22 @@ class DependencyResolver:
 
         for item in info.pkgconfig_deps:
 
+            if info.emul32:
+                console_ui.emit_warning(
+                    "PKGCONFIG32", "Looking for {}".format(item))
+            else:
+                console_ui.emit_warning(
+                    "PKGCONFIG64", "Looking for {}".format(item))
+
             prov = self.get_pkgconfig_provider(info, item)
             if not prov:
                 prov = self.get_pkgconfig_external(info, item)
+                if prov:
+                    console_ui.emit_warning(
+                        "PKGCONFIG", "Found {}".format(prov))
+            else:
+                console_ui.emit_warning(
+                    "PKGCONFIG_INTERNAL", "Found {}".format(prov))
 
             if not prov:
                 console_ui.emit_warning("PKGCONFIG", "Not adding unknown"
