diff --git a/src/graph/gml.hh b/src/graph/gml.hh
index 7bea0ca362305a1eba7bfba69813064363809ea0..62027809dab5885f47da3a01fbf3fcf057b8e02e 100644
--- a/src/graph/gml.hh
+++ b/src/graph/gml.hh
@@ -25,6 +25,7 @@
 #include <boost/variant/recursive_variant.hpp>
 #include <boost/variant/get.hpp>
 #include <boost/spirit/include/support_istream_iterator.hpp>
+#include <boost/regex/v5/unicode_iterator.hpp>
 
 #include <boost/algorithm/string/replace.hpp>
 
