From b113c89af1de2f87589175795a197f6384852a78 Mon Sep 17 00:00:00 2001
From: Christian Hergert <chergert@redhat.com>
Date: Tue, 21 Sep 2021 16:52:16 -0700
Subject: [PATCH] symbol-resolver: fix includes for close/lseek

---
 src/libsysprof/sysprof-symbol-resolver.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/src/libsysprof/sysprof-symbol-resolver.c b/src/libsysprof/sysprof-symbol-resolver.c
index 19e4140..3ba6166 100644
--- a/src/libsysprof/sysprof-symbol-resolver.c
+++ b/src/libsysprof/sysprof-symbol-resolver.c
@@ -20,6 +20,8 @@
 
 #include "config.h"
 
+#include <unistd.h>
+
 #include "sysprof-platform.h"
 #include "sysprof-symbol-resolver.h"
 
-- 
GitLab

