From 18f39ff9cd95122f9f4361000493d3e554a77e99 Mon Sep 17 00:00:00 2001
From: Peter O'Connor <peter@solus-project.com>
Date: Sat, 14 Apr 2018 20:09:50 +1000
Subject: [PATCH] Disable building unshipped code

Signed-off-by: Peter O'Connor <peter@solus-project.com>
---
 CMakeLists.txt | 2 --
 1 file changed, 2 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index d1db64b..646f0db 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -54,9 +54,7 @@ endif ()
 add_subdirectory (src)
 add_subdirectory (pixmaps)
 if (ENABLE_EXAMPLES)
-  add_subdirectory(examples)
 endif()
-add_subdirectory(autotests)
 
 if (BUILD_QCH)
     ecm_install_qch_export(
-- 
2.16.3

