From 3fbc02091e95a49a9a69f5f3339a8fe616b682a2 Mon Sep 17 00:00:00 2001
From: Peter O'Connor <peter@solus-project.com>
Date: Wed, 4 Apr 2018 10:57:28 +1000
Subject: [PATCH] Disable building unshipped code

Signed-off-by: Peter O'Connor <peter@solus-project.com>
---
 CMakeLists.txt | 2 --
 1 file changed, 2 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 8759073..c4d29fe 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -239,8 +239,6 @@ install(FILES   updaters/kscreenlocker.upd DESTINATION ${KDE_INSTALL_DATADIR}/kc
 install(PROGRAMS updaters/ksreenlocker_5_3_separate_autologin.pl
                 DESTINATION ${KDE_INSTALL_DATADIR}/kconf_update)
 
-add_subdirectory(autotests)
-add_subdirectory(tests)
 
 feature_summary(WHAT ALL INCLUDE_QUIET_PACKAGES FATAL_ON_MISSING_REQUIRED_PACKAGES)
 
-- 
2.16.3

