From f35e924b2c92c71d57ba54c4bef8e454f47f171f Mon Sep 17 00:00:00 2001
From: Pierre-Yves <pyu@riseup.net>
Date: Wed, 24 May 2017 12:20:09 +0200
Subject: [PATCH] use gnu libidn2 library

Signed-off-by: Pierre-Yves <pyu@riseup.net>
---
 configure      | 20 ++++++++++----------
 src/text/url.c |  2 +-
 2 files changed, 11 insertions(+), 11 deletions(-)

diff --git a/configure b/configure
index 57cbc4d..0c47c9f 100755
--- a/configure
+++ b/configure
@@ -24966,19 +24966,19 @@ fi
 
 
 pkg_failed=no
-{ $as_echo "$as_me:${as_lineno-$LINENO}: checking for IDN" >&5
-$as_echo_n "checking for IDN... " >&6; }
+{ $as_echo "$as_me:${as_lineno-$LINENO}: checking for IDN2" >&5
+$as_echo_n "checking for IDN2... " >&6; }
 
 if test -n "$IDN_CFLAGS"; then
     pkg_cv_IDN_CFLAGS="$IDN_CFLAGS"
  elif test -n "$PKG_CONFIG"; then
     if test -n "$PKG_CONFIG" && \
-    { { $as_echo "$as_me:${as_lineno-$LINENO}: \$PKG_CONFIG --exists --print-errors \"libidn\""; } >&5
-  ($PKG_CONFIG --exists --print-errors "libidn") 2>&5
+    { { $as_echo "$as_me:${as_lineno-$LINENO}: \$PKG_CONFIG --exists --print-errors \"libidn2\""; } >&5
+  ($PKG_CONFIG --exists --print-errors "libidn2") 2>&5
   ac_status=$?
   $as_echo "$as_me:${as_lineno-$LINENO}: \$? = $ac_status" >&5
   test $ac_status = 0; }; then
-  pkg_cv_IDN_CFLAGS=`$PKG_CONFIG --cflags "libidn" 2>/dev/null`
+  pkg_cv_IDN_CFLAGS=`$PKG_CONFIG --cflags "libidn2" 2>/dev/null`
 		      test "x$?" != "x0" && pkg_failed=yes
 else
   pkg_failed=yes
@@ -24990,12 +24990,12 @@ if test -n "$IDN_LIBS"; then
     pkg_cv_IDN_LIBS="$IDN_LIBS"
  elif test -n "$PKG_CONFIG"; then
     if test -n "$PKG_CONFIG" && \
-    { { $as_echo "$as_me:${as_lineno-$LINENO}: \$PKG_CONFIG --exists --print-errors \"libidn\""; } >&5
-  ($PKG_CONFIG --exists --print-errors "libidn") 2>&5
+    { { $as_echo "$as_me:${as_lineno-$LINENO}: \$PKG_CONFIG --exists --print-errors \"libidn2\""; } >&5
+  ($PKG_CONFIG --exists --print-errors "libidn2") 2>&5
   ac_status=$?
   $as_echo "$as_me:${as_lineno-$LINENO}: \$? = $ac_status" >&5
   test $ac_status = 0; }; then
-  pkg_cv_IDN_LIBS=`$PKG_CONFIG --libs "libidn" 2>/dev/null`
+  pkg_cv_IDN_LIBS=`$PKG_CONFIG --libs "libidn2" 2>/dev/null`
 		      test "x$?" != "x0" && pkg_failed=yes
 else
   pkg_failed=yes
@@ -25016,9 +25016,9 @@ else
         _pkg_short_errors_supported=no
 fi
         if test $_pkg_short_errors_supported = yes; then
-	        IDN_PKG_ERRORS=`$PKG_CONFIG --short-errors --print-errors --cflags --libs "libidn" 2>&1`
+	        IDN_PKG_ERRORS=`$PKG_CONFIG --short-errors --print-errors --cflags --libs "libidn2" 2>&1`
         else
-	        IDN_PKG_ERRORS=`$PKG_CONFIG --print-errors --cflags --libs "libidn" 2>&1`
+	        IDN_PKG_ERRORS=`$PKG_CONFIG --print-errors --cflags --libs "libidn2" 2>&1`
         fi
 	# Put the nasty error message in config.log where it belongs
 	echo "$IDN_PKG_ERRORS" >&5
diff --git a/src/text/url.c b/src/text/url.c
index 5ba11da..460c345 100644
--- a/src/text/url.c
+++ b/src/text/url.c
@@ -499,7 +499,7 @@ void vlc_UrlClean (vlc_url_t *restrict url)
 }
 
 #if defined (HAVE_IDN)
-# include <idna.h>
+# include <idn2.h>
 #elif defined (_WIN32)
 # include <windows.h>
 # include <vlc_charset.h>
-- 
2.13.0

