diff --git a/configure b/configure
index 17116aa..dedb51e 100755
--- a/configure
+++ b/configure
@@ -26364,19 +26364,19 @@ fi
 
 
 pkg_failed=no
-{ $as_echo "$as_me:${as_lineno-$LINENO}: checking for IDN" >&5
-$as_echo_n "checking for IDN... " >&6; }
+{ $as_echo "$as_me:${as_lineno-$LINENO}: checking for IDN2" >&5
+$as_echo_n "checking for IDN2... " >&6; }
 
 if test -n "$IDN_CFLAGS"; then
     pkg_cv_IDN_CFLAGS="$IDN_CFLAGS"
  elif test -n "$PKG_CONFIG"; then
     if test -n "$PKG_CONFIG" && \
-    { { $as_echo "$as_me:${as_lineno-$LINENO}: \$PKG_CONFIG --exists --print-errors \"libidn\""; } >&5
-  ($PKG_CONFIG --exists --print-errors "libidn") 2>&5
+    { { $as_echo "$as_me:${as_lineno-$LINENO}: \$PKG_CONFIG --exists --print-errors \"libidn2\""; } >&5
+  ($PKG_CONFIG --exists --print-errors "libidn2") 2>&5
   ac_status=$?
   $as_echo "$as_me:${as_lineno-$LINENO}: \$? = $ac_status" >&5
   test $ac_status = 0; }; then
-  pkg_cv_IDN_CFLAGS=`$PKG_CONFIG --cflags "libidn" 2>/dev/null`
+  pkg_cv_IDN_CFLAGS=`$PKG_CONFIG --cflags "libidn2" 2>/dev/null`
 		      test "x$?" != "x0" && pkg_failed=yes
 else
   pkg_failed=yes
@@ -26388,12 +26388,12 @@ if test -n "$IDN_LIBS"; then
     pkg_cv_IDN_LIBS="$IDN_LIBS"
  elif test -n "$PKG_CONFIG"; then
     if test -n "$PKG_CONFIG" && \
-    { { $as_echo "$as_me:${as_lineno-$LINENO}: \$PKG_CONFIG --exists --print-errors \"libidn\""; } >&5
-  ($PKG_CONFIG --exists --print-errors "libidn") 2>&5
+    { { $as_echo "$as_me:${as_lineno-$LINENO}: \$PKG_CONFIG --exists --print-errors \"libidn2\""; } >&5
+  ($PKG_CONFIG --exists --print-errors "libidn2") 2>&5
   ac_status=$?
   $as_echo "$as_me:${as_lineno-$LINENO}: \$? = $ac_status" >&5
   test $ac_status = 0; }; then
-  pkg_cv_IDN_LIBS=`$PKG_CONFIG --libs "libidn" 2>/dev/null`
+  pkg_cv_IDN_LIBS=`$PKG_CONFIG --libs "libidn2" 2>/dev/null`
 		      test "x$?" != "x0" && pkg_failed=yes
 else
   pkg_failed=yes
@@ -26414,9 +26414,9 @@ else
         _pkg_short_errors_supported=no
 fi
         if test $_pkg_short_errors_supported = yes; then
-	        IDN_PKG_ERRORS=`$PKG_CONFIG --short-errors --print-errors --cflags --libs "libidn" 2>&1`
+	        IDN_PKG_ERRORS=`$PKG_CONFIG --short-errors --print-errors --cflags --libs "libidn2" 2>&1`
         else
-	        IDN_PKG_ERRORS=`$PKG_CONFIG --print-errors --cflags --libs "libidn" 2>&1`
+	        IDN_PKG_ERRORS=`$PKG_CONFIG --print-errors --cflags --libs "libidn2" 2>&1`
         fi
 	# Put the nasty error message in config.log where it belongs
 	echo "$IDN_PKG_ERRORS" >&5
diff --git a/src/text/url.c b/src/text/url.c
index 44afaba..3a9ad4e 100644
--- a/src/text/url.c
+++ b/src/text/url.c
@@ -872,7 +872,7 @@ char *vlc_uri_fixup(const char *str)
 }
 
 #if defined (HAVE_IDN)
-# include <idna.h>
+# include <idn2.h>
 #elif defined (_WIN32)
 # include <windows.h>
 # include <vlc_charset.h>
@@ -919,9 +919,6 @@ static char *vlc_idna_to_ascii (const char *idn)
         case IDNA_MALLOC_ERROR:
             errno = ENOMEM;
             return NULL;
-        case IDNA_DLOPEN_ERROR:
-            errno = ENOSYS;
-            return NULL;
         default:
             errno = EINVAL;
             return NULL;
