From 842c220f655ff64483f6f43c6643fd95e774b5de Mon Sep 17 00:00:00 2001
From: Alexander Vorobyev <avorobyev@protonmail.com>
Date: Mon, 1 May 2023 01:38:32 +0200
Subject: Use /usr/bin for ecwolf binary

---
 CMakeLists.txt | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index c64a55f..366d393 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -53,7 +53,7 @@ endif()
 # Multiarch support
 if(UNIX AND NOT APPLE)
 	# We are in the games section
-	set(CMAKE_INSTALL_BINDIR "games")
+	set(CMAKE_INSTALL_BINDIR "bin")
 
 	include(GNUInstallDirs)
 	if(CPACK_PACKAGE_NAME)
-- 
2.40.1

