From 90d8f0f329cb7e495a2df6894ed8d91b79d10483 Mon Sep 17 00:00:00 2001
From: Joshua Strobl <joshua@streambits.io>
Date: Thu, 12 Mar 2020 00:56:38 +0200
Subject: [PATCH 1/1] Revert Nautilus bump

---
 configure.ac            | 2 +-
 src/terminal-nautilus.c | 3 ++-
 2 files changed, 3 insertions(+), 2 deletions(-)

diff --git a/configure.ac b/configure.ac
index b0fc75cc..cf9beff9 100644
--- a/configure.ac
+++ b/configure.ac
@@ -198,7 +198,7 @@ AC_ARG_WITH([nautilus-extension],
   [with_nautilus_extension=yes])
 
 if test "$with_nautilus_extension" = "yes"; then
-  NAUTILUS_REQUIRED=3.28.0
+  NAUTILUS_REQUIRED=3.0.0
 
   PKG_CHECK_MODULES([NAUTILUS],[
     libnautilus-extension >= $NAUTILUS_REQUIRED
diff --git a/src/terminal-nautilus.c b/src/terminal-nautilus.c
index b1d642bd..8e743761 100644
--- a/src/terminal-nautilus.c
+++ b/src/terminal-nautilus.c
@@ -24,7 +24,8 @@
 #include <gio/gio.h>
 #include <gtk/gtk.h>
 
-#include <nautilus-extension.h>
+#include <libnautilus-extension/nautilus-menu-provider.h>
+#include <libnautilus-extension/nautilus-extension-types.h>
 
 #include <errno.h>
 #include <fcntl.h>
-- 
2.25.1

