From 8709588cc5ec11e7b1007e7d67aa4df74faf5c68 Mon Sep 17 00:00:00 2001
From: Debarshi Ray <debarshir@gnome.org>
Date: Mon, 23 Feb 2015 18:54:36 +0100
Subject: [PATCH] build: Don't treat warnings as errors

---
 configure.ac | 12 ++++++------
 1 file changed, 6 insertions(+), 6 deletions(-)

diff --git a/configure.ac b/configure.ac
index f22050d6a2e3..d6d936eb0a96 100644
--- a/configure.ac
+++ b/configure.ac
@@ -225,12 +225,12 @@ CC_CHECK_FLAGS_APPEND([WARN_CFLAGS],[CFLAGS],[ \
   -Wall \
   -Wcast-align \
   -Wendif-labels \
-  "-Werror=format=2 -Werror=format-nonliteral -Werror=format-security" \
-  -Werror=implicit-function-declaration \
-  -Werror=init-self \
-  -Werror=missing-include-dirs \
-  -Werror=missing-prototypes \
-  -Werror=pointer-arith \
+  "-Wformat=2 -Wformat-nonliteral -Wformat-security" \
+  -Wimplicit-function-declaration \
+  -Winit-self \
+  -Wmissing-include-dirs \
+  -Wmissing-prototypes \
+  -Wpointer-arith \
   -Wextra \
   -Wfloat-equal \
   -Wimplicit \
-- 
2.19.1

