diff --git a/CMakeLists.txt b/CMakeLists.txt
index d25cd036e32..395967af5da 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -373,7 +373,13 @@ endif()
 
 if(NOT MSVC)
   # Turn some warnings into errors.
-  wl_add_flag(WL_COMPILE_DIAGNOSTICS "-Werror")
+  if(("${CMAKE_CXX_COMPILER_ID}" STREQUAL "GNU") AND (CMAKE_CXX_COMPILER_VERSION VERSION_GREATER_EQUAL 12.0))
+    # Skip this on g++12.0 and 12.1 because of false-positives from system headers.
+    # TODO(Nordfriese): As soon as a new g++ version fixes the bug, reenable -Werror for the working version.
+    message(WARNING "This compiler is known to cause false-positive warnings.")
+  else()
+    wl_add_flag(WL_COMPILE_DIAGNOSTICS "-Werror")
+  endif()
   if(("${CMAKE_CXX_COMPILER_ID}" STREQUAL "GNU") AND (CMAKE_CXX_COMPILER_VERSION VERSION_LESS 10.0))
     # Silence some false positives on older g++
     wl_add_flag(WL_COMPILE_DIAGNOSTICS "-Wno-error=unused-variable")
diff --git a/src/base/log.cc b/src/base/log.cc
index 70854a138d4..c8af4c947f4 100644
--- a/src/base/log.cc
+++ b/src/base/log.cc
@@ -177,7 +177,7 @@ void do_log(const LogType type, const Time& gametime, const char* const fmt, ...
 	assert(logger != nullptr);
 
 	// message type and timestamp
-	char buffer_prefix[32];
+	char buffer_prefix[256];
 	{
 		uint32_t t = gametime.is_valid() ? gametime.get() : SDL_GetTicks();
 		const uint32_t hours = t / (1000 * 60 * 60);
diff --git a/src/editor/editorinteractive.h b/src/editor/editorinteractive.h
index 3d2399cc5e6..4eea7b3f015 100644
--- a/src/editor/editorinteractive.h
+++ b/src/editor/editorinteractive.h
@@ -84,8 +84,8 @@ class EditorInteractive : public InteractiveBase {
 		EditorDecreaseResourcesTool decrease_resources;
 		EditorSetResourcesTool set_resources;
 		EditorIncreaseResourcesTool increase_resources;
-		EditorSetPortSpaceTool set_port_space;
 		EditorUnsetPortSpaceTool unset_port_space;
+		EditorSetPortSpaceTool set_port_space;
 		EditorSetOriginTool set_origin;
 		EditorResizeTool resize;
 	};
diff --git a/src/io/filesystem/disk_filesystem.cc b/src/io/filesystem/disk_filesystem.cc
index 16543897c43..ea9a10c14a0 100644
--- a/src/io/filesystem/disk_filesystem.cc
+++ b/src/io/filesystem/disk_filesystem.cc
@@ -415,8 +415,13 @@ void* RealFSImpl::load(const std::string& fname, size_t& length) {
 		}
 		static_cast<int8_t*>(data)[size] = 0;
 
-		fclose(file);
-		file = nullptr;
+		try {
+			fclose(file);
+			file = nullptr;
+		} catch (...) {
+			file = nullptr;
+			throw;
+		}
 
 		length = size;
 	} catch (...) {
