diff -up widelands-1.1/cmake/WlFunctions.cmake~ widelands-1.1/cmake/WlFunctions.cmake
--- widelands-1.1/cmake/WlFunctions.cmake~	2022-10-22 13:51:16.000000000 +0200
+++ widelands-1.1/cmake/WlFunctions.cmake	2023-03-28 10:31:21.413196378 +0200
@@ -77,7 +77,9 @@ macro(_common_compile_tasks)
     wl_include_directories(${NAME} ${CMAKE_SOURCE_DIR}/src)
     wl_include_directories(${NAME} ${CMAKE_BINARY_DIR}/src)
     if(MINIZIP_STATIC_LIBRARIES)
-      wl_include_directories(${NAME} ${MINIZIP_INCLUDE_DIRS})
+      if (MINIZIP_INCLUDE_DIRS)
+        wl_include_directories(${NAME} ${MINIZIP_INCLUDE_DIRS})
+      endif()
       target_compile_options(${NAME} PUBLIC ${MINIZIP_CFLAGS})
       if("${CMAKE_CXX_COMPILER_ID}" STREQUAL "Clang")
         include(CheckCXXCompilerFlag)
