From bcd65bdd45cb45efd568ed911b009665b061ac9e Mon Sep 17 00:00:00 2001
From: Erik Johnson <github@ekriirke.net> (plus small adjustments by Thomas Staudinger <Staudi.Kaos@gmail.com>
Date: Sat, 4 Sep 2021 00:34:50 +0200
Subject: [PATCH] Fix strict parameters for gui.move

---
 torbrowser_launcher/__init__.py | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/torbrowser_launcher/__init__.py b/torbrowser_launcher/__init__.py
index bff3e8e..eef6a73 100644
--- a/torbrowser_launcher/__init__.py
+++ b/torbrowser_launcher/__init__.py
@@ -90,8 +90,8 @@ def main():
     desktop = app.desktop()
     window_size = gui.size()
     gui.move(
-        (desktop.width() - window_size.width()) / 2,
-        (desktop.height() - window_size.height()) / 2,
+        (desktop.width() - window_size.width()) // 2,
+        (desktop.height() - window_size.height()) // 2,
     )
     gui.show()
 
