From 496f8443292f731fe98bed703181ff385088f6e4 Mon Sep 17 00:00:00 2001
From: Francisco Boni <boboniboni@gmail.com>
Date: Wed, 24 Nov 2021 22:51:22 -0300
Subject: [PATCH 2/5] meson: fix librt find_library check

---
 meson.build | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/meson.build b/meson.build
index 68fa23c..dd12539 100644
--- a/meson.build
+++ b/meson.build
@@ -22,7 +22,7 @@ polkit_dep = dependency('polkit-gobject-1', required: false)
 systemd_dep = dependency('systemd', required: false)
 thread_dep = dependency('threads')
 
-librt_dep = cc.find_library('z')
+librt_dep = cc.find_library('rt')
 cc.check_header('sched.h', dependencies: librt_dep)
 cc.has_function('sched_setscheduler', dependencies: librt_dep)
 
-- 
2.42.0

