From 34dcaf1036184fbb5299f07f5d9ad1b7485f192a Mon Sep 17 00:00:00 2001
From: Algent Albrahimi <algent@protonmail.com>
Date: Tue, 13 Jun 2023 19:51:03 +0200
Subject: [PATCH] makefile

---
 makefile | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/makefile b/makefile
index eb71bba..d2e9a27 100644
--- a/makefile
+++ b/makefile
@@ -1,7 +1,7 @@
 #--------------------------------
 # jhead makefile for Unix
 #--------------------------------
-PREFIX=$(DESTDIR)/usr/local
+PREFIX=$(DESTDIR)/usr
 BINDIR=$(PREFIX)/bin
 DOCDIR=$(PREFIX)/share/doc/jhead
 MANDIR=$(PREFIX)/share/man/man1
-- 
2.40.1

