From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Evan Maddock <maddock.evan@vivaldi.net>
Date: Sat, 1 Nov 2025 20:58:24 -0400
Subject: [PATCH] libpeas girepository2

Signed-off-by: Evan Maddock <maddock.evan@vivaldi.net>
---
 src/plugin/gir_main.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/plugin/gir_main.c b/src/plugin/gir_main.c
index 2afda08..3bad6ce 100644
--- a/src/plugin/gir_main.c
+++ b/src/plugin/gir_main.c
@@ -4,7 +4,7 @@
  *
  */
 
-# include <girepository.h>
+# include <girepository/girepository.h>
 
 # include "astroid_activatable.h"
 
@@ -14,7 +14,7 @@ int main (int argc, char ** argv) {
   GError *error = NULL;
 
   ctx = g_option_context_new (NULL);
-  g_option_context_add_group (ctx, g_irepository_get_option_group ());
+  g_option_context_add_group (ctx, gi_repository_get_option_group ());
 
   if (!g_option_context_parse (ctx, &argc, &argv, &error)) {
     g_print ("astroid girmain: %s\n", error->message);
