From 1dd6e54488072eb91cf406dc78d67a972a4450c5 Mon Sep 17 00:00:00 2001
From: Gavin Zhao <git@gzgz.dev>
Date: Sat, 3 Jan 2026 15:01:25 -0500
Subject: [PATCH] Respect GNUInstallDirs when installing

---
 CMakeLists.txt | 26 ++++++++++++++------------
 1 file changed, 14 insertions(+), 12 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index c3160475b..200378691 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -1316,8 +1316,11 @@ else()
 endif()
 
 # Set directories for installation of libraries and header files.
-set(LIB_DIR ${CMAKE_INSTALL_PREFIX}/lib)
-set(INCLUDE_DIR ${CMAKE_INSTALL_PREFIX}/include)
+include(GNUInstallDirs)
+set(REL_LIB_DIR ${CMAKE_INSTALL_LIBDIR})
+set(LIB_DIR ${CMAKE_INSTALL_PREFIX}/${REL_LIBDIR})
+set(REL_INCLUDE_DIR ${CMAKE_INSTALL_INCLUDEDIR}/amd-blis)
+set(INCLUDE_DIR ${CMAKE_INSTALL_PREFIX}/${REL_INCLUDE_DIR})
 # Set LDFLAGS to be replaced in pc file.
 set(LDFLAGS_STRING ${LDFLAGS})
 # Add OpenMP flags as required.
@@ -1332,7 +1335,6 @@ if(NOT WIN32)
         @ONLY
     )
 endif()
-include(GNUInstallDirs)
 
 if(BUILD_SHARED_LIBS)
     # Build shared library.
@@ -1355,10 +1357,10 @@ if(BUILD_SHARED_LIBS)
     set_target_properties(libblis-shared PROPERTIES PUBLIC_HEADER "${BLIS_PUBLIC_HEADERS}")
     set_target_properties(libblis-shared PROPERTIES OUTPUT_NAME ${LIBBLIS_SHARED})
     # Install targets for shared.
-    install(TARGETS libblis-shared LIBRARY DESTINATION ${CMAKE_INSTALL_PREFIX}/lib
-            ARCHIVE DESTINATION ${CMAKE_INSTALL_PREFIX}/lib
-            RUNTIME DESTINATION ${CMAKE_INSTALL_PREFIX}/lib
-            PUBLIC_HEADER DESTINATION ${CMAKE_INSTALL_PREFIX}/include)
+    install(TARGETS libblis-shared LIBRARY DESTINATION ${REL_LIB_DIR}
+            ARCHIVE DESTINATION ${REL_LIB_DIR}
+            RUNTIME DESTINATION ${REL_LIB_DIR}
+            PUBLIC_HEADER DESTINATION ${REL_INCLUDE_DIR})
     set(libblis_depends libblis-shared)
 endif()
 if(BUILD_STATIC_LIBS OR NOT BUILD_SHARED_LIBS)
@@ -1375,16 +1377,16 @@ if(BUILD_STATIC_LIBS OR NOT BUILD_SHARED_LIBS)
     set_target_properties(libblis-static PROPERTIES PUBLIC_HEADER "${BLIS_PUBLIC_HEADERS}")
     set_target_properties(libblis-static PROPERTIES OUTPUT_NAME ${LIBBLIS_STATIC})
     # Install targets.
-    install(TARGETS libblis-static LIBRARY DESTINATION ${CMAKE_INSTALL_PREFIX}/lib
-            ARCHIVE DESTINATION ${CMAKE_INSTALL_PREFIX}/lib
-            RUNTIME DESTINATION ${CMAKE_INSTALL_PREFIX}/lib
-            PUBLIC_HEADER DESTINATION ${CMAKE_INSTALL_PREFIX}/include)
+    install(TARGETS libblis-static LIBRARY DESTINATION ${REL_LIB_DIR}
+            ARCHIVE DESTINATION ${REL_LIB_DIR}
+            RUNTIME DESTINATION ${REL_LIB_DIR}
+            PUBLIC_HEADER DESTINATION ${REL_INCLUDE_DIR})
     list(APPEND libblis_depends libblis-static)
 endif()
 
 if(NOT WIN32)
     # Install package-config file.
-    install(FILES ${CMAKE_BINARY_DIR}/${PC_OUT_FILE} DESTINATION ${CMAKE_INSTALL_PREFIX}/lib/pkgconfig)
+    install(FILES ${CMAKE_BINARY_DIR}/${PC_OUT_FILE} DESTINATION ${REL_LIB_DIR}/pkgconfig)
 endif()
 
 # Set libblis to the shared or static libblis depending on the option setting.
-- 
2.52.0

