From 1be4ec829bc59edae962f50789eb6c6cde6731be Mon Sep 17 00:00:00 2001
From: Peter O'Connor <peter@solus-project.com>
Date: Sun, 18 Jun 2017 17:28:21 +1000
Subject: [PATCH] Enable #Solus as the default channel

Signed-off-by: Peter O'Connor <peter@solus-project.com>
---
 src/config/preferences.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/config/preferences.cpp b/src/config/preferences.cpp
index 5b2df0c..b2b8d6a 100644
--- a/src/config/preferences.cpp
+++ b/src/config/preferences.cpp
@@ -69,7 +69,7 @@ Preferences::Preferences()
     serverGroup->addServer(server);
     serverGroup->setIdentityId(mIdentity->id());
     Konversation::ChannelSettings channel;
-    channel.setName(QStringLiteral("#konversation"));
+    channel.setName(QStringLiteral("#Solus"));
     serverGroup->addChannel(channel);
     serverGroup->setExpanded(false);
     mServerGroupHash.insert(0, serverGroup);
--
2.13.1
