From 88e8ecf3cd17f108f1291ea8d8cdaca2deff7dfb Mon Sep 17 00:00:00 2001
From: Hans K <hans@communitycomputing.net>
Date: Wed, 19 Aug 2026 00:20:08 -0400
Subject: [PATCH] Use colorful icons on linux

---
 client/ui/tray_icon.go | 14 +++++++-------
 1 file changed, 7 insertions(+), 7 deletions(-)

diff --git a/client/ui/tray_icon.go b/client/ui/tray_icon.go
index 6c7b85d7..b1e1e29d 100644
--- a/client/ui/tray_icon.go
+++ b/client/ui/tray_icon.go
@@ -100,19 +100,19 @@ func (t *Tray) iconForState() (icon, dark []byte) {
 		}
 		switch {
 		case connecting:
-			return pick(iconConnectingMono, iconConnectingMonoDark)
+			return pick(iconConnecting, iconConnectingDark)
 		case errored:
-			return pick(iconErrorMono, iconErrorMonoDark)
+			return pick(iconError, iconErrorDark)
 		case needsLogin:
-			return pick(iconNeedsLoginMono, iconNeedsLoginMonoDark)
+			return iconNeedsLogin, nil
 		case connected && hasUpdate:
-			return pick(iconUpdateConnectedMono, iconUpdateConnectedMonoDark)
+			return pick(iconUpdateConnected, iconUpdateConnectedDark)
 		case connected:
-			return pick(iconConnectedMono, iconConnectedMonoDark)
+			return pick(iconConnected, iconConnectedDark)
 		case hasUpdate:
-			return pick(iconUpdateDisconnectedMono, iconUpdateDisconnectedMonoDark)
+			return pick(iconUpdateDisconnected, iconUpdateDisconnectedDark)
 		default:
-			return pick(iconDisconnectedMono, iconDisconnectedMonoDark)
+			return iconDisconnected, nil
 		}
 	}
 
-- 
2.55.0

