From 58a9d0c1f99f2d4c7fc51b6dbe563447ec60120f Mon Sep 17 00:00:00 2001
From: Michael Barbour <barbour.michael.0@gmail.com>
Date: Wed, 5 Oct 2022 23:34:49 -0400
Subject: [PATCH] ffmpeg.c: Use the new channel layout API for lavc >=
 59.37.100.

---
 ffmpeg.c | 6 +++++-
 1 file changed, 5 insertions(+), 1 deletion(-)

diff --git a/ffmpeg.c b/ffmpeg.c
index 42c5a66..1fd2060 100644
--- a/ffmpeg.c
+++ b/ffmpeg.c
@@ -274,7 +274,11 @@ struct codec * ffmpeg_codec_init(const char *path, const char *type, const char
 	snprintf((char *) c->type, i, "ffmpeg/%s", codec->name);
 	c->enc = av_get_sample_fmt_name(state->cc->sample_fmt);
 	c->fs = state->cc->sample_rate;
-	c->channels = state->cc->channels;
+	#if LIBAVCODEC_VERSION_INT >= AV_VERSION_INT(59, 37, 100)
+		c->channels = state->cc->ch_layout.nb_channels;
+	#else
+		c->channels = state->cc->channels;
+	#endif
 	switch (state->cc->sample_fmt) {
 	case AV_SAMPLE_FMT_U8:
 	case AV_SAMPLE_FMT_U8P:
