https://github.com/babluboy/bookworm/pull/391

diff --git a/src/utils.vala b/src/utils.vala
index 3416da1d..20154bce 100644
--- a/src/utils.vala
+++ b/src/utils.vala
@@ -690,7 +690,7 @@ namespace BookwormApp.Utils {
     }
 
     public static string decodeHTMLChars (string inputString) {
-        string outputString = Soup.URI.decode (inputString);
+        string outputString = GLib.Uri.unescape_string (inputString);
         return outputString;
     }
 
@@ -700,7 +700,7 @@ namespace BookwormApp.Utils {
     }
 
     public static string removeMarkUp (string inputString) {
-        string outputString = Soup.URI.decode (inputString);
+        string outputString = GLib.Uri.unescape_string (inputString);
         //replace the escape char for space present in HTML converted from PDF
         outputString = outputString.replace ("&#160;", " ").replace ("#160;", " ").replace ("&#160", " ");
         return outputString;
diff --git a/src/window.vala b/src/window.vala
index 18b43a1c..3d08d915 100644
--- a/src/window.vala
+++ b/src/window.vala
@@ -463,7 +463,7 @@ public class BookwormApp.AppWindow {
                 WebKit.NavigationAction aNavAction = aNavDecision.get_navigation_action ();
                 WebKit.URIRequest aURIReq = aNavAction.get_request ();
                 string url_clicked_on_webview = BookwormApp.Utils.decodeHTMLChars (aURIReq.get_uri ().strip ());
-                url_clicked_on_webview = Soup.URI.decode (url_clicked_on_webview);
+                url_clicked_on_webview = GLib.Uri.unescape_string (url_clicked_on_webview);
                 debug ("URL Captured:" + url_clicked_on_webview);
                 //Handle external links (not file://) by opening the default browser i.e. http://, ftp://
                 if (url_clicked_on_webview.index_of ("file://") == -1) {

diff --git a/meson.build b/meson.build
index 3708ed5a..43ee66e6 100644
--- a/meson.build
+++ b/meson.build
@@ -62,7 +62,7 @@ executable(
     dependencies: [
         dependency('gtk+-3.0'),
         dependency('gee-0.8'),
-        dependency('webkit2gtk-4.0'),
+        dependency('webkit2gtk-4.1'),
         dependency('poppler-glib'),
         dependency('libxml-2.0'),
         dependency('granite', version: '>=0.5'),
