diff -ruN bookworm-c7c3643760caea4bd26b1d56ed033a52f6e34124-orig/data/scripts/mobi_lib/compatibility_utils.py bookworm-c7c3643760caea4bd26b1d56ed033a52f6e34124/data/scripts/mobi_lib/compatibility_utils.py
--- bookworm-c7c3643760caea4bd26b1d56ed033a52f6e34124-orig/data/scripts/mobi_lib/compatibility_utils.py	2020-04-12 17:25:33.000000000 +1000
+++ bookworm-c7c3643760caea4bd26b1d56ed033a52f6e34124/data/scripts/mobi_lib/compatibility_utils.py	2020-04-14 10:12:09.796899572 +1000
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 # -*- coding: utf-8 -*-
 # vim:ts=4:sw=4:softtabstop=4:smarttab:expandtab
 
diff -ruN bookworm-c7c3643760caea4bd26b1d56ed033a52f6e34124-orig/data/scripts/mobi_lib/__init__.py bookworm-c7c3643760caea4bd26b1d56ed033a52f6e34124/data/scripts/mobi_lib/__init__.py
--- bookworm-c7c3643760caea4bd26b1d56ed033a52f6e34124-orig/data/scripts/mobi_lib/__init__.py	2020-04-12 17:25:33.000000000 +1000
+++ bookworm-c7c3643760caea4bd26b1d56ed033a52f6e34124/data/scripts/mobi_lib/__init__.py	2020-04-14 10:12:09.800899623 +1000
@@ -1,2 +1,2 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 # vim:fileencoding=UTF-8:ts=4:sw=4:sta:et:sts=4:ai
diff -ruN bookworm-c7c3643760caea4bd26b1d56ed033a52f6e34124-orig/data/scripts/mobi_lib/kindleunpack.py bookworm-c7c3643760caea4bd26b1d56ed033a52f6e34124/data/scripts/mobi_lib/kindleunpack.py
--- bookworm-c7c3643760caea4bd26b1d56ed033a52f6e34124-orig/data/scripts/mobi_lib/kindleunpack.py	2020-04-12 17:25:33.000000000 +1000
+++ bookworm-c7c3643760caea4bd26b1d56ed033a52f6e34124/data/scripts/mobi_lib/kindleunpack.py	2020-04-14 10:12:09.790899496 +1000
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 # -*- coding: utf-8 -*-
 # vim:ts=4:sw=4:softtabstop=4:smarttab:expandtab
 
diff -ruN bookworm-c7c3643760caea4bd26b1d56ed033a52f6e34124-orig/data/scripts/mobi_lib/mobi_cover.py bookworm-c7c3643760caea4bd26b1d56ed033a52f6e34124/data/scripts/mobi_lib/mobi_cover.py
--- bookworm-c7c3643760caea4bd26b1d56ed033a52f6e34124-orig/data/scripts/mobi_lib/mobi_cover.py	2020-04-12 17:25:33.000000000 +1000
+++ bookworm-c7c3643760caea4bd26b1d56ed033a52f6e34124/data/scripts/mobi_lib/mobi_cover.py	2020-04-14 10:12:09.740898859 +1000
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 # -*- coding: utf-8 -*-
 # vim:ts=4:sw=4:softtabstop=4:smarttab:expandtab
 
diff -ruN bookworm-c7c3643760caea4bd26b1d56ed033a52f6e34124-orig/data/scripts/mobi_lib/mobi_dict.py bookworm-c7c3643760caea4bd26b1d56ed033a52f6e34124/data/scripts/mobi_lib/mobi_dict.py
--- bookworm-c7c3643760caea4bd26b1d56ed033a52f6e34124-orig/data/scripts/mobi_lib/mobi_dict.py	2020-04-12 17:25:33.000000000 +1000
+++ bookworm-c7c3643760caea4bd26b1d56ed033a52f6e34124/data/scripts/mobi_lib/mobi_dict.py	2020-04-14 10:12:09.736898808 +1000
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 # -*- coding: utf-8 -*-
 # vim:ts=4:sw=4:softtabstop=4:smarttab:expandtab
 
diff -ruN bookworm-c7c3643760caea4bd26b1d56ed033a52f6e34124-orig/data/scripts/mobi_lib/mobi_header.py bookworm-c7c3643760caea4bd26b1d56ed033a52f6e34124/data/scripts/mobi_lib/mobi_header.py
--- bookworm-c7c3643760caea4bd26b1d56ed033a52f6e34124-orig/data/scripts/mobi_lib/mobi_header.py	2020-04-12 17:25:33.000000000 +1000
+++ bookworm-c7c3643760caea4bd26b1d56ed033a52f6e34124/data/scripts/mobi_lib/mobi_header.py	2020-04-14 10:12:09.763899152 +1000
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 # -*- coding: utf-8 -*-
 # vim:ts=4:sw=4:softtabstop=4:smarttab:expandtab
 
diff -ruN bookworm-c7c3643760caea4bd26b1d56ed033a52f6e34124-orig/data/scripts/mobi_lib/mobi_html.py bookworm-c7c3643760caea4bd26b1d56ed033a52f6e34124/data/scripts/mobi_lib/mobi_html.py
--- bookworm-c7c3643760caea4bd26b1d56ed033a52f6e34124-orig/data/scripts/mobi_lib/mobi_html.py	2020-04-12 17:25:33.000000000 +1000
+++ bookworm-c7c3643760caea4bd26b1d56ed033a52f6e34124/data/scripts/mobi_lib/mobi_html.py	2020-04-14 10:12:09.804899674 +1000
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 # -*- coding: utf-8 -*-
 # vim:ts=4:sw=4:softtabstop=4:smarttab:expandtab
 
diff -ruN bookworm-c7c3643760caea4bd26b1d56ed033a52f6e34124-orig/data/scripts/mobi_lib/mobi_index.py bookworm-c7c3643760caea4bd26b1d56ed033a52f6e34124/data/scripts/mobi_lib/mobi_index.py
--- bookworm-c7c3643760caea4bd26b1d56ed033a52f6e34124-orig/data/scripts/mobi_lib/mobi_index.py	2020-04-12 17:25:33.000000000 +1000
+++ bookworm-c7c3643760caea4bd26b1d56ed033a52f6e34124/data/scripts/mobi_lib/mobi_index.py	2020-04-14 10:12:09.742898884 +1000
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 # -*- coding: utf-8 -*-
 # vim:ts=4:sw=4:softtabstop=4:smarttab:expandtab
 
diff -ruN bookworm-c7c3643760caea4bd26b1d56ed033a52f6e34124-orig/data/scripts/mobi_lib/mobi_k8proc.py bookworm-c7c3643760caea4bd26b1d56ed033a52f6e34124/data/scripts/mobi_lib/mobi_k8proc.py
--- bookworm-c7c3643760caea4bd26b1d56ed033a52f6e34124-orig/data/scripts/mobi_lib/mobi_k8proc.py	2020-04-12 17:25:33.000000000 +1000
+++ bookworm-c7c3643760caea4bd26b1d56ed033a52f6e34124/data/scripts/mobi_lib/mobi_k8proc.py	2020-04-14 10:12:09.769899228 +1000
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 # -*- coding: utf-8 -*-
 # vim:ts=4:sw=4:softtabstop=4:smarttab:expandtab
 
diff -ruN bookworm-c7c3643760caea4bd26b1d56ed033a52f6e34124-orig/data/scripts/mobi_lib/mobi_k8resc.py bookworm-c7c3643760caea4bd26b1d56ed033a52f6e34124/data/scripts/mobi_lib/mobi_k8resc.py
--- bookworm-c7c3643760caea4bd26b1d56ed033a52f6e34124-orig/data/scripts/mobi_lib/mobi_k8resc.py	2020-04-12 17:25:33.000000000 +1000
+++ bookworm-c7c3643760caea4bd26b1d56ed033a52f6e34124/data/scripts/mobi_lib/mobi_k8resc.py	2020-04-14 10:12:09.776899318 +1000
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 # -*- coding: utf-8 -*-
 # vim:ts=4:sw=4:softtabstop=4:smarttab:expandtab
 
diff -ruN bookworm-c7c3643760caea4bd26b1d56ed033a52f6e34124-orig/data/scripts/mobi_lib/mobiml2xhtml.py bookworm-c7c3643760caea4bd26b1d56ed033a52f6e34124/data/scripts/mobi_lib/mobiml2xhtml.py
--- bookworm-c7c3643760caea4bd26b1d56ed033a52f6e34124-orig/data/scripts/mobi_lib/mobiml2xhtml.py	2020-04-12 17:25:33.000000000 +1000
+++ bookworm-c7c3643760caea4bd26b1d56ed033a52f6e34124/data/scripts/mobi_lib/mobiml2xhtml.py	2020-04-14 10:12:51.182426724 +1000
@@ -1,4 +1,4 @@
-#! /usr/bin/python
+#!/usr/bin/env python3
 # vim:ts=4:sw=4:softtabstop=4:smarttab:expandtab
 
 
diff -ruN bookworm-c7c3643760caea4bd26b1d56ed033a52f6e34124-orig/data/scripts/mobi_lib/mobi_nav.py bookworm-c7c3643760caea4bd26b1d56ed033a52f6e34124/data/scripts/mobi_lib/mobi_nav.py
--- bookworm-c7c3643760caea4bd26b1d56ed033a52f6e34124-orig/data/scripts/mobi_lib/mobi_nav.py	2020-04-12 17:25:33.000000000 +1000
+++ bookworm-c7c3643760caea4bd26b1d56ed033a52f6e34124/data/scripts/mobi_lib/mobi_nav.py	2020-04-14 10:12:09.747898948 +1000
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 # -*- coding: utf-8 -*-
 # vim:ts=4:sw=4:softtabstop=4:smarttab:expandtab
 
diff -ruN bookworm-c7c3643760caea4bd26b1d56ed033a52f6e34124-orig/data/scripts/mobi_lib/mobi_ncx.py bookworm-c7c3643760caea4bd26b1d56ed033a52f6e34124/data/scripts/mobi_lib/mobi_ncx.py
--- bookworm-c7c3643760caea4bd26b1d56ed033a52f6e34124-orig/data/scripts/mobi_lib/mobi_ncx.py	2020-04-12 17:25:33.000000000 +1000
+++ bookworm-c7c3643760caea4bd26b1d56ed033a52f6e34124/data/scripts/mobi_lib/mobi_ncx.py	2020-04-14 10:12:09.745898923 +1000
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 # -*- coding: utf-8 -*-
 # vim:ts=4:sw=4:softtabstop=4:smarttab:expandtab
 
diff -ruN bookworm-c7c3643760caea4bd26b1d56ed033a52f6e34124-orig/data/scripts/mobi_lib/mobi_opf.py bookworm-c7c3643760caea4bd26b1d56ed033a52f6e34124/data/scripts/mobi_lib/mobi_opf.py
--- bookworm-c7c3643760caea4bd26b1d56ed033a52f6e34124-orig/data/scripts/mobi_lib/mobi_opf.py	2020-04-12 17:25:33.000000000 +1000
+++ bookworm-c7c3643760caea4bd26b1d56ed033a52f6e34124/data/scripts/mobi_lib/mobi_opf.py	2020-04-14 10:12:09.752899012 +1000
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 # -*- coding: utf-8 -*-
 # vim:ts=4:sw=4:softtabstop=4:smarttab:expandtab
 
diff -ruN bookworm-c7c3643760caea4bd26b1d56ed033a52f6e34124-orig/data/scripts/mobi_lib/mobi_pagemap.py bookworm-c7c3643760caea4bd26b1d56ed033a52f6e34124/data/scripts/mobi_lib/mobi_pagemap.py
--- bookworm-c7c3643760caea4bd26b1d56ed033a52f6e34124-orig/data/scripts/mobi_lib/mobi_pagemap.py	2020-04-12 17:25:33.000000000 +1000
+++ bookworm-c7c3643760caea4bd26b1d56ed033a52f6e34124/data/scripts/mobi_lib/mobi_pagemap.py	2020-04-14 10:12:09.800899623 +1000
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 # -*- coding: utf-8 -*-
 # vim:ts=4:sw=4:softtabstop=4:smarttab:expandtab
 
diff -ruN bookworm-c7c3643760caea4bd26b1d56ed033a52f6e34124-orig/data/scripts/mobi_lib/mobi_sectioner.py bookworm-c7c3643760caea4bd26b1d56ed033a52f6e34124/data/scripts/mobi_lib/mobi_sectioner.py
--- bookworm-c7c3643760caea4bd26b1d56ed033a52f6e34124-orig/data/scripts/mobi_lib/mobi_sectioner.py	2020-04-12 17:25:33.000000000 +1000
+++ bookworm-c7c3643760caea4bd26b1d56ed033a52f6e34124/data/scripts/mobi_lib/mobi_sectioner.py	2020-04-14 10:12:09.792899521 +1000
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 # -*- coding: utf-8 -*-
 # vim:ts=4:sw=4:softtabstop=4:smarttab:expandtab
 
diff -ruN bookworm-c7c3643760caea4bd26b1d56ed033a52f6e34124-orig/data/scripts/mobi_lib/mobi_split.py bookworm-c7c3643760caea4bd26b1d56ed033a52f6e34124/data/scripts/mobi_lib/mobi_split.py
--- bookworm-c7c3643760caea4bd26b1d56ed033a52f6e34124-orig/data/scripts/mobi_lib/mobi_split.py	2020-04-12 17:25:33.000000000 +1000
+++ bookworm-c7c3643760caea4bd26b1d56ed033a52f6e34124/data/scripts/mobi_lib/mobi_split.py	2020-04-14 10:12:09.756899063 +1000
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 # -*- coding: utf-8 -*-
 # vim:ts=4:sw=4:softtabstop=4:smarttab:expandtab
 
diff -ruN bookworm-c7c3643760caea4bd26b1d56ed033a52f6e34124-orig/data/scripts/mobi_lib/mobi_uncompress.py bookworm-c7c3643760caea4bd26b1d56ed033a52f6e34124/data/scripts/mobi_lib/mobi_uncompress.py
--- bookworm-c7c3643760caea4bd26b1d56ed033a52f6e34124-orig/data/scripts/mobi_lib/mobi_uncompress.py	2020-04-12 17:25:33.000000000 +1000
+++ bookworm-c7c3643760caea4bd26b1d56ed033a52f6e34124/data/scripts/mobi_lib/mobi_uncompress.py	2020-04-14 10:12:09.771899254 +1000
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 # -*- coding: utf-8 -*-
 # vim:ts=4:sw=4:softtabstop=4:smarttab:expandtab
 
diff -ruN bookworm-c7c3643760caea4bd26b1d56ed033a52f6e34124-orig/data/scripts/mobi_lib/mobi_utils.py bookworm-c7c3643760caea4bd26b1d56ed033a52f6e34124/data/scripts/mobi_lib/mobi_utils.py
--- bookworm-c7c3643760caea4bd26b1d56ed033a52f6e34124-orig/data/scripts/mobi_lib/mobi_utils.py	2020-04-12 17:25:33.000000000 +1000
+++ bookworm-c7c3643760caea4bd26b1d56ed033a52f6e34124/data/scripts/mobi_lib/mobi_utils.py	2020-04-14 10:12:09.798899598 +1000
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 # -*- coding: utf-8 -*-
 # vim:ts=4:sw=4:softtabstop=4:smarttab:expandtab
 # flake8: noqa
diff -ruN bookworm-c7c3643760caea4bd26b1d56ed033a52f6e34124-orig/data/scripts/mobi_lib/unipath.py bookworm-c7c3643760caea4bd26b1d56ed033a52f6e34124/data/scripts/mobi_lib/unipath.py
--- bookworm-c7c3643760caea4bd26b1d56ed033a52f6e34124-orig/data/scripts/mobi_lib/unipath.py	2020-04-12 17:25:33.000000000 +1000
+++ bookworm-c7c3643760caea4bd26b1d56ed033a52f6e34124/data/scripts/mobi_lib/unipath.py	2020-04-14 10:12:09.737898821 +1000
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 # -*- coding: utf-8 -*-
 # vim:ts=4:sw=4:softtabstop=4:smarttab:expandtab
 
diff -ruN bookworm-c7c3643760caea4bd26b1d56ed033a52f6e34124-orig/data/scripts/mobi_lib/unpack_structure.py bookworm-c7c3643760caea4bd26b1d56ed033a52f6e34124/data/scripts/mobi_lib/unpack_structure.py
--- bookworm-c7c3643760caea4bd26b1d56ed033a52f6e34124-orig/data/scripts/mobi_lib/unpack_structure.py	2020-04-12 17:25:33.000000000 +1000
+++ bookworm-c7c3643760caea4bd26b1d56ed033a52f6e34124/data/scripts/mobi_lib/unpack_structure.py	2020-04-14 10:12:09.733898770 +1000
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 # -*- coding: utf-8 -*-
 # vim:ts=4:sw=4:softtabstop=4:smarttab:expandtab
 
