From 289e4172ea3b686c5cc5af825368df58275a83f2 Mon Sep 17 00:00:00 2001
From: Reilly Brogan <reilly@reillybrogan.com>
Date: Fri, 18 Aug 2023 21:08:14 -0500
Subject: [PATCH] Fix reconfigure error

---
 configure.in | 3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

diff --git a/configure.in b/configure.in
index 54d6107..d9c69e3 100644
--- a/configure.in
+++ b/configure.in
@@ -22,6 +22,7 @@ AM_CONFIG_HEADER(config.h)
 
 dnl Checks for programs.
 AC_PROG_CC
+AC_PROG_CXX
 
 dnl AC_PROG_CXX
 dnl AC_PROG_AWK
@@ -1268,7 +1269,6 @@ if test "$disable_atheos" != yes; then
 			AC_CHECK_LIB(syllable, main)
 		fi
 		if test "$ac_cv_lib_atheos_main" = yes -o "$ac_cv_lib_syllable_main" = yes; then
-			AC_PROG_CXX
 			AC_DEFINE(GRDRV_ATHEOS)
 			drivers="$drivers ATHEOS"
 			cf_have_atheos=yes
@@ -1291,7 +1291,6 @@ if test "$disable_haiku" != yes ; then
 		AC_CHECK_LIB(stdc++, main)
 		AC_CHECK_LIB(be, main)
 		if test "$ac_cv_lib_be_main" = yes; then
-			AC_PROG_CXX
 			AC_DEFINE(GRDRV_HAIKU)
 			drivers="$drivers HAIKU"
 			cf_have_haiku=yes
-- 
2.41.0

