From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Silke Hofstra <shofstra@bauwatch.com>
Date: Mon, 2 Jun 2025 15:23:50 +0200
Subject: [PATCH] Use system gtest

---
 src/tests/unit/meson.build | 6 ++----
 1 file changed, 2 insertions(+), 4 deletions(-)

diff --git a/src/tests/unit/meson.build b/src/tests/unit/meson.build
index cd66917..6593576 100644
--- a/src/tests/unit/meson.build
+++ b/src/tests/unit/meson.build
@@ -9,9 +9,7 @@
 #  unit tests
 #
 
-googletest = subproject('gtest')
-gtest_deps = googletest.get_variable('gtest_main_dep')
-gtest_inc = googletest.get_variable('gtest_incdir')
+gtest_deps = dependency('gtest')
 
 unit_tests = executable(
            'unit-tests',
@@ -36,7 +34,7 @@ unit_tests = executable(
                 '../../netcfg/dns/resolver-settings.cpp',
                 '../../netcfg/dns/settings-manager.cpp',
            ],
-           include_directories: [include_dirs, gtest_inc, '../../..'],
+           include_directories: [include_dirs, '../../..'],
            link_with: [
                  common_code,
                  netcfgmgr_lib,
