diff --git a/eng/linux/Generic/usr/bin/otd b/eng/linux/Generic/usr/bin/otd
index 3afb19a1..e0949451 100755
--- a/eng/linux/Generic/usr/bin/otd
+++ b/eng/linux/Generic/usr/bin/otd
@@ -1,3 +1,3 @@
 #!/usr/bin/env sh
 
-/usr/lib/opentabletdriver/OpenTabletDriver.Console "$@"
+DOTNET_ROOT=/usr/lib/dotnet /usr/lib/opentabletdriver/OpenTabletDriver.Console "$@"
diff --git a/eng/linux/Generic/usr/bin/otd-daemon b/eng/linux/Generic/usr/bin/otd-daemon
index 9c3c7426..6e81d8ad 100755
--- a/eng/linux/Generic/usr/bin/otd-daemon
+++ b/eng/linux/Generic/usr/bin/otd-daemon
@@ -1,3 +1,3 @@
 #!/usr/bin/env sh
 
-/usr/lib/opentabletdriver/OpenTabletDriver.Daemon "$@"
+DOTNET_ROOT=/usr/lib/dotnet /usr/lib/opentabletdriver/OpenTabletDriver.Daemon "$@"
diff --git a/eng/linux/Generic/usr/bin/otd-gui b/eng/linux/Generic/usr/bin/otd-gui
index d10b5d79..3831e7d9 100755
--- a/eng/linux/Generic/usr/bin/otd-gui
+++ b/eng/linux/Generic/usr/bin/otd-gui
@@ -1,3 +1,3 @@
 #!/usr/bin/env sh
 
-/usr/lib/opentabletdriver/OpenTabletDriver.UX.Gtk "$@"
+DOTNET_ROOT=/usr/lib/dotnet /usr/lib/opentabletdriver/OpenTabletDriver.UX.Gtk "$@"
