diff --git a/Directory.Build.props b/Directory.Build.props
index dc09822a..1dc8772d 100644
--- a/Directory.Build.props
+++ b/Directory.Build.props
@@ -1,7 +1,7 @@
 <Project>
   <PropertyGroup>
     <VersionBase>0.6.4.0</VersionBase>
-    <FrameworkBase>net6.0</FrameworkBase>
+    <FrameworkBase>net8.0</FrameworkBase>
     <LangVersion>preview</LangVersion>
     <VersionPrefix>$(VersionBase)</VersionPrefix>
   </PropertyGroup>
diff --git a/OpenTabletDriver.Desktop/Interop/Timer/FallbackTimer.cs b/OpenTabletDriver.Desktop/Interop/Timer/FallbackTimer.cs
index b77ce452..bd3a12e8 100644
--- a/OpenTabletDriver.Desktop/Interop/Timer/FallbackTimer.cs
+++ b/OpenTabletDriver.Desktop/Interop/Timer/FallbackTimer.cs
@@ -5,7 +5,7 @@ using OpenTabletDriver.Plugin.Timers;
 
 namespace OpenTabletDriver.Desktop.Interop.Timer
 {
-    internal class FallbackTimer : ITimer, IDisposable
+    internal class FallbackTimer : OpenTabletDriver.Plugin.Timers.ITimer, IDisposable
     {
         private Thread threadTimer;
         private bool runTimer = true;
diff --git a/OpenTabletDriver/OpenTabletDriver.csproj b/OpenTabletDriver/OpenTabletDriver.csproj
index 56df617f..92fde71f 100644
--- a/OpenTabletDriver/OpenTabletDriver.csproj
+++ b/OpenTabletDriver/OpenTabletDriver.csproj
@@ -14,7 +14,7 @@
 
   <ItemGroup Label="NuGet Packages">
     <PackageReference Include="HidSharpCore" Version="1.2.1.1" />
-    <PackageReference Include="Microsoft.Extensions.DependencyInjection" Version="6.0.0-rc.1.21451.13" />
+    <PackageReference Include="Microsoft.Extensions.DependencyInjection" Version="8.0.0" />
     <PackageReference Include="Newtonsoft.Json" Version="13.0.1" />
   </ItemGroup>
 
diff --git a/eng/lib.sh b/eng/lib.sh
index 2f112e3e..673b135c 100755
--- a/eng/lib.sh
+++ b/eng/lib.sh
@@ -17,13 +17,13 @@ REPO_ROOT="$(readlink -f "${ENG_SCRIPT_ROOT}/../")"
 
 ### Build Requirements
 
-DOTNET_VERSION="6.0"
+DOTNET_VERSION="8.0"
 
 # could do away with declare -g, but did it anyway for all of them for consistency
 # with NET_RUNTIME (a global variable without initial value in lib.sh)
 declare -g OUTPUT="dist"
 declare -g CONFIG="Release"
-declare -g FRAMEWORK="net6.0"
+declare -g FRAMEWORK="net8.0"
 declare -g NET_RUNTIME
 declare -g DOG_FOOD="true"
 declare -g BUILD="true"
