--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -42,13 +42,16 @@ endif()
 # Evaluate 'soname' from OSD version
 
     # replace '_' with '.'
-    string(REGEX REPLACE "(_)" "." OSD_SONAME ${OpenSubdiv_VERSION})
+    string(REGEX REPLACE "(_)" "." OSD_NAME ${OpenSubdiv_VERSION})
 
     # remove starting 'v' character
-    string(REGEX REPLACE "^v" "" OSD_SONAME ${OSD_SONAME})
+    string(REGEX REPLACE "^v" "" OSD_NAME ${OSD_NAME})
+
+    # get soname
+    string(REGEX REPLACE "(.*\\..*)\\..*" "\\1" OSD_SONAME ${OSD_NAME})
 
     add_definitions(
-        -DOPENSUBDIV_VERSION_STRING="${OSD_SONAME}"
+        -DOPENSUBDIV_VERSION_STRING="${OSD_NAME}"
     )
 
 #-------------------------------------------------------------------------------
--- a/opensubdiv/CMakeLists.txt
+++ b/opensubdiv/CMakeLists.txt
@@ -187,6 +187,7 @@ if (NOT NO_LIB)
                 PROPERTIES
                     OUTPUT_NAME osdCPU
                     CLEAN_DIRECT_OUTPUT 1
+                    VERSION ${OSD_NAME}
                     SOVERSION ${OSD_SONAME}
                 )
         else()
@@ -218,6 +219,7 @@ if (NOT NO_LIB)
                     PROPERTIES
                         OUTPUT_NAME osdGPU
                         CLEAN_DIRECT_OUTPUT 1
+                        VERSION ${OSD_NAME}
                         SOVERSION ${OSD_SONAME}
                     )
             else()
