From d41bc029ce59e0c31249c24a0699d3969bc7b0ad Mon Sep 17 00:00:00 2001
From: Algent Albrahimi <algent@protonmail.com>
Date: Sat, 2 May 2020 14:28:59 +0200
Subject: [PATCH] Added hpaio to backend/dll.conf.in

---
 backend/dll.conf.in | 1 +
 1 file changed, 1 insertion(+)

diff --git a/backend/dll.conf.in b/backend/dll.conf.in
index 92091cb..8c84518 100644
--- a/backend/dll.conf.in
+++ b/backend/dll.conf.in
@@ -39,6 +39,7 @@ genesys
 #gphoto2
 gt68xx
 hp
+hpaio
 hp3500
 hp3900
 hp4200
-- 
2.26.2

