From b00fb6f3d7b8ff5c462ce6e95e7a6b9dd379f57f Mon Sep 17 00:00:00 2001
From: Joshua Strobl <joshua@streambits.io>
Date: Wed, 18 Sep 2019 09:35:54 +0300
Subject: [PATCH 1/1] Correct cmake install dir

---
 meson.build | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/meson.build b/meson.build
index 499948a..d81b455 100644
--- a/meson.build
+++ b/meson.build
@@ -22,7 +22,7 @@ bindir = join_paths(prefix, get_option('bindir'))
 libdir = join_paths(prefix, get_option('libdir'))
 datadir = join_paths(prefix, get_option('datadir'))
 
-cmakedatadir = join_paths(datadir, 'cmake')
+cmakedatadir = join_paths(libdir, 'cmake')
 
 autoconfdatadir = join_paths(datadir, 'aclocal')
 pkgdir = join_paths(datadir, package_name)
-- 
2.23.0

