diff --git a/configure.ac b/configure.ac
index a707286..80b6e14 100644
--- a/configure.ac
+++ b/configure.ac
@@ -91,7 +91,7 @@ if test -z "$DBLATEX"; then
 fi
 
 dnl check for DocBook DTD and stylesheets in the local catalog.
-JH_CHECK_XML_CATALOG([-//OASIS//DTD DocBook XML V4.3//EN], [DocBook XML DTD V4.3])
+JH_CHECK_XML_CATALOG([-//OASIS//DTD DocBook XML V4.4//EN], [DocBook XML DTD V4.4])
 JH_CHECK_XML_CATALOG([http://docbook.sourceforge.net/release/xsl/current/html/chunk.xsl], [DocBook XSL Stylesheets])
 
 dnl
