From d8f8e8aac6b8cae7d5242bbe5d4f008438d8a0a1 Mon Sep 17 00:00:00 2001
From: Silke Hofstra <silke@slxh.eu>
Date: Wed, 30 Sep 2020 20:36:23 +0200
Subject: [PATCH] Remove building of static library and examples

---
 CMakeLists.txt     |  1 -
 src/CMakeLists.txt | 18 ------------------
 2 files changed, 19 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index dbd5fa5..f7ccbc9 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -400,7 +400,6 @@ message(STATUS "UDEV_RULES_DIR: ${UDEV_RULES_DIR}")
 message(STATUS "USE_HIDAPI: ${USE_HIDAPI}")
 
 subdirs(src)
-subdirs(examples)
 subdirs(tools)
 subdirs(man)
 
diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
index d213abe..3c0079d 100644
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -65,24 +65,6 @@ list(APPEND COMPAT_SOURCES
 	../openbsd-compat/timingsafe_bcmp.c
 )
 
-# static library
-add_library(fido2 STATIC ${FIDO_SOURCES} ${COMPAT_SOURCES})
-target_link_libraries(fido2 ${CBOR_LIBRARIES} ${CRYPTO_LIBRARIES}
-	${UDEV_LIBRARIES} ${BASE_LIBRARIES} ${HIDAPI_LIBRARIES})
-if(WIN32)
-	if (MINGW)
-		target_link_libraries(fido2 wsock32 ws2_32 bcrypt setupapi hid)
-	else()
-		target_link_libraries(fido2 wsock32 ws2_32 bcrypt SetupAPI hid)
-		set_target_properties(fido2 PROPERTIES OUTPUT_NAME fido2_static)
-	endif()
-elseif(APPLE)
-	target_link_libraries(fido2 "-framework CoreFoundation"
-		"-framework IOKit")
-endif()
-install(TARGETS fido2 ARCHIVE DESTINATION ${CMAKE_INSTALL_LIBDIR}
-	LIBRARY DESTINATION ${CMAKE_INSTALL_LIBDIR})
-
 # dynamic library
 add_library(fido2_shared SHARED ${FIDO_SOURCES} ${COMPAT_SOURCES})
 target_link_libraries(fido2_shared ${CBOR_LIBRARIES} ${CRYPTO_LIBRARIES}
-- 
2.28.0

