--- a/Source/FreeImage/PluginEXR.cpp
+++ b/Source/FreeImage/PluginEXR.cpp
@@ -37,7 +37,8 @@
 #include <OpenEXR/ImfRgba.h>
 #include <OpenEXR/ImfArray.h>
 #include <OpenEXR/ImfPreviewImage.h>
-#include <OpenEXR/half.h>
+#include <Imath/half.h>
+#include <Imath/ImathInt64.h>
 
 
 // ==========================================================
--- a/Source/FreeImage/PluginTIFF.cpp
+++ b/Source/FreeImage/PluginTIFF.cpp
@@ -39,7 +39,7 @@
 #include "Utilities.h"
 #include <tiffio.h>
 #include "../Metadata/FreeImageTag.h"
-#include <OpenEXR/half.h>
+#include <Imath/half.h>
 
 #include "FreeImageIO.h"
 #include "PSDParser.h"
