diff -rupN FreeImage/Wrapper/FreeImagePlus/doc/FreeImagePlus.dox FreeImage-new/Wrapper/FreeImagePlus/doc/FreeImagePlus.dox
--- FreeImage/Wrapper/FreeImagePlus/doc/FreeImagePlus.dox	2015-12-31 23:07:18.000000000 +0100
+++ FreeImage-new/Wrapper/FreeImagePlus/doc/FreeImagePlus.dox	2018-07-31 23:35:53.584960114 +0200
@@ -753,7 +753,7 @@ WARN_FORMAT            = "$file:$line: $
 # messages should be written. If left blank the output is written to standard
 # error (stderr).
 
-WARN_LOGFILE           = .\doxygen.log
+WARN_LOGFILE           = ./doxygen.log
 
 #---------------------------------------------------------------------------
 # Configuration options related to the input files
