From c8d95866ea0d97863a90235fc271d4a771f77fb4 Mon Sep 17 00:00:00 2001
From: Joey Riches <josephriches@gmail.com>
Date: Sun, 31 Mar 2024 23:32:43 +0100
Subject: [PATCH 1/1] src/iks.c: Retain py2 piksemel behaviour

---
 src/iks.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/iks.c b/src/iks.c
index d1a1a93..2ec1d9c 100644
--- a/src/iks.c
+++ b/src/iks.c
@@ -698,9 +698,9 @@ iks_string (ikstack *s, iks *x)
 					*t++ = ' ';
 					t = my_strcat (t, IKS_ATTRIB_NAME (y), 0);
 					*t++ = '=';
-					*t++ = '\'';
+					*t++ = '"';
 					t = escape (t, IKS_ATTRIB_VALUE (y), strlen (IKS_ATTRIB_VALUE (y)));
-					*t++ = '\'';
+					*t++ = '"';
 					y = y->next;
 				}
 				if (IKS_TAG_CHILDREN (x)) {
-- 
2.44.0

