From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Reilly Brogan <reilly@reillybrogan.com>
Date: Tue, 23 Sep 2025 12:32:34 -0500
Subject: [PATCH] unbundle sqlite

---
 Cargo.lock | 1 -
 Cargo.toml | 2 +-
 2 files changed, 1 insertion(+), 2 deletions(-)

diff --git a/Cargo.lock b/Cargo.lock
index 14429784f..658cb5f8d 100644
--- a/Cargo.lock
+++ b/Cargo.lock
@@ -3047,7 +3047,6 @@ version = "0.38.1"
 source = "registry+https://github.com/rust-lang/crates.io-index"
 checksum = "f6c19a05435c21ac299d71b6a9c13db3e3f47c520517d58990a462a1397a61db"
 dependencies = [
- "cc",
  "pkg-config",
  "vcpkg",
 ]
diff --git a/Cargo.toml b/Cargo.toml
index b38a96462..9ff25f641 100644
--- a/Cargo.toml
+++ b/Cargo.toml
@@ -88,7 +88,7 @@ proptest = "1.11.0"
 pulldown-cmark = { version = "0.13.3", default-features = false, features = ["html"] }
 rand = "0.10.1"
 regex = "1.12.3"
-rusqlite = { version = "0.40.0", features = ["bundled"] }
+rusqlite = { version = "0.40.0", features = [] }
 rustc-hash = "2.1.2"
 rustc-stable-hash = "0.1.2"
 rustfix = { version = "0.9.7", path = "crates/rustfix" }
