From 8a67fae170f59c050577c7c82bc4844866c3371c Mon Sep 17 00:00:00 2001
From: Fabio Forni <livingsilver94.solus@redaril.me>
Date: Wed, 5 Jun 2019 23:09:27 +0200
Subject: [PATCH] Set correct libdir in bootstrap

---
 src/bootstrap/builder.rs | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/bootstrap/builder.rs b/src/bootstrap/builder.rs
index 5224663..631a406 100644
--- a/src/bootstrap/builder.rs
+++ b/src/bootstrap/builder.rs
@@ -604,7 +604,7 @@ impl<'a> Builder<'a> {
                 let lib = if compiler.stage >= 1 && config.libdir_relative().is_some() {
                     builder.build.config.libdir_relative().unwrap()
                 } else {
-                    Path::new("lib")
+                    Path::new("lib64")
                 };
                 let sysroot = builder
                     .sysroot(self.compiler)
-- 
2.21.0

