From 99cc7efd4451f4f3468812199ba40366024bed4e Mon Sep 17 00:00:00 2001
From: Fabio Forni <livingsilver94.solus@redaril.me>
Date: Thu, 7 Nov 2019 19:15:05 +0100
Subject: [PATCH] Set-correct-libdir-in-bootstrap

---
 src/bootstrap/builder.rs | 3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

diff --git a/src/bootstrap/builder.rs b/src/bootstrap/builder.rs
index 5d586f0..e6428f2 100644
--- a/src/bootstrap/builder.rs
+++ b/src/bootstrap/builder.rs
@@ -615,10 +615,9 @@ impl<'a> Builder<'a> {
             }
 
             fn run(self, builder: &Builder<'_>) -> Interned<PathBuf> {
-                let lib = builder.sysroot_libdir_relative(self.compiler);
                 let sysroot = builder
                     .sysroot(self.compiler)
-                    .join(lib)
+                    .join("lib64")
                     .join("rustlib")
                     .join(self.target)
                     .join("lib");
-- 
2.23.0

