From 8e6c3d3dc45f2df4a1eade634ea94192651ee96c Mon Sep 17 00:00:00 2001
From: Joey Riches <josephriches@gmail.com>
Date: Tue, 18 Feb 2025 13:15:36 +0000
Subject: [PATCH 1/1] use python3 in env

---
 skimage/_build_utils/cythoner.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/skimage/_build_utils/cythoner.py b/skimage/_build_utils/cythoner.py
index 83bc34a48..117c55ee9 100755
--- a/skimage/_build_utils/cythoner.py
+++ b/skimage/_build_utils/cythoner.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 """Scipy variant of Cython command
 
 Cython, as applied to single pyx file.
-- 
2.47.2

