From 1bcfd8f56d7c568774a624eefe77daeeb2319562 Mon Sep 17 00:00:00 2001
From: Joshua Strobl <joshua@streambits.io>
Date: Sat, 27 Mar 2021 13:46:11 +0200
Subject: [PATCH 1/5] Revert libhandy requirement in meson.build

---
 meson.build | 1 -
 1 file changed, 1 deletion(-)

diff --git a/meson.build b/meson.build
index 6a0eff32e..d9d43a7e2 100644
--- a/meson.build
+++ b/meson.build
@@ -120,7 +120,6 @@ gmodule = dependency('gmodule-no-export-2.0', version: glib_ver)
 gnome_autoar = dependency('gnome-autoar-0', version: '>= 0.3.0')
 gnome_desktop = dependency('gnome-desktop-3.0', version: '>= 3.0.0')
 gtk = dependency('gtk+-3.0', version: '>= 3.22.27')
-libhandy = dependency('libhandy-1', version: '>= 1.1.90')
 libportal = []
 if get_option('libportal')
   libportal = dependency('libportal', version: '>= 0.3')
-- 
2.30.2

