From 65ffd25f24aaf552f46d817b83f363da89e9ada3 Mon Sep 17 00:00:00 2001
From: Stefan Ric <stfric369@gmail.com>
Date: Tue, 23 Aug 2016 00:14:44 +0200
Subject: [PATCH] Make icon chooser dialog modal

---
 src/nautilus-properties-window.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/src/nautilus-properties-window.c b/src/nautilus-properties-window.c
index 1214752..1bfb788 100644
--- a/src/nautilus-properties-window.c
+++ b/src/nautilus-properties-window.c
@@ -5185,6 +5185,7 @@ select_image_button_callback (GtkWidget *widget,
 						      g_get_user_special_dir (G_USER_DIRECTORY_PICTURES),
 						      NULL);
 		gtk_window_set_destroy_with_parent (GTK_WINDOW (dialog), TRUE);
+		gtk_window_set_modal (GTK_WINDOW (dialog), TRUE);
 
 		filter = gtk_file_filter_new ();
 		gtk_file_filter_add_pixbuf_formats (filter);
-- 
2.9.2

