From de2d80327f60245d8a03c20b3b5d1025c5941272 Mon Sep 17 00:00:00 2001
From: Joshua Strobl <joshua@streambits.io>
Date: Fri, 17 Sep 2021 22:19:10 +0300
Subject: [PATCH 2/9] Revert libhandy requirement in meson.build

---
 meson.build | 1 -
 1 file changed, 1 deletion(-)

diff --git a/meson.build b/meson.build
index 8cd9c287d..4106bf299 100644
--- a/meson.build
+++ b/meson.build
@@ -120,7 +120,6 @@ gmodule = dependency('gmodule-no-export-2.0', version: glib_ver)
 gnome_autoar = dependency('gnome-autoar-0', version: '>= 0.4.0')
 gnome_desktop = dependency('gnome-desktop-3.0', version: '>= 3.0.0')
 gtk = dependency('gtk+-3.0', version: '>= 3.22.27')
-libhandy = dependency('libhandy-1', version: '>= 1.1.90')
 libportal = []
 if get_option('libportal')
   libportal = dependency('libportal', version: '>= 0.3')
-- 
2.32.0

