From 2ed31bb5da306b807b8e6491881ec011326f7bc5 Mon Sep 17 00:00:00 2001
From: Corey Berla <corey@berla.me>
Date: Fri, 21 Oct 2022 17:19:24 -0700
Subject: [PATCH] list-view: Don't free owned reference

---
 src/nautilus-list-view.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/nautilus-list-view.c b/src/nautilus-list-view.c
index 7960d6ad8..18f860bb8 100644
--- a/src/nautilus-list-view.c
+++ b/src/nautilus-list-view.c
@@ -894,7 +894,7 @@ on_after_sorter_changed (GtkSorter       *sorter,
     g_autoptr (GVariant) state = NULL;
     const gchar *new_sort_text;
     gboolean reversed;
-    g_autofree gchar *current_sort_text = NULL;
+    const gchar *current_sort_text;
 
     if (!self->column_header_was_clicked || self->clicked_column_attribute_q == 0)
     {
-- 
GitLab

