From d42a49b0a3aff39334ba36dc5eae5c39d126ffc1 Mon Sep 17 00:00:00 2001
From: Silke Hofstra <silke@slxh.eu>
Date: Wed, 23 Dec 2020 21:43:05 +0100
Subject: [PATCH] Reduce installed docs

---
 CMakeLists.txt | 6 ------
 1 file changed, 6 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 33220739..240f49e4 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -403,12 +403,6 @@ set(uncrustify_headers
 )
 
 set(uncrustify_docs
-  "${PROJECT_SOURCE_DIR}/AUTHORS"
-  "${PROJECT_SOURCE_DIR}/BUGS"
-  "${PROJECT_SOURCE_DIR}/ChangeLog"
-  "${PROJECT_SOURCE_DIR}/COPYING"
-  "${PROJECT_SOURCE_DIR}/HELP"
-  "${PROJECT_SOURCE_DIR}/README.md"
 )
 
 add_executable(uncrustify ${uncrustify_sources} ${uncrustify_headers})
-- 
2.29.2

