diff -ur a/build.sh b/build.sh
--- a/build.sh	2018-03-03 17:08:49.551347244 +0800
+++ b/build.sh	2018-03-03 17:15:39.556345680 +0800
@@ -546,6 +546,16 @@
   KLAYOUT_VERSION_REV="$KLAYOUT_VERSION_REV"
 )
 
+qmake_options+=(
+  QMAKE_CFLAGS_RELEASE="${CFLAGS}"
+  QMAKE_CXXFLAGS_RELEASE="${CXXFLAGS}"
+  QMAKE_LFLAGS="${LDFLAGS}"
+  QMAKE_LRELEASE=/usr/bin/lrelease-qt4
+  QMAKE_MOC=/usr/bin/moc-qt4
+  QMAKE_RCC=/usr/bin/rcc-qt4
+  QMAKE_UIC=/usr/bin/uic-qt4
+)
+
 if [ $BUILD_EXPERT = 1 ]; then
   qmake_options+=(
     QMAKE_AR="$AR cqs"
