From 3e02fe865bd8451b104296c8a779b157d5ac6ff9 Mon Sep 17 00:00:00 2001
From: Fabio Forni <livingsilver94.solus@redaril.me>
Date: Thu, 15 Aug 2019 11:31:09 +0200
Subject: [PATCH] Remove setup_requires deps

---
 setup.py | 5 -----
 1 file changed, 5 deletions(-)

diff --git a/setup.py b/setup.py
index 64d00b2..1aaaf70 100644
--- a/setup.py
+++ b/setup.py
@@ -13,11 +13,6 @@ setup(
     vcversioner={},
 
     packages=find_packages(),
-    setup_requires=[
-        'pytest-runner',
-        'pytest-asyncio; python_version > "3.4"',
-        'vcversioner'
-    ],
     tests_require=['twisted', 'futures; python_version < "3.0"'],
     include_package_data=True,
 
-- 
2.22.0

