From: Clive Johnston <clivejo@protonmail.com>
Date: Tue, 05 Jun 2018 21:58:17 +0100
Subject: Remove vcversioner, creates errors with solbuild

---
 setup.py | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/setup.py b/setup.py
index a2a0827..e2c7299 100644
--- a/setup.py
+++ b/setup.py
@@ -10,10 +10,10 @@ with open(README_rst, 'r') as f:
 
 setup(
     name="pyee",
-    vcversioner={},
-
+    #vcversioner={},
+    version="3.0.3",
     packages=find_packages(),
-    setup_requires=['pytest-runner', 'vcversioner'],
+    setup_requires=['pytest-runner'],
     tests_require=['twisted'],
     include_package_data=True,
 