diff -up comm-esr31/mail/app/profile/all-thunderbird.js.addons comm-esr31/mail/app/profile/all-thunderbird.js
--- comm-esr31/mail/app/profile/all-thunderbird.js.addons	2014-07-18 02:04:05.000000000 +0200
+++ comm-esr31/mail/app/profile/all-thunderbird.js	2014-07-29 15:29:11.949728141 +0200
@@ -174,7 +174,7 @@ pref("extensions.update.autoUpdateDefaul
 // Disable add-ons installed into the shared user and shared system areas by
 // default. This does not include the application directory. See the SCOPE
 // constants in AddonManager.jsm for values to use here
-pref("extensions.autoDisableScopes", 15);
+pref("extensions.autoDisableScopes", 0);
 
 // Preferences for AMO integration
 pref("extensions.getAddons.cache.enabled", true);
diff -up comm-esr31/mozilla/toolkit/mozapps/extensions/internal/XPIProvider.jsm.addons comm-esr31/mozilla/toolkit/mozapps/extensions/internal/XPIProvider.jsm
--- comm-esr31/mozilla/toolkit/mozapps/extensions/internal/XPIProvider.jsm.addons	2014-07-29 15:29:11.951728146 +0200
+++ comm-esr31/mozilla/toolkit/mozapps/extensions/internal/XPIProvider.jsm	2014-07-29 15:30:51.879030210 +0200
@@ -2102,11 +2102,7 @@ this.XPIProvider = {
 
       AddonManagerPrivate.markProviderSafe(this);
 
-      if (aAppChanged === undefined) {
-        // For new profiles we will never need to show the add-on selection UI
-        Services.prefs.setBoolPref(PREF_SHOWN_SELECTION_UI, true);
-      }
-      else if (aAppChanged && !this.allAppGlobal &&
+      if (aAppChanged && !this.allAppGlobal &&
                Preferences.get(PREF_EM_SHOW_MISMATCH_UI, true)) {
         if (!Preferences.get(PREF_SHOWN_SELECTION_UI, false)) {
           // Flip a flag to indicate that we interrupted startup with an interactive prompt
@@ -2126,6 +2122,9 @@ this.XPIProvider = {
             flushCaches = true;
           }
         }
+      } else {
+        // For new profiles we will never need to show the add-on selection UI
+        Services.prefs.setBoolPref(PREF_SHOWN_SELECTION_UI, true);
       }
 
       if (flushCaches) {
