From c27ab1eb76bd2e5ab9fcc9a26f6c67ca8e02db17 Mon Sep 17 00:00:00 2001
From: Joe Mou <dev@mou.fo>
Date: Thu, 18 Aug 2022 05:47:20 +0000
Subject: [PATCH] Fix set_url_scheme_handler_gnome3 to not associate text/html
 handler

---
 scripts/xdg-settings.in | 1 -
 1 file changed, 1 deletion(-)

diff --git a/scripts/xdg-settings.in b/scripts/xdg-settings.in
index fed5932..ce99019 100755
--- a/scripts/xdg-settings.in
+++ b/scripts/xdg-settings.in
@@ -706,7 +706,6 @@ set_url_scheme_handler_gnome3()
 {
     binary="`desktop_file_to_binary "$2"`"
     [ "$binary" ] || exit_failure_file_missing
-    set_browser_mime "$2" || return
 
     # Set the default browser.
     set_browser_mime "$2" "x-scheme-handler/$1" || return
-- 
GitLab
