From e478015ae544d6aceac063dca0fb6d0c71fb9ca0 Mon Sep 17 00:00:00 2001
From: Ikey Doherty <ikey@solus-project.com>
Date: Tue, 8 Mar 2016 21:41:47 +0000
Subject: [PATCH] Include <cmath> for isnan

Signed-off-by: Ikey Doherty <ikey@solus-project.com>
---
 src/platform/qt/Window.cpp | 6 +++++-
 1 file changed, 5 insertions(+), 1 deletion(-)

diff --git a/src/platform/qt/Window.cpp b/src/platform/qt/Window.cpp
index 210c09d..e1c3ba2 100644
--- a/src/platform/qt/Window.cpp
+++ b/src/platform/qt/Window.cpp
@@ -3,6 +3,10 @@
  * This Source Code Form is subject to the terms of the Mozilla Public
  * License, v. 2.0. If a copy of the MPL was not distributed with this
  * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+#include <cmath>
+#include <cfloat>
+
+
 #include "Window.h"
 
 #include <QDesktopWidget>
@@ -45,7 +49,7 @@ extern "C" {
 
 using namespace QGBA;
 
-#if defined(__WIN32) || defined(__OpenBSD__)
+#if defined(__WIN32) || defined(__OpenBSD__) || defined(__linux__)
 // This is a macro everywhere except MinGW and OpenBSD, it seems
 using std::isnan;
 #endif
-- 
2.7.1

