From a5cce39104ac99d24f449cbcacedc1afcb06e519 Mon Sep 17 00:00:00 2001
From: Ikey Doherty <ikey@evolve-os.com>
Date: Tue, 21 Apr 2015 06:27:19 +0100
Subject: [PATCH] Disable highly buggy fallback flash

---
 src/screenshot-utils.c | 2 --
 1 file changed, 2 deletions(-)

diff --git a/src/screenshot-utils.c b/src/screenshot-utils.c
index 26dd84d..5aafdd1 100644
--- a/src/screenshot-utils.c
+++ b/src/screenshot-utils.c
@@ -582,8 +582,6 @@ screenshot_fallback_get_pixbuf (GdkRectangle *rectangle)
         }
     }
 
-  screenshot_fallback_fire_flash (window, rectangle);
-
   return screenshot;
 }
 
-- 
2.3.2

