--- src/ui/fma-config-tool.desktop.in.old	2017-02-16 00:34:22.000000000 +0200
+++ src/ui/fma-config-tool.desktop.in	2017-11-29 13:21:39.893737689 +0200
@@ -4,6 +4,6 @@
 _Name=FileManager-Actions Configuration Tool
 _Comment=Configure items to be added to the File-Manager context menus
 Exec=fma-config-tool
-Icon=fma-config-tool
+Icon=filemanager-actions
 Categories=System;FileTools;FileManager;
 StartupNotify=true
