--- filemanager-actions-3.4/src/core/fma-icontext.c.orig	2017-02-15 23:43:42.000000000 +0100
+++ filemanager-actions-3.4/src/core/fma-icontext.c	2017-07-29 19:48:37.417268836 +0200
@@ -31,7 +31,7 @@
 #include <config.h>
 #endif

-#ifdef HAVE_GDBUS
+#if 1
 #include <gio/gio.h>
 #else
 # ifdef HAVE_DBUS_GLIB
@@ -45,8 +45,6 @@
 #include <glibtop/proclist.h>
 #include <glibtop/procstate.h>

-#include <libnautilus-extension/nautilus-file-info.h>
-
 #include <api/fma-core-utils.h>
 #include <api/fma-object-api.h>

--- filemanager-actions-3.4/src/core/fma-object-profile.c.orig	2017-02-15 23:34:22.000000000 +0100
+++ filemanager-actions-3.4/src/core/fma-object-profile.c	2017-07-29 19:49:36.023934833 +0200
@@ -34,8 +34,6 @@
 #include <glib/gi18n.h>
 #include <string.h>

-#include <libnautilus-extension/nautilus-file-info.h>
-
 #include <api/fma-core-utils.h>
 #include <api/fma-iio-provider.h>
 #include <api/fma-ifactory-object.h>
