diff --git a/daemon/opensnitchd.service b/daemon/opensnitchd.service
index 3f05fad2..dd22f339 100644
--- a/daemon/opensnitchd.service
+++ b/daemon/opensnitchd.service
@@ -5,8 +5,8 @@ Documentation=https://github.com/evilsocket/opensnitch/wiki
 [Service]
 Type=simple
 PermissionsStartOnly=true
-ExecStartPre=/bin/mkdir -p /etc/opensnitchd/rules
-ExecStart=/usr/local/bin/opensnitchd -rules-path /etc/opensnitchd/rules
+ExecStartPre=/usr/bin/mkdir -p /etc/opensnitchd/rules
+ExecStart=/usr/bin/opensnitchd -rules-path /etc/opensnitchd/rules
 Restart=always
 RestartSec=30
 TimeoutStopSec=10

