diff --git a/daemon/opensnitchd.service b/daemon/opensnitchd.service
--- dev/null
+++ b/daemon/opensnitchd.service
@@ -6,7 +6,7 @@ Documentation=https://github.com/evilsocket/opensnitch/wiki
 Type=simple
 PermissionsStartOnly=true
 ExecStartPre=/bin/mkdir -p /etc/opensnitchd/rules
-ExecStart=/usr/local/bin/opensnitchd -rules-path /etc/opensnitchd/rules
+ExecStart=/usr/bin/opensnitchd -rules-path /etc/opensnitchd/rules
 Restart=always
 RestartSec=30
 TimeoutStopSec=10
