--- /dev/null
+++ b/ui/bin/opensnitch-ui	2025-01-12 15:12:52.917989459 -0600
@@ -172,13 +172,13 @@
 
         if args.socket == None:
             # default
-            args.socket = "unix:///tmp/osui.sock"
+            args.socket = "unix:///run/opensnitch/osui.sock"
 
             addr = cfg.getSettings(Config.DEFAULT_SERVER_ADDR)
             if addr != None and addr != "":
                 if addr.startswith("unix://"):
                     if not os.path.exists(os.path.dirname(addr[7:])):
-                        print("WARNING: unix socket path does not exist, using unix:///tmp/osui.sock, ", addr)
+                        print("WARNING: unix socket path does not exist, using unix:///run/opensnitch/osui.sock, ", addr)
                     else:
                         args.socket = addr
                 else:
