diff --git a/daemon/firewall/config/config.go b/daemon/firewall/config/config.go
diff --git a/ui/bin/opensnitch-ui b/ui/bin/opensnitch-ui
index d71de7e5..11c65afe 100755
--- a/ui/bin/opensnitch-ui
+++ b/ui/bin/opensnitch-ui
@@ -172,13 +172,13 @@ Examples:
 
         if args.socket == None:
             # default
-            args.socket = "unix:///tmp/osui.sock"
+            args.socket = "unix:///run/opensnitch/osui.sock"
 
             addr = cfg.getSettings(Config.DEFAULT_SERVER_ADDR)
             if addr != None and addr != "":
                 if addr.startswith("unix://"):
                     if not os.path.exists(os.path.dirname(addr[7:])):
-                        print("WARNING: unix socket path does not exist, using unix:///tmp/osui.sock, ", addr)
+                        print("WARNING: unix socket path does not exist, using unix:///run/opensnitch/osui.sock, ", addr)
                     else:
                         args.socket = addr
                 else:
diff --git a/daemon/default-config.json b/daemon/default-config.json
index 4c461a77..9d6d014e 100644
--- a/daemon/default-config.json
+++ b/daemon/default-config.json
@@ -1,7 +1,7 @@
 {
     "Server":
     {
-        "Address":"unix:///tmp/osui.sock",
+        "Address":"unix:///run/opensnitch/osui.sock",
         "LogFile":"/var/log/opensnitchd.log"
     },
     "DefaultAction": "allow",
