From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Reilly Brogan <reilly@reillybrogan.com>
Date: Wed, 26 Jun 2024 22:57:56 -0500
Subject: [PATCH] Fix compat with MINIUPNPC API Version 18

---
 source/network/NetServer.cpp | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/source/network/NetServer.cpp b/source/network/NetServer.cpp
index 63868526e9..12fb899e6c 100644
--- a/source/network/NetServer.cpp
+++ b/source/network/NetServer.cpp
@@ -252,7 +252,11 @@ void CNetServerWorker::SetupUPnP()
 	else if ((devlist = upnpDiscover(10000, 0, 0, 0, 0, 0)) != NULL)
 #endif
 	{
+#if defined(MINIUPNPC_API_VERSION) && MINIUPNPC_API_VERSION >= 18
+		ret = UPNP_GetValidIGD(devlist, &urls, &data, internalIPAddress, sizeof(internalIPAddress), nullptr, 0);
+#else
 		ret = UPNP_GetValidIGD(devlist, &urls, &data, internalIPAddress, sizeof(internalIPAddress));
+#endif
 		allocatedUrls = ret != 0; // urls is allocated on non-zero return values
 	}
 	else
