From 20bcf056a66d697fc6a46564d760937d3069f790 Mon Sep 17 00:00:00 2001
From: Alexander Vorobyev <avorobyev@protonmail.com>
Date: Tue, 16 Mar 2021 12:29:49 +0300
Subject: Remove rpath

---
 build/premake/premake5.lua | 18 ------------------
 1 file changed, 18 deletions(-)

diff --git a/build/premake/premake5.lua b/build/premake/premake5.lua
index 36836c1938..2c8c9c04a9 100644
--- a/build/premake/premake5.lua
+++ b/build/premake/premake5.lua
@@ -367,24 +367,6 @@ function project_set_build_flags()
 			if _OPTIONS["prefer-local-libs"] then
 				libdirs { "/usr/local/lib" }
 			end
-
-			-- To use our local shared libraries, they need to be found in the
-			-- runtime dynamic linker path. Add their path to -rpath.
-			if _OPTIONS["libdir"] then
-				linkoptions {"-Wl,-rpath," .. _OPTIONS["libdir"] }
-			else
-				-- On FreeBSD we need to allow use of $ORIGIN
-				if os.istarget("bsd") then
-					linkoptions { "-Wl,-z,origin" }
-				end
-
-				-- Adding the executable path and taking care of correct escaping
-				if _ACTION == "gmake" then
-					linkoptions { "-Wl,-rpath,'$$ORIGIN'" }
-				elseif _ACTION == "codeblocks" then
-					linkoptions { "-Wl,-R\\\\$$$ORIGIN" }
-				end
-			end
 		end
 
 	end
-- 
2.30.2

