From 9bfba0af5bb4b6bb3c44de719eb47f6e80877c00 Mon Sep 17 00:00:00 2001
From: Alexander Vorobyev <avorobyev@protonmail.com>
Date: Tue, 16 Mar 2021 12:05:33 +0300
Subject: Remove check for minor mozjs version

---
 source/scriptinterface/ScriptTypes.h | 13 -------------
 1 file changed, 13 deletions(-)

diff --git a/source/scriptinterface/ScriptTypes.h b/source/scriptinterface/ScriptTypes.h
index 8de1a8ee89..e2f6c75f0f 100644
--- a/source/scriptinterface/ScriptTypes.h
+++ b/source/scriptinterface/ScriptTypes.h
@@ -81,19 +81,6 @@ system-installed version. Make sure you have got all the right files and \
 include paths.
 #endif
 
-#if MOZJS_MINOR_VERSION != 6
-#error Your compiler is trying to use an untested minor version of the \
-SpiderMonkey library. If you are a package maintainer, please make sure \
-to check very carefully that this version does not change the behaviour \
-of the code executed by SpiderMonkey. Different parts of the game (e.g. \
-the multiplayer mode) rely on deterministic behaviour of the JavaScript \
-engine. A simple way for testing this would be playing a network game \
-with one player using the old version and one player using the new \
-version. Another way for testing is running replays and comparing the \
-final hash (check trac.wildfiregames.com/wiki/Debugging#Replaymode). \
-For more information check this link: trac.wildfiregames.com/wiki/Debugging#Outofsync
-#endif
-
 class ScriptInterface;
 
 #endif // INCLUDED_SCRIPTTYPES
-- 
2.30.2

