From 305daa0b0828704d1b7878ad293a6345bb557034 Mon Sep 17 00:00:00 2001
From: Reilly Brogan <reilly@reillybrogan.com>
Date: Sun, 17 Dec 2023 20:40:39 -0600
Subject: [PATCH] Solus: Tmpfiles additions for statelessness

Signed-off-by: Reilly Brogan <reilly@reillybrogan.com>
---
 data/tmpfiles.d/flatpak.conf | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/data/tmpfiles.d/flatpak.conf b/data/tmpfiles.d/flatpak.conf
index 18081e7e..e8ed13ec 100644
--- a/data/tmpfiles.d/flatpak.conf
+++ b/data/tmpfiles.d/flatpak.conf
@@ -1,2 +1,7 @@
 # This is a systemd tmpfiles.d configuration file
 R! /var/tmp/flatpak-cache-*
+
+# Solus additions for statelessness
+d /var/lib/flatpak/exports/share 0755 root root -
+d /var/lib/flatpak/exports/share/applications 0755 root root -
+d /var/lib/flatpak/exports/share/icons 0755 root root -
-- 
2.43.0

