From be3e02b97c86c2b45d73eec58c05acb4643dfdd3 Mon Sep 17 00:00:00 2001
From: Philipp Trulson <philipp@trulson.de>
Date: Tue, 13 Feb 2018 16:02:31 +0100
Subject: [PATCH] Fix pinning to taskbar

---
 org.remmina.Remmina.desktop | 1 +
 1 file changed, 1 insertion(+)

diff --git a/org.remmina.Remmina.desktop b/org.remmina.Remmina.desktop
index 2e43dfc..e568e91 100644
--- a/org.remmina.Remmina.desktop
+++ b/org.remmina.Remmina.desktop
@@ -73,6 +73,7 @@ Type=Application
 Categories=GTK;GNOME;X-GNOME-NetworkSettings;Network;
 Actions=Profile;Tray;Quit;
 Keywords=remote desktop;rdp;vnc;nx;ssh;spice;
+StartupWMClass=remmina
 
 [Desktop Action Profile]
 Name=Create a New Connection Profile
-- 
2.16.1

