diff -up ParaView-v5.8.1/VTK/ThirdParty/freetype/vtk_freetype.h.in.freetype ParaView-v5.8.1/VTK/ThirdParty/freetype/vtk_freetype.h.in
--- ParaView-v5.8.1/VTK/ThirdParty/freetype/vtk_freetype.h.in.freetype	2020-10-31 08:59:37.251895020 -0600
+++ ParaView-v5.8.1/VTK/ThirdParty/freetype/vtk_freetype.h.in	2020-10-31 09:00:38.592349496 -0600
@@ -25,6 +25,15 @@
 
 #if VTK_MODULE_USE_EXTERNAL_vtkfreetype
 # include <ft2build.h>
+/* FT_CALLBACK_DEF no longer exported since freetype-2.10.3 */
+/* has been moved to <freetype/internal/compiler-macros.h> */
+# ifndef FT_CALLBACK_DEF
+#  ifdef __cplusplus
+#   define FT_CALLBACK_DEF( x )  extern "C"  x
+#  else
+#   define FT_CALLBACK_DEF( x )  static  x
+#  endif
+# endif /* FT_CALLBACK_DEF */
 #else
 # include <vtkfreetype/include/ft2build.h>
 #endif
