diff -ru ParaView-v6.0.1.orig/Qt/Core/pqFlatTreeViewEventTranslator.cxx ParaView-v6.0.1/Qt/Core/pqFlatTreeViewEventTranslator.cxx
--- ParaView-v6.0.1.orig/Qt/Core/pqFlatTreeViewEventTranslator.cxx	2025-11-26 17:57:21.429297646 +0100
+++ ParaView-v6.0.1/Qt/Core/pqFlatTreeViewEventTranslator.cxx	2025-11-26 18:15:34.395469488 +0100
@@ -73,8 +73,8 @@
 
       QString info = QString("%1,%2,%3,%4,%5,%6")
                        .arg(mouseEvent->button())
-                       .arg(mouseEvent->buttons())
-                       .arg(mouseEvent->modifiers())
+                       .arg(static_cast<int>(mouseEvent->buttons()))
+                       .arg(static_cast<int>(mouseEvent->modifiers()))
                        .arg(relPt.x())
                        .arg(relPt.y())
                        .arg(idxStr);
diff -ru ParaView-v6.0.1.orig/Qt/Core/pqQVTKWidgetEventTranslator.cxx ParaView-v6.0.1/Qt/Core/pqQVTKWidgetEventTranslator.cxx
--- ParaView-v6.0.1.orig/Qt/Core/pqQVTKWidgetEventTranslator.cxx	2025-11-26 17:57:21.432297664 +0100
+++ ParaView-v6.0.1/Qt/Core/pqQVTKWidgetEventTranslator.cxx	2025-11-26 18:18:12.809916291 +0100
@@ -90,8 +90,8 @@
               .arg(normalized_x)
               .arg(normalized_y)
               .arg(mouseEvent->button())
-              .arg(mouseEvent->buttons())
-              .arg(mouseEvent->modifiers()));
+              .arg(static_cast<int>(mouseEvent->buttons()))
+              .arg(static_cast<int>(mouseEvent->modifiers())));
         }
         return true;
         break;
@@ -118,15 +118,15 @@
               .arg(normalized_x)
               .arg(normalized_y)
               .arg(mouseEvent->button())
-              .arg(mouseEvent->buttons())
-              .arg(mouseEvent->modifiers()));
+              .arg(static_cast<int>(mouseEvent->buttons()))
+              .arg(static_cast<int>(mouseEvent->modifiers())));
           Q_EMIT recordEvent(widget, "mouseRelease",
             QString("(%1,%2,%3,%4,%5)")
               .arg(normalized_x)
               .arg(normalized_y)
               .arg(mouseEvent->button())
-              .arg(mouseEvent->buttons())
-              .arg(mouseEvent->modifiers()));
+              .arg(static_cast<int>(mouseEvent->buttons()))
+              .arg(static_cast<int>(mouseEvent->modifiers())));
         }
         return true;
         break;
diff -ru ParaView-v6.0.1.orig/ThirdParty/QtTesting/vtkqttesting/pqAbstractItemViewEventTranslator.cxx ParaView-v6.0.1/ThirdParty/QtTesting/vtkqttesting/pqAbstractItemViewEventTranslator.cxx
--- ParaView-v6.0.1.orig/ThirdParty/QtTesting/vtkqttesting/pqAbstractItemViewEventTranslator.cxx	2025-11-26 17:57:21.568509864 +0100
+++ ParaView-v6.0.1/ThirdParty/QtTesting/vtkqttesting/pqAbstractItemViewEventTranslator.cxx	2025-11-26 18:08:56.558897084 +0100
@@ -90,8 +90,8 @@
 
       QString info = QString("%1,%2,%3,%4,%5,%6")
                        .arg(mouseEvent->button())
-                       .arg(mouseEvent->buttons())
-                       .arg(mouseEvent->modifiers())
+                       .arg(static_cast<int>(mouseEvent->buttons()))
+                       .arg(static_cast<int>(mouseEvent->modifiers()))
                        .arg(relPt.x())
                        .arg(relPt.y())
                        .arg(idxStr);
diff -ru ParaView-v6.0.1.orig/ThirdParty/QtTesting/vtkqttesting/pqBasicWidgetEventTranslator.cxx ParaView-v6.0.1/ThirdParty/QtTesting/vtkqttesting/pqBasicWidgetEventTranslator.cxx
--- ParaView-v6.0.1.orig/ThirdParty/QtTesting/vtkqttesting/pqBasicWidgetEventTranslator.cxx	2025-11-26 17:57:21.569298527 +0100
+++ ParaView-v6.0.1/ThirdParty/QtTesting/vtkqttesting/pqBasicWidgetEventTranslator.cxx	2025-11-26 18:02:12.574767136 +0100
@@ -53,8 +53,8 @@
 #endif
         QString info = QString("%1,%2,%3,%4,%5")
                          .arg(mouseEvent->button())
-                         .arg(mouseEvent->buttons())
-                         .arg(mouseEvent->modifiers())
+                         .arg(static_cast<int>(mouseEvent->buttons()))
+                         .arg(static_cast<int>(mouseEvent->modifiers()))
                          .arg(pos.x())
                          .arg(pos.y());
 
