From 3b3a6522a96543ee7d696d315b14b4c20ad79da0 Mon Sep 17 00:00:00 2001
From: Ikey Doherty <ikey@solus-project.com>
Date: Sun, 26 Nov 2017 11:27:46 +0000
Subject: [PATCH 1/2] engine: Do NOT override themes

Signed-off-by: Ikey Doherty <ikey@solus-project.com>
---
 src/engine.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/engine.c b/src/engine.c
index ce85998..8f5fe50 100644
--- a/src/engine.c
+++ b/src/engine.c
@@ -90,7 +90,7 @@ static void slate_set_theme(const char *theme_portion)
 
         gtk_style_context_add_provider_for_screen(screen,
                                                   GTK_STYLE_PROVIDER(prov),
-                                                  GTK_STYLE_PROVIDER_PRIORITY_APPLICATION);
+                                                  GTK_STYLE_PROVIDER_PRIORITY_FALLBACK);
         _theme_style_provider = prov;
 end:
         g_free(uri);
-- 
2.15.0

