From 9be09c4c62c22d79037fd9018b25dc06062e4931 Mon Sep 17 00:00:00 2001
From: Ikey Doherty <ikey@solus-project.com>
Date: Sun, 26 Nov 2017 11:31:48 +0000
Subject: [PATCH 2/2] engine: Only explicitly override theme when using Arc
 themeset

Signed-off-by: Ikey Doherty <ikey@solus-project.com>
---
 src/engine.c | 14 +++++++++++---
 1 file changed, 11 insertions(+), 3 deletions(-)

diff --git a/src/engine.c b/src/engine.c
index 8f5fe50..6b80901 100644
--- a/src/engine.c
+++ b/src/engine.c
@@ -66,6 +66,9 @@ static void slate_set_theme(const char *theme_portion)
         gchar *uri = NULL;
         GtkCssProvider *prov = NULL;
         GdkScreen *screen = NULL;
+        GtkSettings *settings = NULL;
+        const char *cur_theme = NULL;
+        guint priority = GTK_STYLE_PROVIDER_PRIORITY_FALLBACK;
 
         uri = slate_theme_form_theme_path(theme_portion);
         file = g_file_new_for_uri(uri);
@@ -80,6 +83,13 @@ static void slate_set_theme(const char *theme_portion)
         }
         screen = gdk_screen_get_default();
 
+        settings = gtk_settings_get_default();
+        g_object_get(settings, "gtk-theme-name", &cur_theme, NULL);
+
+        if (cur_theme && g_str_has_prefix(cur_theme, "Arc")) {
+                priority = GTK_STYLE_PROVIDER_PRIORITY_APPLICATION;
+        }
+
         /* Revert the previous style provider */
         if (_theme_style_provider != NULL) {
                 gtk_style_context_remove_provider_for_screen(screen,
@@ -88,9 +98,7 @@ static void slate_set_theme(const char *theme_portion)
                 _theme_style_provider = NULL;
         }
 
-        gtk_style_context_add_provider_for_screen(screen,
-                                                  GTK_STYLE_PROVIDER(prov),
-                                                  GTK_STYLE_PROVIDER_PRIORITY_FALLBACK);
+        gtk_style_context_add_provider_for_screen(screen, GTK_STYLE_PROVIDER(prov), priority);
         _theme_style_provider = prov;
 end:
         g_free(uri);
-- 
2.15.0

