From 57875e42af03faf387503a547cc2b0db7c5bc3e1 Mon Sep 17 00:00:00 2001
From: Reilly Brogan <reilly@reillybrogan.com>
Date: Thu, 30 Nov 2023 23:33:15 -0600
Subject: [PATCH] Use wlroots-11 package

Signed-off-by: Reilly Brogan <reilly@reillybrogan.com>
---
 meson.build | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/meson.build b/meson.build
index 65bba6e3..80505fd9 100644
--- a/meson.build
+++ b/meson.build
@@ -50,7 +50,7 @@ wayland_server = dependency('wayland-server', version: '>=1.21.0')
 wayland_client = dependency('wayland-client')
 wayland_cursor = dependency('wayland-cursor')
 wayland_protos = dependency('wayland-protocols', version: '>=1.24')
-wlroots = dependency('wlroots', version: wlroots_version)
+wlroots = dependency('wlroots-11', version: wlroots_version)
 xkbcommon = dependency('xkbcommon')
 cairo = dependency('cairo')
 pango = dependency('pango')
-- 
2.43.0

