From 765338859d811ebe24bf064f2a022004fe7f224e Mon Sep 17 00:00:00 2001
From: Alexander Vorobyev <avorobyev@protonmail.com>
Date: Tue, 17 Mar 2020 19:51:32 +0300
Subject: Solus build configuration

---
 makefile | 17 ++++++++---------
 1 file changed, 8 insertions(+), 9 deletions(-)

diff --git a/makefile b/makefile
index dcb1e07..4c43904 100644
--- a/makefile
+++ b/makefile
@@ -1,16 +1,16 @@
 CC=gcc
 INC=-Isrc
-CFLAGS=-rdynamic -fPIC -shared -Wall -std=c99 -fvisibility=hidden $(INC)
-LDFLAGS=-Wl,-z,relro,-z,now
+CFLAGS:=$(CFLAGS) -rdynamic -fPIC -shared -Wall -std=c99 -fvisibility=hidden $(INC)
+LDFLAGS:=$(LDFLAGS) -Wl,-z,relro,-z,now
 LDLIBS=-ldl -lrt
 
-prefix=/usr/local
+prefix=/usr
 exec_prefix=$(prefix)
 bindir=$(exec_prefix)/bin
 libdir=$(exec_prefix)/lib
 datarootdir=$(prefix)/share
-LIB32_PATH=$(libdir)/libstrangle/lib32
-LIB64_PATH=$(libdir)/libstrangle/lib64
+LIB32_PATH=$(prefix)/lib32
+LIB64_PATH=$(prefix)/lib64
 
 SOURCEDIR=src
 BUILDDIR=build
@@ -18,7 +18,7 @@ COMMON_SOURCES=$(wildcard $(SOURCEDIR)/*.c)
 GL_SOURCES=$(COMMON_SOURCES) $(wildcard $(SOURCEDIR)/opengl/*.c)
 
 CXX=g++
-CXXFLAGS= -pthread -rdynamic -fPIC -shared -Wall -std=gnu++17 -fvisibility=hidden -Iinclude $(INC) -DVK_USE_PLATFORM_XLIB_KHR -DHAVE_PTHREAD -DHAVE_TIMESPEC_GET
+CXXFLAGS:=$(CXXFLAGS) -pthread -rdynamic -fPIC -shared -Wall -std=gnu++17 -fvisibility=hidden -Iinclude $(INC) -DVK_USE_PLATFORM_XLIB_KHR -DHAVE_PTHREAD -DHAVE_TIMESPEC_GET
 LDXXFLAGS=
 LDXXLIBS=-lrt
 VK_SOURCES=\
@@ -30,7 +30,7 @@ VK_SOURCES=\
 
 .PHONY: all 32-bit 64-bit native ld clean uninstall install install-32 install-64 install-ld install-common install-native
 
-all: 32-bit 64-bit ld
+all: 32-bit 64-bit
 
 32-bit: \
 	$(BUILDDIR)/libstrangle32.so \
@@ -111,8 +111,7 @@ install: \
 	all \
 	install-common \
 	install-32 \
-	install-64 \
-	install-ld
+	install-64
 
 clean:
 	rm -f $(BUILDDIR)/libstrangle64.so
-- 
2.25.1

