From f21f2a9c608d82a5cc9f61daffbc7dc3b3ce97fb Mon Sep 17 00:00:00 2001
From: Alexander Vorobyev <avorobyev@protonmail.com>
Date: Fri, 16 Nov 2018 08:43:55 +0300
Subject: Solus build configuration

---
 makefile | 14 ++++++--------
 1 file changed, 6 insertions(+), 8 deletions(-)

diff --git a/makefile b/makefile
index 8a3d962..eeb7c8f 100644
--- a/makefile
+++ b/makefile
@@ -1,18 +1,18 @@
 CC=gcc
-CFLAGS=-rdynamic -fPIC -shared -Wall -std=c99 -fvisibility=hidden -DHOOK_DLSYM
-LDFLAGS=-Wl,-z,relro,-z,now
+CFLAGS:=$(CFLAGS) -rdynamic -fPIC -shared -Wall -std=c99 -fvisibility=hidden -DHOOK_DLSYM
+LDFLAGS:=$(LDFLAGS) -Wl,-z,relro,-z,now
 LDLIBS=-ldl -lrt
-prefix=/usr/local
+prefix=/usr
 bindir=$(prefix)/bin
 libdir=$(prefix)/lib
 DOC_PATH=$(prefix)/share/doc/libstrangle
-LIB32_PATH=$(libdir)/libstrangle/lib32
-LIB64_PATH=$(libdir)/libstrangle/lib64
+LIB32_PATH=$(prefix)/lib32
+LIB64_PATH=$(prefix)/lib64
 SOURCEDIR=src/
 BUILDDIR=build/
 SOURCES=$(wildcard $(SOURCEDIR)*.c)
 
-all: $(BUILDDIR)libstrangle64.so $(BUILDDIR)libstrangle32.so $(BUILDDIR)libstrangle.conf
+all: $(BUILDDIR)libstrangle64.so $(BUILDDIR)libstrangle32.so
 
 $(BUILDDIR):
 	mkdir -p $(BUILDDIR)
@@ -28,12 +28,10 @@ $(BUILDDIR)libstrangle32.so: $(BUILDDIR)
 	$(CC) $(CFLAGS) $(LDFLAGS) -m32 -o $(BUILDDIR)libstrangle32.so $(SOURCES) $(LDLIBS)
 
 install: all
-	install -m 0644 -D -T $(BUILDDIR)libstrangle.conf $(DESTDIR)/etc/ld.so.conf.d/libstrangle.conf
 	install -m 0755 -D -T $(BUILDDIR)libstrangle32.so $(DESTDIR)$(LIB32_PATH)/libstrangle.so
 	install -m 0755 -D -T $(BUILDDIR)libstrangle64.so $(DESTDIR)$(LIB64_PATH)/libstrangle.so
 	install -m 0755 -D -T $(SOURCEDIR)strangle.sh $(DESTDIR)$(bindir)/strangle
 	install -m 0644 -D -T COPYING $(DESTDIR)$(DOC_PATH)/LICENSE
-	ldconfig
 
 clean:
 	rm -f $(BUILDDIR)libstrangle64.so
-- 
2.19.1

