From 1a809af65cfadf603c560b76ea9662995e90b2b5 Mon Sep 17 00:00:00 2001
From: Alexander Vorobyev <avorobyev@protonmail.com>
Date: Sun, 18 Nov 2018 19:36:56 +0300
Subject: Use versioned symlink

---
 src/vulkan.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/vulkan.c b/src/vulkan.c
index 50819d9..02c7b95 100644
--- a/src/vulkan.c
+++ b/src/vulkan.c
@@ -29,9 +29,9 @@ along with libstrangle.  If not, see <http://www.gnu.org/licenses/>.
 #include <stdio.h>
 
 void* strangle_vkRequireFunction( const char* name ) {
-	void* handle = dlopen( "libvulkan.so", RTLD_NOW );
+	void* handle = dlopen( "libvulkan.so.1", RTLD_NOW );
 	if ( handle == NULL ) {
-		printf( "Strangle: failed to load libvulkan.so\n" );
+		printf( "Strangle: failed to load libvulkan.so.1\n" );
 	}
 
 	void* func = real_dlsym( handle, name );
-- 
2.19.1

