From 4a5f5ef3e73ab2738d8238e888a7f5e7912c8cd0 Mon Sep 17 00:00:00 2001
From: Marc Mueller <30130371+cdce8p@users.noreply.github.com>
Date: Thu, 17 Oct 2024 10:31:00 +0200
Subject: [PATCH] Update setuptools to 75.2.0 (#2615)

---
 pyproject.toml | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/pyproject.toml b/pyproject.toml
index 61f8c3912b..165d743013 100644
--- a/pyproject.toml
+++ b/pyproject.toml
@@ -1,5 +1,5 @@
 [build-system]
-requires = ["setuptools>=64.0", "wheel>=0.37.1"]
+requires = ["setuptools>=75.2.0"]
 build-backend = "setuptools.build_meta"
 
 [project]
