--- a/Makefile.in.orig	2021-10-29 09:48:49.634834320 +0200
+++ b/Makefile.in	2021-10-29 09:53:51.826848258 +0200
@@ -3,6 +3,7 @@
 bindir=@bindir@
 libdir=@libdir@
 datarootdir=@datarootdir@
+datadir=@datadir@
 mandir=@mandir@
 sysconfdir=@sysconfdir@
 
@@ -134,11 +135,11 @@
 	install -m 0755 -d $(DESTDIR)$(DOCDIR)
 	install -m 0644 -t $(DESTDIR)$(DOCDIR) COPYING README RELNOTES etc/templates/*
 	# profiles and settings
-	install -m 0755 -d $(DESTDIR)$(sysconfdir)/firejail
-	install -m 0644 -t $(DESTDIR)$(sysconfdir)/firejail etc/profile-a-l/*.profile etc/profile-m-z/*.profile etc/inc/*.inc etc/net/*.net etc/firejail.config
-	sh -c "if [ ! -f $(DESTDIR)/$(sysconfdir)/firejail/login.users ]; then install -c -m 0644 etc/login.users $(DESTDIR)/$(sysconfdir)/firejail/.; fi;"
+	install -m 0755 -d $(DESTDIR)$(datadir)/defaults/firejail
+	install -m 0644 -t $(DESTDIR)$(datadir)/defaults/firejail etc/profile-a-l/*.profile etc/profile-m-z/*.profile etc/inc/*.inc etc/net/*.net etc/firejail.config
+	sh -c "if [ ! -f $(DESTDIR)/$(datadir)/defaults/firejail/login.users ]; then install -c -m 0644 etc/login.users $(DESTDIR)/$(datadir)/defaults/firejail/.; fi;"
 ifeq ($(BUSYBOX_WORKAROUND),yes)
-	./mketc.sh $(DESTDIR)$(sysconfdir)/firejail/disable-common.inc
+	./mketc.sh $(DESTDIR)$(datadir)/defaults/firejail/disable-common.inc
 endif
 ifeq ($(HAVE_APPARMOR),-DHAVE_APPARMOR)
 	# install apparmor profile
@@ -191,7 +192,7 @@
 	rm -f $(DESTDIR)$(datarootdir)/bash-completion/completions/firejail
 	rm -f $(DESTDIR)$(datarootdir)/bash-completion/completions/firemon
 	rm -f $(DESTDIR)$(datarootdir)/bash-completion/completions/firecfg
-	@echo "If you want to install a different version of firejail, you might also need to run 'rm -fr $(DESTDIR)$(sysconfdir)/firejail', see #2038."
+	@echo "If you want to install a different version of firejail, you might also need to run 'rm -fr $(DESTDIR)$(datadir)/defaults/firejail', see #2038."
 
 DISTFILES = "src etc m4 platform contrib configure configure.ac Makefile.in install.sh mkman.sh mketc.sh mkdeb.sh.in COPYING README RELNOTES"
 DISTFILES_TEST = "test/Makefile.in test/apps test/apps-x11 test/apps-x11-xorg test/root test/private-lib test/fnetfilter test/fcopy test/environment test/profiles test/utils test/compile test/filters test/network test/fs test/sysutils test/chroot"
--- a/src/firejail/checkcfg.c.orig	2021-10-29 09:57:03.977857121 +0200
+++ b/src/firejail/checkcfg.c	2021-10-29 09:58:00.484859727 +0200
@@ -60,7 +60,12 @@
 		cfg_val[CFG_BROWSER_ALLOW_DRM] = 0;
 
 		// open configuration file
-		const char *fname = SYSCONFDIR "/firejail.config";
+		const char *fname;
+		if (access(SYSCONFDIR "/firejail.config", F_OK) == 0) {
+			fname = SYSCONFDIR "/firejail.config";
+		} else {
+			fname = DATADIR "/defaults/firejail/firejail.config";
+		}
 		fp = fopen(fname, "re");
 		if (!fp) {
 #ifdef HAVE_GLOBALCFG
--- a/src/firejail/profile.c.orig	2021-06-29 08:45:01.000000000 +0200
+++ b/src/firejail/profile.c	2021-10-29 09:59:59.015865194 +0200
@@ -82,7 +82,9 @@
 	if (!rv)
 		// look for a user profile in /etc/firejail directory
 		rv = profile_find(name, SYSCONFDIR, add_ext);
-
+	if (!rv)
+		// look for a user profile in /usr/share/defaults/firejail directory
+		rv = profile_find(name, DATADIR "/defaults/firejail", add_ext);
 	return rv;
 }
 
--- a/src/firejail/restricted_shell.c.orig	2021-06-29 08:45:01.000000000 +0200
+++ b/src/firejail/restricted_shell.c	2021-10-29 09:59:59.016865194 +0200
@@ -34,8 +34,16 @@
 		errExit("asprintf");
 	FILE *fp = fopen(fname, "re");
 	free(fname);
-	if (fp == NULL)
-		return 0;
+	fname = NULL;
+	if (fp == NULL) {
+		// Grab the stateless version
+		if (asprintf(&fname, "%s/defaults/firejail/login.users", DATADIR) == -1)
+			errExit("asprintf");
+		FILE *fp = fopen(fname, "r");
+		free(fname);
+		if (fp == NULL)
+			return 0;
+	}
 
 	int lineno = 0;
 	char buf[MAX_READ];
--- a/src/common.mk.in.orig	2021-10-29 09:55:10.723851897 +0200
+++ b/src/common.mk.in	2021-10-29 09:56:36.478855852 +0200
@@ -6,6 +6,7 @@
 bindir=@bindir@
 libdir=@libdir@
 sysconfdir=@sysconfdir@
+datadir=@datadir@
 
 VERSION=@PACKAGE_VERSION@
 NAME=@PACKAGE_NAME@
@@ -42,6 +43,7 @@
 CFLAGS = @CFLAGS@
 CFLAGS += -ggdb $(HAVE_FATAL_WARNINGS) -O2 -DVERSION='"$(VERSION)"' $(HAVE_GCOV)
 CFLAGS += -DPREFIX='"$(prefix)"' -DSYSCONFDIR='"$(sysconfdir)/firejail"' -DLIBDIR='"$(libdir)"' -DBINDIR='"$(bindir)"'
+CFLAGS += -DDATADIR='"$(datadir)"'
 MANFLAGS = $(HAVE_LTS) $(HAVE_OUTPUT) $(HAVE_X11) $(HAVE_PRIVATE_HOME) $(HAVE_APPARMOR) $(HAVE_OVERLAYFS) $(HAVE_USERTMPFS) $(HAVE_DBUSPROXY) $(HAVE_FIRETUNNEL) $(HAVE_GLOBALCFG) $(HAVE_CHROOT) $(HAVE_NETWORK) $(HAVE_USERNS) $(HAVE_FILE_TRANSFER) $(HAVE_WHITELIST) $(HAVE_SELINUX) $(HAVE_SUID) $(HAVE_FORCE_NONEWPRIVS)
 CFLAGS += $(MANFLAGS)
 CFLAGS += -fstack-protector-all -D_FORTIFY_SOURCE=2 -fPIE -Wformat -Wformat-security
--- a/configure.ac.orig	2021-10-29 09:54:15.224849337 +0200
+++ b/configure.ac	2021-10-29 09:54:56.742851252 +0200
@@ -316,6 +316,7 @@
 Configuration options:
    prefix: $prefix
    sysconfdir: $sysconfdir
+   defaultsdir: $datadir/defaults/firejail
    apparmor: $HAVE_APPARMOR
    SELinux labeling support: $HAVE_SELINUX
    global config: $HAVE_GLOBALCFG
