From: =?UTF-8?q?Mateusz=20Miku=C5=82a?= <mati865@gmail.com>, Fabio Forni <livingsilver94.solus@redaril.me>
Subject: [PATCH] Support a stateless configuration

diff --git a/Makefile.in b/Makefile.in
index d11d537b..d2099870 100644
--- a/Makefile.in
+++ b/Makefile.in
@@ -9,6 +9,7 @@ exec_prefix=@exec_prefix@
 bindir=@bindir@
 libdir=@libdir@
 datarootdir=@datarootdir@
+datadir=@datadir@
 mandir=@mandir@
 sysconfdir=@sysconfdir@
 
@@ -123,12 +124,12 @@ endif
 	install -c -m 0644 README $(DESTDIR)/$(DOCDIR)/.
 	install -c -m 0644 RELNOTES $(DESTDIR)/$(DOCDIR)/.
 	# etc files
-	./mketc.sh $(sysconfdir) $(BUSYBOX_WORKAROUND)
-	install -m 0755 -d $(DESTDIR)/$(sysconfdir)/firejail
+	./mketc.sh $(datadir)/defaults $(BUSYBOX_WORKAROUND)
+	install -m 0755 -d $(DESTDIR)/$(datadir)/defaults/firejail
 	for file in .etc/* etc/firejail.config; do \
-		install -c -m 0644 $$file $(DESTDIR)/$(sysconfdir)/firejail; \
+		install -c -m 0644 $$file $(DESTDIR)/$(datadir)/defaults/firejail; \
 	done
-	sh -c "if [ ! -f $(DESTDIR)/$(sysconfdir)/firejail/login.users ]; then install -c -m 0644 etc/login.users $(DESTDIR)/$(sysconfdir)/firejail/.; fi;"
+	sh -c "if [ ! -f $(DESTDIR)/$(datadir)/defaults/firejail/login.users ]; then install -c -m 0644 etc/login.users $(DESTDIR)/$(datadir)/defaults/firejail/.; fi;"
 	rm -fr .etc
 ifeq ($(HAVE_APPARMOR),-DHAVE_APPARMOR)
 	# install apparmor profile
diff --git a/configure.ac b/configure.ac
index 2b7dd1d2..c78122cf 100644
--- a/configure.ac
+++ b/configure.ac
@@ -183,6 +183,7 @@ echo
 echo "Configuration options:"
 echo "   prefix: $prefix"
 echo "   sysconfdir: $sysconfdir"
+echo "   defaultsdir: $datadir/defaults/firejail"
 echo "   seccomp: $HAVE_SECCOMP"
 echo "   <linux/seccomp.h>: $HAVE_SECCOMP_H"
 echo "   apparmor: $HAVE_APPARMOR"
--- a/src/common.mk.in
+++ b/src/common.mk.in
@@ -5,6 +5,7 @@ prefix=@prefix@
 exec_prefix=@exec_prefix@
 libdir=@libdir@
 sysconfdir=@sysconfdir@
+datadir=@datadir@
 
 VERSION=@PACKAGE_VERSION@
 NAME=@PACKAGE_NAME@
@@ -28,7 +29,7 @@ C_FILE_LIST       = $(sort $(wildcard *.c))
 OBJS = $(C_FILE_LIST:.c=.o)
 BINOBJS =  $(foreach file, $(OBJS), $file)
 
-CFLAGS += -ggdb $(HAVE_FATAL_WARNINGS) -O2 -DVERSION='"$(VERSION)"'  $(HAVE_GCOV) -DPREFIX='"$(prefix)"'  -DSYSCONFDIR='"$(sysconfdir)/firejail"' -DLIBDIR='"$(libdir)"' $(HAVE_X11) $(HAVE_PRIVATE_HOME) $(HAVE_APPARMOR) $(HAVE_OVERLAYFS) $(HAVE_SECCOMP) $(HAVE_GLOBALCFG) $(HAVE_SECCOMP_H) $(HAVE_CHROOT) $(HAVE_NETWORK) $(HAVE_USERNS) $(HAVE_FILE_TRANSFER) $(HAVE_WHITELIST) -fstack-protector-all -D_FORTIFY_SOURCE=2 -fPIE -pie -Wformat -Wformat-security
+CFLAGS += -ggdb $(HAVE_FATAL_WARNINGS) -O2 -DVERSION='"$(VERSION)"'  $(HAVE_GCOV) -DPREFIX='"$(prefix)"'  -DSYSCONFDIR='"$(sysconfdir)/firejail"'   -DDATADIR='"$(datadir)"' -DLIBDIR='"$(libdir)"' $(HAVE_X11) $(HAVE_PRIVATE_HOME) $(HAVE_APPARMOR) $(HAVE_OVERLAYFS) $(HAVE_SECCOMP) $(HAVE_GLOBALCFG) $(HAVE_SECCOMP_H) $(HAVE_CHROOT) $(HAVE_NETWORK) $(HAVE_USERNS) $(HAVE_FILE_TRANSFER) $(HAVE_WHITELIST) -fstack-protector-all -D_FORTIFY_SOURCE=2 -fPIE -pie -Wformat -Wformat-security
 LDFLAGS += -pie -Wl,-z,relro -Wl,-z,now -lpthread
 EXTRA_LDFLAGS +=@EXTRA_LDFLAGS@
 EXTRA_CFLAGS +=@EXTRA_CFLAGS@
diff --git a/src/firejail/checkcfg.c b/src/firejail/checkcfg.c
index f101a845..fa5f681e 100644
--- a/src/firejail/checkcfg.c
+++ b/src/firejail/checkcfg.c
@@ -52,7 +52,12 @@ int checkcfg(int val) {
 		cfg_val[CFG_XPRA_ATTACH] = 0;
 
 		// open configuration file
-		const char *fname = SYSCONFDIR "/firejail.config";
+		const char *fname;
+		if (access(SYSCONFDIR "/firejail.config", F_OK) == 0) {
+			fname = SYSCONFDIR "/firejail.config";
+		} else {
+			fname = DATADIR "/defaults/firejail/firejail.config";
+		}
 		fp = fopen(fname, "r");
 		if (!fp) {
 #ifdef HAVE_GLOBALCFG
--- a/src/firejail/profile.c
+++ b/src/firejail/profile.c
@@ -78,7 +78,9 @@ int profile_find_firejail(const char *name, int add_ext) {
 	if (!rv)
 		// look for a user profile in /etc/firejail directory
 		rv = profile_find(name, SYSCONFDIR, add_ext);
-
+	if (!rv)
+		// look for a user profile in /usr/share/defaults/firejail directory
+		rv = profile_find(name, DATADIR "/defaults/firejail", add_ext);
 	return rv;
 }
 
diff --git a/src/firejail/restricted_shell.c b/src/firejail/restricted_shell.c
index d09a2c7e..4162d14e 100644
--- a/src/firejail/restricted_shell.c
+++ b/src/firejail/restricted_shell.c
@@ -34,8 +34,16 @@ int restricted_shell(const char *user) {
 		errExit("asprintf");
 	FILE *fp = fopen(fname, "r");
 	free(fname);
-	if (fp == NULL)
-		return 0;
+	fname = NULL;
+	if (fp == NULL) {
+		// Grab the stateless version
+		if (asprintf(&fname, "%s/defaults/firejail/login.users", DATADIR) == -1)
+			errExit("asprintf");
+		FILE *fp = fopen(fname, "r");
+		free(fname);
+		if (fp == NULL)
+			return 0;
+	}
 
 	int lineno = 0;
 	char buf[MAX_READ];
-- 
2.16.2

