diff --git a/Makefile b/Makefile
index 57a6928..f939d5a 100644
--- a/Makefile
+++ b/Makefile
@@ -133,15 +133,15 @@ endif
 	install -m 0755 -d $(DESTDIR)$(docdir)
 	install -m 0644 -t $(DESTDIR)$(docdir) COPYING README RELNOTES etc/templates/*
 	# profiles and settings
-	install -m 0755 -d $(DESTDIR)$(sysconfdir)/firejail
-	install -m 0644 -t $(DESTDIR)$(sysconfdir)/firejail src/firecfg/firecfg.config
-	install -m 0644 -t $(DESTDIR)$(sysconfdir)/firejail etc/profile-a-l/*.profile etc/profile-m-z/*.profile etc/inc/*.inc etc/net/*.net etc/firejail.config
-	sh -c "if [ ! -f $(DESTDIR)/$(sysconfdir)/firejail/login.users ]; then install -c -m 0644 etc/login.users $(DESTDIR)/$(sysconfdir)/firejail/.; fi;"
+	install -m 0755 -d $(DESTDIR)$(datadir)/defaults/firejail
+	install -m 0644 -t $(DESTDIR)$(datadir)/defaults/firejail src/firecfg/firecfg.config
+	install -m 0644 -t $(DESTDIR)$(datadir)/defaults/firejail etc/profile-a-l/*.profile etc/profile-m-z/*.profile etc/inc/*.inc etc/net/*.net etc/firejail.config
+	sh -c "if [ ! -f $(DESTDIR)/$(datadir)/defaults/firejail/login.users ]; then install -c -m 0644 etc/login.users $(DESTDIR)/$(datadir)/defaults/firejail/.; fi;"
 ifeq ($(HAVE_IDS),-DHAVE_IDS)
-	install -m 0644 -t $(DESTDIR)$(sysconfdir)/firejail etc/ids.config
+	install -m 0644 -t $(DESTDIR)$(datadir)/defaults/firejail etc/ids.config
 endif
 ifeq ($(BUSYBOX_WORKAROUND),yes)
-	./mketc.sh $(DESTDIR)$(sysconfdir)/firejail/disable-common.inc
+	./mketc.sh $(DESTDIR)$(datadir)/defaults/firejail/disable-common.inc
 endif
 ifeq ($(HAVE_APPARMOR),-DHAVE_APPARMOR)
 	# install apparmor profile
@@ -201,7 +201,7 @@ uninstall: config.mk
 	rm -f $(DESTDIR)$(datarootdir)/zsh/site-functions/_firejail
 	rm -f $(DESTDIR)$(datarootdir)/vim/vimfiles/ftdetect/firejail.vim
 	rm -f $(DESTDIR)$(datarootdir)/vim/vimfiles/syntax/firejail.vim
-	@echo "If you want to install a different version of firejail, you might also need to run 'rm -fr $(DESTDIR)$(sysconfdir)/firejail', see #2038."
+	@echo "If you want to install a different version of firejail, you might also need to run 'rm -fr $(DESTDIR)$(datadir)/defaults/firejail', see #2038."
 
 DISTFILES = \
 COPYING \
diff --git a/config.mk.in b/config.mk.in
index cfef6b8..a472656 100644
--- a/config.mk.in
+++ b/config.mk.in
@@ -17,6 +17,7 @@ exec_prefix=@exec_prefix@
 bindir=@bindir@
 libdir=@libdir@
 datarootdir=@datarootdir@
+datadir=@datadir@
 docdir=@docdir@
 mandir=@mandir@
 sysconfdir=@sysconfdir@
diff --git a/configure.ac b/configure.ac
index 412cdd6..1f2037e 100644
--- a/configure.ac
+++ b/configure.ac
@@ -304,6 +304,7 @@ Compile options:
    install contrib scripts: $HAVE_CONTRIB_INSTALL
    prefix: $prefix
    sysconfdir: $sysconfdir
+   defaultsdir: $datadir/defaults/firejail
    Spectre compiler patch: $HAVE_SPECTRE
 
 Features:
diff --git a/src/firecfg/main.c b/src/firecfg/main.c
index 07e3041..f7313f3 100644
--- a/src/firecfg/main.c
+++ b/src/firecfg/main.c
@@ -176,8 +176,13 @@ static void set_file(const char *name, const char *firejail_exec) {
 // parse /etc/firejail/firecfg.config file
 static void set_links_firecfg(void) {
 	char *cfgfile;
-	if (asprintf(&cfgfile, "%s/firecfg.config", SYSCONFDIR) == -1)
-		errExit("asprintf");
+	if (access(SYSCONFDIR "/firejail.config", F_OK) == 0) {
+		if (asprintf(&cfgfile, "%s/firejail.config", SYSCONFDIR) == -1)
+			errExit("asprintf");
+	} else {
+		if (asprintf(&cfgfile, "%s/defaults/firejail/firejail.config", DATADIR) == -1)
+			errExit("asprintf");
+	}
 
 	char *firejail_exec;
 	if (asprintf(&firejail_exec, "%s/bin/firejail", PREFIX) == -1)
diff --git a/src/firejail/appimage.c b/src/firejail/appimage.c
index 4794735..c5fb2a2 100644
--- a/src/firejail/appimage.c
+++ b/src/firejail/appimage.c
@@ -46,9 +46,16 @@ int appimage_find_profile(const char *archive) {
 	assert(strlen(archive));
 
 	// try to match the name of the archive with the list of programs in /etc/firejail/firecfg.config
-	FILE *fp = fopen(SYSCONFDIR "/firecfg.config", "r");
+	const char *fname;
+	if (access(SYSCONFDIR "/firecfg.config", F_OK) == 0) {
+		fname = SYSCONFDIR "/firecfg.config";
+	} else {
+		fname = DATADIR "/defaults/firejail/firecfg.config";
+	}
+	FILE *fp = fopen(fname, "r");
 	if (!fp) {
-		fprintf(stderr, "Error: cannot find %s, firejail is not correctly installed\n", SYSCONFDIR "/firecfg.config");
+		fprintf(stderr, "Error: cannot find %s or %s, firejail is not correctly installed\n",
+				SYSCONFDIR "/firecfg.config", DATADIR "/defaults/firejail/firecfg.config");
 		exit(1);
 	}
 	char buf[MAXBUF];
diff --git a/src/firejail/checkcfg.c b/src/firejail/checkcfg.c
index ce8446c..55805d8 100644
--- a/src/firejail/checkcfg.c
+++ b/src/firejail/checkcfg.c
@@ -66,7 +66,12 @@ int checkcfg(int val) {
 		cfg_val[CFG_TRACELOG] = 0;
 
 		// open configuration file
-		const char *fname = SYSCONFDIR "/firejail.config";
+		const char *fname;
+		if (access(SYSCONFDIR "/firejail.config", F_OK) == 0) {
+			fname = SYSCONFDIR "/firejail.config";
+		} else {
+			fname = DATADIR "/defaults/firejail/firejail.config";
+		}
 		fp = fopen(fname, "re");
 		if (!fp) {
 #ifdef HAVE_GLOBALCFG
diff --git a/src/firejail/profile.c b/src/firejail/profile.c
index acf206d..633bf50 100644
--- a/src/firejail/profile.c
+++ b/src/firejail/profile.c
@@ -85,6 +85,9 @@ int profile_find_firejail(const char *name, int add_ext) {
 		// look for a user profile in /etc/firejail directory
 		rv = profile_find(name, SYSCONFDIR, add_ext);
 
+	if (!rv)
+		// look for a user profile in /usr/share/defaults/firejail directory
+		rv = profile_find(name, DATADIR "/defaults/firejail", add_ext);
 	return rv;
 #else
 	return profile_find(name, SYSCONFDIR, add_ext);
diff --git a/src/firejail/restricted_shell.c b/src/firejail/restricted_shell.c
index c1340ca..dd5b239 100644
--- a/src/firejail/restricted_shell.c
+++ b/src/firejail/restricted_shell.c
@@ -34,8 +34,16 @@ int restricted_shell(const char *user) {
 		errExit("asprintf");
 	FILE *fp = fopen(fname, "re");
 	free(fname);
-	if (fp == NULL)
-		return 0;
+	fname = NULL;
+	if (fp == NULL) {
+		// Grab the stateless version
+		if (asprintf(&fname, "%s/defaults/firejail/login.users", DATADIR) == -1)
+			errExit("asprintf");
+		FILE *fp = fopen(fname, "r");
+		free(fname);
+		if (fp == NULL)
+			return 0;
+	}
 
 	int lineno = 0;
 	char buf[MAX_READ];
diff --git a/src/prog.mk b/src/prog.mk
index b2ccf61..bb358cf 100644
--- a/src/prog.mk
+++ b/src/prog.mk
@@ -16,6 +16,7 @@ PROG_CFLAGS = \
 	-DPREFIX='"$(prefix)"' -DSYSCONFDIR='"$(sysconfdir)/firejail"' \
 	-DLIBDIR='"$(libdir)"' -DBINDIR='"$(bindir)"' \
 	-DVARDIR='"/var/lib/firejail"' \
+	-DDATADIR='"$(datadir)"' \
 	$(HAVE_GCOV) $(MANFLAGS) \
 	$(EXTRA_CFLAGS)
 
