--- a/Makefile
+++ b/Makefile
@@ -247,17 +247,16 @@ endif
 	$(INSTALL) -m 0755 -d $(DESTDIR)$(docdir)
 	$(INSTALL) -m 0644 -t $(DESTDIR)$(docdir) COPYING README RELNOTES etc/templates/*
 	# profiles and settings
-	$(INSTALL) -m 0755 -d $(DESTDIR)$(sysconfdir)/firejail
-	$(INSTALL) -m 0755 -d $(DESTDIR)$(sysconfdir)/firejail/firecfg.d
-	$(INSTALL) -m 0644 -t $(DESTDIR)$(sysconfdir)/firejail src/firecfg/firecfg.config
-	$(INSTALL) -m 0644 -t $(DESTDIR)$(sysconfdir)/firejail etc/firejail.config
+	$(INSTALL) -m 0755 -d $(DESTDIR)$(datadir)/defaults/firejail
+	$(INSTALL) -m 0644 -t $(DESTDIR)$(datadir)/defaults/firejail src/firecfg/firecfg.config
+	$(INSTALL) -m 0644 -t $(DESTDIR)$(datadir)/defaults/firejail etc/firejail.config
 	@printf 'Installing profiles...\n'
-	@$(INSTALL) -m 0644 -t $(DESTDIR)$(sysconfdir)/firejail $(PROFILES_INC) $(PROFILES_NET) $(PROFILES_PRO)
-	sh -c "if [ ! -f $(DESTDIR)$(sysconfdir)/firejail/login.users ]; then \
-		$(INSTALL) -m 0644 -t $(DESTDIR)$(sysconfdir)/firejail etc/login.users; \
+	@$(INSTALL) -m 0644 -t $(DESTDIR)$(datadir)/defaults/firejail $(PROFILES_INC) $(PROFILES_NET) $(PROFILES_PRO)
+	sh -c "if [ ! -f $(DESTDIR)$(datadir)/defaults/firejail/login.users ]; then \
+		$(INSTALL) -m 0644 -t $(DESTDIR)$(datadir)/defaults/firejail etc/login.users; \
 	fi"
 ifeq ($(BUSYBOX_WORKAROUND),yes)
-	./mketc.sh $(DESTDIR)$(sysconfdir)/firejail/disable-common.inc
+	./mketc.sh $(DESTDIR)$(datadir)/defaults/firejail/disable-common.inc
 endif
 ifeq ($(HAVE_APPARMOR),-DHAVE_APPARMOR)
 	# install apparmor profiles
@@ -311,7 +310,7 @@ uninstall: config.mk
 	$(RM) $(DESTDIR)$(datarootdir)/vim/vimfiles/ftplugin/firejail.vim
 	$(RM) $(DESTDIR)$(datarootdir)/vim/vimfiles/syntax/firejail.vim
 	$(RM) $(DESTDIR)$(datarootdir)/gtksourceview-5/language-specs/firejail-profile.lang
-	@echo "If you want to install a different version of firejail, you might also need to run 'rm -fr $(DESTDIR)$(sysconfdir)/firejail', see #2038."
+	@echo "If you want to install a different version of firejail, you might also need to run 'rm -fr $(DESTDIR)$(datadir)/defaults/firejail', see #2038."
 
 # Note: Keep this list in sync with `paths` in .github/workflows/build.yml.
 DISTFILES = \
diff --git a/config.mk.in b/config.mk.in
index 937d58a..7d4cffc 100644
--- a/config.mk.in
+++ b/config.mk.in
@@ -18,6 +18,7 @@ exec_prefix=@exec_prefix@
 bindir=@bindir@
 libdir=@libdir@
 datarootdir=@datarootdir@
+datadir=@datadir@
 docdir=@docdir@
 mandir=@mandir@
 localstatedir=@localstatedir@
@@ -91,7 +92,8 @@ COMMON_CFLAGS = \
 	-fstack-protector-all \
 	-DPREFIX='"$(prefix)"' -DSYSCONFDIR='"$(sysconfdir)/$(TARNAME)"' \
 	-DLIBDIR='"$(libdir)"' -DBINDIR='"$(bindir)"' \
-	-DVARDIR='"$(localstatedir)/lib/$(TARNAME)"'
+	-DVARDIR='"$(localstatedir)/lib/$(TARNAME)"' \
+	-DDATADIR='"$(datadir)"'
 
 PROG_CFLAGS = \
 	$(COMMON_CFLAGS) \
diff --git a/configure.ac b/configure.ac
index d9a9574..891957b 100644
--- a/configure.ac
+++ b/configure.ac
@@ -271,6 +271,7 @@ Compile options:
    install contrib scripts: $HAVE_CONTRIB_INSTALL
    prefix: $prefix
    sysconfdir: $sysconfdir
+   defaultsdir: $datadir/defaults/firejail
    Spectre compiler patch: $HAVE_SPECTRE
 
 Features:
diff --git a/src/firecfg/firecfg.h b/src/firecfg/firecfg.h
index 3dbbd7f..fe9565f 100644
--- a/src/firecfg/firecfg.h
+++ b/src/firecfg/firecfg.h
@@ -38,6 +38,7 @@
 #define MAX_BUF 4096
 
 // config files
+#define FIRECFG_DEFAULT_CFGFILE DATADIR "/defaults/firejail/firecfg.config"
 #define FIRECFG_CFGFILE SYSCONFDIR "/firecfg.config"
 #define FIRECFG_CONF_GLOB SYSCONFDIR "/firecfg.d/*.conf"
 
diff --git a/src/firecfg/main.c b/src/firecfg/main.c
index 997a3fd..241c01c 100644
--- a/src/firecfg/main.c
+++ b/src/firecfg/main.c
@@ -318,7 +318,10 @@ void parse_config_all(int do_symlink) {
 		return;
 
 	parse_config_glob(FIRECFG_CONF_GLOB, do_symlink);
-	parse_config_file(FIRECFG_CFGFILE, do_symlink);
+	if (access(FIRECFG_CFGFILE, F_OK) == 0)
+		parse_config_file(FIRECFG_CFGFILE, do_symlink);
+	else
+		parse_config_file(FIRECFG_DEFAULT_CFGFILE, do_symlink);
 
 	done_config = 1;
 }
diff --git a/src/firejail/appimage.c b/src/firejail/appimage.c
index 219fab8..7490294 100644
--- a/src/firejail/appimage.c
+++ b/src/firejail/appimage.c
@@ -54,9 +54,15 @@ int appimage_find_profile(const char *archive) {
 		printf("Looking for a %s profile\n", arc);
 
 	// try to match the name of the archive with the list of programs in /etc/firejail/firecfg.config
-	FILE *fp = fopen(SYSCONFDIR "/firecfg.config", "r");
+	const char *fname;
+	if (access(SYSCONFDIR "/firecfg.config", F_OK) == 0)
+		fname = SYSCONFDIR "/firecfg.config";
+	else
+		fname = DATADIR "/defaults/firejail/firecfg.config";
+	FILE *fp = fopen(fname, "r");
 	if (!fp) {
-		fprintf(stderr, "Error: cannot find %s, firejail is not correctly installed\n", SYSCONFDIR "/firecfg.config");
+		fprintf(stderr, "Error: cannot find %s or %s, firejail is not correctly installed\n",
+			SYSCONFDIR "/firecfg.config", DATADIR "/defaults/firejail/firecfg.config");
 		exit(1);
 	}
 	char buf[MAXBUF];
diff --git a/src/firejail/checkcfg.c b/src/firejail/checkcfg.c
index 16c0818..ab62234 100644
--- a/src/firejail/checkcfg.c
+++ b/src/firejail/checkcfg.c
@@ -128,7 +128,11 @@ int checkcfg(int val) {
 		cfg_val[CFG_TRACELOG] = 0;
 
 		// open configuration file
-		const char *fname = SYSCONFDIR "/firejail.config";
+		const char *fname;
+		if (access(SYSCONFDIR "/firejail.config", F_OK) == 0)
+			fname = SYSCONFDIR "/firejail.config";
+		else
+			fname = DATADIR "/defaults/firejail/firejail.config";
 		fp = fopen(fname, "re");
 		if (!fp) {
 			fprintf(stderr, "Warning: Firejail configuration file %s not found, using defaults\n", fname);
diff --git a/src/firejail/profile.c b/src/firejail/profile.c
index 2f8118b..be4a912 100644
--- a/src/firejail/profile.c
+++ b/src/firejail/profile.c
@@ -82,8 +82,8 @@ int profile_find_firejail(const char *name, int add_ext) {
 	free(usercfgdir);
 
 	if (!rv)
-		// look for a user profile in /etc/firejail directory
-		rv = profile_find(name, SYSCONFDIR, add_ext);
+		// look for a user profile in /usr/share/defaults/firejail directory
+		rv = profile_find(name, DATADIR "/defaults/firejail", add_ext);
 
 	return rv;
 #else
diff --git a/src/firejail/restricted_shell.c b/src/firejail/restricted_shell.c
index d8d6dff..e4da400 100644
--- a/src/firejail/restricted_shell.c
+++ b/src/firejail/restricted_shell.c
@@ -34,8 +34,12 @@ int restricted_shell(const char *user) {
 		errExit("asprintf");
 	FILE *fp = fopen(fname, "re");
 	free(fname);
-	if (fp == NULL)
-		return 0;
+	if (fp == NULL) {
+		// fall back to the stateless default
+		fp = fopen(DATADIR "/defaults/firejail/login.users", "re");
+		if (fp == NULL)
+			return 0;
+	}
 
 	int lineno = 0;
 	char buf[MAX_READ];
