--- etc/apparmor/firejail-default.orig	2021-11-18 15:38:03.382294023 +0100
+++ etc/apparmor/firejail-default	2021-11-18 15:38:17.698115051 +0100
@@ -4,7 +4,7 @@
 
 # AppArmor 3.0 uses the @{run} variable in <abstractions/dbus-strict>
 # and <abstractions/dbus-session-strict>.
-#include <tunables/global>
+include if exists <tunables/global>
 
 ##########
 # A simple PID declaration based on Ubuntu's @{pid}
