From 3740f376487dd71c93c09d7df914326247608bf3 Mon Sep 17 00:00:00 2001
From: glitsj16 <glitsj16@users.noreply.github.com>
Date: Sat, 14 Dec 2019 02:08:03 +0000
Subject: [PATCH] Fix --appimage on linux 5.4.x kernel

Fixes #3068.
---
 src/firejail/appimage.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/firejail/appimage.c b/src/firejail/appimage.c
index e886e81da..520960db2 100644
--- a/src/firejail/appimage.c
+++ b/src/firejail/appimage.c
@@ -113,12 +113,12 @@ void appimage_set(const char *appimage) {
 	EUID_ROOT();
 	if (size == 0) {
 		fmessage("Mounting appimage type 1\n");
-		if (mount(devloop, mntdir, "iso9660", flags,  mode) < 0)
+		if (mount(devloop, mntdir, "iso9660", flags, mode) < 0)
 			errExit("mounting appimage");
 	}
 	else {
 		fmessage("Mounting appimage type 2\n");
-		if (mount(devloop, mntdir, "squashfs", flags,  mode) < 0)
+		if (mount(devloop, mntdir, "squashfs", flags, NULL) < 0)
 			errExit("mounting appimage");
 	}
 
