From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Evan Maddock <maddock.evan@vivaldi.net>
Date: Mon, 6 Oct 2025 14:45:39 -0400
Subject: [PATCH 2/3] Fix apparmor profile

Signed-off-by: Evan Maddock <maddock.evan@vivaldi.net>
---
 etc/apparmor/firejail-default | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/etc/apparmor/firejail-default b/etc/apparmor/firejail-default
index 3cc771ed7..eb33adcfb 100644
--- a/etc/apparmor/firejail-default
+++ b/etc/apparmor/firejail-default
@@ -4,7 +4,7 @@
 
 # AppArmor 3.0 uses the @{run} variable in <abstractions/dbus-strict>
 # and <abstractions/dbus-session-strict>.
-#include <tunables/global>
+include if exists <tunables/global>
 
 ##########
 # A simple PID declaration based on Ubuntu's @{pid}
