From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Evan Maddock <maddock.evan@vivaldi.net>
Date: Mon, 6 Oct 2025 14:54:41 -0400
Subject: [PATCH 3/3] Use stateless login.defs

Signed-off-by: Evan Maddock <maddock.evan@vivaldi.net>
---
 src/lib/firejail_user.c | 7 +++++--
 1 file changed, 5 insertions(+), 2 deletions(-)

diff --git a/src/lib/firejail_user.c b/src/lib/firejail_user.c
index 57fb0440e..e9aef1cfb 100644
--- a/src/lib/firejail_user.c
+++ b/src/lib/firejail_user.c
@@ -45,8 +45,11 @@ static void init_uid_gid_min(void) {
 	FILE *fp = fopen("/etc/login.defs", "r");
 	if (!fp) {
 		fp = fopen("/usr/etc/login.defs", "r"); // openSUSE
-		if (!fp)
-			goto errexit;
+		if (!fp) {
+			fp = fopen("/usr/share/defaults/etc/login.defs", "r"); // Solus
+			if (!fp)
+				goto errexit;
+		}
 	}
 
 	char buf[MAXBUF];
