--- a/xbmc/dbwrappers/mysqldataset.h    2019-01-12 12:18:33.000000000 -0500
+++ b/xbmc/dbwrappers/mysqldataset.h    2019-01-19 10:40:41.078632987 -0500
@@ -13,7 +13,7 @@
 #ifdef HAS_MYSQL
 #include "mysql/mysql.h"
 #elif defined(HAS_MARIADB)
-#include <mariadb/mysql.h>
+#include "mysql.h"
 #endif
 
 namespace dbiplus {
--- a/xbmc/dbwrappers/mysqldataset.cpp    2019-01-12 12:18:33.000000000 -0500
+++ b/xbmc/dbwrappers/mysqldataset.cpp    2019-01-19 10:54:52.888268028 -0500
@@ -20,7 +20,7 @@
 #ifdef HAS_MYSQL
 #include "mysql/errmsg.h"
 #elif defined(HAS_MARIADB)
-#include <mariadb/errmsg.h>
+#include "errmsg.h"
 #endif
 
 #ifdef TARGET_POSIX
