diff --git a/cmake/modules/FindCrossGUID.cmake b/cmake/modules/FindCrossGUID.cmake
index 38444f4..4098408 100644
--- a/cmake/modules/FindCrossGUID.cmake
+++ b/cmake/modules/FindCrossGUID.cmake
@@ -45,21 +45,21 @@ if(NOT TARGET CrossGUID::CrossGUID)
       set(CROSSGUID_VERSION ${PC_CROSSGUID_VERSION})
     endif()
 
-    find_path(CROSSGUID_INCLUDE_DIR NAMES crossguid/guid.hpp guid.h
+    find_path(CROSSGUID_INCLUDE_DIR NAMES guid.hpp guid.h
                                     HINTS ${DEPENDS_PATH}/include ${PC_CROSSGUID_INCLUDEDIR}
                                     ${${CORE_PLATFORM_LC}_SEARCH_CONFIG}
                                     NO_CACHE)
-    find_library(CROSSGUID_LIBRARY_RELEASE NAMES crossguid
+    find_library(CROSSGUID_LIBRARY_RELEASE NAMES xg
                                            HINTS ${DEPENDS_PATH}/lib ${PC_CROSSGUID_LIBDIR}
                                            ${${CORE_PLATFORM_LC}_SEARCH_CONFIG}
                                            NO_CACHE)
-    find_library(CROSSGUID_LIBRARY_DEBUG NAMES crossguidd crossguid-dgb
+    find_library(CROSSGUID_LIBRARY_DEBUG NAMES xg
                                          HINTS ${DEPENDS_PATH}/lib ${PC_CROSSGUID_LIBDIR}
                                          ${${CORE_PLATFORM_LC}_SEARCH_CONFIG}
                                          NO_CACHE)
 
     # NEW_CROSSGUID >= 0.2.0 release
-    if(EXISTS "${CROSSGUID_INCLUDE_DIR}/crossguid/guid.hpp")
+    if(EXISTS "${CROSSGUID_INCLUDE_DIR}/guid.hpp")
       list(APPEND _crossguid_definitions HAVE_NEW_CROSSGUID)
     endif()
   endif()
diff --git a/xbmc/utils/StringUtils.cpp b/xbmc/utils/StringUtils.cpp
index b74217f..92010d2 100644
--- a/xbmc/utils/StringUtils.cpp
+++ b/xbmc/utils/StringUtils.cpp
@@ -17,7 +17,7 @@
 //------------------------------------------------------------------------
 
 #ifdef HAVE_NEW_CROSSGUID
-#include <crossguid/guid.hpp>
+#include <guid.hpp>
 #else
 #include <guid.h>
 #endif
