From 8cf988cd4809f68fd54f53db035680d08137a0d0 Mon Sep 17 00:00:00 2001
From: Joey Riches <josephriches@gmail.com>
Date: Thu, 30 Sep 2021 14:59:41 +0100
Subject: [PATCH 1/1] Ensure SSE2 is the minimum baseline

---
 numpy/distutils/ccompiler_opt.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/numpy/distutils/ccompiler_opt.py b/numpy/distutils/ccompiler_opt.py
index e7fd494..d118f6f 100644
--- a/numpy/distutils/ccompiler_opt.py
+++ b/numpy/distutils/ccompiler_opt.py
@@ -218,7 +218,7 @@ class _Config:
     )
     conf_min_features = dict(
         x86 = "SSE SSE2",
-        x64 = "SSE SSE2 SSE3",
+        x64 = "SSE SSE2",
         ppc64 = '', # play it safe
         ppc64le = "VSX VSX2",
         armhf = '', # play it safe
-- 
2.32.0

