From 6302234b71ad97b200670a5b20fec34e87361a12 Mon Sep 17 00:00:00 2001
From: Pierre-Yves <pyu@riseup.net>
Date: Wed, 25 Jul 2018 18:40:13 +0200
Subject: [PATCH] Patch for quazip

Signed-off-by: Pierre-Yves <pyu@riseup.net>
---
 CVAssistant.pro         | 2 +-
 models/resumedocument.h | 2 +-
 2 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/CVAssistant.pro b/CVAssistant.pro
index c63b907..c7630ee 100644
--- a/CVAssistant.pro
+++ b/CVAssistant.pro
@@ -25,7 +25,7 @@ QT       += core gui xml
 
 unix {
     LIBS += -lz
-    LIBS += -lquazip-qt5
+    LIBS += -lquazip5
 }
 
 win32 {
diff --git a/models/resumedocument.h b/models/resumedocument.h
index 33c1b55..b6449f9 100644
--- a/models/resumedocument.h
+++ b/models/resumedocument.h
@@ -24,7 +24,7 @@
 #include "abstract/group.h"
 #include <QDir>
 #include <QList>
-#include <quazip/quazipfile.h>
+#include <quazip5/quazipfile.h>
 #include "../helpers/templator.h"
 
 class ResumeDocument
-- 
2.18.0

