From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Ikey Doherty <ikey@solus-project.com>
Date: Sat, 12 Mar 2016 14:49:00 +0000
Subject: [PATCH] Use a stateless configuration

Signed-off-by: Ikey Doherty <ikey@solus-project.com>
---
 Makefile.am     | 4 ++--
 bash_completion | 4 ++--
 2 files changed, 4 insertions(+), 4 deletions(-)

diff --git a/Makefile.am b/Makefile.am
index 27366b7d..61d864d5 100644
--- a/Makefile.am
+++ b/Makefile.am
@@ -3,10 +3,10 @@ SUBDIRS = completions doc helpers test
 pkgdata_DATA = bash_completion
 
 # Empty, but here just to get the compat dir created with install
-compatdir = $(sysconfdir)/bash_completion.d
+compatdir = $(datadir)/bash-completion/bash_completion.d
 compat_DATA = bash_completion.d/000_bash_completion_compat.bash
 
-profiledir = $(sysconfdir)/profile.d
+profiledir = $(datadir)/defaults/etc/profile.d
 profile_DATA = bash_completion.sh
 
 pkgconfigdir = $(datadir)/pkgconfig
diff --git a/bash_completion b/bash_completion
index 2bc27cc5..941ca734 100644
--- a/bash_completion
+++ b/bash_completion
@@ -3306,13 +3306,13 @@ _comp__init_compat_dirs=()
 if [[ ${BASH_COMPLETION_COMPAT_DIR-} ]]; then
     _comp__init_compat_dirs+=("$BASH_COMPLETION_COMPAT_DIR")
 else
-    _comp__init_compat_dirs+=(/etc/bash_completion.d)
+    _comp__init_compat_dirs+=(/usr/share/bash-completion/bash_completion.d)
     # Similarly as for the "completions" dir, look up from relative to
     # bash_completion, primarily for installed-with-prefix and
     # run-in-place-from-git-clone setups.  Notably we do it after the system
     # location here, in order to prefer in-tree variables and functions.
     if [[ ${BASH_SOURCE%/*} == */share/bash-completion ]]; then
-        _comp__init_compat_dir=${BASH_SOURCE%/share/bash-completion/*}/etc/bash_completion.d
+        _comp__init_compat_dir=/etc/bash_completion.d
     elif [[ $BASH_SOURCE == */* ]]; then
         _comp__init_compat_dir="${BASH_SOURCE%/*}/bash_completion.d"
     else
