From 89674cf739b85046fae6e86d1f4049e6588f6967 Mon Sep 17 00:00:00 2001
From: Ikey Doherty <ikey@solusos.com>
Date: Fri, 16 Aug 2013 21:05:09 +0100
Subject: [PATCH] Add support for extra bash-completions directory

---
 bash_completion |   13 ++++++++++++-
 1 file changed, 12 insertions(+), 1 deletion(-)

diff --git a/bash_completion b/bash_completion
index 5f07ed9..ff1cc2e 100644
--- a/bash_completion
+++ b/bash_completion
@@ -34,8 +34,9 @@
   # readonly variable errors.
   BASH_COMPLETION="${BASH_COMPLETION:-/etc/bash_completion}"
   BASH_COMPLETION_DIR="${BASH_COMPLETION_DIR:=/etc/bash_completion.d}"
+  BASH_COMPLETION_EXTRAS_DIR="${BASH_COMPLETION_EXTRAS_DIR:=/usr/share/bash-completion/completions}"
 } 2>/dev/null || :
-readonly BASH_COMPLETION BASH_COMPLETION_DIR
+readonly BASH_COMPLETION BASH_COMPLETION_DIR BASH_COMPLETION_EXTRAS_DIR
 
 # Set a couple of useful vars
 #
@@ -9287,6 +9288,16 @@ if [ -d $BASH_COMPLETION_DIR -a -r $BASH_COMPLETION_DIR -a \
 fi
 unset i
 
+# source extra completions directory definitions
+if [ -d $BASH_COMPLETION_EXTRAS_DIR -a -r $BASH_COMPLETION_EXTRAS_DIR -a \
+     -x $BASH_COMPLETION_EXTRAS_DIR ]; then
+	for i in $BASH_COMPLETION_EXTRAS_DIR/*; do
+		[[ ${i##*/} != @(*~|*.bak|*.swp|\#*\#|*.dpkg*|.rpm*) ]] &&
+			[ \( -f $i -o -h $i \) -a -r $i ] && . $i
+	done
+fi
+unset i
+
 # source user completion file
 [ $BASH_COMPLETION != ~/.bash_completion -a -r ~/.bash_completion ] \
 	&& . ~/.bash_completion
-- 
1.7.10.4

