From cd5734b3ddc8dcbd8e40265434ab660df50c4544 Mon Sep 17 00:00:00 2001
From: Ikey Doherty <ikey@solus-project.com>
Date: Sat, 12 Mar 2016 14:49:00 +0000
Subject: [PATCH] Use a stateless configuration

Signed-off-by: Ikey Doherty <ikey@solus-project.com>
---
 Makefile.am | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/Makefile.am b/Makefile.am
index dbbba0c..5500a88 100644
--- a/Makefile.am
+++ b/Makefile.am
@@ -5,7 +5,7 @@ pkgdata_DATA = bash_completion
 # Empty, but here just to get the compat dir created with install
 compat_DATA =
 
-profiledir = $(sysconfdir)/profile.d
+profiledir = $(datadir)/defaults/etc/profile.d
 profile_DATA = bash_completion.sh
 
 pkgconfigdir = $(datadir)/pkgconfig
-- 
2.7.1

