From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Reilly Brogan <reilly@reillybrogan.com>
Date: Wed, 13 Mar 2024 13:39:43 -0500
Subject: [PATCH] Fix Wayland appId

---
 src/frontend/qt_sdl/main.cpp | 1 +
 1 file changed, 1 insertion(+)

diff --git a/src/frontend/qt_sdl/main.cpp b/src/frontend/qt_sdl/main.cpp
index e8e30a7e..1b5c6b50 100644
--- a/src/frontend/qt_sdl/main.cpp
+++ b/src/frontend/qt_sdl/main.cpp
@@ -3226,6 +3226,7 @@ void emuStop()
 MelonApplication::MelonApplication(int& argc, char** argv)
     : QApplication(argc, argv)
 {
+    setDesktopFileName(QStringLiteral("net.kuribo64.melonDS"));
     setWindowIcon(QIcon(":/melon-icon"));
 }
 
