From 4e351954af3ddc9730222a88f1e09a6e15be303f Mon Sep 17 00:00:00 2001
From: Pierre-Yves <pyu@riseup.net>
Date: Sat, 26 Jan 2019 21:40:02 +0100
Subject: [PATCH] Disable test_filter_ipc

Signed-off-by: Pierre-Yves <pyu@riseup.net>
---
 Makefile.am | 3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

diff --git a/Makefile.am b/Makefile.am
index 279fb0c..72f507f 100644
--- a/Makefile.am
+++ b/Makefile.am
@@ -789,8 +789,7 @@ test_apps += \
 	tests/test_reqrep_ipc \
 	tests/test_use_fd \
 	tests/test_zmq_poll_fd \
-	tests/test_timeo \
-	tests/test_filter_ipc
+	tests/test_timeo
 
 tests_test_shutdown_stress_SOURCES = tests/test_shutdown_stress.cpp
 tests_test_shutdown_stress_LDADD = src/libzmq.la ${UNITY_LIBS}
-- 
2.20.1

