From 859f54f80df8690a46c6541f5de13268dccd2624 Mon Sep 17 00:00:00 2001
From: Marc Balmer <mbalmer@localhost.localdomain>
Date: Mon, 31 May 2021 10:23:01 +0200
Subject: [PATCH] If OPENPGM_FOUND is set, linke to libpgm.

---
 CMakeLists.txt | 12 ++++++++++++
 1 file changed, 12 insertions(+)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index b3467b1aa..50d21eeb7 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -1440,6 +1440,10 @@ if(BUILD_SHARED)
     target_link_libraries(libzmq ${LIBBSD_LIBRARIES})
   endif()
 
+  if(OPENPGM_FOUND)
+    target_link_libraries(libzmq ${OPENPGM_LIBRARIES})
+  endif()
+
   if(SODIUM_FOUND)
     target_link_libraries(libzmq ${SODIUM_LIBRARIES})
     # On Solaris, libsodium depends on libssp
@@ -1485,6 +1489,10 @@ if(BUILD_STATIC)
     target_link_libraries(libzmq-static ${NSS3_LIBRARIES})
   endif()
 
+  if(OPENPGM_FOUND)
+    target_link_libraries(libzmq-static pgm ${OPENPGM_LIBRARIES})
+  endif()
+
   if(SODIUM_FOUND)
     target_link_libraries(libzmq-static ${SODIUM_LIBRARIES})
     # On Solaris, libsodium depends on libssp
@@ -1553,6 +1561,10 @@ if(BUILD_SHARED)
         target_link_libraries(${perf-tool} ${NSS3_LIBRARIES})
       endif()
 
+      if(OPENPGM_FOUND)
+        target_link_libraries(${perf-tool} ${OPENPGM_LIBRARIES})
+      endif()
+
       if(SODIUM_FOUND)
         target_link_libraries(${perf-tool} ${SODIUM_LIBRARIES})
       endif()
