@@ -, +, @@ 
 pango/break.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)
--- a/pango/break.c	
+++ a/pango/break.c	
@@ -1163,7 +1163,8 @@ pango_default_break (const gchar   *text,
 	   * except where a line break is prohibited, which means we
 	   * effectively break everywhere except inside runs of spaces.
 	   */
-	  attrs[i].is_char_break = TRUE;
+	  if (attrs[i].is_cursor_position)
+	    attrs[i].is_char_break = TRUE;
 
 	  /* Make any necessary replacements first */
 	  if (row_break_type == G_UNICODE_BREAK_UNKNOWN)
@@ -1458,7 +1459,6 @@ pango_default_break (const gchar   *text,
 	    {
 	    case BREAK_PROHIBITED:
 	      /* can't break here */
-	      attrs[i].is_char_break = FALSE;
 	      attrs[i].is_line_break = FALSE;
 	      break;
 
