From 45a3cca96f8db8e09fb0ed0f8bab2acccd353ad7 Mon Sep 17 00:00:00 2001
From: Fabio Forni <livingsilver94.solus@redaril.me>
Date: Mon, 18 Nov 2019 14:44:15 +0100
Subject: [PATCH] Set the correct terminal

---
 src/keyfile.c | 18 +++++++++---------
 1 file changed, 9 insertions(+), 9 deletions(-)

diff --git a/src/keyfile.c b/src/keyfile.c
index 05373af..c5a5a5c 100644
--- a/src/keyfile.c
+++ b/src/keyfile.c
@@ -73,13 +73,6 @@
 #define GEANY_MSGWIN_HEIGHT				208
 #define GEANY_DISK_CHECK_TIMEOUT		30
 #define GEANY_DEFAULT_TOOLS_MAKE		"make"
-#ifdef G_OS_WIN32
-#define GEANY_DEFAULT_TOOLS_TERMINAL	"cmd.exe /Q /C %c"
-#elif defined(__APPLE__)
-#define GEANY_DEFAULT_TOOLS_TERMINAL	"open -a terminal %c"
-#else
-#define GEANY_DEFAULT_TOOLS_TERMINAL	"xterm -e \"/bin/sh %c\""
-#endif
 #ifdef __APPLE__
 #define GEANY_DEFAULT_TOOLS_BROWSER		"open -a safari"
 #define GEANY_DEFAULT_FONT_SYMBOL_LIST	"Helvetica Medium 12"
@@ -943,8 +936,15 @@ static void load_dialog_prefs(GKeyFile *config)
 #endif
 			g_free(tmp_string);
 		}
-		else
-			SETPTR(cmd, g_strdup(GEANY_DEFAULT_TOOLS_TERMINAL));
+		else {
+			if (access("/usr/bin/x-terminal-emulator", X_OK) == 0) {
+				SETPTR(cmd, g_strdup("x-terminal-emulator -e \"/bin/sh %c\""));
+			} else if (access("/usr/bin/mate-terminal-emulator", X_OK) == 0) {
+				SETPTR(cmd, g_strdup("mate-terminal-emulator -e \"/bin/sh %c\""));
+			} else if  (access("/usr/bin/konsole", X_OK) == 0) {
+				SETPTR(cmd, g_strdup("konsole -e \"/bin/sh %c\""));
+			}
+		}
 	}
 	tool_prefs.term_cmd = cmd;
 	tool_prefs.browser_cmd = utils_get_setting_string(config, "tools", "browser_cmd", GEANY_DEFAULT_TOOLS_BROWSER);
-- 
2.23.0

