From 03fa144bde25387c396f71d283c12f364e733bca Mon Sep 17 00:00:00 2001
From: Reilly Brogan <reilly@reillybrogan.com>
Date: Thu, 15 Dec 2022 15:31:10 -0600
Subject: [PATCH] t/test-lib.sh: Don't clobber LD_PRELOAD if already set

---
 t/test-lib.sh | 18 ++++++++++++++++--
 1 file changed, 16 insertions(+), 2 deletions(-)

diff --git a/t/test-lib.sh b/t/test-lib.sh
index 6db377f68b..70b1b8b293 100644
--- a/t/test-lib.sh
+++ b/t/test-lib.sh
@@ -601,8 +601,15 @@ else
 		export MALLOC_CHECK_ MALLOC_PERTURB_
 		if test -n "$_USE_GLIBC_TUNABLES"
 		then
+			if ! test -z "$LD_PRELOAD"
+			then
+				GIT_OLD_LD_PRELOAD=$LD_PRELOAD
+				LD_PRELOAD="$LD_PRELOAD libc_malloc_debug.so.0"
+				export GIT_OLD_LD_PRELOAD
+			else
+				LD_PRELOAD="libc_malloc_debug.so.0"
+			fi
 			g=
-			LD_PRELOAD="libc_malloc_debug.so.0"
 			for t in \
 				glibc.malloc.check=1 \
 				glibc.malloc.perturb=165
@@ -615,7 +622,14 @@ else
 	}
 	teardown_malloc_check () {
 		unset MALLOC_CHECK_ MALLOC_PERTURB_
-		unset LD_PRELOAD GLIBC_TUNABLES
+		unset GLIBC_TUNABLES
+		if ! test -z "$GIT_OLD_LD_PRELOAD"
+		then
+			LD_PRELOAD=$GIT_OLD_LD_PRELOAD
+			unset GIT_OLD_LD_PRELOAD
+		else
+			unset LD_PRELOAD
+		fi
 	}
 fi
 
-- 
2.38.1

