From 3f951447ede9355363331168cb4e70d140bcb16a Mon Sep 17 00:00:00 2001
From: Milan Crha <mcrha@redhat.com>
Date: Fri, 26 Apr 2024 09:15:44 +0200
Subject: [PATCH 6/7] m365: Calendar: Fix a memory leak when loading event/task

---
 src/Microsoft365/calendar/e-cal-backend-m365.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/src/Microsoft365/calendar/e-cal-backend-m365.c b/src/Microsoft365/calendar/e-cal-backend-m365.c
index 7d081f22..c258201f 100644
--- a/src/Microsoft365/calendar/e-cal-backend-m365.c
+++ b/src/Microsoft365/calendar/e-cal-backend-m365.c
@@ -661,6 +661,8 @@ ecb_m365_load_component_sync (ECalMetaBackend *meta_backend,
 		}
 	}
 
+	g_clear_pointer (&item, json_object_unref);
+
 	UNLOCK (cbm365);
 
 	ecb_m365_convert_error_to_client_error (error);
-- 
2.45.0

