From 8bddb71cc29d6b29ce3d236c7f64ff4ebb35715c Mon Sep 17 00:00:00 2001
From: Evan Maddock <maddock.evan@vivaldi.net>
Date: Mon, 19 Dec 2022 16:33:41 -0500
Subject: [PATCH] Add install prefix and only compile nemo translations

Signed-off-by: Evan Maddock <maddock.evan@vivaldi.net>
---
 Makefile | 7 ++++---
 1 file changed, 4 insertions(+), 3 deletions(-)

diff --git a/Makefile b/Makefile
index aad8e780..2f5e4f90 100644
--- a/Makefile
+++ b/Makefile
@@ -1,3 +1,4 @@
+DESTDIR ?= /usr
 
 all: buildmo
 
@@ -5,10 +6,10 @@ buildmo:
 	@echo "Building the mo files"
 	# WARNING: the second sed below will only works correctly with the languages that don't contain "-"
 	for dir in `ls po-export`; do \
-		for file in `ls po-export/$$dir/*.po`; do \
+		for file in `ls po-export/$$dir/nemo*.po`; do \
 			lang=`echo $$file | sed 's/\.po$$//' | sed 's/.*\-//'`; \
-			install -d usr/share/locale/$$lang/LC_MESSAGES/; \
-			msgfmt -o usr/share/locale/$$lang/LC_MESSAGES/$$dir.mo $$file; \
+			install -d $(DESTDIR)/usr/share/locale/$$lang/LC_MESSAGES/; \
+			msgfmt -o $(DESTDIR)/usr/share/locale/$$lang/LC_MESSAGES/$$dir.mo $$file; \
 		done \
 	 done
 
-- 
2.39.0

