From d01e89f3b12b8ee2be9ee144a2f33373e2e64889 Mon Sep 17 00:00:00 2001
From: Ikey Doherty <ikey@solus-project.com>
Date: Sat, 19 Mar 2016 19:32:38 +0000
Subject: [PATCH] Only enable subpixel rendering when a specific file exists

Signed-off-by: Ikey Doherty <ikey@solus-project.com>
---
 src/truetype/ttobjs.c | 6 +++++-
 1 file changed, 5 insertions(+), 1 deletion(-)

diff --git a/src/truetype/ttobjs.c b/src/truetype/ttobjs.c
index 6060d6f..19d022f 100644
--- a/src/truetype/ttobjs.c
+++ b/src/truetype/ttobjs.c
@@ -1302,7 +1302,11 @@
     TT_Driver  driver = (TT_Driver)ttdriver;
 
 #ifdef TT_CONFIG_OPTION_SUBPIXEL_HINTING
-    driver->interpreter_version = TT_INTERPRETER_VERSION_38;
+    if (access("/usr/share/fonts/enable_subpixel_rendering", F_OK) == 0) {
+        driver->interpreter_version = TT_INTERPRETER_VERSION_38;
+    } else {
+        driver->interpreter_version = TT_INTERPRETER_VERSION_35;
+    }
 #else
     driver->interpreter_version = TT_INTERPRETER_VERSION_35;
 #endif
-- 
2.7.3

