From 05d2506a97f007d02e0923d0d311a95323c4632d Mon Sep 17 00:00:00 2001
From: Ikey Doherty <ikey@solus-project.com>
Date: Fri, 9 Dec 2016 02:03:33 +0000
Subject: [PATCH 2/3] config: Enabel subpixel rendering & infinality engine

Signed-off-by: Ikey Doherty <ikey@solus-project.com>
---
 include/freetype/config/ftoption.h | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/include/freetype/config/ftoption.h b/include/freetype/config/ftoption.h
index 12f47a8..dd47603 100644
--- a/include/freetype/config/ftoption.h
+++ b/include/freetype/config/ftoption.h
@@ -126,7 +126,7 @@ FT_BEGIN_HEADER
    * macro is not defined, FreeType offers alternative LCD rendering
    * technology that produces excellent output without LCD filtering.
    */
-/* #define FT_CONFIG_OPTION_SUBPIXEL_RENDERING */
+#define FT_CONFIG_OPTION_SUBPIXEL_RENDERING
 
 
   /**************************************************************************
@@ -658,8 +658,8 @@ FT_BEGIN_HEADER
    * https://www.microsoft.com/typography/cleartype/truetypecleartype.aspx
    */
 /* #define TT_CONFIG_OPTION_SUBPIXEL_HINTING  1         */
-#define TT_CONFIG_OPTION_SUBPIXEL_HINTING  2
-/* #define TT_CONFIG_OPTION_SUBPIXEL_HINTING  ( 1 | 2 ) */
+/* #define TT_CONFIG_OPTION_SUBPIXEL_HINTING  2 */
+#define TT_CONFIG_OPTION_SUBPIXEL_HINTING  ( 1 | 2 )
 
 
   /**************************************************************************
-- 
2.24.1

