From cd7482d48a08970812b1cb01615bf5922525d6dd Mon Sep 17 00:00:00 2001
From: Zach Bacon <zachbacon@vba-m.com>
Date: Sat, 7 Oct 2023 14:14:31 -0400
Subject: [PATCH] config: Enable subpixel rendering * infinality engine

Signed-off-by: Zach Bacon <zachbacon@vba-m.com>
---
 include/freetype/config/ftoption.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/include/freetype/config/ftoption.h b/include/freetype/config/ftoption.h
index 1976b33..0c8f68d 100644
--- a/include/freetype/config/ftoption.h
+++ b/include/freetype/config/ftoption.h
@@ -124,7 +124,7 @@ FT_BEGIN_HEADER
    * rendering technology that produces excellent output.
    */
 /* #define FT_CONFIG_OPTION_SUBPIXEL_RENDERING */
-
+#define FT_CONFIG_OPTION_SUBPIXEL_RENDERING
 
   /**************************************************************************
    *
-- 
2.42.0

