From c76be5334c8abbfdc78d50301779377e2c6908b8 Mon Sep 17 00:00:00 2001
From: Peter O'Connor <peter@solus-project.com>
Date: Wed, 4 Apr 2018 10:57:44 +1000
Subject: [PATCH] Disable building unshipped code

Signed-off-by: Peter O'Connor <peter@solus-project.com>
---
 CMakeLists.txt | 1 -
 1 file changed, 1 deletion(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index a155891..71de748 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -31,6 +31,5 @@ find_package(KF5 ${KF5_MIN_VERSION} REQUIRED COMPONENTS
 find_package(KF5Wayland CONFIG REQUIRED)
 
 add_subdirectory(src)
-add_subdirectory(autotests)
 
 feature_summary(WHAT ALL INCLUDE_QUIET_PACKAGES FATAL_ON_MISSING_REQUIRED_PACKAGES)
-- 
2.16.3

