From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Reilly Brogan <reilly@reillybrogan.com>
Date: Mon, 22 Jul 2024 15:44:40 -0500
Subject: [PATCH] ftbfs protobuf 26.0

---
 CMakeLists.txt           | 3 ++-
 src/MessageTypeStore.cpp | 4 +++-
 2 files changed, 5 insertions(+), 2 deletions(-)

diff --git a/src/MessageTypeStore.cpp b/src/MessageTypeStore.cpp
index 44c94ec..8c7fbfa 100644
--- a/src/MessageTypeStore.cpp
+++ b/src/MessageTypeStore.cpp
@@ -10,6 +10,8 @@
 #include <google/protobuf/compiler/importer.h>
 #include <google/protobuf/dynamic_message.h>
 
+#include <absl/strings/string_view.h>
+
 using namespace Arcus;
 
 /**
@@ -39,7 +41,7 @@ public:
     {
     }
 
-    void AddError(const std::string& filename, int line, int column, const std::string& message) override
+    void RecordError(absl::string_view filename, int line, int column, absl::string_view message) override
     {
         _stream << "[" << filename << " (" << line << "," << column << ")] " << message << std::endl;
         _error_count++;
