diff -up libArcus-4.10.0/cmake/FindSIP.cmake.sip libArcus-4.10.0/cmake/FindSIP.cmake
--- libArcus-4.10.0/cmake/FindSIP.cmake.sip
+++ libArcus-4.10.0/cmake/FindSIP.cmake
@@ -70,7 +70,7 @@ find_path(SIP_INCLUDE_DIRS sip.h
 )

 execute_process(
-    COMMAND ${Python3_EXECUTABLE} -c "import sip; print(sip.SIP_VERSION_STR)"
+    COMMAND ${Python3_EXECUTABLE} -c "import PyQt6.sip; print(PyQt6.sip.SIP_VERSION_STR)"
     RESULT_VARIABLE _process_status
     OUTPUT_VARIABLE _process_output
     OUTPUT_STRIP_TRAILING_WHITESPACE
