Index: pkg/version/version.go
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/pkg/version/version.go b/pkg/version/version.go
--- a/pkg/version/version.go	(revision b437c31f21478a16d1139af09aa31a8da6657cc7)
+++ b/pkg/version/version.go	(date 1784513134073)
@@ -17,7 +17,7 @@
 // This is set to the actual version by GoReleaser, identify by the
 // git tag assigned to the release. Versions built from source will
 // always show master.
-var Version = "master"
+var Version = "1.43.8"

 // Template for the version string.
 var Template = fmt.Sprintf("stripe version %s\n", Version)
