diff --git a/doc/iscsi-gen-initiatorname.8.template b/doc/iscsi-gen-initiatorname.8.template
index 97a6e21..2953c75 100644
--- a/doc/iscsi-gen-initiatorname.8.template
+++ b/doc/iscsi-gen-initiatorname.8.template
@@ -36,10 +36,10 @@ Force overwrite of existing initiator name, if present.
 .TP
 .BI [-p] \fIIQN-PREFIX\fP
 Use \fIIQN-PREFIX\fP as the prefix to the IQN generated,
-instead of the default of \fBiqn.1996-04.de.suse:01\fP.
+instead of the default of \fBiqn.2025-05.local.open-iscsi:solus\fP.
 .SH FILES
 .TP
-@HOMEDIR@/initiatorname.iscsi
+/usr/share/defaults/etc/iscsi/initiatorname.iscsi or @HOMEDIR@/initiatorname.iscsi
 The file containing the initiator name. Do not edit manually.
 .SH "SEE ALSO"
 .BR iscsi-iname (8)
diff --git a/doc/iscsi-iname.8.template b/doc/iscsi-iname.8.template
index b2d1c79..d1f1017 100644
--- a/doc/iscsi-iname.8.template
+++ b/doc/iscsi-iname.8.template
@@ -14,7 +14,7 @@ generates a unique iSCSI node name on every invocation.
 Display help
 .TP
 .BI [-p=]\fIprefix\fP
-Use the prefix passed in instead of the default "@IQN_PREFIX@"
+Use the prefix passed in instead of the default "iqn.2025-05.local.open-iscsi:solus"
 
 .SH AUTHORS
 Open-iSCSI project <http://www.open-iscsi.com/>
diff --git a/doc/iscsiadm.8.template b/doc/iscsiadm.8.template
index 90e1f3f..2118e83 100644
--- a/doc/iscsiadm.8.template
+++ b/doc/iscsiadm.8.template
@@ -185,7 +185,7 @@ while running: the \fBHOMEDIR\fR and the \fBDBROOT\fR. The following
 describes them:
 .TP
 .B Home Directory
-The \fIhome directory\fR for open-iscsi is @HOMEDIR@. This is
+The \fIhome directory\fR for open-iscsi is /usr/share/defaults/etc/iscsi or @HOMEDIR@. This is
 where it keeps its configuration file (\fIiscsid.conf\fR) and its
 initiator name file (\fIinitiatorname.iscsi\fR).
 .TP
@@ -695,10 +695,10 @@ List all sessions in tree format:
 \fBsh#\fR iscsiadm \-\-mode session  \-\-print
 .SH FILES
 .TP
-@HOMEDIR@/iscsid.conf
+/usr/share/defaults/etc/iscsi/iscsid.conf or @HOMEDIR@/iscsid.conf
 The configuration file read by \fBiscsid\fR and \fBiscsiadm\fR on startup.
 .TP
-@HOMEDIR@/initiatorname.iscsi
+/usr/share/defaults/etc/iscsi/initiatorname.iscsi or @HOMEDIR@/initiatorname.iscsi
 The file containing the iSCSI InitiatorName and InitiatorAlias read by
 \fBiscsid\fR and \fBiscsiadm\fR on startup.
 .TP
diff --git a/doc/iscsid.8.template b/doc/iscsid.8.template
index 0d122a0..3debe54 100644
--- a/doc/iscsid.8.template
+++ b/doc/iscsid.8.template
@@ -15,10 +15,14 @@ iSCSI database.
 .TP
 .BI [-c|--config=]\fIconfig\-file\fP
 Read configuration from \fIconfig\-file\fR rather than the default
+\fI/usr/share/defaults/etc/iscsi/iscsid.conf
+or
 \fI@HOMEDIR@/iscsid.conf\fR file.
 .TP
 .BI [-i|--initiatorname=]\fIiname\-file\fP
 Read initiator name from \fIiname\-file\fR rather than the default
+\fI/usr/share/defaults/etc/iscsi/initiatorname.iscsi
+or
 \fI@HOMEDIR@/initiatorname.iscsi\fR file.
 .TP
 .BI [-f|--foreground]
@@ -55,14 +59,14 @@ display version and exit.
 
 .SH FILES
 .TP
-@HOMEDIR@/iscsid.conf
+/usr/share/defaults/etc/iscsi/iscsid.conf or @HOMEDIR@/iscsid.conf
 The configuration file read by
 .B iscsid
 and
 .B iscsiadm
 on startup.
 .TP
-@HOMEDIR@/initiatorname.iscsi
+/usr/share/defaults/etc/iscsi/initiatorname.iscsi or @HOMEDIR@/initiatorname.iscsi
 The file containing the iSCSI initiatorname
 and initiatoralias read by
 .B iscsid
diff --git a/doc/iscsistart.8.template b/doc/iscsistart.8.template
index 62fae4d..45cd78d 100644
--- a/doc/iscsistart.8.template
+++ b/doc/iscsistart.8.template
@@ -14,6 +14,8 @@ sessions used for iSCSI root boot.
 .TP
 .BI [-c|--config=]\fIconfig\-file\fP
 Read configuration from \fIconfig\-file\fR rather than the default
+\fI/usr/share/defaults/etc/iscsi/iscsid.conf
+or
 \fI@HOMEDIR@/iscsid.conf\fR file.
 .TP
 .BI [-i|--initiatorname=]\fIname\fP
