diff --git a/Makefile.conf b/Makefile.conf
index 55d81c5..4df4869 100644
--- a/Makefile.conf
+++ b/Makefile.conf
@@ -45,11 +45,11 @@ endif
 # Installation locations (edit if required)
 
 ifneq ($(HOST),MINGW)
-  prefix=/usr/local
+  prefix=/usr
   bindir=$(prefix)/bin
   incdir=$(prefix)/include
   libdir=$(prefix)/lib
-  docdir=$(prefix)/doc/routino
+  docdir=$(prefix)/share/doc/routino
   datadir=$(prefix)/share/routino
 else
   prefix="c:/Program Files/Routino"
@@ -155,7 +155,7 @@ ifeq ($(HOST),UNIX)
 endif
 
 # Put the current directory in the shared library path for the router using libroutino
-LDFLAGS_LDSO=-Wl,-R.
+
 
 
 # Required for multi-threaded support (comment these two lines out if not required)
