From b816ac120cbb428cfa48a15793b4c5c5d4fc2cfd Mon Sep 17 00:00:00 2001
From: Philipp Trulson <philipp@trulson.de>
Date: Tue, 23 May 2017 12:33:12 +0200
Subject: [PATCH] Fix paths

---
 Makefile.conf | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/Makefile.conf b/Makefile.conf
index a1cfb24..d50b8be 100644
--- a/Makefile.conf
+++ b/Makefile.conf
@@ -45,11 +45,11 @@ endif
 # Installation locations (edit if required)
 
 ifneq ($(HOST),MINGW)
-prefix=/usr/local
+prefix=/usr
 bindir=$(prefix)/bin
 incdir=$(prefix)/include
 libdir=$(prefix)/lib
-docdir=$(prefix)/doc/routino
+docdir=$(prefix)/share/doc/routino
 datadir=$(prefix)/share/routino
 else
 prefix="c:/Program Files/Routino"
@@ -118,7 +118,7 @@ LDFLAGS_SLIM_SONAME=-Wl,-soname=libroutino-slim.so.$(SOVERSION)
 endif
 
 # Put the current directory in the shared library path for the router using libroutino
-LDFLAGS_LDSO=-Wl,-R.
+
 
 
 # Required for multi-threaded support (comment these two lines out if not required)
-- 
2.13.0

