From 36b96923260554f5acd4fdea4ae2db4f00e9a340 Mon Sep 17 00:00:00 2001
From: Peter O'Connor <peter@solus-project.com>
Date: Tue, 2 Oct 2018 13:35:04 +1000
Subject: [PATCH] builder: Export language with UTF-8 notation

Signed-off-by: Peter O'Connor <peter@solus-project.com>
---
 src/builder/util.go | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/builder/util.go b/src/builder/util.go
index a359556..2767381 100644
--- a/src/builder/util.go
+++ b/src/builder/util.go
@@ -150,8 +150,8 @@ func TouchFile(path string) error {
 func SaneEnvironment(username, home string) []string {
 	environment := []string{
 		"PATH=/usr/bin:/usr/sbin:/bin/:/sbin",
-		"LANG=en_US.utf8",
-		"LC_ALL=en_US.utf8",
+		"LANG=en_US.UTF-8",
+		"LC_ALL=en_US.UTF-8",
 		fmt.Sprintf("HOME=%s", home),
 		fmt.Sprintf("USER=%s", username),
 		fmt.Sprintf("USERNAME=%s", username),
-- 
2.19.0

