From cafffbbc31cb6fdbc46e2093a93f8a870af5b312 Mon Sep 17 00:00:00 2001
From: Reilly Brogan <reilly@reillybrogan.com>
Date: Wed, 10 Jan 2024 22:11:54 -0600
Subject: [PATCH] Fix build

Signed-off-by: Reilly Brogan <reilly@reillybrogan.com>
---
 builder/source/simple.go | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/builder/source/simple.go b/builder/source/simple.go
index dd4c17d..7c86997 100644
--- a/builder/source/simple.go
+++ b/builder/source/simple.go
@@ -146,7 +146,7 @@ func (s *SimpleSource) download(destination string) error {
 
 	finalURL := headResp.Request.URL.String()
 	if s.URI != finalURL {
-		slog.Info("Source URL redirected", "uri", finalURL)
+		log.Infof("Source URL redirected: %s\n", finalURL)
 	}
 
 	req, err := grab.NewRequest(destination, finalURL)
-- 
2.43.0

