diff --git a/s_tui/sources/rapl_read.py b/s_tui/sources/rapl_read.py
index c00e3fb..2dda76b 100644
--- a/s_tui/sources/rapl_read.py
+++ b/s_tui/sources/rapl_read.py
@@ -30,7 +30,7 @@ from s_tui.helper_functions import cat
 
 
 INTER_RAPL_DIR = "/sys/class/powercap/intel-rapl/"
-AMD_ENERGY_DIR_GLOB = "/sys/devices/platform/amd_energy.0/hwmon/hwmon*/"
+AMD_ENERGY_DIR_GLOB = "/sys/devices/platform/zenergy.0/hwmon/hwmon*/"
 MICRO_JOULE_IN_JOULE = 1000000.0
 
 
@@ -115,7 +115,7 @@ class AMDEnergyReader:
 
     @staticmethod
     def available():
-        return os.path.exists("/sys/devices/platform/amd_energy.0")
+        return os.path.exists("/sys/devices/platform/zenergy.0")
 
 
 class AMDRaplMsrReader:
