From 1aa400f5779881286a50ee2bf80b632325469de0 Mon Sep 17 00:00:00 2001
From: Pierre-Yves <pyu@riseup.net>
Date: Wed, 19 Jul 2017 18:53:58 +0200
Subject: [PATCH] opensc bash completion

Signed-off-by: Pierre-Yves <pyu@riseup.net>
---
 doc/tools/Makefile.am | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/doc/tools/Makefile.am b/doc/tools/Makefile.am
index cc2b944..159f7d1 100644
--- a/doc/tools/Makefile.am
+++ b/doc/tools/Makefile.am
@@ -13,7 +13,7 @@ man5_MANS = $(patsubst $(srcdir)/%.xml, %, $(wildcard $(srcdir)/*.5.xml))
 endif
 
 completion_DATA = $(patsubst $(srcdir)/%.1.xml, %, $(wildcard $(srcdir)/*.1.xml))
-completiondir = $(sysconfdir)/bash_completion.d
+completiondir = $(datarootdir)/bash-completion/completions
 
 tools.html: $(srcdir)/tools.xml $(wildcard $(srcdir)/*.1.xml) $(wildcard $(srcdir)/*.5.xml)
 	$(XSLTPROC) --nonet --path "$(srcdir)/..:$(xslstylesheetsdir)/html" --xinclude -o $@ html.xsl $<
-- 
2.13.3

