From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Muhammad Alfi Syahrin <malfisya.dev@hotmail.com>
Date: Fri, 13 Jun 2025 22:00:36 +0700
Subject: [PATCH] Use standard directory for resource files

---
 CMakeLists.txt   | 12 ++++++------
 source/Files.cpp |  2 +-
 2 files changed, 7 insertions(+), 7 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index aa7e886..e469de6 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -320,12 +320,12 @@ elseif(UNIX)
 	install(FILES ${CMAKE_CURRENT_BINARY_DIR}/endless-sky.6.gz DESTINATION share/man/man6)
 
 	# Install the resource files.
-	install(DIRECTORY data DESTINATION share/games/endless-sky)
-	install(DIRECTORY images DESTINATION share/games/endless-sky)
-	install(DIRECTORY shaders DESTINATION share/games/endless-sky)
-	install(DIRECTORY sounds DESTINATION share/games/endless-sky)
-	install(FILES credits.txt DESTINATION share/games/endless-sky)
-	install(FILES keys.txt DESTINATION share/games/endless-sky)
+	install(DIRECTORY data DESTINATION share/endless-sky)
+	install(DIRECTORY images DESTINATION share/endless-sky)
+	install(DIRECTORY shaders DESTINATION share/endless-sky)
+	install(DIRECTORY sounds DESTINATION share/endless-sky)
+	install(FILES credits.txt DESTINATION share/endless-sky)
+	install(FILES keys.txt DESTINATION share/endless-sky)
 	install(FILES copyright DESTINATION share/doc/endless-sky)
 	install(FILES changelog DESTINATION share/doc/endless-sky)
 	install(FILES license.txt DESTINATION share/doc/endless-sky)
diff --git a/source/Files.cpp b/source/Files.cpp
index bb03ecf..a00ebd0 100644
--- a/source/Files.cpp
+++ b/source/Files.cpp
@@ -128,7 +128,7 @@ void Files::Init(const char * const *argv)
 		// the end, so parent paths do not include it.
 		static const filesystem::path LOCAL_PATH = "/usr/local";
 		static const filesystem::path STANDARD_PATH = "/usr";
-		static const filesystem::path RESOURCE_PATH = "share/games/endless-sky/";
+		static const filesystem::path RESOURCE_PATH = "share/endless-sky/";
 
 		if(IsParent(LOCAL_PATH, resources))
 			resources = LOCAL_PATH / RESOURCE_PATH;
