--- a/CMakeLists.txt~	2023-02-19 15:37:56.000000000 -0600
+++ b/CMakeLists.txt	2023-02-21 15:25:06.378269708 -0600
@@ -230,11 +230,11 @@
 	install(FILES ${CMAKE_CURRENT_BINARY_DIR}/endless-sky.6.gz DESTINATION share/man/man6)
 
 	# Install the resource files.
-	install(DIRECTORY data DESTINATION share/games/endless-sky)
-	install(DIRECTORY images DESTINATION share/games/endless-sky)
-	install(DIRECTORY sounds DESTINATION share/games/endless-sky)
-	install(FILES credits.txt DESTINATION share/games/endless-sky)
-	install(FILES keys.txt DESTINATION share/games/endless-sky)
+	install(DIRECTORY data DESTINATION share/endless-sky)
+	install(DIRECTORY images DESTINATION share/endless-sky)
+	install(DIRECTORY sounds DESTINATION share/endless-sky)
+	install(FILES credits.txt DESTINATION share/endless-sky)
+	install(FILES keys.txt DESTINATION share/endless-sky)
 	install(FILES copyright DESTINATION share/doc/endless-sky)
 	install(FILES changelog DESTINATION share/doc/endless-sky)
 	install(FILES license.txt DESTINATION share/doc/endless-sky)
--- a/source/Files.cpp~	2025-01-25 16:21:31.000000000 -0600
+++ b/source/Files.cpp	2025-01-27 08:47:01.802746790 -0600
@@ -102,7 +102,7 @@
 		// the executable, but are under the same prefix (/usr or /usr/local).
 		static const filesystem::path LOCAL_PATH = "/usr/local/";
 		static const filesystem::path STANDARD_PATH = "/usr/";
-		static const filesystem::path RESOURCE_PATH = "share/games/endless-sky/";
+		static const filesystem::path RESOURCE_PATH = "share/endless-sky/";
 
 		const auto IsParent = [](const auto parent, const auto child) -> bool {
 			if(distance(child.begin(), child.end()) < distance(parent.begin(), parent.end()))
