From 5a8111f93818dc9a5b9b2134bb41c546d2d85c1a Mon Sep 17 00:00:00 2001
From: Thomas Staudinger <Staudi.Kaos@gmail.com>
Date: Sun, 4 Feb 2024 12:50:33 +0100
Subject: [PATCH] CMakeList.txt: Use FindPython3

Signed-off-by: Thomas Staudinger <Staudi.Kaos@gmail.com>
---
 CMakeLists.txt | 12 ++++--------
 1 file changed, 4 insertions(+), 8 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index c58c29d0..b23d4830 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -370,14 +370,10 @@ if(NOT CMAKE_CROSSCOMPILING)
 	add_subdirectory(tests)
 endif(NOT CMAKE_CROSSCOMPILING)
 
-include(FindPythonInterp)
-if(PYTHONINTERP_FOUND AND PYTHON_VERSION_MAJOR LESS 3)
-	message("Note: Hatari needs at least Python 3 ... ignoring older version")
-	unset(PYTHONINTERP_FOUND)
-endif()
+include(FindPython3)
 set(PYTHON_GTK_FOUND 0)
-if(PYTHONINTERP_FOUND)
-	execute_process(COMMAND ${PYTHON_EXECUTABLE} -c "\
+if(Python3_Interpreter_FOUND)
+	execute_process(COMMAND ${Python3_EXECUTABLE} -c "\
 import gi\n\
 gi.require_version('Gtk', '3.0')\n\
 from gi.repository import Gtk\n\
@@ -390,7 +386,7 @@ from gi.repository import GLib"
 		set(PYTHON_GTK_FOUND 1)
 		add_subdirectory(python-ui)
 	endif()
-endif(PYTHONINTERP_FOUND)
+endif(Python3_Interpreter_FOUND)
 
 if(UNIX AND NOT ENABLE_OSX_BUNDLE)
 	add_subdirectory(share)
-- 
2.43.0