From 0b596a5a8cb1efc23406a82ef897c1a4d3048e8f Mon Sep 17 00:00:00 2001
From: Joshua Strobl <joshua@streambits.io>
Date: Tue, 24 Sep 2019 11:50:09 +0300
Subject: [PATCH 1/1] Remove unnecessary ldconfig, change prefix while we are
 at it.

---
 source/Makefile | 4 +---
 1 file changed, 1 insertion(+), 3 deletions(-)

diff --git a/source/Makefile b/source/Makefile
index 776c067..0c59c78 100644
--- a/source/Makefile
+++ b/source/Makefile
@@ -21,7 +21,7 @@
 # Modify as required.
 #
 SUFFIX := $(shell uname -m | sed -e 's/^unknown/$//' -e 's/^i.86/$//' -e 's/^x86_64/$/64/')
-PREFIX ?= /usr/local
+PREFIX ?= /usr
 INCDIR ?= $(PREFIX)/include
 LIBDIR ?= $(PREFIX)/lib$(SUFFIX)
 
@@ -34,7 +34,6 @@ VERSION = $(MAJVERS).$(MINVERS)
 CPPFLAGS += -I. -D_REENTRANT -D_POSIX_PTHREAD_SEMANTICS
 CPPFLAGS += -DENABLE_VECTOR_MODE 
 CXXFLAGS += -fPIC -Wall -ffast-math -funroll-loops -O2
-CXXFLAGS += -march=native
 LDLFAGS += 
 LDLIBS +=
 
@@ -56,8 +56,8 @@ install:	$(ZITA-CONVOLVER_MIN)
 	install -d $(DESTDIR)$(LIBDIR)
 	install -m 644 $(ZITA-CONVOLVER_H) $(DESTDIR)$(INCDIR)
 	install -m 755 $(ZITA-CONVOLVER_MIN) $(DESTDIR)$(LIBDIR)
-	ldconfig
-	ln -sf $(ZITA-CONVOLVER_MIN) $(DESTDIR)$(LIBDIR)/$(ZITA-CONVOLVER_SO)
+	ln -sv $(ZITA-CONVOLVER_MIN) $(DESTDIR)$(LIBDIR)/$(ZITA-CONVOLVER_SO)
+	ln -sv $(ZITA-CONVOLVER_MIN) $(DESTDIR)$(LIBDIR)/$(ZITA-CONVOLVER_MAJ)
 
 uninstall:
 	rm -rf $(DESTDIR)$(INCDIR)/$(ZITA-CONVOLVER_H)
-- 
2.23.0

