From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Muhammad Alfi Syahrin <malfisya.dev@hotmail.com>
Date: Wed, 16 Apr 2025 10:37:47 +0700
Subject: [PATCH] solus: Use stable channel of rust toolchain

---
 app/rust-toolchain.toml | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/app/rust-toolchain.toml b/app/rust-toolchain.toml
index f9154a1d..fb47e7fa 100644
--- a/app/rust-toolchain.toml
+++ b/app/rust-toolchain.toml
@@ -1,3 +1,3 @@
 [toolchain]
-channel = "1.84.1"
+channel = "stable"
 targets = ["aarch64-linux-android", "armv7-linux-androideabi", "x86_64-linux-android"]
