From b7685a03548a09ad150200a8f7c48117c3e3c4fd Mon Sep 17 00:00:00 2001
From: Reilly Brogan <reilly@reillybrogan.com>
Date: Thu, 30 Nov 2023 23:08:05 -0600
Subject: [PATCH] Co-installable

Signed-off-by: Reilly Brogan <reilly@reillybrogan.com>
---
 meson.build | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/meson.build b/meson.build
index 7988f025..2ecd4faa 100644
--- a/meson.build
+++ b/meson.build
@@ -201,8 +201,8 @@ endif
 pkgconfig = import('pkgconfig')
 pkgconfig.generate(lib_wlr,
 	version: meson.project_version(),
-	filebase: meson.project_name(),
-	name: meson.project_name(),
-	description: 'Wayland compositor library',
+	filebase: meson.project_name() + '-' + soversion.to_string(),
+	name: meson.project_name() + '-' + soversion.to_string(),
+	description: 'Wayland compositor library - 0.16 compatibility',
 	variables: wlr_vars,
 )
-- 
2.43.0

