From b1cd30f09222001fd0f57aaa77df0f2e49e724d9 Mon Sep 17 00:00:00 2001
From: Reilly Brogan <reilly@reillybrogan.com>
Date: Sat, 22 Jul 2023 13:40:02 -0500
Subject: [PATCH] Linux: Fix Wayland window associations

---
 CMakeLists.txt | 3 ++-
 src/main.cpp   | 4 ++++
 2 files changed, 6 insertions(+), 1 deletion(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 499ca7b..ec02398 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -267,7 +267,8 @@ endif()
 
 target_compile_definitions(
   ${PROJECT_NAME}
-  PUBLIC APP_VERSION="${PROJECT_VERSION}${GIT_REV}"
+  PUBLIC APP_VERSION="${PROJECT_VERSION}${GIT_REV}" 
+         APP_ID="${APP_ID}"
          QT_DISABLE_DEPRECATED_BEFORE=0x050900 QSU_INCLUDE_MOC=1)
 
 # Enable useful compiler warnings (except on third-party code).
diff --git a/src/main.cpp b/src/main.cpp
index 78fe0b9..9e239f6 100644
--- a/src/main.cpp
+++ b/src/main.cpp
@@ -16,6 +16,10 @@ int main(int argc, char *argv[])
     app.setApplicationName("Notes");
     app.setApplicationVersion(APP_VERSION);
 
+#if defined(Q_OS_LINUX) && QT_VERSION >= QT_VERSION_CHECK(5, 7, 0)
+	app.setDesktopFileName(QString(APP_ID) + ".desktop");
+#endif
+
 #if QT_VERSION >= QT_VERSION_CHECK(5, 10, 0) && QT_VERSION < QT_VERSION_CHECK(6, 0, 0)
     app.setAttribute(Qt::AA_DisableWindowContextHelpButton);
 #endif
-- 
2.41.0

