From be23bcded2231df7d1a913a8b304be8b34333f4a Mon Sep 17 00:00:00 2001
From: Jarno Turkki <stalebrim@posteo.net>
Date: Wed, 9 Sep 2020 13:12:33 +0300
Subject: [PATCH 1/1] Use native window frame by default

---
 src/mainwindow.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/mainwindow.cpp b/src/mainwindow.cpp
index 5d43830..b5cbc86 100644
--- a/src/mainwindow.cpp
+++ b/src/mainwindow.cpp
@@ -724,7 +724,7 @@ void MainWindow::setupModelView()
  */
 void MainWindow::restoreStates()
 {
-    setUseNativeWindowFrame(m_settingsDatabase->value(QStringLiteral("useNativeWindowFrame"), false).toBool());
+    setUseNativeWindowFrame(m_settingsDatabase->value(QStringLiteral("useNativeWindowFrame"), true).toBool());
 
     if(m_settingsDatabase->value(QStringLiteral("windowGeometry"), "NULL") != "NULL")
         this->restoreGeometry(m_settingsDatabase->value(QStringLiteral("windowGeometry")).toByteArray());
-- 
2.27.0

