From 768db4d81d9b35557ae11aa5679276033cabc01a Mon Sep 17 00:00:00 2001
From: Jarno Turkki <stalebrim@posteo.net>
Date: Wed, 9 Sep 2020 13:33:24 +0300
Subject: [PATCH 1/1] Don't show update window

---
 src/mainwindow.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/mainwindow.cpp b/src/mainwindow.cpp
index b5cbc86..162d5c9 100644
--- a/src/mainwindow.cpp
+++ b/src/mainwindow.cpp
@@ -66,7 +66,7 @@ MainWindow::MainWindow(QWidget *parent) :
     m_isListViewScrollBarHidden(true),
     m_isContentModified(false),
     m_isOperationRunning(false),
-    m_dontShowUpdateWindow(false),
+    m_dontShowUpdateWindow(true),
     m_alwaysStayOnTop(false),
     m_useNativeWindowFrame(false)
 {
-- 
2.27.0

