From d67a8676e58afa7b66d698ea8e6a134bb78a03c3 Mon Sep 17 00:00:00 2001
From: Ikey Doherty <ikey@solus-project.com>
Date: Wed, 24 Jan 2018 05:21:48 +0000
Subject: [PATCH 1/2] lib: Fix nasty Python binding double unref

Signed-off-by: Ikey Doherty <ikey@solus-project.com>
---
 src/lib/manager-plugins.c | 8 ++++++--
 1 file changed, 6 insertions(+), 2 deletions(-)

diff --git a/src/lib/manager-plugins.c b/src/lib/manager-plugins.c
index a05ba28..61987d1 100644
--- a/src/lib/manager-plugins.c
+++ b/src/lib/manager-plugins.c
@@ -155,7 +155,7 @@ static gint ldm_manager_sort_by_priority(gconstpointer a, gconstpointer b)
  * if they can support it. The returned #GPtrArray will free all elements
  * when it itself is freed.
  *
- * Returns: (element-type Ldm.Provider) (transfer full): a list of all possible providers
+ * Returns: (element-type Ldm.Provider) (transfer container): a list of all possible providers
  */
 GPtrArray *ldm_manager_get_providers(LdmManager *self, LdmDevice *device)
 {
@@ -178,7 +178,11 @@ GPtrArray *ldm_manager_get_providers(LdmManager *self, LdmDevice *device)
                         continue;
                 }
 
-                g_ptr_array_add(ret, g_object_ref_sink(provider));
+                if (g_object_is_floating(provider)) {
+                        g_ptr_array_add(ret, g_object_ref_sink(provider));
+                } else {
+                        g_ptr_array_add(ret, provider);
+                }
         }
 
         g_ptr_array_sort(ret, ldm_manager_sort_by_priority);
-- 
2.16.1

