From a1bace38f3b52efb1065433843dfa7bc9c62eeaa Mon Sep 17 00:00:00 2001
From: Ikey Doherty <ikey@solus-project.com>
Date: Fri, 26 Jan 2018 15:49:10 +0000
Subject: [PATCH 2/2] cli: Fix weird offset

Signed-off-by: Ikey Doherty <ikey@solus-project.com>
---
 src/cli/status.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/cli/status.c b/src/cli/status.c
index 00e4982..1f5214b 100644
--- a/src/cli/status.c
+++ b/src/cli/status.c
@@ -192,7 +192,7 @@ static void print_non_gpu(LdmManager *manager, LdmDevice *device)
         for (guint i = 0; i < providers->len; i++) {
                 LdmProvider *provider = providers->pdata[i];
                 fprintf(stdout,
-                        "  \u2558 Provider %02u   : %s\n",
+                        "  \u2558 Provider %02u  : %s\n",
                         i + 1,
                         ldm_provider_get_package(provider));
         }
-- 
2.16.1

