diff -up alpine-2.21/alpine/alpine.c.me alpine-2.21/alpine/alpine.c
diff -up alpine-2.21/pith/smime.c.me alpine-2.21/pith/smime.c
--- alpine-2.21/pith/smime.c.me	2020-02-13 16:54:36.842373446 +0100
+++ alpine-2.21/pith/smime.c	2020-02-13 17:13:37.930022666 +0100
@@ -45,6 +45,7 @@ static char rcsid[] = "$Id: smime.c 1176
 #ifdef PASSFILE
 #include "../pith/imap.h"
 #endif /* PASSFILE */
+#include "../pith/options.h"
 
 #include <openssl/buffer.h>
 #include <openssl/x509v3.h>
