--- a/alpine/smime.c
+++ b/alpine/smime.c
@@ -395,9 +395,9 @@
 	int i;
 
 	bs = X509_get_serialNumber(cert);
-	if (bs->length <= (int)sizeof(long)){
+	if (ASN1_STRING_length(bs) <= (int)sizeof(long)){
 	   l = ASN1_INTEGER_get(bs);
-           if (bs->type == V_ASN1_NEG_INTEGER){
+           if (ASN1_STRING_type(bs) == V_ASN1_NEG_INTEGER){
 	      l = -l;
 	      neg="-";
 	   }
@@ -405,10 +405,10 @@
               neg="";
 	   snprintf(buf, sizeof(buf), " %s%lu (%s0x%lx)", neg, l, neg, l);
 	} else {
-	    snprintf(buf, sizeof(buf), "%s", bs->type == V_ASN1_NEG_INTEGER ? "(Negative)" : "");
-	    for (i = 0; i < bs->length; i++)
-		 snprintf(buf + strlen(buf), sizeof(buf) - strlen(buf), "%02x%s", bs->data[i],
-			i+1 == bs->length ? "" : ":");
+	    snprintf(buf, sizeof(buf), "%s", ASN1_STRING_type(bs) == V_ASN1_NEG_INTEGER ? "(Negative)" : "");
+	    for (i = 0; i < ASN1_STRING_length(bs); i++)
+		 snprintf(buf + strlen(buf), sizeof(buf) - strlen(buf), "%02x%s", ASN1_STRING_get0_data(bs)[i],
+			i+1 == ASN1_STRING_length(bs) ? "" : ":");
 	}
     }
     gf_puts(buf, spc);
