From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Reilly Brogan <reilly@reillybrogan.com>
Date: Mon, 6 May 2024 12:47:32 -0500
Subject: [PATCH] Hack back support for GPUs that don't support modifiers

Signed-off-by: Reilly Brogan <reilly@reillybrogan.com>
---
 src/wayland_backend.cpp | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/src/wayland_backend.cpp b/src/wayland_backend.cpp
index fccf339..c3d5e23 100644
--- a/src/wayland_backend.cpp
+++ b/src/wayland_backend.cpp
@@ -1281,7 +1281,9 @@ namespace gamescope
         else if ( m_FormatModifiers.contains( DRM_FORMAT_ARGB2101010 ) )
             u10BitFormat = VK_FORMAT_A2R10G10B10_UNORM_PACK32;
 
-        assert( u8BitFormat != VK_FORMAT_UNDEFINED );
+        // Hack to fallback to the previously static values if the GPU doesn't support modifiers
+        if ( u8BitFormat == VK_FORMAT_UNDEFINED )
+            u8BitFormat = VK_FORMAT_B8G8R8A8_UNORM;
 
         *pPrimaryPlaneFormat = u10BitFormat != VK_FORMAT_UNDEFINED ? u10BitFormat : u8BitFormat;
         *pOverlayPlaneFormat = u8BitFormat;
