From ec1675e13b0587c8b71e16f45932db3d064dfdf2 Mon Sep 17 00:00:00 2001
From: Joey Riches <josephriches@gmail.com>
Date: Sun, 11 Sep 2022 10:06:41 +0100
Subject: [PATCH 1/1] remove failing cmake test

---
 tests/hit/HIT.cmake | 29 -----------------------------
 1 file changed, 29 deletions(-)

diff --git a/tests/hit/HIT.cmake b/tests/hit/HIT.cmake
index 68c3df3..993fdc9 100755
--- a/tests/hit/HIT.cmake
+++ b/tests/hit/HIT.cmake
@@ -36,35 +36,6 @@ if(CMAKE_VERSION VERSION_GREATER_EQUAL 3.21.0)
 endif()
 message(STATUS "CMAKE HIP ARCHITECTURES: ${CMAKE_HIP_ARCHITECTURES}")
 
-# Identify the GPU Targets.
-# This is done due to limitation of rocm_agent_enumerator
-# While building test parallelly, rocm_agent_enumerator can fail and give out an empty target
-# That results in hipcc building the test for gfx803 (the default target)
-if(NOT DEFINED OFFLOAD_ARCH_STR AND EXISTS "${ROCM_PATH}/bin/rocm_agent_enumerator"
-   AND HIP_PLATFORM STREQUAL "amd" AND UNIX)
-    execute_process(COMMAND ${ROCM_PATH}/bin/rocm_agent_enumerator OUTPUT_VARIABLE HIP_GPU_ARCH
-        RESULT_VARIABLE ROCM_AGENT_ENUM_RESULT)
-    message(STATUS "ROCm Agent Enumurator Result: ${ROCM_AGENT_ENUM_RESULT}")
-    # Trim out gfx000
-    string(REPLACE "gfx000\n" "" HIP_GPU_ARCH ${HIP_GPU_ARCH})
-    if (NOT HIP_GPU_ARCH STREQUAL "")
-        string(LENGTH ${HIP_GPU_ARCH} HIP_GPU_ARCH_LEN)
-        # If string has more gfx target except gfx000
-        if(${HIP_GPU_ARCH_LEN} GREATER_EQUAL 1)
-            string(REGEX REPLACE "\n" ";" HIP_GPU_ARCH_LIST "${HIP_GPU_ARCH}")
-            set(OFFLOAD_ARCH_STR "")
-            foreach(_hip_gpu_arch ${HIP_GPU_ARCH_LIST})
-                set(OFFLOAD_ARCH_STR " ${OFFLOAD_ARCH_STR} --offload-arch=${_hip_gpu_arch} ")
-            endforeach()
-            message(STATUS "Using offload arch string: ${OFFLOAD_ARCH_STR}")
-        endif()
-    else()
-        message(STATUS "ROCm Agent Enumurator found no valid architectures")
-    endif()
-else()
-    message(STATUS "ROCm Agent Enumurator Not Found")
-endif()
-
 #-------------------------------------------------------------------------------
 # Helper macro to parse BUILD instructions
 macro(PARSE_BUILD_COMMAND _target _sources _hipcc_options _clang_options _nvcc_options _link_options _exclude_platforms _exclude_runtime _exclude_compiler _exclude_lib_type _depends _dir)
-- 
2.35.4

