From 3a9866730db35e3e3ad0eee62dc69720ccaf0b04 Mon Sep 17 00:00:00 2001
From: Janek van Oirschot <5994977+JanekvO@users.noreply.github.com>
Date: Tue, 27 Sep 2022 03:16:29 +0100
Subject: [PATCH] Prevent chomp return from overriding HIP_CLANG_TARGET triple
 information (#2958)

---
 bin/hipcc.pl | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/bin/hipcc.pl b/bin/hipcc.pl
index 06097fa0cd..76163d97b6 100755
--- a/bin/hipcc.pl
+++ b/bin/hipcc.pl
@@ -163,7 +163,7 @@ BEGIN
 
     # Figure out the target with which llvm is configured
     $HIP_CLANG_TARGET = `$HIPCC -print-target-triple`;
-    $HIP_CLANG_TARGET = chomp($HIP_CLANG_TARGET);
+    chomp($HIP_CLANG_TARGET);
 
     if (! defined $HIP_CLANG_INCLUDE_PATH) {
         $HIP_CLANG_INCLUDE_PATH = abs_path("$HIP_CLANG_PATH/../lib/clang/$HIP_CLANG_VERSION/include");
