From a2f8d2d498990cbde80599f75d5935f06504f178 Mon Sep 17 00:00:00 2001
From: Jacek Jagosz <jacek.jagosz@outlook.com>
Date: Mon, 1 May 2023 01:18:12 +0200
Subject: [PATCH] Fix hipcc.pl paths for Solus

---
 bin/hipcc.pl | 9 ++-------
 1 file changed, 2 insertions(+), 7 deletions(-)

diff --git a/bin/hipcc.pl b/bin/hipcc.pl
index 645ae62d..0543c1fb 100755
--- a/bin/hipcc.pl
+++ b/bin/hipcc.pl
@@ -612,7 +612,7 @@ if($HIP_PLATFORM eq "amd"){
             $targetsStr = $ENV{HCC_AMDGPU_TARGET};
         } elsif (not $isWindows) {
             # Else try using rocm_agent_enumerator
-            $ROCM_AGENT_ENUM = "${ROCM_PATH}/bin/rocm_agent_enumerator";
+            $ROCM_AGENT_ENUM = "/usr/bin/rocm_agent_enumerator";
             $targetsStr = `${ROCM_AGENT_ENUM} -t GPU`;
             $targetsStr =~ s/\n/,/g;
         }
2.40.1

