diff --git a/hip-config.cmake.in b/hip-config.cmake.in
index 7c4fe7f9..125851bf 100755
--- a/hip-config.cmake.in
+++ b/hip-config.cmake.in
@@ -213,7 +213,7 @@ if(HIP_COMPILER STREQUAL "clang")
         hip_add_interface_compile_flags(hip::device -fms-extensions -fms-compatibility)
         hip_add_interface_compile_flags(hip::device --hip-device-lib-path=\"${HIP_PATH}/lib/bitcode\")
       else()
-        hip_add_interface_compile_flags(hip::device --hip-device-lib-path=\"${AMD_DEVICE_LIBS_PREFIX}/lib\")
+        hip_add_interface_compile_flags(hip::device --hip-device-lib-path=\"${AMD_DEVICE_LIBS_PREFIX}/lib64/amdgcn/bitcode\")
       endif()
     endif()
   endif()
