From def39700c978e8da3e3df93c0c88703f903a744b Mon Sep 17 00:00:00 2001
From: Volker Krause <vkrause@kde.org>
Date: Thu, 23 Apr 2026 17:14:36 +0200
Subject: [PATCH] Add forward compatibility with KMime from KDE Frameworks

---
 CMakeLists.txt     | 6 +++++-
 src/CMakeLists.txt | 2 +-
 2 files changed, 6 insertions(+), 2 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index eda3f3fbd..6465c4657 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -132,7 +132,11 @@ set(QGPGME_NAME "QGpgmeQt6")
 find_package(${QGPGME_NAME} ${GPGME_REQUIRED_VERSION} CONFIG REQUIRED)
 
 find_package(KPim6Libkleo ${LIBKLEO_VERSION} CONFIG REQUIRED)
-find_package(KPim6Mime ${KMIME_WANT_VERSION} CONFIG REQUIRED)
+find_package(KF6Mime ${KF6_WANT_VERSION} CONFIG)
+if(NOT TARGET KF6::Mime)
+    find_package(KPim6Mime ${KMIME_WANT_VERSION} CONFIG REQUIRED)
+    add_library(KF6::Mime ALIAS KPim6::Mime)
+endif()
 find_package(KPim6IdentityManagementCore ${KIDENTITYMANAGEMENT_VERSION} CONFIG)
 find_package(KPim6MailTransport ${KMAILTRANSPORT_VERSION} CONFIG)
 find_package(KPim6AkonadiMime ${AKONADI_MIME_VERSION} CONFIG)
diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
index 34133a52d..ea8582922 100644
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -588,7 +588,7 @@ endif ()
 target_link_libraries(kleopatra_bin
   Gpgmepp
   KPim6::Libkleo
-  KPim6::Mime
+  KF6::Mime
   KPim6::MimeTreeParserWidgets
   KF6::Codecs
   KF6::CoreAddons
-- 
GitLab

