From 1ec128ac6b46a64000dd8bf2200f2d388380992e Mon Sep 17 00:00:00 2001
From: Thomas Staudinger <Staudi.Kaos@gmail.com>
Date: Tue, 23 May 2023 00:10:57 +0200
Subject: [PATCH] Use correct exception

Signed-off-by: Thomas Staudinger <Staudi.Kaos@gmail.com>
---
 moddb/utils.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/moddb/utils.py b/moddb/utils.py
index 7433d33..46d5fbb 100644
--- a/moddb/utils.py
+++ b/moddb/utils.py
@@ -150,7 +150,7 @@ def raise_for_status(response):
             LOGGER.error(response.request.url)
             LOGGER.error(response.request.body)
             raise ModdbException(text["text"])
-    except json.decoder.JSONDecodeError:
+    except requests.exceptions.JSONDecodeError:
         response.raise_for_status()
 
     if (
-- 
2.40.1

