From cf95f4609c911c2968b21ec71f68e9720efec715 Mon Sep 17 00:00:00 2001
From: Thomas Staudinger <Staudi.Kaos@gmail.com>
Date: Fri, 19 May 2023 02:23:57 +0200
Subject: [PATCH] Don't try to parse html response as JSON in error handling

Signed-off-by: Thomas Staudinger <Staudi.Kaos@gmail.com>
---
 moddb/utils.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/moddb/utils.py b/moddb/utils.py
index 7433d33..b532470 100644
--- a/moddb/utils.py
+++ b/moddb/utils.py
@@ -210,7 +210,7 @@ def request(req: requests.Request):
     prepped = prepare_request(req, SESSION)
     r = SESSION.send(prepped)
 
-    raise_for_status(r)
+    r.raise_for_status()
     return r
 
 
-- 
2.40.1

