From 24bae94a6654ab03e3b5cd02d0a763dd4f24956c Mon Sep 17 00:00:00 2001
From: Chris King <bunnyapocalypse@protonmail.com>
Date: Fri, 17 Feb 2023 22:52:40 +0100
Subject: [PATCH] allow newer versions of libraries

---
 setup.py | 10 +++++-----
 1 file changed, 5 insertions(+), 5 deletions(-)

diff --git a/setup.py b/setup.py
index e02927b..649a276 100644
--- a/setup.py
+++ b/setup.py
@@ -25,9 +25,9 @@
     long_description=readme,
     packages=find_packages(include=['moddb', 'moddb.*']),
     install_requires=[
-        'beautifulsoup4 == 4.6.3',
-        'requests>=2.20.0',
-        'toolz == 0.11.2',
-        'pyrate-limiter==2.8.1'
+        'beautifulsoup4 >= 4.6.3',
+        'requests >= 2.20.0',
+        'toolz >= 0.11.2',
+        'pyrate-limiter >= 2.8.1'
       ]
-    )
\ No newline at end of file
+    )
