From 0f900924d4a98e0de4931a903f2e5d0ecbf2afc4 Mon Sep 17 00:00:00 2001
From: Joshua Strobl <joshua@streambits.io>
Date: Tue, 15 Dec 2020 20:40:42 +0200
Subject: [PATCH 1/1] Ensure VMs with nvram are removable.

---
 src/libvirt-machine.vala | 7 -------
 1 file changed, 7 deletions(-)

diff --git a/src/libvirt-machine.vala b/src/libvirt-machine.vala
index fad51144..af991209 100644
--- a/src/libvirt-machine.vala
+++ b/src/libvirt-machine.vala
@@ -522,16 +522,9 @@ public override void delete (bool by_user = true) {
                to avoid blocking the UI */
             App.app.async_launcher.launch.begin ( () => {
                 try {
-                    // This undefines the domain, causing it to be transient if it was running
-#if FLATPAK
                     domain.delete (DomainDeleteFlags.SAVED_STATE |
                                    DomainDeleteFlags.SNAPSHOTS_METADATA |
                                    DomainDeleteFlags.REMOVE_NVRAM);
-#else
-                    // This can go away once libvirt-glib gets a new release https://gitlab.com/libvirt/libvirt-glib/-/issues/1
-                    domain.delete (DomainDeleteFlags.SAVED_STATE |
-                                   DomainDeleteFlags.SNAPSHOTS_METADATA);
-#endif
                 } catch (GLib.Error err) {
                     warning (err.message);
                 }
-- 
2.29.2

