From 72cbee107a82e52164f72cc66ae6b550bf9504c1 Mon Sep 17 00:00:00 2001
From: Joshua Strobl <joshua@streambits.io>
Date: Sat, 17 Apr 2021 04:41:34 +0300
Subject: [PATCH 5/5] Remove rest of libhandy

---
 src/app.vala    |  2 --
 src/meson.build | 16 ----------------
 2 files changed, 18 deletions(-)

diff --git a/src/app.vala b/src/app.vala
index 4764932b..8ae09d04 100644
--- a/src/app.vala
+++ b/src/app.vala
@@ -139,8 +139,6 @@ public App () {
     public override void startup () {
         base.startup ();
 
-        Hdy.init ();
-
         string [] args = {};
         unowned string [] args2 = args;
         Gtk.init (ref args2);
diff --git a/src/meson.build b/src/meson.build
index 5f546576..f95cec39 100644
--- a/src/meson.build
+++ b/src/meson.build
@@ -155,22 +155,6 @@ if spice.found ()
   ]
 endif
 
-libhandy = dependency ('libhandy-1', version: '>= 1.0.0', required: false)
-if not libhandy.found()
-  libhandy = subproject(
-    'libhandy',
-    default_options: [
-      'package_subdir=' + meson.project_name(),
-      'examples=false',
-      'glade_catalog=disabled',
-      'tests=false',
-    ]
-  )
-  dependencies += libhandy.get_variable('libhandy_vapi')
-else
-  dependencies += libhandy
-endif
-
 if get_option('rdp')
   vala_sources += ['rdp-display.vala']
 
-- 
2.32.0

