diff --git a/channels/client/CMakeLists.txt b/channels/client/CMakeLists.txt
index 049bc7b..e37dd84 100644
--- a/channels/client/CMakeLists.txt
+++ b/channels/client/CMakeLists.txt
@@ -24,7 +24,6 @@ set(${MODULE_PREFIX}_SRCS
 	${CMAKE_CURRENT_SOURCE_DIR}/addin.c
 	${CMAKE_CURRENT_SOURCE_DIR}/addin.h)
 
-list(REMOVE_DUPLICATES CHANNEL_STATIC_CLIENT_ENTRIES)
 
 foreach(STATIC_ENTRY ${CHANNEL_STATIC_CLIENT_ENTRIES})
 	foreach(STATIC_MODULE ${CHANNEL_STATIC_CLIENT_MODULES})
