--- a/src/plugins/renderers/rhi/CMakeLists.txt
+++ b/src/plugins/renderers/rhi/CMakeLists.txt
@@ -20,6 +20,8 @@ set(rhiRendererSources graphicshelpers/s
     textures/texture.cpp textures/texture_p.h
 )
 
+find_package(Qt6 REQUIRED COMPONENTS ShaderTools)
+
 #####################################################################
 ## Static library used for Unit Tests:
 #####################################################################
@@ -36,8 +38,6 @@ if(QT_FEATURE_private_tests)
         textures
     )
 
-    find_package(Qt6 REQUIRED COMPONENTS ShaderTools)
-
     target_link_libraries(RhiRendererLib
         PUBLIC
             Qt::3DCore
