From 66bb99ebed411902a50c4d648651a95d676c866b Mon Sep 17 00:00:00 2001
From: Silke Hofstra <silke@slxh.eu>
Date: Wed, 3 Jul 2024 15:20:16 +0200
Subject: [PATCH 04/11] Remove snapd when there are no missing snaps

---
 unsnap | 20 +++++++++++++-------
 1 file changed, 13 insertions(+), 7 deletions(-)

diff --git a/unsnap b/unsnap
index 18ade18..c1d666a 100755
--- a/unsnap
+++ b/unsnap
@@ -376,13 +376,19 @@ function run_scripts() {
         echo -e "${RED}ERROR:${NC} Failed to remove snaps" | tee -a "$LOGFILE"
         exit 14
     fi
-    # echo -e "${GREEN}INFO:${NC} Removing snapd daemon" | tee -a "$LOGFILE"
-    # if "$REMOVESNAPDSCRIPT"; then
-    #     echo -e "${GREEN}INFO:${NC} snapd removed" | tee -a "$LOGFILE"
-    # else
-    #     echo -e "${RED}ERROR:${NC} Failed to remove snapd" | tee -a "$LOGFILE"
-    #     exit 15
-    # fi
+
+    if [ -e "$MISSINGFLATPAK" ]; then
+        echo -e "${YELLOW}WARNING:${NC} the following snaps have no known alternative, will not remove snapd"
+        cat "$MISSINGFLATPAK"
+    else
+        echo -e "${GREEN}INFO:${NC} Removing snapd daemon" | tee -a "$LOGFILE"
+        if "$REMOVESNAPDSCRIPT"; then
+            echo -e "${GREEN}INFO:${NC} snapd removed" | tee -a "$LOGFILE"
+        else
+            echo -e "${RED}ERROR:${NC} Failed to remove snapd" | tee -a "$LOGFILE"
+            exit 15
+        fi
+    fi
 }
 
 function recommend_reboot() {
-- 
2.45.2

