From 97457d80e491dc28ab6f0a797f55c035645f3e66 Mon Sep 17 00:00:00 2001
From: Silke Hofstra <silke@slxh.eu>
Date: Wed, 3 Jul 2024 16:30:12 +0200
Subject: [PATCH 06/11] Load lists from /usr/share/unsnap

---
 unsnap | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/unsnap b/unsnap
index c1d666a..c776ca1 100755
--- a/unsnap
+++ b/unsnap
@@ -30,8 +30,8 @@ function setup_environment() {
     INSTALLEDSNAPLIST="$LOGDIR/allsnaps.txt"
     FILTEREDSNAPLIST="$LOGDIR/filteredsnaps.txt"
     MISSINGFLATPAK="$LOGDIR/missingflatpak.txt"
-    APPLIST="./applist.csv"
-    EXCLUDEDSNAPLIST="./excluded_snaps.txt"
+    APPLIST="/usr/share/unsnap/applist.csv"
+    EXCLUDEDSNAPLIST="/usr/share/unsnap/excluded_snaps.txt"
     BACKUPSCRIPT="$LOGDIR/00-backup"
     INSTALLFLATPACKSCRIPT="$LOGDIR/01-install-flatpak"
     ENABLEFLATHUBSRIPT="$LOGDIR/02-enable-flathub"
-- 
2.45.2

