From b6a1072c5e039bbd7f8f88af9acbd4bba6560957 Mon Sep 17 00:00:00 2001
From: Silke Hofstra <silke@slxh.eu>
Date: Wed, 3 Jul 2024 17:36:41 +0200
Subject: [PATCH 07/11] Fix grep errors when MISSINGFLATPAK is empty

---
 unsnap | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/unsnap b/unsnap
index c776ca1..b1431f6 100755
--- a/unsnap
+++ b/unsnap
@@ -203,7 +203,7 @@ function generate_remove_snaps_script() {
     do
         snap=$(echo "$p" | awk -F ',' '{ print $1}')
         # Check to make sure the snap isn't on the missing list.
-        if grep "$snap" "$MISSINGFLATPAK"; then
+        if [ -e "$MISSINGFLATPAK" ] && grep -q "$snap" "$MISSINGFLATPAK"; then
             echo -e "${YELLOW}WARNING:${NC} $snap snap has no equivalent, not candidate for removing" | tee -a "$LOGFILE"
         else
             echo -n "$snap " >> "$REMOVESNAPSSCRIPT"
-- 
2.45.2

