From 3350f4d3bc1abdd6fd6dc03e96094eb6d0c4c641 Mon Sep 17 00:00:00 2001
From: Silke Hofstra <silke@slxh.eu>
Date: Wed, 3 Jul 2024 17:36:56 +0200
Subject: [PATCH 08/11] Disable systemd units on snapd removal

---
 unsnap | 10 +++++++++-
 1 file changed, 9 insertions(+), 1 deletion(-)

diff --git a/unsnap b/unsnap
index b1431f6..4315d3d 100755
--- a/unsnap
+++ b/unsnap
@@ -224,7 +224,8 @@ function generate_remove_snapd_script() {
     # so we might be better off removing all references to sudo, and inform the user
     # to run the script as "root user or sudo as appropriate".
     echo -e "${GREEN}INFO:${NC} Generating snapd removal script in $REMOVESNAPDSCRIPT" | tee -a "$LOGFILE"
-    echo "$SHEBANG" > "$REMOVESNAPDSCRIPT" 
+    echo "$SHEBANG" > "$REMOVESNAPDSCRIPT"
+
     echo "echo Removing snapd" >> "$REMOVESNAPDSCRIPT" 
     case $PACKAGEMGR in
         apt)
@@ -248,6 +249,13 @@ function generate_remove_snapd_script() {
         *)
             echo -e "${RED}ERROR:${NC} Unable to generate snapd removal script. Unimplemented package manager." | tee -a "$LOGFILE"
     esac
+
+    echo "echo Disabling all systemd units" >> "$REMOVESNAPDSCRIPT"
+    for unit in snapd.{socket,service} snapd.apparmor.service /etc/systemd/system/snap-*.mount
+    do
+        echo "sudo systemctl disable --now $(basename "${unit//\\/\\\\}")" >> "$REMOVESNAPDSCRIPT"
+    done
+
     chmod +x "$REMOVESNAPDSCRIPT"
 }
 
-- 
2.45.2

