From 57a529d4c24bd23c25be14c8f2de3c6245ab2389 Mon Sep 17 00:00:00 2001
From: Silke Hofstra <silke@slxh.eu>
Date: Wed, 3 Jul 2024 18:06:41 +0200
Subject: [PATCH 09/11] Log to /var/log

---
 unsnap | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/unsnap b/unsnap
index 4315d3d..6ad515a 100755
--- a/unsnap
+++ b/unsnap
@@ -25,7 +25,7 @@ function display_warnings() {
 
 function setup_environment() {
     DATETIMESTAMP=$(date +%Y-%m-%d.%H.%M.%S)
-    LOGDIR="./log-$DATETIMESTAMP"
+    LOGDIR="/var/log/unsnap/$DATETIMESTAMP"
     LOGFILE="$LOGDIR/unsnap.log"
     INSTALLEDSNAPLIST="$LOGDIR/allsnaps.txt"
     FILTEREDSNAPLIST="$LOGDIR/filteredsnaps.txt"
@@ -45,7 +45,7 @@ function setup_environment() {
 }
 
 function create_logdir() {
-    mkdir "$LOGDIR"
+    mkdir -p "$LOGDIR"
 }
 
 function audit_snap() {
-- 
2.45.2

