From ddf6882e2e2b26d28861ef20dcc8dcff6d97bbd8 Mon Sep 17 00:00:00 2001
From: Silke Hofstra <silke@slxh.eu>
Date: Wed, 10 Jul 2024 16:24:54 +0200
Subject: [PATCH 10/11] Add unwrapped snap to PATH

---
 unsnap | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/unsnap b/unsnap
index 6ad515a..d4a4d6b 100755
--- a/unsnap
+++ b/unsnap
@@ -13,6 +13,8 @@ CYAN='\033[0;36m'
 WHITE='\033[0;37m'
 NC='\033[0m' # No Color
 
+# Add unwrapped snap to PATH.
+export PATH="/usr/lib64/snapd:${PATH}"
 
 function display_warnings() {
     echo -e "${YELLOW}WARNING:${NC} Care has been taken to ensure this script is safe."
-- 
2.45.2

