From e522db6b2d425e661ff7b4fb88aa7cc4e212995a Mon Sep 17 00:00:00 2001
From: Silke Hofstra <silke@slxh.eu>
Date: Thu, 11 Jul 2024 21:33:56 +0200
Subject: [PATCH 11/11] Extend usage info when running without arguments

---
 unsnap | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/unsnap b/unsnap
index d4a4d6b..330e12e 100755
--- a/unsnap
+++ b/unsnap
@@ -413,6 +413,11 @@ if [ "$1" == "check" ]; then
     echo -e "${GREEN}INFO:${NC} Check requested" | tee -a "$LOGFILE"
     check_applist
 else
+    if [ $# -eq 0 ]; then
+        echo -e "${GREEN}INFO:${NC} running \`unsnap\` without arguments generates removal scripts in $LOGDIR"
+        echo -e "Run \`unsnap auto\` to automatically execute those scripts afterwards."
+        echo -e "See https://help.getsol.us/docs/user/software/third-party/snap for more information.\n"
+    fi
     display_warnings
     create_logdir
 fi
-- 
2.45.2

