From 7f04f066a18b3f398f292846c4fc8cc837c678a3 Mon Sep 17 00:00:00 2001
From: David Mohammed <fossfreedom@users.noreply.github.com>
Date: Thu, 7 Mar 2019 14:56:00 +0000
Subject: [PATCH 1/1] Fix FTBFS with vala 0.43 (#131)

---
 src/list-stack.vala | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/list-stack.vala b/src/list-stack.vala
index 89e1171..83bbeb3 100644
--- a/src/list-stack.vala
+++ b/src/list-stack.vala
@@ -72,7 +72,7 @@ public class ListStack : Gtk.Fixed
 
         return_if_fail (children != null);
 
-        unowned List<Gtk.Widget> prev = children.last ().prev;
+        unowned List<weak Gtk.Widget> prev = children.last ().prev;
         if (prev != null)
             (prev.data as GreeterList).greeter_box.pop ();
     }
-- 
2.21.0

