From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Evan Maddock <maddock.evan@vivaldi.net>
Date: Sat, 13 Dec 2025 12:47:01 -0500
Subject: [PATCH] Lessen opacity of dash-box

Signed-off-by: Evan Maddock <maddock.evan@vivaldi.net>
---
 src/dash-box.vala | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/dash-box.vala b/src/dash-box.vala
index ac7397d..2e4eb3d 100644
--- a/src/dash-box.vala
+++ b/src/dash-box.vala
@@ -311,9 +311,9 @@ public class DashBox : Gtk.Box
         /* Draw darker background with a rounded border */
         c.save ();
         CairoUtils.rounded_rectangle (c, 0, box_y, box_w, box_h, box_r);
-        c.set_source_rgba (0.1, 0.1, 0.1, 0.55);
+        c.set_source_rgba (0.1, 0.1, 0.1, 0.9);
         c.fill_preserve ();
-        c.set_source_rgba (0.4, 0.4, 0.4, 0.4);
+        c.set_source_rgba (0.4, 0.4, 0.4, 0.9);
         c.set_line_width (1);
         c.stroke ();
         c.restore ();
