From 4c90491eb2ad482e8008a8420c1e405dafdfe45a Mon Sep 17 00:00:00 2001
From: Leigh Scott <leigh123linux@gmail.com>
Date: Fri, 2 Aug 2019 08:16:15 +0100
Subject: [PATCH] Fix vala-0.45 compile error (#138)

---
 src/greeter-list.vala | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/greeter-list.vala b/src/greeter-list.vala
index b363b00..cdbbb02 100644
--- a/src/greeter-list.vala
+++ b/src/greeter-list.vala
@@ -164,7 +164,7 @@ public abstract class GreeterList : FadableBox
         }
     }
 
-    public GreeterList (Background bg, MenuBar mb)
+    protected GreeterList (Background bg, MenuBar mb)
     {
         Object (background: bg, menubar: mb);
     }
