From 34c95f22c5f0267f47c5a18edc0859370ed6439e Mon Sep 17 00:00:00 2001
From: Federico Mena Quintero <federico@gnome.org>
Date: Wed, 6 Nov 2019 17:13:15 -0600
Subject: [PATCH] croco.rs: Add struct definition for CRSimpleSel

---
 rsvg_internals/src/croco.rs | 19 ++++++++++++++++---
 1 file changed, 16 insertions(+), 3 deletions(-)

diff --git a/rsvg_internals/src/croco.rs b/rsvg_internals/src/croco.rs
index 892aca9b..21ac8051 100644
--- a/rsvg_internals/src/croco.rs
+++ b/rsvg_internals/src/croco.rs
@@ -6,9 +6,9 @@ use libc;
 
 // Opaque types from libcroco, or those which we only manipulate through libcroco functions
 pub type CRString = gpointer;
-pub type CRSimpleSel = gpointer;
 pub type CRParser = gpointer;
 pub type CRTerm = gpointer;
+pub type CRAdditionalSel = gpointer;
 
 pub type CRStatus = u32;
 
@@ -22,9 +22,22 @@ pub struct CRParsingLocation {
     pub byte_offset: libc::c_uint,
 }
 
+#[repr(C)]
+pub struct CRSimpleSel {
+    pub type_mask: libc::c_int,
+    pub is_case_sentive: gboolean,
+    pub name: CRString,
+    pub combinator: libc::c_int,
+    pub add_sel: CRAdditionalSel,
+    pub specificity: libc::c_ulong,
+    pub next: *mut CRSimpleSel,
+    pub prev: *mut CRSimpleSel,
+    pub location: CRParsingLocation,
+}
+
 #[repr(C)]
 pub struct CRSelector {
-    pub simple_sel: CRSimpleSel,
+    pub simple_sel: *mut CRSimpleSel,
 
     pub next: *mut CRSelector,
     pub prev: *mut CRSelector,
@@ -91,7 +104,7 @@ extern "C" {
     pub fn cr_selector_ref(a_this: *mut CRSelector);
     pub fn cr_selector_unref(a_this: *mut CRSelector) -> gboolean;
 
-    pub fn cr_simple_sel_to_string(a_this: CRSimpleSel) -> *mut libc::c_char;
+    pub fn cr_simple_sel_to_string(a_this: *mut CRSimpleSel) -> *mut libc::c_char;
 
     pub fn cr_string_peek_raw_str(a_this: CRString) -> *const libc::c_char;
 
-- 
2.22.0

