From de5fecc5d98aa4d85514947053cd9976d5b10d09 Mon Sep 17 00:00:00 2001
From: Pierre-Yves <pyu@riseup.net>
Date: Sat, 23 Feb 2019 20:35:41 +0100
Subject: [PATCH] Disable tests 115, 217 and 223

Signed-off-by: Pierre-Yves <pyu@riseup.net>
---
 tests/testsuite | 3 ---
 1 file changed, 3 deletions(-)

diff --git a/tests/testsuite b/tests/testsuite
index a349094..48039c7 100755
--- a/tests/testsuite
+++ b/tests/testsuite
@@ -704,7 +704,6 @@ at_help_all="1;version.at:19;tar version;;
 112;listed01.at:26;--listed for individual files;listed incremental listed01;
 113;listed02.at:28;working --listed;listed incremental listed02;
 114;listed03.at:24;incremental dump when the parent directory is unreadable;listed incremental listed03;
-115;listed04.at:26;--listed-incremental and --one-file-system;listed incremental listed04;
 116;listed05.at:33;--listed-incremental and remounted directories;listed incremental listed05 chdir;
 117;incr03.at:28;renamed files in incrementals;incremental incr03 rename;
 118;incr04.at:29;proper icontents initialization;incremental incr04 icontents;
@@ -806,13 +805,11 @@ at_help_all="1;version.at:19;tar version;;
 214;xattr04.at:26;xattrs: s/open/openat/ regression;xattrs xattr04 chdir;
 215;xattr05.at:28;xattrs: keywords with '=' and '%';xattrs xattr05 chdir;
 216;xattr06.at:27;xattrs: exclude xattrs on create ;xattrs xattr06;
-217;xattr07.at:26;xattrs: xattrs and --skip-old-files;xattrs xattr07;
 218;acls01.at:25;acls: basic functionality;xattrs acls acls01;
 219;acls02.at:25;acls: work with -C;xattrs acls acls02 chdir;
 220;acls03.at:30;acls: default ACLs;xattrs acls acls03 chdir;
 221;selnx01.at:25;selinux: basic store/restore;xattrs selinux selnx01;
 222;selacl01.at:25;acls/selinux: special files & fifos;xattrs selinux acls selacls01;
-223;capabs_raw01.at:25;capabilities: binary store/restore;xattrs capabilities capabs_raw01;
 224;onetop01.at:21;tar --one-top-level;extract onetop onetop01;
 225;onetop02.at:21;tar --one-top-level --show-transformed;extract onetop onetop02;
 226;onetop03.at:21;tar --one-top-level --transform;extract onetop onetop03;
-- 
2.20.1

