diff -up libcryptui-3.12.2/configure.ac.gpgme2fix libcryptui-3.12.2/configure.ac
--- libcryptui-3.12.2/configure.ac.gpgme2fix	2025-09-10 16:11:30.885541487 +0200
+++ libcryptui-3.12.2/configure.ac	2025-09-10 16:11:53.435873888 +0200
@@ -147,7 +147,7 @@ if test $GPGME_CONFIG != "failed" ; then
 	micro=`echo $gpgme_config_version | \
 		sed 's/\([[0-9]]*\)\.\([[0-9]]*\)\.\([[0-9]]*\).*/\3/'`
 
-	if test "$major" -eq "$req_major"; then
+	if test "$major" -ge "$req_major"; then
 		if test "$minor" -ge "$req_minor"; then
 			if test "$micro" -ge "$req_micro"; then
 				ok="yes"
diff -up libcryptui-3.12.2/daemon/seahorse-gpgme-operation.c.gpgme2fix libcryptui-3.12.2/daemon/seahorse-gpgme-operation.c
--- libcryptui-3.12.2/daemon/seahorse-gpgme-operation.c.gpgme2fix	2025-09-10 17:03:02.777681652 +0200
+++ libcryptui-3.12.2/daemon/seahorse-gpgme-operation.c	2025-09-10 17:03:16.624449842 +0200
@@ -302,7 +302,6 @@ event_cb (void *data, gpgme_event_io_t t
         break;
     
     case GPGME_EVENT_NEXT_KEY:
-    case GPGME_EVENT_NEXT_TRUSTITEM:
     default:
         /* Ignore unsupported event types */
         break;
