build/config/compiler/BUILD.gn | 1 -
 1 file changed, 1 deletion(-)

diff --git a/build/config/compiler/BUILD.gn b/build/config/compiler/BUILD.gn
index d40843b..5a1c05a 100644
--- a/build/config/compiler/BUILD.gn
+++ b/build/config/compiler/BUILD.gn
@@ -1543,7 +1543,6 @@ config("default_warnings") {
         # TODO(https://crbug.com/1316298): Re-enable once test failure is figured out
         cflags += [
           "-Xclang",
-          "-no-opaque-pointers",
         ]
       }
