From a25edabaa0b645676b52d066a648908b58fa342d Mon Sep 17 00:00:00 2001
From: Thomas Staudinger <Staudi.Kaos@gmail.com>
Date: Fri, 30 Aug 2024 21:33:04 +0200
Subject: [PATCH] Fix calloc argument order

Signed-off-by: Thomas Staudinger <Staudi.Kaos@gmail.com>
---
 tests/include/test/macros.h | 12 ++++++------
 1 file changed, 6 insertions(+), 6 deletions(-)

diff --git a/tests/include/test/macros.h b/tests/include/test/macros.h
index 894fc6727..56cc6680f 100644
--- a/tests/include/test/macros.h
+++ b/tests/include/test/macros.h
@@ -135,8 +135,8 @@
     do {                                                    \
         TEST_ASSERT((pointer) == NULL);                     \
         if ((item_count) != 0) {                            \
-            (pointer) = mbedtls_calloc(sizeof(*(pointer)),  \
-                                       (item_count));       \
+            (pointer) = mbedtls_calloc((item_count),        \
+                                        sizeof(*(pointer)));\
             TEST_ASSERT((pointer) != NULL);                 \
         }                                                   \
     } while (0)
@@ -165,8 +165,8 @@
 #define TEST_CALLOC_NONNULL(pointer, item_count)            \
     do {                                                    \
         TEST_ASSERT((pointer) == NULL);                     \
-        (pointer) = mbedtls_calloc(sizeof(*(pointer)),      \
-                                   (item_count));           \
+        (pointer) = mbedtls_calloc((item_count),            \
+                                    sizeof(*(pointer)));    \
         if (((pointer) == NULL) && ((item_count) == 0)) {   \
             (pointer) = mbedtls_calloc(1, 1);               \
         }                                                   \
@@ -185,8 +185,8 @@
     do {                                                    \
         TEST_ASSERT((pointer) == NULL);                     \
         if ((item_count) != 0) {                            \
-            (pointer) = mbedtls_calloc(sizeof(*(pointer)),  \
-                                       (item_count));       \
+            (pointer) = mbedtls_calloc((item_count),        \
+                                        sizeof(*(pointer)));\
             TEST_ASSUME((pointer) != NULL);                 \
         }                                                   \
     } while (0)
-- 
2.46.0

