From 66a868b6afb7389db124c521d4ffb84a4ba2bbdf Mon Sep 17 00:00:00 2001
From: Gilles Peskine <Gilles.Peskine@arm.com>
Date: Fri, 29 Sep 2023 17:36:24 +0200
Subject: [PATCH] CMake: fix build with 3rdparty module enabled through a
 custom config

Fixes #8165

Signed-off-by: Gilles Peskine <Gilles.Peskine@arm.com>
---
 3rdparty/CMakeLists.txt                          | 6 +-----

diff --git a/3rdparty/CMakeLists.txt b/3rdparty/CMakeLists.txt
index 18945e52ee75..37480f2cfd98 100644
--- a/3rdparty/CMakeLists.txt
+++ b/3rdparty/CMakeLists.txt
@@ -4,11 +4,7 @@ list (APPEND thirdparty_inc_public)
 list (APPEND thirdparty_inc)
 list (APPEND thirdparty_def)
 
-execute_process(COMMAND ${MBEDTLS_PYTHON_EXECUTABLE} ${CMAKE_CURRENT_SOURCE_DIR}/../scripts/config.py -f ${CMAKE_CURRENT_SOURCE_DIR}/../include/mbedtls/config.h get MBEDTLS_ECDH_VARIANT_EVEREST_ENABLED RESULT_VARIABLE result)
-
-if(${result} EQUAL 0)
-    add_subdirectory(everest)
-endif()
+add_subdirectory(everest)
 
 set(thirdparty_src ${thirdparty_src} PARENT_SCOPE)
 set(thirdparty_lib ${thirdparty_lib} PARENT_SCOPE)
