From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Reilly Brogan <reilly@reillybrogan.com>
Date: Wed, 20 Nov 2024 19:04:50 -0600
Subject: [PATCH] Fix qt6 ldflags

---
 config/qt.m4 | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/config/qt.m4 b/config/qt.m4
index 1f57441845..edb2022f8c 100644
--- a/config/qt.m4
+++ b/config/qt.m4
@@ -397,7 +397,9 @@ EOF1
 	    else
 		AC_SUBST(QT_CORE_INCLUDES)
 		AC_SUBST(QT_CORE_LDFLAGS)
+		QT_CORE_LIBS=$QT_CORE_LIB
 		AC_SUBST(QT_CORE_LIB)
+		AC_SUBST(QT_CORE_LIBS)
 		cat > $lyx_test_qt_pro << EOF2
 QMAKE_EXTRA_VARIABLES = MISSING
 qtHaveModule(core)		{QT += core} else {MISSING += core}
