diff --git a/src/insets/InsetListings.cpp b/src/insets/InsetListings.cpp
index 88ab71a..16d4224 100644
--- a/src/insets/InsetListings.cpp
+++ b/src/insets/InsetListings.cpp
@@ -44,6 +44,7 @@
 
 #include "support/regex.h"
 
+#include <cstring>
 #include <sstream>
 
 using namespace std;
diff --git a/src/lyxfind.cpp b/src/lyxfind.cpp
index a3a0b8b..d3a8db1 100644
--- a/src/lyxfind.cpp
+++ b/src/lyxfind.cpp
@@ -52,6 +52,7 @@
 #include "support/lstrings.h"
 
 #include "support/regex.h"
+#include <iterator>
 
 using namespace std;
 using namespace lyx::support;
-- 
2.35.4

