From a8439aa5343a82722a8a3cd383df6fb99604e5c0 Mon Sep 17 00:00:00 2001
From: "Jan Alexander Steffens (heftig)" <jan.steffens@gmail.com>
Date: Mon, 27 Jan 2020 18:11:05 +0100
Subject: [PATCH 2/4] ZEN: INTERACTIVE: Use BFQ as the elevator for SQ devices

---
 block/elevator.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/block/elevator.c b/block/elevator.c
index bd71f0fc4e4b6..c50d0b7b3d817 100644
--- a/block/elevator.c
+++ b/block/elevator.c
@@ -642,7 +642,11 @@ static struct elevator_type *elevator_get_default(struct request_queue *q)
 	    !blk_mq_is_shared_tags(q->tag_set->flags))
 		return NULL;
 
+#if defined(CONFIG_IOSCHED_BFQ)
+	return elevator_get(q, "bfq", false);
+#else
 	return elevator_get(q, "mq-deadline", false);
+#endif
 }
 
 /*
-- 
2.38.1

