From 5202f9c965b15b4232f18f39f940ccc4837afe90 Mon Sep 17 00:00:00 2001
From: "Jan Alexander Steffens (heftig)" <jan.steffens@gmail.com>
Date: Mon, 27 Jan 2020 18:11:05 +0100
Subject: [PATCH 2/4] ZEN: INTERACTIVE: Use BFQ as the elevator for SQ devices

---
 block/elevator.c | 6 +++++-
 1 file changed, 5 insertions(+), 1 deletion(-)

diff --git a/block/elevator.c b/block/elevator.c
index 24909069f872..4e614cf20685 100644
--- a/block/elevator.c
+++ b/block/elevator.c
@@ -573,7 +573,11 @@ static struct elevator_type *elevator_get_default(struct request_queue *q)
 	    !blk_mq_is_shared_tags(q->tag_set->flags))
 		return NULL;
 
-	return elevator_find_get(q, "mq-deadline");
+#if defined(CONFIG_IOSCHED_BFQ)
+	return elevator_find_get(q, "bfq");
+#else
+ 	return elevator_find_get(q, "mq-deadline");
+#endif
 }
 
 /*
-- 
2.41.0

