From bf79671be178bfc433e8f82b425aa2c4f0f2cd86 Mon Sep 17 00:00:00 2001
From: "Jan Alexander Steffens (heftig)" <heftig@archlinux.org>
Date: Mon, 12 Dec 2022 00:03:03 +0100
Subject: [PATCH 3/4] ZEN: INTERACTIVE: Use Kyber as the elevator for MQ
 devices

---
 block/elevator.c | 6 +++++-
 1 file changed, 5 insertions(+), 1 deletion(-)

diff --git a/block/elevator.c b/block/elevator.c
index c50d0b7b3d817..4d18785cec1db 100644
--- a/block/elevator.c
+++ b/block/elevator.c
@@ -640,7 +640,11 @@ static struct elevator_type *elevator_get_default(struct request_queue *q)
 
 	if (q->nr_hw_queues != 1 &&
 	    !blk_mq_is_shared_tags(q->tag_set->flags))
-		return NULL;
+#if defined(CONFIG_MQ_IOSCHED_KYBER)
+		return elevator_get(q, "kyber", false);
+#else
+		return elevator_get(q, "mq-deadline", false);
+#endif
 
 #if defined(CONFIG_IOSCHED_BFQ)
 	return elevator_get(q, "bfq", false);
-- 
2.38.1

