From 48dd7e08dca3a836ff46f7710879f73aff7f1d1b Mon Sep 17 00:00:00 2001
From: "Jan Alexander Steffens (heftig)" <heftig@archlinux.org>
Date: Mon, 12 Dec 2022 00:03:03 +0100
Subject: [PATCH 3/4] ZEN: INTERACTIVE: Use Kyber as the elevator for MQ
 devices

---
 block/elevator.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/block/elevator.c b/block/elevator.c
index 4e614cf20685..4db1428892f0 100644
--- a/block/elevator.c
+++ b/block/elevator.c
@@ -571,7 +571,11 @@ static struct elevator_type *elevator_get_default(struct request_queue *q)
 
 	if (q->nr_hw_queues != 1 &&
 	    !blk_mq_is_shared_tags(q->tag_set->flags))
+#if defined(CONFIG_MQ_IOSCHED_KYBER)
+		return elevator_find_get(q, "kyber");
+#else
 		return NULL;
+#endif
 
 #if defined(CONFIG_IOSCHED_BFQ)
 	return elevator_find_get(q, "bfq");
-- 
2.41.0

