From 5b55cbd942af092639e72ec9788dee7cf7a37aa6 Mon Sep 17 00:00:00 2001
From: Thomas Staudinger <Staudi.Kaos@gmail.com>
Date: Sun, 15 Jan 2023 20:31:04 +0100
Subject: [PATCH] Remove additional prefix from pkg-config libdir generation

Signed-off-by: Thomas Staudinger <Staudi.Kaos@gmail.com>
---
 CMakeLists.txt | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index dda22431..8e26627a 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -224,7 +224,7 @@ if (INSTALL_PKGCONFIG_MODULE)
 
 	set (prefix ${CMAKE_INSTALL_PREFIX})
 	set (exec_prefix "\$\{prefix\}")
-	set (libdir "\$\{prefix\}/${CMAKE_INSTALL_LIBDIR}")
+	set (libdir ${CMAKE_INSTALL_LIBDIR})
 	set (includedir "\$\{prefix\}/${CMAKE_INSTALL_INCLUDEDIR}")
 	set (VERSION ${PROJECT_VERSION})
 	if (ENABLE_EXTERNAL_LIBS)
-- 
2.39.0

