From 3b64bd94e26c4e07d0cbbecf0c1263fa03f7c05a Mon Sep 17 00:00:00 2001
From: Thomas Staudinger <Staudi.Kaos@gmail.com>
Date: Sun, 15 Jan 2023 23:43:53 +0100
Subject: [PATCH] Handle absolute install paths

---
 CMakeLists.txt | 13 +++++++++++--
 1 file changed, 11 insertions(+), 2 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index dda22431..023d49fa 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -224,8 +224,17 @@ if (INSTALL_PKGCONFIG_MODULE)
 
 	set (prefix ${CMAKE_INSTALL_PREFIX})
 	set (exec_prefix "\$\{prefix\}")
-	set (libdir "\$\{prefix\}/${CMAKE_INSTALL_LIBDIR}")
-	set (includedir "\$\{prefix\}/${CMAKE_INSTALL_INCLUDEDIR}")
+	if (IS_ABSOLUTE "${CMAKE_INSTALL_LIBDIR}")
+		set(libdir "${CMAKE_INSTALL_LIBDIR}")
+	else()
+		set(libdir "\${prefix}/${CMAKE_INSTALL_LIBDIR}")
+	endif()
+
+	if (IS_ABSOLUTE "${CMAKE_INSTALL_INCLUDEDIR}")
+		set(includedir "${CMAKE_INSTALL_INCLUDEDIR}")
+	else()
+		set(includedir "\${prefix}/${CMAKE_INSTALL_INCLUDEDIR}")
+	endif()
 	set (VERSION ${PROJECT_VERSION})
 	if (ENABLE_EXTERNAL_LIBS)
 		set (EXTERNAL_XIPH_REQUIRE "flac ogg vorbis vorbisenc opus")
-- 
2.39.0

