diff --git a/doc/automake.mk b/doc/automake.mk
index 2dfe838..a3b8e24 100644
--- a/doc/automake.mk
+++ b/doc/automake.mk
@@ -86,7 +86,7 @@ dist_docbook_DATA = doc/pspp.xml
 CLEANFILES += pspp-dev.dvi $(docbook_DATA) doc/pspp.info* doc/pspp.xml
 
 
-doc: $(INFO_DEPS) $(DVIS) $(PDFS) $(PNGS) $(HTMLS) $(dist_docbook_DATA)
+doc: $(INFO_DEPS) $(DVIS) $(PDFS) $(HTMLS) $(dist_docbook_DATA)
 PHONY += doc
 
 doc/help-pages-list: $(UI_FILES)
@@ -183,23 +183,19 @@ FIGURE_TXTS = $(FIGURE_SPVS:.spv=.txt)
 FIGURE_TEXIS = $(FIGURE_TXTS:.txt=.texi)
 FIGURE_HTMLS = $(FIGURE_SPVS:.spv=.html)
 FIGURE_PDFS = $(FIGURE_SPVS:.spv=.pdf)
-FIGURE_PNGS = $(FIGURE_SPVS:.spv=.png)
 
 figure-spvs: $(FIGURE_SPVS)
 figure-txts: $(FIGURE_TXTS)
 figure-texis: $(FIGURE_TEXIS)
 figure-htmls: $(FIGURE_HTMLS)
 figure-pdfs: $(FIGURE_PDFS)
-figure-pngs: $(FIGURE_PNGS)
 PHONY += \
 	figure-spv \
 	figure-txts \
 	figure-texis \
 	figure-htmls \
-	figure-pdfs \
-	figure-pngs
+	figure-pdfs
 
-$(top_builddir)/doc/pspp.info:  $(FIGURE_PNGS)
 $(top_builddir)/doc/pspp.ps:    $(FIGURE_TEXIS)
 $(top_builddir)/doc/pspp.dvi:   $(FIGURE_TEXIS)
 $(top_builddir)/doc/pspp.html:  $(FIGURE_HTMLS)
@@ -210,9 +206,8 @@ CLEANFILES += \
 	$(FIGURE_TXTS) \
 	$(FIGURE_SPVS) \
 	$(FIGURE_TEXIS) \
-	$(FIGURE_HTMLS) \
-	$(FIGURE_PNGS)
-SUFFIXES += .sps .spv .txt .html .texi .pdf .png
+	$(FIGURE_HTMLS)
+SUFFIXES += .sps .spv .txt .html .texi .pdf
 
 # Use pspp to process a syntax file into an output file.
 if cross_compiling
@@ -258,7 +253,6 @@ doc/pspp-figures/tutorial7b.spv: doc/pspp-figures/tutorial7.spv $(pspp_output)
 	$(convert) --commands=regression --nth-command=2 --subtypes=coefficients
 
 
-$(FIGURE_PNGS): $(pspp_output)
 $(FIGURE_TXTS): $(pspp_output)
 $(FIGURE_HTMLS): $(pspp_output)
 $(FIGURE_PDFS): $(pspp_output)
@@ -269,8 +263,6 @@ $(FIGURE_TXTS) $(FIGURE_HTMLS): $(pspp_output)
 	$(convert)
 .spv.pdf:
 	$(convert) -O trim=true -O left-margin=0in -O right-margin=0in -O top-margin=0in -O bottom-margin=0in -O paper-size=7.5x99in --table-look=$(srcdir)/doc/tutorial.stt
-.spv.png:
-	$(convert) -O trim=true -O left-margin=0in -O right-margin=0in -O top-margin=0in -O bottom-margin=0in -O paper-size=7.5x99in --table-look=$(srcdir)/doc/tutorial.stt
 EXTRA_DIST += doc/tutorial.stt
 .spv.html:
 	$(convert) -O format=html -O bare=true
@@ -347,10 +339,7 @@ $(top_srcdir)/doc/screenshots/sort-simple.grab \
 $(top_srcdir)/doc/screenshots/sort.grab
 
 
-PDF_SCREENSHOTS =  $(SCREENSHOTS:.grab=-hc.png)
 EPS_SCREENSHOTS =  $(SCREENSHOTS:.grab=-hc.eps)
-HTML_SCREENSHOTS = $(SCREENSHOTS:.grab=-ad.png)
-INFO_SCREENSHOTS = $(SCREENSHOTS:.grab=-ad.png)
 
 doc-make: doc/doc-make.in Makefile
 	$(SED) -e 's|%top_srcdir%|@top_srcdir@|g' \
@@ -362,27 +351,6 @@ doc-make: doc/doc-make.in Makefile
 	$< > $@
 
 
-# Install all the PNG files so that info readers can recognise them
-install-info-local: $(FIGURE_PNGS)
-	$(MKDIR_P) $(DESTDIR)$(infodir)/screenshots
-	for p in $(INFO_SCREENSHOTS); do \
-		$(INSTALL_DATA) $$p $(DESTDIR)$(infodir)/screenshots ;\
-	done
-	$(INSTALL) -d $(DESTDIR)$(infodir)/pspp-figures
-	for p in $(FIGURE_PNGS); do \
-		$(INSTALL_DATA) $$p $(DESTDIR)$(infodir)/pspp-figures ;\
-	done
-
-uninstall-local:
-	for p in $(INFO_SCREENSHOTS); do \
-		f=`basename $$p ` ; \
-		rm -f $(DESTDIR)$(infodir)/screenshots/$$f ; \
-	done
-	for p in $(FIGURE_PNGS); do \
-		f=`basename $$p ` ; \
-		rm -f $(DESTDIR)$(infodir)/pspp-figures/$$f ; \
-	done
-
 EXTRA_DIST+= $(SCREENSHOTS) doc/doc-make.in doc/screengrab
 
-EXTRA_DIST+= $(EPS_SCREENSHOTS) $(PDF_SCREENSHOTS) $(INFO_SCREENSHOTS)
+EXTRA_DIST+= $(EPS_SCREENSHOTS)
