diff --git a/src/editor/kmcomposerwin.cpp b/src/editor/kmcomposerwin.cpp
index df39b57e5..edcf92c40 100644
--- a/src/editor/kmcomposerwin.cpp
+++ b/src/editor/kmcomposerwin.cpp
@@ -429,7 +429,7 @@ KMComposerWin::KMComposerWin(const KMime::Message::Ptr &aMsg, bool lastSignState

     mPluginEditorCheckBeforeSendManagerInterface->setParentWidget(this);

-    mPluginEditorInitManagerInterface->setParent(this);
+    mPluginEditorInitManagerInterface->setParentWidget(this);
     mPluginEditorInitManagerInterface->setRichTextEditor(composerEditorNg);

     mPluginEditorConvertTextManagerInterface->setParentWidget(this);

diff --git a/src/kmmainwidget.cpp b/src/kmmainwidget.cpp
index d35cc9bda..46f23d438 100644
--- a/src/kmmainwidget.cpp
+++ b/src/kmmainwidget.cpp
@@ -2586,7 +2586,9 @@ void KMMainWidget::slotItemsFetchedForActivation(KMCommand *command)
     if (!win) {
         win = new KMReaderMainWin(mFolderDisplayFormatPreference, mFolderHtmlLoadExtPreference);
     }
-    win->viewer()->setWebViewZoomFactor(mMsgView->viewer()->webViewZoomFactor());
+    if (mMsgView) {
+        win->viewer()->setWebViewZoomFactor(mMsgView->viewer()->webViewZoomFactor());
+    }
     const bool useFixedFont = mMsgView ? mMsgView->isFixedFont()
                               : MessageViewer::MessageViewerSettings::self()->useFixedFont();
     win->setUseFixedFont(useFixedFont);

diff --git a/src/kmmainwidget.cpp b/src/kmmainwidget.cpp
index 46f23d438..918bda371 100644
--- a/src/kmmainwidget.cpp
+++ b/src/kmmainwidget.cpp
@@ -2586,7 +2586,7 @@ void KMMainWidget::slotItemsFetchedForActivation(KMCommand *command)
     if (!win) {
         win = new KMReaderMainWin(mFolderDisplayFormatPreference, mFolderHtmlLoadExtPreference);
     }
-    if (mMsgView) {
+    if (mMsgView && mMsgView->viewer()) {
         win->viewer()->setWebViewZoomFactor(mMsgView->viewer()->webViewZoomFactor());
     }
     const bool useFixedFont = mMsgView ? mMsgView->isFixedFont()

diff --git a/src/kmmainwidget.cpp b/src/kmmainwidget.cpp
index c1354d5e5..8caa4433b 100644
--- a/src/kmmainwidget.cpp
+++ b/src/kmmainwidget.cpp
@@ -3603,7 +3603,9 @@ void KMMainWidget::slotPageIsScrolledToBottom(bool isAtBottom)
     if (isAtBottom) {
         slotSelectNextUnreadMessage();
     } else {
-        mMsgView->viewer()->slotJumpDown();
+        if (mMsgView) {
+            mMsgView->viewer()->slotJumpDown();
+        }
     }
 }

