diff --git a/src/kmail_part.cpp b/src/kmail_part.cpp
index 703a1ee..96c4c07 100644
--- a/src/kmail_part.cpp
+++ b/src/kmail_part.cpp
@@ -33,6 +33,7 @@
 
 #include <QVBoxLayout>
 
+#include <KParts/GUIActivateEvent>
 #include <kparts/statusbarextension.h>
 #include <kparts/mainwindow.h>
 #include <kpluginfactory.h>
@@ -129,18 +130,18 @@ bool KMailPart::openFile()
 void KMailPart::guiActivateEvent(KParts::GUIActivateEvent *e)
 {
     KParts::ReadOnlyPart::guiActivateEvent(e);
-    mainWidget->initializeFilterActions();
-    mainWidget->tagActionManager()->createActions();
-    mainWidget->folderShortcutActionManager()->createActions();
-    mainWidget->populateMessageListStatusFilterCombo();
-    mainWidget->initializePluginActions();
-    /*
-    FIXME it doesn't work when we switch component.
-    const QString title = mainWidget->fullCollectionPath();
-    if (!title.isEmpty()) {
-        Q_EMIT setWindowCaption(title);
+    if (e->activated()) {
+        mainWidget->initializeFilterActions();
+        mainWidget->tagActionManager()->createActions();
+        mainWidget->folderShortcutActionManager()->createActions();
+        mainWidget->populateMessageListStatusFilterCombo();
+        mainWidget->initializePluginActions();
+
+        const QString title = mainWidget->fullCollectionPath();
+        if (!title.isEmpty()) {
+            Q_EMIT setWindowCaption(title);
+        }
     }
-    */
 }
 
 void KMailPart::exit()