From 017f88510fb2710bd45f559253df6f65f06163d0 Mon Sep 17 00:00:00 2001
From: Pierre-Yves <pyu@riseup.net>
Date: Fri, 3 Nov 2017 19:12:31 +0100
Subject: [PATCH] remove check gnulib test-getaddrinfo

Signed-off-by: Pierre-Yves <pyu@riseup.net>
---
 gnulib/tests/gnulib.mk | 9 ---------
 1 file changed, 9 deletions(-)

diff --git a/gnulib/tests/gnulib.mk b/gnulib/tests/gnulib.mk
index 452d0b5..e677b3c 100644
--- a/gnulib/tests/gnulib.mk
+++ b/gnulib/tests/gnulib.mk
@@ -562,15 +562,6 @@ EXTRA_DIST += test-fwrite.c signature.h macros.h
 
 ## end   gnulib module fwrite-tests
 
-## begin gnulib module getaddrinfo-tests
-
-GNULIB_TESTS += test-getaddrinfo
-check_PROGRAMS += test-getaddrinfo
-test_getaddrinfo_LDADD = $(LDADD) @GETADDRINFO_LIB@ @LIBINTL@
-EXTRA_DIST += signature.h test-getaddrinfo.c
-
-## end   gnulib module getaddrinfo-tests
-
 ## begin gnulib module getcwd-lgpl-tests
 
 GNULIB_TESTS += test-getcwd-lgpl
-- 
2.15.0

