From 1eb74f530383db698d651d9c7beaf022529ee6b5 Mon Sep 17 00:00:00 2001
From: Fabio Forni <livingsilver94.solus@redaril.me>
Date: Fri, 25 Oct 2019 14:24:12 +0200
Subject: [PATCH] Enable apparmor in qemu.conf

---
 src/qemu/qemu.conf | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)
diff --git a/src/qemu/qemu.conf b/src/qemu/qemu.conf
index b3a3428..87a15e1 100644
--- a/src/qemu/qemu.conf
+++ b/src/qemu/qemu.conf
@@ -419,7 +419,7 @@
 # a special value; security_driver can be set to that value in
 # isolation, but it cannot appear in a list of drivers.
 #
-#security_driver = "selinux"
+security_driver = "apparmor"
 
 # If set to non-zero, then the default security labeling
 # will make guests confined. If set to zero, then guests
-- 
2.23.0
