From 3a142f86bfd64177228c4b0aa843741c10136fa6 Mon Sep 17 00:00:00 2001
From: Campbell Jones <git@serebit.com>
Date: Wed, 6 May 2020 23:25:16 -0400
Subject: [PATCH] disable building gtk docs

---
 Makefile.am  | 4 ++--
 configure.in | 8 +-------
 2 files changed, 3 insertions(+), 9 deletions(-)

diff --git a/Makefile.am b/Makefile.am
index 8cffb3d..fa0a51d 100644
--- a/Makefile.am
+++ b/Makefile.am
@@ -1,7 +1,7 @@
-SUBDIRS = gdl docs po
+SUBDIRS = gdl po
 
 ACLOCAL_AMFLAGS = -I m4
-DISTCHECK_CONFIGURE_FLAGS = --enable-gtk-doc --enable-introspection
+DISTCHECK_CONFIGURE_FLAGS = --enable-introspection
 
 EXTRA_DIST = 			\
 	MAINTAINERS		\
diff --git a/configure.in b/configure.in
index 5169964..79a4159 100644
--- a/configure.in
+++ b/configure.in
@@ -137,9 +137,6 @@ fi
 gdl_images_dir="${packageprefix}/share/gdl/images"
 AC_SUBST(gdl_images_dir)
 
-# check for gtk-doc
-GTK_DOC_CHECK(1.4)
-
 GLIB_PREFIX="`$PKG_CONFIG --variable=prefix glib-2.0`"
 GTK_PREFIX="`$PKG_CONFIG --variable=prefix gtk+-3.0`"
 
@@ -151,10 +148,7 @@ GOBJECT_INTROSPECTION_CHECK([0.6.7])
 AC_OUTPUT([
 Makefile
 gdl/Makefile
-po/Makefile.in 
-docs/Makefile
-docs/reference/Makefile
-docs/reference/version.xml
+po/Makefile.in
 gdl.spec
 gdl-3.0.pc
 ])
-- 
2.26.2

