Fix github issue 1038: https://github.com/mypaint/mypaint/issues/1038

diff --git a/gui/inktool.py b/gui/inktool.py
index b60fb3a..99588bd 100644
--- a/gui/inktool.py
+++ b/gui/inktool.py
@@ -1076,7 +1076,7 @@ class OptionsPresenter (object):
     def _ensure_ui_populated(self):
         if self._options_grid is not None:
             return
-        builder_xml = os.path.splitext(__file__)[0] + ".glade"
+        builder_xml = os.path.join(self._app.datapath, "gui/inktool.glade")
         builder = Gtk.Builder()
         builder.set_translation_domain("mypaint")
         builder.add_from_file(builder_xml)
