From 387660147e1e256760573444d96e29f5551b4054 Mon Sep 17 00:00:00 2001
From: Peter O'Connor <peter@solus-project.com>
Date: Thu, 8 Feb 2018 09:47:30 +1100
Subject: [PATCH] Update path from python3.5 to python3.6

Signed-off-by: Peter O'Connor <peter@solus-project.com>
---
 bin/gufw-pkexec | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/bin/gufw-pkexec b/bin/gufw-pkexec
index ad4e4e1..bb3a6c0 100755
--- a/bin/gufw-pkexec
+++ b/bin/gufw-pkexec
@@ -1,7 +1,7 @@
 #!/bin/bash
 LOCATIONS=(
-"/usr/lib/python3.5/site-packages/gufw/gufw.py"
-"/usr/lib/python3.5/dist-packages/gufw/gufw.py"
+"/usr/lib/python3.6/site-packages/gufw/gufw.py"
+"/usr/lib/python3.6/dist-packages/gufw/gufw.py"
 "/usr/share/gufw/gufw/gufw.py"
 )

--
2.16.1
