diff -uNr a/bin/gufw-pkexec b/bin/gufw-pkexec
--- a/bin/gufw-pkexec	2020-10-28 04:16:48.000000000 +0900
+++ b/bin/gufw-pkexec	2020-12-06 00:57:19.751324000 +0900
@@ -1,2 +1,3 @@
 #!/bin/bash
-python3 /usr/share/gufw/gufw/gufw.py $1
+PYTHON3_VERSION=`python3 -Ic "import sys; sys.stdout.write('{0.major}.{0.minor}'.format(sys.version_info))"`
+python3 /usr/lib/python${PYTHON3_VERSION}/site-packages/gufw/gufw.py $1
