From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Reilly Brogan <reilly@reillybrogan.com>
Date: Fri, 5 Jul 2024 01:47:18 -0500
Subject: [PATCH] solus: Build changes

---
 Cargo.toml | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/Cargo.toml b/Cargo.toml
index 696c012b8..05dd07537 100644
--- a/Cargo.toml
+++ b/Cargo.toml
@@ -277,7 +277,7 @@ sqlite = ["nu-command/sqlite", "nu-cmd-lang/sqlite"]
 
 [profile.release]
 opt-level = "s"     # Optimize for size
-strip = "debuginfo"
+strip = false # Solus build system does it
 lto = "thin"
 
 # build with `cargo build --profile profiling`
