From a96b2f3f62c7407478fb3982189b7d3fc2f46ada Mon Sep 17 00:00:00 2001
From: Joshua Strobl <joshua@streambits.io>
Date: Fri, 30 Oct 2020 15:48:59 +0200
Subject: [PATCH 1/1] Add potential ELAN device for K505ZA / X505ZA

---
 include/linux/input/elan-i2c-ids.h | 12 ++++++++++++
 1 file changed, 12 insertions(+)

diff --git a/include/linux/input/elan-i2c-ids.h b/include/linux/input/elan-i2c-ids.h
index 520858d12680..6df19138509c 100644
--- a/include/linux/input/elan-i2c-ids.h
+++ b/include/linux/input/elan-i2c-ids.h
@@ -77,6 +77,18 @@ static const struct acpi_device_id elan_acpi_id[] = {
 	{ "ELAN0636", 0 }, /* Lenovo V1415-Dali */
 	{ "ELAN0637", 0 }, /* Lenovo V1415-IGLR */
 	{ "ELAN1000", 0 },
+	{ "ELAN1200", 0 },
+	{ "ELAN1201", 0 },
+	{ "ELAN1202", 0 },
+	{ "ELAN1203", 0 },
+	{ "ELAN12A6", 0 },
+	{ "ELAN1300", 0 },
+	{ "ELAN1301", 0 },
+	{ "ELAN2200", 0 },
+	{ "ELAN2201", 0 },
+	{ "ELAN2202", 0 },
+	{ "ELAN2300", 0 },
+	{ "ELAN2301", 0 },
 	{ }
 };
 
-- 
2.29.1

