From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Thomas Staudinger <Staudi.Kaos@gmail.com>
Date: Thu, 12 Sep 2024 10:31:27 +0200
Subject: [PATCH] Remove constraint on sane minor version "0"

Signed-off-by: Thomas Staudinger <Staudi.Kaos@gmail.com>
---
 sane/version.hpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/sane/version.hpp b/sane/version.hpp
index 11b5f29..512855d 100644
--- a/sane/version.hpp
+++ b/sane/version.hpp
@@ -37,7 +37,7 @@ extern "C" {                    // needed until sane-backends-1.0.14
 #include <sane/sane.h>
 }
 
-#if !((SANE_MAJOR == SANE_CURRENT_MAJOR) && (SANE_MINOR == 0))
+#if !(SANE_MAJOR == SANE_CURRENT_MAJOR)
 #error "SANE installation violates versioning portability constraints."
 #endif
 
