From 21b797548182c40c9cfbf6806763b157edd8eebe Mon Sep 17 00:00:00 2001
From: Joshua Strobl <joshua@streambits.io>
Date: Mon, 28 Dec 2020 17:17:01 +0200
Subject: [PATCH 1/1] Tweak bash completion install path.

---
 dbus/Makefile.am | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/dbus/Makefile.am b/dbus/Makefile.am
index 2e8cb27..18c99ea 100644
--- a/dbus/Makefile.am
+++ b/dbus/Makefile.am
@@ -93,7 +93,7 @@ regenerate-built-sources:
         $(GLIB_GENMARSHAL) --prefix=_dbus_g_marshal dbus-gmarshal.list --body >> dbus-gmarshal.c
 
 
-completiondir = $(sysconfdir)/bash_completion.d
+completiondir = $(datadir)/bash-completion/completions/
 if DBUS_BASH_COMPLETION
 libexec_PROGRAMS=dbus-bash-completion-helper
 completion_SCRIPTS=dbus-bash-completion.sh
-- 
2.29.2

