From 0af5b1b163ac51a346745e9255358118b752e918 Mon Sep 17 00:00:00 2001
From: Peter O'Connor <sunnyflunk@gmail.com>
Date: Wed, 14 Dec 2016 16:43:36 +1100
Subject: [PATCH] Force breeze as the default icon theme

---
 skgbasemodeler/skgservices.cpp | 1 +
 1 file changed, 1 insertion(+)

diff --git a/skgbasemodeler/skgservices.cpp b/skgbasemodeler/skgservices.cpp
index 9ad0b89..63f5d1f 100644
--- a/skgbasemodeler/skgservices.cpp
+++ b/skgbasemodeler/skgservices.cpp
@@ -1796,6 +1796,7 @@ QString SKGServices::encodeForUrl(const QString& iString)

 QIcon SKGServices::fromTheme(const QString& iName, const QStringList& iOverlays)
 {
+    QIcon::setThemeName("breeze");
     QIcon output;
     if (!iOverlays.isEmpty()) {
         output = QIcon(new KIconEngine(iName,  KIconLoader::global(), iOverlays));
--
2.11.0
