--- a/cmake/dependencies.cmake
+++ b/cmake/dependencies.cmake
@@ -7,7 +7,6 @@
   filesystem
   iostreams
   regex
-  system
   thread
   REQUIRED
 )
@@ -20,7 +19,6 @@
   ${Boost_FILESYSTEM_LIBRARIES}
   ${Boost_IOSTREAMS_LIBRARIES}
   ${Boost_REGEX_LIBRARIES}
-  ${Boost_SYSTEM_LIBRARIES}
   ${Boost_THREAD_LIBRARIES}
 )
 
--- a/src/core/CMakeLists.txt
+++ b/src/core/CMakeLists.txt
@@ -57,7 +57,6 @@
     Boost::filesystem
     Boost::iostreams
     Boost::regex
-    Boost::system
     Boost::thread
     ZLIB::ZLIB
 )
--- a/src/contrib/CMakeLists.txt
+++ b/src/contrib/CMakeLists.txt
@@ -67,7 +67,6 @@
     Boost::filesystem
     Boost::iostreams
     Boost::regex
-    Boost::system
     Boost::thread
     ZLIB::ZLIB
     lucene++::lucene++)
--- a/src/demo/deletefiles/CMakeLists.txt
+++ b/src/demo/deletefiles/CMakeLists.txt
@@ -38,7 +38,6 @@
     Boost::filesystem
     Boost::iostreams
     Boost::regex
-    Boost::system
     Boost::thread
     ZLIB::ZLIB
     lucene++::lucene++
--- a/src/demo/indexfiles/CMakeLists.txt
+++ b/src/demo/indexfiles/CMakeLists.txt
@@ -39,7 +39,6 @@
     Boost::filesystem
     Boost::iostreams
     Boost::regex
-    Boost::system
     Boost::thread
     ZLIB::ZLIB
     lucene++::lucene++
--- a/src/demo/searchfiles/CMakeLists.txt
+++ b/src/demo/searchfiles/CMakeLists.txt
@@ -38,7 +38,6 @@
     Boost::filesystem
     Boost::iostreams
     Boost::regex
-    Boost::system
     Boost::thread
     ZLIB::ZLIB
     lucene++::lucene++
--- a/src/test/CMakeLists.txt
+++ b/src/test/CMakeLists.txt
@@ -59,7 +59,6 @@
     Boost::filesystem
     Boost::iostreams
     Boost::regex
-    Boost::system
     Boost::thread
     ZLIB::ZLIB
     gtest_main
