From 3a181c5ad28bc2abe18b1eccc43a6c8a71c7ad23 Mon Sep 17 00:00:00 2001
From: Daniel Lobl <daniel.lobl@shopify.com>
Date: Tue, 22 Nov 2022 15:25:37 +0100
Subject: [PATCH] SVT-AV1: pkgconfig: rename SvtEnc private requirement to
 SvtAv1Enc

---
 CMakeLists.txt | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index de3e6daf4..df1764b33 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -130,7 +130,7 @@ if (RAV1E_FOUND)
     list(APPEND REQUIRES_PRIVATE "rav1e")
 endif()
 if (SvtEnc_FOUND)
-    list(APPEND REQUIRES_PRIVATE "SvtEnc")
+    list(APPEND REQUIRES_PRIVATE "SvtAv1Enc")
 endif()
 if (AOM_DECODER_FOUND OR DAV1D_FOUND)
     set(have_avif_decoder yes)
