From 7cd222169e0f256cdf33e13028665d1b6448a6e9 Mon Sep 17 00:00:00 2001
From: David Harder <david@davidjharder.ca>
Date: Sat, 20 Nov 2021 17:49:42 -0600
Subject: [PATCH] Disable wake notifications

---
 lib/solaar/ui/notify.py | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/lib/solaar/ui/notify.py b/lib/solaar/ui/notify.py
index 6d4bfcf..974652c 100644
--- a/lib/solaar/ui/notify.py
+++ b/lib/solaar/ui/notify.py
@@ -135,7 +135,8 @@ if available:
             try:
                 # if _log.isEnabledFor(_DEBUG):
                 #     _log.debug("showing %s", n)
-                n.show()
+                if(reason != _("powered on")):
+                    n.show()
             except Exception:
                 _log.exception('showing %s', n)
 
-- 
2.34.0

