From 8cb73c106bd41beb56e256c7a5fe2162cb528f5e Mon Sep 17 00:00:00 2001
From: Pierre-Yves <pyu@riseup.net>
Date: Mon, 4 Mar 2019 16:35:54 +0100
Subject: [PATCH] data: Allow Caja to show under most GTK desktops

Signed-off-by: Pierre-Yves <pyu@riseup.net>
---
 data/caja-browser.desktop.in | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/data/caja-browser.desktop.in b/data/caja-browser.desktop.in
index 9c0a8f0..0b55c77 100644
--- a/data/caja-browser.desktop.in
+++ b/data/caja-browser.desktop.in
@@ -10,4 +10,4 @@ StartupNotify=false
 Type=Application
 Categories=GTK;System;Utility;Core;
 Keywords=files;browser;manager;MATE;
-OnlyShowIn=MATE;
+OnlyShowIn=MATE;GNOME;Budgie;
-- 
2.21.0

