From 60a45dd150730e9d7be7e000021f8a140deae27b Mon Sep 17 00:00:00 2001
From: Ikey Doherty <ikey@solus-project.com>
Date: Fri, 29 Dec 2017 20:00:47 +0000
Subject: [PATCH] data: Allow Caja to show under most GTK desktops

Signed-off-by: Ikey Doherty <ikey@solus-project.com>
---
 data/caja.desktop.in.in | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/data/caja.desktop.in.in b/data/caja.desktop.in.in
index eff1373..c560dcb 100644
--- a/data/caja.desktop.in.in
+++ b/data/caja.desktop.in.in
@@ -7,7 +7,7 @@ Terminal=false
 Type=Application
 StartupNotify=false
 NoDisplay=true
-OnlyShowIn=MATE;
+OnlyShowIn=MATE;GNOME;Budgie;
 X-MATE-Bugzilla-Bugzilla=MATE
 X-MATE-Bugzilla-Product=caja
 X-MATE-Bugzilla-Component=general
-- 
2.15.1

