From 3cc054726856b1208dd7aa5d81dac72c6651e196 Mon Sep 17 00:00:00 2001
From: Ikey Doherty <ikey@solus-project.com>
Date: Sun, 27 Dec 2015 00:55:30 +0000
Subject: [PATCH] Temp: Show in GNOME so that new Budgie can use nm-applet

Signed-off-by: Ikey Doherty <ikey@solus-project.com>
---
 nm-applet.desktop.in | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/nm-applet.desktop.in b/nm-applet.desktop.in
index 8f94a88..eca880b 100644
--- a/nm-applet.desktop.in
+++ b/nm-applet.desktop.in
@@ -6,7 +6,7 @@ Exec=nm-applet
 Terminal=false
 Type=Application
 NoDisplay=true
-NotShowIn=KDE;GNOME;
+NotShowIn=KDE;
 X-GNOME-Bugzilla-Bugzilla=GNOME
 X-GNOME-Bugzilla-Product=NetworkManager
 X-GNOME-Bugzilla-Component=nm-applet
-- 
2.6.4

