From 62522ebf5f166f1c9c7402a40bcf633c6168af7f Mon Sep 17 00:00:00 2001
From: Ikey Doherty <ikey@solus-project.com>
Date: Fri, 17 Nov 2017 19:15:42 +0000
Subject: [PATCH] Teach it to spell.

Signed-off-by: Ikey Doherty <ikey@solus-project.com>
---
 src/libnm-gtk/meson.build | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/libnm-gtk/meson.build b/src/libnm-gtk/meson.build
index 9aaf219..c0ddba5 100644
--- a/src/libnm-gtk/meson.build
+++ b/src/libnm-gtk/meson.build
@@ -91,7 +91,7 @@ pkg.generate(
   requires: [
     'NetworkManager ' + nm_req_version,
     'libnm-util ' + nm_req_version,
-    'linm-glib ' + nm_req_version,
+    'libnm-glib ' + nm_req_version,
     'gtk+-3.0 ' + gtk_req_version
   ],
   variables: 'exec_prefix=' + nma_prefix,
-- 
2.15.0

