From edf937e16e3dd6a7720656f456e9cb2bc303155d Mon Sep 17 00:00:00 2001
From: "Munoz, Obed N" <obed.n.munoz@intel.com>
Date: Mon, 1 Aug 2016 11:13:01 -0500
Subject: [PATCH] default config

Signed-off-by: Munoz, Obed N <obed.n.munoz@intel.com>
---
 docs/conf/httpd.conf.in | 19 ++++++++++++++-----
 1 file changed, 14 insertions(+), 5 deletions(-)

diff --git a/docs/conf/httpd.conf.in b/docs/conf/httpd.conf.in
index 966d2c3..113854d 100644
--- a/docs/conf/httpd.conf.in
+++ b/docs/conf/httpd.conf.in
@@ -65,6 +65,8 @@ Listen @@Port@@
 #
 @@LoadModule@@
 
+Protocols h2 h2c http/1.1
+
 <IfModule unixd_module>
 #
 # If you wish httpd to run as a different user or group, you must run
@@ -74,8 +76,8 @@ Listen @@Port@@
 # It is usually good practice to create a dedicated user and group for
 # running httpd, as with most system services.
 #
-User daemon
-Group daemon
+User httpd
+Group httpd
 
 </IfModule>
 
@@ -181,7 +183,7 @@ DocumentRoot "@exp_htdocsdir@"
 # logged here.  If you *do* define an error logfile for a <VirtualHost>
 # container, that host's errors will be logged there and not here.
 #
-ErrorLog "@rel_logfiledir@/error_log"
+ErrorLog "/var/log/httpd/error_log"
 
 #
 # LogLevel: Control the number of messages logged to the error_log.
@@ -210,13 +212,13 @@ LogLevel warn
     # define per-<VirtualHost> access logfiles, transactions will be
     # logged therein and *not* in this file.
     #
-    CustomLog "@rel_logfiledir@/access_log" common
+    CustomLog "/var/log/httpd/access_log" common
 
     #
     # If you prefer a logfile with access, agent, and referer information
     # (Combined Logfile Format) you can use the following directive.
     #
-    #CustomLog "@rel_logfiledir@/access_log" combined
+    #CustomLog "/var/log/httpd/access_log" combined
 </IfModule>
 
 <IfModule alias_module>
@@ -407,3 +409,10 @@ SSLRandomSeed startup builtin
 SSLRandomSeed connect builtin
 </IfModule>
 
+# Virtual hosts
+IncludeOptional /usr/share/defaults/httpd/conf.d/*.conf
+IncludeOptional /usr/share/defaults/httpd/conf.modules.d/*.conf
+IncludeOptional /etc/httpd/conf.d/*.conf
+IncludeOptional /etc/httpd/conf.modules.d/*.conf
+
+PidFile /run/httpd/httpd.pid
-- 
2.9.0

