--- a/server/config.c    2018-03-16 19:26:28.016598998 +0100
+++ b/server/config.c   2018-03-16 19:26:22.631643000 +0100
@@ -1930,8 +1930,11 @@
          */
         rv = apr_dir_open(&dirp, path, ptemp);
         if (rv != APR_SUCCESS) {
-            return apr_psprintf(p, "Could not open config directory %s: %pm",
-                                path, &rv);
+            if (!optional)
+                return apr_psprintf(p, "Could not open config directory %s: %pm",
+                                    path, &rv);
+            else
+                return NULL;
         }
 
         candidates = apr_array_make(ptemp, 1, sizeof(fnames));
