diff --git a/server/config.c b/server/config.c
index dada063..697488c 100644
--- a/server/config.c
+++ b/server/config.c
@@ -1848,8 +1848,11 @@ static const char *process_resource_config_nofnmatch(server_rec *s,
          */
         rv = apr_dir_open(&dirp, path, ptemp);
         if (rv != APR_SUCCESS) {
-            return apr_psprintf(p, "Could not open config directory %s: %pm",
-                                path, &rv);
+            if (!optional)
+                return apr_psprintf(p, "Could not open config directory %s: %pm",
+                                    path, &rv);
+            else
+                return NULL;
         }
 
         candidates = apr_array_make(ptemp, 1, sizeof(fnames));
@@ -1934,8 +1937,11 @@ static const char *process_resource_config_fnmatch(server_rec *s,
      */
     rv = apr_dir_open(&dirp, path, ptemp);
     if (rv != APR_SUCCESS) {
-        return apr_psprintf(p, "Could not open config directory %s: %pm",
-                            path, &rv);
+            if (!optional)
+                return apr_psprintf(p, "Could not open config directory %s: %pm",
+                                    path, &rv);
+            else
+                return NULL;
     }
 
     candidates = apr_array_make(ptemp, 1, sizeof(fnames));
