From 76a27cae0e6951a7b6b25d25307cd952879c3fd7 Mon Sep 17 00:00:00 2001
From: Alberto Murillo <alberto.murillo.silva@intel.com>
Date: Wed, 22 Apr 2015 11:17:51 -0500
Subject: [PATCH] Look fo envvars in /etc/httpd

---
 support/apachectl.in | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/support/apachectl.in b/support/apachectl.in
index 3281c2e..52eb9dc 100644
--- a/support/apachectl.in
+++ b/support/apachectl.in
@@ -45,8 +45,8 @@ ARGV="$@"
 HTTPD='@exp_sbindir@/@progname@'
 #
 # pick up any necessary environment variables
-if test -f @exp_sbindir@/envvars; then
-  . @exp_sbindir@/envvars
+if test -f /etc/httpd/envvars; then
+  . /etc/httpd/envvars
 fi
 #
 # a command that outputs a formatted text version of the HTML at the
-- 
2.1.0

