From f42813e78df56d50ce614e50a2b20ee2c4fd44e7 Mon Sep 17 00:00:00 2001
From: Reilly Brogan <Reilly@reillybrogan.com>
Date: Sun, 20 Aug 2023 22:49:33 -0500
Subject: [PATCH] configure: Fix missing runstatedir

---
 configure | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/configure b/configure
index ef0efec..ff02817 100755
--- a/configure
+++ b/configure
@@ -1317,6 +1317,11 @@ do
   | -silent | --silent | --silen | --sile | --sil)
     silent=yes ;;
 
+  -runstatedir | --runstatedir | --runstatedi | --runstated | --runstate)
+    ac_prev=runstatedir ;;
+  -runstatedir=* | --runstatedir=* | --runstatedi=* | --runstated=* | --runstate=*)
+    runstatedir=$ac_optarg ;;
+
   -sbindir | --sbindir | --sbindi | --sbind | --sbin | --sbi | --sb)
     ac_prev=sbindir ;;
   -sbindir=* | --sbindir=* | --sbindi=* | --sbind=* | --sbin=* \
-- 
2.41.0

