From e45b53bbf5788272b15ddc9d69400185e7ea566a Mon Sep 17 00:00:00 2001
From: Nazar Stasiv <nazar@autistici.org>
Date: Mon, 31 Jan 2022 09:43:30 +0200
Subject: [PATCH] vendored-ssl

---
 api/Cargo.toml | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/api/Cargo.toml b/api/Cargo.toml
index 8662a9e..51ff86f 100644
--- a/api/Cargo.toml
+++ b/api/Cargo.toml
@@ -18,7 +18,7 @@ serde = { version = "1.0", features=["derive"] }
 futures = "0.3"
 
 http = "0.1"
-curl = { version = "0.4", default-features=false, features = ["static-ssl"] }
+curl = { version = "0.4" }
 isahc = "0.8"
 
 [dev-dependencies]
-- 
2.34.0

