From 2175866f05561f1f0ba34c4c89f55f6f9bc4f530 Mon Sep 17 00:00:00 2001
From: Peter O'Connor <peter@solus-project.com>
Date: Sat, 14 Apr 2018 20:03:04 +1000
Subject: [PATCH] Disable building unshipped code

Signed-off-by: Peter O'Connor <peter@solus-project.com>
---
 CMakeLists.txt | 2 --
 1 file changed, 2 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index f5b53e7..98e85d6 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -90,8 +90,6 @@ endif()
 kde_enable_exceptions()
 
 add_subdirectory(src)
-add_subdirectory(autotests)
-add_subdirectory(tests)
 
 # create a Config.cmake and a ConfigVersion.cmake file and install them
 set(CMAKECONFIG_INSTALL_DIR "${KDE_INSTALL_CMAKEPACKAGEDIR}/KF5CoreAddons")
-- 
2.16.3

