From 339ea5e70b8c048df0afa20d8fc088c2902786e9 Mon Sep 17 00:00:00 2001
From: David Faure <faure@kde.org>
Date: Sun, 2 Oct 2016 10:00:58 +0200
Subject: [PATCH] autotest: fix memory leak, detected by ASAN

---
 autotests/kjobtest.cpp | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/autotests/kjobtest.cpp b/autotests/kjobtest.cpp
index 88be4ac..139b9be 100644
--- a/autotests/kjobtest.cpp
+++ b/autotests/kjobtest.cpp
@@ -276,6 +276,7 @@ void KJobTest::testDelegateUsage()
     TestJob *job1 = new TestJob;
     TestJob *job2 = new TestJob;
     TestJobUiDelegate *delegate = new TestJobUiDelegate;
+    QPointer<TestJobUiDelegate> guard(delegate);
 
     QVERIFY(job1->uiDelegate() == 0);
     job1->setUiDelegate(delegate);
@@ -284,6 +285,10 @@ void KJobTest::testDelegateUsage()
     QVERIFY(job2->uiDelegate() == 0);
     job2->setUiDelegate(delegate);
     QVERIFY(job2->uiDelegate() == 0);
+
+    delete job1;
+    delete job2;
+    QVERIFY(guard.isNull()); // deleted by job1
 }
 
 void KJobTest::testNestedExec()
