From 5f475032b7c5f2c3680a7a8b856335a97f56ca2c Mon Sep 17 00:00:00 2001
From: David Faure <faure@kde.org>
Date: Sun, 2 Oct 2016 10:07:54 +0200
Subject: [PATCH] [autotest] fix memory leak

---
 autotests/kurlmimedatatest.cpp | 1 +
 1 file changed, 1 insertion(+)

diff --git a/autotests/kurlmimedatatest.cpp b/autotests/kurlmimedatatest.cpp
index 5e55d9e..264879f 100644
--- a/autotests/kurlmimedatatest.cpp
+++ b/autotests/kurlmimedatatest.cpp
@@ -135,4 +135,5 @@ void KUrlMimeDataTest::testMostLocalUrlList()
         QCOMPARE(qurls[i], static_cast<QUrl>(localUrls[i]));
     }
 
+    delete mimeData;
 }
