From a06cef31cc4c908bc9b76bd9d103fe9c60e0953f Mon Sep 17 00:00:00 2001
From: Montel Laurent <montel@kde.org>
Date: Tue, 11 Oct 2016 11:11:08 +0200
Subject: [PATCH] Add more autotests

---
 autotests/ktexttohtmltest.cpp | 15 +++++++++++++++
 1 file changed, 15 insertions(+)

diff --git a/autotests/ktexttohtmltest.cpp b/autotests/ktexttohtmltest.cpp
index c5690e8..0179a00 100644
--- a/autotests/ktexttohtmltest.cpp
+++ b/autotests/ktexttohtmltest.cpp
@@ -392,6 +392,21 @@ void KTextToHTMLTest::testHtmlConvert_data()
                                << KTextToHTML::Options(KTextToHTML::PreserveSpaces)
                                << "https://\"><!--";

+   QTest::newRow("url-exec-html-2") << "https://192.168.1.1:\"><!--"
+                               << KTextToHTML::Options(KTextToHTML::PreserveSpaces)
+                               << "https://192.168.1.1:\"><!--";
+
+   QTest::newRow("url-exec-html-3") << "https://<IP>:\"><!--"
+                               << KTextToHTML::Options(KTextToHTML::PreserveSpaces)
+                               << "https://<IP>:\"><!--";
+
+   QTest::newRow("url-exec-html-4") << "https://<IP>:/\"><!--"
+                               << KTextToHTML::Options(KTextToHTML::PreserveSpaces)
+                               << "https://<IP>:/\"><!--";
+
+   QTest::newRow("url-exec-html-5") << "https://<IP>:/\"><script>alert(1);</script><!--"
+                               << KTextToHTML::Options(KTextToHTML::PreserveSpaces)
+                               << "https://<IP>:/\"><script>alert(1);</script><!--";
 }
