From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Ikey Doherty <ikey@solus-project.com>
Date: Sat, 19 Mar 2016 23:58:19 +0000
Subject: [PATCH] Support local configs too for stateless implementation

Signed-off-by: Ikey Doherty <ikey@solus-project.com>
---
 conf.d/50-user.conf | 7 +++++++
 src/fcinit.c        | 2 ++
 2 files changed, 9 insertions(+)

diff --git a/conf.d/50-user.conf b/conf.d/50-user.conf
index d019f4d..cd7ad9f 100644
--- a/conf.d/50-user.conf
+++ b/conf.d/50-user.conf
@@ -2,12 +2,19 @@
 <!DOCTYPE fontconfig SYSTEM "urn:fontconfig:fonts.dtd">
 <fontconfig>
   <description>Load per-user customization files</description>
+	<!-- global OS wide defaults -->
+	<include ignore_missing="no">/usr/share/fonts/conf.d</include>
+	<include ignore_missing="no">/usr/share/fonts/fonts.conf</include>
+
 	<!--
 	    Load per-user customization files where stored on XDG Base Directory
 	    specification compliant places. it should be usually:
 	      $HOME/.config/fontconfig/conf.d
 	      $HOME/.config/fontconfig/fonts.conf
 	-->
+	<include ignore_missing="yes">/etc/fonts/conf.d</include>
+	<include ignore_missing="yes">/etc/fonts/fonts.conf</include>
+
 	<include ignore_missing="yes" prefix="xdg">fontconfig/conf.d</include>
 	<include ignore_missing="yes" prefix="xdg">fontconfig/fonts.conf</include>
 	<!-- the following elements will be removed in the future -->
diff --git a/src/fcinit.c b/src/fcinit.c
index c05cdc5..d9153ca 100644
--- a/src/fcinit.c
+++ b/src/fcinit.c
@@ -45,6 +45,8 @@ FcInitFallbackConfig (const FcChar8 *sysroot)
 	"  <dir prefix=\"xdg\">fonts</dir>" \
 	"  <cachedir>" FC_CACHEDIR "</cachedir>" \
 	"  <cachedir prefix=\"xdg\">fontconfig</cachedir>" \
+    "  <include ignore_missing=\"no\">/usr/share/fonts/conf.d</include>" \
+    "  <include ignore_missing=\"no\">/usr/share/fonts/fonts.conf</include>" \
 	"  <include ignore_missing=\"yes\">" CONFIGDIR "</include>" \
 	"  <include ignore_missing=\"yes\" prefix=\"xdg\">fontconfig/conf.d</include>" \
 	"  <include ignore_missing=\"yes\" prefix=\"xdg\">fontconfig/fonts.conf</include>" \
