From 748db20cb47d67967f006e91e10900db8226f0a5 Mon Sep 17 00:00:00 2001
From: Ikey Doherty <ikey@solus-project.com>
Date: Sat, 19 Mar 2016 23:58:19 +0000
Subject: [PATCH] Support local configs too for stateless implementation

Signed-off-by: Ikey Doherty <ikey@solus-project.com>
---
 conf.d/50-user.conf | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/conf.d/50-user.conf b/conf.d/50-user.conf
index 07c9182..6a3e8b2 100644
--- a/conf.d/50-user.conf
+++ b/conf.d/50-user.conf
@@ -7,6 +7,9 @@
 	      $HOME/.config/fontconfig/conf.d
 	      $HOME/.config/fontconfig/fonts.conf
 	-->
+	<include ignore_missing="yes">/etc/fonts/conf.d</include>
+	<include ignore_missing="yes">/etc/fonts/fonts.conf</include>
+
 	<include ignore_missing="yes" prefix="xdg">fontconfig/conf.d</include>
 	<include ignore_missing="yes" prefix="xdg">fontconfig/fonts.conf</include>
 	<!-- the following elements will be removed in the future -->
-- 
2.7.3

