From a63fd722003b68318bf6ca93a5cb990a569d319f Mon Sep 17 00:00:00 2001
From: Ikey Doherty <ikey@solus-project.com>
Date: Sat, 19 Mar 2016 18:12:48 +0000
Subject: [PATCH] Enforce Solus preferences

Signed-off-by: Ikey Doherty <ikey@solus-project.com>
---
 conf.d/60-latin.conf | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/conf.d/60-latin.conf b/conf.d/60-latin.conf
index 8604f70..bb1ad54 100644
--- a/conf.d/60-latin.conf
+++ b/conf.d/60-latin.conf
@@ -17,6 +17,7 @@
 	<alias>
 		<family>sans-serif</family>
 		<prefer>
+			<family>Clear Sans</family>
 			<family>DejaVu Sans</family>
 			<family>Bitstream Vera Sans</family>
 			<family>Verdana</family>
@@ -34,6 +35,8 @@
 	<alias>
 		<family>monospace</family>
 		<prefer>
+			<family>Hack</family>
+			<family>Droid Sans Mono</family>
 			<family>DejaVu Sans Mono</family>
 			<family>Bitstream Vera Sans Mono</family>
 			<family>Inconsolata</family>
-- 
2.7.3

