From 91066af5516d2c29979a3fead2c540799b9d3b74 Mon Sep 17 00:00:00 2001
From: Ikey Doherty <ikey@solus-project.com>
Date: Fri, 9 Dec 2016 02:38:46 +0000
Subject: [PATCH 2/3] Enforce sane defaults for Solus

Signed-off-by: Ikey Doherty <ikey@solus-project.com>
---
 conf.d/60-latin.conf | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/conf.d/60-latin.conf b/conf.d/60-latin.conf
index 35600ea..b766fc0 100644
--- a/conf.d/60-latin.conf
+++ b/conf.d/60-latin.conf
@@ -4,6 +4,7 @@
 	<alias>
 		<family>serif</family>
 		<prefer>
+			<family>Clear Sans</family>
 			<family>Bitstream Vera Serif</family>
 			<family>DejaVu Serif</family>
 			<family>Times New Roman</family>
@@ -17,6 +18,7 @@
 	<alias>
 		<family>sans-serif</family>
 		<prefer>
+			<family>Clear Sans</family>
 			<family>Bitstream Vera Sans</family>
 			<family>DejaVu Sans</family>
 			<family>Verdana</family>
@@ -34,6 +36,8 @@
 	<alias>
 		<family>monospace</family>
 		<prefer>
+			<family>Hack</family>
+			<family>Droid Sans Mono</family>
 			<family>Bitstream Vera Sans Mono</family>
 			<family>DejaVu Sans Mono</family>
 			<family>Inconsolata</family>
-- 
2.11.0

