Index: fontconfig-2.11.0/conf.d/40-nonlatin.conf
===================================================================
--- fontconfig-2.11.0.orig/conf.d/40-nonlatin.conf	2013-11-06 11:30:46.228295938 +0100
+++ fontconfig-2.11.0/conf.d/40-nonlatin.conf	2013-11-06 11:30:46.220295937 +0100
@@ -161,6 +161,10 @@
 		<default><family>sans-serif</family></default>
 	</alias>
 	<alias>
+		<family>MgOpen Moderna</family>
+		<default><family>sans-serif</family></default>
+	</alias>
+	<alias>
 		<family>Sazanami Gothic</family>
 		<default><family>sans-serif</family></default>
 	</alias>
Index: fontconfig-2.11.0/conf.d/60-latin.conf
===================================================================
--- fontconfig-2.11.0.orig/conf.d/60-latin.conf	2013-11-06 11:30:46.228295938 +0100
+++ fontconfig-2.11.0/conf.d/60-latin.conf	2013-11-06 11:30:46.220295937 +0100
@@ -4,8 +4,8 @@
 	<alias>
 		<family>serif</family>
 		<prefer>
-			<family>Bitstream Vera Serif</family>
 			<family>DejaVu Serif</family>
+			<family>Bitstream Vera Serif</family>
 			<family>Times New Roman</family>
 			<family>Thorndale AMT</family>
 			<family>Luxi Serif</family>
@@ -16,8 +16,8 @@
 	<alias>
 		<family>sans-serif</family>
 		<prefer>
-			<family>Bitstream Vera Sans</family>
 			<family>DejaVu Sans</family>
+			<family>Bitstream Vera Sans</family>
 			<family>Verdana</family>
 			<family>Arial</family>
 			<family>Albany AMT</family>
@@ -32,8 +32,8 @@
 	<alias>
 		<family>monospace</family>
 		<prefer>
-			<family>Bitstream Vera Sans Mono</family>
 			<family>DejaVu Sans Mono</family>
+			<family>Bitstream Vera Sans Mono</family>
 			<family>Inconsolata</family>
 			<family>Andale Mono</family>
 			<family>Courier New</family>
Index: fontconfig-2.11.0/conf.d/Makefile.am
===================================================================
--- fontconfig-2.11.0.orig/conf.d/Makefile.am	2013-11-06 11:30:46.228295938 +0100
+++ fontconfig-2.11.0/conf.d/Makefile.am	2013-11-06 11:30:46.220295937 +0100
@@ -26,6 +26,9 @@
 DOC_FILES = $(DOC_SOURCES:.in=)
 
 CONF_LINKS = \
+	10-antialias.conf \
+	10-hinting.conf \
+	10-hinting-slight.conf \
 	10-scale-bitmap-fonts.conf \
 	20-unhint-small-vera.conf \
 	30-urw-aliases.conf \
@@ -50,7 +53,12 @@
 
 templatedir = $(TEMPLATEDIR)
 template_DATA =				\
+	10-antialias.conf		\
 	10-autohint.conf		\
+	10-hinting.conf		\
+	10-hinting-full.conf		\
+	10-hinting-medium.conf		\
+	10-hinting-slight.conf		\
 	10-no-sub-pixel.conf		\
 	10-scale-bitmap-fonts.conf	\
 	10-sub-pixel-bgr.conf		\
@@ -70,6 +78,7 @@
 	49-sansserif.conf		\
 	50-user.conf			\
 	51-local.conf			\
+	53-monospace-lcd-filter.conf			\
 	60-latin.conf			\
 	65-fonts-persian.conf		\
 	65-khmer.conf			\
Index: fontconfig-2.11.0/conf.d/Makefile.in
===================================================================
--- fontconfig-2.11.0.orig/conf.d/Makefile.in	2013-11-06 11:30:46.228295938 +0100
+++ fontconfig-2.11.0/conf.d/Makefile.in	2013-11-06 11:30:46.224295937 +0100
@@ -327,6 +327,9 @@
 DOC_SOURCES = README.in
 DOC_FILES = $(DOC_SOURCES:.in=)
 CONF_LINKS = \
+	10-antialias.conf \
+	10-hinting.conf \
+	10-hinting-slight.conf \
 	10-scale-bitmap-fonts.conf \
 	20-unhint-small-vera.conf \
 	30-urw-aliases.conf \
@@ -349,7 +352,12 @@
 config_DATA = $(DOC_FILES)
 templatedir = $(TEMPLATEDIR)
 template_DATA = \
+	10-antialias.conf		\
 	10-autohint.conf		\
+	10-hinting.conf		\
+	10-hinting-full.conf		\
+	10-hinting-medium.conf		\
+	10-hinting-slight.conf		\
 	10-no-sub-pixel.conf		\
 	10-scale-bitmap-fonts.conf	\
 	10-sub-pixel-bgr.conf		\
@@ -369,6 +377,7 @@
 	49-sansserif.conf		\
 	50-user.conf			\
 	51-local.conf			\
+	53-monospace-lcd-filter.conf			\
 	60-latin.conf			\
 	65-fonts-persian.conf		\
 	65-khmer.conf			\
