Index: fontconfig-2.10.93/conf.d/53-monospace-lcd-filter.conf
===================================================================
--- /dev/null	1970-01-01 00:00:00.000000000 +0000
+++ fontconfig-2.10.93/conf.d/53-monospace-lcd-filter.conf	2013-05-21 13:46:49.330358589 +0200
@@ -0,0 +1,22 @@
+<?xml version="1.0"?>
+<!DOCTYPE fontconfig SYSTEM "fonts.dtd">
+<!-- conf.d/monospace-lcd-filter.conf -->
+<fontconfig>
+<!--  Use legacy LCD filter on smaller Monospace fonts -->
+  <match target="font">
+    <test name="family">
+      <string>DejaVu Sans Mono</string>
+      <string>Bitstream Vera Sans Mono</string>
+    </test>
+    <test name="pixelsize" compare="less_eq">
+      <double>12.0</double>
+    </test>
+
+    <edit name="lcd_filter" mode="assign">
+      <const>lcdlegacy</const>
+    </edit>
+    <edit name="hintstyle" mode="assign">
+      <const>hintfull</const>
+    </edit>
+  </match>
+</fontconfig>
