Index: fontconfig-2.10.93/src/fcname.c
===================================================================
--- fontconfig-2.10.93.orig/src/fcname.c	2013-05-21 13:46:51.050358515 +0200
+++ fontconfig-2.10.93/src/fcname.c	2013-05-21 13:46:51.046358515 +0200
@@ -188,6 +188,7 @@
     { (FcChar8 *) "lcddefault",	    "lcdfilter",    FC_LCD_DEFAULT },
     { (FcChar8 *) "lcdlight",	    "lcdfilter",    FC_LCD_LIGHT },
     { (FcChar8 *) "lcdlegacy",	    "lcdfilter",    FC_LCD_LEGACY },
+    { (FcChar8 *) "lcdfilterlegacy",	    "lcdfilter",    FC_LCD_LEGACY },
 };
 
 #define NUM_FC_CONSTANTS   (sizeof _FcBaseConstants/sizeof _FcBaseConstants[0])
