Index: fontconfig-2.11.0/src/fcname.c
===================================================================
--- fontconfig-2.11.0.orig/src/fcname.c	2013-11-06 11:30:58.368296342 +0100
+++ fontconfig-2.11.0/src/fcname.c	2013-11-06 11:30:58.364296342 +0100
@@ -190,6 +190,7 @@
     { (FcChar8 *) "lcddefault",	    "lcdfilter",    FC_LCD_DEFAULT },
     { (FcChar8 *) "lcdlight",	    "lcdfilter",    FC_LCD_LIGHT },
     { (FcChar8 *) "lcdlegacy",	    "lcdfilter",    FC_LCD_LEGACY },
+    { (FcChar8 *) "lcdfilterlegacy",	    "lcdfilter",    FC_LCD_LEGACY },
 };
 
 #define NUM_FC_CONSTANTS   (sizeof _FcBaseConstants/sizeof _FcBaseConstants[0])
