Index: fontconfig-2.11.0/conf.d/Makefile.am
===================================================================
--- fontconfig-2.11.0.orig/conf.d/Makefile.am	2013-11-06 11:31:03.312296507 +0100
+++ fontconfig-2.11.0/conf.d/Makefile.am	2013-11-06 11:31:03.308296506 +0100
@@ -30,6 +30,7 @@
 	10-hinting.conf \
 	10-hinting-slight.conf \
 	10-scale-bitmap-fonts.conf \
+	11-lcdfilter-default.conf \
 	20-unhint-small-vera.conf \
 	30-urw-aliases.conf \
 	30-metric-aliases.conf \
Index: fontconfig-2.11.0/conf.d/Makefile.in
===================================================================
--- fontconfig-2.11.0.orig/conf.d/Makefile.in	2013-11-06 11:31:03.312296507 +0100
+++ fontconfig-2.11.0/conf.d/Makefile.in	2013-11-06 11:31:03.308296506 +0100
@@ -331,6 +331,7 @@
 	10-hinting.conf \
 	10-hinting-slight.conf \
 	10-scale-bitmap-fonts.conf \
+	11-lcdfilter-default.conf \
 	20-unhint-small-vera.conf \
 	30-urw-aliases.conf \
 	30-metric-aliases.conf \
