Index: fontconfig-2.10.93/conf.d/Makefile.am
===================================================================
--- fontconfig-2.10.93.orig/conf.d/Makefile.am	2013-05-21 13:46:55.182358336 +0200
+++ fontconfig-2.10.93/conf.d/Makefile.am	2013-05-21 13:46:55.178358336 +0200
@@ -30,6 +30,7 @@
 	10-hinting.conf \
 	10-hinting-slight.conf \
 	10-scale-bitmap-fonts.conf \
+	11-lcdfilter-default.conf \
 	20-unhint-small-vera.conf \
 	30-urw-aliases.conf \
 	30-metric-aliases.conf \
Index: fontconfig-2.10.93/conf.d/Makefile.in
===================================================================
--- fontconfig-2.10.93.orig/conf.d/Makefile.in	2013-05-21 13:46:55.182358336 +0200
+++ fontconfig-2.10.93/conf.d/Makefile.in	2013-05-21 13:46:55.178358336 +0200
@@ -304,6 +304,7 @@
 	10-hinting.conf \
 	10-hinting-slight.conf \
 	10-scale-bitmap-fonts.conf \
+	11-lcdfilter-default.conf \
 	20-unhint-small-vera.conf \
 	30-urw-aliases.conf \
 	30-metric-aliases.conf \
