Index: fontconfig-2.10.93/conf.d/Makefile.am
===================================================================
--- fontconfig-2.10.93.orig/conf.d/Makefile.am	2013-05-21 13:46:57.634358230 +0200
+++ fontconfig-2.10.93/conf.d/Makefile.am	2013-05-21 13:46:57.626358230 +0200
@@ -43,6 +43,7 @@
 	65-fonts-persian.conf \
 	65-nonlatin.conf \
 	69-unifont.conf \
+	70-no-bitmaps.conf \
 	80-delicious.conf \
 	90-synthetic.conf
 
Index: fontconfig-2.10.93/conf.d/Makefile.in
===================================================================
--- fontconfig-2.10.93.orig/conf.d/Makefile.in	2013-05-21 13:46:57.634358230 +0200
+++ fontconfig-2.10.93/conf.d/Makefile.in	2013-05-21 13:46:57.626358230 +0200
@@ -317,6 +317,7 @@
 	65-fonts-persian.conf \
 	65-nonlatin.conf \
 	69-unifont.conf \
+	70-no-bitmaps.conf \
 	80-delicious.conf \
 	90-synthetic.conf
 
