Index: fontconfig-2.11.0/conf.d/Makefile.am
===================================================================
--- fontconfig-2.11.0.orig/conf.d/Makefile.am	2013-11-06 11:31:06.052296598 +0100
+++ fontconfig-2.11.0/conf.d/Makefile.am	2013-11-06 11:31:06.044296598 +0100
@@ -43,6 +43,7 @@
 	65-fonts-persian.conf \
 	65-nonlatin.conf \
 	69-unifont.conf \
+	70-no-bitmaps.conf \
 	80-delicious.conf \
 	90-synthetic.conf
 
Index: fontconfig-2.11.0/conf.d/Makefile.in
===================================================================
--- fontconfig-2.11.0.orig/conf.d/Makefile.in	2013-11-06 11:31:06.052296598 +0100
+++ fontconfig-2.11.0/conf.d/Makefile.in	2013-11-06 11:31:06.048296598 +0100
@@ -344,6 +344,7 @@
 	65-fonts-persian.conf \
 	65-nonlatin.conf \
 	69-unifont.conf \
+	70-no-bitmaps.conf \
 	80-delicious.conf \
 	90-synthetic.conf
 
