From 97ad8cd89b47213fe211d7a08082a313f6393ff6 Mon Sep 17 00:00:00 2001
From: Akira TAGOH <akira@tagoh.org>
Date: Wed, 10 Jun 2026 22:06:16 +0900
Subject: [PATCH 1/2] ci: set SOURCE_DATE_EPOCH to the build script

---
 .gitlab-ci/build.py | 1 +
 1 file changed, 1 insertion(+)

diff --git a/.gitlab-ci/build.py b/.gitlab-ci/build.py
index 17692c39..3652bf1c 100755
--- a/.gitlab-ci/build.py
+++ b/.gitlab-ci/build.py
@@ -110,6 +110,7 @@ class Build:
         os.environ.setdefault("CI_MERGE_REQUEST_PROJECT_URL", "https://gitlab.freedesktop.org/fontconfig/fontconfig")
         os.environ.setdefault("CI_COMMIT_REF_NAME", "main")
         os.environ.setdefault("BUILDLOG", str(pathlib.Path(os.environ["BUILDDIR"]) / "fc-build.log"))
+        os.environ.setdefault("SOURCE_DATE_EPOCH", str(int(time.time())))
 
         self.builddir = pathlib.Path(os.environ["BUILDDIR"])
         self.prefix = pathlib.Path(os.environ["PREFIX"])
-- 
GitLab


From c6c94e13b4a0b0aaa15823ea1093ee0eb17a5944 Mon Sep 17 00:00:00 2001
From: Akira TAGOH <akira@tagoh.org>
Date: Wed, 10 Jun 2026 22:39:31 +0900
Subject: [PATCH 2/2] test: unset SOURCE_DATE_EPOCH for some test cases

They depend on the timestamp change.
We can't fix the timestamp with SOURCE_DATE_EPOCH for them.
---
 test/test_basic.py | 6 +++++-
 1 file changed, 5 insertions(+), 1 deletion(-)

diff --git a/test/test_basic.py b/test/test_basic.py
index ebb2c3d8..30ab3080 100644
--- a/test/test_basic.py
+++ b/test/test_basic.py
@@ -163,6 +163,8 @@ def test_with_complicated_dir_structure(fctest, fcfont):
 
 
 def test_subdir_with_out_of_date_cache(fctest, fcfont):
+    # This test case depends on the timestamp. Do not fix with SOURCE_DATE_EPOCH
+    fctest.env.pop('SOURCE_DATE_EPOCH')
     fctest.setup()
     fctest.install_font([], 'a')
     for ret, stdout, stderr in fctest.run_cache([str(Path(fctest.fontdir.name) / 'a')]):
@@ -188,7 +190,9 @@ def test_subdir_with_out_of_date_cache(fctest, fcfont):
         assert out == out_expected
 
 
-def test_new_file_with_out_of_date_cache(fctest, fcfont):
+def test_new_file_with_out_of_date_cache(monkeypatch, fctest, fcfont):
+    # This test case depends on the timestamp. Do not fix with SOURCE_DATE_EPOCH
+    fctest.env.pop('SOURCE_DATE_EPOCH')
     fctest.setup()
     fctest.install_font(fcfont.fonts[0], '.')
     for ret, stdout, stderr in fctest.run_cache([fctest.fontdir.name]):
-- 
GitLab

