--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -1,5 +1,3 @@
 # Generated from src.pro.
 
-add_subdirectory(glslang)
-add_subdirectory(SPIRV-Cross)
 add_subdirectory(shadertools)
--- a/src/shadertools/CMakeLists.txt
+++ b/src/shadertools/CMakeLists.txt
@@ -4,6 +4,17 @@
 ## ShaderTools Module:
 #####################################################################
 
+find_package(glslang QUIET)
+find_package(PkgConfig QUIET)
+
+pkg_check_modules(SPV_X spirv-cross-c-shared)
+
+find_path(GLSL_INCLUDE_DIR NAMES SPIRV/GlslangToSpv.h
+  PATHS
+    /usr/local/include/glslang
+    /usr/include/glslang
+)
+
 qt_internal_add_module(ShaderTools
     EXCEPTIONS
     SOURCES
@@ -16,16 +27,21 @@ qt_internal_add_module(ShaderTools
     DEFINES
         QT_BUILD_SHADERTOOLS_LIB
     INCLUDE_DIRECTORIES
-        ../3rdparty/SPIRV-Cross
-        ../3rdparty/glslang
+        ${SPV_X_INCLUDE_DIRS}
+        ${GLSL_INCLUDE_DIR}
     LIBRARIES
         Qt::GuiPrivate
 # special case begin
-        Qt::BundledSpirv_Cross
-        Qt::BundledGlslang_Glslang
-        Qt::BundledGlslang_Spirv
-        Qt::BundledGlslang_Osdependent
-        Qt::BundledGlslang_Oglcompiler
+        ${SPV_X_LIBRARIES}
+        glslang
+        MachineIndependent
+        GenericCodeGen
+        SPIRV
+        SPVRemapper
+        OSDependent
+        OGLCompiler
+        HLSL
+        glslang
 # special case end
     PUBLIC_LIBRARIES
         Qt::Core
--- a/src/shadertools/qspirvcompiler.cpp
+++ b/src/shadertools/qspirvcompiler.cpp
@@ -178,8 +178,6 @@ bool QSpirvCompilerPrivate::readFile(con
     return true;
 }
 
-using namespace QtShaderTools;
-
 class Includer : public glslang::TShader::Includer
 {
 public:
