From b911373b6271615ae62eb9ddef306692c75aa2e9 Mon Sep 17 00:00:00 2001
From: Jacob <Jacalz@users.noreply.github.com>
Date: Wed, 20 Mar 2019 20:15:11 +0100
Subject: [PATCH] 0001-update-desktop-entry.patch

---
 pacman.desktop | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/pacman.desktop b/pacman.desktop
index 4727340..02d88f0 100644
--- a/pacman.desktop
+++ b/pacman.desktop
@@ -1,7 +1,7 @@
 [Desktop Entry]
-Name=Pacman (SDL)
+Name=Pacman
 Comment=Yet another pacman clone in C/C++ and SDL
 Exec=pacman
-Icon=/usr/local/share/pacman/gfx/pacman_desktop.png
+Icon=/usr/share/pacman/gfx/pacman_desktop.png
 Type=Application
 Categories=Game;ArcadeGame;