diff --color -ur collada-dom-2.5.0/CMakeLists.txt collada-dom-2.5.0.minizip/CMakeLists.txt
--- collada-dom-2.5.0/CMakeLists.txt	2017-10-21 07:05:24.000000000 +0200
+++ collada-dom-2.5.0.minizip/CMakeLists.txt	2022-04-12 20:23:53.174323323 +0200
@@ -220,15 +220,7 @@
   endif()
 endif()
 
-pkg_check_modules(minizip minizip)
-if(minizip_FOUND)
-  set(MINIZIP_INCLUDE_DIR ${minizip_INCLUDE_DIRS})
-else()
-  message(STATUS "compiling minizip from sources and linking statically")
-  add_subdirectory(dom/external-libs/minizip-1.1)
-  set(MINIZIP_INCLUDE_DIR ${CMAKE_CURRENT_SOURCE_DIR}/dom/external-libs/minizip-1.1 ${ZLIB_INCLUDE_DIR})
-endif()
-
+pkg_check_modules(minizip REQUIRED minizip)
 pkg_check_modules(liburiparser liburiparser)
 
 if(liburiparser_FOUND)
diff --color -ur collada-dom-2.5.0/dom/include/dae/daeZAEUncompressHandler.h collada-dom-2.5.0.minizip/dom/include/dae/daeZAEUncompressHandler.h
--- collada-dom-2.5.0/dom/include/dae/daeZAEUncompressHandler.h	2017-10-21 07:05:24.000000000 +0200
+++ collada-dom-2.5.0.minizip/dom/include/dae/daeZAEUncompressHandler.h	2022-04-12 20:35:24.083885697 +0200
@@ -9,7 +9,7 @@
 #ifndef __DAE_ZAE_UNCOMPRESS_HANDLER_H__
 #define __DAE_ZAE_UNCOMPRESS_HANDLER_H__
 
-#include <unzip.h>
+#include <minizip/unzip.h>
 #include <libxml/xmlreader.h>
 #include <dae/daeURI.h>
 
diff --color -ur collada-dom-2.5.0/dom/src/modules/LIBXMLPlugin/daeLIBXMLPlugin.cpp collada-dom-2.5.0.minizip/dom/src/modules/LIBXMLPlugin/daeLIBXMLPlugin.cpp
--- collada-dom-2.5.0/dom/src/modules/LIBXMLPlugin/daeLIBXMLPlugin.cpp	2017-10-21 07:05:24.000000000 +0200
+++ collada-dom-2.5.0.minizip/dom/src/modules/LIBXMLPlugin/daeLIBXMLPlugin.cpp	2022-04-12 20:34:33.767143363 +0200
@@ -32,7 +32,7 @@
 #include <iomanip>
 using namespace std;
 
-#include <zip.h> // for saving compressed files
+#include <minizip/zip.h> // for saving compressed files
 #ifdef _WIN32
 #include <iowin32.h>
 #else
