diff --color -ur collada-dom-2.5.0.minizip/dom/src/modules/LIBXMLPlugin/daeLIBXMLPlugin.cpp collada-dom-2.5.0.minizip.include-zlib/dom/src/modules/LIBXMLPlugin/daeLIBXMLPlugin.cpp
--- collada-dom-2.5.0.minizip/dom/src/modules/LIBXMLPlugin/daeLIBXMLPlugin.cpp	2022-04-12 20:34:33.767143363 +0200
+++ collada-dom-2.5.0.minizip.include-zlib/dom/src/modules/LIBXMLPlugin/daeLIBXMLPlugin.cpp	2022-04-12 20:40:12.034389822 +0200
@@ -33,6 +33,7 @@
 using namespace std;
 
 #include <minizip/zip.h> // for saving compressed files
+#include <zlib.h>
 #ifdef _WIN32
 #include <iowin32.h>
 #else
