From 2e191eb8a45994c094fb6a31e63b68964f1d20f5 Mon Sep 17 00:00:00 2001
From: Ikey Doherty <ikey.doherty@gmail.com>
Date: Thu, 4 Sep 2014 21:55:34 +0100
Subject: [PATCH] ifconfig: Force use of /proc/net/dev

Signed-off-by: Ikey Doherty <ikey.doherty@gmail.com>
---
 ifconfig/system/linux.h | 1 +
 1 file changed, 1 insertion(+)

diff --git a/ifconfig/system/linux.h b/ifconfig/system/linux.h
index f909d17..09aebce 100644
--- a/ifconfig/system/linux.h
+++ b/ifconfig/system/linux.h
@@ -117,3 +117,4 @@ _IU_EXTRN (tx_heartbeat_errors)
 _IU_EXTRN (tx_window_errors)
 
 #endif
+#define PATH_PROCNET_DEV "/proc/net/dev"
-- 
1.8.4

