From a84c183bb25201c46fa64316be619b10fdec674f Mon Sep 17 00:00:00 2001
From: Christophe Giboudeaux <christophe@krop.fr>
Date: Fri, 9 Mar 2018 14:23:30 +0100
Subject: [PATCH] Also detect fluidsynth 1.1.xx

---
 cmake/FindFluidSynth.cmake | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/cmake/FindFluidSynth.cmake b/cmake/FindFluidSynth.cmake
index 27135a3..d0f3e86 100644
--- a/cmake/FindFluidSynth.cmake
+++ b/cmake/FindFluidSynth.cmake
@@ -73,13 +73,13 @@ find_path(FluidSynth_INCLUDE_DIRS
   HINTS ${PC_FluidSynth_INCLUDEDIR}
 )
 
-#set(FluidSynth_VERSION "${PC_FluidSynth_VERSION}")
+set(FluidSynth_VERSION "${PC_FluidSynth_VERSION}")
 
 if(NOT FluidSynth_VERSION)
   if(EXISTS "${FluidSynth_INCLUDE_DIRS}/fluidsynth/version.h")
     file(STRINGS "${FluidSynth_INCLUDE_DIRS}/fluidsynth/version.h" _FLUIDSYNTH_VERSION_H
-         REGEX "^#define[ ]+FLUIDSYNTH_VERSION[ ]+\"[0-9].[0-9].[0-9]\"")
-    string(REGEX REPLACE "^#define[ ]+FLUIDSYNTH_VERSION[ ]+\"([0-9].[0-9].[0-9])\".*" "\\1" FluidSynth_VERSION "${_FLUIDSYNTH_VERSION_H}")
+         REGEX "^#define[ ]+FLUIDSYNTH_VERSION.*$")
+    string(REGEX REPLACE "^#define[ ]+FLUIDSYNTH_VERSION[ ]+\"([0-9]+.[0-9]+.[0-9]+)\".*" "\\1" FluidSynth_VERSION "${_FLUIDSYNTH_VERSION_H}")
     unset(_FLUIDSYNTH_VERSION_H)
   endif()
 endif()
