From 8e4d12bed4a84930d9c9a1dbfb4a88d13e6af73c Mon Sep 17 00:00:00 2001
From: Silke Hofstra <silke@slxh.eu>
Date: Sun, 10 Jan 2021 18:58:09 +0100
Subject: [PATCH] support exfatprogs

---
 src/udiskslinuxfsinfo.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/udiskslinuxfsinfo.c b/src/udiskslinuxfsinfo.c
index 15af26c8..96a540e4 100644
--- a/src/udiskslinuxfsinfo.c
+++ b/src/udiskslinuxfsinfo.c
@@ -117,11 +117,11 @@ const FSInfo _fs_info[] =
     },
     {
       FS_EXFAT,
-      "exfatlabel $DEVICE $LABEL",
+      "tune.exfat -L $LABEL $DEVICE",
       NULL,
       FALSE, /* supports_online_label_rename */
       FALSE, /* supports_owners */
-      "mkexfatfs -n $LABEL $DEVICE",
+      "mkfs.exfat -L $LABEL $DEVICE",
       NULL,
       NULL, /* option_no_discard */
     },
-- 
2.30.0

