From 2ccf32d6f4647208ec6cf22be775d345cc3f6368 Mon Sep 17 00:00:00 2001
From: Thomas Staudinger <Staudi.Kaos@gmail.com>
Date: Fri, 8 Apr 2022 18:49:54 +0200
Subject: [PATCH] Fix go install

---
 build/build_linux.sh | 18 +++++++++++++++---
 1 file changed, 15 insertions(+), 3 deletions(-)

diff --git a/build/build_linux.sh b/build/build_linux.sh
index 093a56ce..7b555f1f 100755
--- a/build/build_linux.sh
+++ b/build/build_linux.sh
@@ -53,9 +53,21 @@ else
 	export GOPATH=$PWD:$GOPATH
 fi
 
-(cd "$PWD/src/github.com/visualfc/gotools" && go install -ldflags "-s" -v)
-(cd "$PWD/src/github.com/visualfc/gocode" && go install -ldflags "-s" -v)
-(cd "$PWD/src/github.com/fatih/gomodifytags" && go install -ldflags "-s" -v)
+go install -ldflags "-s" -v github.com/visualfc/gotools@latest
+
+if [ $? -ge 1 ]; then
+	echo 'error, go install fail'
+	exit 1
+fi
+
+go install -ldflags "-s" -v github.com/visualfc/gocode@latest
+
+if [ $? -ge 1 ]; then
+	echo 'error, go install fail'
+	exit 1
+fi
+
+go install -ldflags "-s" -v github.com/fatih/gomodifytags@latest
 
 if [ $? -ge 1 ]; then
 	echo 'error, go install fail'
-- 
2.35.1

