From 49b96d467655e7d6e1c66fd2b4e161a76dabdea6 Mon Sep 17 00:00:00 2001
From: Campbell Jones <git@serebit.com>
Date: Tue, 23 Feb 2021 15:30:39 -0500
Subject: [PATCH] Enable tiered compilation

---
 src/hotspot/cpu/x86/c1_globals_x86.hpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/hotspot/cpu/x86/c1_globals_x86.hpp b/src/hotspot/cpu/x86/c1_globals_x86.hpp
index 5ee965bff..ae5212fa0 100644
--- a/src/hotspot/cpu/x86/c1_globals_x86.hpp
+++ b/src/hotspot/cpu/x86/c1_globals_x86.hpp
@@ -39,7 +39,7 @@ define_pd_global(bool, InlineIntrinsics,               true );
 define_pd_global(bool, PreferInterpreterNativeStubs,   false);
 define_pd_global(bool, ProfileTraps,                   false);
 define_pd_global(bool, UseOnStackReplacement,          true );
-define_pd_global(bool, TieredCompilation,              false);
+define_pd_global(bool, TieredCompilation,              true);
 define_pd_global(intx, CompileThreshold,               1500 );
 
 define_pd_global(intx,   OnStackReplacePercentage,     933  );
-- 
2.30.0

