diff -pruN a/src/build_mozc.py b/src/build_mozc.py
--- a/src/build_mozc.py	2020-11-19 13:18:22.391113319 +0900
+++ b/src/build_mozc.py	2020-11-19 13:20:00.583082777 +0900
@@ -579,7 +579,7 @@ def BuildWithNinja(options, targets):
 
   for target in targets:
     (_, target_name) = target.split(':')
-    RunOrDie([ninja, '-C', build_arg, target_name])
+    RunOrDie([ninja, '-v', '-C', build_arg, target_name])
 
 
 def BuildOnWindows(targets):
@@ -588,7 +588,7 @@ def BuildOnWindows(targets):
 
   for target in targets:
     (build_arg, target_name) = target.split(':')
-    RunOrDie([ninja, '-C', build_arg, target_name])
+    RunOrDie([ninja, '-v', '-C', build_arg, target_name])
 
 
 def BuildMain(options, targets):
