diff -pruN a/src/gyp/common.gypi b/src/gyp/common.gypi
--- a/src/gyp/common.gypi	2021-11-15 17:48:33.103785797 +0900
+++ b/src/gyp/common.gypi	2021-11-15 17:54:09.104908953 +0900
@@ -73,9 +73,6 @@
       '-Wno-deprecated-declarations',
       '-Wwrite-strings',
       '<@(extra_warning_cflags)',
-
-      '-Wno-unknown-warning-option',
-      '-Wno-inconsistent-missing-override',
     ],
 
     # gcc_cflags will be shared with Mac and Linux
@@ -105,7 +102,6 @@
     ],
     # Libraries for GNU/Linux environment.
     'linux_ldflags': [
-      '-lc++',
       '-pthread',
       '-Wl,--build-id',
     ],
@@ -210,9 +206,11 @@
               '-Wno-covered-switch-default',
               '-Wno-unnamed-type-template-args',
               '-Wno-c++11-narrowing',
+	      '-std=c++17',
+	      '-stdlib=libc++',
             ],
           }],
-          ['compiler_target=="clang" or compiler_target=="gcc"', {
+          ['compiler_target=="gcc"', {
             'cflags_cc': [
               '-std=c++17',
             ],
@@ -230,9 +228,11 @@
               '-Wno-covered-switch-default',
               '-Wno-unnamed-type-template-args',
               '-Wno-c++11-narrowing',
+	      '-std=c++17',
+	      '-stdlib=libc++',
             ],
           }],
-          ['compiler_host=="clang" or compiler_host=="gcc"', {
+          ['compiler_host=="gcc"', {
             'cflags_cc': [
               '-std=c++17',
             ],
