diff -pruN a/src/base/init_mozc.cc b/src/base/init_mozc.cc
--- a/src/base/init_mozc.cc	2023-06-15 19:21:31.000000000 +0900
+++ b/src/base/init_mozc.cc	2023-08-17 21:59:52.675769639 +0900
@@ -87,7 +87,8 @@ std::string GetLogFilePathFromProgramNam
 void ParseCommandLineFlags(int argc, char **argv) {
   absl::flags_internal::ParseCommandLineImpl(
       argc, argv,
-      absl::flags_internal::ArgvListAction::kRemoveParsedArgs,
+      // Abseil 20230802.0 does not use ArgvListAction
+      // absl::flags_internal::ArgvListAction::kRemoveParsedArgs,
       // Suppress help messages invoked by --help and others.
       // Use UsageFlagsAction::kHandleUsage to enable it.
       absl::flags_internal::UsageFlagsAction::kIgnoreUsage,
