diff -Naur a/src/dictionary/file/codec.cc b/src/dictionary/file/codec.cc
--- a/src/dictionary/file/codec.cc	2023-06-15 06:21:31.000000000 -0400
+++ b/src/dictionary/file/codec.cc	2024-01-24 10:33:35.503269086 -0500
@@ -42,6 +42,7 @@
 #include "dictionary/file/codec_util.h"
 #include "dictionary/file/section.h"
 #include "absl/status/status.h"
+#include "absl/strings/str_cat.h"
 #include "absl/strings/string_view.h"
 
 namespace mozc {
diff -Naur a/src/rewriter/a11y_description_rewriter.cc b/src/rewriter/a11y_description_rewriter.cc
--- a/src/rewriter/a11y_description_rewriter.cc	2023-06-15 06:21:31.000000000 -0400
+++ b/src/rewriter/a11y_description_rewriter.cc	2024-01-24 10:41:40.010311197 -0500
@@ -37,6 +37,7 @@
 #include "base/util.h"
 #include "data_manager/serialized_dictionary.h"
 #include "protocol/commands.pb.h"
+#include "absl/strings/str_cat.h"
 #include "absl/strings/string_view.h"
 
 namespace mozc {
diff -Naur a/src/rewriter/unicode_rewriter.cc b/src/rewriter/unicode_rewriter.cc
--- a/src/rewriter/unicode_rewriter.cc	2023-06-15 06:21:31.000000000 -0400
+++ b/src/rewriter/unicode_rewriter.cc	2024-01-24 12:11:24.699434633 -0500
@@ -45,6 +45,7 @@
 #include "absl/strings/ascii.h"
 #include "absl/strings/match.h"
 #include "absl/strings/numbers.h"
+#include "absl/strings/str_cat.h"
 #include "absl/strings/str_format.h"
 #include "absl/strings/string_view.h"
 
diff -Naur a/src/rewriter/version_rewriter.cc b/src/rewriter/version_rewriter.cc
--- a/src/rewriter/version_rewriter.cc	2023-06-15 06:21:31.000000000 -0400
+++ b/src/rewriter/version_rewriter.cc	2024-01-24 12:01:26.279796975 -0500
@@ -38,6 +38,7 @@
 #include "converter/segments.h"
 #include "request/conversion_request.h"
 #include "absl/container/flat_hash_map.h"
+#include "absl/strings/str_cat.h"
 #include "absl/strings/string_view.h"
 
 namespace mozc {
