diff -pruN a/src/build_mozc.py b/src/build_mozc.py
--- a/src/build_mozc.py	2023-06-23 22:37:22.303384091 +0900
+++ b/src/build_mozc.py	2023-06-23 22:38:13.003206120 +0900
@@ -77,14 +77,14 @@ EXT_THIRD_PARTY_DIR = os.path.join(MOZC_
 #   https://github.com/google/mozc/issues/567
 # Qt5 candidate window built by Bazel is the alternative.
 #   https://github.com/google/mozc/blob/master/docs/build_mozc_in_docker.md
-USE_DEPRECATED_GTK_RENDERER = False
+USE_DEPRECATED_GTK_RENDERER = True
 
 # Ibus build is no longer supported by GYP build.
 # The build rules and code will be removed in future.
 #   https://github.com/google/mozc/issues/567
 # Bazel build is the alternative.
 #   https://github.com/google/mozc/blob/master/docs/build_mozc_in_docker.md
-USE_UNSUPPORTED_IBUS_BUILD = False
+USE_UNSUPPORTED_IBUS_BUILD = True
 
 sys.path.append(SRC_DIR)
 
@@ -172,7 +172,7 @@ def GetGypFileNames(options):
   elif options.target_platform == 'Linux':
     gyp_file_names.extend(glob.glob('%s/unix/emacs/*.gyp' % SRC_DIR))
     if USE_UNSUPPORTED_IBUS_BUILD:
-      gyp_file_names.extend('%s/unix/ibus/*.gyp' % SRC_DIR)
+      gyp_file_names.extend(glob.glob('%s/unix/ibus/*.gyp' % SRC_DIR))
   gyp_file_names.sort()
   return gyp_file_names
 
diff -pruN a/src/unix/ibus/ibus.gyp b/src/unix/ibus/ibus.gyp
--- a/src/unix/ibus/ibus.gyp	2023-06-15 19:21:31.000000000 +0900
+++ b/src/unix/ibus/ibus.gyp	2023-06-23 22:38:00.131997436 +0900
@@ -239,45 +239,6 @@
       ],
     },
     {
-      'target_name': 'ibus_mozc_test',
-      'type': 'executable',
-      'sources': [
-        'key_event_handler_test.cc',
-        'key_translator_test.cc',
-        'message_translator_test.cc',
-        'mozc_engine_test.cc',
-        'path_util_test.cc',
-        'surrounding_text_util_test.cc',
-      ],
-      'dependencies': [
-        '../../base/absl.gyp:absl_strings',
-        '../../base/base.gyp:base',
-        '../../client/client.gyp:client',
-        '../../client/client.gyp:client_mock',
-        '../../protocol/protocol.gyp:commands_proto',
-        '../../testing/testing.gyp:gtest_main',
-        'ibus_mozc_lib',
-      ],
-      'variables': {
-        'test_size': 'small',
-      },
-    },
-    # Test cases meta target: this target is referred from gyp/tests.gyp
-    {
-      'target_name': 'ibus_all_test',
-      'type': 'none',
-      'dependencies': [
-        'ibus_mozc_test',
-      ],
-      'conditions': [
-        ['enable_gtk_renderer==1', {
-          'dependencies': [
-            'candidate_window_handler_test',
-          ],
-        }],
-      ],
-    },
-    {
       # Meta target to set up build environment for ibus. Required 'cflags'
       # and 'link_settings' will be automatically injected into any target
       # which directly or indirectly depends on this target.
