From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Muhammad Alfi Syahrin <malfisya.dev@hotmail.com>
Date: Tue, 15 Apr 2025 08:01:14 +0700
Subject: [PATCH] Hide desktop entry by default

---
 src/app/vte.desktop.in | 1 +
 1 file changed, 1 insertion(+)

diff --git a/src/app/vte.desktop.in b/src/app/vte.desktop.in
index 0c425fa..faeb09b 100644
--- a/src/app/vte.desktop.in
+++ b/src/app/vte.desktop.in
@@ -16,3 +16,4 @@ X-TerminalArgTitle=--title
 X-TerminalArgDir=--working-directory
 # X-TerminalArgHold=
 X-KDE-AuthorizeAction=shell_access
+NoDisplay=true
