Index: prepare_vscode.sh
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/prepare_vscode.sh b/prepare_vscode.sh
--- a/prepare_vscode.sh	(revision eb5d6e23a9abe76460a22e41cccacf7a7d5fea96)
+++ b/prepare_vscode.sh	(date 1780507632660)
@@ -215,7 +215,7 @@
   if [[ "${CI_BUILD}" != "no" && "${OS_NAME}" == "osx" ]]; then
     CXX=clang++ npm ci && break
   else
-    npm ci && break
+    npm-22 ci && break
   fi
 
   if [[ $i == 5 ]]; then
