Index: prepare_vscode.sh
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/prepare_vscode.sh b/prepare_vscode.sh
--- a/prepare_vscode.sh	(revision 4015f2d0191311733aa5dbb2abde8101dce63eef)
+++ b/prepare_vscode.sh	(date 1783902719143)
@@ -213,7 +213,7 @@
   if [[ "${CI_BUILD}" != "no" && "${OS_NAME}" == "osx" ]]; then
     CXX=clang++ npm ci && break
   else
-    npm ci && break
+    npm-24 ci && break
   fi
 
   if [[ $i == 5 ]]; then
