Index: src/stable/resources/linux/code.desktop
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/src/stable/resources/linux/code.desktop b/src/stable/resources/linux/code.desktop
--- a/src/stable/resources/linux/code.desktop	(revision eb967d7bffbcbaec6f9aa89ef7515b3cfe46dd0b)
+++ b/src/stable/resources/linux/code.desktop	(date 1783646401966)
@@ -1,15 +1,15 @@
 [Desktop Entry]
-Name=@@NAME_LONG@@
+Name=VSCodium
 Comment=Code Editing. Redefined.
 GenericName=Text Editor
 GenericName[ar]=محرر نصوص
-Exec=@@EXEC@@ %F
-Icon=@@ICON@@
+Exec=/usr/bin/codium %F
+Icon=vscodium
 Type=Application
 StartupNotify=false
-StartupWMClass=@@NAME_SHORT@@
+StartupWMClass=VSCodium
 Categories=TextEditor;Development;IDE;
-MimeType=text/plain;inode/directory;application/x-@@NAME@@-workspace;
+MimeType=text/plain;inode/directory;application/x-codium-workspace;
 Actions=new-empty-window;
 Keywords=vscodium;codium;vscode;

@@ -25,5 +25,5 @@
 Name[ru]=Новое пустое окно
 Name[zh_CN]=新建空窗口
 Name[zh_TW]=開新空視窗
-Exec=@@EXEC@@ --new-window %F
-Icon=@@ICON@@
+Exec=/usr/bin/codium --new-window %F
+Icon=vscodium
