Index: src/stable/resources/linux/code-url-handler.desktop
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/src/stable/resources/linux/code-url-handler.desktop b/src/stable/resources/linux/code-url-handler.desktop
--- a/src/stable/resources/linux/code-url-handler.desktop	(revision 4015f2d0191311733aa5dbb2abde8101dce63eef)
+++ b/src/stable/resources/linux/code-url-handler.desktop	(date 1783907950911)
@@ -1,12 +1,12 @@
 [Desktop Entry]
-Name=@@NAME_LONG@@ - URL Handler
+Name=VSCodium - URL Handler
 Comment=Code Editing. Redefined.
 GenericName=Text Editor
-Exec=@@EXEC@@ --open-url %U
-Icon=@@ICON@@
+Exec=/usr/bin/codium --open-url %U
+Icon=vscodium
 Type=Application
 NoDisplay=true
 StartupNotify=true
 Categories=Utility;TextEditor;Development;IDE;
-MimeType=x-scheme-handler/@@URLPROTOCOL@@;
+MimeType=x-scheme-handler/vscodium;
 Keywords=vscodium;codium;vscode;