From 2f15f67b65ac2d91cffdbaaa907dd94bac73cbcb Mon Sep 17 00:00:00 2001
From: Campbell Jones <git@serebit.com>
Date: Thu, 6 Aug 2020 16:11:39 -0400
Subject: [PATCH] Ensure java home is set

---
 _gradle                | 4 ++++
 gradle-completion.bash | 4 ++++
 2 files changed, 8 insertions(+)

diff --git a/_gradle b/_gradle
index e1fc3d3..f661cfa 100644
--- a/_gradle
+++ b/_gradle
@@ -86,6 +86,10 @@ __gradle-generate-script-cache() {
 __gradle-generate-tasks-cache() {
     __gradle-set-files-checksum
 
+    if [[ -z $JAVA_HOME ]]; then
+        export JAVA_HOME="/usr/lib64/openjdk-17"
+    fi
+
     # Use Gradle wrapper when it exists.
     local gradle_cmd="gradle"
     if [[ -x "$project_root_dir/gradlew" ]]; then
diff --git a/gradle-completion.bash b/gradle-completion.bash
index e392346..cfc02b0 100644
--- a/gradle-completion.bash
+++ b/gradle-completion.bash
@@ -273,6 +273,10 @@ __gradle-notify-tasks-cache-build() {
 __gradle-generate-tasks-cache() {
     __gradle-set-files-checksum
 
+    if [[ -z $JAVA_HOME ]]; then
+        export JAVA_HOME="/usr/lib64/openjdk-17"
+    fi
+
     # Use Gradle wrapper when it exists.
     local gradle_cmd="gradle"
     if [[ -x "$project_root_dir/gradlew" ]]; then
-- 
2.27.0

