diff --git a/subprojects/native/src/main/java/org/gradle/internal/nativeintegration/console/NativePlatformConsoleDetector.java b/subprojects/native/src/main/java/org/gradle/internal/nativeintegration/console/NativePlatformConsoleDetector.java
index 9a33da7801e..d61017f2414 100644
--- a/subprojects/native/src/main/java/org/gradle/internal/nativeintegration/console/NativePlatformConsoleDetector.java
+++ b/subprojects/native/src/main/java/org/gradle/internal/nativeintegration/console/NativePlatformConsoleDetector.java
@@ -16,6 +16,8 @@
 
 package org.gradle.internal.nativeintegration.console;
 
+import net.rubygrapefruit.platform.NativeException;
+import net.rubygrapefruit.platform.terminal.TerminalOutput;
 import net.rubygrapefruit.platform.terminal.Terminals;
 import org.gradle.internal.os.OperatingSystem;
 
@@ -40,12 +42,20 @@ public ConsoleMetaData getConsole() {
 
         boolean stdout = terminals.isTerminal(Stdout);
         boolean stderr = terminals.isTerminal(Stderr);
-        if (stdout) {
-            return new NativePlatformConsoleMetaData(stdout, stderr, terminals.getTerminal(Stdout));
-        } else if (stderr) {
-            return new NativePlatformConsoleMetaData(stdout, stderr, terminals.getTerminal(Stderr));
+
+        TerminalOutput terminal;
+
+        try {
+            if (stdout) {
+                return new NativePlatformConsoleMetaData(stdout, stderr, terminals.getTerminal(Stdout));
+            } else if (stderr) {
+                return new NativePlatformConsoleMetaData(stdout, stderr, terminals.getTerminal(Stderr));
+            } else {
+                return null;
+            }
+        } catch (NativeException ex) {
+            return null;
         }
-        return null;
     }
 
     @Override
