diff -ru iotop-0.6.orign/sbin/iotop iotop-0.6/sbin/iotop
--- iotop-0.6.orign/sbin/iotop	2013-05-27 00:44:18.000000000 +0200
+++ iotop-0.6/sbin/iotop	2015-11-15 22:47:24.468058681 +0100
@@ -1,4 +1,4 @@
-#!/usr/bin/python
+#!/usr/bin/python3
 # iotop: Display I/O usage of processes in a top like UI
 # Copyright (c) 2007, 2008 Guillaume Chazarain <guichaz@gmail.com>, GPLv2
 # See iotop --help for some help
