diff -up iotop-0.6/iotop/ui.py.batchprintutf8 iotop-0.6/iotop/ui.py
--- iotop-0.6/iotop/ui.py.batchprintutf8	2016-11-14 11:21:23.690185257 +0100
+++ iotop-0.6/iotop/ui.py	2016-11-14 11:23:26.511040007 +0100
@@ -444,7 +444,7 @@ class IOTopUI(object):
                 if self.options.quiet <= int(first_time):
                     print(''.join(titles))
             for l in lines:
-                print(l)
+                print(l.encode('utf-8'))
             sys.stdout.flush()
         else:
             self.win.erase()
