From 13405f447376a049ca02811278d2025ac67dd979 Mon Sep 17 00:00:00 2001
From: Ikey Doherty <ikey@solus-project.com>
Date: Thu, 17 Mar 2016 15:09:45 +0000
Subject: [PATCH] Support a stateless configuration

Signed-off-by: Ikey Doherty <ikey@solus-project.com>
---
 Makefile.in  | 10 ++++++----
 bin/quilt.in |  2 +-
 2 files changed, 7 insertions(+), 5 deletions(-)

diff --git a/Makefile.in b/Makefile.in
index 416da20..f3911da 100644
--- a/Makefile.in
+++ b/Makefile.in
@@ -265,6 +265,7 @@ $(patsubst %.in,%,$(wildcard bin/*.in quilt/*.in quilt/scripts/*.in)) :: Makefil
 		-e 's:@STAT_HARDLINK''@:$(STAT_HARDLINK):g'		\
 		-e 's:@VERSION''@:$(VERSION):g'				\
 		-e 's:@ETCDIR''@:$(etcdir):g'				\
+		-e 's:@DATADIR''@:$(datadir):g'				\
 		-e 's:@LOCALEDIR''@:$(localedir):g'			\
 		-e 's:@DOCSUBDIR''@:$(docdir):g'			\
 		$< > $@
@@ -329,11 +330,12 @@ endif
 	$(INSTALL) -d $(BUILD_ROOT)$(mandir)/man1
 	$(INSTALL) -m 644 $(MAN1) $(BUILD_ROOT)$(mandir)/man1/
 
-	$(INSTALL) -d $(BUILD_ROOT)$(etcdir)
-	$(INSTALL) -d $(BUILD_ROOT)$(etcdir)/bash_completion.d
+	$(INSTALL) -d $(BUILD_ROOT)$(datadir)
+	$(INSTALL) -d $(BUILD_ROOT)$(datadir)/bash-completion
+	$(INSTALL) -d $(BUILD_ROOT)$(datadir)/bash-completion/completions
 	$(INSTALL) -m 644 bash_completion				\
-		   $(BUILD_ROOT)$(etcdir)/bash_completion.d/quilt
-	$(INSTALL) -m 644 quilt.quiltrc $(BUILD_ROOT)$(etcdir)/
+		   $(BUILD_ROOT)$(datadir)/bash-completion/completions/quilt
+	$(INSTALL) -D -m 644 quilt.quiltrc $(BUILD_ROOT)$(datadir)/defaults/quilt/quiltrc
 	$(INSTALL) -d $(BUILD_ROOT)$(emacsdir)/
 	$(INSTALL) -m 644 lib/quilt.el $(BUILD_ROOT)$(emacsdir)/
 
diff --git a/bin/quilt.in b/bin/quilt.in
index 2271833..4f5201b 100644
--- a/bin/quilt.in
+++ b/bin/quilt.in
@@ -22,7 +22,7 @@ export QUILT_DIR
 
 if [ -z "$QUILTRC" ]
 then
-	for QUILTRC in $HOME/.quiltrc @ETCDIR@/quilt.quiltrc; do
+	for QUILTRC in $HOME/.quiltrc @ETCDIR@/quilt.quiltrc @DATADIR@/defaults/quilt/quilt.rc; do
 		[ -e $QUILTRC ] && break
 	done
 	export QUILTRC
-- 
2.7.3

