diff --git a/kernel/common/inc/nv-lock.h b/kernel/common/inc/nv-lock.h
index f2ef557..2880faf 100644
--- a/kernel/common/inc/nv-lock.h
+++ b/kernel/common/inc/nv-lock.h
@@ -13,7 +13,11 @@
 
 #include "conftest.h"
 
+#include <linux/version.h>
 #include <linux/spinlock.h>
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(4, 14, 0)
+#include <linux/sched/task_stack.h>
+#endif
 
 #if defined(NV_LINUX_SEMAPHORE_H_PRESENT)
 #include <linux/semaphore.h>
