diff --git a/src/lib/probe.c b/src/lib/probe.c
index b85abc1..1ed130f 100644
--- a/src/lib/probe.c
+++ b/src/lib/probe.c
@@ -72,9 +72,9 @@ static char *cbm_get_luks_uuid(const char *part)
         /* i.e. /sys/block/dm-1/slaves/dm-0/slaves/sdb1/dev
          * or /sys/block/dm-1/slaves/sdb1/dev
          */
-        npath = string_printf("%s/block/%s/slaves/*{,/slaves/*}/dev", sys, part);
+        npath = string_printf("%s/block/%s/slaves/*/slaves/*/dev", sys, part);
 
-        glob(npath, GLOB_DOOFFS | GLOB_BRACE, NULL, &glo);
+        glob(npath, GLOB_DOOFFS, NULL, &glo);
 
         if (glo.gl_pathc < 1) {
                 globfree(&glo);
