From 193d8fce1e0f46a0bded23d8d37fc607200e2284 Mon Sep 17 00:00:00 2001
From: Silke Hofstra <silke@slxh.eu>
Date: Wed, 29 Jul 2020 19:36:48 +0200
Subject: [PATCH 01/13] Fix LUKS glob order

---
 src/lib/probe.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/lib/probe.c b/src/lib/probe.c
index a6ffc70..f1eff4d 100644
--- a/src/lib/probe.c
+++ b/src/lib/probe.c
@@ -72,7 +72,7 @@ static char *cbm_get_luks_uuid(const char *part)
         /* i.e. /sys/block/dm-1/slaves/dm-0/slaves/sdb1/dev
          * or /sys/block/dm-1/slaves/sdb1/dev
          */
-        npath = string_printf("%s/block/%s/slaves/*{,/slaves/*}/dev", sys, part);
+        npath = string_printf("%s/block/%s/slaves/*{/slaves/*,}/dev", sys, part);
 
         glob(npath, GLOB_DOOFFS | GLOB_BRACE, NULL, &glo);
 
-- 
2.40.1

