From 6c0a383788be4c70396512cb4b74225ee084284f Mon Sep 17 00:00:00 2001
From: Ikey Doherty <michael.i.doherty@intel.com>
Date: Fri, 16 Jun 2017 15:08:33 +0100
Subject: [PATCH 2/3] cli: Return the correct value when set_timeout works

Signed-off-by: Ikey Doherty <michael.i.doherty@intel.com>
---
 src/cli/ops/timeout.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/cli/ops/timeout.c b/src/cli/ops/timeout.c
index 42cbc36..8b5d8c9 100644
--- a/src/cli/ops/timeout.c
+++ b/src/cli/ops/timeout.c
@@ -89,7 +89,7 @@ bool cbm_command_set_timeout(int argc, char **argv)
                 fprintf(stdout, "New timeout value is: %d\n", n_val);
         }
 
-        return false;
+        return true;
 }
 
 bool cbm_command_get_timeout(int argc, char **argv)
-- 
2.13.1

