From c8268601c89fde4b7e6db1ac7d7d1c8bc3d3dfc3 Mon Sep 17 00:00:00 2001
From: Ikey Doherty <michael.i.doherty@intel.com>
Date: Fri, 16 Jun 2017 15:08:48 +0100
Subject: [PATCH 3/3] tests: Run clang-format to deduplicate header includes

Signed-off-by: Ikey Doherty <michael.i.doherty@intel.com>
---
 tests/check-core.c | 1 -
 1 file changed, 1 deletion(-)

diff --git a/tests/check-core.c b/tests/check-core.c
index f49d725..60c49b5 100644
--- a/tests/check-core.c
+++ b/tests/check-core.c
@@ -23,7 +23,6 @@
 #include "nica/array.h"
 #include "nica/files.h"
 #include "util.h"
-#include "util.h"
 #include "writer.h"
 
 #include "blkid-harness.h"
-- 
2.13.1

