From 756c06ffcfe1742f215ed2f90da051d28cacd416 Mon Sep 17 00:00:00 2001
From: Pierre-Yves <pyu@riseup.net>
Date: Sun, 28 Apr 2019 18:30:24 +0200
Subject: [PATCH 1/2] Fix for languages other than English

Signed-off-by: Pierre-Yves <pyu@riseup.net>
---
 menulibre/util.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/menulibre/util.py b/menulibre/util.py
index dcca52d..29e547a 100644
--- a/menulibre/util.py
+++ b/menulibre/util.py
@@ -96,7 +96,7 @@ def getRelatedKeys(menu_item_type, key_only=False):
 def escapeText(text):
     if text is None:
         return ""
-    return GLib.markup_escape_text(text, len(text))
+    return GLib.markup_escape_text(text)
 
 
 def getProcessUsername(process):
-- 
2.21.0

