From f1b796e4c66ffb6c3c875ac54df0ac59c13f2d6e Mon Sep 17 00:00:00 2001
From: Pierre-Yves <pyu@riseup.net>
Date: Sun, 28 Apr 2019 18:32:17 +0200
Subject: [PATCH 2/2] Should only be visible on Gnome and Budgie

Signed-off-by: Pierre-Yves <pyu@riseup.net>
---
 menulibre.desktop.in | 1 -
 1 file changed, 1 deletion(-)

diff --git a/menulibre.desktop.in b/menulibre.desktop.in
index 292dc38..2507f85 100644
--- a/menulibre.desktop.in
+++ b/menulibre.desktop.in
@@ -4,7 +4,6 @@ Type=Application
 _Name=Menu Editor
 _Comment=Add or remove applications from the menu
 Icon=menulibre
-OnlyShowIn=Budgie;Cinnamon;GNOME;KDE;LXDE;LXQt;MATE;Pantheon;Unity;XFCE;
 Exec=menulibre
 Categories=GNOME;GTK;Settings;DesktopSettings;Utility;
 Keywords=Configuration;Menu;User;
-- 
2.21.0

