From 99ebcff78637b1581aef3ba7c02c2553874eb8a1 Mon Sep 17 00:00:00 2001
From: "F. von Gellhorn" <flinux@vongellhorn.ch>
Date: Sun, 20 Sep 2020 17:04:54 +0200
Subject: [PATCH 1/1] Set a nicer theme as default

---
 src/preferences.cpp | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/src/preferences.cpp b/src/preferences.cpp
index 7aee5fa..bb20eb9 100644
--- a/src/preferences.cpp
+++ b/src/preferences.cpp
@@ -411,7 +411,7 @@ void Preferences::reset() {
 
 #if STYLE_SWITCHING
 	#if QT_VERSION >= 0x050000
-	style = "Fusion";
+	style = "Breeze";
 	#else
 	style="";
 	#endif
@@ -486,8 +486,8 @@ void Preferences::reset() {
 
 	allow_video_movement = false;
 
-	gui = "DefaultGUI";
-	iconset = "H2O";
+	gui = "MiniGUI";
+	iconset = "PapirusDark";
 
 
 #if USE_MINIMUMSIZE
-- 
2.27.0

