From 8c8b203974606d2e43fe63def1efe548fc4ec19b Mon Sep 17 00:00:00 2001
From: Martin Ziemer <gituser@horrad.de>
Date: Mon, 16 Mar 2026 12:09:08 +0100
Subject: [PATCH] Remove unused requirement

---
 pyproject.toml | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/pyproject.toml b/pyproject.toml
index 32d329ba..fafa258c 100644
--- a/pyproject.toml
+++ b/pyproject.toml
@@ -1,5 +1,5 @@
 [build-system]
-requires = ["setuptools", "wheel", "pytest-runner"]
+requires = ["setuptools", "wheel"]
 build-backend = "setuptools.build_meta"
 
 [tool.black]
