From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Auke Kok <auke-jan.h.kok@intel.com>
Date: Wed, 7 Nov 2018 14:38:03 -0800
Subject: [PATCH] Make SSH stateless.

---
 servconf.c | 7 +++----
 1 file changed, 3 insertions(+), 4 deletions(-)

diff --git a/servconf.c b/servconf.c
index 4d1910f..aeab8da 100644
--- a/servconf.c
+++ b/servconf.c
@@ -416,10 +416,6 @@ fill_default_server_options(ServerOptions *options)
 		    &options->authorized_keys_files,
 		    &options->num_authkeys_files,
 		    _PATH_SSH_USER_PERMITTED_KEYS);
-		opt_array_append("[default]", 0, "AuthorizedKeysFiles",
-		    &options->authorized_keys_files,
-		    &options->num_authkeys_files,
-		    _PATH_SSH_USER_PERMITTED_KEYS2);
 	}
 	if (options->permit_tun == -1)
 		options->permit_tun = SSH_TUNMODE_NO;
@@ -2486,6 +2482,9 @@ load_server_config(const char *filename, struct sshbuf *conf)
 
 	debug2_f("filename %s", filename);
 	if ((f = fopen(filename, "r")) == NULL) {
+		if (errno == ENOENT)
+                       return;
+
 		perror(filename);
 		exit(1);
 	}
