From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Reilly Brogan <reilly@reillybrogan.com>
Date: Tue, 7 Jul 2026 14:14:18 -0500
Subject: [PATCH] Use config drop-ins

---
 ssh_config  | 7 +++++++
 sshd_config | 7 +++++++
 2 files changed, 14 insertions(+)

diff --git a/ssh_config b/ssh_config
index 238a0c5e3..637c2ed3d 100644
--- a/ssh_config
+++ b/ssh_config
@@ -43,3 +43,10 @@
 #   ProxyCommand ssh -q -W %h:%p gateway.example.com
 #   RekeyLimit 1G 1h
 #   UserKnownHostsFile ~/.ssh/known_hosts.d/%k
+
+# Package drop-ins
+Include /usr/share/defaults/ssh/ssh_config.d/*.conf
+
+# To modify the system-wide ssh configuration, create a  *.conf  file under
+#  /etc/ssh/ssh_config.d/  which will be automatically included below
+Include /etc/ssh/ssh_config.d/*.conf
diff --git a/sshd_config b/sshd_config
index 0f4a3a724..8cd8cd96e 100644
--- a/sshd_config
+++ b/sshd_config
@@ -116,3 +116,10 @@ Subsystem	sftp	/usr/libexec/sftp-server
 #	AllowTcpForwarding no
 #	PermitTTY no
 #	ForceCommand cvs server
+
+# Package drop-ins
+Include /usr/share/defaults/ssh/sshd_config.d/*.conf
+
+# To modify the system-wide sshd configuration, create a  *.conf  file under
+#  /etc/ssh/sshd_config.d/  which will be automatically included below
+Include /etc/ssh/sshd_config.d/*.conf
