From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Auke Kok <auke-jan.h.kok@intel.com>
Date: Wed, 7 Nov 2018 14:42:50 -0800
Subject: [PATCH] Default default secure ciphers.

---
 cipher.h   | 6 ++++++
 servconf.c | 2 ++
 2 files changed, 8 insertions(+)

diff --git a/cipher.h b/cipher.h
index 6533ff2bb..e0eba8f21 100644
--- a/cipher.h
+++ b/cipher.h
@@ -47,6 +47,12 @@
 #define CIPHER_ENCRYPT		1
 #define CIPHER_DECRYPT		0
 
+#define SECURE_CIPHERS 	"chacha20-poly1305@openssh.com,"\
+						"blowfish-cbc,"\
+						"aes256-gcm@openssh.com,"\
+						"aes128-gcm@openssh.com,"\
+						"aes256-ctr,aes192-ctr,aes128-ctr"
+
 struct sshcipher;
 struct sshcipher_ctx;
 
diff --git a/servconf.c b/servconf.c
index 48ec8c4ec..0ebbd3c1e 100644
--- a/servconf.c
+++ b/servconf.c
@@ -469,6 +469,8 @@ fill_default_server_options(ServerOptions *options)
 		options->ip_qos_interactive = IPTOS_DSCP_EF;
 	if (options->ip_qos_bulk == -1)
 		options->ip_qos_bulk = IPTOS_DSCP_CS0;
+	if (options->ciphers == NULL)
+		options->ciphers = xstrdup(SECURE_CIPHERS);
 	if (options->version_addendum == NULL)
 		options->version_addendum = xstrdup("");
 	if (options->fwd_opts.streamlocal_bind_mask == (mode_t)-1)
