From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Auke Kok <auke-jan.h.kok@intel.com>
Date: Wed, 7 Nov 2018 14:43:20 -0800
Subject: [PATCH] Always use PAM by default.

---
 servconf.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/servconf.c b/servconf.c
index 82df98a..d39f018 100644
--- a/servconf.c
+++ b/servconf.c
@@ -277,7 +277,7 @@ fill_default_server_options(ServerOptions *options)
 
 	/* Portable-specific options */
 	if (options->use_pam == -1)
-		options->use_pam = 0;
+		options->use_pam = 1;
 
 	/* Standard Options */
 	if (options->num_host_key_files == 0) {
