From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Auke Kok <auke-jan.h.kok@intel.com>
Date: Wed, 7 Nov 2018 14:43:20 -0800
Subject: [PATCH] Always use PAM by default.

---
 servconf.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/servconf.h b/servconf.h
index 9e64e4673..db329e4b2 100644
--- a/servconf.h
+++ b/servconf.h
@@ -272,7 +272,7 @@ SSHCONF_ALIAS(KeepAlive, TCPKeepAlive, SSHCFG_GLOBAL)
 
 #ifdef USE_PAM
 #define SSHD_CONFIG_ENTRIES_PAM \
-SSHCONF_INTFLAG(use_pam, UsePAM, SSHCFG_GLOBAL, 0, SSHCFG_COPY_NONE) \
+SSHCONF_INTFLAG(use_pam, UsePAM, SSHCFG_GLOBAL, 1, SSHCFG_COPY_NONE) \
 SSHCONF_STRING(pam_service_name, PAMServiceName, SSHCFG_GLOBAL, SSHCFG_COPY_NONE)
 #else
 #define SSHD_CONFIG_ENTRIES_PAM \
