From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: William Douglas <william.douglas@intel.com>
Date: Thu, 27 Oct 2016 18:21:20 +0000
Subject: [PATCH] Set default server keep alive

---
 readconf.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/readconf.c b/readconf.c
index 0f27652..f4cec35 100644
--- a/readconf.c
+++ b/readconf.c
@@ -2584,7 +2584,7 @@ fill_default_options(Options * options)
 	if (options->verify_host_key_dns == -1)
 		options->verify_host_key_dns = 0;
 	if (options->server_alive_interval == -1)
-		options->server_alive_interval = 0;
+		options->server_alive_interval = 120;
 	if (options->server_alive_count_max == -1)
 		options->server_alive_count_max = 3;
 	if (options->control_master == -1)
