From 57892a9d68f2e2a5150a97a6eeff731dcdf28385 Mon Sep 17 00:00:00 2001
From: Reilly Brogan <reilly@reillybrogan.com>
Date: Sat, 24 Sep 2022 15:27:48 -0500
Subject: [PATCH 1/1] Update sshd_config to reflect UsePAM yes default

---
 sshd_config | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/sshd_config b/sshd_config
index c423eba1..f8cec78f 100644
--- a/sshd_config
+++ b/sshd_config
@@ -79,7 +79,7 @@ AuthorizedKeysFile	.ssh/authorized_keys
 # If you just want the PAM account and session checks to run without
 # PAM authentication, then enable this but set PasswordAuthentication
 # and KbdInteractiveAuthentication to 'no'.
-#UsePAM no
+#UsePAM yes
 
 #AllowAgentForwarding yes
 #AllowTcpForwarding yes
-- 
2.35.4

