From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Reilly Brogan <reilly@reillybrogan.com>
Date: Mon, 1 Jul 2024 13:29:12 -0500
Subject: [PATCH] Load stateless config file

---
 pathnames.h | 1 +
 sshd.c      | 9 +++++++--
 2 files changed, 8 insertions(+), 2 deletions(-)

diff --git a/pathnames.h b/pathnames.h
index 61c5f846..9de4a279 100644
--- a/pathnames.h
+++ b/pathnames.h
@@ -35,6 +35,7 @@
  * should be world-readable.
  */
 #define _PATH_SERVER_CONFIG_FILE	SSHDIR "/sshd_config"
+#define _PATH_SERVER_CONFIG_FILE_STATELESS "/usr/share/defaults/etc/ssh/sshd_config"
 #define _PATH_HOST_CONFIG_FILE		SSHDIR "/ssh_config"
 #define _PATH_HOST_DSA_KEY_FILE		SSHDIR "/ssh_host_dsa_key"
 #define _PATH_HOST_ECDSA_KEY_FILE	SSHDIR "/ssh_host_ecdsa_key"
diff --git a/sshd.c b/sshd.c
index ed54fc6d..6f5d6734 100644
--- a/sshd.c
+++ b/sshd.c
@@ -1400,8 +1400,13 @@ main(int ac, char **av)
 	/* Fetch our configuration */
 	if ((cfg = sshbuf_new()) == NULL)
 		fatal("sshbuf_new config failed");
-	if (strcasecmp(config_file_name, "none") != 0)
-		load_server_config(config_file_name, cfg);
+	if (strcasecmp(config_file_name, "none") != 0) {
+		if (access(config_file_name, R_OK) != 0 && strcmp(config_file_name, _PATH_SERVER_CONFIG_FILE) == 0) {
+			load_server_config(_PATH_SERVER_CONFIG_FILE_STATELESS, cfg);
+		} else {
+			load_server_config(config_file_name, cfg);
+		}
+	}
 
 	parse_server_config(&options, config_file_name, cfg,
 	    &includes, NULL, 0);
