From 10821825e5d80dd9112ac43cd444ea77b4b09df8 Mon Sep 17 00:00:00 2001
From: Joey Riches <josephriches@gmail.com>
Date: Fri, 15 Mar 2024 22:58:08 +0000
Subject: [PATCH 1/1] policy: Change group to sudo and allow package-remove

---
 policy/org.freedesktop.packagekit.rules | 7 ++++++-
 1 file changed, 6 insertions(+), 1 deletion(-)

diff --git a/policy/org.freedesktop.packagekit.rules b/policy/org.freedesktop.packagekit.rules
index 6a1c8a701..73fdbd51c 100644
--- a/policy/org.freedesktop.packagekit.rules
+++ b/policy/org.freedesktop.packagekit.rules
@@ -1,7 +1,12 @@
 polkit.addRule(function(action, subject) {
     if (action.id == "org.freedesktop.packagekit.package-install" &&
         subject.active == true && subject.local == true &&
-        subject.isInGroup("wheel")) {
+        subject.isInGroup("sudo")) {
+            return polkit.Result.YES;
+    }
+    if (action.id == "org.freedesktop.packagekit.package-remove" &&
+        subject.active == true && subject.local == true &&
+        subject.isInGroup("sudo")) {
             return polkit.Result.YES;
     }
 });
-- 
2.44.0

