From bf2444ac0f0a2cd7cdfe2e391765b2342e164231 Mon Sep 17 00:00:00 2001
From: Dejan <dejan.mzc@gmail.com>
Date: Thu, 9 Jul 2020 23:31:22 +0200
Subject: [PATCH] dataL: add another device ID for G203

---
 data/svgs/svg-lookup.ini | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/data/svgs/svg-lookup.ini b/data/svgs/svg-lookup.ini
index 07979b7..8061c86 100644
--- a/data/svgs/svg-lookup.ini
+++ b/data/svgs/svg-lookup.ini
@@ -3,7 +3,7 @@ DeviceMatch=usb:1ea7:4011
 Svg=fallback.svg
 
 [Logitech G102/G103]
-DeviceMatch=usb:046d:c084
+DeviceMatch=usb:046d:c084;usb:046d:c092
 Svg=logitech-g102-g203.svg
 
 [Logitech G300]
