From 5c2544e7258372f85eb9f791d613730e28c45853 Mon Sep 17 00:00:00 2001
From: Joey Riches <josephriches@gmail.com>
Date: Wed, 20 Apr 2022 11:32:12 +0100
Subject: [PATCH 1/1] Call the right cython executable

---
 tools/cythonize.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/tools/cythonize.py b/tools/cythonize.py
index 402824c..fb8d8a7 100755
--- a/tools/cythonize.py
+++ b/tools/cythonize.py
@@ -97,7 +97,7 @@ def process_pyx(fromfile, tofile, cwd):
 
     try:
         try:
-            r = subprocess.call(['cython'] + flags + ["-o", tofile, fromfile], cwd=cwd)
+            r = subprocess.call(['cython3'] + flags + ["-o", tofile, fromfile], cwd=cwd)
             if r != 0:
                 raise Exception('Cython failed')
         except OSError as e:
-- 
2.35.3

