From c68c86f35e7e9b62817705a4cb1fb5c2b2fb4870 Mon Sep 17 00:00:00 2001
From: Ikey Doherty <ikey@solus-project.com>
Date: Fri, 16 Dec 2016 05:04:48 +0000
Subject: [PATCH] Support stateless glibc ldconfig configuration

Signed-off-by: Ikey Doherty <ikey@solus-project.com>
---
 dracut.sh | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/dracut.sh b/dracut.sh
index 5e2feba..37d0ee4 100755
--- a/dracut.sh
+++ b/dracut.sh
@@ -1514,7 +1514,7 @@ done
 
 if [[ $kernel_only != yes ]]; then
     # make sure that library links are correct and up to date
-    for f in /etc/ld.so.conf /etc/ld.so.conf.d/*; do
+    for f in /etc/ld.so.conf /etc/ld.so.conf.d/* /usr/share/defaults/etc/ld.so.conf /usr/share/ld.so.conf.d/*; do
         [[ -f $f ]] && inst_simple "$f"
     done
     if ! ldconfig -r "$initdir"; then
-- 
2.11.0

