From 3c1b76e519522fb65aaed5d6e7d46914c40dd6d5 Mon Sep 17 00:00:00 2001
From: daper <david@daper.email>
Date: Wed, 21 Feb 2018 13:43:33 +0100
Subject: [PATCH] multipath: Use stateless configuration for non-host-only
 initrds

Signed-off-by: daper <david@daper.email>
---
 modules.d/90multipath/module-setup.sh | 1 +
 1 file changed, 1 insertion(+)

diff --git a/modules.d/90multipath/module-setup.sh b/modules.d/90multipath/module-setup.sh
index 370cab92..819bd698 100755
--- a/modules.d/90multipath/module-setup.sh
+++ b/modules.d/90multipath/module-setup.sh
@@ -103,6 +103,7 @@ install() {
         xdrgetprio \
         xdrgetuid \
         /etc/xdrdevices.conf \
+        /usr/share/defaults/multipath/multipath.conf \
         /etc/multipath.conf \
         /etc/multipath/* \
         "$config_dir"/* \
-- 
2.39.0

