From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Reilly Brogan <reilly@reillybrogan.com>
Date: Mon, 23 Sep 2024 15:52:46 -0500
Subject: [PATCH] Fix linking with system curl

---
 lib/CMakeLists.txt | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/lib/CMakeLists.txt b/lib/CMakeLists.txt
index e1ad148..b579024 100644
--- a/lib/CMakeLists.txt
+++ b/lib/CMakeLists.txt
@@ -162,7 +162,7 @@ else()
 endif()
 
 if (CHIAKI_USE_SYSTEM_CURL)
-	target_link_libraries(chiaki-lib CURL::libcurl_shared)
+	target_link_libraries(chiaki-lib ${CURL_LIBRARIES})
 else()
 	target_link_libraries(chiaki-lib CURL::libcurl_static)
 endif()
