Index: manaplus-2.1.3.17/src/net/ea/chatrecv.h
===================================================================
--- manaplus-2.1.3.17.orig/src/net/ea/chatrecv.h
+++ manaplus-2.1.3.17/src/net/ea/chatrecv.h
@@ -34,6 +34,7 @@
 
 #include <string>
 #include <queue>
+#include <cstdint>
 #ifndef __time_t_defined
 #include <sys/time.h>
 #endif  // __time_t_defined
Index: manaplus-2.1.3.17/src/net/ea/traderecv.h
===================================================================
--- manaplus-2.1.3.17.orig/src/net/ea/traderecv.h
+++ manaplus-2.1.3.17/src/net/ea/traderecv.h
@@ -33,6 +33,7 @@
 #endif  // defined(__GXX_EXPERIMENTAL_CXX0X__)
 
 #include <string>
+#include <cstdint>
 
 namespace Net
 {

