From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Reilly Brogan <reilly@reillybrogan.com>
Date: Tue, 30 Jul 2024 19:12:55 -0500
Subject: [PATCH] solus: Restore library version

---
 meson.build | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/meson.build b/meson.build
index a65565e..c396875 100644
--- a/meson.build
+++ b/meson.build
@@ -168,6 +168,7 @@ library('pcsclite_real',
   include_directories : incdir,
   dependencies : threads_dep,
   c_args: '-DLIBPCSCLITE -DSIMCLIST_NO_DUMPRESTORE',
+  version : meson.project_version(),
   soversion : 1,
   install : true)
 
@@ -181,6 +182,7 @@ libpcsclite = library('pcsclite',
   libpcsclite_src,
   dependencies : dl_deps,
   include_directories : incdir,
+  version : meson.project_version(),
   soversion : 1,
   install : true)
 
@@ -194,6 +196,7 @@ library('pcscspy',
   sources : ['src/spy/libpcscspy.c', 'src/sys_unix.c'],
   dependencies : dl_deps,
   include_directories : incdir,
+  version : meson.project_version(),
   soversion : 0,
   install : true)
 
