From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Ikey Doherty <ikey@serpentos.com>
Date: Fri, 22 Nov 2024 21:57:37 +0000
Subject: [PATCH] Set plymouth defaults

Signed-off-by: Ikey Doherty <ikey@serpentos.com>
---
 src/plymouthd.defaults | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/src/plymouthd.defaults b/src/plymouthd.defaults
index 7f247fd0..6ab498d3 100644
--- a/src/plymouthd.defaults
+++ b/src/plymouthd.defaults
@@ -1,6 +1,7 @@
 # Distribution defaults. Changes to this file will get overwritten during
 # upgrades.
 [Daemon]
-Theme=spinner
+UseSimpledrm=1
+Theme=bgrt
 ShowDelay=0
 DeviceTimeout=8
