From e252680cee4849e0bcbc41aa113f20672b758a63 Mon Sep 17 00:00:00 2001
From: Thomas Staudinger <Staudi.Kaos@gmail.com>
Date: Tue, 10 Jan 2023 23:15:15 +0100
Subject: [PATCH] Use stateless configuration for im-multipress

---
 meson.build                            |  5 +++--
 modules/input/gtkimcontextmultipress.c | 17 ++++++++++++-----
 2 files changed, 15 insertions(+), 7 deletions(-)

diff --git a/meson.build b/meson.build
index aed48fc3f6..0e1e6b0ad9 100644
--- a/meson.build
+++ b/meson.build
@@ -751,7 +751,8 @@ immodules = [
   [ 'ipa', files([ immodule_srcdir + 'imipa.c' ]) ],
   [ 'multipress', files( [ immodule_srcdir + 'gtkimcontextmultipress.c', immodule_srcdir + 'immultipress.c' ]), true,
                     ['-DMULTIPRESS_LOCALEDIR="@0@"'.format(gtk_mplocaledir),
-                     '-DMULTIPRESS_CONFDIR="@0@/gtk-3.0"'.format(gtk_sysconfdir)] ],
+                     '-DMULTIPRESS_CONFDIR="@0@/gtk-2.0"'.format(gtk_sysconfdir),
+                     '-DSYSTEM_MULTIPRESS_CONFDIR="@0@/gtk-3.0"'.format(gtk_datadir)] ],
   [ 'thai', files([ immodule_srcdir + 'thai-charprop.c', immodule_srcdir + 'gtkimcontextthai.c', immodule_srcdir + 'imthai.c' ]) ],
   [ 'ti-er', files([ immodule_srcdir + 'imti-er.c' ]) ],
   [ 'ti-et', files([ immodule_srcdir + 'imti-et.c' ]) ],
@@ -764,7 +765,7 @@ immodules = [
   [ 'xim', files([ immodule_srcdir + 'gtkimcontextxim.c', immodule_srcdir + 'imxim.c' ]), x11_enabled ],
 ]
 
-install_data(join_paths(immodule_srcdir, 'im-multipress.conf'), install_dir: join_paths(gtk_sysconfdir, 'gtk-3.0'))
+install_data(join_paths(immodule_srcdir, 'im-multipress.conf'), install_dir: join_paths(gtk_datadir, 'gtk-3.0'))
 
 # Check for bind_textdomain_codeset, including -lintl if GLib brings it in by
 # doing the same check as glib. We can't check that by linking to glib because
diff --git a/modules/input/gtkimcontextmultipress.c b/modules/input/gtkimcontextmultipress.c
index b92570bdbd..2b0f2a87ff 100644
--- a/modules/input/gtkimcontextmultipress.c
+++ b/modules/input/gtkimcontextmultipress.c
@@ -25,6 +25,7 @@
 
 #define AUTOMATIC_COMPOSE_TIMEOUT 1 /* seconds */
 #define CONFIGURATION_FILENAME MULTIPRESS_CONFDIR G_DIR_SEPARATOR_S "im-multipress.conf"
+#define STATELESS_CONFIGURATION_FILENAME SYSTEM_MULTIPRESS_CONFDIR G_DIR_SEPARATOR_S "im-multipress.conf"
 
 /* This contains rows of characters that can be entered by pressing
  * a particular key repeatedly.  Each row has one key (such as GDK_a),
@@ -384,12 +385,18 @@ load_config (GtkImContextMultipress *self)
   gsize     i;
 
   key_file = g_key_file_new ();
+  const char *conf_filename = NULL;
 
-  if (!g_key_file_load_from_file (key_file, CONFIGURATION_FILENAME,
+  if (g_file_test(CONFIGURATION_FILENAME, G_FILE_TEST_EXISTS))
+    conf_filename = CONFIGURATION_FILENAME;
+  else
+    conf_filename = STATELESS_CONFIGURATION_FILENAME;
+
+  if (!g_key_file_load_from_file (key_file, conf_filename,
                                   G_KEY_FILE_NONE, &error))
     {
       g_warning ("Error while trying to open the %s configuration file: %s",
-                 CONFIGURATION_FILENAME, error->message);
+                 conf_filename, error->message);
       g_error_free (error);
       g_key_file_free (key_file);
       return;
@@ -400,7 +407,7 @@ load_config (GtkImContextMultipress *self)
   if (error != NULL)
     {
       g_warning ("Error while trying to read the %s configuration file: %s",
-                 CONFIGURATION_FILENAME, error->message);
+                 conf_filename, error->message);
       g_error_free (error);
       g_key_file_free (key_file);
       return;
@@ -417,7 +424,7 @@ load_config (GtkImContextMultipress *self)
         {
           g_warning ("Error while trying to read the %s configuration file: "
                      "invalid key name \"%s\"",
-                     CONFIGURATION_FILENAME, keys[i]);
+                     conf_filename, keys[i]);
           continue;
         }
 
@@ -427,7 +434,7 @@ load_config (GtkImContextMultipress *self)
       if (error != NULL)
         {
           g_warning ("Error while trying to read the %s configuration file: %s",
-                     CONFIGURATION_FILENAME, error->message);
+                     conf_filename, error->message);
           g_error_free (error);
           error = NULL;
           g_slice_free (KeySequence, seq);
-- 
2.39.0

