From d2e9015b920827fef8b8a87b471af61be2a73829 Mon Sep 17 00:00:00 2001
From: Campbell Jones <git@serebit.com>
Date: Wed, 30 Sep 2020 11:44:41 -0400
Subject: [PATCH] fix tooltip issue with various de panels

---
 gdk/gdkwindowimpl.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/gdk/gdkwindowimpl.c b/gdk/gdkwindowimpl.c
index c42c5aa218..eb02307173 100644
--- a/gdk/gdkwindowimpl.c
+++ b/gdk/gdkwindowimpl.c
@@ -166,7 +166,7 @@ maybe_flip_position (gint      bounds_pos,
   *flipped = TRUE;
   secondary = rect_pos + (1 - rect_sign) * rect_size / 2 - offset - (1 - window_sign) * window_size / 2;
 
-  if (secondary >= bounds_pos && secondary + window_size <= bounds_pos + bounds_size)
+  if (secondary >= bounds_pos && secondary + window_size <= bounds_pos + bounds_size || primary > bounds_pos + bounds_size)
     return secondary;
 
   *flipped = FALSE;
-- 
GitLab

