@@ -, +, @@ 
 the widget's
 gtk/gtkgesture.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)
--- a/gtk/gtkgesture.c	
+++ a/gtk/gtkgesture.c	
@@ -554,7 +554,7 @@ _gtk_gesture_update_point (GtkGesture     *gesture,
   widget_window = _find_widget_window (gesture, event->any.window);
 
   if (!widget_window)
-    return FALSE;
+    widget_window = event->any.window;
 
   touchpad = EVENT_IS_TOUCHPAD_GESTURE (event);
 
