From 2b6059dee3a95591d787e8b8c931cd68c059d43f Mon Sep 17 00:00:00 2001
From: Yen Chi Hsuan <yan12125@gmail.com>
Date: Fri, 2 Jun 2017 01:28:39 +0800
Subject: [PATCH 1/1] :bug: Fix building with Qt 5.9 (closes #528)

---
 tests/modeltest.cpp | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/tests/modeltest.cpp b/tests/modeltest.cpp
index 360a7be..6bf8124 100644
--- a/tests/modeltest.cpp
+++ b/tests/modeltest.cpp
@@ -448,7 +448,8 @@ void ModelTest::data()
     QVariant textAlignmentVariant = model->data ( model->index ( 0, 0 ), Qt::TextAlignmentRole );
     if ( textAlignmentVariant.isValid() ) {
         int alignment = textAlignmentVariant.toInt();
-        QCOMPARE( alignment, ( alignment & ( Qt::AlignHorizontal_Mask | Qt::AlignVertical_Mask ) ) );
+        QCOMPARE( alignment, static_cast<int>( alignment & ( Qt::AlignHorizontal_Mask
+                                                           | Qt::AlignVertical_Mask ) ) );
     }
 
     // General Purpose roles that should return a QColor
-- 
2.13.1

