From 46a2f1b004409a805d86e86e21cd2318bf538971 Mon Sep 17 00:00:00 2001
From: Algent Albrahimi <algent@protonmail.com>
Date: Sun, 8 Aug 2021 18:36:41 +0200
Subject: [PATCH] Fix Makefile

---
 Makefile | 15 ++++++++-------
 1 file changed, 8 insertions(+), 7 deletions(-)

diff --git a/Makefile b/Makefile
index a639ca9..44b802b 100644
--- a/Makefile
+++ b/Makefile
@@ -12,8 +12,6 @@ cursors-bloom-fantacy:
 	cd cursors-src/bloom-fantacy/loginspinner/ && make
 
 check: check-same-icon check-perm
-	gtk-update-icon-cache bloom || exit 101
-	gtk-update-icon-cache bloom-dark || exit 101
 	-rm -f bloom/icon-theme.cache
 	-rm -f bloom-dark/icon-theme.cache
 
@@ -24,7 +22,7 @@ check-same-icon:
 	find bloom -type f | xargs md5sum | sort | uniq --check-chars=32 -d
 	find bloom-dark -type f | xargs md5sum | sort | uniq --check-chars=32 -d
 
-check-perm: hicolor-links
+check-perm:
 	@echo "Fix icon files permission"
 	find bloom -type f -exec chmod 644 {} \;
 	find bloom-dark -type f -exec chmod 644 {} \;
@@ -47,13 +45,16 @@ install-icons:
 	cp -r bloom/* $(DESTDIR)$(PREFIX)/share/icons/bloom
 	mkdir -p $(DESTDIR)$(PREFIX)/share/icons/bloom-dark
 	cp -r bloom-dark/* $(DESTDIR)$(PREFIX)/share/icons/bloom-dark
-	mkdir -p $(DESTDIR)$(PREFIX)/share/icons/Vintage
-	cp -r Vintage/* $(DESTDIR)$(PREFIX)/share/icons/Vintage
+	mkdir -p $(DESTDIR)$(PREFIX)/share/icons/deepin
+	cp -r bloom-classic/* $(DESTDIR)$(PREFIX)/share/icons/deepin
+	mkdir -p $(DESTDIR)$(PREFIX)/share/icons/deepin-dark
+	cp -r bloom-classic-dark/* $(DESTDIR)$(PREFIX)/share/icons/deepin-dark
+	mkdir -p $(DESTDIR)$(PREFIX)/share/icons/Sea
+	cp -r Sea/* $(DESTDIR)$(PREFIX)/share/icons/Sea
 
 install-cursors:
 	mkdir -p $(DESTDIR)$(PREFIX)/share/icons/bloom
 	cp -r bloom/cursors $(DESTDIR)$(PREFIX)/share/icons/bloom
 	install -m644 bloom/cursor.theme $(DESTDIR)$(PREFIX)/share/icons/bloom/cursor.theme
 
-hicolor-links:
-	./tools/hicolor.links bloom hicolor.list ./
+install: install-icons install-cursors
-- 
2.32.0

