From 85c6353be65352bf93103588cf2f831e730a3df7 Mon Sep 17 00:00:00 2001
From: Joshua Strobl <joshua@stroblindustries.com>
Date: Wed, 22 Feb 2017 01:01:00 +0200
Subject: [PATCH 1/1] Override liblastfm paths.

---
 CMakeModules/FindLibLastFm.cmake | 12 ++++--------
 1 file changed, 4 insertions(+), 8 deletions(-)

diff --git a/CMakeModules/FindLibLastFm.cmake b/CMakeModules/FindLibLastFm.cmake
index a3488350e..738eebe10 100644
--- a/CMakeModules/FindLibLastFm.cmake
+++ b/CMakeModules/FindLibLastFm.cmake
@@ -7,21 +7,17 @@
 # (c) Dominik Schmidt <dev@dominik-schmidt.de>
 #
 
-if( TOMAHAWK_QT5 )
-    set(LASTFM_LIB_SUFFIX "5")
-endif()
-
 # Include dir
 find_path(LIBLASTFM_INCLUDE_DIR
   # Track.h doesn't exist in liblastfm-0.3.1, was called Track back then
-  NAMES lastfm${LASTFM_LIB_SUFFIX}/Track.h
-  PATHS ${KDE4_INCLUDE_DIR}
+  NAMES lastfm/Track.h
+  PATHS /usr/include/
 )
 
 # Finally the library itself
 find_library(LIBLASTFM_LIBRARY
-  NAMES lastfm${LASTFM_LIB_SUFFIX}
-  PATHS ${KDE4_LIB_DIR}
+  NAMES lastfm5
+  PATHS /usr/lib64/
 )
 
 set(LIBLASTFM_LIBRARIES ${LIBLASTFM_LIBRARY})
-- 
2.11.1

