From e3c96f40564e1572f2f36dfab224fd1b285f4476 Mon Sep 17 00:00:00 2001
From: Ikey Doherty <ikey@solus-project.com>
Date: Tue, 15 Mar 2016 22:07:06 +0000
Subject: [PATCH] pkexec: Support a stateless configuration

Signed-off-by: Ikey Doherty <ikey@solus-project.com>
---
 src/programs/pkexec.c | 12 ++++++++++--
 1 file changed, 10 insertions(+), 2 deletions(-)

diff --git a/src/programs/pkexec.c b/src/programs/pkexec.c
index f42a3bb..67f3824 100644
--- a/src/programs/pkexec.c
+++ b/src/programs/pkexec.c
@@ -308,19 +308,27 @@ is_valid_shell (const gchar *shell)
   gchar **shells;
   GError *error;
   guint n;
+  const gchar *shellFile = NULL;
 
   ret = FALSE;
 
   contents = NULL;
   shells = NULL;
 
+  if (g_file_test ("/etc/shells", G_FILE_TEST_EXISTS))
+    {
+      shellFile = "/etc/shells";
+    } else {
+      shellFile = "//usr/share/defaults/etc/shells";
+    }
+
   error = NULL;
-  if (!g_file_get_contents ("/etc/shells",
+  if (!g_file_get_contents (shellFile,
                             &contents,
                             NULL, /* gsize *length */
                             &error))
     {
-      g_printerr ("Error getting contents of /etc/shells: %s\n", error->message);
+      g_printerr ("Error getting contents of %s: %s\n", shellFile, error->message);
       g_error_free (error);
       goto out;
     }
-- 
2.7.1

