From 69fb14be4b9234ae11fb0162090c6be618ed5168 Mon Sep 17 00:00:00 2001
From: Ikey Doherty <ikey@solus-project.com>
Date: Thu, 19 Oct 2017 17:41:50 +0100
Subject: [PATCH 3/3] data: Use modern stateless dbus `system.d` directory

Signed-off-by: Ikey Doherty <ikey@solus-project.com>
---
 data/Makefile.am | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/data/Makefile.am b/data/Makefile.am
index fe0f1d5..f2b27ed 100644
--- a/data/Makefile.am
+++ b/data/Makefile.am
@@ -15,7 +15,7 @@ $(service_DATA): $(service_in_files) Makefile
 
 # ----------------------------------------------------------------------------------------------------
 
-dbusconfdir = $(sysconfdir)/dbus-1/system.d
+dbusconfdir = $(datadir)/dbus-1/system.d
 dbusconf_in_files = org.freedesktop.PolicyKit1.conf.in
 dbusconf_DATA = $(dbusconf_in_files:.conf.in=.conf)
 
-- 
2.14.2

