diff --git a/configure.ac b/configure.ac
index 5ca36d7..8a66dc3 100644
--- a/configure.ac
+++ b/configure.ac
@@ -163,26 +163,6 @@ if test "$enable_libsystemd_login" != "no"; then
   fi
 fi
 
-AS_IF([test "x$cross_compiling" != "xyes" ], [
-  AS_IF([test "$have_libsystemd" = "yes"], [
-    AS_IF([test ! -d /sys/fs/cgroup/systemd/ ], [
-      AS_IF([test "$enable_libsystemd_login" = "yes"], [
-        AC_MSG_WARN([libsystemd requested but system does not appear to be using systemd])
-      ], [
-        AC_MSG_ERROR([libsystemd autoconfigured, but system does not appear to use systemd])
-      ])
-    ])
-  ], [
-    AS_IF([test -d /sys/fs/cgroup/systemd/ ], [
-      AS_IF([test "$enable_libsystemd_login" = "no" ], [
-        AC_MSG_WARN([ConsoleKit requested but system appears to use systemd])
-      ], [
-        AC_MSG_ERROR([ConsoleKit autoconfigured, but systemd is in use (missing libsystemd or libsystemd-login pkg-config?)])
-      ])
-    ])
-  ])
-])
-
 AC_SUBST(LIBSYSTEMD_CFLAGS)
 AC_SUBST(LIBSYSTEMD_LIBS)
 AM_CONDITIONAL(HAVE_LIBSYSTEMD, [test "$have_libsystemd" = "yes"], [Using libsystemd])
