From 3e48b3a5743dbd0012f36553cc3b9957ebd9c88d Mon Sep 17 00:00:00 2001
From: Peter O'Connor <sunnyflunk@gmail.com>
Date: Sun, 30 Oct 2016 17:03:15 +1100
Subject: [PATCH] Support stateless config directories

---
 src/corelib/io/qstandardpaths_unix.cpp | 1 +
 1 file changed, 1 insertion(+)

Index: qtbase-opensource-src-5.6.2/src/corelib/io/qstandardpaths_unix.cpp
===================================================================
--- qtbase-opensource-src-5.6.2.orig/src/corelib/io/qstandardpaths_unix.cpp
+++ qtbase-opensource-src-5.6.2/src/corelib/io/qstandardpaths_unix.cpp
@@ -304,6 +304,7 @@ static QStringList xdgConfigDirs()
         dirs.append(QString::fromLatin1("/etc/xdg"));
     else
         dirs = xdgConfigDirs.split(QLatin1Char(':'));
+    dirs.append(QString::fromLatin1("/usr/share/xdg"));
     return dirs;
 }

