From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Muhammad Alfi Syahrin <malfisya.dev@hotmail.com>
Date: Tue, 22 Apr 2025 20:27:12 +0700
Subject: [PATCH] Remove hardcoded install paths

---
 CMakeLists.txt | 2 --
 1 file changed, 2 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 964921d..5d79ddc 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -133,8 +133,6 @@ if(UNIX AND NOT APPLE)
     # If CMAKE_INSTALL_PREFIX is not set, it will set rpath to /opt/....
     # Note: using cmake --install --prefix /some/prefix will not set rpath relative to that prefix
     # The above can be used for building packages: in which install dir is a path construction folder that will not be there in packages
-    set(CMAKE_INSTALL_LIBDIR "opt/megacmd/lib")
-    set(CMAKE_INSTALL_BINDIR "usr/bin") #override default "bin"
 
     if(CMAKE_INSTALL_PREFIX_INITIALIZED_TO_DEFAULT)
         message(STATUS "Overriding default CMAKE_INSTALL_PREFIX to /")
