From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Muhammad Alfi Syahrin <malfisya.dev@hotmail.com>
Date: Tue, 22 Apr 2025 20:27:42 +0700
Subject: [PATCH] Fix PCRE and ZLib linking

---
 CMakeLists.txt | 1 +
 1 file changed, 1 insertion(+)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 5d79ddc..152c520 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -381,6 +381,7 @@ target_link_libraries(mega-exec LMegacmdClient)
 target_link_libraries(mega-cmd PUBLIC MEGA::SDKlib LMEGAcmdCommonUtils)
 target_link_libraries(mega-cmd-updater PUBLIC MEGA::SDKlib LMEGAcmdCommonUtils)
 target_link_libraries(LMegacmdServer PUBLIC MEGA::SDKlib LMEGAcmdCommonUtils)
+target_link_libraries(LMegacmdServer PUBLIC pcrecpp z)
 
 if (ENABLE_MEGACMD_TESTS)
   target_link_libraries(LMegacmdServer PUBLIC LMegacmdTestsCommon)
