diff --git a/pytest.ini b/pytest.ini
index 0929e44..4573df6 100644
--- a/pytest.ini
+++ b/pytest.ini
@@ -1,4 +1,4 @@
 [pytest]
-addopts = -v --flake8 --cov-report xml --cov-report=term-missing --cov=josepy --cov-config .coveragerc
+addopts = -v
 norecursedirs = *.egg .eggs dist build docs .tox
 flake8-ignore = E501
diff --git a/setup.py b/setup.py
index d15dde4..f6a6fd5 100644
--- a/setup.py
+++ b/setup.py
@@ -20,11 +20,6 @@ install_requires = [
 ]
 
 testing_requires = [
-    'coverage>=4.0',
-    'pytest-cache>=1.0',
-    'pytest-cov',
-    'flake8',
-    'pytest-flake8>=0.5',
     'pytest>=2.8.0',
     'mock',
 ]
