From 2e1fa1db78588f0a45cb5154eaf0965114bb4f56 Mon Sep 17 00:00:00 2001
From: Peter O'Connor <peter@solus-project.com>
Date: Sat, 21 Apr 2018 09:31:55 +1000
Subject: [PATCH] Disable building unshipped code

Signed-off-by: Peter O'Connor <peter@solus-project.com>
---
 CMakeLists.txt | 3 ---
 1 file changed, 3 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 106acbf..39b6859 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -76,9 +76,6 @@ set(libknumber_la_SRCS
 	${kcalc_SOURCE_DIR}/knumber/knumber_operators.cpp
 )
 
-add_subdirectory( knumber )
-# add_subdirectory( tests )
-
 set(kcalc_KDEINIT_SRCS ${libknumber_la_SRCS}
    kcalc.cpp 
    bitbutton.cpp
-- 
2.16.3

