From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Reilly Brogan <reilly@reillybrogan.com>
Date: Fri, 15 Mar 2024 15:12:01 -0500
Subject: [PATCH] solus: Use local toolchain

Signed-off-by: Reilly Brogan <reilly@reillybrogan.com>
---
 go.env | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/go.env b/go.env
index 6ff2b921d4..bdfb992653 100644
--- a/go.env
+++ b/go.env
@@ -9,4 +9,5 @@ GOSUMDB=sum.golang.org
 
 # Automatically download newer toolchains as directed by go.mod files.
 # See https://go.dev/doc/toolchain for details.
-GOTOOLCHAIN=auto
+# Solus: Use locally installed golang toolchain by default
+GOTOOLCHAIN=local
