From 29426e6ba7c162d488dc56d597b18af49976ecca Mon Sep 17 00:00:00 2001
From: Joshua Strobl <joshua@streambits.io>
Date: Tue, 16 Feb 2021 23:27:07 +0200
Subject: [PATCH] os-test: Solus uses /usr/bin/hostname, conform and don't fail

---
 src/os/os_test.go | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/os/os_test.go b/src/os/os_test.go
index 63427deb6e..e3e61d956f 100644
--- a/src/os/os_test.go
+++ b/src/os/os_test.go
@@ -1676,7 +1676,7 @@ func runBinHostname(t *testing.T) string {
 		t.Fatal(err)
 	}
 	defer r.Close()
-	const path = "/bin/hostname"
+	const path = "/usr/bin/hostname"
 	argv := []string{"hostname"}
 	if runtime.GOOS == "aix" {
 		argv = []string{"hostname", "-s"}
@@ -1705,7 +1705,7 @@ func runBinHostname(t *testing.T) string {
 		output = output[0 : n-1]
 	}
 	if output == "" {
-		t.Fatalf("/bin/hostname produced no output")
+		t.Fatalf("/usr/bin/hostname produced no output")
 	}
 
 	return output
-- 
2.35.1

