From 5a382f1d84189945484bcf2eced7211113c3eef9 Mon Sep 17 00:00:00 2001
From: Ikey Doherty <ikey@solus-project.com>
Date: Thu, 8 Sep 2016 21:38:19 +0100
Subject: [PATCH 2/2] Stop forcing Clang. The sanitizer there is bricked

Signed-off-by: Ikey Doherty <ikey@solus-project.com>
---
 misc/cgo/testsanitizers/test.bash | 5 +----
 1 file changed, 1 insertion(+), 4 deletions(-)

diff --git a/misc/cgo/testsanitizers/test.bash b/misc/cgo/testsanitizers/test.bash
index 78747d1..f2b2622 100755
--- a/misc/cgo/testsanitizers/test.bash
+++ b/misc/cgo/testsanitizers/test.bash
@@ -8,11 +8,8 @@
 
 set -e
 
-# The sanitizers were originally developed with clang, so prefer it.
+# NO. Clang. NO.
 CC=cc
-if test -x "$(type -p clang)"; then
-  CC=clang
-fi
 export CC
 
 msan=yes
-- 
2.10.0

