From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Reilly Brogan <reilly@reillybrogan.com>
Date: Fri, 25 Nov 2022 15:06:28 -0600
Subject: [PATCH] Better follow FreeDesktop specification

---
 CMakeLists.txt | 11 ++++-------
 src/main.cpp   |  4 ----
 2 files changed, 4 insertions(+), 11 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index e374177a..c513bace 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -972,14 +972,11 @@ if(APPLE)
 endif()
 
 if(UNIX AND NOT APPLE)
-    if(FLATPAK)
-        set(APPID "im.nheko.Nheko")
-        set_target_properties(nheko PROPERTIES OUTPUT_NAME "${APPID}")
-    else()
-        set(APPID "nheko")
-    endif()
+    set(APPID "im.nheko.Nheko")
+
+    set_target_properties(nheko PROPERTIES OUTPUT_NAME "${APPID}")
+    install(TARGETS nheko RUNTIME)
 
-    install(TARGETS nheko RUNTIME DESTINATION "${CMAKE_INSTALL_BINDIR}")
     install(FILES "resources/nheko-16.png" DESTINATION "${CMAKE_INSTALL_DATAROOTDIR}/icons/hicolor/16x16/apps" RENAME "${APPID}.png")
     install(FILES "resources/nheko-32.png" DESTINATION "${CMAKE_INSTALL_DATAROOTDIR}/icons/hicolor/32x32/apps" RENAME "${APPID}.png")
     install(FILES "resources/nheko-48.png" DESTINATION "${CMAKE_INSTALL_DATAROOTDIR}/icons/hicolor/48x48/apps" RENAME "${APPID}.png")
diff --git a/src/main.cpp b/src/main.cpp
index 8ea15901..ea21e9c7 100644
--- a/src/main.cpp
+++ b/src/main.cpp
@@ -304,11 +304,7 @@ main(int argc, char *argv[])
 #if !defined(Q_OS_MACOS)
     app.setWindowIcon(QIcon::fromTheme(QStringLiteral("nheko"), QIcon{":/logos/nheko.png"}));
 #endif
-#ifdef NHEKO_FLATPAK
     app.setDesktopFileName(QStringLiteral("im.nheko.Nheko"));
-#else
-    app.setDesktopFileName(QStringLiteral("nheko"));
-#endif
 
     http::init();
 
