From 3812a38a996065c8827abb5b763d484a240acfd2 Mon Sep 17 00:00:00 2001
From: Reilly Brogan <reilly@reillybrogan.com>
Date: Fri, 25 Nov 2022 15:02:23 -0600
Subject: [PATCH 1/1] Default to system theme

---
 src/UserSettingsPage.h | 8 ++------
 1 file changed, 2 insertions(+), 6 deletions(-)

diff --git a/src/UserSettingsPage.h b/src/UserSettingsPage.h
index 34a792eb..87c83df8 100644
--- a/src/UserSettingsPage.h
+++ b/src/UserSettingsPage.h
@@ -323,12 +323,8 @@ signals:
     void exposeDBusApiChanged(bool state);
 
 private:
-    // Default to system theme if QT_QPA_PLATFORMTHEME var is set.
-    QString defaultTheme_ = QProcessEnvironment::systemEnvironment()
-                                .value(QStringLiteral("QT_QPA_PLATFORMTHEME"), QLatin1String(""))
-                                .isEmpty()
-                              ? "light"
-                              : "system";
+    // Default to system theme
+    QString defaultTheme_ = "system";
     QString theme_;
     bool messageHoverHighlight_;
     bool enlargeEmojiOnlyMessages_;
-- 
2.38.1

