From 0397083192a99dbfa3e19e9b8d33806e6c876089 Mon Sep 17 00:00:00 2001
From: Pierre-Yves <pyu@riseup.net>
Date: Mon, 17 Dec 2018 18:47:46 +0100
Subject: [PATCH] Makefile.in use our CFLAGS

Signed-off-by: Pierre-Yves <pyu@riseup.net>
---
 src/Makefile.in | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/Makefile.in b/src/Makefile.in
index 41c7589..3b4ecd3 100644
--- a/src/Makefile.in
+++ b/src/Makefile.in
@@ -6,7 +6,7 @@ package = @PACKAGE_NAME@
 version = @PACKAGE_VERSION@
 
 CC = @CC@
-CFLAGS = -Wall -fPIC -fno-stack-protector
+CFLAGS = @CFLAGS@
 LIBFLAGS = -c
 DBGFLAGS = -g
 
-- 
2.20.0

