From acf31e8218105a02b784625fb1d7ff7b05d87ad8 Mon Sep 17 00:00:00 2001
From: Troy Harvey <harveydevel@gmail.com>
Date: Sat, 21 May 2022 00:10:56 +1000
Subject: [PATCH] Unset no opaque pointers

---
 build/config/compiler/BUILD.gn | 1 -
 1 file changed, 1 deletion(-)

diff --git a/build/config/compiler/BUILD.gn b/build/config/compiler/BUILD.gn
index d40843b..5a1c05a 100644
--- a/build/config/compiler/BUILD.gn
+++ b/build/config/compiler/BUILD.gn
@@ -1543,7 +1543,6 @@ config("default_warnings") {
         # TODO(https://crbug.com/1316298): Re-enable once test failure is figured out
         cflags += [
           "-Xclang",
-          "-no-opaque-pointers",
         ]
       }
 
-- 
2.35.3

