From cd31491307a15a70d370e89f3732363c3d8f011c Mon Sep 17 00:00:00 2001
From: Evan Maddock <maddock.evan@vivaldi.net>
Date: Wed, 10 Feb 2021 18:43:02 -0500
Subject: [PATCH] Remove appstreamcli because we don't have it in Solus

---
 data/meson.build | 30 ------------------------------
 1 file changed, 30 deletions(-)

diff --git a/data/meson.build b/data/meson.build
index 81e9b410..4c2306ea 100644
--- a/data/meson.build
+++ b/data/meson.build
@@ -33,36 +33,6 @@ if desktop_file_validate.found()
   )
 endif
 
-# Merge release information into MetaInfo file
-tilix_metainfo_name = '@0@.appdata.xml'.format(project_id)
-appstreamcli = find_program('appstreamcli')
-metainfo_with_releases = custom_target('metainfo-news-merge',
-    input : ['../NEWS', 'metainfo/@0@.in'.format(tilix_metainfo_name)],
-    output : ['untranslated-@0@.appdata.xml'.format(project_id)],
-    command : [appstreamcli, 'news-to-metainfo', '--limit=6', '@INPUT0@', '@INPUT1@', '@OUTPUT@']
-)
-
-# Install the MetaInfo file
-metainfo_file = i18n.merge_file(
-    tilix_metainfo_name,
-    output: tilix_metainfo_name,
-    input: metainfo_with_releases,
-    po_dir: meson.source_root() / 'po',
-    install: true,
-    install_dir: datadir / 'metainfo'
-)
-
-# Validate MetaInfo file
-if appstreamcli.found()
-  test (
-    'Validate metainfo file',
-    appstreamcli,
-    args: ['validate', '--no-net',
-           join_paths(meson.current_source_dir(), 'metainfo', '@0@.in'.format(tilix_metainfo_name))
-    ]
-  )
-endif
-
 # Build resources
 gresource = gnome.compile_resources(
     'tilix',
-- 
2.30.0

