From f073fdb09c112f057f6a01b8eb1f7aa027dcbce2 Mon Sep 17 00:00:00 2001
From: Evan Maddock <maddock.evan@vivaldi.net>
Date: Tue, 22 Feb 2022 13:15:19 -0500
Subject: [PATCH] Remove appstreamcli because we don't have it in Solus

Signed-off-by: Evan Maddock <maddock.evan@vivaldi.net>
---
 data/meson.build | 32 ++++++++++++++++----------------
 1 file changed, 16 insertions(+), 16 deletions(-)

diff --git a/data/meson.build b/data/meson.build
index 9b2ea4b2..1e584785 100644
--- a/data/meson.build
+++ b/data/meson.build
@@ -32,24 +32,24 @@ endif
 
 # Merge release information into MetaInfo file
 tilix_metainfo_name = '@0@.appdata.xml'.format(project_id)
-appstreamcli = find_program('appstreamcli')
-metainfo_with_releases = custom_target('metainfo-news-merge',
-    input : ['../NEWS', 'metainfo/@0@.in'.format(tilix_metainfo_name)],
-    output : ['untranslated-@0@.appdata.xml'.format(project_id)],
-    command : [appstreamcli, 'news-to-metainfo', '--limit=6', '@INPUT0@', '@INPUT1@', '@OUTPUT@']
-)
+appstreamcli = find_program('appstreamcli', required: false)
+if appstreamcli.found()
+  metainfo_with_releases = custom_target('metainfo-news-merge',
+      input : ['../NEWS', 'metainfo/@0@.in'.format(tilix_metainfo_name)],
+      output : ['untranslated-@0@.appdata.xml'.format(project_id)],
+      command : [appstreamcli, 'news-to-metainfo', '--limit=6', '@INPUT0@', '@INPUT1@', '@OUTPUT@']
+  )
 
-# Install the MetaInfo file
-metainfo_file = i18n.merge_file(
-    output: tilix_metainfo_name,
-    input: metainfo_with_releases,
-    po_dir: meson.project_source_root() / 'po',
-    install: true,
-    install_dir: datadir / 'metainfo'
-)
+  # Install the MetaInfo file
+  metainfo_file = i18n.merge_file(
+      output: tilix_metainfo_name,
+      input: metainfo_with_releases,
+      po_dir: meson.project_source_root() / 'po',
+      install: true,
+      install_dir: datadir / 'metainfo'
+  )
 
-# Validate MetaInfo file
-if appstreamcli.found()
+  # Validate MetaInfo file
   test (
     'Validate metainfo file',
     appstreamcli,
-- 
2.35.1

