From e7ad21d07bdce8fe5b8bdbd38f08527beef656ec Mon Sep 17 00:00:00 2001
From: Campbell Jones <git@serebit.com>
Date: Wed, 20 Jan 2021 14:56:57 -0500
Subject: [PATCH] Add jcenter to repositories

---
 build.gradle | 1 +
 1 file changed, 1 insertion(+)

diff --git a/build.gradle b/build.gradle
index 49146e7dc..ae50100ea 100644
--- a/build.gradle
+++ b/build.gradle
@@ -32,6 +32,7 @@ allprojects {
 
     repositories {
 		mavenCentral()
+		jcenter()
 		maven { url 'https://resources.knopflerfish.org/repo/maven2/release/' }
 		maven { url "${project.gradle.gradleUserHomeDir}/local-artifacts" }
 		maven { url 'https://dl.bintray.com/freeplane/freeplane' }
-- 
2.30.0

