From 2eaf23ac314124ca473b7ffe9e41ef3a910b2823 Mon Sep 17 00:00:00 2001
From: Ivan Trepakov <liontiger23@gmail.com>
Date: Sat, 16 May 2026 18:59:52 +0700
Subject: [PATCH] Gradle 9.x.x fixes

---
 freeplane_debughelper/build.gradle | 10 ++++++----
 1 file changed, 6 insertions(+), 4 deletions(-)

diff --git a/freeplane_debughelper/build.gradle b/freeplane_debughelper/build.gradle
index 45a8885775..b47379e7a4 100644
--- a/freeplane_debughelper/build.gradle
+++ b/freeplane_debughelper/build.gradle
@@ -7,0 +8,2 @@ repositories {
+def freeplaneProject = project(':freeplane')
+
@@ -17,4 +19,4 @@ dependencies {
-	runtimeOnly project(':freeplane').sourceSets.viewer.resources.sourceDirectories
-	runtimeOnly project(':freeplane').sourceSets.editor.resources.sourceDirectories
-	runtimeOnly project(':freeplane').sourceSets.external.resources.sourceDirectories
-	runtimeOnly files("${project(':freeplane').buildDir}/emoji/xml")
+    runtimeOnly freeplaneProject.sourceSets.viewer.resources.sourceDirectories
+    runtimeOnly freeplaneProject.sourceSets.editor.resources.sourceDirectories
+    runtimeOnly freeplaneProject.sourceSets.external.resources.sourceDirectories
+    runtimeOnly files("${freeplaneProject.buildDir}/emoji/xml")
-- 
2.54.0

