From fab12f7205c4724af1fa7012354dc0129ffae3ac Mon Sep 17 00:00:00 2001
From: Joshua Strobl <joshua@stroblindustries.com>
Date: Mon, 20 Mar 2017 20:48:53 +0200
Subject: [PATCH 1/1] Make use of DESTDIR and PREFIX. Enable statelessness.

---
 Makefile | 14 +++++++-------
 config.c | 22 ++++++++++------------
 2 files changed, 17 insertions(+), 19 deletions(-)

diff --git a/Makefile b/Makefile
index 1a20f4f..1054bad 100755
--- a/Makefile
+++ b/Makefile
@@ -1,4 +1,4 @@
-
+PREFIX = /usr
 RAW_CC = gcc
 RAW_FLAGS = -Wall -O2
 LINK_OPT = 
@@ -24,9 +24,9 @@ MAN_PAGES = $(NAME).8.gz
 RAW_FLAGS += -DVERSION=\"$(VERSION)\"
 
 # Where we get installed
-BIN = /usr/local/bin
-MAN = /usr/share/man/man8
-CONF= /usr/local/etc
+BIN = $(DESTDIR)$(PREFIX)/bin
+MAN = $(DESTDIR)$(PREFIX)/share/man/man8
+CONF= $(DESTDIR)/usr/share/defaults/foremost
 # Setup for compiling and cross-compiling for Windows
 # The CR_ prefix refers to cross compiling from OSX to Windows
 CR_CC = $(CR_BASE)/gcc
@@ -118,9 +118,9 @@ foremost: $(OBJ)
 #---------------------------------------------------------------------
 
 install: goals
-	install -m 755 $(NAME) $(BIN)
-	install -m 444 $(MAN_PAGES) $(MAN)
-	install -m 444 foremost.conf $(CONF)
+	install -Dm00755 $(NAME) $(BIN)/$(NAME)
+	install -Dm00644 $(MAN_PAGES) $(MAN)/$(MAN_PAGES)
+	install -Dm00644 foremost.conf $(CONF)/$(NAME).conf
 macinstall: BIN = /usr/local/bin/
 macinstall: MAN = /usr/share/man/man1/
 macinstall: CONF = /usr/local/etc/
diff --git a/config.c b/config.c
index 9a4bc3f..4c47901 100755
--- a/config.c
+++ b/config.c
@@ -284,19 +284,17 @@ int load_config_file(f_state *s)
 	{
 
 		/*Can't find  a conf in the current directory
-    * So lets try the /usr/local/etc*/
-#ifdef __WIN32
-		set_config_file(s, "/Program Files/foremost/foremost.conf");
-#else
-		set_config_file(s, "/usr/local/etc/foremost.conf");
-#endif
-		if ((f = fopen(get_config_file(s), "r")) == NULL)
-			{
-			print_error(s, get_config_file(s), strerror(errno));
-			free(buffer);
-			return TRUE;
-			}
+    * So lets try the /etc*/
+		set_config_file(s, "/etc/foremost.conf");
+
+		if ((f = fopen(get_config_file(s), "r")) == NULL) {
+			set_config_file(s, "/usr/share/defaults/foremost/foremost.conf");
 
+			if ((f = fopen(get_config_file(s), "r")) == NULL) {
+				print_error(s, get_config_file(s), strerror(errno));
+				return TRUE;
+			}
+		}
 	}
 
 	while (fgets(buffer, MAX_STRING_LENGTH, f))
-- 
2.12.0

