From 5e8fb255dc6bc9b7084a05b8dc24f31417542e5e Mon Sep 17 00:00:00 2001
From: hedmo <hedmoo@yahoo.com>
Date: Fri, 8 Jan 2021 07:34:17 +0000
Subject: [PATCH] Update xlsx_revision_context.cpp to build with gcc-11

---
 src/liborcus/xlsx_revision_context.cpp | 1 +
 1 file changed, 1 insertion(+)

diff --git a/src/liborcus/xlsx_revision_context.cpp b/src/liborcus/xlsx_revision_context.cpp
index edfd9eed..0b6ec0af 100644
--- a/src/liborcus/xlsx_revision_context.cpp
+++ b/src/liborcus/xlsx_revision_context.cpp
@@ -16,6 +16,7 @@
 #include "orcus/global.hpp"
 
 #include <iostream>
+#include <limits>
 
 using namespace std;
 
-- 
GitLab

