From 50506d534230d90de2b0a2af3fb072428c942a4e Mon Sep 17 00:00:00 2001
From: hedmo <hedmoo@yahoo.com>
Date: Fri, 8 Jan 2021 07:30:16 +0000
Subject: [PATCH] Update orcus_xlsx.cpp to build with gcc-11

---
 src/liborcus/orcus_xlsx.cpp | 1 +
 1 file changed, 1 insertion(+)

diff --git a/src/liborcus/orcus_xlsx.cpp b/src/liborcus/orcus_xlsx.cpp
index 6df72a32..a59b2b9d 100644
--- a/src/liborcus/orcus_xlsx.cpp
+++ b/src/liborcus/orcus_xlsx.cpp
@@ -32,6 +32,7 @@
 
 #include <cstdlib>
 #include <iostream>
+#include <limits>
 #include <string>
 #include <cstring>
 #include <sstream>
-- 
GitLab

