From 3bfa1a000e4639fe68781c584356ceea7c6b1a72 Mon Sep 17 00:00:00 2001
From: hedmo <hedmoo@yahoo.com>
Date: Fri, 8 Jan 2021 07:32:53 +0000
Subject: [PATCH] Update xls_xml_context.cpp to build with gcc-11

---
 src/liborcus/xls_xml_context.cpp | 1 +
 1 file changed, 1 insertion(+)

diff --git a/src/liborcus/xls_xml_context.cpp b/src/liborcus/xls_xml_context.cpp
index 538f5607..ca3dc4d0 100644
--- a/src/liborcus/xls_xml_context.cpp
+++ b/src/liborcus/xls_xml_context.cpp
@@ -16,6 +16,7 @@
 #include <mdds/sorted_string_map.hpp>
 
 #include <iostream>
+#include <limits>
 
 using namespace std;
 namespace ss = orcus::spreadsheet;
-- 
GitLab

