From d62ca21c500383fbe23fbb88a6840bda5e679f8f Mon Sep 17 00:00:00 2001
From: Jens Lody <jens@codeblocks.org>
Date: Fri, 10 Feb 2017 21:24:29 +0100
Subject: Build fix for gcc 7.


diff --git a/src/include/scripting/sqplus/sqplus.h b/src/include/scripting/sqplus/sqplus.h
index 73f37b2..ae7d3c7 100644
--- a/src/include/scripting/sqplus/sqplus.h
+++ b/src/include/scripting/sqplus/sqplus.h
@@ -154,10 +154,10 @@ struct ScriptStringVar : ScriptStringVarBase {
     return safeStringCopy(s,_s.s,MaxLength);
   }
   bool operator == (const ScriptStringVar & _s) {
-    return _strcmp(s,_s.s) == 0;
+    return strcmp(s,_s.s) == 0;
   }
   bool compareCaseInsensitive(const ScriptStringVar & _s) {
-    return _stricmp(s,_s.s) == 0;
+    return strcasecmp(s,_s.s) == 0;
   }
 };
 
