diff --color -Naur a/src/mime/codeblocks.desktop b/src/mime/codeblocks.desktop
--- a/src/mime/codeblocks.desktop	2020-03-28 08:36:06.000000000 -0400
+++ b/src/mime/codeblocks.desktop	2023-07-25 06:31:23.007846036 -0400
@@ -4,7 +4,7 @@
 Name=Code::Blocks IDE
 GenericName=Integrated development environment
 Comment=Configurable and extensible IDE
-Exec=codeblocks %F
+Exec=env GDK_BACKEND=x11 codeblocks %F
 Icon=codeblocks
 Terminal=false
 X-MultipleArgs=false