diff --git a/libxfce4util/Makefile.am b/libxfce4util/Makefile.am
index d669bdb..7d5f5a9 100644
--- a/libxfce4util/Makefile.am
+++ b/libxfce4util/Makefile.am
@@ -8,7 +8,7 @@ AM_CPPFLAGS =								\
 	-DDEFAULT_XDG_DATA_HOME=\"~/.local/share\"			\
 	-DDEFAULT_XDG_CONFIG_HOME=\"~/.config\"				\
 	-DDEFAULT_XDG_DATA_DIRS=\"$(datadir):/usr/local/share:/usr/share\" \
-	-DDEFAULT_XDG_CONFIG_DIRS=\"$(sysconfdir)/xdg:$(sysconfdir)/xfce4:/etc/xdg\" \
+	-DDEFAULT_XDG_CONFIG_DIRS=\"$(sysconfdir)/xdg:$(sysconfdir)/defaults/xfce4\" \
 	-DDATADIR=\"$(datadir)\"					\
 	-DSYSCONFDIR=\"$(sysconfdir)\"					\
 	-DPACKAGE_LOCALE_DIR=\"$(localedir)\"				\
