From 37b4460d0334e04c3a73264512a8f379d353f528 Mon Sep 17 00:00:00 2001
From: Christophe Giboudeaux <christophe@krop.fr>
Date: Sat, 7 Dec 2019 23:26:24 +0100
Subject: [PATCH] Fix build with recent pango versions.

---
 CMakeLists.txt                | 2 ++
 cmake_modules/FindPANGO.cmake | 2 ++
 2 files changed, 4 insertions(+)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 36b5f93..d844fcc 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -77,6 +77,8 @@ INCLUDE_DIRECTORIES(
   ${NOTIFY_INCLUDE_DIRS}
 )
 
+ADD_DEFINITIONS(${PANGO_CFLAGS})
+
 ADD_SUBDIRECTORY(configuration)
 ADD_SUBDIRECTORY(gaminggearfxcontrol)
 ADD_SUBDIRECTORY(gaminggearfxinfo)
diff --git a/cmake_modules/FindPANGO.cmake b/cmake_modules/FindPANGO.cmake
index 2b8e880..911c459 100644
--- a/cmake_modules/FindPANGO.cmake
+++ b/cmake_modules/FindPANGO.cmake
@@ -16,6 +16,8 @@ FIND_PATH(PANGO_INCLUDE_DIRS pango/pango.h
   PATH_SUFFIXES pango-1.0
 )
 
+SET(PANGO_CFLAGS "${PKG_PANGO_CFLAGS}")
+
 FIND_LIBRARY(PANGO_LIBRARIES pango-1.0
   HINTS ${PKG_PANGO_LIBRARY_DIRS}
 )
-- 
2.24.0
