From 7e25f162ab4885c65baf1b68a3bdabddfd3fb6cb Mon Sep 17 00:00:00 2001
From: Ikey Doherty <ikey@solus-project.com>
Date: Fri, 3 Nov 2017 20:10:55 +0000
Subject: [PATCH 1/2] Fix broken installation of systemd unit

Signed-off-by: Ikey Doherty <ikey@solus-project.com>
---
 data/meson.build | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/data/meson.build b/data/meson.build
index 83487b1..1f4b81f 100644
--- a/data/meson.build
+++ b/data/meson.build
@@ -5,7 +5,7 @@ data_conf.set('SBINDIR', path_sbindir)
 
 configure_file(
     input: 'qol-assist-migration.service.in',
-    output: 'qol-assist-migration.service.in',
+    output: 'qol-assist-migration.service',
     configuration: data_conf,
     install_dir: with_systemd_system_unit_dir,
 )
-- 
2.15.0

