From 7dc5f4cbb480ab0ee4a15515054068cdf46e953a Mon Sep 17 00:00:00 2001
From: Ikey Doherty <ikey@solus-project.com>
Date: Thu, 16 Nov 2017 07:02:30 +0000
Subject: [PATCH 1/2] cli: Fix compilation issue with musl

Signed-off-by: Ikey Doherty <ikey@solus-project.com>
---
 src/cli/main.c | 1 -
 1 file changed, 1 deletion(-)

diff --git a/src/cli/main.c b/src/cli/main.c
index 2bea3a2..51203b5 100644
--- a/src/cli/main.c
+++ b/src/cli/main.c
@@ -66,7 +66,6 @@ int main(__qol_unused__ int argc, __qol_unused__ char **argv)
         /* Before we go anywhere, kill stdin */
         if (stdin && fileno(stdin) >= 0) {
                 close(fileno(stdin));
-                stdin = NULL;
         }
 
         if (argc < 2) {
-- 
2.15.0

