--- a/res/share/batch/freedesktop_integration/peazip.desktop
+++ b/res/share/batch/freedesktop_integration/peazip.desktop
@@ -1,8 +1,8 @@
 [Desktop Entry]
 Name=PeaZip
 Comment=PeaZip archive manager
-Exec=peazip %F
-TryExec=peazip
+Exec=/usr/bin/peazip %F
+TryExec=/usr/bin/peazip
 Icon=peazip
 Type=Application
 Terminal=false
@@ -13,20 +13,20 @@
 
 [Desktop Action add]
 Name=PeaZip, add to archive
-Exec=peazip -add2archive %F
-Icon=peazip_add
+Exec=/usr/bin/peazip -add2archive %F
+Icon=peazip
 
 [Desktop Action extract]
 Name=PeaZip, extract archive(s)
-Exec=peazip -ext2main %F
-Icon=peazip_extract
+Exec=/usr/bin/peazip -ext2main %F
+Icon=peazip
 
 [Desktop Action extracthere]
 Name=PeaZip, extract here
-Exec=peazip -ext2here %F
-Icon=peazip_extract
+Exec=/usr/bin/peazip -ext2here %F
+Icon=peazip
 
 [Desktop Action extractheresmart]
 Name=PeaZip, extract here (smart new folder)
-Exec=peazip -ext2folder %F
-Icon=peazip_extract
+Exec=/usr/bin/peazip -ext2folder %F
+Icon=peazip
