From 955809b995c07d43edf67edaa92c6bf3dc0ace1c Mon Sep 17 00:00:00 2001
From: Brian Prodoehl <bprodoehl@connectify.me>
Date: Mon, 7 Dec 2020 15:51:05 -0500
Subject: [PATCH] Reorder speedtest servers URLs, preferring c.speedtest.net
 domain

---
 speedtest.py | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/speedtest.py b/speedtest.py
index 92a2be0b9..5ba65f793 100755
--- a/speedtest.py
+++ b/speedtest.py
@@ -1246,10 +1246,10 @@ def get_servers(self, servers=None, exclude=None):
                     )
 
         urls = [
-            '://www.speedtest.net/speedtest-servers-static.php',
             'http://c.speedtest.net/speedtest-servers-static.php',
-            '://www.speedtest.net/speedtest-servers.php',
             'http://c.speedtest.net/speedtest-servers.php',
+            '://www.speedtest.net/speedtest-servers-static.php',
+            '://www.speedtest.net/speedtest-servers.php',
         ]
 
         headers = {}
