From 14a4b45fc65025cfa2d5dd260b41034ef08d0b38 Mon Sep 17 00:00:00 2001
From: Peter O'Connor <peter@solus-project.com>
Date: Tue, 24 Apr 2018 16:10:52 +1000
Subject: [PATCH] Fix path for nice

Signed-off-by: Peter O'Connor <peter@solus-project.com>
---
 src/jobs/meltjob.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/jobs/meltjob.cpp b/src/jobs/meltjob.cpp
index e3c58a7..7979732 100644
--- a/src/jobs/meltjob.cpp
+++ b/src/jobs/meltjob.cpp
@@ -69,7 +69,7 @@ void MeltJob::start()
     QProcess::start(meltPath.absoluteFilePath(), args);
 #else
     args.prepend(meltPath.absoluteFilePath());
-    QProcess::start("/usr/bin/nice", args);
+    QProcess::start("/bin/nice", args);
 #endif
     AbstractJob::start();
 }
-- 
2.16.3

