From 0243dec4a8c43e20e9cd2e3b9a59d4fcd8a5cdea Mon Sep 17 00:00:00 2001
From: Alexander Vorobyev <avorobyev@protonmail.com>
Date: Tue, 12 Nov 2019 19:27:40 +0300
Subject: Fix lua binary

---
 Makefile | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/Makefile b/Makefile
index 26d333c..3a08061 100644
--- a/Makefile
+++ b/Makefile
@@ -3,7 +3,7 @@ PREFIX ?= /usr
 all: mpv check
 
 mpv: gen.lua
-	@env MPV_BASHCOMPGEN_VERBOSE=1 lua $< > $@
+	@env MPV_BASHCOMPGEN_VERBOSE=1 luajit $< > $@
 
 check: mpv
 	@echo Checking Bash syntax...
-- 
2.23.0

