From 2dac61951eb706780060ddb335d9586678fc985b Mon Sep 17 00:00:00 2001
From: Alexander Vorobyev <avorobyev@protonmail.com>
Date: Sun, 22 Nov 2020 14:12:30 +0300
Subject: Patch youtube-dl references

---
 player/lua/ytdl_hook.lua | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/player/lua/ytdl_hook.lua b/player/lua/ytdl_hook.lua
index e9a7b3aac7..6420015e60 100644
--- a/player/lua/ytdl_hook.lua
+++ b/player/lua/ytdl_hook.lua
@@ -8,7 +8,7 @@ local o = {
     use_manifests = false,
     all_formats = false,
     force_all_formats = true,
-    ytdl_path = "youtube-dl",
+    ytdl_path = "youtube-dlc",
 }
 
 local ytdl = {
-- 
2.29.2

