From 6840d7b65449878e576dade611e02b38678e8a19 Mon Sep 17 00:00:00 2001
From: Joshua Strobl <joshua@streambits.io>
Date: Tue, 27 Oct 2020 22:11:30 +0200
Subject: [PATCH 1/1] Patch youtube-dl references

---
 player/lua/ytdl_hook.lua | 10 +++++-----
 1 file changed, 5 insertions(+), 5 deletions(-)

diff --git a/player/lua/ytdl_hook.lua b/player/lua/ytdl_hook.lua
index 7e77eb01ba..ec5693e02b 100644
--- a/player/lua/ytdl_hook.lua
+++ b/player/lua/ytdl_hook.lua
@@ -10,7 +10,7 @@ local o = {
 options.read_options(o)
 
 local ytdl = {
-    path = "youtube-dl",
+    path = "youtube-dlc",
     searched = false,
     blacklisted = {}
 }
@@ -465,7 +465,7 @@ local function check_version(ytdl_path)
     end
     local version_ts = os.time{year=year, month=month, day=day}
     if (os.difftime(os.time(), version_ts) > 60*60*24*90) then
-        msg.warn("It appears that your youtube-dl version is severely out of date.")
+        msg.warn("It appears that your youtube-dlc version is severely out of date.")
     end
 end
 
@@ -477,7 +477,7 @@ function run_ytdl_hook(url)
         local exesuf = (package.config:sub(1,1) == '\\') and '.exe' or ''
         local ytdl_mcd = mp.find_config_file("youtube-dl" .. exesuf)
         if not (ytdl_mcd == nil) then
-            msg.verbose("found youtube-dl at: " .. ytdl_mcd)
+            msg.verbose("found youtube-dlc at: " .. ytdl_mcd)
             ytdl.path = ytdl_mcd
         end
         ytdl.searched = true
@@ -545,7 +545,7 @@ function run_ytdl_hook(url)
         -- trim our stderr to avoid spurious newlines
         ytdl_err = result.stderr:gsub("^%s*(.-)%s*$", "%1")
         msg.error(ytdl_err)
-        local err = "youtube-dl failed: "
+        local err = "youtube-dlc failed: "
         if result.error_string and result.error_string == "init" then
             err = err .. "not found or not enough permissions"
         elseif not result.killed_by_us then
@@ -568,7 +568,7 @@ function run_ytdl_hook(url)
         return
     end
 
-    msg.verbose("youtube-dl succeeded!")
+    msg.verbose("youtube-dlc succeeded!")
     msg.debug('ytdl parsing took '..os.clock()-start_time..' seconds')
 
     json["proxy"] = json["proxy"] or proxy
-- 
2.29.1

