--- a/options/path.c
+++ b/options/path.c
@@ -63,6 +63,7 @@ static const char *const config_dirs[] = {
     "osxbundle",
     "exe_dir",
     "global",
+    "stateless",
 };
 
 // Return a platform specific path using a path type as defined in osdep/path.h.
--- a/osdep/path-unix.c
+++ b/osdep/path-unix.c
@@ -133,6 +133,8 @@ const char *mp_get_platform_path_unix(void *talloc_ctx, const char *type)
         return mpv_state;
     if (strcmp(type, "global") == 0)
         return MPV_CONFDIR;
+    if (strcmp(type, "stateless") == 0)
+        return "/usr/share/mpv";
     if (strcmp(type, "desktop") == 0)
         return mpv_desktop;
     return NULL;
