From ddfecdb65aefdf6011a432dff73a31d013db3394 Mon Sep 17 00:00:00 2001
From: Fabio Forni <livingsilver94.solus@redaril.me>
Date: Thu, 19 Aug 2021 00:06:28 +0200
Subject: [PATCH] Use ayatana-appindicator

---
 AppIndicator.xs | 2 +-
 Makefile.PL     | 6 +++---
 2 files changed, 4 insertions(+), 4 deletions(-)

diff --git a/AppIndicator.xs b/AppIndicator.xs
index fb2fb68..3b2a104 100644
--- a/AppIndicator.xs
+++ b/AppIndicator.xs
@@ -3,7 +3,7 @@
 #include "XSUB.h"
 
 #include <glib.h>
-#include <libappindicator/app-indicator.h>
+#include <libayatana-appindicator/app-indicator.h>
 #include <gtk/gtk.h>
 #include <gperl.h>
 
diff --git a/Makefile.PL b/Makefile.PL
index 0d3a76d..c8809a9 100644
--- a/Makefile.PL
+++ b/Makefile.PL
@@ -3,18 +3,18 @@ use ExtUtils::MakeMaker;
 # See lib/ExtUtils/MakeMaker.pm for details of how to influence
 # the contents of the Makefile that is written.
 
-open my $fh,"pkg-config --cflags appindicator-0.1 gtk+-2.0 2>/dev/null |";
+open my $fh,"pkg-config --cflags ayatana-appindicator-0.1 gtk+-2.0 2>/dev/null |";
 my $inc=<$fh>;
 chomp $inc;
 close $fh;
 
-open $fh,"pkg-config --libs appindicator-0.1 gtk+-2.0 2>/dev/null |";
+open $fh,"pkg-config --libs ayatana-appindicator-0.1 gtk+-2.0 2>/dev/null |";
 my $lib=<$fh>;
 chomp $lib;
 close $fh;
 
 if (not($inc)) {
-	die "pkg-config --cflags appindicator-0.1 gtk+-2.0 didn't succeed, make sure it is righly installed";
+	die "pkg-config --cflags ayatana-appindicator-0.1 gtk+-2.0 didn't succeed, make sure it is righly installed";
 }
 
 
-- 
2.32.0

