commit ddc444b19693ebe686d0f748eb5898768d61a7cb
Author: Maik Wöhl <maik.woehl@outlook.de>
Date:   Sat Jan 20 15:19:18 2024 +0100

    Make project buildable

diff --git a/MANIFEST.in b/MANIFEST.in
new file mode 100644
index 0000000..fca4e5b
--- /dev/null
+++ b/MANIFEST.in
@@ -0,0 +1,2 @@
+graft src/ui
+graft src/resources
diff --git a/pyproject.toml b/pyproject.toml
new file mode 100644
index 0000000..e0f1aac
--- /dev/null
+++ b/pyproject.toml
@@ -0,0 +1,14 @@
+[project]
+name = "onedrive-gui"
+version = "1.0.3"
+description = "A simple GUI for Linux OneDrive Client, with multi-account support."
+readme = "README.md"
+requires-python = ">=3.8"
+license = {file = "LICENSE"}
+
+[project.scripts]
+onedrive-gui = "OneDriveGUI:main"
+
+[build-system]
+requires = ["setuptools >= 61.0"]
+build-backend = "setuptools.build_meta"
diff --git a/src/OneDriveGUI.py b/src/OneDriveGUI.py
index 608aa39..5fd04d7 100644
--- a/src/OneDriveGUI.py
+++ b/src/OneDriveGUI.py
@@ -3164,7 +3164,8 @@ def shorten_path(path, limit):
     return path
 
 
-if __name__ == "__main__":
+def main():
+    global gui_settings
     gui_settings = read_gui_settings()
 
     logging.basicConfig(
@@ -3173,21 +3174,31 @@ if __name__ == "__main__":
         level=config_debug_level(),
     )
 
+    global client_bin_path
     client_bin_path = config_client_bin_path()
+    global client_version
     client_version = get_installed_client_version()
+    global global_config
     global_config = create_global_config()
 
     if len(global_config) > 0:
         save_global_config()
 
+    global temp_global_config
     temp_global_config = copy.deepcopy(global_config)
 
+    global app
     app = QApplication(sys.argv)
     app.setApplicationName("OneDriveGUI")
     app.setWindowIcon(QIcon(DIR_PATH + "/resources/images/icons8-clouds-48.png"))
 
+    global main_window
     main_window = MainWindow()
     main_window_start_state()
+    global profile_settings_window
     profile_settings_window = ProfileSettingsWindow()
 
     app.exec()
+
+if __name__ == "__main__":
+    main()
diff --git a/src/resources/OneDriveGUI.desktop b/src/resources/OneDriveGUI.desktop
index 5feef90..44f9bf2 100644
--- a/src/resources/OneDriveGUI.desktop
+++ b/src/resources/OneDriveGUI.desktop
@@ -1,8 +1,7 @@
 [Desktop Entry]
 Name=OneDriveGUI
 StartupNotify=true
-Exec=/usr/bin/python3 /home/bob/host_share/Python/OneDriveGUI/src/OneDriveGUI.py
+Exec=/usr/bin/onedrive-gui
 Terminal=false
-Path=/home/bob/host_share/Python/OneDriveGUI/src/resources/images
-Icon=/home/bob/host_share/Python/OneDriveGUI/src/resources/images/icons8-clouds-48.png
-Type=Application
\ No newline at end of file
+Icon=onedrive-gui
+Type=Application
