commit 8847e5a1f2a96aabca97c7bddbee9c05a48d2583
Author: Maik Wöhl <maik.woehl@outlook.de>
Date:   Sun Jan 21 16:50:39 2024 +0100

    Create package directory

diff --git a/MANIFEST.in b/MANIFEST.in
index fca4e5b..6aa67f5 100644
--- a/MANIFEST.in
+++ b/MANIFEST.in
@@ -1,2 +1,2 @@
-graft src/ui
-graft src/resources
+graft src/onedrive_gui/ui
+graft src/onedrive_gui/resources
diff --git a/pyproject.toml b/pyproject.toml
index e0f1aac..b84da14 100644
--- a/pyproject.toml
+++ b/pyproject.toml
@@ -7,7 +7,7 @@ requires-python = ">=3.8"
 license = {file = "LICENSE"}
 
 [project.scripts]
-onedrive-gui = "OneDriveGUI:main"
+onedrive-gui = "onedrive_gui.OneDriveGUI:main"
 
 [build-system]
 requires = ["setuptools >= 61.0"]
diff --git a/src/OneDriveGUI.py b/src/onedrive_gui/OneDriveGUI.py
similarity index 99%
rename from src/OneDriveGUI.py
rename to src/onedrive_gui/OneDriveGUI.py
index 5fd04d7..60c0d0e 100644
--- a/src/OneDriveGUI.py
+++ b/src/onedrive_gui/OneDriveGUI.py
@@ -43,26 +43,26 @@ from urllib3 import HTTPSConnectionPool
 # TODO: Split into multiple files once all main features are implemented.
 
 # Imports for main window.
-from ui.ui_mainwindow import Ui_MainWindow
-from ui.ui_list_item_widget import Ui_list_item_widget
-from ui.ui_process_status_page import Ui_status_page
+from .ui.ui_mainwindow import Ui_MainWindow
+from .ui.ui_list_item_widget import Ui_list_item_widget
+from .ui.ui_process_status_page import Ui_status_page
 
 
 # Imports for Profiles windows.
-from ui.ui_profile_settings_window import Ui_profile_settings_window
-from ui.ui_profile_settings_page import Ui_profile_settings
+from .ui.ui_profile_settings_window import Ui_profile_settings_window
+from .ui.ui_profile_settings_page import Ui_profile_settings
 
 # Imports for GUI settings window
-from ui.ui_gui_settings_window import Ui_gui_settings_window
+from .ui.ui_gui_settings_window import Ui_gui_settings_window
 
 # Import for login windows.
 # Don't use WebEngine login window when running from AppImage.
 DIR_PATH = os.path.dirname(os.path.realpath(__file__))
 APPIMAGE = True if "tmp/.mount_one" in DIR_PATH.lower() else False
 if APPIMAGE:
-    from ui.ui_external_login import Ui_ExternalLoginWindow
+    from .ui.ui_external_login import Ui_ExternalLoginWindow
 else:
-    from ui.ui_login import Ui_LoginWindow
+    from .ui.ui_login import Ui_LoginWindow
 
 
 PROFILES_FILE = os.path.expanduser("~/.config/onedrive-gui/profiles")
diff --git a/src/onedrive_gui/__init__.py b/src/onedrive_gui/__init__.py
new file mode 100644
index 0000000..e69de29
diff --git a/src/resources/OneDriveGUI.desktop b/src/onedrive_gui/resources/OneDriveGUI.desktop
similarity index 100%
rename from src/resources/OneDriveGUI.desktop
rename to src/onedrive_gui/resources/OneDriveGUI.desktop
diff --git a/src/resources/default_config b/src/onedrive_gui/resources/default_config
similarity index 100%
rename from src/resources/default_config
rename to src/onedrive_gui/resources/default_config
diff --git a/src/resources/images/OneDriveGUI.ico b/src/onedrive_gui/resources/images/OneDriveGUI.ico
similarity index 100%
rename from src/resources/images/OneDriveGUI.ico
rename to src/onedrive_gui/resources/images/OneDriveGUI.ico
diff --git a/src/resources/images/OneDriveGUI.png b/src/onedrive_gui/resources/images/OneDriveGUI.png
similarity index 100%
rename from src/resources/images/OneDriveGUI.png
rename to src/onedrive_gui/resources/images/OneDriveGUI.png
diff --git a/src/resources/images/account.png b/src/onedrive_gui/resources/images/account.png
similarity index 100%
rename from src/resources/images/account.png
rename to src/onedrive_gui/resources/images/account.png
diff --git a/src/resources/images/close-filled.png b/src/onedrive_gui/resources/images/close-filled.png
similarity index 100%
rename from src/resources/images/close-filled.png
rename to src/onedrive_gui/resources/images/close-filled.png
diff --git a/src/resources/images/folder.png b/src/onedrive_gui/resources/images/folder.png
similarity index 100%
rename from src/resources/images/folder.png
rename to src/onedrive_gui/resources/images/folder.png
diff --git a/src/resources/images/gear.png b/src/onedrive_gui/resources/images/gear.png
similarity index 100%
rename from src/resources/images/gear.png
rename to src/onedrive_gui/resources/images/gear.png
diff --git a/src/resources/images/icons8-cloud-cross-40_2.png b/src/onedrive_gui/resources/images/icons8-cloud-cross-40_2.png
similarity index 100%
rename from src/resources/images/icons8-cloud-cross-40_2.png
rename to src/onedrive_gui/resources/images/icons8-cloud-cross-40_2.png
diff --git a/src/resources/images/icons8-cloud-done-40_2.png b/src/onedrive_gui/resources/images/icons8-cloud-done-40_2.png
similarity index 100%
rename from src/resources/images/icons8-cloud-done-40_2.png
rename to src/onedrive_gui/resources/images/icons8-cloud-done-40_2.png
diff --git a/src/resources/images/icons8-cloud-sync-40_2.png b/src/onedrive_gui/resources/images/icons8-cloud-sync-40_2.png
similarity index 100%
rename from src/resources/images/icons8-cloud-sync-40_2.png
rename to src/onedrive_gui/resources/images/icons8-cloud-sync-40_2.png
diff --git a/src/resources/images/icons8-clouds-48.ico b/src/onedrive_gui/resources/images/icons8-clouds-48.ico
similarity index 100%
rename from src/resources/images/icons8-clouds-48.ico
rename to src/onedrive_gui/resources/images/icons8-clouds-48.ico
diff --git a/src/resources/images/icons8-clouds-48.png b/src/onedrive_gui/resources/images/icons8-clouds-48.png
similarity index 100%
rename from src/resources/images/icons8-clouds-48.png
rename to src/onedrive_gui/resources/images/icons8-clouds-48.png
diff --git a/src/resources/images/icons8-clouds-80-dark-edge.png b/src/onedrive_gui/resources/images/icons8-clouds-80-dark-edge.png
similarity index 100%
rename from src/resources/images/icons8-clouds-80-dark-edge.png
rename to src/onedrive_gui/resources/images/icons8-clouds-80-dark-edge.png
diff --git a/src/resources/images/icons8-green-circle-48.png b/src/onedrive_gui/resources/images/icons8-green-circle-48.png
similarity index 100%
rename from src/resources/images/icons8-green-circle-48.png
rename to src/onedrive_gui/resources/images/icons8-green-circle-48.png
diff --git a/src/resources/images/icons8-red-circle-48.png b/src/onedrive_gui/resources/images/icons8-red-circle-48.png
similarity index 100%
rename from src/resources/images/icons8-red-circle-48.png
rename to src/onedrive_gui/resources/images/icons8-red-circle-48.png
diff --git a/src/resources/images/menu.png b/src/onedrive_gui/resources/images/menu.png
similarity index 100%
rename from src/resources/images/menu.png
rename to src/onedrive_gui/resources/images/menu.png
diff --git a/src/resources/images/pause.png b/src/onedrive_gui/resources/images/pause.png
similarity index 100%
rename from src/resources/images/pause.png
rename to src/onedrive_gui/resources/images/pause.png
diff --git a/src/resources/images/play.png b/src/onedrive_gui/resources/images/play.png
similarity index 100%
rename from src/resources/images/play.png
rename to src/onedrive_gui/resources/images/play.png
diff --git a/src/resources/images/quit.png b/src/onedrive_gui/resources/images/quit.png
similarity index 100%
rename from src/resources/images/quit.png
rename to src/onedrive_gui/resources/images/quit.png
diff --git a/src/resources/images/stop.png b/src/onedrive_gui/resources/images/stop.png
similarity index 100%
rename from src/resources/images/stop.png
rename to src/onedrive_gui/resources/images/stop.png
diff --git a/src/resources/images/storage.png b/src/onedrive_gui/resources/images/storage.png
similarity index 100%
rename from src/resources/images/storage.png
rename to src/onedrive_gui/resources/images/storage.png
diff --git a/src/resources/images/user-account.png b/src/onedrive_gui/resources/images/user-account.png
similarity index 100%
rename from src/resources/images/user-account.png
rename to src/onedrive_gui/resources/images/user-account.png
diff --git a/src/resources/images/warning.png b/src/onedrive_gui/resources/images/warning.png
similarity index 100%
rename from src/resources/images/warning.png
rename to src/onedrive_gui/resources/images/warning.png
diff --git a/src/ui/create_new_profile.ui b/src/onedrive_gui/ui/create_new_profile.ui
similarity index 100%
rename from src/ui/create_new_profile.ui
rename to src/onedrive_gui/ui/create_new_profile.ui
diff --git a/src/ui/external_login.ui b/src/onedrive_gui/ui/external_login.ui
similarity index 100%
rename from src/ui/external_login.ui
rename to src/onedrive_gui/ui/external_login.ui
diff --git a/src/ui/gui_settings_window.ui b/src/onedrive_gui/ui/gui_settings_window.ui
similarity index 100%
rename from src/ui/gui_settings_window.ui
rename to src/onedrive_gui/ui/gui_settings_window.ui
diff --git a/src/ui/import_existing_profile.ui b/src/onedrive_gui/ui/import_existing_profile.ui
similarity index 100%
rename from src/ui/import_existing_profile.ui
rename to src/onedrive_gui/ui/import_existing_profile.ui
diff --git a/src/ui/list_item_widget.ui b/src/onedrive_gui/ui/list_item_widget.ui
similarity index 100%
rename from src/ui/list_item_widget.ui
rename to src/onedrive_gui/ui/list_item_widget.ui
diff --git a/src/ui/login.ui b/src/onedrive_gui/ui/login.ui
similarity index 100%
rename from src/ui/login.ui
rename to src/onedrive_gui/ui/login.ui
diff --git a/src/ui/mainwindow.ui b/src/onedrive_gui/ui/mainwindow.ui
similarity index 100%
rename from src/ui/mainwindow.ui
rename to src/onedrive_gui/ui/mainwindow.ui
diff --git a/src/ui/mainwindow.ui.autosave b/src/onedrive_gui/ui/mainwindow.ui.autosave
similarity index 100%
rename from src/ui/mainwindow.ui.autosave
rename to src/onedrive_gui/ui/mainwindow.ui.autosave
diff --git a/src/ui/process_status_page.ui b/src/onedrive_gui/ui/process_status_page.ui
similarity index 100%
rename from src/ui/process_status_page.ui
rename to src/onedrive_gui/ui/process_status_page.ui
diff --git a/src/ui/profile_settings_page.ui b/src/onedrive_gui/ui/profile_settings_page.ui
similarity index 100%
rename from src/ui/profile_settings_page.ui
rename to src/onedrive_gui/ui/profile_settings_page.ui
diff --git a/src/ui/profile_settings_window.ui b/src/onedrive_gui/ui/profile_settings_window.ui
similarity index 100%
rename from src/ui/profile_settings_window.ui
rename to src/onedrive_gui/ui/profile_settings_window.ui
diff --git a/src/ui/setup_wizard.ui b/src/onedrive_gui/ui/setup_wizard.ui
similarity index 100%
rename from src/ui/setup_wizard.ui
rename to src/onedrive_gui/ui/setup_wizard.ui
diff --git a/src/ui/test.ui b/src/onedrive_gui/ui/test.ui
similarity index 100%
rename from src/ui/test.ui
rename to src/onedrive_gui/ui/test.ui
diff --git a/src/ui/ui_create_new_profile.py b/src/onedrive_gui/ui/ui_create_new_profile.py
similarity index 100%
rename from src/ui/ui_create_new_profile.py
rename to src/onedrive_gui/ui/ui_create_new_profile.py
diff --git a/src/ui/ui_external_login.py b/src/onedrive_gui/ui/ui_external_login.py
similarity index 100%
rename from src/ui/ui_external_login.py
rename to src/onedrive_gui/ui/ui_external_login.py
diff --git a/src/ui/ui_gui_settings_window.py b/src/onedrive_gui/ui/ui_gui_settings_window.py
similarity index 100%
rename from src/ui/ui_gui_settings_window.py
rename to src/onedrive_gui/ui/ui_gui_settings_window.py
diff --git a/src/ui/ui_import_existing_profile.py b/src/onedrive_gui/ui/ui_import_existing_profile.py
similarity index 100%
rename from src/ui/ui_import_existing_profile.py
rename to src/onedrive_gui/ui/ui_import_existing_profile.py
diff --git a/src/ui/ui_list_item_widget.py b/src/onedrive_gui/ui/ui_list_item_widget.py
similarity index 100%
rename from src/ui/ui_list_item_widget.py
rename to src/onedrive_gui/ui/ui_list_item_widget.py
diff --git a/src/ui/ui_login.py b/src/onedrive_gui/ui/ui_login.py
similarity index 100%
rename from src/ui/ui_login.py
rename to src/onedrive_gui/ui/ui_login.py
diff --git a/src/ui/ui_mainwindow.py b/src/onedrive_gui/ui/ui_mainwindow.py
similarity index 100%
rename from src/ui/ui_mainwindow.py
rename to src/onedrive_gui/ui/ui_mainwindow.py
diff --git a/src/ui/ui_process_status_page.py b/src/onedrive_gui/ui/ui_process_status_page.py
similarity index 100%
rename from src/ui/ui_process_status_page.py
rename to src/onedrive_gui/ui/ui_process_status_page.py
diff --git a/src/ui/ui_profile_settings_page.py b/src/onedrive_gui/ui/ui_profile_settings_page.py
similarity index 100%
rename from src/ui/ui_profile_settings_page.py
rename to src/onedrive_gui/ui/ui_profile_settings_page.py
diff --git a/src/ui/ui_profile_settings_window.py b/src/onedrive_gui/ui/ui_profile_settings_window.py
similarity index 100%
rename from src/ui/ui_profile_settings_window.py
rename to src/onedrive_gui/ui/ui_profile_settings_window.py
