From 0380f1d7d96d9643ebee7d32971919a12321fa6e Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Maik=20W=C3=B6hl?= <maik.woehl@outlook.de>
Date: Fri, 17 Jul 2026 20:39:18 +0200
Subject: [PATCH 3/4] Introduce main method to call
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

Signed-off-by: Maik Wöhl <maik.woehl@outlook.de>
---
 src/onedrive_gui/OneDriveGUI.py | 24 ++++++++++--------------
 1 file changed, 10 insertions(+), 14 deletions(-)

diff --git a/src/onedrive_gui/OneDriveGUI.py b/src/onedrive_gui/OneDriveGUI.py
index 6a05a59..26c734a 100644
--- a/src/onedrive_gui/OneDriveGUI.py
+++ b/src/onedrive_gui/OneDriveGUI.py
@@ -26,11 +26,16 @@ from .options import gui_settings, global_config, version
 from .global_config import save_global_config
 from .main_window import MainWindow
 
+workers = {}
+
+def main():
+    logging.info(f"Starting OneDriveGUI v{version}")
 
-def main_window_start_state():
-    # Determine if OneDriveGUI should start maximized, minimized to tray or minimized to taskbar/dock.
-    # This should help ensure the GUI does not just disappear on Gnome without system tray extension.
+    if len(global_config) > 0:
+        save_global_config(global_config)
 
+    main_window = MainWindow()
+    
     if gui_settings.get("start_minimized") == "True" or len(global_config) == 0:
         try:
             if main_window.tray.isSystemTrayAvailable():
@@ -45,16 +50,7 @@ def main_window_start_state():
         logging.info("[GUI] Starting OneDriveGUI maximized")
 
 
-workers = {}
-
+    app.exec()
 
 if __name__ == "__main__":
-    logging.info(f"Starting OneDriveGUI v{version}")
-
-    if len(global_config) > 0:
-        save_global_config(global_config)
-
-    main_window = MainWindow()
-    main_window_start_state()
-
-    app.exec()
+    main()
\ No newline at end of file
-- 
2.55.0

