From: Fabian Vogt <fvogt@suse.com>
Subject: Also use a linker version script for libQt5WebKit

To maintain ABI compatibility with the older libQt5WebKit5 packages.

https://github.com/annulen/webkit/issues/597

diff --git a/Source/WebKit/CMakeLists.txt b/Source/WebKit/CMakeLists.txt
index 82dc1e662a5..bfe480c7cf4 100644
--- a/Source/WebKit/CMakeLists.txt
+++ b/Source/WebKit/CMakeLists.txt
@@ -78,6 +78,13 @@ if (${PORT} STREQUAL "Qt")
     if (ENABLE_WEBKIT2)
         add_dependencies(WebKit WebKit2)
     endif ()
+
+    set(VERSION_SCRIPT "${CMAKE_BINARY_DIR}/QtWebKit.version")
+    add_custom_command(TARGET WebKit PRE_LINK
+        COMMAND ${PERL_EXECUTABLE} ${TOOLS_DIR}/qt/generate-version-script.pl ${Qt5_VERSION} > ${VERSION_SCRIPT}
+        VERBATIM
+    )
+    set_target_properties(WebKit PROPERTIES LINK_FLAGS -Wl,--version-script,${VERSION_SCRIPT})
 endif ()
 
 add_dependencies(WebKit WebCore)
diff --git a/Source/WebKit2/CMakeLists.txt b/Source/WebKit2/CMakeLists.txt
index e5d08ed68ba..6736689a2d1 100644
--- a/Source/WebKit2/CMakeLists.txt
+++ b/Source/WebKit2/CMakeLists.txt
@@ -776,6 +776,13 @@ if (WebKit2_VERSION_SCRIPT)
     ADD_TARGET_PROPERTIES(WebKit2 LINK_FLAGS "${WebKit2_VERSION_SCRIPT}")
 endif ()
 
+set(VERSION_SCRIPT "${CMAKE_BINARY_DIR}/QtWebKit2.version")
+add_custom_command(TARGET WebKit2 PRE_LINK
+    COMMAND ${PERL_EXECUTABLE} ${TOOLS_DIR}/qt/generate-version-script.pl ${Qt5_VERSION} > ${VERSION_SCRIPT}
+    VERBATIM
+)
+set_target_properties(WebKit2 PROPERTIES LINK_FLAGS -Wl,--version-script,${VERSION_SCRIPT})
+
 add_executable(WebProcess ${WebProcess_SOURCES})
 add_webkit2_prefix_header(WebProcess)
 target_link_libraries(WebProcess ${WebProcess_LIBRARIES})
