--- a/src/diagnostics/src/SOS/extensions/hostcoreclr.cpp 2026-01-31 14:26:09.052734212 -0800
+++ b/src/diagnostics/src/SOS/extensions/hostcoreclr.cpp 2026-01-31 14:26:13.120580598 -0800
@@ -126,6 +126,7 @@
 #endif
         "/usr/local/share/dotnet"
 #else
+        "/usr/lib64/dotnet",
         "/rh-dotnet90/root/usr/bin/dotnet",
         "/rh-dotnet80/root/usr/bin/dotnet",
         "/rh-dotnet60/root/usr/bin/dotnet",