diff --git a/src/diagnostics/src/SOS/extensions/hostcoreclr.cpp b/src/diagnostics/src/SOS/extensions/hostcoreclr.cpp
index 3bd499dc27..f174bfa7ea 100644
--- a/src/diagnostics/src/SOS/extensions/hostcoreclr.cpp
+++ b/src/diagnostics/src/SOS/extensions/hostcoreclr.cpp
@@ -127,6 +127,7 @@ namespace RuntimeHostingConstants
 #endif
         "/usr/local/share/dotnet"
 #else
+        "/usr/lib64/dotnet",
         "/rh-dotnet60/root/usr/bin/dotnet",
         "/rh-dotnet70/root/usr/bin/dotnet",
         "/rh-dotnet80/root/usr/bin/dotnet",
diff --git a/src/runtime/src/native/corehost/hostmisc/pal.unix.cpp b/src/runtime/src/native/corehost/hostmisc/pal.unix.cpp
index 613902b5ea..ffd14e14db 100644
--- a/src/runtime/src/native/corehost/hostmisc/pal.unix.cpp
+++ b/src/runtime/src/native/corehost/hostmisc/pal.unix.cpp
@@ -608,7 +608,7 @@ bool pal::get_default_installation_dir_for_arch(pal::architecture arch, pal::str
         recv->assign(_X("/usr/local/share/dotnet"));
     }
 #else
-    recv->assign(_X("/usr/share/dotnet"));
+    recv->assign(_X("/usr/lib64/dotnet"));
 #endif
     return true;
 }
