From 7e77648eeddf6724eb93045d6bf9355ce432e07d Mon Sep 17 00:00:00 2001
From: Joshua Strobl <joshua@stroblindustries.com>
Date: Fri, 10 Jun 2016 04:07:31 +0300
Subject: [PATCH 1/1] Change to correct dirs.

---
 install.sh | 13 ++++++-------
 1 file changed, 6 insertions(+), 7 deletions(-)

diff --git a/install.sh b/install.sh
index 96477a8..36cb024 100755
--- a/install.sh
+++ b/install.sh
@@ -52,13 +52,12 @@ if [ $# -eq 1 ] ; then
       mkdir -p $configdir
       ;;
     *)
-      bindir="$1/nim/bin"
-      configdir="$1/nim/config"
-      libdir="$1/nim/lib"
-      docdir="$1/nim/doc"
-      datadir="$1/nim/data"
-      nimbleDir="$1/nim"
-      mkdir -p $1/nim
+      bindir="$1/usr/bin"
+      configdir="$1/etc"
+      libdir="$1/usr/lib64/nim"
+      docdir="$1/usr/share/nim/doc"
+      datadir="$1/usr/share/nim/data"
+      nimbleDir="$1/usr/lib64/nim"
       mkdir -p $bindir
       mkdir -p $configdir
       ;;
-- 
2.8.3

