diff --git a/install.sh b/install.sh
index 5934353..4b19df0 100755
--- a/install.sh
+++ b/install.sh
@@ -104,10 +104,6 @@ if [ $# -eq 1 ] ; then
 
   cp bin/nim $bindir/nim
   chmod 755 $bindir/nim
-  cp bin/nim-gdb $bindir/nim-gdb
-  chmod 755 $bindir/nim-gdb
-  cp bin/nim-gdb.bash $bindir/nim-gdb.bash
-  chmod 755 $bindir/nim-gdb.bash
   cp config/nim.cfg $configdir/nim.cfg
   chmod 644 $configdir/nim.cfg
   cp config/nimdoc.cfg $configdir/nimdoc.cfg
