diff -Nru nim-0.17.2.orig/tests/async/tnewasyncudp.nim nim-0.17.2/tests/async/tnewasyncudp.nim
--- nim-0.17.2.orig/tests/async/tnewasyncudp.nim	2017-09-07 21:45:44.000000000 +0300
+++ nim-0.17.2/tests/async/tnewasyncudp.nim	2017-11-09 19:53:36.659185006 +0300
@@ -86,7 +86,7 @@
     size = 0
     var grammString = $buffer
     if grammString.startswith("Message ") and
-       saddr.sin_addr.s_addr == 0x100007F:
+       saddr.sin_addr.s_addr == nativesockets.ntohl(INADDR_LOOPBACK):
       await sendTo(server, addr grammString[0], len(grammString),
                    cast[ptr SockAddr](addr saddr), slen)
       inc(msgCount)
