*** install.sh	2015-05-05 03:43:14.000000000 +1000
--- install-eopkg.sh	2015-08-15 13:47:25.783834176 +1000
***************
*** 39,51 ****
        datadir=/usr/local/share/nim/data
        ;;
      *)
!       bindir="$1/nim/bin"
!       configdir="$1/nim/config"
!       libdir="$1/nim/lib"
!       docdir="$1/nim/doc"
!       datadir="$1/nim/data"
        
-       mkdir -p $1/nim
        mkdir -p $bindir
        mkdir -p $configdir
        ;;
--- 39,50 ----
        datadir=/usr/local/share/nim/data
        ;;
      *)
!       bindir="$1/usr/bin"
!       configdir="$1/etc"
!       libdir="$1/usr/lib/nim"
!       docdir="$1/usr/share/nim/doc"
!       datadir="$1/usr/share/nim/data"
        
        mkdir -p $bindir
        mkdir -p $configdir
        ;;
