From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Auke Kok <auke-jan.h.kok@intel.com>
Date: Mon, 18 May 2020 10:30:44 -0700
Subject: [PATCH] Attempt to make nim parse a system-wide fallback config file.

---
 compiler/nimconf.nim | 1 +
 1 file changed, 1 insertion(+)

diff --git a/compiler/nimconf.nim b/compiler/nimconf.nim
index c3591a9..bfc6763 100644
--- a/compiler/nimconf.nim
+++ b/compiler/nimconf.nim
@@ -237,6 +237,7 @@ proc getSystemConfigPath*(conf: ConfigRef; filename: RelativeFile): AbsoluteFile
   when defined(unix):
     if not fileExists(result): result = p / RelativeDir"etc/nim" / filename
     if not fileExists(result): result = AbsoluteDir"/etc/nim" / filename
+    if not fileExists(result): result = AbsoluteDir"/usr/share/defaults/nim" / filename
 
 proc loadConfigs*(cfg: RelativeFile; cache: IdentCache; conf: ConfigRef) =
   setDefaultLibpath(conf)
