From 9cd1e8909f0c536e465662c1edd949dce554e990 Mon Sep 17 00:00:00 2001
From: Clement Lefebvre <clement.lefebvre@linuxmint.com>
Date: Wed, 27 Nov 2024 12:04:42 +0000
Subject: [PATCH] nemo-share: Fix build

---
 nemo-share/src/nemo-share.c | 2 --
 1 file changed, 2 deletions(-)

diff --git a/nemo-share/src/nemo-share.c b/nemo-share/src/nemo-share.c
index 870d8736..5077473c 100644
--- a/nemo-share/src/nemo-share.c
+++ b/nemo-share/src/nemo-share.c
@@ -32,8 +32,6 @@
 #include <libnemo-extension/nemo-property-page-provider.h>
 #include <libnemo-extension/nemo-name-and-desc-provider.h>
 
-#include <libcinnamon-desktop/gnome-installer.h>
-
 #include "nemo-share.h"
 
 #include <glib/gi18n-lib.h>
