From 91ddf5cffa99afb3e74900f8080627708de21593 Mon Sep 17 00:00:00 2001
From: Pierre-Yves <pyu@riseup.net>
Date: Thu, 26 Jul 2018 21:54:26 +0200
Subject: [PATCH] Support build against quazip-qt5

Signed-off-by: Pierre-Yves <pyu@riseup.net>
---
 texstudio.pro | 10 +++++++---
 1 file changed, 7 insertions(+), 3 deletions(-)

diff --git a/texstudio.pro b/texstudio.pro
index abeba823..fe61b904 100644
--- a/texstudio.pro
+++ b/texstudio.pro
@@ -285,9 +285,13 @@ isEmpty(USE_SYSTEM_QUAZIP) {
   DEFINES += QUAZIP_STATIC
   include(src/quazip/quazip/quazip.pri)
 } else {
-  isEmpty(QUAZIP_LIB): QUAZIP_LIB = -lquazip
-  isEmpty(QUAZIP_INCLUDE): QUAZIP_INCLUDE = $${PREFIX}/include/quazip
-
+  greaterThan(QT_MAJOR_VERSION, 4) { #Qt5
+    isEmpty(QUAZIP_LIB): QUAZIP_LIB = -lquazip5
+    isEmpty(QUAZIP_INCLUDE): QUAZIP_INCLUDE = $${PREFIX}/include/quazip5
+  } else { #Qt4
+    isEmpty(QUAZIP_LIB): QUAZIP_LIB = -lquazip
+    isEmpty(QUAZIP_INCLUDE): QUAZIP_INCLUDE = $${PREFIX}/include/quazip
+  }
   INCLUDEPATH += $${QUAZIP_INCLUDE}
   LIBS += $${QUAZIP_LIB}
 }
