diff -r 209eafc9748b -r b236375ad1a8 src/enblend.cc
--- a/src/enblend.cc	Sat Nov 23 16:26:14 2019 +0100
+++ b/src/enblend.cc	Tue Nov 26 14:00:20 2019 +0100
@@ -231,12 +231,10 @@
 }
 
 
-#define DUMP_GLOBAL_VARIABLES(...) dump_global_variables(__FILE__, __LINE__, ##__VA_ARGS__)
-void dump_global_variables(const char* file, unsigned line,
-                           std::ostream& out = std::cout)
+void dump_global_variables(std::ostream& out = std::cout)
 {
     out <<
-        "+ " << file << ":" << line << ": state of global variables\n" <<
+        "+ State of global variables\n" <<
         "+ Verbose = " << Verbose << ", option \"--verbose\"\n" <<
         "+ OutputFileName = <" << OutputFileName << ">\n" <<
         "+ OutputMaskFileName = <" << OutputMaskFileName.value_or("<not defined>") << ">\n" <<
@@ -1805,7 +1803,7 @@
     }
 
     if (parameter::as_boolean("dump-global-variables", false)) {
-        DUMP_GLOBAL_VARIABLES();
+        dump_global_variables();
     }
 
     sig.check();
diff -r 209eafc9748b -r b236375ad1a8 src/enfuse.cc
--- a/src/enfuse.cc	Sat Nov 23 16:26:14 2019 +0100
+++ b/src/enfuse.cc	Tue Nov 26 14:00:20 2019 +0100
@@ -204,12 +204,10 @@
 double* enblend::Histogram<InputPixelType, ResultPixelType>::precomputedEntropy = nullptr;
 
 
-#define DUMP_GLOBAL_VARIABLES(...) dump_global_variables(__FILE__, __LINE__, ##__VA_ARGS__)
-void dump_global_variables(const char* file, unsigned line,
-                           std::ostream& out = std::cout)
+void dump_global_variables(std::ostream& out = std::cout)
 {
     out <<
-        "+ " << file << ":" << line << ": state of global variables\n" <<
+        "+ State of global variables\n" <<
         "+ Verbose = " << Verbose << ", option \"--verbose\"\n" <<
         "+ OutputFileName = <" << OutputFileName << ">\n" <<
         "+ OutputMaskFileName = <" << OutputMaskFileName.value_or("<not defined>") << ">\n" <<
@@ -2029,7 +2027,7 @@
     }
 
     if (parameter::as_boolean("dump-global-variables", false)) {
-        DUMP_GLOBAL_VARIABLES();
+        dump_global_variables();
     }
 
     sig.check();
 
 
 

