diff -r ef8470f9d682 src/anneal.h
--- a/src/anneal.h	Sun Feb 12 15:25:10 2017 +0100
+++ b/src/anneal.h	Tue Jul 28 10:48:23 2020 +0100
@@ -29,6 +29,7 @@
 #include <algorithm>
 #include <limits>
 #include <vector>
+#include <limits>
 
 #ifdef _WIN32
 #include <cmath>
diff -r ef8470f9d682 src/exposure_weight.cc
--- a/src/exposure_weight.cc	Sun Feb 12 15:25:10 2017 +0100
+++ b/src/exposure_weight.cc	Tue Jul 28 10:48:23 2020 +0100
@@ -21,6 +21,7 @@
 
 #include <cassert>
 #include <iostream>
+#include <limits>
 
 #include "global.h"
 #include "openmp_def.h"         // omp::atomic_t
diff -r ef8470f9d682 src/fixmath.h
--- a/src/fixmath.h	Sun Feb 12 15:25:10 2017 +0100
+++ b/src/fixmath.h	Tue Jul 28 10:48:23 2020 +0100
@@ -27,6 +27,7 @@
 #endif
 
 #include <cmath>
+#include <limits>
 
 #include <time.h>
 
diff -r ef8470f9d682 src/minimizer.h
--- a/src/minimizer.h	Sun Feb 12 15:25:10 2017 +0100
+++ b/src/minimizer.h	Tue Jul 28 10:48:23 2020 +0100
@@ -26,6 +26,7 @@
 #include <config.h>
 #endif
 
+#include <limits>
 #include <string>
 #include <stdexcept>
 #include <vector>
diff -r ef8470f9d682 src/opencl.h
--- a/src/opencl.h	Sun Feb 12 15:25:10 2017 +0100
+++ b/src/opencl.h	Tue Jul 28 10:48:23 2020 +0100
@@ -37,6 +37,7 @@
 #include <thread>
 #include <type_traits>          // std::add_const
 #include <vector>
+#include <limits>
 
 #define __CL_ENABLE_EXCEPTIONS
 
diff -r ef8470f9d682 src/openmp_vigra.h
--- a/src/openmp_vigra.h	Sun Feb 12 15:25:10 2017 +0100
+++ b/src/openmp_vigra.h	Tue Jul 28 10:48:23 2020 +0100
@@ -31,6 +31,7 @@
 #include <vigra/combineimages.hxx>
 #include <vigra/convolution.hxx>
 #include <vigra/distancetransform.hxx>
+#include <limits>
 
 #include "openmp_def.h"
 
diff -r ef8470f9d682 src/self_test.cc
--- a/src/self_test.cc	Sun Feb 12 15:25:10 2017 +0100
+++ b/src/self_test.cc	Tue Jul 28 10:48:23 2020 +0100
@@ -34,6 +34,7 @@
 #include <cmath>                // fabsf
 #include <iostream>
 #include <string>
+#include <limits>
 
 #include "self_test.h"
 
