From 7826e19de3794e1fbda6e5cac3ea6a2392dea2bc Mon Sep 17 00:00:00 2001
From: Peter O'Connor <peter@solus-project.com>
Date: Sat, 14 Apr 2018 20:10:07 +1000
Subject: [PATCH] Disable building unshipped code

Signed-off-by: Peter O'Connor <peter@solus-project.com>
---
 CMakeLists.txt | 2 --
 1 file changed, 2 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 3dfa8ca..f270f8b 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -54,8 +54,6 @@ remove_definitions(-DQT_NO_CAST_FROM_ASCII)
 find_package(KF5DocTools ${KF5_DEP_VERSION})
 
 add_subdirectory(src)
-add_subdirectory(autotests)
-add_subdirectory(tests)
 if(KF5DocTools_FOUND)
     add_subdirectory(docs)
 endif()
-- 
2.16.3

