From 5e83734a080554a5d52cc78df4a029a85dd2d33c Mon Sep 17 00:00:00 2001
From: Philipp Trulson <philipp@trulson.de>
Date: Fri, 22 Feb 2019 11:58:56 +0100
Subject: [PATCH] change-dest-for-mgc-file

---
 configure.ac | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/configure.ac b/configure.ac
index ec296aa4..e619d389 100644
--- a/configure.ac
+++ b/configure.ac
@@ -71,7 +71,7 @@ esac
 AC_SUBST(MINGW)
 AM_CONDITIONAL(MINGW, test "$MINGW" = 1)

-AC_SUBST([pkgdatadir], ['$(datadir)/misc'])
+AC_SUBST([pkgdatadir], ['$(datadir)/file'])
 AC_SUBST(fsect)
 AM_CONDITIONAL(FSECT5, test x$fsect = x5)

--
2.20.1
