From 1c1685a4417cf2902756505b1eea232d84519b51 Mon Sep 17 00:00:00 2001
From: Algent Albrahimi <algent@protonmail.com>
Date: Sat, 19 Feb 2022 21:20:59 +0100
Subject: [PATCH] fix install dir paths

---
 budgie-takeabreak/budgie_takeabreak.py | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/budgie-takeabreak/budgie_takeabreak.py b/budgie-takeabreak/budgie_takeabreak.py
index edd0104..d6ae4c9 100644
--- a/budgie-takeabreak/budgie_takeabreak.py
+++ b/budgie-takeabreak/budgie_takeabreak.py
@@ -263,9 +263,9 @@ class BudgieTakeaBreakApplet(Budgie.Applet):
         try:
             # I know, old school, but it works well
             pid = subprocess.check_output([
-                "/usr/bin/pgrep", "-f", "-u", user, self.runnerpath,
+                "/bin/pgrep", "-f", "-u", user, self.runnerpath,
             ]).decode("utf-8").strip()
-            subprocess.Popen(["/usr/bin/kill", pid])
+            subprocess.Popen(["/bin/kill", pid])
         except subprocess.CalledProcessError:
             pass
 
-- 
2.35.1

