diff --git a/budgie-takeabreak/budgie_takeabreak.py b/budgie-takeabreak/budgie_takeabreak.py
index 2bc25c7..0fb0bcf 100644
--- a/budgie-takeabreak/budgie_takeabreak.py
+++ b/budgie-takeabreak/budgie_takeabreak.py
@@ -266,7 +266,7 @@ class BudgieTakeaBreakApplet(Budgie.Applet):
             pid = subprocess.check_output([
                 "/usr/bin/pgrep", "-f", "-u", user, self.runnerpath,
             ]).decode("utf-8").strip()
-            subprocess.Popen(["/usr/bin/kill", pid])
+            subprocess.Popen(["/bin/kill", pid])
         except subprocess.CalledProcessError:
             pass
 
-- 
2.41.0
