diff --git a/budgie-extras-daemon/src/meson.build b/budgie-extras-daemon/src/meson.build
index f176202..661be12 100644
--- a/budgie-extras-daemon/src/meson.build
+++ b/budgie-extras-daemon/src/meson.build
@@ -6,7 +6,7 @@ custom_target('desktop-file-daemon',
     output : 'budgie-extras-daemon.desktop',
     command : [intltool, '--desktop-style', podir, '@INPUT@', '@OUTPUT@'],
     install : true,
-    install_dir : join_paths(get_option('sysconfdir'), 'xdg', 'autostart'))
+    install_dir : join_paths(get_option('datadir'), 'xdg', 'autostart'))
 
 subdir('config')
 
diff --git a/budgie-takeabreak/budgie_takeabreak.py b/budgie-takeabreak/budgie_takeabreak.py
index e518582..f37a4b5 100644
--- a/budgie-takeabreak/budgie_takeabreak.py
+++ b/budgie-takeabreak/budgie_takeabreak.py
@@ -233,9 +233,9 @@ class BudgieTakeaBreakApplet(Budgie.Applet):
         try:
             # I know, old school, but it works well
             pid = subprocess.check_output([
-                "/usr/bin/pgrep", "-f", "-u", user, app,
+                "/bin/pgrep", "-f", "-u", user, app,
             ]).decode("utf-8").strip()
-            subprocess.Popen(["/usr/bin/kill", pid])
+            subprocess.Popen(["/bin/kill", pid])
         except subprocess.CalledProcessError:
             pass
         if state:
diff --git a/budgie-weathershow/src/weathershow/WeatherShow.vala b/budgie-weathershow/src/weathershow/WeatherShow.vala
index 88814fe..7bfb4cf 100644
--- a/budgie-weathershow/src/weathershow/WeatherShow.vala
+++ b/budgie-weathershow/src/weathershow/WeatherShow.vala
@@ -77,7 +77,7 @@ namespace WeatherShowFunctions {
     }
 
     private bool check_onwindow(string path) {
-        string cmd_check = "/usr/bin/pgrep -f " + path;
+        string cmd_check = "/bin/pgrep -f " + path;
         string output;
         try {
             GLib.Process.spawn_command_line_sync(cmd_check, out output);
@@ -97,7 +97,7 @@ namespace WeatherShowFunctions {
         if (win_exists) {
             try {
                 Process.spawn_command_line_async(
-                    "/usr/bin/pkill -f ".concat(path));
+                    "/bin/pkill -f ".concat(path));
             }
             catch (SpawnError e) {
                 /* nothing to be done */
diff --git a/budgie-window-shuffler/data/meson.build b/budgie-window-shuffler/data/meson.build
index 9b528f5..d3cd6cf 100644
--- a/budgie-window-shuffler/data/meson.build
+++ b/budgie-window-shuffler/data/meson.build
@@ -9,7 +9,7 @@ shufflertargetauto = custom_target('shufflerautostart',
   input : 'shufflerdaemon-autostart.desktop.in',
   command : [substprog, '@INPUT@', LIB_INSTALL_DIR, '@OUTPUT@', podir],
   install : true,
-  install_dir : join_paths(sysconfdir, 'xdg', 'autostart')
+  install_dir : join_paths(datadir, 'xdg', 'autostart')
 )
 
 shufflerguitargetauto = custom_target('shufflerguiautostart',
@@ -17,7 +17,7 @@ shufflerguitargetauto = custom_target('shufflerguiautostart',
   input : 'shufflergui-autostart.desktop.in',
   command : [substprog, '@INPUT@', LIB_INSTALL_DIR, '@OUTPUT@', podir],
   install : true,
-  install_dir : join_paths(sysconfdir, 'xdg', 'autostart')
+  install_dir : join_paths(datadir, 'xdg', 'autostart')
 )
 
 shufflertargetw = custom_target('shufflercontrols',
diff --git a/budgie-window-shuffler/src/jump.vala b/budgie-window-shuffler/src/jump.vala
index b761c6a..9d4117c 100644
--- a/budgie-window-shuffler/src/jump.vala
+++ b/budgie-window-shuffler/src/jump.vala
@@ -47,7 +47,7 @@ namespace JumpActive {
     }
 
     private bool procruns (string processname) {
-        string cmd = @"/usr/bin/pgrep -f $processname";
+        string cmd = @"/bin/pgrep -f $processname";
         string output;
         try {
             GLib.Process.spawn_command_line_sync(cmd, out output);
@@ -233,4 +233,4 @@ namespace JumpActive {
             stderr.printf ("%s\n", e.message);
         }
     }
-}
\ No newline at end of file
+}
diff --git a/budgie-window-shuffler/src/shuffler_control.vala b/budgie-window-shuffler/src/shuffler_control.vala
index 66dcfc2..9235784 100644
--- a/budgie-window-shuffler/src/shuffler_control.vala
+++ b/budgie-window-shuffler/src/shuffler_control.vala
@@ -591,7 +591,7 @@ namespace ShufflerControls {
         }
 
         private bool procruns (string processname) {
-            string cmd = @"/usr/bin/pgrep -f $processname";
+            string cmd = @"/bin/pgrep -f $processname";
             string output;
             try {
                 GLib.Process.spawn_command_line_sync(cmd, out output);
-- 
2.26.0

