--- a/budgie-takeabreak/takeabreak_run
+++ b/budgie-takeabreak/takeabreak_run
@@ -107,14 +107,14 @@ def take_a_break():
             elif mode == "lock":
                 subprocess_tasks(
                     "run_async",
-                    ["gnome-screensaver-command", "-l"]
+                    ["budgie-screensaver-command", "-l"]
                 )
     time.sleep(sleeptime)
     for scr in screens:
         if auto_unlock:
             subprocess_tasks(
                 "run_async",
-                ["gnome-screensaver-command", "-d"]
+                ["budgie-screensaver-command", "-d"]
             )
         time.sleep(2)
         if mode == "rotate":
@@ -138,7 +138,7 @@ if from_unidle:
 
 
 def lockscreen_check():
-    lockproc = "gnome-screensaver-dialog"
+    lockproc = "budgie-screensaver-dialog"
     try:
         return lockproc in (p.name() for p in psutil.process_iter())
     except psutil.NoSuchProcess:
--
diff --git a/budgie-takeabreak/takeabreak_run b/budgie-takeabreak/takeabreak_run
old mode 100644
new mode 100755
-- 
2.41.0

