From 83061f56ff190be691e9b9401168311b0becf15e Mon Sep 17 00:00:00 2001
From: Thomas Staudinger <Staudi.Kaos@gmail.com>
Date: Tue, 24 Dec 2024 21:52:43 +0100
Subject: [PATCH] budgie-hotcorners: Fix formatting of defaults file JSON

Signed-off-by: Thomas Staudinger <Staudi.Kaos@gmail.com>
---
 budgie-hotcorners/misc/defaults.in | 26 +++++++++++++-------------
 1 file changed, 13 insertions(+), 13 deletions(-)

diff --git a/budgie-hotcorners/misc/defaults.in b/budgie-hotcorners/misc/defaults.in
index 57e03807..90bac834 100644
--- a/budgie-hotcorners/misc/defaults.in
+++ b/budgie-hotcorners/misc/defaults.in
@@ -1,13 +1,13 @@
-{'command': '@bdesktopdir@/showdesktop/showdesktop', 'name': 'Show Desktop'}
-{'command': '@libdir@/budgie-previews/previews_triggers hotcorners_all', 'name': 'Exposé all windows', 'gsettingsboolean': 'org.ubuntubudgie.budgie-wpreviews enable-previews'}
-{'command': '@libdir@/budgie-previews/previews_triggers hotcorners_current', 'name': 'Exposé current application', 'gsettingsboolean': 'org.ubuntubudgie.budgie-wpreviews enable-previews'}
-{'command': 'budgie-desktop-settings', 'name':'Budgie Desktop Settings'}
-{'command': 'dbus-send --session --type="method_call" --dest=org.gnome.ScreenSaver /org/gnome/ScreenSaver org.gnome.ScreenSaver.Lock', 'name': 'Lock screen'}
-{'command': '@libdir@/budgie-window-shuffler/togglegui', 'name': 'Toggle Shuffler GUI', 'gsettingsboolean': 'org.ubuntubudgie.windowshuffler runshufflergui'}
-{'command': '@libdir@/budgie-window-shuffler/grid_all', 'name': 'All windows in grid'}
-{'command': '@libdir@/budgie-window-shuffler/tile_active 0 0 2 2', 'name': 'Window to topleft'}
-{'command': '@libdir@/budgie-window-shuffler/tile_active 1 0 2 2', 'name': 'Window to topright'}
-{'command': '@libdir@/budgie-window-shuffler/tile_active 1 1 2 2', 'name': 'Window to bottomright'}
-{'command': '@libdir@/budgie-window-shuffler/tile_active 0 1 2 2', 'name': 'Window to bottomleft'}
-{'command': 'dbus-send --session --type="method_call" --dest=org.budgie_desktop.Panel /org/budgie_desktop/Raven org.budgie_desktop.Raven.ToggleAppletView', 'name': 'Toggle Raven - AppletView'}
-{'command': 'dbus-send --session --type="method_call" --dest=org.budgie_desktop.Panel /org/budgie_desktop/Raven org.budgie_desktop.Raven.ToggleNotificationsView', 'name': 'Toggle Raven - Notifications'}
+{"command": "@bdesktopdir@/showdesktop/showdesktop", "name": "Show Desktop"}
+{"command": "@libdir@/budgie-previews/previews_triggers hotcorners_all", "name": "Exposé all windows", "gsettingsboolean": "org.ubuntubudgie.budgie-wpreviews enable-previews"}
+{"command": "@libdir@/budgie-previews/previews_triggers hotcorners_current", "name": "Exposé current application", "gsettingsboolean": "org.ubuntubudgie.budgie-wpreviews enable-previews"}
+{"command": "budgie-desktop-settings", "name":"Budgie Desktop Settings"}
+{"command": "dbus-send --session --type=\"method_call\" --dest=org.gnome.ScreenSaver /org/gnome/ScreenSaver org.gnome.ScreenSaver.Lock", "name": "Lock screen"}
+{"command": "@libdir@/budgie-window-shuffler/togglegui", "name": "Toggle Shuffler GUI", "gsettingsboolean": "org.ubuntubudgie.windowshuffler runshufflergui"}
+{"command": "@libdir@/budgie-window-shuffler/grid_all", "name": "All windows in grid"}
+{"command": "@libdir@/budgie-window-shuffler/tile_active 0 0 2 2", "name": "Window to topleft"}
+{"command": "@libdir@/budgie-window-shuffler/tile_active 1 0 2 2", "name": "Window to topright"}
+{"command": "@libdir@/budgie-window-shuffler/tile_active 1 1 2 2", "name": "Window to bottomright"}
+{"command": "@libdir@/budgie-window-shuffler/tile_active 0 1 2 2", "name": "Window to bottomleft"}
+{"command": "dbus-send --session --type=\"method_call\" --dest=org.budgie_desktop.Panel /org/budgie_desktop/Raven org.budgie_desktop.Raven.ToggleAppletView", "name": "Toggle Raven - AppletView"}
+{"command": "dbus-send --session --type=\"method_call\" --dest=org.budgie_desktop.Panel /org/budgie_desktop/Raven org.budgie_desktop.Raven.ToggleNotificationsView", "name": "Toggle Raven - Notifications"}
