diff --git a/budgie-hotcorners/applet/src/HotCornersApplet.vala b/budgie-hotcorners/applet/src/HotCornersApplet.vala
index a8e483e..ad61657 100755
--- a/budgie-hotcorners/applet/src/HotCornersApplet.vala
+++ b/budgie-hotcorners/applet/src/HotCornersApplet.vala
@@ -69,10 +69,7 @@ namespace HotCornersApplet {
 
             Button settingsbutton = new Gtk.Button();
             settingsbutton.clicked.connect(()=> {
-                run_command(
-                    Config.PACKAGE_LIBEXECDIR +
-                    "/budgie-hotcorners/budgie-hotcorners-settingswindow"
-                );
+                run_command("/usr/lib64/budgie-extras/budgie-hotcorners/budgie-hotcorners-settingswindow");
                 if (popover != null) {
                     popover.hide();
                 }
-- 
2.40.1

