From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Evan Maddock <maddock.evan@vivaldi.net>
Date: Fri, 31 Oct 2025 17:56:43 -0400
Subject: [PATCH] Replace gnome screensaver with budgie screensaver

Signed-off-by: Evan Maddock <maddock.evan@vivaldi.net>
---
 budgie-takeabreak/takeabreak_run | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/budgie-takeabreak/takeabreak_run b/budgie-takeabreak/takeabreak_run
index 694c763..d434af4 100755
--- a/budgie-takeabreak/takeabreak_run
+++ b/budgie-takeabreak/takeabreak_run
@@ -173,13 +173,13 @@ def take_a_break():
                 else:
                     subprocess_tasks(
                         "run_async",
-                        ["gnome-screensaver-command", "-l"])
+                        ["budgie-screensaver-command", "-l"])
     time.sleep(sleeptime)
     for scr in screens:
         if auto_unlock:
             subprocess_tasks(
                 "run_async",
-                ["gnome-screensaver-command", "-d"]
+                ["budgie-screensaver-command", "-d"]
             )
         time.sleep(2)
         if mode == "rotate":
@@ -213,7 +213,7 @@ if from_unidle:
 
 
 def lockscreen_check():
-    lockproc = "gnome-screensaver-dialog"
+    lockproc = "budgie-screensaver-dialog"
     try:
         return lockproc in (p.name() for p in psutil.process_iter())
     except psutil.NoSuchProcess:
