diff --git a/budgie-window-shuffler/applet/src/ShufflerApplet.vala b/budgie-window-shuffler/applet/src/ShufflerApplet.vala
index 84c7e36..91ce258 100644
--- a/budgie-window-shuffler/applet/src/ShufflerApplet.vala
+++ b/budgie-window-shuffler/applet/src/ShufflerApplet.vala
@@ -68,7 +68,7 @@ namespace ShufflerApplet {
     }
 
     private bool procruns (string processname) {
-        string cmd = Config.PACKAGE_BINDIR + @"/pgrep -f $processname";
+        string cmd = @"pgrep -f $processname";
         string output;
         try {
             GLib.Process.spawn_command_line_sync(cmd, out output);
diff --git a/budgie-window-shuffler/src/jump.vala b/budgie-window-shuffler/src/jump.vala
index f68e4f0..342d033 100644
--- a/budgie-window-shuffler/src/jump.vala
+++ b/budgie-window-shuffler/src/jump.vala
@@ -46,7 +46,7 @@ namespace JumpActive {
     }
 
     private bool procruns (string processname) {
-        string cmd = Config.PACKAGE_BINDIR + @"/pgrep -f $processname";
+        string cmd = @"pgrep -f $processname";
         string output;
         try {
             GLib.Process.spawn_command_line_sync(cmd, out output);
@@ -228,4 +228,4 @@ namespace JumpActive {
             stderr.printf ("%s\n", e.message);
         }
     }
-}
\ No newline at end of file
+}
diff --git a/budgie-wpreviews/src/previews_controls.vala b/budgie-wpreviews/src/previews_controls.vala
index 3166d29..9a1050f 100644
--- a/budgie-wpreviews/src/previews_controls.vala
+++ b/budgie-wpreviews/src/previews_controls.vala
@@ -115,7 +115,7 @@ namespace PreviewsControls {
         }
 
         private bool procruns (string processname) {
-            string cmd = Config.PACKAGE_BINDIR + @"/pgrep -f $processname";
+            string cmd = @"pgrep -f $processname";
             string output;
             try {
                 GLib.Process.spawn_command_line_sync(cmd, out output);
@@ -163,4 +163,4 @@ namespace PreviewsControls {
         controls.show_all();
         Gtk.main();
     }
-}
\ No newline at end of file
+}
-- 
2.40.1

