diff --git a/budgie-hotcorners/src/HotCornersApplet.vala b/budgie-hotcorners/src/HotCornersApplet.vala
index 676cd82..e54313a 100644
--- a/budgie-hotcorners/src/HotCornersApplet.vala
+++ b/budgie-hotcorners/src/HotCornersApplet.vala
@@ -35,7 +35,7 @@ namespace HCSupport {
     }
 
     private bool locked () {
-        string cmd = Config.PACKAGE_BINDIR + "/pgrep -f budgie-screensaver-dialog";
+        string cmd = "pgrep -f budgie-screensaver-dialog";
         string output;
         try {
             GLib.Process.spawn_command_line_sync(cmd, out output);
@@ -426,7 +426,7 @@ namespace HotCornersApplet {
     }
 
     private bool procruns (string processname) {
-        string cmd = Config.PACKAGE_BINDIR + @"/pgrep -f $processname";
+        string cmd = @"pgrep -f $processname";
         string output;
         try {
             GLib.Process.spawn_command_line_sync(cmd, out output);
diff --git a/budgie-weathershow/src/weathershow/WeatherShow.vala b/budgie-weathershow/src/weathershow/WeatherShow.vala
index aa14d47..0caecaa 100644
--- a/budgie-weathershow/src/weathershow/WeatherShow.vala
+++ b/budgie-weathershow/src/weathershow/WeatherShow.vala
@@ -77,7 +77,7 @@ namespace WeatherShowFunctions {
     }
 
     private bool check_onwindow(string path) {
-        string cmd_check = Config.PACKAGE_BINDIR + "/pgrep -f " + path;
+        string cmd_check = "pgrep -f " + path;
         string output;
         try {
             GLib.Process.spawn_command_line_sync(cmd_check, out output);
diff --git a/budgie-window-shuffler/applet/src/ShufflerApplet.vala b/budgie-window-shuffler/applet/src/ShufflerApplet.vala
index ea80aa6..2f303d7 100644
--- a/budgie-window-shuffler/applet/src/ShufflerApplet.vala
+++ b/budgie-window-shuffler/applet/src/ShufflerApplet.vala
@@ -68,7 +68,7 @@ namespace ShufflerApplet {
     }
 
     private bool procruns (string processname) {
-        string cmd = Config.PACKAGE_BINDIR + @"/pgrep -f $processname";
+        string cmd = @"pgrep -f $processname";
         string output;
         try {
             GLib.Process.spawn_command_line_sync(cmd, out output);
diff --git a/budgie-window-shuffler/src/jump.vala b/budgie-window-shuffler/src/jump.vala
index 29e1a82..73b6350 100644
--- a/budgie-window-shuffler/src/jump.vala
+++ b/budgie-window-shuffler/src/jump.vala
@@ -46,7 +46,7 @@ namespace JumpActive {
     }
 
     private bool procruns (string processname) {
-        string cmd = Config.PACKAGE_BINDIR + @"/pgrep -f $processname";
+        string cmd = @"pgrep -f $processname";
         string output;
         try {
             GLib.Process.spawn_command_line_sync(cmd, out output);
diff --git a/budgie-wpreviews/src/previews_controls.vala b/budgie-wpreviews/src/previews_controls.vala
index a315e89..245ee7d 100644
--- a/budgie-wpreviews/src/previews_controls.vala
+++ b/budgie-wpreviews/src/previews_controls.vala
@@ -115,7 +115,7 @@ namespace PreviewsControls {
         }
 
         private bool procruns (string processname) {
-            string cmd = Config.PACKAGE_BINDIR + @"/pgrep -f $processname";
+            string cmd = @"pgrep -f $processname";
             string output;
             try {
                 GLib.Process.spawn_command_line_sync(cmd, out output);
-- 
2.35.3

