From 447bb4e57ebc3d59c211b659032cb1875231e520 Mon Sep 17 00:00:00 2001
From: Volker Krause <vkrause@kde.org>
Date: Wed, 15 Apr 2026 18:02:20 +0200
Subject: [PATCH] Add forward compatibility for KMime from KDE Frameworks

---
 CMakeLists.txt     | 6 +++++-
 src/CMakeLists.txt | 2 +-
 2 files changed, 6 insertions(+), 2 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 17a491a9..f82bff65 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -85,7 +85,11 @@ ecm_set_disabled_deprecation_versions(QT 6.9.0
     KF 6.19.0
 )
 
-find_package(KPim6Mime ${KPIM_MIME_VERSION} CONFIG REQUIRED)
+find_package(KF6Mime ${KF_MIN_VERSION} CONFIG)
+if(NOT TARGET KF6::Mime)
+    find_package(KPim6Mime ${KPIM_MIME_VERSION} CONFIG REQUIRED)
+    add_library(KF6::Mime ALIAS KPim6::Mime)
+endif()
 
 find_package(md4c)
 set_package_properties(md4c PROPERTIES
diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
index 3d0e7b7c..1d2cca0c 100644
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -98,7 +98,7 @@ target_link_libraries(marknote_static
         KF6::CoreAddons
         KF6::IconThemes
         KF6::ColorScheme
-        KPim6::Mime
+        KF6::Mime
         KirigamiApp
         KirigamiAddonsStatefulApp
         settingsplugin
-- 
GitLab

