From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Reilly Brogan <reilly@reillybrogan.com>
Date: Mon, 10 Jun 2024 15:13:21 -0500
Subject: [PATCH] solus: Check stateless dir if XDG_CONFIG_DIRS isn't set

Signed-off-by: Reilly Brogan <reilly@reillybrogan.com>
---
 surfraw.IN | 14 +++++++++-----
 1 file changed, 9 insertions(+), 5 deletions(-)

diff --git a/surfraw.IN b/surfraw.IN
index 01130c0..1a7b5a7 100755
--- a/surfraw.IN
+++ b/surfraw.IN
@@ -36,10 +36,14 @@ find_global_conf () {
 		break
 	    fi
 	done
-    elif test -r "@sysconfdir@/xdg/surfraw/$base"; then # XDG default
-	conf="@sysconfdir@/xdg/surfraw/$base"
-    elif test -r "@sysconfdir@/surfraw.$base"; then # old surfraw default
-	conf="@sysconfdir@/surfraw.$base"
+    elif test -r "@syconfdir@/xdg/surfraw/$base"; then # XDG default
+	conf="@syconfdir@/xdg/surfraw/$base"
+    elif test -r "@syconfdir@/surfraw.$base"; then # old surfraw default
+	conf="@syconfdir@/surfraw.$base"
+    elif test -r "/usr/share/xdg/surfraw/$base"; then # XDG default
+	conf="/usr/share/xdg/surfraw/$base"
+    elif test -r "/usr/share/surfraw.$base"; then # old surfraw default
+	conf="/usr/share/surfraw.$base"
     fi
     echo "$conf"
 }
@@ -52,7 +56,7 @@ load_global_conf () {
 	conf=$(find_global_conf conf)
     fi
     if test -z "$conf" || ! test -r "$conf"; then
-	err "couldn't find global config in @sysconfdir@/xdg/surfraw/conf or \$XDG_CONFIG_DIRS"
+	err "couldn't find global config in /usr/share/xdg/surfraw/conf or \$XDG_CONFIG_DIRS"
     fi
     . "$conf"
 }
