From 18b3335fb70e8e8db6c27d1e158daaaeb62bee32 Mon Sep 17 00:00:00 2001
From: Ikey Doherty <ikey@solus-project.com>
Date: Sun, 20 Mar 2016 00:08:50 +0000
Subject: [PATCH] Use absolute paths for font directory and do not make
 assumptions

Signed-off-by: Ikey Doherty <ikey@solus-project.com>
---
 Makefile.am  | 2 +-
 configure.ac | 4 ++--
 2 files changed, 3 insertions(+), 3 deletions(-)

diff --git a/Makefile.am b/Makefile.am
index 4720fdf..a9b8a12 100644
--- a/Makefile.am
+++ b/Makefile.am
@@ -333,7 +333,7 @@ EXTRA_DIST += \
 pkgfilterdir = $(CUPS_SERVERBIN)/filter
 pkgfilter_PROGRAMS = pdftoopvp
 
-pkgfontconfigdir = $(sysconfdir)/$(FONTDIR)
+pkgfontconfigdir = $(FONTDIR)
 pkgfontconfig_DATA = filter/pdftoopvp/99pdftoopvp.conf
 
 pdftoopvp_SOURCES = \
diff --git a/configure.ac b/configure.ac
index 3cef542..70aeffa 100644
--- a/configure.ac
+++ b/configure.ac
@@ -65,7 +65,7 @@ PKG_PROG_PKG_CONFIG([0.20])
 AC_ARG_WITH([fontdir],
 	[AS_HELP_STRING([--with-fontdir=path], [Specify path to font config directory (default: fonts/conf.d/).])],
 	[FONTDIR="$withval"],
-	[FONTDIR="fonts/conf.d"]
+	[FONTDIR="${sysconfdir}/fonts/conf.d"]
 )
 AC_SUBST(FONTDIR)
 
@@ -716,7 +716,7 @@ Environment settings:
 	LDFLAGS:         ${LDFLAGS}
 Build configuration:
 	cups-config:     ${with_cups_config}
-	font directory:  ${sysconfdir}/${FONTDIR}
+	font directory:  ${FONTDIR}
 	init directory:  ${INITDDIR}
 	cups dom socket: ${CUPS_DEFAULT_DOMAINSOCKET}
 	gs-path:         ${with_gs_path}
-- 
2.7.3

