From 018283c2ce27169f2930413e7ad4c784ff5c0ae8 Mon Sep 17 00:00:00 2001
From: Pierre-Yves <pyu@riseup.net>
Date: Fri, 5 Jul 2019 23:02:13 +0200
Subject: [PATCH] Use absolute paths for font directory and do not make
 assumptions

Signed-off-by: Pierre-Yves <pyu@riseup.net>
---
 configure.ac | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/configure.ac b/configure.ac
index 3973db6..d930655 100644
--- a/configure.ac
+++ b/configure.ac
@@ -73,7 +73,7 @@ AM_GNU_GETTEXT([external])
 AC_ARG_WITH([fontdir],
 	[AS_HELP_STRING([--with-fontdir=path], [Specify path to font config directory (default: fonts/conf.d/).])],
 	[FONTDIR="$withval"],
-	[FONTDIR="fonts/conf.d"]
+	[FONTDIR="${sysconfdir}/fonts/conf.d"]
 )
 AC_SUBST(FONTDIR)
 
@@ -919,7 +919,7 @@ Environment settings:
 	LDFLAGS:                                   ${LDFLAGS}
 Build configuration:
 	cups-config:     ${with_cups_config}
-	font directory:  ${sysconfdir}/${FONTDIR}
+	font directory:  ${FONTDIR}
 	foomatic:        ${enable_foomatic}
 	init directory:  ${INITDDIR}
 	cups dom socket: ${CUPS_DEFAULT_DOMAINSOCKET}
-- 
2.34.0

