From 37b69c3652f049d469ede15e8fcd358085818931 Mon Sep 17 00:00:00 2001
From: Ikey Doherty <ikey@solus-project.com>
Date: Sun, 24 Apr 2016 10:12:34 +0100
Subject: [PATCH] Use absolute paths for font directory and do not make
 assumptions

Signed-off-by: Ikey Doherty <ikey@solus-project.com>
---
 Makefile.am  | 2 +-
 configure.ac | 4 ++--
 2 files changed, 3 insertions(+), 3 deletions(-)

diff --git a/Makefile.am b/Makefile.am
index 0fd6922..fca3259 100644
--- a/Makefile.am
+++ b/Makefile.am
@@ -388,7 +388,7 @@ EXTRA_DIST += \
 pkgfilterdir = $(CUPS_SERVERBIN)/filter
 pkgfilter_PROGRAMS = pdftoopvp
 
-pkgfontconfigdir = $(sysconfdir)/$(FONTDIR)
+pkgfontconfigdir = $(FONTDIR)
 pkgfontconfig_DATA = filter/pdftoopvp/99pdftoopvp.conf
 
 pdftoopvp_SOURCES = \
diff --git a/configure.ac b/configure.ac
index bbbedff..c8470fb 100644
--- a/configure.ac
+++ b/configure.ac
@@ -66,7 +66,7 @@ PKG_PROG_PKG_CONFIG([0.20])
 AC_ARG_WITH([fontdir],
 	[AS_HELP_STRING([--with-fontdir=path], [Specify path to font config directory (default: fonts/conf.d/).])],
 	[FONTDIR="$withval"],
-	[FONTDIR="fonts/conf.d"]
+	[FONTDIR="${sysconfdir}/fonts/conf.d"]
 )
 AC_SUBST(FONTDIR)
 
@@ -755,7 +755,7 @@ Environment settings:
 	LDFLAGS:         ${LDFLAGS}
 Build configuration:
 	cups-config:     ${with_cups_config}
-	font directory:  ${sysconfdir}/${FONTDIR}
+	font directory:  ${FONTDIR}
 	foomatic:        ${enable_foomatic}
 	init directory:  ${INITDDIR}
 	cups dom socket: ${CUPS_DEFAULT_DOMAINSOCKET}
-- 
2.8.1

