From d764ad3d84ca5fa093d26d20b82c913ece68bf56 Mon Sep 17 00:00:00 2001
From: Ikey Doherty <ikey@solus-project.com>
Date: Tue, 10 Oct 2017 01:29:12 +0100
Subject: [PATCH 2/2] utils: Fix broken systemd unit

Signed-off-by: Ikey Doherty <ikey@solus-project.com>
---
 utils/cups-browsed.service | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/utils/cups-browsed.service b/utils/cups-browsed.service
index 3bf0656..d445a30 100644
--- a/utils/cups-browsed.service
+++ b/utils/cups-browsed.service
@@ -1,7 +1,7 @@
 [Unit]
 Description=Make remote CUPS printers available locally
-Requires=cups.service
-After=cups.service avahi-daemon.service network-online.target
+Requires=org.cups.cupsd.service
+After=org.cups.cupsd.service avahi-daemon.service
 Wants=avahi-daemon.service network-online.target
 
 [Service]
-- 
2.34.0
