From 65105dd5707e84178c14bb1cb26e344b48bfb6c8 Mon Sep 17 00:00:00 2001
From: Ikey Doherty <ikey.doherty@gmail.com>
Date: Wed, 20 Aug 2014 13:47:36 +0100
Subject: [PATCH] Do not hide System Settings category from menu listings

Signed-off-by: Ikey Doherty <ikey.doherty@gmail.com>
---
 desktop-directories/X-GNOME-SystemSettings.directory.in | 1 -
 1 file changed, 1 deletion(-)

diff --git a/desktop-directories/X-GNOME-SystemSettings.directory.in b/desktop-directories/X-GNOME-SystemSettings.directory.in
index acfbffe..fe0151b 100644
--- a/desktop-directories/X-GNOME-SystemSettings.directory.in
+++ b/desktop-directories/X-GNOME-SystemSettings.directory.in
@@ -2,4 +2,3 @@
 Name=System Settings
 Icon=gnome-settings
 Type=Directory
-NoDisplay=true
-- 
1.8.4

