From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Muhammad Alfi Syahrin <malfisya.dev@hotmail.com>
Date: Sat, 1 Feb 2025 13:06:17 +0700
Subject: [PATCH] Don't override icons at all, leave that to the desktop

---
 src/Win_GParted.cc | 9 ---------
 1 file changed, 9 deletions(-)

diff --git a/src/Win_GParted.cc b/src/Win_GParted.cc
index 2c962f0..4d5967e 100644
--- a/src/Win_GParted.cc
+++ b/src/Win_GParted.cc
@@ -88,15 +88,6 @@ Win_GParted::Win_GParted( const std::vector<Glib::ustring> & user_devices )
 	//==== GUI =========================
 	this ->set_title( _("GParted") );
 	this ->set_default_size( 775, 500 );
-	
-	try
-	{
-		this ->set_default_icon_name( "gparted" ) ;
-	}
-	catch ( Glib::Exception & e )
-	{
-		std::cerr << Utils::convert_ustring(e.what()) << std::endl;
-	}
 
 	// Pack the main box
 	vbox_main.set_orientation(Gtk::ORIENTATION_VERTICAL);
-- 
2.47.2

