From f05d1b8d5dd2d490b099ad07fa72bae6d7a470de Mon Sep 17 00:00:00 2001
From: Ikey Doherty <ikey@solus-project.com>
Date: Tue, 21 Apr 2015 23:13:10 +0100
Subject: [PATCH] Support pkexec

Signed-off-by: Ikey Doherty <ikey@solus-project.com>
---
 configure.ac | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/configure.ac b/configure.ac
index b5569a2..25a8249 100644
--- a/configure.ac
+++ b/configure.ac
@@ -20,7 +20,7 @@ AM_PROG_LIBTOOL
 dnl======================
 dnl checks for other programs
 dnl======================
-AC_CHECK_PROGS([GKSUPROG], [gksudo gksu kdesudo], [])
+AC_CHECK_PROGS([GKSUPROG], [gksudo gksu kdesudo pkexec], [])
 
 
 dnl======================
-- 
2.3.2

