From 66f8ecc4399ac53d2d74ed83b955bb478653ede2 Mon Sep 17 00:00:00 2001
From: Ikey Doherty <ikey@solus-project.com>
Date: Sat, 31 Oct 2015 17:37:55 +0000
Subject: [PATCH 1/2] Support pkexec

Signed-off-by: Ikey Doherty <ikey@solus-project.com>
---
 gparted.desktop.in.in | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/gparted.desktop.in.in b/gparted.desktop.in.in
index 4b41fe4..87bcd0f 100644
--- a/gparted.desktop.in.in
+++ b/gparted.desktop.in.in
@@ -3,7 +3,7 @@ _Name=GParted
 _GenericName=Partition Editor
 _X-GNOME-FullName=GParted Partition Editor
 _Comment=Create, reorganize, and delete partitions
-Exec=@gksuprog@ @installdir@/gparted %f
+Exec=@installdir@/gparted_wrapper %f
 Icon=gparted
 Terminal=false
 Type=Application
-- 
2.8.3

