From c285d19d44ccb721cb8c64576aa98b7198416bfd Mon Sep 17 00:00:00 2001
From: Ikey Doherty <ikey@solus-project.com>
Date: Sat, 31 Oct 2015 15:58:02 +0000
Subject: [PATCH] Support pkexec

Signed-off-by: Ikey Doherty <ikey@solus-project.com>
---
 configure.ac | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/configure.ac b/configure.ac
index 8a3b280..d24582b 100644
--- a/configure.ac
+++ b/configure.ac
@@ -23,7 +23,7 @@ PKG_CHECK_EXISTS
 dnl======================
 dnl checks for other programs
 dnl======================
-AC_CHECK_PROGS([GKSUPROG], [gksudo gksu kdesudo "xdg-su -c"], [])
+AC_CHECK_PROGS([GKSUPROG], [gksudo gksu kdesudo pkexec "xdg-su -c"], [])
 
 
 dnl======================
-- 
2.6.2

