From 68f6ad776ee5d25b7d0dbbfeb1b993e776a80d09 Mon Sep 17 00:00:00 2001
From: Pierre-Yves <pyu@riseup.net>
Date: Sun, 23 Jun 2019 23:07:55 +0200
Subject: [PATCH] Don't override icons at all, leave that to the desktop

Signed-off-by: Pierre-Yves <pyu@riseup.net>
---
 src/Win_GParted.cc | 9 ---------
 1 file changed, 9 deletions(-)

diff --git a/src/Win_GParted.cc b/src/Win_GParted.cc
index b5288e8..42b0fff 100644
--- a/src/Win_GParted.cc
+++ b/src/Win_GParted.cc
@@ -85,15 +85,6 @@ Win_GParted::Win_GParted( const std::vector<Glib::ustring> & user_devices )
 	//==== GUI =========================
 	this ->set_title( _("GParted") );
 	this ->set_default_size( 775, 500 );
-	
-	try
-	{
-		this ->set_default_icon_name( "gparted" ) ;
-	}
-	catch ( Glib::Exception & e )
-	{
-		std::cout << e .what() << std::endl ;
-	}
 
 	// Pack the main box
 	vbox_main.set_orientation(Gtk::ORIENTATION_VERTICAL);
-- 
2.22.0

