From 3846d8bd207d6eccf3f5df9e301f99eeb84770a9 Mon Sep 17 00:00:00 2001
From: Pierre-Yves <pyu@riseup.net>
Date: Thu, 23 Aug 2018 17:59:38 +0200
Subject: [PATCH 2/2] Don't override icons at all, leave that to the desktop

Signed-off-by: Pierre-Yves <pyu@riseup.net>
---
 src/Win_GParted.cc | 9 ---------
 1 file changed, 9 deletions(-)

diff --git a/src/Win_GParted.cc b/src/Win_GParted.cc
index dec3ef9..663ac92 100644
--- a/src/Win_GParted.cc
+++ b/src/Win_GParted.cc
@@ -87,15 +87,6 @@ Win_GParted::Win_GParted( const std::vector<Glib::ustring> & user_devices )
 	this ->set_title( _("GParted") );
 	this ->set_default_size( 775, 500 );
 	
-	try
-	{
-		this ->set_default_icon_name( "gparted" ) ;
-	}
-	catch ( Glib::Exception & e )
-	{
-		std::cout << e .what() << std::endl ;
-	}
-	
 	//Pack the main box
 	this ->add( vbox_main ); 
 	
-- 
2.18.0

