From d76bec7cf39561942fba458ec34b13ee7de0642e Mon Sep 17 00:00:00 2001
From: Ikey Doherty <ikey@solus-project.com>
Date: Sat, 11 Jun 2016 22:10:40 +0100
Subject: [PATCH 2/2] Don't override icons at all, leave that to the desktop

Signed-off-by: Ikey Doherty <ikey@solus-project.com>
---
 src/Win_GParted.cc | 13 -------------
 1 file changed, 13 deletions(-)

diff --git a/src/Win_GParted.cc b/src/Win_GParted.cc
index 63162b3..ebcc046 100644
--- a/src/Win_GParted.cc
+++ b/src/Win_GParted.cc
@@ -77,19 +77,6 @@ Win_GParted::Win_GParted( const std::vector<Glib::ustring> & user_devices )
 	this ->set_title( _("GParted") );
 	this ->set_default_size( 775, 500 );
 	
-	try
-	{
-#ifdef HAVE_SET_DEFAULT_ICON_NAME
-		this ->set_default_icon_name( "gparted" ) ;
-#else
-		this ->set_icon_from_file( GNOME_ICONDIR "/gparted.png" ) ;
-#endif
-	}
-	catch ( Glib::Exception & e )
-	{
-		std::cout << e .what() << std::endl ;
-	}
-	
 	//Pack the main box
 	this ->add( vbox_main ); 
 	
-- 
2.8.3

