From c5d1f4774bd9a6b76d7d88323deaf6861fb876d5 Mon Sep 17 00:00:00 2001
From: Fabio Forni <livingsilver94.solus@redaril.me>
Date: Fri, 1 May 2020 15:18:34 +0200
Subject: [PATCH] Run with openjdk-8

---
 build/linux/dist/arduino | 7 +------
 1 file changed, 1 insertion(+), 6 deletions(-)

diff --git a/build/linux/dist/arduino b/build/linux/dist/arduino
index f70c650..805589e 100755
--- a/build/linux/dist/arduino
+++ b/build/linux/dist/arduino
@@ -19,11 +19,6 @@ export PATH="${APPDIR}/java/bin:${PATH}"
 
 export JAVA_TOOL_OPTIONS=`echo $JAVA_TOOL_OPTIONS | sed 's|-javaagent:/usr/share/java/jayatanaag.jar||g'`
 
-JAVA=java
-if [ -x "$APPDIR/java/bin/java" ]; then
-  JAVA=$APPDIR/java/bin/java
-fi
-
 # Collect options to java in an array, to properly handle whitespace in options
 JAVA_OPTIONS=("-DAPP_DIR=$APPDIR")
 
@@ -32,5 +27,5 @@ if [[ "$@" != *"--"* ]] ; then
 	JAVA_OPTIONS+=("-splash:$APPDIR/lib/splash.png")
 fi
 
-"$JAVA" "${JAVA_OPTIONS[@]}" processing.app.Base "$@"
+/usr/lib64/openjdk-8/bin/java "${JAVA_OPTIONS[@]}" processing.app.Base "$@"
 
-- 
2.26.2

