From 957d2e1841d1afbfb42900393abbb7659b83a0fa Mon Sep 17 00:00:00 2001
From: livingsilver94 <fabio0x@keemail.me>
Date: Mon, 18 Jun 2018 22:11:38 +0200
Subject: [PATCH 2/2] Force Ubuntu version

---
 cibuild.sh    | 3 +++
 init-tools.sh | 3 +++
 2 files changed, 6 insertions(+)

diff --git a/cibuild.sh b/cibuild.sh
index 18705c1..e3db393 100755
--- a/cibuild.sh
+++ b/cibuild.sh
@@ -127,6 +127,9 @@ get_current_linux_name() {
             echo "opensuse.42.1"
             return 0
         fi
+    elif [ "$(cat /etc/*-release | grep -cim1 solus)" -eq 1 ]; then
+        echo "ubuntu.16.04"
+        return 0
     fi
 
     # Cannot determine Linux distribution, assuming Ubuntu 14.04.
diff --git a/init-tools.sh b/init-tools.sh
index 3dea1da..3dd0fc1 100755
--- a/init-tools.sh
+++ b/init-tools.sh
@@ -58,6 +58,9 @@ get_current_linux_name() {
             echo "opensuse.42.1"
             return 0
         fi
+    elif [ "$(cat /etc/*-release | grep -cim1 solus)" -eq 1 ]; then
+        echo "ubuntu.16.04"
+        return 0
     fi
 
     # Cannot determine Linux distribution, assuming Ubuntu 14.04.
-- 
2.17.1

