diff --git a/mono/create_bootstrap.sh b/mono/create_bootstrap.sh
old mode 100644
new mode 100755
index 5ae5344e..2f4fba9a
--- a/mono/create_bootstrap.sh
+++ b/mono/create_bootstrap.sh
@@ -1,21 +1,9 @@
-#!/bin/sh
-
-# This creates a bootstrap from an exising mono installation
-# This is just to ensure that we have the correct "matched" Roslyn
-
+#!/bin/bash
 TMP_DIR=`mktemp -d`
 MSBUILD_DST_DIR=${TMP_DIR}/msbuild
 
-MONO_PREFIX_SRC_DIR=`dirname $(which mono)`
-MONO_PREFIX_SRC_DIR="${MONO_PREFIX_SRC_DIR}/../"
+MONO_PREFIX_SRC_DIR=/usr
 MSBUILD_BIN_SRC_DIR=$MONO_PREFIX_SRC_DIR/lib/mono/msbuild/15.0/bin
-
-ID=`msbuild -version | head -n 1 | awk -F\( '{print $3}' | cut -d \  -f 1 | sed -e 's,\/,-,'`
-ZIP_PATH=${PWD}/mono_msbuild_${ID}.zip
-
-echo "Building bootstrap from $MONO_PREFIX_SRC_DIR"
-
-# -L so that we get the linked files in Roslyn/
 cp -L -R $MSBUILD_BIN_SRC_DIR $MSBUILD_DST_DIR
 
 # don't fallback to dotnet, everything is self contained here
@@ -23,7 +11,7 @@ rm -Rf $MSBUILD_DST_DIR/SdkResolvers/Microsoft.DotNet.MSBuildSdkResolver
 
 mkdir $MSBUILD_DST_DIR/Extensions
 
-cp -R $MONO_PREFIX_SRC_DIR/lib/mono/xbuild/ $MSBUILD_DST_DIR/Extensions
+cp -r $MONO_PREFIX_SRC_DIR/lib/mono/xbuild/* $MSBUILD_DST_DIR/Extensions
 rm -Rf $MSBUILD_DST_DIR/Extensions/1[24].0
 
 # adjust System.Reflection.Metadata.dll to be a link, so we use exactly
@@ -40,6 +28,6 @@ MONO_GC_PARAMS="nursery-size=64m,$MONO_GC_PARAMS" exec mono $MONO_OPTIONS $THIS_
 EOL
 
 chmod +x $MSBUILD_DST_DIR/msbuild
-
-(cd $TMP_DIR; zip -r ${ZIP_PATH} msbuild)
-rm -Rf $TMP_DIR
+mkdir -p artifacts
+mv $MSBUILD_DST_DIR artifacts/mono-msbuild
+chmod +x artifacts/mono-msbuild/MSBuild.dll
