From 554184c8230ce723ff4734770e86399c44f524c5 Mon Sep 17 00:00:00 2001
From: livingsilver94 <fabio0x@keemail.me>
Date: Thu, 21 Jun 2018 22:53:41 +0200
Subject: [PATCH 2/2] Force verbosity to diagnostic

---
 cibuild.sh | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/cibuild.sh b/cibuild.sh
index 18705c1..514f4ca 100755
--- a/cibuild.sh
+++ b/cibuild.sh
@@ -314,7 +314,7 @@ LOCAL_BUILD_LOG_PATH="$THIS_SCRIPT_PATH"/msbuild_local_build-"$host".log
 LOCAL_BUILD_BINLOG_PATH="$THIS_SCRIPT_PATH"/msbuild_rebuild-"$host".binlog
 MOVE_LOG_PATH="$THIS_SCRIPT_PATH"/msbuild_move_bootstrap-"$host".log
 
-BUILD_MSBUILD_ARGS="$PROJECT_FILE_ARG /p:OS=$OS_ARG /p:Configuration=$CONFIGURATION /p:OverrideToolHost=$RUNTIME_HOST /verbosity:minimal $EXTRA_ARGS"
+BUILD_MSBUILD_ARGS="$PROJECT_FILE_ARG /p:OS=$OS_ARG /p:Configuration=$CONFIGURATION /p:OverrideToolHost=$RUNTIME_HOST /verbosity:diagnostic $EXTRA_ARGS"
 
 setHome
 
@@ -334,7 +334,7 @@ fi
 
 echo
 echo "** Moving bootstrapped MSBuild to the bootstrap folder"
-MOVE_MSBUILD_ARGS="$BOOTSTRAP_FILE_ARG /p:OS=$OS_ARG /p:Configuration=$CONFIGURATION /p:OverrideToolHost=$RUNTIME_HOST /verbosity:minimal"
+MOVE_MSBUILD_ARGS="$BOOTSTRAP_FILE_ARG /p:OS=$OS_ARG /p:Configuration=$CONFIGURATION /p:OverrideToolHost=$RUNTIME_HOST /verbosity:diagnostic"
 runMSBuildWith "$RUNTIME_HOST" "$RUNTIME_HOST_ARGS" "$MSBUILD_EXE" "$MOVE_MSBUILD_ARGS" "$MOVE_LOG_PATH"
 
 if [[ $BOOTSTRAP_ONLY = true ]]; then
-- 
2.17.1

