diff --git a/mono/build/extract_and_copy_hostfxr.sh b/mono/build/extract_and_copy_hostfxr.sh
index d4b70026..45a67e8a 100755
--- a/mono/build/extract_and_copy_hostfxr.sh
+++ b/mono/build/extract_and_copy_hostfxr.sh
@@ -37,15 +37,7 @@ function GetDotNetInstallScript {
   _GetDotNetInstallScript="$install_script"
 }
 
-TMPDIR=`mktemp -d`
-DOTNET_DIR=$TMPDIR/.dotnet
+DOTNET_DIR=REPLACE_ME
 
-OLDCWD=`pwd`
-cd $TMPDIR
 
-GetDotNetInstallScript $TMPDIR
-bash ./dotnet-install.sh --version $1 --install-dir $DOTNET_DIR --architecture x64 --runtime dotnet --skip-non-versioned-files
 find $DOTNET_DIR -name libhostfxr.dylib | xargs -I {} cp -v {} $DESTDIR
-
-cd $OLDCWD
-rm -Rf $TMPDIR
diff --git a/mono/build/get_sdk_files.sh b/mono/build/get_sdk_files.sh
index 57cf505c..748eb520 100755
--- a/mono/build/get_sdk_files.sh
+++ b/mono/build/get_sdk_files.sh
@@ -17,16 +17,7 @@ source $THIS_DIR/../../eng/common/tools.sh
 ReadGlobalVersion "dotnet"
 dotnet_sdk_version=$_ReadGlobalVersion
 
-TMPDIR=`mktemp -d`
-DOTNET_DIR=$TMPDIR/.dotnet-sdk
+DOTNET_DIR=REPLACE_ME
 
-OLDCWD=`pwd`
-cd $TMPDIR
-
-GetDotNetInstallScript $TMPDIR
-bash ./dotnet-install.sh --version $dotnet_sdk_version --install-dir $DOTNET_DIR --architecture x64 --skip-non-versioned-files
 find $DOTNET_DIR -name Microsoft.NETCoreSdk.BundledVersions.props -exec cp -v {} $1 \;
 find $DOTNET_DIR -name RuntimeIdentifierGraph.json -exec cp -v {} $1 \;
-
-cd $OLDCWD
-#rm -Rf $TMPDIR
