From 2e44f7e1a42f2df7fbf150cab11ef738b242fe97 Mon Sep 17 00:00:00 2001
From: Reilly Brogan <reilly@reillybrogan.com>
Date: Sun, 15 Oct 2023 21:01:25 -0500
Subject: [PATCH] Fix compile error

---
 common.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/common.cpp b/common.cpp
index 2447fd2..f3200b6 100644
--- a/common.cpp
+++ b/common.cpp
@@ -202,7 +202,7 @@ int rom_unzip(const char *path, const char *extraction_directory)
 
    char read_buffer[8192];
 
-   uLong i;
+   ulong i;
    for (i = 0; i < global_info.number_entry; ++i)
    {
       unz_file_info file_info;
-- 
2.42.0

