From 024676a8577d4ce37a704aa040b4706e85a0f192 Mon Sep 17 00:00:00 2001
From: "F. von Gellhorn" <flinux@vongellhorn.ch>
Date: Sun, 12 Jan 2020 19:09:21 +0100
Subject: [PATCH 1/1] Fix build

---
 Makefile | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/Makefile b/Makefile
index f10b4d0..17b8a40 100644
--- a/Makefile
+++ b/Makefile
@@ -500,11 +500,11 @@ clean:
 	rm -f *.d */*.d */*/*.d */*/*/*.d */*/*/*/*.d */*/*/*/*/*.d
 
 strip:
-	$(STRIP) $(TARGET_NAME).out
+	$(STRIP) $(TARGET_NAME)
 
 install: strip
 	$(INSTALL) -m 0755 -d $(DESTDIR)$(PREFIX)/$(BINDIR)
-	$(INSTALL) -m 555 $(TARGET_NAME).out $(DESTDIR)$(PREFIX)/$(BINDIR)/$(TARGET_NAME)
+	$(INSTALL) -m 555 $(TARGET_NAME) $(DESTDIR)$(PREFIX)/$(BINDIR)/$(TARGET_NAME)
 	$(INSTALL) -m 0755 -d $(DESTDIR)$(PREFIX)/$(MANDIR)
 	$(INSTALL) -m 644 Assets/$(TARGET_NAME).6 $(DESTDIR)$(PREFIX)/$(MANDIR) 
 
-- 
2.24.1

