From f86ce5580ec065b6a0a82e62096a01d68eeaa83f Mon Sep 17 00:00:00 2001
From: Jacob <Jacalz@users.noreply.github.com>
Date: Sat, 16 Feb 2019 17:14:21 +0100
Subject: [PATCH] Make the desktop file actually work

---
 Assets/mrboom.desktop | 3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

diff --git a/Assets/mrboom.desktop b/Assets/mrboom.desktop
index 8a6067f..8244955 100644
--- a/Assets/mrboom.desktop
+++ b/Assets/mrboom.desktop
@@ -1,8 +1,7 @@
 [Desktop Entry]
 Name=Mr.Boom
 Comment=8 player Bomberman clone
-Exec=/usr/games/mrboom
-TryExec=/usr/games/mrboom
+Exec=mrboom
 Icon=mrboom
 Terminal=false
 Categories=Game;ArcadeGame;ActionGame;