From 9ece9578fabc3bb3ba5ff6f5db4139eae5b0b942 Mon Sep 17 00:00:00 2001
From: Reilly Brogan <Reilly@reillybrogan.com.com>
Date: Sun, 20 Aug 2023 19:23:34 -0500
Subject: [PATCH] configure: Fix missing runstatedir

---
 configure | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/configure b/configure
index 272bb84..c071bc5 100755
--- a/configure
+++ b/configure
@@ -2387,6 +2387,11 @@ do
   | -silent | --silent | --silen | --sile | --sil)
     silent=yes ;;
 
+  -runstatedir | --runstatedir | --runstatedi | --runstated | --runstate)
+    ac_prev=runstatedir ;;
+  -runstatedir=* | --runstatedir=* | --runstatedi=* | --runstated=* | --runstate=*)
+    runstatedir=$ac_optarg ;;
+
   -sbindir | --sbindir | --sbindi | --sbind | --sbin | --sbi | --sb)
     ac_prev=sbindir ;;
   -sbindir=* | --sbindir=* | --sbindi=* | --sbind=* | --sbin=* \
-- 
2.41.0

