From 98895da4d684b8c53a55217e96192d2693d8f55a Mon Sep 17 00:00:00 2001
From: Louie Lu <git@louie.lu>
Date: Wed, 23 Feb 2022 11:11:34 +0800
Subject: [PATCH] Fix custom label not working, #2044

---
 guake/guake_app.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/guake/guake_app.py b/guake/guake_app.py
index f623fa3c..6a9be59b 100644
--- a/guake/guake_app.py
+++ b/guake/guake_app.py
@@ -1441,7 +1441,7 @@ def restore_tabs(self, filename="session.json", suppress_notify=False):
                     for index, tab in enumerate(tabs):
                         if tab.get("panes", False):
                             box, page_num, term = nb.new_page_with_focus(
-                                label=tab["label"], empty=True
+                                label=tab["label"], user_set=tab["custom_label_set"], empty=True
                             )
                             box.restore_box_layout(box.child, tab["panes"])
                         else:
