diff --git a/Makefile b/Makefile
index 38c46c0..cab23d2 100644
--- a/Makefile
+++ b/Makefile
@@ -127,13 +127,13 @@ install-schemas:
 	install -dm755                                       "$(DESTDIR)$(datadir)/applications"
 	install -Dm644 "$(DEV_DATA_DIR)/guake.desktop"       "$(DESTDIR)$(datadir)/applications/"
 	install -Dm644 "$(DEV_DATA_DIR)/guake-prefs.desktop" "$(DESTDIR)$(datadir)/applications/"
-	install -dm755                                       "$(DESTDIR)/metainfo/"
-	install -Dm644 "$(DEV_DATA_DIR)/guake.desktop.metainfo.xml"  "$(DESTDIR)/metainfo/"
+	install -dm755                                       "$(DESTDIR)${datadir}/metainfo/"
+	install -Dm644 "$(DEV_DATA_DIR)/guake.desktop.metainfo.xml"  "$(DESTDIR)/${datadir}/metainfo/"
 	install -dm755                                 "$(DESTDIR)$(IMAGE_DIR)"
 	install -Dm644 "$(DEV_DATA_DIR)"/pixmaps/*.png "$(DESTDIR)$(IMAGE_DIR)/"
 	install -Dm644 "$(DEV_DATA_DIR)"/pixmaps/*.svg "$(DESTDIR)$(IMAGE_DIR)/"
-	install -dm755                                     "$(DESTDIR)$(PREFIX)/share/pixmaps"
-	install -Dm644 "$(DEV_DATA_DIR)/pixmaps/guake.png" "$(DESTDIR)$(PREFIX)/share/pixmaps/"
+	install -dm755                                     "$(DESTDIR)$(datadir)/pixmaps"
+	install -Dm644 "$(DEV_DATA_DIR)/pixmaps/guake.png" "$(DESTDIR)$(datadir)/pixmaps/"
 	install -dm755                                           "$(DESTDIR)$(SHARE_DIR)"
 	install -Dm644 "$(DEV_DATA_DIR)/autostart-guake.desktop" "$(DESTDIR)$(SHARE_DIR)/"
 	install -dm755                           "$(DESTDIR)$(GLADE_DIR)"
@@ -160,7 +160,7 @@ purge-system: uninstall-system reset
 uninstall-schemas:
 	rm -f "$(DESTDIR)$(datadir)/applications/guake.desktop"
 	rm -f "$(DESTDIR)$(datadir)/applications/guake-prefs.desktop"
-	rm -f "$(DESTDIR)/metainfo/guake.desktop.metainfo.xml"
+	rm -f "$(DESTDIR)${datadir}/metainfo/guake.desktop.metainfo.xml"
 	rm -f "$(DESTDIR)$(datadir)/pixmaps/guake.png"
 	rm -fr "$(DESTDIR)$(IMAGE_DIR)"
 	rm -fr "$(DESTDIR)$(SHARE_DIR)"
