From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Reilly Brogan <reilly@reillybrogan.com>
Date: Tue, 7 May 2024 18:54:00 -0500
Subject: [PATCH] Fix use with nuitka

Signed-off-by: Reilly Brogan <reilly@reillybrogan.com>
---
 xattr/lib.py | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/xattr/lib.py b/xattr/lib.py
index b40d46a..44ca32d 100644
--- a/xattr/lib.py
+++ b/xattr/lib.py
@@ -1,7 +1,8 @@
 import os
 import sys
 
-from ._lib import lib, ffi
+import _cffi_backend # noqa: F401  # required by _lib
+from ._lib import lib, ffi # type: ignore
 
 XATTR_NOFOLLOW = lib.XATTR_XATTR_NOFOLLOW
 XATTR_CREATE = lib.XATTR_XATTR_CREATE
