--- a/configure.ac
+++ b/configure.ac
@@ -16,6 +16,10 @@ AX_CHECK_GNU_MAKE([], [AC_MSG_ERROR([mak
 # Set a value even if not found, so that an invocation via build-aux/missing works
 AC_PATH_PROG([HELP2MAN], [help2man], [help2man])
 
+# Configure the Perl module destination dir
+PMDIR=$($PERL -MConfig -e 'print $Config{installvendorlib}')
+AC_SUBST([PMDIR])
+
 # Generate output files
 AC_CONFIG_FILES([
   Makefile
--- a/Makefile.am
+++ b/Makefile.am
@@ -8,7 +8,8 @@ ACLOCAL_AMFLAGS = -I m4
 CLOC = cloc --autoconf
 
 bin_SCRIPTS = pstops psbook psjoin psresize psselect psnup epsffit extractres includeres
-dist_pkgdata_DATA = PSUtils.pm
+pmdir = $(PMDIR)
+dist_pm_DATA = PSUtils.pm
 
 man_MANS = psutils.1 \
 	psbook.1 psselect.1 pstops.1 epsffit.1 psnup.1 psresize.1 \
