--- a/configure.ac	2016-01-15 21:28:52.000000000 +0100
+++ b/configure.ac	2019-03-13 18:44:51.290457304 +0100
@@ -118,22 +118,6 @@
 CFLAGS="$CFLAGS $ARCH"
 
 # KDE dialogs support
-AC_ARG_WITH(kde, AC_HELP_STRING([--with-kde], [Compile with KDE support]), [kde=$withval], [kde=no])
-if test "x$kde" = "xdefault"; then
-  case $host_os in
-    *mingw*)
-      kde=no
-      ;;
-    *)
-      kde=yes
-      ;;
-  esac
-fi
-if test "x$kde" = "xyes"; then
-        AC_PROG_CXX
-        AC_PATH_KDE
-        AC_DEFINE(WITH_KDE, 1, [Use KDE support])
-fi
 AM_CONDITIONAL(USE_KDE, test "x$kde" = "xyes")
 AC_SUBST(kde)
 
--- a/src/Makefile.am	2016-03-14 20:55:57.000000000 +0100
+++ b/src/Makefile.am	2019-03-13 18:45:56.889452864 +0100
@@ -9,16 +9,13 @@
 tilp_CPPFLAGS = -I$(top_srcdir)/intl \
 	@TICABLES_CFLAGS@ @TIFILES_CFLAGS@ @TICALCS_CFLAGS@ @TICONV_CFLAGS@ \
 	@GLIB_CFLAGS@ @GTK_CFLAGS@ \
-	@KDE_INCLUDES@ @QT_INCLUDES@ \
 	-DSHARE_DIR=\"$(pkgdatadir)\" \
 	-DLOCALEDIR=\"$(datadir)/locale\" \
 	-DSYSCONFDIR=\"$(sysconfdir)\" \
 	-DGTK_DISABLE_DEPRECATED
 tilp_LDFLAGS = -export-dynamic
 tilp_LDADD = @TICABLES_LIBS@ @TIFILES_LIBS@ @TICALCS_LIBS@ @TICONV_LIBS@ \
-	@GLIB_LIBS@ @GTK_LIBS@ \
-	@LIB_KDECORE@ @LIB_KDEUI@ @LIB_KIO@ @LIB_QT@ @KDE_LDFLAGS@ \
-	@QT_LDFLAGS@ @X_LDFLAGS@ @LIBZ@
+	@GLIB_LIBS@ @GTK_LIBS@ @LIBZ@
 tilp_SOURCES = *.h \
 	tilp_calcs.c tilp_cmdline.c tilp_config.c tilp_error.c \
 	tilp_files.c tilp_gif.c tilp_main.c \