From 8f43885b05c8c398225c1b702570391f67dd65f9 Mon Sep 17 00:00:00 2001
From: Condor <condor@stz-bg.com>
Date: Thu, 5 Sep 2024 20:20:02 +0300
Subject: [PATCH] Fix issue 1620

---
 mlir/lib/Conversion/TosaToRock/CMakeLists.txt | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/mlir/lib/Conversion/TosaToRock/CMakeLists.txt b/mlir/lib/Conversion/TosaToRock/CMakeLists.txt
index ec003c79cf45..8bf8d205253e 100644
--- a/mlir/lib/Conversion/TosaToRock/CMakeLists.txt
+++ b/mlir/lib/Conversion/TosaToRock/CMakeLists.txt
@@ -7,7 +7,9 @@ add_rocmlir_conversion_library(MLIRTosaToRock
   ${MLIR_MAIN_INCLUDE_DIR}/mlir/IR
 
   DEPENDS
+  MLIRRockPassIncGen
   MLIRConversionPassIncGen
+  RocMLIRConversionPassIncGen
 
   LINK_LIBS PUBLIC
   MLIRIR
