From 372f9a44022eafca437b753d5e61eee9315594fc Mon Sep 17 00:00:00 2001
From: Peter O'Connor <peter@solus-project.com>
Date: Mon, 8 Oct 2018 13:32:34 +1100
Subject: [PATCH] Enable stack-protector-strong by default

Signed-off-by: Peter O'Connor <peter@solus-project.com>
---
 clang/lib/Driver/ToolChains/Linux.h | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/lib/Driver/ToolChains/Linux.h b/lib/Driver/ToolChains/Linux.h
index 4c6199469..e04ca3343 100644
--- a/lib/Driver/ToolChains/Linux.h
+++ b/lib/Driver/ToolChains/Linux.h
@@ -40,6 +40,9 @@ public:
   bool isPIEDefault() const override;
   bool isNoExecStackDefault() const override;
   bool IsMathErrnoDefault() const override;
+  LangOptions::StackProtectorMode GetDefaultStackProtectorLevel(bool KernelOrKext) const override {
+    return LangOptions::SSPStrong;
+  }
   SanitizerMask getSupportedSanitizers() const override;
   void addProfileRTLibs(const llvm::opt::ArgList &Args,
                         llvm::opt::ArgStringList &CmdArgs) const override;
-- 
2.22.0

