From 9b8f9ceef6250ddb797ee7a8cd21303457ddf088 Mon Sep 17 00:00:00 2001
From: Ikey Doherty <ikey@solus-project.com>
Date: Thu, 16 Jul 2015 21:12:49 +0100
Subject: [PATCH] Use /usr/lib64 for dynamic linker

Signed-off-by: Ikey Doherty <ikey@solus-project.com>
---
 lib/Driver/Tools.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/lib/Driver/Tools.cpp b/lib/Driver/Tools.cpp
index 198e82e..43ba583 100644
--- a/lib/Driver/Tools.cpp
+++ b/lib/Driver/Tools.cpp
@@ -7228,7 +7228,7 @@ static StringRef getLinuxDynamicLinker(const ArgList &Args,
            ToolChain.getTriple().getEnvironment() == llvm::Triple::GNUX32)
     return "/libx32/ld-linux-x32.so.2";
   else
-    return "/lib64/ld-linux-x86-64.so.2";
+    return "/usr/lib64/ld-linux-x86-64.so.2";
 }
 
 static void AddRunTimeLibs(const ToolChain &TC, const Driver &D,
-- 
2.4.5

