From eada0144b7654148ff224fa39d99c2c0e6d85771 Mon Sep 17 00:00:00 2001
From: Joey Riches <josephriches@gmail.com>
Date: Fri, 3 Sep 2021 11:25:52 +0100
Subject: [PATCH 1/1] Ignore --copy-dt-needed-entries

---
 lld/ELF/Options.td | 1 +
 1 file changed, 1 insertion(+)

diff --git a/lld/ELF/Options.td b/lld/ELF/Options.td
index b5c1be5c6..30d752f0d 100644
--- a/lld/ELF/Options.td
+++ b/lld/ELF/Options.td
@@ -693,6 +693,7 @@ def plugin_opt_eq : J<"plugin-opt=">;
 
 // Options listed below are silently ignored for now for compatibility.
 def: F<"detect-odr-violations">;
+def: F<"copy-dt-needed-entries">;
 def: Flag<["-"], "g">;
 def: F<"long-plt">;
 def: F<"no-add-needed">;
-- 
2.32.0

