From 1298875096e536ec45a45fe6af0567161cc1c562 Mon Sep 17 00:00:00 2001
From: Ikey Doherty <ikey@solus-project.com>
Date: Sat, 8 Oct 2016 00:53:24 +0100
Subject: [PATCH 1/2] Use /usr/lib64 for dynamic linker

Signed-off-by: Ikey Doherty <ikey@solus-project.com>
---
 lib/Driver/ToolChains.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/lib/Driver/ToolChains.cpp b/lib/Driver/ToolChains.cpp
index 347aa29..c828812 100644
--- a/lib/Driver/ToolChains.cpp
+++ b/lib/Driver/ToolChains.cpp
@@ -4326,7 +4326,7 @@ std::string Linux::getDynamicLinker(const ArgList &Args) const {
   if (Distro == Exherbo && (Triple.getVendor() == llvm::Triple::UnknownVendor ||
                             Triple.getVendor() == llvm::Triple::PC))
     return "/usr/" + Triple.str() + "/lib/" + Loader;
-  return "/" + LibDir + "/" + Loader;
+  return "/usr/" + LibDir + "/" + Loader;
 }
 
 void Linux::AddClangSystemIncludeArgs(const ArgList &DriverArgs,
-- 
2.10.0

