From d339ca9a7cab44df015793c066c290995be30cab Mon Sep 17 00:00:00 2001
From: Joey Riches <josephriches@gmail.com>
Date: Sat, 28 May 2022 12:49:57 +0100
Subject: [PATCH 1/1] Silently ignore copy dt needed entries

---
 lld/ELF/Options.td | 1 +
 1 file changed, 1 insertion(+)

diff --git a/lld/ELF/Options.td b/lld/ELF/Options.td
index 73e7b2dde..7c451212a 100644
--- a/lld/ELF/Options.td
+++ b/lld/ELF/Options.td
@@ -690,6 +690,7 @@ def: Flag<["-"], "dc">;
 def: Flag<["-"], "g">;
 def: F<"long-plt">;
 def: FF<"no-add-needed">;
+def: F<"copy-dt-needed-entries">;
 def: F<"no-copy-dt-needed-entries">;
 def: F<"no-ctors-in-init-array">;
 def: F<"no-keep-memory">;
-- 
2.35.3

