From 079693c530350e8a88a11dbb5ec49fe529547a98 Mon Sep 17 00:00:00 2001
From: Ikey Doherty <ikey@solus-project.com>
Date: Sun, 8 Oct 2017 22:57:49 +0100
Subject: [PATCH] llvm: Let the linker find the correct ffi as cmake breaks it

Signed-off-by: Ikey Doherty <ikey@solus-project.com>
---
 lib/ExecutionEngine/Interpreter/CMakeLists.txt | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/lib/ExecutionEngine/Interpreter/CMakeLists.txt b/lib/ExecutionEngine/Interpreter/CMakeLists.txt
index 7456b3d..59fe90c 100644
--- a/lib/ExecutionEngine/Interpreter/CMakeLists.txt
+++ b/lib/ExecutionEngine/Interpreter/CMakeLists.txt
@@ -16,5 +16,5 @@ add_llvm_library(LLVMInterpreter
   )
 
 if( LLVM_ENABLE_FFI )
-  target_link_libraries( LLVMInterpreter PRIVATE ${FFI_LIBRARY_PATH} )
+  target_link_libraries( LLVMInterpreter PRIVATE ffi )
 endif()
-- 
2.14.2

