From 1d2bac35b4232c438495e11440722dbe1179378c Mon Sep 17 00:00:00 2001
From: Ikey Doherty <ikey@solus-project.com>
Date: Sat, 8 Oct 2016 02:08:53 +0100
Subject: [PATCH] Completely ignore the borky FFI_LIBRARY_PATH

Signed-off-by: Ikey Doherty <ikey@solus-project.com>
---
 lib/ExecutionEngine/Interpreter/CMakeLists.txt | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/lib/ExecutionEngine/Interpreter/CMakeLists.txt b/lib/ExecutionEngine/Interpreter/CMakeLists.txt
index 4dbc2df..33a83df 100644
--- a/lib/ExecutionEngine/Interpreter/CMakeLists.txt
+++ b/lib/ExecutionEngine/Interpreter/CMakeLists.txt
@@ -13,7 +13,7 @@ add_llvm_library(LLVMInterpreter
   )
 
 if( LLVM_ENABLE_FFI )
-  target_link_libraries( LLVMInterpreter PRIVATE ${FFI_LIBRARY_PATH} )
+  target_link_libraries( LLVMInterpreter PRIVATE ffi )
 endif()
 
 add_dependencies(LLVMInterpreter intrinsics_gen)
-- 
2.10.0

