From b92631e4eca403b0d6169df7865d4eead0411ff1 Mon Sep 17 00:00:00 2001
From: Ikey Doherty <ikey@solus-project.com>
Date: Sat, 8 Oct 2016 00:56:27 +0100
Subject: [PATCH 2/2] Implement Solus's default toolchain options

Signed-off-by: Ikey Doherty <ikey@solus-project.com>
---
 lib/Driver/ToolChains.cpp | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/lib/Driver/ToolChains.cpp b/lib/Driver/ToolChains.cpp
index c828812..b09bd24 100644
--- a/lib/Driver/ToolChains.cpp
+++ b/lib/Driver/ToolChains.cpp
@@ -4044,6 +4044,9 @@ Linux::Linux(const Driver &D, const llvm::Triple &Triple, const ArgList &Args)
   if (IsRedhat(Distro) && Distro != RHEL5 && Distro != RHEL6)
     ExtraOpts.push_back("--no-add-needed");
 
+  /* Solus Operating System support */
+  ExtraOpts.push_back("--hash-style=gnu");
+
 #ifdef ENABLE_LINKER_BUILD_ID
   ExtraOpts.push_back("--build-id");
 #endif
-- 
2.10.0

