From 276c1e284eec36af55ed1bad2c3671ac5765e211 Mon Sep 17 00:00:00 2001
From: Ikey Doherty <ikey@solus-project.com>
Date: Sat, 18 Jul 2015 01:45:55 +0100
Subject: [PATCH 2/2] Implement Solus's default toolchain options

Signed-off-by: Ikey Doherty <ikey@solus-project.com>
---
 lib/Driver/ToolChains.cpp | 8 ++++++--
 1 file changed, 6 insertions(+), 2 deletions(-)

diff --git a/lib/Driver/ToolChains.cpp b/lib/Driver/ToolChains.cpp
index 99c7b8e..1287d88 100644
--- a/lib/Driver/ToolChains.cpp
+++ b/lib/Driver/ToolChains.cpp
@@ -3682,7 +3682,7 @@ Linux::Linux(const Driver &D, const llvm::Triple &Triple, const ArgList &Args)
   // .gnu.hash needs symbols to be grouped by hash code whereas the MIPS
   // ABI requires a mapping between the GOT and the symbol table.
   // Android loader does not support .gnu.hash.
-  if (!IsMips && !IsAndroid) {
+  /*if (!IsMips && !IsAndroid) {
     if (IsRedhat(Distro) || IsOpenSUSE(Distro) ||
         (IsUbuntu(Distro) && Distro >= UbuntuMaverick))
       ExtraOpts.push_back("--hash-style=gnu");
@@ -3701,7 +3701,11 @@ Linux::Linux(const Driver &D, const llvm::Triple &Triple, const ArgList &Args)
     ExtraOpts.push_back("--build-id");
 
   if (IsOpenSUSE(Distro))
-    ExtraOpts.push_back("--enable-new-dtags");
+    ExtraOpts.push_back("--enable-new-dtags");*/
+    
+  /* Solus Operating System support */
+  ExtraOpts.push_back("--hash-style=gnu");
+  ExtraOpts.push_back("--build-id");
 
   // The selection of paths to try here is designed to match the patterns which
   // the GCC driver itself uses, as this is part of the GCC-compatible driver.
-- 
2.7.3

