From 36c43f1f09aa4117b7ec894ec61a31f01baddf87 Mon Sep 17 00:00:00 2001
From: Ikey Doherty <ikey@solus-project.com>
Date: Thu, 1 Jun 2017 00:54:35 +0100
Subject: [PATCH 2/3] cfe: Use the GNU hash style for Solus binary build-ids

Signed-off-by: Ikey Doherty <ikey@solus-project.com>
---
 lib/Driver/ToolChains.cpp | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/lib/Driver/ToolChains.cpp b/lib/Driver/ToolChains.cpp
index 0648b14..098fcc3 100644
--- a/lib/Driver/ToolChains.cpp
+++ b/lib/Driver/ToolChains.cpp
@@ -4139,6 +4139,9 @@ Linux::Linux(const Driver &D, const llvm::Triple &Triple, const ArgList &Args)
     if (Distro.IsDebian() || Distro.IsOpenSUSE() || Distro == Distro::UbuntuLucid ||
         Distro == Distro::UbuntuJaunty || Distro == Distro::UbuntuKarmic)
       ExtraOpts.push_back("--hash-style=both");
+
+    // Solus defaults to GNU hash style
+    ExtraOpts.push_back("--hash-style=gnu");
   }
 
   if (Distro.IsRedhat() && Distro != Distro::RHEL5 && Distro != Distro::RHEL6)
-- 
2.13.0

