From 6f6cd15d6ba5515ece1a4bbd968732c5299cd2e2 Mon Sep 17 00:00:00 2001
From: Ikey Doherty <ikey@solus-project.com>
Date: Sun, 8 Oct 2017 22:56:20 +0100
Subject: [PATCH 3/3] cfe: Link properly to multilib xml2 library

Signed-off-by: Ikey Doherty <ikey@solus-project.com>
---
 tools/c-index-test/CMakeLists.txt | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/tools/c-index-test/CMakeLists.txt b/tools/c-index-test/CMakeLists.txt
index ad990e0..b2bdc14 100644
--- a/tools/c-index-test/CMakeLists.txt
+++ b/tools/c-index-test/CMakeLists.txt
@@ -39,7 +39,7 @@ set_target_properties(c-index-test
 # If libxml2 is available, make it available for c-index-test.
 if (CLANG_HAVE_LIBXML)
   include_directories(SYSTEM ${LIBXML2_INCLUDE_DIR})
-  target_link_libraries(c-index-test ${LIBXML2_LIBRARIES})
+  target_link_libraries(c-index-test xml2)
 endif()
 
 if (NOT LLVM_INSTALL_TOOLCHAIN_ONLY)
-- 
2.14.2

