From 040be8b657a96aebfbd8a7af3bda8f0d6247801b Mon Sep 17 00:00:00 2001
From: Ikey Doherty <ikey@solus-project.com>
Date: Fri, 2 Jun 2017 01:26:57 +0100
Subject: [PATCH 3/3] cfe: Link properly to multilib xml2 library

Signed-off-by: Ikey Doherty <ikey@solus-project.com>
---
 tools/c-index-test/CMakeLists.txt | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/tools/c-index-test/CMakeLists.txt b/tools/c-index-test/CMakeLists.txt
index e0df503..417914b 100644
--- a/tools/c-index-test/CMakeLists.txt
+++ b/tools/c-index-test/CMakeLists.txt
@@ -36,7 +36,7 @@ set_target_properties(c-index-test
 # If libxml2 is available, make it available for c-index-test.
 if (CLANG_HAVE_LIBXML)
   include_directories(SYSTEM ${LIBXML2_INCLUDE_DIR})
-  target_link_libraries(c-index-test ${LIBXML2_LIBRARIES})
+  target_link_libraries(c-index-test xml2)
 endif()
 
 if (NOT LLVM_INSTALL_TOOLCHAIN_ONLY)
-- 
2.13.0

