From 5246858c4648610c6a4fb7770d07f975ba8a8d93 Mon Sep 17 00:00:00 2001
From: Rex Morgan <rex@rexflex.net>
Date: Thu, 1 Jun 2017 17:04:23 -0500
Subject: [PATCH] Fixing build error with gcc7.

---
 source/Utility/TaskPool.cpp | 1 +
 1 file changed, 1 insertion(+)

diff --git a/source/Utility/TaskPool.cpp b/source/Utility/TaskPool.cpp
index c5c63a2..a0ba444 100644
--- a/source/Utility/TaskPool.cpp
+++ b/source/Utility/TaskPool.cpp
@@ -7,6 +7,7 @@
 //
 //===----------------------------------------------------------------------===//
 
+#include <functional>
 #include "lldb/Utility/TaskPool.h"
 
 namespace
-- 
2.13.0