--- Makefile	2019-07-12 02:16:46.000000000 +0200
+++ ../Makefile	2020-02-10 15:03:48.671709047 +0100
@@ -11,3 +11,4 @@
 P8IGN = E251,E201,E302,E202,E126,E127,E203,E226
-PREFIX ?= /usr
+DESTDIR ?= /
+PREFIX ?= $(DESTDIR)/usr
 BINDIR ?= $(PREFIX)/bin
@@ -57,3 +58,3 @@
 install: install-mnexec install-manpages
-	$(PYTHON) setup.py install
+	$(PYTHON) setup.py install --prefix=/usr --root="$(DESTDIR)" --optimize=1 install
 
