From 518962a81099941a5d636d0b20c7afa9c08d36cf Mon Sep 17 00:00:00 2001
From: Simon Schneegans <code@simonschneegans.de>
Date: Tue, 2 Jan 2024 20:32:05 +0100
Subject: [PATCH] :bug: Fix invisible settings button on Plasma 6

---
 kwin/metadata.json.in | 1 +
 1 file changed, 1 insertion(+)

diff --git a/kwin/metadata.json.in b/kwin/metadata.json.in
index 7da0e59..c3cdf07 100755
--- a/kwin/metadata.json.in
+++ b/kwin/metadata.json.in
@@ -20,6 +20,7 @@
        "KWin/Effect"
      ]
   },
+  "X-KDE-ConfigModule": "kcm_kwin4_genericscripted",
   "X-KDE-Ordering": "60",
   "X-KDE-PluginKeyword": "%DIR_NAME%",
   "X-KWin-Config-TranslationDomain": "burn-my-windows",
