From eb933572f5b169de6bbc4a925d94770aca82f30a Mon Sep 17 00:00:00 2001
From: Coelacanthus <CoelacanthusHex@gmail.com>
Date: Sun, 19 Nov 2023 20:54:56 +0800
Subject: [PATCH] Fix compatibility with Plasma 6

resolve #401

Signed-off-by: Coelacanthus <CoelacanthusHex@gmail.com>
---
 kwin/metadata.json.in | 1 +
 1 file changed, 1 insertion(+)

diff --git a/kwin/metadata.json.in b/kwin/metadata.json.in
index 395d5806..7da0e59f 100755
--- a/kwin/metadata.json.in
+++ b/kwin/metadata.json.in
@@ -1,4 +1,5 @@
 {
+  "KPackageStructure": "KWin/Effect",
   "KPlugin": {
      "Name": "%NAME%",
      "Description": "%DESCRIPTION%",
