From 73d4cf52e379e9aec5cc16e020d75d5fbf47aed2 Mon Sep 17 00:00:00 2001
From: Joshua Strobl <joshua@streambits.io>
Date: Tue, 24 Sep 2019 11:58:32 +0300
Subject: [PATCH 1/1] Patch out ldconfig, change prefix while we are at it.

---
 source/Makefile | 4 +---
 1 file changed, 1 insertion(+), 3 deletions(-)

diff --git a/source/Makefile b/source/Makefile
index cd54ff5..06833d3 100644
--- a/source/Makefile
+++ b/source/Makefile
@@ -21,7 +21,7 @@
 # Modify as required.
 #
 SUFFIX := $(shell uname -m | sed -e 's/^unknown/$//' -e 's/^i.86/$//' -e 's/^x86_64/$/64/')
-PREFIX ?= /usr/local
+PREFIX ?= /usr
 INCDIR ?= $(PREFIX)/include
 LIBDIR ?= $(PREFIX)/lib$(SUFFIX)
 
@@ -33,7 +33,6 @@ DISTDIR = zita-resampler-$(VERSION)
 
 CPPFLAGS += -I. -D_REENTRANT -D_POSIX_PTHREAD_SEMANTICS
 CXXFLAGS += -Wall -fPIC -O2 -ffast-math
-CXXFLAGS += -march=native
 LDFLAGS += 
 LDLIBS +=
 
@@ -58,7 +57,6 @@ install:	$(ZITA-RESAMPLER_MIN)
 	install -d $(DESTDIR)$(LIBDIR)
 	install -m 644 $(ZITA-RESAMPLER_H)   $(DESTDIR)$(INCDIR)/zita-resampler
 	install -m 755 $(ZITA-RESAMPLER_MIN) $(DESTDIR)$(LIBDIR)
-	ldconfig
 	ln -sf $(ZITA-RESAMPLER_MIN) $(DESTDIR)$(LIBDIR)/$(ZITA-RESAMPLER_SO)
 
 uninstall:
-- 
2.23.0

