From 63194bab650c62bf41863d15fdcfb561eb39c3b1 Mon Sep 17 00:00:00 2001
From: Muhammad Alfi Syahrin <malfisya.dev@hotmail.com>
Date: Fri, 16 Feb 2024 05:48:40 +0700
Subject: [PATCH] Patch prefix to use usr

---
 source/Makefile | 3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

diff --git a/source/Makefile b/source/Makefile
index f9f1d88..46d1aef 100644
--- a/source/Makefile
+++ b/source/Makefile
@@ -21,7 +21,7 @@
 # Modify as required.
 #
 SUFFIX := $(shell uname -m | sed -e 's/^unknown/$//' -e 's/^i.86/$//' -e 's/^x86_64/$/64/')
-PREFIX ?= /usr/local
+PREFIX ?= /usr
 INCDIR ?= $(PREFIX)/include
 LIBDIR ?= $(PREFIX)/lib$(SUFFIX)
 
@@ -34,7 +34,6 @@ CPPFLAGS += -I. -D_REENTRANT -D_POSIX_PTHREAD_SEMANTICS
 CPPFLAGS += -DENABLE_SSE2
 #CPPFLAGS += -DENABLE_NEON  
 CXXFLAGS += -Wall -fPIC -O2 -ffast-math
-CXXFLAGS += -march=native
 LDFLAGS += 
 LDLIBS +=
 
-- 
2.43.1

