From 53b0594f77edcbe63e67c9edcc17fd216dad29a3 Mon Sep 17 00:00:00 2001
From: Shannon Huggins <phate@houseofsyn.com>
Date: Sat, 27 Mar 2021 09:35:12 +1100
Subject: [PATCH] Add stateless configuration support

---
 minidlna.c | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/minidlna.c b/minidlna.c
index b2769ae..001f6a4 100644
--- a/minidlna.c
+++ b/minidlna.c
@@ -550,6 +550,7 @@ init(int argc, char **argv)
 	struct sigaction sa;
 	const char * presurl = NULL;
 	const char * optionsfile = "/etc/minidlna.conf";
+	const char * defaultfile = "/usr/share/defaults/etc/minidlna.conf";
 	char mac_str[13];
 	char *string, *word;
 	char *path;
@@ -594,9 +595,8 @@ init(int argc, char **argv)
 	 * command line arguments have final say */
 	if (readoptionsfile(optionsfile) < 0)
 	{
-		/* only error if file exists or using -f */
-		if(access(optionsfile, F_OK) == 0 || options_flag)
-			DPRINTF(E_FATAL, L_GENERAL, "Error reading configuration file %s\n", optionsfile);
+		readoptionsfile(defaultfile);
+
 	}
 
 	for (i=0; i<num_options; i++)
-- 
2.30.2

