--- a/btf_encoder.c
+++ b/btf_encoder.c
@@ -1967,6 +1967,7 @@
 		return btf_encoder__add_ref_type(encoder, BTF_KIND_PTR, ref_type_id, NULL, false);
 	case DW_TAG_restrict_type:
 		return btf_encoder__add_ref_type(encoder, BTF_KIND_RESTRICT, ref_type_id, NULL, false);
+	case DW_TAG_atomic_type:
 	case DW_TAG_volatile_type:
 		return btf_encoder__add_ref_type(encoder, BTF_KIND_VOLATILE, ref_type_id, NULL, false);
 	case DW_TAG_typedef:
