diff --git a/Cargo.lock b/Cargo.lock
index c491f7e..15ecbff 100644
--- a/Cargo.lock
+++ b/Cargo.lock
@@ -586,7 +586,7 @@ dependencies = [
 
 [[package]]
 name = "obs-cmd"
-version = "0.18.1"
+version = "0.18.4"
 dependencies = [
  "clap",
  "obws",
diff --git a/Cargo.toml b/Cargo.toml
index a801105..361f794 100644
--- a/Cargo.toml
+++ b/Cargo.toml
@@ -1,6 +1,6 @@
 [package]
 name = "obs-cmd"
-version = "0.18.1"
+version = "0.18.4"
 edition = "2021"
 description = "A minimal command to control obs via obs-websocket"
 authors = ["Luigi Maselli <luigi@grigio.org>"]
