--- a/containers.conf
+++ b/containers.conf
@@ -262,7 +262,7 @@
 
 # Path to directory where CNI plugin binaries are located.
 #
-#cni_plugin_dirs = ["/usr/libexec/cni"]
+cni_plugin_dirs = ["/usr/lib64/cni"]
 
 # The network name of the default CNI network to attach pods to.
 #
@@ -427,7 +427,7 @@
 
 # Default OCI runtime
 #
-#runtime = "crun"
+runtime = "runc"
 
 # List of the OCI runtimes that support --format=json.  When json is supported
 # engine will use it for reporting nicer errors.
