--- a/containers.conf
+++ b/containers.conf
@@ -238,7 +238,7 @@ default_sysctls = [
 
 # Path to directory where CNI plugin binaries are located.
 #
-# cni_plugin_dirs = ["/usr/libexec/cni"]
+cni_plugin_dirs = ["/usr/lib64/cni"]
 
 # The network name of the default CNI network to attach pods to.
 # default_network = "podman"
@@ -398,7 +398,7 @@ default_sysctls = [
 
 # Default OCI runtime
 #
-# runtime = "crun"
+runtime = "runc"
 
 # List of the OCI runtimes that support --format=json.  When json is supported
 # engine will use it for reporting nicer errors.
