--- a/containers.conf
+++ b/containers.conf
@@ -327,7 +327,7 @@ default_sysctls = [
 # Before changing this value all containers must be stopped otherwise it is likely that
 # iptables rules and network interfaces might leak on the host. A reboot will fix this.
 #
-#network_backend = ""
+network_backend = "netavark"
 
 # Path to directory where CNI plugin binaries are located.
 #
@@ -671,7 +671,7 @@ default_sysctls = [
 
 # Default OCI runtime
 #
-#runtime = "crun"
+runtime = "runc"
 
 # List of the OCI runtimes that support --format=json. When json is supported
 # engine will use it for reporting nicer errors.
