--- a/registries.conf
+++ b/registries.conf
@@ -16,7 +16,10 @@
 # create accounts with arbitrary names). This will prevent an image from being
 # spoofed, squatted or otherwise made insecure.  If it is necessary to use one
 # of these registries, it should be added at the end of the list.
-#
+
+[registries.search]
+registries = ['docker.io', 'gcr.io', 'quay.io', 'registry.fedoraproject.org', 'registry.access.redhat.com', 'registry.centos.org']
+
 # # An array of host[:port] registries to try when pulling an unqualified image, in order.
 # unqualified-search-registries = ["example.com"]
 #
