diff --git a/mk/nvidia-modprobe.mk b/mk/nvidia-modprobe.mk
index ad399de..489275e 100644
--- a/mk/nvidia-modprobe.mk
+++ b/mk/nvidia-modprobe.mk
@@ -29,14 +29,6 @@ CFLAGS   := -O2 -g -fdata-sections -ffunction-sections -fstack-protector -fno-st
 
 LIB_OBJS := $(LIB_SRCS:.c=.o)
 
-$(SRCS_DIR)/.download_stamp:
-	$(MKDIR) -p $(SRCS_DIR)
-	$(CURL) --progress-bar -fSL $(URL) | \
-	$(TAR) -C $(SRCS_DIR) --strip-components=1 -xz $(PREFIX)/modprobe-utils
-	@touch $@
-
-$(LIB_SRCS): $(SRCS_DIR)/.download_stamp
-
 ##### Public rules #####
 
 .PHONY: all install clean
