From 9443620eaefc50a9de2027410c9fb97964c6a98f Mon Sep 17 00:00:00 2001
From: Pierre-Yves <pyu@riseup.net>
Date: Mon, 18 Dec 2017 19:09:14 +0100
Subject: [PATCH] fix install libdir path

Signed-off-by: Pierre-Yves <pyu@riseup.net>
---
 makefile.in | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/makefile.in b/makefile.in
index ac47bc2..332c41d 100644
--- a/makefile.in
+++ b/makefile.in
@@ -1188,9 +1188,9 @@ help:
 install: all
 	@echo "installing..."
 	@echo ""
-	mkdir -p $(DESTDIR)$(exec_prefix)$(libdir)
+	mkdir -p $(DESTDIR)$(libdir)
 	mkdir -p $(DESTDIR)$(prefix)/include/liquid
-	install -m 644 -p $(SHARED_LIB) libliquid.a $(DESTDIR)$(exec_prefix)$(libdir)
+	install -m 644 -p $(SHARED_LIB) libliquid.a $(DESTDIR)$(libdir)
 	install -m 644 -p $(addprefix include/,$(headers_install)) $(DESTDIR)$(prefix)/include/liquid
 	@echo ""
 	@echo "---------------------------------------------------------"
@@ -1213,8 +1213,8 @@ install: all
 uninstall:
 	@echo "uninstalling..."
 	$(RM) $(addprefix $(DESTDIR)$(prefix)/include/liquid/, $(headers_install))
-	$(RM) $(DESTDIR)$(exec_prefix)$(libdir)/libliquid.a
-	$(RM) $(DESTDIR)$(exec_prefix)$(libdir)/$(SHARED_LIB)
+	$(RM) $(DESTDIR)$(libdir)/libliquid.a
+	$(RM) $(DESTDIR)$(libdir)/$(SHARED_LIB)
 	@echo "done."
 
 ##
-- 
2.15.1

