From eae5d497ba44f152fc4b9d2cd9400e9dedf858f0 Mon Sep 17 00:00:00 2001
From: Joey Riches <josephriches@gmail.com>
Date: Thu, 16 Nov 2023 16:23:20 +0000
Subject: [PATCH 1/1] packagekit: Hardcode some solus thingies

---
 plugins/packagekit/packagekit-common.c | 7 +++++++
 1 file changed, 7 insertions(+)

diff --git a/plugins/packagekit/packagekit-common.c b/plugins/packagekit/packagekit-common.c
index 445487b12..638ec1e76 100644
--- a/plugins/packagekit/packagekit-common.c
+++ b/plugins/packagekit/packagekit-common.c
@@ -508,6 +508,9 @@ gs_plugin_packagekit_refine_details_app (GsPlugin *plugin,
 	guint j;
 	guint64 download_size = 0, install_size = 0;
 
+	/* Solus Hack: Packagekit provides no way to get repo url so hardcode it for now */
+	gs_app_set_origin_hostname (app, "https://cdn.getsol.us");
+
 	/* @source_ids can have as many as 200 elements (google-noto); typically
 	 * it has 1 or 2
 	 *
@@ -587,6 +590,10 @@ gs_plugin_packagekit_set_packaging_format (GsPlugin *plugin, GsApp *app)
 	} else if (gs_plugin_check_distro_id (plugin, "debian") ||
 	           gs_plugin_check_distro_id (plugin, "ubuntu")) {
 		gs_app_set_metadata (app, "GnomeSoftware::PackagingFormat", "deb");
+	} else if (gs_plugin_check_distro_id (plugin, "solus")) {
+		gs_app_set_metadata (app, "GnomeSoftware::PackagingFormat", "eopkg");
+		gs_app_set_metadata (app, "GnomeSoftware::PackagingIcon", "distributor-logo-solus");
+		return;
 	} else {
 		return;
 	}
-- 
2.42.0

