From 861d3b2d6642789c0be32d7a03501940c8916fb8 Mon Sep 17 00:00:00 2001
From: Troy Harvey <harveydevel@gmail.com>
Date: Sun, 8 Jun 2025 12:21:15 +1000
Subject: [PATCH] Show console apps

Signed-off-by: Troy Harvey <harveydevel@gmail.com>
---
 lib/gs-plugin-loader.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/lib/gs-plugin-loader.c b/lib/gs-plugin-loader.c
index e2d08eaec..a2645087b 100644
--- a/lib/gs-plugin-loader.c
+++ b/lib/gs-plugin-loader.c
@@ -759,12 +759,13 @@ gs_plugin_loader_app_is_valid (GsApp               *app,
 		return FALSE;
 	}
 
-	/* never show CLI apps */
+	/* never show CLI apps
 	if (gs_app_get_kind (app) == AS_COMPONENT_KIND_CONSOLE_APP) {
 		g_debug ("app invalid as console %s",
 			 gs_plugin_loader_get_app_str (app));
 		return FALSE;
 	}
+	*/
 
 	/* don't show unknown state */
 	if (gs_app_get_state (app) == GS_APP_STATE_UNKNOWN) {
-- 
2.49.0

