From 772ebaaf47dfe9e0facd5812451f328497c4b16c Mon Sep 17 00:00:00 2001
From: Alexander Vorobyev <avorobyev@protonmail.com>
Date: Wed, 19 Sep 2018 20:46:00 +0300
Subject: Enable asar

---
 Gruntfile.js | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/Gruntfile.js b/Gruntfile.js
index f328f6a..dd03fb5 100644
--- a/Gruntfile.js
+++ b/Gruntfile.js
@@ -214,7 +214,7 @@ module.exports = function(grunt) {
 
       options: {
         arch: 'all',
-        asar: false,
+        asar: true,
         publish: null,
       },
     },
-- 
2.18.0

