From 600bfe81d454eaa12a6d8f907643d4ed3bc97f31 Mon Sep 17 00:00:00 2001
From: Joey Riches <josephriches@gmail.com>
Date: Sun, 20 Aug 2017 16:52:21 +0100
Subject: [PATCH 1/1] Wire: enable asar for linux

---
 Gruntfile.coffee | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/Gruntfile.coffee b/Gruntfile.coffee
index 8b4a338..38ab574 100644
--- a/Gruntfile.coffee
+++ b/Gruntfile.coffee
@@ -113,7 +113,7 @@ module.exports = (grunt) ->
 
     electronbuilder:
       options:
-        asar: false
+        asar: true
         arch: 'all'
       linux_prod:
         options:
-- 
2.14.1

