From 07dc3accb6fa4440050a7ec0ba5460e6a3b57e8e Mon Sep 17 00:00:00 2001
From: Alexander Vorobyev <avorobyev@protonmail.com>
Date: Thu, 14 Mar 2019 21:40:35 +0300
Subject: Enable asar

---
 Gruntfile.js | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/Gruntfile.js b/Gruntfile.js
index 917ee3c..29e84b6 100644
--- a/Gruntfile.js
+++ b/Gruntfile.js
@@ -287,7 +287,7 @@ module.exports = function(grunt) {
 
       options: {
         arch: 'x64',
-        asar: false,
+        asar: true,
         publish: null,
       },
     },
-- 
2.21.0

