From e78483818b063c5a8a8b3b7cc0de473561a7b2f6 Mon Sep 17 00:00:00 2001
From: Pierre-Yves <pyu@riseup.net>
Date: Fri, 9 Nov 2018 22:39:53 +0100
Subject: [PATCH] Disable creation of static library

Signed-off-by: Pierre-Yves <pyu@riseup.net>
---
 src/wscript_build | 4 ----
 1 file changed, 4 deletions(-)

diff --git a/src/wscript_build b/src/wscript_build
index 895c191..30f02db 100644
--- a/src/wscript_build
+++ b/src/wscript_build
@@ -35,10 +35,6 @@ else:
     # linux, darwin, android, mingw, ...
     build_features = ['cstlib', 'cshlib']
 
-# also install static lib
-from waflib.Tools.c import cstlib
-cstlib.inst_to = '${LIBDIR}'
-
 for target in build_features:
     ctx(features = 'c ' + target,
             use = uselib + ['lib_objects'],
-- 
2.19.1

