From 70abdc76e38763dfe4d745e0cd562f473e1b176c Mon Sep 17 00:00:00 2001
From: Michael Meinertzhagen <mmeinert@protonmail.com>
Date: Wed, 19 Apr 2017 09:34:43 +0200
Subject: [PATCH 1/1] Disable creation of static library

---
 src/wscript_build | 5 -----
 1 file changed, 5 deletions(-)

diff --git a/src/wscript_build b/src/wscript_build
index 0a7d72a..6786ff0 100644
--- a/src/wscript_build
+++ b/src/wscript_build
@@ -34,11 +34,6 @@ else:
     # linux, darwin, android, mingw, ...
     build_features = ['cstlib', 'cshlib']
 
-# also install static lib
-from waflib.Tools.c import cstlib
-from waflib.Tools.fc import fcstlib
-fcstlib.inst_to = cstlib.inst_to = '${LIBDIR}'
-
 for target in build_features:
     ctx(features = 'c ' + target,
             use = uselib + ['lib_objects'],
-- 
2.12.2

