From 6d7f9b89899e73b94121436c9494fe50f7b0bdbf Mon Sep 17 00:00:00 2001
From: Reilly Brogan <reilly@reillybrogan.com>
Date: Sat, 20 Jan 2024 18:43:57 -0600
Subject: [PATCH] Update ADE

---
 modules/gapi/cmake/DownloadADE.cmake | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/modules/gapi/cmake/DownloadADE.cmake b/modules/gapi/cmake/DownloadADE.cmake
index aa24e949ea..a4f2dc2433 100644
--- a/modules/gapi/cmake/DownloadADE.cmake
+++ b/modules/gapi/cmake/DownloadADE.cmake
@@ -1,7 +1,7 @@
 set(ade_src_dir "${OpenCV_BINARY_DIR}/3rdparty/ade")
-set(ade_filename "v0.1.1f.zip")
-set(ade_subdir "ade-0.1.1f")
-set(ade_md5 "b624b995ec9c439cbc2e9e6ee940d3a2")
+set(ade_filename "v0.1.2d.zip")
+set(ade_subdir "ade-0.1.2d")
+set(ade_md5 "dbb095a8bf3008e91edbbf45d8d34885")
 ocv_download(FILENAME ${ade_filename}
              HASH ${ade_md5}
              URL
-- 
2.43.0

