diff --git a/org.synfig.SynfigStudio.desktop.in b/org.synfig.SynfigStudio.desktop.in
index 4e3dfcc..7a85315 100644
--- a/org.synfig.SynfigStudio.desktop.in
+++ b/org.synfig.SynfigStudio.desktop.in
@@ -7,6 +7,7 @@ Terminal=false
 Type=Application
 Icon=org.synfig.SynfigStudio
 Categories=GTK;Graphics;2DGraphics;VectorGraphics;
+StartupWMClass=synfigstudio
 StartupNotify=true
 MimeType=application/x-sif;
 Keywords=2D;Vector;Animation;Cartoon;Anime;FX;Draw;Flash;ToonBoom;Moho;
