diff --git a/BaseTools/GNUmakefile b/BaseTools/GNUmakefile
index 790d33a..4e8461a 100644
--- a/BaseTools/GNUmakefile
+++ b/BaseTools/GNUmakefile
@@ -19,17 +19,15 @@ SOURCE_SUBDIRS := $(patsubst %,Source/%,$(sort $(LANGUAGES)))
 SUBDIRS := $(SOURCE_SUBDIRS) Tests
 CLEAN_SUBDIRS := $(patsubst %,%-clean,$(sort $(SUBDIRS)))
 
+Tests: $(SOURCE_SUBDIRS)
+
 .PHONY: subdirs $(SUBDIRS)
 subdirs: $(SUBDIRS)
 $(SUBDIRS):
-	$(MAKE) -C $@
+	$(MAKE) -C $@
 
 .PHONY: $(CLEAN_SUBDIRS)
 $(CLEAN_SUBDIRS):
-	-$(MAKE) -C $(@:-clean=) clean
+	-$(MAKE) -C $(@:-clean=) clean
 
 clean:  $(CLEAN_SUBDIRS)
-
-test:
-	@$(MAKE) -C Tests
-
diff --git a/BaseTools/Source/C/GNUmakefile b/BaseTools/Source/C/GNUmakefile
index 83e188c..45e3857 100644
--- a/BaseTools/Source/C/GNUmakefile
+++ b/BaseTools/Source/C/GNUmakefile
@@ -44,10 +44,11 @@ MAKEROOT = .
 
 include Makefiles/header.makefile
 
-all: makerootdir subdirs $(MAKEROOT)/libs
-	@echo Finished building BaseTools C Tools with ARCH=$(ARCH)
+all: makerootdir subdirs
+	@echo Finished building BaseTools C Tools with ARCH=$(ARCH)
 
 LIBRARIES = Common
+VFRAUTOGEN = VfrCompile/VfrLexer.h
 # NON_BUILDABLE_APPLICATIONS = GenBootSector BootSectImage
 APPLICATIONS = \
   GnuGenBootSector \
@@ -70,25 +71,31 @@ APPLICATIONS = \
 
 SUBDIRS := $(LIBRARIES) $(APPLICATIONS)
 
+$(LIBRARIES): $(MAKEROOT)/libs
+$(APPLICATIONS): $(LIBRARIES) $(MAKEROOT)/bin $(VFRAUTOGEN)
+
 .PHONY: outputdirs
 makerootdir:
-	-mkdir -p $(MAKEROOT)
+	-mkdir -p $(MAKEROOT)
 
 .PHONY: subdirs $(SUBDIRS)
 subdirs: $(SUBDIRS)
 $(SUBDIRS):
-	$(MAKE) -C $@
+	$(MAKE) -C $@
 
 .PHONY: $(patsubst %,%-clean,$(sort $(SUBDIRS)))
 $(patsubst %,%-clean,$(sort $(SUBDIRS))):
-	-$(MAKE) -C $(@:-clean=) clean
+	-$(MAKE) -C $(@:-clean=) clean
+
+$(VFRAUTOGEN): VfrCompile/VfrSyntax.g
+	$(MAKE) -C VfrCompile VfrLexer.h
 
 clean:  $(patsubst %,%-clean,$(sort $(SUBDIRS)))
 
 clean: localClean
 
 localClean:
-	rm -f $(MAKEROOT)/bin/*
-	-rmdir $(MAKEROOT)/libs $(MAKEROOT)/bin
+	rm -f $(MAKEROOT)/bin/*
+	-rmdir $(MAKEROOT)/libs $(MAKEROOT)/bin
 
 include Makefiles/footer.makefile
