From 4917af8d9f2776f73bc6878d72ed87adf02e41da Mon Sep 17 00:00:00 2001
From: Thomas Staudinger <Staudi.Kaos@gmail.com>
Date: Sat, 10 Jun 2023 00:45:45 +0200
Subject: [PATCH] Disable unsupported ld flag

---
 build/config/compiler/BUILD.gn | 8 --------
 1 file changed, 8 deletions(-)

diff --git a/build/config/compiler/BUILD.gn b/build/config/compiler/BUILD.gn
index 28c2255ba..0e3ee4840 100644
--- a/build/config/compiler/BUILD.gn
+++ b/build/config/compiler/BUILD.gn
@@ -726,14 +726,6 @@ config("compiler") {
       }
 
       ldflags += [ "-Wl,-mllvm,-import-instr-limit=$import_instr_limit" ]
-
-      if (!is_chromeos) {
-        # TODO(https://crbug.com/972449): turn on for ChromeOS when that
-        # toolchain has this flag.
-        # We only use one version of LLVM within a build so there's no need to
-        # upgrade debug info, which can be expensive since it runs the verifier.
-        ldflags += [ "-Wl,-mllvm,-disable-auto-upgrade-debug-info" ]
-      }
     }
 
     # TODO(https://crbug.com/1211155): investigate why this isn't effective on
-- 
2.40.1
