From ae27a7c8c987b7b8ae07f69833ff070205c5419a Mon Sep 17 00:00:00 2001
From: Troy Harvey <harveydevel@gmail.com>
Date: Tue, 16 Apr 2024 11:12:12 +1000
Subject: [PATCH] drop flags unsupported by clang16

Signed-off-by: Troy Harvey <harveydevel@gmail.com>
---
 build/config/compiler/BUILD.gn | 10 +---------
 1 file changed, 1 insertion(+), 9 deletions(-)

diff --git a/build/config/compiler/BUILD.gn b/build/config/compiler/BUILD.gn
index bd7dd25..9df6e64 100644
--- a/build/config/compiler/BUILD.gn
+++ b/build/config/compiler/BUILD.gn
@@ -618,15 +618,10 @@ config("compiler") {
     # TODO(crbug.com/1491036): This isn't supported by Cronet's mainline llvm version.
     if (default_toolchain != "//build/toolchain/cros:target" &&
         !llvm_android_mainline) {
-      cflags += [
-        "-mllvm",
-        "-split-threshold-for-reg-with-hint=0",
-      ]
       if (use_thin_lto && is_a_target_toolchain) {
         if (is_win) {
           ldflags += [ "-mllvm:-split-threshold-for-reg-with-hint=0" ]
         } else {
-          ldflags += [ "-Wl,-mllvm,-split-threshold-for-reg-with-hint=0" ]
         }
       }
     }
@@ -765,7 +760,6 @@ config("compiler") {
       ldflags += [
         "/opt:lldltojobs=all",
         "-mllvm:-import-instr-limit=$import_instr_limit",
-        "-mllvm:-disable-auto-upgrade-debug-info",
       ]
     } else {
       ldflags += [ "-flto=thin" ]
@@ -800,7 +794,6 @@ config("compiler") {
 
       # We only use one version of LLVM within a build so there's no need to
       # upgrade debug info, which can be expensive since it runs the verifier.
-      ldflags += [ "-Wl,-mllvm,-disable-auto-upgrade-debug-info" ]
     }
 
     # TODO(https://crbug.com/1211155): investigate why this isn't effective on
@@ -1892,7 +1885,6 @@ config("default_warnings") {
         "-Wno-vla-extension",
 
         # TODO(https://crbug.com/1490607): Fix and re-enable.
-        "-Wno-thread-safety-reference-return",
       ]
 
       cflags_cc += [
@@ -1903,7 +1895,7 @@ config("default_warnings") {
       if (!is_nacl) {
         cflags_cc += [
           # TODO(https://crbug.com/1513724): Fix and re-enable.
-          "-Wno-c++11-narrowing-const-reference",
+          "-Wno-c++11-narrowing",
         ]
       }
     }
-- 
2.44.0

