From 69405d35580fa8b248e3b82704ba8bc42b411e72 Mon Sep 17 00:00:00 2001
From: Pierre-Yves <pyu@riseup.net>
Date: Wed, 31 Jan 2018 15:58:01 +0100
Subject: [PATCH] Remove check-if-root due to issue with fakeroot

Signed-off-by: Pierre-Yves <pyu@riseup.net>
---
 Makefile.in | 12 ++----------
 1 file changed, 2 insertions(+), 10 deletions(-)

diff --git a/Makefile.in b/Makefile.in
index 768861f..45132a4 100644
--- a/Makefile.in
+++ b/Makefile.in
@@ -9,7 +9,7 @@
 
 gb_Top_MODULE_CHECK_TARGETS := slowcheck unitcheck subsequentcheck perfcheck uicheck screenshot
 
-.PHONY : all check-if-root bootstrap gbuild build build-non-l10n-only build-l10n-only check clean clean-build clean-host test-install distclean distro-pack-install docs download etags fetch get-submodules id install install-gdb-printers install-strip tags debugrun help showmodules translations packageinfo internal.clean $(gb_Top_MODULE_CHECK_TARGETS)
+.PHONY : all bootstrap gbuild build build-non-l10n-only build-l10n-only check clean clean-build clean-host test-install distclean distro-pack-install docs download etags fetch get-submodules id install install-gdb-printers install-strip tags debugrun help showmodules translations packageinfo internal.clean $(gb_Top_MODULE_CHECK_TARGETS)
 
 MAKECMDGOALS?=all
 build_goal:=$(if $(filter build check,$(MAKECMDGOALS)),all)\
@@ -58,14 +58,6 @@ else # MAKE_RESTARTS
 
 all: build
 
-check-if-root:
-	@if test `id -u` = 0 && ! grep -q 'lxc\|docker' /proc/self/cgroup; then \
-		echo; \
-		echo 'No. You make ME a sandwich.'; \
-		echo; \
-		exit 1; \
-	fi
-
 gb_Side ?= host
 
 include $(BUILDDIR)/config_$(gb_Side).mk
@@ -254,7 +246,7 @@ endif
 #
 # Bootstrap
 #
-bootstrap: check-if-root compilerplugins
+bootstrap: compilerplugins
 
 #
 # Build
-- 
2.16.1

