From d64fa6c5735ec55432e638221f2750fc0006cd60 Mon Sep 17 00:00:00 2001
From: "Bryan T. Meyers" <bmeyers@datadrake.com>
Date: Mon, 20 May 2019 15:26:19 -0400
Subject: [PATCH] Remove includes mangling

---
 configure.ac | 1 -
 1 file changed, 1 deletion(-)

diff --git a/configure.ac b/configure.ac
index c582db0..ed1388d 100644
--- a/configure.ac
+++ b/configure.ac
@@ -11287,7 +11287,6 @@ then
     fi
 
     QT5_CFLAGS="-I$qt5_incdir -DQT_CLEAN_NAMESPACE -DQT_THREAD_SUPPORT"
-    QT5_CFLAGS=$(printf '%s' "$QT5_CFLAGS" | sed -e "s/-I/${ISYSTEM?}/g")
     QT5_LIBS="-L$qt5_libdir -lQt5Core -lQt5Gui -lQt5Widgets -lQt5Network"
 
     dnl Check for Meta Object Compiler
-- 
2.21.0


