From 4fa7adcbda78977c39d04f3b38b8ff315c2f4060 Mon Sep 17 00:00:00 2001
From: Ikey Doherty <ikey@solus-project.com>
Date: Sun, 17 May 2015 19:14:20 +0100
Subject: [PATCH] Ensure settings.ini directory exists before writing to it

Signed-off-by: Ikey Doherty <ikey@solus-project.com>
---
 gtweak/gtksettings.py | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/gtweak/gtksettings.py b/gtweak/gtksettings.py
index bcec9f1..d160df0 100644
--- a/gtweak/gtksettings.py
+++ b/gtweak/gtksettings.py
@@ -15,7 +15,7 @@
 # You should have received a copy of the GNU General Public License
 # along with gnome-tweak-tool.  If not, see <http://www.gnu.org/licenses/>.
 
-import os.path
+import os
 import logging
 
 from gi.repository import GLib
@@ -56,6 +56,8 @@ class GtkSettingsManager:
         keyfile.set_integer(SETTINGS_GROUP_NAME, key, value)
 
         try:
+            if not os.path.exists(os.path.dirname(self._path)):
+                os.makedirs(os.path.dirname(self._path))
             data = keyfile.to_data()
             GLib.file_set_contents(self._path, data[0])
         except:
-- 
2.3.2

