From 3ee2441d22255af65d6fcaeba6d751cbf73f66fa Mon Sep 17 00:00:00 2001
From: "Bryan T. Meyers" <bmeyers@datadrake.com>
Date: Mon, 8 Aug 2016 08:51:22 -0400
Subject: [PATCH 2/2] fix FindQCA

---
 cmake/FindQCA.cmake | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/cmake/FindQCA.cmake b/cmake/FindQCA.cmake
index 241ddc9..8f9bc39 100644
--- a/cmake/FindQCA.cmake
+++ b/cmake/FindQCA.cmake
@@ -21,7 +21,7 @@ if(QCA_INCLUDE_DIR AND QCA_LIBRARY)
 else(QCA_INCLUDE_DIR AND QCA_LIBRARY)
 
   if(ENABLE_QT5)
-    set(QCA_LIBRARY_NAMES qca-qt5 qca2-qt5)
+    set(QCA_LIBRARY_NAMES qca qca2 qca-qt5 qca2-qt5)
   else(ENABLE_QT5)
     set(QCA_LIBRARY_NAMES qca qca2)
   endif(ENABLE_QT5)
-- 
2.9.2

