--- a/src/libutils/oiiohelpers/oiiohelpers.cpp
+++ b/src/libutils/oiiohelpers/oiiohelpers.cpp
@@ -3,9 +3,9 @@
 
 
 #include <OpenColorIO/OpenColorIO.h>
+#include <Imath/half.h>
 
 
-#include "OpenEXR/half.h"
 #include "oiiohelpers.h"
 #include "utils/StringUtils.h"
 
--- a/src/OpenColorIO/BitDepthUtils.h
+++ b/src/OpenColorIO/BitDepthUtils.h
@@ -6,8 +6,7 @@
 #define INCLUDED_OCIO_BIT_DEPTH_UTILS_H
 
 #include <OpenColorIO/OpenColorIO.h>
-
-#include "OpenEXR/half.h"
+#include <Imath/half.h>
 
 
 namespace OCIO_NAMESPACE
--- a/src/OpenColorIO/MathUtils.h
+++ b/src/OpenColorIO/MathUtils.h
@@ -8,8 +8,7 @@
 #include <cmath>
 
 #include <OpenColorIO/OpenColorIO.h>
-
-#include "OpenEXR/half.h"
+#include <Imath/half.h>
 
 namespace OCIO_NAMESPACE
 {
--- a/src/OpenColorIO/transforms/builtins/OpHelpers.cpp
+++ b/src/OpenColorIO/transforms/builtins/OpHelpers.cpp
@@ -3,8 +3,8 @@
 
 
 #include <OpenColorIO/OpenColorIO.h>
+#include <Imath/half.h>
 
-#include "OpenEXR/half.h"
 #include "ops/lut1d/Lut1DOp.h"
 #include "transforms/builtins/OpHelpers.h"
 
--- a/src/OpenColorIO/transforms/builtins/ACES.cpp
+++ b/src/OpenColorIO/transforms/builtins/ACES.cpp
@@ -5,8 +5,8 @@
 #include <cmath>
 
 #include <OpenColorIO/OpenColorIO.h>
+#include <Imath/half.h>
 
-#include "OpenEXR/half.h"
 #include "ops/fixedfunction/FixedFunctionOp.h"
 #include "ops/gradingrgbcurve/GradingRGBCurveOp.h"
 #include "ops/log/LogOp.h"
--- a/src/OpenColorIO/ops/range/RangeOp.cpp
+++ b/src/OpenColorIO/ops/range/RangeOp.cpp
@@ -5,9 +5,9 @@
 #include <sstream>
 
 #include <OpenColorIO/OpenColorIO.h>
+#include <Imath/half.h>
 
 #include "GpuShaderUtils.h"
-#include "OpenEXR/half.h"
 #include "HashUtils.h"
 #include "MathUtils.h"
 #include "ops/lut1d/Lut1DOpData.h"
--- a/src/apps/ocioconvert/main.cpp
+++ b/src/apps/ocioconvert/main.cpp
@@ -24,7 +24,8 @@ namespace OIIO = OIIO_NAMESPACE;
 #endif // OCIO_GPU_ENABLED
 
 #include "oiiohelpers.h"
-#include "OpenEXR/half.h"
+
+#include <Imath/half.h>
 
 
 // Array of non OpenColorIO arguments.
--- a/src/apps/ocioperf/main.cpp
+++ b/src/apps/ocioperf/main.cpp
@@ -9,9 +9,9 @@
 #if (OIIO_VERSION < 10100)
 namespace OIIO = OIIO_NAMESPACE;
 #endif
+#include <Imath/half.h>
 
 #include "apputils/argparse.h"
-#include "OpenEXR/half.h"
 #include "oiiohelpers.h"
 #include "utils/StringUtils.h"
 
--- a/share/cmake/modules/FindHalf.cmake
+++ b/share/cmake/modules/FindHalf.cmake
@@ -48,13 +48,13 @@ if(NOT OCIO_INSTALL_EXT_PACKAGES STREQUA
         # Find include directory
         find_path(Half_INCLUDE_DIR
             NAMES
-                OpenEXR/half.h
+                Imath/half.h
             HINTS
                 ${Half_ROOT}
                 ${PC_Half_INCLUDE_DIRS}
             PATH_SUFFIXES
                 include
-                OpenEXR/include
+                Imath/include
         )
 
         # Lib names to search for
@@ -82,6 +82,7 @@ if(NOT OCIO_INSTALL_EXT_PACKAGES STREQUA
             NAMES
                 ${_Half_STATIC_LIB_NAMES} 
                 ${_Half_LIB_NAMES}
+                Imath
             HINTS
                 ${Half_ROOT}
                 ${PC_Half_LIBRARY_DIRS}
