From f68e7d270bc32b38deda90b683f4ba070fc474c9 Mon Sep 17 00:00:00 2001
From: Ikey Doherty <ikey@solus-project.com>
Date: Thu, 16 Jul 2015 20:18:04 +0100
Subject: [PATCH] Use /usr/lib{64,32,x32} for linker locations on Solus (modern
 /usr)

Signed-off-by: Ikey Doherty <ikey@solus-project.com>
---
 gcc/config/i386/linux64.h | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/gcc/config/i386/linux64.h b/gcc/config/i386/linux64.h
index a90171e..d88347a 100644
--- a/gcc/config/i386/linux64.h
+++ b/gcc/config/i386/linux64.h
@@ -27,6 +27,6 @@ see the files COPYING3 and COPYING.RUNTIME respectively.  If not, see
 #define GNU_USER_LINK_EMULATION64 "elf_x86_64"
 #define GNU_USER_LINK_EMULATIONX32 "elf32_x86_64"
 
-#define GLIBC_DYNAMIC_LINKER32 "/lib/ld-linux.so.2"
-#define GLIBC_DYNAMIC_LINKER64 "/lib64/ld-linux-x86-64.so.2"
-#define GLIBC_DYNAMIC_LINKERX32 "/libx32/ld-linux-x32.so.2"
+#define GLIBC_DYNAMIC_LINKER32 "/usr/lib32/ld-linux.so.2"
+#define GLIBC_DYNAMIC_LINKER64 "/usr/lib64/ld-linux-x86-64.so.2"
+#define GLIBC_DYNAMIC_LINKERX32 "/usr/libx32/ld-linux-x32.so.2"
-- 
2.4.5

