--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -66,7 +66,9 @@ else()
     list(APPEND library-src src/gauss_transform_make_default.cpp)
 endif()
 
+include(GNUInstallDirs)
+
-add_library(Library-C++ STATIC ${library-src})
+add_library(Library-C++ SHARED ${library-src})
 set_target_properties(Library-C++ PROPERTIES
     OUTPUT_NAME cpd
     VERSION ${CPD_VERSION}
@@ -102,9 +104,9 @@ endif()
 
 
 # Install
-install(TARGETS Library-C++ DESTINATION lib EXPORT cpd-targets)
+install(TARGETS Library-C++ DESTINATION ${CMAKE_INSTALL_LIBDIR} EXPORT cpd-targets)
 install(DIRECTORY include/cpd DESTINATION include)
-install(EXPORT cpd-targets NAMESPACE Cpd:: DESTINATION lib/cmake/cpd)
+install(EXPORT cpd-targets NAMESPACE Cpd:: DESTINATION ${CMAKE_INSTALL_LIBDIR}/cmake/cpd)
 
 
 # Docs
