--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -904,8 +904,8 @@ if(WITH_FOLLY_DISTRIBUTED_MUTEX)
     third-party/folly/folly/synchronization/WaitOptions.cpp)
 endif()
 
-set(ROCKSDB_STATIC_LIB rocksdb${ARTIFACT_SUFFIX})
-set(ROCKSDB_SHARED_LIB rocksdb-shared${ARTIFACT_SUFFIX})
+set(ROCKSDB_STATIC_LIB rocksdb-static${ARTIFACT_SUFFIX})
+set(ROCKSDB_SHARED_LIB rocksdb${ARTIFACT_SUFFIX})
 
 option(ROCKSDB_BUILD_SHARED "Build shared versions of the RocksDB libraries" ON)
 
@@ -1005,14 +1005,6 @@ if(NOT WIN32 OR ROCKSDB_INSTALL_ON_WINDO
 
   install(DIRECTORY "${PROJECT_SOURCE_DIR}/cmake/modules" COMPONENT devel DESTINATION ${package_config_destination})
 
-  install(
-    TARGETS ${ROCKSDB_STATIC_LIB}
-    EXPORT RocksDBTargets
-    COMPONENT devel
-    ARCHIVE DESTINATION "${CMAKE_INSTALL_LIBDIR}"
-    INCLUDES DESTINATION "${CMAKE_INSTALL_INCLUDEDIR}"
-  )
-
   if(ROCKSDB_BUILD_SHARED)
     install(
       TARGETS ${ROCKSDB_SHARED_LIB}
