diff --git a/src/libs/ui/mainwindow.cpp b/src/libs/ui/mainwindow.cpp
index d217c54..56ce048 100644
--- a/src/libs/ui/mainwindow.cpp
+++ b/src/libs/ui/mainwindow.cpp
@@ -194,7 +194,7 @@ MainWindow::MainWindow(Core::Application *app, QWidget *parent)
     ui->splitter->restoreState(m_settings->verticalSplitterGeometry);
 
     // Setup web settings.
-    auto webSettings = new Browser::Settings(m_settings, this);
+    new Browser::Settings(m_settings, this);
 
     // Setup web bridge.
     m_webBridge = new Browser::WebBridge(this);