diff --git a/src/libs/ui/docsetsdialog.cpp b/src/libs/ui/docsetsdialog.cpp
index 8c9fe63..bc35375 100644
--- a/src/libs/ui/docsetsdialog.cpp
+++ b/src/libs/ui/docsetsdialog.cpp
@@ -360,8 +360,8 @@ void DocsetsDialog::downloadCompleted()
         QTemporaryFile *tmpFile = m_tmpFiles[docsetName];
         if (!tmpFile) {
             tmpFile = new QTemporaryFile(QStringLiteral("%1/%2.XXXXXX.tmp").arg(Core::Application::cacheLocation(), docsetName), this);
-            tmpFile->open();
-            m_tmpFiles.insert(docsetName, tmpFile);
+            if (tmpFile->open())
+                m_tmpFiles.insert(docsetName, tmpFile);
         }
 
         while (reply->bytesAvailable()) {
@@ -403,8 +403,8 @@ void DocsetsDialog::downloadProgress(qint64 received, qint64 total)
         QTemporaryFile *tmpFile = m_tmpFiles[docsetName];
         if (!tmpFile) {
             tmpFile = new QTemporaryFile(QStringLiteral("%1/%2.XXXXXX.tmp").arg(Core::Application::cacheLocation(), docsetName), this);
-            tmpFile->open();
-            m_tmpFiles.insert(docsetName, tmpFile);
+            if (tmpFile->open())
+                m_tmpFiles.insert(docsetName, tmpFile);
         }
 
         tmpFile->write(reply->read(received));
diff --git a/src/libs/ui/qxtglobalshortcut/CMakeLists.txt b/src/libs/ui/qxtglobalshortcut/CMakeLists.txt
index 476c2f5..e88ce73 100644
--- a/src/libs/ui/qxtglobalshortcut/CMakeLists.txt
+++ b/src/libs/ui/qxtglobalshortcut/CMakeLists.txt
@@ -29,7 +29,7 @@ endif()
 
 add_library(QxtGlobalShortcut STATIC ${QxtGlobalShortcut_SOURCES})
 
-find_package(Qt${QT_VERSION_MAJOR} COMPONENTS Gui REQUIRED)
+find_package(Qt${QT_VERSION_MAJOR} COMPONENTS Gui GuiPrivate REQUIRED)
 target_link_libraries(QxtGlobalShortcut Qt${QT_VERSION_MAJOR}::Gui)
 
 if(APPLE)
