From af4708dd5973242843c1c394c46b1274c73b8640 Mon Sep 17 00:00:00 2001
From: Tim Klocke <taaem@mailbox.org>
Date: Tue, 13 Mar 2018 15:42:43 -0300
Subject: [PATCH] Don't override the default workspace switching shortcut

---
 app/localShortcuts.js | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/app/localShortcuts.js b/app/localShortcuts.js
index 2f12a0fa3..657917ef9 100644
--- a/app/localShortcuts.js
+++ b/app/localShortcuts.js
@@ -18,8 +18,8 @@ module.exports.register = (win) => {
     ['F6', messages.SHORTCUT_FOCUS_URL],
     ['CmdOrCtrl+Shift+]', messages.SHORTCUT_NEXT_TAB],
     ['CmdOrCtrl+Shift+[', messages.SHORTCUT_PREV_TAB],
-    ['CmdOrCtrl+Alt+Right', messages.SHORTCUT_NEXT_TAB],
-    ['CmdOrCtrl+Alt+Left', messages.SHORTCUT_PREV_TAB],
+    ['CmdOrCtrl+Right', messages.SHORTCUT_NEXT_TAB],
+    ['CmdOrCtrl+Left', messages.SHORTCUT_PREV_TAB],
     ['Ctrl+PageDown', messages.SHORTCUT_NEXT_TAB],
     ['Ctrl+PageUp', messages.SHORTCUT_PREV_TAB],
     ['CmdOrCtrl+9', messages.SHORTCUT_SET_ACTIVE_FRAME_TO_LAST],
-- 
2.16.2

