commit 00a1024fc93978330f94e7346135a7feec8c0241
Author: David Faure <faure@kde.org>
Date:   Fri May 3 19:40:20 2019 +0200

    Fix compilation after Collection::referenced was removed from Akonadi

diff --git a/tests/testlib/akonadifakestorage.cpp b/tests/testlib/akonadifakestorage.cpp
index e64c363..8fe1258 100644
--- a/tests/testlib/akonadifakestorage.cpp
+++ b/tests/testlib/akonadifakestorage.cpp
@@ -508,7 +508,7 @@ Akonadi::Collection::List AkonadiFakeStorage::collectChildren(const Akonadi::Col
     auto collections = Akonadi::Collection::List();
 
     foreach (const auto &child, m_data->childCollections(findId(root))) {
-        if (child.enabled() || child.referenced())
+        if (child.enabled())
             collections << m_data->collection(findId(child));
         collections += collectChildren(child);
     }
