From 567523cec634405db7cd4e151518778a485277db Mon Sep 17 00:00:00 2001
From: Christophe Giboudeaux <christophe@krop.fr>
Date: Fri, 11 Oct 2019 10:29:17 +0200
Subject: [PATCH] Look for AkonadiContact

---
 CMakeLists.txt | 1 +
 1 file changed, 1 insertion(+)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 0b639c4..2f13e28 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -67,6 +67,7 @@ include_directories (
 
 find_package(KF5 REQUIRED COMPONENTS
     AkonadiCalendar
+    AkonadiContact
     AkonadiNotes
     AkonadiSearch
     IdentityManagement
-- 
2.23.0

