--- zanshin-0.4.1/CMakeLists.txt	2016-09-06 16:21:19.000000000 +0200
+++ zanshin-0.4.1.new/CMakeLists.txt	2016-11-21 14:37:27.169696650 +0100
@@ -16,6 +16,21 @@ include(ECMMarkAsTest)
 include(ECMPoQmTools)
 
 find_package(Qt5 ${QT_REQUIRED_VERSION} CONFIG REQUIRED Core Gui Widgets Qml Test)
+find_package(KF5 REQUIRED COMPONENTS
+    AkonadiCalendar
+    AkonadiContact
+    AkonadiNotes
+    AkonadiSearch
+    IdentityManagement
+    KontactInterface
+    Ldap
+    Runner
+    Wallet
+    WindowSystem
+    I18n
+    KDELibs4Support
+)
+find_package(KF5Akonadi "5.1" CONFIG REQUIRED)
 find_package(Boost REQUIRED)
 macro(assert_min_ver version)
     set(error_msg "${CMAKE_CXX_COMPILER} ${CMAKE_CXX_COMPILER_VERSION} not supported")
@@ -64,21 +75,6 @@ include_directories (
     3rdparty/kdepim/libkdepim/
 )
 
-find_package(KF5 REQUIRED COMPONENTS
-    AkonadiCalendar
-    AkonadiNotes
-    AkonadiSearch
-    IdentityManagement
-    KontactInterface
-    Ldap
-    Runner
-    Wallet
-    WindowSystem
-    I18n
-    KDELibs4Support
-)
-find_package(KF5Akonadi "5.1" CONFIG REQUIRED)
-
 add_subdirectory(3rdparty)
 add_subdirectory(src)
