diff -rupN qmapshack-1.11.1/src/qmapshack/helpers/CToolBarSetupDialog.cpp qmapshack-1.11.1-new/src/qmapshack/helpers/CToolBarSetupDialog.cpp
--- qmapshack-1.11.1/src/qmapshack/helpers/CToolBarSetupDialog.cpp	2018-04-10 13:32:15.000000000 +0200
+++ qmapshack-1.11.1-new/src/qmapshack/helpers/CToolBarSetupDialog.cpp	2018-07-01 21:20:59.462555036 +0200
@@ -20,6 +20,8 @@
 #include "helpers/CToolBarConfig.h"
 #include "helpers/CToolBarSetupDialog.h"
 
+#include <QAction>
+
 bool CToolBarSetupDialog::CItemFilter::shouldBeMoved(QListWidgetItem *item)
 {
     CDialogItem * dialogItem = dynamic_cast<CDialogItem *>(item);
