From 58d250bc4569fd2ebd08122042f0222f54949d28 Mon Sep 17 00:00:00 2001
From: Reilly Brogan <reilly@reillybrogan.com>
Date: Sat, 10 Feb 2024 12:46:03 -0600
Subject: [PATCH] Allow Solus system libenet

---
 CMakeLists.txt | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 5ca6b1282c..e0a87e9974 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -659,7 +659,8 @@ endif()
 
 dolphin_find_optional_system_library(pugixml Externals/pugixml)
 
-dolphin_find_optional_system_library_pkgconfig(ENET libenet>=1.3.18 enet::enet Externals/enet)
+# System libenet on Solus has been patched for ENET_SOCKOPT_TTL support
+dolphin_find_optional_system_library_pkgconfig(ENET libenet>=1.3.17 enet::enet Externals/enet)
 
 dolphin_find_optional_system_library_pkgconfig(xxhash libxxhash>=0.8.2 xxhash::xxhash Externals/xxhash)
 
-- 
2.43.0

