From d1f6636f6d695febb0b44ccf030cc5b59ea68dc1 Mon Sep 17 00:00:00 2001
From: Joshua Strobl <joshua@stroblindustries.com>
Date: Sat, 13 Apr 2019 20:21:09 +0300
Subject: [PATCH 1/1] Test removing spidermonkey validation.

---
 configure.ac | 34 ----------------------------------
 1 file changed, 34 deletions(-)

diff --git a/configure.ac b/configure.ac
index 9e012ba..581f87d 100644
--- a/configure.ac
+++ b/configure.ac
@@ -80,40 +80,6 @@ CPPFLAGS="$GJS_CFLAGS"
 LIBS_save="$LIBS"
 LIBS="$GJS_LIBS"
 
-dnl If SpiderMonkey was compiled with --enable-debug, then we need to define
-dnl -DDEBUG before including js-config.h.
-AC_MSG_CHECKING([whether SpiderMonkey was configured with --enable-debug])
-AC_COMPILE_IFELSE([AC_LANG_SOURCE([[
-#include <js-config.h>
-#ifdef JS_DEBUG
-#error debug yes, if we did not already error out due to DEBUG not being defined
-#endif
-]])], [
-  AC_MSG_RESULT([no])
-  GJS_DEBUG_DEF=
-], [
-  AC_DEFINE([HAVE_DEBUG_SPIDERMONKEY], [1],
-    [Define to 1 if SpiderMonkey was compiled with --enable-debug])
-  GJS_DEBUG_DEF='#define DEBUG 1'
-  AC_MSG_RESULT([yes])
-])
-
-# Sanity check for SpiderMonkey
-AC_MSG_CHECKING([if a minimal SpiderMonkey program compiles and runs])
-AC_RUN_IFELSE([AC_LANG_PROGRAM([
-$GJS_DEBUG_DEF
-#include <js/Initialization.h>
-  ], [
-if (!JS_Init()) exit(1);
-JS_ShutDown();
-  ])],
-  [AC_MSG_RESULT([yes])],
-  [AC_MSG_RESULT([no])
-    AC_MSG_ERROR([Check if you are using the recommended configuration for
-SpiderMonkey. See the file org.gnome.Sdk.json.in in
-https://gitlab.gnome.org/GNOME/gnome-sdk-images])],
-  [AC_MSG_RESULT([cross-compiling, unable to determine])])
-
 LIBS="$LIBS_save"
 CPPFLAGS="$CPPFLAGS_save"
 
-- 
2.21.0

