--- gjs-1.48.0/gjs/jsapi-util.cpp~	2017-04-29 12:55:17.000000000 +0000
+++ gjs-1.48.0/gjs/jsapi-util.cpp	2017-04-29 13:18:46.049215763 +0000
@@ -810,7 +810,7 @@
          * other hand, if swapping is going on, better
          * to GC.
          */
-        if (rss_size > linux_rss_trigger) {
+        if (rss_size > linux_rss_trigger || now - last_gc_check_time > 500 * 16666) {
             linux_rss_trigger = (gulong) MIN(G_MAXULONG, rss_size * 1.25);
             JS_GC(JS_GetRuntime(context));
             last_gc_time = now;
