From bd695e77fcbb0198afd21f25e77672fb0685d98b Mon Sep 17 00:00:00 2001
From: Joshua Strobl <joshua@stroblindustries.com>
Date: Sat, 20 Apr 2019 20:29:17 +0300
Subject: [PATCH 1/1] Fix zsh completion path

---
 quodlibet/gdist/zsh_completions.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/quodlibet/gdist/zsh_completions.py b/quodlibet/gdist/zsh_completions.py
index fd959db..45b5782 100644
--- a/quodlibet/gdist/zsh_completions.py
+++ b/quodlibet/gdist/zsh_completions.py
@@ -45,7 +45,7 @@ class install_zsh_completions(Command):
 
     def run(self):
         basepath = os.path.join(
-            self.install_dir, 'share', 'zsh', 'vendor-completions')
+            self.install_dir, 'share', 'zsh', 'site-functions')
         out = self.mkpath(basepath)
         self.outfiles.extend(out or [])
         for src, dest in self.zsh_completions:
-- 
2.21.0

