From 7d40b82a63c600a8e56a65be8d70b8e889ac174d Mon Sep 17 00:00:00 2001
From: Pierre-Yves <pyu@riseup.net>
Date: Fri, 2 Nov 2018 21:51:31 +0100
Subject: [PATCH] Fix zsh completion path

Signed-off-by: Pierre-Yves <pyu@riseup.net>
---
 gdist/zsh_completions.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/gdist/zsh_completions.py b/gdist/zsh_completions.py
index fd959db..45b5782 100644
--- a/gdist/zsh_completions.py
+++ b/gdist/zsh_completions.py
@@ -45,7 +45,7 @@ class install_zsh_completions(Command):
 
     def run(self):
         basepath = os.path.join(
-            self.install_dir, 'share', 'zsh', 'vendor-completions')
+            self.install_dir, 'share', 'zsh', 'site-functions')
         out = self.mkpath(basepath)
         self.outfiles.extend(out or [])
         for src, dest in self.zsh_completions:
-- 
2.19.1

