From 7678dc2bcd0e8c0a4e21168a7f1c76bdf55ae246 Mon Sep 17 00:00:00 2001
From: Dirk Mueller <dmueller@suse.com>
Date: Tue, 2 Aug 2022 05:38:53 +0200
Subject: [PATCH] Set soversion number to project version (#708)

---
 src/CMakeLists.txt | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
index 11d14e4b..c0a0b10c 100644
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -132,9 +132,9 @@ set_target_properties(
     OUTPUT_NAME
       opencc
     VERSION
-      1.1.3
+      ${OPENCC_VERSION_MAJOR}.${OPENCC_VERSION_MINOR}.${OPENCC_VERSION_REVISION}
     SOVERSION
-      1.1
+      ${OPENCC_VERSION_MAJOR}.${OPENCC_VERSION_MINOR}
 )
 
 # Installation
