From bf7dfe5f8bebd5b4980ac02f12c0634c105617aa Mon Sep 17 00:00:00 2001
From: Peter O'Connor <peter@solus-project.com>
Date: Tue, 12 Jun 2018 18:57:34 +1000
Subject: [PATCH] Disable building unshipped code

Signed-off-by: Peter O'Connor <peter@solus-project.com>
---
 CMakeLists.txt | 1 -
 1 file changed, 1 deletion(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 2a56e9f..0002ca1 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -50,7 +50,6 @@ add_subdirectory(src)
 # Enable unit testing
 if (BUILD_TESTING)
     add_subdirectory(autotests)
-    add_subdirectory(tests)
 endif ()
 
 # Create a Config.cmake and a ConfigVersion.cmake file and install them
-- 
2.17.1

