From 59e423ea39088cd46f1903b1094127d8fda28d73 Mon Sep 17 00:00:00 2001
From: Troy Harvey <harvey@getsol.us>
Date: Sun, 2 Aug 2026 12:30:45 +1000
Subject: [PATCH] operations: Add modules.softdep to ignore list

Signed-off-by: Troy Harvey <harvey@getsol.us>
---
 pisi/operations/check.py | 1 +
 1 file changed, 1 insertion(+)

diff --git a/pisi/operations/check.py b/pisi/operations/check.py
index 51162d47..c7a728a8 100644
--- a/pisi/operations/check.py
+++ b/pisi/operations/check.py
@@ -33,6 +33,7 @@ def file_corrupted(pfile):
     "modules.dep.bin",
     "modules.symbols",
     "modules.symbols.bin",
+    "modules.softdep",
 ]
 
 _top_level_dirs = [
