From 870feab44a5abcf58ba58990ce2f3610a89d48ed Mon Sep 17 00:00:00 2001
From: Joshua Strobl <joshua@streambits.io>
Date: Thu, 30 Jul 2020 14:10:48 +0300
Subject: [PATCH 1/1] Use a stateless configuration

---
 src/Makefile.am | 2 +-
 src/rcfile.c    | 9 +++++++--
 2 files changed, 8 insertions(+), 3 deletions(-)

diff --git a/src/Makefile.am b/src/Makefile.am
index c6098cb..bc393fe 100644
--- a/src/Makefile.am
+++ b/src/Makefile.am
@@ -1,5 +1,5 @@
 AM_CPPFLAGS = -DLOCALEDIR=\"$(localedir)\" -DSYSCONFDIR=\"$(sysconfdir)\" \
-	-I$(top_builddir)/lib -I$(top_srcdir)/lib
+	-DDATADIR=\"$(datadir)\" -I$(top_builddir)/lib -I$(top_srcdir)/lib
 
 CLEANFILES = revision.h
 
diff --git a/src/rcfile.c b/src/rcfile.c
index 733db37..443df9e 100644
--- a/src/rcfile.c
+++ b/src/rcfile.c
@@ -1684,8 +1684,13 @@ void do_rcfiles(void)
 		nanorc = get_full_path(custom_nanorc);
 		if (access(nanorc, F_OK) != 0)
 			die(_("Specified rcfile does not exist\n"));
-	} else
-		nanorc = mallocstrcpy(nanorc, SYSCONFDIR "/nanorc");
+	} else {
+		if (access(SYSCONFDIR "/nanorc", F_OK) == 0) {
+			nanorc = mallocstrcpy(nanorc, SYSCONFDIR "/nanorc");
+		} else {
+			nanorc = mallocstrcpy(nanorc, DATADIR "/defaults/nano/nanorc");
+		}
+	}
 
 	if (is_good_file(nanorc))
 		parse_one_nanorc();
-- 
2.27.0

