From 58c6a817525a25b0259c2d8e3617d6fd67435d48 Mon Sep 17 00:00:00 2001
From: Michael Meinertzhagen <mmeinert@protonmail.com>
Date: Sat, 7 Jan 2017 13:08:27 +0100
Subject: [PATCH 1/1] Use a stateless configuration

---
 src/Makefile.am | 2 +-
 src/rcfile.c    | 6 +++++-
 2 files changed, 6 insertions(+), 2 deletions(-)

diff --git a/src/Makefile.am b/src/Makefile.am
index 22baa16f..11ec9e7b 100644
--- a/src/Makefile.am
+++ b/src/Makefile.am
@@ -1,4 +1,4 @@
-AM_CPPFLAGS = -DLOCALEDIR=\"$(localedir)\" -DSYSCONFDIR=\"$(sysconfdir)\"
+AM_CPPFLAGS = -DLOCALEDIR=\"$(localedir)\" -DSYSCONFDIR=\"$(sysconfdir)\" -DDATADIR=\"$(datadir)\"
 
 CLEANFILES = revision.h
 
diff --git a/src/rcfile.c b/src/rcfile.c
index a9426d22..8ec59112 100644
--- a/src/rcfile.c
+++ b/src/rcfile.c
@@ -1258,7 +1258,11 @@ void parse_one_nanorc(void)
 /* First read the system-wide rcfile, then the user's rcfile. */
 void do_rcfiles(void)
 {
-    nanorc = mallocstrcpy(nanorc, SYSCONFDIR "/nanorc");
+    if (access(SYSCONFDIR "/nanorc", F_OK) == 0) {
+		nanorc = mallocstrcpy(nanorc, SYSCONFDIR "/nanorc");
+    } else {
+		nanorc = mallocstrcpy(nanorc, DATADIR "/defaults/nano/nanorc");
+    }
 
     /* Process the system-wide nanorc. */
     parse_one_nanorc();
-- 
2.11.0

