From 4ac2719e63f679875decc8847aa13d778d08a484 Mon Sep 17 00:00:00 2001
From: Ikey Doherty <ikey@solus-project.com>
Date: Sat, 12 Mar 2016 15:28:09 +0000
Subject: [PATCH] Use a stateless configuration

Signed-off-by: Ikey Doherty <ikey@solus-project.com>
---
 src/Makefile.am | 2 +-
 src/rcfile.c    | 6 +++++-
 2 files changed, 6 insertions(+), 2 deletions(-)

diff --git a/src/Makefile.am b/src/Makefile.am
index df65845..181db1a 100644
--- a/src/Makefile.am
+++ b/src/Makefile.am
@@ -1,4 +1,4 @@
-AM_CPPFLAGS = -DLOCALEDIR=\"$(localedir)\" -DSYSCONFDIR=\"$(sysconfdir)\"
+AM_CPPFLAGS = -DLOCALEDIR=\"$(localedir)\" -DSYSCONFDIR=\"$(sysconfdir)\" -DDATADIR=\"$(datadir)\"
 
 bin_PROGRAMS = 	nano
 nano_SOURCES =	browser.c \
diff --git a/src/rcfile.c b/src/rcfile.c
index 7c10b3d..8662683 100644
--- a/src/rcfile.c
+++ b/src/rcfile.c
@@ -1375,7 +1375,11 @@ void do_rcfile(void)
     struct stat rcinfo;
     FILE *rcstream;
 
-    nanorc = mallocstrcpy(nanorc, SYSCONFDIR "/nanorc");
+    if (access(SYSCONFDIR "/nanorc", F_OK) == 0) {
+		nanorc = mallocstrcpy(nanorc, SYSCONFDIR "/nanorc");
+    } else {
+		nanorc = mallocstrcpy(nanorc, DATADIR "defaults/nano/nanorc");
+    }
 
     /* Don't open directories, character files, or block files. */
     if (stat(nanorc, &rcinfo) != -1) {
-- 
2.7.1

