From 0d58ce1cb19d31c44a6edea61efc669108605445 Mon Sep 17 00:00:00 2001
From: Peter O'Connor <peter@solus-project.com>
Date: Thu, 5 Jul 2018 21:25:31 +1000
Subject: [PATCH] Point dhclient-script path to where it's installed

Signed-off-by: Peter O'Connor <peter@solus-project.com>
---
 includes/dhcpd.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/includes/dhcpd.h b/includes/dhcpd.h
index 5930e6a..3cd2d10 100644
--- a/includes/dhcpd.h
+++ b/includes/dhcpd.h
@@ -1568,7 +1568,7 @@ typedef unsigned char option_mask [16];
 #endif
 
 #ifndef _PATH_DHCLIENT_SCRIPT
-#define _PATH_DHCLIENT_SCRIPT	"/sbin/dhclient-script"
+#define _PATH_DHCLIENT_SCRIPT	"/usr/bin/dhclient-script"
 #endif
 
 #ifndef _PATH_DHCLIENT_PID
-- 
2.18.0

