From e5a39bd81c6cf14fcd404ab8deaa06eb739fe32f Mon Sep 17 00:00:00 2001
From: Joey Riches <josephriches@gmail.com>
Date: Mon, 11 Jan 2021 18:26:29 +0000
Subject: [PATCH 1/1] fixup path

---
 docs/manpages/meson.build | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/docs/manpages/meson.build b/docs/manpages/meson.build
index 0b8a648..831d085 100644
--- a/docs/manpages/meson.build
+++ b/docs/manpages/meson.build
@@ -43,7 +43,7 @@ foreach m : daemon_manpages + cli_manpages
     command: [xsltproc,
               '--output', '@OUTPUT@',
               '--stringparam', 'man.authors.section.enabled', '0',
-              '/etc/asciidoc/docbook-xsl/manpage.xsl', '@INPUT@'],
+              '/usr/share/asciidoc/docbook-xsl/manpage.xsl', '@INPUT@'],
     input: xml,
     output: manpage,
     install: true,
-- 
2.30.0

