From 5788fb32f75a9425c9adec469bb2cfd21be8ea9d Mon Sep 17 00:00:00 2001
From: Pierre-Yves <pyu@riseup.net>
Date: Mon, 6 Nov 2017 17:35:24 +0100
Subject: [PATCH] fix VERSION_TAG compilation problem

Signed-off-by: Pierre-Yves <pyu@riseup.net>
---
 src/Makefile | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/Makefile b/src/Makefile
index 5519ab9..d588e2e 100644
--- a/src/Makefile
+++ b/src/Makefile
@@ -127,7 +127,7 @@ COMPTIME                := $(shell date +%s)
 # the value will be something like this: "tag: vX.Y.Z, refs/pull/Y/head"
 
 VERSION_EXPORT          := tag: v4.0.0
-VERSION_TAG             := $(shell test -d .git && git describe --tags --dirty=+ || echo "$(VERSION_EXPORT)" | $(SED) 's/.*: v\([\.0-9]*\),.*/v\1/')
+VERSION_TAG             := $(shell test -d .git && git describe --tags --dirty=+ || echo "$(VERSION_EXPORT)" | $(SED) 's/.*: v\([\.0-9]*\).*/v\1/')
 
 ##
 ## General compiler and linker options
-- 
2.15.0

