From 40331fb3879fa682505a86601862a5afce747ced Mon Sep 17 00:00:00 2001
From: "F. von Gellhorn" <flinux@vongellhorn.ch>
Date: Tue, 28 Jan 2020 22:56:45 +0100
Subject: [PATCH 1/1] Set breeze-dark as default gtk3 theme

---
 kconf_update/main.cpp | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/kconf_update/main.cpp b/kconf_update/main.cpp
index 479082d..893ef91 100644
--- a/kconf_update/main.cpp
+++ b/kconf_update/main.cpp
@@ -180,13 +180,14 @@ int setGtk3()
     out << QStringLiteral("[Settings]\n");
     out << QStringLiteral("gtk-cursor-theme-name=breeze_cursors\n");
     out << QStringLiteral("gtk-font-name=Noto Sans 10\n"); // matches plasma-workspace:startkde/startkde.cmake
-    out << QStringLiteral("gtk-theme-name=")+gtk3Theme+QStringLiteral("\n");
+    out << QStringLiteral("gtk-theme-name=Breeze-Dark\n");
     out << QStringLiteral("gtk-icon-theme-name=breeze\n");
     out << QStringLiteral("gtk-fallback-icon-theme=gnome\n");
     out << QStringLiteral("gtk-toolbar-style=GTK_TOOLBAR_ICONS\n");
     out << QStringLiteral("gtk-menu-images=1\n");
     out << QStringLiteral("gtk-button-images=1\n");
     out << QStringLiteral("gtk-primary-button-warps-slider=0\n");
+    out << QStringLiteral("gtk-application-prefer-dark-theme=false\n");
     qCDebug(GTKBREEZE) << "gtk3rc written";
 
     return 0;
-- 
2.25.0

