From dbfbfca64763e8038da5da53a30a6ac58fa4b02a Mon Sep 17 00:00:00 2001
From: "F. von Gellhorn" <flinux@vongellhorn.ch>
Date: Mon, 27 Jan 2020 23:19:34 +0100
Subject: [PATCH 1/1] Set Breeze-Dark as default gtk3 theme

---
 kconf_update/main.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/kconf_update/main.cpp b/kconf_update/main.cpp
index 479082d..bf3d139 100644
--- a/kconf_update/main.cpp
+++ b/kconf_update/main.cpp
@@ -152,7 +152,7 @@ int setGtk2()
  */
 int setGtk3()
 {
-    const QString gtk3Theme = QStringLiteral("Breeze");
+    const QString gtk3Theme = QStringLiteral("Breeze-Dark");
 
     if (!isGTK3ThemeInstalled(gtk3Theme)) {
         qCDebug(GTKBREEZE) << "not found, quitting";
-- 
2.25.0

