From 71013dcd82466e2958ed8945b7a7bc276084e0fd Mon Sep 17 00:00:00 2001
From: Heiko Becker <heiko.becker@kde.org>
Date: Wed, 6 Mar 2024 01:52:39 +0100
Subject: [PATCH] Don't require a non-existent ECM version

---
 CMakeLists.txt | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 8f86c54..8e63b6e 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -4,7 +4,7 @@ project(breeze-gtk)
 set(PROJECT_VERSION "6.0.1")
 set(PROJECT_VERSION_MAJOR 6)
 
-find_package(ECM 6.0.1 REQUIRED NO_MODULE)
+find_package(ECM 6.0.0 REQUIRED NO_MODULE)
 include(FeatureSummary)
 
 set(CMAKE_MODULE_PATH ${ECM_MODULE_PATH}  "${CMAKE_SOURCE_DIR}/cmake")
-- 
GitLab

