From b3d42657c545b5854729a9dd9dd9a18d67cdc5cb Mon Sep 17 00:00:00 2001
From: Peter O'Connor <peter@solus-project.com>
Date: Thu, 2 Nov 2017 22:23:16 +1100
Subject: [PATCH] Fix desktop icon location and rm 'snap' binary

Signed-off-by: Peter O'Connor <peter@solus-project.com>
---
 Makefile       | 1 -
 fotoxx.desktop | 2 +-
 2 files changed, 1 insertion(+), 2 deletions(-)

diff --git a/Makefile b/Makefile
index cd75b94..5af6eec 100644
--- a/Makefile
+++ b/Makefile
@@ -97,7 +97,6 @@ install: fotoxx uninstall
 	mkdir -p  $(DESTDIR)$(PREFIX)/share/applications
 	mkdir -p  $(DESTDIR)$(APPDATADIR)
 	cp -f  fotoxx $(DESTDIR)$(BINDIR)
-	cp -f  fotoxx $(DESTDIR)$(BINDIR)/fotoxx-snap            ### snap package name
 	cp -f -R  data/* $(DESTDIR)$(DATADIR)
 	cp -f -R  icons/* $(DESTDIR)$(ICONDIR)
 	cp -f -R  locales/* $(DESTDIR)$(LOCALESDIR)
diff --git a/fotoxx.desktop b/fotoxx.desktop
index adb56dc..fc11d1a 100644
--- a/fotoxx.desktop
+++ b/fotoxx.desktop
@@ -8,7 +8,7 @@ Type=Application
 Terminal=false
 MimeType=image/bmp;image/gif;image/tiff;image/jpeg;image/png;
 Exec=fotoxx %f
-Icon=/usr/share/icons/fotoxx.png
+Icon=/usr/share/fotoxx/icons/fotoxx.png
 
 GenericName[de]=Foto Bearbeitung
 GenericName[fr]=Gestionnaire de photos
--
2.15.0
