From 1de008f2d5e5de28fd017e8712e016ffde4ad096 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Florian=20M=C3=BCllner?= <fmuellner@gnome.org>
Date: Sun, 8 Mar 2020 01:21:49 +0100
Subject: [PATCH] Bump extensions-tool version as well

---
 subprojects/extensions-tool/meson.build | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/subprojects/extensions-tool/meson.build b/subprojects/extensions-tool/meson.build
index ca7e67ce22..bca8a8a367 100644
--- a/subprojects/extensions-tool/meson.build
+++ b/subprojects/extensions-tool/meson.build
@@ -1,5 +1,5 @@
 project('gnome-extensions-tool', 'c',
-  version: '3.35.1',
+  version: '3.36.0',
   meson_version: '>= 0.47.0',
   license: 'GPLv2+'
 )
-- 
2.24.1

