From 6f9a23e08a5c55bcf4a6fca305ed64cd0e927cda Mon Sep 17 00:00:00 2001
From: Joey Riches <josephriches@gmail.com>
Date: Fri, 21 Apr 2023 14:08:32 +0100
Subject: [PATCH 1/1] XDG dirs

---
 src/chatty/Chatty.java | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/chatty/Chatty.java b/src/chatty/Chatty.java
index b815917..dd3cfd3 100644
--- a/src/chatty/Chatty.java
+++ b/src/chatty/Chatty.java
@@ -212,7 +212,7 @@ public class Chatty {
     // Paths
     //--------------------------
     public enum PathType {
-        SETTINGS(() -> Paths.get(System.getProperty("user.home"), ".chatty"), null),
+        SETTINGS(() -> Paths.get(System.getProperty("user.home"), ".config", "chatty"), null),
         WORKING(() -> Paths.get(System.getProperty("user.dir")), null),
         BACKUP(() -> getUserDataDirectory().resolve("backup"), null),
         IMAGE(() -> getWorkingDirectory().resolve("img"), "imgPath"),
-- 
2.39.0

