From 80dd666671e480b77e9bd9d81b3f4b0cdb21100a Mon Sep 17 00:00:00 2001
From: Joey Riches <josephriches@gmail.com>
Date: Sat, 6 Jan 2018 13:05:04 +0000
Subject: [PATCH 1/1] Set proper user directory for unix

---
 src/chatty/Chatty.java | 6 ++++--
 1 file changed, 4 insertions(+), 2 deletions(-)

diff --git a/src/chatty/Chatty.java b/src/chatty/Chatty.java
index b1e4a6c..c3b12e8 100644
--- a/src/chatty/Chatty.java
+++ b/src/chatty/Chatty.java
@@ -200,8 +200,10 @@ public class Chatty {
             return settingsDir + File.separator;
         }
         String dir = System.getProperty("user.home")
-                + File.separator 
-                + ".chatty" 
+                + File.separator
+                + ".config"
+                + File.separator
+                + "chatty"
                 + File.separator;
         new File(dir).mkdirs();
         return dir;
-- 
2.15.1

