From 16b8ad49aef2183b50c4d73e2045c24331cf7b61 Mon Sep 17 00:00:00 2001
From: Ivan Trepakov <liontiger23@gmail.com>
Date: Sat, 16 May 2026 14:37:52 +0700
Subject: [PATCH 2/2] Upgrade shadow to 8.3.7 which supports Gradle 9

---
 build.gradle | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/build.gradle b/build.gradle
index d52ae958..27b6ec8b 100644
--- a/build.gradle
+++ b/build.gradle
@@ -1,7 +1,7 @@
 import java.util.regex.Matcher
 
 plugins {
-    id 'com.github.johnrengelman.shadow' version '8.1.1'
+    id 'com.gradleup.shadow' version '8.3.7'
     id "org.gradle.crypto.checksum" version "1.4.0"
     id "java"
 }
-- 
2.54.0

