From 496e59e5d77bdcd23b361d1a39424dc6d1bc6031 Mon Sep 17 00:00:00 2001
From: Pierre-Yves <pyu@riseup.net>
Date: Tue, 23 Apr 2019 19:49:04 +0200
Subject: [PATCH] Patch qomui for stateless iproute2

Signed-off-by: Pierre-Yves <pyu@riseup.net>
---
 qomui/bypass.py | 6 +++++-
 1 file changed, 5 insertions(+), 1 deletion(-)

diff --git a/qomui/bypass.py b/qomui/bypass.py
index bb90b19..daa9382 100755
--- a/qomui/bypass.py
+++ b/qomui/bypass.py
@@ -41,7 +41,11 @@ def create_cgroup(user, group, interface, gw=None,  gw_6=None, default_int=None)
         logging.debug("Bypass: No routing table added - table bypass_qomui already exists")
 
     except CalledProcessError:
-        with open("/etc/iproute2/rt_tables", "a") as rt_tables:
+        ip2_tables = "/etc/iproute2/rt_tables"
+        if not os.path.isfile(ip2_tables):
+            ip2_tables = "/usr/share/defaults/iproute2/rt_tables"
+
+        with open(ip2_tables, "a") as rt_tables:
             rt_tables.write("11 bypass_qomui\n")
         logging.debug("Bypass: Created new routing table")
 
-- 
2.21.0

