From ad205c48568e56eca7e7bfdc58e69ad937ca3064 Mon Sep 17 00:00:00 2001
From: arvindcheru <90783369+arvindcheru@users.noreply.github.com>
Date: Mon, 3 Apr 2023 12:18:21 -0400
Subject: [PATCH] Fix to enable library target softlinks in hipfft package

---
 library/CMakeLists.txt | 10 +++++-----
 1 file changed, 5 insertions(+), 5 deletions(-)

diff --git a/library/CMakeLists.txt b/library/CMakeLists.txt
index e797694..3e38ca7 100644
--- a/library/CMakeLists.txt
+++ b/library/CMakeLists.txt
@@ -49,11 +49,6 @@ set( hipfft_headers_public
 
 source_group( "Header Files\\Public" FILES ${hipfft_headers_public} )
 
-# nvcc can not recognize shared libraray file name with suffix other than *.so when linking.
-#if (NOT BUILD_WITH_COMPILER STREQUAL "HIP-NVCC")
-#  rocm_setsoversion(hipfft ${hipfft_SOVERSION})
-#endif()
-
 # Include sources
 include( src/CMakeLists.txt )
 
@@ -121,6 +116,11 @@ if( NOT BUILD_WITH_COMPILER STREQUAL "HIP-NVCC" )
     PROPERTIES CXX_VISIBILITY_PRESET "hidden" VISIBILITY_INLINES_HIDDEN ON )
 endif()
 
+# nvcc can not recognize shared libraray file name with suffix other than *.so when linking.
+if (NOT BUILD_WITH_COMPILER STREQUAL "HIP-NVCC")
+  rocm_set_soversion(hipfft ${hipfft_SOVERSION})
+endif()
+
 # Generate export header
 include( GenerateExportHeader )
 generate_export_header( hipfft EXPORT_FILE_NAME ${PROJECT_BINARY_DIR}/include/hipfft/hipfft-export.h )
