--- USD-20.11/cmake/macros/Public.cmake.orig	2020-10-14 11:25:19.000000000 -0700
+++ USD-20.11/cmake/macros/Public.cmake	2020-11-07 22:19:37.199113982 -0800
@@ -921,8 +921,10 @@ function(pxr_toplevel_prologue)
                     FOLDER "${folder}"
                     PREFIX "${PXR_LIB_PREFIX}"
                     IMPORT_PREFIX "${PXR_LIB_PREFIX}"
+                    VERSION "${PXR_MAJOR_VERSION}.${PXR_MINOR_VERSION}.${PXR_PATCH_VERSION}"
+                    SOVERSION "${PXR_MAJOR_VERSION}"
             )
-            _get_install_dir("lib" libInstallPrefix)
+            _get_install_dir("lib${LIB_SUFFIX}" libInstallPrefix)
             install(
                 TARGETS usd_ms
                 LIBRARY DESTINATION ${libInstallPrefix}
