--- a/cmake/modules/FindOpenEXR.cmake
+++ b/cmake/modules/FindOpenEXR.cmake
@@ -23,7 +23,7 @@
 #
 
 find_path(OPENEXR_INCLUDE_DIR
-    OpenEXR/half.h
+    OpenEXR/ImfImage.h
 HINTS
     "${OPENEXR_LOCATION}"
     "$ENV{OPENEXR_LOCATION}"
@@ -60,9 +60,7 @@ foreach(OPENEXR_LIB
     Half
     Iex
     Imath
-    IlmImf
     IlmThread
-    IlmImfUtil
     IexMath
     )
 
@@ -89,6 +87,8 @@ endforeach(OPENEXR_LIB)
 # So #include <half.h> works
 list(APPEND OPENEXR_INCLUDE_DIRS ${OPENEXR_INCLUDE_DIR})
 list(APPEND OPENEXR_INCLUDE_DIRS ${OPENEXR_INCLUDE_DIR}/OpenEXR)
+list(APPEND OPENEXR_INCLUDE_DIRS /usr/include/Imath)
+list(APPEND OPENEXR_INCLUDE_DIRS ${OPENEXR_INCLUDE_DIR}/Imath)
 
 include(FindPackageHandleStandardArgs)
 find_package_handle_standard_args(OpenEXR
