--- a/cmake/macros/Private.cmake
+++ b/cmake/macros/Private.cmake
@@ -1118,10 +1118,10 @@ function(_pxr_library NAME)
 
     # Figure plugin/resource install paths
     if (isPlugin)
-        _get_install_dir("plugin" pluginInstallPrefix)
+        _get_install_dir("lib/usd/plugin" pluginInstallPrefix)
         if (NOT PXR_INSTALL_SUBDIR)
             # XXX --- Why this difference?
-            _get_install_dir("plugin/usd" pluginInstallPrefix)
+            _get_install_dir("lib/usd/plugin/usd" pluginInstallPrefix)
         endif()
     else()
         _get_install_dir("lib/usd" pluginInstallPrefix)
