--- a/cmake/macros/Private.cmake
+++ b/cmake/macros/Private.cmake
@@ -847,7 +847,11 @@ function(_pxr_target_link_libraries NAME
                 elseif(CMAKE_COMPILER_IS_GNUCXX)
                     list(APPEND final -Wl,--whole-archive ${lib} -Wl,--no-whole-archive)
                 elseif("${CMAKE_CXX_COMPILER_ID}" MATCHES "Clang")
-                    list(APPEND final -Wl,-force_load ${lib})
+                    if(APPLE)
+                        list(APPEND final -Wl,-force_load ${lib})
+                    else()
+                        list(APPEND final -Wl,--whole-archive ${lib} -Wl,--no-whole-archive)
+                    endif()
                 else()
                     # Unknown platform.
                     list(APPEND final ${lib})
--- a/cmake/macros/Public.cmake
+++ b/cmake/macros/Public.cmake
@@ -985,10 +985,17 @@ function(pxr_toplevel_epilogue)
                     -Wl,--whole-archive $<TARGET_FILE:usd_m> -Wl,--no-whole-archive
             )
         elseif("${CMAKE_CXX_COMPILER_ID}" MATCHES "Clang")
-            target_link_libraries(usd_ms
-                PRIVATE
-                    -Wl,-force_load $<TARGET_FILE:usd_m>
-            )
+            if(APPLE)
+                target_link_libraries(usd_ms
+                    PRIVATE
+                        -Wl,-force_load $<TARGET_FILE:usd_m>
+                )
+            else()
+                target_link_libraries(usd_ms
+                    PRIVATE
+                        -Wl,--whole-archive $<TARGET_FILE:usd_m> -Wl,--no-whole-archive
+                )
+            endif()
         endif()
 
         # Since we didn't add a dependency to usd_ms on usd_m above, we
