diff -up perl-5.38.0-RC2/cpan/Compress-Raw-Zlib/Zlib.xs.omv~ perl-5.38.0-RC2/cpan/Compress-Raw-Zlib/Zlib.xs
--- perl-5.38.0-RC2/cpan/Compress-Raw-Zlib/Zlib.xs.omv~	2023-06-28 20:04:31.712114994 +0200
+++ perl-5.38.0-RC2/cpan/Compress-Raw-Zlib/Zlib.xs	2023-06-28 20:05:09.133004375 +0200
@@ -37,6 +37,10 @@
 #  include "zlib.h"
 #endif
 
+// Perl expects an integer (0xF), but it's actually just an F...
+#undef ZLIBNG_VER_STATUS
+#define ZLIBNG_VER_STATUS 0xf
+
 
 /* zlib prior to 1.06 doesn't know about z_off_t */
 #ifndef z_off_t
