From 24880b4812c67fb2188b62fc601fe67c196249a7 Mon Sep 17 00:00:00 2001
From: Ikey Doherty <ikey@solus-project.com>
Date: Sun, 7 Aug 2016 03:17:20 +0100
Subject: [PATCH] Don't explicitly force libpng16, ask for libpng.

Signed-off-by: Ikey Doherty <ikey@solus-project.com>
---
 configure.ac | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/configure.ac b/configure.ac
index 536fdf5..60098bf 100644
--- a/configure.ac
+++ b/configure.ac
@@ -96,7 +96,7 @@ AC_DEFINE([GOA_API_IS_SUBJECT_TO_CHANGE], [], [We are aware that GOA's API can c
 PKG_CHECK_MODULES(GRILO, [grilo-0.3 >= $GRILO_MIN_VERSION])
 PKG_CHECK_MODULES(GTK, [gtk+-3.0 >= $GTK_MIN_VERSION])
 PKG_CHECK_MODULES(GTK_UNIX_PRINT, [gtk+-unix-print-3.0])
-PKG_CHECK_MODULES(PNG, [libpng16])
+PKG_CHECK_MODULES(PNG, [libpng])
 PKG_CHECK_MODULES(TRACKER, [tracker-control-1.0 tracker-sparql-1.0])
 
 AX_REQUIRE_DEFINED([LIBGD_INIT])
-- 
2.9.2

