From 39658f27620dd81c9405a0700eba48fa37797040 Mon Sep 17 00:00:00 2001
From: Reilly Brogan <reilly@reillybrogan.com>
Date: Wed, 1 Nov 2023 11:59:20 -0500
Subject: [PATCH] Fix build by including qtsan

---
 src/corelib/thread/thread.pri | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/src/corelib/thread/thread.pri b/src/corelib/thread/thread.pri
index 25cf68a324..8027a71e17 100644
--- a/src/corelib/thread/thread.pri
+++ b/src/corelib/thread/thread.pri
@@ -7,7 +7,8 @@ HEADERS += \
     thread/qthread.h \
     thread/qthreadstorage.h \
     thread/qwaitcondition_p.h \
-    thread/qwaitcondition.h
+    thread/qwaitcondition.h \
+    thread/qtsan_impl.h
 
 SOURCES += \
     thread/qrunnable.cpp \
-- 
2.42.0

