From c192d49568cba4eabd6e3059103648b4f2f88114 Mon Sep 17 00:00:00 2001
From: Reilly Brogan <reilly@reillybrogan.com>
Date: Wed, 13 Dec 2023 18:08:53 -0600
Subject: [PATCH] Solus: Skip adding boot partition

Signed-off-by: Reilly Brogan <reilly@reillybrogan.com>
---
 src/modules/fstab/main.py | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/src/modules/fstab/main.py b/src/modules/fstab/main.py
index 1fd1d695a..0a25e2bb6 100755
--- a/src/modules/fstab/main.py
+++ b/src/modules/fstab/main.py
@@ -256,6 +256,12 @@ class FstabGenerator(object):
         disk_name = disk_name_for_partition(partition)
         is_ssd = disk_name in self.ssd_disks
 
+        # On Solus we want to skip adding the boot partition to fstab if installing on UEFI
+        # clr-boot-manager will mount it as needed.
+        fw_type = libcalamares.globalstorage.value("firmwareType")
+        if mount_point == "/boot" and fw_type == "efi":
+            return None
+
         # Swap partitions are called "linuxswap" by parted.
         # That "fs" is visible in GS, but that gets mapped
         # to "swap", above, because that's the spelling needed in /etc/fstab
-- 
2.43.0

