From 48a18ee87cc0f83ffc82b9a0d9fd36488f4605c8 Mon Sep 17 00:00:00 2001
From: Evan Maddock <maddock.evan@vivaldi.net>
Date: Thu, 21 Dec 2023 11:51:42 -0500
Subject: [PATCH] partition: Hide Volume Group buttons

Signed-off-by: Evan Maddock <maddock.evan@vivaldi.net>
---
 src/modules/partition/gui/PartitionPage.cpp | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/src/modules/partition/gui/PartitionPage.cpp b/src/modules/partition/gui/PartitionPage.cpp
index 0b3cf2478..11b078554 100644
--- a/src/modules/partition/gui/PartitionPage.cpp
+++ b/src/modules/partition/gui/PartitionPage.cpp
@@ -213,6 +213,11 @@ PartitionPage::updateButtons()
     m_ui->resizeVolumeGroupButton->setEnabled( currentDeviceIsVG && !isVGdeactivated );
     m_ui->deactivateVolumeGroupButton->setEnabled( currentDeviceIsVG && isDeactivable && !isVGdeactivated );
     m_ui->removeVolumeGroupButton->setEnabled( currentDeviceIsVG && isRemovable );
+
+    m_ui->newVolumeGroupButton->hide();
+    m_ui->resizeVolumeGroupButton->hide();
+    m_ui->deactivateVolumeGroupButton->hide();
+    m_ui->removeVolumeGroupButton->hide();
 }
 
 void
-- 
2.43.0

