From c9eb081aeba508813e4b8109c016625078756fa4 Mon Sep 17 00:00:00 2001
From: Evan Maddock <maddock.evan@vivaldi.net>
Date: Sun, 12 Nov 2023 15:54:00 -0500
Subject: [PATCH] bootloader: Add clr-boot-manager support

Signed-off-by: Evan Maddock <maddock.evan@vivaldi.net>
---
 src/modules/bootloader/main.py | 40 +++++++++++++++++++++++++++++++++-
 1 file changed, 39 insertions(+), 1 deletion(-)

diff --git a/src/modules/bootloader/main.py b/src/modules/bootloader/main.py
index d7f4a716c..1602fdd1c 100644
--- a/src/modules/bootloader/main.py
+++ b/src/modules/bootloader/main.py
@@ -466,6 +466,40 @@ def efi_boot_next():
         subprocess.call([boot_mgr, "-n", boot_entry])
 
 
+def install_clr_boot_manager(efi_directory):
+    """
+    Installs clr-boot-manager as the bootloader for EFI systems.
+
+    :param efi_directory:
+    """
+    libcalamares.utils.debug("Bootloader: clr-boot-manager")
+
+    installation_root_path = libcalamares.globalstorage.value("rootMountPoint")
+    kernel_dir = os.path.join(installation_root_path, "etc", "kernel", "cmdline.d")
+
+    os.makedirs(kernel_dir, mode=0o00755, exist_ok=True)
+    
+    kernel_resume_file = os.path.join(kernel_dir, "10_resume.conf")
+
+    # Write out the resume= parameter for clr-boot-manager
+    partitions = libcalamares.globalstorage.value("partitions")
+    swap_uuid = ""
+
+    # Get the UUID of the swap partition, if present
+    for partition in partitions:
+        if partition["fs"] == "linuxswap" and not partition.get("claimed", None):
+            continue
+        has_luks = "luksMapperName" in partition
+        if partition["fs"] == "linuxswap" and not has_luks:
+            swap_uuid = partition["uuid"]
+
+    if swap_uuid != "":
+        with open(kernel_resume_file, "w") as resume_file:
+            resume_file.write("resume=UUID={}\n".format(swap_uuid))
+
+    check_target_env_call(["clr-boot-manager", "update"])
+
+
 def install_systemd_boot(efi_directory):
     """
     Installs systemd-boot as bootloader for EFI setups.
@@ -740,7 +774,11 @@ def prepare_bootloader(fw_type):
     efi_boot_loader = libcalamares.job.configuration["efiBootLoader"]
     efi_directory = libcalamares.globalstorage.value("efiSystemPartition")
 
-    if efi_boot_loader == "systemd-boot" and fw_type == "efi":
+    if efi_boot_loader == "clr-boot-manager":
+        if fw_type != "efi":
+            install_grub(efi_directory, fw_type)
+        install_clr_boot_manager(efi_directory)
+    elif efi_boot_loader == "systemd-boot" and fw_type == "efi":
         install_systemd_boot(efi_directory)
     elif efi_boot_loader == "sb-shim" and fw_type == "efi":
         install_secureboot(efi_directory)
-- 
2.43.0

