From 6f5b0eb4d670fa91e07efe1e01b38c56bf38aca8 Mon Sep 17 00:00:00 2001
From: Evan Maddock <maddock.evan@vivaldi.net>
Date: Mon, 18 Sep 2023 16:20:40 -0400
Subject: [PATCH] Don't spawn a new shell when starting

Signed-off-by: Evan Maddock <maddock.evan@vivaldi.net>
---
 calamares.desktop    | 2 +-
 calamares.desktop.in | 2 +-
 2 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/calamares.desktop b/calamares.desktop
index 97ad28de8..f3bfcf45b 100644
--- a/calamares.desktop
+++ b/calamares.desktop
@@ -5,7 +5,7 @@ Name=Install System
 GenericName=System Installer
 Keywords=calamares;system;installer;
 TryExec=calamares
-Exec=sh -c "pkexec calamares"
+Exec=/usr/bin/calamares-gui
 Comment=Calamares — System Installer
 Icon=calamares
 Terminal=false
diff --git a/calamares.desktop.in b/calamares.desktop.in
index ed1d4def8..e1bb7717a 100644
--- a/calamares.desktop.in
+++ b/calamares.desktop.in
@@ -5,7 +5,7 @@ Name=Install System
 GenericName=System Installer
 Keywords=calamares;system;installer;
 TryExec=calamares
-Exec=sh -c "pkexec calamares"
+Exec=/usr/bin/calamares-gui
 Comment=Calamares — System Installer
 Icon=calamares
 Terminal=false
-- 
2.42.0

