diff -up qtwebengine-everywhere-src-5.15.15-20231006/src/3rdparty/chromium/third_party/perfetto/include/perfetto/ext/base/subprocess.h.omv~ qtwebengine-everywhere-src-5.15.15-20231006/src/3rdparty/chromium/third_party/perfetto/include/perfetto/ext/base/subprocess.h
--- qtwebengine-everywhere-src-5.15.15-20231006/src/3rdparty/chromium/third_party/perfetto/include/perfetto/ext/base/subprocess.h.omv~	2023-10-06 08:03:12.262741993 +0200
+++ qtwebengine-everywhere-src-5.15.15-20231006/src/3rdparty/chromium/third_party/perfetto/include/perfetto/ext/base/subprocess.h	2023-10-06 08:03:40.829053990 +0200
@@ -173,7 +173,7 @@ class Subprocess {
   };
 
   explicit Subprocess(std::initializer_list<std::string> exec_cmd = {});
-  Subprocess(Subprocess&&);
+  Subprocess(Subprocess&&) noexcept;
   Subprocess& operator=(Subprocess&&);
   ~Subprocess();  // It will KillAndWaitForTermination() if still alive.
 
diff -up qtwebengine-everywhere-src-5.15.16-20231006/src/3rdparty/chromium/gpu/command_buffer/service/program_manager.cc.omv~ qtwebengine-everywhere-src-5.15.16-20231006/src/3rdparty/chromium/gpu/command_buffer/service/program_manager.cc
--- qtwebengine-everywhere-src-5.15.16-20231006/src/3rdparty/chromium/gpu/command_buffer/service/program_manager.cc.omv~	2023-10-06 13:45:38.089892995 +0200
+++ qtwebengine-everywhere-src-5.15.16-20231006/src/3rdparty/chromium/gpu/command_buffer/service/program_manager.cc	2023-10-06 13:54:51.406090097 +0200
@@ -620,7 +620,7 @@ std::string Program::ProcessLogInfo(cons
       output += hashed_name;
   }
 
-  return output + input.as_string();
+  return output + static_cast<std::string>(input);
 }
 
 void Program::UpdateLogInfo() {
diff -up qtwebengine-everywhere-src-5.15.16-20231006/src/3rdparty/chromium/third_party/crashpad/crashpad/compat/linux/sys/mman_memfd_create.cc.omv~ qtwebengine-everywhere-src-5.15.16-20231006/src/3rdparty/chromium/third_party/crashpad/crashpad/compat/linux/sys/mman_memfd_create.cc
--- qtwebengine-everywhere-src-5.15.16-20231006/src/3rdparty/chromium/third_party/crashpad/crashpad/compat/linux/sys/mman_memfd_create.cc.omv~	2023-10-06 13:32:03.062719282 +0200
+++ qtwebengine-everywhere-src-5.15.16-20231006/src/3rdparty/chromium/third_party/crashpad/crashpad/compat/linux/sys/mman_memfd_create.cc	2023-10-06 13:35:15.949763581 +0200
@@ -14,6 +14,7 @@
 
 #include <sys/mman.h>
 
+#if 0 // current glibc has memfd_create
 #include <dlfcn.h>
 #include <sys/syscall.h>
 #include <unistd.h>
@@ -24,7 +25,7 @@
 
 extern "C" {
 
-int memfd_create(const char* name, unsigned int flags) {
+int memfd_create(const char* name, unsigned int flags) throw() {
   static const crashpad::NoCfiIcall<decltype(memfd_create)*> next_memfd_create(
       dlsym(RTLD_NEXT, "memfd_create"));
   return next_memfd_create ? next_memfd_create(name, flags)
@@ -34,3 +35,4 @@ int memfd_create(const char* name, unsig
 }  // extern "C"
 
 #endif  // __GLIBC__
+#endif
