diff -up qtwebengine-everywhere-src-5.15.15-20230613/src/3rdparty/chromium/base/memory/ref_counted.h.omv~ qtwebengine-everywhere-src-5.15.15-20230613/src/3rdparty/chromium/base/memory/ref_counted.h
diff -up qtwebengine-everywhere-src-5.15.15-20230613/src/3rdparty/chromium/base/threading/thread_local_internal.h.omv~ qtwebengine-everywhere-src-5.15.15-20230613/src/3rdparty/chromium/base/threading/thread_local_internal.h
--- qtwebengine-everywhere-src-5.15.15-20230613/src/3rdparty/chromium/base/threading/thread_local_internal.h.omv~	2023-06-13 18:52:30.233909174 +0200
+++ qtwebengine-everywhere-src-5.15.15-20230613/src/3rdparty/chromium/base/threading/thread_local_internal.h	2023-06-13 18:53:11.044240640 +0200
@@ -70,7 +70,9 @@ class CheckedThreadLocalOwnedPointer {
 
   std::atomic_int num_assigned_threads_{0};
 
-  DISALLOW_COPY_AND_ASSIGN(CheckedThreadLocalOwnedPointer<T>);
+  //DISALLOW_COPY_AND_ASSIGN(CheckedThreadLocalOwnedPointer<T>);
+  CheckedThreadLocalOwnedPointer(const CheckedThreadLocalOwnedPointer<T>&) = delete;
+  CheckedThreadLocalOwnedPointer<T>& operator=(const CheckedThreadLocalOwnedPointer<T>&) = delete;
 };
 
 }  // namespace internal
