From 01433247e812cdfeaec8dfc2ee6ab8c84354735d Mon Sep 17 00:00:00 2001
From: Ikey Doherty <ikey@solus-project.com>
Date: Sun, 16 Oct 2016 07:23:37 +0100
Subject: [PATCH] Fix discoverability in MATE

Signed-off-by: Ikey Doherty <ikey@solus-project.com>
---
 data/solus-sc.desktop | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/data/solus-sc.desktop b/data/solus-sc.desktop
index 0fe1fc1..e2aa1e5 100644
--- a/data/solus-sc.desktop
+++ b/data/solus-sc.desktop
@@ -7,5 +7,5 @@ Terminal=false
 Type=Application
 StartupNotify=false
 Icon=system-software-install
-Categories=GNOME;GTK;System;PackageManager;
+Categories=PackageManager;System;Settings;
 Keywords=Updates;Upgrade;Sources;Repositories;Preferences;Install;Uninstall;Program;Software;App;Store;
-- 
2.10.1

