From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Thomas Staudinger <Staudi.Kaos@gmail.com>
Date: Thu, 26 Sep 2024 00:48:56 +0200
Subject: [PATCH] Remove repo_data files from POTFILES.in

Signed-off-by: Thomas Staudinger <Staudi.Kaos@gmail.com>
---
 po/POTFILES.in | 3 ---
 1 file changed, 3 deletions(-)

diff --git a/po/POTFILES.in b/po/POTFILES.in
index ab10fa0..e41f5bb 100644
--- a/po/POTFILES.in
+++ b/po/POTFILES.in
@@ -1,6 +1,3 @@
-repo_data/components.xml.in
-repo_data/distribution.xml.in
-repo_data/groups.xml.in
 solus_sc/application.py
 solus_sc/application.py
 solus_sc/available_view.py
