From 8d64b1a89165c3505e84aa2c6401a876d83c9754 Mon Sep 17 00:00:00 2001
From: Joshua Strobl <joshua@stroblindustries.com>
Date: Wed, 5 Dec 2018 07:41:43 +0200
Subject: [PATCH 1/1] Add a desktop action for going to the Updates view.

---
 data/solus-sc.desktop | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/data/solus-sc.desktop b/data/solus-sc.desktop
index 01282de..ccb028f 100644
--- a/data/solus-sc.desktop
+++ b/data/solus-sc.desktop
@@ -9,3 +9,8 @@ StartupNotify=true
 Icon=system-software-install
 Categories=X-GNOME-Settings-Panel;PackageManager;System;Settings
 Keywords=Updates;Upgrade;Sources;Repositories;Preferences;Install;Uninstall;Program;Software;App;Store;
+Actions=view-updates;
+
+[Desktop Action view-updates]
+Name=Updates
+Exec=solus-sc --update-view
\ No newline at end of file
-- 
2.19.2

