From 9f394a04454ff6bb60e22fa6c957dcf61abf8b35 Mon Sep 17 00:00:00 2001
From: Staudey <Staudi.Kaos@gmail.com>
Date: Tue, 20 Feb 2018 11:02:59 +0100
Subject: [PATCH 01/29] Fix SC printing "Removed x of x" when installing

---
 solus_sc/basket.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/solus_sc/basket.py b/solus_sc/basket.py
index 4cb1574..2f733f9 100644
--- a/solus_sc/basket.py
+++ b/solus_sc/basket.py
@@ -234,7 +234,7 @@ class BasketView(Gtk.Revealer):
                     lab = _("Removed {} of {}: {}")
                 elif cmd == 'installed':
                     # "Installed 1 of 10: bash"
-                    lab = _("Removed {} of {}: {}")
+                    lab = _("Installed {} of {}: {}")
                 ffmt = lab.format(self.current_package, self.total_packages,
                                   what)
                 self.set_progress(prog, ffmt)
-- 
2.16.2

