From 3ae06a4d45fe154e44d8df3f0c5fabb6c134ce7a Mon Sep 17 00:00:00 2001
From: Troy Harvey <harveydevel@gmail.com>
Date: Fri, 22 Aug 2025 14:37:53 +1000
Subject: [PATCH] Add exception handling to markdown conversion

Signed-off-by: Troy Harvey <harveydevel@gmail.com>
---
 solus_sc/changelog.py | 28 ++++++++++++++++------------
 1 file changed, 16 insertions(+), 12 deletions(-)

diff --git a/solus_sc/changelog.py b/solus_sc/changelog.py
index 58dd2be..54c6d20 100755
--- a/solus_sc/changelog.py
+++ b/solus_sc/changelog.py
@@ -81,18 +81,22 @@ class ScChangelogEntry(Gtk.EventBox):
                 r = r.split(id)[1].strip()
                 break
 
-            r = MARKUP_URI_HIT.sub(r'<a href="\2">\1</a>', r)
-            r = MARKUP_CODE_HIT.sub(r'<tt>\1</tt>', r)
-            r = MARKUP_BOLD_HIT.sub(r'<b>\1</b>', r)
-            r = MARKUP_ITALIC_HIT.sub(r'<i>\1</i>', r)
-            r = MARKUP_GITHUB_CHECKED_TASK_HIT.sub(r'&#x1F5F9; \1', r)
-            r = MARKUP_GITHUB_UNCHECKED_TASK_HIT.sub(r'&#x2610; \1', r)
-            r = MARKUP_GITHUB_TITLE_HIT.sub(r'<i>\1</i>\n', r)
-            r = MARKUP_SIGNED_OFF_BY_HIT.sub(r'<small><b>{}</b> <i>\1</i></small>'.format(GIT_SIGNED_OFF_BY_LINE), r)
-            r = CVE_HIT.sub(r' <a href="{}\1">\1</a>'.format(CVE_URI), r)
-            r = BUG_HIT.sub(r' <a href="{}/T\1">T\1</a>'.format(PHAB_URI), r)
-            r = DIFF_HIT.sub(r' <a href="{}/D\1">D\1</a>'.format(PHAB_URI), r)
-            r = GITHUB_ISSUE_HIT.sub(r'<a href="{}/issues/\1">#\1</a>'.format(GITHUB_URI), r)
+            try:
+                r = MARKUP_URI_HIT.sub(r'<a href="\2">\1</a>', r)
+                r = MARKUP_CODE_HIT.sub(r'<tt>\1</tt>', r)
+                r = MARKUP_BOLD_HIT.sub(r'<b>\1</b>', r)
+                r = MARKUP_ITALIC_HIT.sub(r'<i>\1</i>', r)
+                r = MARKUP_GITHUB_CHECKED_TASK_HIT.sub(r'&#x1F5F9; \1', r)
+                r = MARKUP_GITHUB_UNCHECKED_TASK_HIT.sub(r'&#x2610; \1', r)
+                r = MARKUP_GITHUB_TITLE_HIT.sub(r'<i>\1</i>\n', r)
+                r = MARKUP_SIGNED_OFF_BY_HIT.sub(r'<small><b>{}</b> <i>\1</i></small>'.format(GIT_SIGNED_OFF_BY_LINE), r)
+                r = CVE_HIT.sub(r' <a href="{}\1">\1</a>'.format(CVE_URI), r)
+                r = BUG_HIT.sub(r' <a href="{}/T\1">T\1</a>'.format(PHAB_URI), r)
+                r = DIFF_HIT.sub(r' <a href="{}/D\1">D\1</a>'.format(PHAB_URI), r)
+                r = GITHUB_ISSUE_HIT.sub(r'<a href="{}/issues/\1">#\1</a>'.format(GITHUB_URI), r)
+            except Exception as e:
+                print "Problem with line:", r
+                print "Error:", e
 
             # Check if this is a bullet point
             if (r.startswith("- ") or r.startswith("* ")) and len(r) > 2:
-- 
2.49.1

