From d8987c9f27ee177b6f6e4296710bb5701b12e309 Mon Sep 17 00:00:00 2001
From: Peter O'Connor <peter@solus-project.com>
Date: Mon, 30 Oct 2017 15:37:06 +1100
Subject: [PATCH] Enable autostart in Plasma session

Signed-off-by: Peter O'Connor <peter@solus-project.com>
---
 data/solus-update.desktop | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/data/solus-update.desktop b/data/solus-update.desktop
index 13f9666..93f9562 100644
--- a/data/solus-update.desktop
+++ b/data/solus-update.desktop
@@ -6,7 +6,7 @@ Icon=software-update-available
 Exec=solus-update-checker
 TryExec=solus-update-checker
 NoDisplay=true
-OnlyShowIn=Budgie;GNOME;MATE;
+OnlyShowIn=Budgie;GNOME;KDE;MATE;
 X-GNOME-Autostart-Phase=Application
 X-GNOME-Autostart-Notify=true
 AutostartCondition=GSettings com.solus-project.software-center check-updates
