From 843efc20bedac241159045dd2d9e953c9802a2c6 Mon Sep 17 00:00:00 2001
From: Ikey Doherty <ikey@solus-project.com>
Date: Sun, 19 Jun 2016 23:39:23 +0100
Subject: [PATCH 2/2] Gonna pretend like I haven't been fixing this for over a
 year

Signed-off-by: Ikey Doherty <ikey@solus-project.com>
---
 eopkg_assist/__init__.py | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/eopkg_assist/__init__.py b/eopkg_assist/__init__.py
index d36d7b6..bb7b6e5 100644
--- a/eopkg_assist/__init__.py
+++ b/eopkg_assist/__init__.py
@@ -11,5 +11,8 @@
 #  (at your option) any later version.
 #
 
+import os
 import gi.repository
 gi.require_version('Gio', '2.0')
+
+os.environ['PATH'] = '/sbin:/usr/sbin:/bin:/usr/bin'
-- 
2.8.4

