From 9cca0e25462091c0665031f94bf6d5840bec00f3 Mon Sep 17 00:00:00 2001
From: Ikey Doherty <ikey@solus-project.com>
Date: Sun, 28 May 2017 21:51:30 +0100
Subject: [PATCH 2/4] changelog: Don't swallow greedily, and add bolding
 support

Signed-off-by: Ikey Doherty <ikey@solus-project.com>
---
 solus_sc/changelog.py | 12 ++++++------
 1 file changed, 6 insertions(+), 6 deletions(-)

diff --git a/solus_sc/changelog.py b/solus_sc/changelog.py
index df24c71..eec1018 100755
--- a/solus_sc/changelog.py
+++ b/solus_sc/changelog.py
@@ -33,8 +33,9 @@ GENERAL_URI = re.compile(r"""(?i)\b((?:[a-z][\w-]+:(?:/{1,3}|[a-z0-9%])|www\d\
 ))*\))+(?:\(([^\s()<>]+|(\([^\s()<>]+\)))*\)|[^\s`!()\[\]{};:'".,<>\
 ?«»“”‘’]))""")
 
-MARKUP_URI_HIT = re.compile(".*\[({})\]\(({})\).*".format("[^\]]+", ".*"))
-MARKUP_CODE_HIT = re.compile(r".*`([^`]+)`.*")
+MARKUP_URI_HIT = re.compile("\[({})\]\(({})\)".format("[^\]]+", ".*"))
+MARKUP_CODE_HIT = re.compile(r"`([^`]+)`")
+MARKUP_BOLD_HIT = re.compile(r"\*{2}([^\*{2}]+)\*{2}")
 
 
 class ScChangelogEntry(Gtk.EventBox):
@@ -61,10 +62,9 @@ class ScChangelogEntry(Gtk.EventBox):
                 r = r.split(id)[1].strip()
                 break
 
-            if MARKUP_URI_HIT.match(r):
-                r = MARKUP_URI_HIT.sub(r'<a href="\2">\1</a>', r)
-            if MARKUP_CODE_HIT.match(r):
-                r = MARKUP_CODE_HIT.sub(r'<tt>\1</tt>', r)
+            r = MARKUP_URI_HIT.sub(r'<a href="\2">\1</a>', r)
+            r = MARKUP_CODE_HIT.sub(r'<tt>\1</tt>', r)
+            r = MARKUP_BOLD_HIT.sub(r'<b>\1</b>', r)
 
             # Check if this is a bullet point
             if (r.startswith("- ") or r.startswith("* ")) and len(r) > 2:
-- 
2.13.0

