From dbacaff629c1417610b063b4d061b6a296146fef Mon Sep 17 00:00:00 2001
From: Ikey Doherty <ikey@solus-project.com>
Date: Sun, 23 Oct 2016 05:57:13 +0100
Subject: [PATCH 2/2] Ensure we locally record timestamp of last_checked

Signed-off-by: Ikey Doherty <ikey@solus-project.com>
---
 solus_update/application.py | 1 +
 1 file changed, 1 insertion(+)

diff --git a/solus_update/application.py b/solus_update/application.py
index ce3dfbd..585dcd4 100644
--- a/solus_update/application.py
+++ b/solus_update/application.py
@@ -318,6 +318,7 @@ class ScUpdateApp(Gio.Application):
         timestamp = time.time()
         variant = GLib.Variant.new_int64(timestamp)
         self.settings.set_value("last-checked", variant)
+        self.last_checked = timestamp
 
     def is_update_check_required(self):
         """ Determine if an update is required at all"""
-- 
2.10.1

