From 3f18ddf0b8022a21765173ec1b80c634766cd166 Mon Sep 17 00:00:00 2001
From: Ikey Doherty <ikey@solus-project.com>
Date: Sun, 16 Oct 2016 07:39:17 +0100
Subject: [PATCH 2/2] Ensure we appear in both MATE & GNOME

Signed-off-by: Ikey Doherty <ikey@solus-project.com>
---
 data/solus-sc.desktop | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/data/solus-sc.desktop b/data/solus-sc.desktop
index e2aa1e5..ca210a5 100644
--- a/data/solus-sc.desktop
+++ b/data/solus-sc.desktop
@@ -7,5 +7,5 @@ Terminal=false
 Type=Application
 StartupNotify=false
 Icon=system-software-install
-Categories=PackageManager;System;Settings;
+Categories=X-GNOME-Settings-Panel;PackageManager;System;Settings
 Keywords=Updates;Upgrade;Sources;Repositories;Preferences;Install;Uninstall;Program;Software;App;Store;
-- 
2.10.1

