From eec196833e0fcbe99230af532651b52a96268f53 Mon Sep 17 00:00:00 2001
From: Ikey Doherty <ikey@solus-project.com>
Date: Sat, 2 Jul 2016 16:26:27 +0100
Subject: [PATCH 3/3] Forcibly lower the time to package anything

Signed-off-by: Ikey Doherty <ikey@solus-project.com>
---
 eopkg_assist/backend.py | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/eopkg_assist/backend.py b/eopkg_assist/backend.py
index 48f76fe..feec8d9 100644
--- a/eopkg_assist/backend.py
+++ b/eopkg_assist/backend.py
@@ -170,6 +170,8 @@ class EopkgAssistService(dbus.service.Object):
         ui = EopkgUiMonitor(ok, ok)
         pisi.context.ui = ui
         pisi.context.config.values.general.ignore_safety = True
+        pisi.context.config.values.build.compressionlevel = 1
+        pisi.context.config.values.build.generateDebug = False
         options = pisi.config.Options()
         options.output_dir = tempfile.mkdtemp(suffix='sc')
         pisi.api.set_options(options)
-- 
2.9.0

