From ed591bfeeb66b8652939d88854fe118418ebcea2 Mon Sep 17 00:00:00 2001
From: Ikey Doherty <ikey@solus-project.com>
Date: Tue, 30 May 2017 00:18:36 +0100
Subject: [PATCH 4/4] changelog: Don't consume the last `)` for markup URIs

Signed-off-by: Ikey Doherty <ikey@solus-project.com>
---
 solus_sc/changelog.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/solus_sc/changelog.py b/solus_sc/changelog.py
index 53a2383..4c9c772 100755
--- a/solus_sc/changelog.py
+++ b/solus_sc/changelog.py
@@ -33,7 +33,7 @@ GENERAL_URI = re.compile(r"""(?i)\b((?:[a-z][\w-]+:(?:/{1,3}|[a-z0-9%])|www\d\
 ))*\))+(?:\(([^\s()<>]+|(\([^\s()<>]+\)))*\)|[^\s`!()\[\]{};:'".,<>\
 ?«»“”‘’]))""")
 
-MARKUP_URI_HIT = re.compile("\[({})\]\(({})\)".format("[^\]]+", ".*"))
+MARKUP_URI_HIT = re.compile(r"\[({})\]\(({})\)".format("[^\]]+", "[^)]+"))
 MARKUP_CODE_HIT = re.compile(r"`([^`]+)`")
 MARKUP_BOLD_HIT = re.compile(r"\*{2}([^\*{2}]+)\*{2}")
 
-- 
2.13.0

