From 1d547486d2200f0aec4e34e507fcede8b303237b Mon Sep 17 00:00:00 2001
From: "Georg Grabler (STiAT)" <ggrabler@gmail.com>
Date: Mon, 3 Apr 2017 09:43:10 +0200
Subject: [PATCH 7/8] Fix #T2697: action_show_updates takes no extra parameters

---
 solus_update/application.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/solus_update/application.py b/solus_update/application.py
index 1ff9db5..07b89b9 100644
--- a/solus_update/application.py
+++ b/solus_update/application.py
@@ -310,7 +310,7 @@ class ScUpdateApp(Gio.Application):
         self.notification = Notify.Notification.new(title, body, icon_name)
         self.notification.set_timeout(UPDATE_NOTIF_TIMEOUT)
         self.notification.add_action("open-sc", _("Open Software Center"),
-                                     self.action_show_updates)
+                                     self.action_show_updates, None)
         self.notification.show()
 
     def store_update_time(self):
-- 
2.12.2

