From e2a072f139005713271b9843e8adc2bccbc90693 Mon Sep 17 00:00:00 2001
From: Pierre-Yves <pyu@riseup.net>
Date: Thu, 22 Mar 2018 17:40:46 +0100
Subject: [PATCH] Use libdir variable in pktsetup

Signed-off-by: Pierre-Yves <pyu@riseup.net>
---
 pktsetup/Makefile.am | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/pktsetup/Makefile.am b/pktsetup/Makefile.am
index 344dbfe..72a8d33 100644
--- a/pktsetup/Makefile.am
+++ b/pktsetup/Makefile.am
@@ -3,8 +3,8 @@ pktsetup_SOURCES = pktsetup.c
 EXTRA_DIST = pktsetup.rules
 
 install-data-local:
-	mkdir -p "$(DESTDIR)/lib/udev/rules.d"
-	$(INSTALL_DATA) "$(srcdir)/pktsetup.rules" "$(DESTDIR)/lib/udev/rules.d/80-pktsetup.rules"
+	mkdir -p "$(DESTDIR)$(libdir)/udev/rules.d"
+	$(INSTALL_DATA) "$(srcdir)/pktsetup.rules" "$(DESTDIR)$(libdir)/udev/rules.d/80-pktsetup.rules"
 
 uninstall-local:
-	rm -f "$(DESTDIR)/lib/udev/rules.d/80-pktsetup.rules"
+	rm -f "$(DESTDIR)$(libdir)/udev/rules.d/80-pktsetup.rules"
-- 
2.16.2

