From d707ca130937f5904acebf7c00fd940252706dea Mon Sep 17 00:00:00 2001
From: Christophe Giboudeaux <christophe@krop.fr>
Date: Sat, 7 Dec 2019 23:19:55 +0100
Subject: [PATCH] Fix build with recent pango releases.

---
 CMakeLists.txt                | 2 ++
 cmake_modules/FindPANGO.cmake | 2 ++
 2 files changed, 4 insertions(+)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 0557162..742b3d3 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -117,6 +117,8 @@ INCLUDE_DIRECTORIES(
   ${M_INCLUDE_DIR}
 )
 
+ADD_DEFINITIONS(${PANGO_CFLAGS})
+
 MACRO(add_dbus_bindings name)
   ADD_CUSTOM_COMMAND(
     OUTPUT
diff --git a/cmake_modules/FindPANGO.cmake b/cmake_modules/FindPANGO.cmake
index 2b8e880..911c459 100644
--- a/cmake_modules/FindPANGO.cmake
+++ b/cmake_modules/FindPANGO.cmake
@@ -16,6 +16,8 @@ FIND_PATH(PANGO_INCLUDE_DIRS pango/pango.h
   PATH_SUFFIXES pango-1.0
 )
 
+SET(PANGO_CFLAGS "${PKG_PANGO_CFLAGS}")
+
 FIND_LIBRARY(PANGO_LIBRARIES pango-1.0
   HINTS ${PKG_PANGO_LIBRARY_DIRS}
 )

