From 60735353d0086eea88ed1b7b2b835986315db2a1 Mon Sep 17 00:00:00 2001
From: Pierre-Yves <pyu@riseup.net>
Date: Tue, 16 Apr 2019 19:10:20 +0200
Subject: [PATCH] bootstrap: add solus

Signed-off-by: Pierre-Yves <pyu@riseup.net>
---
 bootstrap.sh | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/bootstrap.sh b/bootstrap.sh
index 92dcfec..22416d8 100755
--- a/bootstrap.sh
+++ b/bootstrap.sh
@@ -30,6 +30,10 @@ for i in $ID $ID_LIKE; do
 			PKG_CONFIG_32=i586-slackware-linux-gnu-pkg-config
 			PKG_CONFIG_64=x86_64-slackware-linux-gnu-pkg-config
 			;;
+		solus)
+			PKG_CONFIG_32=pkg-config
+			PKG_CONFIG_64=x86_64-solus-linux-gnu-pkg-config
+			;;
 		*)
 			continue
 			;;
-- 
2.21.0

