From fdcdb8d38597d123f63bf6f844333ca2fd8b1d60 Mon Sep 17 00:00:00 2001
From: Algent Albrahimi <algent@protonmail.com>
Date: Sat, 19 Jun 2021 21:17:16 +0200
Subject: [PATCH] fix build with python 3.7

---
 src/calibre/utils/formatter.py | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/src/calibre/utils/formatter.py b/src/calibre/utils/formatter.py
index 1b8674c..5a5d391 100644
--- a/src/calibre/utils/formatter.py
+++ b/src/calibre/utils/formatter.py
@@ -925,7 +925,8 @@ class _Interpreter(object):
 
     def do_node_first_non_empty(self, prog):
         for expr in prog.expression_list:
-            if v := self.expr(expr):
+            v = self.expr(expr)
+            if v:
                 if (self.break_reporter):
                     self.break_reporter(prog.node_name, v, prog.line_number)
                 return v
-- 
2.30.2

