From d0acc0c0358dbe361fe12179a08492a2aeab0392 Mon Sep 17 00:00:00 2001
From: Algent Albrahimi <algent@protonmail.com>
Date: Sat, 10 Jul 2021 23:34:32 +0200
Subject: [PATCH] calibre no update

---
 update.py | 14 --------------
 1 file changed, 14 deletions(-)

diff --git a/src/calibre/gui2/update.py b/src/calibre/gui2/update.py
index b5d847a..9b31105 100644
--- a/src/calibre/gui2/update.py
+++ b/src/calibre/gui2/update.py
@@ -85,20 +85,6 @@ class CheckForUpdates(Thread):
         while not self.shutdown_event.is_set():
             calibre_update_version = NO_CALIBRE_UPDATE
             plugins_update_found = 0
-            try:
-                version = get_newest_version()
-                if version[:2] > numeric_version[:2]:
-                    calibre_update_version = version
-            except Exception as e:
-                prints('Failed to check for calibre update:', as_unicode(e))
-            try:
-                update_plugins = get_plugin_updates_available(raise_error=True)
-                if update_plugins is not None:
-                    plugins_update_found = len(update_plugins)
-            except Exception as e:
-                prints('Failed to check for plugin update:', as_unicode(e))
-            if calibre_update_version != NO_CALIBRE_UPDATE or plugins_update_found > 0:
-                self.signal.update_found.emit(calibre_update_version, plugins_update_found)
             self.shutdown_event.wait(self.INTERVAL)
 
     def shutdown(self):
-- 
2.32.0

