From 9684e5fa7647614480ef02bcd7fac3e818a534e2 Mon Sep 17 00:00:00 2001
From: Fabio Forni <livingsilver94.solus@tuta.io>
Date: Mon, 22 Oct 2018 19:23:04 +0200
Subject: [PATCH] Use system theme

---
 src/calibre/gui2/__init__.py | 5 +----
 1 file changed, 1 insertion(+), 4 deletions(-)

diff --git a/src/calibre/gui2/__init__.py b/src/calibre/gui2/__init__.py
index 0b77e7b..47fd7ed 100644
--- a/src/calibre/gui2/__init__.py
+++ b/src/calibre/gui2/__init__.py
@@ -1055,10 +1055,7 @@ class Application(QApplication):
         self.set_palette(dark_palette())
 
     def setup_styles(self, force_calibre_style):
-        if iswindows or ismacos:
-            using_calibre_style = gprefs['ui_style'] != 'system'
-        else:
-            using_calibre_style = os.environ.get('CALIBRE_USE_SYSTEM_THEME', '0') == '0'
+        using_calibre_style = False
         if force_calibre_style:
             using_calibre_style = True
         if using_calibre_style:
-- 
2.32.0

