--- a/setup/build_environment.py
+++ b/setup/build_environment.py
@@ -253,5 +253,8 @@
         ' PODOFO_LIB_DIR environment variables.',
     )
 )
+piper_error = None if any(os.path.exists(os.path.join(d, 'onnxruntime_cxx_api.h')) for d in piper_inc_dirs) else (
+    'onnxruntime not found on your system. Piper TTS functionality will not be available.',
+)
 podofo_inc_dirs = [podofo_inc, os.path.dirname(podofo_inc)]
 podofo_lib_dirs = [podofo_lib]
--- a/setup/extensions.json
+++ b/setup/extensions.json
@@ -138,6 +138,8 @@
         "name": "piper",
         "sources": "calibre/utils/tts/piper.cpp",
 		"needs_c++": "17",
+        "error": "!piper_error",
+        "optional": true,
         "libraries": "!piper_libs",
         "lib_dirs": "!piper_lib_dirs",
         "inc_dirs": "!piper_inc_dirs"
