diff -ru calibre-8.7.0.orig/src/calibre/headless/CMakeLists.txt calibre-8.7.0/src/calibre/headless/CMakeLists.txt
--- calibre-8.7.0.orig/src/calibre/headless/CMakeLists.txt	2025-10-05 11:10:39.597203419 +0200
+++ calibre-8.7.0/src/calibre/headless/CMakeLists.txt	2025-10-05 11:11:01.519392394 +0200
@@ -2,6 +2,9 @@
 project(headless)
 set(CMAKE_AUTOMOC ON)
 find_package(Qt6Gui REQUIRED)
+if(Qt6_VERSION VERSION_GREATER_EQUAL 6.10)
+  find_package(Qt6 COMPONENTS CorePrivate GuiPrivate REQUIRED)
+endif()
 add_library(headless MODULE main.cpp headless_backingstore.cpp headless_integration.cpp)
 set_property(TARGET headless PROPERTY QT_PLUGIN_TYPE "platforms")
 set_property(TARGET headless PROPERTY QT_PLUGIN_CLASS_NAME "HeadlessIntegrationPlugin")
