From bfc759a6ed83608636363f6eca852f80bd38ce2a Mon Sep 17 00:00:00 2001
From: Joey Riches <josephriches@gmail.com>
Date: Sun, 15 Mar 2026 17:59:25 +0000
Subject: [PATCH 1/1] test/fuzzing: Increase timeout

---
 test/fuzzing/meson.build | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/test/fuzzing/meson.build b/test/fuzzing/meson.build
index 2ed5f4ff5..d5e374a6f 100644
--- a/test/fuzzing/meson.build
+++ b/test/fuzzing/meson.build
@@ -105,6 +105,7 @@ if not get_option('raster').disabled()
       args: chunk,
       workdir: meson.current_build_dir() / '..' / '..',
       protocol: 'tap',
+      timeout: 600,
       suite: ['fuzzing'],
     )
     i += 1
@@ -119,6 +120,7 @@ if not get_option('vector').disabled()
       args: chunk,
       workdir: meson.current_build_dir() / '..' / '..',
       protocol: 'tap',
+      timeout: 600,
       suite: ['fuzzing'],
     )
     i += 1
-- 
2.53.0

