From 1f45d98be209adf9c973bc380a80c705941bb260 Mon Sep 17 00:00:00 2001
From: Troy Harvey <harveydevel@gmail.com>
Date: Wed, 16 Aug 2023 17:55:31 +1000
Subject: [PATCH] Use python3 and avoid deprecation warnings

---
 dstat | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/dstat b/dstat
index 9359965..be23c10 100755
--- a/dstat
+++ b/dstat
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 
 ### This program is free software; you can redistribute it and/or
 ### modify it under the terms of the GNU General Public License
@@ -19,7 +19,7 @@
 from __future__ import absolute_import, division, generators, print_function
 __metaclass__ = type
 
-import collections
+import collections.abc
 import fnmatch
 import getopt
 import getpass
@@ -512,7 +512,7 @@ class dstat:
                 scale = self.scales[i]
             else:
                 scale = self.scale
-            if isinstance(self.val[name], collections.Sequence) and not isinstance(self.val[name], six.string_types):
+            if isinstance(self.val[name], collections.abc.Sequence) and not isinstance(self.val[name], six.string_types):
                 line = line + cprintlist(self.val[name], ctype, self.width, scale)
                 sep = theme['frame'] + char['colon']
                 if i + 1 != len(self.vars):
-- 
2.41.0

