From db3b3eaa004c315104bd9c3bdea1448c44b26860 Mon Sep 17 00:00:00 2001
From: Pierre-Yves <pyu@riseup.net>
Date: Sun, 11 Mar 2018 15:37:01 +0100
Subject: [PATCH] Use gdk_display_beep() in place of obsolete gdk_beep()

Signed-off-by: Pierre-Yves <pyu@riseup.net>
---
 gtk.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/gtk.c b/gtk.c
index 3078e51..6e6741e 100644
--- a/gtk.c
+++ b/gtk.c
@@ -2152,7 +2152,7 @@ static void menu_copy_event(GtkMenuItem *menuitem, gpointer data)
     if (text) {
 	write_clip(fe, text);
     } else {
-	gdk_beep();
+	gdk_display_beep(gdk_display_get_default());
     }
 }
 
-- 
2.16.2

