From e4e440d38dea7f75653569f5f1ae97cfa44a5b5f Mon Sep 17 00:00:00 2001
From: Pierre-Yves <pyu@riseup.net>
Date: Sun, 19 Apr 2020 17:43:10 +0200
Subject: [PATCH] Use stateless directories

Signed-off-by: Pierre-Yves <pyu@riseup.net>
---
 Makefile.am | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/Makefile.am b/Makefile.am
index f7ffce2..70fb51f 100644
--- a/Makefile.am
+++ b/Makefile.am
@@ -2,5 +2,5 @@ SUBDIRS = src
 
 dist_man_MANS = xiccd.8
 
-autostartdir = $(sysconfdir)/xdg/autostart
+autostartdir = $(datadir)/xdg/autostart
 autostart_DATA = xiccd.desktop
-- 
2.26.1

