--- a/src/shared_conversions.h
+++ b/src/shared_conversions.h
@@ -29,7 +29,8 @@ namespace Kolab {
 typedef boost::shared_ptr<cDateTime> cDateTimePtr;
 
 typedef ::xsd::cxx::tree::type type;
-#if (XSD_INT_VERSION >= 4000000L)
+#if (defined(LIBXSD_VERSION) && LIBXSD_VERSION >= 4000000L) || \
+    (defined(XSD_INT_VERSION) && XSD_INT_VERSION >= 4000000L)
 typedef ::xsd::cxx::tree::simple_type< char, type > simple_type;
 #else
 typedef ::xsd::cxx::tree::simple_type< type > simple_type;
