Author: Andriy Grytsenko <andrej@rep.kiev.ua>
Description: DBus policy user should be not "0" but "root".

--- a/org.freedesktop.SystemToolsBackends.conf
+++ b/org.freedesktop.SystemToolsBackends.conf
@@ -22,14 +22,14 @@
     <allow own="org.freedesktop.SystemToolsBackends.UsersConfig2"/>
     <allow own="org.freedesktop.SystemToolsBackends.SelfConfig2"/>
     <allow own="org.freedesktop.SystemToolsBackends.Platform"/>
-    <allow send_interface="org.freedesktop.SystemToolsBackends.Platform" send_member="getPlatform"/>
+    <allow send_destination="org.freedesktop.SystemToolsBackends.Platform" send_member="getPlatform"/>
     -->
 
     <!-- Only allow talking to the dispatcher -->
     <allow send_destination="org.freedesktop.SystemToolsBackends"/>
   </policy>
 
-  <policy user="0">
+  <policy user="root">
     <!-- only root is allowed to run the dispatcher -->
     <allow own="org.freedesktop.SystemToolsBackends"/>
 
