From 3f3f5d6ff8695e99ddf533054943afe96a234c1d Mon Sep 17 00:00:00 2001
From: David Harder <davidjharder@gmail.com>
Date: Sat, 24 Aug 2019 17:31:28 -0500
Subject: [PATCH] Fix version number

---
 Makefile | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/Makefile b/Makefile
index 2564b8c..8f25131 100644
--- a/Makefile
+++ b/Makefile
@@ -1,4 +1,4 @@
-VERSION=1.9.6
+VERSION=1.9.7
 CFLAGS+=-g -DVERSION='"$(VERSION)"' -Wall -Wextra -Werror -Wno-unused-parameter
 LDFLAGS+=-static
 INCLUDE+=-Iinclude
-- 
2.23.0

