From d4b16ea605eadf62bf22eae36c5acd2f6715d6a5 Mon Sep 17 00:00:00 2001
From: Reilly Brogan <reilly@reillybrogan.com>
Date: Fri, 29 Dec 2023 17:19:48 -0600
Subject: [PATCH] Fix Wayland appId

Signed-off-by: Reilly Brogan <reilly@reillybrogan.com>
---
 src/yuzu/main.cpp | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/src/yuzu/main.cpp b/src/yuzu/main.cpp
index 059fcf041..142bb2484 100644
--- a/src/yuzu/main.cpp
+++ b/src/yuzu/main.cpp
@@ -5342,6 +5342,9 @@ int main(int argc, char* argv[]) {
     if (QString::fromLocal8Bit(qgetenv("DISPLAY")).isEmpty()) {
         qputenv("DISPLAY", ":0");
     }
+
+    // Fix the Wayland appId. This needs to match the name of the .desktop file sans the .desktop suffix.
+    QGuiApplication::setDesktopFileName(QStringLiteral("org.yuzu_emu.yuzu"));
 #endif
 
     SetHighDPIAttributes();
-- 
2.43.0

