From 2b50e584055596c0a9899c66bdf1038fe457cc81 Mon Sep 17 00:00:00 2001
From: Joey Riches <josephriches@gmail.com>
Date: Thu, 7 May 2026 12:33:40 +0100
Subject: [PATCH 1/1] Add bindings for download size property

---
 src/details.cpp | 5 +++++
 src/details.h   | 2 ++
 2 files changed, 7 insertions(+)

diff --git a/src/details.cpp b/src/details.cpp
index 18b6434..9bff82a 100644
--- a/src/details.cpp
+++ b/src/details.cpp
@@ -66,3 +66,8 @@ qulonglong PackageKit::Details::size() const
 {
     return value(QLatin1String("size")).toULongLong();
 }
+
+qulonglong PackageKit::Details::downloadSize() const
+{
+    return value(QLatin1String("download-size")).toULongLong();
+}
diff --git a/src/details.h b/src/details.h
index 67c7a57..c258ab9 100644
--- a/src/details.h
+++ b/src/details.h
@@ -48,6 +48,8 @@ public:
 
     qulonglong size() const;
 
+    qulonglong downloadSize() const;
+
 };
 
 }
-- 
2.53.0

