From 51f887e550ea85a3189a120b8b77cdab2aefbdd7 Mon Sep 17 00:00:00 2001
From: Fabio Forni <livingsilver94.solus@redaril.me>
Date: Sun, 24 Nov 2019 01:40:42 +0100
Subject: [PATCH] Default to openjdk-17

---
 apache-maven/src/bin/mvn | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/apache-maven/src/bin/mvn b/apache-maven/src/bin/mvn
index a554c66..c9dae7d 100755
--- a/apache-maven/src/bin/mvn
+++ b/apache-maven/src/bin/mvn
@@ -90,7 +90,7 @@ if $mingw ; then
 fi
 
 if [ -z "$JAVA_HOME" ] ; then
-  JAVACMD="`\\unset -f command; \\command -v java`"
+  JAVACMD='/usr/lib64/openjdk-17/bin/java'
 else
   JAVACMD="$JAVA_HOME/bin/java"
 fi
-- 
2.24.0

