diff --git a/apps/desktop/src/main/native-messaging.main.ts b/apps/desktop/src/main/native-messaging.main.ts
index 93525164ff..492d08a92e 100644
--- a/apps/desktop/src/main/native-messaging.main.ts
+++ b/apps/desktop/src/main/native-messaging.main.ts
@@ -417,26 +417,7 @@ export class NativeMessagingMain {
   }
 
   private binaryPath() {
-    const ext = process.platform === "win32" ? ".exe" : "";
-
-    if (isDev()) {
-      const devPath = path.join(
-        this.appPath,
-        "..",
-        "desktop_native",
-        "target",
-        "debug",
-        `desktop_proxy${ext}`,
-      );
-
-      // isDev() returns true when using a production build with ELECTRON_IS_DEV=1,
-      // so we need to fall back to the prod binary if the dev binary doesn't exist.
-      if (existsSync(devPath)) {
-        return devPath;
-      }
-    }
-
-    return path.join(path.dirname(this.exePath), `desktop_proxy${ext}`);
+    return "/usr/share/bitwarden-desktop/resources/app.asar.unpacked/desktop_proxy";
   }
 
   private homedir() {

