--- clients/tsconfig.json.old	2023-04-26 13:30:57.000000000 +0200
+++ clients/tsconfig.json	2023-04-26 20:15:39.879489082 +0200
@@ -6,7 +6,7 @@
     "target": "ES2016",
     "module": "ES2020",
     "lib": ["es5", "es6", "es7", "dom", "ES2021"],
-    "sourceMap": true,
+    "sourceMap": false,
     "allowSyntheticDefaultImports": true,
     "experimentalDecorators": true,
     "emitDecoratorMetadata": true,
--- clients/apps/desktop/webpack.renderer.js.old	2023-04-26 13:30:57.000000000 +0200
+++ clients/apps/desktop/webpack.renderer.js	2023-04-26 20:17:42.900973173 +0200
@@ -61,7 +61,7 @@
 
 const renderer = {
   mode: NODE_ENV,
-  devtool: "source-map",
+  devtool: false,
   target: "web",
   node: {
     __dirname: false,
@@ -138,7 +138,7 @@
     new AngularWebpackPlugin({
       tsConfigPath: "tsconfig.renderer.json",
       entryModule: "src/app/app.module#AppModule",
-      sourceMap: true,
+      sourceMap: false,
     }),
     // ref: https://github.com/angular/angular/issues/20357
     new webpack.ContextReplacementPlugin(
@@ -150,9 +150,6 @@
       filename: "index.html",
       chunks: ["app/vendor", "app/main"],
     }),
-    new webpack.SourceMapDevToolPlugin({
-      include: ["app/main.js"],
-    }),
     new MiniCssExtractPlugin({
       filename: "[name].[contenthash].css",
       chunkFilename: "[id].[contenthash].css",
--- clients/apps/desktop/tsconfig.json.old	2023-12-06 21:48:09.744042358 +0100
+++ clients/apps/desktop/tsconfig.json	2023-12-06 21:53:51.341419937 +0100
@@ -6,7 +6,7 @@
     "experimentalDecorators": true,
     "module": "ES2020",
     "target": "ES2016",
-    "sourceMap": true,
+    "removeComments": true,
     "types": [],
     "baseUrl": ".",
     "paths": {
