diff --git a/apps/desktop/webpack.renderer.js b/apps/desktop/webpack.renderer.js
index 555e1ed38b..35cd4d1c24 100644
--- a/apps/desktop/webpack.renderer.js
+++ b/apps/desktop/webpack.renderer.js
@@ -61,7 +61,7 @@ const common = {
 
 const renderer = {
   mode: NODE_ENV,
-  devtool: "source-map",
+  devtool: false,
   target: "web",
   node: {
     __dirname: false,
@@ -158,7 +158,7 @@ const renderer = {
     new AngularWebpackPlugin({
       tsConfigPath: "tsconfig.renderer.json",
       entryModule: "src/app/app.module#AppModule",
-      sourceMap: true,
+      sourceMap: false,
     }),
     // ref: https://github.com/angular/angular/issues/20357
     new webpack.ContextReplacementPlugin(
@@ -170,9 +170,6 @@ const renderer = {
       filename: "index.html",
       chunks: ["app/vendor", "app/main"],
     }),
-    new webpack.SourceMapDevToolPlugin({
-      include: ["app/main.js"],
-    }),
     new MiniCssExtractPlugin({
       filename: "[name].[contenthash].css",
       chunkFilename: "[id].[contenthash].css",
diff --git a/tsconfig.base.json b/tsconfig.base.json
index c462ab97d3..48719c4a17 100644
--- a/tsconfig.base.json
+++ b/tsconfig.base.json
@@ -15,7 +15,8 @@
     "baseUrl": ".",
     "resolveJsonModule": true,
     "allowJs": true,
-    "sourceMap": true,
+    "sourceMap": false,
+    "removeComments": true,
     "skipLibCheck": true,
     "paths": {
       "@bitwarden/admin-console/common": ["./libs/admin-console/src/common"],
