diff --git a/apps/desktop/webpack.base.js b/apps/desktop/webpack.base.js
index c9da84c..587b20a 100644
--- a/apps/desktop/webpack.base.js
+++ b/apps/desktop/webpack.base.js
@@ -167,7 +167,7 @@ module.exports.buildConfig = function buildConfig(params) {
   const rendererConfig = {
     name: "renderer",
     mode: NODE_ENV,
-    devtool: "source-map",
+    devtool: false,
     target: "web",
     node: {
       __dirname: false,
@@ -298,7 +298,7 @@ module.exports.buildConfig = function buildConfig(params) {
       new AngularWebpackPlugin({
         tsConfigPath: params.renderer.tsConfig,
         entryModule: params.renderer.entryModule,
-        sourceMap: true,
+        sourceMap: false,
       }),
       // ref: https://github.com/angular/angular/issues/20357
       new webpack.ContextReplacementPlugin(
@@ -310,9 +310,6 @@ module.exports.buildConfig = function buildConfig(params) {
         filename: "index.html",
         chunks: ["app/vendor", "app/main"],
       }),
-      new webpack.SourceMapDevToolPlugin({
-        include: ["app/main.js"],
-      }),
       new MiniCssExtractPlugin({
         filename: "[name].[contenthash].css",
         chunkFilename: "[id].[contenthash].css",
diff --git a/tsconfig.base.json b/tsconfig.base.json
index 2d105d4..a81dfda 100644
--- a/tsconfig.base.json
+++ b/tsconfig.base.json
@@ -15,7 +15,7 @@
     "baseUrl": ".",
     "resolveJsonModule": true,
     "allowJs": true,
-    "sourceMap": true,
+    "removeComments": true,
     "skipLibCheck": true,
     "paths": {
       "@bitwarden/admin-console/common": ["./libs/admin-console/src/common"],
