diff --git a/apps/desktop/webpack.base.js b/apps/desktop/webpack.base.js
index c9da84cd2e..fdf70ab19f 100644
--- a/apps/desktop/webpack.base.js
+++ b/apps/desktop/webpack.base.js
@@ -1,6 +1,5 @@
 const path = require("path");
 const webpack = require("webpack");
-const { merge } = require("webpack-merge");
 const HtmlWebpackPlugin = require("html-webpack-plugin");
 const MiniCssExtractPlugin = require("mini-css-extract-plugin");
 const CopyWebpackPlugin = require("copy-webpack-plugin");
@@ -167,7 +166,7 @@ module.exports.buildConfig = function buildConfig(params) {
   const rendererConfig = {
     name: "renderer",
     mode: NODE_ENV,
-    devtool: "source-map",
+    devtool: false,
     target: "web",
     node: {
       __dirname: false,
@@ -310,9 +309,6 @@ module.exports.buildConfig = function buildConfig(params) {
         filename: "index.html",
         chunks: ["app/vendor", "app/main"],
       }),
-      new webpack.SourceMapDevToolPlugin({
-        include: ["app/main.js"],
-      }),
       new MiniCssExtractPlugin({
         filename: "[name].[contenthash].css",
         chunkFilename: "[id].[contenthash].css",
