From 13c4ad18c9d1fc2cc9108eedaa8dc758b7c17ff6 Mon Sep 17 00:00:00 2001
From: beardypig <beardypig@users.noreply.github.com>
Date: Wed, 15 Feb 2017 12:43:29 +0000
Subject: [PATCH] setup: choose pycountry as a dependency using an environment
 variable

---
 setup.py | 7 +++++--
 1 file changed, 5 insertions(+), 2 deletions(-)

diff --git a/setup.py b/setup.py
index 7f0a168..7f1bf12 100644
--- a/setup.py
+++ b/setup.py
@@ -34,8 +34,11 @@
     deps.append("backports.shutil_get_terminal_size")
 
 # for localization
-deps.append("iso-639")
-deps.append("iso3166")
+if environ.get("STREAMLINK_USE_PYCOUNTRY"):
+    deps.append("pycountry")
+else:
+    deps.append("iso-639")
+    deps.append("iso3166")
 
 # When we build an egg for the Win32 bootstrap we don't want dependency
 # information built into it.
