diff --git a/Makefile b/Makefile
index 089ebe557..76b19d946 100644
--- a/Makefile
+++ b/Makefile
@@ -25,7 +25,7 @@ endif
 # Make version suffix -beta.NNNN.CCCCCCCC (N=Commit number, C=Commit)
 VERSION_SUFFIX := -beta.$(shell git rev-list --count HEAD).$(shell git show --no-patch --no-notes --pretty='%h' HEAD)
 # TAG is current version + commit number + commit + branch
-TAG := $(VERSION)$(VERSION_SUFFIX)$(TAG_BRANCH)
+TAG := $(VERSION)
 ifdef RELEASE_TAG
 	TAG := $(RELEASE_TAG)
 endif
