From bc152e8a7cc7ee9b484fe6f16bc6d6d5ed108c7a Mon Sep 17 00:00:00 2001
From: Ikey Doherty <ikey@solus-project.com>
Date: Sun, 8 Oct 2017 22:55:32 +0100
Subject: [PATCH 2/3] cfe: Use the GNU hash style for Solus binary build-ids

Signed-off-by: Ikey Doherty <ikey@solus-project.com>
---
 lib/Driver/ToolChains/Linux.cpp | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/lib/Driver/ToolChains/Linux.cpp b/lib/Driver/ToolChains/Linux.cpp
index bd414de..feb8cc8 100644
--- a/lib/Driver/ToolChains/Linux.cpp
+++ b/lib/Driver/ToolChains/Linux.cpp
@@ -239,6 +239,9 @@ Linux::Linux(const Driver &D, const llvm::Triple &Triple, const ArgList &Args)
     if (Distro.IsDebian() || Distro.IsOpenSUSE() || Distro == Distro::UbuntuLucid ||
         Distro == Distro::UbuntuJaunty || Distro == Distro::UbuntuKarmic)
       ExtraOpts.push_back("--hash-style=both");
+
+    // Solus defaults to GNU hash style
+    ExtraOpts.push_back("--hash-style=gnu");
   }
 
   if (Distro.IsRedhat() && Distro != Distro::RHEL5 && Distro != Distro::RHEL6)
-- 
2.14.2

