From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Muhammad Alfi Syahrin <malfisya.dev@hotmail.com>
Date: Fri, 11 Apr 2025 06:04:51 +0700
Subject: [PATCH] Add missing include

---
 kstars/kstarsinit.cpp | 1 +
 1 file changed, 1 insertion(+)

diff --git a/kstars/kstarsinit.cpp b/kstars/kstarsinit.cpp
index 1926739..ddeb379 100644
--- a/kstars/kstarsinit.cpp
+++ b/kstars/kstarsinit.cpp
@@ -33,6 +33,7 @@
 #include <KToggleAction>
 #include <KToolBar>
 
+#include <QActionGroup>
 #include <QMenu>
 #include <QStatusBar>
 
