--- a/src/sys/unix.rs
+++ b/src/sys/unix.rs
@@ -462,7 +462,7 @@ pub unsafe fn get_proc_address<T>(handle: NonNull<c_void>, name: &CStr) -> app
 
 pub fn load_libicuuc() -> apperr::Result<NonNull<c_void>> {
-    unsafe { load_library(c"libicuuc.so") }
+    unsafe { load_library(c"libicuuc.so.76") }
 }
 
 pub fn load_libicui18n() -> apperr::Result<NonNull<c_void>> {
-    unsafe { load_library(c"libicui18n.so") }
+    unsafe { load_library(c"libicui18n.so.76") }
}

