diff --git a/test/Makefile.inc b/test/Makefile.inc
index 820afdb..c625a68 100644
--- a/test/Makefile.inc
+++ b/test/Makefile.inc
@@ -16,7 +16,6 @@ TESTSOURCES = ares-test-main.cc	\
   ares-test-parse-txt.cc		\
   ares-test-parse-uri.cc		\
   ares-test-misc.cc			\
-  ares-test-live.cc			\
   ares-test-mock.cc			\
   ares-test-mock-ai.cc			\
   ares-test-internal.cc		\
diff --git a/test/ares-test-misc.cc b/test/ares-test-misc.cc
index b11b04b..b7cdd80 100644
--- a/test/ares-test-misc.cc
+++ b/test/ares-test-misc.cc
@@ -118,43 +118,6 @@ TEST_F(DefaultChannelTest, SetServersCSV) {
   EXPECT_EQ(ARES_EBADSTR, ares_set_servers_csv(channel_, "2.3.4.5,1.2.3.4:Z,3.4.5.6"));
 }
 
-TEST_F(DefaultChannelTest, TimeoutValue) {
-  struct timeval tinfo;
-  tinfo.tv_sec = 0;
-  tinfo.tv_usec = 0;
-  struct timeval tmax;
-  tmax.tv_sec = 0;
-  tmax.tv_usec = 10;
-  struct timeval* pt;
-
-  // No timers => get max back.
-  pt = ares_timeout(channel_, &tmax, &tinfo);
-  EXPECT_EQ(&tmax, pt);
-  EXPECT_EQ(0, pt->tv_sec);
-  EXPECT_EQ(10, pt->tv_usec);
-
-  pt = ares_timeout(channel_, nullptr, &tinfo);
-  EXPECT_EQ(nullptr, pt);
-
-  HostResult result;
-  ares_gethostbyname(channel_, "www.google.com.", AF_INET, HostCallback, &result);
-
-  // Now there's a timer running.
-  pt = ares_timeout(channel_, &tmax, &tinfo);
-  EXPECT_EQ(&tmax, pt);
-  EXPECT_EQ(0, pt->tv_sec);
-  EXPECT_EQ(10, pt->tv_usec);
-
-  tmax.tv_sec = 100;
-  pt = ares_timeout(channel_, &tmax, &tinfo);
-  EXPECT_EQ(&tinfo, pt);
-
-  pt = ares_timeout(channel_, nullptr, &tinfo);
-  EXPECT_EQ(&tinfo, pt);
-
-  Process();
-}
-
 TEST_F(LibraryTest, InetNtoP) {
   struct in_addr addr;
   addr.s_addr = htonl(0x01020304);
