From 2356d6d908ec864cf2eda9dc3b89ec73ecf30d70 Mon Sep 17 00:00:00 2001
From: Joshua Strobl <joshua@stroblindustries.com>
Date: Mon, 8 May 2017 15:12:19 +0300
Subject: [PATCH] Resolve issues with inability to signin with Google.


diff --git a/Source/WebCore/platform/UserAgentQuirks.cpp b/Source/WebCore/platform/UserAgentQuirks.cpp
index 4284a68..283f400 100644
--- a/Source/WebCore/platform/UserAgentQuirks.cpp
+++ b/Source/WebCore/platform/UserAgentQuirks.cpp
@@ -79,7 +79,7 @@ static bool urlRequiresChromeBrowser(const URL& url)
 
 static bool urlRequiresFirefoxBrowser(const URL& url)
 {
-    return isGoogle(url);
+    return isGoogle(url) && url.host() != "accounts.google.com";
 }
 
 static bool urlRequiresMacintoshPlatform(const URL& url)
-- 
2.12.2

