From: Adrian Perez de Castro <aperez@igalia.com>
Subject: Fix link error in 2.50.2
Origin: https://github.com/WebKit/WebKit/commit/730bffd856d2a1e56dd3bd2a0702282f19c5242a
Index: webkitgtk/Source/WebCore/Modules/credentialmanagement/BasicCredential.idl
===================================================================
--- webkitgtk.orig/Source/WebCore/Modules/credentialmanagement/BasicCredential.idl
+++ webkitgtk/Source/WebCore/Modules/credentialmanagement/BasicCredential.idl
@@ -25,7 +25,6 @@
 
 [
     InterfaceName=Credential,
-    CustomToJSObject,
     EnabledByQuirk=shouldExposeCredentialsContainer,
     Exposed=Window,
     SecureContext
