From e92dd63559a78342aa8b9cd4907efa6be0f4765b Mon Sep 17 00:00:00 2001
From: Gavin Zhao <git@gzgz.dev>
Date: Sun, 14 Dec 2025 11:37:03 -0500
Subject: [PATCH 2/2] [rocfft] Use envvar ROCFFT_CONCURRENCY for setting
 concurrency

---
 projects/rocfft/shared/concurrency.h | 9 +++++++++
 1 file changed, 9 insertions(+)

diff --git a/projects/rocfft/shared/concurrency.h b/projects/rocfft/shared/concurrency.h
index a36c7c1d9b..1f4f729df5 100644
--- a/projects/rocfft/shared/concurrency.h
+++ b/projects/rocfft/shared/concurrency.h
@@ -20,6 +20,7 @@
 
 #pragma once
 
+#include <string>
 #include <thread>
 
 #ifndef WIN32
@@ -32,6 +33,14 @@
 // return std::thread::hardware_concurrency().
 static unsigned int rocfft_concurrency()
 {
+	const char* env_concurrency = std::getenv("ROCFFT_CONCURRENCY");
+	if (env_concurrency)
+	{
+		try {
+			return (unsigned int)(std::stoi(env_concurrency));
+		} catch (...) {}
+	}
+
 #ifndef WIN32
     cpu_set_t cpuset;
     if(sched_getaffinity(0, sizeof(cpuset), &cpuset) == 0)
-- 
2.52.0

