diff -up firefox-40.0/mozilla-release/widget/gtk/gtk3drawing.c.old firefox-40.0/mozilla-release/widget/gtk/gtk3drawing.c
--- firefox-40.0/mozilla-release/widget/gtk/gtk3drawing.c.old	2015-08-07 13:50:18.000000000 +0200
+++ firefox-40.0/mozilla-release/widget/gtk/gtk3drawing.c	2015-08-07 13:53:05.920824588 +0200
@@ -2643,13 +2643,6 @@ moz_gtk_get_widget_border(GtkThemeWidget
             ensure_entry_widget();
             style = gtk_widget_get_style_context(gEntryWidget);
             moz_gtk_add_style_border(style, left, top, right, bottom);
-
-            /* Use the document padding in HTML
-               and GTK style padding in XUL. */
-            if (!inhtml) {
-                moz_gtk_add_style_padding(style, left, top, right, bottom);
-            }
-
             return MOZ_GTK_SUCCESS;
         }
     case MOZ_GTK_TREEVIEW:
