From 0a880afedde7694518db4269db280c1e657b4713 Mon Sep 17 00:00:00 2001
From: "Jan Alexander Steffens (heftig)" <jan.steffens@gmail.com>
Date: Mon, 27 Jan 2020 18:11:05 +0100
Subject: [PATCH 05/10] ZEN: INTERACTIVE: Use BFQ as the elevator for SQ
 devices

---
 block/elevator.c | 6 +++++-
 1 file changed, 5 insertions(+), 1 deletion(-)

diff --git a/block/elevator.c b/block/elevator.c
index 5ff093cb3cf8..483fe7ed8141 100644
--- a/block/elevator.c
+++ b/block/elevator.c
@@ -576,7 +576,11 @@ static struct elevator_type *elevator_get_default(struct request_queue *q)
 	    !blk_mq_is_shared_tags(q->tag_set->flags))
 		return NULL;
 
-	return elevator_find_get(q, "mq-deadline");
+#if defined(CONFIG_IOSCHED_BFQ)
+	return elevator_find_get(q, "bfq");
+#else
+ 	return elevator_find_get(q, "mq-deadline");
+#endif
 }
 
 /*
-- 
2.42.0

