--- a/cmake/draco_targets.cmake
+++ b/cmake/draco_targets.cmake
@@ -320,7 +320,7 @@ macro(draco_add_library)
     # VERSION and SOVERSION as necessary
     if((lib_TYPE STREQUAL BUNDLE OR lib_TYPE STREQUAL SHARED) AND NOT MSVC)
       set_target_properties(${lib_NAME}
-                            PROPERTIES VERSION ${DRACO_SOVERSION} SOVERSION
+                            PROPERTIES VERSION ${DRACO_VERSION} SOVERSION
                                        ${DRACO_SOVERSION_MAJOR})
     endif()
   endif()
