--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -1,4 +1,4 @@
-cmake_minimum_required(VERSION 3.12 FATAL_ERROR)
+cmake_minimum_required(VERSION 3.20.5 FATAL_ERROR)
 
 # Draco requires C++11.
 set(CMAKE_CXX_STANDARD 11)
--- a/cmake/draco_targets.cmake
+++ b/cmake/draco_targets.cmake
@@ -320,6 +320,7 @@ macro(draco_add_library)
 
   if(lib_TYPE STREQUAL SHARED AND NOT MSVC)
     set_target_properties(${lib_NAME} PROPERTIES SOVERSION ${DRACO_SOVERSION})
+    set_target_properties(${lib_NAME} PROPERTIES VERSION ${DRACO_VERSION})
   endif()
 
   if(BUILD_SHARED_LIBS AND (MSVC OR WIN32))
