diff --git a/include/settings/types/LayerIndex.h b/include/settings/types/LayerIndex.h
index 1e48bbc..adcffaf 100644
--- a/include/settings/types/LayerIndex.h
+++ b/include/settings/types/LayerIndex.h
@@ -5,6 +5,7 @@
 #define LAYERINDEX_H
 
 #include <functional>
+#include <fmt/format.h>
 
 namespace cura
 {
@@ -109,6 +110,12 @@ struct LayerIndex
     int value = 0;
 };
 
+constexpr auto format_as(LayerIndex index)
+{
+    return index.value;
+}
+
+
 }
 
 namespace std
