From 4b9f0fed275eca87525142ed2911c7072f6a5f80 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Gabriel=20F=C3=A9ron?= <feron.gabriel@gmail.com>
Date: Tue, 18 Jun 2019 22:54:59 +0200
Subject: [PATCH] [PATCH 2/3] Don't use -static-libstdc++

---
 CMakeLists.txt | 4 ----
 1 file changed, 4 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 99d234ec..ae9c3ece 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -101,10 +101,6 @@ if (ENABLE_MORE_COMPILER_OPTIMIZATION_FLAGS AND NOT (CMAKE_BUILD_TYPE_UPPER MATC
     endif()
 endif ()
 
-if(NOT APPLE AND NOT WIN32)
-    set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -static-libstdc++")
-endif()
-
 if (WIN32)
     add_definitions(-DNOMINMAX)
 endif()
-- 
2.21.0

