diff --git a/tensorboard/pip_package/setup.py b/tensorboard/pip_package/setup.py
index 2ded104..9817954 100644
--- a/tensorboard/pip_package/setup.py
+++ b/tensorboard/pip_package/setup.py
@@ -30,7 +30,7 @@ REQUIRED_PACKAGES = [
     'markdown >= 2.6.8',
     'numpy >= 1.12.0',
     'protobuf >= 3.6.0',
-    'setuptools >= 41.0.0',
+    'setuptools >= 40.6.2',
     'six >= 1.10.0',
     'werkzeug >= 0.11.15',
     # python3 specifically requires wheel 0.26
