From 2562fa4ddeba5b5945d9482baa4c26a414f5e831 Mon Sep 17 00:00:00 2001
From: Glyph <glyph@twistedmatrix.com>
Date: Sat, 11 Jun 2022 03:29:28 -0700
Subject: [PATCH] again

---
 setup.py | 5 ++---
 1 file changed, 2 insertions(+), 3 deletions(-)

diff --git a/setup.py b/setup.py
index ea3ffa8..c2f9816 100644
--- a/setup.py
+++ b/setup.py
@@ -11,13 +11,12 @@
     description="""
     Self-service finite-state machines for the programmer on the go.
     """.strip(),
-    long_description=long_description,
+    readme='README.md',
     packages=find_packages(exclude=[]),
     package_dir={'automat': 'automat'},
     setup_requires=[
-        'setuptools-scm',
-        'm2r',
         'wheel',
+        'setuptools-scm',
     ],
     install_requires=[
         "attrs>=19.2.0",
