From 0b740ea1b1f7e9532175db23193edfb3e83f49a6 Mon Sep 17 00:00:00 2001
From: Glyph <glyph@twistedmatrix.com>
Date: Sat, 11 Jun 2022 03:26:34 -0700
Subject: [PATCH] uh, what

---
 setup.py | 1 +
 1 file changed, 1 insertion(+)

diff --git a/setup.py b/setup.py
index 3fb7fdb..5cf7a45 100644
--- a/setup.py
+++ b/setup.py
@@ -25,6 +25,7 @@
     setup_requires=[
         'setuptools-scm',
         'm2r',
+        'wheel',
     ],
     install_requires=[
         "attrs>=19.2.0",
