From 1a687e23fef2895c8acf2f012d77a0f598d28642 Mon Sep 17 00:00:00 2001
From: Glyph <glyph@twistedmatrix.com>
Date: Sat, 11 Jun 2022 03:29:14 -0700
Subject: [PATCH] try to modernize

---
 setup.py | 8 --------
 1 file changed, 8 deletions(-)

diff --git a/setup.py b/setup.py
index 5cf7a45..ea3ffa8 100644
--- a/setup.py
+++ b/setup.py
@@ -4,14 +4,6 @@
 
 from setuptools import setup, find_packages
 
-try:
-    from m2r import parse_from_file
-    long_description = parse_from_file('README.md')
-except(IOError, ImportError):
-    print("\n\n!!! m2r not found, long_description is bad, don't upload this to PyPI !!!\n\n")
-    import io
-    long_description = io.open('README.md', encoding="utf-8").read()
-
 setup(
     name='Automat',
     use_scm_version=True,
