From: Luigi Baldoni <aloisio@gmx.com>
Date: 2018-06-18 13:44:22 +0200
Subject: Add missing include statement
Upstream: submitted

Qt 5.11 requires more explicit include statements

---
Index: mpc-qt-18.03/mainwindow.cpp
===================================================================
--- mpc-qt-18.03.orig/mainwindow.cpp
+++ mpc-qt-18.03/mainwindow.cpp
@@ -20,6 +20,7 @@
 #include <QMessageBox>
 #include <QLibraryInfo>
 #include <QToolTip>
+#include <QStyle>
 
 using namespace Helpers;
 
