From 6e312619b03ea85ef38978cbd782476ff74df450 Mon Sep 17 00:00:00 2001
From: Joey Riches <josephriches@gmail.com>
Date: Mon, 26 May 2025 12:01:20 +0100
Subject: [PATCH 1/1] etc: Add "Categories" to .desktop file

---
 etc/nsxiv.desktop | 1 +
 1 file changed, 1 insertion(+)

diff --git a/etc/nsxiv.desktop b/etc/nsxiv.desktop
index de90437..c455588 100644
--- a/etc/nsxiv.desktop
+++ b/etc/nsxiv.desktop
@@ -6,3 +6,4 @@ Exec=nsxiv %F
 MimeType=image/bmp;image/gif;image/jpeg;image/jpg;image/png;image/tiff;image/x-bmp;image/x-portable-anymap;image/x-portable-bitmap;image/x-portable-graymap;image/x-tga;image/x-xpixmap;image/webp;image/heic;image/svg+xml;application/postscript;image/jp2;image/jxl;image/avif;image/heif;
 NoDisplay=true
 Icon=nsxiv
+Categories=Graphics;Viewer
-- 
2.49.0

