From 773edfd533acf258f1d785092506c0746a93808c Mon Sep 17 00:00:00 2001
From: Jaime van Kessel <nallath@gmail.com>
Date: Mon, 8 Jun 2020 10:24:59 +0200
Subject: [PATCH] Prevent crash when camera tool was not found

---
 cura/CuraApplication.py | 5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

diff --git a/cura/CuraApplication.py b/cura/CuraApplication.py
index de7da13d1c..75b1d67697 100755
--- a/cura/CuraApplication.py
+++ b/cura/CuraApplication.py
@@ -886,8 +886,9 @@ def runWithGUI(self):
 
         # Initialize camera tool
         camera_tool = controller.getTool("CameraTool")
-        camera_tool.setOrigin(Vector(0, 100, 0))
-        camera_tool.setZoomRange(0.1, 2000)
+        if camera_tool:
+            camera_tool.setOrigin(Vector(0, 100, 0))
+            camera_tool.setZoomRange(0.1, 2000)
 
         # Initialize camera animations
         self._camera_animation = CameraAnimation.CameraAnimation()
