From 15c760dc994c4c4a240ddbb7aafeeaf303540460 Mon Sep 17 00:00:00 2001
From: Nikolai Magnussen <nikolai@magnussen.tf>
Date: Wed, 16 Aug 2017 12:59:06 +0200
Subject: [PATCH] Fixed a comparison where the wrong driver will be displayed

---
 doflicky/bundleset.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/doflicky/bundleset.py b/doflicky/bundleset.py
index 05f03f5..e501435 100644
--- a/doflicky/bundleset.py
+++ b/doflicky/bundleset.py
@@ -60,7 +60,7 @@ class BundleSet:
             otherDrivers = [x for x in self.uniqueDrivers if x.get_base() == b]
 
             for existing in otherDrivers:
-                if existing.get_priority() > driver:
+                if existing.get_priority() > driver.get_priority():
                     print("DEBUG: {} shadowed by existing {}".format(
                         driver.get_name(), existing.get_name()))
                     return
-- 
2.14.1

