From e404a3ab3c7edfb6d6021927e861ed9ed2bd1923 Mon Sep 17 00:00:00 2001
From: Ikey Doherty <ikey@solus-project.com>
Date: Sun, 16 Oct 2016 07:20:48 +0100
Subject: [PATCH] Fix discoverability for MATE desktop

Signed-off-by: Ikey Doherty <ikey@solus-project.com>
---
 doflicky.desktop | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/doflicky.desktop b/doflicky.desktop
index 502d00a..65e83eb 100644
--- a/doflicky.desktop
+++ b/doflicky.desktop
@@ -7,5 +7,5 @@ Terminal=false
 Type=Application
 StartupNotify=false
 Icon=cs-cat-hardware
-Categories=GNOME;GTK;System;PackageManager;
+Categories=GTK;Settings;HardwareSettings;
 Keywords=Driver;GPU;Hardware;Preferences;Install;Uninstall;Program;Software;App;Store;
-- 
2.10.1

