From d05c9d097277ec30409b4b4c46634ab66fa5986d Mon Sep 17 00:00:00 2001
From: Ikey Doherty <ikey@solus-project.com>
Date: Mon, 17 Apr 2017 05:43:14 +0100
Subject: [PATCH] doflicky: Fix broken signal name

Signed-off-by: Ikey Doherty <ikey@solus-project.com>
---
 doflicky/window.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/doflicky/window.py b/doflicky/window.py
index a20eff7..34af14c 100644
--- a/doflicky/window.py
+++ b/doflicky/window.py
@@ -133,7 +133,7 @@ closed source code."""
         # update page..
         page = OpPage()
         self.op_page = page
-        self.op_page.connect('complete', self.finished_handler)
+        self.op_page.connect('basket-changed', self.finished_handler)
         self.op_page.connect('cancelled', self.cancelled_handler)
         self.stack.add_named(page, "operations")
         self.show_all()
-- 
2.12.2

