From 1943768d8668aa390032bc032dfc5b92eebede03 Mon Sep 17 00:00:00 2001
From: Ikey Doherty <ikey@solus-project.com>
Date: Sun, 16 Oct 2016 07:51:19 +0100
Subject: [PATCH 2/2] Ensure we show in both MATE & Budgie

Signed-off-by: Ikey Doherty <ikey@solus-project.com>
---
 doflicky.desktop | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/doflicky.desktop b/doflicky.desktop
index 65e83eb..d90270c 100644
--- a/doflicky.desktop
+++ b/doflicky.desktop
@@ -7,5 +7,5 @@ Terminal=false
 Type=Application
 StartupNotify=false
 Icon=cs-cat-hardware
-Categories=GTK;Settings;HardwareSettings;
+Categories=GTK;Settings;HardwareSettings;X-GNOME-Settings-Panel;
 Keywords=Driver;GPU;Hardware;Preferences;Install;Uninstall;Program;Software;App;Store;
-- 
2.10.1

