From c45b61223b49e470e5f97a870bbbafce3952836b Mon Sep 17 00:00:00 2001
From: Reilly Brogan <reilly@reillybrogan.com>
Date: Tue, 23 May 2023 10:37:21 -0500
Subject: [PATCH] Solus tweak to launchfeedback config

---
 kcms/cursortheme/launchfeedbacksettings.kcfg | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/kcms/cursortheme/launchfeedbacksettings.kcfg b/kcms/cursortheme/launchfeedbacksettings.kcfg
index 0d70303de..fc29d8ba2 100644
--- a/kcms/cursortheme/launchfeedbacksettings.kcfg
+++ b/kcms/cursortheme/launchfeedbacksettings.kcfg
@@ -13,7 +13,7 @@ SPDX-License-Identifier: LGPL-2.0-only
       <default>false</default>
     </entry>
     <entry name="bouncing" key="Bouncing" type="Bool">
-      <default>true</default>
+      <default>false</default>
     </entry>
     <entry name="cursorTimeout" key="Timeout" type="Int">
       <label>Timeout in seconds</label>
-- 
2.40.1

