From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Reilly Brogan <reilly@reillybrogan.com>
Date: Sat, 2 Mar 2024 11:35:42 -0600
Subject: [PATCH] Use stateless dir for menu

---
 menu/desktop/CMakeLists.txt | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/menu/desktop/CMakeLists.txt b/menu/desktop/CMakeLists.txt
index 911532bacc..4aff4bebc6 100644
--- a/menu/desktop/CMakeLists.txt
+++ b/menu/desktop/CMakeLists.txt
@@ -42,4 +42,4 @@
    kf5-utilities-xutils.directory
 DESTINATION  ${KDE_INSTALL_DESKTOPDIR} )
 
-install( FILES plasma-applications.menu  DESTINATION  ${KDE_INSTALL_SYSCONFDIR}/xdg/menus )
+install( FILES plasma-applications.menu  DESTINATION  /usr/share/xdg/menus )
