From 14ba6766d9764e00d53783b98e6334faa564e22a Mon Sep 17 00:00:00 2001
From: Peter O'Connor <peter@solus-project.com>
Date: Fri, 19 Jan 2018 10:49:00 +1100
Subject: [PATCH] Use plasma wallpaper as background

Signed-off-by: Peter O'Connor <peter@solus-project.com>
---
 sddm-theme/theme.conf | 5 ++---
 1 file changed, 2 insertions(+), 3 deletions(-)

diff --git a/sddm-theme/theme.conf b/sddm-theme/theme.conf
index c293b25..4978480 100644
--- a/sddm-theme/theme.conf
+++ b/sddm-theme/theme.conf
@@ -1,4 +1,3 @@
 [General]
-type=color
-color=#1d99f3
-background=
+background=/usr/share/wallpapers/Next/contents/images/3840x2160.png
+type=image
-- 
2.15.1

