From 11c98900aab40a20114117869e7c8f0b37a4f94d Mon Sep 17 00:00:00 2001
From: "flinux@vongellhorn.ch" <flinux@vongellhorn.ch>
Date: Tue, 30 Jul 2019 18:26:33 +0200
Subject: [PATCH 1/1] Change default background

---
 sddm-theme/theme.conf.cmake | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/sddm-theme/theme.conf.cmake b/sddm-theme/theme.conf.cmake
index ea9a943..8aefea9 100644
--- a/sddm-theme/theme.conf.cmake
+++ b/sddm-theme/theme.conf.cmake
@@ -2,4 +2,4 @@
 type=image
 color=#1d99f3
 fontSize=10
-background=${CMAKE_INSTALL_PREFIX}/${WALLPAPER_INSTALL_DIR}/Next/contents/images/5120x2880.png
+background=${CMAKE_INSTALL_PREFIX}/${WALLPAPER_INSTALL_DIR}/SolusPlasma/contents/images/1920x1080.png
-- 
2.22.0

