From 71a88b64e9f851d12e0f3435ddcec5268da101ab Mon Sep 17 00:00:00 2001
From: "F. von Gellhorn" <flinux@vongellhorn.ch>
Date: Sun, 16 Feb 2020 00:12:39 +0100
Subject: [PATCH 1/1] Change default sddm background

---
 sddm-theme/theme.conf.cmake | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/sddm-theme/theme.conf.cmake b/sddm-theme/theme.conf.cmake
index 8494a5c..f3a54df 100644
--- a/sddm-theme/theme.conf.cmake
+++ b/sddm-theme/theme.conf.cmake
@@ -4,5 +4,5 @@ logo=${KDE_INSTALL_FULL_DATADIR}/sddm/themes/breeze/default-logo.svg
 type=image
 color=#1d99f3
 fontSize=10
-background=${KDE_INSTALL_FULL_WALLPAPERDIR}/Next/contents/images/5120x2880.jpg
+background=${KDE_INSTALL_FULL_WALLPAPERDIR}/SolusPlasma/contents/images/1920x1080.jpg
 needsFullUserModel=false
-- 
2.25.0

