From 9b287377343fd796d359e67de54d477d8084cfcd Mon Sep 17 00:00:00 2001
From: Benjamin Barenblat <bbaren@google.com>
Date: Fri, 3 Aug 2018 11:22:37 -0400
Subject: [PATCH] Add FAT to mountpoint file system whitelist

---
 util/fusermount.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/util/fusermount.c b/util/fusermount.c
index 8dce92f6..45b54e3a 100644
--- a/util/fusermount.c
+++ b/util/fusermount.c
@@ -1033,6 +1033,7 @@ static int check_perm(const char **mntp, struct stat *stbuf, int *mountpoint_fd)
 		0x9123683E /* BTRFS_SUPER_MAGIC */,
 		0x00C36400 /* CEPH_SUPER_MAGIC */,
 		0xFF534D42 /* CIFS_MAGIC_NUMBER */,
+		0X00004D44 /* MSDOS_SUPER_MAGIC */,
 		0x0000F15F /* ECRYPTFS_SUPER_MAGIC */,
 		0x0000EF53 /* EXT[234]_SUPER_MAGIC */,
 		0xF2F52010 /* F2FS_SUPER_MAGIC */,
