=== modified file 'Onboard/Indicator.py'
--- old/Onboard/Indicator.py	2017-02-14 21:42:14 +0000
+++ new/Onboard/Indicator.py	2023-09-07 15:10:09 +0000
@@ -294,7 +294,10 @@
         BackendBase.__init__(self, menu)
 
         try:
-            from gi.repository import AppIndicator3 as AppIndicator
+            try:
+                from gi.repository import AyatanaAppIndicator3 as AppIndicator
+            except ImportError:
+                from gi.repository import AppIndicator3 as AppIndicator
         except ImportError as ex:
             raise RuntimeError(ex)
 

=== modified file 'Onboard/Version.py'
--- old/Onboard/Version.py	2016-02-29 11:54:24 +0000
+++ new/Onboard/Version.py	2023-09-07 15:08:21 +0000
@@ -35,7 +35,10 @@
 
     # AppIndicator3 is not required
     try:
-        gi.require_version('AppIndicator3', '0.1')
+        try:
+            gi.require_version('AyatanaAppIndicator3', '0.1')
+        except ValueError:
+            gi.require_version('AppIndicator3', '0.1')
     except ValueError:
         pass
 

