From 6e650618f109785b0b7e72d09c88dcbed1c0ca37 Mon Sep 17 00:00:00 2001
From: Peter O'Connor <peter@solus-project.com>
Date: Wed, 7 Feb 2018 23:21:34 +1100
Subject: [PATCH] Use installed path for ckb-animations

Signed-off-by: Peter O'Connor <peter@solus-project.com>
---
 src/ckb/animscript.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/ckb/animscript.cpp b/src/ckb/animscript.cpp
index f49a64c..03263ef 100644
--- a/src/ckb/animscript.cpp
+++ b/src/ckb/animscript.cpp
@@ -30,7 +30,7 @@ QString AnimScript::path(){
 #ifdef __APPLE__
     return QDir(QApplication::applicationDirPath() + "/../Resources").absoluteFilePath("ckb-animations");
 #else
-    return QDir("/usr/lib").absoluteFilePath("ckb-animations");
+    return QDir("/usr/share/ckb").absoluteFilePath("ckb-animations");
 #endif
 }

--
2.16.1
