From 479f79915c4d127e2373b8ce79a651f649a65837 Mon Sep 17 00:00:00 2001
From: Peter O'Connor <peter@solus-project.com>
Date: Mon, 28 May 2018 17:56:50 +1000
Subject: [PATCH] Use Solus flags

Signed-off-by: Peter O'Connor <peter@solus-project.com>
---
 libraries/liblmdb/Makefile | 7 +------
 1 file changed, 1 insertion(+), 6 deletions(-)

diff --git a/libraries/liblmdb/Makefile b/libraries/liblmdb/Makefile
index f254511..f2fb937 100644
--- a/libraries/liblmdb/Makefile
+++ b/libraries/liblmdb/Makefile
@@ -18,12 +18,7 @@
 # There may be other macros in mdb.c of interest. You should
 # read mdb.c before changing any of them.
 #
-CC	= gcc
-AR	= ar
-W	= -W -Wall -Wno-unused-parameter -Wbad-function-cast -Wuninitialized
-THREADS = -pthread
-OPT = -O2 -g
-CFLAGS	= $(THREADS) $(OPT) $(W) $(XCFLAGS)
+CFLAGS += -pthread
 LDLIBS	=
 SOLIBS	=
 SOEXT	= .so
--
2.17.0
