From 8133d42c1a0253503581bbb9445ae981a9c453cc Mon Sep 17 00:00:00 2001
From: Evan Maddock <maddock.evan@vivaldi.net>
Date: Tue, 23 Nov 2021 13:49:31 -0500
Subject: [PATCH] Remove update command functionality

Signed-off-by: Evan Maddock <maddock.evan@vivaldi.net>
---
 cmd/commands.go | 49 -------------------------------------------------
 1 file changed, 49 deletions(-)

diff --git a/cmd/commands.go b/cmd/commands.go
index 1e2c40de..1cbe6cb8 100644
--- a/cmd/commands.go
+++ b/cmd/commands.go
@@ -288,55 +288,6 @@ is always printed to stdout.`,
 			return fs
 		}(),
 	})
-
-	RegisterCommand(Command{
-		Name:  "upgrade",
-		Func:  cmdUpgrade,
-		Short: "Upgrade Caddy (EXPERIMENTAL)",
-		Long: `
-Downloads an updated Caddy binary with the same modules/plugins at the
-latest versions. EXPERIMENTAL: May be changed or removed.`,
-		Flags: func() *flag.FlagSet {
-			fs := flag.NewFlagSet("upgrade", flag.ExitOnError)
-			fs.Bool("keep-backup", false, "Keep the backed up binary, instead of deleting it")
-			return fs
-		}(),
-	})
-
-	RegisterCommand(Command{
-		Name:  "add-package",
-		Func:  cmdAddPackage,
-		Usage: "<packages...>",
-		Short: "Adds Caddy packages (EXPERIMENTAL)",
-		Long: `
-Downloads an updated Caddy binary with the specified packages (module/plugin)
-added. Retains existing packages. Returns an error if the any of packages are 
-already included. EXPERIMENTAL: May be changed or removed.
-`,
-		Flags: func() *flag.FlagSet {
-			fs := flag.NewFlagSet("add-package", flag.ExitOnError)
-			fs.Bool("keep-backup", false, "Keep the backed up binary, instead of deleting it")
-			return fs
-		}(),
-	})
-
-	RegisterCommand(Command{
-		Name:  "remove-package",
-		Func:  cmdRemovePackage,
-		Usage: "<packages...>",
-		Short: "Removes Caddy packages (EXPERIMENTAL)",
-		Long: `
-Downloads an updated Caddy binaries without the specified packages (module/plugin). 
-Returns an error if any of the packages are not included. 
-EXPERIMENTAL: May be changed or removed.
-`,
-		Flags: func() *flag.FlagSet {
-			fs := flag.NewFlagSet("remove-package", flag.ExitOnError)
-			fs.Bool("keep-backup", false, "Keep the backed up binary, instead of deleting it")
-			return fs
-		}(),
-	})
-
 }
 
 // RegisterCommand registers the command cmd.
-- 
2.34.0

