From 033f5581238eac67084a289009f77b0191bcd467 Mon Sep 17 00:00:00 2001
From: Evan Maddock <maddock.evan@vivaldi.net>
Date: Wed, 30 Aug 2023 15:36:29 -0400
Subject: [PATCH] Remove update command functionality

Signed-off-by: Evan Maddock <maddock.evan@vivaldi.net>
---
 cmd/commands.go | 44 --------------------------------------------
 1 file changed, 44 deletions(-)

diff --git a/cmd/commands.go b/cmd/commands.go
index 0885577e..270a83c0 100644
--- a/cmd/commands.go
+++ b/cmd/commands.go
@@ -383,50 +383,6 @@ is always printed to stdout.
 		},
 	})
 
-	RegisterCommand(Command{
-		Name:  "upgrade",
-		Short: "Upgrade Caddy (EXPERIMENTAL)",
-		Long: `
-Downloads an updated Caddy binary with the same modules/plugins at the
-latest versions. EXPERIMENTAL: May be changed or removed.
-`,
-		CobraFunc: func(cmd *cobra.Command) {
-			cmd.Flags().BoolP("keep-backup", "k", false, "Keep the backed up binary, instead of deleting it")
-			cmd.RunE = WrapCommandFuncForCobra(cmdUpgrade)
-		},
-	})
-
-	RegisterCommand(Command{
-		Name:  "add-package",
-		Usage: "<packages...>",
-		Short: "Adds Caddy packages (EXPERIMENTAL)",
-		Long: `
-Downloads an updated Caddy binary with the specified packages (module/plugin)
-added. Retains existing packages. Returns an error if the any of packages are 
-already included. EXPERIMENTAL: May be changed or removed.
-`,
-		CobraFunc: func(cmd *cobra.Command) {
-			cmd.Flags().BoolP("keep-backup", "k", false, "Keep the backed up binary, instead of deleting it")
-			cmd.RunE = WrapCommandFuncForCobra(cmdAddPackage)
-		},
-	})
-
-	RegisterCommand(Command{
-		Name:  "remove-package",
-		Func:  cmdRemovePackage,
-		Usage: "<packages...>",
-		Short: "Removes Caddy packages (EXPERIMENTAL)",
-		Long: `
-Downloads an updated Caddy binaries without the specified packages (module/plugin). 
-Returns an error if any of the packages are not included. 
-EXPERIMENTAL: May be changed or removed.
-`,
-		CobraFunc: func(cmd *cobra.Command) {
-			cmd.Flags().BoolP("keep-backup", "k", false, "Keep the backed up binary, instead of deleting it")
-			cmd.RunE = WrapCommandFuncForCobra(cmdRemovePackage)
-		},
-	})
-
 	RegisterCommand(Command{
 		Name:  "manpage",
 		Usage: "--directory <path>",
-- 
2.42.0

