From 70493600072206127f55177a602b2941ebdb90f0 Mon Sep 17 00:00:00 2001
From: Pierre-Yves <pyu@riseup.net>
Date: Tue, 29 Dec 2020 16:46:52 +0100
Subject: [PATCH] ensurepip: Always install pip/setuptools ignoring installed
 packages

Signed-off-by: Pierre-Yves <pyu@riseup.net>
---
 Lib/ensurepip/__init__.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/Lib/ensurepip/__init__.py b/Lib/ensurepip/__init__.py
index 94d40b0..e3dc428 100644
--- a/Lib/ensurepip/__init__.py
+++ b/Lib/ensurepip/__init__.py
@@ -114,7 +114,7 @@ def _bootstrap(*, root=None, upgrade=False, user=False,
             additional_paths.append(os.path.join(tmpdir, wheel_name))
 
         # Construct the arguments to be passed to the pip command
-        args = ["install", "--no-cache-dir", "--no-index", "--find-links", tmpdir]
+        args = ["install", "-I", "--no-index", "--find-links", tmpdir]
         if root:
             args += ["--root", root]
         if upgrade:
-- 
2.29.2

