---
 Lib/distutils/util.py |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

Index: Python-3.11.8/Lib/distutils/util.py
===================================================================
--- Python-3.11.8.orig/Lib/distutils/util.py
+++ Python-3.11.8/Lib/distutils/util.py
@@ -436,7 +436,7 @@ byte_compile(files, optimize=%r, force=%
     else:
         from py_compile import compile
 
-        for file in py_files:
+        for file in sorted(py_files):
             if file[-3:] != ".py":
                 # This lets us be lazy and not filter filenames in
                 # the "install_lib" command.
