From 9474076d5bdbbd43070f554b297c98c215bc06f1 Mon Sep 17 00:00:00 2001
From: "flinux@vongellhorn.ch" <flinux@vongellhorn.ch>
Date: Sun, 13 Oct 2019 19:03:15 +0200
Subject: [PATCH 2/2] We use qmake rather than qmake-qt5

---
 modules/ECMQueryQmake.cmake | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/modules/ECMQueryQmake.cmake b/modules/ECMQueryQmake.cmake
index 74a6df8..1f8ebf7 100644
--- a/modules/ECMQueryQmake.cmake
+++ b/modules/ECMQueryQmake.cmake
@@ -1,7 +1,7 @@
 find_package(Qt5Core QUIET)
 
 if (Qt5Core_FOUND)
-    set(_qmake_executable_default "qmake-qt5")
+    set(_qmake_executable_default "qmake")
 endif ()
 if (TARGET Qt5::qmake)
     get_target_property(_qmake_executable_default Qt5::qmake LOCATION)
-- 
2.23.0

