From bad9604f5679e11572106e7d0da2efa515238d24 Mon Sep 17 00:00:00 2001
From: Campbell Jones <git@serebit.com>
Date: Tue, 22 Sep 2020 10:41:47 -0400
Subject: [PATCH] dont use lto and or pgo

---
 CMakeLists.txt | 5 -----
 1 file changed, 5 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index d6f7c596..1884ce13 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -439,11 +439,6 @@ IF("${CMAKE_MAJOR_VERSION}.${CMAKE_MINOR_VERSION}" VERSION_GREATER 3.7)
 endif()
 
 
-#
-# Configure the build system to use LTO and/or PGO while building LDC
-#
-include(HandleLTOPGOBuildOptions)
-
 #
 # Enable Dynamic compilation if supported for this platform and LLVM version.
 #
-- 
2.27.0

