From 2a53312fa99b749fe67f352adbec63bd68df104b Mon Sep 17 00:00:00 2001
From: Joey Riches <josephriches@gmail.com>
Date: Sat, 9 Oct 2021 11:18:01 +0100
Subject: [PATCH 1/1] dont use lto and or pgo

---
 CMakeLists.txt | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 55bb50b..cfe0c5f 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -444,7 +444,7 @@ endif()
 # Configure the build system to use LTO and/or PGO while building LDC
 #
 set(DFLAGS_LDC "")
-include(HandleLTOPGOBuildOptions)
+#include(HandleLTOPGOBuildOptions)
 
 #
 # Enable Dynamic compilation if supported for this platform and LLVM version.
-- 
2.32.0

