From 5c2f2fc1860f1a14d5a0abac71b6b49f7bd9e3cf Mon Sep 17 00:00:00 2001
From: Joey Riches <josephriches@gmail.com>
Date: Tue, 9 Oct 2018 21:05:57 +0100
Subject: [PATCH] bash-hack

---
 CMakeLists.txt | 11 -----------
 1 file changed, 11 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index cf2e53f1c..0ea36e2e8 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -608,17 +608,6 @@ if(MSVC)
     configure_file(vcbuild/x86.bat     ${PROJECT_BINARY_DIR}/bin/x86.bat     COPYONLY)
 endif()
 
-if(${CMAKE_SYSTEM_NAME} MATCHES "Linux")
-    find_package(bash-completion QUIET)
-    if(NOT BASH_COMPLETION_FOUND)
-        set(BASH_COMPLETION_COMPLETIONSDIR "${CONF_INST_DIR}/bash_completion.d")
-        if(LINUX_DISTRIBUTION_IS_GENTOO)
-            set(BASH_COMPLETION_COMPLETIONSDIR "/usr/share/bash-completion")
-        endif()
-    endif()
-    install(DIRECTORY bash_completion.d/ DESTINATION ${BASH_COMPLETION_COMPLETIONSDIR})
-endif()
-
 #
 # Packaging
 #
