diff -ru ldc-1.15.0-src~/driver/configfile.cpp ldc-1.15.0-src/driver/configfile.cpp
--- ldc-1.15.0-src~/driver/configfile.cpp	2019-04-06 08:58:35.000000000 -0700
+++ ldc-1.15.0-src/driver/configfile.cpp	2019-08-12 12:12:34.802450861 -0700
@@ -166,6 +166,10 @@
   // try /etc/ldc (absolute path)
   p = "/etc/ldc";
   APPEND_FILENAME_AND_RETURN_IF_EXISTS
+
+  // try /usr/share/defaults/etc (stateless path)
+  p = "/usr/share/defaults/etc";
+  APPEND_FILENAME_AND_RETURN_IF_EXISTS
 #endif
 
 #undef APPEND_FILENAME_AND_RETURN_IF_EXISTS
