Index: cairo-1.12.8/util/cairo-fdr/Makefile.am
===================================================================
--- cairo-1.12.8.orig/util/cairo-fdr/Makefile.am
+++ cairo-1.12.8/util/cairo-fdr/Makefile.am
@@ -9,7 +9,7 @@ AM_CPPFLAGS = -I$(top_srcdir)/src \
 cairo_fdr_la_SOURCES = fdr.c
 cairo_fdr_la_CPPFLAGS = $(AM_CPPFLAGS)
 cairo_fdr_la_CFLAGS = $(CAIRO_CFLAGS)
-cairo_fdr_la_LDFLAGS = -module -no-undefined
+cairo_fdr_la_LDFLAGS = -module -no-undefined -avoid-version
 if CAIRO_HAS_DL
 cairo_fdr_la_LIBADD = -ldl
 endif
Index: cairo-1.12.8/util/cairo-trace/Makefile.am
===================================================================
--- cairo-1.12.8.orig/util/cairo-trace/Makefile.am
+++ cairo-1.12.8/util/cairo-trace/Makefile.am
@@ -11,7 +11,7 @@ libcairo_trace_la_SOURCES = trace.c
 libcairo_trace_la_CPPFLAGS = -DCAIRO_TRACE_OUTDIR="\"$(cairooutdir)\"" \
 			  $(AM_CPPFLAGS)
 libcairo_trace_la_CFLAGS = $(CAIRO_CFLAGS) $(real_pthread_CFLAGS)
-libcairo_trace_la_LDFLAGS = -no-undefined
+libcairo_trace_la_LDFLAGS = -no-undefined -module -avoid-version
 
 libcairo_trace_la_LIBADD = $(real_pthread_LIBS) -lz
 if CAIRO_HAS_DL
Index: cairo-1.12.8/util/cairo-sphinx/Makefile.am
===================================================================
--- cairo-1.12.8.orig/util/cairo-sphinx/Makefile.am
+++ cairo-1.12.8/util/cairo-sphinx/Makefile.am
@@ -11,7 +11,7 @@ AM_CPPFLAGS = -I$(top_srcdir)/src \
 cairo_sphinx_la_SOURCES = fdr.c
 cairo_sphinx_la_CPPFLAGS = $(AM_CPPFLAGS)
 cairo_sphinx_la_CFLAGS = $(CAIRO_CFLAGS)
-cairo_sphinx_la_LDFLAGS = -module -no-undefined
+cairo_sphinx_la_LDFLAGS = -module -no-undefined -avoid-version
 if CAIRO_HAS_DL
 cairo_sphinx_la_LIBADD = -ldl
 endif
