From 20ffb52bc4d77d99eacc10b8b39e66fe876f1ca8 Mon Sep 17 00:00:00 2001
From: Reilly Brogan <reilly@reillybrogan.com>
Date: Sat, 14 Oct 2023 19:22:11 -0500
Subject: [PATCH] minizip-ng compat

---
 src/Exporters/ExportEPUB.cpp | 5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

diff --git a/src/Exporters/ExportEPUB.cpp b/src/Exporters/ExportEPUB.cpp
index 1ff1055d1..e26b92205 100644
--- a/src/Exporters/ExportEPUB.cpp
+++ b/src/Exporters/ExportEPUB.cpp
@@ -27,7 +27,8 @@
 #include <string>
 #include <string.h>
 
-#include <zip.h>
+#include <minizip/zip.h>
+#include <zlib.h>
 #ifdef _WIN32
 #include <iowin32.h>
 #endif
@@ -145,7 +146,7 @@ void ExportEPUB::SaveFolderAsEpubToLocation(const QString &fullfolderpath, const
     fileInfo.tmz_date.tm_year = timeNow.date().year();
 
      // Write the mimetype. This must be uncompressed and the first entry in the archive.
-    if (zipOpenNewFileInZip64(zfile, "mimetype", &fileInfo, NULL, 0, NULL, 0, NULL, Z_NO_COMPRESSION, 0, 0) != ZIP_OK) {
+    if (zipOpenNewFileInZip_64(zfile, "mimetype", &fileInfo, NULL, 0, NULL, 0, NULL, Z_NO_COMPRESSION, 0, 0) != ZIP_OK) {
         zipClose(zfile, NULL);
         QFile::remove(tempFile);
         throw(CannotStoreFile("mimetype"));
-- 
2.42.0

