From c5f2ec64ad25276409a2501886192f5dc7c08b31 Mon Sep 17 00:00:00 2001
From: Kevin Hendricks <kevinhendricks@users.noreply.github.com>
Date: Tue, 21 Mar 2017 15:02:43 -0400
Subject: [PATCH] finally update the build version

---
 CMakeLists.txt | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index af9e488..8ada3af 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -29,7 +29,7 @@ set( CMAKE_MODULE_PATH ${CMAKE_MODULE_PATH} "${CMAKE_CURRENT_SOURCE_DIR}/cmake_e
 
 set( SIGIL_MAJOR_VERSION 0 )
 set( SIGIL_MINOR_VERSION 9 )
-set( SIGIL_REVISION_VERSION 7 )
+set( SIGIL_REVISION_VERSION 8 )
 set( SIGIL_FULL_VERSION ${SIGIL_MAJOR_VERSION}.${SIGIL_MINOR_VERSION}.${SIGIL_REVISION_VERSION} )
 
 if( UNIX AND NOT APPLE )
