From 86d266c2e0ad4cd1a85a415244782355f4f092fb Mon Sep 17 00:00:00 2001
From: Ikey Doherty <ikey@solus-project.com>
Date: Sat, 15 Oct 2016 22:57:48 +0100
Subject: [PATCH] Do not start on virtualized environments

Signed-off-by: Ikey Doherty <ikey@solus-project.com>
---
 data/thermald.service.in | 1 +
 1 file changed, 1 insertion(+)

diff --git a/data/thermald.service.in b/data/thermald.service.in
index 17d640a..27c9332 100644
--- a/data/thermald.service.in
+++ b/data/thermald.service.in
@@ -1,5 +1,6 @@
 [Unit]
 Description=Thermal Daemon Service
+ConditionVirtualization=no
 
 [Service]
 Type=dbus
-- 
2.10.1

