diff --git a/bin/coin-config b/bin/coin-config
index 810914e..534a34f 100755
--- a/bin/coin-config
+++ b/bin/coin-config
@@ -17,7 +17,7 @@
 program="$0"
 me=`echo "$0" | sed 's,^.*[/\\],,g'`
 wd=`echo "$0" | sed 's,/[^/]*$,,'`
-prefix=`cd "$wd/.."; pwd`
+prefix="/usr/"
 
 hello=`echo hello | fmt -t -w70 2>&1`
 if test x"$hello" = x"hello"; then
diff --git a/include/Inventor/SbBasic.h b/include/Inventor/SbBasic.h
index d038bb1..a3c8072 100644
--- a/include/Inventor/SbBasic.h
+++ b/include/Inventor/SbBasic.h
@@ -24,6 +24,7 @@
  *
 \**************************************************************************/
 
+#include <Inventor/C/errors/debugerror.h>
 #include <Inventor/C/basic.h>
 
 /* ********************************************************************** */
diff --git a/src/fonts/freetype.cpp b/src/fonts/freetype.cpp
index 6f4bf27..0a7adc8 100644
--- a/src/fonts/freetype.cpp
+++ b/src/fonts/freetype.cpp
@@ -32,6 +32,8 @@
 
    20050613 mortene. */
 
+#include <cstdlib>
+#include <cmath>
 #include "fonts/freetype.h"
 
 #ifdef HAVE_CONFIG_H
