From 5d574ec8c33a84e3212d288b45bfb7879ec3a2cb Mon Sep 17 00:00:00 2001
From: Ikey Doherty <ikey@solus-project.com>
Date: Mon, 27 Jul 2015 20:02:53 +0100
Subject: [PATCH] Harden compiler flags by default and go full relro now

Signed-off-by: Ikey Doherty <ikey@solus-project.com>
---
 pisi/configfile.py | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/pisi/configfile.py b/pisi/configfile.py
index 922bdb5..4402d4e 100644
--- a/pisi/configfile.py
+++ b/pisi/configfile.py
@@ -85,9 +85,9 @@ class BuildDefaults:
     jobs = "-j5"
     generateDebug = False
     enableSandbox = False # Dropping sandbox support soon
-    cflags = "-mtune=generic -march=x86-64 -g2 -O2 -pipe -fPIC -fomit-frame-pointer -fstack-protector -D_FORTIFY_SOURCE=2"
-    cxxflags = "-mtune=generic -march=x86-64 -g2 -O2 -pipe -fPIC -fomit-frame-pointer -fstack-protector -D_FORTIFY_SOURCE=2"
-    ldflags = "-Wl,-O1 -Wl,-z,relro -Wl,--hash-style=gnu -Wl,--as-needed -Wl,--sort-common"
+    cflags = "-mtune=generic -march=x86-64 -g2 -O2 -pipe -fPIC -Wformat -Wformat-security -fomit-frame-pointer -fstack-protector --param ssp-buffer-size=4 -fexceptions -D_FORTIFY_SOURCE=2 -feliminate-unused-debug-types -Wno-error"
+    cxxflags = "-mtune=generic -march=x86-64 -g2 -O2 -pipe -fPIC -Wformat -Wformat-security -fomit-frame-pointer -fstack-protector --param ssp-buffer-size=4 -fexceptions -D_FORTIFY_SOURCE=2 -feliminate-unused-debug-types -Wno-error"
+    ldflags = "-Wl,--copy-dt-needed-entries -Wl,-O1 -Wl,-z,relro -Wl,-z,now"
     buildhelper = None
     compressionlevel = 1
     fallback = "https://www.solus-project.com/source"
-- 
2.4.6

