From c2d5bd69bbd5646e9308524f37e1b70509aa2665 Mon Sep 17 00:00:00 2001
From: Joshua Strobl <joshua@streambits.io>
Date: Sat, 11 Jul 2020 00:06:58 +0300
Subject: [PATCH 1/1] Retry up to 3 times, reduce timeout from 5 minutes to 2
 minutes.

---
 pisi/fetcher.py | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/pisi/fetcher.py b/pisi/fetcher.py
index e8c816e..8a89142 100644
--- a/pisi/fetcher.py
+++ b/pisi/fetcher.py
@@ -130,6 +130,7 @@ class Fetcher:
                            http_headers = self._get_http_headers(),
                            ftp_headers  = self._get_ftp_headers(),
                            proxies      = self._get_proxies(),
+                           retry = 3,
                            timeout      = timeout,
                            user_agent   = 'eopkg Fetcher/' + pisi.__version__)
         except urlgrabber.grabber.URLGrabError:
@@ -165,6 +166,8 @@ class Fetcher:
                            throttle     = self._get_bandwith_limit(),
                            reget        = self._test_range_support(),
                            copy_local   = 1,
+                           retry = 3, # retry 3 times
+                           timeout = 120, # Reduce from default of 5 minutes to 2 minutes
                            user_agent   = 'eopkg Fetcher/' + pisi.__version__)
         except urlgrabber.grabber.URLGrabError, e:
             raise FetchError(_('Could not fetch destination file "%s": %s') % (self.url.get_uri(), e))
-- 
2.27.0

