From 0387aa6f105f7b2279e8e5265b3222c092d22785 Mon Sep 17 00:00:00 2001
From: Ikey Doherty <ikey@solus-project.com>
Date: Sat, 21 Apr 2018 13:33:10 +0100
Subject: [PATCH] Gormless feckin eejit altogether

Signed-off-by: Ikey Doherty <ikey@solus-project.com>
---
 pisi/db/filesdb.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/pisi/db/filesdb.py b/pisi/db/filesdb.py
index 4efeef0..2079e59 100644
--- a/pisi/db/filesdb.py
+++ b/pisi/db/filesdb.py
@@ -67,7 +67,7 @@ class FilesDB(lazydb.LazyDB):
     def get_pkgconfig32_provider(self, pkgconfigName):
         """ get_pkgconfig32_provider will try known paths to find the provider
             of a given pkgconfig32 name """
-        fp = "usr/lib32/pkgconfig" + pkgconfigName + ".pc"
+        fp = "usr/lib32/pkgconfig/" + pkgconfigName + ".pc"
         if self.has_file(fp):
             return self.get_file(fp)
         return None
-- 
2.16.3

