diff --git a/pisi/archive.py b/pisi/archive.py
index 831369b..3c8c393 100644
--- a/pisi/archive.py
+++ b/pisi/archive.py
@@ -384,13 +384,13 @@ class ArchiveTar(ArchiveBase):
 
             try:
                 self.tar.extract(tarinfo)
-            except OSError, e:
+            except (IOError, OSError), e:
                 # Handle the case where an upper directory cannot
                 # be created because of a conflict with an existing
                 # regular file or symlink. In this case, remove
                 # the old file and retry extracting.
 
-                if e.errno != errno.EEXIST:
+                if e.errno != errno.EEXIST and e.errno != err.ENOTDIR:
                     raise
 
                 # For the path "a/b/c", upper_dirs will be ["a", "a/b"].
