From 58879f15a36a36dc4f35b02303452589a294f750 Mon Sep 17 00:00:00 2001
From: Justin Zobel <justin.zobel@gmail.com>
Date: Thu, 23 Apr 2015 08:11:46 +0930
Subject: [PATCH 1/4] branding update

---
 README | 10 +++++-----
 1 file changed, 5 insertions(+), 5 deletions(-)

diff --git a/README b/README
index 04bc0b7..6d48c74 100644
--- a/README
+++ b/README
@@ -1,4 +1,4 @@
-Evolve OS Package Manager
+Solus Operating System Package Manager
 -------------------------
 
 Fork of the PiSi Package Manager, originally from Pardus Linux,
@@ -9,11 +9,11 @@ NOTE: Due to talks with the "Pisilinux" project maintainers, who claim
 that PiSi is being actively "maintained" (developed doesn't seem the correct
 word here) this project should now be considered a FORK of the original
 Pardus PiSi package manager. Note this is not a new effort, and merely
-continues its life through SolusOS into Evolve OS.
+continues its life through SolusOS into Solus Operating System.
 
-In essence, it is within Evolve OS's interests to keep developing and
+In essence, it is within Solus Operating System's interests to keep developing and
 changing this package manager (and eventually a full rewrite) to advance
-practices and improve the Evolve OS project on the whole.
+practices and improve the Solus Operating System project on the whole.
 
 Other projects will continue to use the *original* PiSi package manager,
 as the aim of some of these projects is to preserve everything about their
@@ -23,7 +23,7 @@ interests in completely rewriting it and ripping pieces out.
 In time, the contents of this repository will change significantly, as
 I now have total "carte blanche" to rewrite the entire package manager
 in a more sensible fashion without fear of reprisal of projects not
-related to Evolve OS.
+related to Solus Operating System.
 
 The last release of this fork under the name "PiSi" will be 3.2, after
 which the changes below will be initiated and a new name will be taken,
-- 
2.3.2

