From 74fc945ec9b8990def5b025663ef93f72116e32c Mon Sep 17 00:00:00 2001
From: Ikey Doherty <ikey@solus-project.com>
Date: Sat, 5 Mar 2016 03:09:25 +0000
Subject: [PATCH] Set -march=core2 and -mtune=ivybridge

Signed-off-by: Ikey Doherty <ikey@solus-project.com>
---
 pisi/configfile.py | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/pisi/configfile.py b/pisi/configfile.py
index 00dae9b..db7e7cc 100644
--- a/pisi/configfile.py
+++ b/pisi/configfile.py
@@ -85,8 +85,8 @@ class BuildDefaults:
     jobs = "-j5"
     generateDebug = False
     enableSandbox = False # Dropping sandbox support soon
-    cflags = "-mtune=generic -march=x86-64 -g2 -O2 -pipe -fPIC -Wformat -Wformat-security -fomit-frame-pointer -fstack-protector-strong --param ssp-buffer-size=4 -fexceptions -D_FORTIFY_SOURCE=2 -feliminate-unused-debug-types -Wno-error"
-    cxxflags = "-mtune=generic -march=x86-64 -g2 -O2 -pipe -fPIC -Wformat -Wformat-security -fomit-frame-pointer -fstack-protector-strong --param ssp-buffer-size=4 -fexceptions -D_FORTIFY_SOURCE=2 -feliminate-unused-debug-types -Wno-error"
+    cflags = "-mtune=ivybridge -march=core2 -g2 -O2 -pipe -fPIC -Wformat -Wformat-security -fomit-frame-pointer -fstack-protector-strong --param ssp-buffer-size=4 -fexceptions -D_FORTIFY_SOURCE=2 -feliminate-unused-debug-types -Wno-error"
+    cxxflags = "-mtune=ivybridge -march=core2 -g2 -O2 -pipe -fPIC -Wformat -Wformat-security -fomit-frame-pointer -fstack-protector-strong --param ssp-buffer-size=4 -fexceptions -D_FORTIFY_SOURCE=2 -feliminate-unused-debug-types -Wno-error"
     ldflags = "-Wl,--copy-dt-needed-entries -Wl,-O1 -Wl,-z,relro -Wl,-z,now"
     buildhelper = None
     compressionlevel = 1
-- 
2.7.1

