From f8106d57511e56b08b1e87fb84c3d2e2ffb2704c Mon Sep 17 00:00:00 2001
From: Ikey Doherty <ikey@solus-project.com>
Date: Sat, 7 Jan 2017 22:44:57 +0000
Subject: [PATCH] config: Enable ccache by default, help people out

Signed-off-by: Ikey Doherty <ikey@solus-project.com>
---
 pisi/configfile.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/pisi/configfile.py b/pisi/configfile.py
index d17c31c..12d5339 100644
--- a/pisi/configfile.py
+++ b/pisi/configfile.py
@@ -88,7 +88,7 @@ class BuildDefaults:
     cflags = "-mtune=generic -march=x86-64 -ftree-vectorize -g2 -O2 -pipe -fPIC -Wformat -Wformat-security -fno-omit-frame-pointer -fstack-protector-strong --param ssp-buffer-size=4 -fexceptions -D_FORTIFY_SOURCE=2 -feliminate-unused-debug-types -Wno-error -Wp,-D_REENTRANT"
     cxxflags = "-mtune=generic -march=x86-64 -ftree-vectorize -g2 -O2 -pipe -fPIC -Wformat -Wformat-security -fno-omit-frame-pointer -fstack-protector-strong --param ssp-buffer-size=4 -fexceptions -D_FORTIFY_SOURCE=2 -feliminate-unused-debug-types -Wno-error -Wp,-D_REENTRANT"
     ldflags = "-Wl,--copy-dt-needed-entries -Wl,-O1 -Wl,-z,relro -Wl,-z,now"
-    buildhelper = None
+    buildhelper = "ccache"
     compressionlevel = 1
     fallback = "https://www.solus-project.com/source"
     ignored_build_types = ""
-- 
2.11.0

