From f1199edd3579821a5afdc2bea37a6cc644367fc6 Mon Sep 17 00:00:00 2001
From: Joey Riches <josephriches@gmail.com>
Date: Sun, 13 Feb 2022 11:46:35 +0000
Subject: [PATCH 1/1] fetcher: remove debug line

---
 pisi/fetcher.py | 1 -
 1 file changed, 1 deletion(-)

diff --git a/pisi/fetcher.py b/pisi/fetcher.py
index 2219a1f..2ada7f4 100644
--- a/pisi/fetcher.py
+++ b/pisi/fetcher.py
@@ -233,7 +233,6 @@ class Fetcher:
     def _get_retry_attempts(self):
         retry_attempts = ctx.config.options.retry_attempts or ctx.config.values.general.retry_attempts
         if retry_attempts and retry_attempts != "0":
-            ctx.ui.warning(_("retry attempts is %s") % retry_attempts)
             return int(retry_attempts)
         else:
             return 5
-- 
2.35.1

