From 6d5f5c763feb7b5828a7c08fa970e82e88bcbb13 Mon Sep 17 00:00:00 2001
From: Ikey Doherty <ikey@solus-project.com>
Date: Thu, 7 Apr 2016 07:34:25 +0100
Subject: [PATCH 1/2] Step 1, ensure our install archive has a fixed timestamp.

Signed-off-by: Ikey Doherty <ikey@solus-project.com>
---
 pisi/package.py | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/pisi/package.py b/pisi/package.py
index e49d781..60d1b66 100644
--- a/pisi/package.py
+++ b/pisi/package.py
@@ -28,6 +28,8 @@ import pisi.files
 import pisi.util as util
 import fetcher
 
+# 1980, Jan 1, for zip
+TSTAMP_META = 315532800
 
 class Error(pisi.Error):
     pass
@@ -151,6 +153,7 @@ class Package:
             self.install_archive.close()
             arcpath = self.install_archive_path
             arcname = os.path.basename(arcpath)
+            os.utime(arcpath, (TSTAMP_META, TSTAMP_META))
             self.add_to_package(arcpath, arcname)
 
         self.impl.close()
-- 
2.7.3

