From 28bbda54505a98830d85a4ca47bc167475878217 Mon Sep 17 00:00:00 2001
From: Ikey Doherty <ikey@solus-project.com>
Date: Fri, 26 Jan 2018 10:01:07 +0000
Subject: [PATCH] configfile: Drop unnecessary LD flag that breaks stuff

Signed-off-by: Ikey Doherty <ikey@solus-project.com>
---
 pisi/configfile.py | 3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

diff --git a/pisi/configfile.py b/pisi/configfile.py
index 9f12a00..52d1e74 100644
--- a/pisi/configfile.py
+++ b/pisi/configfile.py
@@ -87,7 +87,7 @@ class BuildDefaults:
     enableSandbox = False # Dropping sandbox support soon
     cflags = "-mtune=generic -march=x86-64 -g2 -O2 -pipe -fPIC -Wformat -Wformat-security -D_FORTIFY_SOURCE=2 -fstack-protector-strong --param ssp-buffer-size=32 -fasynchronous-unwind-tables -ftree-vectorize -feliminate-unused-debug-types -Wall -Wno-error -Wp,-D_REENTRANT"
     cxxflags = "-mtune=generic -march=x86-64 -g2 -O2 -pipe -fPIC -D_FORTIFY_SOURCE=2 -fstack-protector-strong --param ssp-buffer-size=32 -fasynchronous-unwind-tables -ftree-vectorize -feliminate-unused-debug-types -Wall -Wno-error -Wp,-D_REENTRANT"
-    ldflags = "-Wl,--copy-dt-needed-entries -Wl,-O1 -Wl,-z,relro -Wl,-z,now -Wl,-z,max-page-size=0x1000 -Wl,-Bsymbolic-functions"
+    ldflags = "-Wl,-O1 -Wl,-z,relro -Wl,-z,now -Wl,-z,max-page-size=0x1000 -Wl,-Bsymbolic-functions"
     buildhelper = "ccache"
     compressionlevel = 1
     fallback = "https://www.solus-project.com/source"
@@ -314,4 +314,3 @@ class ConfigurationFile(object):
                 fp.write(sect.getvalue())
 
         fp.close()
-        
-- 
2.16.1

