From a10ea43ee775499c6ca09809b1e5001103ece55f Mon Sep 17 00:00:00 2001
From: Joey Riches <josephriches@gmail.com>
Date: Mon, 14 Feb 2022 17:29:27 +0000
Subject: [PATCH 1/2] Fix download speed calculation for partial files

---
 pisi/fetcher.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/pisi/fetcher.py b/pisi/fetcher.py
index 2ada7f4..8755847 100644
--- a/pisi/fetcher.py
+++ b/pisi/fetcher.py
@@ -72,7 +72,7 @@ class FetchHandler:
 
         if int(self.now()) != int(self.last_updated) and self.size > 0:
             try:
-                self.rate, self.symbol = util.human_readable_rate(self.size / (self.now() - self.s_time))
+                self.rate, self.symbol = util.human_readable_rate((self.size - self.exist_size) / (self.now() - self.s_time))
             except ZeroDivisionError:
                 return
             if self.total_size:
-- 
2.35.1

