From 8886235f2c9bd1a76099e0f8f3a4d8cac6c2e38c Mon Sep 17 00:00:00 2001
From: Ikey Doherty <ikey@solus-project.com>
Date: Fri, 17 Jul 2015 00:49:30 +0100
Subject: [PATCH] Enforce LD_AS_NEEDED by default

---
 pisi/actionsapi/variables.py | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/pisi/actionsapi/variables.py b/pisi/actionsapi/variables.py
index c56d1ca..f11bb23 100644
--- a/pisi/actionsapi/variables.py
+++ b/pisi/actionsapi/variables.py
@@ -34,6 +34,8 @@ def exportFlags():
     os.environ['USER_LDFLAGS'] = values.build.ldflags
     os.environ['JOBS'] = values.build.jobs
 
+    os.environ['LD_AS_NEEDED'] = "1"
+
     # http://liste.pardus.org.tr/gelistirici/2009-January/016442.html
     os.environ['CC'] = "%s-gcc" % values.build.host
     os.environ['CXX'] = "%s-g++" % values.build.host
-- 
2.4.6

