From 55f317a39923fe5733522ddf9bb37450ee4bd6b5 Mon Sep 17 00:00:00 2001
From: Ikey Doherty <ikey@solus-project.com>
Date: Sun, 26 Nov 2017 17:29:32 +0000
Subject: [PATCH 2/4] get: Lock to multiprocessing CPU count, not higher.

Signed-off-by: Ikey Doherty <ikey@solus-project.com>
---
 pisi/actionsapi/get.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/pisi/actionsapi/get.py b/pisi/actionsapi/get.py
index a82a5c3..63929f4 100644
--- a/pisi/actionsapi/get.py
+++ b/pisi/actionsapi/get.py
@@ -126,7 +126,7 @@ def makeJOBS():
     if env.jobs == "auto":
         procs = 2
         try:
-            procs = multiprocessing.cpu_count() + 1
+            procs = multiprocessing.cpu_count()
         except Exception, e:
             ctx.ui.warning("Unable to retrieve CPU count: %s" % e)
         return "-j%s" % procs
-- 
2.15.0

