From c85b6c77da0d6cc8a2ac3f760bd309dae8958959 Mon Sep 17 00:00:00 2001
From: Ikey Doherty <ikey@solus-project.com>
Date: Sat, 9 Apr 2016 23:45:04 +0100
Subject: [PATCH 2/2] Only set utime when requested

Signed-off-by: Ikey Doherty <ikey@solus-project.com>
---
 pisi/package.py | 8 +++++---
 1 file changed, 5 insertions(+), 3 deletions(-)

diff --git a/pisi/package.py b/pisi/package.py
index 60d1b66..c4a2490 100644
--- a/pisi/package.py
+++ b/pisi/package.py
@@ -28,8 +28,6 @@ import pisi.files
 import pisi.util as util
 import fetcher
 
-# 1980, Jan 1, for zip
-TSTAMP_META = 315532800
 
 class Error(pisi.Error):
     pass
@@ -42,6 +40,8 @@ class Package:
     formats = ("1.0", "1.1", "1.2")
     default_format = "1.2"
 
+    history_timestamp = None
+
     @staticmethod
     def archive_name_and_format(package_format):
         if package_format == "1.2":
@@ -153,7 +153,9 @@ class Package:
             self.install_archive.close()
             arcpath = self.install_archive_path
             arcname = os.path.basename(arcpath)
-            os.utime(arcpath, (TSTAMP_META, TSTAMP_META))
+            if self.history_timestamp is not None:
+                tstamp = self.history_timestamp
+                os.utime(arcpath, (tstamp, tstamp))
             self.add_to_package(arcpath, arcname)
 
         self.impl.close()
-- 
2.7.3

