From c6d572514acde6359cd56345da5e95f1cc02d259 Mon Sep 17 00:00:00 2001
From: Ikey Doherty <ikey.doherty@gmail.com>
Date: Mon, 8 Sep 2014 21:22:33 +0100
Subject: [PATCH 2/3] build: Fix warn() vs warning() issue

---
 pisi/operations/build.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/pisi/operations/build.py b/pisi/operations/build.py
index cf284ab..d24e2b9 100644
--- a/pisi/operations/build.py
+++ b/pisi/operations/build.py
@@ -1026,7 +1026,7 @@ class Builder:
                     pcPath = util.join_path(self.pkg_install_dir(), path)
                     code,out,err = pisi.util.run_batch("%s --modversion %s" % (pkgconfigExec, pcPath))
                     if code != 0:
-                        ctx.ui.warn(_("Unable to obtain pkgconfig module version for %s") % pcName)
+                        ctx.ui.warning(_("Unable to obtain pkgconfig module version for %s") % pcName)
                     else:
                         pkgconfig.version = out.strip().rstrip()
                     metadata.package.providesPkgConfig.append(pkgconfig)
-- 
1.8.4

