From ce9d59f1517e004976d52cc721f1b617a7bfb24d Mon Sep 17 00:00:00 2001
From: Justin Zobel <justin.zobel@gmail.com>
Date: Thu, 23 Apr 2015 08:13:07 +0930
Subject: [PATCH 2/4] branding update

---
 PKG-INFO | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/PKG-INFO b/PKG-INFO
index 0b490a6..4ccbd4f 100644
--- a/PKG-INFO
+++ b/PKG-INFO
@@ -2,9 +2,9 @@ Metadata-Version: 1.0
 Name: pisi
 Version: 3.2
 Summary: PiSi (Packages Installed Successfully as Intended)
-Home-page: http://www.evolve-os.com
+Home-page: https://www.solus-project.com
 Author: Ikey Doherty
-Author-email: ikey@solusos.com
+Author-email: ikey@solus-project.com
 License: GNU GPL2
-Description: PiSi is the package management system of Evolve OS, originally written for Pardus Linux by the Pardus Developers.
+Description: PiSi is the package management system of Solus Operating System, originally written for Pardus Linux by the Pardus Developers.
 Platform: UNKNOWN
-- 
2.3.2

