From 96ebe4dad7dd85393a71d7e118c11ecb8ed11740 Mon Sep 17 00:00:00 2001
From: Peter O'Connor <sunnyflunk@gmail.com>
Date: Wed, 9 Jan 2019 12:05:23 +1100
Subject: [PATCH 2/3] Sort common symbols by alignment in builds

---
 pisi/configfile.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/pisi/configfile.py b/pisi/configfile.py
index 9f12a00..c3bf120 100644
--- a/pisi/configfile.py
+++ b/pisi/configfile.py
@@ -87,7 +87,7 @@ class BuildDefaults:
     enableSandbox = False # Dropping sandbox support soon
     cflags = "-mtune=generic -march=x86-64 -g2 -O2 -pipe -fPIC -Wformat -Wformat-security -D_FORTIFY_SOURCE=2 -fstack-protector-strong --param ssp-buffer-size=32 -fasynchronous-unwind-tables -ftree-vectorize -feliminate-unused-debug-types -Wall -Wno-error -Wp,-D_REENTRANT"
     cxxflags = "-mtune=generic -march=x86-64 -g2 -O2 -pipe -fPIC -D_FORTIFY_SOURCE=2 -fstack-protector-strong --param ssp-buffer-size=32 -fasynchronous-unwind-tables -ftree-vectorize -feliminate-unused-debug-types -Wall -Wno-error -Wp,-D_REENTRANT"
-    ldflags = "-Wl,--copy-dt-needed-entries -Wl,-O1 -Wl,-z,relro -Wl,-z,now -Wl,-z,max-page-size=0x1000 -Wl,-Bsymbolic-functions"
+    ldflags = "-Wl,--copy-dt-needed-entries -Wl,-O1 -Wl,-z,relro -Wl,-z,now -Wl,-z,max-page-size=0x1000 -Wl,-Bsymbolic-functions -Wl,--sort-common"
     buildhelper = "ccache"
     compressionlevel = 1
     fallback = "https://www.solus-project.com/source"
-- 
2.20.1

