From b8c726b50ed4adb5e03f74e2aa7a3b9ee78b8d1e Mon Sep 17 00:00:00 2001
From: Ikey Doherty <ikey@solus-project.com>
Date: Sat, 15 Apr 2017 02:29:25 +0100
Subject: [PATCH 3/3] configfile: Enable -malign-data=abi

Signed-off-by: Ikey Doherty <ikey@solus-project.com>
---
 pisi/configfile.py | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/pisi/configfile.py b/pisi/configfile.py
index 52cc039..66ae8ee 100644
--- a/pisi/configfile.py
+++ b/pisi/configfile.py
@@ -85,8 +85,8 @@ class BuildDefaults:
     jobs = "-j5"
     generateDebug = False
     enableSandbox = False # Dropping sandbox support soon
-    cflags = "-mtune=generic -march=x86-64 -O2 -g -feliminate-unused-debug-types -pipe -fPIC -Wall -Wp,-D_FORTIFY_SOURCE=2 -fexceptions -fstack-protector --param=ssp-buffer-size=32 -Wformat -Wformat-security -fasynchronous-unwind-tables -ftree-loop-vectorize -ftree-vectorize -fopt-info-vec -fno-omit-frame-pointer -Wno-error -Wp,-D_REENTRANT"
-    cxxflags = "-mtune=generic -march=x86-64 -O2 -g -feliminate-unused-debug-types -pipe -fPIC -Wall -Wp,-D_FORTIFY_SOURCE=2 -fexceptions -fstack-protector --param=ssp-buffer-size=32 -fasynchronous-unwind-tables -ftree-loop-vectorize -ftree-vectorize -fopt-info-vec -fno-omit-frame-pointer -Wno-error -Wp,-D_REENTRANT"
+    cflags = "-mtune=generic -march=x86-64 -malign-data=abi -O2 -g -feliminate-unused-debug-types -pipe -fPIC -Wall -Wp,-D_FORTIFY_SOURCE=2 -fexceptions -fstack-protector --param=ssp-buffer-size=32 -Wformat -Wformat-security -fasynchronous-unwind-tables -ftree-loop-vectorize -ftree-vectorize -fopt-info-vec -fno-omit-frame-pointer -Wno-error -Wp,-D_REENTRANT"
+    cxxflags = "-mtune=generic -march=x86-64 -malign-data=abi -O2 -g -feliminate-unused-debug-types -pipe -fPIC -Wall -Wp,-D_FORTIFY_SOURCE=2 -fexceptions -fstack-protector --param=ssp-buffer-size=32 -fasynchronous-unwind-tables -ftree-loop-vectorize -ftree-vectorize -fopt-info-vec -fno-omit-frame-pointer -Wno-error -Wp,-D_REENTRANT"
     ldflags = "-Wl,--copy-dt-needed-entries -Wl,-z,relro -Wl,-z,now -Wl,-z,max-page-size=0x1000"
     buildhelper = "ccache"
     compressionlevel = 1
-- 
2.12.2

