From b69c93ae3e08faddcb3bfc48ab98cfa0b09a25b2 Mon Sep 17 00:00:00 2001
From: Peter O'Connor <sunnyflunk@gmail.com>
Date: Wed, 9 Jan 2019 12:08:33 +1100
Subject: [PATCH 3/3] Change fallback to getsol.us in case it is ever used

---
 pisi/configfile.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/pisi/configfile.py b/pisi/configfile.py
index c3bf120..7a4b563 100644
--- a/pisi/configfile.py
+++ b/pisi/configfile.py
@@ -90,7 +90,7 @@ class BuildDefaults:
     ldflags = "-Wl,--copy-dt-needed-entries -Wl,-O1 -Wl,-z,relro -Wl,-z,now -Wl,-z,max-page-size=0x1000 -Wl,-Bsymbolic-functions -Wl,--sort-common"
     buildhelper = "ccache"
     compressionlevel = 1
-    fallback = "https://www.solus-project.com/source"
+    fallback = "https://getsol.us/sources"
     ignored_build_types = ""
 
 class DirectoriesDefaults:
-- 
2.20.1

