From d0fe2ee983f5bcbc0fb3f47ee26caebe1e3295c6 Mon Sep 17 00:00:00 2001
From: Ikey Doherty <ikey@solus-project.com>
Date: Sun, 26 Nov 2017 17:30:00 +0000
Subject: [PATCH 3/4] configfile: Default build jobs to automatic detection now

Signed-off-by: Ikey Doherty <ikey@solus-project.com>
---
 pisi/configfile.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/pisi/configfile.py b/pisi/configfile.py
index c4f34eb..9f12a00 100644
--- a/pisi/configfile.py
+++ b/pisi/configfile.py
@@ -82,7 +82,7 @@ class BuildDefaults:
     """Default values for [build] section"""
     build_host = "solus"
     host = "x86_64-solus-linux"
-    jobs = "-j5"
+    jobs = "auto"
     generateDebug = False
     enableSandbox = False # Dropping sandbox support soon
     cflags = "-mtune=generic -march=x86-64 -g2 -O2 -pipe -fPIC -Wformat -Wformat-security -D_FORTIFY_SOURCE=2 -fstack-protector-strong --param ssp-buffer-size=32 -fasynchronous-unwind-tables -ftree-vectorize -feliminate-unused-debug-types -Wall -Wno-error -Wp,-D_REENTRANT"
-- 
2.15.0

