From 594680e3d3e1cf41a0b4ef71ac93e2edaf20999e Mon Sep 17 00:00:00 2001
From: Ikey Doherty <ikey@evolve-os.com>
Date: Thu, 15 Jan 2015 07:29:38 +0000
Subject: [PATCH 3/3] build: Ensure installdb is used, not packagedb

---
 pisi/operations/build.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/pisi/operations/build.py b/pisi/operations/build.py
index d24e2b9..d657578 100644
--- a/pisi/operations/build.py
+++ b/pisi/operations/build.py
@@ -1056,7 +1056,7 @@ class Builder:
                         if not found and dep not in metadata.package.packageDependencies:
                             newDep = pisi.dependency.Dependency()
                             newDep.package = dep
-                            pkg = self.packagedb.get_package(dep)
+                            pkg = self.installdb.get_package(dep)
                             newDep.releaseFrom = pkg.release
                             metadata.package.packageDependencies.append(newDep)
                             ctx.ui.debug("%s depends on %s (>= release %s)" % (metadata.package.name, dep, pkg.release))
-- 
1.8.4

