From d0e6037270b2e8713bbb9e787c5b5a94744cf3c3 Mon Sep 17 00:00:00 2001
From: Ikey Doherty <ikey@evolve-os.com>
Date: Mon, 27 Apr 2015 03:45:54 +0100
Subject: [PATCH 4/4] configfile: Use correct short distro prefix

---
 pisi/configfile.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/pisi/configfile.py b/pisi/configfile.py
index 1678819..06aec03 100644
--- a/pisi/configfile.py
+++ b/pisi/configfile.py
@@ -65,7 +65,7 @@ class GeneralDefaults:
     """Default values for [general] section"""
     destinationdirectory = "/"
     autoclean = False
-    distribution = "Solus Operating System"
+    distribution = "Solus"
     distribution_release = "1"
     distribution_id = "1"
     architecture = "x86_64"
-- 
2.3.2

