From 7e6a1f89f0c49b02125264d89184921a8b9bd09a Mon Sep 17 00:00:00 2001
From: Ikey Doherty <ikey@solus-project.com>
Date: Sun, 26 Nov 2017 17:30:24 +0000
Subject: [PATCH 4/4] Raise undetected jobCount to 4

Signed-off-by: Ikey Doherty <ikey@solus-project.com>
---
 pisi/actionsapi/get.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/pisi/actionsapi/get.py b/pisi/actionsapi/get.py
index 63929f4..b426314 100644
--- a/pisi/actionsapi/get.py
+++ b/pisi/actionsapi/get.py
@@ -124,7 +124,7 @@ def LDFLAGS():
 def makeJOBS():
     # Note: "auto" only works when /sys is mounted.
     if env.jobs == "auto":
-        procs = 2
+        procs = 4
         try:
             procs = multiprocessing.cpu_count()
         except Exception, e:
-- 
2.15.0

