From 0def4851e6f5f3ccc821081709d267521a7388ef Mon Sep 17 00:00:00 2001
From: Ikey Doherty <ikey@solus-project.com>
Date: Thu, 27 Apr 2017 07:47:39 +0100
Subject: [PATCH 4/7] Enforce a standard flag set going forward

Signed-off-by: Ikey Doherty <ikey@solus-project.com>
---
 pisi/configfile.py | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/pisi/configfile.py b/pisi/configfile.py
index 66ae8ee..c1e63e4 100644
--- a/pisi/configfile.py
+++ b/pisi/configfile.py
@@ -85,9 +85,9 @@ class BuildDefaults:
     jobs = "-j5"
     generateDebug = False
     enableSandbox = False # Dropping sandbox support soon
-    cflags = "-mtune=generic -march=x86-64 -malign-data=abi -O2 -g -feliminate-unused-debug-types -pipe -fPIC -Wall -Wp,-D_FORTIFY_SOURCE=2 -fexceptions -fstack-protector --param=ssp-buffer-size=32 -Wformat -Wformat-security -fasynchronous-unwind-tables -ftree-loop-vectorize -ftree-vectorize -fopt-info-vec -fno-omit-frame-pointer -Wno-error -Wp,-D_REENTRANT"
-    cxxflags = "-mtune=generic -march=x86-64 -malign-data=abi -O2 -g -feliminate-unused-debug-types -pipe -fPIC -Wall -Wp,-D_FORTIFY_SOURCE=2 -fexceptions -fstack-protector --param=ssp-buffer-size=32 -fasynchronous-unwind-tables -ftree-loop-vectorize -ftree-vectorize -fopt-info-vec -fno-omit-frame-pointer -Wno-error -Wp,-D_REENTRANT"
-    ldflags = "-Wl,--copy-dt-needed-entries -Wl,-z,relro -Wl,-z,now -Wl,-z,max-page-size=0x1000"
+    cflags = "-mtune=generic -march=x86-64 -g2 -O2 -pipe -fPIC -Wformat -Wformat-security -malign-data=abi -fno-omit-frame-pointer -fexceptions -fstack-protector --param ssp-buffer-size=32 -fasynchronous-unwind-tables -ftree-loop-vectorize -ftree-vectorize -feliminate-unused-debug-types -Wall -Wno-error -Wp,-D_REENTRANT"
+    cxxflags = "-mtune=generic -march=x86-64 -g2 -O2 -pipe -fPIC -malign-data=abi -fno-omit-frame-pointer -fexceptions -fstack-protector --param ssp-buffer-size=32 -fasynchronous-unwind-tables -ftree-loop-vectorize -ftree-vectorize -feliminate-unused-debug-types -Wall -Wno-error -Wp,-D_REENTRANT"
+    ldflags = "-Wl,--copy-dt-needed-entries -Wl,-O1 -Wl,-z,relro -Wl,-z,now -Wl,-z,max-page-size=0x1000"
     buildhelper = "ccache"
     compressionlevel = 1
     fallback = "https://www.solus-project.com/source"
-- 
2.13.0

