From b14938a7b183ac6c489688d1e7ade9bafaac0227 Mon Sep 17 00:00:00 2001
From: Ikey Doherty <ikey@solus-project.com>
Date: Fri, 8 Dec 2017 16:43:52 +0000
Subject: [PATCH 6/6] cli: Only grab the length for a valid list

Signed-off-by: Ikey Doherty <ikey@solus-project.com>
---
 pisi/cli/listinstalled.py | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/pisi/cli/listinstalled.py b/pisi/cli/listinstalled.py
index 3a13bf9..d9825f7 100644
--- a/pisi/cli/listinstalled.py
+++ b/pisi/cli/listinstalled.py
@@ -102,7 +102,8 @@ Usage: list-installed
         orphans = list_orphans()
         keys = orphans.keys()
         keys.sort()
-        maxlen = max([len(x) for x in keys])
+        if keys and len(keys) > 0:
+            maxlen = max([len(x) for x in keys])
 
         for orphan in keys:
             owner = orphans[orphan]
-- 
2.15.1

