From 6b7f85c3c9ab31f87424013f477fd3584652669e Mon Sep 17 00:00:00 2001
From: Ikey Doherty <ikey@evolve-os.com>
Date: Sun, 22 Feb 2015 10:28:24 +0000
Subject: [PATCH] build: Workaround dodgy magic for EXIF tags

---
 pisi/operations/build.py | 39 ++++++++++++++++++++++-----------------
 1 file changed, 22 insertions(+), 17 deletions(-)

diff --git a/pisi/operations/build.py b/pisi/operations/build.py
index d657578..6530f59 100644
--- a/pisi/operations/build.py
+++ b/pisi/operations/build.py
@@ -1043,23 +1043,28 @@ class Builder:
                 if not os.path.exists(fullpath):
                     # Dodgy symlinks
                     continue
-                filemagic = magic.from_file(fullpath)
-                if "SB executable" in filemagic or "SB shared object" in filemagic:
-                    ctx.ui.debug("Checking %s for binary dependencies" % fullpath)
-                    bindeps = self.get_binary_deps(fullpath)
-                    for dep in bindeps:
-                        found = False
-                        for depen in metadata.package.packageDependencies:
-                            if depen.package == dep:
-                                found = True
-                                break
-                        if not found and dep not in metadata.package.packageDependencies:
-                            newDep = pisi.dependency.Dependency()
-                            newDep.package = dep
-                            pkg = self.installdb.get_package(dep)
-                            newDep.releaseFrom = pkg.release
-                            metadata.package.packageDependencies.append(newDep)
-                            ctx.ui.debug("%s depends on %s (>= release %s)" % (metadata.package.name, dep, pkg.release))
+                filemagic = None
+                try:
+                    filemagic = magic.from_file(fullpath)
+                except:
+                    pass
+                if filemagic:
+                    if "SB executable" in filemagic or "SB shared object" in filemagic:
+                        ctx.ui.debug("Checking %s for binary dependencies" % fullpath)
+                        bindeps = self.get_binary_deps(fullpath)
+                        for dep in bindeps:
+                            found = False
+                            for depen in metadata.package.packageDependencies:
+                                if depen.package == dep:
+                                    found = True
+                                    break
+                            if not found and dep not in metadata.package.packageDependencies:
+                                newDep = pisi.dependency.Dependency()
+                                newDep.package = dep
+                                pkg = self.installdb.get_package(dep)
+                                newDep.releaseFrom = pkg.release
+                                metadata.package.packageDependencies.append(newDep)
+                                ctx.ui.debug("%s depends on %s (>= release %s)" % (metadata.package.name, dep, pkg.release))
 
         # Seems insane iterating again for requirements, but we must ensure we grab
         # all pkgconfig files first! (also this is just a small list of known pc files :)
-- 
2.3.2

