From 7248b6742b9ceace62e4976f1a9d10bafe0fc52e Mon Sep 17 00:00:00 2001
From: Joshua Strobl <joshua@stroblindustries.com>
Date: Sun, 30 Dec 2018 17:15:32 +0200
Subject: [PATCH 2/2] Add potential ELAN device for K505ZA / X505ZA

---
 drivers/input/mouse/elan_i2c_core.c | 12 ++++++++++++
 1 file changed, 12 insertions(+)

diff --git a/drivers/input/mouse/elan_i2c_core.c b/drivers/input/mouse/elan_i2c_core.c
index a94b6494e71a..22ac056b1a66 100644
--- a/drivers/input/mouse/elan_i2c_core.c
+++ b/drivers/input/mouse/elan_i2c_core.c
@@ -1353,6 +1353,18 @@ static const struct acpi_device_id elan_acpi_id[] = {
 	{ "ELAN0621", 0 },
 	{ "ELAN0622", 0 },
 	{ "ELAN1000", 0 },
+	{ "ELAN1200", 0 },
+	{ "ELAN1201", 0 },
+	{ "ELAN1202", 0 },
+	{ "ELAN1203", 0 },
+	{ "ELAN12A6", 0 },
+	{ "ELAN1300", 0 },
+	{ "ELAN1301", 0 },
+	{ "ELAN2200", 0 },
+	{ "ELAN2201", 0 },
+	{ "ELAN2202", 0 },
+	{ "ELAN2300", 0 },
+	{ "ELAN2301", 0 },
 	{ }
 };
 MODULE_DEVICE_TABLE(acpi, elan_acpi_id);
-- 
2.20.1

