diff --git a/src/config/bluetooth.lua.d/30-bluez-monitor.lua b/src/config/bluetooth.lua.d/30-bluez-monitor.lua
index b40026c20625b18f71f85b05a1cb9962bd2e13fe..a870aa5d934e25b2ffabdec94dda5269481073de 100644
--- a/src/config/bluetooth.lua.d/30-bluez-monitor.lua
+++ b/src/config/bluetooth.lua.d/30-bluez-monitor.lua
@@ -3,7 +3,7 @@ bluez_monitor.properties = {}
 bluez_monitor.rules = {}
 
 function bluez_monitor.enable()
-  if not bluez_monitor.enabled then
+  if bluez_monitor.enabled == false then
     return
   end
 
diff --git a/src/config/main.lua.d/20-default-access.lua b/src/config/main.lua.d/20-default-access.lua
index a6ffb2ee61809a98ccb8b6eb90ba2983f1d7c8b4..0a7eb95575d8bb384a50aa6790b79e6e08ef2dea 100644
--- a/src/config/main.lua.d/20-default-access.lua
+++ b/src/config/main.lua.d/20-default-access.lua
@@ -3,7 +3,7 @@ default_access.properties = {}
 default_access.rules = {}
 
 function default_access.enable()
-  if not default_access.enabled then
+  if default_access.enabled == false then
     return
   end
 
diff --git a/src/config/main.lua.d/30-alsa-monitor.lua b/src/config/main.lua.d/30-alsa-monitor.lua
index da0b2c7026ede8ce366a7a7f8b66c06e74764dea..8e45e434b2f8174229949b4c776c8932df7b3ea4 100644
--- a/src/config/main.lua.d/30-alsa-monitor.lua
+++ b/src/config/main.lua.d/30-alsa-monitor.lua
@@ -3,7 +3,7 @@ alsa_monitor.properties = {}
 alsa_monitor.rules = {}
 
 function alsa_monitor.enable()
-  if not alsa_monitor.enabled then
+  if alsa_monitor.enabled == false then
     return
   end
 
diff --git a/src/config/main.lua.d/30-libcamera-monitor.lua b/src/config/main.lua.d/30-libcamera-monitor.lua
index 4a8257ffe2cbfe385b8d5c1c1ed6556497431968..cd820a8345ea92f7dba88543316070385bea1da5 100644
--- a/src/config/main.lua.d/30-libcamera-monitor.lua
+++ b/src/config/main.lua.d/30-libcamera-monitor.lua
@@ -3,7 +3,7 @@ libcamera_monitor.properties = {}
 libcamera_monitor.rules = {}
 
 function libcamera_monitor.enable()
-  if not libcamera_monitor.enabled then
+  if libcamera_monitor.enabled == false then
     return
   end
 
diff --git a/src/config/main.lua.d/30-v4l2-monitor.lua b/src/config/main.lua.d/30-v4l2-monitor.lua
index 7cfd4bcddd528658da6cbbd0f58c6a3d3a481ebf..3fbdc9e75628dde599956367ccf930dfe26f60c0 100644
--- a/src/config/main.lua.d/30-v4l2-monitor.lua
+++ b/src/config/main.lua.d/30-v4l2-monitor.lua
@@ -3,7 +3,7 @@ v4l2_monitor.properties = {}
 v4l2_monitor.rules = {}
 
 function v4l2_monitor.enable()
-  if not v4l2_monitor.enabled then
+  if v4l2_monitor.enabled == false then
     return
   end
 
diff --git a/src/config/main.lua.d/40-device-defaults.lua b/src/config/main.lua.d/40-device-defaults.lua
index 55aafe8538d8e75c369f89f3a2911a0e0e7a3c1e..2204c4ac250f816389b4dff1bca7888e0c28e7c9 100644
--- a/src/config/main.lua.d/40-device-defaults.lua
+++ b/src/config/main.lua.d/40-device-defaults.lua
@@ -38,7 +38,7 @@ device_defaults.persistent_profiles = {
 }
 
 function device_defaults.enable()
-  if not device_defaults.enabled then
+  if device_defaults.enabled == false then
     return
   end
 
diff --git a/src/config/main.lua.d/40-stream-defaults.lua b/src/config/main.lua.d/40-stream-defaults.lua
index 2975f4f6dafcc92ac72a54e54b4f99185b1444ec..307d83df505c8559093f430f08f9af5a7e57896e 100644
--- a/src/config/main.lua.d/40-stream-defaults.lua
+++ b/src/config/main.lua.d/40-stream-defaults.lua
@@ -25,7 +25,7 @@ stream_defaults.rules = {
 }
 
 function stream_defaults.enable()
-  if not stream_defaults.enabled then
+  if stream_defaults.enabled == false then
     return
   end
 
diff --git a/src/config/policy.lua.d/10-default-policy.lua b/src/config/policy.lua.d/10-default-policy.lua
index 7b00e94ee7787f040356e590db96238a58599397..edfdeb1c9fc3c3e5da1083bb81ca8c3ccaf6c027 100644
--- a/src/config/policy.lua.d/10-default-policy.lua
+++ b/src/config/policy.lua.d/10-default-policy.lua
@@ -38,7 +38,7 @@ bluetooth_policy.policy = {
 }
 
 function default_policy.enable()
-  if not default_policy.enabled then
+  if default_policy.enabled == false then
     return
   end
 
