From 709c1bd247a9f76823117cd78fe4796dca25e10e Mon Sep 17 00:00:00 2001
From: Thomas Stromberg <tstromberg@google.com>
Date: Thu, 20 Jun 2019 15:47:41 +0800
Subject: [PATCH 1/2] Update Makefile to use golangci-lint v1.17.1

---
 Makefile | 14 ++------------
 1 file changed, 2 insertions(+), 12 deletions(-)

diff --git a/Makefile b/Makefile
index 28c4b7334..6e4f95c94 100755
--- a/Makefile
+++ b/Makefile
@@ -222,19 +222,9 @@ fmt:
 vet:
 	@go vet $(SOURCE_PACKAGES)
 
-# Once v1.16.1+ is released, replace with
-# curl -sfL https://install.goreleaser.com/github.com/golangci/golangci-lint.sh \
-#  | bash -s -- -b out/linters v1.16.1
+
 out/linters/golangci-lint:
-	mkdir -p out/linters
-	cd out/linters
-	test -f go.mod || go mod init linters
-	go get -u github.com/golangci/golangci-lint/cmd/golangci-lint@692dacb773b703162c091c2d8c59f9cd2d6801db 2>&1 \
-	  | grep -v "go: finding"
-	test -x $(GOPATH)/bin/golangci-lint \
-	  || curl -sfL https://install.goreleaser.com/github.com/golangci/golangci-lint.sh \
-	  | sh -s -- -b $(GOPATH)/bin v1.16.0
-	cp -f $(GOPATH)/bin/golangci-lint out/linters/golangci-lint
+	curl -sfL https://install.goreleaser.com/github.com/golangci/golangci-lint.sh | bash -s -- -b out/linters v1.17.1
 
 .PHONY: lint
 lint: pkg/minikube/assets/assets.go out/linters/golangci-lint
-- 
2.22.0

