From 3dbb0109dfa6c6c6b85c0e620121d47d69ce13e9 Mon Sep 17 00:00:00 2001
From: Thomas Stromberg <tstromberg@google.com>
Date: Thu, 20 Jun 2019 16:15:23 +0800
Subject: [PATCH 2/2] Simplify the Make code, use a GOLINT_VERSION var

---
 Makefile | 5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

diff --git a/Makefile b/Makefile
index 6e4f95c94..8b6cfb3bc 100755
--- a/Makefile
+++ b/Makefile
@@ -41,6 +41,7 @@ MINIKUBE_UPLOAD_LOCATION := gs://${MINIKUBE_BUCKET}
 KERNEL_VERSION ?= 4.16.14
 
 GO_VERSION ?= $(shell go version | cut -d' ' -f3 | sed -e 's/go//')
+GOLINT_VERSION ?= v1.17.1
 export GO111MODULE := on
 
 GOOS ?= $(shell go env GOOS)
@@ -222,9 +223,9 @@ fmt:
 vet:
 	@go vet $(SOURCE_PACKAGES)
 
-
 out/linters/golangci-lint:
-	curl -sfL https://install.goreleaser.com/github.com/golangci/golangci-lint.sh | bash -s -- -b out/linters v1.17.1
+	mkdir -p out/linters
+	curl -sfL https://raw.githubusercontent.com/golangci/golangci-lint/master/install.sh | sh -s -- -b out/linters $(GOLINT_VERSION)
 
 .PHONY: lint
 lint: pkg/minikube/assets/assets.go out/linters/golangci-lint
-- 
2.22.0

