diff -up source/shared_lib/sources/platform/common/platform_common.cpp.orig source/shared_lib/sources/platform/common/platform_common.cpp
--- source/shared_lib/sources/platform/common/platform_common.cpp.orig 2016-06-24 07:56:09.496569134 -0300
+++ source/shared_lib/sources/platform/common/platform_common.cpp  2016-06-24 07:56:11.962569290 -0300
@@ -127,2 +127,1 @@
-	system_time_point system_now = std::chrono::system_clock::now();
-	return std::chrono::system_clock::to_time_t(system_now);
+	return std::chrono::system_clock::to_time_t(std::chrono::system_clock::now());
