From 5260702eed3175de5b48aa671eb6176f7e233af2 Mon Sep 17 00:00:00 2001
From: Joshua Strobl <joshua@streambits.io>
Date: Fri, 13 Mar 2020 18:04:38 +0200
Subject: [PATCH 1/1] Don't force docs to be enabled

---
 meson.build | 1 -
 1 file changed, 1 deletion(-)

diff --git a/meson.build b/meson.build
index 61c3f13..9e8a046 100644
--- a/meson.build
+++ b/meson.build
@@ -35,7 +35,6 @@ configure_file(input: 'libportal.pc.in',
                install_dir: join_paths(get_option('libdir'), 'pkgconfig'))
 
 subdir('libportal')
-subdir('doc')
 if get_option('build-portal-test')
   subdir('portal-test')
 endif
-- 
2.25.1

