From ae94cf124c7a0d564f9a736efd9882495fc82ea3 Mon Sep 17 00:00:00 2001
From: Milan Nikolic <gen2brain@gmail.com>
Date: Wed, 22 Dec 2021 14:05:53 +0100
Subject: [PATCH] Use integers for move(), issue #191

---
 src/m64py/frontend/mainwindow.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/m64py/frontend/mainwindow.py b/src/m64py/frontend/mainwindow.py
index 903ebbc..8ef8aa4 100644
--- a/src/m64py/frontend/mainwindow.py
+++ b/src/m64py/frontend/mainwindow.py
@@ -177,7 +177,7 @@ def center_widget(self):
         width, height = size.width(), size.height()
         dwidth, dheight = desktop.width(), desktop.height()
         cw, ch = (dwidth/2)-(width/2), (dheight/2)-(height/2)
-        self.move(cw, ch)
+        self.move(int(cw), int(ch))
 
     def connect_signals(self):
         """Connects signals."""
