From 516e89102e50ed9ec4634d020aa6a3f7dc12baf8 Mon Sep 17 00:00:00 2001
From: Milan Nikolic <gen2brain@gmail.com>
Date: Wed, 22 Dec 2021 14:15:33 +0100
Subject: [PATCH] Use integer, issue #191

---
 src/m64py/frontend/romreader.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/m64py/frontend/romreader.py b/src/m64py/frontend/romreader.py
index ff861d0..84de37a 100644
--- a/src/m64py/frontend/romreader.py
+++ b/src/m64py/frontend/romreader.py
@@ -123,7 +123,7 @@ def read_files(self):
                 log.warn(str(err))
                 continue
             percent = float(filenum) / float(num_files) * 100
-            self.parent.progressBar.valueChanged.emit(percent)
+            self.parent.progressBar.valueChanged.emit(int(percent))
         self.exit()
 
     def stop(self):
