From 7c21a9c07f9db708926e8c8b01f2f0aa6d00822f Mon Sep 17 00:00:00 2001
From: James Lee <jamesl33info@gmail.com>
Date: Sun, 31 Mar 2019 21:18:22 +0100
Subject: [PATCH] Don't enable tlp service on virtualised platforms

---
 tlp.service.in | 1 +
 1 file changed, 1 insertion(+)

diff --git a/tlp.service.in b/tlp.service.in
index 9ac3470..4aec4b8 100644
--- a/tlp.service.in
+++ b/tlp.service.in
@@ -5,6 +5,7 @@
 
 [Unit]
 Description=TLP system startup/shutdown
+ConditionVirtualization=no
 After=multi-user.target bluetooth.service NetworkManager.service
 Before=shutdown.target
 Documentation=https://linrunner.de/tlp
-- 
2.21.0

