From c3693b61877b47371fc24f7ea6d3f603550978dc Mon Sep 17 00:00:00 2001
From: Ikey Doherty <ikey@solus-project.com>
Date: Sat, 15 Oct 2016 22:34:32 +0100
Subject: [PATCH] Don't enable tlp service on virtualised platforms

Signed-off-by: Ikey Doherty <ikey@solus-project.com>
---
 tlp.service.in | 1 +
 1 file changed, 1 insertion(+)

diff --git a/tlp.service.in b/tlp.service.in
index 7dc4b2c..14012c1 100644
--- a/tlp.service.in
+++ b/tlp.service.in
@@ -6,6 +6,7 @@
 [Unit]
 Description=TLP system startup/shutdown
 Wants=bluetooth.service NetworkManager.service
+ConditionVirtualization=no
 After=multi-user.target bluetooth.service NetworkManager.service
 Before=shutdown.target
 Documentation=http://linrunner.de/tlp
-- 
2.10.1

