From 25102f4b1e8db6831ea50b3aaf8b04cb7dd8aa65 Mon Sep 17 00:00:00 2001
From: palto42 <12280188+palto42@users.noreply.github.com>
Date: Sat, 3 Dec 2022 11:48:59 +0100
Subject: [PATCH 2/2] enable_default_settings_in_conf

---
 tlp.conf | 10 +++++-----
 1 file changed, 5 insertions(+), 5 deletions(-)

diff --git a/tlp.conf b/tlp.conf
index e7f2c8a..92e3e1c 100644
--- a/tlp.conf
+++ b/tlp.conf
@@ -312,8 +312,8 @@
 # Wi-Fi power saving mode: on=enable, off=disable.
 # Default: off (AC), on (BAT)
 
-#WIFI_PWR_ON_AC=off
-#WIFI_PWR_ON_BAT=on
+WIFI_PWR_ON_AC=off
+WIFI_PWR_ON_BAT=off
 
 # Disable Wake-on-LAN: Y/N.
 # Default: Y
@@ -326,8 +326,8 @@
 # systems without PulseAudio may require 10.
 # Default: 1
 
-#SOUND_POWER_SAVE_ON_AC=1
-#SOUND_POWER_SAVE_ON_BAT=1
+SOUND_POWER_SAVE_ON_AC=0
+SOUND_POWER_SAVE_ON_BAT=0
 
 # Disable controller too (HDA only): Y/N.
 # Note: effective only when SOUND_POWER_SAVE_ON_AC/BAT is activated.
@@ -378,7 +378,7 @@
 # Set to 0 to disable, 1 to enable USB autosuspend feature.
 # Default: 1
 
-#USB_AUTOSUSPEND=1
+USB_AUTOSUSPEND=0
 
 # Exclude listed devices from USB autosuspend (separate with spaces).
 # Use lsusb to get the ids.
-- 
2.35.5

