From 36ca2f40bc658ed898f25c5c0934898dee0c47ca Mon Sep 17 00:00:00 2001
From: Silke Hofstra <silke@slxh.eu>
Date: Sat, 17 Oct 2020 12:53:33 +0200
Subject: [PATCH 1/4] Ensure /boot is mounted

---
 data/fwupd.service.in | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/data/fwupd.service.in b/data/fwupd.service.in
index 26379e44d..1110fbf2a 100644
--- a/data/fwupd.service.in
+++ b/data/fwupd.service.in
@@ -1,7 +1,8 @@
 [Unit]
 Description=Firmware update daemon
 Documentation=https://fwupd.org/
-After=dbus.service
+After=dbus.service clr-boot-manager-mount-boot.service
+Wants=clr-boot-manager-mount-boot.service
 Before=display-manager.service
 ConditionVirtualization=!container
 
-- 
2.47.0

