From f768876a05f373a964e531cf9ba21f27347ec777 Mon Sep 17 00:00:00 2001
From: Silke Hofstra <silke@slxh.eu>
Date: Wed, 26 Apr 2023 12:23:48 +0200
Subject: [PATCH 2/4] Disable tests that fail in Solbuild chroot

---
 plugins/thunderbolt/meson.build | 1 -
 1 file changed, 1 deletion(-)

diff --git a/plugins/thunderbolt/meson.build b/plugins/thunderbolt/meson.build
index 696caef9d..54fb710ee 100644
--- a/plugins/thunderbolt/meson.build
+++ b/plugins/thunderbolt/meson.build
@@ -51,6 +51,5 @@ if get_option('tests') and run_sanitize_unsafe_tests and umockdev.found() and gi
   else
     env.prepend('LD_PRELOAD', 'libumockdev-preload.so.0')
   endif
-  test('thunderbolt-self-test', e, env: env, timeout: 120)
 endif
 endif
-- 
2.41.0

