From 331b8e86475a4b1d2327dba767a8c8d7ecc7597e Mon Sep 17 00:00:00 2001
From: Silke Hofstra <silke@slxh.eu>
Date: Wed, 26 Apr 2023 12:23:48 +0200
Subject: [PATCH 2/5] Disable tests that fail in Solbuild chroot

---
 plugins/thunderbolt/meson.build | 1 -
 1 file changed, 1 deletion(-)

diff --git a/plugins/thunderbolt/meson.build b/plugins/thunderbolt/meson.build
index e7b527110..c98ef72c7 100644
--- a/plugins/thunderbolt/meson.build
+++ b/plugins/thunderbolt/meson.build
@@ -52,6 +52,5 @@ if get_option('tests') and run_sanitize_unsafe_tests and umockdev.found() and gi
   else
     env.prepend('LD_PRELOAD', 'libumockdev-preload.so.0')
   endif
-  test('thunderbolt-self-test', e, env: env, timeout: 120)
 endif
 endif
-- 
2.43.0

