From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Reilly Brogan <reilly@reillybrogan.com>
Date: Fri, 19 Jul 2024 13:57:56 -0500
Subject: [PATCH] libtool no rpath

---
 ltmain.sh | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/ltmain.sh b/ltmain.sh
index 70990740b6c..905e66df16d 100644
--- a/ltmain.sh
+++ b/ltmain.sh
@@ -7103,6 +7103,7 @@ EOF
 	  rpath="$finalize_rpath"
 	  test "$mode" != relink && rpath="$compile_rpath$rpath"
 	  for libdir in $rpath; do
+	    case "$libdir" in /usr/lib|/usr/lib32|/usr/lib64|/usr/lib/../lib|/usr/lib/../lib32|/usr/lib/../lib64|/usr/lib/../lib64/lp64d) continue;; esac
 	    if test -n "$hardcode_libdir_flag_spec"; then
 	      if test -n "$hardcode_libdir_separator"; then
 		if test -z "$hardcode_libdirs"; then
@@ -7798,6 +7799,7 @@ EOF
       rpath=
       hardcode_libdirs=
       for libdir in $compile_rpath $finalize_rpath; do
+	  case "$libdir" in /usr/lib|/usr/lib32|/usr/lib64|/usr/lib/../lib|/usr/lib/../lib32|/usr/lib/../lib64|/usr/lib/../lib64/lp64d) continue;; esac
 	if test -n "$hardcode_libdir_flag_spec"; then
 	  if test -n "$hardcode_libdir_separator"; then
 	    if test -z "$hardcode_libdirs"; then
@@ -7849,6 +7851,7 @@ EOF
       rpath=
       hardcode_libdirs=
       for libdir in $finalize_rpath; do
+	  case "$libdir" in /usr/lib|/usr/lib32|/usr/lib64|/usr/lib/../lib|/usr/lib/../lib32|/usr/lib/../lib64|/usr/lib/../lib64/lp64d) continue;; esac
 	if test -n "$hardcode_libdir_flag_spec"; then
 	  if test -n "$hardcode_libdir_separator"; then
 	    if test -z "$hardcode_libdirs"; then
