diff --git a/build/Linux+BSD/org.musescore.MuseScore.appdata.xml.in b/build/Linux+BSD/org.musescore.MuseScore.appdata.xml.in
index 2ff55728b0..bdb3c4ace8 100644
--- a/build/Linux+BSD/org.musescore.MuseScore.appdata.xml.in
+++ b/build/Linux+BSD/org.musescore.MuseScore.appdata.xml.in
@@ -7,13 +7,12 @@
 
         After editing, run either of these commands to check for errors in this file:
             $  appstreamcli validate [filename]
-            $  appstream-util validate-relax [filename]
+            $  appstream-util validate [filename]
     -->
     <id>org.musescore.MuseScore@MUSESCORE_INSTALL_SUFFIX@</id>
     <metadata_license>CC0-1.0</metadata_license><!-- License of this file. See <project_license> for MuseScore license. -->
     <name>MuseScore</name>
     <summary>Create, play and print beautiful sheet music</summary>
-    <icon type="stock">mscore@MUSESCORE_INSTALL_SUFFIX@</icon>
     <description>
         <p>MuseScore is cross-platform, multi-lingual, open source music notation software. It features an easy to use WYSIWYG editor with audio score playback for results that look and sound beautiful. It supports unlimited staves with up to four voices each, dynamics, articulations, lyrics, chords, lead sheet notation, import/export of MIDI and MusicXML, export to PDF and WAV, plus online score sharing.</p>
         <p>MuseScore can upload scores directly to the score sharing site musescore.com. Program support is provided on musescore.org.</p>
@@ -115,6 +114,15 @@
     </kudos>
     <!-- Only <releases> below this point please! -->
     <releases>
+        <release date="2024-01-24" version="4.2.1" />
+        <release date="2023-12-18" version="4.2.0" />
+        <release date="2023-07-26" version="4.1.1">
+            <url>https://github.com/musescore/MuseScore/releases/tag/v4.1.1</url>
+        </release>
+        <release date="2023-07-12" version="4.1.0">
+            <url>https://github.com/musescore/MuseScore/releases/tag/v4.1.0</url>
+        </release>
+        <release date="2023-03-13" version="4.0.2" />
         <release date="2023-01-13" version="4.0.1">
             <url>https://musescore.org/en/4.0.1</url>
         </release>
@@ -136,9 +137,6 @@
         <release date="2020-10-09" version="3.5.1">
             <url>https://musescore.org/en/3.5.1</url>
         </release>
-        <release date="2020-10-09" version="3.5.1">
-            <url>https://musescore.org/en/3.5.1</url>
-        </release>
         <release date="2020-08-07" version="3.5">
             <url>https://musescore.org/en/3.5</url>
         </release>
