From 153aaf88bf9bf5c6c4b118bd871f8cf1eafdcab5 Mon Sep 17 00:00:00 2001
From: Quentin Rameau <quinq@fifth.space>
Date: Mon, 4 Feb 2019 00:29:26 +0100
Subject: [PATCH 1/6] Close when the embedding window is destroyed

---
 LICENSE | 2 +-
 dmenu.c | 7 ++++++-
 2 files changed, 7 insertions(+), 2 deletions(-)

diff --git a/LICENSE b/LICENSE
index 6ed8ad3..9762166 100644
--- a/LICENSE
+++ b/LICENSE
@@ -9,7 +9,7 @@ MIT/X Consortium License
 © 2009 Evan Gates <evan.gates@gmail.com>
 © 2010-2012 Connor Lane Smith <cls@lubutu.com>
 © 2014-2019 Hiltjo Posthuma <hiltjo@codemadness.org>
-© 2015-2018 Quentin Rameau <quinq@fifth.space>
+© 2015-2019 Quentin Rameau <quinq@fifth.space>
 
 Permission is hereby granted, free of charge, to any person obtaining a
 copy of this software and associated documentation files (the "Software"),
diff --git a/dmenu.c b/dmenu.c
index 6b8f51b..f803149 100644
--- a/dmenu.c
+++ b/dmenu.c
@@ -556,6 +556,11 @@ run(void)
 		if (XFilterEvent(&ev, None))
 			continue;
 		switch(ev.type) {
+		case DestroyNotify:
+			if (ev.xdestroywindow.window != win)
+				break;
+			cleanup();
+			exit(1);
 		case Expose:
 			if (ev.xexpose.count == 0)
 				drw_map(drw, win, 0, 0, mw, mh);
@@ -667,7 +672,7 @@ setup(void)
 	XMapRaised(dpy, win);
 	XSetInputFocus(dpy, win, RevertToParent, CurrentTime);
 	if (embed) {
-		XSelectInput(dpy, parentwin, FocusChangeMask);
+		XSelectInput(dpy, parentwin, FocusChangeMask | SubstructureNotifyMask);
 		if (XQueryTree(dpy, parentwin, &dw, &w, &dws, &du) && dws) {
 			for (i = 0; i < du && dws[i] != win; ++i)
 				XSelectInput(dpy, dws[i], FocusChangeMask);
-- 
2.27.0

