From a9b1de384ae1ad30805e893af5cd6ea9b87c89da Mon Sep 17 00:00:00 2001
From: Hiltjo Posthuma <hiltjo@codemadness.org>
Date: Tue, 12 Feb 2019 22:58:35 +0100
Subject: [PATCH 4/6] improve xopenim error message

die() already prints a newline.
---
 dmenu.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/dmenu.c b/dmenu.c
index ae56f4f..3bfd74d 100644
--- a/dmenu.c
+++ b/dmenu.c
@@ -671,7 +671,7 @@ setup(void)
 		if ((xim = XOpenIM(dpy, NULL, NULL, NULL)) == NULL) {
 			XSetLocaleModifiers("@im=");
 			if ((xim = XOpenIM(dpy, NULL, NULL, NULL)) == NULL)
-				die("XOpenIM failed. Could not open input device.\n");
+				die("XOpenIM failed: could not open input device");
 		}
 	}
 
-- 
2.27.0

