From a0296f126a4187d9aeb7c8708e281219f29e09c7 Mon Sep 17 00:00:00 2001
From: Joshua Strobl <joshua@streambits.io>
Date: Thu, 16 Sep 2021 01:15:21 +0300
Subject: [PATCH 1/1] Switch to using explicit dependency func instead of using
 find_library

---
 meson.build | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/meson.build b/meson.build
index 12d3e91d..4e30488b 100644
--- a/meson.build
+++ b/meson.build
@@ -65,7 +65,7 @@ edj_files = []
 
 if get_option('nls') == true
   subdir('po')
-   terminology_dependencies += cc.find_library('intl', required: false)
+  terminology_dependencies += dependency('intl', required: false)
 endif
 
 foreach efl_dep: efl_deps
-- 
2.32.0

