--- a/site/source/conf.py
+++ b/site/source/conf.py
@@ -288,8 +288,10 @@
 # One entry per manual page. List of tuples
 # (source start file, name, description, authors, manual section).
 man_pages = [
-    ('index', 'emscripten', u'Emscripten Documentation',
-     [u'Hamish Willee'], 1)
+    ('docs/tools_reference/emcc', 'emcc', u'Emscripten Compiler Documentation',
+     [u'The Emscripten Authors'], 1),
+    ('docs/tools_reference/em++', 'em++', u'Emscripten Compiler Documentation',
+     [u'The Emscripten Authors'], 1),
 ]
 
 # If true, show URL addresses after external links.
--- a/site/source/docs/tools_reference/index.rst
+++ b/site/source/docs/tools_reference/index.rst
@@ -22,5 +22,6 @@
 
    emsdk
    emcc
+   em++
    emcmdprompt
 
--- a/site/source/docs/tools_reference/emcc.rst
+++ b/site/source/docs/tools_reference/emcc.rst
@@ -24,7 +24,7 @@
 
 Most `clang options <http://linux.die.net/man/1/clang>`_ will work, as will `gcc options <https://gcc.gnu.org/onlinedocs/gcc/Option-Summary.html#Option-Summary>`_, for example: ::
 
-  # Display this information
+  Display this information
   emcc --help
 
   Display compiler version information
