--- a/emrun.py
+++ b/emrun.py
@@ -587,7 +587,7 @@ class HTTPHandler(SimpleHTTPRequestHandl
     # A browser has navigated to this page - check which PID got spawned for
     # the browser
     global navigation_has_occurred
-    if not navigation_has_occurred and current_browser_processes is None:
+    if not navigation_has_occurred and current_browser_processes is None and not emrun_options.no_browser:
       detect_browser_processes()
 
     navigation_has_occurred = True
