--- a/tools/system_libs.py
+++ b/tools/system_libs.py
@@ -953,6 +953,9 @@ class libunwind(NoExceptLibrary, MTLibra
   cflags = ['-Oz', '-D_LIBUNWIND_DISABLE_VISIBILITY_ANNOTATIONS']
   src_dir = ['system', 'lib', 'libunwind', 'src']
   src_files = ['Unwind-wasm.cpp']
+  includes = [
+    ['system', 'lib', 'libunwind', 'include'],
+  ]
 
   def __init__(self, **kwargs):
     super(libunwind, self).__init__(**kwargs)
