From 676979f0743f0f35b69512fa8abfa97573638547 Mon Sep 17 00:00:00 2001
From: Troy Harvey <harveydevel@gmail.com>
Date: Wed, 17 Aug 2022 10:23:05 +1000
Subject: [PATCH] Fix arg

---
 src/shares.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/shares.c b/src/shares.c
index bac0d1d..0efed3c 100644
--- a/src/shares.c
+++ b/src/shares.c
@@ -623,7 +623,7 @@ add_share (ShareInfo *info, GError **error)
 		return FALSE;
 
 	argv[0] = "add";
-	argv[1] = "-l";
+	argv[1] = "--long";
 	argv[2] = info->share_name;
 	argv[3] = info->path;
 	argv[4] = info->comment;
-- 
2.35.4

