From 68cb921bf05fbb02b51e98308ecd85b9b1090c22 Mon Sep 17 00:00:00 2001
From: "F. von Gellhorn" <flinux@vongellhorn.ch>
Date: Sun, 12 Apr 2020 21:45:03 +0200
Subject: [PATCH 1/1] Fix build with gcc

---
 buttonlist.h | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/buttonlist.h b/buttonlist.h
index 7bb37b3..980213f 100644
--- a/buttonlist.h
+++ b/buttonlist.h
@@ -16,7 +16,7 @@ public:
 
 	void Add_Button(Uint16 x, Uint16 y, Uint16 width, Uint16 height, 
 						void (*callback)(void)) {
-		struct button *belem;
+		button *belem;
 		
 		for ( belem=&button_list; belem->next; belem=belem->next );
 		belem->next = new button;
@@ -30,7 +30,7 @@ public:
 	}
 
 	void Activate_Button(Uint16 x, Uint16 y) {
-		struct button *belem;
+		button *belem;
 
 		for ( belem=button_list.next; belem; belem=belem->next ) {
 			if ( (x >= belem->x1) && (x <= belem->x2) &&
@@ -42,7 +42,7 @@ public:
 	}
 
 	void Delete_Buttons(void) {
-		struct button *belem, *btemp;
+		button *belem, *btemp;
 
 		for ( belem=button_list.next; belem; ) {
 			btemp = belem;
-- 
2.26.0

