From b5b436bdf23e0bd86922e7eefe3ec2400e968bac Mon Sep 17 00:00:00 2001
From: posophe <p.drouand@gmail.com>
Date: Sun, 1 Mar 2015 20:26:19 +0100
Subject: [PATCH] Fix deprecated code

---
 buttonlist.h | 20 ++++++++++----------
 1 file changed, 10 insertions(+), 10 deletions(-)

diff --git a/buttonlist.h b/buttonlist.h
index 7bb37b3..f7ab5ba 100644
--- a/buttonlist.h
+++ b/buttonlist.h
@@ -7,6 +7,13 @@
 class ButtonList {
 
 public:
+typedef struct _button {
+		/* Sensitive area */
+		Uint16 x1, y1;
+		Uint16 x2, y2;
+		void (*callback)(void);
+		struct _button *next;
+	} button;
 	ButtonList() {
 		button_list.next = NULL;
 	}
@@ -16,7 +23,7 @@ class ButtonList {
 
 	void Add_Button(Uint16 x, Uint16 y, Uint16 width, Uint16 height, 
 						void (*callback)(void)) {
-		struct button *belem;
+		button *belem;
 		
 		for ( belem=&button_list; belem->next; belem=belem->next );
 		belem->next = new button;
@@ -30,7 +37,7 @@ class ButtonList {
 	}
 
 	void Activate_Button(Uint16 x, Uint16 y) {
-		struct button *belem;
+		button *belem;
 
 		for ( belem=button_list.next; belem; belem=belem->next ) {
 			if ( (x >= belem->x1) && (x <= belem->x2) &&
@@ -42,7 +49,7 @@ class ButtonList {
 	}
 
 	void Delete_Buttons(void) {
-		struct button *belem, *btemp;
+		button *belem, *btemp;
 
 		for ( belem=button_list.next; belem; ) {
 			btemp = belem;
@@ -53,12 +60,5 @@ class ButtonList {
 	}
 	
 private:
-	typedef struct button {
-		/* Sensitive area */
-		Uint16 x1, y1;
-		Uint16 x2, y2;
-		void (*callback)(void);
-		struct button *next;
-	} button;
 	button button_list;
 };

