--- Maelstrom-3.0.6/screenlib/SDL_FrameBuf.cpp.ook	2006-02-14 01:50:13.000000000 -0500
+++ Maelstrom-3.0.6/screenlib/SDL_FrameBuf.cpp	2006-02-14 01:50:58.000000000 -0500
@@ -848,8 +848,8 @@
 			for ( i=0; i<dirtymaplen; ++i ) {
 				if ( dirtymap[i] != NULL ) {
 					dirtymap[i] = (SDL_Rect *)(
-					((int)dirtymap[i]-(int)updatelist) +
-								(int)newlist
+					((long)dirtymap[i]-(long)updatelist) +
+								(long)newlist
 					);
 				}
 			}
