diff -up Maelstrom-3.0.6/Makefile.am.BAD Maelstrom-3.0.6/Makefile.am
--- Maelstrom-3.0.6/Makefile.am.BAD	2009-12-06 17:37:05.042119272 -0500
+++ Maelstrom-3.0.6/Makefile.am	2009-12-06 17:37:39.040120370 -0500
@@ -65,7 +65,7 @@ DIST_SUBDIRS = $(SUBDIRS) Images Docs
 
 # Special install rule for the game
 install:
-	make install_gamedata target=@GAME_INSTALLDIR@
+	make install_gamedata target=$(DESTDIR)@GAME_INSTALLDIR@
 
 install_gamedata:
 	sh mkinstalldirs $(target)/
diff -up Maelstrom-3.0.6/Makefile.in.BAD Maelstrom-3.0.6/Makefile.in
--- Maelstrom-3.0.6/Makefile.in.BAD	2009-12-06 17:37:58.555119622 -0500
+++ Maelstrom-3.0.6/Makefile.in	2009-12-06 17:38:04.208119720 -0500
@@ -431,7 +431,7 @@ maintainer-clean-generic clean mostlycle
 
 # Special install rule for the game
 install:
-	make install_gamedata target=@GAME_INSTALLDIR@
+	make install_gamedata target=$(DESTDIR)@GAME_INSTALLDIR@
 
 install_gamedata:
 	sh mkinstalldirs $(target)/
