From 44c9efc0f9bb35d3e159a19cdd985c4cfb110099 Mon Sep 17 00:00:00 2001
From: Ikey Doherty <ikey@solus-project.com>
Date: Thu, 24 Nov 2016 18:51:38 +0000
Subject: [PATCH] Fix build failures

Signed-off-by: Ikey Doherty <ikey@solus-project.com>
---
 Maelstrom-netd.c | 5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

diff --git a/Maelstrom-netd.c b/Maelstrom-netd.c
index fde5c05..b681b4b 100644
--- a/Maelstrom-netd.c
+++ b/Maelstrom-netd.c
@@ -2,6 +2,7 @@
 /* Here we go... */
 
 #include <stdio.h>
+#include <stdlib.h>
 #include <string.h>
 #include <signal.h>
 #include <errno.h>
@@ -20,7 +21,7 @@
          the same.  Otherwise we crash.
 */
 
-#include "protocol.h"
+#include "netlogic/protocol.h"
 
 #define MAX_CONNECTIONS		64
 #define UNCONNECTED		0
@@ -201,7 +202,7 @@ main(int argc, char *argv[])
 	memset(&serv_addr, 0, sizeof(serv_addr));
 	serv_addr.sin_family      = AF_INET;
 	serv_addr.sin_addr.s_addr = htonl(INADDR_ANY);
-	serv_addr.sin_port        = htons(NETPLAY_PORT-1);
+	serv_addr.sin_port        = htons(NETPLAY_PORT -1);
 	if (bind(netfd, (struct sockaddr *)&serv_addr, sizeof(serv_addr)) < 0) {
 		perror("Can't bind local address");
 		exit(3);
-- 
2.10.2

