From 56d5188ac2a53153253e24b09f032904bfae91f8 Mon Sep 17 00:00:00 2001
From: Peter O'Connor <peter@solus-project.com>
Date: Fri, 16 Mar 2018 12:40:53 +1100
Subject: [PATCH] Use locale.h with glibc-2.26

Signed-off-by: Peter O'Connor <peter@solus-project.com>
---
 source/i18n/digitlst.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/source/i18n/digitlst.cpp b/source/i18n/digitlst.cpp
index 4d7c3f4..2292c01 100644
--- a/source/i18n/digitlst.cpp
+++ b/source/i18n/digitlst.cpp
@@ -64,7 +64,7 @@
 # if U_PLATFORM_USES_ONLY_WIN32_API || U_PLATFORM == U_PF_CYGWIN
 #   include <locale.h>
 # else
-#   include <xlocale.h>
+#   include <locale.h>
 # endif
 #endif
 
-- 
2.16.2

