diff --git a/opt/brother/Printers/HLL2390DW/cupswrapper/brother-HLL2390DW-cups-en.ppd b/opt/brother/Printers/HLL2390DW/cupswrapper/brother-HLL2390DW-cups-en.ppd
index 1ad3a3e..1e0c7d1 100644
--- a/opt/brother/Printers/HLL2390DW/cupswrapper/brother-HLL2390DW-cups-en.ppd
+++ b/opt/brother/Printers/HLL2390DW/cupswrapper/brother-HLL2390DW-cups-en.ppd
@@ -25,9 +25,9 @@
 *cupsFilter:    "application/vnd.cups-postscript 0 brother_lpdwrapper_HLL2390DW"
 *cupsFilter:    "application/vnd.cups-pdf 0 brother_lpdwrapper_HLL2390DW"
 *PSVersion: "(3010.106) 3"
-*ModelName: "HLL2390DW"
-*NickName: "Brother HLL2390DW for CUPS "
-*ShortNickName: "Brother HLL2390DW"
+*ModelName: "HL-L2390DW"
+*NickName: "Brother HL-L2390DW for CUPS "
+*ShortNickName: "Brother HL-L2390DW"
 
 *%==== Basic Device Capabilities =============
 *LanguageLevel: "3"
