From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Reilly Brogan <reilly@reillybrogan.com>
Date: Mon, 25 Nov 2024 14:21:39 -0600
Subject: [PATCH] Improve compat with ppp 2.5.x

Signed-off-by: Reilly Brogan <reilly@reillybrogan.com>
---
 src/nm-fortisslvpn-service.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/nm-fortisslvpn-service.c b/src/nm-fortisslvpn-service.c
index a8483c2..0f37366 100644
--- a/src/nm-fortisslvpn-service.c
+++ b/src/nm-fortisslvpn-service.c
@@ -225,7 +225,7 @@ run_openfortivpn (NMFortisslvpnPlugin *plugin, NMSettingVpn *s_vpn, GError **err
 	g_ptr_array_add (argv, (gpointer) g_strdup ("-c"));
 	g_ptr_array_add (argv, (gpointer) g_strdup (priv->config_file));
 
-	g_ptr_array_add (argv, (gpointer) g_strdup ("--no-routes"));
+	g_ptr_array_add (argv, (gpointer) g_strdup ("--pppd-accept-remote"));
 	g_ptr_array_add (argv, (gpointer) g_strdup ("--no-dns"));
 	ip4_config = nm_connection_get_setting_ip4_config (priv->connection);
 	if (!nm_setting_ip_config_get_ignore_auto_dns (ip4_config)) {
