From 09cb118d02f11dd62ea8be0740be83c9fbd33b8a Mon Sep 17 00:00:00 2001
From: Nazar Stasiv <nazar@autistici.org>
Date: Sat, 14 Jan 2023 08:46:31 +0200
Subject: [PATCH] compile-failure

---
 lib/duden/IFileSystem.h | 1 +
 1 file changed, 1 insertion(+)

diff --git a/lib/duden/IFileSystem.h b/lib/duden/IFileSystem.h
index 2399476..4e1415a 100644
--- a/lib/duden/IFileSystem.h
+++ b/lib/duden/IFileSystem.h
@@ -5,6 +5,7 @@
 #include <optional>
 #include <set>
 #include <map>
+#include <memory>
 
 namespace duden {
 
-- 
2.39.0

