diff -Naur ./data/meson.build ./data/meson.build
--- ./data/meson.build	2023-01-22 15:37:05.000000000 -0500
+++ ./data/meson.build	2023-07-16 11:05:00.715448500 -0400
@@ -8,7 +8,7 @@
         conf.set('demo_agent', '')
     endif
 
-    conf_dir = join_paths(sysconfdir, 'geoclue')
+    conf_dir = join_paths(datadir, 'defaults', 'geoclue')
     configure_file(output: 'geoclue.conf',
                    input: 'geoclue.conf.in',
                    configuration: conf,
diff -Naur ./demo/meson.build ./demo/meson.build
--- ./demo/meson.build	2023-01-22 15:37:05.000000000 -0500
+++ ./demo/meson.build	2023-07-16 11:05:43.021762000 -0400
@@ -54,7 +54,7 @@
                                    install_dir: desktop_dir)
 
     # Also install in the autostart directory.
-    autostart_dir = join_paths(sysconfdir, 'xdg', 'autostart')
+    autostart_dir = join_paths(datadir, 'xdg', 'autostart')
     meson.add_install_script('install-file.py',
                              desktop_file.full_path(),
                              autostart_dir)
diff -Naur ./meson.build ./meson.build
--- ./meson.build	2023-01-22 15:37:05.000000000 -0500
+++ ./meson.build	2023-07-16 11:06:17.553250900 -0400
@@ -28,6 +28,7 @@
 conf.set_quoted('PACKAGE_URL', 'https://gitlab.freedesktop.org/geoclue/geoclue/wikis/home')
 conf.set_quoted('PACKAGE_BUGREPORT', 'https://gitlab.freedesktop.org/geoclue/geoclue/issues/new')
 conf.set_quoted('TEST_SRCDIR', meson.project_source_root() + '/data/')
+conf.set_quoted('DATADIR', datadir)
 conf.set_quoted('LOCALEDIR', localedir)
 conf.set_quoted('SYSCONFDIR', sysconfdir)
 conf.set_quoted('MOZILLA_API_KEY', get_option('mozilla-api-key'))
diff -Naur ./src/gclue-config.c ./src/gclue-config.c
--- ./src/gclue-config.c	2023-01-22 15:37:05.000000000 -0500
+++ ./src/gclue-config.c	2023-07-16 11:20:09.563762900 -0400
@@ -26,7 +26,8 @@
 
 #include "gclue-config.h"
 
-#define CONFIG_FILE_PATH SYSCONFDIR "/geoclue/geoclue.conf"
+#define CONFIG_FILE_PATH            SYSCONFDIR "/geoclue/geoclue.conf"
+#define DEFAULT_CONFIG_FILE_PATH    DATADIR "/defaults/geoclue/geoclue.conf"
 #define CONFIG_D_DIRECTORY SYSCONFDIR "/geoclue/conf.d/"
 
 /* This class will be responsible for fetching configuration. */
@@ -554,12 +555,20 @@
         g_autoptr(GArray) files = NULL;
         char *name;
         gsize i;
+        
+        const gchar *conf_path = NULL;
+	
+	        if (g_file_test (CONFIG_FILE_PATH, G_FILE_TEST_EXISTS)) {
+	                conf_path = CONFIG_FILE_PATH;
+	        } else {
+	                conf_path = DEFAULT_CONFIG_FILE_PATH;
+	        }
 
         config->priv = gclue_config_get_instance_private (config);
         config->priv->key_file = g_key_file_new ();
 
         /* Load config file from default path, log all missing parameters */
-        load_config_file (config, CONFIG_FILE_PATH, TRUE);
+        load_config_file (config, conf_path, TRUE);
 
         /*
          * Apply config overwrites from conf.d style config files,