diff --git a/data/meson.build b/data/meson.build
index b22ff55..0b48836 100644
--- a/data/meson.build
+++ b/data/meson.build
@@ -14,7 +14,7 @@ if get_option('enable-backend')
         conf.set('default_wifi_enable', 'false')
     endif
 
-    conf_dir = join_paths(sysconfdir, 'geoclue')
+    conf_dir = join_paths(datadir, 'defaults', 'geoclue')
     configure_file(output: 'geoclue.conf',
                    input: 'geoclue.conf.in',
                    configuration: conf,
diff --git a/demo/meson.build b/demo/meson.build
index 1427fbe..8612474 100644
--- a/demo/meson.build
+++ b/demo/meson.build
@@ -54,7 +54,7 @@ if get_option('demo-agent')
                                    install_dir: desktop_dir)
 
     # Also install in the autostart directory.
-    autostart_dir = join_paths(sysconfdir, 'xdg', 'autostart')
+    autostart_dir = join_paths(datadir, 'xdg', 'autostart')
     meson.add_install_script('install-file.py',
                              desktop_file.full_path(),
                              autostart_dir)
diff --git a/meson.build b/meson.build
index c561572..9e4315f 100644
--- a/meson.build
+++ b/meson.build
@@ -28,6 +28,7 @@ conf.set_quoted('PACKAGE_STRING', 'geoclue ' + gclue_version)
 conf.set_quoted('PACKAGE_URL', 'https://gitlab.freedesktop.org/geoclue/geoclue/wikis/home')
 conf.set_quoted('PACKAGE_BUGREPORT', 'https://gitlab.freedesktop.org/geoclue/geoclue/issues/new')
 conf.set_quoted('TEST_SRCDIR', meson.project_source_root() + '/data/')
+conf.set_quoted('DATADIR', datadir)
 conf.set_quoted('LOCALEDIR', localedir)
 conf.set_quoted('SYSCONFDIR', sysconfdir)
 conf.set_quoted('DEFAULT_WIFI_URL', get_option('default-wifi-url'))
diff --git a/src/gclue-config.c b/src/gclue-config.c
index acd6009..49ef75b 100644
--- a/src/gclue-config.c
+++ b/src/gclue-config.c
@@ -27,6 +27,7 @@
 #include "gclue-config.h"
 
 #define CONFIG_FILE_PATH SYSCONFDIR "/geoclue/geoclue.conf"
+#define DEFAULT_CONFIG_FILE_PATH DATADIR "/defaults/geoclue/geoclue.conf"
 #define CONFIG_D_DIRECTORY SYSCONFDIR "/geoclue/conf.d/"
 
 /* This class will be responsible for fetching configuration. */
@@ -530,6 +531,14 @@ gclue_config_init (GClueConfig *config)
         char *name;
         gsize i;
 
+        const gchar *conf_path = NULL;
+
+        if (g_file_test (CONFIG_FILE_PATH, G_FILE_TEST_EXISTS)) {
+                conf_path = CONFIG_FILE_PATH;
+        } else {
+                conf_path = DEFAULT_CONFIG_FILE_PATH;
+        }
+
         config->priv = priv;
 
         /* Sources should be enabled by default */
@@ -548,7 +557,7 @@ gclue_config_init (GClueConfig *config)
 
         /* Load config file from default path, log all missing parameters */
         priv->key_file = g_key_file_new ();
-        load_config_file (config, CONFIG_FILE_PATH);
+        load_config_file (config, conf_path);
 
         /*
          * Apply config overwrites from conf.d style config files,
