From 814c28cced39afdb433eb641eebe267a5bf89783 Mon Sep 17 00:00:00 2001
From: Joshua Strobl <joshua@streambits.io>
Date: Mon, 16 Mar 2020 20:09:04 +0200
Subject: [PATCH 1/1] Add redshift to default config

---
 data/geoclue.conf.in | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/data/geoclue.conf.in b/data/geoclue.conf.in
index bebe471..a257e9c 100644
--- a/data/geoclue.conf.in
+++ b/data/geoclue.conf.in
@@ -128,3 +128,8 @@ users=
 allowed=true
 system=false
 users=
+
+[redshift]
+allowed=true
+system=false
+users=
-- 
2.25.1

