From 459a0f3d4e2693966fded5bb3dd00d45d69591a7 Mon Sep 17 00:00:00 2001
From: Ikey Doherty <ikey@solus-project.com>
Date: Sun, 3 Dec 2017 16:12:41 +0000
Subject: [PATCH] shared: Support Linux 4.14

Signed-off-by: Ikey Doherty <ikey@solus-project.com>
---
 src/shared/linux_osl.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/shared/linux_osl.c b/src/shared/linux_osl.c
index 6157d18..15f64cb 100644
--- a/src/shared/linux_osl.c
+++ b/src/shared/linux_osl.c
@@ -1076,7 +1076,7 @@ osl_os_get_image_block(char *buf, int len, void *image)
 	if (!image)
 		return 0;
 
-	rdlen = kernel_read(fp, fp->f_pos, buf, len);
+	rdlen = kernel_read(fp, buf, len, &fp->f_pos);
 	if (rdlen > 0)
 		fp->f_pos += rdlen;
 
-- 
2.15.1

