--- broadcom-sta.orig/src/wl/sys/wl_linux.c	2015-06-11 16:08:48.824052643 +0800
+++ broadcom-sta/src/wl/sys/wl_linux.c	2015-06-11 16:09:33.444049717 +0800
@@ -1307,7 +1307,7 @@
 	dev->priv = priv_link;
 #else
 
-	dev = alloc_netdev(sizeof(priv_link_t), intf_name, ether_setup);
+	dev = alloc_netdev(sizeof(priv_link_t), intf_name, NET_NAME_UNKNOWN, ether_setup);
 	if (!dev) {
 		WL_ERROR(("wl%d: %s: alloc_netdev failed\n",
 			(wl->pub)?wl->pub->unit:wlif->subunit, __FUNCTION__));
