From 182c05c5eae366c5ab9ca9d170623ed944e24d99 Mon Sep 17 00:00:00 2001
From: Joshua Strobl <joshua@streambits.io>
Date: Thu, 18 Feb 2021 23:19:54 +0200
Subject: [PATCH 1/1] Add shell completions to correct place

---
 setup.py | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/setup.py b/setup.py
index af68b485e..21e16a419 100644
--- a/setup.py
+++ b/setup.py
@@ -59,9 +59,9 @@ if len(sys.argv) >= 2 and sys.argv[1] == 'py2exe':
     params = py2exe_params
 else:
     files_spec = [
-        ('etc/bash_completion.d', ['youtube-dl.bash-completion']),
-        ('etc/fish/completions', ['youtube-dl.fish']),
-        ('share/doc/youtube_dl', ['README.txt']),
+        ('share/bash-completion/completions', ['youtube-dl.bash-completion']),
+        ('share/fish/completions', ['youtube-dl.fish']),
+        ('share/zsh/site-functions', ['youtube-dl.zsh']),
         ('share/man/man1', ['youtube-dl.1'])
     ]
     root = os.path.dirname(os.path.abspath(__file__))
-- 
2.30.0

