From f54db9e2b06f92d4a193ade8a85e02028d53fa3c Mon Sep 17 00:00:00 2001
From: Joey Riches <josephriches@gmail.com>
Date: Sun, 18 Jun 2023 13:36:42 +0100
Subject: [PATCH 1/1] daemon: Stateless systemd service path

---
 src/daemon.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/daemon.c b/src/daemon.c
index 92a5c06..ce60745 100644
--- a/src/daemon.c
+++ b/src/daemon.c
@@ -55,7 +55,7 @@
 #define PATH_PASSWD "passwd"
 #define PATH_SHADOW "shadow"
 #define PATH_GROUP "/etc/group"
-#define PATH_DM     "/etc/systemd/system/display-manager.service"
+#define PATH_DM     "/usr/lib64/systemd/system/display-manager.service"
 
 enum
 {
-- 
2.40.1

