From afe45d3f1d03f8e3efd9bbd723b92477d020a343 Mon Sep 17 00:00:00 2001
From: Joey Riches <josephriches@gmail.com>
Date: Wed, 16 Nov 2022 10:11:21 +0000
Subject: [PATCH 1/1] Change passwd path to /bin/, since that is where our 
 binary is.

---
 src/user.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/user.c b/src/user.c
index 199e54d..5149b24 100644
--- a/src/user.c
+++ b/src/user.c
@@ -2262,7 +2262,7 @@ user_change_password_mode_authorized_cb (Daemon                *daemon,
                 if (mode == PASSWORD_MODE_SET_AT_LOGIN ||
                     mode == PASSWORD_MODE_NONE) {
 
-                        argv[0] = "/usr/bin/passwd";
+                        argv[0] = "/bin/passwd";
                         argv[1] = "-d";
                         argv[2] = "--";
                         argv[3] = accounts_user_get_user_name (ACCOUNTS_USER (user));
-- 
2.35.5

