From c4b9e33dd5dec82ecd402465fc5b742bc1c3dd74 Mon Sep 17 00:00:00 2001
From: Joshua Strobl <joshua@stroblindustries.com>
Date: Sun, 25 Jun 2017 19:11:40 +0300
Subject: [PATCH 1/1] Change passwd path to /bin/, since that is where our
 binary is.

---
 src/user.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/user.c b/src/user.c
index 917d427..d56d0f2 100644
--- a/src/user.c
+++ b/src/user.c
@@ -2396,7 +2396,7 @@ user_change_password_mode_authorized_cb (Daemon                *daemon,

                 if (mode == PASSWORD_MODE_SET_AT_LOGIN ||
                     mode == PASSWORD_MODE_NONE) {
-                        argv[0] = "/usr/bin/passwd";
+                        argv[0] = "/bin/passwd";
                         argv[1] = "-d";
                         argv[2] = "--";
                         argv[3] = accounts_user_get_user_name (ACCOUNTS_USER (user));
--
2.40.1
