iff --git a/plugins/svg.c b/plugins/svg.c
index a0eb715..39fed02 100644
--- a/plugins/svg.c
+++ b/plugins/svg.c
@@ -31,7 +31,10 @@
 #ifndef RSVG_CAIRO_H
 #include <librsvg/rsvg-cairo.h>
 #endif
-#ifndef LIBRSVG_FEATURES_H
+#if !defined(RSVG_FEATURES_H) && !defined(LIBRSVG_FEATURES_H)
+#include <librsvg/rsvg-features.h>
+#endif
+#if !defined(RSVG_FEATURES_H) && !defined(LIBRSVG_FEATURES_H)
 #include <librsvg/librsvg-features.h>
 #endif
