From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Reilly Brogan <reilly@reillybrogan.com>
Date: Sat, 9 Nov 2024 16:26:55 -0600
Subject: [PATCH] Service tweaks

---
 systemd/waydroid-container.service | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/systemd/waydroid-container.service b/systemd/waydroid-container.service
index 4a86383..c263608 100644
--- a/systemd/waydroid-container.service
+++ b/systemd/waydroid-container.service
@@ -1,5 +1,7 @@
 [Unit]
 Description=Waydroid Container
+Wants=waydroid-pre.service
+After=waydroid-pre.service
 
 [Service]
 UMask=0022
