From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Reilly Brogan <reilly@reillybrogan.com>
Date: Tue, 23 Apr 2024 18:34:01 -0500
Subject: [PATCH] Stateless

---
 Makefile.am           | 2 +-
 daemon/cups-browsed.c | 8 +++++++-
 2 files changed, 8 insertions(+), 2 deletions(-)

diff --git a/Makefile.am b/Makefile.am
index 5e86aedb..61f2bb81 100644
--- a/Makefile.am
+++ b/Makefile.am
@@ -47,7 +47,7 @@ implicitclass_CFLAGS = \
 # ===========
 # CUPS Config
 # ===========
-pkgcupsserverrootdir = $(CUPS_SERVERROOT)
+pkgcupsserverrootdir = "/usr/share/defaults/cups/"
 pkgcupsserverroot_DATA = \
 	daemon/cups-browsed.conf
 
diff --git a/daemon/cups-browsed.c b/daemon/cups-browsed.c
index b06461b4..0b1412af 100644
--- a/daemon/cups-browsed.c
+++ b/daemon/cups-browsed.c
@@ -33,6 +33,7 @@
 #include <signal.h>
 #include <regex.h>
 #include <pthread.h>
+#include <unistd.h>
 
 #include <glib.h>
 
@@ -12600,7 +12601,12 @@ read_configuration (const char *filename)
   cluster_t *cluster = NULL;
 
   if (!filename)
-    filename = CUPS_SERVERROOT "/cups-browsed.conf";
+    {
+      filename = CUPS_SERVERROOT "/cups-browsed.conf";
+      if (access(filename, F_OK) != 0) {
+        filename = "/usr/share/defaults/cups/cups-browsed.conf";
+      }
+    }
 
   if ((fp = cupsFileOpen(filename, "r")) == NULL)
   {
