From 4ccd64b65b4672ce211bf56ee1ca6e1a5f8ebf5c Mon Sep 17 00:00:00 2001
From: Zdenek Dohnal <zdohnal@redhat.com>
Date: Wed, 7 Feb 2024 14:59:41 +0100
Subject: [PATCH 1/2] Init variables which can be later used uninitialized

---
 daemon/cups-browsed.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/daemon/cups-browsed.c b/daemon/cups-browsed.c
index 29fd34a1..a144abae 100644
--- a/daemon/cups-browsed.c
+++ b/daemon/cups-browsed.c
@@ -6935,7 +6935,7 @@ on_job_state (CupsNotifier *object,
 
       // The priority order for the PDLs is the same as in the
       // PPD generator in ppd/ppd-generator.c of libppd
-      document_format = (char *)malloc(sizeof(char) * 32);
+      document_format = (char *)calloc(32, sizeof(char));
       if (cupsArrayFind(pdl_list, "application/vnd.cups-pdf"))
 	strcpy(document_format, "application/vnd.cups-pdf");
       else if (cupsArrayFind(pdl_list, "image/urf"))
@@ -10951,7 +10951,7 @@ resolve_callback(void* arg)
   // Called whenever a service has been resolved successfully
 
   // New remote printer found
-  AvahiStringList *rp_entry, *adminurl_entry;
+  AvahiStringList *rp_entry = NULL, *adminurl_entry = NULL;
   char *rp_key, *rp_value, *adminurl_key, *adminurl_value;
 
   debug_printf("Avahi Resolver: Service '%s' of type '%s' in domain '%s' with host name '%s' and port %d on interface '%s' (%s).\n",

From 281080bf212b531bb9f7d7f51ba6ea634418380f Mon Sep 17 00:00:00 2001
From: Zdenek Dohnal <zdohnal@redhat.com>
Date: Wed, 7 Feb 2024 16:42:08 +0100
Subject: [PATCH 2/2] Fix memory leak found by valgrind

---
 daemon/cups-browsed.c | 8 +++++++-
 1 file changed, 7 insertions(+), 1 deletion(-)

diff --git a/daemon/cups-browsed.c b/daemon/cups-browsed.c
index a144abae..e297448c 100644
--- a/daemon/cups-browsed.c
+++ b/daemon/cups-browsed.c
@@ -10889,7 +10889,7 @@ resolve_callback(void* arg)
   AVAHI_GCC_UNUSED void* userdata = a->userdata;
 
   char ifname[IF_NAMESIZE];
-  AvahiStringList *uuid_entry, *printer_type_entry;
+  AvahiStringList *uuid_entry = NULL, *printer_type_entry;
   char *uuid_key, *uuid_value;
 
   debug_printf("resolve_callback() in THREAD %ld\n", pthread_self());
@@ -11166,6 +11166,12 @@ resolve_callback(void* arg)
   }
 
  ignore:
+  if (uuid_entry)
+  {
+    avahi_free(uuid_key);
+    avahi_free(uuid_value);
+  }
+
   if (a->name) free((char*)a->name);
   if (a->type) free((char*)a->type);
   if (a->domain) free((char*)a->domain);
