From 817797571b1137052261480bd2371067bf39b0dd Mon Sep 17 00:00:00 2001
From: Peter O'Connor <peter@solus-project.com>
Date: Fri, 20 Apr 2018 10:16:16 +1000
Subject: [PATCH] Disable building unshipped code

Signed-off-by: Peter O'Connor <peter@solus-project.com>
---
 CMakeLists.txt | 1 -
 1 file changed, 1 deletion(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 673ef9d..db3d2e0 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -95,7 +95,6 @@ set_target_properties(komparediff2 PROPERTIES
 
 install(TARGETS komparediff2 EXPORT LibKompareDiff2Targets ${INSTALL_TARGETS_DEFAULT_ARGS})
 
-add_subdirectory(tests)
 
 install(FILES
     ${CMAKE_CURRENT_BINARY_DIR}/diff2_export.h
-- 
2.16.3

