From 116b5528928786085cff33ca4a2dec2fb6017ab1 Mon Sep 17 00:00:00 2001
From: Ikey Doherty <ikey@solus-project.com>
Date: Sun, 3 Dec 2017 11:17:30 +0000
Subject: [PATCH] Support stateless /etc/shells file

Signed-off-by: Ikey Doherty <ikey@solus-project.com>
---
 src/legacy/getusershell.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/src/legacy/getusershell.c b/src/legacy/getusershell.c
index 5fecdec..ea2a5fd 100644
--- a/src/legacy/getusershell.c
+++ b/src/legacy/getusershell.c
@@ -17,6 +17,7 @@ void endusershell(void)
 void setusershell(void)
 {
 	if (!f) f = fopen("/etc/shells", "rbe");
+	if (!f) f = fopen("/usr/share/defaults/etc/shells", "rbe");
 	if (!f) f = fmemopen((void *)defshells, sizeof defshells - 1, "rb");
 }
 
-- 
2.15.1

