diff --git a/Makefile b/Makefile
index bb1c6c9..09e3571 100644
--- a/Makefile
+++ b/Makefile
@@ -23,7 +23,6 @@ install:
 	install -Dm00644 internal/assets/svg/icon.svg $(DESTDIR)$(PREFIX)/share/icons/hicolor/scalable/apps/$(APPID).svg
 	install -Dm00644 internal/assets/unix/$(APPID).desktop $(DESTDIR)$(PREFIX)/share/applications/$(APPID).desktop
 	install -Dm00644 internal/assets/unix/$(APPID).appdata.xml $(DESTDIR)$(PREFIX)/share/appdata/$(APPID).appdata.xml
-	sudo gtk-update-icon-cache -f /usr/share/icons/hicolor/
 
 uninstall:
 	-rm $(DESTDIR)$(PREFIX)/bin/$(NAME)
