From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Reilly Brogan <reilly@reillybrogan.com>
Date: Thu, 12 Sep 2024 15:06:41 -0500
Subject: [PATCH] temp

Signed-off-by: Reilly Brogan <reilly@reillybrogan.com>
---
 CMakeLists.txt | 43 ++++++++++++++++++++-----------------------
 1 file changed, 20 insertions(+), 23 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 97af9025..f8e7fe3f 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -15,7 +15,6 @@ project(elisa
     LANGUAGES CXX)
 
 set(REQUIRED_KF_VERSION "6.3")
-set(REQUIRED_KA_VERSION "0.11.76") # KirigamiAddons Version
 find_package(ECM ${REQUIRED_KF_VERSION} REQUIRED NO_MODULE)
 set(CMAKE_MODULE_PATH "${CMAKE_CURRENT_SOURCE_DIR}/cmake" ${ECM_MODULE_PATH})
 
@@ -29,9 +28,24 @@ include(ECMAddTests)
 include(ECMQtDeclareLoggingCategory)
 include(ECMGenerateQmlTypes)
 include(ECMAddAndroidApk)
+include(ECMFindQmlModule)
 
 set(REQUIRED_QT_VERSION "6.5.0")
-find_package(Qt6 ${REQUIRED_QT_VERSION} CONFIG REQUIRED Core Network Qml Quick Test Sql Multimedia Svg Gui Widgets QuickTest Concurrent QuickControls2)
+find_package(Qt6 ${REQUIRED_QT_VERSION} CONFIG REQUIRED COMPONENTS
+    Concurrent
+    Core
+    Gui
+    Multimedia
+    Network
+    Qml
+    Quick
+    QuickControls2
+    QuickTest
+    Sql
+    Svg
+    Test
+    Widgets
+)
 
 find_package(Qt6Core ${REQUIRED_QT_VERSION} CONFIG REQUIRED Private)
 
@@ -48,26 +62,6 @@ if (Qt6DBus_FOUND)
 set(QtDBus_FOUND 1)
 endif()
 
-find_package(Qt6QuickWidgets ${REQUIRED_QT_VERSION} CONFIG QUIET)
-set_package_properties(Qt6QuickWidgets PROPERTIES
-    DESCRIPTION "Qt6 Quick Widgets is needed at runtime to provide the interface."
-    TYPE RUNTIME)
-
-find_package(Qt6QuickControls2 ${REQUIRED_QT_VERSION} CONFIG QUIET)
-set_package_properties(Qt6QuickControls2 PROPERTIES
-    DESCRIPTION "Qt6 Quick Controls version 2 is needed at runtime to provide the interface."
-    TYPE RUNTIME)
-
-find_package(KF6Kirigami ${REQUIRED_KF_VERSION} CONFIG QUIET)
-set_package_properties(KF6Kirigami PROPERTIES
-    DESCRIPTION "KF Kirigami is needed to provide the mobile UI components."
-    TYPE REQUIRED)
-
-find_package(KF6KirigamiAddons ${REQUIRED_KA_VERSION} CONFIG QUIET)
-set_package_properties(KF6KirigamiAddons PROPERTIES
-    DESCRIPTION "KF KirigamiAddons is needed to provide the about KDE dialogue."
-    TYPE REQUIRED)
-
 find_package(KF6I18n ${REQUIRED_KF_VERSION} CONFIG QUIET)
 set_package_properties(KF6I18n PROPERTIES
     DESCRIPTION "KF text internationalization library."
@@ -123,7 +117,7 @@ if (NOT ANDROID)
     find_package(KF6QQC2DesktopStyle ${REQUIRED_KF_VERSION} CONFIG QUIET)
     set_package_properties(KF6QQC2DesktopStyle PROPERTIES
         DESCRIPTION "KDE QtQuick style for desktop"
-        TYPE REQUIRED)
+        TYPE RUNTIME)
 endif()
 
 find_package(UPNPQT CONFIG QUIET)
@@ -142,6 +136,9 @@ set_package_properties(LIBVLC PROPERTIES
     URL "https://www.videolan.org/vlc/libvlc.html"
     TYPE RECOMMENDED)
 
+ecm_find_qmlmodule(org.kde.kirigami 2.19)
+ecm_find_qmlmodule(org.kde.kirigamiaddons.formcard 1.0)
+
 include(FeatureSummary)
 include(GenerateExportHeader)
 include(ECMSetupVersion)
