From d35741451b5657a8f26cb31b03f74c5ea1607636 Mon Sep 17 00:00:00 2001
From: Joshua Strobl <joshua@stroblindustries.com>
Date: Sat, 3 Mar 2018 19:03:48 +0200
Subject: [PATCH 1/1] Don't require prompts.

---
 configure | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/configure b/configure
index b6968669..76b68f67 100755
--- a/configure
+++ b/configure
@@ -83,7 +83,7 @@ de_error=0
 wd=$PWD
 
 # update icons for dark/light GTK themes
-read -p "Use icons for dark GTK+ themes? [y/N] " yn
+yn=y
 case $yn in
     [Yy]* )
         rm "$wd/actions/22x22" "$wd/devices/scalable" "$wd/panel/16" \
@@ -109,7 +109,7 @@ case $yn in
         echo "Updated theme to use icons for light GTK+ themes."
         ;;
 esac
-read -p "Use icons for a dark coloured panel? [y/N] " yn
+yn=y
 case $yn in
     [Yy]* )
         ln -sfr "$wd/panel/16-dark" "$wd/panel/16"
@@ -129,7 +129,7 @@ cd "$wd/places/scalable"
 ! distromatch && alt_distromatch
 
 # prompt for desktop environment integration instead of distro
-read -p "Use distributor logo from desktop environment? [y/N] " yn
+yn=y
 case $yn in
     [Yy]* )
         case $DE in
-- 
2.16.2

