From 60dc342c63722d7c8c69543a7ddd958968b49589 Mon Sep 17 00:00:00 2001
From: Joshua Strobl <joshua@stroblindustries.com>
Date: Sun, 25 Dec 2016 16:24:36 +0200
Subject: [PATCH 1/1] Drop request for input during desktop environment
 integration.

---
 configure | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/configure b/configure
index 53bdd82..97cf711 100644
--- a/configure
+++ b/configure
@@ -86,7 +86,7 @@ alt_distromatch() {
 ! distromatch && alt_distromatch
 
 # prompt for desktop environment integration instead of distro
-read -p "Use distributor logo from desktop environment? [y/N] " yn
+yn=y
 case $yn in
     [Yy]* )
         case $DE in
-- 
2.11.0

