--- a/build/bootstrap.proj
+++ b/build/bootstrap.proj
@@ -8,7 +8,7 @@
     </PropertyGroup>
 
     <ItemGroup>
-        <PackageDownload Include="ILMerge" version="[3.0.21]" />
+        <PackageDownload Include="ILRepack" version="[2.0.18]" />
         <PackageDownload Include="Lucene.Net" version="[3.0.3]" />
         <PackageDownload Include="MicroBuild.Core" version="[0.2.0]" />
         <PackageDownload Include="Microsoft.Build" version="[15.1.262-preview5]" />
