--- a/build/common.project.props
+++ b/build/common.project.props
@@ -29,7 +29,7 @@
     <NupkgOutputDirectory Condition=" '$(BuildRTM)' == 'true' ">$(ArtifactsDirectory)ReleaseNupkgs\</NupkgOutputDirectory>
     <SolutionPackagesFolder>$(RepositoryRootDirectory)packages\</SolutionPackagesFolder>
     <XunitConsoleExePath>$(SolutionPackagesFolder)xunit.runner.console\2.4.1\tools\net452\xunit.console.x86.exe</XunitConsoleExePath>
-    <ILMergeExePath>$(SolutionPackagesFolder)ilmerge\3.0.21\tools\net452\ILMerge.exe</ILMergeExePath>
+    <ILMergeExePath>$(SolutionPackagesFolder)ilrepack\2.0.18\tools\ILRepack.exe</ILMergeExePath>
     <XunitXmlLoggerDirectory>$(SolutionPackagesFolder)xunitxml.testlogger\2.0.0\build\_common</XunitXmlLoggerDirectory>
     <NuGetBuildTasksPackTargets Condition="Exists('$(SolutionPackagesFolder)nuget.build.tasks.pack\4.9.2\build\NuGet.Build.Tasks.Pack.targets')">$(SolutionPackagesFolder)nuget.build.tasks.pack\4.9.2\build\NuGet.Build.Tasks.Pack.targets</NuGetBuildTasksPackTargets>
     <EnlistmentRoot>$(RepositoryRootDirectory)</EnlistmentRoot>
--- a/src/NuGet.Clients/NuGet.CommandLine/NuGet.CommandLine.csproj
+++ b/src/NuGet.Clients/NuGet.CommandLine/NuGet.CommandLine.csproj
@@ -121,10 +121,15 @@
     </EmbeddedResource>
   </ItemGroup>
 
+  <Target Name="CopyNuGetExe" AfterTargets="Build">
+    <Copy SourceFiles="$(OutputPath)NuGet.exe" DestinationFolder="$(MSBuildThisFileDirectory)" />
+    <Copy SourceFiles="@(MergeInclude)" DestinationFolder="$(MSBuildThisFileDirectory)" />
+  </Target>
+
   <!-- 	Since we are moving some code and strings from NuGet.CommandLine to NuGet.Commands, we opted to go through normal localization process
         (build .resources.dll) and then add them to the ILMerged nuget.exe -->
   <!-- This will also be called from CI build, after assemblies are localized, since our test infra takes nuget.exe before Localization  -->
-  <Target Name="ILMergeNuGetExe" AfterTargets="Build" Condition="'$(BuildingInsideVisualStudio)' != 'true' and '$(SkipILMergeOfNuGetExe)' != 'true'" >
+  <Target Name="ILMergeNuGetExe" AfterTargets="CopyNuGetExe" Condition="'$(BuildingInsideVisualStudio)' != 'true' and '$(SkipILMergeOfNuGetExe)' != 'true'" >
     <PropertyGroup>
       <!-- when done after build, no localizedartifacts are built yet, so expected localized artifact count is 0. -->
       <ExpectedLocalizedArtifactCount Condition="'$(ExpectedLocalizedArtifactCount)' == ''">0</ExpectedLocalizedArtifactCount>
@@ -138,7 +143,7 @@
     <Error Text="Satellite assemblies count ILMerged into NuGet.exe should be $(ExpectedLocalizedArtifactCount), but was: @(LocalizedArtifacts-&gt;Count())" 
            Condition="'@(LocalizedArtifacts-&gt;Count())' != '$(ExpectedLocalizedArtifactCount)'" />
     <PropertyGroup>
-      <PathToBuiltNuGetExe>$(OutputPath)NuGet.exe</PathToBuiltNuGetExe>
-      <IlmergeCommand>$(ILMergeExePath) /lib:$(OutputPath) /out:$(ArtifactsDirectory)$(VsixOutputDirName)\NuGet.exe @(MergeAllowDup -> '/allowdup:%(Identity)', ' ') /log:$(OutputPath)IlMergeLog.txt</IlmergeCommand>
+      <PathToBuiltNuGetExe>NuGet.exe</PathToBuiltNuGetExe>
+      <IlmergeCommand>$(ILMergeExePath) /lib:$(OutputPath) /lib:\usr\lib64\mono\4.5\Facades /out:$(ArtifactsDirectory)$(VsixOutputDirName)\NuGet.exe @(MergeAllowDup -> '/allowdup:%(Identity)', ' ') /log:$(OutputPath)IlMergeLog.txt</IlmergeCommand>
       <IlmergeCommand Condition="Exists($(MS_PFX_PATH))">$(IlmergeCommand) /delaysign /keyfile:$(MS_PFX_PATH)</IlmergeCommand>
       <!-- LocalizedArtifacts need fullpath, since there will be duplicate file names -->
       <IlmergeCommand>$(IlmergeCommand) $(PathToBuiltNuGetExe) @(BuildArtifacts->'%(filename)%(extension)', ' ') @(LocalizedArtifacts->'%(fullpath)', ' ')</IlmergeCommand>
--- a/src/NuGet.Clients/NuGet.CommandLine/ilmerge.props
+++ b/src/NuGet.Clients/NuGet.CommandLine/ilmerge.props
@@ -75,5 +75,40 @@
     <MergeAllowdup Include="NuGetFrameworkUtility"/>
     <MergeAllowdup Include="OneWayCompatibilityMappingEntry"/>
     <MergeAllowdup Include="Strings"/>
+
+    <MergeAllowdup Include="NuGet.Frameworks.CompatibilityMappingComparer"/>
+    <MergeAllowdup Include="NuGet.Frameworks.FrameworkPrecedenceSorter"/>
+    <MergeAllowdup Include="NuGet.Frameworks.FrameworkRangeComparer"/>
+    <MergeAllowdup Include="NuGet.Frameworks.NuGetFrameworkFullComparer"/>
+    <MergeAllowdup Include="NuGet.Frameworks.NuGetFrameworkNameComparer"/>
+    <MergeAllowdup Include="NuGet.Frameworks.NuGetFrameworkSorter"/>
+    <MergeAllowdup Include="NuGet.Frameworks.CompatibilityListProvider"/>
+    <MergeAllowdup Include="NuGet.Frameworks.CompatibilityProvider"/>
+    <MergeAllowdup Include="NuGet.Frameworks.CompatibilityTable"/>
+    <MergeAllowdup Include="NuGet.Frameworks.IFrameworkCompatibilityListProvider"/>
+    <MergeAllowdup Include="NuGet.Frameworks.IFrameworkCompatibilityProvider"/>
+    <MergeAllowdup Include="NuGet.Frameworks.IFrameworkMappings"/>
+    <MergeAllowdup Include="NuGet.Frameworks.IFrameworkNameProvider"/>
+    <MergeAllowdup Include="NuGet.Frameworks.IFrameworkSpecific"/>
+    <MergeAllowdup Include="NuGet.Frameworks.IFrameworkTargetable"/>
+    <MergeAllowdup Include="NuGet.Frameworks.IPortableFrameworkMappings"/>
+    <MergeAllowdup Include="NuGet.Frameworks.DefaultCompatibilityProvider"/>
+    <MergeAllowdup Include="NuGet.Frameworks.DefaultFrameworkMappings"/>
+    <MergeAllowdup Include="NuGet.Frameworks.DefaultFrameworkNameProvider"/>
+    <MergeAllowdup Include="NuGet.Frameworks.DefaultPortableFrameworkMappings"/>
+    <MergeAllowdup Include="NuGet.Frameworks.FallbackFramework"/>
+    <MergeAllowdup Include="NuGet.Frameworks.FrameworkConstants"/>
+    <MergeAllowdup Include="NuGet.Frameworks.FrameworkException"/>
+    <MergeAllowdup Include="NuGet.Frameworks.FrameworkExpander"/>
+    <MergeAllowdup Include="NuGet.Frameworks.NuGetFrameworkExtensions"/>
+    <MergeAllowdup Include="NuGet.Frameworks.FrameworkNameHelpers"/>
+    <MergeAllowdup Include="NuGet.Frameworks.FrameworkNameProvider"/>
+    <MergeAllowdup Include="NuGet.Frameworks.FrameworkRange"/>
+    <MergeAllowdup Include="NuGet.Frameworks.FrameworkReducer"/>
+    <MergeAllowdup Include="NuGet.Frameworks.FrameworkRuntimePair"/>
+    <MergeAllowdup Include="NuGet.Frameworks.FrameworkSpecificMapping"/>
+    <MergeAllowdup Include="NuGet.Frameworks.NuGetFramework"/>
+    <MergeAllowdup Include="NuGet.Frameworks.NuGetFrameworkUtility"/>
+    <MergeAllowdup Include="NuGet.Frameworks.OneWayCompatibilityMappingEntry"/>
   </ItemGroup>
 </Project>
