diff --git a/src/daemon/daemon.conf.in b/src/daemon/daemon.conf.in
index a955523..f8eb631 100644
--- a/src/daemon/daemon.conf.in
+++ b/src/daemon/daemon.conf.in
@@ -53,7 +53,7 @@ ifelse(@HAVE_DBUS@, 1, [dnl
 ; log-time = no
 ; log-backtrace = 0
 
-; resample-method = speex-float-1
+resample-method = soxr-mq
 ; avoid-resampling = false
 ; enable-remixing = yes
 ; remixing-use-all-sink-channels = yes
