From 50facaae47bfe4f73fa35d0b43409b651a4fdb07 Mon Sep 17 00:00:00 2001
From: Ikey Doherty <ikey@solus-project.com>
Date: Sun, 29 Nov 2015 19:13:29 +0000
Subject: [PATCH] Disable flat volumes as they're making folks unhappy

Signed-off-by: Ikey Doherty <ikey@solus-project.com>
---
 src/daemon/daemon.conf.in | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/daemon/daemon.conf.in b/src/daemon/daemon.conf.in
index 067738d..5c065ed 100644
--- a/src/daemon/daemon.conf.in
+++ b/src/daemon/daemon.conf.in
@@ -56,7 +56,7 @@ ifelse(@HAVE_DBUS@, 1, [dnl
 ; enable-remixing = yes
 ; enable-lfe-remixing = no
 
-; flat-volumes = yes
+flat-volumes = no
 
 ifelse(@HAVE_SYS_RESOURCE_H@, 1, [dnl
 ; rlimit-fsize = -1
-- 
2.6.3

