diff --git a/src/daemon/default.pa.in b/src/daemon/default.pa.in
index 030334f36..08b4ae6cd 100755
--- a/src/daemon/default.pa.in
+++ b/src/daemon/default.pa.in
@@ -170,6 +170,12 @@ load-module module-filter-heuristics
 load-module module-filter-apply
 ])dnl
 
+### Enable D-Bus support if an equalizer module is found.
+### This is needed for qpaeq (pulseaudio equalizer)
+.ifexists module-equalizer-sink@PA_SOEXT@
+load-module module-dbus-protocol
+.endif
+
 ### Make some devices default
 #set-default-sink output
 #set-default-source input
