From 9370de69d3b475b27ca85f06ec4f538ae4b1510e Mon Sep 17 00:00:00 2001
From: Joshua Strobl <joshua@stroblindustries.com>
Date: Thu, 18 Jan 2018 22:06:11 +0200
Subject: [PATCH 1/1] Ensure gpicview opens in center.

---
 src/main-win.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/src/main-win.c b/src/main-win.c
index 32f6433..af1e512 100644
--- a/src/main-win.c
+++ b/src/main-win.c
@@ -172,6 +172,7 @@ void main_win_init( MainWin*mw )
         gtk_window_set_icon_from_file((GtkWindow*)mw, PACKAGE_DATA_DIR "/icons/hicolor/48x48/apps/gpicview.png", NULL);
     }
     gtk_window_set_default_size( (GtkWindow*)mw, 640, 480 );
+    gtk_window_set_position( (GtkWindow*)mw, GTK_WIN_POS_CENTER_ALWAYS);
 
 #if GTK_CHECK_VERSION(3, 0, 0)
     GtkWidget* box = gtk_box_new( GTK_ORIENTATION_VERTICAL, 0 );
-- 
2.15.1

