From 87e2fdfd9ba21a5eadc4a503996722a7dbe0741d Mon Sep 17 00:00:00 2001
From: Joey Riches <josephriches@gmail.com>
Date: Thu, 26 May 2022 19:44:01 +0100
Subject: [PATCH 1/1] Missing include

---
 IO/OMF/core/OMFElement.cxx | 1 +
 1 file changed, 1 insertion(+)

diff --git a/IO/OMF/core/OMFElement.cxx b/IO/OMF/core/OMFElement.cxx
index 0f5f03d3..385af929 100644
--- a/IO/OMF/core/OMFElement.cxx
+++ b/IO/OMF/core/OMFElement.cxx
@@ -30,6 +30,7 @@
 #include "vtk_jsoncpp.h"
 #include "vtksys/SystemTools.hxx"
 
+#include <array>
 #include <numeric>
 #include <string>
 #include <vector>
-- 
2.35.3

