Index: b/CMake/vtkCompilerExtras.cmake
===================================================================
--- a/CMake/vtkCompilerExtras.cmake	2016-01-28 08:17:38.000000000 -0800
+++ b/CMake/vtkCompilerExtras.cmake	2016-09-19 20:54:32.434279075 -0700
@@ -44,14 +44,14 @@
   option(VTK_USE_GCC_VISIBILITY "Use GCC visibility support if available." OFF)
   mark_as_advanced(VTK_USE_GCC_VISIBILITY)
 
-  if(${_gcc_version} VERSION_GREATER 4.2.0 AND BUILD_SHARED_LIBS
-    AND HAVE_GCC_VISIBILITY AND VTK_USE_GCC_VISIBILITY
-    AND NOT MINGW AND NOT CYGWIN)
-    # Should only be set if GCC is newer than 4.2.0
-    set(VTK_ABI_CXX_FLAGS "-fvisibility=hidden -fvisibility-inlines-hidden")
-  else()
-    set(VTK_ABI_CXX_FLAGS "")
-  endif()
+  #  if(${_gcc_version} VERSION_GREATER 4.2.0 AND BUILD_SHARED_LIBS
+  # AND HAVE_GCC_VISIBILITY AND VTK_USE_GCC_VISIBILITY
+  # AND NOT MINGW AND NOT CYGWIN)
+  # # Should only be set if GCC is newer than 4.2.0
+  # set(VTK_ABI_CXX_FLAGS "-fvisibility=hidden -fvisibility-inlines-hidden")
+  # else()
+  # set(VTK_ABI_CXX_FLAGS "")
+  #endif()
 
   # Set up the debug CXX_FLAGS for extra warnings
   option(VTK_EXTRA_COMPILER_WARNINGS
