From b2f5d0768f222998d9839433a10cb956874c4ceb Mon Sep 17 00:00:00 2001
From: Ryan Krattiger <ryan.krattiger@kitware.com>
Date: Fri, 15 Sep 2023 16:53:08 -0500
Subject: [PATCH] ZSpace: Use vtkDynamicLoader

---
 .../vtkZSpaceCoreCompatibilitySDKManager.cxx   | 18 +++++++++---------
 .../vtkZSpaceCoreCompatibilitySDKManager.h     |  8 ++++----
 2 files changed, 13 insertions(+), 13 deletions(-)

diff --git a/Rendering/ZSpace/vtkZSpaceCoreCompatibilitySDKManager.cxx b/Rendering/ZSpace/vtkZSpaceCoreCompatibilitySDKManager.cxx
index d38d6e68d1..7b9935ac40 100644
--- a/Rendering/ZSpace/vtkZSpaceCoreCompatibilitySDKManager.cxx
+++ b/Rendering/ZSpace/vtkZSpaceCoreCompatibilitySDKManager.cxx
@@ -55,12 +55,12 @@ vtkZSpaceCoreCompatibilitySDKManager::~vtkZSpaceCoreCompatibilitySDKManager()
 
 //------------------------------------------------------------------------------
 bool vtkZSpaceCoreCompatibilitySDKManager::loadZspaceCoreCompatibilityEntryPoints(
-  const char* zSpaceCoreCompatDllFilePath, HMODULE& dllModuleHandle,
+  const char* zSpaceCoreCompatDllFilePath, vtkLibHandle& dllModuleHandle,
   zSpaceCoreCompatEntryPoints& entryPoints)
 {
-  dllModuleHandle = LoadLibraryA(zSpaceCoreCompatDllFilePath);
+  dllModuleHandle = vtkDynamicLoader::LoadLibrary(zSpaceCoreCompatDllFilePath);
 
-  if (dllModuleHandle == nullptr)
+  if (!dllModuleHandle)
   {
     // If the release variant of the zSpace Core Compatibility API DLL
     // could not be loaded, attempt to the debug variant instead.
@@ -68,11 +68,11 @@ bool vtkZSpaceCoreCompatibilitySDKManager::loadZspaceCoreCompatibilityEntryPoint
     std::string zSpaceCoreCompatDllDebugFilePath(zSpaceCoreCompatDllFilePath);
     zSpaceCoreCompatDllDebugFilePath.append("_D");
 
-    dllModuleHandle = LoadLibraryA(zSpaceCoreCompatDllDebugFilePath.c_str());
+    dllModuleHandle = vtkDynamicLoader::LoadLibrary(zSpaceCoreCompatDllDebugFilePath);
 
-    if (dllModuleHandle == nullptr)
+    if (!dllModuleHandle)
     {
-      vtkErrorMacro(<< "Win32 Error : "
+      vtkErrorMacro(<< "System Error : "
                     << "Failed to load zSpace Core Compatibility API DLL.");
 
       return false;
@@ -81,12 +81,12 @@ bool vtkZSpaceCoreCompatibilitySDKManager::loadZspaceCoreCompatibilityEntryPoint
 
 #define ZC_COMPAT_SAMPLE_LOCAL_LOAD_ENTRY_POINT(undecoratedFuncName)                               \
   {                                                                                                \
-    void* entryPointProcAddress =                                                                  \
-      GetProcAddress(dllModuleHandle, "zccompat" #undecoratedFuncName);                            \
+    vtkSymbolPointer entryPointProcAddress =                                                       \
+      vtkDynamicLoader::GetSymbolAddress(dllModuleHandle, "zccompat" #undecoratedFuncName);        \
                                                                                                    \
     if (entryPointProcAddress == nullptr)                                                          \
     {                                                                                              \
-      vtkErrorMacro(<< "Win32 Error : "                                                            \
+      vtkErrorMacro(<< "System Error : "                                                           \
                     << "Failed to get zSpace Core Compatibility entry point "                      \
                     << "proc address for entry point "                                             \
                     << "\"zccompat" << #undecoratedFuncName << "\".");                             \
diff --git a/Rendering/ZSpace/vtkZSpaceCoreCompatibilitySDKManager.h b/Rendering/ZSpace/vtkZSpaceCoreCompatibilitySDKManager.h
index af9116c927..3a660fa170 100644
--- a/Rendering/ZSpace/vtkZSpaceCoreCompatibilitySDKManager.h
+++ b/Rendering/ZSpace/vtkZSpaceCoreCompatibilitySDKManager.h
@@ -15,6 +15,7 @@
 #ifndef vtkZSpaceCoreCompatibilitySDKManager_h
 #define vtkZSpaceCoreCompatibilitySDKManager_h
 
+#include "vtkDynamicLoader.h"
 #include "vtkZSpaceSDKManager.h"
 
 // Disable "anonymous struct/union" warning on zSpace compat headers
@@ -22,8 +23,7 @@
 #include "zSpaceCoreCompatibility.h" // zSpace header
 #pragma warning(default : 4201)
 
-#include <vector>    // for std::vector
-#include <windows.h> // for HMODULE
+#include <vector> // for std::vector
 
 VTK_ABI_NAMESPACE_BEGIN
 
@@ -116,7 +116,7 @@ protected:
   zSpaceCoreCompatEntryPoints EntryPts;
 
   // Handle to the zSpace Core Compatibility API dynamic library (.dll).
-  HMODULE zSpaceCoreCompatDllModuleHandle;
+  vtkLibHandle zSpaceCoreCompatDllModuleHandle;
 
   // Handle to the current window
   HWND WidondowHandle;
@@ -126,7 +126,7 @@ protected:
    * zSpace Core Compatibility API entry point functions (at runtime)
    */
   bool loadZspaceCoreCompatibilityEntryPoints(const char* zSpaceCoreCompatDllFilePath,
-    HMODULE& zSpaceCoreCompatDllModuleHandle, zSpaceCoreCompatEntryPoints& entryPoints);
+    vtkLibHandle& zSpaceCoreCompatDllModuleHandle, zSpaceCoreCompatEntryPoints& entryPoints);
 
   /**
    * zSpace stores matrix in column-major format (as OpenGL). The matrix
-- 
2.43.0

