Index: b/CMake/GenerateExportHeader.cmake
===================================================================
--- a/CMake/GenerateExportHeader.cmake	2016-01-28 08:17:38.000000000 -0800
+++ b/CMake/GenerateExportHeader.cmake	2016-09-19 20:59:14.709640864 -0700
@@ -175,9 +175,9 @@
         _gcc_version "${_gcc_version_info}")
     endif()
 
-    if(${_gcc_version} VERSION_LESS "4.2")
-      set(GCC_TOO_OLD TRUE)
-    endif()
+    #if(${_gcc_version} VERSION_LESS "4.2")
+    # set(GCC_TOO_OLD TRUE)
+    #endif()
   endif()
 
   if(CMAKE_CXX_COMPILER_ID MATCHES "Intel")
