From c7d6a8d81367a4ed92163c059aa3181386eabc24 Mon Sep 17 00:00:00 2001
From: Ben Boeckel <ben.boeckel@kitware.com>
Date: Mon, 3 May 2021 11:55:27 -0400
Subject: [PATCH] vtkDataArrayPrivate: include <limits> for std::numeric_limits

See: #18194
---
 Common/Core/vtkDataArrayPrivate.txx | 1 +
 1 file changed, 1 insertion(+)

diff --git a/Common/Core/vtkDataArrayPrivate.txx b/Common/Core/vtkDataArrayPrivate.txx
index eb366f1c6d0..6709f7f3ac1 100644
--- a/Common/Core/vtkDataArrayPrivate.txx
+++ b/Common/Core/vtkDataArrayPrivate.txx
@@ -24,6 +24,7 @@
 #include <algorithm>
 #include <array>
 #include <cassert> // for assert()
+#include <limits>
 #include <vector>
 
 namespace vtkDataArrayPrivate
-- 
GitLab

