From 29bfddf455fb909ce32ede0b8cddcb55e0bb794d Mon Sep 17 00:00:00 2001
From: Michael Migliore <michael.migliore@kitware.com>
Date: Fri, 2 Oct 2020 15:59:22 +0200
Subject: [PATCH] Add error when PROJ4 string is not valid

---
 Geovis/Core/vtkGeoProjection.cxx | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/Geovis/Core/vtkGeoProjection.cxx b/Geovis/Core/vtkGeoProjection.cxx
index d819a76948c..7ff6526a5d3 100644
--- a/Geovis/Core/vtkGeoProjection.cxx
+++ b/Geovis/Core/vtkGeoProjection.cxx
@@ -192,6 +192,10 @@ int vtkGeoProjection::UpdateProjection()
   if (this->PROJ4String && strlen(this->PROJ4String))
   {
     this->Projection = pj_init_plus(this->PROJ4String);
+    if (!this->Projection)
+    {
+      vtkErrorMacro("Cannot set projection with string " << this->PROJ4String);
+    }
   }
   else
   {
-- 
GitLab

