--- a/vtkm/thirdparty/diy/vtkmdiy/CMakeLists.txt
+++ b/vtkm/thirdparty/diy/vtkmdiy/CMakeLists.txt
@@ -141,7 +141,7 @@ function(add_diy_mpi_library use_mpi)
         set (has_mpi_val 0)
     endif()
 
-    add_library(${lib_name} ${sources})
+    add_library(${lib_name} STATIC ${sources})
     set_target_properties(${lib_name} PROPERTIES POSITION_INDEPENDENT_CODE ON)
     target_compile_features(${lib_name} PRIVATE cxx_std_14)
     target_compile_definitions(${lib_name}
