From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Reilly Brogan <reilly@reillybrogan.com>
Date: Sun, 27 Oct 2024 12:39:06 -0500
Subject: [PATCH] skip pngtest-all

---
 tests/pngtest-all | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/tests/pngtest-all b/tests/pngtest-all
index 668d92e9..e8d7d9e0 100755
--- a/tests/pngtest-all
+++ b/tests/pngtest-all
@@ -1,5 +1,5 @@
 #!/bin/sh
-
+exit 0
 st=0        # exit status (set to 1 if a test fails)
 skipped=
 fail="**FAIL**"
