From 008648bd83ee12578ece8bfa22921cf017fd270e Mon Sep 17 00:00:00 2001
From: Joey Riches <josephriches@gmail.com>
Date: Fri, 10 Nov 2023 16:37:57 +0000
Subject: [PATCH 1/1] asb-plugin-font: Add glob match for .ttc files

---
 libappstream-builder/plugins/asb-plugin-font.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/libappstream-builder/plugins/asb-plugin-font.c b/libappstream-builder/plugins/asb-plugin-font.c
index 2e64151..299e411 100644
--- a/libappstream-builder/plugins/asb-plugin-font.c
+++ b/libappstream-builder/plugins/asb-plugin-font.c
@@ -34,6 +34,7 @@ void
 asb_plugin_add_globs (AsbPlugin *plugin, GPtrArray *globs)
 {
 	asb_plugin_add_glob (globs, "/usr/share/fonts/*/*.otf");
+	asb_plugin_add_glob (globs, "/usr/share/fonts/*/*.ttc");
 	asb_plugin_add_glob (globs, "/usr/share/fonts/*/*.ttf");
 }
 
@@ -42,6 +43,8 @@ _asb_plugin_check_filename (const gchar *filename)
 {
 	if (asb_plugin_match_glob ("/usr/share/fonts/*/*.otf", filename))
 		return TRUE;
+	if (asb_plugin_match_glob ("/usr/share/fonts/*/*.ttc", filename))
+		return TRUE;
 	if (asb_plugin_match_glob ("/usr/share/fonts/*/*.ttf", filename))
 		return TRUE;
 	return FALSE;
-- 
2.42.0

