From 0e1a9e9e18b13f3b26f05d8404cde8a0fffa2910 Mon Sep 17 00:00:00 2001
From: Ikey Doherty <ikey@solus-project.com>
Date: Sun, 3 Jul 2016 16:50:44 +0100
Subject: [PATCH 1/2] gs-manager: Use the user's screensaver settings now

Signed-off-by: Ikey Doherty <ikey@solus-project.com>
---
 src/gs-manager.c | 18 ++----------------
 1 file changed, 2 insertions(+), 16 deletions(-)

diff --git a/src/gs-manager.c b/src/gs-manager.c
index ed3c1b9..ebb7a5a 100644
--- a/src/gs-manager.c
+++ b/src/gs-manager.c
@@ -570,23 +570,9 @@ static GSettings *
 get_system_settings (void)
 {
         GSettings *settings;
-        gchar **keys;
-        gchar **k;
 
-        /* FIXME: we need to bind system settings instead of user but
-         *        that's currently impossible, not implemented yet.
-         *        Hence, reset to system default values.
-         */
-        /* TODO: Ideally we would like to bind some other key, screensaver-specific. */
-        settings = g_settings_new ("org.gnome.desktop.background");
-
-        g_settings_delay (settings);
-
-        keys = g_settings_list_keys (settings);
-        for (k = keys; *k; k++) {
-                g_settings_reset (settings, *k);
-        }
-        g_strfreev (keys);
+        /* Use the screensaver preferences */
+        settings = g_settings_new ("org.gnome.desktop.screensaver");
 
         return settings;
 }
-- 
2.9.0

