From 022698cb8e72e4926ac7b720ea296239832b0846 Mon Sep 17 00:00:00 2001
From: Ikey Doherty <ikey@solus-project.com>
Date: Sun, 3 Jul 2016 17:22:01 +0100
Subject: [PATCH 2/3] Restore missing g_settings_delay

Signed-off-by: Ikey Doherty <ikey@solus-project.com>
---
 src/gs-manager.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/src/gs-manager.c b/src/gs-manager.c
index ebb7a5a..3b67735 100644
--- a/src/gs-manager.c
+++ b/src/gs-manager.c
@@ -574,6 +574,8 @@ get_system_settings (void)
         /* Use the screensaver preferences */
         settings = g_settings_new ("org.gnome.desktop.screensaver");
 
+        g_settings_delay (settings);
+
         return settings;
 }
 
-- 
2.9.0

