From 0ae0f4d0467fd7ca73c632be09a2680f4e817525 Mon Sep 17 00:00:00 2001
From: Ikey Doherty <ikey@solus-project.com>
Date: Sun, 3 Jul 2016 17:29:50 +0100
Subject: [PATCH 3/3] gs-manager: Ensure we actually reload the gnome-bg now

Signed-off-by: Ikey Doherty <ikey@solus-project.com>
---
 src/gs-manager.c | 7 +------
 1 file changed, 1 insertion(+), 6 deletions(-)

diff --git a/src/gs-manager.c b/src/gs-manager.c
index 3b67735..1b735e4 100644
--- a/src/gs-manager.c
+++ b/src/gs-manager.c
@@ -555,14 +555,9 @@ background_settings_change_event_cb (GSettings *settings,
                                      gint       n_keys,
                                      GSManager   *manager)
 {
-#if 0
-        /* FIXME: since we bind user settings instead of system ones,
-         *        watching for changes is no longer valid.
-         */
+        /* Reload gnome-bg due to lock-screen change */
         gnome_bg_load_from_preferences (manager->priv->bg,
                                         manager->priv->settings);
-#endif
-
         return FALSE;
 }
 
-- 
2.9.0

