From 5936cf5d149b36f7e00a3393629d1585254f9da9 Mon Sep 17 00:00:00 2001
From: Evan Maddock <maddock.evan@vivaldi.net>
Date: Sun, 28 Jan 2024 11:50:08 -0500
Subject: [PATCH] Fix grep error for /etc/shells on Solus

Signed-off-by: Evan Maddock <maddock.evan@vivaldi.net>
---
 gnome-session/budgie-session.in | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/gnome-session/budgie-session.in b/gnome-session/budgie-session.in
index 8a91e061..cbcf8f27 100755
--- a/gnome-session/budgie-session.in
+++ b/gnome-session/budgie-session.in
@@ -3,7 +3,7 @@
 if [ "x$XDG_SESSION_TYPE" = "xwayland" ] &&
    [ "x$XDG_SESSION_CLASS" != "xgreeter" ] &&
    [  -n "$SHELL" ] &&
-   grep -q "$SHELL" /etc/shells &&
+   grep -q "$SHELL" /usr/share/defaults/etc/shells &&
    ! (echo "$SHELL" | grep -q "false") &&
    ! (echo "$SHELL" | grep -q "nologin"); then
   if [ "$1" != '-l' ]; then
-- 
2.43.0

