From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Evan Maddock <maddock.evan@vivaldi.net>
Date: Mon, 24 Mar 2025 14:10:11 -0400
Subject: [PATCH] labwc: Set snapping range to 10

Signed-off-by: Evan Maddock <maddock.evan@vivaldi.net>
---
 labwc/labwc-rc.xml | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/labwc/labwc-rc.xml b/labwc/labwc-rc.xml
index 453597fb..c6d69acd 100644
--- a/labwc/labwc-rc.xml
+++ b/labwc/labwc-rc.xml
@@ -64,4 +64,8 @@
     </keybind>
   </keyboard>
 
+  <snapping>
+    <range>10</range>
+  </snapping>
+
 </labwc_config>
