From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Evan Maddock <maddock.evan@vivaldi.net>
Date: Thu, 27 Mar 2025 10:53:54 -0400
Subject: [PATCH] scripts: Only use dbus-run-session if address is unset or
 empty

Signed-off-by: Evan Maddock <maddock.evan@vivaldi.net>
---
 scripts/startxfce4.in | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/scripts/startxfce4.in b/scripts/startxfce4.in
index 45360b20..6711f347 100644
--- a/scripts/startxfce4.in
+++ b/scripts/startxfce4.in
@@ -161,7 +161,10 @@ then
     # Also: Xfce will use its own config directory and config file
     # to avoid conflict with current labwc setup and avoid launching 
     # anything from ~/.config/labwc/autostart
-    XFCE4_SESSION_COMPOSITOR="dbus-run-session -- ${OPTS:-labwc --config-dir ${XDG_CONFIG_HOME:-${HOME}/.config}/xfce4/labwc --config ${XDG_CONFIG_HOME:-${HOME}/.config}/xfce4/labwc/rc.xml $startup_option xfce4-session}"
+    XFCE4_SESSION_COMPOSITOR="${OPTS:-labwc --config-dir ${XDG_CONFIG_HOME:-${HOME}/.config}/xfce4/labwc --config ${XDG_CONFIG_HOME:-${HOME}/.config}/xfce4/labwc/rc.xml $startup_option xfce4-session}"
+    if [ -z "$DBUS_SESSION_BUS_ADDRESS" ]; then
+       XFCE4_SESSION_COMPOSITOR="dbus-run-session -- ${XFCE4_SESSION_COMPOSITOR}"
+    fi
   fi
 fi
 
