From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Evan Maddock <maddock.evan@vivaldi.net>
Date: Mon, 24 Mar 2025 14:23:37 -0400
Subject: [PATCH] labwc: Use Breeze cursors

Signed-off-by: Evan Maddock <maddock.evan@vivaldi.net>
---
 labwc/labwc-environment | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/labwc/labwc-environment b/labwc/labwc-environment
index cd449229..83d5424c 100644
--- a/labwc/labwc-environment
+++ b/labwc/labwc-environment
@@ -4,4 +4,4 @@
 XKB_DEFAULT_LAYOUT=en
 
 # Set cursor theme
-XCURSOR_THEME=Adwaita
+XCURSOR_THEME=breeze_cursors
