From 2d36b73c59f5a1899cd5cdb3f416e3e8640d3612 Mon Sep 17 00:00:00 2001
From: Evan Maddock <maddock.evan@vivaldi.net>
Date: Sat, 22 Apr 2023 11:23:45 -0400
Subject: [PATCH] Use new msgpack libname

Signed-off-by: Evan Maddock <maddock.evan@vivaldi.net>
---
 configure.ac | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/configure.ac b/configure.ac
index 028d5559..9bbe2ae8 100644
--- a/configure.ac
+++ b/configure.ac
@@ -201,7 +201,7 @@ fi
 
 PKG_CHECK_MODULES(
   MSGPACK,
-  msgpack >= 1.1.0,
+  msgpack-c >= 1.1.0,
   [
     CPPFLAGS="$MSGPACK_CFLAGS $CPPFLAGS"
     LIBS="$MSGPACK_LIBS $LIBS"
-- 
2.39.0

