--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -62,6 +62,7 @@ project(OpenVDB LANGUAGES CXX VERSION ${
 ###### OpenVDB Build/Component Options
 
 include(CMakeDependentOption)
+include(CMakePackageConfigHelpers)
 include(GNUInstallDirs)
 
 # todo epydoc and pdflatex
@@ -371,6 +372,12 @@ list(APPEND CMAKE_MODULE_PATH "${CMAKE_C
 # Add cmake modules to installation command
 # @todo fix our glew cmake module
 
+write_basic_package_version_file(
+    OpenVDBConfigVersion.cmake
+    VERSION ${PROJECT_VERSION}
+    COMPATIBILITY SameMajorVersion
+)
+
 if(OPENVDB_INSTALL_CMAKE_MODULES)
   set(OPENVDB_CMAKE_MODULES
     cmake/FindBlosc.cmake
@@ -384,6 +391,8 @@ if(OPENVDB_INSTALL_CMAKE_MODULES)
     cmake/OpenVDBHoudiniSetup.cmake
     cmake/OpenVDBMayaSetup.cmake
     cmake/OpenVDBUtils.cmake
+    cmake/OpenVDBConfig.cmake
+    ${PROJECT_BINARY_DIR}/OpenVDBConfigVersion.cmake
   )
   install(FILES ${OPENVDB_CMAKE_MODULES} DESTINATION ${CMAKE_INSTALL_LIBDIR}/cmake/OpenVDB)
 endif()
--- a/cmake/FindOpenVDB.cmake
+++ b/cmake/FindOpenVDB.cmake
@@ -104,7 +104,9 @@ may be provided to tell this module wher
 
 #]=======================================================================]
 
-cmake_minimum_required(VERSION 3.15)
+cmake_policy(PUSH)
+cmake_policy(VERSION 3.15)
+
 include(GNUInstallDirs)
 
 
@@ -788,3 +790,5 @@ endif()
 
 unset(_OPENVDB_VISIBLE_DEPENDENCIES)
 unset(_OPENVDB_HIDDEN_DEPENDENCIES)
+
+cmake_policy(POP)
--- a/cmake/OpenVDBConfig.cmake
+++ b/cmake/OpenVDBConfig.cmake
@@ -3,8 +3,8 @@
 #
 #[=======================================================================[.rst:
 
-FindOpenVDB
------------
+OpenVDBConfig
+-------------
 
 Find OpenVDB include dirs, libraries and settings
 
@@ -104,7 +104,9 @@ may be provided to tell this module wher
 
 #]=======================================================================]
 
-cmake_minimum_required(VERSION 3.15)
+cmake_policy(PUSH)
+cmake_policy(VERSION 3.15)
+
 include(GNUInstallDirs)
 
 
@@ -179,12 +181,16 @@ endif()
   pkg_check_modules(PC_OpenVDB QUIET OpenVDB)
 endif()
 
+## Stash input CMAKE_MODULE_PATH
+set(OPENVDB_ORIGINAL_CMAKE_MODULE_PATH ${CMAKE_MODULE_PATH})
+set(CMAKE_MODULE_PATH ${CMAKE_MODULE_PATH} ${CMAKE_CURRENT_LIST_DIR})
+
 # This CMake module supports being called from external packages AND from
 # within the OpenVDB repository for building openvdb components with the
 # core library build disabled. Determine where we are being called from:
 #
-# (repo structure = <root>/cmake/FindOpenVDB.cmake)
-# (inst structure = <root>/lib/cmake/OpenVDB/FindOpenVDB.cmake)
+# (repo structure = <root>/cmake/OpenVDBConfig.cmake)
+# (inst structure = <root>/lib/cmake/OpenVDB/OpenVDBConfig.cmake)
 
 get_filename_component(_DIR_NAME ${CMAKE_CURRENT_LIST_DIR} NAME)
 
@@ -194,7 +196,7 @@ elseif(${_DIR_NAME} STREQUAL "OpenVDB")
   # Set the install variable to track directories if this is being called from
   # an installed location and from another package. The expected installation
   # directory structure is:
-  #  <root>/lib/cmake/OpenVDB/FindOpenVDB.cmake
+  #  <root>/lib/cmake/OpenVDB/OpenVDBConfig.cmake
   #  <root>/include
   #  <root>/bin
   get_filename_component(_IMPORT_PREFIX ${CMAKE_CURRENT_LIST_DIR} DIRECTORY)
@@ -786,5 +788,10 @@ endif()
   endif()
 endif()
 
+## Reset CMAKE_MODULE_PATH
+set(CMAKE_MODULE_PATH ${OPENVDB_ORIGINAL_CMAKE_MODULE_PATH})
+
 unset(_OPENVDB_VISIBLE_DEPENDENCIES)
 unset(_OPENVDB_HIDDEN_DEPENDENCIES)
+
+cmake_policy(POP)
