From ff96d54d3f320e5c22111cd95ae257a7a05c9035 Mon Sep 17 00:00:00 2001
From: Damien Pollet <damien.pollet@gmail.com>
Date: Thu, 26 Sep 2024 13:53:35 +0200
Subject: [PATCH] Allow using optimist 3.1

---
 lolcat.gemspec | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/lolcat.gemspec b/lolcat.gemspec
index 633ba3f..929366b 100644
--- a/lolcat.gemspec
+++ b/lolcat.gemspec
@@ -13,7 +13,7 @@ Gem::Specification.new do |s|
 
   s.add_development_dependency "rake"
   s.add_dependency "paint", "~> 2.1"
-  s.add_dependency "optimist", "~> 3.0.0"
+  s.add_dependency "optimist", "~> 3.0"
   s.add_dependency "manpages", "~> 0.6.1"
 
   s.files         = `git ls-files`.split("\n")