From 3861e78c3e50a521cbdb9699c67f0d678bd2fee2 Mon Sep 17 00:00:00 2001
From: Ikey Doherty <ikey@solusos.com>
Date: Mon, 9 Sep 2013 19:47:21 +0100
Subject: [PATCH] Allow Solus to remain named "Solus"

---
 util/grub.d/10_linux.in | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/util/grub.d/10_linux.in b/util/grub.d/10_linux.in
index 14402e8..32c575d 100644
--- a/util/grub.d/10_linux.in
+++ b/util/grub.d/10_linux.in
@@ -30,6 +30,9 @@ CLASS="--class gnu-linux --class gnu --class os"
 
 if [ "x${GRUB_DISTRIBUTOR}" = "x" ] ; then
   OS=GNU/Linux
+elif [ "x${GRUB_DISTRIBUTOR}" = "xSolus" ] ; then
+  OS="${GRUB_DISTRIBUTOR}"
+  CLASS="--class $(echo ${GRUB_DISTRIBUTOR} | tr 'A-Z' 'a-z' | cut -d' ' -f1) ${CLASS}"
 else
   OS="${GRUB_DISTRIBUTOR} GNU/Linux"
   CLASS="--class $(echo ${GRUB_DISTRIBUTOR} | tr 'A-Z' 'a-z' | cut -d' ' -f1) ${CLASS}"
-- 
1.8.3.2

