From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Reilly Brogan <reilly@reillybrogan.com>
Date: Mon, 30 Mar 2026 14:47:38 -0500
Subject: [PATCH] typer only

---
 pyproject.toml | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/pyproject.toml b/pyproject.toml
index b4e0a95..f9edd74 100644
--- a/pyproject.toml
+++ b/pyproject.toml
@@ -26,7 +26,7 @@ dependencies = [
 	"jaraco.context >= 4.1",
 	"jaraco.functools",
 	"more_itertools",
-	"typer-slim",
+	"typer",
 ]
 dynamic = ["version"]
 
