From 7d43e0317b4da16dd2c6a63487b45922ece438be Mon Sep 17 00:00:00 2001
From: Ikey Doherty <ikey@solus-project.com>
Date: Wed, 3 Aug 2016 02:17:25 +0100
Subject: [PATCH 1/2] stdlib: Add missing curly braves for GCC6

Signed-off-by: Ikey Doherty <ikey@solus-project.com>
---
 stdlib/setenv.c | 5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

diff --git a/stdlib/setenv.c b/stdlib/setenv.c
index da61ee0..bfd8fec 100644
--- a/stdlib/setenv.c
+++ b/stdlib/setenv.c
@@ -276,7 +276,7 @@ unsetenv (const char *name)
   LOCK;
 
   ep = __environ;
-  if (ep != NULL)
+  if (ep != NULL) {
     while (*ep != NULL)
       if (!strncmp (*ep, name, len) && (*ep)[len] == '=')
 	{
@@ -288,8 +288,9 @@ unsetenv (const char *name)
 	  while (*dp++);
 	  /* Continue the loop in case NAME appears again.  */
 	}
-      else
+  } else {
 	++ep;
+  }
 
   UNLOCK;
 
-- 
2.9.2

