From c6561fb843be1a1e71b3cff49a926bbfd57979d4 Mon Sep 17 00:00:00 2001
From: Ikey Doherty <ikey@solus-project.com>
Date: Mon, 14 Dec 2015 13:12:22 +0000
Subject: [PATCH] ldd: Force correct RTLDLIST for Solus

Signed-off-by: Ikey Doherty <ikey@solus-project.com>
---
 elf/ldd.bash.in | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/elf/ldd.bash.in b/elf/ldd.bash.in
index 7fb6c15..56eef6f 100644
--- a/elf/ldd.bash.in
+++ b/elf/ldd.bash.in
@@ -26,7 +26,7 @@
 TEXTDOMAIN=libc
 TEXTDOMAINDIR=@TEXTDOMAINDIR@
 
-RTLDLIST=@RTLD@
+RTLDLIST="/usr/lib32/ld-linux.so.2 /usr/lib64/ld-linux-x86-64.so.2 /usr/libx32/ld-linux-x32.so.2"
 warn=
 bind_now=
 verbose=
-- 
2.6.4

