From 93eadc95b70577786fb9e5e5407d3aa08730f8eb Mon Sep 17 00:00:00 2001
From: Ikey Doherty <ikey@solus-project.com>
Date: Wed, 3 Aug 2016 02:25:44 +0100
Subject: [PATCH 2/2] nis: Fix use of braces with GCC6

Signed-off-by: Ikey Doherty <ikey@solus-project.com>
---
 nis/nis_call.c | 7 +++++--
 1 file changed, 5 insertions(+), 2 deletions(-)

diff --git a/nis/nis_call.c b/nis/nis_call.c
index 3fa37e4..6894af3 100644
--- a/nis/nis_call.c
+++ b/nis/nis_call.c
@@ -679,7 +679,7 @@ nis_server_cache_add (const_nis_name name, int search_parent,
 
   /* Choose which entry should be evicted from the cache.  */
   loc = &nis_server_cache[0];
-  if (*loc != NULL)
+  if (*loc != NULL) {
     for (i = 1; i < 16; ++i)
       if (nis_server_cache[i] == NULL)
 	{
@@ -689,7 +689,10 @@ nis_server_cache_add (const_nis_name name, int search_parent,
       else if ((*loc)->uses > nis_server_cache[i]->uses
 	       || ((*loc)->uses == nis_server_cache[i]->uses
 		   && (*loc)->expires > nis_server_cache[i]->expires))
-	loc = &nis_server_cache[i];
+        {
+            loc = &nis_server_cache[i];
+        }
+  }
   old = *loc;
   *loc = new;
 
-- 
2.9.2

