diff --git a/nis/nis_call.c b/nis/nis_call.c
index 3fa37e4..bd89fe5 100644
--- a/nis/nis_call.c
+++ b/nis/nis_call.c
@@ -679,13 +679,14 @@ nis_server_cache_add (const_nis_name name, int search_parent,
 
   /* Choose which entry should be evicted from the cache.  */
   loc = &nis_server_cache[0];
-  if (*loc != NULL)
+  if (*loc != NULL) {
     for (i = 1; i < 16; ++i)
       if (nis_server_cache[i] == NULL)
 	{
 	  loc = &nis_server_cache[i];
 	  break;
 	}
+    }
       else if ((*loc)->uses > nis_server_cache[i]->uses
 	       || ((*loc)->uses == nis_server_cache[i]->uses
 		   && (*loc)->expires > nis_server_cache[i]->expires))
diff --git a/stdlib/setenv.c b/stdlib/setenv.c
index da61ee0..50a8642 100644
--- a/stdlib/setenv.c
+++ b/stdlib/setenv.c
@@ -276,7 +276,7 @@ unsetenv (const char *name)
   LOCK;
 
   ep = __environ;
-  if (ep != NULL)
+  if (ep != NULL) {
     while (*ep != NULL)
       if (!strncmp (*ep, name, len) && (*ep)[len] == '=')
 	{
@@ -290,7 +290,7 @@ unsetenv (const char *name)
 	}
       else
 	++ep;
-
+  }
   UNLOCK;
 
   return 0;
