From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Thomas Staudinger <Staudi.Kaos@gmail.com>
Date: Fri, 28 Jun 2024 11:44:02 +0200
Subject: [PATCH] Update paths in service file

Signed-off-by: Thomas Staudinger <Staudi.Kaos@gmail.com>
---
 parser/apparmor.service | 12 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/parser/apparmor.service b/parser/apparmor.service
index 15b9f74..752eda1 100644
--- a/parser/apparmor.service
+++ b/parser/apparmor.service
@@ -9,8 +9,8 @@ ConditionSecurity=apparmor
 
 [Service]
 Type=oneshot
-ExecStart=/lib/apparmor/apparmor.systemd reload
-ExecReload=/lib/apparmor/apparmor.systemd reload
+ExecStart=/usr/lib64/apparmor/apparmor.systemd reload
+ExecReload=/usr/lib64/apparmor/apparmor.systemd reload
 
 # systemd maps 'restart' to 'stop; start' which means removing AppArmor confinement
 # from running processes (and not being able to re-apply it later).
