commit fea70d8cab64009ded2d47b5a64b6201b052a5e9
Author: Maik Wöhl <maik.woehl@outlook.de>
Date:   Sun Sep 10 18:17:31 2023 +0200

    MakeEnable xvfb for solus

diff --git a/fs/etc/xpra/conf.d/55_server_x11.conf.in b/fs/etc/xpra/conf.d/55_server_x11.conf.in
index 90c52de8d..0bb4bba62 100644
--- a/fs/etc/xpra/conf.d/55_server_x11.conf.in
+++ b/fs/etc/xpra/conf.d/55_server_x11.conf.in
@@ -49,4 +49,7 @@ fake-xinerama = %(fake_xinerama)s
 #xvfb = %(xdummy_command)s
 #
 # Selecting virtual X server:
-xvfb = %(xvfb_command)s
+xvfb = Xvfb -nolisten tcp -noreset \
+        +extension GLX +extension Composite \
+        -auth $XAUTHORITY \
+        -screen 0 8192x4096x24+32
