diff --git a/fs/etc/xpra/conf.d/55_server_x11.conf.in b/fs/etc/xpra/conf.d/55_server_x11.conf.in
index 569df6332..884faa47d 100644
--- a/fs/etc/xpra/conf.d/55_server_x11.conf.in
+++ b/fs/etc/xpra/conf.d/55_server_x11.conf.in
@@ -52,4 +52,7 @@ fake-xinerama = %(fake_xinerama)s
 #        -config %(conf_dir)s/xorg.conf
 #
 # Selecting virtual X server:
-xvfb = %(xvfb_command)s
+xvfb = Xvfb -nolisten tcp -noreset \
+        +extension GLX +extension Composite \
+        -auth $XAUTHORITY \
+        -screen 0 8192x4096x24+32
