diff --git a/setup.py b/setup.py
index 0177c1109..5d4ebacd4 100755
--- a/setup.py
+++ b/setup.py
@@ -811,8 +811,12 @@ def get_base_conf_dir(install_dir, stripbuildroot=True):
 
 def get_conf_dir(install_dir, stripbuildroot=True):
     dirs = get_base_conf_dir(install_dir, stripbuildroot)
-    if "etc" not in dirs:
-        dirs.append("etc")
+    if "usr" not in dirs:
+        dirs.append("usr")
+    if "share" not in dirs:
+        dirs.append("share")
+    if "defaults" not in dirs:
+        dirs.append("defaults")
     dirs.append("xpra")
     return os.path.join(*dirs)
 
@@ -1626,13 +1630,13 @@ else:
                 if nvfbc_ENABLED:
                     etc_xpra_files.append("nvfbc.keys")
                 for x in etc_xpra_files:
-                    copytodir("fs/etc/xpra/%s" % x, "/etc/xpra")
-                copytodir("fs/etc/X11/xorg.conf.d/90-xpra-virtual.conf", "/etc/X11/xorg.conf.d/")
+                    copytodir("fs/etc/xpra/%s" % x, "/usr/share/defaults/xpra")
+                copytodir("fs/etc/X11/xorg.conf.d/90-xpra-virtual.conf", "/usr/share/X11/xorg.conf.d/")
 
             if pam_ENABLED:
-                copytodir("fs/etc/pam.d/xpra", "/etc/pam.d")
+                copytodir("fs/etc/pam.d/xpra", "/usr/share/defaults/etc/pam.d")
 
-            systemd_dir = "/lib/systemd/system"
+            systemd_dir = "/usr/lib64/systemd/system"
             if service_ENABLED:
                 #Linux init service:
                 subs = {}
@@ -1644,7 +1648,7 @@ else:
                     cdir = "/etc/sysconfig"
                 else:
                     cdir = "/etc/default"
-                copytodir("fs/etc/sysconfig/xpra", cdir)
+                copytodir("fs/etc/sysconfig/xpra.env", "/usr/share/defaults/xpra")
                 if cdir!="/etc/sysconfig":
                     #also replace the reference to it in the service file below
                     subs[b"/etc/sysconfig"] = cdir.encode()
@@ -1660,7 +1664,7 @@ else:
             if sd_listen_ENABLED:
                 copytodir("fs/lib/systemd/system/xpra.socket", systemd_dir)
             if dbus_ENABLED and proxy_ENABLED:
-                copytodir("fs/etc/dbus-1/system.d/xpra.conf", "/etc/dbus-1/system.d")
+                copytodir("fs/etc/dbus-1/system.d/xpra.conf", "/usr/share/dbus-1/system.d")
 
             if docs_ENABLED:
                 if WIN32 or OSX:
@@ -1672,7 +1676,7 @@ else:
 
             if data_ENABLED:
                 for d in ("http-headers", "content-type", "content-categories"):
-                    dirtodir("fs/etc/xpra/%s" % d, "/etc/xpra/%s" % d)
+                    dirtodir("fs/etc/xpra/%s" % d, "/usr/share/defaults/xpra/%s" % d)
 
     # add build_conf to build step
     cmdclass.update({
