diff --git a/setup.py b/setup.py
index 029abc94f..c9bc4f03f 100755
--- a/setup.py
+++ b/setup.py
@@ -1615,12 +1615,12 @@ else:
                     etc_xpra_files.append("nvfbc.keys")
                 for x in etc_xpra_files:
                     copytodir("fs/etc/xpra/%s" % x, "/etc/xpra")
-                copytodir("fs/etc/X11/xorg.conf.d/90-xpra-virtual.conf", "/etc/X11/xorg.conf.d/")
+                copytodir("fs/etc/X11/xorg.conf.d/90-xpra-virtual.conf", "/usr/share/X11/xorg.conf.d/")
 
             if pam_ENABLED:
                 copytodir("fs/etc/pam.d/xpra", "/etc/pam.d")
 
-            systemd_dir = "/lib/systemd/system"
+            systemd_dir = "/usr/lib64/systemd/system"
             if service_ENABLED:
                 #Linux init service:
                 subs = {}
@@ -1632,7 +1632,7 @@ else:
                     cdir = "/etc/sysconfig"
                 else:
                     cdir = "/etc/default"
-                copytodir("fs/etc/sysconfig/xpra", cdir)
+                copytodir("fs/etc/sysconfig/xpra", "/usr/share/defaults")
                 if cdir!="/etc/sysconfig":
                     #also replace the reference to it in the service file below
                     subs[b"/etc/sysconfig"] = cdir.encode()
@@ -1648,7 +1648,7 @@ else:
             if sd_listen_ENABLED:
                 copytodir("fs/lib/systemd/system/xpra.socket", systemd_dir)
             if dbus_ENABLED and proxy_ENABLED:
-                copytodir("fs/etc/dbus-1/system.d/xpra.conf", "/etc/dbus-1/system.d")
+                copytodir("fs/etc/dbus-1/system.d/xpra.conf", "/usr/share/dbus-1/system.d")
 
             if docs_ENABLED:
                 if WIN32 or OSX:
