commit 2cea2ea3727cb471d15ec4d6b97bbe0b3ef7a373
Author: Maik Wöhl <maik.woehl@outlook.de>
Date:   Sun Sep 10 18:07:30 2023 +0200

    Rewire install paths

diff --git a/setup.py b/setup.py
index e254bf99c..9330425ad 100755
--- a/setup.py
+++ b/setup.py
@@ -1003,8 +1003,12 @@ def get_base_conf_dir(install_dir, stripbuildroot=True):
 
 def get_conf_dir(install_dir, stripbuildroot=True):
     dirs = get_base_conf_dir(install_dir, stripbuildroot)
-    if "etc" not in dirs:
-        dirs.append("etc")
+    if "usr" not in dirs:
+        dirs.append("usr")
+    if "share" not in dirs:
+        dirs.append("share")
+    if "defaults" not in dirs:
+        dirs.append("defaults")
     dirs.append("xpra")
     return os.path.join(*dirs)
 
@@ -1730,7 +1734,7 @@ if WIN32:
     if data_ENABLED:
         add_data_files("share/metainfo",      ["fs/share/metainfo/xpra.appdata.xml"])
         for d in ("http-headers", "content-type", "content-categories", "content-parent"):
-            add_data_files(f"etc/xpra/{d}", glob.glob(f"fs/etc/xpra/{d}/*"))
+            add_data_files(f"usr/share/defaults/xpra/{d}", glob.glob(f"fs/etc/xpra/{d}/*"))
 
     #END OF win32
 #*******************************************************************************
@@ -1873,13 +1877,13 @@ else:
                 else:
                     addconf("xorg.conf")
                 for src, dst_name in etc_xpra_files.items():
-                    copytodir(f"fs/etc/xpra/{src}", "/etc/xpra", dst_name=dst_name)
-                copytodir("fs/etc/X11/xorg.conf.d/90-xpra-virtual.conf", "/etc/X11/xorg.conf.d/")
+                    copytodir(f"fs/etc/xpra/{src}", "/usr/share/defaults/xpra", dst_name=dst_name)
+                copytodir("fs/etc/X11/xorg.conf.d/90-xpra-virtual.conf", "/usr/share/X11/xorg.conf.d/")
 
             if pam_ENABLED:
-                copytodir("fs/etc/pam.d/xpra", "/etc/pam.d")
+                copytodir("fs/etc/pam.d/xpra", "/usr/share/defaults/pam.d")
 
-            systemd_dir = "/lib/systemd/system"
+            systemd_dir = "/usr/lib64/systemd/system"
             if is_openSUSE():
                 systemd_dir = "__UNITDIR__"
             if service_ENABLED:
@@ -1889,10 +1893,8 @@ else:
                     cdir = "/etc/sysconfig"
                 elif is_Debian() or is_Ubuntu():
                     cdir = "/etc/default"
-                elif os.path.exists("/etc/sysconfig"):
-                    cdir = "/etc/sysconfig"
                 else:
-                    cdir = "/etc/default"
+                    cdir = "/usr/share/defaults/xpra"
                 if is_openSUSE():
                     #openSUSE does things differently:
                     cdir = "__FILLUPDIR__"
@@ -1900,7 +1902,7 @@ else:
                     os.chmod("fs/etc/sysconfig/sysconfig.xpra", 0o644)
                     copytodir("fs/etc/sysconfig/sysconfig.xpra", cdir)
                 else:
-                    copytodir("fs/etc/sysconfig/xpra", cdir)
+                    copytodir("fs/etc/sysconfig/xpra.env", cdir)
                 if cdir!="/etc/sysconfig":
                     #also replace the reference to it in the service file below
                     subs[b"/etc/sysconfig"] = cdir.encode()
@@ -1916,7 +1918,7 @@ else:
             if sd_listen_ENABLED:
                 copytodir("fs/lib/systemd/system/xpra.socket", systemd_dir)
             if dbus_ENABLED and proxy_ENABLED:
-                copytodir("fs/etc/dbus-1/system.d/xpra.conf", "/etc/dbus-1/system.d")
+                copytodir("fs/etc/dbus-1/system.d/xpra.conf", "/usr/share/dbus-1/system.d")
 
             if docs_ENABLED:
                 doc_dir = f"{install_dir}/share/doc/xpra/"
@@ -1924,7 +1926,7 @@ else:
 
             if data_ENABLED:
                 for etc_dir in ("http-headers", "content-type", "content-categories", "content-parent"):
-                    dirtodir(f"fs/etc/xpra/{etc_dir}", f"/etc/xpra/{etc_dir}")
+                    dirtodir(f"fs/etc/xpra/{etc_dir}", f"/usr/share/defaults/xpra/{etc_dir}")
 
     # add build_conf to build step
     cmdclass.update({
