commit 2d72af8f1e472eaf50e2182ef1cbcb7eb4152579
Author: Maik Wöhl <maik.woehl@outlook.de>
Date:   Sun Sep 10 18:15:02 2023 +0200

    Update config documentation

diff --git a/xpra/scripts/config.py b/xpra/scripts/config.py
index 32c49230e..835b6cf59 100755
--- a/xpra/scripts/config.py
+++ b/xpra/scripts/config.py
@@ -432,6 +432,7 @@ def get_xpra_defaults_dirs(username:Optional[str]=None, uid=None, gid=None):
     # load config files in this order (the later ones override earlier ones):
     # * application defaults   (ie: "/Volumes/Xpra/Xpra.app/Contents/Resources/" on OSX)
     #                          (ie: "C:\Program Files\Xpra\" on win32)
+    #                          (ie: "/usr/share/defaults/xpra" on solus)
     #                          (ie: None on others)
     # * system defaults        (ie: "/etc/xpra" on Posix - not on OSX)
     #                          (ie: "/Library/Application Support/Xpra" on OSX)
