diff --git a/xpra/scripts/config.py b/xpra/scripts/config.py
index 089b5ce1f..b38951384 100755
--- a/xpra/scripts/config.py
+++ b/xpra/scripts/config.py
@@ -427,6 +427,7 @@ def get_xpra_defaults_dirs(username=None, uid=None, gid=None):
     # load config files in this order (the later ones override earlier ones):
     # * application defaults   (ie: "/Volumes/Xpra/Xpra.app/Contents/Resources/" on OSX)
     #                          (ie: "C:\Program Files\Xpra\" on win32)
+    #                          (ie: "/usr/share/defaults/xpra" on solus)
     #                          (ie: None on others)
     # * system defaults        (ie: "/etc/xpra" on Posix - not on OSX)
     #                          (ie: "/Library/Application Support/Xpra" on OSX)
