From 1a1faefe2506264a44cdc6fae455ab99f7b6768c Mon Sep 17 00:00:00 2001
From: Algent Albrahimi <algent@protonmail.com>
Date: Sun, 10 Nov 2019 21:29:21 +0100
Subject: [PATCH] Fix gprename Makefile

---
 Makefile | 120 +++++++++++++++++++++++++++----------------------------
 1 file changed, 59 insertions(+), 61 deletions(-)

diff --git a/Makefile b/Makefile
index 34bdb19..46e4893 100644
--- a/Makefile
+++ b/Makefile
@@ -44,72 +44,70 @@ install: build
 	msgfmt -o build/locale/zh_CN.mo	locale/zh_CN.po
 
 	# Create directories
-	install -d "$(DESTDIR)"
-	install -d "$(DESTDIR)/bin"
-	install -d "/usr/share"
-	install -d "/usr/share/icons"
-	install -d "$(DESTDIR)/share/applications"
-	install -d "$(DESTDIR)/share/man"
-	install -d "$(DESTDIR)/share/man/man1"
-	install -d "$(DESTDIR)/share/locale/ca/LC_MESSAGES"
-	install -d "$(DESTDIR)/share/locale/de/LC_MESSAGES"
-	install -d "$(DESTDIR)/share/locale/es/LC_MESSAGES"
-	install -d "$(DESTDIR)/share/locale/et/LC_MESSAGES"
-	install -d "$(DESTDIR)/share/locale/fr/LC_MESSAGES"
-	install -d "$(DESTDIR)/share/locale/id/LC_MESSAGES"
-	install -d "$(DESTDIR)/share/locale/it/LC_MESSAGES"
-	install -d "$(DESTDIR)/share/locale/nl/LC_MESSAGES"
-	install -d "$(DESTDIR)/share/locale/pl/LC_MESSAGES"
-	install -d "$(DESTDIR)/share/locale/pt_BR/LC_MESSAGES"
-	install -d "$(DESTDIR)/share/locale/ro/LC_MESSAGES"
-	install -d "$(DESTDIR)/share/locale/ru/LC_MESSAGES"
-	install -d "$(DESTDIR)/share/locale/sv/LC_MESSAGES"
-	install -d "$(DESTDIR)/share/locale/zh_CN/LC_MESSAGES"
+	install -d "$(DESTDIR)$(PREFIX)"
+	install -d "$(DESTDIR)$(PREFIX)/bin"
+	install -d "$(DESTDIR)/usr/share"
+	install -d "$(DESTDIR)/usr/share/icons"
+	install -d "$(DESTDIR)/usr/share/applications"
+	install -d "$(DESTDIR)$(PREFIX)/share/man"
+	install -d "$(DESTDIR)$(PREFIX)/share/man/man1"
+	install -d "$(DESTDIR)$(PREFIX)/share/locale/ca/LC_MESSAGES"
+	install -d "$(DESTDIR)$(PREFIX)/share/locale/de/LC_MESSAGES"
+	install -d "$(DESTDIR)$(PREFIX)/share/locale/es/LC_MESSAGES"
+	install -d "$(DESTDIR)$(PREFIX)/share/locale/et/LC_MESSAGES"
+	install -d "$(DESTDIR)$(PREFIX)/share/locale/fr/LC_MESSAGES"
+	install -d "$(DESTDIR)$(PREFIX)/share/locale/id/LC_MESSAGES"
+	install -d "$(DESTDIR)$(PREFIX)/share/locale/it/LC_MESSAGES"
+	install -d "$(DESTDIR)$(PREFIX)/share/locale/nl/LC_MESSAGES"
+	install -d "$(DESTDIR)$(PREFIX)/share/locale/pl/LC_MESSAGES"
+	install -d "$(DESTDIR)$(PREFIX)/share/locale/pt_BR/LC_MESSAGES"
+	install -d "$(DESTDIR)$(PREFIX)/share/locale/ro/LC_MESSAGES"
+	install -d "$(DESTDIR)$(PREFIX)/share/locale/ru/LC_MESSAGES"
+	install -d "$(DESTDIR)$(PREFIX)/share/locale/sv/LC_MESSAGES"
+	install -d "$(DESTDIR)$(PREFIX)/share/locale/zh_CN/LC_MESSAGES"
 
 	# Copy all files in the filesystem
-	install -m 755 build/gprename                "$(DESTDIR)/bin/"
-	install -m 644 gprename-nautilus-actions.xml "$(DESTDIR)/share/applications/"
-	install -m 644 icon/gprename.png             "/usr/share/icons/"
-	install -m 644 man/gprename.1                "$(DESTDIR)/share/man/man1/"
-	install -m 644 build/locale/ca.mo            "$(DESTDIR)/share/locale/ca/LC_MESSAGES/gprename.mo"
-	install -m 644 build/locale/de.mo            "$(DESTDIR)/share/locale/de/LC_MESSAGES/gprename.mo"
-	install -m 644 build/locale/es.mo            "$(DESTDIR)/share/locale/es/LC_MESSAGES/gprename.mo"
-	install -m 644 build/locale/et.mo            "$(DESTDIR)/share/locale/et/LC_MESSAGES/gprename.mo"
-	install -m 644 build/locale/fr.mo            "$(DESTDIR)/share/locale/fr/LC_MESSAGES/gprename.mo"
-	install -m 644 build/locale/id.mo            "$(DESTDIR)/share/locale/id/LC_MESSAGES/gprename.mo"
-	install -m 644 build/locale/it.mo            "$(DESTDIR)/share/locale/it/LC_MESSAGES/gprename.mo"
-	install -m 644 build/locale/nl.mo            "$(DESTDIR)/share/locale/nl/LC_MESSAGES/gprename.mo"
-	install -m 644 build/locale/pl.mo            "$(DESTDIR)/share/locale/pl/LC_MESSAGES/gprename.mo"
-	install -m 644 build/locale/pt_BR.mo         "$(DESTDIR)/share/locale/pt_BR/LC_MESSAGES/gprename.mo"
-	install -m 644 build/locale/ro.mo            "$(DESTDIR)/share/locale/ro/LC_MESSAGES/gprename.mo"
-	install -m 644 build/locale/ru.mo            "$(DESTDIR)/share/locale/ru/LC_MESSAGES/gprename.mo"
-	install -m 644 build/locale/sv.mo            "$(DESTDIR)/share/locale/sv/LC_MESSAGES/gprename.mo"
-	install -m 644 build/locale/zh_CN.mo         "$(DESTDIR)/share/locale/zh_CN/LC_MESSAGES/gprename.mo"
-	desktop-file-install bin/gprename.desktop
-	update-desktop-database
+	install -m 755 build/gprename                "$(DESTDIR)$(PREFIX)/bin/"
+	install -m 644 gprename-nautilus-actions.xml "$(DESTDIR)/usr/share/applications/"
+	install -m 644 icon/gprename.png             "$(DESTDIR)/usr/share/icons/"
+	install -m 644 man/gprename.1                "$(DESTDIR)$(PREFIX)/share/man/man1/"
+	install -m 644 build/locale/ca.mo            "$(DESTDIR)$(PREFIX)/share/locale/ca/LC_MESSAGES/gprename.mo"
+	install -m 644 build/locale/de.mo            "$(DESTDIR)$(PREFIX)/share/locale/de/LC_MESSAGES/gprename.mo"
+	install -m 644 build/locale/es.mo            "$(DESTDIR)$(PREFIX)/share/locale/es/LC_MESSAGES/gprename.mo"
+	install -m 644 build/locale/et.mo            "$(DESTDIR)$(PREFIX)/share/locale/et/LC_MESSAGES/gprename.mo"
+	install -m 644 build/locale/fr.mo            "$(DESTDIR)$(PREFIX)/share/locale/fr/LC_MESSAGES/gprename.mo"
+	install -m 644 build/locale/id.mo            "$(DESTDIR)$(PREFIX)/share/locale/id/LC_MESSAGES/gprename.mo"
+	install -m 644 build/locale/it.mo            "$(DESTDIR)$(PREFIX)/share/locale/it/LC_MESSAGES/gprename.mo"
+	install -m 644 build/locale/nl.mo            "$(DESTDIR)$(PREFIX)/share/locale/nl/LC_MESSAGES/gprename.mo"
+	install -m 644 build/locale/pl.mo            "$(DESTDIR)$(PREFIX)/share/locale/pl/LC_MESSAGES/gprename.mo"
+	install -m 644 build/locale/pt_BR.mo         "$(DESTDIR)$(PREFIX)/share/locale/pt_BR/LC_MESSAGES/gprename.mo"
+	install -m 644 build/locale/ro.mo            "$(DESTDIR)$(PREFIX)/share/locale/ro/LC_MESSAGES/gprename.mo"
+	install -m 644 build/locale/ru.mo            "$(DESTDIR)$(PREFIX)/share/locale/ru/LC_MESSAGES/gprename.mo"
+	install -m 644 build/locale/sv.mo            "$(DESTDIR)$(PREFIX)/share/locale/sv/LC_MESSAGES/gprename.mo"
+	install -m 644 build/locale/zh_CN.mo         "$(DESTDIR)$(PREFIX)/share/locale/zh_CN/LC_MESSAGES/gprename.mo"
+	install -m 644 bin/gprename.desktop          "$(DESTDIR)/usr/share/applications/gprename.desktop"
 	@echo "Installation completed."
 
 uninstall: clean
-	rm -f  "$(DESTDIR)/bin/gprename"
-	rm -f  "/usr/share/applications/gprename.desktop"
-	rm -f  "$(DESTDIR)/share/applications/gprename-nautilus-actions.xml"
-	rm -f  "$(DESTDIR)/share/man/man1/gprename.1"
-	rm -f  "$(DESTDIR)/share/locale/ca/LC_MESSAGES/gprename.mo"
-	rm -f  "$(DESTDIR)/share/locale/de/LC_MESSAGES/gprename.mo"
-	rm -f  "$(DESTDIR)/share/locale/es/LC_MESSAGES/gprename.mo"
-	rm -f  "$(DESTDIR)/share/locale/et/LC_MESSAGES/gprename.mo"
-	rm -f  "$(DESTDIR)/share/locale/fr/LC_MESSAGES/gprename.mo"
-	rm -f  "$(DESTDIR)/share/locale/id/LC_MESSAGES/gprename.mo"
-	rm -f  "$(DESTDIR)/share/locale/it/LC_MESSAGES/gprename.mo"
-	rm -f  "$(DESTDIR)/share/locale/nl/LC_MESSAGES/gprename.mo"
-	rm -f  "$(DESTDIR)/share/locale/pl/LC_MESSAGES/gprename.mo"
-	rm -f  "$(DESTDIR)/share/locale/pt_BR/LC_MESSAGES/gprename.mo"
-	rm -f  "$(DESTDIR)/share/locale/ro/LC_MESSAGES/gprename.mo"
-	rm -f  "$(DESTDIR)/share/locale/ru/LC_MESSAGES/gprename.mo"
-	rm -f  "$(DESTDIR)/share/locale/sv/LC_MESSAGES/gprename.mo"
-	rm -f  "$(DESTDIR)/share/locale/zh_CN/LC_MESSAGES/gprename.mo"
-	rm -fr "/usr/share/icons/gprename"
-	update-desktop-database
+	rm -f  "$(DESTDIR)$(PREFIX)/bin/gprename"
+	rm -f  "$(DESTDIR)/usr/share/applications/gprename.desktop"
+	rm -f  "$(DESTDIR)/usr/share/applications/gprename-nautilus-actions.xml"
+	rm -f  "$(DESTDIR)$(PREFIX)/share/man/man1/gprename.1"
+	rm -f  "$(DESTDIR)$(PREFIX)/share/locale/ca/LC_MESSAGES/gprename.mo"
+	rm -f  "$(DESTDIR)$(PREFIX)/share/locale/de/LC_MESSAGES/gprename.mo"
+	rm -f  "$(DESTDIR)$(PREFIX)/share/locale/es/LC_MESSAGES/gprename.mo"
+	rm -f  "$(DESTDIR)$(PREFIX)/share/locale/et/LC_MESSAGES/gprename.mo"
+	rm -f  "$(DESTDIR)$(PREFIX)/share/locale/fr/LC_MESSAGES/gprename.mo"
+	rm -f  "$(DESTDIR)$(PREFIX)/share/locale/id/LC_MESSAGES/gprename.mo"
+	rm -f  "$(DESTDIR)$(PREFIX)/share/locale/it/LC_MESSAGES/gprename.mo"
+	rm -f  "$(DESTDIR)$(PREFIX)/share/locale/nl/LC_MESSAGES/gprename.mo"
+	rm -f  "$(DESTDIR)$(PREFIX)/share/locale/pl/LC_MESSAGES/gprename.mo"
+	rm -f  "$(DESTDIR)$(PREFIX)/share/locale/pt_BR/LC_MESSAGES/gprename.mo"
+	rm -f  "$(DESTDIR)$(PREFIX)/share/locale/ro/LC_MESSAGES/gprename.mo"
+	rm -f  "$(DESTDIR)$(PREFIX)/share/locale/ru/LC_MESSAGES/gprename.mo"
+	rm -f  "$(DESTDIR)$(PREFIX)/share/locale/sv/LC_MESSAGES/gprename.mo"
+	rm -f  "$(DESTDIR)$(PREFIX)/share/locale/zh_CN/LC_MESSAGES/gprename.mo"
+	rm -fr "$(DESTDIR)/usr/share/icons/gprename"
 
 clean:
 	rm -rf build
-- 
2.23.0

