From 1f7879fc32c6ed67a0c9bd897ed49210bdaf2a3c Mon Sep 17 00:00:00 2001
From: "Beatrice T. Meyers" <root@datadrake.com>
Date: Thu, 7 Apr 2022 00:53:29 -0400
Subject: [PATCH] Switch test to newer time API to fix compilation

---
 tests/testDailyRollingFileAppender.cpp | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/tests/testDailyRollingFileAppender.cpp b/tests/testDailyRollingFileAppender.cpp
index 9958f41..41fbeaf 100644
--- a/tests/testDailyRollingFileAppender.cpp
+++ b/tests/testDailyRollingFileAppender.cpp
@@ -232,7 +232,8 @@ namespace OnlyManualTesting {
 
 		now += seconds;
 
-		if (stime(&now) == -1) {
+		struct timespec ts = { .tv_sec = now };
+		if (clock_settime(CLOCK_REALTIME, &ts) == -1) {
 			std::cerr << "Can not set date. Need admin privileges?" << std::endl;
 			return -1;
 		}
-- 
2.35.1

