From da81ce9ce4212873e40561663aed195a20aae615 Mon Sep 17 00:00:00 2001
From: Fabio Forni <livingsilver94.solus@redaril.me>
Date: Mon, 22 Nov 2021 21:30:20 +0100
Subject: [PATCH] Read vimrc from /etc/vim

---
 src/feature.h | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/src/feature.h b/src/feature.h
index 36dd371..cc2ed15 100644
--- a/src/feature.h
+++ b/src/feature.h
@@ -819,22 +819,22 @@
 /*
  * SYS_VIMRC_FILE	Name of the user-defined system-wide .vimrc file.
  */
-// #define SYS_VIMRC_FILE	"/etc/vimrc"
+#define SYS_VIMRC_FILE	"/etc/vim/vimrc"
 
 /*
  * SYS_VIMRC_DEFAULT_FILE	Name of the default (stateless) system-wide .vimrc file.
  */
-// #define SYS_VIMRC_DEFAULT_FILE	"/usr/share/defaults/vim/vimrc"
+#define SYS_VIMRC_DEFAULT_FILE	"/usr/share/defaults/vim/vimrc"
 
 /*
  * SYS_GVIMRC_FILE	Name of the user-defined system-wide .gvimrc file.
  */
-// #define SYS_GVIMRC_FILE	"/etc/gvimrc"
+#define SYS_GVIMRC_FILE	"/etc/vim/gvimrc"
 
 /*
  * SYS_GVIMRC_DEFAULT_FILE	Name of the default (stateless) system-wide .gvimrc file.
  */
-// #define SYS_GVIMRC_DEFAULT_FILE	"/usr/share/defaults/vim/gvimrc"
+#define SYS_GVIMRC_DEFAULT_FILE	"/usr/share/defaults/vim/gvimrc"
 
 /*
  * DFLT_HELPFILE	Name of the help file.
-- 
2.34.0

