diff '--color=auto' -pruN a/src/feature.h b/src/feature.h
--- a/src/feature.h	2026-02-05 19:36:42.223982706 -0500
+++ b/src/feature.h	2026-02-05 19:37:41.684279672 -0500
@@ -703,22 +703,22 @@
 /*
  * SYS_VIMRC_FILE	Name of the user-defined system-wide .vimrc file.
  */
-// #define SYS_VIMRC_FILE	"/etc/vimrc"
+#define SYS_VIMRC_FILE	"/etc/vim/vimrc"
 
 /*
  * SYS_VIMRC_DEFAULT_FILE  Name of the default (stateless) system-wide .vimrc file.
  */
-// #define SYS_VIMRC_DEFAULT_FILE  "/usr/share/defaults/vim/vimrc"
+#define SYS_VIMRC_DEFAULT_FILE  "/usr/share/defaults/vim/vimrc"
 
 /*
  * SYS_GVIMRC_FILE	Name of the user-defined system-wide .gvimrc file.
  */
-// #define SYS_GVIMRC_FILE	"/etc/gvimrc"
+#define SYS_GVIMRC_FILE	"/etc/vim/gvimrc"
 
 /*
  * SYS_GVIMRC_DEFAULT_FILE Name of the default (stateless) system-wide .gvimrc file.
  */
-// #define SYS_GVIMRC_DEFAULT_FILE "/usr/share/defaults/vim/gvimrc"
+#define SYS_GVIMRC_DEFAULT_FILE "/usr/share/defaults/vim/gvimrc"
 
 /*
  * DFLT_HELPFILE	Name of the help file.
