diff --git a/install.sh b/install.sh
index 877210d..17ce256 100755
--- a/install.sh
+++ b/install.sh
@@ -26,11 +26,6 @@ verbose_echo() {
     fi
 }
 
-if [ "$(id -u)" -eq 0 ]; then
-    INSTALL_PREFIX="${INSTALL_PREFIX:-/usr/local}"
-else
-    INSTALL_PREFIX="${INSTALL_PREFIX:-$HOME/.local}"
-fi
 verbose_echo "Installing Lyrebird to prefix: ${INSTALL_PREFIX}"
 
 BIN_PATH="$INSTALL_PREFIX/bin"
@@ -82,9 +77,6 @@ if [ -d "/etc/lyrebird/" ]; then
     warning_echo "/etc/lyrebird/ is now deprecated, please relocate contents to ~/.config/lyrebird/ and delete"
 fi
 
-# Required pip3 modules space separated
-REQUIRED_PIP_MODULES="toml"
-
 # Create all of the directories if they don't exist
 if [ ! -d "$BIN_PATH" ]; then
     verbose_echo "Creating binary path: $BIN_PATH"
@@ -109,14 +101,6 @@ if [ ! -d "$DESKTOP_PATH" ]; then
     fi
 fi
 
-install_python_modules() {
-    verbose_echo "Installing Python modules: $REQUIRED_PIP_MODULES"
-    if [ $DRYRUN != 1 ]; then
-        # Var not included in quotes so it installs each module
-        pip3 install --prefix $INSTALL_PREFIX $REQUIRED_PIP_MODULES
-    fi
-}
-
 install_binary_source() {
     verbose_echo "Copying app/ to: $SHARE_PATH"
     if [ $DRYRUN != 1 ]; then cp -rf app "$SHARE_PATH"; fi
@@ -136,7 +120,7 @@ install_binary_source() {
 
 install_desktop() {
     verbose_echo "Copying lyrebird.desktop to: $DESKTOP_PATH"
-    if [ $DRYRUN != 1 ]; then BIN_PATH=$BIN_PATH SHARE_PATH=$SHARE_PATH envsubst < lyrebird.desktop > $DESKTOP_PATH/lyrebird.desktop; fi
+    if [ $DRYRUN != 1 ]; then BIN_PATH=/usr/bin/ SHARE_PATH=/usr/share/lyrebird envsubst < lyrebird.desktop > $DESKTOP_PATH/lyrebird.desktop; fi
 
     verbose_echo "Setting permission 644 for: $DESKTOP_PATH/lyrebird.desktop"
     if [ $DRYRUN != 1 ]; then chmod -R 644 "$DESKTOP_PATH/lyrebird.desktop"; fi
