From 543c6edc3519cf46bd3dc2a13d2a10f25774f97d Mon Sep 17 00:00:00 2001
From: "F. von Gellhorn" <flinux@vongellhorn.ch>
Date: Sun, 26 Jan 2020 02:47:49 +0100
Subject: [PATCH 1/1] Fix apparmor for akonadi

---
 apparmor/mysqld_akonadi | 1 +
 1 file changed, 1 insertion(+)

diff --git a/apparmor/mysqld_akonadi b/apparmor/mysqld_akonadi
index c769d87..7a7abc5 100644
--- a/apparmor/mysqld_akonadi
+++ b/apparmor/mysqld_akonadi
@@ -30,4 +30,5 @@ profile mysqld_akonadi {
   /usr/share/mysql/** r,
   owner @{xdg_data_home}/akonadi/** rwk,
   owner @{PROC}/@{pid}/loginuid r,
+  /{,var/}run/user/[0-9]*/akonadi** rwk,
 }
-- 
2.25.0

