From 8694be9e10450b780ee27397f56fb4a8737dd271 Mon Sep 17 00:00:00 2001
From: Peter O'Connor <peter@solus-project.com>
Date: Wed, 4 Apr 2018 10:57:21 +1000
Subject: [PATCH] Disable building unshipped code

Signed-off-by: Peter O'Connor <peter@solus-project.com>
---
 CMakeLists.txt | 1 -
 1 file changed, 1 deletion(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 233e45d..1913c61 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -28,7 +28,6 @@ add_subdirectory(icons)
 add_subdirectory(kcm)
 #add_subdirectory(plasma)
 add_subdirectory(kded)
-add_subdirectory(tests)
 add_subdirectory(console)
 
 install( FILES kscreen.categories DESTINATION ${KDE_INSTALL_CONFDIR} )
-- 
2.16.3

