From 056722da90834c3626441723897d0e023715a237 Mon Sep 17 00:00:00 2001
From: Peter O'Connor <sunnyflunk@gmail.com>
Date: Wed, 14 Dec 2016 19:14:40 +1100
Subject: [PATCH] Use breeze icons by default

---
 shell/main.cpp | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/shell/main.cpp b/shell/main.cpp
index cb240cc..c48c0b0 100644
--- a/shell/main.cpp
+++ b/shell/main.cpp
@@ -31,6 +31,8 @@ int main(int argc, char** argv)
 {
     QApplication app(argc, argv);

+    QIcon::setThemeName("breeze");
+
     KLocalizedString::setApplicationDomain("okular");

     KAboutData aboutData = okularAboutData();
