diff -ruNb a/inxi b/inxi
--- a/inxi	2014-05-23 00:46:30.000000000 +0100
+++ b/inxi	2014-07-13 18:21:45.893867961 +0100
@@ -601,7 +601,7 @@
 DISTROS_DERIVED="antix-version aptosid-version kanotix-version knoppix-version mandrake-release pardus-release porteus-version sabayon-release siduction-version sidux-version slitaz-release solusos-release turbolinux-release zenwalk-version"
 # debian_version excluded from DISTROS_PRIMARY so Debian can fall through to /etc/issue detection. Same goes for Ubuntu.
 DISTROS_EXCLUDE_LIST="debian_version ubuntu_version"
-DISTROS_PRIMARY="arch-release gentoo-release redhat-release slackware-version SuSE-release"
+DISTROS_PRIMARY="arch-release gentoo-release redhat-release slackware-version SuSE-release solus-release"
 DISTROS_LSB_GOOD="mandrake-release mandriva-release mandrakelinux-release"
 # this is being used both by core distros and derived distros now, eg, solusos 1 uses it for solusos id, while
 # debian, solusos base, uses it as well, so we have to know which it is.
@@ -8046,7 +8046,7 @@
 	local repo_file='' repo_data_working='' repo_data_working2='' repo_line='' repo_files=''
 	local repo_name=''
 	local apt_file='/etc/apt/sources.list' yum_repo_dir='/etc/yum.repos.d/' yum_conf='/etc/yum.conf'
-	local pacman_conf='/etc/pacman.conf' pacman_repo_dir='/etc/pacman.d/' pisi_dir='/etc/pisi/'
+	local pacman_conf='/etc/pacman.conf' pacman_repo_dir='/etc/pacman.d/' pisi_dir='/etc/pisi/' eopkg_dir='/etc/eopkg/'
 	local zypp_repo_dir='/etc/zypp/repos.d/' freebsd_conf='/etc/portsnap.conf'
 	
 	# apt - debian, buntus, also sometimes some yum/rpm repos may create apt repos here as well
@@ -8174,7 +8174,7 @@
 		# execute line breaks
 		REPO_DATA="$( echo -e $repo_data_working2 )"
 	# pisi - pardus
-	elif [[ -d $pisi_dir && -n $( type -p pisi ) ]];then
+	elif [[ -d $pisi_dir && -n $( type -p pisi ) || -d $eopkg_dir && -n $( type -p pisi) ]];then
 		REPO_DATA="$( pisi list-repo )"
 		# now we need to create the structure: repo info: repo path
 		# we do that by looping through the lines of the output and then
@@ -8198,7 +8198,11 @@
 			if [[ -n $( grep '://' <<< $repo_line ) ]];then
 				repo_data_working="$repo_data_working:$repo_line\n"
 			else
+				if [[ -d $pisi_dir ]];then
 				repo_data_working="${repo_data_working}pisi repo:$repo_line"
+				else
+					repo_data_working="${repo_data_working}eopkg repo:$repo_line"
+				fi
 			fi
 		done <<< "$REPO_DATA"
 		# echo and execute the line breaks inserted
