From 8b6af4242b6977391d14441f8f88204e10421e6a Mon Sep 17 00:00:00 2001
From: Sebastiano Vigna <sebastiano.vigna@gmail.com>
Date: Thu, 20 Nov 2025 16:36:20 +0700
Subject: [PATCH] Added explicit dependency from $(MAINH) for ansi.o

---
 src/makefile | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/src/makefile b/src/makefile
index ed593e5..4f38c7c 100644
--- a/src/makefile
+++ b/src/makefile
@@ -156,6 +156,8 @@ enums.h names.c names.h hash.c hash.h help.c help.h ext.c: ne.texinfo version.te
 	perl info2src.pl
 	rm -f ne.info*
 
+ansi.o: $(MAINH)
+
 actions.o: $(MAINH) support.h keycodes.h names.h names.c errors.h errors.c protos.h version.h
 
 autocomp.o: $(MAINH) support.h protos.h

