From 5a13948467e01c060296ce01528338f1c45d9005 Mon Sep 17 00:00:00 2001
From: Ivan Trepakov <liontiger23@gmail.com>
Date: Fri, 21 Nov 2025 09:57:26 +0700
Subject: [PATCH] Separate enums.h and other files that depend on it for
 parallel building

---
 src/makefile | 7 +++++--
 1 file changed, 5 insertions(+), 2 deletions(-)

diff --git a/src/makefile b/src/makefile
index 4f38c7c..194ecbf 100644
--- a/src/makefile
+++ b/src/makefile
@@ -146,12 +146,15 @@ ne.texinfo:
 version.texinfo: ../doc/version.texinfo
 	-ln -sf ../doc/version.texinfo
 
-SYNH = syn_types.h syn_hash.h syn_regex.h syn_utf8.h syn_utils.h
+ENUMH = enums.h
+SYNH = syn_types.h syn_hash.h syn_regex.h syn_utf8.h syn_utils.h $(ENUMH)
 MAINH = ne.h debug.h enums.h errors.h utf8.h syntax.h keycodes.h termchar.h info2cap.h $(SYNH)
 
 ext.c: ext.c.in
 
-enums.h names.c names.h hash.c hash.h help.c help.h ext.c: ne.texinfo version.texinfo info2src.pl
+names.c names.h hash.c hash.h help.c help.h ext.c: $(ENUMH)
+
+$(ENUMH): ne.texinfo version.texinfo info2src.pl
 	makeinfo -D autohelp ne.texinfo
 	perl info2src.pl
 	rm -f ne.info*

