From 5eaae202abf9209fbd5df6608c0360775df4f6c6 Mon Sep 17 00:00:00 2001
From: Joshua Strobl <joshua@stroblindustries.com>
Date: Wed, 17 Apr 2019 09:38:33 +0300
Subject: [PATCH 1/1] Ensure we use the system keyboard layout.

---
 data/ibus.schemas.in | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/data/ibus.schemas.in b/data/ibus.schemas.in
index f4a019d0..d98fa771 100644
--- a/data/ibus.schemas.in
+++ b/data/ibus.schemas.in
@@ -325,7 +325,7 @@
       <applyto>/desktop/ibus/general/use_system_keyboard_layout</applyto>
       <owner>ibus</owner>
       <type>bool</type>
-      <default>false</default>
+      <default>true</default>
       <locale name="C">
         <short>Use system keyboard layout</short>
 	    <long>Use system keyboard (XKB) layout</long>
-- 
2.21.0

