From e880ac712407a2f676d8c60442fcb8c629d067c1 Mon Sep 17 00:00:00 2001
From: Gavin Zhao <git@gzgz.dev>
Date: Sun, 22 Dec 2024 18:55:13 -0500
Subject: [PATCH] Don't build libhsakmt-staticdrm

Signed-off-by: Gavin Zhao <git@gzgz.dev>
---
 CMakeLists.txt | 74 +++++++++++++++++++++++++-------------------------
 1 file changed, 37 insertions(+), 37 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index d10078a..5f37b11 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -289,43 +289,43 @@ configure_file ( libhsakmt.pc.in libhsakmt.pc @ONLY )
 
 install ( FILES ${CMAKE_CURRENT_BINARY_DIR}/libhsakmt.pc DESTINATION ${CMAKE_INSTALL_LIBDIR}/pkgconfig COMPONENT devel)
 
-## Create separate target file for static builds
-## In static builds, libdrm and libdrm_amdgpu need to be linked statically
-add_library ( ${HSAKMT_STATIC_DRM_TARGET}  STATIC ${HSAKMT_SRC})
-
-target_include_directories( ${HSAKMT_STATIC_DRM_TARGET}
-  PUBLIC
-  $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/include>
-  $<INSTALL_INTERFACE:${CMAKE_INSTALL_INCLUDEDIR}>
-  PRIVATE
-  ${CMAKE_CURRENT_SOURCE_DIR}/src )
-
-## Set the VERSION and SOVERSION values
-set_property(TARGET ${HSAKMT_STATIC_DRM_TARGET} PROPERTY LINK_FLAGS ${HSAKMT_LINK_FLAGS}
-             PROPERTY VERSION "${LIB_VERSION_STRING}"
-             PROPERTY SOVERSION "${LIB_VERSION_MAJOR}" )
-
-#Additional search path for static libraries
-if(${DISTRO_ID} MATCHES "ubuntu")
-    set(AMDGPU_STATIC_LIB_PATHS "-L/opt/amdgpu/lib/x86_64-linux-gnu")
-else()
-    set(AMDGPU_STATIC_LIB_PATHS "-L/opt/amdgpu/lib64" "-L/opt/amdgpu/lib")
-endif()
-# Link drm_amdgpu and drm library statically
-target_link_libraries ( ${HSAKMT_STATIC_DRM_TARGET}
-  PRIVATE pthread rt c numa ${CMAKE_DL_LIBS}
-  INTERFACE -Wl,-Bstatic ${AMDGPU_STATIC_LIB_PATHS} ${DRM_AMDGPU_LDFLAGS} ${DRM_LDFLAGS} -Wl,-Bdynamic
-)
-target_compile_options(${HSAKMT_STATIC_DRM_TARGET} PRIVATE ${DRM_CFLAGS} ${HSAKMT_C_FLAGS})
-
-install ( TARGETS ${HSAKMT_STATIC_DRM_TARGET} EXPORT ${HSAKMT_STATIC_DRM_TARGET}Targets
-          ARCHIVE DESTINATION ${CMAKE_INSTALL_LIBDIR} COMPONENT devel
-          LIBRARY DESTINATION ${CMAKE_INSTALL_LIBDIR} COMPONENT devel)
-install ( EXPORT ${HSAKMT_STATIC_DRM_TARGET}Targets
-  FILE ${HSAKMT_STATIC_DRM_TARGET}Targets.cmake
-  NAMESPACE ${HSAKMT_STATIC_DRM_TARGET}::
-  DESTINATION ${CMAKE_INSTALL_LIBDIR}/cmake/${HSAKMT_TARGET}
-  COMPONENT devel)
+# ## Create separate target file for static builds
+# ## In static builds, libdrm and libdrm_amdgpu need to be linked statically
+# add_library ( ${HSAKMT_STATIC_DRM_TARGET}  STATIC ${HSAKMT_SRC})
+# 
+# target_include_directories( ${HSAKMT_STATIC_DRM_TARGET}
+#   PUBLIC
+#   $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/include>
+#   $<INSTALL_INTERFACE:${CMAKE_INSTALL_INCLUDEDIR}>
+#   PRIVATE
+#   ${CMAKE_CURRENT_SOURCE_DIR}/src )
+# 
+# ## Set the VERSION and SOVERSION values
+# set_property(TARGET ${HSAKMT_STATIC_DRM_TARGET} PROPERTY LINK_FLAGS ${HSAKMT_LINK_FLAGS}
+#              PROPERTY VERSION "${LIB_VERSION_STRING}"
+#              PROPERTY SOVERSION "${LIB_VERSION_MAJOR}" )
+# 
+# #Additional search path for static libraries
+# if(${DISTRO_ID} MATCHES "ubuntu")
+#     set(AMDGPU_STATIC_LIB_PATHS "-L/opt/amdgpu/lib/x86_64-linux-gnu")
+# else()
+#     set(AMDGPU_STATIC_LIB_PATHS "-L/opt/amdgpu/lib64" "-L/opt/amdgpu/lib")
+# endif()
+# # Link drm_amdgpu and drm library statically
+# target_link_libraries ( ${HSAKMT_STATIC_DRM_TARGET}
+#   PRIVATE pthread rt c numa ${CMAKE_DL_LIBS}
+#   INTERFACE -Wl,-Bstatic ${AMDGPU_STATIC_LIB_PATHS} ${DRM_AMDGPU_LDFLAGS} ${DRM_LDFLAGS} -Wl,-Bdynamic
+# )
+# target_compile_options(${HSAKMT_STATIC_DRM_TARGET} PRIVATE ${DRM_CFLAGS} ${HSAKMT_C_FLAGS})
+# 
+# install ( TARGETS ${HSAKMT_STATIC_DRM_TARGET} EXPORT ${HSAKMT_STATIC_DRM_TARGET}Targets
+#           ARCHIVE DESTINATION ${CMAKE_INSTALL_LIBDIR} COMPONENT devel
+#           LIBRARY DESTINATION ${CMAKE_INSTALL_LIBDIR} COMPONENT devel)
+# install ( EXPORT ${HSAKMT_STATIC_DRM_TARGET}Targets
+#   FILE ${HSAKMT_STATIC_DRM_TARGET}Targets.cmake
+#   NAMESPACE ${HSAKMT_STATIC_DRM_TARGET}::
+#   DESTINATION ${CMAKE_INSTALL_LIBDIR}/cmake/${HSAKMT_TARGET}
+#   COMPONENT devel)
 
 ###########################
 # Packaging directives
-- 
2.47.0

