From e95dad3926c69a60b5ae16467e31f20696c0ce35 Mon Sep 17 00:00:00 2001
From: Joshua Strobl <joshua@streambits.io>
Date: Tue, 10 Nov 2020 19:26:54 +0200
Subject: [PATCH 1/1] Don't perform node version check

---
 build/npm/preinstall.js | 7 -------
 1 file changed, 7 deletions(-)

diff --git a/build/npm/preinstall.js b/build/npm/preinstall.js
index cb88d37ade..678e67b457 100644
--- a/build/npm/preinstall.js
+++ b/build/npm/preinstall.js
@@ -5,13 +5,6 @@
 
 let err = false;
 
-const majorNodeVersion = parseInt(/^(\d+)\./.exec(process.versions.node)[1]);
-
-if (majorNodeVersion < 10 || majorNodeVersion >= 13) {
-	console.error('\033[1;31m*** Please use node >=10 and <=12.\033[0;0m');
-	err = true;
-}
-
 const cp = require('child_process');
 const yarnVersion = cp.execSync('yarn -v', { encoding: 'utf8' }).trim();
 const parsedYarnVersion = /^(\d+)\.(\d+)\./.exec(yarnVersion);
-- 
2.29.1

