From a2bbf582bd0740d7052c12c5df16b1a4e2a24107 Mon Sep 17 00:00:00 2001
From: Joshua Strobl <joshua@stroblindustries.com>
Date: Wed, 13 Feb 2019 01:42:20 +0200
Subject: [PATCH 1/1] Patch out node check.

---
 build/npm/preinstall.js | 5 -----
 1 file changed, 5 deletions(-)

diff --git a/build/npm/preinstall.js b/build/npm/preinstall.js
index 3b0d7dc538..6c8b5b46a4 100644
--- a/build/npm/preinstall.js
+++ b/build/npm/preinstall.js
@@ -7,11 +7,6 @@ let err = false;
 
 const majorNodeVersion = parseInt(/^(\d+)\./.exec(process.versions.node)[1]);
 
-if (majorNodeVersion < 8 || majorNodeVersion >= 11) {
-	console.error('\033[1;31m*** Please use node >=8 and <11.\033[0;0m');
-	err = true;
-}
-
 const cp = require('child_process');
 const yarnVersion = cp.execSync('yarn -v', { encoding: 'utf8' }).trim();
 const parsedYarnVersion = /^(\d+)\.(\d+)\./.exec(yarnVersion);
-- 
2.20.1

