From 7d6e9286ae7e573b89269fb56f8ef7d46d1c0e1e Mon Sep 17 00:00:00 2001
From: Evan Maddock <maddock.evan@vivaldi.net>
Date: Thu, 24 Jun 2021 19:21:05 -0400
Subject: [PATCH] Make wrapped_lines static to fix compile errors

Signed-off-by: Evan Maddock <maddock.evan@vivaldi.net>
---
 src/t4k_common.h   | 2 +-
 src/t4k_linewrap.c | 2 +-
 2 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/t4k_common.h b/src/t4k_common.h
index d4676e7..ef8bbd6 100644
--- a/src/t4k_common.h
+++ b/src/t4k_common.h
@@ -239,7 +239,7 @@ MFStrategy;
 #define MAX_LINES 128     //!< Maximum lines to wrap.
 #define MAX_LINEWIDTH 256 //!< Maximum characters of each line.
 
-char wrapped_lines[MAX_LINES][MAX_LINEWIDTH]; //!< Global buffer for wrapped lines.
+static char wrapped_lines[MAX_LINES][MAX_LINEWIDTH]; //!< Global buffer for wrapped lines.
 
 //TODO separate headers for different areas a la SDL?
 
diff --git a/src/t4k_linewrap.c b/src/t4k_linewrap.c
index 5f80ad1..e1c8464 100644
--- a/src/t4k_linewrap.c
+++ b/src/t4k_linewrap.c
@@ -32,7 +32,7 @@ along with this program.  If not, see <http://www.gnu.org/licenses/>.  */
 #include <stdio.h>
 
 static char wrapped_lines0[MAX_LINES][MAX_LINEWIDTH];  // for internal storage
-char wrapped_lines[MAX_LINES][MAX_LINEWIDTH]; // publicly available!
+static char wrapped_lines[MAX_LINES][MAX_LINEWIDTH]; // publicly available!
 
 
 
-- 
2.32.0

