From a4149721f883f468f657cc75a722c55a9154dad2 Mon Sep 17 00:00:00 2001
From: Ikey Doherty <ikey@solus-project.com>
Date: Sat, 27 May 2017 11:30:31 +0100
Subject: [PATCH] launcher: Fallback to the correct path within Solus

Signed-off-by: Ikey Doherty <ikey@solus-project.com>
---
 fs-uae-launcher | 5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

diff --git a/fs-uae-launcher b/fs-uae-launcher
index b1071e7..ba1a334 100755
--- a/fs-uae-launcher
+++ b/fs-uae-launcher
@@ -3,8 +3,9 @@
 import os
 import sys
 
-if os.environ.get("FSGS_PYTHONPATH", ""):
-    sys.path.insert(0, os.environ.get("FSGS_PYTHONPATH"))
+# Always fall back to a sane path
+fs_ua_path = os.environ.get("FSGS_PYTHONPATH", "/usr/share/fs-uae-launcher")
+sys.path.insert(0, fs_ua_path)
 
 from fs_uae_launcher.main import main
 main()
-- 
2.13.0

