From b3246e1e2c5531455781d9a26ab5e92ef2b186db Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Luk=C3=A1=C5=A1=20Lalinsk=C3=BD?= <lalinsky@gmail.com>
Date: Fri, 4 Nov 2022 21:25:40 +0100
Subject: [PATCH] Only set FFmpeg include dirs if building tools

---
 tests/CMakeLists.txt | 13 ++++++-------
 1 file changed, 6 insertions(+), 7 deletions(-)

diff --git a/tests/CMakeLists.txt b/tests/CMakeLists.txt
index 0afb060..123e643 100644
--- a/tests/CMakeLists.txt
+++ b/tests/CMakeLists.txt
@@ -10,13 +10,6 @@ include_directories(
 	${CMAKE_CURRENT_SOURCE_DIR}
 )
 
-include_directories(
-	${FFMPEG_LIBAVFORMAT_INCLUDE_DIRS}
-	${FFMPEG_LIBAVCODEC_INCLUDE_DIRS}
-	${FFMPEG_LIBAVUTIL_INCLUDE_DIRS}
-	${AUDIO_PROCESSOR_INCLUDE_DIRS}
-)
-
 set(SRCS
 	main.cpp
 	test_api.cpp
@@ -45,6 +38,12 @@ set(SRCS
 
 if(BUILD_TOOLS)
 	set(SRCS ${SRCS} ../src/audio/ffmpeg_audio_reader_test.cpp)
+    include_directories(
+        ${FFMPEG_LIBAVFORMAT_INCLUDE_DIRS}
+        ${FFMPEG_LIBAVCODEC_INCLUDE_DIRS}
+        ${FFMPEG_LIBAVUTIL_INCLUDE_DIRS}
+        ${AUDIO_PROCESSOR_INCLUDE_DIRS}
+    )
 	link_libraries(fpcalc_libs)
 endif()
 
