diff -uNrp a/xed/xed-app.c b/xed/xed-app.c
--- a/xed/xed-app.c	2025-08-26 17:05:48.000000000 +0100
+++ b/xed/xed-app.c	2025-09-09 01:03:41.042962295 +0100
@@ -38,7 +38,7 @@
 #include <gtksourceview/gtksource.h>
 
 #ifdef ENABLE_INTROSPECTION
-#include <girepository.h>
+#include <girepository/girepository.h>
 #endif
 
 #include "xed-app.h"
@@ -991,7 +991,7 @@ xed_app_init (XedApp *app)
     g_application_add_main_option_entries (G_APPLICATION (app), options);
 
 #ifdef ENABLE_INTROSPECTION
-    g_application_add_option_group (G_APPLICATION (app), g_irepository_get_option_group ());
+    g_application_add_option_group (G_APPLICATION (app), gi_repository_get_option_group ());
 #endif
 
     load_accels ();
diff -uNrp a/xed/xed-plugins-engine.c b/xed/xed-plugins-engine.c
--- a/xed/xed-plugins-engine.c	2025-08-26 17:05:48.000000000 +0100
+++ b/xed/xed-plugins-engine.c	2025-09-09 01:06:40.371484541 +0100
@@ -31,7 +31,7 @@
 #include <config.h>
 #include <string.h>
 #include <glib/gi18n.h>
-#include <girepository.h>
+#include <girepository/girepository.h>
 
 #include "xed-plugins-engine.h"
 #include "xed-debug.h"
@@ -66,7 +66,7 @@ xed_plugins_engine_init (XedPluginsEngin
 
     typelib_dir = g_build_filename (xed_dirs_get_xed_lib_dir (), "girepository-1.0", NULL);
 
-    if (!g_irepository_require_private (g_irepository_get_default (), typelib_dir, "Xed", "1.0", 0, &error))
+    if (!gi_repository_require_private (gi_repository_dup_default (), typelib_dir, "Xed", "1.0", 0, &error))
     {
         g_warning ("Could not load Xed repository: %s", error->message);
         g_error_free (error);
@@ -76,14 +76,14 @@ xed_plugins_engine_init (XedPluginsEngin
     g_free (typelib_dir);
 
     /* This should be moved to libpeas */
-    if (!g_irepository_require (g_irepository_get_default (), "Peas", "1.0", 0, &error))
+    if (!gi_repository_require (gi_repository_dup_default (), "Peas", "1.0", 0, &error))
     {
         g_warning ("Could not load Peas repository: %s", error->message);
         g_error_free (error);
         error = NULL;
     }
 
-    if (!g_irepository_require (g_irepository_get_default (), "PeasGtk", "1.0", 0, &error))
+    if (!gi_repository_require (gi_repository_dup_default (), "PeasGtk", "1.0", 0, &error))
     {
         g_warning ("Could not load PeasGtk repository: %s", error->message);
         g_error_free (error);
