diff --git a/src/land/engine/ArcanistLandEngine.php b/src/land/engine/ArcanistLandEngine.php
index 5722073b..8473606b 100644
--- a/src/land/engine/ArcanistLandEngine.php
+++ b/src/land/engine/ArcanistLandEngine.php
@@ -282,56 +282,6 @@ abstract class ArcanistLandEngine
     $viewer = $this->getViewer();
     $viewer_phid = $viewer->getPHID();
 
-    $unauthored = array();
-    foreach ($revision_refs as $revision_ref) {
-      $author_phid = $revision_ref->getAuthorPHID();
-      if ($author_phid !== $viewer_phid) {
-        $unauthored[] = $revision_ref;
-      }
-    }
-
-    if ($unauthored) {
-      $this->getWorkflow()->loadHardpoints(
-        $unauthored,
-        array(
-          ArcanistRevisionRef::HARDPOINT_AUTHORREF,
-        ));
-
-      echo tsprintf(
-        "\n%!\n%W\n\n",
-        pht('NOT REVISION AUTHOR'),
-        pht(
-          'You are landing revisions which you ("%s") are not the author of:',
-          $viewer->getMonogram()));
-
-      foreach ($unauthored as $revision_ref) {
-        $display_ref = $revision_ref->newRefView();
-
-        $author_ref = $revision_ref->getAuthorRef();
-        if ($author_ref) {
-          $display_ref->appendLine(
-            pht(
-              'Author: %s',
-              $author_ref->getMonogram()));
-        }
-
-        echo tsprintf('%s', $display_ref);
-      }
-
-      echo tsprintf(
-        "\n%?\n",
-        pht(
-          'Use "Commandeer" in the web interface to become the author of '.
-          'a revision.'));
-
-      $query = pht('Land revisions you are not the author of?');
-
-      $this->getWorkflow()
-        ->getPrompt('arc.land.unauthored')
-        ->setQuery($query)
-        ->execute();
-    }
-
     $planned = array();
     $published = array();
     $not_accepted = array();
