From 38a4088289d7a560dc2f052c941752c4cf0a1986 Mon Sep 17 00:00:00 2001
From: Campbell Jones <git@serebit.com>
Date: Sun, 2 Aug 2020 00:05:04 -0400
Subject: [PATCH 1/1] Use XDG compliant config location if it already exists

---
 .../ArcanistConfigurationManager.php          | 22 +++++++++++++++----
 .../ArcanistInstallCertificateWorkflow.php    |  2 +-
 2 files changed, 19 insertions(+), 5 deletions(-)

diff --git a/src/configuration/ArcanistConfigurationManager.php b/src/configuration/ArcanistConfigurationManager.php
index 68ebde4e..f857c00e 100644
--- a/src/configuration/ArcanistConfigurationManager.php
+++ b/src/configuration/ArcanistConfigurationManager.php
@@ -194,10 +194,10 @@ final class ArcanistConfigurationManager extends Phobject {
             $prompt = pht(
               "File permissions on your %s are too open. ".
               "Fix them by chmod'ing to 600?",
-              '~/.arcrc');
+              $user_config_path);
             if (!phutil_console_confirm($prompt, $default_no = false)) {
               throw new ArcanistUsageException(
-                pht('Set %s to file mode 600.', '~/.arcrc'));
+                pht('Set %s to file mode 600.', $user_config_path));
             }
             execx('chmod 600 %s', $user_config_path);
 
@@ -213,7 +213,7 @@ final class ArcanistConfigurationManager extends Phobject {
           $user_config = phutil_json_decode($user_config_data);
         } catch (PhutilJSONParserException $ex) {
           throw new PhutilProxyException(
-            pht("Your '%s' file is not a valid JSON file.", '~/.arcrc'),
+            pht("Your '%s' file is not a valid JSON file.", $user_config_path),
             $ex);
         }
       } else {
@@ -258,6 +258,7 @@ final class ArcanistConfigurationManager extends Phobject {
   }
 
   public function getUserConfigurationFileLocation() {
+    $console = PhutilConsole::getConsole();
     if ($this->customArcrcFilename !== null) {
       return $this->customArcrcFilename;
     }
@@ -265,7 +266,20 @@ final class ArcanistConfigurationManager extends Phobject {
     if (phutil_is_windows()) {
       return getenv('APPDATA').'/.arcrc';
     } else {
-      return getenv('HOME').'/.arcrc';
+      $home_dir = getenv('HOME');
+      $alt_config = $home_dir.'/.config/arcrc';
+      $xdg_config_dir = getenv('XDG_CONFIG_DIR');
+
+      if ($xdg_config_dir != FALSE) {
+        $xdg_config = $xdg_config_dir.'/arcrc';
+        if (file_exists($xdg_config)) {
+          return $xdg_config;
+        }
+      } elseif (file_exists($alt_config)) {
+        return $alt_config;
+      } else {
+        return $home_dir.'/.arcrc';
+      }
     }
   }
 
diff --git a/src/workflow/ArcanistInstallCertificateWorkflow.php b/src/workflow/ArcanistInstallCertificateWorkflow.php
index 49bb5045..64fcd2ad 100644
--- a/src/workflow/ArcanistInstallCertificateWorkflow.php
+++ b/src/workflow/ArcanistInstallCertificateWorkflow.php
@@ -167,7 +167,7 @@ EOTEXT
       );
     }
 
-    echo pht('Writing %s...', '~/.arcrc')."\n";
+    echo pht('Writing %s...', $config_path)."\n";
     $configuration_manager->writeUserConfigurationFile($config);
 
     if ($is_token_auth) {
-- 
2.39.0

