From b54bdbc6638b89d8d8f67e942e8b7fd88c410866 Mon Sep 17 00:00:00 2001
From: Ikey Doherty <ikey@solus-project.com>
Date: Sun, 7 May 2017 17:03:07 +0100
Subject: [PATCH] Switch branches after first creating them for Initial Commit

Signed-off-by: Ikey Doherty <ikey@solus-project.com>
---
 src/workflow/ArcanistPatchWorkflow.php | 12 +++++++++---
 1 file changed, 9 insertions(+), 3 deletions(-)

diff --git a/src/workflow/ArcanistPatchWorkflow.php b/src/workflow/ArcanistPatchWorkflow.php
index c1a8f878..da2d787e 100644
--- a/src/workflow/ArcanistPatchWorkflow.php
+++ b/src/workflow/ArcanistPatchWorkflow.php
@@ -303,15 +303,21 @@ EOTEXT
       $branch_name = $this->getBranchName($bundle);
       $base_revision = $bundle->getBaseRevision();
 
-      if ($base_revision && $has_base_revision) {
+      // We may be applying the initial commit to this repository, so skip
+      // the base revision in this instance.
+      if ($base_revision && $has_base_revision && $base_revision != ArcanistGitAPI::GIT_MAGIC_ROOT_COMMIT) {
         $base_revision = $repository_api->getCanonicalRevisionName(
           $base_revision);
         $repository_api->execxLocal(
-          'checkout -b %s %s',
+          'branch %s %s',
           $branch_name,
           $base_revision);
+        $repository_api->execxLocal(
+          'checkout %s',
+          $branch_name);
       } else {
-        $repository_api->execxLocal('checkout -b %s', $branch_name);
+        $repository_api->execxLocal('branch %s', $branch_name);
+        $repository_api->execxLocal('checkout %s', $branch_name);
       }
 
       echo phutil_console_format(
-- 
2.12.2

