From c947eed68559152ac95bdab207b206794ee9a5c2 Mon Sep 17 00:00:00 2001
From: Jens Georg <mail@jensge.org>
Date: Mon, 14 Jan 2019 15:06:09 +0100
Subject: [PATCH] Fix compiler warning in ls-profiles

---
 tools/gupnp-dlna-ls-profiles.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/tools/gupnp-dlna-ls-profiles.c b/tools/gupnp-dlna-ls-profiles.c
index 22b9924..74d1315 100644
--- a/tools/gupnp-dlna-ls-profiles.c
+++ b/tools/gupnp-dlna-ls-profiles.c
@@ -39,7 +39,7 @@
 static gboolean relaxed = FALSE;
 
 static void
-print_profile (GUPnPDLNAProfile *profile)
+print_profile (GUPnPDLNAProfile *profile, gpointer user_data)
 {
         g_print ("%s %-30s%-35s\n",
                  gupnp_dlna_profile_get_extended (profile) ? "*" : " ",
-- 
2.18.1

