From 35ac1470b366fa133ec7af09ac375ab769565f3d Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Abdulkadir=20Furkan=20=C5=9Eanl=C4=B1?=
 <abdulkadirfsanli@protonmail.com>
Date: Sun, 30 Jun 2019 23:49:08 +0200
Subject: [PATCH] Fix ldconfig call.

---
 librarian.sh | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/librarian.sh b/librarian.sh
index 96fdae4..272316c 100755
--- a/librarian.sh
+++ b/librarian.sh
@@ -29,6 +29,6 @@ files)  echo "$FULLNAME" "$LIBNAME" "$LIBNAME.$MAJOR"
 install)/usr/bin/install -c $FULLNAME "$1"
 	/bin/ln -s -f $FULLNAME $1/$LIBNAME.$MAJOR
 	/bin/ln -s -f $FULLNAME $1/$LIBNAME
-	/sbin/ldconfig "$1"
+	/sbin/ldconfig -n "$1"
 	;;
 esac
-- 
2.22.0

