From 55cea9c59fb35999f186f44af40f512d7f23871b Mon Sep 17 00:00:00 2001
From: Balint Reczey <balint.reczey@canonical.com>
Date: Sat, 10 Oct 2020 21:21:48 +0200
Subject: [PATCH] Honor system's LDFLAGS in librarian.sh

Fixes #228.
---
 configure.inc | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/configure.inc b/configure.inc
index 97ff6b9..ac0f973 100755
--- a/configure.inc
+++ b/configure.inc
@@ -1428,7 +1428,7 @@ LIBNAME=\$LIBRARY.dylib
 FULLNAME=\$LIBNAME
 
 case "\$ACTION" in
-make)   FLAGS="$AC_CFLAGS -dynamiclib"
+make)   FLAGS="$AC_CFLAGS $AC_LDFLAGS -dynamiclib"
 	VFLAGS="-current_version \$VERSION -compatibility_version \$MAJOR"
 
 	rm -f \$LIBRARY
@@ -1483,7 +1483,7 @@ LIBNAME=\$LIBRARY.so
 FULLNAME=\$LIBNAME.\$VERSION
 
 case "\$ACTION" in
-make)   FLAGS="$AC_CFLAGS -shared"
+make)   FLAGS="$AC_CFLAGS $AC_LDFLAGS -shared"
 	unset VFLAGS
 	test "$USE_SONAME" && VFLAGS="-Wl,-soname,\$LIBNAME.\$MAJOR"
 
-- 
2.29.2

