From 7c027d3961e237b520d405066befe298c877c0e7 Mon Sep 17 00:00:00 2001
From: Reilly Brogan <Reilly@reillybrogan.com>
Date: Mon, 21 Aug 2023 21:48:08 -0600
Subject: [PATCH] Fix build with gcc 10

---
 ui-gtk/UgtkBanner.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/ui-gtk/UgtkBanner.h b/ui-gtk/UgtkBanner.h
index 06fa0e5..600795d 100644
--- a/ui-gtk/UgtkBanner.h
+++ b/ui-gtk/UgtkBanner.h
@@ -64,7 +64,7 @@ struct UgtkBanner {
 		UgetRssFeed*    feed;
 		UgetRssItem*    item;
 	} rss;
-} banner;
+};
 
 void  ugtk_banner_init (UgtkBanner* banner);
 
-- 
2.41.0

