From c747c54ccf83fe9f0ea7633e7c4dd06a19d310a4 Mon Sep 17 00:00:00 2001
From: Ikey Doherty <ikey@solus-project.com>
Date: Sun, 3 Dec 2017 16:21:12 +0000
Subject: [PATCH] kernel: Port to 4.14 kernel_read API

Signed-off-by: Ikey Doherty <ikey@solus-project.com>
---
 kernel/nvidia/nv-gpu-numa.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/kernel/nvidia/nv-gpu-numa.c b/kernel/nvidia/nv-gpu-numa.c
index 1463d2d..2b2507b 100644
--- a/kernel/nvidia/nv-gpu-numa.c
+++ b/kernel/nvidia/nv-gpu-numa.c
@@ -112,6 +112,7 @@ static NV_STATUS bad_idea_read_string_from_file(const char *path_to_file,
 {
     struct file *filp;
     int read_count;
+    loff_t pos = 0;
 
     filp = filp_open(path_to_file, O_RDONLY, 0);
     if (IS_ERR(filp)) 
@@ -120,7 +121,7 @@ static NV_STATUS bad_idea_read_string_from_file(const char *path_to_file,
         return NV_ERR_NO_VALID_PATH;
     }
 
-    read_count = kernel_read(filp, 0, read_buffer, read_buffer_size - 1);
+    read_count = kernel_read(filp, read_buffer, read_buffer_size - 1, &pos);
 
     filp_close(filp, NULL);
 
-- 
2.15.1

