From 3adffc1d5302d293cda7fe0ef95c356c4e2d3b6c Mon Sep 17 00:00:00 2001
From: Joey Riches <josephriches@gmail.com>
Date: Sat, 28 May 2022 12:49:57 +0100
Subject: [PATCH 01/15] Silently ignore copy dt needed entries

---
 lld/ELF/Options.td | 1 +
 1 file changed, 1 insertion(+)

diff --git a/lld/ELF/Options.td b/lld/ELF/Options.td
index 93883d4c6ff5..898a19c1a157 100644
--- a/lld/ELF/Options.td
+++ b/lld/ELF/Options.td
@@ -815,6 +815,7 @@ def: Flag<["-"], "d">;
 def: Flag<["-"], "g">;
 def: F<"long-plt">;
 def: FF<"no-add-needed">;
+def: F<"copy-dt-needed-entries">;
 def: F<"no-copy-dt-needed-entries">;
 def: F<"no-ctors-in-init-array">;
 def: F<"no-keep-memory">;
-- 
2.52.0

