From 9d65b46ba0bfc3d177fe3452cfdeb70f6b5612f2 Mon Sep 17 00:00:00 2001
From: Gavin Zhao <git@gzgz.dev>
Date: Fri, 17 Jul 2026 21:06:33 -0700
Subject: [PATCH 11/15] Set Solus version string

Solus 21.1.8-135
---
 clang/CMakeLists.txt | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/clang/CMakeLists.txt b/clang/CMakeLists.txt
index f9e2a60831ea..e0d05d2890ca 100644
--- a/clang/CMakeLists.txt
+++ b/clang/CMakeLists.txt
@@ -356,7 +356,7 @@ set(CLANG_SYSTEMZ_DEFAULT_ARCH "z10" CACHE STRING "SystemZ Default Arch")
 set(CLANG_VENDOR ${PACKAGE_VENDOR} CACHE STRING
   "Vendor-specific text for showing with version information.")
 
-set(CLANG_REPOSITORY_STRING "@@SOLUS_VERSION_STRING@@" CACHE STRING
+set(CLANG_REPOSITORY_STRING "Solus 21.1.8-135" CACHE STRING
   "Vendor-specific text for showing the repository the source is taken from.")
 
 if(CLANG_REPOSITORY_STRING)
-- 
2.54.0

