diff --git a/catch/CMakeLists.txt b/catch/CMakeLists.txt
index baca620..7fc5a5a 100644
--- a/catch/CMakeLists.txt
+++ b/catch/CMakeLists.txt
@@ -159,7 +159,8 @@ if (WIN32)
 endif()
 
 if(HIP_PLATFORM STREQUAL "amd")
-    add_compile_options(-Wall -Wextra -Werror -Wno-deprecated -Wno-option-ignored)
+    add_compile_options(-w)
+    add_compile_definitions(HT_AMD)
 endif()
 
 cmake_policy(PUSH)
@@ -253,7 +254,7 @@ else()
         set(config_file ${CMAKE_SOURCE_DIR}/hipTestMain/config/${configToUse})
         set(config_gen_path ${CMAKE_CURRENT_BINARY_DIR}/${CATCH_BUILD_DIR}/hipTestMain/config)
         set(json_file ${config_gen_path}/${configToUse}_${arch}.json)
-        set(cmd "${ROCM_PATH}/llvm/bin/clang-cpp -P -D${arch} -DGITHASH=\\\"${HIP_VERSION_GITHASH}\\\" ${config_file}>${json_file}")
+        set(cmd "${ROCM_PATH}/lib64/llvm-rocm/bin/clang-cpp -P -D${arch} -DGITHASH=\\\"${HIP_VERSION_GITHASH}\\\" ${config_file}>${json_file}")
         message(${cmd})
         execute_process(COMMAND bash -c ${cmd}
         RESULT_VARIABLE json_result)
