From 51182c681bb792ec9163293ffc0c3eba4c97b607 Mon Sep 17 00:00:00 2001
From: Peter O'Connor <peter@solus-project.com>
Date: Fri, 12 Jan 2018 20:35:05 +1100
Subject: [PATCH] Include stateless samba directory

Signed-off-by: Peter O'Connor <peter@solus-project.com>
---
 src/core/ksambashare.cpp | 1 +
 1 file changed, 1 insertion(+)

diff --git a/src/core/ksambashare.cpp b/src/core/ksambashare.cpp
index 3940fec..37f4e74 100644
--- a/src/core/ksambashare.cpp
+++ b/src/core/ksambashare.cpp
@@ -41,6 +41,7 @@
 static const char *const DefaultSambaConfigFilePathList[] = {
     "/etc/samba/smb.conf",
     "/etc/smb.conf",
+    "/usr/share/defaults/samba/smb.conf",
     "/usr/local/etc/smb.conf",
     "/usr/local/samba/lib/smb.conf",
     "/usr/samba/lib/smb.conf",
-- 
2.15.1

