From 09d0a8b27484663a59ad9710392805b00c147017 Mon Sep 17 00:00:00 2001
From: "flinux@vongellhorn.ch" <flinux@vongellhorn.ch>
Date: Sun, 13 Oct 2019 19:43:02 +0200
Subject: [PATCH 1/1] Include stateless samba directory

---
 src/core/ksambashare.cpp | 1 +
 1 file changed, 1 insertion(+)

diff --git a/src/core/ksambashare.cpp b/src/core/ksambashare.cpp
index e09d6cf..8520e06 100644
--- a/src/core/ksambashare.cpp
+++ b/src/core/ksambashare.cpp
@@ -44,6 +44,7 @@ Q_LOGGING_CATEGORY(KIO_CORE_SAMBASHARE, "kf5.kio.core.sambashare", QtWarningMsg)
 static const char *const DefaultSambaConfigFilePathList[] = {
     "/etc/samba/smb.conf",
     "/etc/smb.conf",
+    "/usr/share/defaults/samba/smb.conf",
     "/usr/local/etc/smb.conf",
     "/usr/local/samba/lib/smb.conf",
     "/usr/samba/lib/smb.conf",
-- 
2.23.0

