From de147a539ade8bde5121d9b70fb3c78bdf12ebbd Mon Sep 17 00:00:00 2001
From: Peter O'Connor <peter@solus-project.com>
Date: Thu, 13 Jul 2017 14:54:28 +1000
Subject: [PATCH] Enable standard directories as default places

Signed-off-by: Peter O'Connor <peter@solus-project.com>
---
 src/filewidgets/kfileplacesmodel.cpp | 18 ++++++++++++++++++
 1 file changed, 18 insertions(+)

diff --git a/src/filewidgets/kfileplacesmodel.cpp b/src/filewidgets/kfileplacesmodel.cpp
index 4fc5fc1..d667b12 100644
--- a/src/filewidgets/kfileplacesmodel.cpp
+++ b/src/filewidgets/kfileplacesmodel.cpp
@@ -108,6 +108,24 @@ KFilePlacesModel::KFilePlacesModel(QObject *parent)
                                               QStringLiteral("Home"), I18N_NOOP2("KFile System Bookmarks", "Home"),
                                               QUrl::fromLocalFile(QDir::homePath()), QStringLiteral("user-home"));
         KFilePlacesItem::createSystemBookmark(d->bookmarkManager,
+                                              QStringLiteral("Desktop"), I18N_NOOP2("KFile System Bookmarks", "Desktop"),
+                                              QUrl::fromLocalFile(QDir::homePath() + "/Desktop" ), QStringLiteral("user-desktop"));
+        KFilePlacesItem::createSystemBookmark(d->bookmarkManager,
+                                              QStringLiteral("Documents"), I18N_NOOP2("KFile System Bookmarks", "Documents"),
+                                              QUrl::fromLocalFile(QDir::homePath() + "/Documents" ), QStringLiteral("folder-documents"));
+        KFilePlacesItem::createSystemBookmark(d->bookmarkManager,
+                                              QStringLiteral("Downloads"), I18N_NOOP2("KFile System Bookmarks", "Downloads"),
+                                              QUrl::fromLocalFile(QDir::homePath() + "/Downloads" ), QStringLiteral("folder-download"));
+        KFilePlacesItem::createSystemBookmark(d->bookmarkManager,
+                                              QStringLiteral("Music"), I18N_NOOP2("KFile System Bookmarks", "Music"),
+                                              QUrl::fromLocalFile(QDir::homePath() + "/Music" ), QStringLiteral("folder-music"));
+        KFilePlacesItem::createSystemBookmark(d->bookmarkManager,
+                                              QStringLiteral("Pictures"), I18N_NOOP2("KFile System Bookmarks", "Pictures"),
+                                              QUrl::fromLocalFile(QDir::homePath() + "/Pictures" ), QStringLiteral("folder-pictures"));
+        KFilePlacesItem::createSystemBookmark(d->bookmarkManager,
+                                              QStringLiteral("Videos"), I18N_NOOP2("KFile System Bookmarks", "Videos"),
+                                              QUrl::fromLocalFile(QDir::homePath() + "/Videos" ), QStringLiteral("folder-pictures"));
+        KFilePlacesItem::createSystemBookmark(d->bookmarkManager,
                                               QStringLiteral("Network"), I18N_NOOP2("KFile System Bookmarks", "Network"),
                                               QUrl(QStringLiteral("remote:/")), QStringLiteral("network-workgroup"));
 #if defined(_WIN32_WCE)
-- 
2.13.3

