diff --git a/src/ioslaves/file/file_unix.cpp b/src/ioslaves/file/file_unix.cpp
index 87c47e7e740970421f1124d8cf9af123c4dedf0a..c0bc64354d09b8b16110fa5d2f3a216a14b306d6 100644
--- a/src/ioslaves/file/file_unix.cpp
+++ b/src/ioslaves/file/file_unix.cpp
@@ -328,6 +328,9 @@ inline static time_t stat_mtime(const QT_STATBUF &buf)
 static bool isOnCifsMount(const QString &filePath)
 {
     const auto mount = KMountPoint::currentMountPoints().findByPath(filePath);
+    if (!mount) {
+        return false;
+    }
     return mount->mountType() == QStringLiteral("cifs") || mount->mountType() == QStringLiteral("smb3");
 }
 
