From 80392799f102266be8ec1b97caace693b40f6ba3 Mon Sep 17 00:00:00 2001
From: Thomas Staudinger <Staudi.Kaos@gmail.com>
Date: Fri, 26 Aug 2022 00:17:14 +0200
Subject: [PATCH] Fix build with GCC 12

---
 test_util.cpp | 1 +
 1 file changed, 1 insertion(+)

diff --git a/test_util.cpp b/test_util.cpp
index 0909263..58b491d 100644
--- a/test_util.cpp
+++ b/test_util.cpp
@@ -2,6 +2,7 @@
 #include <math.h>
 #include <stdio.h>
 #include <algorithm>
+#include <memory>
 #include <epoxy/gl.h>
 #include <gtest/gtest.h>
 #include <gtest/gtest-message.h>
-- 
2.35.4

