From 1f483e1d335374482845d0325ac8b13af3162c53 Mon Sep 17 00:00:00 2001
From: OpenSauce04 <opensauce04@gmail.com>
Date: Tue, 21 Oct 2025 14:08:29 +0100
Subject: [PATCH] cmake: Fixed build issues on *nix introduced by Qt 6.10.0

---
 src/citra_qt/CMakeLists.txt | 11 ++++++-----
 1 file changed, 6 insertions(+), 5 deletions(-)

diff --git a/src/citra_qt/CMakeLists.txt b/src/citra_qt/CMakeLists.txt
index 4796fc922e..7dc2b777b0 100644
--- a/src/citra_qt/CMakeLists.txt
+++ b/src/citra_qt/CMakeLists.txt
@@ -279,11 +279,12 @@ endif()
 
 if (NOT WIN32)
     find_package(Qt6 REQUIRED COMPONENTS Core Gui Widgets)
-    target_include_directories(citra_qt PRIVATE ${Qt6Gui_PRIVATE_INCLUDE_DIRS})
-endif()
-
-if (APPLE)
-    target_link_libraries(citra_qt PRIVATE Qt6::GuiPrivate)
+    if(Qt6_VERSION VERSION_GREATER_EQUAL "6.10.0")
+        find_package(Qt6 REQUIRED COMPONENTS GuiPrivate)
+        target_link_libraries(citra_qt PRIVATE Qt6::GuiPrivate)
+    else()
+        target_include_directories(citra_qt PRIVATE ${Qt6Gui_PRIVATE_INCLUDE_DIRS})
+    endif()
 endif()
 
 if (UNIX AND NOT APPLE)
