From 02faa68210789af309c9b19dda74b06c4db06254 Mon Sep 17 00:00:00 2001
From: Peter O'Connor <peter@solus-project.com>
Date: Sat, 14 Apr 2018 20:05:38 +1000
Subject: [PATCH] Disable building unshipped code

Signed-off-by: Peter O'Connor <peter@solus-project.com>
---
 CMakeLists.txt | 2 --
 1 file changed, 2 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index f92c2b0..081d997 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -42,7 +42,6 @@ if(STATIC_LIBRARY)
     add_definitions(-DQT_PLUGIN)
     add_definitions(-DQT_STATICPLUGIN=1)
 else(STATIC_LIBRARY)
-    add_subdirectory(autotests)
 endif(STATIC_LIBRARY)
 
 ################# set KDE specific information #################
@@ -111,7 +110,6 @@ add_definitions(-DQT_NO_URL_CAST_FROM_STRING)
 add_subdirectory(src)
 
 if (BUILD_EXAMPLES AND NOT STATIC_LIBRARY)
-    add_subdirectory(examples)
 endif()
 
 if (IS_DIRECTORY "${CMAKE_CURRENT_SOURCE_DIR}/po")
-- 
2.16.3

