From 596424c2e8e67012dacda6ae24bde86aacd1d528 Mon Sep 17 00:00:00 2001
From: Joshua Strobl <joshua@stroblindustries.com>
Date: Mon, 23 Apr 2018 07:56:26 +0300
Subject: [PATCH 1/1] Stop registering mimetype to open directories.

---
 data/org.gnome.baobab.desktop.in | 1 -
 1 file changed, 1 deletion(-)

diff --git a/data/org.gnome.baobab.desktop.in b/data/org.gnome.baobab.desktop.in
index d5290c3..2d0ced4 100644
--- a/data/org.gnome.baobab.desktop.in
+++ b/data/org.gnome.baobab.desktop.in
@@ -10,7 +10,6 @@ Icon=baobab
 Terminal=false
 Type=Application
 StartupNotify=true
-MimeType=inode/directory;
 Categories=GTK;GNOME;System;Filesystem;X-GNOME-Utilities;
 NotShowIn=KDE;
 DBusActivatable=true
-- 
2.16.3

