From 6dfd544024b1e5b20a92770ebce8db718a4c398e Mon Sep 17 00:00:00 2001
From: Ikey Doherty <ikey@solus-project.com>
Date: Fri, 17 Jul 2015 01:06:09 +0100
Subject: [PATCH] Deal with lightdm's weirdess by being GNOME. Vs Budgie:GNOME

---
 data/budgie-desktop.desktop.in | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/data/budgie-desktop.desktop.in b/data/budgie-desktop.desktop.in
index 75ecd01..8c166a5 100644
--- a/data/budgie-desktop.desktop.in
+++ b/data/budgie-desktop.desktop.in
@@ -5,4 +5,4 @@ Exec=@prefix@/bin/budgie-session
 TryExec=@prefix@/bin/budgie-session
 Icon=
 Type=Application
-DesktopNames=Budgie;GNOME
+DesktopNames=GNOME
-- 
2.4.6

