From e6d9939658c919f71cc2366a6ec1440b752edc5f Mon Sep 17 00:00:00 2001
From: Ikey Doherty <ikey@evolve-os.com>
Date: Sun, 8 Mar 2015 20:46:59 +0000
Subject: [PATCH 1/2] session: Force "gnome" as XDG_CURRENT_DESKTOP when absent
 (fix Qt apps)

---
 session/BudgieSession.vala | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/session/BudgieSession.vala b/session/BudgieSession.vala
index af60764..d8d6c62 100644
--- a/session/BudgieSession.vala
+++ b/session/BudgieSession.vala
@@ -72,6 +72,10 @@ public class Session : GLib.Application
             }
         }
 
+        if (Environment.get_variable("XDG_CURRENT_DESKTOP") == null) {
+            Environment.set_variable("XDG_CURRENT_DESKTOP", "gnome", true);
+        }
+
         if (should_logout) {
             do_logout();
             return 0;
-- 
1.8.4

