From 01e1f2bd59cc6d6cc1b00df3e8c42cd9fbeaf83d Mon Sep 17 00:00:00 2001
From: David Mohammed <fossfreedom@users.noreply.github.com>
Date: Wed, 20 Jul 2022 10:31:27 +0100
Subject: [PATCH] Obtain custom timezone using the schema key name (resolves
 #164) (#165)

---
 src/applets/clock/ClockApplet.vala | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/applets/clock/ClockApplet.vala b/src/applets/clock/ClockApplet.vala
index 5c1c666a..a520f02c 100644
--- a/src/applets/clock/ClockApplet.vala
+++ b/src/applets/clock/ClockApplet.vala
@@ -211,7 +211,7 @@ public class ClockApplet : Budgie.Applet {
 				case "use-custom-timezone":
 				case "custom-timezone":
 					if (settings.get_boolean("use-custom-timezone")) {
-						string custom_tz = settings.get_string("custom_timezone");
+						string custom_tz = settings.get_string("custom-timezone");
 						try {
 							this.clock_timezone = new TimeZone.identifier(custom_tz);
 						} catch (Error e) {
