From 0ebf191b5664432c7aaef7f88088c49553e606d7 Mon Sep 17 00:00:00 2001
From: Ikey Doherty <ikey@solus-project.com>
Date: Thu, 23 Jun 2016 16:16:44 +0100
Subject: [PATCH 1/4] Hopefully fix update issue with battery

Signed-off-by: Ikey Doherty <ikey@solus-project.com>
---
 panel/applets/status/PowerIndicator.vala | 9 +++++----
 1 file changed, 5 insertions(+), 4 deletions(-)

diff --git a/panel/applets/status/PowerIndicator.vala b/panel/applets/status/PowerIndicator.vala
index ab3f063..ab004cb 100644
--- a/panel/applets/status/PowerIndicator.vala
+++ b/panel/applets/status/PowerIndicator.vala
@@ -15,14 +15,15 @@ public class BatteryIcon : Gtk.Image
     public unowned Up.Device battery { protected set; public get; }
 
     public BatteryIcon(Up.Device battery) {
-        this.battery = battery;
-        this.update_ui();
+        this.update_ui(battery);
     }
 
-    public void update_ui()
+    public void update_ui(Up.Device battery)
     {
         string tip;
 
+        this.battery = battery;
+
         // Determine the icon to use for this battery
         string image_name;
         if (battery.percentage <= 10) {
@@ -107,7 +108,7 @@ public class PowerIndicator : Gtk.Bin
     {
         if (devices.contains(device.serial)) {
             /* Treated as a change event */
-            devices.lookup(device.serial).update_ui();
+            devices.lookup(device.serial).update_ui(device);
             return;
         }
         if (!this.is_interesting(device)) {
-- 
2.8.4

