From 14489d8c62787a56c04632836406b5afea9d95c3 Mon Sep 17 00:00:00 2001
From: Evan Maddock <maddock.evan@vivaldi.net>
Date: Sun, 28 Jan 2024 12:26:22 -0500
Subject: [PATCH] sendto: Use stateless path for autostart directory

Signed-off-by: Evan Maddock <maddock.evan@vivaldi.net>
---
 src/dialogs/sendto/meson.build | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/dialogs/sendto/meson.build b/src/dialogs/sendto/meson.build
index e9e92357..774a50e8 100644
--- a/src/dialogs/sendto/meson.build
+++ b/src/dialogs/sendto/meson.build
@@ -36,7 +36,7 @@ executable(
 
 install_data(
     'org.buddiesofbudgie.sendto-daemon.desktop',
-    install_dir: join_paths(confdir, 'xdg', 'autostart')
+    install_dir: xdg_appdir
 )
 
 install_data(
-- 
2.43.0

