From b992485b812a0f103ba744a7721c34c8afe63377 Mon Sep 17 00:00:00 2001
From: Ikey Doherty <ikey@solus-project.com>
Date: Sun, 28 Feb 2016 00:57:34 +0000
Subject: [PATCH 1/2] applets/tray: Force the correct colours

Signed-off-by: Ikey Doherty <ikey@solus-project.com>
---
 panel/applets/tray/TrayApplet.vala | 14 ++++++++++++++
 1 file changed, 14 insertions(+)

diff --git a/panel/applets/tray/TrayApplet.vala b/panel/applets/tray/TrayApplet.vala
index 6f8ba54..aa45d4b 100644
--- a/panel/applets/tray/TrayApplet.vala
+++ b/panel/applets/tray/TrayApplet.vala
@@ -95,6 +95,19 @@ public class TrayApplet : Budgie.Applet
             return;
         }
         tray.set_icon_size(icon_size);
+
+        Gdk.Color fg;
+        Gdk.Color warning;
+        Gdk.Color error;
+        Gdk.Color success;
+
+        Gdk.Color.parse("white", out fg);
+        Gdk.Color.parse("red", out error);
+        Gdk.Color.parse("orange", out warning);
+        Gdk.Color.parse("white", out success);
+
+        tray.set_colors(fg, error, warning, success);
+
         tray.set_padding(5);
         box.add(tray);
         show_all();
@@ -104,6 +117,7 @@ public class TrayApplet : Budgie.Applet
             return;
         }
         win.queue_draw();
+        tray.force_redraw();
     }
 }
 
-- 
2.7.1

