From b1f9f5b808b5b6c8e70f6677a0ee22dab555017b Mon Sep 17 00:00:00 2001
From: Ikey Doherty <ikey@solus-project.com>
Date: Sat, 12 Mar 2016 15:52:44 +0000
Subject: [PATCH] Support stateless configuration

Signed-off-by: Ikey Doherty <ikey@solus-project.com>
---
 configure.ac        | 16 ++++++++++++++++
 session/Makefile.am |  4 ++--
 2 files changed, 18 insertions(+), 2 deletions(-)

diff --git a/configure.ac b/configure.ac
index 8ed0578..ed77eb6 100644
--- a/configure.ac
+++ b/configure.ac
@@ -117,6 +117,19 @@ GTK_DOC_CHECK([1.14],[--flavour no-tmpl])
 AM_CONDITIONAL([ENABLE_GTK_DOC], false)
 ])
 
+AC_ARG_ENABLE(stateless, AS_HELP_STRING([--enable-stateless], [enable stateless configuration]))
+if test "x$enable_stateless" = "xyes" ; then
+    AC_DEFINE([STATELESS], [1], [Using stateless configuration])
+    adl_RECURSIVE_EVAL(["${datadir}/xdg"], [XDGDIR])
+else
+    enable_stateless="no"
+    adl_RECURSIVE_EVAL(["${sysconfdir}/xdg"], [XDGDIR])
+fi
+
+AC_SUBST(XDGDIR)
+
+AM_CONDITIONAL([STATELESS_CONFIG], [test x$enable_stateless = "xyes"])
+
 AC_CONFIG_FILES([Makefile
                 config/Makefile
                 data/budgie-panel.desktop
@@ -168,6 +181,9 @@ AC_MSG_RESULT([
         cflags:                 ${CFLAGS}
         ldflags:                ${LDFLAGS}
 
+        stateless:              ${enable_stateless}
+        xdg directory:          ${XDGDIR}
+
         Plugins:
         =======
 
diff --git a/session/Makefile.am b/session/Makefile.am
index 42387c1..e479b5e 100644
--- a/session/Makefile.am
+++ b/session/Makefile.am
@@ -38,8 +38,8 @@ gnomesessiondir = $(datadir)/gnome-session/sessions
 gnomesession_DATA = \
 	budgie-desktop.session
 
-# TODO: Permit stateless on Solus
-autostartdir = $(sysconfdir)/xdg/autostart
+autostartdir = $(XDGDIR)/autostart
+
 autostart_DATA = \
 	budgie-desktop-nm-applet.desktop
 
-- 
2.7.1

