From 88895c5769d8090ddda1bc74deab64ea7f485469 Mon Sep 17 00:00:00 2001
From: Ikey Doherty <ikey@solus-project.com>
Date: Tue, 28 Jun 2016 18:19:05 +0100
Subject: [PATCH 2/4] Forcibly redraw the power applet

Signed-off-by: Ikey Doherty <ikey@solus-project.com>
---
 panel/applets/status/PowerIndicator.vala | 1 +
 1 file changed, 1 insertion(+)

diff --git a/panel/applets/status/PowerIndicator.vala b/panel/applets/status/PowerIndicator.vala
index ab004cb..ccf8603 100644
--- a/panel/applets/status/PowerIndicator.vala
+++ b/panel/applets/status/PowerIndicator.vala
@@ -63,6 +63,7 @@ public class BatteryIcon : Gtk.Image
 
         set_tooltip_text(tip);
         set_from_icon_name(image_name, Gtk.IconSize.MENU);
+        this.queue_draw();
     }
 }
 
-- 
2.8.4

