From 762b9856a71c8229324a4fec1a1992c4f12b7ae6 Mon Sep 17 00:00:00 2001
From: Ikey Doherty <ikey@solus-project.com>
Date: Sun, 3 Jul 2016 18:34:17 +0100
Subject: [PATCH 2/2] raven: Actually update the date so I don't miss my rent

Signed-off-by: Ikey Doherty <ikey@solus-project.com>
---
 raven/calendar.vala | 17 +++++++++++++----
 1 file changed, 13 insertions(+), 4 deletions(-)

diff --git a/raven/calendar.vala b/raven/calendar.vala
index 7ad6010..a305618 100644
--- a/raven/calendar.vala
+++ b/raven/calendar.vala
@@ -13,16 +13,15 @@ public class CalendarWidget : Gtk.Box
 {
 
     private Gtk.Calendar? cal = null;
+    private Budgie.HeaderWidget? header = null;
 
-    private const string date_format = "%e %b %Y";
+    private static const string date_format = "%e %b %Y";
 
     public CalendarWidget()
     {
         Object(orientation: Gtk.Orientation.VERTICAL);
-        /* TODO: Fix icon */
-
         var time = new DateTime.now_local();
-        var header = new Budgie.HeaderWidget(time.format(date_format), "x-office-calendar-symbolic", false);
+        header = new Budgie.HeaderWidget(time.format(date_format), "x-office-calendar-symbolic", false);
         var expander = new Budgie.RavenExpander(header);
         this.pack_start(expander, false, false, 0);
 
@@ -32,6 +31,16 @@ public class CalendarWidget : Gtk.Box
         ebox.get_style_context().add_class("raven-background");
         ebox.add(cal);
         expander.add(ebox);
+
+        Timeout.add_seconds_full(GLib.Priority.LOW, 30, this.update_date);
+    }
+
+    private bool update_date()
+    {
+        var time = new DateTime.now_local();
+        var strf = time.format(date_format);
+        header.text = strf;
+        return true;
     }
 
 } // End class
-- 
2.9.0

