From 6bfadb2ab7d4ef7f8f0bd7859070d29ac64526ad Mon Sep 17 00:00:00 2001
From: Ikey Doherty <ikey@solus-project.com>
Date: Tue, 12 Jul 2016 15:30:23 +0100
Subject: [PATCH 2/5] theme: Ensure we link against GTK (issue #465)

Signed-off-by: Ikey Doherty <ikey@solus-project.com>
---
 theme/Makefile.am | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/theme/Makefile.am b/theme/Makefile.am
index a93d0cd..f284668 100644
--- a/theme/Makefile.am
+++ b/theme/Makefile.am
@@ -67,6 +67,9 @@ libbudgietheme_la_SOURCES = \
 libbudgietheme_la_CFLAGS = \
 	$(BUDGIE_BASE_CFLAGS)
 
+libbudgietheme_la_LIBADD = \
+	$(BUDGIE_BASE_LIBS)
+
 # Theme headers
 budgie_devdir = $(includedir)/budgie-desktop
 budgie_dev_HEADERS = \
-- 
2.9.1

