From 772ae3a3232d48738177984e3e54dc0b3bf7b60e Mon Sep 17 00:00:00 2001
From: Ikey Doherty <ikey@evolve-os.com>
Date: Mon, 9 Mar 2015 04:38:55 +0000
Subject: [PATCH 2/2] session: Make Java, Google Chrome and XDG happy by lying
 a bit

---
 session/BudgieSession.vala | 8 ++++++++
 1 file changed, 8 insertions(+)

diff --git a/session/BudgieSession.vala b/session/BudgieSession.vala
index d8d6c62..7cb0b23 100644
--- a/session/BudgieSession.vala
+++ b/session/BudgieSession.vala
@@ -72,10 +72,18 @@ public class Session : GLib.Application
             }
         }
 
+        // Make Qt, etc, happy
         if (Environment.get_variable("XDG_CURRENT_DESKTOP") == null) {
             Environment.set_variable("XDG_CURRENT_DESKTOP", "gnome", true);
         }
 
+        /* Make Java, Google Chrome, and xdg-open happy.
+           Further more, whoever thought falling back to a web browser in xdg-open
+           was a good idea... MASSIVE ROUND OF APPLAUSE.... */
+        if (Environment.get_variable("GNOME_DESKTOP_SESSION_ID") == null) {
+            Environment.set_variable("GNOME_DESKTOP_SESSION_ID", "this-is-deprecated", true);
+        }
+
         if (should_logout) {
             do_logout();
             return 0;
-- 
1.8.4

