From 6a0460abb5f998f4e733dad17e3bb93631e4b41a Mon Sep 17 00:00:00 2001
From: Ikey Doherty <ikey@solus-project.com>
Date: Tue, 28 Jun 2016 18:41:04 +0100
Subject: [PATCH 3/4] Try to brute force the notify signal

Signed-off-by: Ikey Doherty <ikey@solus-project.com>
---
 panel/applets/status/PowerIndicator.vala | 9 +++++++++
 1 file changed, 9 insertions(+)

diff --git a/panel/applets/status/PowerIndicator.vala b/panel/applets/status/PowerIndicator.vala
index ccf8603..548952f 100644
--- a/panel/applets/status/PowerIndicator.vala
+++ b/panel/applets/status/PowerIndicator.vala
@@ -16,6 +16,15 @@ public class BatteryIcon : Gtk.Image
 
     public BatteryIcon(Up.Device battery) {
         this.update_ui(battery);
+
+        battery.notify.connect(this.on_battery_change);
+    }
+
+    private void on_battery_change(Object o, ParamSpec sp)
+    {
+        message("BATTERY CHANGE");
+        this.battery.refresh_sync(null);
+        this.update_ui(this.battery);
     }
 
     public void update_ui(Up.Device battery)
-- 
2.8.4

