From 449d0a65cc1a0c109f1c725867ba1dcb899f83ae Mon Sep 17 00:00:00 2001
From: Ikey Doherty <ikey@solus-project.com>
Date: Tue, 28 Jun 2016 18:42:29 +0100
Subject: [PATCH 4/4] Remove spam and clean it up

Signed-off-by: Ikey Doherty <ikey@solus-project.com>
---
 panel/applets/status/PowerIndicator.vala | 7 ++++++-
 1 file changed, 6 insertions(+), 1 deletion(-)

diff --git a/panel/applets/status/PowerIndicator.vala b/panel/applets/status/PowerIndicator.vala
index 548952f..1452ed2 100644
--- a/panel/applets/status/PowerIndicator.vala
+++ b/panel/applets/status/PowerIndicator.vala
@@ -13,6 +13,7 @@ public class BatteryIcon : Gtk.Image
 {
     /** The battery associated with this icon */
     public unowned Up.Device battery { protected set; public get; }
+    bool changing = false;
 
     public BatteryIcon(Up.Device battery) {
         this.update_ui(battery);
@@ -22,9 +23,13 @@ public class BatteryIcon : Gtk.Image
 
     private void on_battery_change(Object o, ParamSpec sp)
     {
-        message("BATTERY CHANGE");
+        if (this.changing) {
+            return;
+        }
+        this.changing = true;
         this.battery.refresh_sync(null);
         this.update_ui(this.battery);
+        this.changing = false;
     }
 
     public void update_ui(Up.Device battery)
-- 
2.8.4

