From dc72ca8d734d3d2a33c1f892f147fe5bb711c748 Mon Sep 17 00:00:00 2001
From: Ikey Doherty <ikey@solus-project.com>
Date: Wed, 19 Oct 2016 20:36:56 +0100
Subject: [PATCH 6/6] Ensure we build again..

Signed-off-by: Ikey Doherty <ikey@solus-project.com>
---
 panel/Makefile.am  | 1 +
 polkit/Makefile.am | 1 +
 2 files changed, 2 insertions(+)

diff --git a/panel/Makefile.am b/panel/Makefile.am
index 3a3a2df..0eac8d0 100644
--- a/panel/Makefile.am
+++ b/panel/Makefile.am
@@ -70,6 +70,7 @@ budgie_panel_LDADD = \
 
 budgie_panel_VALAFLAGS = \
 	--pkg gtk+-3.0 \
+	--pkg gdk-x11-3.0 \
 	--pkg gio-unix-2.0 \
 	--pkg libuuid \
 	--pkg libpeas-gtk-1.0 \
diff --git a/polkit/Makefile.am b/polkit/Makefile.am
index 4858f64..df6a22f 100644
--- a/polkit/Makefile.am
+++ b/polkit/Makefile.am
@@ -62,6 +62,7 @@ applications_DATA = \
 
 budgie_polkit_dialog_VALAFLAGS = \
 	--pkg gtk+-3.0 \
+	--pkg gdk-x11-3.0 \
 	--pkg polkit-gobject-1 \
 	--pkg polkit-agent-1 \
 	--pkg posix \
-- 
2.10.1

