From 5a56d6dd36f42ca6da1eb753f5f0cb44f1abab26 Mon Sep 17 00:00:00 2001
From: Campbell Jones <git@serebit.com>
Date: Tue, 8 Dec 2020 22:47:18 -0500
Subject: [PATCH] Fix issue with KeePass causing panel segfaults (#2072)

---
 src/applets/tray/carbontray/child.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/applets/tray/carbontray/child.c b/src/applets/tray/carbontray/child.c
index 52a117cf..97be14d5 100644
--- a/src/applets/tray/carbontray/child.c
+++ b/src/applets/tray/carbontray/child.c
@@ -168,7 +168,7 @@ static void carbon_child_class_init(CarbonChildClass* klass) {
 }
 
 static bool set_wmclass(CarbonChild* self, Display* xdisplay) {
-	XClassHint ch;
+	XClassHint ch = {};
 
 	GdkDisplay* display = gdk_display_get_default();
 	gdk_x11_display_error_trap_push(display);
