From 3bcd2fff2e2e78349c1629ef9a4fd28d8296b834 Mon Sep 17 00:00:00 2001
From: Joshua Strobl <me@joshuastrobl.com>
Date: Tue, 6 Feb 2024 22:34:06 +0200
Subject: [PATCH] fix: set budgie run dialog to skip pager and taskbar

fixes #529
---
 src/dialogs/run/RunDialog.vala | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/src/dialogs/run/RunDialog.vala b/src/dialogs/run/RunDialog.vala
index 5138e50a5..faa3be5c0 100644
--- a/src/dialogs/run/RunDialog.vala
+++ b/src/dialogs/run/RunDialog.vala
@@ -44,6 +44,8 @@ namespace Budgie {
 		construct {
 			set_keep_above(true);
 			set_position(Gtk.WindowPosition.CENTER);
+			set_skip_pager_hint(true);
+			set_skip_taskbar_hint(true);
 			Gdk.Visual? visual = screen.get_rgba_visual();
 			if (visual != null) {
 				this.set_visual(visual);
