diff --git a/wm/core.c b/wm/core.c
index 9515318..b3220e7 100644
--- a/wm/core.c
+++ b/wm/core.c
@@ -107,7 +107,7 @@ static void budgie_wm_dispose(GObject *object)
 
         /* Any stray lists the tab module might have */
         tabs_clean();
-        budgie_keys_end();
+        //budgie_keys_end();
         budgie_menus_end(BUDGIE_WM(object));
 
         G_OBJECT_CLASS(budgie_wm_parent_class)->dispose(object);
@@ -163,7 +163,7 @@ static void budgie_wm_start(MetaPlugin *plugin)
                 (MetaKeyHandlerFunc)switch_windows, self, NULL);
 
         /* Handle keys.. */
-        budgie_keys_init(meta_screen_get_display(screen));
+        //budgie_keys_init(meta_screen_get_display(screen));
         budgie_menus_init(self);
         g_signal_connect(meta_screen_get_display(screen), "overlay-key",
             G_CALLBACK(overlay_cb), NULL);
