From dbdc01ef25af12b1b7cc4800b0d49ca1195aa51d Mon Sep 17 00:00:00 2001
From: Algent Albrahimi <algent@protonmail.com>
Date: Sat, 3 Dec 2022 13:56:01 +0100
Subject: [PATCH] Use qpdf 11.2.0

Signed-off-by: Algent Albrahimi <algent@protonmail.com>
---
 pyproject.toml | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/pyproject.toml b/pyproject.toml
index c044a00..63c3062 100644
--- a/pyproject.toml
+++ b/pyproject.toml
@@ -49,7 +49,7 @@ skip = ["cp36-*", "pp39-*", "*musllinux*", "pp*-manylinux*_aarch64", "cp311-many
 test-skip = "*_arm64 *_universal2:arm64"
 
 [tool.cibuildwheel.environment]
-QPDF_MIN_VERSION = "11.1.0"
+QPDF_MIN_VERSION = "11.2.0"
 QPDF_VERSION = "11.1.1"
 QPDF_PATTERN = "https://github.com/qpdf/qpdf/releases/download/vVERSION/qpdf-VERSION.tar.gz"
 JPEG_RELEASE = "https://www.ijg.org/files/jpegsrc.v9e.tar.gz"
-- 
2.35.5

