From a35004e01268fc57a3a5a7c7086f34907546443f Mon Sep 17 00:00:00 2001
From: Algent Albrahimi <algent@protonmail.com>
Date: Sat, 26 Jun 2021 14:18:00 +0200
Subject: [PATCH] Fix tests in setup.cfg

---
 setup.cfg | 2 --
 1 file changed, 2 deletions(-)

diff --git a/setup.cfg b/setup.cfg
index 102b503..6e94d83 100644
--- a/setup.cfg
+++ b/setup.cfg
@@ -80,8 +80,6 @@ test=pytest
 [tool:pytest]
 norecursedirs = lib .pc .git venv
 testpaths = tests
-addopts =
-	-n auto
 
 [coverage:run]
 
-- 
2.32.0

