From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Reilly Brogan <reilly@reillybrogan.com>
Date: Wed, 26 Jun 2024 23:11:02 -0500
Subject: [PATCH] Fix compat with MINIUPNPC API Version 18

---
 source/netclient.c | 21 +++++++++++++++++++--
 1 file changed, 19 insertions(+), 2 deletions(-)

diff --git a/source/netclient.c b/source/netclient.c
index 74167ec..8e1e02b 100644
--- a/source/netclient.c
+++ b/source/netclient.c
@@ -507,8 +507,21 @@ static Q_BOOL upnp_init() {
      *
      */
     device_list = upnpDiscover(2000, NULL, NULL, 0, 0, 2, NULL);
-#    else
-#      error "Unsupported miniupnpc API version.  Consider disabling UPNP (--disable-upnp), or file an issue at https://github.com/klamonte/qodem/issues" .
+#      else
+#        if (MINIUPNPC_API_VERSION == 18)
+    /*
+     * Version 18
+     *
+     * upnpDiscover(int delay, const char * multicastif,
+     *              const char * minissdpdsock, int localport,
+     *              int ipv6, unsigned char ttl,
+     *              int * error);
+     *
+     */
+    device_list = upnpDiscover(2000, NULL, NULL, 0, 0, 2, NULL);
+#      else
+#        error "Unsupported miniupnpc API version.  Consider disabling UPNP (--disable-upnp), or file an issue at https://github.com/klamonte/qodem/issues" .
+#        endif
 #      endif
 #    endif
 #  endif
@@ -517,7 +530,11 @@ static Q_BOOL upnp_init() {
     if (device_list != NULL) {
 
         rc = UPNP_GetValidIGD(device_list, &upnp_urls, &upnp_igd_datas,
+#if (MINIUPNPC_API_VERSION >= 18)
+                              local_host, sizeof(local_host), NULL, 0);
+#else
                               local_host, sizeof(local_host));
+#endif
 
         switch (rc) {
 
