From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Ikey Doherty <ikey@solus-project.com>
Date: Fri, 16 Dec 2016 05:04:48 +0000
Subject: [PATCH] Support stateless glibc ldconfig configuration

Signed-off-by: Ikey Doherty <ikey@solus-project.com>
---
 dracut-init.sh | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/dracut-init.sh b/dracut-init.sh
index 1ce0d7ed..28c935a6 100755
--- a/dracut-init.sh
+++ b/dracut-init.sh
@@ -579,7 +579,7 @@ inst_rules_wildcard() {
 build_ld_cache() {
     local dstdir="${dstdir:-"$initdir"}"
 
-    for f in "$dracutsysrootdir"/etc/ld.so.conf "$dracutsysrootdir"/etc/ld.so.conf.d/*; do
+    for f in "$dracutsysrootdir"/etc/ld.so.conf "$dracutsysrootdir"/etc/ld.so.conf.d/* "$dracutsysrootdir"/usr/share/defaults/etc/ld.so.conf "$dracutsysrootdir"/usr/share/ld.so.conf.d/*; do
         [[ -f $f ]] && inst_simple "${f#"$dracutsysrootdir"}"
     done
     if ! $DRACUT_LDCONFIG -r "$initdir" -f /etc/ld.so.conf; then
