From 4cf0841a7869b6400877f350075188c610134a9e Mon Sep 17 00:00:00 2001
From: Reilly Brogan <reilly@reillybrogan.com>
Date: Mon, 11 Dec 2023 14:38:23 -0600
Subject: [PATCH] Solus: De-compress kbd files

---
 modules.d/10i18n/module-setup.sh | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/modules.d/10i18n/module-setup.sh b/modules.d/10i18n/module-setup.sh
index ac456112..710be3db 100755
--- a/modules.d/10i18n/module-setup.sh
+++ b/modules.d/10i18n/module-setup.sh
@@ -142,6 +142,10 @@ install() {
         for _src in "${KBDSUBDIRS[@]}"; do
             inst_dir "${kbddir}/$_src"
             $DRACUT_CP -L -t "${initdir}/${kbddir}/$_src" "${dracutsysrootdir}${kbddir}/$_src"/*
+
+            # Solus: De-compress everything since we compress the initrd anyway
+            find "${initdir}${kbddir}/$_src" -name '*.gz' -print0 | \
+                xargs -0 gzip -f -d
         done
 
         # remove unnecessary files
-- 
2.43.0

