From 00412b82d11e2ddd977185c2a3cf383c9f4b9691 Mon Sep 17 00:00:00 2001
From: Peter O'Connor <peter@solus-project.com>
Date: Thu, 18 Oct 2018 18:17:05 +1100
Subject: [PATCH] Ensure modprobe.d directory exists before copying file

Signed-off-by: Peter O'Connor <peter@solus-project.com>
---
 hdajackretask/apply-changes.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/hdajackretask/apply-changes.c b/hdajackretask/apply-changes.c
index aa291ce..66642bb 100644
--- a/hdajackretask/apply-changes.c
+++ b/hdajackretask/apply-changes.c
@@ -253,6 +253,7 @@ static const gchar* retask_conf =
 static const gchar* install_script =
 "#!/bin/sh\n"
 "mv %s/hda-jack-retask.fw /lib/firmware/hda-jack-retask.fw\n 2>>%s\n"
+"mkdir -p /etc/modprobe.d\n"
 "mv %s/hda-jack-retask.conf /etc/modprobe.d/hda-jack-retask.conf 2>>%s\n";
 
 gboolean apply_changes_boot(pin_configs_t* pins, int entries, int card, int device, 
--
2.19.1
