diff --git a/build.gradle b/build.gradle
index 7c79298..7106d95 100644
--- a/build.gradle
+++ b/build.gradle
@@ -195,13 +195,13 @@ def rpm(def platform, def arch) {
 }
 
 packageTask('linux', [], 'ext/rocksaw/lib/linux/librocksaw.so') {
-    deb('linux', 'i386')
-    rpm('linux', 'i386')
+    //deb('linux', 'i386')
+    //rpm('linux', 'i386')
 }
 
 packageTask('linux64', [], 'ext/rocksaw/lib/linux64/librocksaw.so') {
-    deb('linux64', 'amd64')
-    rpm('linux64', 'x86_64')
+    //deb('linux64', 'amd64')
+    //rpm('linux64', 'x86_64')
 }
 
 packageTask('win32', 'lib/jna-win32.jar', 'ext/rocksaw/lib/rocksaw.dll') {
