diff --git a/build.gradle b/build.gradle
index f87bcb47..4f08e98f 100644
--- a/build.gradle
+++ b/build.gradle
@@ -98,20 +98,7 @@ def packageTask(def platform, def moreJars, def moreLibs, Closure doMore) {
         )
         with jar
 
-        def rtJar = new File(System.getProperty('java.home') + "/jre/lib/rt.jar")
-        if (!rtJar.exists()) rtJar = new File(System.getProperty('java.home') + "/lib/rt.jar") // support for Zulu JVM in Github workflow
-        if (!rtJar.exists()) rtJar = new File('/usr/lib/jvm/java-8-oracle/jre/lib/rt.jar') // try Debian/Ubuntu version of Java 8 if current Java doesn't have rt.jar
-        if (!rtJar.exists()) rtJar = new File('/usr/lib/jvm/java-1.8.0-openjdk-amd64/jre/lib/rt.jar') // try Debian/Ubuntu version of OpenJDK 8
-        // if (!rtJar.exists()) rtJar = new File('/usr/lib/jvm/java-11-openjdk-amd64/jmods/java.base.jmod') // Java 9+ needs more jmods (awt)
-
-        if (rtJar.exists() && platform != "any") {
-            println "Using $rtJar for proguard minimization"
-            finalizedBy minimizeTask(rtJar, platform, doMore)
-        }
-        else {
-            println "WARNING: not minimizing, Java8 rt.jar not found on the system"
-            doLast(doMore)
-        }
+        doLast(doMore)
     }
 }
 
@@ -194,12 +181,12 @@ def rpm(def platform, def arch) {
 }
 
 packageTask('linux64', [], 'ext/rocksaw/lib/linux64/librocksaw.so') {
-    deb('linux64', 'amd64')
-    rpm('linux64', 'x86_64')
+    //deb('linux64', 'amd64')
+    //rpm('linux64', 'x86_64')
 }
 
 packageTask('any', [], []) {
-    deb('any', 'all', 'libswt-gtk-4-java, libswt-cairo-gtk-4-jni,')
+    //deb('any', 'all', 'libswt-gtk-4-java, libswt-cairo-gtk-4-jni,')
 }
 
 packageTask('win32', 'lib/jna-win32.jar', 'ext/rocksaw/lib/rocksaw.dll') {
