From dc434b20fd16e297f7f203e5a0e0993e7d3a94d3 Mon Sep 17 00:00:00 2001
From: Ivan Trepakov <liontiger23@gmail.com>
Date: Sat, 16 May 2026 18:34:12 +0700
Subject: [PATCH] Gradle 9.x.x fixes

---
 build.gradle | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/build.gradle b/build.gradle
index 344e87f..58e19ab 100644
--- a/build.gradle
+++ b/build.gradle
@@ -29,7 +29,7 @@ String exec(def line, String dir = ".", boolean failOnError = true) {
 
 version = exec("git describe --tags", ".", false) ?: "unknown"
 
-sourceCompatibility = targetCompatibility = 17
+java.sourceCompatibility = java.targetCompatibility = 17
 
 def javaModules = ["java.base", "java.prefs", "java.logging", "jdk.crypto.ec"]
 project.ext.set("javaModules", javaModules)
-- 
2.54.0

