diff --git a/build.gradle b/build.gradle
index 61753185..71c1828c 100644
--- a/build.gradle
+++ b/build.gradle
@@ -27,7 +27,7 @@ String exec(def line, String dir = ".", boolean failOnError = true) {
     }
 }
 
-version = exec("git describe --tags", ".", false) ?: "unknown"
+version = ipscanVersion
 
 sourceCompatibility = 11
 targetCompatibility = 11
@@ -137,10 +137,7 @@ def packageTask(String platform, Closure doMore) {
         )
         with jar
 
-        if (platform != "any")
-            finalizedBy minimizeTask(platform, doMore)
-        else
-            doLast(doMore)
+        doLast(doMore)
     }
 }
 
@@ -253,12 +250,12 @@ def macLauncher(def platform) {
 }
 
 packageTask('linux64') {
-    deb('linux64', 'amd64')
-    rpm('linux64', 'x86_64')
+    //deb('linux64', 'amd64')
+    //rpm('linux64', 'x86_64')
 }
 
 packageTask('any') {
-    deb('any', 'all', 'libswt-gtk-4-java, libswt-cairo-gtk-4-jni,')
+    //deb('any', 'all', 'libswt-gtk-4-java, libswt-cairo-gtk-4-jni,')
 }
 
 packageTask('win64') {
