From d8f2754b28fa54bac7b04fbd82f8690b51d238ab Mon Sep 17 00:00:00 2001
From: Algent Albrahimi <algent@protonmail.com>
Date: Thu, 10 Mar 2022 18:45:11 +0100
Subject: [PATCH] Fix game version

Signed-off-by: Algent Albrahimi <algent@protonmail.com>
---
 src/opentyrian_version.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/opentyrian_version.h b/src/opentyrian_version.h
index 543cfa4..a67b946 100644
--- a/src/opentyrian_version.h
+++ b/src/opentyrian_version.h
@@ -2,7 +2,7 @@
 #define OPENTYRIAN_VERSION_H
 
 #ifndef OPENTYRIAN_VERSION
-#define OPENTYRIAN_VERSION "v2.1.???"
+#define OPENTYRIAN_VERSION "v2.1.20220309"
 #endif
 
 #endif // OPENTYRIAN_VERSION_H
-- 
2.35.1

