From 0aacf663e5377a0caab5fad70d760d9a667cad9d Mon Sep 17 00:00:00 2001
From: Peter O'Connor <peter@solus-project.com>
Date: Fri, 16 Mar 2018 16:13:04 +1100
Subject: [PATCH] Fix glibc 2.26 build

---
 src/framework/mlt_property.h | 6 ++++--
 1 file changed, 4 insertions(+), 2 deletions(-)

diff --git a/src/framework/mlt_property.h b/src/framework/mlt_property.h
index 404d513..39fefd9 100644
--- a/src/framework/mlt_property.h
+++ b/src/framework/mlt_property.h
@@ -30,8 +30,10 @@
 #include <sys/param.h>
 #endif
 
-#if defined(__GLIBC__) || defined(__APPLE__) || (__FreeBSD_version >= 900506)
-#include <xlocale.h>
+#if (defined(__GLIBC__) && !defined(__APPLE__)) || HAVE_LOCALE_H
+#  include <locale.h>
+#elif defined(__APPLE__) || (__FreeBSD_version >= 900506)
+#  include <xlocale.h>
 #else
 typedef char* locale_t;
 #endif
-- 
2.16.2

