From b8276cde66a1e786c3e3dfb0dbe571b4321fdea0 Mon Sep 17 00:00:00 2001
From: "F. von Gellhorn" <flinux@vongellhorn.ch>
Date: Sat, 28 Dec 2019 18:03:05 +0100
Subject: [PATCH 2/2] Fix make install for lumas

---
 src/modules/lumas/create_lumas | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/src/modules/lumas/create_lumas b/src/modules/lumas/create_lumas
index 7c25e83..a86c847 100755
--- a/src/modules/lumas/create_lumas
+++ b/src/modules/lumas/create_lumas
@@ -2,6 +2,12 @@
 
 [ \( -d PAL \) -a \( ! $0 -nt .executed \) ] && exit 0
 
+if [ "$(uname -s)" = "Darwin" ]; then
+	export DYLD_LIBRARY_PATH=$LD_LIBRARY_PATH:../../framework
+else
+	export LD_LIBRARY_PATH=$LD_LIBRARY_PATH:../../framework
+fi
+
 bpp=16
 [ -f .8bit ] && bpp=8
 
-- 
2.24.1

