From ca702da321716613efedc1d20ae13f8a35646fa1 Mon Sep 17 00:00:00 2001
From: Joshua Strobl <joshua@stroblindustries.com>
Date: Sat, 2 Sep 2017 12:39:29 +0300
Subject: [PATCH 1/1] Fix register URL. 404s without a slash.

---
 lutris/gui/lutriswindow.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/lutris/gui/lutriswindow.py b/lutris/gui/lutriswindow.py
index 01e671ea..0474b534 100644
--- a/lutris/gui/lutriswindow.py
+++ b/lutris/gui/lutriswindow.py
@@ -163,7 +163,7 @@ class LutrisWindow(Gtk.ApplicationWindow):
                 lambda *x: self._open_browser('https://lutris.net/games/')
             ),
             'register-account': Action(
-                lambda *x: self._open_browser('https://lutris.net/user/register')
+                lambda *x: self._open_browser('https://lutris.net/user/register/')
             ),
 
             'disconnect': Action(self.on_disconnect),
-- 
2.14.1

