From 9328c3268164f1201f7bdb3f6c28f4b7bd314200 Mon Sep 17 00:00:00 2001
From: Mathieu Comandon <strycore@gmail.com>
Date: Fri, 15 Jan 2021 13:42:17 -0800
Subject: [PATCH] Check for available DXVK versions

---
 lutris/util/wine/dxvk.py | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/lutris/util/wine/dxvk.py b/lutris/util/wine/dxvk.py
index 73682a772..31b361aee 100644
--- a/lutris/util/wine/dxvk.py
+++ b/lutris/util/wine/dxvk.py
@@ -53,7 +53,8 @@ def version(self):
         """Return version of DXVK (latest known version if not provided)"""
         if self._version:
             return self._version
-        return self.versions[0]
+        if self.versions:
+            return self.versions[0]
 
     @property
     def dxvk_path(self):
