From d7aabb70b991a8c9e262ad8adbe4bc33fa123f02 Mon Sep 17 00:00:00 2001
From: Muhammad Alfi Syahrin <malfisya.dev@hotmail.com>
Date: Thu, 25 Jan 2024 14:42:35 +0700
Subject: [PATCH] patch Makefile

---
 Makefile | 20 +++++++++++---------
 1 file changed, 11 insertions(+), 9 deletions(-)

diff --git a/Makefile b/Makefile
index a8d81b8..f3cfd83 100644
--- a/Makefile
+++ b/Makefile
@@ -31,7 +31,7 @@ DESTDIR ?= # root dir
 
 sharedir := $(DESTDIR)$(PREFIX)/share
 bindir := $(DESTDIR)$(PREFIX)/bin
-execdir := $(DESTDIR)$(PREFIX)/libexec
+execdir := bindir
 
 
 all: build
@@ -58,20 +58,22 @@ build:
 install:
 	install -d $(bindir)
 	install -d $(execdir)
-	install -d $(sharedir)/pixmaps
+	install -d $(sharedir)/icons/hicolor/128x128/apps
+	install -d $(sharedir)/icons/hicolor/32x32/apps
+	install -d $(sharedir)/icons/hicolor/16x16/apps
 	install -d $(sharedir)/applications
-	install -d $(sharedir)/appdata
-	install -d $(sharedir)/playonlinux/bin
+	install -d $(sharedir)/metainfo
+	install -d $(sharedir)/playonlinux
 	install -d $(sharedir)/man/man1
 	install -d $(sharedir)/locale
 	$(GZIP) -c ./doc/playonlinux-pkg.1 > $(sharedir)/man/man1/playonlinux-pkg.1.gz
 	$(GZIP) -c ./doc/playonlinux.1 > $(sharedir)/man/man1/playonlinux.1.gz
 	cp ./etc/PlayOnLinux.desktop $(sharedir)/applications/PlayOnLinux.desktop
-	cp ./etc/PlayOnLinux.appdata.xml $(sharedir)/appdata/PlayOnLinux.appdata.xml
-	cp ./etc/playonlinux.png $(sharedir)/pixmaps/playonlinux.png
-	cp ./etc/playonlinux16.png $(sharedir)/pixmaps/playonlinux16.png
-	cp ./etc/playonlinux32.png $(sharedir)/pixmaps/playonlinux32.png
-	cp ./bin/{playonlinux,playonlinux-pkg} $(bindir)/
+	cp ./etc/PlayOnLinux.appdata.xml $(sharedir)/metainfo/PlayOnLinux.appdata.xml
+	cp ./etc/playonlinux.png $(sharedir)/icons/hicolor/128x128/apps/playonlinux.png
+	cp ./etc/playonlinux16.png $(sharedir)/icons/hicolor/16x16/apps/playonlinux.png
+	cp ./etc/playonlinux32.png $(sharedir)/icons/hicolor/32x32/apps/playonlinux.png
+	cp -p ./bin/{playonlinux,playonlinux-pkg} $(bindir)/
 	cp ./bin/playonlinux-check_dd $(execdir)/
 	cp ./{playonlinux*,README.md,TRANSLATORS,CHANGELOG.md,LICENCE} $(sharedir)/playonlinux/
 	cp -R ./{bash,etc,lib,plugins,python,resources,tests} $(sharedir)/playonlinux/
-- 
2.43.0

