From 8c91c14e582e494674fdbd9e3e2b1a88c0eab947 Mon Sep 17 00:00:00 2001
From: Muhammad Alfi Syahrin <malfisya.dev@hotmail.com>
Date: Thu, 25 Jan 2024 14:43:19 +0700
Subject: [PATCH] patch desktop file

---
 etc/PlayOnLinux.desktop | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/etc/PlayOnLinux.desktop b/etc/PlayOnLinux.desktop
index 490108c..f2ceba1 100644
--- a/etc/PlayOnLinux.desktop
+++ b/etc/PlayOnLinux.desktop
@@ -5,4 +5,6 @@ Comment=Front-end application for the wine
 Type=Application
 Exec=playonlinux %F
 Icon=playonlinux
+Terminal=false
+StartupWMClass=PlayOnLinux
 Categories=Utility;Emulator;
-- 
2.43.0

