From 6b7daa9c2b966482f0bbc6c614dda7cf8cee9be0 Mon Sep 17 00:00:00 2001
From: Evan Maddock <maddock.evan@vivaldi.net>
Date: Thu, 24 Jun 2021 19:56:33 -0400
Subject: [PATCH] Fix building

Signed-off-by: Evan Maddock <maddock.evan@vivaldi.net>
---
 src/menu_lan.c | 2 +-
 src/network.c  | 2 +-
 2 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/menu_lan.c b/src/menu_lan.c
index 001ed23..3965890 100644
--- a/src/menu_lan.c
+++ b/src/menu_lan.c
@@ -37,7 +37,7 @@ along with this program.  If not, see <http://www.gnu.org/licenses/>.  */
 
 
 /* lan_player_type now defined in network.h */
-lan_player_type lan_player_info[MAX_CLIENTS];
+static lan_player_type lan_player_info[MAX_CLIENTS];
 
 /* Local function prototypes: ------------------- */
 void draw_player_table(void);
diff --git a/src/network.c b/src/network.c
index 689ace7..dab0ac1 100644
--- a/src/network.c
+++ b/src/network.c
@@ -54,7 +54,7 @@ static int connected_server = -1;
 static int my_index = -1;
 
 /* Keep track of other connected players: */
-lan_player_type lan_player_info[MAX_CLIENTS];
+static lan_player_type lan_player_info[MAX_CLIENTS];
 
 /* Local function prototypes: */
 int say_to_server(char *statement);
-- 
2.32.0

