--- src/titlescreen.c~	2024-02-05 14:10:46.782527009 -0600
+++ src/titlescreen.c	2024-02-05 14:25:13.788723383 -0600
@@ -195,7 +195,7 @@
   /* load backgrounds */
   T4K_LoadBothBkgds(bkg_path, &fs_bkg, &win_bkg);
   T4K_SetMenuSounds(NULL, sounds[SND_POP], sounds[SND_TOCK]);
-  T4K_OnResolutionSwitch(&HandleTitleScreenResSwitch);
+  T4K_OnResolutionSwitch((ResSwitchCallback)&HandleTitleScreenResSwitch);
   
   if(fs_bkg == NULL || win_bkg == NULL)
   {
--- src/menu.c~	2011-05-03 19:15:06.000000000 -0500
+++ src/menu.c	2024-02-05 14:27:25.615839350 -0600
@@ -112,7 +112,7 @@
     which,
     return_choice,
     &DrawTitleScreen, 
-    &HandleTitleScreenEvents, 
+    (int (*)(SDL_Event *))&HandleTitleScreenEvents, 
     &HandleTitleScreenAnimations, 
     &handle_activity);
 }
@@ -228,7 +228,7 @@
   }
 
   //re-register resolution switcher
-  T4K_OnResolutionSwitch(&HandleTitleScreenResSwitch);
+  T4K_OnResolutionSwitch((ResSwitchCallback)&HandleTitleScreenResSwitch);
   //redraw if necessary
   RenderTitleScreen();
     
@@ -243,7 +243,7 @@
 int run_academy(void)
 {
   int chosen_lesson = -1;
-  T4K_OnResolutionSwitch(&HandleTitleScreenResSwitch);
+  T4K_OnResolutionSwitch((ResSwitchCallback)&HandleTitleScreenResSwitch);
 
   chosen_lesson = run_menu(MENU_LESSONS, true);
   while (chosen_lesson >= 0)
@@ -263,7 +263,7 @@
 
       T4K_OnResolutionSwitch(NULL);
       game();
-      T4K_OnResolutionSwitch(&HandleTitleScreenResSwitch);
+      T4K_OnResolutionSwitch((ResSwitchCallback)&HandleTitleScreenResSwitch);
 
       /* If successful, display Gold Star for this lesson! */
       if (MC_MissionAccomplished())
--- src/menu_lan.c~	2011-05-03 19:15:10.000000000 -0500
+++ src/menu_lan.c	2024-02-05 14:42:43.069686903 -0600
@@ -147,7 +147,7 @@
 		    NULL, "Server Selection", NULL);
 	    T4K_PrerenderMenu(MENU_SERVERSELECT);
 	    server_choice = T4K_RunMenu(MENU_SERVERSELECT, true, &DrawTitleScreen,
-		    &HandleTitleScreenEvents, &HandleTitleScreenAnimations, NULL);
+		    (int (*)(SDL_Event *))&HandleTitleScreenEvents, &HandleTitleScreenAnimations, NULL);
 
 	    if(!LAN_AutoSetup(server_choice))
 	    {
