From 2516d6576764c9c36dad2fb5986f4b5f4be80d26 Mon Sep 17 00:00:00 2001
From: Algent Albrahimi <algent@protonmail.com>
Date: Sun, 20 Feb 2022 20:08:17 +0100
Subject: [PATCH] Fix homepage

---
 data/meson.build | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/data/meson.build b/data/meson.build
index 2906308..ee19288 100644
--- a/data/meson.build
+++ b/data/meson.build
@@ -74,7 +74,7 @@ install_data(
 ui_config = configuration_data()
 ui_config.set('app-id', application_id)
 ui_config.set('version', meson.project_version() + version_suffix)
-ui_config.set('package_url', 'https://somas.pages.gitlab.gnome.org/apostrophe/')
+ui_config.set('package_url', 'https://world.pages.gitlab.gnome.org/apostrophe/')
 ui_preconfigured_files = files(
   'ui/About.ui.in'
 )
-- 
2.35.1

