diff --git a/crypto/cms/cms_kari.c b/crypto/cms/cms_kari.c
index 2cfcdb2..04c7723 100644
--- a/crypto/cms/cms_kari.c
+++ b/crypto/cms/cms_kari.c
@@ -401,9 +401,7 @@ static int cms_wrap_init(CMS_KeyAgreeRecipientInfo *kari,
      * Pick a cipher based on content encryption cipher. If it is DES3 use
      * DES3 wrap otherwise use AES wrap similar to key size.
      */
-    if (EVP_CIPHER_type(cipher) == NID_des_ede3_cbc)
-        kekcipher = EVP_des_ede3_wrap();
-    else if (keylen <= 16)
+    if (keylen <= 16)
         kekcipher = EVP_aes_128_wrap();
     else if (keylen <= 24)
         kekcipher = EVP_aes_192_wrap();
diff --git a/ssl/ssl.h b/ssl/ssl.h
index 5ef56fa..0d10213 100644
--- a/ssl/ssl.h
+++ b/ssl/ssl.h
@@ -338,7 +338,8 @@ extern "C" {
  * The following cipher list is used by default. It also is substituted when
  * an application-defined cipher list string starts with 'DEFAULT'.
  */
-# define SSL_DEFAULT_CIPHER_LIST "ALL:!EXPORT:!LOW:!aNULL:!eNULL:!SSLv2"
+# define SSL_DEFAULT_CIPHER_LIST "ALL:!EXPORT:!LOW:!aNULL:!eNULL:!SSLv2:!EXPORT:!RC2:!DES:!3DES"
+
 /*
  * As of OpenSSL 1.0.0, ssl_create_cipher_list() in ssl/ssl_ciph.c always
  * starts with a reasonable order, and all we have to do for DEFAULT is
diff --git a/test/Makefile b/test/Makefile
index e566bab..b14bbf3 100644
--- a/test/Makefile
+++ b/test/Makefile
@@ -152,7 +152,7 @@ alltests: \
 	test_enc test_x509 test_rsa test_crl test_sid \
 	test_gen test_req test_pkcs7 test_verify test_dh test_dsa \
 	test_ss test_ca test_engine test_evp test_evp_extra test_ssl test_tsa test_ige \
-	test_jpake test_srp test_cms test_ocsp test_v3name test_heartbeat \
+	test_jpake test_srp test_ocsp test_v3name test_heartbeat \
 	test_constant_time test_verify_extra test_clienthello test_sslv2conftest
 
 test_evp: $(EVPTEST)$(EXE_EXT) evptests.txt
diff --git a/test/tests.com b/test/tests.com
index 6e1c818..55a0f00 100644
--- a/test/tests.com
+++ b/test/tests.com
@@ -56,7 +56,7 @@ $	    tests := -
 	test_enc,test_x509,test_rsa,test_crl,test_sid,-
 	test_gen,test_req,test_pkcs7,test_verify,test_dh,test_dsa,-
 	test_ss,test_ca,test_engine,test_evp,test_ssl,test_tsa,test_ige,-
-	test_jpake,test_srp,test_cms,test_ocsp,test_v3name,test_heartbeat,-
+	test_jpake,test_srp,test_ocsp,test_v3name,test_heartbeat,-
 	test_constant_time
 $	endif
 $	tests = f$edit(tests,"COLLAPSE")
@@ -360,12 +360,6 @@ $ test_jpake:
 $	write sys$output "Test JPAKE"
 $	mcr 'texe_dir''jpaketest'
 $	return
-$ test_cms:
-$	write sys$output "CMS consistency test"
-$	! Define the logical name used to find openssl.exe in the perl script.
-$	define /user_mode osslx 'exe_dir'
-$	perl CMS-TEST.PL
-$	return
 $ test_srp: 
 $	write sys$output "Test SRP"
 $	mcr 'texe_dir''srptest'
