diff -up ./tests/test_client.py.relaxed ./tests/test_client.py
--- ./tests/test_client.py.relaxed	2018-09-19 05:16:40.000000000 +0100
+++ ./tests/test_client.py	2018-10-09 09:47:51.837994509 +0100
@@ -33,7 +33,7 @@ import warnings
 import weakref
 from tempfile import mkstemp
 
-from pytest_relaxed import raises
+from pytest import raises
 
 import paramiko
 from paramiko.pkey import PublicBlob
@@ -662,10 +662,10 @@ class PasswordPassphraseTests(ClientTest
 
     # TODO: more granular exception pending #387; should be signaling "no auth
     # methods available" because no key and no password
-    @raises(SSHException)
     def test_passphrase_kwarg_not_used_for_password_auth(self):
-        # Using the "right" password in the "wrong" field shouldn't work.
-        self._test_connection(passphrase="pygmalion")
+        with raises(SSHException):
+            # Using the "right" password in the "wrong" field shouldn't work.
+            self._test_connection(passphrase='pygmalion')
 
     def test_passphrase_kwarg_used_for_key_passphrase(self):
         # Straightforward again, with new passphrase kwarg.
@@ -683,14 +683,14 @@ class PasswordPassphraseTests(ClientTest
             password="television",
         )
 
-    @raises(AuthenticationException)  # TODO: more granular
     def test_password_kwarg_not_used_for_passphrase_when_passphrase_kwarg_given(
         self
     ):  # noqa
-        # Sanity: if we're given both fields, the password field is NOT used as
-        # a passphrase.
-        self._test_connection(
-            key_filename=_support("test_rsa_password.key"),
-            password="television",
-            passphrase="wat? lol no",
-        )
+        with raises(AuthenticationException): # TODO: more granular
+            # Sanity: if we're given both fields, the password field is NOT used as
+            # a passphrase.
+            self._test_connection(
+                key_filename=_support('test_rsa_password.key'),
+                password='television',
+                passphrase='wat? lol no',
+            )
