From fe7ac5b4d9112d75115420a0d22dedb900a8f3d2 Mon Sep 17 00:00:00 2001
From: Laurent Montel <montel@kde.org>
Date: Wed, 13 Jun 2018 13:18:51 +0200
Subject: [PATCH] Fix dnd attachment with qt5.11

Bug reported by David
---
 webengineviewer/src/webengineview.cpp | 16 +++++++++++-----
 1 file changed, 11 insertions(+), 5 deletions(-)

diff --git a/webengineviewer/src/webengineview.cpp b/webengineviewer/src/webengineview.cpp
index 78a0e3f8..a314aed5 100644
--- a/webengineviewer/src/webengineview.cpp
+++ b/webengineviewer/src/webengineview.cpp
@@ -166,13 +166,19 @@ void WebEngineView::forwardMouseReleaseEvent(QMouseEvent *event)
 
 bool WebEngineView::eventFilter(QObject *obj, QEvent *event)
 {
+    // Keyboard events are sent to parent widget
+    if (obj == this && event->type() == QEvent::ParentChange && parentWidget()) {
+        parentWidget()->installEventFilter(this);
+    }
+
     // Hack to find widget that receives input events
     if (obj == this && event->type() == QEvent::ChildAdded) {
-        QWidget *child = dynamic_cast<QWidget *>(static_cast<QChildEvent *>(event)->child());
-        if (child && child->inherits("QtWebEngineCore::RenderWidgetHostViewQtDelegateWidget")) {
-            d->mCurrentWidget = child;
-            d->mCurrentWidget->installEventFilter(this);
-        }
+        QTimer::singleShot(0, this, [this]() {
+            if (focusProxy() && d->mCurrentWidget != focusProxy()) {
+                d->mCurrentWidget = focusProxy();
+                d->mCurrentWidget->installEventFilter(this);
+            }
+        });
     }
 
     // Forward events to WebEngineView
