From b1fdb365ed2280390447ede5f450434fe240b035 Mon Sep 17 00:00:00 2001
From: Reilly Brogan <reilly@reillybrogan.com>
Date: Fri, 15 Dec 2023 15:46:22 -0600
Subject: [PATCH] Fix amateur hour shit

---
 makefile | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/makefile b/makefile
index ce54a02..f9dbe56 100644
--- a/makefile
+++ b/makefile
@@ -1,14 +1,14 @@
 #
 # Makefile for UNIX - unrar
 
-# Linux using GCC
-CXX=c++
-CXXFLAGS=-march=native -O2 -std=c++11 -Wno-logical-op-parentheses -Wno-switch -Wno-dangling-else
+# What the fuck
+CXX := $(CXX)
+CXXFLAGS := $(CXXFLAGS)
 LIBFLAGS=-fPIC
 DEFINES=-D_FILE_OFFSET_BITS=64 -D_LARGEFILE_SOURCE -DRAR_SMP
 STRIP=strip
 AR=ar
-LDFLAGS=-pthread
+LDFLAGS := $(LDFLAGS)
 DESTDIR=/usr
 
 ##########################
-- 
2.43.0

