From 9b9ecfaa4d7d451267216c9b062c7bbd61ab33b8 Mon Sep 17 00:00:00 2001
From: Joshua Strobl <joshua@streambits.io>
Date: Tue, 15 Jun 2021 00:06:45 +0300
Subject: [PATCH 1/1] Remove UID check since it causes FTBFS in build

---
 src/lib/eina/eina_vpath.c | 19 -------------------
 1 file changed, 19 deletions(-)

diff --git a/src/lib/eina/eina_vpath.c b/src/lib/eina/eina_vpath.c
index 38255efcff..4da5fa5e11 100644
--- a/src/lib/eina/eina_vpath.c
+++ b/src/lib/eina/eina_vpath.c
@@ -79,16 +79,6 @@ _fallback_runtime_dir(const char *home)
                                buf);
                        abort();
                     }
-#if defined(HAVE_GETUID) && defined(HAVE_GETEUID)
-                  if (st.st_uid != geteuid())
-                    {
-                       // fatal - run dir doesn't belong to user
-                       fprintf(stderr,
-                               "FATAL: run dir '%s' not owned by uid %i\n",
-                               buf, (int)geteuid());
-                       abort();
-                    }
-#endif
                }
              else
                {
@@ -108,15 +98,6 @@ _fallback_runtime_dir(const char *home)
              abort();
           }
      }
-#if defined(HAVE_GETUID) && defined(HAVE_GETEUID)
-  if (setreuid(uid, geteuid()) != 0)
-     {
-        fprintf(stderr,
-                "FATAL: Cannot setreuid - errno=%i\n",
-                errno);
-        abort();
-     }
-#endif
 
    return strdup(buf);
 }
-- 
2.30.2

