diff -ruNb a/docs/Makefile.in b/docs/Makefile.in
--- a/docs/Makefile.in	2012-09-22 17:48:02.000000000 -0400
+++ b/docs/Makefile.in	2013-04-08 23:02:53.591998660 -0400
@@ -449,7 +449,7 @@
 	$(COMPILE) -DALL_TERM_DOC -c $(srcdir)/checkdoc.c
 
 # install section
-install: install-gih install-info install-pdf
+install: install-gih install-pdf
 
 install-gih: gnuplot.gih
 	$(top_srcdir)/mkinstalldirs $(DESTDIR)$(GIHDIR)
