From bc38ca59a96af59506ff63ff0b24be1763e1772b Mon Sep 17 00:00:00 2001
From: Pierre-Yves <pyu@riseup.net>
Date: Tue, 31 Jul 2018 22:14:33 +0200
Subject: [PATCH] Patch for qt

Signed-off-by: Pierre-Yves <pyu@riseup.net>
---
 qtgui/ssearch_w.cpp | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/qtgui/ssearch_w.cpp b/qtgui/ssearch_w.cpp
index 2d0c97c..abef286 100644
--- a/qtgui/ssearch_w.cpp
+++ b/qtgui/ssearch_w.cpp
@@ -32,6 +32,7 @@
 #include <qevent.h>
 #include <QTimer>
 #include <QCompleter>
+#include <QAbstractItemView>
 
 #include "log.h"
 #include "guiutils.h"
@@ -710,7 +711,7 @@ bool SSearch::eventFilter(QObject *target, QEvent *event)
     LOGDEB2("SSearch::eventFilter: target "  << (target) << " ("  << (queryText->lineEdit()) << ") type "  << (eventTypeToStr(event->type())) << "\n" );
 #endif
 
-    if (target == queryText->view()) {
+    if (target == (QObject*)(queryText->view())) {
 	if (event->type() == QEvent::Hide) {
 	    // List was closed. If we were displaying completions, need
 	    // to reset state.
-- 
2.18.0

