--- desktop/recoll-searchgui.desktop.old	2025-05-11 11:32:05.133786447 +0000
+++ desktop/recoll-searchgui.desktop	2025-05-11 11:32:25.047940594 +0000
@@ -3,7 +3,7 @@
 Comment=Find documents by specifying search terms
 Comment[ru]=Поиск документов по заданным условиям
 Comment[de]=Finde Dokumente durch Angeben von Suchbegriffen
-Exec=recoll
+Exec=env QT_QPA_PLATFORM=xcb recoll
 GenericName=Local Text Search
 GenericName[ru]=Локальный текстовый поиск
 GenericName[de]=Lokale Textsuche
