From b8eadc9f89e8f7ab0338eacda9f98a6caea76883 Mon Sep 17 00:00:00 2001
From: Andreas Baumann <mail@andreasbaumann.cc>
Date: Thu, 2 Jun 2022 13:33:11 +0200
Subject: [PATCH libaio 01/26] harness: use off64_t instead of off_t in test
 23.t (EINVAL on 32-bit)

---
 harness/cases/23.t | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/harness/cases/23.t b/harness/cases/23.t
index cc5a8a1..9bbb05c 100644
--- a/harness/cases/23.t
+++ b/harness/cases/23.t
@@ -72,7 +72,7 @@ static void fail_errno(const char *format, ...)
 static void *thrproc2(void *arg)
 {
 	for (;;) {
-		off_t offset = 0;
+		off64_t offset = 0;
 
 		pthread_barrier_wait(&barrier);
 		if (exiting)
@@ -92,7 +92,7 @@ static void *thrproc3(void *arg)
 {
 	for (;;) {
 		char c;
-		off_t offset = 0;
+		off64_t offset = 0;
 
 		pthread_barrier_wait(&barrier);
 		if (exiting)
-- 
2.43.0

