From f483509a17b2fb7ea072d0ba5d9a875335e71aaa Mon Sep 17 00:00:00 2001
From: Guillem Jover <guillem@hadrons.org>
Date: Wed, 14 Aug 2019 03:30:09 +0200
Subject: [PATCH libaio 14/26] harness: Add KERNEL_RW_POINTER for various
 32-bit architectures

Signed-off-by: Guillem Jover <guillem@hadrons.org>
---
 harness/main.c | 9 +++++++++
 1 file changed, 9 insertions(+)

diff --git a/harness/main.c b/harness/main.c
index 82e9c69..0eed7bd 100644
--- a/harness/main.c
+++ b/harness/main.c
@@ -12,7 +12,16 @@
 #include <libaio.h>
 
 #if __LP64__ == 0
+#if defined(__i386__) || defined(__powerpc__)
 #define KERNEL_RW_POINTER	((void *)0xc0010000)
+#elif defined(__arm__) || defined(__s390__)
+#define KERNEL_RW_POINTER	((void *)0x00010000)
+#elif defined(__sparc__)
+#define KERNEL_RW_POINTER	((void *)0xf0010000)
+#else
+#warning Not really sure where kernel memory is.  Guessing.
+#define KERNEL_RW_POINTER	((void *)0xc0010000)
+#endif
 #else
 //#warning Not really sure where kernel memory is.  Guessing.
 #define KERNEL_RW_POINTER	((void *)0xffffffff81000000)
-- 
2.43.0

