From 1b5cf07a7809fecb8ef14faad5941aa462e25683 Mon Sep 17 00:00:00 2001
From: Pierre-Yves <pyu@riseup.net>
Date: Sun, 6 May 2018 09:21:41 +0200
Subject: [PATCH] Fix some paths

Signed-off-by: Pierre-Yves <pyu@riseup.net>
---
 devices/mounter/CMakeLists.txt | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/devices/mounter/CMakeLists.txt b/devices/mounter/CMakeLists.txt
index 04627b3..75ddc2a 100644
--- a/devices/mounter/CMakeLists.txt
+++ b/devices/mounter/CMakeLists.txt
@@ -2,10 +2,10 @@ set(CANTATA_MOUNTER_SRCS main.cpp mounter.cpp)
 qt5_add_dbus_adaptor(CANTATA_MOUNTER_SRCS mpd.cantata.mounter.xml mounter.h Mounter )
 
 add_executable( cantata-mounter ${CANTATA_MOUNTER_SRCS} )
-install(TARGETS cantata-mounter RUNTIME DESTINATION lib/cantata)
+install(TARGETS cantata-mounter RUNTIME DESTINATION lib64/cantata)
 configure_file(mpd.cantata.mounter.service.cmake ${CMAKE_CURRENT_BINARY_DIR}/mpd.cantata.mounter.service)
 install(FILES ${CMAKE_CURRENT_BINARY_DIR}/mpd.cantata.mounter.service DESTINATION ${SHARE_INSTALL_PREFIX}/dbus-1/system-services/ )
-install(FILES ${CMAKE_CURRENT_SOURCE_DIR}/mpd.cantata.mounter.conf DESTINATION /etc/dbus-1/system.d/ )
+install(FILES ${CMAKE_CURRENT_SOURCE_DIR}/mpd.cantata.mounter.conf DESTINATION ${SHARE_INSTALL_PREFIX}/dbus-1/system.d/ )
 
 target_link_libraries(cantata-mounter ${Qt5Core_LIBRARIES} ${Qt5DBus_LIBRARIES})
 include_directories( ${QTINCLUDES} ${CMAKE_CURRENT_SOURCE_DIR} ${CMAKE_CURRENT_BINARY_DIR} ${CMAKE_BINARY_DIR})
-- 
2.16.3

