diff --git a/src/gst/gstpipewireformat.c b/src/gst/gstpipewireformat.c
index b4ec24f23be93eb5d38d729064b9a0ac5b432583..1c63013bd9a94860285b8464868bd5c84c5f1a3a 100644
--- a/src/gst/gstpipewireformat.c
+++ b/src/gst/gstpipewireformat.c
@@ -1209,10 +1209,11 @@ filter_dmabuf_caps (GstCapsFeatures *features,
   return TRUE;
 }
 
-GstCaps *
-gst_caps_sanitize (GstCaps *caps)
+void
+gst_caps_sanitize (GstCaps **caps)
 {
-  caps = gst_caps_make_writable (caps);
-  gst_caps_filter_and_map_in_place (caps, filter_dmabuf_caps, NULL);
-  return caps;
+  g_return_if_fail (GST_IS_CAPS (*caps));
+
+  *caps = gst_caps_make_writable (*caps);
+  gst_caps_filter_and_map_in_place (*caps, filter_dmabuf_caps, NULL);
 }
diff --git a/src/gst/gstpipewireformat.h b/src/gst/gstpipewireformat.h
index ca76b70c2f06aaf67e0d78d038af5e6cdff07c00..1c3a239baeaf5ce5856a0ec517f4dc5a0bb901b0 100644
--- a/src/gst/gstpipewireformat.h
+++ b/src/gst/gstpipewireformat.h
@@ -15,7 +15,7 @@ GPtrArray *      gst_caps_to_format_all  (GstCaps *caps);
 
 GstCaps *        gst_caps_from_format    (const struct spa_pod *format);
 
-GstCaps *        gst_caps_sanitize       (GstCaps *caps);
+void             gst_caps_sanitize       (GstCaps **caps);
 
 G_END_DECLS
 
diff --git a/src/gst/gstpipewiresrc.c b/src/gst/gstpipewiresrc.c
index b8e1a271fb392a9153b6ae7d991109be4a28c272..dd7b942cdf736f7e2cc70fd11f201427b264a63b 100644
--- a/src/gst/gstpipewiresrc.c
+++ b/src/gst/gstpipewiresrc.c
@@ -846,9 +846,9 @@ gst_pipewire_src_negotiate (GstBaseSrc * basesrc)
   }
 
   GST_DEBUG_OBJECT (basesrc, "have common caps: %" GST_PTR_FORMAT, caps);
-  gst_caps_sanitize (caps);
+  gst_caps_sanitize (&caps);
 
-  if (caps == NULL || gst_caps_is_empty (caps))
+  if (gst_caps_is_empty (caps))
     goto no_common_caps;
 
   GST_DEBUG_OBJECT (basesrc, "have common caps (sanitized): %" GST_PTR_FORMAT, caps);
