diff -ru PyQt6_WebEngine-6.8.0.orig/sip/QtWebEngineCore/qwebengineframe.sip PyQt6_WebEngine-6.8.0/sip/QtWebEngineCore/qwebengineframe.sip
--- PyQt6_WebEngine-6.8.0.orig/sip/QtWebEngineCore/qwebengineframe.sip	2024-12-13 08:18:04.682729252 +0100
+++ PyQt6_WebEngine-6.8.0/sip/QtWebEngineCore/qwebengineframe.sip	2024-12-13 08:18:18.679705449 +0100
@@ -25,6 +25,7 @@
 class QWebEngineFrame /NoDefaultCtors/
 {
 %TypeHeaderCode
+#include <qvariant.h>
 #include <qwebengineframe.h>
 %End
 
