--- configure.ac.old	2016-05-24 23:43:41.265711304 -0700
+++ configure.ac	2016-05-25 00:03:39.128437427 -0700
@@ -942,7 +942,8 @@
 # Glib2 keep changing mutex APIs so we also need to disable 'deprecated'
 # warnings for now (-Wno-deprecated-declarations).
 for TEST_CFLAG in -Wno-pointer-sign -Wno-unused-value -fno-strict-aliasing \
-    -Wno-unknown-pragmas -Wno-uninitialized -Wno-deprecated-declarations; do
+    -Wno-unknown-pragmas -Wno-uninitialized -Wno-deprecated-declarations \
+    -Wno-unused-const-variable; do
     AC_MSG_CHECKING([for GCC flag $TEST_CFLAG])
     ORIGINAL_CFLAGS="$CFLAGS"
     CFLAGS="$CFLAGS $TEST_CFLAG"
