diff --git a/src/gui/src/ActionDialog.cpp b/src/gui/src/ActionDialog.cpp
index 9495033..8c21cb8 100644
--- a/src/gui/src/ActionDialog.cpp
+++ b/src/gui/src/ActionDialog.cpp
@@ -26,6 +26,8 @@
 #include <QtCore>
 #include <QtGui>
 
+#include <QButtonGroup>
+
 ActionDialog::ActionDialog(QWidget* parent, ServerConfig& config, Hotkey& hotkey, Action& action) :
     QDialog(parent, Qt::WindowTitleHint | Qt::WindowSystemMenuHint),
     Ui::ActionDialogBase(),
diff --git a/src/gui/src/ScreenSetupView.cpp b/src/gui/src/ScreenSetupView.cpp
index f26d970..0f0f352 100644
--- a/src/gui/src/ScreenSetupView.cpp
+++ b/src/gui/src/ScreenSetupView.cpp
@@ -23,6 +23,8 @@
 #include <QtCore>
 #include <QtGui>
 
+#include <QHeaderView>
+
 ScreenSetupView::ScreenSetupView(QWidget* parent) :
     QTableView(parent)
 {
