From db82964da6187c47d1a43df42a50ee701618e4c9 Mon Sep 17 00:00:00 2001
From: Ikey Doherty <ikey@solus-project.com>
Date: Wed, 31 Aug 2016 06:46:28 +0100
Subject: [PATCH] rtgui: Use the system theme by default

Signed-off-by: Ikey Doherty <ikey@solus-project.com>
---
 rtgui/options.cc | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/rtgui/options.cc b/rtgui/options.cc
index 8a4b843..ae83f32 100644
--- a/rtgui/options.cc
+++ b/rtgui/options.cc
@@ -308,7 +308,7 @@ void Options::setDefaults () {
     overwriteOutputFile = false;		// if TRUE, existing output JPGs/PNGs are overwritten, instead of adding ..-1.jpg, -2.jpg etc.
     theme = "25-Gray-Gray";
     slimUI = false;
-    useSystemTheme = false;
+    useSystemTheme = true;
     maxThumbnailHeight = 400;
     maxCacheEntries = 20000;
     thumbInterp = 1;
-- 
2.9.2

