From cf9413c6322c5e10c27c88a1ebc1ca6e127bc8c6 Mon Sep 17 00:00:00 2001
From: Arjan van de Ven <arjan@linux.intel.com>
Date: Sat, 18 Jun 2016 17:33:33 +0000
Subject: [PATCH 1/4] more udev children workers

---
 units/systemd-udevd.service.in | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/units/systemd-udevd.service.in b/units/systemd-udevd.service.in
index a0ee9e0a50..9cf5c6e131 100644
--- a/units/systemd-udevd.service.in
+++ b/units/systemd-udevd.service.in
@@ -25,7 +25,7 @@ OOMScoreAdjust=-1000
 Sockets=systemd-udevd-control.socket systemd-udevd-kernel.socket
 Restart=always
 RestartSec=0
-ExecStart={{ROOTLIBEXECDIR}}/systemd-udevd
+ExecStart={{ROOTLIBEXECDIR}}/systemd-udevd --children-max=16
 KillMode=mixed
 TasksMax=infinity
 PrivateMounts=yes
-- 
2.43.0

