From 235cd74f3a12fc9ed839e92d48d4db36df056b5c Mon Sep 17 00:00:00 2001
From: Arjan van de Ven <arjan@linux.intel.com>
Date: Wed, 13 Jun 2018 13:42:12 -0700
Subject: [PATCH 3/4] Notify systemd earlier that resolved is ready.

---
 src/resolve/resolved.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/resolve/resolved.c b/src/resolve/resolved.c
index 1625c5189d..47b15e892a 100644
--- a/src/resolve/resolved.c
+++ b/src/resolve/resolved.c
@@ -69,6 +69,8 @@ static int run(int argc, char *argv[]) {
 
         assert_se(sigprocmask_many(SIG_BLOCK, NULL, SIGTERM, SIGINT, SIGUSR1, SIGUSR2, SIGRTMIN+1, SIGRTMIN+18, -1) >= 0);
 
+        notify_stop = notify_start(NOTIFY_READY, NOTIFY_STOPPING);
+
         r = manager_new(&m);
         if (r < 0)
                 return log_error_errno(r, "Could not create manager: %m");
@@ -87,8 +89,6 @@ static int run(int argc, char *argv[]) {
         if (r < 0)
                 return log_error_errno(r, "Failed to drop remaining caps: %m");
 
-        notify_stop = notify_start(NOTIFY_READY, NOTIFY_STOPPING);
-
         r = sd_event_loop(m->event);
         if (r < 0)
                 return log_error_errno(r, "Event loop failed: %m");
-- 
2.43.0

