--- a/src/mongo/rpc/SConscript
+++ b/src/mongo/rpc/SConscript
@@ -3,6 +3,7 @@
 Import("env")
 Import("wiredtiger")
 Import("get_option")
+Import("use_system_version_of_library")
 
 env = env.Clone()
 
@@ -22,9 +23,15 @@
 )
 
 protoEnv = env.Clone()
+wiredtigerLib = []
 if wiredtiger:
     protoEnv.InjectThirdParty(libraries=['wiredtiger'])
 
+    if not use_system_version_of_library('wiredtiger'):
+        wiredtigerLib = '$BUILD_DIR/third_party/wiredtiger/wiredtiger_checksum'
+    else:
+        wiredtigerLib = '$BUILD_DIR/third_party/shim_wiredtiger'
+
 protoEnv.Library(
     target=[
         'protocol',
@@ -42,7 +49,7 @@
         '$BUILD_DIR/mongo/bson/util/bson_extract',
         '$BUILD_DIR/mongo/db/bson/dotted_path_support',
         '$BUILD_DIR/mongo/db/server_options_core',
-        '$BUILD_DIR/third_party/wiredtiger/wiredtiger_checksum' if wiredtiger else [],
+        wiredtigerLib,
     ],
 )
 
@@ -148,7 +155,7 @@
     ],
     LIBDEPS=[
         '$BUILD_DIR/mongo/client/clientdriver_minimal',
-        '$BUILD_DIR/third_party/wiredtiger/wiredtiger_checksum',
+        '$BUILD_DIR/third_party/wiredtiger/wiredtiger_checksum' if not use_system_version_of_library('wiredtiger') else '$BUILD_DIR/third_party/shim_wiredtiger',
         'rpc',
     ],
 )
