diff -ruN mongodb-mongo-35d46d2.a/src/mongo/db/exec/plan_stats.h mongodb-mongo-35d46d2.b/src/mongo/db/exec/plan_stats.h
--- mongodb-mongo-35d46d2.a/src/mongo/db/exec/plan_stats.h	2021-05-03 09:41:41.000000000 +0200
+++ mongodb-mongo-35d46d2.b/src/mongo/db/exec/plan_stats.h	2021-06-13 13:57:33.854285142 +0200
@@ -33,6 +33,7 @@
 #include <cstdlib>
 #include <string>
 #include <vector>
+#include <optional>
 
 #include "mongo/db/index/multikey_paths.h"
 #include "mongo/db/jsobj.h"
