--- a/src/third_party/SConscript
+++ b/src/third_party/SConscript
@@ -376,8 +376,10 @@ abseilEnv = env
 if use_system_version_of_library("abseil-cpp"):
     abseilEnv = env.Clone(
         SYSLIBDEPS=[
-            env['LIBDEPS_ABSL_CONTAINER_SYSLIBDEP'],
-            env['LIBDEPS_ABSL_HASH_SYSLIBDEP'],
+            "absl_city",
+            "absl_hash",
+            "absl_raw_hash_set",
+            "absl_throw_delegate",
         ])
 else:
     abseilDirectory = 'abseil-cpp-master'
