--- a/src/mongo/shell/shell_utils_extended.cpp
+++ b/src/mongo/shell/shell_utils_extended.cpp
@@ -36,6 +36,7 @@
 #include <sys/types.h>
 #endif
 
+#include <boost/algorithm.hpp>
 #include <boost/filesystem.hpp>
 #include <fmt/format.h>
 #include <fstream>
@@ -304,9 +305,18 @@ BSONObj copyDir(const BSONObj& args, voi
 
     BSONObjIterator it(args);
     const std::string source = it.next().str();
-    const std::string destination = it.next().str();
+    const boost::filesystem::path destination = boost::filesystem::path(it.next().str());
 
-    boost::filesystem::copy(source, destination, boost::filesystem::copy_options::recursive);
+    for (const auto& dirEnt : boost::make_iterator_range(boost::filesystem::recursive_directory_iterator{source}, {})) {
+        const boost::filesystem::path& path = dirEnt.path();
+        std::string relativePath = path.string();
+        boost::algorithm::replace_first(relativePath, source, "");
+        if (!boost::filesystem::is_directory(path)) {
+            boost::filesystem::copy_file(path, destination / relativePath);
+        } else {
+            boost::filesystem::copy_directory(path, destination / relativePath);
+        }
+    }
 
     return undefinedReturn;
 }
