From 8070d5af50949351fee1900c4438e9ea0a63faf7 Mon Sep 17 00:00:00 2001
From: Thomas Staudinger <Staudi.Kaos@gmail.com>
Date: Sat, 11 Mar 2023 00:54:46 +0100
Subject: [PATCH] Add additional terminals to gio

---
 gio/gdesktopappinfo.c | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/gio/gdesktopappinfo.c b/gio/gdesktopappinfo.c
index 30fcb2937..62e1c5eff 100644
--- a/gio/gdesktopappinfo.c
+++ b/gio/gdesktopappinfo.c
@@ -2710,7 +2710,10 @@ prepend_terminal_to_vector (int          *argc,
     { "color-xterm", "-e" },
     { "rxvt", "-e" },
     { "dtterm", "-e" },
-    { "xterm", "-e" }
+    { "xterm", "-e" },
+    { "alacritty", "-e" },
+    { "kitty", NULL },
+    { "cool-retro-term", "-e" }
   };
 
   g_return_val_if_fail (argc != NULL, FALSE);
-- 
2.39.0

