From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Muhammad Alfi Syahrin <malfisya.dev@hotmail.com>
Date: Sun, 9 Nov 2025 09:49:55 +0700
Subject: [PATCH] Add additional terminals to gio

---
 gio/gdesktopappinfo.c | 6 +++++-
 1 file changed, 5 insertions(+), 1 deletion(-)

diff --git a/gio/gdesktopappinfo.c b/gio/gdesktopappinfo.c
index 1e5a752..bfbc585 100644
--- a/gio/gdesktopappinfo.c
+++ b/gio/gdesktopappinfo.c
@@ -2720,6 +2720,7 @@ prepend_terminal_to_vector (int          *argc,
     const char *exec_arg;
   } known_terminals[] = {
     { "xdg-terminal-exec", NULL },
+    { "ptyxis", "--" },
     { "kgx", "-e" },
     { "gnome-terminal", "--" },
     { "mate-terminal", "-x" },
@@ -2730,7 +2731,10 @@ prepend_terminal_to_vector (int          *argc,
     { "color-xterm", "-e" },
     { "rxvt", "-e" },
     { "dtterm", "-e" },
-    { "xterm", "-e" }
+    { "xterm", "-e" },
+    { "alacritty", "-e" },
+    { "kitty", NULL },
+    { "cool-retro-term", "-e" }
   };
 
   g_return_val_if_fail (argc != NULL, FALSE);
