From fe09b70c41d2bfe8698bde5cf2e9f4d584fcc5c1 Mon Sep 17 00:00:00 2001
From: Ikey Doherty <ikey@solus-project.com>
Date: Sat, 11 Jul 2015 13:40:21 +0100
Subject: [PATCH 1/2] Support stateless XDG config directory

Signed-off-by: Ikey Doherty <ikey@solus-project.com>
---
 glib/gutils.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/glib/gutils.c b/glib/gutils.c
index 9f7b7e3..96fd7ed 100644
--- a/glib/gutils.c
+++ b/glib/gutils.c
@@ -2023,7 +2023,7 @@ g_get_system_config_dirs (void)
       conf_dirs = (gchar *) g_getenv ("XDG_CONFIG_DIRS");
 
       if (!conf_dirs || !conf_dirs[0])
-          conf_dirs = "/etc/xdg";
+          conf_dirs = "/usr/share/xdg" G_SEARCHPATH_SEPARATOR_S "/etc/xdg";
 
       conf_dir_vector = g_strsplit (conf_dirs, G_SEARCHPATH_SEPARATOR_S, 0);
 #endif
-- 
2.14.1

