From 52ea8f52706f32ab46d8ad85bc7a40bb5145262f Mon Sep 17 00:00:00 2001
From: Joshua Strobl <joshua@stroblindustries.com>
Date: Sat, 20 Apr 2019 12:36:27 +0300
Subject: [PATCH 1/1] Support stateless XDG and snap directories.

---
 glib/gutils.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/glib/gutils.c b/glib/gutils.c
index 2e2d457..3ef96e0 100644
--- a/glib/gutils.c
+++ b/glib/gutils.c
@@ -2080,7 +2080,7 @@ g_build_system_data_dirs (void)
    */
 #ifndef G_OS_WIN32
   if (!data_dirs || !data_dirs[0])
-    data_dirs = "/usr/local/share/:/usr/share/";
+    data_dirs = "/usr/local/share/:/usr/share/:/var/lib/snapd/desktop/";
 
   data_dir_vector = g_strsplit (data_dirs, G_SEARCHPATH_SEPARATOR_S, 0);
 #else
@@ -2174,7 +2174,7 @@ g_build_system_config_dirs (void)
     }
 #else
   if (!conf_dirs || !conf_dirs[0])
-    conf_dirs = "/etc/xdg";
+    conf_dirs = "/usr/share/xdg" G_SEARCHPATH_SEPARATOR_S "/etc/xdg";
 
   conf_dir_vector = g_strsplit (conf_dirs, G_SEARCHPATH_SEPARATOR_S, 0);
 #endif
-- 
2.21.0

