From 4c6e7fe3039966c9f4124b574c8bd5c1c31f802c Mon Sep 17 00:00:00 2001
From: Ikey Doherty <ikey@solus-project.com>
Date: Sat, 11 Jul 2015 13:40:21 +0100
Subject: [PATCH] Support stateless XDG config directory

Signed-off-by: Ikey Doherty <ikey@solus-project.com>
---
 glib/gutils.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/glib/gutils.c b/glib/gutils.c
index e229d0d..59b2cec 100644
--- a/glib/gutils.c
+++ b/glib/gutils.c
@@ -2000,7 +2000,7 @@ g_get_system_config_dirs (void)
       conf_dirs = (gchar *) g_getenv ("XDG_CONFIG_DIRS");
 
       if (!conf_dirs || !conf_dirs[0])
-          conf_dirs = "/etc/xdg";
+          conf_dirs = "/usr/share/xdg" G_SEARCHPATH_SEPARATOR_S "/etc/xdg";
 
       conf_dir_vector = g_strsplit (conf_dirs, G_SEARCHPATH_SEPARATOR_S, 0);
 #endif
-- 
2.4.5

