From b043806c27b6d4500706c936e346a9a08f9b064f Mon Sep 17 00:00:00 2001
From: Joshua Strobl <joshua@streambits.io>
Date: Mon, 22 Mar 2021 14:45:11 +0200
Subject: [PATCH 1/1] Add additional terminals to GIO

---
 gio/gdesktopappinfo.c | 10 ++++++++++
 1 file changed, 10 insertions(+)

diff --git a/gio/gdesktopappinfo.c b/gio/gdesktopappinfo.c
index 1a4b97918..797095238 100644
--- a/gio/gdesktopappinfo.c
+++ b/gio/gdesktopappinfo.c
@@ -2604,6 +2604,16 @@ prepend_terminal_to_vector (int    *argc,
             check = g_find_program_in_path ("tilix");
           if (check == NULL)
             check = g_find_program_in_path ("konsole");
+          if (check == NULL)
+            check = g_find_program_in_path ("alacritty");
+          if (check == NULL)
+            check = g_find_program_in_path ("kitty");
+          if (check == NULL)
+            check = g_find_program_in_path ("roxterm");
+          if (check == NULL)
+            check = g_find_program_in_path ("lilyterm");
+          if (check == NULL)
+            check = g_find_program_in_path ("cool-retro-term");
           if (check == NULL)
             check = g_find_program_in_path ("nxterm");
           if (check == NULL)
-- 
2.30.2

