From a089758abc6aadfce47bb6f1e698551bb9162c9f Mon Sep 17 00:00:00 2001
From: Algent Albrahimi <algent@protonmail.com>
Date: Mon, 21 Aug 2023 22:34:53 +0200
Subject: [PATCH] use podofo-09

---
 Makefile | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/Makefile b/Makefile
index ede0d7b..eb021e2 100644
--- a/Makefile
+++ b/Makefile
@@ -927,14 +927,14 @@ WARNFLAGS = -Wall -Wshadow
 PKG_CONFIG_LIBS := $(shell $(PKG_CONFIG) --cflags $(LIBS_ALL))
 CXXFLAGS += $(DEBUGFLAGS) $(DEFINES) $(OPTIMIZE) $(PKG_CONFIG_LIBS) -MP -MD -pthread $(WARNFLAGS) -std=c++17
 CFLAGS = $(filter-out -Wsuggest-override, $(filter-out -std=%,$(CXXFLAGS))) -std=c99
-LDFLAGS += -lm -lpthread -lstdc++ -L/usr/lib/podofo-0.9
+LDFLAGS += -lm -lpthread -lstdc++ -L/usr/lib/podofo-09
 GLIB_COMPILE_RESOURCES := $(shell $(PKG_CONFIG) --variable=glib_compile_resources gio-2.0)
 
 ifeq ($(DEBUG),1)
 	CXXFLAGS += -DUUID_PTR_CHECK -DCONNECTION_CHECK
 endif
 
-CXXFLAGS += -I/usr/include/podofo-0.9
+CXXFLAGS += -I/usr/include/podofo-09
 
 ifeq ($(OS),Windows_NT)
 	WITH_SPNAV ?= 0
-- 
2.41.0

