diff -ur ufw-0.35/tests/check-requirements ufw-0.35-python35/tests/check-requirements
--- ufw-0.35/tests/check-requirements	2016-02-18 06:26:44.000000000 +0100
+++ ufw-0.35-python36/tests/check-requirements	2016-10-06 03:19:38.200835663 +0200
@@ -50,7 +50,7 @@
 # check python
 found_python="no"
 echo -n "Has python: "
-for exe in python2.7 python2.6 python2.5 python3.2 python3.3 python3.4 python3.5 python python3 ; do
+for exe in python2.7 python2.6 python2.5 python3.4 python3.5 python3.6 python python3 ; do
     if ! which $exe >/dev/null 2>&1; then
         continue
     fi
@@ -59,7 +59,7 @@
         echo "pass (binary: $exe, version: $v, py2)"
         found_python="yes"
         break
-    elif echo "$v" | grep -q "^3.[234]"; then
+    elif echo "$v" | grep -q "^3.[456]"; then
         echo "pass (binary: $exe, version: $v, py3)"
         found_python="yes"
         break
