From 06d533628b1f3a75d06cbb29773dc6aaa2916fc3 Mon Sep 17 00:00:00 2001
From: Eddie Kohler <ekohler@gmail.com>
Date: Tue, 10 Oct 2023 22:05:48 -0400
Subject: [PATCH] Refuse to position a frame so it goes out of bounds.

---
 src/support.c | 23 ++++++++++++++++++-----
 1 file changed, 18 insertions(+), 5 deletions(-)

diff --git a/src/support.c b/src/support.c
index 94997b1..0abd08f 100644
--- a/src/support.c
+++ b/src/support.c
@@ -1618,7 +1618,8 @@ merge_frame_interval(Gt_Frameset *fset, int f1, int f2,
       desti->comment = 0;
     }
     if (fr->comment) {
-      if (!desti->comment) desti->comment = Gif_NewComment();
+      if (!desti->comment)
+        desti->comment = Gif_NewComment();
       merge_comments(desti->comment, fr->comment);
       /* delete the comment early to help with memory; set field to 0 so we
          don't re-free it later */
@@ -1628,10 +1629,22 @@ merge_frame_interval(Gt_Frameset *fset, int f1, int f2,
 
     if (fr->interlacing >= 0)
       desti->interlace = fr->interlacing;
-    if (fr->left >= 0)
-      desti->left = fr->left + (fr->position_is_offset ? desti->left : 0);
-    if (fr->top >= 0)
-      desti->top = fr->top + (fr->position_is_offset ? desti->top : 0);
+    if (fr->left >= 0) {
+      int left = fr->left + (fr->position_is_offset ? desti->left : 0);
+      if (left + desti->width > 65535) {
+        error(1, "left position %d out of range", left);
+        return 0;
+      }
+      desti->left = left;
+    }
+    if (fr->top >= 0) {
+      int top = fr->top + (fr->position_is_offset ? desti->top : 0);
+      if (top + desti->height > 65535) {
+        error(1, "top position %d out of range", top);
+        return 0;
+      }
+      desti->top = top;
+    }
 
     if (fr->delay >= 0)
       desti->delay = fr->delay;
