From cf17d29ff13f8d407ec04b8bc4c482a500020ed2 Mon Sep 17 00:00:00 2001
From: Ikey Doherty <ikey@solus-project.com>
Date: Sun, 17 May 2015 17:24:06 +0100
Subject: [PATCH 1/2] installer: Fix ESP mount issue

---
 os_installer/installer.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/os_installer/installer.py b/os_installer/installer.py
index bde06a0..cc3366f 100644
--- a/os_installer/installer.py
+++ b/os_installer/installer.py
@@ -308,7 +308,7 @@ class InstallerEngine:
                     else:                                                    
                         fstab.write("%s\t%s\t%s\t%s\t%s\t%s\n" % (partition_uuid, partition.mount_as, partition.type, fstab_mount_options, "0", fstab_fsck_option))
             if self.efi_mode and setup.grub_device is not None:
-                fstab.write("%s\t/boot/efi\tvfat\t0\t0\n" % self.get_uuid(setup.grub_device))
+                fstab.write("%s\t/boot/efi\tvfat\tdefaults\t0\t0\n" % self.get_uuid(setup.grub_device))
             fstab.close()
             
             # write host+hostname infos
-- 
2.3.2

