From 9470964760291608356fa5df51f025a5bf999ff1 Mon Sep 17 00:00:00 2001
From: Ikey Doherty <ikey@solus-project.com>
Date: Thu, 2 Jul 2015 07:56:29 +0100
Subject: [PATCH] Fix major bork in UEFI installation

---
 os_installer/installer.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/os_installer/installer.py b/os_installer/installer.py
index f7faee7..7a9554e 100644
--- a/os_installer/installer.py
+++ b/os_installer/installer.py
@@ -473,7 +473,7 @@ EndSection\n""" % (setup.keyboard_model, setup.keyboard_layout))
         with open(solfile, "w") as solconf:
             solconf.write("title Solus\nlinux /solus/kernel\ninitrd /solus/initramfs\noptions root=%s quiet\n" % self.root_partition)
         kver = os.uname()[2]
-        sdir = get_solus_dir("/target/boot/efi")
+        sdir = self.get_solus_dir("/target/boot/efi")
         if not os.path.exists(sdir):
             os.makedirs(sdir)
         kernel = "/source/boot/kernel-%s" % kver
-- 
2.4.5

