From 37e09de7d8a6ef1295ea4d81fec23985a717c872 Mon Sep 17 00:00:00 2001
From: Ikey Doherty <ikey@solus-project.com>
Date: Mon, 29 Feb 2016 08:17:27 +0000
Subject: [PATCH] Well done, numnuts.

Signed-off-by: Ikey Doherty <ikey@solus-project.com>
---
 os_installer/installer.py | 1 +
 1 file changed, 1 insertion(+)

diff --git a/os_installer/installer.py b/os_installer/installer.py
index bfc74c8..1c1ac8e 100644
--- a/os_installer/installer.py
+++ b/os_installer/installer.py
@@ -309,6 +309,7 @@ class InstallerEngine:
 
                     # systemd/initramfs take care of /
                     if partition.mount_as == "/":
+                        self.root_partition = partition_uuid
                         continue
 
                     # systemd auto-detects swap on GPT systems
-- 
2.7.1

