From f6ffa4116bbd1453f754da8d727b6d0dabdbdb03 Mon Sep 17 00:00:00 2001
From: Ikey Doherty <ikey@solus-project.com>
Date: Mon, 20 Jun 2016 21:58:10 +0100
Subject: [PATCH] diskman: Don't parse empty values

Signed-off-by: Ikey Doherty <ikey@solus-project.com>
---
 os_installer2/diskman.py | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/os_installer2/diskman.py b/os_installer2/diskman.py
index 6fa2c4f..5c3e4cc 100644
--- a/os_installer2/diskman.py
+++ b/os_installer2/diskman.py
@@ -625,6 +625,8 @@ class DiskManager:
                 key = splits[0].lower()
                 val = "=".join(splits[1:]).strip()
 
+                if len(val) == 0:
+                    continue
                 if val[0] == "\"":
                     val = val[1:]
                 if val[-1] == "\"":
-- 
2.8.4

