From fa3e0c4bc083c14df78fc1f15f36ede2d5a2aac2 Mon Sep 17 00:00:00 2001
From: Ikey Doherty <ikey@solus-project.com>
Date: Wed, 15 Feb 2017 07:59:20 +0000
Subject: [PATCH 01/35] postinstall: Remove redundant discard flag (fixes #29)

Signed-off-by: Ikey Doherty <ikey@solus-project.com>
---
 os_installer2/postinstall.py | 5 -----
 1 file changed, 5 deletions(-)

diff --git a/os_installer2/postinstall.py b/os_installer2/postinstall.py
index ebbc131..98a6c99 100644
--- a/os_installer2/postinstall.py
+++ b/os_installer2/postinstall.py
@@ -575,13 +575,8 @@ class PostInstallFstab(PostInstallStep):
 
         appends = []
 
-        # Determine if SSD optimizations should be considered
         dev_path = self.info.strategy.drive.path
-        ssd = DiskManager.is_device_ssd(dev_path)
-
         ext4_ops = "rw,relatime,errors=remount-ro"
-        if ssd:
-            ext4_ops = "discard,{}".format(ext4_ops)
 
         # Add the ESP to /boot/efi
         if strat.is_uefi() and self.info.bootloader_install:
-- 
2.13.3

