From 4087635e97b2a3ec29814eb97a5838a10cc3b90b Mon Sep 17 00:00:00 2001
From: Ikey Doherty <ikey@solus-project.com>
Date: Wed, 19 Oct 2016 04:44:02 +0100
Subject: [PATCH] Fix branding of UEFI installation

Signed-off-by: Ikey Doherty <ikey@solus-project.com>
---
 os_installer2/postinstall.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/os_installer2/postinstall.py b/os_installer2/postinstall.py
index 6009e6e..ebbc131 100644
--- a/os_installer2/postinstall.py
+++ b/os_installer2/postinstall.py
@@ -860,7 +860,7 @@ class PostInstallBootloader(PostInstallStep):
         try:
             with open(solfile, "w") as solconf:
                 conf = [
-                    "title Solus 1.2",
+                    "title Solus 1.2.1",
                     "linux /solus/kernel",
                     "initrd /solus/initramfs",
                     "options {}".format(cmdline)
-- 
2.10.1

