From 240f2769fe6adee452f888561088b237c4418e2d Mon Sep 17 00:00:00 2001
From: Ikey Doherty <ikey@solus-project.com>
Date: Tue, 30 Jun 2015 06:36:33 +0100
Subject: [PATCH] Update branding for test images

Signed-off-by: Ikey Doherty <ikey@solus-project.com>
---
 dist/install.conf | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/dist/install.conf b/dist/install.conf
index 0bc9ac0..60dc43a 100644
--- a/dist/install.conf
+++ b/dist/install.conf
@@ -1,14 +1,14 @@
 # Defines the distribution name and version..
 [Branding]
 Name=Solus Operating System
-Version=beta2
+Version=unstable
 Theme=LightTheme
 
 [Configuration]
 LiveUser=live
 
 [LightTheme]
-Widgets=evopop-light-gtk-theme
+Widgets=Arc
 SymbolicIcons=true
 IconTheme=EvoPop
 DarkControls=false
-- 
2.4.5

