From 71fa2a2da11a2d259e1411e0351c4f5868ce501c Mon Sep 17 00:00:00 2001
From: Ikey Doherty <ikey@solus-project.com>
Date: Wed, 12 Oct 2016 03:35:47 +0100
Subject: [PATCH 1/8] disksops: --allow-discards is for open only, not create

Signed-off-by: Ikey Doherty <ikey@solus-project.com>
---
 os_installer2/diskops.py | 6 ------
 1 file changed, 6 deletions(-)

diff --git a/os_installer2/diskops.py b/os_installer2/diskops.py
index d089518..f2fb2e2 100644
--- a/os_installer2/diskops.py
+++ b/os_installer2/diskops.py
@@ -12,7 +12,6 @@
 #
 
 from os_installer2 import format_size_local
-from .diskman import DiskManager
 import parted
 import subprocess
 import tempfile
@@ -343,11 +342,6 @@ class DiskOpCreateLUKSContainer(DiskOpCreatePartition):
             # Not wanting to run into issues..
             cmd += " --force-password --batch-mode"
 
-            # SSD? --allow-discards
-            dev_path = self.device.path
-            if DiskManager.is_device_ssd(dev_path):
-                cmd += " --allow-discards"
-
             subprocess.check_call(cmd, shell=True)
 
             # Now we want to open it so it's usable.
-- 
2.10.1

