From 4659e0655b91e769bf26c95f8a829ae612683f58 Mon Sep 17 00:00:00 2001
From: Ikey Doherty <ikey@solus-project.com>
Date: Wed, 1 Jul 2015 06:32:07 +0100
Subject: [PATCH] Fix case sensitivity with EFI installation (dual-boot)

Signed-off-by: Ikey Doherty <ikey@solus-project.com>
---
 os_installer/installer.py | 43 +++++++++++++++++++++++++++++++++++++------
 1 file changed, 37 insertions(+), 6 deletions(-)

diff --git a/os_installer/installer.py b/os_installer/installer.py
index cc3366f..f7faee7 100644
--- a/os_installer/installer.py
+++ b/os_installer/installer.py
@@ -419,18 +419,50 @@ EndSection\n""" % (setup.keyboard_model, setup.keyboard_layout))
         grubfh.writelines(grub_output)
         grubfh.close()
 
+
+    def get_ichild(self, root,child):
+        t1 = os.path.join(root, child)
+        if os.path.exists(t1) or not os.path.exists(root):
+            return t1
+        try:
+            for i in os.listdir(root):
+                i2 = i.lower()
+                if i2 == child:
+                    return os.path.join(root, i)
+        except Exception, ex:
+            print("Error obtaining %s dir: %s" % (child, ex))
+        return t1
+
+    def get_efi_dir(self, base):
+        return self.get_ichild(base, "efi")
+
+    def get_loader_dir(self, base):
+        return self.get_ichild(base, "loader")
+
+    def get_efi_boot_dir(self, base):
+        return self.get_ichild(self.get_efi_dir(base), "boot")
+
+    def get_efi_boot_file(self, base):
+        return self.get_ichild(self.get_efi_boot_dir(base), "bootx64.efi")
+
+    def get_loader_entries(self, base):
+        return self.get_ichild(self.get_loader_dir(base), "entries")
+
+    def get_solus_dir(self, base):
+        return self.get_ichild(base, "solus")
+
     def do_gummy(self, our_total, our_current):
         self.update_progress(pulse=True, total=our_total, current=our_current, message=_("Configuring bootloader"))
-        tgt = "/target/boot/efi/efi/boot/bootx64.efi"
+        tgt = self.get_efi_boot_file("/target/boot/efi")
         if not os.path.exists(tgt):
-            dirn = "/target/boot/efi/efi/boot/"
+            dirn = self.get_efi_boot_dir("/target/boot/efi")
             if not os.path.exists(dirn):
                 os.makedirs(dirn)
             shutil.copy("/usr/lib/gummiboot/gummibootx64.efi", tgt)
 
-        entfile = "/target/boot/efi/loader/default.conf"
+        entfile = os.path.join(self.get_loader_dir("/target/boot/efi"), "default.conf")
         if not os.path.exists(entfile):
-            dirn = "/target/boot/efi/loader"
+            dirn = self.get_loader_dir("/target/boot/efi")
             if not os.path.exists(dirn):
                 os.makedirs(dirn)
             with open(entfile, "w") as defconf:
@@ -441,8 +473,7 @@ EndSection\n""" % (setup.keyboard_model, setup.keyboard_layout))
         with open(solfile, "w") as solconf:
             solconf.write("title Solus\nlinux /solus/kernel\ninitrd /solus/initramfs\noptions root=%s quiet\n" % self.root_partition)
         kver = os.uname()[2]
-        print kver
-        sdir = "/target/boot/efi/solus"
+        sdir = get_solus_dir("/target/boot/efi")
         if not os.path.exists(sdir):
             os.makedirs(sdir)
         kernel = "/source/boot/kernel-%s" % kver
-- 
2.4.5

