From edbefd7af24149b1ba9454e8a839e40124735257 Mon Sep 17 00:00:00 2001
From: Ikey Doherty <ikey@solus-project.com>
Date: Tue, 1 Mar 2016 03:38:58 +0000
Subject: [PATCH 2/6] Enforce lightdm

Signed-off-by: Ikey Doherty <ikey@solus-project.com>
---
 data/lightdm.conf | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/data/lightdm.conf b/data/lightdm.conf
index 7fe9c98..7f84a5f 100644
--- a/data/lightdm.conf
+++ b/data/lightdm.conf
@@ -102,7 +102,7 @@
 #greeter-allow-guest=true
 #greeter-show-manual-login=false
 #greeter-show-remote-login=true
-#user-session=default
+user-session=budgie-desktop
 #allow-user-switching=true
 #allow-guest=true
 #guest-session=
-- 
2.8.1

