From 3e82c65b284fce593538c22d94958a026d3a042f Mon Sep 17 00:00:00 2001
From: Ikey Doherty <ikey@solus-project.com>
Date: Sun, 17 May 2015 17:26:06 +0100
Subject: [PATCH 2/2] disks: Ensure we refresh ESP

---
 os_installer/pages/disks.py | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/os_installer/pages/disks.py b/os_installer/pages/disks.py
index cbc119b..1cd3551 100644
--- a/os_installer/pages/disks.py
+++ b/os_installer/pages/disks.py
@@ -228,6 +228,7 @@ class DiskPage(BasePage):
         os.system("gparted %s" % self.target_disk)
         self.installer.can_go_forward(False)
         self.build_partitions()
+        self.build_esp()
 
     def _partition_selected(self, selection):
         model, treeiter = selection.get_selected()
@@ -301,7 +302,8 @@ class DiskPage(BasePage):
                     if fs.type in ["fat", "fat32"]:
                         f = partition.getFlag(parted.PARTITION_BOOT)
                         if f:
-                            esp.append(partition.path)
+                            if partition.path not in esp:
+                                esp.append(partition.path)
                 partition = partition.nextPartition()
         self.installer.suggestions["esp"] = esp
 
-- 
2.3.2

