From cd0f3c2d8c13fc10c08695ced1885910bd870c5f Mon Sep 17 00:00:00 2001
From: Ikey Doherty <ikey@solus-project.com>
Date: Fri, 10 Jun 2016 23:08:32 +0100
Subject: [PATCH 3/6] Preserve the .disk member

Signed-off-by: Ikey Doherty <ikey@solus-project.com>
---
 os_installer2/diskops.py | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/os_installer2/diskops.py b/os_installer2/diskops.py
index 7eb143b..74403ea 100644
--- a/os_installer2/diskops.py
+++ b/os_installer2/diskops.py
@@ -22,10 +22,12 @@ class BaseDiskOp:
     device = None
     errors = None
     part_offset = 0
+    disk = None
 
     def __init__(self, device):
         self.device = device
-        pass
+        if device.disk is not None:
+            self.disk = device.disk
 
     def describe(self):
         """ Describe this operation """
-- 
2.8.3

