From a1af63ec5805c51fa649103f6b1b8ceef53d29d4 Mon Sep 17 00:00:00 2001
From: Ikey Doherty <ikey@solus-project.com>
Date: Tue, 1 Mar 2016 03:40:27 +0000
Subject: [PATCH 3/5] Fix the UEFI installation options (ro+timeout)

Signed-off-by: Ikey Doherty <ikey@solus-project.com>
---
 os_installer/installer.py | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/os_installer/installer.py b/os_installer/installer.py
index 1c1ac8e..2e3b2a6 100644
--- a/os_installer/installer.py
+++ b/os_installer/installer.py
@@ -506,14 +506,14 @@ EndSection\n""" % (setup.keyboard_model, setup.keyboard_layout))
             if not os.path.exists(dirn):
                 os.makedirs(dirn)
             with open(entfile, "w") as defconf:
-                defconf.write("default solus\ntimeout 4\n")
+                defconf.write("timeout 4\ndefault solus\n")
 
         # Write the Solus entry
         solfile = "/target/boot/efi/loader/entries/solus.conf"
         if not os.path.exists(os.path.dirname(solfile)):
             os.makedirs(os.path.dirname(solfile))
         with open(solfile, "w") as solconf:
-            solconf.write("title Solus Operating System\nlinux /solus/kernel\ninitrd /solus/initramfs\noptions root=%s quiet\n" % self.root_partition)
+            solconf.write("title Solus Operating System 1.1\nlinux /solus/kernel\ninitrd /solus/initramfs\noptions root=%s quiet ro\n" % self.root_partition)
 
         # Now install the Solus kernel/initramfs.
         kver = os.uname()[2]
-- 
2.8.1

