From ac32e2bf36640b9f7a2367686983442bf638f8c0 Mon Sep 17 00:00:00 2001
From: Ikey Doherty <ikey@solus-project.com>
Date: Fri, 10 Jun 2016 23:09:24 +0100
Subject: [PATCH 4/5] Incorporate missing SVGs

Signed-off-by: Ikey Doherty <ikey@solus-project.com>
---
 setup.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/setup.py b/setup.py
index d30b07e..e27c2af 100644
--- a/setup.py
+++ b/setup.py
@@ -11,5 +11,5 @@ setup(
     packages        = ['os_installer2', 'os_installer2.pages'],
     scripts         = ['os-installer-gtk'],
     classifiers     = [ "License :: OSI Approved :: GPL-2.0 License"],
-    package_data    = {'os_installer2': ['data/*.png']},
+    package_data    = {'os_installer2': ['data/*.png', 'data/*.svg']},
 )
-- 
2.8.3

