From 975abe3321e5d5fda45c48498a7e5dfb5f554684 Mon Sep 17 00:00:00 2001
From: Ikey Doherty <ikey@solus-project.com>
Date: Mon, 27 Mar 2017 23:56:24 +0100
Subject: [PATCH 5/6] progress: stop overwriting original errors

Signed-off-by: Ikey Doherty <ikey@solus-project.com>
---
 os_installer2/pages/progress.py | 1 -
 1 file changed, 1 deletion(-)

diff --git a/os_installer2/pages/progress.py b/os_installer2/pages/progress.py
index 7e7e5b9..7336c14 100644
--- a/os_installer2/pages/progress.py
+++ b/os_installer2/pages/progress.py
@@ -721,7 +721,6 @@ class InstallerProgressPage(BasePage):
         # Copy source -> target
         if not self.copy_system():
             self.filesystem_copying = False
-            self.set_error_message("Failed to copy filesystem")
             self.unmount_all()
             self.installing = False
             return False
-- 
2.12.2

