From 37267e852507484c144bedc50ba5399018f70224 Mon Sep 17 00:00:00 2001
From: Ikey Doherty <ikey@solus-project.com>
Date: Sat, 30 Apr 2016 14:59:18 +0100
Subject: [PATCH 5/5] Remove the livecd configuration on post-install

Signed-off-by: Ikey Doherty <ikey@solus-project.com>
---
 os_installer/installer.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/os_installer/installer.py b/os_installer/installer.py
index 5235cc3..9607953 100644
--- a/os_installer/installer.py
+++ b/os_installer/installer.py
@@ -277,7 +277,7 @@ class InstallerEngine:
             # Again Solus specific, but remove the installer from the image
             our_current += 1
             self.update_progress(total=our_total, current=our_current, message=_("Removing live configuration (installer)"))
-            self.do_run_in_chroot("eopkg remove os-installer --ignore-comar")
+            self.do_run_in_chroot("eopkg remove os-installer budgie-desktop-branding-livecd --ignore-comar")
             # Ensure we clean this due to livecd overrides
             self.do_run_in_chroot("glib-compile-schemas /usr/share/glib-2.0/schemas")
 
-- 
2.8.1

