From 8a016995b56aa31ec6dd767d9914d7a482ab0498 Mon Sep 17 00:00:00 2001
From: Ikey Doherty <ikey@solus-project.com>
Date: Tue, 28 Mar 2017 00:02:10 +0100
Subject: [PATCH 6/8] postinstall: Add nasty hack to trick CBM into using our
 specified ESP

Signed-off-by: Ikey Doherty <ikey@solus-project.com>
---
 os_installer2/postinstall.py | 13 +++++++++++++
 1 file changed, 13 insertions(+)

diff --git a/os_installer2/postinstall.py b/os_installer2/postinstall.py
index 2bb5562..4ca4d17 100644
--- a/os_installer2/postinstall.py
+++ b/os_installer2/postinstall.py
@@ -779,6 +779,19 @@ class PostInstallBootloader(PostInstallStep):
         if not updated_uefi:
             self.set_errors("Failed to install goofiboot")
             return False
+
+        # Tell clr-boot-manager where our ESP is for now
+        try:
+            if not os.path.exists("/dev/disk/by-partlabel"):
+                os.makedirs("/dev/disk/by-partlabel", 00755)
+            if os.path.exists("/dev/disk/by-partlabel/ESP"):
+                os.unlink("/dev/disk/by-partlabel/ESP")
+            os.symlink("/dev/{}".format(self.installer.locate_esp()),
+                       "/dev/disk/by-partlabel/ESP")
+        except Exception as e:
+            self.set_errors("Failed to simulate ESP link: {}".format(e))
+            return False
+
         # Proxy back to CBM
         return self.apply_boot_loader()
 
-- 
2.12.2

