From 3289a005093d7c9cca4b144894b756b7e545f30d Mon Sep 17 00:00:00 2001
From: Ikey Doherty <ikey@solus-project.com>
Date: Fri, 10 Jun 2016 23:30:22 +0100
Subject: [PATCH 06/14] Implement the BIOS bootloader install (aka GRUB..)

Signed-off-by: Ikey Doherty <ikey@solus-project.com>
---
 os_installer2/postinstall.py | 14 ++++++++++++--
 1 file changed, 12 insertions(+), 2 deletions(-)

diff --git a/os_installer2/postinstall.py b/os_installer2/postinstall.py
index 50014e9..75438f9 100644
--- a/os_installer2/postinstall.py
+++ b/os_installer2/postinstall.py
@@ -592,8 +592,18 @@ class PostInstallBootloader(PostInstallStep):
 
     def apply_bios(self):
         """ Take the BIOS approach to bootloader configuration """
-        self.set_errors("Sorry, still working on this BIOS doflickey!")
-        return False
+        if not self.info.bootloader_install:
+            return True
+        cmd = "grub-install \"{}\"".format(self.info.bootloader_sz)
+        if not self.run_in_chroot(cmd):
+            self.set_errors("Failed to install GRUB bootloader")
+            return False
+
+        cmd = "grub-mkconfig -o /boot/grub/grub.cfg"
+        if not self.run_in_chroot(cmd):
+            self.set_errors("Failed to update GRUB bootloader configuration")
+            return False
+        return True
 
     def apply_uefi(self):
         """ Take the UEFI approach to bootloader configuration """
-- 
2.8.3

