From 07d1c806d51b91520c430b52b5b577dca57475e5 Mon Sep 17 00:00:00 2001
From: Ikey Doherty <ikey@solus-project.com>
Date: Sat, 30 Apr 2016 15:16:15 +0100
Subject: [PATCH 6/6] Fix syntax issue

Signed-off-by: Ikey Doherty <ikey@solus-project.com>
---
 os_installer/installer.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/os_installer/installer.py b/os_installer/installer.py
index 9607953..70ae38a 100644
--- a/os_installer/installer.py
+++ b/os_installer/installer.py
@@ -75,7 +75,7 @@ class InstallerEngine:
                         p = Popen(cmd, shell=True)
                         p.wait()
                     except Exception as e:
-                        print("Failed setting mount-count: {}".format(e)
+                        print("Failed setting mount-count: {}".format(e))
                                         
     def step_mount_partitions(self, setup):
         # Mount the installation media
-- 
2.8.1

