From 4e3e38503e74a0a4ddf35595cfffade7c9e131e6 Mon Sep 17 00:00:00 2001
From: Ikey Doherty <ikey@solus-project.com>
Date: Fri, 14 Oct 2016 07:00:47 +0100
Subject: [PATCH 07/10] diskops: Fix broken string for DiskOpFormatSwapLate

Signed-off-by: Ikey Doherty <ikey@solus-project.com>
---
 os_installer2/diskops.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/os_installer2/diskops.py b/os_installer2/diskops.py
index f2fb2e2..08de4fc 100644
--- a/os_installer2/diskops.py
+++ b/os_installer2/diskops.py
@@ -758,7 +758,7 @@ class DiskOpFormatSwapLate(DiskOpFormatPartition):
         DiskOpFormatPartition.__init__(self, device, part, "swap")
 
     def describe(self):
-        return "Use {} as {} swap partition".format(
+        return "Format {} as {} partition".format(
             self.part.path, self.format_type)
 
     def apply(self, disk, simulate):
-- 
2.10.1

