From 15739dd0efdf6eaa8b7d9fe23b6ae09cb44ded5b Mon Sep 17 00:00:00 2001
From: Ikey Doherty <ikey@solus-project.com>
Date: Tue, 28 Mar 2017 00:15:02 +0100
Subject: [PATCH 07/31] postinstall: Fix broken ESP emulation

Signed-off-by: Ikey Doherty <ikey@solus-project.com>
---
 os_installer2/postinstall.py | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/os_installer2/postinstall.py b/os_installer2/postinstall.py
index 4ca4d17..fbe2fa5 100644
--- a/os_installer2/postinstall.py
+++ b/os_installer2/postinstall.py
@@ -784,9 +784,9 @@ class PostInstallBootloader(PostInstallStep):
         try:
             if not os.path.exists("/dev/disk/by-partlabel"):
                 os.makedirs("/dev/disk/by-partlabel", 00755)
-            if os.path.exists("/dev/disk/by-partlabel/ESP"):
-                os.unlink("/dev/disk/by-partlabel/ESP")
-            os.symlink("/dev/{}".format(self.installer.locate_esp()),
+            if os.path.lexists("/dev/disk/by-partlabel/ESP"):
+                os.remove("/dev/disk/by-partlabel/ESP")
+            os.symlink(self.installer.locate_esp(),
                        "/dev/disk/by-partlabel/ESP")
         except Exception as e:
             self.set_errors("Failed to simulate ESP link: {}".format(e))
-- 
2.13.2

