From d10023a15c5e2aedd2cd33f817f2a6534b43f8f1 Mon Sep 17 00:00:00 2001
From: Ikey Doherty <ikey@solus-project.com>
Date: Tue, 28 Mar 2017 00:16:01 +0100
Subject: [PATCH 08/11] progress: Temporarily return to /boot/efi mountpoint

Signed-off-by: Ikey Doherty <ikey@solus-project.com>
---
 os_installer2/pages/progress.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/os_installer2/pages/progress.py b/os_installer2/pages/progress.py
index 7336c14..f62f818 100644
--- a/os_installer2/pages/progress.py
+++ b/os_installer2/pages/progress.py
@@ -244,7 +244,7 @@ class InstallerProgressPage(BasePage):
         """ Mount the ESP into the root """
         self.set_display_string("Mounting EFI System Partition")
         root = self.get_installer_target_filesystem()
-        fpath = os.path.join(root, "boot")
+        fpath = os.path.join(root, "boot/efi")
         if not os.path.exists(fpath):
             try:
                 os.makedirs(fpath, mode=0o755)
-- 
2.12.2

