From dfb70334e8730bfa0a2802f481a9a9b5cef88771 Mon Sep 17 00:00:00 2001
From: Ikey Doherty <ikey@solus-project.com>
Date: Fri, 10 Jun 2016 23:54:59 +0100
Subject: [PATCH 08/17] Ensure bootloader install is enabled by default..

Signed-off-by: Ikey Doherty <ikey@solus-project.com>
---
 os_installer2/mainwindow.py | 1 +
 1 file changed, 1 insertion(+)

diff --git a/os_installer2/mainwindow.py b/os_installer2/mainwindow.py
index 17e069e..0083dc7 100644
--- a/os_installer2/mainwindow.py
+++ b/os_installer2/mainwindow.py
@@ -76,6 +76,7 @@ class InstallInfo:
 
     def __init__(self):
         self.users = list()
+        self.bootloader_install = True
 
 
 class MainWindow(Gtk.ApplicationWindow):
-- 
2.8.3

