From f56d8ac9a66a04a233bf433c136b4a9bd50959a8 Mon Sep 17 00:00:00 2001
From: Ikey Doherty <ikey@solus-project.com>
Date: Sat, 15 Oct 2016 21:07:37 +0100
Subject: [PATCH 08/11] mainwindow: Explicitly set icon name for the MATE
 Desktop

Signed-off-by: Ikey Doherty <ikey@solus-project.com>
---
 os_installer2/mainwindow.py | 1 +
 1 file changed, 1 insertion(+)

diff --git a/os_installer2/mainwindow.py b/os_installer2/mainwindow.py
index 7fb3358..0d1aff9 100644
--- a/os_installer2/mainwindow.py
+++ b/os_installer2/mainwindow.py
@@ -114,6 +114,7 @@ class MainWindow(Gtk.ApplicationWindow):
         self.application = app
 
         self.set_title("Installer")
+        self.set_icon_name("system-software-install")
         self.connect("delete-event", self.quit_handler)
 
         self.set_position(Gtk.WindowPosition.CENTER)
-- 
2.10.1

