From ced95943fd608a6fd066c65afa5f4dd19e22c1b6 Mon Sep 17 00:00:00 2001
From: Ikey Doherty <ikey@solus-project.com>
Date: Sat, 11 Jun 2016 00:08:12 +0100
Subject: [PATCH 10/20] Actually create the user's home directory, I hear it's
 useful.

Signed-off-by: Ikey Doherty <ikey@solus-project.com>
---
 os_installer2/postinstall.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/os_installer2/postinstall.py b/os_installer2/postinstall.py
index a4ae5e9..eee7616 100644
--- a/os_installer2/postinstall.py
+++ b/os_installer2/postinstall.py
@@ -389,7 +389,7 @@ class PostInstallUsers(PostInstallStep):
             if user.admin:
                 groups.extend(self.admin_groups)
 
-            cmd = "useradd -s {} -c '{}' -G {} {}".format(
+            cmd = "useradd -s {} -c '{}' -G {} -m {}".format(
                 "/bin/bash",  # Default shell in Solus
                 user.realname,
                 ",".join(groups),
-- 
2.8.3

