From ad85dd8e2ae70201ba7bc4b93fb10c8c782433e0 Mon Sep 17 00:00:00 2001
From: Ikey Doherty <ikey@solus-project.com>
Date: Tue, 28 Mar 2017 00:32:14 +0100
Subject: [PATCH 10/35] postinstall: Actually return True.

Signed-off-by: Ikey Doherty <ikey@solus-project.com>
---
 os_installer2/postinstall.py | 1 +
 1 file changed, 1 insertion(+)

diff --git a/os_installer2/postinstall.py b/os_installer2/postinstall.py
index f4c4c81..0ac5d9d 100644
--- a/os_installer2/postinstall.py
+++ b/os_installer2/postinstall.py
@@ -795,6 +795,7 @@ class PostInstallBootloader(PostInstallStep):
         if not updated_uefi:
             self.set_errors("Failed to install goofiboot")
             return False
+        return True
 
     def get_ichild(self, root, child):
         t1 = os.path.join(root, child)
-- 
2.13.3

