From 04479c00f6e24fe0ffa7e3bbd3c7cca3ca9605a7 Mon Sep 17 00:00:00 2001
From: Ikey Doherty <ikey@solus-project.com>
Date: Sun, 16 Oct 2016 00:10:38 +0100
Subject: [PATCH 10/11] postinstall: Remove ability to replace lightdm.conf

Signed-off-by: Ikey Doherty <ikey@solus-project.com>
---
 os_installer2/postinstall.py | 1 -
 1 file changed, 1 deletion(-)

diff --git a/os_installer2/postinstall.py b/os_installer2/postinstall.py
index 05f4891..b8a37b8 100644
--- a/os_installer2/postinstall.py
+++ b/os_installer2/postinstall.py
@@ -139,7 +139,6 @@ class PostInstallRemoveLiveConfig(PostInstallStep):
 
         self.original_source = "/usr/share/os-installer"
         self.modified_files = [
-            "/etc/lightdm/lightdm.conf",
             "/etc/gdm/custom.conf"
         ]
 
-- 
2.10.1

