From 5f7fc4ba68c082d6e3ca16735eeb2cfa314c6b13 Mon Sep 17 00:00:00 2001
From: Ikey Doherty <ikey@solus-project.com>
Date: Sat, 11 Jun 2016 00:16:59 +0100
Subject: [PATCH 11/21] Write out the root partition in the fstab

Signed-off-by: Ikey Doherty <ikey@solus-project.com>
---
 os_installer2/postinstall.py | 13 +++++++++++--
 1 file changed, 11 insertions(+), 2 deletions(-)

diff --git a/os_installer2/postinstall.py b/os_installer2/postinstall.py
index eee7616..bd64e64 100644
--- a/os_installer2/postinstall.py
+++ b/os_installer2/postinstall.py
@@ -509,7 +509,8 @@ FSTAB_HEADER = """
 # /dev/ROOT   /            ext3    noatime        0 1
 # /dev/SWAP   none         swap    sw             0 0
 # /dev/fd0    /mnt/floppy  auto    noauto         0 0
-proc\t/proc\tproc\tdefaults\t0\t0
+none        /proc        proc    nosuid,noexec  0 0
+none        /dev/shm     tmpfs   defaults       0 0
 """
 
 
@@ -542,7 +543,7 @@ class PostInstallFstab(PostInstallStep):
 
         for op in strat.get_operations():
             # TODO: Add custom mountpoints here!
-            # Skip / and swap for GPT/UEFI
+            # Skip swap for GPT/UEFI
             if disk.type == "gpt" and strat.is_uefi():
                 continue
 
@@ -563,6 +564,14 @@ class PostInstallFstab(PostInstallStep):
             else:
                 appends.append("{}\tswap\tswap\tsw\t0\t0".format(swap_path))
 
+        # Add the root partition last
+        root = strat.get_root_partition()
+        uuid = self.get_part_uuid(root)
+        appends.append("# {} at time of installation".format(root))
+        appends.append(
+            "UUID={}\t/\text4\trw,relatime,errors=remount-ro\t0\t0".format(
+                uuid))
+
         fp = os.path.join(self.installer.get_installer_target_filesystem(),
                           "etc/fstab")
 
-- 
2.8.3

