From fc936d75f3be677aa6b260876db8ab767220ef8a Mon Sep 17 00:00:00 2001
From: Ikey Doherty <ikey@solus-project.com>
Date: Tue, 28 Mar 2017 00:41:08 +0100
Subject: [PATCH 11/14] postinstall: Stop writing the /boot mount in fstab for
 the ESP

Signed-off-by: Ikey Doherty <ikey@solus-project.com>
---
 os_installer2/postinstall.py | 11 -----------
 1 file changed, 11 deletions(-)

diff --git a/os_installer2/postinstall.py b/os_installer2/postinstall.py
index 0ac5d9d..ee98ad4 100644
--- a/os_installer2/postinstall.py
+++ b/os_installer2/postinstall.py
@@ -577,17 +577,6 @@ class PostInstallFstab(PostInstallStep):
 
         ext4_ops = "rw,relatime,errors=remount-ro"
 
-        # Add the ESP to /boot
-        if strat.is_uefi() and self.info.bootloader_install:
-            esp = self.installer.locate_esp()
-            uuid = get_part_uuid(esp, True)
-            if uuid:
-                esp_ent = "PARTUUID={}\t/boot\tvfat\tdefaults\t0\t0"
-                appends.append(esp_ent.format(uuid))
-            else:
-                esp_ent = "{}\t/boot\tvfat\tdefaults\t0\t0"
-                appends.append(esp_ent.format(esp))
-
         for op in strat.get_operations():
             # TODO: Add custom mountpoints here!
             # Skip swap for GPT/UEFI
-- 
2.12.2

