From da6a09320eb61acee7269c1734f1b4bbbfd1a5ca Mon Sep 17 00:00:00 2001
From: Ikey Doherty <ikey@solus-project.com>
Date: Sat, 11 Jun 2016 00:22:03 +0100
Subject: [PATCH 12/25] Fix slightly minor issue^HMonumental fuckup

Signed-off-by: Ikey Doherty <ikey@solus-project.com>
---
 os_installer2/postinstall.py | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/os_installer2/postinstall.py b/os_installer2/postinstall.py
index bd64e64..f05215e 100644
--- a/os_installer2/postinstall.py
+++ b/os_installer2/postinstall.py
@@ -557,7 +557,7 @@ class PostInstallFstab(PostInstallStep):
             if not swap_path:
                 continue
 
-            uuid = self.get_part_uuid(swap_path)
+            uuid = get_part_uuid(swap_path)
             if uuid:
                 im = "UUID={}\tswap\tswap\tsw\t0\t0".format(swap_path)
                 appends.append(im)
@@ -566,7 +566,7 @@ class PostInstallFstab(PostInstallStep):
 
         # Add the root partition last
         root = strat.get_root_partition()
-        uuid = self.get_part_uuid(root)
+        uuid = get_part_uuid(root)
         appends.append("# {} at time of installation".format(root))
         appends.append(
             "UUID={}\t/\text4\trw,relatime,errors=remount-ro\t0\t0".format(
-- 
2.8.3

