From 05ea943c4a41bc7edd6d353ec2daebe100fb372e Mon Sep 17 00:00:00 2001
From: Ikey Doherty <ikey@solus-project.com>
Date: Sat, 11 Jun 2016 11:08:48 +0100
Subject: [PATCH 13/25] postinstall: Allow nvvar modification

Signed-off-by: Ikey Doherty <ikey@solus-project.com>
---
 os_installer2/postinstall.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/os_installer2/postinstall.py b/os_installer2/postinstall.py
index f05215e..72bc7b3 100644
--- a/os_installer2/postinstall.py
+++ b/os_installer2/postinstall.py
@@ -625,7 +625,7 @@ class PostInstallBootloader(PostInstallStep):
         uuid = get_part_uuid(root_part, True)
 
         espt = self.installer.get_esp_target()
-        cmd = "goofiboot install --path=\"{}\" --no-variables".format(espt)
+        cmd = "goofiboot install --path=\"{}\"".format(espt)
         try:
             subprocess.check_call(cmd, shell=True)
         except Exception as e:
-- 
2.8.3

