From 8a7e038c5324498f880821159c67f5e5a44e43eb Mon Sep 17 00:00:00 2001
From: Ikey Doherty <ikey@solus-project.com>
Date: Sat, 11 Jun 2016 11:11:58 +0100
Subject: [PATCH 14/24] diskops: Ensure root partition is set bootable on msdos
 disk

Signed-off-by: Ikey Doherty <ikey@solus-project.com>
---
 os_installer2/diskops.py | 14 ++++++++++++++
 1 file changed, 14 insertions(+)

diff --git a/os_installer2/diskops.py b/os_installer2/diskops.py
index fae9cff..dc7221f 100644
--- a/os_installer2/diskops.py
+++ b/os_installer2/diskops.py
@@ -229,6 +229,20 @@ class DiskOpCreateRoot(DiskOpCreatePartition):
             return False
         return True
 
+    def apply(self, disk, simulate):
+        """ Create root partition, set to bootable on msdos """
+        b = DiskOpCreatePartition.apply(self, disk, simulate)
+        if not b:
+            return b
+        if disk.type != "msdos":
+            return True
+        try:
+            self.part.setFlag(parted.PARTITION_BOOT)
+        except Exception as e:
+            self.set_errors("Cannot set root as bootable: {}".format(e))
+            return False
+        return True
+
 
 class DiskOpUseSwap(BaseDiskOp):
     """ Use an existing swap paritition """
-- 
2.8.3

