From 76ca1ee02cd6b5c58cd88e62cb68d895b289e3e9 Mon Sep 17 00:00:00 2001
From: Ikey Doherty <ikey@solus-project.com>
Date: Tue, 28 Mar 2017 22:21:25 +0100
Subject: [PATCH 14/18] progress: Make the permissions error more informative

Signed-off-by: Ikey Doherty <ikey@solus-project.com>
---
 os_installer2/pages/progress.py | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/os_installer2/pages/progress.py b/os_installer2/pages/progress.py
index 7336c14..fe436db 100644
--- a/os_installer2/pages/progress.py
+++ b/os_installer2/pages/progress.py
@@ -472,7 +472,8 @@ class InstallerProgressPage(BasePage):
                     # Update progress
                     self.filesystem_copied_size += st.st_size
                 except Exception as ex:
-                    self.set_error_message("Permissions issue: {}".format(ex))
+                    self.set_error_message("Permissions issue: {} {}".format(
+                        ex, d))
                     return False
 
         self.set_display_string("Finalizing file copy")
-- 
2.13.2

