From 15c98ed5f5b0e29ea2776aa003225e93002e76de Mon Sep 17 00:00:00 2001
From: Ikey Doherty <ikey@solus-project.com>
Date: Fri, 7 Apr 2017 02:33:34 +0100
Subject: [PATCH 15/31] postinstall: Ensure we remove GNOME Desktop branding
 too now

Signed-off-by: Ikey Doherty <ikey@solus-project.com>
---
 os_installer2/postinstall.py | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/os_installer2/postinstall.py b/os_installer2/postinstall.py
index cb05ca1..6abad66 100644
--- a/os_installer2/postinstall.py
+++ b/os_installer2/postinstall.py
@@ -134,6 +134,8 @@ class PostInstallRemoveLiveConfig(PostInstallStep):
         # Find the branding package to nuke
         if os.path.exists("/usr/bin/mate-panel"):
             self.live_packages.append("mate-desktop-branding-livecd")
+        elif os.path.exists("/usr/bin/gnome-shell"):
+            self.live_packages.append("gnome-desktop-branding-livecd")
         else:
             self.live_packages.append("budgie-desktop-branding-livecd")
 
-- 
2.13.2

