From 7ea1eeff7b12910eba3a4a9a3a06b377751db07c Mon Sep 17 00:00:00 2001
From: Ikey Doherty <ikey@solus-project.com>
Date: Sat, 11 Jun 2016 11:37:01 +0100
Subject: [PATCH 15/15] Switch that to using UUID only

Signed-off-by: Ikey Doherty <ikey@solus-project.com>
---
 os_installer2/postinstall.py | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/os_installer2/postinstall.py b/os_installer2/postinstall.py
index 72bc7b3..e3c4fea 100644
--- a/os_installer2/postinstall.py
+++ b/os_installer2/postinstall.py
@@ -622,7 +622,7 @@ class PostInstallBootloader(PostInstallStep):
         """ Take the UEFI approach to bootloader configuration """
         bpath = self.installer.get_installer_target_filesystem()
         root_part = self.info.strategy.get_root_partition()
-        uuid = get_part_uuid(root_part, True)
+        uuid = get_part_uuid(root_part)
 
         espt = self.installer.get_esp_target()
         cmd = "goofiboot install --path=\"{}\"".format(espt)
@@ -657,7 +657,7 @@ class PostInstallBootloader(PostInstallStep):
                 conf = [
                     "title Solus 1.2",
                     "linux /solus/kernel",
-                    "options root=PARTUUID={} quiet ro".format(uuid)
+                    "options root=UUID={} quiet ro".format(uuid)
                 ]
                 solconf.write("\n".join(conf) + "\n")
         except Exception as e:
-- 
2.8.3

