From 777b21c69f60dcd1ca3b5222a150b2b24a2c7f5d Mon Sep 17 00:00:00 2001
From: Ikey Doherty <ikey@solus-project.com>
Date: Sat, 11 Jun 2016 11:52:41 +0100
Subject: [PATCH 16/23] Add missing initramfs

Signed-off-by: Ikey Doherty <ikey@solus-project.com>
---
 os_installer2/postinstall.py | 1 +
 1 file changed, 1 insertion(+)

diff --git a/os_installer2/postinstall.py b/os_installer2/postinstall.py
index e3c4fea..ccb82f5 100644
--- a/os_installer2/postinstall.py
+++ b/os_installer2/postinstall.py
@@ -657,6 +657,7 @@ class PostInstallBootloader(PostInstallStep):
                 conf = [
                     "title Solus 1.2",
                     "linux /solus/kernel",
+                    "initrd /solus/initramfs",
                     "options root=UUID={} quiet ro".format(uuid)
                 ]
                 solconf.write("\n".join(conf) + "\n")
-- 
2.8.3

