From bb0d731d9040f715e13f30dcf57610bab242e136 Mon Sep 17 00:00:00 2001
From: Ikey Doherty <ikey@solus-project.com>
Date: Fri, 7 Jul 2017 02:45:16 +0100
Subject: [PATCH 17/18] Grow a headerbar, everyone is doing it

Signed-off-by: Ikey Doherty <ikey@solus-project.com>
---
 os_installer2/mainwindow.py | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/os_installer2/mainwindow.py b/os_installer2/mainwindow.py
index 0d1aff9..ade27bd 100644
--- a/os_installer2/mainwindow.py
+++ b/os_installer2/mainwindow.py
@@ -113,6 +113,10 @@ class MainWindow(Gtk.ApplicationWindow):
         Gtk.ApplicationWindow.__init__(self, application=app)
         self.application = app
 
+        header = Gtk.HeaderBar.new()
+        header.set_show_close_button(True)
+        self.set_titlebar(header)
+
         self.set_title("Installer")
         self.set_icon_name("system-software-install")
         self.connect("delete-event", self.quit_handler)
-- 
2.13.2

