From 86aea412b21602369c2b461ba486481f2ab79244 Mon Sep 17 00:00:00 2001
From: Ikey Doherty <ikey@solus-project.com>
Date: Sat, 11 Jun 2016 11:54:27 +0100
Subject: [PATCH 17/25] Only make / bootable after creating the filesystem on
 msdos disk

Signed-off-by: Ikey Doherty <ikey@solus-project.com>
---
 os_installer2/diskops.py | 8 --------
 1 file changed, 8 deletions(-)

diff --git a/os_installer2/diskops.py b/os_installer2/diskops.py
index dc7221f..29f070b 100644
--- a/os_installer2/diskops.py
+++ b/os_installer2/diskops.py
@@ -227,13 +227,6 @@ class DiskOpCreateRoot(DiskOpCreatePartition):
         except Exception as e:
             self.set_errors("{}: {}".format(self.part.path, e))
             return False
-        return True
-
-    def apply(self, disk, simulate):
-        """ Create root partition, set to bootable on msdos """
-        b = DiskOpCreatePartition.apply(self, disk, simulate)
-        if not b:
-            return b
         if disk.type != "msdos":
             return True
         try:
@@ -241,7 +234,6 @@ class DiskOpCreateRoot(DiskOpCreatePartition):
         except Exception as e:
             self.set_errors("Cannot set root as bootable: {}".format(e))
             return False
-        return True
 
 
 class DiskOpUseSwap(BaseDiskOp):
-- 
2.8.3

