From af7c633e731d7d3d408d027dd1041f17d3562b01 Mon Sep 17 00:00:00 2001
From: Ikey Doherty <ikey@solus-project.com>
Date: Sat, 11 Jun 2016 12:21:44 +0100
Subject: [PATCH 18/21] Try to fix that somewhat fatal issue

Signed-off-by: Ikey Doherty <ikey@solus-project.com>
---
 os_installer2/diskops.py | 14 ++++++++++++++
 1 file changed, 14 insertions(+)

diff --git a/os_installer2/diskops.py b/os_installer2/diskops.py
index 29f070b..b46858e 100644
--- a/os_installer2/diskops.py
+++ b/os_installer2/diskops.py
@@ -235,6 +235,20 @@ class DiskOpCreateRoot(DiskOpCreatePartition):
             self.set_errors("Cannot set root as bootable: {}".format(e))
             return False
 
+    def apply(self, disk, simulate):
+        """ Create root partition  """
+        b = DiskOpCreatePartition.apply(self, disk, simulate)
+        if not b:
+            return b
+        if disk.type != "msdos":
+            return True
+        try:
+            self.part.setFlag(parted.PARTITION_BOOT)
+        except Exception as e:
+            self.set_errors("Cannot set root as bootable: {}".format(e))
+            return False
+        return True
+
 
 class DiskOpUseSwap(BaseDiskOp):
     """ Use an existing swap paritition """
-- 
2.8.3

