From dc0abd1be7823ef311969a1e6c30a43c33c76632 Mon Sep 17 00:00:00 2001
From: Ikey Doherty <ikey@solus-project.com>
Date: Fri, 14 Jul 2017 11:39:32 +0100
Subject: [PATCH 19/35] Stop forcing dark theme

Signed-off-by: Ikey Doherty <ikey@solus-project.com>
---
 os_installer2/mainwindow.py | 2 --
 1 file changed, 2 deletions(-)

diff --git a/os_installer2/mainwindow.py b/os_installer2/mainwindow.py
index 090c13b..6cf2a3c 100644
--- a/os_installer2/mainwindow.py
+++ b/os_installer2/mainwindow.py
@@ -116,8 +116,6 @@ class MainWindow(Gtk.ApplicationWindow):
         Gtk.ApplicationWindow.__init__(self, application=app)
         self.application = app
 
-        Gtk.Settings.get_default().set_property("gtk-application-prefer-dark-theme", True)
-
         self.headerbar = Gtk.HeaderBar.new()
         self.headerbar.set_show_close_button(True)
         self.set_titlebar(self.headerbar)
-- 
2.13.3

