From 44705c7e989495dcc07bc99c41dc8fc8fa0ad6be Mon Sep 17 00:00:00 2001
From: Ikey Doherty <ikey@solus-project.com>
Date: Fri, 14 Jul 2017 11:40:37 +0100
Subject: [PATCH 20/31] Remove the ugly headerbar hacks

Signed-off-by: Ikey Doherty <ikey@solus-project.com>
---
 os_installer2/mainwindow.py | 23 -----------------------
 1 file changed, 23 deletions(-)

diff --git a/os_installer2/mainwindow.py b/os_installer2/mainwindow.py
index 6cf2a3c..3a3703d 100644
--- a/os_installer2/mainwindow.py
+++ b/os_installer2/mainwindow.py
@@ -103,9 +103,6 @@ class MainWindow(Gtk.ApplicationWindow):
     can_quit = True
     is_final_step = False
 
-    image_step = None
-    label_step = None
-
     def quit_handler(self, w, udata=None):
         """ Ensure quit stuff is sane ... """
         if not self.can_quit:
@@ -116,21 +113,6 @@ class MainWindow(Gtk.ApplicationWindow):
         Gtk.ApplicationWindow.__init__(self, application=app)
         self.application = app
 
-        self.headerbar = Gtk.HeaderBar.new()
-        self.headerbar.set_show_close_button(True)
-        self.set_titlebar(self.headerbar)
-
-        self.image_step = Gtk.Image.new_from_icon_name("system-software-install", Gtk.IconSize.LARGE_TOOLBAR)
-        self.image_step.set_pixel_size(32)
-        self.image_step.set_property("margin", 4)
-        self.label_step = Gtk.Label.new("")
-
-        box = Gtk.Box.new(Gtk.Orientation.HORIZONTAL, 0)
-        box.pack_start(self.label_step, False, False, 0)
-        self.headerbar.set_custom_title(box)
-        self.headerbar.pack_start(self.image_step)
-
-        self.set_title("")
         self.set_icon_name("system-software-install")
         self.connect("delete-event", self.quit_handler)
 
@@ -285,11 +267,6 @@ class MainWindow(Gtk.ApplicationWindow):
         else:
             self.set_can_previous(True)
         page.prepare(self.info)
-        mk = u"<span font-size='x-large'>{}</span>".format(page.get_title())
-        self.label_step.set_markup(mk)
-        self.image_step.set_from_icon_name(page.get_icon_name(),
-                                           Gtk.IconSize.LARGE_TOOLBAR)
-        self.image_step.set_pixel_size(32)
         self.installer_stack.set_visible_child_name(page.get_name())
 
     def set_can_previous(self, can_prev):
-- 
2.13.2

