From 1655eb755ba90e167fe887541660162b24aa541c Mon Sep 17 00:00:00 2001
From: Ikey Doherty <ikey@solus-project.com>
Date: Sat, 11 Jun 2016 12:48:34 +0100
Subject: [PATCH 20/21] Remove that it wont work

Signed-off-by: Ikey Doherty <ikey@solus-project.com>
---
 os_installer2/diskops.py | 8 +-------
 1 file changed, 1 insertion(+), 7 deletions(-)

diff --git a/os_installer2/diskops.py b/os_installer2/diskops.py
index b46858e..c58027a 100644
--- a/os_installer2/diskops.py
+++ b/os_installer2/diskops.py
@@ -227,13 +227,7 @@ class DiskOpCreateRoot(DiskOpCreatePartition):
         except Exception as e:
             self.set_errors("{}: {}".format(self.part.path, e))
             return False
-        if disk.type != "msdos":
-            return True
-        try:
-            self.part.setFlag(parted.PARTITION_BOOT)
-        except Exception as e:
-            self.set_errors("Cannot set root as bootable: {}".format(e))
-            return False
+        return True
 
     def apply(self, disk, simulate):
         """ Create root partition  """
-- 
2.8.3

