From 649442b107c0375dbba0248ff0594555b8a16174 Mon Sep 17 00:00:00 2001
From: Ikey Doherty <ikey@solus-project.com>
Date: Fri, 14 Jul 2017 11:42:34 +0100
Subject: [PATCH 21/33] Add proper styling for buttons

Signed-off-by: Ikey Doherty <ikey@solus-project.com>
---
 os_installer2/mainwindow.py | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/os_installer2/mainwindow.py b/os_installer2/mainwindow.py
index 3a3703d..cbd3d18 100644
--- a/os_installer2/mainwindow.py
+++ b/os_installer2/mainwindow.py
@@ -139,7 +139,10 @@ class MainWindow(Gtk.ApplicationWindow):
         bbox.set_halign(Gtk.Align.END)
         self.installer_page.pack_end(bbox, False, 0, 0)
         self.prev_button = Gtk.Button.new_with_label("Previous")
+        self.prev_button.get_style_context().add_class("flat")
         self.next_button = Gtk.Button.new_with_label("Next")
+        self.next_button.get_style_context().add_class("flat")
+        self.next_button.get_style_context().add_class("suggested-action")
         bbox.add(self.prev_button)
         bbox.add(self.next_button)
         bbox.set_margin_bottom(20)
-- 
2.13.3

