From 28b63d6ac6979c7b0d0ca993ed18258b08cc1ae1 Mon Sep 17 00:00:00 2001
From: Ikey Doherty <ikey@solus-project.com>
Date: Sat, 11 Jun 2016 13:14:01 +0100
Subject: [PATCH 22/24] Still fallback to an install if update fails for any
 reason

Signed-off-by: Ikey Doherty <ikey@solus-project.com>
---
 os_installer2/postinstall.py | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/os_installer2/postinstall.py b/os_installer2/postinstall.py
index a26159b..283f13c 100644
--- a/os_installer2/postinstall.py
+++ b/os_installer2/postinstall.py
@@ -632,7 +632,9 @@ class PostInstallBootloader(PostInstallStep):
             # Update the existing goofiboot stuff, fallback to no nvvars mod
             commands = [
                 "goofiboot update --path=\"{}\"".format(espt),
-                "goofiboot update --path=\"{}\" --no-variables".format(espt)
+                "goofiboot update --path=\"{}\" --no-variables".format(espt),
+                "goofiboot install --path=\"{}\"".format(espt),
+                "goofiboot install --path=\"{}\" --no-variables".format(espt)
             ]
             # Install a fresh goofiboot, fallback to no nvvars mod
         else:
-- 
2.8.3

