From 0e8973b1954ea360f5d31250c3cf6175551b93f2 Mon Sep 17 00:00:00 2001
From: Ikey Doherty <ikey@solus-project.com>
Date: Fri, 14 Jul 2017 11:42:41 +0100
Subject: [PATCH 22/37] Deliberately break init check so I can hack on the UI

Signed-off-by: Ikey Doherty <ikey@solus-project.com>
---
 os-installer-gtk | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/os-installer-gtk b/os-installer-gtk
index a9143fe..43484b6 100755
--- a/os-installer-gtk
+++ b/os-installer-gtk
@@ -26,7 +26,7 @@ if __name__ == "__main__":
         sys.exit(1)
 
     # No source filesystem? No cookies for you!
-    if not os.path.exists(SOURCE_FILESYSTEM):
+    if os.path.exists(SOURCE_FILESYSTEM):
         msg = "Source file system is missing, cannot continue.\n\n{}".format(
             SOURCE_FILESYSTEM)
 
-- 
2.13.3

