From 664aaae9f01bfd2a61474ce200d8ff569a50838c Mon Sep 17 00:00:00 2001
From: Ikey Doherty <ikey@solus-project.com>
Date: Fri, 14 Jul 2017 11:45:33 +0100
Subject: [PATCH 23/33] Add some default CSS styling

Signed-off-by: Ikey Doherty <ikey@solus-project.com>
---
 os_installer2/mainwindow.py | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/os_installer2/mainwindow.py b/os_installer2/mainwindow.py
index cbd3d18..3c7f6a7 100644
--- a/os_installer2/mainwindow.py
+++ b/os_installer2/mainwindow.py
@@ -154,6 +154,8 @@ class MainWindow(Gtk.ApplicationWindow):
         self.info = InstallInfo()
         self.info.owner = self
 
+        self.get_style_context().add_class("installer-window")
+
         # Hook up actions
         self.prev_button.connect("clicked", lambda x: self.prev_page())
         self.next_button.connect("clicked", lambda x: self.next_page())
-- 
2.13.3

