From e8a9187e12e33ae4d81e8227d60f9f715d3c2af2 Mon Sep 17 00:00:00 2001
From: Ikey Doherty <ikey@solus-project.com>
Date: Sat, 11 Jun 2016 15:04:15 +0100
Subject: [PATCH 23/25] Actually use the UUID..

Signed-off-by: Ikey Doherty <ikey@solus-project.com>
---
 os_installer2/postinstall.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/os_installer2/postinstall.py b/os_installer2/postinstall.py
index 283f13c..bd8812f 100644
--- a/os_installer2/postinstall.py
+++ b/os_installer2/postinstall.py
@@ -559,7 +559,7 @@ class PostInstallFstab(PostInstallStep):
 
             uuid = get_part_uuid(swap_path)
             if uuid:
-                im = "UUID={}\tswap\tswap\tsw\t0\t0".format(swap_path)
+                im = "UUID={}\tswap\tswap\tsw\t0\t0".format(uuid)
                 appends.append(im)
             else:
                 appends.append("{}\tswap\tswap\tsw\t0\t0".format(swap_path))
-- 
2.8.3

