From 89cfc30e8b55be8e2cd939557f3bd1ca402d7caa Mon Sep 17 00:00:00 2001
From: Ikey Doherty <ikey@solus-project.com>
Date: Sat, 11 Jun 2016 16:24:41 +0100
Subject: [PATCH 25/25] Fix bad, bad wording (thanks DataDrake)

Signed-off-by: Ikey Doherty <ikey@solus-project.com>
---
 os_installer2/pages/complete.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/os_installer2/pages/complete.py b/os_installer2/pages/complete.py
index 51e6665..d342bab 100644
--- a/os_installer2/pages/complete.py
+++ b/os_installer2/pages/complete.py
@@ -40,7 +40,7 @@ class InstallationCompletePage(BasePage):
         lab.set_property("margin-top", 5)
         lab.set_property("margin-bottom", 5)
 
-        lab = Gtk.Label("Restart and remove any installation media to start "
+        lab = Gtk.Label("Restart and then remove any installation media to start "
                         "using your new operating system.")
         lab.set_use_markup(True)
         lab.set_halign(Gtk.Align.CENTER)
-- 
2.8.3

