From 92a92c7aab660f1411d2a7ff04d4ca77f5120186 Mon Sep 17 00:00:00 2001
From: Ikey Doherty <ikey@solus-project.com>
Date: Sat, 11 Jun 2016 21:05:02 +0100
Subject: [PATCH 27/29] diskman: Ensure we really *do* skip the boot device

Signed-off-by: Ikey Doherty <ikey@solus-project.com>
---
 os_installer2/diskman.py | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/os_installer2/diskman.py b/os_installer2/diskman.py
index f5f8a09..b7d1974 100644
--- a/os_installer2/diskman.py
+++ b/os_installer2/diskman.py
@@ -732,8 +732,14 @@ class DiskManager:
             if whence == "/" or whence.startswith("/run/initramfs"):
                 blacklist.append(mpoint)
 
+        # i.e. /dev/sda, full write mount
+        if device.path in blacklist:
+            print("DEBUG: Skipping boot disk")
+            return None
+
         # Could be a disk without a label
         if disk:
+            # i.e. /dev/sda1, partition mount
             for partition in disk.partitions:
                 if partition.path in blacklist:
                     print("DEBUG: Skipping boot disk")
-- 
2.8.3

