From 60cedb048394bfac6405de898f8676b8aa46919c Mon Sep 17 00:00:00 2001
From: Ikey Doherty <ikey@solus-project.com>
Date: Sat, 11 Jun 2016 21:05:28 +0100
Subject: [PATCH 28/29] pages/complete: Fix pep8 unhappiness

Signed-off-by: Ikey Doherty <ikey@solus-project.com>
---
 os_installer2/pages/complete.py | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/os_installer2/pages/complete.py b/os_installer2/pages/complete.py
index d342bab..53c7662 100644
--- a/os_installer2/pages/complete.py
+++ b/os_installer2/pages/complete.py
@@ -40,8 +40,8 @@ class InstallationCompletePage(BasePage):
         lab.set_property("margin-top", 5)
         lab.set_property("margin-bottom", 5)
 
-        lab = Gtk.Label("Restart and then remove any installation media to start "
-                        "using your new operating system.")
+        lab = Gtk.Label("Restart and then remove any installation media to "
+                        "start using your new operating system.")
         lab.set_use_markup(True)
         lab.set_halign(Gtk.Align.CENTER)
         lab.set_property("xalign", 0.5)
-- 
2.8.3

