From 5c133eaa05fddb8359b6192d4c80ce8540455fb8 Mon Sep 17 00:00:00 2001
From: Ikey Doherty <ikey@solus-project.com>
Date: Fri, 14 Jul 2017 15:17:25 +0100
Subject: [PATCH 28/35] Don't override can_next as we have no welcome page now

Signed-off-by: Ikey Doherty <ikey@solus-project.com>
---
 os_installer2/mainwindow.py | 5 -----
 1 file changed, 5 deletions(-)

diff --git a/os_installer2/mainwindow.py b/os_installer2/mainwindow.py
index f8f92a3..1a5ea9b 100644
--- a/os_installer2/mainwindow.py
+++ b/os_installer2/mainwindow.py
@@ -253,11 +253,6 @@ class MainWindow(Gtk.ApplicationWindow):
             except Exception as e:
                 print("Fatal exception initialising: {}".format(e))
 
-        # Allow next again
-        Gdk.threads_enter()
-        self.set_can_next(True)
-        Gdk.threads_leave()
-
 
     def add_installer_page(self, page):
         """ Work a page into the set """
-- 
2.13.3

