From 43fb815eaab70b40897101ee1f97ca27fdbbc99e Mon Sep 17 00:00:00 2001
From: Ikey Doherty <ikey@solus-project.com>
Date: Fri, 14 Jul 2017 15:47:19 +0100
Subject: [PATCH 30/36] Unneuter the installer once more

Signed-off-by: Ikey Doherty <ikey@solus-project.com>
---
 os-installer-gtk | 2 +-
 setup.py         | 2 +-
 2 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/os-installer-gtk b/os-installer-gtk
index ee8c0a6..0f3e543 100755
--- a/os-installer-gtk
+++ b/os-installer-gtk
@@ -41,7 +41,7 @@ if __name__ == "__main__":
         sys.exit(1)
 
     # No source filesystem? No cookies for you!
-    if os.path.exists(SOURCE_FILESYSTEM):
+    if not os.path.exists(SOURCE_FILESYSTEM):
         msg = "Source file system is missing, cannot continue.\n\n{}".format(
             SOURCE_FILESYSTEM)
 
diff --git a/setup.py b/setup.py
index 3767d1d..b03a28c 100644
--- a/setup.py
+++ b/setup.py
@@ -11,5 +11,5 @@ setup(
     packages        = ['os_installer2', 'os_installer2.pages'],
     scripts         = ['os-installer-gtk'],
     classifiers     = [ "License :: OSI Approved :: GPL-2.0 License"],
-    package_data    = {'os_installer2': ['data/*.png', 'data/*.svg']},
+    package_data    = {'os_installer2': ['data/*.png', 'data/*.svg', 'data/*.css']},
 )
-- 
2.13.3

