From e16c21261de00ea1ec2caaeda89d6dc09b9b1f2f Mon Sep 17 00:00:00 2001
From: Ikey Doherty <ikey@solus-project.com>
Date: Fri, 14 Jul 2017 15:56:28 +0100
Subject: [PATCH 31/36] Use the new icon provided by Budgie Desktop package

Signed-off-by: Ikey Doherty <ikey@solus-project.com>
---
 os_installer2/mainwindow.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/os_installer2/mainwindow.py b/os_installer2/mainwindow.py
index 1a5ea9b..932d52f 100644
--- a/os_installer2/mainwindow.py
+++ b/os_installer2/mainwindow.py
@@ -150,7 +150,7 @@ class MainWindow(Gtk.ApplicationWindow):
         self.headerbox.pack_start(self.box_labels, True, True, 0)
 
         # Vanity! TODO: Select correct icon ..
-        img_vanity = Gtk.Image.new_from_icon_name("budgie-desktop_badge-symbolic", Gtk.IconSize.LARGE_TOOLBAR)
+        img_vanity = Gtk.Image.new_from_icon_name("budgie-desktop-symbolic", Gtk.IconSize.LARGE_TOOLBAR)
         img_vanity.set_property("margin", 8)
         img_vanity.set_property("margin-top", 0)
         lab_vanity = Gtk.Label.new("Solus Budgie")
-- 
2.13.3

