From e8be67ce2e43560d48541951e8aed7e9900fe47a Mon Sep 17 00:00:00 2001
From: Ikey Doherty <ikey@solus-project.com>
Date: Sat, 11 Jun 2016 23:23:32 +0100
Subject: [PATCH 33/84] Stop overriding the icon. Be polite. >_>

Signed-off-by: Ikey Doherty <ikey@solus-project.com>
---
 os_installer2/mainwindow.py | 5 -----
 1 file changed, 5 deletions(-)

diff --git a/os_installer2/mainwindow.py b/os_installer2/mainwindow.py
index 0083dc7..b3b4033 100644
--- a/os_installer2/mainwindow.py
+++ b/os_installer2/mainwindow.py
@@ -26,7 +26,6 @@ from .pages.users import InstallerUsersPage
 from .pages.summary import InstallerSummaryPage
 from .pages.progress import InstallerProgressPage
 from .pages.complete import InstallationCompletePage
-from . import join_resource_path as jrp
 import sys
 import threading
 import traceback
@@ -111,10 +110,6 @@ class MainWindow(Gtk.ApplicationWindow):
     def __init__(self, app):
         Gtk.ApplicationWindow.__init__(self, application=app)
         self.application = app
-        try:
-            self.set_icon_from_file(jrp("install-solus-192-arc-style.svg"))
-        except:
-            pass
 
         self.set_title("Installer")
         self.connect("delete-event", self.quit_handler)
-- 
2.8.3

