From 29cb1764d2e0e601e45d8dd298f34f5f910519b4 Mon Sep 17 00:00:00 2001
From: Ikey Doherty <ikey@solus-project.com>
Date: Sun, 12 Jun 2016 00:05:41 +0100
Subject: [PATCH 34/80] Try with -p not -y

Signed-off-by: Ikey Doherty <ikey@solus-project.com>
---
 os_installer2/diskops.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/os_installer2/diskops.py b/os_installer2/diskops.py
index b379fb2..3fa3326 100644
--- a/os_installer2/diskops.py
+++ b/os_installer2/diskops.py
@@ -344,7 +344,7 @@ class DiskOpResizeOS(BaseDiskOp):
                     self.new_part_off = self.part.geometry.end
                     return True
                 # check it first
-                cmd1 = "/sbin/e2fsck -f -y {}".format(self.part.path)
+                cmd1 = "/sbin/e2fsck -f -p {}".format(self.part.path)
                 try:
                     subprocess.check_output(cmd1, shell=True,
                                             stderr=subprocess.STDOUT)
-- 
2.8.3

