From 5b52caae7850cd99f6c981775d2e5a9d6e45ed97 Mon Sep 17 00:00:00 2001
From: Ikey Doherty <ikey@solus-project.com>
Date: Mon, 17 Jul 2017 17:30:45 +0100
Subject: [PATCH 34/36] Fix my inner derp

Signed-off-by: Ikey Doherty <ikey@solus-project.com>
---
 os_installer2/mainwindow.py | 1 +
 1 file changed, 1 insertion(+)

diff --git a/os_installer2/mainwindow.py b/os_installer2/mainwindow.py
index a91bd36..2ba6289 100644
--- a/os_installer2/mainwindow.py
+++ b/os_installer2/mainwindow.py
@@ -28,6 +28,7 @@ from .pages.complete import InstallationCompletePage
 import sys
 import threading
 import traceback
+import os
 
 class FancyLabel(Gtk.Label):
 
-- 
2.13.3

