From 384c95fa9bfe9a6cb627d98dcaf800e0e69b50f9 Mon Sep 17 00:00:00 2001
From: Ikey Doherty <ikey@solus-project.com>
Date: Sun, 12 Jun 2016 00:30:33 +0100
Subject: [PATCH 35/85] Don't chew the last 100 mb of disk..

Signed-off-by: Ikey Doherty <ikey@solus-project.com>
---
 os_installer2/pages/partitioning.py | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/os_installer2/pages/partitioning.py b/os_installer2/pages/partitioning.py
index b9d066f..5bed386 100644
--- a/os_installer2/pages/partitioning.py
+++ b/os_installer2/pages/partitioning.py
@@ -236,7 +236,10 @@ class DualBootPage(Gtk.VBox):
 
         val = self.spin.get_value()
         avail = self.info.strategy.candidate_part.size
-        GB = 1000.0 * 1000.0 * 1000.0
+
+        MB = 1000.0 * 1000.0
+        avail -= 100 * MB
+        GB = MB * 1000.0
 
         nval = (avail / GB) - val
         ssize = format_size_local(nval * GB, double_precision=True)
-- 
2.8.3

