From 2c7fc1e6df26f49f6321ff403b8605f476bf3049 Mon Sep 17 00:00:00 2001
From: Ikey Doherty <ikey@solus-project.com>
Date: Mon, 17 Jul 2017 17:52:06 +0100
Subject: [PATCH 35/35] Set correct vanity icons for MATE and GNOME

Signed-off-by: Ikey Doherty <ikey@solus-project.com>
---
 os_installer2/mainwindow.py | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/os_installer2/mainwindow.py b/os_installer2/mainwindow.py
index 2ba6289..02b647a 100644
--- a/os_installer2/mainwindow.py
+++ b/os_installer2/mainwindow.py
@@ -157,8 +157,10 @@ class MainWindow(Gtk.ApplicationWindow):
             vanity_icon = "budgie-desktop-symbolic"
             vanity_string = "Solus Budgie"
         elif os.path.exists("/usr/bin/gnome-shell"):
+            vanity_icon = "desktop-environment-gnome"
             vanity_string = "Solus GNOME"
         elif os.path.exists("/usr/bin/mate-panel"):
+            vanity_icon = "mate"
             vanity_string = "Solus MATE"
         else:
             vanity_icon = "start-here-solus"
-- 
2.13.3

