From 39014442a9b3c9b6adebbfedc3fa0b42e37c777e Mon Sep 17 00:00:00 2001
From: Ikey Doherty <ikey@solus-project.com>
Date: Sun, 12 Jun 2016 11:13:01 +0100
Subject: [PATCH 37/80] Lie about UEFI so I can install to my external

Signed-off-by: Ikey Doherty <ikey@solus-project.com>
---
 os_installer2/diskman.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/os_installer2/diskman.py b/os_installer2/diskman.py
index f145f31..5c90f4a 100644
--- a/os_installer2/diskman.py
+++ b/os_installer2/diskman.py
@@ -321,7 +321,7 @@ class DiskManager:
         }
 
         # Set up UEFI knowledge
-        if os.path.exists("/sys/firmware/efi"):
+        if not os.path.exists("/sys/firmware/efi"):
             self.is_uefi = True
             pl_f = "/sys/firmware/efi/fw_platform_size"
             if os.path.exists(pl_f):
-- 
2.8.3

