From 52be6bcc9fa82f460453b76f3436e042f07dac58 Mon Sep 17 00:00:00 2001
From: Ikey Doherty <ikey@solus-project.com>
Date: Sun, 12 Jun 2016 11:14:17 +0100
Subject: [PATCH 38/84] Lie about rootfs location, again for madman testing

Signed-off-by: Ikey Doherty <ikey@solus-project.com>
---
 os_installer2/__init__.py | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/os_installer2/__init__.py b/os_installer2/__init__.py
index 64c4a64..68da5ea 100644
--- a/os_installer2/__init__.py
+++ b/os_installer2/__init__.py
@@ -21,7 +21,10 @@ gi.require_version('TimezoneMap', '1.0')
 
 
 # The path of the source filesystem
-SOURCE_FILESYSTEM = "/run/initramfs/live/LiveOS/squashfs.img"
+# SOURCE_FILESYSTEM = "/run/initramfs/live/LiveOS/squashfs.img"
+
+SOURCE_FILESYSTEM = \
+    "/home/ikey/Solus/primary/IMAGEROOT/deploy/LiveOS/squashfs.img"
 
 # The guy inside that is actually our filesystem to copy
 INNER_FILESYSTEM = "LiveOS/rootfs.img"
-- 
2.8.3

