From 21ca51e23401feff91706c4afa86724f64167e9a Mon Sep 17 00:00:00 2001
From: Ikey Doherty <ikey@solus-project.com>
Date: Sun, 12 Jun 2016 11:20:55 +0100
Subject: [PATCH 39/51] Lock dual boot to the correct disk type

Signed-off-by: Ikey Doherty <ikey@solus-project.com>
---
 os_installer2/strategy.py | 9 ++++++---
 1 file changed, 6 insertions(+), 3 deletions(-)

diff --git a/os_installer2/strategy.py b/os_installer2/strategy.py
index dcb204c..266bae5 100644
--- a/os_installer2/strategy.py
+++ b/os_installer2/strategy.py
@@ -410,9 +410,12 @@ class DualBootStrategy(DiskStrategy):
         if not self.drive.disk:
             return False
 
-        # Dual-boot will require GPT
-        if self.is_uefi() and self.drive.disk.type != "gpt":
-            return False
+        if self.is_uefi():
+            if self.drive.disk.type != "gpt":
+                return False
+        else:
+            if self.drive.disk.type != "msdos":
+                return False
 
         self.potential_spots = []
         # The absolute minimum number of partitions we need (swap = bonus.)
-- 
2.8.3

