From f46a94838efcf755a7796f5f5f843e8c82adef18 Mon Sep 17 00:00:00 2001
From: Ikey Doherty <ikey@solus-project.com>
Date: Sun, 12 Jun 2016 11:21:41 +0100
Subject: [PATCH 40/81] Same for ReplaceOS

Signed-off-by: Ikey Doherty <ikey@solus-project.com>
---
 os_installer2/strategy.py | 10 +++++++---
 1 file changed, 7 insertions(+), 3 deletions(-)

diff --git a/os_installer2/strategy.py b/os_installer2/strategy.py
index 266bae5..4896667 100644
--- a/os_installer2/strategy.py
+++ b/os_installer2/strategy.py
@@ -502,9 +502,13 @@ class ReplaceOSStrategy(DiskStrategy):
         # Require table
         if not self.drive.disk:
             return False
-        if self.is_uefi() and self.drive.disk.type != "gpt":
-            # This disk is no good
-            return False
+
+        if self.is_uefi():
+            if self.drive.disk.type != "gpt":
+                return False
+        else:
+            if self.drive.disk.type != "msdos":
+                return False
 
         self.potential_spots = []
 
-- 
2.8.3

