From 5a1d0a839fdd1b58239f18c8cc233d43017d7e9e Mon Sep 17 00:00:00 2001
From: Ikey Doherty <ikey@solus-project.com>
Date: Sun, 12 Jun 2016 11:27:18 +0100
Subject: [PATCH 43/84] Learn2syntax

Signed-off-by: Ikey Doherty <ikey@solus-project.com>
---
 os_installer2/diskops.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/os_installer2/diskops.py b/os_installer2/diskops.py
index 42ba416..f4456f2 100644
--- a/os_installer2/diskops.py
+++ b/os_installer2/diskops.py
@@ -360,7 +360,7 @@ class DiskOpResizeOS(BaseDiskOp):
                     self.set_errors(ex)
                     return False
 
-                print("Resizing to constraint of {}".format(new_size)
+                print("Resizing to constraint of {}".format(new_size))
                 c = parted.Constraint(device=self.device)
                 c_start = self.part.geometry.start
                 c_end = self.part.geometry.start + nlen
-- 
2.8.3

