From 9e8e4a8a030c916ca0c1092f64aebc9be2e90c9c Mon Sep 17 00:00:00 2001
From: Ikey Doherty <ikey@solus-project.com>
Date: Sun, 12 Jun 2016 11:35:22 +0100
Subject: [PATCH 44/80] So, we meet again.

Signed-off-by: Ikey Doherty <ikey@solus-project.com>
---
 os_installer2/diskops.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/os_installer2/diskops.py b/os_installer2/diskops.py
index f4456f2..03b1802 100644
--- a/os_installer2/diskops.py
+++ b/os_installer2/diskops.py
@@ -368,7 +368,7 @@ class DiskOpResizeOS(BaseDiskOp):
                                                     constraint=c,
                                                     start=c_start,
                                                     end=c_end)
-                self.new_part_off = self.part.geometry.end
+                self.new_part_off = c_end
             else:
                 return False
         except Exception as e:
-- 
2.8.3

