From 75ef49fad98a4b9ce7113d6cc7b008e358531687 Mon Sep 17 00:00:00 2001
From: Ikey Doherty <ikey@solus-project.com>
Date: Sun, 12 Jun 2016 11:59:45 +0100
Subject: [PATCH 45/84] Use long for the bytes and stop pandering to feckin
 politics

Signed-off-by: Ikey Doherty <ikey@solus-project.com>
---
 os_installer2/diskops.py | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/os_installer2/diskops.py b/os_installer2/diskops.py
index 03b1802..7665e5c 100644
--- a/os_installer2/diskops.py
+++ b/os_installer2/diskops.py
@@ -295,14 +295,14 @@ class DiskOpResizeOS(BaseDiskOp):
             cmd = None
 
             if self.part.fileSystem.type == "ntfs":
-                newSz = str(int(self.their_size / 1000))
+                newSz = str(long(self.their_size))
 
                 prefix = "/usr/sbin"
                 check_cmd = "{}/ntfsresize -i -f --force -v {} {}".format(
                     prefix,
                     "--no-action" if simulate else "", self.part.path)
 
-                resize_cmd = "{}/ntfsresize {} -f -f -b --size {}k {}".format(
+                resize_cmd = "{}/ntfsresize {} -f -f -b --size {} {}".format(
                     prefix,
                     "--no-action" if simulate else "", newSz, self.part.path)
 
@@ -350,8 +350,8 @@ class DiskOpResizeOS(BaseDiskOp):
                     self.set_errors(ex)
                     return False
 
-                new_size = str(int(self.their_size / 1024))
-                cmd = "/sbin/resize2fs {} {}K".format(
+                new_size = str(long(self.their_size))
+                cmd = "/sbin/resize2fs {} {}".format(
                     self.part.path, new_size)
                 try:
                     subprocess.check_call(cmd, shell=True)
-- 
2.8.3

