From 8cefe6623c6fe8c950e803ecee99f9634ca52666 Mon Sep 17 00:00:00 2001
From: Ikey Doherty <ikey@solus-project.com>
Date: Sun, 12 Jun 2016 19:31:29 +0100
Subject: [PATCH 53/84] Update the header

Signed-off-by: Ikey Doherty <ikey@solus-project.com>
---
 os_installer2/pages/partitioning.py | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/os_installer2/pages/partitioning.py b/os_installer2/pages/partitioning.py
index 248f5a8..dd5d724 100644
--- a/os_installer2/pages/partitioning.py
+++ b/os_installer2/pages/partitioning.py
@@ -45,7 +45,10 @@ class ManualPage(Gtk.VBox):
         Gtk.VBox.__init__(self)
 
         lab = Gtk.Label("Select custom mount points to use with Solus from "
-                        "the available partition selection below.")
+                        "the available partition selection below.\n"
+                        "Only available system partitions and swap will be "
+                        "displayed here. Use gparted to edit partitions\nin "
+                        "order for them to be displayed.")
         self.pack_start(lab, False, False, 0)
         lab.set_margin_top(20)
         lab.set_margin_bottom(20)
-- 
2.8.3

