From 9b0d3b9023413851e35ec6667119812868587f00 Mon Sep 17 00:00:00 2001
From: Ikey Doherty <ikey@solus-project.com>
Date: Sun, 12 Jun 2016 19:34:17 +0100
Subject: [PATCH 54/81] Plug in the switcher logic

Signed-off-by: Ikey Doherty <ikey@solus-project.com>
---
 os_installer2/pages/partitioning.py | 20 ++++++++++++++++++++
 1 file changed, 20 insertions(+)

diff --git a/os_installer2/pages/partitioning.py b/os_installer2/pages/partitioning.py
index dd5d724..4d6b204 100644
--- a/os_installer2/pages/partitioning.py
+++ b/os_installer2/pages/partitioning.py
@@ -153,6 +153,26 @@ class ManualPage(Gtk.VBox):
         model = self.treeview.get_model()
         model[path][4] = text
 
+        if text == '/':
+            self.set_root_partition(path)
+        elif text == 'swap':
+            self.set_swap_partition(path)
+        elif text == '/home':
+            self.set_home_partition(path)
+
+    def set_root_partition(self, path):
+        """ Update the root partition """
+        print("Updating root partition")
+        pass
+
+    def set_swap_partition(self, path):
+        """ Update the swap partition """
+        pass
+
+    def set_home_partition(self, path):
+        """ Update the home partition """
+        pass
+
     def push_partition(self, drive, part):
         model = self.treeview.get_model()
         os = None
-- 
2.8.3

