From a1094a995583ae8c89172dd5dc1bd2cea63dae54 Mon Sep 17 00:00:00 2001
From: Ikey Doherty <ikey@solus-project.com>
Date: Sun, 12 Jun 2016 19:45:06 +0100
Subject: [PATCH 56/84] Turn Format into a checkbox

Signed-off-by: Ikey Doherty <ikey@solus-project.com>
---
 os_installer2/pages/partitioning.py | 13 ++++++-------
 1 file changed, 6 insertions(+), 7 deletions(-)

diff --git a/os_installer2/pages/partitioning.py b/os_installer2/pages/partitioning.py
index 6707732..5a47b1f 100644
--- a/os_installer2/pages/partitioning.py
+++ b/os_installer2/pages/partitioning.py
@@ -28,7 +28,7 @@ import threading
 INDEX_PARTITION_PATH = 0
 INDEX_PARTITION_TYPE = 1
 INDEX_PARTITION_DESCRIPTION = 2
-INDEX_PARTITION_FORMAT_AS = 3
+INDEX_PARTITION_FORMAT = 3
 INDEX_PARTITION_MOUNT_AS = 4
 INDEX_PARTITION_SIZE = 5
 INDEX_PARTITION_FREE_SPACE = 6
@@ -97,9 +97,8 @@ class ManualPage(Gtk.VBox):
         self.treeview.append_column(self.column6)
 
         # format
-        ren = Gtk.CellRendererText()
+        ren = Gtk.CellRendererToggle()
         self.column7 = Gtk.TreeViewColumn("Format?", ren)
-        self.column7.add_attribute(ren, "markup", INDEX_PARTITION_FORMAT_AS)
         self.treeview.append_column(self.column7)
 
         # size
@@ -208,7 +207,7 @@ class ManualPage(Gtk.VBox):
             part.path,
             fsname,
             os,
-            None,
+            False,
             None,
             part.sizeString,
             part.freespace_string,
@@ -217,7 +216,7 @@ class ManualPage(Gtk.VBox):
         INDEX_PARTITION_PATH = 0
         INDEX_PARTITION_TYPE = 1
         INDEX_PARTITION_DESCRIPTION = 2
-        INDEX_PARTITION_FORMAT_AS = 3
+        INDEX_PARTITION_FORMAT = 3
         INDEX_PARTITION_MOUNT_AS = 4
         INDEX_PARTITION_SIZE = 5
         INDEX_PARTITION_FREE_SPACE = 6
@@ -232,7 +231,7 @@ class ManualPage(Gtk.VBox):
             part.path,
             "swap",
             None,
-            None,
+            False,
             None,
             partSize,
             None,
@@ -241,7 +240,7 @@ class ManualPage(Gtk.VBox):
     def populate_ui(self):
         prober = self.info.prober
 
-        model = Gtk.ListStore(str, str, str, str,
+        model = Gtk.ListStore(str, str, str, bool,
                               str, str, str, SystemPartition)
         self.treeview.set_model(model)
         for drive in prober.drives:
-- 
2.8.3

