From 175857c2e832a25827655b1021429c3cb115a96c Mon Sep 17 00:00:00 2001
From: Ikey Doherty <ikey@solus-project.com>
Date: Sun, 12 Jun 2016 19:59:31 +0100
Subject: [PATCH 58/84] Ensure / is *always* marked as format

Signed-off-by: Ikey Doherty <ikey@solus-project.com>
---
 os_installer2/pages/partitioning.py | 7 +++++--
 1 file changed, 5 insertions(+), 2 deletions(-)

diff --git a/os_installer2/pages/partitioning.py b/os_installer2/pages/partitioning.py
index 510f96b..7db03bb 100644
--- a/os_installer2/pages/partitioning.py
+++ b/os_installer2/pages/partitioning.py
@@ -98,7 +98,8 @@ class ManualPage(Gtk.VBox):
 
         # format
         ren = Gtk.CellRendererToggle()
-        self.column7 = Gtk.TreeViewColumn("Format", ren)
+        self.column7 = Gtk.TreeViewColumn("Format", ren,
+                                          active=INDEX_PARTITION_FORMAT)
         self.treeview.append_column(self.column7)
 
         # size
@@ -168,11 +169,13 @@ class ManualPage(Gtk.VBox):
             skip_part = p[INDEX_PARTITION_PATH]
             skip_mount = p[INDEX_PARTITION_MOUNT_AS]
             if skip_part == active_part:
-                # TODO: Force format!
+                # Mandatory format of /
+                p[INDEX_PARTITION_FORMAT] = True
                 continue
             # Reset anyone trying to be root..
             if skip_mount == '/':
                 p[INDEX_PARTITION_MOUNT_AS] = None
+                p[INDEX_PARTITION_FORMAT] = False
 
     def set_swap_partition(self, path):
         """ Update the swap partition """
-- 
2.8.3

