From 976a61bff2cb3d593ab50a22750d714fcf98dddd Mon Sep 17 00:00:00 2001
From: Ikey Doherty <ikey@solus-project.com>
Date: Sun, 12 Jun 2016 20:29:27 +0100
Subject: [PATCH 60/85] Underp previous

Signed-off-by: Ikey Doherty <ikey@solus-project.com>
---
 os_installer2/pages/system.py | 7 +++----
 1 file changed, 3 insertions(+), 4 deletions(-)

diff --git a/os_installer2/pages/system.py b/os_installer2/pages/system.py
index ade3296..44cf05f 100644
--- a/os_installer2/pages/system.py
+++ b/os_installer2/pages/system.py
@@ -104,7 +104,9 @@ class InstallerSystemPage(BasePage):
 
     def check_forward(self):
         """ Determine if we can forward/back """
-        bs = [self.info.hostname, self.info.bootloader]
+        bs = [self.info.hostname]
+        if self.check_boot.get_active():
+            bs.append(self.info.bootloader)
         misfires = [x for x in bs if not x]
         if len(misfires) == 0:
             self.info.owner.set_can_next(True)
@@ -174,6 +176,3 @@ class InstallerSystemPage(BasePage):
         if err:
             self.error_label2.set_label(
                 "Failed to find location for bootloader: {}".format(err))
-        else:
-            self.error_label2.set_label(
-                "Cannot find a valid bootloader location (MBR disk)")
-- 
2.8.3

