From 7442a6f87299f1ffbdd1129d5153a61655062c49 Mon Sep 17 00:00:00 2001
From: Ikey Doherty <ikey@solus-project.com>
Date: Sun, 12 Jun 2016 20:31:32 +0100
Subject: [PATCH 61/81] Ununderp

Signed-off-by: Ikey Doherty <ikey@solus-project.com>
---
 os_installer2/pages/system.py | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/os_installer2/pages/system.py b/os_installer2/pages/system.py
index 44cf05f..fff88b4 100644
--- a/os_installer2/pages/system.py
+++ b/os_installer2/pages/system.py
@@ -176,3 +176,7 @@ class InstallerSystemPage(BasePage):
         if err:
             self.error_label2.set_label(
                 "Failed to find location for bootloader: {}".format(err))
+        else:
+            self.check_boot.set_active(False)
+            self.error_label2.set_label(
+                "Warning: Cannot find a valid bootloader location (MBR disk)")
-- 
2.8.3

