From 9cfc10c29319f69e8cc9a0f9731b41293077fb53 Mon Sep 17 00:00:00 2001
From: Ikey Doherty <ikey@solus-project.com>
Date: Sun, 12 Jun 2016 20:46:04 +0100
Subject: [PATCH 63/85] Update for swap

Signed-off-by: Ikey Doherty <ikey@solus-project.com>
---
 os_installer2/pages/partitioning.py | 13 ++++++++++++-
 1 file changed, 12 insertions(+), 1 deletion(-)

diff --git a/os_installer2/pages/partitioning.py b/os_installer2/pages/partitioning.py
index 656a969..cdd0860 100644
--- a/os_installer2/pages/partitioning.py
+++ b/os_installer2/pages/partitioning.py
@@ -192,7 +192,18 @@ class ManualPage(Gtk.VBox):
 
     def set_swap_partition(self, path):
         """ Update the swap partition """
-        pass
+        model = self.treeview.get_model()
+        row = model[path]
+        active_part = row[INDEX_PARTITION_PATH]
+        self.selection_swap = active_part
+        for p in model:
+            skip_part = p[INDEX_PARTITION_PATH]
+            skip_mount = p[INDEX_PARTITION_MOUNT_AS]
+            if skip_part == active_part:
+                continue
+            # Reset anyone trying to be swap..
+            if skip_mount == 'swap':
+                p[INDEX_PARTITION_MOUNT_AS] = None
 
     def set_home_partition(self, path):
         """ Update the home partition """
-- 
2.8.3

