From 5d6f548aa79f2c77680bbc335eec57dce6cc5835 Mon Sep 17 00:00:00 2001
From: Ikey Doherty <ikey@solus-project.com>
Date: Sun, 12 Jun 2016 21:08:06 +0100
Subject: [PATCH 65/80] Preliminary checking of root size for teh sanity

Signed-off-by: Ikey Doherty <ikey@solus-project.com>
---
 os_installer2/pages/partitioning.py | 31 +++++++++++++++++++++++++------
 1 file changed, 25 insertions(+), 6 deletions(-)

diff --git a/os_installer2/pages/partitioning.py b/os_installer2/pages/partitioning.py
index df4c5b2..9bc6f18 100644
--- a/os_installer2/pages/partitioning.py
+++ b/os_installer2/pages/partitioning.py
@@ -33,6 +33,7 @@ INDEX_PARTITION_MOUNT_AS = 4
 INDEX_PARTITION_SIZE = 5
 INDEX_PARTITION_FREE_SPACE = 6
 INDEX_PARTITION_OBJECT = 7
+INDEX_PARTITION_SIZE_NUM = 8
 
 
 class ManualPage(Gtk.VBox):
@@ -44,6 +45,7 @@ class ManualPage(Gtk.VBox):
 
     selection_home = None
     selection_root = None
+    selection_root_size = None
     selection_swap = None
 
     def __init__(self):
@@ -183,6 +185,7 @@ class ManualPage(Gtk.VBox):
         row = model[path]
         active_part = row[INDEX_PARTITION_PATH]
         self.selection_root = active_part
+        self.selection_root_size = row[INDEX_PARTITION_SIZE_NUM]
         for p in model:
             skip_part = p[INDEX_PARTITION_PATH]
             skip_mount = p[INDEX_PARTITION_MOUNT_AS]
@@ -266,6 +269,8 @@ class ManualPage(Gtk.VBox):
         if fs and fs.type:
             fsname = fs.type
 
+        partSizeActual = part.partition.getLength() * \
+            part.partition.disk.device.sectorSize
         model.append([
             part.path,
             fsname,
@@ -274,7 +279,9 @@ class ManualPage(Gtk.VBox):
             None,
             part.sizeString,
             part.freespace_string,
-            part])
+            part,
+            partSizeActual
+        ])
         """
         INDEX_PARTITION_PATH = 0
         INDEX_PARTITION_TYPE = 1
@@ -284,12 +291,14 @@ class ManualPage(Gtk.VBox):
         INDEX_PARTITION_SIZE = 5
         INDEX_PARTITION_FREE_SPACE = 6
         INDEX_PARTITION_OBJECT = 7
+        INDEX_PARTITION_SIZE_NUM = 8
         """
 
     def push_swap(self, part):
         model = self.treeview.get_model()
-        partSize = format_size_local(part.getLength() *
-                                     part.disk.device.sectorSize)
+
+        partSizeActual = part.getLength() * part.disk.device.sectorSize
+        partSize = format_size_local(partSizeActual)
         model.append([
             part.path,
             "swap",
@@ -298,13 +307,15 @@ class ManualPage(Gtk.VBox):
             None,
             partSize,
             None,
-            None])
+            None,
+            partSizeActual
+        ])
 
     def populate_ui(self):
         prober = self.info.prober
 
         model = Gtk.ListStore(str, str, str, bool,
-                              str, str, str, SystemPartition)
+                              str, str, str, SystemPartition, long)
         self.treeview.set_model(model)
         for drive in prober.drives:
             for swap in drive.get_swap_partitions():
@@ -344,7 +355,15 @@ class ManualPage(Gtk.VBox):
             if sel == "":
                 continue
             lab.append(display.format(sel))
-        self.selection_label.set_markup("\t\t".join(lab))
+        labe = "\t\t".join(lab)
+        if self.selection_root:
+            # Can't go forward so nuh
+            if self.selection_root_size < MIN_REQUIRED_SIZE:
+                labe += "\n<b>Root partition is not at least 10GB</b>"
+        else:
+            # Can't go forward.. ?
+            print("NO fwd for you!")
+        self.selection_label.set_markup(labe)
 
 
 class DualBootPage(Gtk.VBox):
-- 
2.8.3

