From 25c2e64e175e2a7b154f09bb9a1ede900248c358 Mon Sep 17 00:00:00 2001
From: Ikey Doherty <ikey@solus-project.com>
Date: Sun, 12 Jun 2016 21:24:13 +0100
Subject: [PATCH 66/84] Fix the strategy to not create ESP on Custom user
 partitioning

Signed-off-by: Ikey Doherty <ikey@solus-project.com>
---
 os_installer2/strategy.py | 11 +++++++----
 1 file changed, 7 insertions(+), 4 deletions(-)

diff --git a/os_installer2/strategy.py b/os_installer2/strategy.py
index 10b3984..7b689ab 100644
--- a/os_installer2/strategy.py
+++ b/os_installer2/strategy.py
@@ -166,15 +166,18 @@ class DiskStrategy:
             paths.reverse()
             return paths
         esps = self.dp.collect_esp()
-        if len(esps) == 0:
+        if len(esps) == 0 and not isinstance(self, UserPartitionStrategy):
             # Create a new ESP
             return [("Create new ESP on {}".format(self.drive.path), "c")]
         cand = self.get_suitable_esp()
         # Have an ESP and it's not good enough for use.
         if not cand:
-            self.set_errors(
-                "ESP is too small: {} free space remaining".format(
-                    cand.freespace_string))
+            if esps:
+                self.set_errors(
+                    "ESP is too small: {} free space remaining".format(
+                        esps[0].freespace_string))
+            else:
+                self.set_errors("No usable ESP found on this system")
             return []
         return [(self.dsc(cand), cand.path)]
 
-- 
2.8.3

