From dc19e8e582511ec2086fb01c0d6eb5a95eb9c81d Mon Sep 17 00:00:00 2001
From: Ikey Doherty <ikey@solus-project.com>
Date: Sun, 12 Jun 2016 21:27:48 +0100
Subject: [PATCH 67/81] Only update when required

Signed-off-by: Ikey Doherty <ikey@solus-project.com>
---
 os_installer2/pages/partitioning.py | 6 +++++-
 1 file changed, 5 insertions(+), 1 deletion(-)

diff --git a/os_installer2/pages/partitioning.py b/os_installer2/pages/partitioning.py
index 9bc6f18..3012a07 100644
--- a/os_installer2/pages/partitioning.py
+++ b/os_installer2/pages/partitioning.py
@@ -47,6 +47,7 @@ class ManualPage(Gtk.VBox):
     selection_root = None
     selection_root_size = None
     selection_swap = None
+    cur_strategy = None
 
     def __init__(self):
         Gtk.VBox.__init__(self)
@@ -336,7 +337,10 @@ class ManualPage(Gtk.VBox):
     def update_strategy(self, info):
         self.info = info
         info.owner.set_can_next(False)
-        self.populate_ui()
+        if self.info.strategy != self.cur_strategy:
+            self.populate_ui()
+        self.cur_strategy = self.info.strategy
+        self.update_selection()
 
     def update_selection(self):
         """ Test if we can move forward now, i.e. everything is valid... """
-- 
2.8.3

