From 488b71744fcc1125a94f52d920d13840ec43cbfe Mon Sep 17 00:00:00 2001
From: Ikey Doherty <ikey@solus-project.com>
Date: Sun, 12 Jun 2016 21:53:25 +0100
Subject: [PATCH 70/80] Finish the navigation

Signed-off-by: Ikey Doherty <ikey@solus-project.com>
---
 os_installer2/pages/partitioning.py | 41 ++++++++++++++++++++++++++++++-------
 1 file changed, 34 insertions(+), 7 deletions(-)

diff --git a/os_installer2/pages/partitioning.py b/os_installer2/pages/partitioning.py
index 672b0d3..88ca4b5 100644
--- a/os_installer2/pages/partitioning.py
+++ b/os_installer2/pages/partitioning.py
@@ -113,6 +113,7 @@ class ManualPage(Gtk.VBox):
 
         # format
         ren = Gtk.CellRendererToggle()
+        ren.connect("toggled", self.on_format_toggled)
         self.column7 = Gtk.TreeViewColumn("Format", ren,
                                           active=INDEX_PARTITION_FORMAT)
         self.treeview.append_column(self.column7)
@@ -141,6 +142,23 @@ class ManualPage(Gtk.VBox):
         toolbar.add(gparted)
         self.pack_start(toolbar, False, False, 0)
 
+    def on_format_toggled(self, widget, path):
+        model = self.treeview.get_model()
+        row = model[path]
+        mpoint = row[INDEX_PARTITION_MOUNT_AS]
+        if mpoint not in ['/home', 'swap']:
+            return
+        acceptables = {
+            '/home': ACCEPTABLE_FS_TYPES,
+            'swap': ['swap']
+        }
+        # Don't allow overriding the forced format.
+        fs = row[INDEX_PARTITION_TYPE]
+        if fs not in acceptables[mpoint]:
+            return
+
+        row[INDEX_PARTITION_FORMAT] = not row[INDEX_PARTITION_FORMAT]
+
     def launch_gparted(self, btn, udata=None):
         """ Send off a thread """
         self.info.owner.set_sensitive(False)
@@ -384,14 +402,23 @@ class ManualPage(Gtk.VBox):
                 continue
             lab.append(display.format(sel))
         labe = "\t\t".join(lab)
-        if self.selection_root:
-            # Can't go forward so nuh
-            if self.selection_root_size < MIN_REQUIRED_SIZE:
-                labe += "\n<b>Root partition is not at least 10GB</b>"
-        else:
-            # Can't go forward.. ?
-            print("NO fwd for you!")
+
+        # Don't allow forward
+        if not self.selection_root:
+            self.info.owner.set_can_next(False)
+            self.selection_label.set_markup(labe)
+            return
+
+        # Can't go forward so nuh
+        if self.selection_root_size < MIN_REQUIRED_SIZE:
+            labe += "\n<b>Root partition is not at least 10GB</b>"
+            self.selection_label.set_markup(labe)
+            self.info.owner.set_can_next(False)
+            return
+
+        # Now we can go forward.
         self.selection_label.set_markup(labe)
+        self.info.owner.set_can_next(True)
 
 
 class DualBootPage(Gtk.VBox):
-- 
2.8.3

