From fce411e863b4201fb65200f6d77089911efbdf35 Mon Sep 17 00:00:00 2001
From: Ikey Doherty <ikey@solus-project.com>
Date: Sun, 12 Jun 2016 22:58:30 +0100
Subject: [PATCH 76/85] We'll just pretend that didn't happen

Signed-off-by: Ikey Doherty <ikey@solus-project.com>
---
 os_installer2/strategy.py | 7 +++++++
 1 file changed, 7 insertions(+)

diff --git a/os_installer2/strategy.py b/os_installer2/strategy.py
index 1521add..9f69fbd 100644
--- a/os_installer2/strategy.py
+++ b/os_installer2/strategy.py
@@ -597,6 +597,13 @@ class UserPartitionStrategy(DiskStrategy):
                 continue
             return op.path
 
+    def get_root_partition(self):
+        """ Get our root partition """
+        for op in self.get_operations():
+            if isinstance(op, DiskOpFormatRoot):
+                return op.part.path
+        return None
+
     def set_root_partition(self, part):
         """ Set the root partition to use """
         self.root_part = part
-- 
2.8.3

