From 01103a19017a360627b8b0cf0722903066102997 Mon Sep 17 00:00:00 2001
From: Ikey Doherty <ikey@solus-project.com>
Date: Sun, 12 Jun 2016 23:03:01 +0100
Subject: [PATCH 77/85] Unbugger /home fstab

Signed-off-by: Ikey Doherty <ikey@solus-project.com>
---
 os_installer2/postinstall.py | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/os_installer2/postinstall.py b/os_installer2/postinstall.py
index cf6c66d..02387e1 100644
--- a/os_installer2/postinstall.py
+++ b/os_installer2/postinstall.py
@@ -547,8 +547,9 @@ class PostInstallFstab(PostInstallStep):
             if isinstance(op, DiskOpUseHome):
                 huuid = get_part_uuid(op.home_part.path)
                 fs = op.home_part_fs
-                i = "UUID={}\t/\t{}\trw,relatime,errors=remount-ro\t0\t0"
+                i = "UUID={}\t/home\t{}\trw,relatime,errors=remount-ro\t0\t0"
                 appends.append(i.format(huuid, fs))
+                continue
             if disk.type == "gpt" and strat.is_uefi():
                 continue
 
-- 
2.8.3

