From 0749fbf3b4dde15e40b9ce804b21db6d25702931 Mon Sep 17 00:00:00 2001
From: Ikey Doherty <ikey@solus-project.com>
Date: Sun, 12 Jun 2016 23:06:52 +0100
Subject: [PATCH 79/81] Ensure we actually nuke /home/live

Signed-off-by: Ikey Doherty <ikey@solus-project.com>
---
 os_installer2/pages/progress.py | 18 ++++++++++++++++++
 1 file changed, 18 insertions(+)

diff --git a/os_installer2/pages/progress.py b/os_installer2/pages/progress.py
index 4bb5206..b721dd8 100644
--- a/os_installer2/pages/progress.py
+++ b/os_installer2/pages/progress.py
@@ -36,6 +36,7 @@ import os
 import stat
 import parted
 import sys
+import shutil
 
 # Update 5 times a second, vs every byte copied..
 UPDATE_FREQUENCY = 1000 / 5
@@ -582,6 +583,17 @@ class InstallerProgressPage(BasePage):
         print("DEBUG: / ({}) mounted at {}".format(root, target))
         return True
 
+    def maybe_nuke_live(self):
+        fpath = os.path.join(self.get_installer_target_filesystem(),
+                             "home/live")
+        if not os.path.exists(fpath):
+            return True
+        try:
+            shutil.rmtree(fpath)
+        except Exception:
+            return False
+        return True
+
     def maybe_mount_home(self):
         strategy = self.info.strategy
         home = strategy.get_home_dir()
@@ -656,6 +668,12 @@ class InstallerProgressPage(BasePage):
             return False
         self.filesystem_copying = False
 
+        if not self.maybe_nuke_live():
+            self.set_error_message("Failed to clean /home")
+            self.unmount_all()
+            self.installing = False
+            return False
+
         if not self.maybe_mount_home():
             self.set_error_message("Failed to mount /home")
             self.unmount_all()
-- 
2.8.3

