From 5350e620d89e1d965afb2ce25a8442feb2efbb16 Mon Sep 17 00:00:00 2001
From: Ikey Doherty <ikey@solus-project.com>
Date: Mon, 13 Jun 2016 00:13:39 +0100
Subject: [PATCH 81/85] Explicitly enable external access

Signed-off-by: Ikey Doherty <ikey@solus-project.com>
---
 os_installer2/pages/partitioning.py | 15 +++++++++++++--
 1 file changed, 13 insertions(+), 2 deletions(-)

diff --git a/os_installer2/pages/partitioning.py b/os_installer2/pages/partitioning.py
index 76f6566..e180357 100644
--- a/os_installer2/pages/partitioning.py
+++ b/os_installer2/pages/partitioning.py
@@ -170,13 +170,24 @@ class ManualPage(Gtk.VBox):
         """ Launch this on the thread so the UI doesnt lock """
         perms = self.info.owner.get_perms_manager()
         perms.up_permissions()
+        dm = self.info.owner.get_disk_manager()
+        prober = self.info.prober
+        # Start external access
+        for drive in prober.drives:
+            if not drive.device:
+                continue
+            drive.device.beginExternalAccess()
+
         try:
             os.system("gparted")
         except:
             pass
 
-        perms = self.info.owner.get_perms_manager()
-        dm = self.info.owner.get_disk_manager()
+        # Stop access
+        for drive in prober.drives:
+            if not drive.device:
+                continue
+            drive.device.endExternalAccess()
 
         prober = DriveProber(dm)
         self.info.prober = prober
-- 
2.8.3

