From 760b29b63047af3ec7fa8f1d1129f74de0ca33fd Mon Sep 17 00:00:00 2001
From: Ikey Doherty <ikey@solus-project.com>
Date: Mon, 13 Jun 2016 00:17:38 +0100
Subject: [PATCH 82/84] Fix startup spam

Signed-off-by: Ikey Doherty <ikey@solus-project.com>
---
 os_installer2/pages/partitioning.py | 5 ++---
 1 file changed, 2 insertions(+), 3 deletions(-)

diff --git a/os_installer2/pages/partitioning.py b/os_installer2/pages/partitioning.py
index e180357..7dbe75f 100644
--- a/os_installer2/pages/partitioning.py
+++ b/os_installer2/pages/partitioning.py
@@ -208,7 +208,6 @@ class ManualPage(Gtk.VBox):
         # TODO: Refresh
 
     def on_mount_changed(self, widget, path, text):
-        print("{} = {}".format(path, text))
         model = self.treeview.get_model()
 
         if text is None or text.strip() == '':
@@ -375,7 +374,6 @@ class ManualPage(Gtk.VBox):
         for drive in prober.drives:
             for swap in drive.get_swap_partitions():
                 try:
-                    print(swap)
                     self.push_swap(swap)
                 except Exception as e:
                     print("Swap problem: {}".format(e))
@@ -383,10 +381,11 @@ class ManualPage(Gtk.VBox):
             for part in sorted(drive.partitions):
                 try:
                     part_prop = drive.partitions[part]
+                    if not part_prop:
+                        continue
                     self.push_partition(drive, part_prop)
                 except Exception as e:
                     print("Init problem: {}".format(e))
-            print(drive)
 
     def update_strategy(self, info):
         self.info = info
-- 
2.8.3

