diff --git a/etc/meson.build b/etc/meson.build
index 59966f8..29dd18b 100644
--- a/etc/meson.build
+++ b/etc/meson.build
@@ -1 +1 @@
-install_data('UPower.conf', install_dir: get_option('sysconfdir') / 'UPower')
+install_data('UPower.conf', install_dir: get_option('datadir') / 'UPower')
diff --git a/meson.build b/meson.build
index 04aed85..96174a3 100644
--- a/meson.build
+++ b/meson.build
@@ -29,6 +29,7 @@ cdata.set_quoted('GETTEXT_PACKAGE', meson.project_name())
 cdata.set_quoted('PACKAGE_VERSION', meson.project_version())
 cdata.set_quoted('VERSION', meson.project_version())
 cdata.set_quoted('PACKAGE_SYSCONF_DIR', get_option('sysconfdir'))
+cdata.set_quoted('PACKAGE_DATA_DIR', get_option('datadir'))
 
 glib_min_version = '2.56'
 
diff --git a/src/up-config.c b/src/up-config.c
index f842ede..505d58e 100644
--- a/src/up-config.c
+++ b/src/up-config.c
@@ -102,8 +102,14 @@ up_config_init (UpConfig *config)
 	config->priv->keyfile = g_key_file_new ();
 
 	filename = g_strdup (g_getenv ("UPOWER_CONF_FILE_NAME"));
-	if (filename == NULL)
+	if (filename == NULL) {
 		filename = g_build_filename (PACKAGE_SYSCONF_DIR,"UPower", "UPower.conf", NULL);
+		
+		if (!g_file_test(filename, G_FILE_TEST_EXISTS)) {
+					g_free(filename);
+					filename = g_build_filename (PACKAGE_DATA_DIR, "UPower", "UPower.conf", NULL);
+		}
+	}
 
 	/* load */
 	ret = g_key_file_load_from_file (config->priv->keyfile,
