From 8d73f349b942f22486d209e9fd28eb6a9f9095db Mon Sep 17 00:00:00 2001
From: Algent Albrahimi <algent@protonmail.com>
Date: Sun, 27 Mar 2022 17:55:18 +0200
Subject: [PATCH] pam d location

---
 configs/Makefile | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/configs/Makefile b/configs/Makefile
index b75a3e1..abfc4b5 100644
--- a/configs/Makefile
+++ b/configs/Makefile
@@ -103,9 +103,9 @@ local-install-base: $(CONFFILES) $(CONFDSUBDIRFILES) $(LOGROTATECONFFILE)
 	fi \
 	fi
 ifeq ($(USE_AUTHPAM),true)
-	@if test ! -f $(DESTDIR)/etc/pam.d/pluto ; then \
-		mkdir -p $(DESTDIR)/etc/pam.d/ ; \
-		$(INSTALL) $(INSTCONFFLAGS) pam.d/pluto $(DESTDIR)/etc/pam.d/pluto ; \
+	@if test ! -f $(DESTDIR)/usr/share/defaults/etc/pam.d/pluto ; then \
+		mkdir -p $(DESTDIR)/usr/share/defaults/etc/pam.d/ ; \
+		$(INSTALL) $(INSTCONFFLAGS) pam.d/pluto $(DESTDIR)/usr/share/defaults/etc/pam.d/pluto ; \
 	else \
 		echo -e "\n************************** WARNING ***********************************" ; \
 		echo "We are not installing a new copy of the pam.d/pluto file, as one" ; \
-- 
2.35.1

