diff --git a/Makefile b/Makefile
index 7487215..1cd36b8 100644
--- a/Makefile
+++ b/Makefile
@@ -220,9 +220,9 @@ local-install:
 		echo -e "**********************************************************************\n" ; \
 	fi \
 	fi
-	@if test ! -f $(DESTDIR)/etc/pam.d/pluto ; then \
-		mkdir -p $(DESTDIR)/etc/pam.d/ ; \
-		$(INSTALL) $(INSTCONFFLAGS) pam.d/pluto $(DESTDIR)/etc/pam.d/pluto ; \
+	@if test ! -f $(DESTDIR)/usr/share/defaults/etc/pam.d/pluto ; then \
+		mkdir -p $(DESTDIR)/usr/share/defaults/etc/pam.d/ ; \
+		$(INSTALL) $(INSTCONFFLAGS) pam.d/pluto $(DESTDIR)/usr/share/defaults/etc/pam.d/pluto ; \
 	else \
 		echo -e "\n************************** WARNING ***********************************" ; \
 		echo "We are not installing a new copy of the pam.d/pluto file, as one" ; \
