--- lazarus-2.2.6--orig/Makefile.fpc	2023-03-04 14:24:23.000000000 +0100
+++ lazarus-2.2.6--patched/Makefile.fpc	2023-03-07 19:25:14.938976319 +0100
@@ -19,18 +19,18 @@
 files=$(wildcard *$(OEXT)) $(wildcard *$(PPUEXT)) $(wildcard *$(RSTEXT)) $(wildcard *.lfm)
 
 [install]
 fpcpackage=n
-basedir=share/lazarus
+basedir=$(_LIB)/lazarus
 
 [dist]
 destdir=$(BASEDIR)/dist
 
 [prerules]
 RCPP?=$(strip $(firstword cpp$(SRCEXEEXT)))
 #
 ifndef LAZARUS_INSTALL_DIR
-LAZARUS_INSTALL_DIR=$(INSTALL_PREFIX)/share/lazarus
+LAZARUS_INSTALL_DIR=$(INSTALL_PREFIX)/$(_LIB)/lazarus
 INSTALL_MAN_DIR=$(INSTALL_PREFIX)/share/man
 ifneq ($(findstring $(OS_TARGET),win32 win64),)
 LAZARUS_INSTALL_DIR=C:\lazarus
 endif
@@ -286,10 +286,11 @@
 installbase:
 ifneq ($(findstring $(OS_TARGET),win32 win64),)
 
 else
+	$(MKDIR) $(INSTALL_PREFIX)/$(_LIB)
+	$(MKDIR) $(INSTALL_PREFIX)/$(_LIB)/lazarus
 	$(MKDIR) $(INSTALL_PREFIX)/share
-	$(MKDIR) $(INSTALL_PREFIX)/share/lazarus
 	$(MKDIR) $(INSTALL_PREFIX)/share/applications
 	$(MKDIR) $(INSTALL_PREFIX)/share/pixmaps
 	$(MKDIR) $(INSTALL_PREFIX)/share/mime/packages
 	$(MKDIR) $(INSTALL_PREFIX)/share/icons/hicolor/48x48/mimetypes
