diff --git a/cmake/modules/TestSignalType.c b/cmake/modules/TestSignalType.c
index aca0bf4ed..36f485312 100644
--- a/cmake/modules/TestSignalType.c
+++ b/cmake/modules/TestSignalType.c
@@ -4,10 +4,9 @@
 # undef signal
 #endif
 #ifdef __cplusplus
-extern "C" void (*signal (int, void (*)(int)))(int);
-#else
-void (*signal ()) ();
+extern "C"
 #endif
+void (*signal (int, void (*)(int)))(int);
 
 int
 main ()
