--- a/components/bsa/bsa_file.cpp
+++ b/components/bsa/bsa_file.cpp
@@ -26,6 +26,7 @@
 #include <components/files/constrainedfilestream.hpp>
 
 #include <cassert>
+#include <algorithm>
 
 #include <boost/filesystem/path.hpp>
 #include <boost/filesystem/fstream.hpp>
diff -ru openmw-openmw-0.48.0.orig/apps/openmw/mwstate/charactermanager.hpp openmw-openmw-0.48.0/apps/openmw/mwstate/charactermanager.hpp
--- openmw-openmw-0.48.0.orig/apps/openmw/mwstate/charactermanager.hpp	2024-08-30 22:11:44.591374022 +0200
+++ openmw-openmw-0.48.0/apps/openmw/mwstate/charactermanager.hpp	2024-08-30 22:12:01.901694900 +0200
@@ -2,6 +2,7 @@
 #define GAME_STATE_CHARACTERMANAGER_H
 
 #include <boost/filesystem/path.hpp>
+#include <list>
 
 #include "character.hpp"
 
