From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Evan Maddock <maddock.evan@vivaldi.net>
Date: Sun, 15 Jun 2025 19:58:29 -0400
Subject: [PATCH 4/4] backends/packagekit: Check for nullptr when getting
 driver category

Signed-off-by: Evan Maddock <maddock.evan@vivaldi.net>
---
 .../backends/PackageKitBackend/PackageKitBackend.cpp        | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/libdiscover/backends/PackageKitBackend/PackageKitBackend.cpp b/libdiscover/backends/PackageKitBackend/PackageKitBackend.cpp
index 76bf22ea4..751bf8187 100644
--- a/libdiscover/backends/PackageKitBackend/PackageKitBackend.cpp
+++ b/libdiscover/backends/PackageKitBackend/PackageKitBackend.cpp
@@ -242,6 +242,12 @@ PackageKitBackend::PackageKitBackend(QObject *parent)
     // Hide the drivers category if there's no drivers
     connect(CategoryModel::global(), &CategoryModel::rootCategoriesChanged, this, [this] {
         auto driversCategory = CategoryModel::global()->findCategoryByName(QStringLiteral("Hardware Drivers"));
+
+        // Will be null on subsequent runs
+        if (!driversCategory) {
+            return;
+        }
+
         AbstractResourcesBackend::Filters filter;
         filter.category = driversCategory;
         auto stream = search(filter);
