From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Evan Maddock <maddock.evan@vivaldi.net>
Date: Wed, 5 Mar 2025 16:10:10 -0500
Subject: [PATCH 5/6] Remove unused function

Signed-off-by: Evan Maddock <maddock.evan@vivaldi.net>
---
 .../PackageKitBackend/AppPackageKitResource.cpp      | 12 ------------
 1 file changed, 12 deletions(-)

diff --git a/libdiscover/backends/PackageKitBackend/AppPackageKitResource.cpp b/libdiscover/backends/PackageKitBackend/AppPackageKitResource.cpp
index f04434053..3b685fb41 100644
--- a/libdiscover/backends/PackageKitBackend/AppPackageKitResource.cpp
+++ b/libdiscover/backends/PackageKitBackend/AppPackageKitResource.cpp
@@ -116,18 +116,6 @@ QStringList AppPackageKitResource::mimetypes() const
 
 static constexpr auto s_addonKinds = {AppStream::Component::KindAddon, AppStream::Component::KindCodec};
 
-static bool hasDeviceModaliases(const AppStream::Component &comp)
-{
-    const auto mods = comp.provided(AppStream::Provided::KindModalias);
-    auto sys = AppStream::SystemInfo();
-    for (const auto &mod : mods.items()) {
-        if (sys.hasDeviceMatchingModalias(mod)) {
-            return true;
-        }
-    }
-    return false;
-}
-
 bool AppPackageKitResource::hasCategory(const QString &category) const
 {
     if (m_appdata.kind() != AppStream::Component::KindAddon && category == QStringLiteral("Application"))
