From b92b9ccf6cdf3b0218151ffe4502e99fabeff12d Mon Sep 17 00:00:00 2001
From: Peter O'Connor <peter@solus-project.com>
Date: Fri, 20 Apr 2018 15:37:38 +1000
Subject: [PATCH] Disable building unshipped code

Signed-off-by: Peter O'Connor <peter@solus-project.com>
---
 CMakeLists.txt                    | 1 -
 libkdegamesprivate/CMakeLists.txt | 2 --
 2 files changed, 3 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 0309139..a1125dd 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -51,7 +51,6 @@ add_subdirectory( declarativeimports )
 add_subdirectory( highscore )
 add_subdirectory( includes )
 add_subdirectory( libkdegamesprivate )
-add_subdirectory( tests )
 
 include_directories(
     ${CMAKE_CURRENT_SOURCE_DIR}/highscore
diff --git a/libkdegamesprivate/CMakeLists.txt b/libkdegamesprivate/CMakeLists.txt
index b122f37..cdc6cc5 100644
--- a/libkdegamesprivate/CMakeLists.txt
+++ b/libkdegamesprivate/CMakeLists.txt
@@ -1,8 +1,6 @@
 enable_testing()
 project(libkdegamesprivate)
 
-add_subdirectory(tests)
-
 # NOTE: The libkdegamesprivate target is compiled in the parent directory,
 # because CMake can't cope with exported libraries in two different
 # directories.
-- 
2.16.3

