From 6efa420f4c6c172ca2eb42ced43801454329cc6c Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Abdulkadir=20Furkan=20=C5=9Eanl=C4=B1?=
 <abdulkadirfsanli@protonmail.com>
Date: Sat, 9 Feb 2019 19:13:29 +0100
Subject: [PATCH] Fix default --enable-mediactrl

---
 configure | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/configure b/configure
index a5152f5..10127c5 100755
--- a/configure
+++ b/configure
@@ -4233,7 +4233,7 @@ DEFAULT_wxUSE_UNICODE_UTF8_LOCALE=no
 
 DEFAULT_wxUSE_ARTPROVIDER_TANGO=auto
 DEFAULT_wxUSE_OPENGL=auto
-DEFAULT_wxUSE_MEDIACTRL=auto
+DEFAULT_wxUSE_MEDIACTRL=no
 DEFAULT_wxUSE_COMPILER_TLS=auto
 DEFAULT_wxUSE_HOTKEY=auto
 DEFAULT_wxUSE_METAFILE=auto
-- 
2.20.1

