--- a/src/visudo/mod.rs
+++ b/src/visudo/mod.rs
@@ -80,7 +80,7 @@ pub fn main() {
 fn check(file_arg: Option<&str>, perms: bool, owner: bool) -> io::Result<()> {
     let mut sudoers_path = file_arg
         .map(PathBuf::from)
-        .unwrap_or_else(candidate_sudoers_file);
+        .unwrap_or_else(|| PathBuf::from("/etc/sudoers.d/visudo"));
 
     let sudoers_file = File::open(if sudoers_path == Path::new("-") {
         // portability: /dev/stdin 'almost POSIX' and exists on BSD and Linux systems
@@ -144,7 +144,7 @@ fn check(file_arg: Option<&str>, perms: bool, owner: bool) -> io::Result<()> {
 fn run(file_arg: Option<&str>, perms: bool, owner: bool) -> io::Result<()> {
     let sudoers_path = &file_arg
         .map(PathBuf::from)
-        .unwrap_or_else(candidate_sudoers_file);
+        .unwrap_or_else(|| PathBuf::from("/etc/sudoers.d/visudo"));
 
     let (sudoers_file, existed) = if sudoers_path.exists() {
         let file = File::options()
