From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Evan Maddock <maddock.evan@vivaldi.net>
Date: Mon, 20 Apr 2026 15:46:30 -0400
Subject: [PATCH] build: Update to Budgie 10.10 series

Signed-off-by: Evan Maddock <maddock.evan@vivaldi.net>
---
 plugin/meson.build | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/plugin/meson.build b/plugin/meson.build
index 1884880..4b525bb 100644
--- a/plugin/meson.build
+++ b/plugin/meson.build
@@ -13,7 +13,7 @@ applet_cputemp_sources = [
 applet_cputemp_deps = [
     dependency('gtk+-3.0', version: '>= 3.24.0'),
     dependency('libpeas-2', version: '>= 1.99.0'),
-    dependency('budgie-2.0', version: '>= 3')
+    dependency('budgie-3.0', version: '>= 4')
 ]
 
 shared_library(
