From 70fdbed2826b43f9804094427d68a7c1e5c521e5 Mon Sep 17 00:00:00 2001
From: Reilly Brogan <reilly@reillybrogan.com>
Date: Sun, 15 Oct 2023 21:38:25 -0500
Subject: [PATCH] Fix compile error

---
 cmake/ETLBuildMod.cmake | 10 ++++++++++
 1 file changed, 10 insertions(+)

diff --git a/cmake/ETLBuildMod.cmake b/cmake/ETLBuildMod.cmake
index e46cdd173..0fe884560 100644
--- a/cmake/ETLBuildMod.cmake
+++ b/cmake/ETLBuildMod.cmake
@@ -51,6 +51,16 @@ if(NOT ANDROID)
 			"src/luasql/ls_sqlite3.c"
 		)
 		set(QAGAME_SRC ${QAGAME_SRC} ${LUASQL_SRC})
+	elseif(FEATURE_DBMS)
+		target_compile_definitions(qagame PRIVATE FEATURE_DBMS)
+
+		if(BUNDLED_SQLITE3)
+			target_link_libraries(qagame bundled_sqlite3)
+		else() # BUNDLED_SQLITE3
+			find_package(SQLite3 REQUIRED)
+			target_link_libraries(qagame ${SQLITE3_LIBRARY})
+			target_include_directories(qagame PUBLIC ${SQLITE3_INCLUDE_DIR})
+		endif()
 	endif()
 
 	if(FEATURE_SERVERMDX)
-- 
2.42.0

