From 6e847d0608cb6901d59be8eafdc734b35bf5e991 Mon Sep 17 00:00:00 2001
From: Tobias Stoeckmann <tobias@stoeckmann.org>
Date: Thu, 18 May 2023 17:55:21 +0200
Subject: [PATCH] pam_shells: Plug econf memory leak

Signed-off-by: Tobias Stoeckmann <tobias@stoeckmann.org>
---
 modules/pam_shells/pam_shells.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/modules/pam_shells/pam_shells.c b/modules/pam_shells/pam_shells.c
index abebdd0c..9320d350 100644
--- a/modules/pam_shells/pam_shells.c
+++ b/modules/pam_shells/pam_shells.c
@@ -104,6 +104,7 @@ static int perform_check(pam_handle_t *pamh)
         if (!retval)
 	   break;
     }
+    econf_free (keys);
     econf_free (key_file);
 #else
     char shellFileLine[256];
-- 
2.42.0

