From 4c7aa3b2c66b245a3faeb321d31cc0d031341e48 Mon Sep 17 00:00:00 2001
From: Flaviu Tamas <me@flaviutamas.com>
Date: Sun, 12 Nov 2023 16:07:19 -0500
Subject: [PATCH] pam_shells: improve error message on denial

---
 modules/pam_shells/pam_shells.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/modules/pam_shells/pam_shells.c b/modules/pam_shells/pam_shells.c
index 548757048..2fae67cfc 100644
--- a/modules/pam_shells/pam_shells.c
+++ b/modules/pam_shells/pam_shells.c
@@ -173,6 +173,7 @@ static int perform_check(pam_handle_t *pamh)
 #endif
 
     if (retval) {
+	pam_syslog(pamh, LOG_NOTICE, "User has an invalid shell '%s'", userShell);
 	return PAM_AUTH_ERR;
     } else {
 	return PAM_SUCCESS;
