diff --git a/driver/razeraccessory_driver.c b/driver/razeraccessory_driver.c
index 8a4ad54..513dde8 100644
--- a/driver/razeraccessory_driver.c
+++ b/driver/razeraccessory_driver.c
@@ -25,6 +25,7 @@
 #include <linux/usb/input.h>
 #include <linux/hid.h>
 #include <linux/random.h>
+#include <linux/version.h>
 
 #include "razeraccessory_driver.h"
 #include "razercommon.h"
@@ -2079,7 +2080,9 @@ MODULE_DEVICE_TABLE(hid, razer_devices);
 static struct hid_driver razer_accessory_driver = {
     .name = "razeraccessory",
     .id_table = razer_devices,
+#if LINUX_VERSION_CODE > KERNEL_VERSION(5, 3, 0)
     .match = razer_accessory_match,
+#endif
     .probe = razer_accessory_probe,
     .remove = razer_accessory_disconnect,
     .raw_event = razer_raw_event,
