diff --git a/src/libide/tweaks/ide-tweaks-window.c b/src/libide/tweaks/ide-tweaks-window.c
index f2553a396e96988ce3d8bc6a40f15267684985cb..dcb55bb3516b53d67976d72575aa24155904d678 100644
--- a/src/libide/tweaks/ide-tweaks-window.c
+++ b/src/libide/tweaks/ide-tweaks-window.c
@@ -32,8 +32,6 @@
 #include "ide-tweaks-panel-list-private.h"
 #include "ide-tweaks-window.h"
 
-G_DEFINE_AUTOPTR_CLEANUP_FUNC (GtkStackPage, g_object_unref)
-
 struct _IdeTweaksWindow
 {
   AdwWindow               parent_instance;
