diff -uraN gnome-builder-43.2/meson.build omv-gnome-builder-43.2/meson.build
--- gnome-builder-43.2/meson.build	2022-10-05 02:10:26.000000000 +0200
+++ omv-gnome-builder-43.2/meson.build	2022-10-10 17:20:55.093692654 +0200
@@ -177,7 +177,6 @@
   '-Werror=redundant-decls',
   '-Werror=return-type',
   '-Werror=sequence-point',
-  '-Werror=shadow',
   '-Werror=trigraphs',
   '-Werror=undef',
   '-Werror=write-strings',
