From 6fe096f730138d21b94ede53fba03be96b9e7f0d Mon Sep 17 00:00:00 2001
From: Ikey Doherty <ikey@solus-project.com>
Date: Sun, 29 May 2016 06:24:15 +0100
Subject: [PATCH] Use a stateless implementation for OpenCL provider
 integration

Signed-off-by: Ikey Doherty <ikey@solus-project.com>
---
 ocl_icd_loader.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/ocl_icd_loader.c b/ocl_icd_loader.c
index e42ed3f..46e9f17 100644
--- a/ocl_icd_loader.c
+++ b/ocl_icd_loader.c
@@ -49,7 +49,7 @@ SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
 #define DEBUG_OCL_ICD_PROVIDE_DUMP_FIELD
 #include "ocl_icd_debug.h"
 
-#define ETC_OPENCL_VENDORS "/etc/OpenCL/vendors"
+#define ETC_OPENCL_VENDORS "/usr/share/OpenCL/vendors"
 
 int debug_ocl_icd_mask=0;
 
-- 
2.8.3

