From bc121fc1f8cf0afc48361d3d2e5851afcd589c66 Mon Sep 17 00:00:00 2001
From: Brice Videau <bvideau@anl.gov>
Date: Tue, 23 May 2023 10:37:22 -0500
Subject: [PATCH] Add CL_NO_NON_ICD_DISPATCH_EXTENSION_PROTOTYPES to avoid non
 exported function definitions.

---
 Makefile.am | 1 +
 1 file changed, 1 insertion(+)

diff --git a/Makefile.am b/Makefile.am
index 9bb7100..7248d4d 100644
--- a/Makefile.am
+++ b/Makefile.am
@@ -42,6 +42,7 @@ stamp-generator: icd_generator.rb
 # ignore the warning in OpenCL headers when using old interface
 libOpenCL_la_CFLAGS= $(NO_DEPRECATED_DECLARATIONS_FLAGS) \
 	$(AM_CFLAGS) $(PTHREAD_CFLAGS) \
+	-DCL_NO_NON_ICD_DISPATCH_EXTENSION_PROTOTYPES \
 	-DETC_OPENCL_VENDORS=\"@OCL_ICD_VENDORDIR@\" \
 	-DETC_OPENCL_LAYERS=\"@OCL_ICD_LAYERDIR@\"
 
