From 8a8a09d44f91ab1b6993be76d42f58b67d061a66 Mon Sep 17 00:00:00 2001
From: Brice Videau <bvideau@anl.gov>
Date: Thu, 10 Nov 2022 11:15:36 -0600
Subject: [PATCH] FIX ABI issues.

---
 ocl_icd_loader.c | 12 ++++++------
 1 file changed, 6 insertions(+), 6 deletions(-)

diff --git a/ocl_icd_loader.c b/ocl_icd_loader.c
index b90f799..c252e8c 100644
--- a/ocl_icd_loader.c
+++ b/ocl_icd_loader.c
@@ -891,12 +891,12 @@ getDefaultPlatformID() {
   typeof(name) name##_hid __attribute__ ((alias (#name), visibility("hidden")))
 #endif
 
-typedef enum {
-  CL_ICDL_OCL_VERSION=1,
-  CL_ICDL_VERSION=2,
-  CL_ICDL_NAME=3,
-  CL_ICDL_VENDOR=4,
-} cl_icdl_info;
+typedef cl_uint cl_icdl_info;
+
+#define CL_ICDL_OCL_VERSION 1
+#define CL_ICDL_VERSION     2
+#define CL_ICDL_NAME        3
+#define CL_ICDL_VENDOR      4
 
 static cl_int clGetICDLoaderInfoOCLICD(
   cl_icdl_info     param_name,
