From aca316400f2c2b6d08e80af93486452095f582bf Mon Sep 17 00:00:00 2001
From: Ikey Doherty <ikey@solus-project.com>
Date: Sun, 16 Oct 2016 09:13:44 +0100
Subject: [PATCH] Support a stateless configuration

Signed-off-by: Ikey Doherty <ikey@solus-project.com>
---
 configure.ac         |  2 +-
 src/server/speechd.c | 15 ++++++++++-----
 2 files changed, 11 insertions(+), 6 deletions(-)

diff --git a/configure.ac b/configure.ac
index d3f019a..5491c7d 100644
--- a/configure.ac
+++ b/configure.ac
@@ -342,7 +342,7 @@ GETTEXT_PACKAGE=speech-dispatcher
 AC_SUBST([GETTEXT_PACKAGE])
 
 # Paths for configuration files:
-spdconfdir='${sysconfdir}/speech-dispatcher'
+spdconfdir='${datadir}/defaults/speech-dispatcher'
 AC_SUBST([spdconfdir])
 clientconfdir='${spdconfdir}/clients'
 AC_SUBST([clientconfdir])
diff --git a/src/server/speechd.c b/src/server/speechd.c
index 111a7cb..da403d9 100644
--- a/src/server/speechd.c
+++ b/src/server/speechd.c
@@ -1069,12 +1069,17 @@ int main(int argc, char *argv[])
 			if (!g_file_test
 			    (test_speechd_conf_file, G_FILE_TEST_IS_REGULAR)) {
 				/* If the local configuration file doesn't exist, read the global configuration */
-				if (strcmp(SYS_CONF, ""))
+				if (g_file_test("/etc/speech-dispatcher", G_FILE_TEST_IS_DIR)) {
+					if (strcmp(SYS_CONF, ""))
+						SpeechdOptions.conf_dir =
+						g_strdup(SYS_CONF);
+					else
+						SpeechdOptions.conf_dir =
+						g_strdup("/etc/speech-dispatcher/");
+				} else {
 					SpeechdOptions.conf_dir =
-					    g_strdup(SYS_CONF);
-				else
-					SpeechdOptions.conf_dir =
-					    g_strdup("/etc/speech-dispatcher/");
+					g_strdup("/usr/share/defaults/speech-dispatcher/");
+				}
 			}
 			g_free(test_speechd_conf_file);
 		}
-- 
2.10.1

