From 1f24134226d9213d7372d2bc90ec35089e267c42 Mon Sep 17 00:00:00 2001
From: Ikey Doherty <ikey@solus-project.com>
Date: Sun, 16 Oct 2016 10:33:18 +0100
Subject: [PATCH 2/2] Apply default configuration

Signed-off-by: Ikey Doherty <ikey@solus-project.com>
---
 config/speechd.conf | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/config/speechd.conf b/config/speechd.conf
index b1ab847..6cf424e 100644
--- a/config/speechd.conf
+++ b/config/speechd.conf
@@ -211,7 +211,7 @@ DefaultVolume 100
 #    either relative (to etc/speech-dispatcher/modules/) or absolute
 
 #AddModule "espeak"       "sd_espeak"   "espeak.conf"
-#AddModule "espeak-ng"    "sd_espeak-ng" "espeak-ng.conf"
+AddModule "espeak-ng"    "sd_espeak-ng" "espeak-ng.conf"
 #AddModule "festival"     "sd_festival"  "festival.conf"
 #AddModule "flite"        "sd_flite"     "flite.conf"
 #AddModule "ivona"	 "sd_ivona"    "ivona.conf"
@@ -240,7 +240,7 @@ DefaultVolume 100
 # The DefaultModule selects which output module is the default.  You
 # must use one of the names of the modules loaded with AddModule.
 
-DefaultModule espeak
+DefaultModule espeak-ng
 
 # The LanguageDefaultModule selects which output modules are prefered
 # for specified languages.
-- 
2.10.1

