From 544356103e788f62a74690216ee52af259f11798 Mon Sep 17 00:00:00 2001
From: Martin Reboredo <yakoyoku@gmail.com>
Date: Tue, 1 Sep 2020 16:25:09 -0300
Subject: [PATCH] LKJ

Signed-off-by: Martin Reboredo <yakoyoku@gmail.com>
---
 config/speechd.conf | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/config/speechd.conf b/config/speechd.conf
index 49e18ce6..3412e0ae 100644
--- a/config/speechd.conf
+++ b/config/speechd.conf
@@ -251,7 +251,7 @@ SymbolsPreprocFile "orca-chars.dic"
 #    either relative (to etc/speech-dispatcher/modules/) or absolute
 
 #AddModule "espeak"                   "sd_espeak"    "espeak.conf"
-#AddModule "espeak-ng"                "sd_espeak-ng" "espeak-ng.conf"
+AddModule "espeak-ng"                "sd_espeak-ng" "espeak-ng.conf"
 #AddModule "festival"                 "sd_festival"  "festival.conf"
 #AddModule "flite"                    "sd_flite"     "flite.conf"
 #AddModule "ivona"                    "sd_ivona"     "ivona.conf"
-- 
2.27.0

