From 28490d8977abce841982cc131fbe511c37a9b955 Mon Sep 17 00:00:00 2001
From: "F. von Gellhorn" <flinux@vongellhorn.ch>
Date: Sat, 2 Feb 2019 01:06:14 +0100
Subject: [PATCH 1/1] LKJ

---
 config/speechd.conf | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/config/speechd.conf b/config/speechd.conf
index 089b080..05886d8 100644
--- a/config/speechd.conf
+++ b/config/speechd.conf
@@ -224,7 +224,7 @@ DefaultVolume 100
 #    either relative (to etc/speech-dispatcher/modules/) or absolute
 
 #AddModule "espeak"                   "sd_espeak"    "espeak.conf"
-#AddModule "espeak-ng"                "sd_espeak-ng" "espeak-ng.conf"
+AddModule "espeak-ng"                "sd_espeak-ng" "espeak-ng.conf"
 #AddModule "festival"                 "sd_festival"  "festival.conf"
 #AddModule "flite"                    "sd_flite"     "flite.conf"
 #AddModule "ivona"                    "sd_ivona"     "ivona.conf"
-- 
2.20.1

