From 9887dbf8252b3609f11fa84c563f57be0beef6fd Mon Sep 17 00:00:00 2001
From: Nazar Stasiv <nazar@autistici.org>
Date: Wed, 2 Feb 2022 12:08:42 +0200
Subject: [PATCH] config patch

---
 config/speechd.conf | 6 ++++--
 1 file changed, 4 insertions(+), 2 deletions(-)

diff --git a/config/speechd.conf b/config/speechd.conf
index b7a514f..a5b091a 100644
--- a/config/speechd.conf
+++ b/config/speechd.conf
@@ -256,7 +256,7 @@ SymbolsPreprocFile "orca-chars.dic"
 #    or ~/.config/speech-dispatcher) or absolute
 
 #AddModule "espeak"                   "sd_espeak"    "espeak.conf"
-#AddModule "espeak-ng"                "sd_espeak-ng" "espeak-ng.conf"
+AddModule "espeak-ng"                "sd_espeak-ng" "espeak-ng.conf"
 #AddModule "festival"                 "sd_festival"  "festival.conf"
 #AddModule "flite"                    "sd_flite"     "flite.conf"
 #AddModule "ivona"                    "sd_ivona"     "ivona.conf"
@@ -271,7 +271,7 @@ SymbolsPreprocFile "orca-chars.dic"
 #AddModule "kali"                     "sd_kali"      "kali.conf"
 #AddModule "mary-generic"             "sd_generic"   "mary-generic.conf"
 #AddModule "baratinoo"                "sd_baratinoo" "baratinoo.conf"
-#AddModule "rhvoice"                  "sd_rhvoice"   "rhvoice.conf"
+AddModule "rhvoice"                  "sd_rhvoice"   "rhvoice.conf"
 #AddModule "voxin"                    "sd_voxin"     "voxin.conf"
 
 # DO NOT REMOVE the following line unless you have
@@ -297,6 +297,8 @@ SymbolsPreprocFile "orca-chars.dic"
 #LanguageDefaultModule "en"  "espeak"
 #LanguageDefaultModule "cs"  "festival"
 #LanguageDefaultModule "es"  "festival"
+#LanguageDefaultModule "ru"  "rhvoice"
+#LanguageDefaultModule "uk"  "rhvoice"
 
 # -----CLIENT SPECIFIC CONFIGURATION-----
 
-- 
2.34.0

