diff --git a/Makefile b/Makefile
index b3f548b7..c81e78bc 100644
--- a/Makefile
+++ b/Makefile
@@ -50,9 +50,9 @@ BUILD_FLAGS_LAUNCHER+=-ldflags '${GO_LDFLAGS_LAUNCHER}'
 DEPLOY_DIR:=cmd/${TARGET_CMD}/deploy
 DIRNAME:=$(shell basename ${CURDIR})
 
-LAUNCHER_EXE:=proton-bridge
-BRIDGE_EXE=bridge
-BRIDGE_GUI_EXE_NAME=bridge-gui
+LAUNCHER_EXE:=protonmail-bridge
+BRIDGE_EXE=proton-bridge
+BRIDGE_GUI_EXE_NAME=proton-bridge-gui
 BRIDGE_GUI_EXE=${BRIDGE_GUI_EXE_NAME}
 LAUNCHER_PATH:=cmd/launcher
 
diff --git a/cmd/launcher/main.go b/cmd/launcher/main.go
index e1cb0984..a6c6c1d1 100644
--- a/cmd/launcher/main.go
+++ b/cmd/launcher/main.go
@@ -44,8 +44,8 @@ import (
 
 const (
 	appName = "Proton Mail Launcher"
-	exeName = "bridge"
-	guiName = "bridge-gui"
+	exeName = "proton-bridge"
+	guiName = "proton-bridge-gui"
 
 	FlagCLI                 = "cli"
 	FlagCLIShort            = "c"
diff --git a/internal/frontend/bridge-gui/FindQt.cmake b/internal/frontend/bridge-gui/FindQt.cmake
index 351d7b7b..3cbe1b0d 100644
--- a/internal/frontend/bridge-gui/FindQt.cmake
+++ b/internal/frontend/bridge-gui/FindQt.cmake
@@ -3,6 +3,6 @@ if (NOT QMAKE_EXE)
     message(FATAL_ERROR "Could not locate qmake executable, make sure you have Qt 6 installed in that qmake is in your PATH environment variable.")
 endif()
 message(STATUS "Found qmake at ${QMAKE_EXE}")
-execute_process(COMMAND "${QMAKE_EXE}" -query QT_INSTALL_PREFIX OUTPUT_VARIABLE QT_DIR OUTPUT_STRIP_TRAILING_WHITESPACE)
+execute_process(COMMAND "${QMAKE_EXE}" -query QT_HOST_DATA OUTPUT_VARIABLE QT_DIR OUTPUT_STRIP_TRAILING_WHITESPACE)
 
 set(CMAKE_PREFIX_PATH  ${QT_DIR} ${CMAKE_PREFIX_PATH})
diff --git a/internal/frontend/bridge-gui/bridge-gui/DeployLinux.cmake b/internal/frontend/bridge-gui/bridge-gui/DeployLinux.cmake
index 10e1ea16..c318eaf6 100644
--- a/internal/frontend/bridge-gui/bridge-gui/DeployLinux.cmake
+++ b/internal/frontend/bridge-gui/bridge-gui/DeployLinux.cmake
@@ -54,9 +54,9 @@ AppendQt6Lib("libQt6Gui.so.6")
 AppendQt6Lib("libQt6Core.so.6")
 AppendQt6Lib("libQt6QuickTemplates2.so.6")
 AppendQt6Lib("libQt6DBus.so.6")
-AppendQt6Lib("libicui18n.so.56")
-AppendQt6Lib("libicuuc.so.56")
-AppendQt6Lib("libicudata.so.56")
+AppendQt6Lib("libicui18n.so")
+AppendQt6Lib("libicuuc.so")
+AppendQt6Lib("libicudata.so")
 AppendQt6Lib("libQt6XcbQpa.so.6")
 AppendQt6Lib("libQt6WaylandClient.so.6")
 AppendQt6Lib("libQt6WlShellIntegration.so.6")
diff --git a/internal/frontend/bridge-gui/bridge-gui/main.cpp b/internal/frontend/bridge-gui/bridge-gui/main.cpp
index acc3cf36..b0490f89 100644
--- a/internal/frontend/bridge-gui/bridge-gui/main.cpp
+++ b/internal/frontend/bridge-gui/bridge-gui/main.cpp
@@ -53,7 +53,7 @@ QString const exeSuffix;
 
 QString const bridgeLock = "bridge-v3.lock"; ///< The file name used for the bridge-gui lock file.
 QString const bridgeGUILock = "bridge-v3-gui.lock"; ///< The file name used for the bridge-gui lock file.
-QString const exeName = "bridge" + exeSuffix; ///< The bridge executable file name.*
+QString const exeName = "proton-bridge" + exeSuffix; ///< The bridge executable file name.*
 qint64 const grpcServiceConfigWaitDelayMs = 180000; ///< The wait delay for the gRPC config file in milliseconds.
 QString const waitFlag = "--wait"; ///< The wait command-line flag.
 
diff --git a/internal/frontend/bridge-gui/bridgepp/CMakeLists.txt b/internal/frontend/bridge-gui/bridgepp/CMakeLists.txt
index c1c4a04f..99b0c497 100644
--- a/internal/frontend/bridge-gui/bridgepp/CMakeLists.txt
+++ b/internal/frontend/bridge-gui/bridgepp/CMakeLists.txt
@@ -48,7 +48,7 @@ qt_standard_project_setup()
 #****************************************************************************************************************************************************
 # gRPC / Protobuf
 #****************************************************************************************************************************************************
-find_package(Protobuf CONFIG REQUIRED)
+find_package(Protobuf REQUIRED)
 message(STATUS "Using protobuf ${Protobuf_VERSION}")
 
 find_package(gRPC CONFIG REQUIRED)
