diff --git a/internal/frontend/bridge-gui/bridge-gui/main.cpp b/internal/frontend/bridge-gui/bridge-gui/main.cpp
index ff017dd1..9075d78e 100644
--- a/internal/frontend/bridge-gui/bridge-gui/main.cpp
+++ b/internal/frontend/bridge-gui/bridge-gui/main.cpp
@@ -81,6 +81,7 @@ void initQtApplication() {
     QGuiApplication::setOrganizationName(PROJECT_VENDOR);
     QGuiApplication::setOrganizationDomain("proton.ch");
     QGuiApplication::setQuitOnLastWindowClosed(false);
+    QGuiApplication::setDesktopFileName("ch.protonmail.protonmail-bridge");
 
 #ifndef Q_OS_MACOS
 
