From 8cdf6ba3a8c990a2329f567a728755f2751506b1 Mon Sep 17 00:00:00 2001
From: Campbell Jones <git@serebit.com>
Date: Sat, 27 Mar 2021 14:30:56 -0400
Subject: [PATCH] set llvm version to 11.1.0

---
 CMakeLists.txt | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index ecebb4c..79c792c 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -1,6 +1,6 @@
 cmake_minimum_required(VERSION 3.3)
 
-set (BASE_LLVM_VERSION 11.0.0)
+set (BASE_LLVM_VERSION 11.1.0)
 set(LLVM_SPIRV_VERSION ${BASE_LLVM_VERSION}.0)
 
 option(LLVM_SPIRV_INCLUDE_TESTS
-- 
2.30.2

