From f0d5b864959480a0c1b1b4538a543aa4bf6756b4 Mon Sep 17 00:00:00 2001
From: Rye Mutt <rye@alchemyviewer.org>
Date: Wed, 27 Dec 2023 21:38:00 -0500
Subject: [PATCH] Fix memory corruption introduced in
 61e181c8ae93dbf56040336179c9954078bd1399

---
 inflate.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/inflate.c b/inflate.c
index af5abf11a5..fe55c498e3 100644
--- a/inflate.c
+++ b/inflate.c
@@ -1350,8 +1350,8 @@ int32_t Z_EXPORT PREFIX(inflateCopy)(PREFIX3(stream) *dest, PREFIX3(stream) *sou
     copy->next = copy->codes + (state->next - state->codes);
 
     /* window */
+    copy->window = NULL;
     if (state->window != NULL) {
-        copy->window = NULL;
         if (PREFIX(inflate_ensure_window)(copy)) {
             ZFREE_STATE(source, copy);
             return Z_MEM_ERROR;
