diff --git a/FL/x.H b/FL/x.H
index 702ac86..99e691f 100644
--- a/FL/x.H
+++ b/FL/x.H
@@ -106,7 +106,7 @@ FL_EXPORT Fl_Region XRectangleRegion(int x, int y, int w, int h); // in fl_rect.
 // With Xlib / X11 fonts, fl_xfont will return the current selected font.
 // With XFT / X11 fonts, fl_xfont will attempt to return the bitmap "core" font most
 // similar to (usually the same as) the current XFT font.
-class Fl_XFont_On_Demand
+class FL_EXPORT Fl_XFont_On_Demand
 {
 public:
   Fl_XFont_On_Demand(XFontStruct* p = NULL) : ptr(p) { }
diff --git a/configure b/configure
index 1e85302..f545ec9 100755
--- a/configure
+++ b/configure
@@ -3690,24 +3690,24 @@ fi
    ;;
 esac
 
-LINKFLTK="../lib/libfltk.a"
-LINKFLTKFORMS="../lib/libfltk_forms.a"
-LINKFLTKGL="../lib/libfltk_gl.a"
-LINKFLTKIMG="../lib/libfltk_images.a"
+LINKFLTK="../lib/libfltk.so"
+LINKFLTKFORMS="../lib/libfltk_forms.so"
+LINKFLTKGL="../lib/libfltk_gl.so"
+LINKFLTKIMG="../lib/libfltk_images.so"
 GLDEMOS="gldemos"
 
-LIBEXT=".a"
-LIBNAME="../lib/libfltk.a"
-FLLIBNAME="../lib/libfltk_forms.a"
-GLLIBNAME="../lib/libfltk_gl.a"
-IMGLIBNAME="../lib/libfltk_images.a"
-CAIROLIBNAME="../lib/libfltk_cairo.a"
+LIBEXT=".so"
+LIBNAME="../lib/libfltk.so"
+FLLIBNAME="../lib/libfltk_forms.so"
+GLLIBNAME="../lib/libfltk_gl.so"
+IMGLIBNAME="../lib/libfltk_images.so"
+CAIROLIBNAME="../lib/libfltk_cairo.so"
 
-LIBBASENAME="libfltk.a"
-FLLIBBASENAME="libfltk_forms.a"
-GLLIBBASENAME="libfltk_gl.a"
-IMGLIBBASENAME="libfltk_images.a"
-CAIROLIBBASENAME="libfltk_cairo.a"
+LIBBASENAME="libfltk.so"
+FLLIBBASENAME="libfltk_forms.so"
+GLLIBBASENAME="libfltk_gl.so"
+IMGLIBBASENAME="libfltk_images.so"
+CAIROLIBBASENAME="libfltk_cairo.so"
 
 CAIRODIR=""
 CAIROFLAGS=""
@@ -3735,7 +3735,7 @@ if test x$enable_cairoext = xyes; then
 	  CAIROFLAGS="`pkg-config --cflags cairo`"
 	  CAIROLIBS="-lcairo -lpixman-1"
 	  CXXFLAGS="$CAIROFLAGS $CXXFLAGS"
-	  LINKFLTKCAIRO="../lib/libfltk_cairo.a"
+	  LINKFLTKCAIRO="../lib/libfltk_cairo.so"
 	  FLTKCAIROOPTION="-L ../cairo -lfltk_cairo$SHAREDSUFFIX"
 	  LIBS="$CAIROLIBS $LIBS"
 	  	  LINKFLTK+=" $LINKFLTKCAIRO"
@@ -3747,7 +3747,7 @@ else
 	  CAIROFLAGS="`pkg-config --cflags cairo`"
 	  CAIROLIBS="-lcairo -lpixman-1"
 	  CXXFLAGS="$CAIROFLAGS $CXXFLAGS"
-	  LINKFLTKCAIRO="../lib/libfltk_cairo.a"
+	  LINKFLTKCAIRO="../lib/libfltk_cairo.so"
 	  FLTKCAIROOPTION="-L ../cairo -lfltk_cairo$SHAREDSUFFIX"
     fi
 fi
@@ -4062,7 +4062,7 @@ else
     PICFLAG=0
     SHAREDSUFFIX=""
     FLUID="fluid"
-    LINKSHARED="$LINKFLTKCAIRO ../lib/libfltk_images.a ../lib/libfltk_forms.a ../lib/libfltk.a"
+    LINKSHARED="$LINKFLTKCAIRO ../lib/libfltk_images.so ../lib/libfltk_forms.so ../lib/libfltk.so"
 fi
 
 if test "$fltk_cross_compiling" = "yes"; then
diff --git a/configure.ac b/configure.ac
index 70ff5f9..6bc05c6 100644
--- a/configure.ac
+++ b/configure.ac
@@ -95,24 +95,24 @@ case $host_os in
 esac
 
 dnl Define the libraries and link options we will need.
-LINKFLTK="../lib/libfltk.a"
-LINKFLTKFORMS="../lib/libfltk_forms.a"
-LINKFLTKGL="../lib/libfltk_gl.a"
-LINKFLTKIMG="../lib/libfltk_images.a"
+LINKFLTK="../lib/libfltk.so"
+LINKFLTKFORMS="../lib/libfltk_forms.so"
+LINKFLTKGL="../lib/libfltk_gl.so"
+LINKFLTKIMG="../lib/libfltk_images.so"
 GLDEMOS="gldemos"
 
-LIBEXT=".a"
-LIBNAME="../lib/libfltk.a"
-FLLIBNAME="../lib/libfltk_forms.a"
-GLLIBNAME="../lib/libfltk_gl.a"
-IMGLIBNAME="../lib/libfltk_images.a"
-CAIROLIBNAME="../lib/libfltk_cairo.a"
+LIBEXT=".so"
+LIBNAME="../lib/libfltk.so"
+FLLIBNAME="../lib/libfltk_forms.so"
+GLLIBNAME="../lib/libfltk_gl.so"
+IMGLIBNAME="../lib/libfltk_images.so"
+CAIROLIBNAME="../lib/libfltk_cairo.so"
 
-LIBBASENAME="libfltk.a"
-FLLIBBASENAME="libfltk_forms.a"
-GLLIBBASENAME="libfltk_gl.a"
-IMGLIBBASENAME="libfltk_images.a"
-CAIROLIBBASENAME="libfltk_cairo.a"
+LIBBASENAME="libfltk.so"
+FLLIBBASENAME="libfltk_forms.so"
+GLLIBBASENAME="libfltk_gl.so"
+IMGLIBBASENAME="libfltk_images.so"
+CAIROLIBBASENAME="libfltk_cairo.so"
 
 dnl Check for Cairo library unless disabled...
 CAIRODIR=""
@@ -133,7 +133,7 @@ if test x$enable_cairoext = xyes; then
 	  CAIROFLAGS="`pkg-config --cflags cairo`"
 	  CAIROLIBS="-lcairo -lpixman-1"
 	  CXXFLAGS="$CAIROFLAGS $CXXFLAGS"
-	  LINKFLTKCAIRO="../lib/libfltk_cairo.a"
+	  LINKFLTKCAIRO="../lib/libfltk_cairo.so"
 	  FLTKCAIROOPTION="-L ../cairo -lfltk_cairo$SHAREDSUFFIX"
 	  LIBS="$CAIROLIBS $LIBS"
 	  dnl $LINKFLTKCAIRO
@@ -147,7 +147,7 @@ else
 	  CAIROFLAGS="`pkg-config --cflags cairo`"
 	  CAIROLIBS="-lcairo -lpixman-1"
 	  CXXFLAGS="$CAIROFLAGS $CXXFLAGS"
-	  LINKFLTKCAIRO="../lib/libfltk_cairo.a"
+	  LINKFLTKCAIRO="../lib/libfltk_cairo.so"
 	  FLTKCAIROOPTION="-L ../cairo -lfltk_cairo$SHAREDSUFFIX"
     fi
 fi
@@ -377,7 +377,7 @@ else
     PICFLAG=0
     SHAREDSUFFIX=""
     FLUID="fluid"
-    LINKSHARED="$LINKFLTKCAIRO ../lib/libfltk_images.a ../lib/libfltk_forms.a ../lib/libfltk.a"
+    LINKSHARED="$LINKFLTKCAIRO ../lib/libfltk_images.so ../lib/libfltk_forms.so ../lib/libfltk.so"
 fi
 
 dnl Define the fluid executable used when building the test programs.
diff --git a/fltk-config.in b/fltk-config.in
index a33de79..b40148d 100644
--- a/fltk-config.in
+++ b/fltk-config.in
@@ -53,21 +53,6 @@ CAIROFLAGS="@CAIROFLAGS@"
 if test -f "$selfdir/FL/Fl_Window.H"; then
 	includedir="$selfdir"
 	libdir="$selfdir/lib"
-
-	if test -f "$libdir/libfltk_jpeg.a"; then
-		CFLAGS="-I$includedir/jpeg $CFLAGS"
-		CXXFLAGS="-I$includedir/jpeg $CXXFLAGS"
-	fi
-
-	if test -f "$libdir/libfltk_z.a"; then
-		CFLAGS="-I$includedir/zlib $CFLAGS"
-		CXXFLAGS="-I$includedir/zlib $CXXFLAGS"
-	fi
-
-	if test -f "$libdir/libfltk_png.a"; then
-		CFLAGS="-I$includedir/png $CFLAGS"
-		CXXFLAGS="-I$includedir/png $CXXFLAGS"
-	fi
 fi
 
 if test -d $includedir/FL/images; then
@@ -75,11 +60,6 @@ if test -d $includedir/FL/images; then
 	CXXFLAGS="-I$includedir/FL/images $CXXFLAGS"
 fi
 
-if test -f "$libdir/libfltk_cairo.a"; then
-	CFLAGS="$CAIROFLAGS $CFLAGS"
-	CXXFLAGS="$CAIROFLAGS $CXXFLAGS"
-fi
-
 # libraries to link with:
 LIBNAME="@LIBNAME@"
 DSONAME="@DSONAME@"
@@ -233,29 +213,23 @@ else
 fi
 
 # Calculate needed libraries
-LDSTATIC="$libdir/libfltk.a $LDLIBS"
 LDLIBS="-lfltk$SHAREDSUFFIX $LDLIBS"
 
 if test x$use_forms = xyes; then
     LDLIBS="-lfltk_forms$SHAREDSUFFIX $LDLIBS"
-    LDSTATIC="$libdir/libfltk_forms.a $LDSTATIC"
 fi
 if test x$use_gl = xyes; then
     LDLIBS="-lfltk_gl$SHAREDSUFFIX @GLLIBS@ $LDLIBS"
-    LDSTATIC="$libdir/libfltk_gl.a @GLLIBS@ $LDSTATIC"
 fi
 if test x$use_images = xyes; then
     LDLIBS="-lfltk_images$SHAREDSUFFIX $IMAGELIBS $LDLIBS"
-    LDSTATIC="$libdir/libfltk_images.a $STATICIMAGELIBS $LDSTATIC"
 fi
 
 if test x$use_cairo = xyes; then
     LDLIBS="-lfltk_cairo$SHAREDSUFFIX $CAIROLIBS $LDLIBS"
-    LDSTATIC="$libdir/libfltk_cairo.a $CAIROLIBS $LDSTATIC"
 fi
 
 LDLIBS="$DSOLINK $LDFLAGS $libs $LDLIBS"
-LDSTATIC="$LDFLAGS $LDSTATIC"
 
 # Answer to user requests
 if test -n "$echo_help"; then
@@ -369,26 +343,26 @@ if test "$echo_ldstaticflags" = "yes"; then
 fi
 
 if test "$echo_libs" = "yes"; then
-    USELIBS="$libdir/libfltk.a"
+    USELIBS="$libdir/libfltk.so"
 
     if test x$use_forms = xyes; then
-        USELIBS="$libdir/libfltk_forms.a $USELIBS"
+        USELIBS="$libdir/libfltk_forms.so $USELIBS"
     fi
 
     if test x$use_gl = xyes; then
-        USELIBS="$libdir/libfltk_gl.a $USELIBS"
+        USELIBS="$libdir/libfltk_gl.so $USELIBS"
     fi
 
     if test x$use_cairo = xyes; then
-        USELIBS="$libdir/libfltk_cairo.a $USELIBS"
+        USELIBS="$libdir/libfltk_cairo.so $USELIBS"
     fi
 
     if test x$use_images = xyes; then
-        USELIBS="$libdir/libfltk_images.a $USELIBS"
+        USELIBS="$libdir/libfltk_images.so $USELIBS"
 
         for lib in fltk_jpeg fltk_png fltk_z; do
-            if test -f $libdir/lib$lib.a; then
-                USELIBS="$libdir/lib$lib.a $USELIBS"
+            if test -f $libdir/lib$lib.so; then
+                USELIBS="$libdir/lib$lib.so $USELIBS"
             fi
 	done
     fi
diff --git a/fltk.list.in b/fltk.list.in
index 9f2a823..2a9eb05 100644
--- a/fltk.list.in
+++ b/fltk.list.in
@@ -54,7 +54,7 @@ $ZLIB=@ZLIB@
 
 %if DSONAME
 %system aix
-f 0555 root sys $libdir/libfltk_s.a src/libfltk_s.a nostrip()
+f 0555 root sys $libdir/libfltk_s.a src/libfltk_s.xa nostrip()
 %system hpux
 f 0555 root sys $libdir/libfltk.sl.@FL_DSO_VERSION@ src/libfltk.sl.@FL_DSO_VERSION@ nostrip()
 %system darwin
@@ -138,23 +138,23 @@ f 0444 root sys $mandir/man1/fltk-config.1 documentation/src/fltk-config.man
 f 0444 root sys $mandir/man3/fltk.3 documentation/src/fltk.man
 
 # Library files
-f 0444 root sys $libdir/libfltk.a lib/libfltk.a
-f 0444 root sys $libdir/libfltk_forms.a lib/libfltk_forms.a
-f 0444 root sys $libdir/libfltk_images.a lib/libfltk_images.a
+f 0444 root sys $libdir/libfltk.a lib/libfltk.so
+f 0444 root sys $libdir/libfltk_forms.a lib/libfltk_forms.so
+f 0444 root sys $libdir/libfltk_images.a lib/libfltk_images.so
 %if GLLIBNAME
-f 0444 root sys $libdir/libfltk_gl.a lib/libfltk_gl.a
+f 0444 root sys $libdir/libfltk_gl.a lib/libfltk_gl.so
 %endif
 
 %if JPEG
-f 0444 root sys $libdir/libfltk_jpeg.a lib/libfltk_jpeg.a
+f 0444 root sys $libdir/libfltk_jpeg.a lib/libfltk_jpeg.so
 %endif
 
 %if PNG
-f 0444 root sys $libdir/libfltk_png.a lib/libfltk_png.a
+f 0444 root sys $libdir/libfltk_png.a lib/libfltk_png.so
 %endif
 
 %if ZLIB
-f 0444 root sys $libdir/libfltk_z.a lib/libfltk_z.a
+f 0444 root sys $libdir/libfltk_z.a lib/libfltk_z.so
 %endif
 
 %if DSONAME
diff --git a/test/Makefile b/test/Makefile
index be49743..3bede9c 100644
--- a/test/Makefile
+++ b/test/Makefile
@@ -302,7 +302,7 @@ buttons$(EXEEXT): buttons.o
 
 blocks$(EXEEXT): blocks.o
 	echo Linking $@...
-	$(CXX) $(ARCHFLAGS) $(CXXFLAGS) $(LDFLAGS) blocks.o -o $@ $(AUDIOLIBS) $(LINKFLTK) $(LDLIBS)
+	$(CXX) $(ARCHFLAGS) $(CXXFLAGS) $(LDFLAGS) blocks.o -o $@ $(AUDIOLIBS) $(LINKSHARED) $(LDLIBS)
 	$(OSX_ONLY) $(RM) -f -r blocks.app
 	$(OSX_ONLY) mkdir -p blocks.app/Contents/MacOS blocks.app/Contents/Resources
 	$(OSX_ONLY) $(INSTALL_BIN) blocks$(EXEEXT) blocks.app/Contents/MacOS
@@ -311,7 +311,7 @@ blocks$(EXEEXT): blocks.o
 
 checkers$(EXEEXT): checkers.o
 	echo Linking $@...
-	$(CXX) $(ARCHFLAGS) $(CXXFLAGS) $(LDFLAGS) checkers.o -o $@ $(LINKFLTK) $(LDLIBS)
+	$(CXX) $(ARCHFLAGS) $(CXXFLAGS) $(LDFLAGS) checkers.o -o $@ $(LINKSHARED) $(LDLIBS)
 	$(OSX_ONLY) $(RM) -f -r checkers.app
 	$(OSX_ONLY) mkdir -p checkers.app/Contents/MacOS checkers.app/Contents/Resources
 	$(OSX_ONLY) $(INSTALL_BIN) checkers$(EXEEXT) checkers.app/Contents/MacOS
@@ -342,14 +342,14 @@ demo$(EXEEXT): demo.o
 
 device$(EXEEXT): device.o $(IMGLIBNAME)
 	echo Linking $@...
-	$(CXX) $(ARCHFLAGS) $(CXXFLAGS) $(LDFLAGS) device.o -o $@ $(LINKFLTKIMG) $(LDLIBS)
+	$(CXX) $(ARCHFLAGS) $(CXXFLAGS) $(LDFLAGS) device.o -o $@ $(LINKSHARED) $(LDLIBS)
 	$(OSX_ONLY) ../fltk-config --post $@
 
 doublebuffer$(EXEEXT): doublebuffer.o
 
 editor$(EXEEXT): editor.o
 	echo Linking $@...
-	$(CXX) $(ARCHFLAGS) $(CXXFLAGS) $(LDFLAGS) editor.o -o $@ $(LINKFLTKIMG) $(LDLIBS)
+	$(CXX) $(ARCHFLAGS) $(CXXFLAGS) $(LDFLAGS) editor.o -o $@ $(LINKSHARED) $(LDLIBS)
 	$(OSX_ONLY) ../fltk-config --post $@
 	$(OSX_ONLY) cp -f mac-resources/editor.plist editor.app/Contents/Info.plist
 
@@ -358,7 +358,7 @@ fast_slow.cxx:	fast_slow.fl ../fluid/fluid$(EXEEXT)
 
 file_chooser$(EXEEXT): file_chooser.o $(IMGLIBNAME)
 	echo Linking $@...
-	$(CXX) $(ARCHFLAGS) $(CXXFLAGS) $(LDFLAGS) file_chooser.o -o $@ $(LINKFLTKIMG) $(LDLIBS)
+	$(CXX) $(ARCHFLAGS) $(CXXFLAGS) $(LDFLAGS) file_chooser.o -o $@ $(LINKSHARED) $(LDLIBS)
 	$(OSX_ONLY) ../fltk-config --post $@
 
 fltk-versions$(EXEEXT): fltk-versions.o
@@ -374,7 +374,7 @@ hello$(EXEEXT): hello.o
 
 help$(EXEEXT): help.o $(IMGLIBNAME)
 	echo Linking $@...
-	$(CXX) $(ARCHFLAGS) $(CXXFLAGS) $(LDFLAGS) help.o -o $@ $(LINKFLTKIMG) $(LDLIBS)
+	$(CXX) $(ARCHFLAGS) $(CXXFLAGS) $(LDFLAGS) help.o -o $@ $(LINKSHARED) $(LDLIBS)
 	$(OSX_ONLY) ../fltk-config --post $@
 	$(OSX_ONLY) mkdir -p help.app/Contents/Resources
 	$(OSX_ONLY) cp -f help_dialog.html help.app/Contents/Resources/
@@ -394,7 +394,7 @@ input_choice$(EXEEXT): input_choice.o
 
 keyboard$(EXEEXT): keyboard_ui.o keyboard.o
 	echo Linking $@...
-	$(CXX) $(ARCHFLAGS) $(CXXFLAGS) $(LDFLAGS) -o $@ keyboard.o keyboard_ui.o $(LINKFLTK) $(LDLIBS)
+	$(CXX) $(ARCHFLAGS) $(CXXFLAGS) $(LDFLAGS) -o $@ keyboard.o keyboard_ui.o $(LINKSHARED) $(LDLIBS)
 	$(OSX_ONLY) ../fltk-config --post $@
 keyboard_ui.o:	keyboard_ui.h
 keyboard_ui.cxx:	keyboard_ui.fl ../fluid/fluid$(EXEEXT)
@@ -410,7 +410,7 @@ list_visuals$(EXEEXT): list_visuals.o
 
 mandelbrot$(EXEEXT): mandelbrot_ui.o mandelbrot.o
 	echo Linking $@...
-	$(CXX) $(ARCHFLAGS) $(CXXFLAGS) $(LDFLAGS) -o $@ mandelbrot.o mandelbrot_ui.o $(LINKFLTK) $(LDLIBS)
+	$(CXX) $(ARCHFLAGS) $(CXXFLAGS) $(LDFLAGS) -o $@ mandelbrot.o mandelbrot_ui.o $(LINKSHARED) $(LDLIBS)
 	$(OSX_ONLY) ../fltk-config --post $@
 mandelbrot_ui.o:	mandelbrot_ui.h
 mandelbrot_ui.cxx:	mandelbrot_ui.fl ../fluid/fluid$(EXEEXT)
@@ -423,7 +423,7 @@ minimum$(EXEEXT): minimum.o
 
 native-filechooser$(EXEEXT): native-filechooser.o $(IMGLIBNAME)
 	echo Linking $@...
-	$(CXX) $(ARCHFLAGS) $(CXXFLAGS) $(LDFLAGS) native-filechooser.o -o $@ $(LINKFLTKIMG) $(LDLIBS)
+	$(CXX) $(ARCHFLAGS) $(CXXFLAGS) $(LDFLAGS) native-filechooser.o -o $@ $(LINKSHARED) $(LDLIBS)
 	$(OSX_ONLY) ../fltk-config --post $@
 
 navigation$(EXEEXT): navigation.o
@@ -443,7 +443,7 @@ pixmap$(EXEEXT): pixmap.o
 
 pixmap_browser$(EXEEXT): pixmap_browser.o $(IMGLIBNAME)
 	echo Linking $@...
-	$(CXX) $(ARCHFLAGS) $(CXXFLAGS) $(LDFLAGS) pixmap_browser.o -o $@ $(LINKFLTKIMG) $(LDLIBS)
+	$(CXX) $(ARCHFLAGS) $(CXXFLAGS) $(LDFLAGS) pixmap_browser.o -o $@ $(LINKSHARED) $(LDLIBS)
 	$(OSX_ONLY) ../fltk-config --post $@
 
 preferences$(EXEEXT):	preferences.o
@@ -467,7 +467,7 @@ subwindow$(EXEEXT): subwindow.o
 
 sudoku: sudoku.o
 	echo Linking $@...
-	$(CXX) $(ARCHFLAGS) $(CXXFLAGS) $(LDFLAGS) sudoku.o -o $@ $(AUDIOLIBS) $(LINKFLTKIMG) $(LDLIBS)
+	$(CXX) $(ARCHFLAGS) $(CXXFLAGS) $(LDFLAGS) sudoku.o -o $@ $(AUDIOLIBS) $(LINKSHARED) $(LDLIBS)
 	$(OSX_ONLY) $(RM) -f -r sudoku.app
 	$(OSX_ONLY) mkdir -p sudoku.app/Contents/MacOS sudoku.app/Contents/Resources
 	$(OSX_ONLY) $(INSTALL_BIN) sudoku$(EXEEXT) sudoku.app/Contents/MacOS
@@ -477,7 +477,7 @@ sudoku: sudoku.o
 sudoku.exe: sudoku.o sudoku.rc
 	echo Linking $@...
 	$(RC) sudoku.rc sudokures.o
-	$(CXX) $(ARCHFLAGS) $(CXXFLAGS) $(LDFLAGS) sudoku.o sudokures.o -o $@ $(AUDIOLIBS) $(LINKFLTKIMG) $(LDLIBS)
+	$(CXX) $(ARCHFLAGS) $(CXXFLAGS) $(LDFLAGS) sudoku.o sudokures.o -o $@ $(AUDIOLIBS) $(LINKSHARED) $(LDLIBS)
 
 symbols$(EXEEXT): symbols.o
 
@@ -511,7 +511,7 @@ CubeView$(EXEEXT): CubeMain.o CubeView.o CubeViewUI.o
 	echo Linking $@...
 	$(CXX) $(ARCHFLAGS) $(CXXFLAGS) $(LDFLAGS) -o $@ \
 		CubeMain.o CubeView.o CubeViewUI.o \
-		$(LINKFLTKGL) $(LINKFLTK) $(GLDLIBS)
+		$(LINKFLTKGL) $(LINKSHARED) $(GLDLIBS)
 	$(OSX_ONLY) ../fltk-config --post $@
 CubeMain.o: CubeViewUI.h CubeView.h CubeViewUI.cxx
 CubeView.o: CubeView.h
@@ -520,35 +520,35 @@ CubeViewUI.cxx:	CubeViewUI.fl ../fluid/fluid$(EXEEXT)
 
 cube$(EXEEXT): cube.o
 	echo Linking $@...
-	$(CXX) $(ARCHFLAGS) $(CXXFLAGS) $(LDFLAGS) -o $@ cube.o $(LINKFLTKGL) $(LINKFLTK) $(GLDLIBS)
+	$(CXX) $(ARCHFLAGS) $(CXXFLAGS) $(LDFLAGS) -o $@ cube.o $(LINKFLTKGL) $(LINKSHARED) $(GLDLIBS)
 	$(OSX_ONLY) ../fltk-config --post $@
 
 fractals$(EXEEXT): fractals.o fracviewer.o
 	echo Linking $@...
-	$(CXX) $(ARCHFLAGS) $(CXXFLAGS) $(LDFLAGS) -o $@ fractals.o fracviewer.o $(LINKFLTKGL) $(LINKFLTK) $(GLDLIBS)
+	$(CXX) $(ARCHFLAGS) $(CXXFLAGS) $(LDFLAGS) -o $@ fractals.o fracviewer.o $(LINKFLTKGL) $(LINKSHARED) $(GLDLIBS)
 	$(OSX_ONLY) ../fltk-config --post $@
 
 fullscreen$(EXEEXT): fullscreen.o
 	echo Linking $@...
-	$(CXX) $(ARCHFLAGS) $(CXXFLAGS) $(LDFLAGS) -o $@ fullscreen.o $(LINKFLTKGL) $(LINKFLTK) $(GLDLIBS)
+	$(CXX) $(ARCHFLAGS) $(CXXFLAGS) $(LDFLAGS) -o $@ fullscreen.o $(LINKFLTKGL) $(LINKSHARED) $(GLDLIBS)
 	$(OSX_ONLY) ../fltk-config --post $@
 
 glpuzzle$(EXEEXT): glpuzzle.o
 	echo Linking $@...
-	$(CXX) $(ARCHFLAGS) $(CXXFLAGS) $(LDFLAGS) -o $@ glpuzzle.o $(LINKFLTKGL) $(LINKFLTK) $(GLDLIBS)
+	$(CXX) $(ARCHFLAGS) $(CXXFLAGS) $(LDFLAGS) -o $@ glpuzzle.o $(LINKFLTKGL) $(LINKSHARED) $(GLDLIBS)
 	$(OSX_ONLY) ../fltk-config --post $@
 
 gl_overlay$(EXEEXT): gl_overlay.o
 	echo Linking $@...
-	$(CXX) $(ARCHFLAGS) $(CXXFLAGS) $(LDFLAGS) -o $@ gl_overlay.o $(LINKFLTKGL) $(LINKFLTK) $(GLDLIBS)
+	$(CXX) $(ARCHFLAGS) $(CXXFLAGS) $(LDFLAGS) -o $@ gl_overlay.o $(LINKFLTKGL) $(LINKSHARED) $(GLDLIBS)
 	$(OSX_ONLY) ../fltk-config --post $@
 
 shape$(EXEEXT): shape.o
 	echo Linking $@...
-	$(CXX) $(ARCHFLAGS) $(CXXFLAGS) $(LDFLAGS) -o $@ shape.o $(LINKFLTKGL) $(LINKFLTK) $(GLDLIBS)
+	$(CXX) $(ARCHFLAGS) $(CXXFLAGS) $(LDFLAGS) -o $@ shape.o $(LINKFLTKGL) $(LINKSHARED) $(GLDLIBS)
 	$(OSX_ONLY) ../fltk-config --post $@
 
 cairo_test$(EXEEXT): cairo_test.o
 	echo Linking $@...
-	$(CXX) $(ARCHFLAGS) $(CXXFLAGS) $(CAIROFLAGS) $(LDFLAGS) -o $@ cairo_test.o $(LINKFLTK) $(LINKFLTKCAIRO) $(GLDLIBS)
+	$(CXX) $(ARCHFLAGS) $(CXXFLAGS) $(CAIROFLAGS) $(LDFLAGS) -o $@ cairo_test.o $(LINKSHARED) $(LINKFLTKCAIRO) $(GLDLIBS)
 	$(OSX_ONLY) ../fltk-config --post $@
-- 
2.35.1

