From 67955d96a74dc117171f4b6c9623bb833ea59903 Mon Sep 17 00:00:00 2001
From: Thomas Staudinger <Staudi.Kaos@gmail.com>
Date: Wed, 4 Jan 2023 01:51:49 +0100
Subject: [PATCH] Enable stateless config

---
 meson.build    | 2 +-
 src/config.cpp | 1 +
 2 files changed, 2 insertions(+), 1 deletion(-)

diff --git a/meson.build b/meson.build
index d6b9e17..59c891f 100644
--- a/meson.build
+++ b/meson.build
@@ -355,7 +355,7 @@ executable(
 install_data(
     './resources/config',
     './resources/style.css',
-    install_dir: sysconfdir + '/xdg/waybar'
+    install_dir: '/usr/share/defaults/waybar'
 )
 
 scdoc = dependency('scdoc', version: '>=1.9.2', native: true, required: get_option('man-pages'))
diff --git a/src/config.cpp b/src/config.cpp
index a4d9090..0d4c2a2 100644
--- a/src/config.cpp
+++ b/src/config.cpp
@@ -17,6 +17,7 @@ namespace waybar {
 const std::vector<std::string> Config::CONFIG_DIRS = {
     "$XDG_CONFIG_HOME/waybar/", "$HOME/.config/waybar/",   "$HOME/waybar/",
     "/etc/xdg/waybar/",         SYSCONFDIR "/xdg/waybar/", "./resources/",
+    "/usr/share/defaults/waybar/"
 };
 
 const char *Config::CONFIG_PATH_ENV = "WAYBAR_CONFIG_DIR";
-- 
2.39.0

