diff --git a/data/org.gnome.Four-in-a-row.appdata.xml.in b/data/org.gnome.Four-in-a-row.appdata.xml.in
index 1c820ba..5fc6869 100644
--- a/data/org.gnome.Four-in-a-row.appdata.xml.in
+++ b/data/org.gnome.Four-in-a-row.appdata.xml.in
@@ -20,49 +20,32 @@
   </description>
   <screenshots>
     <screenshot height="447" width="700" type="default">
-      <image>https://people.gnome.org/~mcatanzaro/four-in-a-row.png</image>
+      <image>https://static.gnome.org/appdata/mcatanzaro-old/four-in-a-row.png</image>
     </screenshot>
   </screenshots>
   <provides>
     <id>four-in-a-row.desktop</id>
   </provides>
   <launchable type="desktop-id">org.gnome.Four-in-a-row.desktop</launchable>
   <url type="homepage">https://wiki.gnome.org/Apps/Four-in-a-row</url>
   <url type="bugtracker">https://gitlab.gnome.org/GNOME/four-in-a-row/issues</url>
-  <url type="translate">https://wiki.gnome.org/TranslationProject</url>
-  <url type="donation">https://www.gnome.org/friends/</url>
+  <url type="translate">https://l10n.gnome.org/module/four-in-a-row/</url>
+  <url type="donation">https://www.gnome.org/donate/</url>
   <url type="help">https://help.gnome.org/users/four-in-a-row/stable/</url>
+  <url type="vcs-browser">https://gitlab.gnome.org/GNOME/four-in-a-row</url>
   <update_contact>mcatanzaro_at_gnome.org</update_contact>
   <project_group>GNOME</project_group>
   <developer_name>The GNOME Project</developer_name>
+  <developer id="org.gnome">
+    <name>The GNOME Project</name>
+  </developer>
   <kudos>
     <kudo>HiDpiIcon</kudo>
     <kudo>ModernToolkit</kudo>
     <kudo>UserDocs</kudo>
   </kudos>
   <translation type="gettext">four-in-a-row</translation>
-  <content_rating type="oars-1.0">
-    <content_attribute id="violence-cartoon">none</content_attribute>
-    <content_attribute id="violence-fantasy">none</content_attribute>
-    <content_attribute id="violence-realistic">none</content_attribute>
-    <content_attribute id="violence-bloodshed">none</content_attribute>
-    <content_attribute id="violence-sexual">none</content_attribute>
-    <content_attribute id="drugs-alcohol">none</content_attribute>
-    <content_attribute id="drugs-narcotics">none</content_attribute>
-    <content_attribute id="drugs-tobacco">none</content_attribute>
-    <content_attribute id="sex-nudity">none</content_attribute>
-    <content_attribute id="sex-themes">none</content_attribute>
-    <content_attribute id="language-profanity">none</content_attribute>
-    <content_attribute id="language-humor">none</content_attribute>
-    <content_attribute id="language-discrimination">none</content_attribute>
-    <content_attribute id="social-chat">none</content_attribute>
-    <content_attribute id="social-info">none</content_attribute>
-    <content_attribute id="social-audio">none</content_attribute>
-    <content_attribute id="social-location">none</content_attribute>
-    <content_attribute id="social-contacts">none</content_attribute>
-    <content_attribute id="money-purchasing">none</content_attribute>
-    <content_attribute id="money-gambling">none</content_attribute>
-  </content_rating>
+  <content_rating type="oars-1.1" />
 
   ​<releases>
     <!-- should be kept in versioning inverted order -->
