diff --git a/src/four-in-a-row.vala b/src/four-in-a-row.vala
index cd527f0af464557033cd0f43b84dbaca298d5bd4..dca251acac6b087f29711ce833d96e718a2e218e 100644
--- a/src/four-in-a-row.vala
+++ b/src/four-in-a-row.vala
@@ -140,6 +140,7 @@ private class FourInARow : Gtk.Application
         Intl.textdomain (GETTEXT_PACKAGE);
 
         Environment.set_application_name (PROGRAM_NAME);
+        Environment.set_prgname ("org.gnome.Four-in-a-row");
         Window.set_default_icon_name ("org.gnome.Four-in-a-row");
 
         return new FourInARow ().run (args);
