From be0c6654230aa7581b89477f726ad82d226d712e Mon Sep 17 00:00:00 2001
From: Peter O'Connor <peter@solus-project.com>
Date: Sat, 14 Apr 2018 20:04:12 +1000
Subject: [PATCH] Disable building unshipped code

Signed-off-by: Peter O'Connor <peter@solus-project.com>
---
 CMakeLists.txt | 2 --
 1 file changed, 2 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 16e6514..d83a5f1 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -47,14 +47,12 @@ endif()
 
 add_subdirectory(src)
 if (BUILD_TESTING)
-     add_subdirectory(autotests)
 endif()
 if (NOT CMAKE_CROSSCOMPILING)
 	add_subdirectory(data)
 endif()
 
 if(TARGET Qt5::Widgets)
-    add_subdirectory(examples)
 endif()
 
 
-- 
2.16.3

