From 850e11d3fa71e7f5b3ca4c23d8bc34312c956dbb Mon Sep 17 00:00:00 2001
From: "flinux@vongellhorn.ch" <flinux@vongellhorn.ch>
Date: Sun, 13 Oct 2019 19:37:07 +0200
Subject: [PATCH 1/1] Enable spell check by default

---
 src/core/settings.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/core/settings.cpp b/src/core/settings.cpp
index 698a946..8e9e927 100644
--- a/src/core/settings.cpp
+++ b/src/core/settings.cpp
@@ -297,7 +297,7 @@ void Settings::restore()
     d->backgroundCheckerEnabled
         = settings.value(QStringLiteral("backgroundCheckerEnabled"), true).toBool();
     d->checkerEnabledByDefault
-        = settings.value(QStringLiteral("checkerEnabledByDefault"), false).toBool();
+        = settings.value(QStringLiteral("checkerEnabledByDefault"), true).toBool();
     d->disablePercentage
         = settings.value(QStringLiteral("Sonnet_AsYouTypeDisablePercentage"), 90).toInt();
     d->disableWordCount
-- 
2.23.0

