From 338602f1a6b2419262b7efe8df4f3b4c87ee7df9 Mon Sep 17 00:00:00 2001
From: Peter O'Connor <peter@solus-project.com>
Date: Thu, 2 Nov 2017 13:57:49 +1100
Subject: [PATCH] Enable spell check by default

Signed-off-by: Peter O'Connor <peter@solus-project.com>
---
 src/core/settings.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/core/settings.cpp b/src/core/settings.cpp
index 8614e35..8c9a021 100644
--- a/src/core/settings.cpp
+++ b/src/core/settings.cpp
@@ -279,7 +279,7 @@ void Settings::restore()
     d->checkUppercase = settings.value(QStringLiteral("checkUppercase"), true).toBool();
     d->skipRunTogether = settings.value(QStringLiteral("skipRunTogether"), true).toBool();
     d->backgroundCheckerEnabled = settings.value(QStringLiteral("backgroundCheckerEnabled"), true).toBool();
-    d->checkerEnabledByDefault = settings.value(QStringLiteral("checkerEnabledByDefault"), false).toBool();
+    d->checkerEnabledByDefault = settings.value(QStringLiteral("checkerEnabledByDefault"), true).toBool();
     d->disablePercentage = settings.value(QStringLiteral("Sonnet_AsYouTypeDisablePercentage"), 90).toInt();
     d->disableWordCount = settings.value(QStringLiteral("Sonnet_AsYouTypeDisableWordCount"), 100).toInt();
     d->autodetectLanguage = settings.value(QStringLiteral("autodetectLanguage"), true).toBool();
--
2.14.3
