From 6b5903d484277aae9a845b76bae6fbd668a964ec Mon Sep 17 00:00:00 2001
From: Peter O'Connor <peter@solus-project.com>
Date: Sun, 14 Jan 2018 09:53:11 +1100
Subject: [PATCH] Enable spell check by default

Signed-off-by: Peter O'Connor <peter@solus-project.com>
---
 src/core/settings.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/core/settings.cpp b/src/core/settings.cpp
index 4793cd1..d2d125e 100644
--- a/src/core/settings.cpp
+++ b/src/core/settings.cpp
@@ -279,7 +279,7 @@ void Settings::restore()
     d->backgroundCheckerEnabled
         = settings.value(QStringLiteral("backgroundCheckerEnabled"), true).toBool();
     d->checkerEnabledByDefault
-        = settings.value(QStringLiteral("checkerEnabledByDefault"), false).toBool();
+        = settings.value(QStringLiteral("checkerEnabledByDefault"), true).toBool();
     d->disablePercentage
         = settings.value(QStringLiteral("Sonnet_AsYouTypeDisablePercentage"), 90).toInt();
     d->disableWordCount
-- 
2.15.1

