--- Makefile	2016-08-13 09:58:21.000000000 -0400
+++ /home/cfielder/project/git/moreutils/Makefile	2017-07-16 05:59:38.961360030 -0400
@@ -1,6 +1,6 @@
-BINS=isutf8 ifdata ifne pee sponge mispipe lckdo parallel errno
+BINS=isutf8 ifdata ifne pee sponge mispipe lckdo errno
 PERLSCRIPTS=vidir vipe ts combine zrun chronic
-MANS=sponge.1 vidir.1 vipe.1 isutf8.1 ts.1 combine.1 ifdata.1 ifne.1 pee.1 zrun.1 chronic.1 mispipe.1 lckdo.1 parallel.1 errno.1
+MANS=sponge.1 vidir.1 vipe.1 isutf8.1 combine.1 ifdata.1 ifne.1 pee.1 zrun.1 chronic.1 mispipe.1 lckdo.1 errno.1
 CFLAGS?=-O2 -g -Wall
 INSTALL_BIN?=install -s
 PREFIX?=/usr
@@ -8,7 +8,7 @@
 ifneq (,$(findstring CYGWIN,$(shell uname)))
 	DOCBOOKXSL?=/usr/share/sgml/docbook/xsl-stylesheets
 else
-	DOCBOOKXSL?=/usr/share/xml/docbook/stylesheet/docbook-xsl
+	DOCBOOKXSL?=/usr/share/xml/docbook/xsl-stylesheets-1.79.2
 endif
 
 DOCBOOK2XMAN=xsltproc --param man.authors.section.enabled 0 $(DOCBOOKXSL)/manpages/docbook.xsl
