From 970c02329c7b322dfff878a41ab9e830ebb67427 Mon Sep 17 00:00:00 2001
From: Fabio Forni <livingsilver94.solus@redaril.me>
Date: Fri, 20 Aug 2021 22:59:01 +0200
Subject: [PATCH] Use AyatanaAppIndicator

---
 safeeyes/plugins/trayicon/plugin.py | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/safeeyes/plugins/trayicon/plugin.py b/safeeyes/plugins/trayicon/plugin.py
index 1461133..64b9191 100644
--- a/safeeyes/plugins/trayicon/plugin.py
+++ b/safeeyes/plugins/trayicon/plugin.py
@@ -19,8 +19,8 @@
 import datetime
 import gi
 gi.require_version('Gtk', '3.0')
-gi.require_version('AppIndicator3', '0.1')
-from gi.repository import AppIndicator3 as appindicator
+gi.require_version('AyatanaAppIndicator3', '0.1')
+from gi.repository import AyatanaAppIndicator3 as appindicator
 from gi.repository import Gtk
 import logging
 from safeeyes import utility
-- 
2.32.0

