--- src/i18n.h~	2025-04-19 14:51:18.000000000 -0500
+++ src/i18n.h	2026-03-18 15:58:56.025290275 -0500
@@ -185,7 +185,8 @@
 extern const char *lang_prefixes[NUM_LANGS];
 extern int need_own_font;
 extern int need_right_to_left;  // Right-justify
-extern const char *lang_prefix, *short_lang_prefix;
+extern const char *lang_prefix;
+extern char *short_lang_prefix;
 
 typedef struct w_langs
 {
@@ -194,7 +195,7 @@
   int need_right_to_left;
   int lang_y_nudge;
   const char *lang_prefix;
-  const char *short_lang_prefix;
+  char *short_lang_prefix;
 } w_langs;
 
 extern w_langs wished_langs[255];
--- src/i18n.c~	2025-04-19 14:51:18.000000000 -0500
+++ src/i18n.c	2026-03-18 16:00:02.676082411 -0500
@@ -256,7 +256,8 @@
 
 int need_own_font;
 int need_right_to_left;
-const char *lang_prefix, *short_lang_prefix;
+const char *lang_prefix;
+char *short_lang_prefix;
 
 w_langs wished_langs[255];
 
