--- magic/src/blind.c~	2024-10-24 01:30:40.000000000 -0500
+++ magic/src/blind.c	2025-02-03 12:47:38.300767632 -0600
@@ -27,6 +27,9 @@
   Last updated: October 7, 2024
 */
 
+#include <stdio.h>
+#include <string.h>
+
 #include "tp_magic_api.h"
 #include "SDL_image.h"
 #include "SDL_mixer.h"
--- magic/src/checkerboard.c~	2024-10-24 01:30:40.000000000 -0500
+++ magic/src/checkerboard.c	2025-02-03 12:51:44.802029995 -0600
@@ -28,6 +28,9 @@
   Last updated: October 7, 2024
 */
 
+#include <stdio.h>
+#include <string.h>
+
 #include "tp_magic_api.h"
 #include "SDL_image.h"
 #include "SDL_mixer.h"
--- magic/src/confetti.c~	2024-10-24 01:30:40.000000000 -0500
+++ magic/src/confetti.c	2025-02-03 12:58:37.819739152 -0600
@@ -3,6 +3,8 @@
 */
 
 #include <time.h>               //For time()
+#include <stdio.h>
+#include <string.h>
 
 #include "tp_magic_api.h"
 #include "SDL_image.h"
--- magic/src/crescent.c~	2024-10-24 01:30:40.000000000 -0500
+++ magic/src/crescent.c	2025-02-03 13:02:56.063501372 -0600
@@ -27,6 +27,7 @@
 
 #include <stdio.h>
 #include <string.h>
+#include <math.h>
 #include "tp_magic_api.h"
 #include "SDL_image.h"
 #include "SDL_mixer.h"
--- magic/src/dither.c~	2024-10-24 01:30:40.000000000 -0500
+++ magic/src/dither.c	2025-02-03 13:07:20.163212839 -0600
@@ -15,6 +15,7 @@
 #include <stdio.h>
 #include <string.h>
 #include <libintl.h>
+#include <math.h>
 
 #include "tp_magic_api.h"
 #include "SDL_image.h"
--- magic/src/fisheye.c~	2024-10-24 01:30:40.000000000 -0500
+++ magic/src/fisheye.c	2025-02-03 13:11:30.913040007 -0600
@@ -29,6 +29,8 @@
 */
 
 #include <math.h>
+#include <stdio.h>
+#include <string.h>
 #include "tp_magic_api.h"
 #include "SDL_image.h"
 #include "SDL_mixer.h"
--- magic/src/fade_darken.c~	2024-10-24 01:30:40.000000000 -0500
+++ magic/src/fade_darken.c	2025-02-03 13:14:53.862667789 -0600
@@ -28,6 +28,7 @@
 
 #include <stdio.h>
 #include <string.h>
+#include <math.h>
 #include "tp_magic_api.h"
 #include "SDL_image.h"
 #include "SDL_mixer.h"
--- magic/src/fractal.c~	2024-10-24 01:30:40.000000000 -0500
+++ magic/src/fractal.c	2025-02-03 13:20:23.299560628 -0600
@@ -27,6 +27,7 @@
 
 #include <stdio.h>
 #include <string.h>
+#include <math.h>
 #include "tp_magic_api.h"
 #include "SDL_image.h"
 #include "SDL_mixer.h"
--- magic/src/fold.c~	2024-10-24 01:30:40.000000000 -0500
+++ magic/src/fold.c	2025-02-03 13:24:02.358159425 -0600
@@ -11,6 +11,9 @@
 
 //sound playing needs fixing.
 
+#include <stdio.h>
+#include <string.h>
+
 #include "tp_magic_api.h"
 #include "SDL_image.h"
 #include "SDL_mixer.h"
--- magic/src/fretwork.c~	2024-10-24 01:30:40.000000000 -0500
+++ magic/src/fretwork.c	2025-02-03 13:28:01.947626904 -0600
@@ -8,6 +8,8 @@
 #include "SDL_image.h"
 #include "SDL_mixer.h"
 #include <stdbool.h>
+#include <stdio.h>
+#include <string.h>
 
 #define SEG_NONE 0
 
--- magic/src/n_pt_persp.c~	2024-10-24 01:30:40.000000000 -0500
+++ magic/src/n_pt_persp.c	2025-02-03 13:31:54.148228724 -0600
@@ -23,6 +23,7 @@
 #include <stdio.h>
 #include <string.h>
 #include <libintl.h>
+#include <math.h>
 
 #include "tp_magic_api.h"
 #include "SDL_image.h"
--- magic/src/n_pt_persp.c~	2025-02-03 13:36:48.074512096 -0600
+++ magic/src/n_pt_persp.c	2025-02-03 13:37:33.733245435 -0600
@@ -24,6 +24,7 @@
 #include <string.h>
 #include <libintl.h>
 #include <math.h>
+#include <float.h>
 
 #include "tp_magic_api.h"
 #include "SDL_image.h"
--- magic/src/pixels.c~	2024-10-24 01:30:40.000000000 -0500
+++ magic/src/pixels.c	2025-02-03 13:41:42.889790279 -0600
@@ -30,6 +30,7 @@
 
 #include <stdio.h>
 #include <string.h>
+#include <math.h>
 #include "tp_magic_api.h"
 #include "SDL_image.h"
 #include "SDL_mixer.h"
--- magic/src/puzzle.c~	2024-10-24 01:30:40.000000000 -0500
+++ magic/src/puzzle.c	2025-02-03 13:45:34.040440279 -0600
@@ -30,6 +30,8 @@
 
 
 #include <time.h>               //for time()
+#include <stdio.h>
+#include <string.h>
 #include "tp_magic_api.h"
 #include "SDL_image.h"
 #include "SDL_mixer.h"
--- magic/src/rails.c~	2024-10-24 01:30:40.000000000 -0500
+++ magic/src/rails.c	2025-02-03 13:49:13.217198352 -0600
@@ -9,6 +9,9 @@
 #include "SDL_image.h"
 #include "SDL_mixer.h"
 
+#include <stdio.h>
+#include <string.h>
+
 #define SEG_NONE 0
 
 #define SEG_LEFT 1
--- magic/src/ribbon.c~	2024-10-24 01:30:40.000000000 -0500
+++ magic/src/ribbon.c	2025-02-03 13:57:35.276389397 -0600
@@ -28,6 +28,7 @@
 
 #include <stdio.h>
 #include <string.h>
+#include <math.h>
 #include "tp_magic_api.h"
 #include "SDL_image.h"
 #include "SDL_mixer.h"
--- magic/src/rosette.c~	2024-10-24 01:30:40.000000000 -0500
+++ magic/src/rosette.c	2025-02-03 14:01:01.504235593 -0600
@@ -34,6 +34,8 @@
 #include "tp_magic_api.h"
 #include "SDL_image.h"
 #include "SDL_mixer.h"
+#include <stdio.h>
+#include <string.h>
 #include <math.h>               //for sin, cos, ...
 
 static int ROSETTE_R = 8;       //circle's diameter
--- magic/src/spiral.c~	2024-10-24 01:30:40.000000000 -0500
+++ magic/src/spiral.c	2025-02-03 14:05:52.819585681 -0600
@@ -29,6 +29,7 @@
 
 #include <stdio.h>
 #include <string.h>
+#include <math.h>
 #include "tp_magic_api.h"
 #include "SDL_image.h"
 #include "SDL_mixer.h"
--- magic/src/stretch.c~	2024-10-24 01:30:40.000000000 -0500
+++ magic/src/stretch.c	2025-02-03 14:10:08.329114863 -0600
@@ -28,6 +28,9 @@
   Last updated: October 7, 2024
 */
 
+#include <stdio.h>
+#include <string.h>
+
 #include "tp_magic_api.h"
 #include "SDL_image.h"
 #include "SDL_mixer.h"
--- magic/src/spraypaint.c~	2024-10-24 01:30:40.000000000 -0500
+++ magic/src/spraypaint.c	2025-02-03 14:14:31.527599794 -0600
@@ -27,6 +27,7 @@
 
 #include <stdio.h>
 #include <string.h>
+#include <math.h>
 #include "tp_magic_api.h"
 #include "SDL_image.h"
 #include "SDL_mixer.h"
--- magic/src/string.c~	2024-10-24 01:30:40.000000000 -0500
+++ magic/src/string.c	2025-02-03 14:18:09.575344630 -0600
@@ -7,6 +7,9 @@
 #include "SDL_image.h"
 #include "SDL_mixer.h"
 
+#include <stdio.h>
+#include <string.h>
+
 unsigned int img_w, img_h;
 static Uint8 string_r, string_g, string_b;
 static int string_ox, string_oy;
--- magic/src/tv.c~	2024-10-24 01:30:40.000000000 -0500
+++ magic/src/tv.c	2025-02-03 14:22:32.168833011 -0600
@@ -33,6 +33,8 @@
 #include "SDL_image.h"
 #include "SDL_mixer.h"
 #include <math.h>
+#include <stdio.h>
+#include <string.h>
 
 static int tv_radius = 16;
 
--- magic/src/trochoids.c~	2024-10-24 01:30:40.000000000 -0500
+++ magic/src/trochoids.c	2025-02-03 15:19:48.988517684 -0600
@@ -13,6 +13,7 @@
 #include <stdio.h>
 #include <string.h>
 #include <libintl.h>
+#include <math.h>
 
 #include "tp_magic_api.h"
 #include "SDL_image.h"
--- src/fill.c~	2024-10-24 01:30:40.000000000 -0500
+++ src/fill.c	2025-02-03 15:40:27.391554949 -0600
@@ -33,6 +33,7 @@
 
 #include <stdio.h>
 #include <string.h>
+#include <stdlib.h>
 
 /* math.h makes y1 an obscure function! */
 #define y1 evil_y1
--- magic/src/bloom.c~	2024-10-24 01:30:40.000000000 -0500
+++ magic/src/bloom.c	2025-02-03 15:53:37.201114240 -0600
@@ -6,6 +6,7 @@
    Last updated: January 16, 2024
 */
 
+#include <stdlib.h>
 #include <stdio.h>
 #include <string.h>
 #include <libintl.h>
--- magic/src/alien.c~	2024-10-24 01:30:40.000000000 -0500
+++ magic/src/alien.c	2025-02-03 15:54:20.541870590 -0600
@@ -28,6 +28,7 @@
   Last updated: October 7, 2024
 */
 
+#include <stdlib.h>
 #include <stdio.h>
 #include <string.h>
 #include <libintl.h>
--- magic/src/ascii.c~	2024-10-24 01:30:40.000000000 -0500
+++ magic/src/ascii.c	2025-02-03 15:54:53.680684330 -0600
@@ -33,6 +33,7 @@
 #define DEBUG_PRINTF(...)
 #endif
 
+#include <stdlib.h>
 #include <stdio.h>
 #include <string.h>
 #include "tp_magic_api.h"
--- magic/src/checkerboard.c~	2025-02-03 16:04:10.465553765 -0600
+++ magic/src/checkerboard.c	2025-02-03 16:04:40.656395075 -0600
@@ -28,6 +28,7 @@
   Last updated: October 7, 2024
 */
 
+#include <stdlib.h>
 #include <stdio.h>
 #include <string.h>
 
--- magic/src/confetti.c~	2025-02-03 16:11:57.553912535 -0600
+++ magic/src/confetti.c	2025-02-03 16:12:29.350804638 -0600
@@ -5,6 +5,7 @@
 #include <time.h>               //For time()
 #include <stdio.h>
 #include <string.h>
+#include <stdlib.h>
 
 #include "tp_magic_api.h"
 #include "SDL_image.h"
--- magic/src/dither.c~	2025-02-03 16:11:57.554912532 -0600
+++ magic/src/dither.c	2025-02-03 16:13:00.671698355 -0600
@@ -12,6 +12,7 @@
 /* Inclusion of header files */
 /* ---------------------------------------------------------------------- */
 
+#include <stdlib.h>
 #include <stdio.h>
 #include <string.h>
 #include <libintl.h>
--- magic/src/tv.c~	2025-02-03 16:24:00.931457868 -0600
+++ magic/src/tv.c	2025-02-03 16:25:04.786241189 -0600
@@ -35,6 +35,7 @@
 #include <math.h>
 #include <stdio.h>
 #include <string.h>
+#include <stdlib.h>
 
 static int tv_radius = 16;
 
--- magic/src/toothpaste.c~	2024-10-24 01:30:40.000000000 -0500
+++ magic/src/toothpaste.c	2025-02-03 16:25:33.245144617 -0600
@@ -31,6 +31,7 @@
 #include <stdio.h>
 #include <string.h>
 #include <libintl.h>
+#include <stdlib.h>
 #include "tp_magic_api.h"
 #include "SDL_image.h"
 #include "SDL_mixer.h"
--- magic/src/fretwork.c~	2025-02-03 16:33:02.489620127 -0600
+++ magic/src/fretwork.c	2025-02-03 16:33:32.110519611 -0600
@@ -10,6 +10,7 @@
 #include <stdbool.h>
 #include <stdio.h>
 #include <string.h>
+#include <stdlib.h>
 
 #define SEG_NONE 0
 
--- magic/src/googlyeyes.c~	2024-10-24 01:30:40.000000000 -0500
+++ magic/src/googlyeyes.c	2025-02-03 16:33:59.607426309 -0600
@@ -10,6 +10,7 @@
 #include <string.h>
 #include <libintl.h>
 #include <math.h>
+#include <stdlib.h>
 
 #include "tp_magic_api.h"
 #include "SDL_image.h"
--- magic/src/polyfill.c~	2024-10-24 01:30:40.000000000 -0500
+++ magic/src/polyfill.c	2025-02-03 16:42:48.416398591 -0600
@@ -14,6 +14,7 @@
 #include <stdio.h>
 #include <string.h>
 #include <libintl.h>
+#include <stdlib.h>
 
 #include "tp_magic_api.h"
 #include "SDL_image.h"
--- magic/src/puzzle.c~	2025-02-03 16:41:56.379627065 -0600
+++ magic/src/puzzle.c	2025-02-03 16:43:16.617274773 -0600
@@ -32,6 +32,7 @@
 #include <time.h>               //for time()
 #include <stdio.h>
 #include <string.h>
+#include <stdlib.h>
 #include "tp_magic_api.h"
 #include "SDL_image.h"
 #include "SDL_mixer.h"
--- magic/src/rails.c~	2025-02-03 16:41:56.379627065 -0600
+++ magic/src/rails.c	2025-02-03 16:43:36.502187467 -0600
@@ -11,6 +11,7 @@
 
 #include <stdio.h>
 #include <string.h>
+#include <stdlib.h>
 
 #define SEG_NONE 0
 
--- magic/src/smudge.c~	2024-10-24 01:30:40.000000000 -0500
+++ magic/src/smudge.c	2025-02-03 16:44:04.889062843 -0600
@@ -32,6 +32,7 @@
 
 #include <stdio.h>
 #include <string.h>
+#include <stdlib.h>
 #include "tp_magic_api.h"
 #include "SDL_image.h"
 #include "SDL_mixer.h"
--- magic/src/string.c~	2025-02-03 16:41:56.381627056 -0600
+++ magic/src/string.c	2025-02-03 16:44:33.394937690 -0600
@@ -9,6 +9,7 @@
 
 #include <stdio.h>
 #include <string.h>
+#include <stdlib.h>
 
 unsigned int img_w, img_h;
 static Uint8 string_r, string_g, string_b;
--- magic/src/tessell.c~	2024-10-24 01:30:40.000000000 -0500
+++ magic/src/tessell.c	2025-02-03 16:44:57.536831697 -0600
@@ -28,6 +28,7 @@
 
 #include <stdio.h>
 #include <string.h>
+#include <stdlib.h>
 #include "tp_magic_api.h"
 #include "SDL_image.h"
 #include "SDL_mixer.h"
--- magic/src/maze.c~	2024-10-24 01:30:40.000000000 -0500
+++ magic/src/maze.c	2025-02-03 16:45:18.120741333 -0600
@@ -9,6 +9,7 @@
 #include <string.h>
 #include <libintl.h>
 #include <math.h>
+#include <stdlib.h>
 
 #include "tp_magic_api.h"
 #include "SDL_image.h"
--- magic/src/kaleidox.c~	2024-10-24 01:30:40.000000000 -0500
+++ magic/src/kaleidox.c	2025-02-03 16:45:41.451638895 -0600
@@ -11,6 +11,7 @@
 #include <string.h>
 #include <libintl.h>
 #include <math.h>
+#include <stdlib.h>
 
 #include "tp_magic_api.h"
 #include "SDL_image.h"
--- magic/src/mosaic_shaped.c~	2024-10-24 01:30:40.000000000 -0500
+++ magic/src/mosaic_shaped.c	2025-02-03 16:46:12.631502015 -0600
@@ -42,7 +42,7 @@
 #include <math.h>
 #include <limits.h>
 #include <time.h>
-
+#include <stdlib.h>
 
 #ifndef gettext_noop
 #define gettext_noop(String) String
--- magic/src/n_pt_persp.c~	2025-02-03 16:41:56.378627069 -0600
+++ magic/src/n_pt_persp.c	2025-02-03 16:46:51.087333197 -0600
@@ -25,6 +25,7 @@
 #include <libintl.h>
 #include <math.h>
 #include <float.h>
+#include <stdlib.h>
 
 #include "tp_magic_api.h"
 #include "SDL_image.h"
--- magic/src/realrainbow.c~	2024-10-24 01:30:40.000000000 -0500
+++ magic/src/realrainbow.c	2025-02-03 16:47:20.956202051 -0600
@@ -17,6 +17,7 @@
 #include <stdio.h>
 #include <string.h>
 #include <math.h>
+#include <stdlib.h>
 #include "SDL_image.h"
 
 #include "tp_magic_api.h"
--- magic/src/mosaic.c~	2024-10-24 01:30:40.000000000 -0500
+++ magic/src/mosaic.c	2025-02-03 16:47:59.561032551 -0600
@@ -38,7 +38,7 @@
 #include <math.h>
 #include <limits.h>
 #include <time.h>
-
+#include <stdlib.h>
 
 #ifndef gettext_noop
 #define gettext_noop(String) String
--- magic/src/noise.c~	2024-10-24 01:30:40.000000000 -0500
+++ magic/src/noise.c	2025-02-03 16:48:33.874881890 -0600
@@ -37,6 +37,7 @@
 #include <math.h>
 #include <limits.h>
 #include <time.h>
+#include <stdlib.h>
 
 #ifndef gettext_noop
 #define gettext_noop(String) String
--- magic/src/spiral.c~	2025-02-03 16:41:56.380627060 -0600
+++ magic/src/spiral.c	2025-02-03 16:49:07.585733881 -0600
@@ -30,6 +30,7 @@
 #include <stdio.h>
 #include <string.h>
 #include <math.h>
+#include <stdlib.h>
 #include "tp_magic_api.h"
 #include "SDL_image.h"
 #include "SDL_mixer.h"
--- magic/src/rain.c~	2024-10-24 01:30:40.000000000 -0500
+++ magic/src/rain.c	2025-02-03 16:49:33.124621752 -0600
@@ -37,6 +37,7 @@
 #include <math.h>
 #include <limits.h>
 #include <time.h>
+#include <stdlib.h>
 
 #ifndef gettext_noop
 #define gettext_noop(String) String
--- magic/src/spraypaint.c~	2025-02-03 16:41:56.380627060 -0600
+++ magic/src/spraypaint.c	2025-02-03 16:50:01.294498066 -0600
@@ -28,6 +28,7 @@
 #include <stdio.h>
 #include <string.h>
 #include <math.h>
+#include <stdlib.h>
 #include "tp_magic_api.h"
 #include "SDL_image.h"
 #include "SDL_mixer.h"
--- magic/src/snow.c~	2024-10-24 01:30:40.000000000 -0500
+++ magic/src/snow.c	2025-02-03 16:50:26.275388393 -0600
@@ -43,6 +43,7 @@
 #include <math.h>
 #include <limits.h>
 #include <time.h>
+#include <stdlib.h>
 
 #ifndef gettext_noop
 #define gettext_noop(String) String
--- magic/src/trochoids.c~	2025-02-03 16:41:56.381627056 -0600
+++ magic/src/trochoids.c	2025-02-03 16:50:46.297300482 -0600
@@ -14,6 +14,7 @@
 #include <string.h>
 #include <libintl.h>
 #include <math.h>
+#include <stdlib.h>
 
 #include "tp_magic_api.h"
 #include "SDL_image.h"
--- magic/src/swirls.c~	2024-10-24 01:30:40.000000000 -0500
+++ magic/src/swirls.c	2025-02-03 16:51:11.730188816 -0600
@@ -14,6 +14,7 @@
 #include <string.h>
 #include <libintl.h>
 #include <math.h>
+#include <stdlib.h>
 
 #include "tp_magic_api.h"
 #include "SDL_image.h"
--- magic/src/lightning.c~	2024-10-24 01:30:40.000000000 -0500
+++ magic/src/lightning.c	2025-02-03 16:57:15.947567484 -0600
@@ -10,6 +10,7 @@
 #include <string.h>
 #include <libintl.h>
 #include <math.h>
+#include <stdlib.h>
 
 #include "tp_magic_api.h"
 #include "SDL_image.h"
