From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Reilly Brogan <reilly@reillybrogan.com>
Date: Thu, 1 Aug 2024 18:27:42 -0500
Subject: [PATCH] stateless config

---
 greetd/src/config/mod.rs | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/greetd/src/config/mod.rs b/greetd/src/config/mod.rs
index 2d1d645..2849501 100644
--- a/greetd/src/config/mod.rs
+++ b/greetd/src/config/mod.rs
@@ -238,7 +238,8 @@ pub fn read_config() -> Result<Config, Error> {
     let config_str = match matches.opt_str("config") {
         Some(v) => read_to_string(v),
         None => read_to_string("/etc/greetd/greetd.conf")
-            .or_else(|_| read_to_string("/etc/greetd/config.toml")),
+                    .or_else(|_| read_to_string("/etc/greetd/config.toml")
+                    .or_else(|_| read_to_string("/usr/share/defaults/greetd/config.toml")))
     }?;
     let mut file = parse_config(&config_str)?;
 
