From d781ae002accd91f660ed0d482148f9da551dbaf Mon Sep 17 00:00:00 2001
From: Joshua Strobl <joshua@stroblindustries.com>
Date: Fri, 17 May 2019 12:25:16 +0300
Subject: [PATCH 1/1] Force usage of sudo over runuser on Solus

---
 overlord/snapshotstate/backend/helpers.go | 13 +++----------
 1 file changed, 3 insertions(+), 10 deletions(-)

diff --git a/overlord/snapshotstate/backend/helpers.go b/overlord/snapshotstate/backend/helpers.go
index b0a9db53a..516e6c623 100644
--- a/overlord/snapshotstate/backend/helpers.go
+++ b/overlord/snapshotstate/backend/helpers.go
@@ -171,17 +171,10 @@ var (
 )
 
 func pickUserWrapper() string {
-	// runuser and sudo happen to work the same way in this case.  The main
-	// reason to prefer runuser over sudo is that runuser is part of
-	// util-linux, which is considered essential, whereas sudo is an addon
-	// which could be removed.  However util-linux < 2.23 does not have
-	// runuser, and we support some distros that ship things older than that
-	// (e.g. Ubuntu 14.04)
-	for _, cmd := range []string{"runuser", "sudo"} {
-		if lp, err := execLookPath(cmd); err == nil {
-			return lp
-		}
+	if lp, err := execLookPath("sudo"); err == nil {
+		return lp
 	}
+
 	return ""
 }
 
-- 
2.21.0

