From 492abc107a1a904e1e072c1604574294e6ec8685 Mon Sep 17 00:00:00 2001
From: Ikey Doherty <ikey@solus-project.com>
Date: Thu, 25 Jan 2018 12:15:47 +0000
Subject: [PATCH 2/4] interfaces/builtin: Fix typo in steam-support interface

Signed-off-by: Ikey Doherty <ikey@solus-project.com>
---
 interfaces/builtin/steam_support.go | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/interfaces/builtin/steam_support.go b/interfaces/builtin/steam_support.go
index 5d8c75fd0..312761bb7 100644
--- a/interfaces/builtin/steam_support.go
+++ b/interfaces/builtin/steam_support.go
@@ -183,7 +183,7 @@ ptrace (trace, read) peer=snap.@{SNAP_NAME}.**,
 # AppArmor Limitations #
 ########################
 
-# Games Dev Tycoon reads /proc/$ppid/environ but unfortunately AppArmor
+# Game Dev Tycoon reads /proc/$ppid/environ but unfortunately AppArmor
 # doesn't have any such qualifier.
 @{PROC}/[0-9]*/environ r,
 `
-- 
2.16.2

