From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Silke Hofstra <silke@slxh.eu>
Date: Wed, 12 Aug 2026 21:52:13 +0200
Subject: [PATCH 4/4] apparmor: Prefer network_v9 instead of network

---
 sandbox/apparmor/apparmor.go | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/sandbox/apparmor/apparmor.go b/sandbox/apparmor/apparmor.go
index 3797f2a5b7..a3c8ff2a43 100644
--- a/sandbox/apparmor/apparmor.go
+++ b/sandbox/apparmor/apparmor.go
@@ -547,7 +547,7 @@ var (
 		"file",
 		"mount",
 		"namespaces",
-		"network",
+		"network_v9",
 		"ptrace",
 		"signal",
 	}
