--- cmake/FindDocBookXSL.cmake	2019-09-13 14:59:01.000000000 +0200
+++ cmake/FindDocBookXSL.cmake	2019-10-19 23:22:08.644980325 +0200
@@ -5,6 +5,7 @@
     /usr/local/share/xsl/docbook # FreeBSD
     /usr/share/xml/docbook/stylesheet/nwalsh/current # OpenSUSE
     /usr/share/sgml/docbook/xsl-ns-stylesheets # Fedora
+    /usr/share/xml/docbook/xsl-stylesheets-1.79.2/ # Solus
     )
 find_package_handle_standard_args(DocBookXSL
     "Could NOT find DocBook XSL stylesheets" DOCBOOKXSL_DIR)
