--- pentobi/unix/CMakeLists.txt	2019-10-20 00:19:04.391604195 +0200
+++ pentobi/unix/CMakeLists.txt	2019-10-20 00:02:22.601017112 +0200
@@ -7,7 +7,7 @@
 get_filename_component(GETTEXT_BIN_DIR ${GETTEXT_MSGFMT_EXECUTABLE} DIRECTORY)
 get_filename_component(GETTEXT_INSTALL_DIR ${GETTEXT_BIN_DIR} DIRECTORY)
 find_file(METAINFO_ITS NAMES metainfo.its appdata.its
-    HINTS ${GETTEXT_INSTALL_DIR}/share/gettext/its
+    HINTS ${GETTEXT_INSTALL_DIR}/share/gettext-0.20/its
     )
 if(NOT METAINFO_ITS)
     message(FATAL_ERROR
