From 1c8fc3f30630dec1526979d190c280902712b99d Mon Sep 17 00:00:00 2001
From: Joshua Strobl <joshua@stroblindustries.com>
Date: Wed, 17 Apr 2019 12:50:33 +0300
Subject: [PATCH 1/1] Force Xorg use on the login screen

---
 data/gdm.conf-custom.in | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/data/gdm.conf-custom.in b/data/gdm.conf-custom.in
index 9b63ba95..4f220684 100644
--- a/data/gdm.conf-custom.in
+++ b/data/gdm.conf-custom.in
@@ -2,7 +2,7 @@
 
 [daemon]
 # Uncomment the line below to force the login screen to use Xorg
-#WaylandEnable=false
+WaylandEnable=false
 
 [security]
 
-- 
2.21.0

