From 611c9926a6507781dc050f18fcce8f37dddca1b9 Mon Sep 17 00:00:00 2001
From: Joshua Strobl <joshua@stroblindustries.com>
Date: Fri, 3 May 2019 22:38:13 +0300
Subject: [PATCH 1/1] Tweak threshold and add logging.

From https://gitlab.gnome.org/GNOME/gdm/commit/ee19a9749fb2c17f350ae596e3b55f1404adac39
---
 daemon/gdm-display.c | 9 +++++++--
 1 file changed, 7 insertions(+), 2 deletions(-)

diff --git a/daemon/gdm-display.c b/daemon/gdm-display.c
index a8747785..b3bcb44e 100644
--- a/daemon/gdm-display.c
+++ b/daemon/gdm-display.c
@@ -51,6 +51,8 @@
 
 #define GNOME_SESSION_SESSIONS_PATH DATADIR "/gnome-session/sessions"
 
+#define DISPLAY_FAIL_THRESHOLD 10
+
 typedef struct _GdmDisplayPrivate
 {
         GObject               parent;
@@ -696,10 +698,13 @@ gdm_display_unmanage (GdmDisplay *self)
         }
 
         elapsed = g_timer_elapsed (priv->server_timer, NULL);
-        if (elapsed < 3) {
-                g_warning ("GdmDisplay: display lasted %lf seconds", elapsed);
+        g_debug ("GdmDisplay: display lasted %lf seconds", elapsed);
+
+        if (elapsed < DISPLAY_FAIL_THRESHOLD) {
+                g_warning ("GdmDisplay: Less than the %d second threshold, failing", DISPLAY_FAIL_THRESHOLD);
                 _gdm_display_set_status (self, GDM_DISPLAY_FAILED);
         } else {
+                g_debug ("GdmDisplay: display lasted more than %d second threshold, unmanaging", DISPLAY_FAIL_THRESHOLD);
                 _gdm_display_set_status (self, GDM_DISPLAY_UNMANAGED);
         }
 
-- 
2.21.0

