From b11bb90f5ac33dd223e818e4d4b3b5f74914e327 Mon Sep 17 00:00:00 2001
From: Ikey Doherty <ikey@solus-project.com>
Date: Sun, 8 Nov 2015 21:45:41 +0000
Subject: [PATCH] Force Xorg use on the login screen

Signed-off-by: Ikey Doherty <ikey@solus-project.com>
---
 data/gdm.conf-custom.in | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/data/gdm.conf-custom.in b/data/gdm.conf-custom.in
index c9e2864..37d148d 100644
--- a/data/gdm.conf-custom.in
+++ b/data/gdm.conf-custom.in
@@ -2,7 +2,7 @@
 
 [daemon]
 # Uncoment the line below to force the login screen to use Xorg
-#WaylandEnable=false
+WaylandEnable=false
 
 [security]
 
-- 
2.6.3

