From 4262f77250312fac3182578c5ea12391986d26a6 Mon Sep 17 00:00:00 2001
From: Ikey Doherty <ikey@solus-project.com>
Date: Wed, 2 Dec 2015 03:15:27 +0000
Subject: [PATCH 3/3] Default to budgie-desktop session

Signed-off-by: Ikey Doherty <ikey@solus-project.com>
---
 data/gdm.conf-custom.in | 1 +
 1 file changed, 1 insertion(+)

diff --git a/data/gdm.conf-custom.in b/data/gdm.conf-custom.in
index 37d148d..0abed58 100644
--- a/data/gdm.conf-custom.in
+++ b/data/gdm.conf-custom.in
@@ -3,6 +3,7 @@
 [daemon]
 # Uncoment the line below to force the login screen to use Xorg
 WaylandEnable=false
+DefaultSession=budgie-desktop.desktop
 
 [security]
 
-- 
2.6.3

