From 0e31d869aa20e13b346af6ef6aa11cf72ba2e833 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Tomasz=20Mi=C4=85sko?= <tomasz.miasko@gmail.com>
Date: Wed, 6 Mar 2019 00:00:00 +0000
Subject: [PATCH] Remove erroneous NULL from session search directories

Fixes issue #470.
---
 libgdm/gdm-sessions.c | 2 --
 1 file changed, 2 deletions(-)

diff --git a/libgdm/gdm-sessions.c b/libgdm/gdm-sessions.c
index afbc5c09..46713db5 100644
--- a/libgdm/gdm-sessions.c
+++ b/libgdm/gdm-sessions.c
@@ -239,7 +239,6 @@ collect_sessions (void)
                 DMCONFDIR "/Sessions/",
                 DATADIR "/gdm/BuiltInSessions/",
                 DATADIR "/xsessions/",
-                NULL
         };
 
         xorg_search_array = g_array_new (TRUE, TRUE, sizeof (char *));
@@ -256,7 +255,6 @@ collect_sessions (void)
 #ifdef ENABLE_WAYLAND_SUPPORT
         const char *wayland_search_dirs[] = {
                 DATADIR "/wayland-sessions/",
-                NULL
         };
 
         wayland_search_array = g_array_new (TRUE, TRUE, sizeof (char *));
-- 
2.18.1

