From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Ivan Trepakov <liontiger23@gmail.com>
Date: Tue, 18 Nov 2025 12:15:22 +0700
Subject: [PATCH] Fix C23 build problems

---
 unix/configure | 1 +
 unix/unxcfg.h  | 2 +-
 2 files changed, 2 insertions(+), 1 deletion(-)

diff --git a/unix/configure b/unix/configure
index 785d8dd..2de909a 100755
--- a/unix/configure
+++ b/unix/configure
@@ -419,6 +419,7 @@ $CC $CFLAGS -c conftest.c >/dev/null 2>/dev/null
 
 echo Check for directory libraries
 cat > conftest.c << _EOF_
+#include <dirent.h>
 int main() { return closedir(opendir(".")); }
 _EOF_
 
diff --git a/unix/unxcfg.h b/unix/unxcfg.h
index e39b283..0d6643f 100644
--- a/unix/unxcfg.h
+++ b/unix/unxcfg.h
@@ -117,7 +117,7 @@ typedef struct stat z_stat;
 #  endif
 #else
 #  include <time.h>
-   struct tm *gmtime(), *localtime();
+   struct tm *gmtime(const time_t* timer), *localtime(const time_t* timer);
 #endif
 
 #if (defined(BSD4_4) || (defined(SYSV) && defined(MODERN)))
