From 02dc0a7f0f80b4cefa178535b2aaf0b25901c2b7 Mon Sep 17 00:00:00 2001
From: Joshua Strobl <joshua@streambits.io>
Date: Wed, 22 Jan 2020 14:44:22 +0200
Subject: [PATCH 1/1] Patch out keybinder support. Breaks Ctrl+F in other
 applications.

---
 alttoolbar_type.py | 14 --------------
 1 file changed, 14 deletions(-)

diff --git a/alttoolbar_type.py b/alttoolbar_type.py
index 00832c0..4299381 100644
--- a/alttoolbar_type.py
+++ b/alttoolbar_type.py
@@ -1392,20 +1392,6 @@ class AltToolbarHeaderBar(AltToolbarShared):
         # hook the key-press for the application window
         self.shell.props.window.connect("key-press-event", self._on_key_press)
 
-        try:
-            import gi
-            gi.require_version('Keybinder', '3.0')
-            from gi.repository import Keybinder
-
-            # bind the zoom keyboard shortcuts
-            Keybinder.init()
-            if Keybinder.supported():
-                Keybinder.bind("<Ctrl>f", self._on_search)
-        except ValueError:
-            # no need to worry if keybinder is not installed - just
-            # that the toggle does not work
-            return
-
     def add_always_visible_source(self, source):
         """
            remember which sources always have the song-category buttons enabled
-- 
2.25.0

