From 2a223670082d7c7dd34c7cca0b5f4b4793164594 Mon Sep 17 00:00:00 2001
From: Ikey Doherty <ikey@solus-project.com>
Date: Tue, 1 Mar 2016 03:52:58 +0000
Subject: [PATCH 1/2] Activate by default

Signed-off-by: Ikey Doherty <ikey@solus-project.com>
---
 alternative-toolbar.plugin.in | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/alternative-toolbar.plugin.in b/alternative-toolbar.plugin.in
index 365c705..a48d0e2 100644
--- a/alternative-toolbar.plugin.in
+++ b/alternative-toolbar.plugin.in
@@ -10,3 +10,6 @@ Copyright=© 2016 fossfreedom
 Website=http://github.com/fossfreedom/alternative-toolbar
 Help=https://github.com/fossfreedom/alternative-toolbar/blob/master/README.md
 Version=0.16.2
+
+[RB]
+InitiallyEnabled=true
-- 
2.7.1

