From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Reilly Brogan <reilly@reillybrogan.com>
Date: Sat, 3 May 2025 00:57:11 -0500
Subject: [PATCH] Force python3

---
 gui/GameConqueror.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/gui/GameConqueror.py b/gui/GameConqueror.py
index 47f3f82..cbb7d49 100644
--- a/gui/GameConqueror.py
+++ b/gui/GameConqueror.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 """
     Game Conqueror: a graphical game cheating tool, using scanmem as its backend
     
