From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Evan Maddock <maddock.evan@vivaldi.net>
Date: Fri, 31 Oct 2025 19:15:50 -0400
Subject: [PATCH] build: Port to libpeas-2 and budgie-2.0

Signed-off-by: Evan Maddock <maddock.evan@vivaldi.net>
---
 src/meson.build | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/meson.build b/src/meson.build
index dfaca89..b72f5c2 100644
--- a/src/meson.build
+++ b/src/meson.build
@@ -23,7 +23,7 @@ calendar_applet_sources = ['Applet.vala',
                'AppletSettings.vala']
 
 calendar_applet_deps = [dependency('gtk+-3.0', version: '>=3.18'),
-            dependency('budgie-1.0', version: '>=2')]
+            dependency('budgie-2.0', version: '>=3')]
 
 calendar_applet_vala_args = ['--target-glib=2.50', config_header]
 
