diff --git a/librawstudio/rs-exif.cc b/librawstudio/rs-exif.cc
index 8ad23a16..142c011e 100644
--- a/librawstudio/rs-exif.cc
+++ b/librawstudio/rs-exif.cc
@@ -21,6 +21,8 @@
 #include <iomanip>
 #include <exiv2/image.hpp>
 #include <exiv2/exif.hpp>
+#include <exiv2/exiv2.hpp>
+#include <exiv2/error.hpp>
 #include "rs-exif.h"
 #include <assert.h>
 #include "rs-library.h"
diff --git a/plugins/load-gdk/exiv2-colorspace.cpp b/plugins/load-gdk/exiv2-colorspace.cpp
index a34ea0b7..58586ff2 100644
--- a/plugins/load-gdk/exiv2-colorspace.cpp
+++ b/plugins/load-gdk/exiv2-colorspace.cpp
@@ -22,6 +22,8 @@
 #include <iomanip>
 #include <exiv2/image.hpp>
 #include <exiv2/exif.hpp>
+#include <exiv2/exiv2.hpp>
+#include <exiv2/error.hpp>
 #include <assert.h>
 #include "exiv2-colorspace.h"
 #include <math.h>
diff --git a/plugins/load-png/exiv2-colorspace.cpp b/plugins/load-png/exiv2-colorspace.cpp
index a34ea0b7..58586ff2 100644
--- a/plugins/load-png/exiv2-colorspace.cpp
+++ b/plugins/load-png/exiv2-colorspace.cpp
@@ -22,6 +22,8 @@
 #include <iomanip>
 #include <exiv2/image.hpp>
 #include <exiv2/exif.hpp>
+#include <exiv2/exiv2.hpp>
+#include <exiv2/error.hpp>
 #include <assert.h>
 #include "exiv2-colorspace.h"
 #include <math.h>
diff --git a/plugins/meta-exiv2/exiv2-metadata.cpp b/plugins/meta-exiv2/exiv2-metadata.cpp
index 8a2fc762..85cf21c2 100644
--- a/plugins/meta-exiv2/exiv2-metadata.cpp
+++ b/plugins/meta-exiv2/exiv2-metadata.cpp
@@ -22,6 +22,8 @@
 #include <iomanip>
 #include <exiv2/image.hpp>
 #include <exiv2/exif.hpp>
+#include <exiv2/exiv2.hpp>
+#include <exiv2/error.hpp>
 #include <assert.h>
 #include "exiv2-metadata.h"
 #include <math.h>
