From 12d0d6486d17c26504154ee11c0b5d078200ff3f Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?S=C3=A9rgio=20M=2E=20Basto?= <sergio@serjux.com>
Date: Thu, 27 May 2021 00:11:21 +0100
Subject: [PATCH] Second fix for glib 2.68 we just need remove 2 libraries from
 the wrapper

---
 librawstudio/rawstudio.h | 8 ++++++++
 1 file changed, 8 insertions(+)

diff --git a/librawstudio/rawstudio.h b/librawstudio/rawstudio.h
index 4b58d9f6..22e8206b 100644
--- a/librawstudio/rawstudio.h
+++ b/librawstudio/rawstudio.h
@@ -23,6 +23,10 @@
 #include <sys/types.h>
 #include "rs-types.h"
 
+#ifdef  __cplusplus
+extern "C" {
+#endif
+
 #include "rs-macros.h"
 
 #include "rs-debug.h"
@@ -74,4 +78,8 @@
 
 #include "x86-cpu.h"
 
+#ifdef  __cplusplus
+} /* extern "c" */
+#endif
+
 #endif /* RAWSTUDIO_H */
