diff -ur rawstudio-c140a5eb64901e07db5190db20f9884e86e5dcae.orig/librawstudio/rs-curve.c rawstudio-c140a5eb64901e07db5190db20f9884e86e5dcae/librawstudio/rs-curve.c
--- rawstudio-c140a5eb64901e07db5190db20f9884e86e5dcae.orig/librawstudio/rs-curve.c	2021-06-02 23:43:01.000000000 +0200
+++ rawstudio-c140a5eb64901e07db5190db20f9884e86e5dcae/librawstudio/rs-curve.c	2024-01-09 13:04:49.880221963 +0100
@@ -21,6 +21,7 @@
 #include <rawstudio.h>
 #include <math.h>
 #include <libxml/encoding.h>
+#include <libxml/parser.h>
 #include <libxml/xmlwriter.h>
 #include <string.h> /* memset() */
 
diff -ur rawstudio-c140a5eb64901e07db5190db20f9884e86e5dcae.orig/librawstudio/rs-lens-db.c rawstudio-c140a5eb64901e07db5190db20f9884e86e5dcae/librawstudio/rs-lens-db.c
--- rawstudio-c140a5eb64901e07db5190db20f9884e86e5dcae.orig/librawstudio/rs-lens-db.c	2021-06-02 23:43:01.000000000 +0200
+++ rawstudio-c140a5eb64901e07db5190db20f9884e86e5dcae/librawstudio/rs-lens-db.c	2024-01-09 12:59:54.619241429 +0100
@@ -19,6 +19,7 @@
 
 #include <rawstudio.h>
 #include <libxml/encoding.h>
+#include <libxml/parser.h>
 #include <libxml/xmlwriter.h>
 #include "config.h"
 #include "rs-lens-db.h"
diff -ur rawstudio-c140a5eb64901e07db5190db20f9884e86e5dcae.orig/librawstudio/rs-lens-fix.c rawstudio-c140a5eb64901e07db5190db20f9884e86e5dcae/librawstudio/rs-lens-fix.c
--- rawstudio-c140a5eb64901e07db5190db20f9884e86e5dcae.orig/librawstudio/rs-lens-fix.c	2021-06-02 23:43:01.000000000 +0200
+++ rawstudio-c140a5eb64901e07db5190db20f9884e86e5dcae/librawstudio/rs-lens-fix.c	2024-01-09 13:01:15.219416911 +0100
@@ -19,6 +19,7 @@
 
 #include <rawstudio.h>
 #include <libxml/encoding.h>
+#include <libxml/tree.h>
 #include "config.h"
 
 static GHashTable *lens_fix_hash_table;
diff -ur rawstudio-c140a5eb64901e07db5190db20f9884e86e5dcae.orig/librawstudio/rs-library.c rawstudio-c140a5eb64901e07db5190db20f9884e86e5dcae/librawstudio/rs-library.c
--- rawstudio-c140a5eb64901e07db5190db20f9884e86e5dcae.orig/librawstudio/rs-library.c	2021-06-02 23:43:01.000000000 +0200
+++ rawstudio-c140a5eb64901e07db5190db20f9884e86e5dcae/librawstudio/rs-library.c	2024-01-09 13:02:36.693583821 +0100
@@ -60,6 +60,7 @@
 #include "gettext.h"
 #include "rs-debug.h"
 #include <libxml/encoding.h>
+#include <libxml/parser.h>
 #include <libxml/xmlwriter.h>
 #include <sqlite3.h>
 
diff -ur rawstudio-c140a5eb64901e07db5190db20f9884e86e5dcae.orig/librawstudio/rs-metadata.c rawstudio-c140a5eb64901e07db5190db20f9884e86e5dcae/librawstudio/rs-metadata.c
--- rawstudio-c140a5eb64901e07db5190db20f9884e86e5dcae.orig/librawstudio/rs-metadata.c	2021-06-02 23:43:01.000000000 +0200
+++ rawstudio-c140a5eb64901e07db5190db20f9884e86e5dcae/librawstudio/rs-metadata.c	2024-01-09 12:56:38.947244136 +0100
@@ -21,6 +21,7 @@
 #include <glib/gstdio.h> /* g_unlink() */
 #include <config.h>
 #include <libxml/encoding.h>
+#include <libxml/parser.h>
 #include <libxml/xmlwriter.h>
 #include "gettext.h"
 
diff -ur rawstudio-c140a5eb64901e07db5190db20f9884e86e5dcae.orig/librawstudio/rs-profile-camera.c rawstudio-c140a5eb64901e07db5190db20f9884e86e5dcae/librawstudio/rs-profile-camera.c
--- rawstudio-c140a5eb64901e07db5190db20f9884e86e5dcae.orig/librawstudio/rs-profile-camera.c	2021-06-02 23:43:01.000000000 +0200
+++ rawstudio-c140a5eb64901e07db5190db20f9884e86e5dcae/librawstudio/rs-profile-camera.c	2024-01-09 13:03:45.895876219 +0100
@@ -20,6 +20,7 @@
 #include <glib-2.0/glib.h>
 #include "config.h"
 #include <libxml/encoding.h>
+#include <libxml/parser.h>
 #include <libxml/xmlwriter.h>
 #include "rs-utils.h"
 
diff -ur rawstudio-c140a5eb64901e07db5190db20f9884e86e5dcae.orig/profiles/write-dcp.c rawstudio-c140a5eb64901e07db5190db20f9884e86e5dcae/profiles/write-dcp.c
--- rawstudio-c140a5eb64901e07db5190db20f9884e86e5dcae.orig/profiles/write-dcp.c	2021-06-02 23:43:01.000000000 +0200
+++ rawstudio-c140a5eb64901e07db5190db20f9884e86e5dcae/profiles/write-dcp.c	2024-01-09 13:17:27.218467219 +0100
@@ -20,6 +20,7 @@
 #include <glib.h>
 #include <tiffio.h>
 #include <libxml/encoding.h>
+#include <libxml/tree.h>
 
 #define DCPTAG_PROFILE_NAME 0xc6f8
 #define DCPTAG_PROFILE_EMBED_POLICY 0xc6fd
diff -ur rawstudio-c140a5eb64901e07db5190db20f9884e86e5dcae.orig/src/rs-batch.c rawstudio-c140a5eb64901e07db5190db20f9884e86e5dcae/src/rs-batch.c
--- rawstudio-c140a5eb64901e07db5190db20f9884e86e5dcae.orig/src/rs-batch.c	2021-06-02 23:43:01.000000000 +0200
+++ rawstudio-c140a5eb64901e07db5190db20f9884e86e5dcae/src/rs-batch.c	2024-01-09 13:11:26.551162933 +0100
@@ -23,6 +23,7 @@
 #include <gtk/gtk.h>
 #include <config.h>
 #include <libxml/encoding.h>
+#include <libxml/parser.h>
 #include <libxml/xmlwriter.h>
 #include "application.h"
 #include "rs-batch.h"
diff -ur rawstudio-c140a5eb64901e07db5190db20f9884e86e5dcae.orig/src/rs-cache.c rawstudio-c140a5eb64901e07db5190db20f9884e86e5dcae/src/rs-cache.c
--- rawstudio-c140a5eb64901e07db5190db20f9884e86e5dcae.orig/src/rs-cache.c	2021-06-02 23:43:01.000000000 +0200
+++ rawstudio-c140a5eb64901e07db5190db20f9884e86e5dcae/src/rs-cache.c	2024-01-09 13:08:29.441976914 +0100
@@ -20,6 +20,7 @@
 #include <rawstudio.h>
 #include <glib.h>
 #include <libxml/encoding.h>
+#include <libxml/parser.h>
 #include <libxml/xmlwriter.h>
 #include "application.h"
 #include "rs-cache.h"
diff -ur rawstudio-c140a5eb64901e07db5190db20f9884e86e5dcae.orig/src/rs-camera-db.c rawstudio-c140a5eb64901e07db5190db20f9884e86e5dcae/src/rs-camera-db.c
--- rawstudio-c140a5eb64901e07db5190db20f9884e86e5dcae.orig/src/rs-camera-db.c	2021-06-02 23:43:01.000000000 +0200
+++ rawstudio-c140a5eb64901e07db5190db20f9884e86e5dcae/src/rs-camera-db.c	2024-01-09 13:10:03.617012744 +0100
@@ -20,6 +20,7 @@
 #include "config.h"
 #include "gettext.h"
 #include <libxml/encoding.h>
+#include <libxml/parser.h>
 #include <libxml/xmlwriter.h>
 #include "rs-camera-db.h"
 #include "rs-photo.h"
diff -ur rawstudio-c140a5eb64901e07db5190db20f9884e86e5dcae.orig/src/rs-store.c rawstudio-c140a5eb64901e07db5190db20f9884e86e5dcae/src/rs-store.c
--- rawstudio-c140a5eb64901e07db5190db20f9884e86e5dcae.orig/src/rs-store.c	2021-06-02 23:43:01.000000000 +0200
+++ rawstudio-c140a5eb64901e07db5190db20f9884e86e5dcae/src/rs-store.c	2024-01-09 13:11:37.297052819 +0100
@@ -22,6 +22,7 @@
 #include <glib/gprintf.h>
 #include <config.h>
 #include <libxml/encoding.h>
+#include <libxml/parser.h>
 #include <libxml/xmlwriter.h>
 #include <glib.h>
 #include <math.h>
