diff -ur rawstudio-c140a5eb64901e07db5190db20f9884e86e5dcae.orig/librawstudio/rs-curve.c rawstudio-c140a5eb64901e07db5190db20f9884e86e5dcae/librawstudio/rs-curve.c
--- rawstudio-c140a5eb64901e07db5190db20f9884e86e5dcae.orig/librawstudio/rs-curve.c	2024-01-09 13:04:49.880221963 +0100
+++ rawstudio-c140a5eb64901e07db5190db20f9884e86e5dcae/librawstudio/rs-curve.c	2024-01-09 13:32:00.191433885 +0100
@@ -58,7 +58,7 @@
 static gboolean rs_curve_size_allocate_helper(RSCurveWidget *curve);
 static void rs_curve_size_allocate(GtkWidget *widget, GtkAllocation *allocation, gpointer user_data);
 static void rs_curve_changed(RSCurveWidget *curve);
-static gboolean rs_curve_widget_draw(RSCurveWidget *curve, cairo_t *cr);
+static gboolean rs_curve_widget_draw(GtkWidget *curve, cairo_t *cr);
 static gboolean rs_curve_widget_button_press(GtkWidget *widget, GdkEventButton *event);
 static gboolean rs_curve_widget_button_release(GtkWidget *widget, GdkEventButton *event);
 static gboolean rs_curve_widget_motion_notify(GtkWidget *widget, GdkEventMotion *event);
@@ -808,13 +808,9 @@
  * Draw everything
  */
 static gboolean
-rs_curve_widget_draw(RSCurveWidget *curve, cairo_t *cr)
+rs_curve_widget_draw(GtkWidget *widget, cairo_t *cr)
 {
-	GtkWidget *widget;
-	g_return_if_fail (curve != NULL);
-	g_return_if_fail (RS_IS_CURVE_WIDGET(curve));
-
-	widget = GTK_WIDGET(curve);
+	g_return_val_if_fail (widget != NULL, TRUE);
 
 	if (gtk_widget_get_visible(widget) && gtk_widget_get_realized(widget))
 	{
@@ -827,6 +823,7 @@
 		/* Draw the curve */
 		rs_curve_draw_spline(widget, cr);
 	}
+	return TRUE;
 }
   
 static gboolean
