From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Evan Maddock <maddock.evan@vivaldi.net>
Date: Mon, 20 Apr 2026 16:38:12 -0400
Subject: [PATCH] build: Port to libpeas-2 and budgie-3.0

Signed-off-by: Evan Maddock <maddock.evan@vivaldi.net>
---
 meson.build | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/meson.build b/meson.build
index 3a435c8..b05ae47 100644
--- a/meson.build
+++ b/meson.build
@@ -17,11 +17,11 @@ shared_library(
   srcs,
   dependencies: [
     dependency('gtk+-3.0'),
-    dependency('budgie-1.0', version: '>=2'),
-    dependency('libpeas-1.0', version: '>=1.8.0')
+    dependency('budgie-3.0', version: '>=4'),
+    dependency('libpeas-2', version: '>=1.99.0')
   ],
   vala_args: [
-    '--pkg', 'libpeas-1.0',
+    '--pkg', 'libpeas-2',
     '--pkg', 'gtk+-3.0'
   ],
   install: true,
