From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Evan Maddock <maddock.evan@vivaldi.net>
Date: Sat, 1 Nov 2025 21:33:21 -0400
Subject: [PATCH] build: Port to libpeas-2 and budgie-2.0

Signed-off-by: Evan Maddock <maddock.evan@vivaldi.net>
---
 meson.build | 8 ++------
 1 file changed, 2 insertions(+), 6 deletions(-)

diff --git a/meson.build b/meson.build
index 3a435c8..83eae1f 100644
--- a/meson.build
+++ b/meson.build
@@ -17,12 +17,8 @@ shared_library(
   srcs,
   dependencies: [
     dependency('gtk+-3.0'),
-    dependency('budgie-1.0', version: '>=2'),
-    dependency('libpeas-1.0', version: '>=1.8.0')
-  ],
-  vala_args: [
-    '--pkg', 'libpeas-1.0',
-    '--pkg', 'gtk+-3.0'
+    dependency('budgie-2.0', version: '>=3'),
+    dependency('libpeas-2', version: '>=1.99.0')
   ],
   install: true,
   install_dir: applet_dir
