From b16861c2bf4d6b6076f929c7edee1c740ddccdd9 Mon Sep 17 00:00:00 2001
From: Muhammad Alfi Syahrin <malfisya.dev@hotmail.com>
Date: Thu, 11 Jan 2024 07:12:36 +0700
Subject: [PATCH] fix  icon in overview

---
 meta/kraft.desktop | 1 +
 1 file changed, 1 insertion(+)

diff --git a/meta/kraft.desktop b/meta/kraft.desktop
index 974df97..0030522 100644
--- a/meta/kraft.desktop
+++ b/meta/kraft.desktop
@@ -2,6 +2,7 @@
 Type=Application
 Exec=kraft %u 
 Icon=kraft
+StartupWMClass=kraft
 X-DocPath=kraft/index.html
 Terminal=false
 Name=Kraft
-- 
2.43.0

