From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Thomas Staudinger <Staudi.Kaos@gmail.com>
Date: Thu, 9 May 2024 14:22:52 +0200
Subject: [PATCH] Use sudo group instead of wheel

---
 data/configs/blueman.rules | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/data/configs/blueman.rules b/data/configs/blueman.rules
index 3b0862dc..a58701c3 100644
--- a/data/configs/blueman.rules
+++ b/data/configs/blueman.rules
@@ -4,7 +4,7 @@ polkit.addRule(function(action, subject) {
          action.id == "org.blueman.dhcp.client" ||
          action.id == "org.blueman.rfkill.setstate" ||
          action.id == "org.blueman.pppd.pppconnect") &&
-        subject.isInGroup("wheel")) {
+        subject.isInGroup("sudo")) {
         return polkit.Result.YES;
     }
 });
