From 100488f82f696b2a820ef4f2d20482913ebaf337 Mon Sep 17 00:00:00 2001
From: Ikey Doherty <ikey@solus-project.com>
Date: Thu, 1 Mar 2018 12:15:46 +0000
Subject: [PATCH] Use local cmake modules

Signed-off-by: Ikey Doherty <ikey@solus-project.com>
---
 CMakeLists.txt | 1 +
 1 file changed, 1 insertion(+)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index f1c5f7a..7b0e63c 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -2,6 +2,7 @@ cmake_minimum_required (VERSION 2.8)
 cmake_policy (VERSION 2.8)
 
 project (com.github.artemanufrij.playmyvideos)
+list (APPEND CMAKE_MODULE_PATH ${CMAKE_SOURCE_DIR}/cmake)
 include (GNUInstallDirs)
 set (DATADIR "${CMAKE_INSTALL_FULL_DATAROOTDIR}")
 set (PKGDATADIR "${DATADIR}/${CMAKE_PROJECT_NAME}")
-- 
2.16.2

