From e960c5168654211a8429e8089ba160e3cc89dc37 Mon Sep 17 00:00:00 2001
From: Dylan Araps <dylan.araps@gmail.com>
Date: Sun, 8 Mar 2020 16:13:19 +0200
Subject: [PATCH] neofetch: Fix uptime seconds on Linux. Closes #1422

---
 neofetch | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/neofetch b/neofetch
index 1bcb1a10..614d51e7 100755
--- a/neofetch
+++ b/neofetch
@@ -1396,7 +1396,7 @@ get_uptime() {
 
     uptime=${d:+$d, }${h:+$h, }$m
     uptime=${uptime%', '}
-    uptime=${uptime:-$seconds secs}
+    uptime=${uptime:-$s secs}
 
     # Make the output of uptime smaller.
     case $uptime_shorthand in
