From e09146142f9abdf0ecf0badc4e3c3a5e3c51c23a Mon Sep 17 00:00:00 2001
From: Dylan Araps <dylan.araps@gmail.com>
Date: Sun, 8 Mar 2020 17:42:00 +0200
Subject: [PATCH] neofetch: Prefer lsb-release

---
 neofetch | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/neofetch b/neofetch
index 614d51e7..0a25c0b1 100755
--- a/neofetch
+++ b/neofetch
@@ -979,8 +979,8 @@ get_distro() {
                     -f /etc/lsb-release ]]; then
 
                 # Source the os-release file
-                for file in /usr/lib/os-release  /etc/os-release \
-                            /etc/openwrt_release /etc/lsb-release; do
+                for file in /etc/lsb-release /usr/lib/os-release \
+                            /etc/os-release  /etc/openwrt_release; do
                     source "$file" && break
                 done
 
