From 5241b24bf4e8c21ba5a4eaf3464102d8316b2b76 Mon Sep 17 00:00:00 2001
From: Dylan Araps <dylan.araps@gmail.com>
Date: Mon, 9 Mar 2020 10:58:31 +0200
Subject: [PATCH] neofetch: Enable DE version by default

---
 neofetch | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/neofetch b/neofetch
index 0a25c0b1..88ce6965 100755
--- a/neofetch
+++ b/neofetch
@@ -398,10 +398,10 @@ public_ip_timeout=2
 
 # Show Desktop Environment version
 #
-# Default: 'off'
+# Default: 'on'
 # Values:  'on', 'off'
 # Flag:    --de_version
-de_version="off"
+de_version="on"
 
 
 # Disk
