From 0c038451000e42a93323268a8e233062cea4783c Mon Sep 17 00:00:00 2001
From: Muhammad Herdiansyah <herdiansyah@openmailbox.org>
Date: Fri, 3 Mar 2017 14:43:10 +0700
Subject: [PATCH] Packages: Used better method for checking packages in eopkg
 (#669)

---
 neofetch | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/neofetch b/neofetch
index 9d39edc..8a8aa53 100755
--- a/neofetch
+++ b/neofetch
@@ -468,7 +468,7 @@ get_packages() {
                 packages="$((packages+=$(ls -d -1 /Programs/*/ | wc -l)))"
 
             type -p eopkg >/dev/null && \
-                packages="$((packages+=$(eopkg list-installed | wc -l)))"
+                packages="$((packages+=$(ls -1 /var/lib/eopkg/package | wc -l)))"
 
             type -p pkg_info >/dev/null && \
                 packages="$((packages+=$(pkg_info | wc -l)))"
