From d70387c00855448e1467740d81464e7993b7ab1f Mon Sep 17 00:00:00 2001
From: James Lee <jamesl33info@gmail.com>
Date: Mon, 29 Oct 2018 16:45:39 +0000
Subject: [PATCH] Add Solus liability waiver

---
 intel-undervolt.conf | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/intel-undervolt.conf b/intel-undervolt.conf
index 5ea90c7..3753b68 100644
--- a/intel-undervolt.conf
+++ b/intel-undervolt.conf
@@ -1,3 +1,9 @@
+# DISCLAIMER - Solus/The Solus team do not condone the use of intel-undervolt
+# without doing proper research beforehand. If done incorrectly you could
+# permanently damage your hardware. Solus and/or The Solus team take no
+# responsibility for any damage caused to your hardware from the incorrect
+# use of the intel-undervolt software.
+
 # CPU Undervolting
 # Usage: apply ${index} ${display_name} ${undervolt_value}
 # Example: apply 2 'CPU Cache' -25.84
--
2.19.1
