From b064c8368f3c401510b9313abc6a269832d57449 Mon Sep 17 00:00:00 2001
From: James Lee <jamesl33info@gmail.com>
Date: Sat, 6 Jul 2019 21:10:34 +0100
Subject: [PATCH] Add Solus liability waiver

---
 intel-undervolt.conf | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/intel-undervolt.conf b/intel-undervolt.conf
index ccb5065..c35905c 100644
--- a/intel-undervolt.conf
+++ b/intel-undervolt.conf
@@ -1,3 +1,9 @@
+# DISCLAIMER - Solus/The Solus team do not condone the use of intel-undervolt
+# without doing proper research beforehand. If done incorrectly you could
+# permanently damage your hardware. Solus and/or The Solus team take no
+# responsibility for any damage caused to your hardware from the incorrect
+# use of the intel-undervolt software.
+
 # Enable or Disable Triggers (elogind)
 # Usage: enable [yes/no]

--
2.22.0

