--- slang-orig/CMakeLists.txt	2026-09-24 21:00:43.472000000 -0400
+++ slang-work/CMakeLists.txt	2026-09-24 21:01:22.828226734 -0400
@@ -619,10 +619,9 @@
 endif()
 
 if(${SLANG_USE_SYSTEM_LZ4})
-    find_package(lz4 REQUIRED)
-    if(NOT TARGET lz4_static)
-        add_library(lz4_static ALIAS LZ4::lz4)
-    endif()
+    find_package(PkgConfig REQUIRED)
+    pkg_check_modules(lz4 REQUIRED IMPORTED_TARGET liblz4)
+    add_library(lz4_static ALIAS PkgConfig::lz4)
 endif()
 
 if(${SLANG_USE_SYSTEM_VULKAN_HEADERS})
@@ -830,11 +829,6 @@
     DESTINATION share/doc/slang
     PATTERN ".*" EXCLUDE
 )
-install(
-    DIRECTORY "${slang_SOURCE_DIR}/include"
-    DESTINATION .
-    PATTERN ".*" EXCLUDE
-)
 include(SlangUserSkills)
 slang_install_user_skills()
 include(CPack)
@@ -904,7 +898,7 @@
         # update this destination.
         install(
             FILES "${PROJECT_BINARY_DIR}/slang-compiler.pc"
-            DESTINATION "lib/pkgconfig"
+            DESTINATION "${CMAKE_INSTALL_LIBDIR}/pkgconfig"
         )
     endif()
 endif()
