--- slang-orig/extras/pkgconfig/slang-compiler.pc.in	2026-09-24 21:00:43.576446799 -0400
+++ slang-work/extras/pkgconfig/slang-compiler.pc.in	2026-09-24 21:01:22.828279074 -0400
@@ -1,7 +1,7 @@
 prefix=${pcfiledir}/../..
 exec_prefix=${prefix}
-libdir=${prefix}/lib
-includedir=${prefix}/include
+libdir=${prefix}/@CMAKE_INSTALL_LIBDIR@
+includedir=${prefix}/@CMAKE_INSTALL_INCLUDEDIR@
 
 Name: slang-compiler
 Description: Slang shading language compiler library
