--- slang-orig/cmake/SlangTarget.cmake	2026-09-24 21:00:43.473631528 -0400
+++ slang-work/cmake/SlangTarget.cmake	2026-09-24 21:01:22.828342984 -0400
@@ -723,7 +723,7 @@
 # Linux systems) relies on runtime libraries being at "$ORIGIN/../lib". This
 # could be improved by setting at configure-time that path to be the relative
 # path from CM_I_RD to CM_I_LD.
-set(library_subdir lib)
+set(library_subdir ${CMAKE_INSTALL_LIBDIR})
 set(runtime_subdir bin)
 
 # On Windows, because there's no RPATH, place modules in bin, next to the
