--- slang-orig/source/slang/CMakeLists.txt	2026-09-24 21:00:43.587262638 -0400
+++ slang-work/source/slang/CMakeLists.txt	2026-09-24 21:01:22.828407324 -0400
@@ -499,8 +499,8 @@
     install(
         TARGETS slang-proxy
         RUNTIME DESTINATION bin
-        ARCHIVE DESTINATION lib
-        LIBRARY DESTINATION lib
+        ARCHIVE DESTINATION ${CMAKE_INSTALL_LIBDIR}
+        LIBRARY DESTINATION ${CMAKE_INSTALL_LIBDIR}
     )
 
     message(
