--- slang-orig/source/standard-modules/CMakeLists.txt	2026-09-24 21:00:43.615409540 -0400
+++ slang-work/source/standard-modules/CMakeLists.txt	2026-09-24 21:01:22.828443744 -0400
@@ -19,7 +19,7 @@
 if(CMAKE_SYSTEM_NAME STREQUAL "Windows")
     set(slang_library_dir "bin")
 else()
-    set(slang_library_dir "lib")
+    set(slang_library_dir ${CMAKE_INSTALL_LIBDIR})
 endif()
 
 # Set install directory using the platform-specific library directory
