--- slang-orig/source/slang-glslang/slang-glslang.cpp	2026-09-24 21:00:43.586208197 -0400
+++ slang-work/source/slang-glslang/slang-glslang.cpp	2026-09-24 21:01:22.828494804 -0400
@@ -1,7 +1,7 @@
 // slang-glslang.cpp
 #include "slang-glslang.h"
 
-#include "SPIRV/GlslangToSpv.h"
+#include "glslang/SPIRV/GlslangToSpv.h"
 #include "glslang/Public/ShaderLang.h"
 #include "slang.h"
 #include "spirv-tools/libspirv.h"
