From 85bf70698ed94f73ea6b84534911af81cf7aaa18 Mon Sep 17 00:00:00 2001
From: Pete Alexandrou <pete@ozmartians.com>
Date: Mon, 13 Dec 2021 14:02:34 +1100
Subject: [PATCH] Python 3.10 fix for QSize() float bug

---
 vidcutter/videoslider.py | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/vidcutter/videoslider.py b/vidcutter/videoslider.py
index a846a7c1..86154622 100644
--- a/vidcutter/videoslider.py
+++ b/vidcutter/videoslider.py
@@ -246,8 +246,8 @@ def clearProgress(self) -> None:
     def initThumbs(self) -> None:
         framesize = self.parent.videoService.framesize()
         thumbsize = QSize(
-            VideoService.config.thumbnails['TIMELINE'].height() * (framesize.width() / framesize.height()),
-            VideoService.config.thumbnails['TIMELINE'].height())
+            int(VideoService.config.thumbnails['TIMELINE'].height() * (framesize.width() / framesize.height())),
+            int(VideoService.config.thumbnails['TIMELINE'].height()))
         positions, frametimes = [], []
         thumbs = int(math.ceil((self.rect().width() - (self.offset * 2)) / thumbsize.width()))
         for pos in range(thumbs):
