From aec5fbaf1c4f22bcfdf70b17f11b10441aed8c61 Mon Sep 17 00:00:00 2001
From: Shannon Huggins <x@ptr.me>
Date: Sat, 10 May 2025 15:24:38 +1000
Subject: [PATCH] device_type edge handling for NoneType

Signed-off-by: Shannon Huggins <x@ptr.me>
---

 ui5/devmgr5.py | 18 +++++++++---------
 1 file changed, 9 insertions(+), 9 deletions(-)

diff --git a/ui5/devmgr5.py b/ui5/devmgr5.py
index 8a19840..93fbf9c 100644
--- a/ui5/devmgr5.py
+++ b/ui5/devmgr5.py
@@ -820,7 +820,7 @@ class DevMgr5(Ui_MainWindow_Derived, Ui_MainWindow, QMainWindow):
     def updateWindowTitle(self):
         if self.cur_device is None or self.cur_printer is None:
             return
-        if self.cur_device.device_type == DEVICE_TYPE_FAX:
+        if hasattr(self.cur_device,'device_type') and self.cur_device.device_type == DEVICE_TYPE_FAX:
                 self.setWindowTitle(self.__tr("HP Device Manager - %s (Fax)"%self.cur_device.model_ui))
         else:
             if self.cur_device.fax_type:
@@ -853,7 +853,7 @@ class DevMgr5(Ui_MainWindow_Derived, Ui_MainWindow, QMainWindow):
             self.updatePrinterCombos()
             self.updateCurrentTab()
             self.statusBar().showMessage(self.cur_device_uri)
-            if self.cur_device.device_type == DEVICE_TYPE_PRINTER:
+            if  hasattr(self.cur_device, 'device_type') and self.cur_device.device_type == DEVICE_TYPE_PRINTER:
                 self.Tabs.setTabText(self.Tabs.indexOf(self.Settings), QApplication.translate("MainWindow", "Print Settings", None))
                 self.Tabs.setTabText(self.Tabs.indexOf(self.Control), QApplication.translate("MainWindow", "Printer Control", None))
             else:
@@ -1784,7 +1784,7 @@ class DevMgr5(Ui_MainWindow_Derived, Ui_MainWindow, QMainWindow):
     def updatePrintSettingsTab(self):
         beginWaitCursor()
         try:
-            if self.cur_device.device_type == DEVICE_TYPE_PRINTER:
+            if hasattr(self.cur_device,'device_type') and self.cur_device.device_type == DEVICE_TYPE_PRINTER:
                 self.PrintSettingsPrinterNameLabel.setText(self.__tr("Printer Name:"))
             else:
                 self.PrintSettingsPrinterNameLabel.setText(self.__tr("Fax Name:"))
@@ -1882,7 +1882,7 @@ class DevMgr5(Ui_MainWindow_Derived, Ui_MainWindow, QMainWindow):
     def updatePrintControlTab(self):
         if self.cur_device is None or self.cur_printer is None:
             return
-        if self.cur_device.device_type == DEVICE_TYPE_PRINTER:
+        if hasattr(self.cur_device,'device_type') and self.cur_device.device_type == DEVICE_TYPE_PRINTER:
             self.PrintControlPrinterNameLabel.setText(self.__tr("Printer Name:"))
             self.groupBox.setTitle(QApplication.translate("MainWindow", "Printer Queue Control", None))
         else:
@@ -1947,7 +1947,7 @@ class DevMgr5(Ui_MainWindow_Derived, Ui_MainWindow, QMainWindow):
         
         default_printer = cups.getDefaultPrinter()
             
-        if self.cur_device.device_type == DEVICE_TYPE_PRINTER:
+        if hasattr(self.cur_device,'device_type') and self.cur_device.device_type == DEVICE_TYPE_PRINTER:
             device_string = "Printer"
         else:
             device_string = "Fax"
@@ -2008,7 +2008,7 @@ class DevMgr5(Ui_MainWindow_Derived, Ui_MainWindow, QMainWindow):
             else:
                 result, result_str = cups.cups_operation(cups.start, GUI_MODE, 'qt4', self, self.cur_printer)
                 if result == cups.IPP_OK:
-                    if self.cur_device.device_type == DEVICE_TYPE_PRINTER:
+                    if hasattr(self.cur_device,'device_type') and self.cur_device.device_type == DEVICE_TYPE_PRINTER:
                         e = EVENT_PRINTER_QUEUE_STARTED
                     else:
                         e = EVENT_FAX_QUEUE_STARTED
@@ -2031,7 +2031,7 @@ class DevMgr5(Ui_MainWindow_Derived, Ui_MainWindow, QMainWindow):
             if self.printer_accepting:
                 result, result_str = cups.cups_operation(cups.reject, GUI_MODE, 'qt4', self, self.cur_printer)
                 if result == cups.IPP_OK:
-                    if self.cur_device.device_type == DEVICE_TYPE_PRINTER:
+                    if hasattr(self.cur_device,'device_type') and self.cur_device.device_type == DEVICE_TYPE_PRINTER:
                         e = EVENT_PRINTER_QUEUE_REJECTING_JOBS
                     else:
                         e = EVENT_FAX_QUEUE_REJECTING_JOBS
@@ -2039,7 +2039,7 @@ class DevMgr5(Ui_MainWindow_Derived, Ui_MainWindow, QMainWindow):
             else:
                 result, result_str = cups.cups_operation(cups.accept, GUI_MODE, 'qt4', self, self.cur_printer)
                 if result == cups.IPP_OK:
-                    if self.cur_device.device_type == DEVICE_TYPE_PRINTER:
+                    if hasattr(self.cur_device,'device_type') and self.cur_device.device_type == DEVICE_TYPE_PRINTER:
                         e = EVENT_PRINTER_QUEUE_ACCEPTING_JOBS
                     else:
                         e = EVENT_FAX_QUEUE_ACCEPTING_JOBS
@@ -2065,7 +2065,7 @@ class DevMgr5(Ui_MainWindow_Derived, Ui_MainWindow, QMainWindow):
                 cups.releaseCupsInstance()
             else:
                 self.updatePrintController()
-                if self.cur_device.device_type == DEVICE_TYPE_PRINTER:
+                if hasattr(self.cur_device,'device_type') and self.cur_device.device_type == DEVICE_TYPE_PRINTER:
                     e = EVENT_PRINTER_QUEUE_SET_AS_DEFAULT
                 else:
                     e = EVENT_FAX_QUEUE_SET_AS_DEFAULT
-- 
2.52.0

