From e215af334ad4765a2488307d7a0415e2caa86336 Mon Sep 17 00:00:00 2001
From: Pierre-Yves <pyu@riseup.net>
Date: Mon, 12 Mar 2018 21:14:07 +0100
Subject: [PATCH] Ensure sudo use on Solus

Signed-off-by: Pierre-Yves <pyu@riseup.net>
---
 base/password.py | 1 +
 1 file changed, 1 insertion(+)

diff --git a/base/password.py b/base/password.py
index 42c42f8..967b295 100644
--- a/base/password.py
+++ b/base/password.py
@@ -37,6 +37,7 @@ AUTH_TYPES = {'mepis': 'su',
               'suse': 'su',
               'mandriva': 'su',
               'fedora': 'su',
+              'solus': 'sudo',
               'redhat': 'su',
               'rhel': 'su',
               'slackware': 'su',
-- 
2.16.2

