From aec5fbaf1c4f22bcfdf70b17f11b10441aed8c61 Mon Sep 17 00:00:00 2001
From: Shannon Huggins <x@ptr.me>
Date: Sat, 10 May 2025 15:24:38 +1000
Subject: [PATCH] device_type edge handling for NoneType

Signed-off-by: Shannon Huggins <x@ptr.me>
---
 ui5/devmgr5.py | 20 ++++++++++----------
 1 file changed, 10 insertions(+), 10 deletions(-)

diff --git a/ui5/devmgr5.py b/ui5/devmgr5.py
index b50eecc..20748b1 100644
--- a/ui5/devmgr5.py
+++ b/ui5/devmgr5.py
@@ -818,7 +818,7 @@ class DevMgr5(Ui_MainWindow_Derived, Ui_MainWindow, QMainWindow):
 
 
     def updateWindowTitle(self):
-        if self.cur_device.device_type == DEVICE_TYPE_FAX:
+        if hasattr(self.cur_device,'device_type') and self.cur_device.device_type == DEVICE_TYPE_FAX:
                 self.setWindowTitle(self.__tr("HP Device Manager - %s (Fax)"%self.cur_device.model_ui))
         else:
             if self.cur_device.fax_type:
@@ -851,7 +851,7 @@ class DevMgr5(Ui_MainWindow_Derived, Ui_MainWindow, QMainWindow):
             self.updatePrinterCombos()
             self.updateCurrentTab()
             self.statusBar().showMessage(self.cur_device_uri)
-            if self.cur_device.device_type == DEVICE_TYPE_PRINTER:
+            if  hasattr(self.cur_device, 'device_type') and self.cur_device.device_type == DEVICE_TYPE_PRINTER:
                 self.Tabs.setTabText(self.Tabs.indexOf(self.Settings), QApplication.translate("MainWindow", "Print Settings", None))
                 self.Tabs.setTabText(self.Tabs.indexOf(self.Control), QApplication.translate("MainWindow", "Printer Control", None))
             else:
@@ -1782,7 +1782,7 @@ class DevMgr5(Ui_MainWindow_Derived, Ui_MainWindow, QMainWindow):
     def updatePrintSettingsTab(self):
         beginWaitCursor()
         try:
-            if self.cur_device.device_type == DEVICE_TYPE_PRINTER:
+            if hasattr(self.cur_device,'device_type') and self.cur_device.device_type == DEVICE_TYPE_PRINTER:
                 self.PrintSettingsPrinterNameLabel.setText(self.__tr("Printer Name:"))
             else:
                 self.PrintSettingsPrinterNameLabel.setText(self.__tr("Fax Name:"))
@@ -1878,7 +1878,7 @@ class DevMgr5(Ui_MainWindow_Derived, Ui_MainWindow, QMainWindow):
 
 
     def updatePrintControlTab(self):
-        if self.cur_device.device_type == DEVICE_TYPE_PRINTER:
+        if hasattr(self.cur_device,'device_type') and self.cur_device.device_type == DEVICE_TYPE_PRINTER:
             self.PrintControlPrinterNameLabel.setText(self.__tr("Printer Name:"))
             self.groupBox.setTitle(QApplication.translate("MainWindow", "Printer Queue Control", None))
         else:
@@ -1943,7 +1943,7 @@ class DevMgr5(Ui_MainWindow_Derived, Ui_MainWindow, QMainWindow):
         
         default_printer = cups.getDefaultPrinter()
             
-        if self.cur_device.device_type == DEVICE_TYPE_PRINTER:
+        if hasattr(self.cur_device,'device_type') and self.cur_device.device_type == DEVICE_TYPE_PRINTER:
             device_string = "Printer"
         else:
             device_string = "Fax"
@@ -1996,7 +1996,7 @@ class DevMgr5(Ui_MainWindow_Derived, Ui_MainWindow, QMainWindow):
             if self.printer_state in (cups.IPP_PRINTER_STATE_IDLE, cups.IPP_PRINTER_STATE_PROCESSING):
                 result, result_str = cups.cups_operation(cups.stop, GUI_MODE, 'qt4', self, self.cur_printer)
                 if result == cups.IPP_OK:
-                    if self.cur_device.device_type == DEVICE_TYPE_PRINTER:
+                    if hasattr(self.cur_device,'device_type') and self.cur_device.device_type == DEVICE_TYPE_PRINTER:
                         e = EVENT_PRINTER_QUEUE_STOPPED
                     else:
                         e = EVENT_FAX_QUEUE_STOPPED
@@ -2004,7 +2004,7 @@ class DevMgr5(Ui_MainWindow_Derived, Ui_MainWindow, QMainWindow):
             else:
                 result, result_str = cups.cups_operation(cups.start, GUI_MODE, 'qt4', self, self.cur_printer)
                 if result == cups.IPP_OK:
-                    if self.cur_device.device_type == DEVICE_TYPE_PRINTER:
+                    if hasattr(self.cur_device,'device_type') and self.cur_device.device_type == DEVICE_TYPE_PRINTER:
                         e = EVENT_PRINTER_QUEUE_STARTED
                     else:
                         e = EVENT_FAX_QUEUE_STARTED
@@ -2027,7 +2027,7 @@ class DevMgr5(Ui_MainWindow_Derived, Ui_MainWindow, QMainWindow):
             if self.printer_accepting:
                 result, result_str = cups.cups_operation(cups.reject, GUI_MODE, 'qt4', self, self.cur_printer)
                 if result == cups.IPP_OK:
-                    if self.cur_device.device_type == DEVICE_TYPE_PRINTER:
+                    if hasattr(self.cur_device,'device_type') and self.cur_device.device_type == DEVICE_TYPE_PRINTER:
                         e = EVENT_PRINTER_QUEUE_REJECTING_JOBS
                     else:
                         e = EVENT_FAX_QUEUE_REJECTING_JOBS
@@ -2035,7 +2035,7 @@ class DevMgr5(Ui_MainWindow_Derived, Ui_MainWindow, QMainWindow):
             else:
                 result, result_str = cups.cups_operation(cups.accept, GUI_MODE, 'qt4', self, self.cur_printer)
                 if result == cups.IPP_OK:
-                    if self.cur_device.device_type == DEVICE_TYPE_PRINTER:
+                    if hasattr(self.cur_device,'device_type') and self.cur_device.device_type == DEVICE_TYPE_PRINTER:
                         e = EVENT_PRINTER_QUEUE_ACCEPTING_JOBS
                     else:
                         e = EVENT_FAX_QUEUE_ACCEPTING_JOBS
@@ -2061,7 +2061,7 @@ class DevMgr5(Ui_MainWindow_Derived, Ui_MainWindow, QMainWindow):
                 cups.releaseCupsInstance()
             else:
                 self.updatePrintController()
-                if self.cur_device.device_type == DEVICE_TYPE_PRINTER:
+                if hasattr(self.cur_device,'device_type') and self.cur_device.device_type == DEVICE_TYPE_PRINTER:
                     e = EVENT_PRINTER_QUEUE_SET_AS_DEFAULT
                 else:
                     e = EVENT_FAX_QUEUE_SET_AS_DEFAULT
-- 
2.49.0

