From 16af20cc70f963680b3cad96c1704f53ef777a9c Mon Sep 17 00:00:00 2001
From: Pierre-Yves <pyu@riseup.net>
Date: Fri, 3 Aug 2018 16:54:50 +0200
Subject: [PATCH] Ensure sudo use on Solus

Signed-off-by: Pierre-Yves <pyu@riseup.net>
---
 base/password.py | 1 +
 1 file changed, 1 insertion(+)

diff --git a/base/password.py b/base/password.py
index cebc5cc..a7ddcde 100644
--- a/base/password.py
+++ b/base/password.py
@@ -34,6 +34,7 @@ PASSWORD_RETRY_COUNT = 3
 
 AUTH_TYPES = {'mepis': 'su',
               'debian': 'su',
+              'solus': 'sudo',
               'suse': 'su',
               'mandriva': 'su',
               'fedora': 'su',
-- 
2.18.0

