From 7e139126cdbc2ce2b81c170c3fb91fa53afac805 Mon Sep 17 00:00:00 2001
From: Tracey Clark <tracey_dev@tlcnet.info>
Date: Sat, 29 Apr 2023 16:25:49 -0500
Subject: [PATCH] Ensure-sudo-use-on-Solus

---
 base/utils.py | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/base/utils.py b/base/utils.py
index 25546f5..26c8d4a 100644
--- a/base/utils.py
+++ b/base/utils.py
@@ -2616,10 +2616,10 @@ def readAuthType():
         authType = 'su'
         user = os.getenv('USER')
         try:
-            members = getgrnam('wheel').gr_mem
+            members = getgrnam('sudo').gr_mem
         except KeyError:
             try:
-                members = getgrnam('sudo').gr_mem
+                members = getgrnam('wheel').gr_mem
             except:
                 log.warn(" can not get user group " )
                 return authType     
-- 
2.40.1

