diff -Pdpru hplip-3.15.11.orig/io/hpmud/jd.c hplip-3.15.11/io/hpmud/jd.c
--- hplip-3.15.11.orig/io/hpmud/jd.c	2015-11-12 11:29:02.000000000 +0100
+++ hplip-3.15.11/io/hpmud/jd.c	2015-11-19 16:49:59.902956345 +0100
@@ -31,6 +31,7 @@
 #endif
 
 #include <signal.h>
+#include <stdlib.h>
 #include "hpmud.h"
 #include "hpmudi.h"
 
diff -Pdpru hplip-3.15.11.orig/protocol/discovery/mdns.c hplip-3.15.11/protocol/discovery/mdns.c
--- hplip-3.15.11.orig/protocol/discovery/mdns.c	2015-11-12 11:29:01.000000000 +0100
+++ hplip-3.15.11/protocol/discovery/mdns.c	2015-11-19 16:46:02.785546885 +0100
@@ -24,7 +24,11 @@
  Author: Sanjay Kumar
  \*****************************************************************************/
 
-//#include <stdio.h>
+#ifndef _GNU_SOURCE
+#define _GNU_SOURCE
+#endif
+
+#include <stdio.h>
 #include <string.h>
 #include <syslog.h>
 #include <sys/socket.h>
@@ -32,6 +36,7 @@
 #include <netdb.h>
 #include <arpa/inet.h>
 #include "mdns.h"
+#include <unistd.h>
 
 /* Convert "www.google.com" to "3www6google3com". */
 static int mdns_convert_name_to_dns(const char *name, int name_size, char *dns_name)
diff -Pdpru hplip-3.15.11.orig/protocol/hp_ipp.c hplip-3.15.11/protocol/hp_ipp.c
--- hplip-3.15.11.orig/protocol/hp_ipp.c	2015-11-12 11:29:01.000000000 +0100
+++ hplip-3.15.11/protocol/hp_ipp.c	2015-11-19 16:45:57.890095775 +0100
@@ -18,6 +18,9 @@ Boston, MA 02110-1301, USA.
 
 \******************************************************************************/
 
+#ifndef _GNU_SOURCE
+#define _GNU_SOURCE
+#endif
 
 #include <cups/cups.h>
 #include <cups/language.h>
diff -Pdpru hplip-3.15.11.orig/scan/sane/bb_ledm.c hplip-3.15.11/scan/sane/bb_ledm.c
--- hplip-3.15.11.orig/scan/sane/bb_ledm.c	2015-11-12 11:28:47.000000000 +0100
+++ hplip-3.15.11/scan/sane/bb_ledm.c	2015-11-19 16:48:12.798968122 +0100
@@ -27,6 +27,7 @@
 # include "xml.h"
 # include <stdlib.h>
 
+# include <unistd.h>
 # include <stdint.h>
 
 # define _STRINGIZE(x) #x
diff -Pdpru hplip-3.15.11.orig/scan/sane/sclpml.c hplip-3.15.11/scan/sane/sclpml.c
--- hplip-3.15.11.orig/scan/sane/sclpml.c	2015-11-12 11:28:47.000000000 +0100
+++ hplip-3.15.11/scan/sane/sclpml.c	2015-11-19 16:48:16.290576562 +0100
@@ -25,11 +25,16 @@
 
 \************************************************************************************/
 
+#ifndef _GNU_SOURCE
+#define _GNU_SOURCE
+#endif
+
 #include <stdio.h>
 #include <string.h>
 #include <unistd.h>
 #include <fcntl.h>
 #include <string.h>
+#include <sys/time.h>
 #include "sane.h"
 #include "saneopts.h"
 #include "common.h"

