From f3560f22c7f9a114f8cb120a7261ef56b41b982e Mon Sep 17 00:00:00 2001
From: Pierre-Yves <pyu@riseup.net>
Date: Mon, 5 Jun 2017 23:26:23 +0200
Subject: [PATCH] disable upgrade

Signed-off-by: Pierre-Yves <pyu@riseup.net>
---
 upgrade.py | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/upgrade.py b/upgrade.py
index ad67b24..4e77c69 100755
--- a/upgrade.py
+++ b/upgrade.py
@@ -207,6 +207,9 @@ if os.geteuid() == 0:
 if os.getenv("HPLIP_DEBUG"):
     log.set_level('debug')
 
+log.error("HPLIP upgrade is disabled by Solus for security reasons. New versions are distributed via the Software Center")
+clean_exit(1)
+
 for o, a in opts:
     if o in ('-h', '--help'):
         usage()
-- 
2.13.0

