diff --git a/base/module.py b/base/module.py
index 04ac5b8..56be168 100644
--- a/base/module.py
+++ b/base/module.py
@@ -301,6 +301,9 @@ class Module(object):
             if UI_TOOLKIT_QT4 in self.supported_ui_toolkits and UI_TOOLKIT_QT4 in self.installed_ui_toolkits:
                 long_params.extend(['qt4', 'use-qt4'])
 
+            if UI_TOOLKIT_QT5 in self.supported_ui_toolkits and UI_TOOLKIT_QT5 in self.installed_ui_toolkits:
+                long_params.extend(['qt5', 'use-qt5'])
+
         if extra_params is not None:
             params = ''.join([params, extra_params])
 
