diff --git a/installer/pluginhandler.py b/installer/pluginhandler.py
index c2e2a94..d10d1b9 100755
--- a/installer/pluginhandler.py
+++ b/installer/pluginhandler.py
@@ -36,7 +36,7 @@ def get_checksum(s):
 
 
 PLUGIN_STATE_FILE = '/var/lib/hp/hplip.state'
-PLUGIN_FALLBACK_LOCATION = 'https://developers.hp.com/sites/default/files/'
+PLUGIN_FALLBACK_LOCATION = 'https://developers.hp.com/sites/default/files/2026-05/'
