diff --git a/CMakeLists.txt b/CMakeLists.txt
index 831a61a0..ac558d6d 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -335,7 +335,6 @@ add_custom_target(generate_helper_files DEPENDS
     vk_dispatch_table_helper.h
     vk_extension_helper.h
     vk_typemap_helper.h
-    spirv_tools_commit_id.h
     )
 
 # Rules to build generated helper files
@@ -349,7 +348,6 @@ run_vk_xml_generate(helper_file_generator.py vk_enum_string_helper.h)
 run_vk_xml_generate(helper_file_generator.py vk_object_types.h)
 run_vk_xml_generate(helper_file_generator.py vk_extension_helper.h)
 run_vk_xml_generate(helper_file_generator.py vk_typemap_helper.h)
-run_external_revision_generate(${EXTERNAL_SOURCE_ROOT}/glslang/External/spirv-tools SPIRV_TOOLS_COMMIT_ID spirv_tools_commit_id.h)
 
 if(NOT WIN32)
     include(GNUInstallDirs)
