--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -368,10 +368,10 @@ CONFIGURE_FILE(${PROJECT_SOURCE_DIR}/con
 
 # export targets:
 INSTALL(EXPORT vigra-targets
-        DESTINATION lib${LIB_SUFFIX}/vigra)
+        DESTINATION lib${LIB_SUFFIX}/cmake/vigra)
 INSTALL(FILES ${PROJECT_BINARY_DIR}/lib/vigra/CMake/VigraConfig.cmake
             ${PROJECT_BINARY_DIR}/lib/vigra/CMake/VigraConfigVersion.cmake
-        DESTINATION lib${LIB_SUFFIX}/vigra)
+        DESTINATION lib${LIB_SUFFIX}/cmake/vigra)
 EXPORT(TARGETS vigraimpex FILE vigra-targets.cmake)
 
 ##################################################
