From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Evan Maddock <maddock.evan@vivaldi.net>
Date: Wed, 20 Mar 2024 19:39:00 -0400
Subject: [PATCH] Use C++17

Signed-off-by: Evan Maddock <maddock.evan@vivaldi.net>
---
 ricochet.pro | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/ricochet.pro b/ricochet.pro
index 03ac281..0237805 100644
--- a/ricochet.pro
+++ b/ricochet.pro
@@ -36,7 +36,7 @@ lessThan(QT_MAJOR_VERSION,5)|lessThan(QT_MINOR_VERSION,1) {
 TARGET = ricochet
 TEMPLATE = app
 QT += core gui network quick widgets multimedia
-CONFIG += c++11
+CONFIG += c++17
 
 VERSION = 1.1.4
 
