From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Reilly Brogan <reilly@reillybrogan.com>
Date: Wed, 26 Jun 2024 22:34:25 -0500
Subject: [PATCH] Fix compat with MINIUPNPC API Version 18

---
 extra/upnpc.cpp | 6 ++++--
 1 file changed, 4 insertions(+), 2 deletions(-)

diff --git a/extra/upnpc.cpp b/extra/upnpc.cpp
index 46bad852..f7537f4a 100644
--- a/extra/upnpc.cpp
+++ b/extra/upnpc.cpp
@@ -52,9 +52,11 @@ bool UPnPc::init()
 
     if (!devices)
         return false;
-
+#if (MINIUPNPC_API_VERSION >= 18)
+    bool ret = UPNP_GetValidIGD(devices, &urls, &data, nullptr, 0, nullptr, 0);
+#else
     bool ret = UPNP_GetValidIGD(devices, &urls, &data, nullptr, 0);
-
+#endif
     freeUPNPDevlist(devices);
 
     return ret;
