From b74dcdd3f6b05522388729141c29286829c302fc Mon Sep 17 00:00:00 2001
From: Ian Kent <raven@themaw.net>
Date: Fri, 28 Feb 2020 09:16:06 +0800
Subject: [PATCH 06/13] autofs-5.1.6 - correct fsf address

---
 CHANGELOG | 1 +
 COPYING   | 4 ++--
 COPYRIGHT | 6 +++---
 3 files changed, 6 insertions(+), 5 deletions(-)

diff --git a/CHANGELOG b/CHANGELOG
index 3eb3409..f53f9ad 100644
--- a/CHANGELOG
+++ b/CHANGELOG
@@ -4,6 +4,7 @@ xx/xx/2020 autofs-5.1.7
 - fix program map multi-mount lookup after mount fail.
 - fix browse dir not re-created on symlink expire.
 - fix a regression with map instance lookup.
+- correct fsf address.
 
 07/10/2019 autofs-5.1.6
 - support strictexpire mount option.
diff --git a/COPYING b/COPYING
index 60549be..66885ff 100644
--- a/COPYING
+++ b/COPYING
@@ -2,7 +2,7 @@
 		       Version 2, June 1991
 
  Copyright (C) 1989, 1991 Free Software Foundation, Inc.
-                       59 Temple Place, Suite 330, Boston, MA  02111-1307  USA
+                       51 Franklin Street, Fifth Floor, Boston, MA 02110-1301, USA
  Everyone is permitted to copy and distribute verbatim copies
  of this license document, but changing it is not allowed.
 
@@ -305,7 +305,7 @@ the "copyright" line and a pointer to where the full notice is found.
 
     You should have received a copy of the GNU General Public License
     along with this program; if not, write to the Free Software
-    Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA
+    Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA
 
 
 Also add information on how to contact you by electronic and paper mail.
diff --git a/COPYRIGHT b/COPYRIGHT
index 37bd5d8..b0dc106 100644
--- a/COPYRIGHT
+++ b/COPYRIGHT
@@ -4,9 +4,9 @@ For all software in this distribution unless otherwise indicated:
 
    This program is free software; you can redistribute it and/or modify
    it under the terms of the GNU General Public License as published by
-   the Free Software Foundation, Inc., 675 Mass Ave, Cambridge MA 02139,
-   USA; either version 2 of the License, or (at your option) any later
-   version.
+   the Free Software Foundation, Inc., 51 Franklin Street, Fifth Floor,
+   Boston, MA  02110-1301, USA; either version 2 of the License, or
+   (at your option) any later version.
 
    This program is distributed in the hope that it will be useful,
    but WITHOUT ANY WARRANTY; without even the implied warranty of
-- 
2.27.0

