From 1d90510ecf89537675459b071481f183afee8901 Mon Sep 17 00:00:00 2001
From: Thomas Staudinger <Staudi.Kaos@gmail.com>
Date: Thu, 23 Jun 2022 19:56:23 +0200
Subject: [PATCH] Include <string>

---
 gold/errors.h | 1 +
 1 file changed, 1 insertion(+)

diff --git a/gold/errors.h b/gold/errors.h
index c26b5586..d1322029 100644
--- a/gold/errors.h
+++ b/gold/errors.h
@@ -26,6 +26,7 @@
 #include <cstdarg>
 
 #include "gold-threads.h"
+#include <string>
 
 namespace gold
 {
-- 
2.35.3

