From 91ef3f0859c700f70b7c9d2d756c71d00a0c5a6a Mon Sep 17 00:00:00 2001
From: Gavin Zhao <git@gzgz.dev>
Date: Mon, 25 Dec 2023 09:39:12 -0500
Subject: [PATCH] Use fallback libraries for unoptimize architectures

Signed-off-by: Gavin Zhao <git@gzgz.dev>
---
 Tensile/TensileCreateLibrary.py | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/Tensile/TensileCreateLibrary.py b/Tensile/TensileCreateLibrary.py
index 9aa0f02..31b2de5 100644
--- a/Tensile/TensileCreateLibrary.py
+++ b/Tensile/TensileCreateLibrary.py
@@ -1083,6 +1083,11 @@ def generateLogicDataAndSolutions(logicFiles, args):
       for key, value in masterLibraries.items():
         if key != "fallback":
           value.merge(deepcopy(masterLibraries["fallback"]))
+      for architectureName in parseArchitecturesFromArgs(args.Architecture, True):
+        if architectureName not in masterLibraries:
+          print("Using fallback for arch: " + architectureName)
+          masterLibraries[architectureName] = deepcopy(masterLibraries["fallback"])
+          masterLibraries[architectureName].version = args.version
 
       masterLibraries.pop("fallback")
 
-- 
2.43.0

