From d8fb0e04b3a721ff5f500ae37e1f3a0238f40488 Mon Sep 17 00:00:00 2001
From: Gavin Zhao <git@gzgz.dev>
Date: Thu, 20 Nov 2025 03:39:10 -0500
Subject: [PATCH] [tensile] Extend ISA compatibility by coercing arches

---
 .../Source/lib/source/hip/HipHardware.cpp     | 23 +++++++++++++++++++
 1 file changed, 23 insertions(+)

diff --git a/shared/tensile/Tensile/Source/lib/source/hip/HipHardware.cpp b/shared/tensile/Tensile/Source/lib/source/hip/HipHardware.cpp
index 411d7c5751..e4989f3571 100644
--- a/shared/tensile/Tensile/Source/lib/source/hip/HipHardware.cpp
+++ b/shared/tensile/Tensile/Source/lib/source/hip/HipHardware.cpp
@@ -24,6 +24,8 @@
  *
  *******************************************************************************/
 
+#include <string>
+#include <cstring>
 #include <Tensile/AMDGPU.hpp>
 #include <Tensile/hip/HipHardware.hpp>
 #include <Tensile/hip/HipUtils.hpp>
@@ -71,6 +73,27 @@ namespace Tensile
                                           deviceId));
             }
 #endif
+            std::string archName(prop.gcnArchName);
+            size_t pos = std::string::npos;
+            if((pos = archName.find("gfx103")) != std::string::npos)
+            {
+                std::strcpy(prop.gcnArchName, "gfx1030");
+            }
+            else if((pos = archName.find("gfx101")) != std::string::npos)
+            {
+                std::strcpy(prop.gcnArchName, "gfx1010");
+            }
+            else if((pos = archName.find("gfx90")) != std::string::npos)
+            {
+                constexpr int cmpIdx = std::char_traits<char>::length("gfx90");
+                if (pos + cmpIdx < archName.size())
+                {
+                    if(archName.at(pos + cmpIdx) == '2' || archName.at(pos + cmpIdx) == '9' || archName.at(pos + cmpIdx) == 'c')
+                    {
+                        std::strcpy(prop.gcnArchName, "gfx900");
+                    }
+                }
+            }
 
             return GetDevice(prop);
         }
-- 
2.51.2

