From ed2403a39c3c3d619e97558d53844e7cc4dcff40 Mon Sep 17 00:00:00 2001
From: James Lee <jamesl33info@gmail.com>
Date: Tue, 11 Dec 2018 00:41:13 +0000
Subject: [PATCH] fix hunspell 1.7.0

---
 lisp/textmodes/ispell.el | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/lisp/textmodes/ispell.el b/lisp/textmodes/ispell.el
index 3674a7080b..4591986493 100644
--- a/lisp/textmodes/ispell.el
+++ b/lisp/textmodes/ispell.el
@@ -1111,7 +1111,7 @@ dictionary from that list was found."
 				 null-device
 				 t
 				 nil
-				 "-D")
+				 "-D" null-device)
 	    (buffer-string))
 	  "[\n\r]+"
 	  t))
-- 
2.19.2

