From 947d0b17533baa8fd09a9b9dcf7ac607eb9f7a0f Mon Sep 17 00:00:00 2001
From: Joshua Strobl <joshua@streambits.io>
Date: Thu, 18 Jun 2020 17:42:33 +0300
Subject: [PATCH 1/1] Update libplist and libusbmuxd pkgconfigs

---
 CMakeLists.txt | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 6b5c9276..3ae267c1 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -127,8 +127,8 @@ pkg_check_modules(CHROMAPRINT libchromaprint)
 pkg_check_modules(LIBGPOD libgpod-1.0>=0.7.92)
 pkg_check_modules(LIBMTP libmtp>=1.0)
 pkg_check_modules(LIBIMOBILEDEVICE libimobiledevice-1.0)
-pkg_check_modules(LIBUSBMUXD libusbmuxd)
-pkg_check_modules(LIBPLIST libplist)
+pkg_check_modules(LIBUSBMUXD libusbmuxd-2.0)
+pkg_check_modules(LIBPLIST libplist-2.0)
 find_package(Gettext)
 find_package(FFTW3)
 
-- 
2.27.0

