diff --git a/Adwaita-blue/index.theme b/Adwaita-blue/index.theme
index 46550d6..fb811a8 100644
--- a/Adwaita-blue/index.theme
+++ b/Adwaita-blue/index.theme
@@ -2,7 +2,7 @@
 Name=Adwaita-blue
 Comment=The Only One
 Example=folder
-Inherits=Adwaita,AdwaitaLegacy,hicolor
+Inherits=MoreWaita,Adwaita,AdwaitaLegacy,hicolor
 Hidden=true
 
 # KDE Specific Stuff
diff --git a/Adwaita-brown/index.theme b/Adwaita-brown/index.theme
index 65e7926..eb0d371 100644
--- a/Adwaita-brown/index.theme
+++ b/Adwaita-brown/index.theme
@@ -2,7 +2,7 @@
 Name=Adwaita-brown
 Comment=The Only One
 Example=folder
-Inherits=Adwaita,Adwaita-blue,AdwaitaLegacy,hicolor
+Inherits=MoreWaita,Adwaita,Adwaita-blue,AdwaitaLegacy,hicolor
 Hidden=true
 
 # KDE Specific Stuff
diff --git a/Adwaita-green/index.theme b/Adwaita-green/index.theme
index 01f16b7..4c8e913 100644
--- a/Adwaita-green/index.theme
+++ b/Adwaita-green/index.theme
@@ -2,7 +2,7 @@
 Name=Adwaita-green
 Comment=The Only One
 Example=folder
-Inherits=Adwaita,Adwaita-blue,AdwaitaLegacy,hicolor
+Inherits=MoreWaita,Adwaita,Adwaita-blue,AdwaitaLegacy,hicolor
 Hidden=true
 
 # KDE Specific Stuff
diff --git a/Adwaita-orange/index.theme b/Adwaita-orange/index.theme
index 0d247b7..0bf75cf 100644
--- a/Adwaita-orange/index.theme
+++ b/Adwaita-orange/index.theme
@@ -2,7 +2,7 @@
 Name=Adwaita-orange
 Comment=The Only One
 Example=folder
-Inherits=Adwaita,Adwaita-blue,AdwaitaLegacy,hicolor
+Inherits=MoreWaita,Adwaita,Adwaita-blue,AdwaitaLegacy,hicolor
 Hidden=true
 
 # KDE Specific Stuff
diff --git a/Adwaita-pink/index.theme b/Adwaita-pink/index.theme
index b905bdc..9f8a392 100644
--- a/Adwaita-pink/index.theme
+++ b/Adwaita-pink/index.theme
@@ -2,7 +2,7 @@
 Name=Adwaita-pink
 Comment=The Only One
 Example=folder
-Inherits=Adwaita,Adwaita-blue,AdwaitaLegacy,hicolor
+Inherits=MoreWaita,Adwaita,Adwaita-blue,AdwaitaLegacy,hicolor
 Hidden=true
 
 # KDE Specific Stuff
diff --git a/Adwaita-purple/index.theme b/Adwaita-purple/index.theme
index 6b09a17..427e84a 100644
--- a/Adwaita-purple/index.theme
+++ b/Adwaita-purple/index.theme
@@ -2,7 +2,7 @@
 Name=Adwaita-purple
 Comment=The Only One
 Example=folder
-Inherits=Adwaita,Adwaita-blue,AdwaitaLegacy,hicolor
+Inherits=MoreWaita,Adwaita,Adwaita-blue,AdwaitaLegacy,hicolor
 Hidden=true
 
 # KDE Specific Stuff
diff --git a/Adwaita-red/index.theme b/Adwaita-red/index.theme
index 5ca462e..5858ff6 100644
--- a/Adwaita-red/index.theme
+++ b/Adwaita-red/index.theme
@@ -2,7 +2,7 @@
 Name=Adwaita-red
 Comment=The Only One
 Example=folder
-Inherits=Adwaita,Adwaita-blue,AdwaitaLegacy,hicolor
+Inherits=MoreWaita,Adwaita,Adwaita-blue,AdwaitaLegacy,hicolor
 Hidden=true
 
 # KDE Specific Stuff
diff --git a/Adwaita-slate/index.theme b/Adwaita-slate/index.theme
index 9c91b4b..b76a496 100644
--- a/Adwaita-slate/index.theme
+++ b/Adwaita-slate/index.theme
@@ -2,7 +2,7 @@
 Name=Adwaita-slate
 Comment=The Only One
 Example=folder
-Inherits=Adwaita,Adwaita-blue,AdwaitaLegacy,hicolor
+Inherits=MoreWaita,Adwaita,Adwaita-blue,AdwaitaLegacy,hicolor
 Hidden=true
 
 # KDE Specific Stuff
diff --git a/Adwaita-teal/index.theme b/Adwaita-teal/index.theme
index 01e67c3..5bc94eb 100644
--- a/Adwaita-teal/index.theme
+++ b/Adwaita-teal/index.theme
@@ -2,7 +2,7 @@
 Name=Adwaita-teal
 Comment=The Only One
 Example=folder
-Inherits=Adwaita,Adwaita-blue,AdwaitaLegacy,hicolor
+Inherits=MoreWaita,Adwaita,Adwaita-blue,AdwaitaLegacy,hicolor
 Hidden=true
 
 # KDE Specific Stuff
diff --git a/Adwaita-yellow/index.theme b/Adwaita-yellow/index.theme
index cba6024..01fbdab 100644
--- a/Adwaita-yellow/index.theme
+++ b/Adwaita-yellow/index.theme
@@ -2,7 +2,7 @@
 Name=Adwaita-yellow
 Comment=The Only One
 Example=folder
-Inherits=Adwaita,Adwaita-blue,AdwaitaLegacy,hicolor
+Inherits=MoreWaita,Adwaita,Adwaita-blue,AdwaitaLegacy,hicolor
 Hidden=true
 
 # KDE Specific Stuff
