diff -rupN --no-dereference qwt-6.3.0/designer/designer.pro qwt-6.3.0-new/designer/designer.pro
--- qwt-6.3.0/designer/designer.pro	2024-05-08 09:00:29.909579246 +0200
+++ qwt-6.3.0-new/designer/designer.pro	2026-04-04 00:45:00.397986349 +0200
@@ -98,7 +98,7 @@ contains(QWT_CONFIG, QwtDesigner) {
         # compile the path for finding the Qwt library
         # into the plugin. Not supported on Windows !
 
-        QMAKE_RPATHDIR *= $${QWT_INSTALL_LIBS}
+        #QMAKE_RPATHDIR *= $${QWT_INSTALL_LIBS}
         qwtAddLibrary($${QWT_OUT_ROOT}/lib, qwt)
 
         contains(QWT_CONFIG, QwtDll) {
