diff -rupN --no-dereference qwt-6.3.0/qwtbuild.pri qwt-6.3.0-new/qwtbuild.pri
--- qwt-6.3.0/qwtbuild.pri	2024-05-08 09:00:29.925579632 +0200
+++ qwt-6.3.0-new/qwtbuild.pri	2026-04-04 00:45:00.156080513 +0200
@@ -14,7 +14,7 @@
 CONFIG           += qt
 CONFIG           += warn_on
 CONFIG           += no_keywords
-CONFIG           += silent
+# CONFIG           += silent
 CONFIG           -= depend_includepath
 
 # CONFIG += sanitize
diff -rupN --no-dereference qwt-6.3.0/qwtconfig.pri qwt-6.3.0-new/qwtconfig.pri
--- qwt-6.3.0/qwtconfig.pri	2024-05-08 09:00:29.929579728 +0200
+++ qwt-6.3.0-new/qwtconfig.pri	2026-04-04 00:45:00.156360747 +0200
@@ -19,7 +19,7 @@ QWT_VERSION      = $${QWT_VER_MAJ}.$${QW
 QWT_INSTALL_PREFIX = $$[QT_INSTALL_PREFIX]
 
 unix {
-    QWT_INSTALL_PREFIX    = /usr/local/qwt-$$QWT_VERSION
+    # QWT_INSTALL_PREFIX    = /usr/local/qwt-$$QWT_VERSION
     # QWT_INSTALL_PREFIX = /usr/local/qwt-$$QWT_VERSION-qt-$$QT_VERSION
 }
 
@@ -28,9 +28,9 @@ win32 {
     # QWT_INSTALL_PREFIX = C:/Qwt-$$QWT_VERSION-qt-$$QT_VERSION
 }
 
-QWT_INSTALL_DOCS      = $${QWT_INSTALL_PREFIX}/doc
-QWT_INSTALL_HEADERS   = $${QWT_INSTALL_PREFIX}/include
-QWT_INSTALL_LIBS      = $${QWT_INSTALL_PREFIX}/lib
+QWT_INSTALL_DOCS      = $$[QT_INSTALL_DOCS]/html
+QWT_INSTALL_HEADERS   = $$[QT_INSTALL_HEADERS]/qwt
+QWT_INSTALL_LIBS      = $$[QT_INSTALL_LIBS]
 
 ######################################################################
 # Designer plugin
@@ -42,7 +42,8 @@ QWT_INSTALL_LIBS      = $${QWT_INSTALL_P
 # runtime environment of designer/creator.
 ######################################################################
 
-QWT_INSTALL_PLUGINS   = $${QWT_INSTALL_PREFIX}/plugins/designer
+#QWT_INSTALL_PLUGINS   = $${QWT_INSTALL_PREFIX}/plugins/designer
+QWT_INSTALL_PLUGINS   = $$[QT_INSTALL_PLUGINS]/designer
 
 # linux distributors often organize the Qt installation
 # their way and QT_INSTALL_PREFIX doesn't offer a good
@@ -63,8 +64,8 @@ QWT_INSTALL_PLUGINS   = $${QWT_INSTALL_P
 # with every Qt upgrade.
 ######################################################################
 
-QWT_INSTALL_FEATURES  = $${QWT_INSTALL_PREFIX}/features
-# QWT_INSTALL_FEATURES  = $$[QT_INSTALL_PREFIX]/features
+#QWT_INSTALL_FEATURES  = $${QWT_INSTALL_PREFIX}/features
+QWT_INSTALL_FEATURES  = $$[QMAKE_MKSPECS]/features
 
 ######################################################################
 # Build the static/shared libraries.
