diff --git a/src/copyout.c b/src/copyout.c
index 1f0987a..0e51336 100644
--- a/src/copyout.c
+++ b/src/copyout.c
@@ -21,6 +21,7 @@
 
 #include <stdio.h>
 #include <sys/types.h>
+#include <sys/sysmacros.h>
 #include <sys/stat.h>
 #include "filetypes.h"
 #include "cpiohdr.h"
diff --git a/src/copypass.c b/src/copypass.c
index c5a9899..799b2c2 100644
--- a/src/copypass.c
+++ b/src/copypass.c
@@ -21,6 +21,7 @@
 
 #include <stdio.h>
 #include <sys/types.h>
+#include <sys/sysmacros.h>
 #include <sys/stat.h>
 #include "filetypes.h"
 #include "cpiohdr.h"
diff --git a/src/util.c b/src/util.c
index 6ff6032..5df68eb 100644
--- a/src/util.c
+++ b/src/util.c
@@ -22,6 +22,7 @@
 #include <stdio.h>
 #include <sys/types.h>
 #include <sys/stat.h>
+#include <sys/sysmacros.h>
 #include "cpiohdr.h"
 #include "dstring.h"
 #include "extern.h"
