diff --git a/plugins/notify.pm b/plugins/notify.pm
index 7f192c5..6356bc1 100644
--- a/plugins/notify.pm
+++ b/plugins/notify.pm
@@ -29,7 +29,7 @@ my ($Daemon_name,$can_actions,$can_body);
 sub Start
 {	$notify=Gtk2::Notify->new('empty','empty');
 	#$notify->set_urgency('low');
-	#$notify->set_category('music'); #is there a standard category for that ?
+	$notify->set_category('x-gnome.music'); #is there a standard category for that ?
 	my ($name, $vendor, $version, $spec_version)= Gtk2::Notify->get_server_info;
 	$Daemon_name= "$name $version ($vendor)";
 	my @caps = Gtk2::Notify->get_server_caps;
