From 042601544636f8d22813795614b1c5b529b0848b Mon Sep 17 00:00:00 2001
From: Lennart Poettering <lennart@poettering.net>
Date: Tue, 8 Aug 2023 14:18:38 +0200
Subject: [PATCH 1/7] vconsole-setup: port to main-func.h boilerplate

---
 src/vconsole/vconsole-setup.c | 13 +++++++------
 1 file changed, 7 insertions(+), 6 deletions(-)

diff --git a/src/vconsole/vconsole-setup.c b/src/vconsole/vconsole-setup.c
index 3b71afec36..9580c8d2c9 100644
--- a/src/vconsole/vconsole-setup.c
+++ b/src/vconsole/vconsole-setup.c
@@ -29,6 +29,7 @@
 #include "locale-util.h"
 #include "lock-util.h"
 #include "log.h"
+#include "main-func.h"
 #include "proc-cmdline.h"
 #include "process-util.h"
 #include "signal-util.h"
@@ -571,7 +572,7 @@ static int verify_source_vc(char **ret_path, const char *src_vc) {
         return TAKE_FD(fd);
 }
 
-int main(int argc, char **argv) {
+static int run(int argc, char **argv) {
         _cleanup_(context_done) Context c = {};
         _cleanup_free_ char *vc = NULL;
         _cleanup_close_ int fd = -EBADF;
@@ -588,7 +589,7 @@ int main(int argc, char **argv) {
         else
                 fd = find_source_vc(&vc, &idx);
         if (fd < 0)
-                return EXIT_FAILURE;
+                return fd;
 
         utf8 = is_locale_utf8();
 
@@ -597,10 +598,8 @@ int main(int argc, char **argv) {
         /* Take lock around the remaining operation to avoid being interrupted by a tty reset operation
          * performed for services with TTYVHangup=yes. */
         r = lock_generic(fd, LOCK_BSD, LOCK_EX);
-        if (r < 0) {
-                log_error_errno(r, "Failed to lock console: %m");
-                return EXIT_FAILURE;
-        }
+        if (r < 0)
+                return log_error_errno(r, "Failed to lock console: %m");
 
         (void) toggle_utf8_sysfs(utf8);
         (void) toggle_utf8_vc(vc, fd, utf8);
@@ -622,3 +621,5 @@ int main(int argc, char **argv) {
 
         return IN_SET(r, 0, EX_OSERR) && keyboard_ok ? EXIT_SUCCESS : EXIT_FAILURE;
 }
+
+DEFINE_MAIN_FUNCTION_WITH_POSITIVE_FAILURE(run);
-- 
2.43.0

