From 0957cf5f13066f828e8805db91ced0d56358eaf1 Mon Sep 17 00:00:00 2001
From: Peter O'Connor <sunnyflunk@serpentos.com>
Date: Mon, 21 Feb 2022 01:48:09 +0300
Subject: [PATCH] Don't add GLIBC_PRIVATE_DT_RELR symbol for RELR

Signed-off-by: Peter O'Connor <sunnyflunk@serpentos.com>
---
 bfd/elflink.c | 7 -------
 1 file changed, 7 deletions(-)

diff --git a/bfd/elflink.c b/bfd/elflink.c
index f8521426cad..7a2565e3df2 100644
--- a/bfd/elflink.c
+++ b/bfd/elflink.c
@@ -7018,13 +7018,6 @@ bfd_elf_size_dynamic_sections (bfd *output_bfd,
       if (sinfo.failed)
 	return false;
 
-      if (info->enable_dt_relr)
-	{
-	  elf_link_add_dt_relr_dependency (&sinfo);
-	  if (sinfo.failed)
-	    return false;
-	}
-
       if (elf_tdata (output_bfd)->verref == NULL)
 	s->flags |= SEC_EXCLUDE;
       else
-- 
2.34.1

