From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Reilly Brogan <reilly@reillybrogan.com>
Date: Fri, 13 Dec 2024 12:42:26 -0600
Subject: [PATCH] Proper stateless

---
 gprofng/src/Settings.cc | 21 +++++++++++++++------
 1 file changed, 15 insertions(+), 6 deletions(-)

diff --git a/gprofng/src/Settings.cc b/gprofng/src/Settings.cc
index 6d1d357a17..d62c5fdacb 100644
--- a/gprofng/src/Settings.cc
+++ b/gprofng/src/Settings.cc
@@ -413,13 +413,22 @@ Settings::read_rc (bool ipc_or_rdt_mode)
   rc_path = dbe_sprintf (NTXT ("%s/gprofng.rc"), sysconfdir);
   if (access (rc_path, R_OK | F_OK) != 0)
     {
-      StringBuilder sb;
-      sb.sprintf (GTXT ("Warning: Default gprofng.rc file (%s) missing; configuration error "), rc_path);
-      Emsg *m = new Emsg (CMSG_COMMENT, sb);
-      app->get_comments_queue ()->append (m);
+      free (rc_path);
+      // Read vendor file
+      const char *vendordir = getenv("GPROFNG_VENDORDIR");
+      if (vendordir == NULL)
+        vendordir = "/usr/share/defaults/gprofng";
+      rc_path = dbe_sprintf (NTXT ("%s/gprofng.rc"), vendordir);
+      if (access (rc_path, R_OK | F_OK) != 0)
+        {
+          StringBuilder sb;
+          sb.sprintf (GTXT ("Warning: Default gprofng.rc file (%s) missing; configuration error "), rc_path);
+          Emsg *m = new Emsg (CMSG_COMMENT, sb);
+          app->get_comments_queue ()->append (m);
+        }
+      else
+        set_rc (rc_path, false, app->get_comments_queue (), override);
     }
-  else
-    set_rc (rc_path, false, app->get_comments_queue (), override);
   free (rc_path);
   is_loexpand_default = true;
   if (str_printmode == NULL)
