diff -rup binutils.orig/etc/texi2pod.pl binutils-2.37/etc/texi2pod.pl
--- binutils.orig/etc/texi2pod.pl	2021-08-10 10:15:38.063964450 +0100
+++ binutils-2.37/etc/texi2pod.pl	2021-08-10 16:52:51.705688992 +0100
@@ -59,6 +59,8 @@ while ($_ = shift) {
 	    $flag = shift;
 	}
         push (@ipath, $flag);
+    } elsif (/^--no-split$/) {
+       # ignore option for makeinfo compatibility
     } elsif (/^-/) {
 	usage();
     } else {
