From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Reilly Brogan <reilly@reillybrogan.com>
Date: Sun, 8 Dec 2024 17:52:51 -0600
Subject: [PATCH] Ensure that edit-config works even if called from direct
 location

---
 system/edit-config | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/system/edit-config b/system/edit-config
index 3944810d2..96140673a 100755
--- a/system/edit-config
+++ b/system/edit-config
@@ -5,7 +5,7 @@
 
 set -e
 
-script_dir="$(CDPATH="" cd -- "$(dirname -- "$0")" && pwd -P)"
+script_dir="${NETDATA_USER_CONFIG_DIR:-/etc/netdata}"
 
 usage() {
   check_directories
