From 482ee914ebc011a3b5e22b09a8dedbef191097a9 Mon Sep 17 00:00:00 2001
From: Ikey Doherty <ikey@solusos.com>
Date: Fri, 23 Aug 2013 19:11:18 +0100
Subject: [PATCH] Fix race condition for concurrent builds

---
 Makefile.in |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/Makefile.in b/Makefile.in
index 328aea1..9705929 100644
--- a/Makefile.in
+++ b/Makefile.in
@@ -423,7 +423,7 @@ TAGS:
 ##  (AUTOMATICALLY UPDATED - DO NOT EDIT)
 ##
 
-$(LOBJS): Makefile
+$(LOBJS): pth_p.h Makefile
 
 # DO NOT REMOVE
 pth_attr.lo: pth_attr.c pth_p.h pth_vers.c pth.h pth_acdef.h pth_acmac.h
-- 
1.7.10.4

