--- a/lib/mhash_int.h
+++ b/lib/mhash_int.h
@@ -36,7 +36,6 @@ void *hmac_mhash_end(MHASH thread);
 
 /* Key generation functions */
 
-#define EMPTY_PROTOTYPES 1
 
 #if defined(EMPTY_PROTOTYPES)
 
@@ -63,6 +62,9 @@ mutils_error mhash_keygen_ext(keygenid algorithm, KEYGEN data,
 
 mutils_word8 *mhash_get_keygen_name(hashid type);
 mutils_word32 mhash_get_keygen_salt_size(keygenid type);
+mutils_boolean mhash_keygen_uses_salt(keygenid type);
+mutils_boolean mhash_keygen_uses_count(keygenid type);
+mutils_boolean mhash_keygen_uses_hash_algorithm(keygenid type);
 mutils_word32 mhash_keygen_count(void);
 mutils_boolean mhash_keygen_uses_salt(keygenid type);
 mutils_boolean mhash_keygen_uses_count(keygenid type);
