diff --git a/hack/install_catatonit.sh b/hack/install_catatonit.sh
index a35e349f5..2c2845c35 100755
--- a/hack/install_catatonit.sh
+++ b/hack/install_catatonit.sh
@@ -1,7 +1,7 @@
 #!/usr/bin/env bash
-BASE_PATH="/usr/libexec/podman"
+BASE_PATH=$DESTDIR"/usr/libexec/podman"
 CATATONIT_PATH="${BASE_PATH}/catatonit"
-CATATONIT_VERSION="v0.1.7"
+CATATONIT_VERSION="v0.2.0"
 set -e
 
 if [ -f $CATATONIT_PATH ] && [ -z "$1" ]; then
