From 64e6f47f43079b435f2a21ee9236a80c83578cf1 Mon Sep 17 00:00:00 2001
From: Reilly Brogan <reilly@reillybrogan.com>
Date: Fri, 29 Sep 2023 16:43:47 -0500
Subject: [PATCH] Add GPG 2.4

---
 meson.build | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/meson.build b/meson.build
index c1efbfb..7fe718b 100644
--- a/meson.build
+++ b/meson.build
@@ -26,7 +26,7 @@ config_h_dir = include_directories('.')
 # Dependencies
 min_glib_version = '2.44'
 min_gtk_version = '3.18'
-accepted_gpg_versions= ['1.2.0', '1.4.0', '2.0.0', '2.1.0', '2.2.0', '2.3.0']
+accepted_gpg_versions= ['1.2.0', '1.4.0', '2.0.0', '2.1.0', '2.2.0', '2.3.0', '2.4.0']
 min_gpgme_version = '1.0.0'
 
 libnautilus_extension = dependency('libnautilus-extension-4', version: '>= 43.rc', required: false)
-- 
2.42.0

