diff -ruNb a/src/session-child.c b/src/session-child.c
--- a/src/session-child.c	2013-04-23 23:54:53.000000000 +0100
+++ b/src/session-child.c	2013-06-05 22:12:04.347716417 +0100
@@ -13,7 +13,9 @@
 #include <grp.h>
 #include <glib.h>
 #include <security/pam_appl.h>
+#if 0
 #include <utmpx.h>
+#endif
 #include <sys/mman.h>
 
 #include "configuration.h"
@@ -581,6 +583,7 @@
     /* Wait for the command to complete (blocks) */
     if (child_pid > 0)
     {
+        #if 0
         /* Log to utmp */
         if (g_strcmp0 (class, XDG_SESSION_CLASS_GREETER) != 0)
         {
@@ -606,10 +609,11 @@
                 g_printerr ("Failed to write utmpx: %s\n", strerror (errno));
             endutxent ();
         }
-
+        #endif
         waitpid (child_pid, &return_code, 0);
         child_pid = 0;
 
+        #if 0
         /* Log to utmp */
         if (g_strcmp0 (class, XDG_SESSION_CLASS_GREETER) != 0)
         {
@@ -635,6 +639,7 @@
                 g_printerr ("Failed to write utmpx: %s\n", strerror (errno));
             endutxent ();
         }
+        #endif
     }
 
     /* Remove X authority */
