From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Reilly Brogan <reilly@reillybrogan.com>
Date: Wed, 6 Aug 2025 18:41:42 -0500
Subject: [PATCH] Don't build 32bit EFI stub

---
 meson.build | 8 --------
 1 file changed, 8 deletions(-)

diff --git a/meson.build b/meson.build
index 7ede6f7a96..d7790353e8 100644
--- a/meson.build
+++ b/meson.build
@@ -1929,14 +1929,6 @@ conf.set_quoted('EFI_MACHINE_TYPE_NAME', have ? efi_arch : '')
 
 efi_arch_alt = ''
 efi_cpu_family_alt = ''
-if have and efi_arch == 'x64' and cc.links('''
-                #include <limits.h>
-                int main(int argc, char *argv[]) {
-                        return __builtin_popcount(argc - CHAR_MAX);
-                }''', args : ['-m32', '-march=i686'], name : '32bit build possible')
-        efi_arch_alt = 'ia32'
-        efi_cpu_family_alt = 'x86'
-endif
 
 pefile = pymod.find_installation('python3', required: false, modules : ['pefile'])
 
