From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Reilly Brogan <reilly@reillybrogan.com>
Date: Wed, 16 Oct 2024 17:10:38 -0500
Subject: [PATCH] Remove /etc/os-release from tmpfiles

Signed-off-by: Reilly Brogan <reilly@reillybrogan.com>
---
 tmpfiles.d/etc.conf.in | 1 -
 1 file changed, 1 deletion(-)

diff --git a/tmpfiles.d/etc.conf.in b/tmpfiles.d/etc.conf.in
index 9c1ed67ab7..f194e6b992 100644
--- a/tmpfiles.d/etc.conf.in
+++ b/tmpfiles.d/etc.conf.in
@@ -7,7 +7,6 @@
 
 # See tmpfiles.d(5) for details.
 
-L /etc/os-release - - - - ../usr/lib/os-release
 L+ /etc/mtab - - - - ../proc/self/mounts
 {% if HAVE_SMACK_RUN_LABEL %}
 t /etc/mtab - - - - security.SMACK64=_
