From 40ee27a24b0fb55f9128fc6b0548390f22cb64f9 Mon Sep 17 00:00:00 2001
From: Reilly Brogan <reilly@reillybrogan.com>
Date: Tue, 11 Nov 2025 11:42:35 +0000
Subject: [PATCH 1/1] systemd: Support sysconfdir for system-sleep scripts

Signed-off-by: Reilly Brogan <reilly@reillybrogan.com>
---
 meson.build       | 1 +
 src/sleep/sleep.c | 1 +
 2 files changed, 2 insertions(+)

diff --git a/meson.build b/meson.build
index c289d7f53a..96dfdf25a1 100644
--- a/meson.build
+++ b/meson.build
@@ -308,6 +308,7 @@ conf.set_quoted('SYSTEM_GENERATOR_DIR',                       systemgeneratordir
 conf.set_quoted('SYSTEM_PRESET_DIR',                          systempresetdir)
 conf.set_quoted('SYSTEM_SHUTDOWN_PATH',                       systemshutdowndir)
 conf.set_quoted('SYSTEM_SLEEP_PATH',                          systemsleepdir)
+conf.set_quoted('SYSTEM_SLEEP_PATH_USER',                     pkgsysconfdir / 'system-sleep')
 conf.set_quoted('SYSTEM_SYSVINIT_PATH',                       sysvinit_path)
 conf.set_quoted('SYSTEM_SYSVRCND_PATH',                       sysvrcnd_path)
 conf.set_quoted('SYSTEM_SYSVRCLOCAL_PATH',                    sysvrclocal_path)
diff --git a/src/sleep/sleep.c b/src/sleep/sleep.c
index 3bfe479088..fedfa5ecda 100644
--- a/src/sleep/sleep.c
+++ b/src/sleep/sleep.c
@@ -247,6 +247,7 @@ static int execute(
                 NULL
         };
         static const char* const dirs[] = {
+                SYSTEM_SLEEP_PATH_USER,
                 SYSTEM_SLEEP_PATH,
                 NULL
         };
-- 
2.51.0

