From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Ikey Doherty <ikey@serpentos.com>
Date: Wed, 11 Sep 2024 01:48:11 +0000
Subject: [PATCH] tmpfiles: Allow nsswitch clobber at non-boot interval

Signed-off-by: Ikey Doherty <ikey@serpentos.com>
---
 tmpfiles.d/etc.conf.in | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/tmpfiles.d/etc.conf.in b/tmpfiles.d/etc.conf.in
index 9c1ed67ab7..284b6eb0ca 100644
--- a/tmpfiles.d/etc.conf.in
+++ b/tmpfiles.d/etc.conf.in
@@ -13,7 +13,7 @@ L+ /etc/mtab - - - - ../proc/self/mounts
 t /etc/mtab - - - - security.SMACK64=_
 {% endif %}
 C! /etc/locale.conf - - - -
-C! /etc/nsswitch.conf - - - -
+C /etc/nsswitch.conf - - - -
 C! /etc/vconsole.conf - - - -
 {% if HAVE_PAM %}
 C! /etc/pam.d - - - -
