From dd6ea30e83bd64de17469bea191970eacd2e0110 Mon Sep 17 00:00:00 2001
From: Joey Riches <josephriches@gmail.com>
Date: Wed, 10 May 2023 17:47:20 +0100
Subject: [PATCH 09/14] path-lookup: Include stateless xdg path by default

---
 src/basic/path-lookup.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/src/basic/path-lookup.c b/src/basic/path-lookup.c
index 7d158a8295..78867a9522 100644
--- a/src/basic/path-lookup.c
+++ b/src/basic/path-lookup.c
@@ -122,7 +122,8 @@ int xdg_user_dirs(char ***ret_config_dirs, char ***ret_data_dirs) {
         if (e)
                 config_dirs = strv_split(e, ":");
         else
-                config_dirs = strv_new("/etc/xdg");
+                config_dirs = strv_new("/usr/share/xdg",
+                                       "/etc/xdg");
         if (!config_dirs)
                 return -ENOMEM;
 
-- 
2.43.0

