From 4e63b2f5453d86a66a64d178136ce033e0ac8536 Mon Sep 17 00:00:00 2001
From: Reilly Brogan <reilly@reillybrogan.com>
Date: Wed, 13 Dec 2023 20:31:00 -0600
Subject: [PATCH 14/15] Solus: Ensure /var/log/journal is created

---
 tmpfiles.d/systemd.conf.in | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/tmpfiles.d/systemd.conf.in b/tmpfiles.d/systemd.conf.in
index d36f0694af..49582f542b 100644
--- a/tmpfiles.d/systemd.conf.in
+++ b/tmpfiles.d/systemd.conf.in
@@ -36,7 +36,7 @@ A+ /run/log/journal/%m - - - - d:group:wheel:r-x,group:wheel:r-X
 {% endif %}
 {% endif %}
 
-z /var/log/journal 2755 root systemd-journal - -
+d /var/log/journal 2755 root systemd-journal - -
 z /var/log/journal/%m 2755 root systemd-journal - -
 z /var/log/journal/%m/system.journal 0640 root systemd-journal - -
 {% if HAVE_ACL %}
-- 
2.47.0

