From 0a929c9792212dbfe644493ad940ffe41e3f0e3d Mon Sep 17 00:00:00 2001
From: Reilly Brogan <reilly@reillybrogan.com>
Date: Fri, 29 Sep 2023 13:31:04 -0500
Subject: [PATCH] Allow extra options to be passed to setup.py

---
 lang/python/Makefile.am | 1 +
 1 file changed, 1 insertion(+)

diff --git a/lang/python/Makefile.am b/lang/python/Makefile.am
index 68b98e8e..deaa571c 100644
--- a/lang/python/Makefile.am
+++ b/lang/python/Makefile.am
@@ -97,6 +97,7 @@ install-exec-local:
 	  --build-base="$$(basename "$${PYTHON}")-gpg" \
 	  install \
 	  --prefix "$(DESTDIR)$(prefix)" \
+	  $${SETUP_PY_EXTRA_OPTS:-} \
 	  --verbose ; \
 	done
 
-- 
2.42.0

