From bc597c78aae5075f8b5f7c106b583e9ec239d46d Mon Sep 17 00:00:00 2001
From: Reilly Brogan <reilly@reillybrogan.com>
Date: Fri, 29 Sep 2023 14:36:51 -0500
Subject: [PATCH] Disable generation of python2 bindings

---
 configure.ac | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/configure.ac b/configure.ac
index ae4f9648..95938b30 100644
--- a/configure.ac
+++ b/configure.ac
@@ -559,7 +559,7 @@ if test "$found_py" = "1"; then
 	if test "$found_py" = "1" -o "$found_py3" = "1"; then
 	  # Reset everything, so that we can look for another Python.
           m4_foreach([mym4pythonver],
-                     [[2.7],[3.4],[3.5],[3.6],[3.7],[3.8],[3.9],[3.10],
+                     [[3.4],[3.5],[3.6],[3.7],[3.8],[3.9],[3.10],
                       [3.11],[3.12],[all]],
            [unset PYTHON
 	    unset PYTHON_VERSION
-- 
2.42.0

