From e13216dcaa455e08368dcde256a6896d6e8918a1 Mon Sep 17 00:00:00 2001
From: Graeme Gott <graeme@gottcode.org>
Date: Fri, 14 Feb 2025 14:12:17 -0500
Subject: [PATCH] Fix menu not shown on correct monitor (Issue #154)

---
 panel-plugin/window.cpp | 6 +++++-
 1 file changed, 5 insertions(+), 1 deletion(-)

diff --git a/panel-plugin/window.cpp b/panel-plugin/window.cpp
index da17fb9..f4f48ad 100644
--- a/panel-plugin/window.cpp
+++ b/panel-plugin/window.cpp
@@ -524,6 +524,7 @@ void WhiskerMenu::Window::show(const Position position)
 	m_recent->get_view()->reload_icon_size();
 	m_applications->get_view()->reload_icon_size();
 
+	GdkMonitor* monitor_gdk = nullptr;
 	if (position == PositionAtButton)
 	{
 		// Wait up to half a second for auto-hidden panels to be shown
@@ -543,6 +544,9 @@ void WhiskerMenu::Window::show(const Position position)
 
 		// Fetch position
 		m_plugin->get_menu_position(&m_geometry.x, &m_geometry.y);
+
+		GtkWidget* widget = m_plugin->get_button();
+		monitor_gdk = gdk_display_get_monitor_at_window(gtk_widget_get_display(widget), gtk_widget_get_window(widget));
 	}
 	else
 	{
@@ -551,10 +555,10 @@ void WhiskerMenu::Window::show(const Position position)
 		GdkSeat* seat = gdk_display_get_default_seat(display);
 		GdkDevice* device = gdk_seat_get_pointer(seat);
 		gdk_device_get_position(device, nullptr, &m_geometry.x, &m_geometry.y);
+		monitor_gdk = gdk_display_get_monitor_at_point(display, m_geometry.x, m_geometry.y);
 	}
 
 	// Resize window if necessary, and also prevent it from being larger than screen
-	GdkMonitor* monitor_gdk = gdk_display_get_monitor_at_point(gdk_display_get_default(), m_geometry.x, m_geometry.y);
 #ifdef HAVE_GTK_LAYER_SHELL
 	if (gtk_layer_is_supported())
 	{
-- 
GitLab

