Index: b/tests/all_funcs
===================================================================
--- a/tests/all_funcs	2016-07-05 14:05:33.000000000 -0700
+++ b/tests/all_funcs	2016-09-28 16:22:20.921632541 -0700
@@ -1,8 +1,7 @@
 #!/bin/bash
 
 function proceed {
-  read -p "proceed ? [n] " response
-  if [ "$response" != "y" ]; then exit -1; fi
+    :
 }
 
 make clean tests_only EXTRA_CFLAGS='-DHASH_FUNCTION=HASH_BER'; proceed
