From b8487b6dcbb7a1264a9d0bf5352b72b3a09686f9 Mon Sep 17 00:00:00 2001
From: Joey Riches <josephriches@gmail.com>
Date: Thu, 26 Aug 2021 11:40:10 +0100
Subject: [PATCH 1/1] Add missing includes for gcc 11

---
 src/libs/common/MuleDebug.cpp | 1 +
 1 file changed, 1 insertion(+)

diff --git a/src/libs/common/MuleDebug.cpp b/src/libs/common/MuleDebug.cpp
index f04e375..f5c3cad 100644
--- a/src/libs/common/MuleDebug.cpp
+++ b/src/libs/common/MuleDebug.cpp
@@ -24,6 +24,7 @@
 //
 
 #include <cstdlib>			// Needed for std::abort()
+#include <exception>			// Needed for std::terminate()
 
 #include "config.h"			// Needed for HAVE_CXXABI and HAVE_EXECINFO
 
-- 
2.32.0

