From 9bcd20152bbeac7b62c5f68889b50023c3de642a Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?D=C3=A9vai=20Tam=C3=A1s?= <gonosztopi@amule.org>
Date: Sat, 27 Feb 2021 19:13:42 +0100
Subject: [PATCH] cmake: Add missing Romanian translation

---
 po/CMakeLists.txt | 1 +
 1 file changed, 1 insertion(+)

diff --git a/po/CMakeLists.txt b/po/CMakeLists.txt
index 950855b09b..136f5b8b0a 100644
--- a/po/CMakeLists.txt
+++ b/po/CMakeLists.txt
@@ -32,6 +32,7 @@ FUNCTION (SetTranslationOption PoName DefaultMode)
 	STRING (REGEX REPLACE "^pl$"	"Polish"		LangName "${LangName}")
 	STRING (REGEX REPLACE "^pt_BR$"	"Portuguese (Brazilian)" LangName "${LangName}")
 	STRING (REGEX REPLACE "^pt_PT$"	"Portuguese"		LangName "${LangName}")
+	STRING (REGEX REPLACE "^ro$"	"Romanian"		LangName "${LangName}")
 	STRING (REGEX REPLACE "^ru$"	"Russian"		LangName "${LangName}")
 	STRING (REGEX REPLACE "^sl$"	"Slovenian"		LangName "${LangName}")
 	STRING (REGEX REPLACE "^sq$"	"Albanian"		LangName "${LangName}")
