From a667c615384cf53e7f126fa569d6a37895b57f6e Mon Sep 17 00:00:00 2001
From: Vollstrecker <werner@vollstreckernet.de>
Date: Sun, 25 Apr 2021 12:55:46 +0200
Subject: [PATCH] Install desktop icons also

---
 src/CMakeLists.txt | 10 ++++++++++
 1 file changed, 10 insertions(+)

diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
index ba02b0185..8950e5b27 100644
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -222,6 +222,16 @@ if (BUILD_MONOLITHIC)
 	install (TARGETS amule
 		RUNTIME DESTINATION bin
 	)
+
+	install (FILES aMule.xpm
+		DESTINATION "${CMAKE_INSTALL_DATADIR}/pixmaps"
+		RENAME amule.xpm
+	)
+
+	install (FILES aMule.xpm
+		DESTINATION "${CMAKE_INSTALL_DATADIR}/pixmaps"
+		RENAME amulegui.xpm
+	)
 endif (BUILD_MONOLITHIC)
 
 if (BUILD_REMOTEGUI)
