From d22e0575619dbbd8471d755e7d7cbf97b2a333f8 Mon Sep 17 00:00:00 2001
From: Vollstrecker <werner@vollstreckernet.de>
Date: Sat, 24 Apr 2021 15:40:33 +0200
Subject: [PATCH] Intendation

---
 src/utils/cas/CMakeLists.txt | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/utils/cas/CMakeLists.txt b/src/utils/cas/CMakeLists.txt
index 34c3f7da09..94199abbfa 100644
--- a/src/utils/cas/CMakeLists.txt
+++ b/src/utils/cas/CMakeLists.txt
@@ -16,7 +16,7 @@ if (WIN32)
 endif()
 
 target_compile_definitions (cas
-		PRIVATE ${gdlib_CFLAGS}
+	PRIVATE ${gdlib_CFLAGS}
 )
 
 target_include_directories (cas
