From f186b3a2ce26e12d4765ed193a2949443fd5b09c Mon Sep 17 00:00:00 2001
From: Vollstrecker <werner@vollstreckernet.de>
Date: Sun, 25 Apr 2021 14:50:13 +0200
Subject: [PATCH] Fix *.mo installation name and prevent double-install

---
 po/CMakeLists.txt | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/po/CMakeLists.txt b/po/CMakeLists.txt
index 412fb284b..d896b851a 100644
--- a/po/CMakeLists.txt
+++ b/po/CMakeLists.txt
@@ -63,9 +63,9 @@ foreach (PO_FILE ${PO_FILES})
 
 		install (FILES ${CMAKE_CURRENT_BINARY_DIR}/${PO_FILE}.gmo
 			DESTINATION ${CMAKE_INSTALL_LOCALEDIR}/${PO_FILE}/LC_MESSAGES/
-			RENAME ${PO_FILE}.mo
+			RENAME amule.mo
 		)
 	endif()
 endforeach()
 
-gettext_process_po_files ("" ALL INSTALL_DESTINATION ${CMAKE_INSTALL_LOCALEDIR} PO_FILES ${PO_BUILD})
+gettext_process_po_files ("" ALL PO_FILES ${PO_BUILD})
