diff -up irrlicht-1.8.4/source/Irrlicht/COSOperator.cpp.sysctl irrlicht-1.8.4/source/Irrlicht/COSOperator.cpp
--- irrlicht-1.8.4/source/Irrlicht/COSOperator.cpp.sysctl	2020-08-06 12:11:59.218021970 -0400
+++ irrlicht-1.8.4/source/Irrlicht/COSOperator.cpp	2020-08-06 12:13:16.768492942 -0400
@@ -13,9 +13,11 @@
 #include <unistd.h>
 #ifndef _IRR_SOLARIS_PLATFORM_
 #include <sys/types.h>
+#ifdef _IRR_OSX_PLATFORM_
 #include <sys/sysctl.h>
 #endif
 #endif
+#endif
 
 #if defined(_IRR_COMPILE_WITH_X11_DEVICE_)
 #include "CIrrDeviceLinux.h"
