From 09ac7209063ba5b85e8e699515cdfa5decd72b79 Mon Sep 17 00:00:00 2001
From: Reilly Brogan <reilly@reillybrogan.com>
Date: Wed, 4 Oct 2023 12:14:34 -0500
Subject: [PATCH] Stateless nsswitch.conf

---
 nss/nss_database.c | 6 ++++--
 resolv/netdb.h     | 1 +
 2 files changed, 5 insertions(+), 2 deletions(-)

diff --git a/nss/nss_database.c b/nss/nss_database.c
index f2ed2f2c25..01494b1b2c 100644
--- a/nss/nss_database.c
+++ b/nss/nss_database.c
@@ -304,6 +304,8 @@ nss_database_reload (struct nss_database_data *staging,
                      struct file_change_detection *initial)
 {
   FILE *fp = fopen (_PATH_NSSWITCH_CONF, "rce");
+  if (fp == NULL)
+    fp = fopen (_VENDOR_NSSWITCH_CONF, "rce");
   if (fp == NULL)
     switch (errno)
       {
@@ -406,8 +408,8 @@ nss_database_check_reload_and_get (struct nss_database_state *local,
       return true;
     }
 
-  struct file_change_detection initial;
-  if (!__file_change_detection_for_path (&initial, _PATH_NSSWITCH_CONF))
+  struct file_change_detection initial, initial2;
+  if (!__file_change_detection_for_path (&initial, _PATH_NSSWITCH_CONF) && !__file_change_detection_for_path (&initial2, _VENDOR_NSSWITCH_CONF))
     return false;
 
   __libc_lock_lock (local->lock);
diff --git a/resolv/netdb.h b/resolv/netdb.h
index ac268f035f..9a3840eef8 100644
--- a/resolv/netdb.h
+++ b/resolv/netdb.h
@@ -44,6 +44,7 @@
 #define	_PATH_HOSTS		"/etc/hosts"
 #define	_PATH_NETWORKS		"/etc/networks"
 #define	_PATH_NSSWITCH_CONF	"/etc/nsswitch.conf"
+#define	_VENDOR_NSSWITCH_CONF	"/usr/share/defaults/etc/nsswitch.conf"
 #define	_PATH_PROTOCOLS		"/etc/protocols"
 #define	_PATH_SERVICES		"/etc/services"
 
-- 
2.42.0

