From 0647213cbaccc4480bd8a8041b0e920929797faa Mon Sep 17 00:00:00 2001
From: Reilly Brogan <reilly@reillybrogan.com>
Date: Fri, 15 Dec 2023 16:38:58 -0600
Subject: [PATCH] Stateless NSS

Signed-off-by: Reilly Brogan <reilly@reillybrogan.com>
---
 nss/nss_files/files-XXX.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/nss/nss_files/files-XXX.c b/nss/nss_files/files-XXX.c
index 2ed7184ca9..f12929dfe9 100644
--- a/nss/nss_files/files-XXX.c
+++ b/nss/nss_files/files-XXX.c
@@ -40,6 +40,7 @@
 #define ENTNAME_r	CONCAT(ENTNAME,_r)
 
 #define DATAFILE	"/etc/" DATABASE
+#define ALTDATAFILE	"/usr/share/defaults/etc/" DATABASE
 
 #ifdef NEED_H_ERRNO
 # include <netdb.h>
@@ -75,6 +76,9 @@ internal_setent (FILE **stream)
     {
       *stream = __nss_files_fopen (DATAFILE);
 
+      if (*stream == NULL)
+        *stream = __nss_files_fopen (ALTDATAFILE);
+
       if (*stream == NULL)
 	status = errno == EAGAIN ? NSS_STATUS_TRYAGAIN : NSS_STATUS_UNAVAIL;
     }
-- 
2.43.0

