diff --git a/Makefile b/Makefile
index 9174e266b1..081abdf725 100644
--- a/Makefile
+++ b/Makefile
@@ -97,7 +97,6 @@
 # RELEASE_FLAGS is flags for release builds.
 RELEASE_FLAGS =
 WARNINGS = \
-  -Werror -Wall -Wextra \
   -Wformat-signedness \
   -Wlogical-op \
   -Wmissing-declarations \
@@ -109,7 +108,8 @@ WARNINGS = \
   -Wsuggest-override \
   -Wunused-macros \
   -Wzero-as-null-pointer-constant \
-  -Wno-unknown-warning-option
+  -Wno-unknown-warning-option \
+  -Wno-array-bounds
 # Uncomment below to disable warnings
 #WARNINGS = -w
 DEBUGSYMS = -g
