From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Reilly Brogan <reilly@reillybrogan.com>
Date: Thu, 13 Jun 2024 21:07:39 -0500
Subject: [PATCH] solus: Use solus-specific libexec dir

Signed-off-by: Reilly Brogan <reilly@reillybrogan.com>
---
 crates/cli/src/main.rs | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/crates/cli/src/main.rs b/crates/cli/src/main.rs
index 4132be300..25179bf6e 100644
--- a/crates/cli/src/main.rs
+++ b/crates/cli/src/main.rs
@@ -194,7 +194,7 @@ mod linux {
                     .and_then(Path::parent)
                     .ok_or_else(|| anyhow!("no parent path for cli"))?;
 
-                match dir.join("libexec").join("zed-editor").canonicalize() {
+                match dir.join("lib64").join("zed").join("zed-editor").canonicalize() {
                     Ok(path) => Ok(path),
                     // In development cli and zed are in the ./target/ directory together
                     Err(e) => match cli.parent().unwrap().join("zed").canonicalize() {
