diff --git a/install.sh b/install.sh
index a306665..274b279 100755
--- a/install.sh
+++ b/install.sh
@@ -11,31 +11,6 @@
 VERSION=3.0.0
 FULLVERSION=3.0.0
 
-# some useful functions
-# ask displays 1st parameter, and ask new value for variable, whose name is
-# in the second parameter.
-ask ()
-{
-  askvar=$2
-  eval old=\$$askvar
-  eval printf \""$1 [$old] : "\"
-  read $askvar
-  eval test -z \"\$$askvar\" && eval $askvar=\'$old\'
-}
-
-# yesno gives 1 on no, 0 on yes $1 gives text to display.
-yesno ()
-{
-  while true; do
-  printf "$1 (Y/n) ? "
-  read ans
-  case X"$ans" in
-   X|Xy|XY) return 0;;
-   Xn|XN) return 1;;
-  esac
-  done
-}
-#
 #
 #
 CMDTAR="tar"
@@ -128,12 +103,6 @@ installbinary ()
   echo "Installing compiler and RTL for $FPCTARGET..."
   unztarfromtar "$BINARYTAR" "${CROSSPREFIX}base.$1.tar.gz" "$PREFIX"
 
-  if [ -f "binutils-${CROSSPREFIX}$1.tar.gz" ]; then
-    if yesno "Install Cross binutils"; then
-      unztar "binutils-${CROSSPREFIX}$1.tar.gz" "$PREFIX"
-    fi
-  fi
-
   # Install symlink
   rm -f "$EXECDIR/ppc${PPCSUFFIX}"
   ln -sf "$LIBDIR/ppc${PPCSUFFIX}" "$EXECDIR/ppc${PPCSUFFIX}"
@@ -183,13 +152,6 @@ installbinary ()
     chmod u=srx,g=rx,o=rx "$PREFIX/bin/grab_vcsa"
   fi
 
-  ide=`$TAR -tf $BINARYTAR | grep "${CROSSPREFIX}ide.$1.tar.gz"`
-  if [ "$ide" = "${CROSSPREFIX}ide.$1.tar.gz" ]; then
-    if yesno "Install Textmode IDE"; then
-      unztarfromtar "$BINARYTAR" "${CROSSPREFIX}ide.$1.tar.gz" "$PREFIX"
-    fi
-  fi
-
   rm -f *."$1".tar.gz
 }
 
@@ -246,12 +208,8 @@ case "$OSNAME" in
      fi
   ;;
   *)
-     # Install in /usr/local or /usr ?
-     if checkpath /usr/local/bin; then
-         PREFIX=/usr/local
-     else
-         PREFIX=/usr
-     fi
+     # Install in /usr
+     PREFIX=fpc_bootstrap
   ;;
 esac
 
@@ -260,15 +218,6 @@ if [ ! -w "$PREFIX" ]; then
   PREFIX="$HOME/fpc-$VERSION"
 fi
 
-case "$OSNAME" in
-  haiku)
-     ask "Install prefix (/boot/common or /boot/home/config) " PREFIX
-  ;;
-  *)
-     ask "Install prefix (/usr or /usr/local) " PREFIX
-  ;;
-esac
-
 # Support ~ expansion
 PREFIX=`eval echo $PREFIX`
 export PREFIX
@@ -323,29 +272,6 @@ done
 echo Done.
 echo
 
-# Install the documentation. Optional.
-if [ -f doc-pdf.tar.gz ]; then
-  if yesno "Install documentation"; then
-    echo Installing documentation in "$DOCDIR" ...
-    makedirhierarch "$DOCDIR"
-    unztar doc-pdf.tar.gz "$DOCDIR" "--strip 1"
-    echo Done.
-  fi
-fi
-echo
-
-# Install the demos. Optional.
-if [ -f demo.tar.gz ]; then
-  if yesno "Install demos"; then
-    ask "Install demos in" DEMODIR
-    echo Installing demos in "$DEMODIR" ...
-    makedirhierarch "$DEMODIR"
-    unztar demo.tar.gz "$DEMODIR"
-    echo Done.
-  fi
-fi
-echo
-
 # Install /etc/fpc.cfg, this is done using the samplecfg script
 if [ "$cross" = "" ]; then
   "$LIBDIR/samplecfg" "$LIBDIR"
