From c93dfd0b84438e9d1e139da39d99d32b1beb3088 Mon Sep 17 00:00:00 2001
From: Ikey Doherty <ikey@solus-project.com>
Date: Wed, 29 Jun 2016 21:52:46 +0100
Subject: [PATCH] Underp buildsystem

Signed-off-by: Ikey Doherty <ikey@solus-project.com>
---
 Makefile              |  3 ++-
 config/Makefile.linux | 16 +++++-----------
 2 files changed, 7 insertions(+), 12 deletions(-)

diff --git a/Makefile b/Makefile
index 6a9803c..bba0e52 100644
--- a/Makefile
+++ b/Makefile
@@ -42,8 +42,9 @@ endif
 
 GLEW_PREFIX ?= /usr
 GLEW_DEST ?= /usr
+ALIBDIR    ?= lib
 BINDIR    ?= $(GLEW_DEST)/bin
-LIBDIR    ?= $(GLEW_DEST)/lib
+LIBDIR    ?= $(GLEW_DEST)/$(ALIBDIR)
 INCDIR    ?= $(GLEW_DEST)/include/GL
 
 ifneq ($(GLEW_NO_GLU), -DGLEW_NO_GLU)
diff --git a/config/Makefile.linux b/config/Makefile.linux
index 55e4a23..499012a 100644
--- a/config/Makefile.linux
+++ b/config/Makefile.linux
@@ -1,20 +1,14 @@
 NAME = $(GLEW_NAME)
-CC = cc
-LD = cc
+CC ?= cc
+LD ?= cc
 M_ARCH ?= $(shell uname -m)
-ARCH64 = false
-ifeq (x86_64,${M_ARCH})
-  ARCH64 = true
-endif
-ifeq (ppc64,${M_ARCH})
-  ARCH64 = true
-endif
+ARCH64 ?= true
 ifeq (${ARCH64},true)
   LDFLAGS.EXTRA = -L/usr/X11R6/lib64 -L/usr/lib64
   LIBDIR = $(GLEW_DEST)/lib64
 else
-  LDFLAGS.EXTRA = -L/usr/X11R6/lib -L/usr/lib
-  LIBDIR = $(GLEW_DEST)/lib
+  LDFLAGS.EXTRA = -L/usr/X11R6/lib32 -L/usr/lib32
+  LIBDIR = $(GLEW_DEST)/lib32
 endif
 LDFLAGS.GL = -lGL -lX11
 LDFLAGS.STATIC = -Wl,-Bstatic
-- 
2.9.0

