From aa1383797c0cc8f2a418921bdabb245be2cd461c Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?S=C3=A9bastien=20Noel?= <sebastien@twolife.be>
Date: Sat, 25 Apr 2026 10:41:25 +0200
Subject: [PATCH] fix build failure with freerdp 3.25

---
 tests/test-client-rdp.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/tests/test-client-rdp.c b/tests/test-client-rdp.c
index a06a1879..fce13af7 100644
--- a/tests/test-client-rdp.c
+++ b/tests/test-client-rdp.c
@@ -249,7 +249,7 @@ static BOOL
 rdp_client_authenticate (freerdp  *instance,
                          char    **username,
                          char    **password,
-                         char    **domain)
+                         char    **domain, rdp_auth_reason reason)
 {
   /* Credentials were already parsed from the command line */
 
@@ -307,7 +307,7 @@ rdp_client_new (freerdp    *instance,
   instance->PreConnect = rdp_client_pre_connect;
   instance->PostConnect = rdp_client_post_connect;
   instance->PostDisconnect = rdp_client_post_disconnect;
-  instance->Authenticate = rdp_client_authenticate;
+  instance->AuthenticateEx = rdp_client_authenticate;
   instance->VerifyCertificateEx = rdp_client_verify_certificate_ex;
   instance->VerifyChangedCertificateEx = rdp_client_verify_changed_certificate_ex;
 
-- 
GitLab