--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -280,10 +280,10 @@ if(NOT WIN32)
   install(TARGETS pano13 LIBRARY COMPONENT libraries)
   set(_pano_target pano13)
   #create and install static library
-  add_library(pano13a STATIC ${pano13_hdr} ${pano13_src})
-  set_target_properties(pano13a PROPERTIES OUTPUT_NAME pano13)
-  install(TARGETS pano13a ARCHIVE COMPONENT libraries)
+#  add_library(pano13a STATIC ${pano13_hdr} ${pano13_src})
+#  set_target_properties(pano13a PROPERTIES OUTPUT_NAME pano13)
+#  install(TARGETS pano13a ARCHIVE COMPONENT libraries)
 else()
   IF(BUILD_DLL)

