From e81659c97f375c1ab4750a9c3fb352fc2f6c9eff Mon Sep 17 00:00:00 2001
From: Joshua Strobl <joshua@stroblindustries.com>
Date: Sat, 8 Oct 2016 12:15:38 +0300
Subject: [PATCH 1/1] Fix nethash.

---
 scripts/cmd5.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/scripts/cmd5.py b/scripts/cmd5.py
index 5ac4cd4..0ed935d 100644
--- a/scripts/cmd5.py
+++ b/scripts/cmd5.py
@@ -30,6 +30,6 @@ for filename in sys.argv[1:]:
 
 hash = hashlib.md5(f).hexdigest().lower()[16:]
 #TODO 0.7: improve nethash creation
-if hash == "3dc531e4296de555":
+if hash == "3dc531e4296de555" or hash == "e42d81cd67b8c7bc":
 	hash = "626fce9a778df4d4"
 print('#define GAME_NETVERSION_HASH "%s"' % hash)
-- 
2.10.1

