From 42d0231eccfb76cb556b185e25987c8eef08fe7a Mon Sep 17 00:00:00 2001
From: Alexander Vorobyev <avorobyev@protonmail.com>
Date: Wed, 25 Nov 2020 22:13:37 +0300
Subject: Fix multiarch

---
 Makefile.in | 10 ++++++----
 1 file changed, 6 insertions(+), 4 deletions(-)

diff --git a/Makefile.in b/Makefile.in
index 96089c8..f0fc799 100644
--- a/Makefile.in
+++ b/Makefile.in
@@ -41,7 +41,9 @@ default:	@TARGETS@
 CC		= @CC@
 CXX		= @CXX@
 CFLAGS		= @CFLAGS@
-CXXFLAGS	= -std=c++11 -I. @CXXFLAGS@ @SNDFILE_CFLAGS@ 
+CXXFLAGS	= -std=c++11 -I. @CXXFLAGS@ @SNDFILE_CFLAGS@
+SUFFIX      := $(shell uname -m | sed -e 's/^unknown/$//' -e 's/^i.86/$//' -e 's/^x86_64/$/64/')
+LIBDIR		?= lib$(SUFFIX)
 
 # ar, ranlib
 #
@@ -74,8 +76,8 @@ INSTALL_PREFIX	 	  = @prefix@
 INSTALL_API_HEADERS	  = $(INSTALL_PREFIX)/include/vamp
 INSTALL_SDK_HEADERS	  = $(INSTALL_PREFIX)/include/vamp-sdk
 INSTALL_HOSTSDK_HEADERS	  = $(INSTALL_PREFIX)/include/vamp-hostsdk
-INSTALL_SDK_LIBS	  = $(INSTALL_PREFIX)/lib
-INSTALL_PLUGINS		  = $(INSTALL_PREFIX)/lib/vamp
+INSTALL_SDK_LIBS	  = $(INSTALL_PREFIX)/$(LIBDIR)
+INSTALL_PLUGINS		  = $(INSTALL_PREFIX)/$(LIBDIR)/vamp
 INSTALL_BINARIES	  = $(INSTALL_PREFIX)/bin 
 
 INSTALL_SDK_LIBNAME	  = libvamp-sdk.so.2.10.0
@@ -90,7 +92,7 @@ INSTALL_HOSTSDK_LINK_DEV  = libvamp-hostsdk.so
 INSTALL_HOSTSDK_STATIC    = libvamp-hostsdk.a
 INSTALL_HOSTSDK_LA        = libvamp-hostsdk.la
 
-INSTALL_PKGCONFIG	  = $(INSTALL_PREFIX)/lib/pkgconfig
+INSTALL_PKGCONFIG	  = $(INSTALL_PREFIX)/$(LIBDIR)/pkgconfig
 
 # Flags required to tell the compiler to create a dynamically loadable object
 #
-- 
2.29.2

