From eb378b91ef76b755eda6180cce40e48d027eaccc Mon Sep 17 00:00:00 2001
From: NoXPhasma <noxphasma@live.de>
Date: Fri, 13 Jan 2023 22:11:32 +0100
Subject: [PATCH] Add support for GameSir Xbox controllers

---
 transport/wired.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/transport/wired.c b/transport/wired.c
index d71e2d5..924d6a1 100644
--- a/transport/wired.c
+++ b/transport/wired.c
@@ -558,6 +558,7 @@ static const struct usb_device_id xone_wired_id_table[] = {
 	{ XONE_WIRED_VENDOR(0x3285) }, /* Nacon */
 	{ XONE_WIRED_VENDOR(0x2dc8) }, /* 8BitDo */
 	{ XONE_WIRED_VENDOR(0x2e95) }, /* SCUF */
+	{ XONE_WIRED_VENDOR(0x3537) }, /* GameSir */
 	{ },
 };
 
