From 2b11239db8521995a52e4a9751d3446143339893 Mon Sep 17 00:00:00 2001
From: Ikey Doherty <ikey@solus-project.com>
Date: Sun, 6 Mar 2016 16:02:16 +0000
Subject: [PATCH] libkmod: Support /usr/lib to enable stateless configs on
 Solus

Signed-off-by: Ikey Doherty <ikey@solus-project.com>
---
 libkmod/libkmod.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/libkmod/libkmod.c b/libkmod/libkmod.c
index fe68134..49fc865 100644
--- a/libkmod/libkmod.c
+++ b/libkmod/libkmod.c
@@ -61,7 +61,7 @@ static struct _index_files {
 static const char *default_config_paths[] = {
 	SYSCONFDIR "/modprobe.d",
 	"/run/modprobe.d",
-	"/lib/modprobe.d",
+	"/usr/lib/modprobe.d",
 	NULL
 };
 
-- 
2.7.1

