From 7b38a7deaed40f9f5a1522771ced55f0c3a165ce Mon Sep 17 00:00:00 2001
From: Reilly Brogan <reilly@reillybrogan.com>
Date: Sat, 5 Aug 2023 15:20:12 -0500
Subject: [PATCH] Fix grep error for /etc/shells on Solus

---
 gnome-session/gnome-session.in | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/gnome-session/gnome-session.in b/gnome-session/gnome-session.in
index ddd1a591..d335d3a1 100755
--- a/gnome-session/gnome-session.in
+++ b/gnome-session/gnome-session.in
@@ -3,7 +3,7 @@
 if [ "x$XDG_SESSION_TYPE" = "xwayland" ] &&
    [ "x$XDG_SESSION_CLASS" != "xgreeter" ] &&
    [  -n "$SHELL" ] &&
-   grep -q "$SHELL" /etc/shells &&
+   grep -q "$SHELL" /usr/share/defaults/etc/shells &&
    ! (echo "$SHELL" | grep -q "false") &&
    ! (echo "$SHELL" | grep -q "nologin"); then
   if [ "$1" != '-l' ]; then
-- 
2.41.0

