From 35a7e737f1f13dc7da0cb6e764b7a25708388598 Mon Sep 17 00:00:00 2001
From: Reilly Brogan <reilly@reillybrogan.com>
Date: Sun, 14 Jan 2024 19:57:45 -0600
Subject: [PATCH] Fix build with minizip-ng

---
 base/zlib_help.h | 1 +
 1 file changed, 1 insertion(+)

diff --git a/base/zlib_help.h b/base/zlib_help.h
index af871ee..e58d660 100644
--- a/base/zlib_help.h
+++ b/base/zlib_help.h
@@ -7,6 +7,7 @@
 #pragma once
 
 #include <zip.h>
+#include <zlib.h>
 #include <unzip.h>
 #include "logs.h"
 
-- 
2.43.0

