From: Kentaro Hayashi <kenhys@xdump.org>
Date: Sat, 29 Mar 2025 22:45:52 +0900
Subject: Always disable update check on startup

Forwarded: not-needed
Author: HAYASHI Kentaro <kenhys@xdump.org>
---
 src/apps/deskflow-gui/MainWindow.cpp | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/src/apps/deskflow-gui/MainWindow.cpp b/src/apps/deskflow-gui/MainWindow.cpp
index 5b0d730..f902b18 100644
--- a/src/apps/deskflow-gui/MainWindow.cpp
+++ b/src/apps/deskflow-gui/MainWindow.cpp
@@ -605,6 +605,8 @@ void MainWindow::serverConnectionConfigureClient(const QString &clientName)
 void MainWindow::open()
 {
 
+  // Always disable update check on startup
+  Settings::setValue(Settings::Gui::AutoUpdateCheck, false);
   if (!Settings::value(Settings::Gui::AutoUpdateCheck).isValid()) {
     showAndActivate();
     Settings::setValue(Settings::Gui::AutoUpdateCheck, messages::showUpdateCheckOption(this));
