diff --git a/src/lib/gui/MainWindow.cpp b/src/lib/gui/MainWindow.cpp
index 5b0d730e..f902b18d 100644
--- a/src/lib/gui/MainWindow.cpp
+++ b/src/lib/gui/MainWindow.cpp
@@ -605,6 +605,8 @@ void MainWindow::serverConnectionConfigureClient(const QString &clientName)
 void MainWindow::open()
 {
 
+  // Always disable update check on startup
+  Settings::setValue(Settings::Gui::AutoUpdateCheck, false);
   if (!Settings::value(Settings::Gui::AutoUpdateCheck).isValid()) {
     showAndActivate();
     Settings::setValue(Settings::Gui::AutoUpdateCheck, messages::showUpdateCheckOption(this));
-- 
2.47.2

