From: Kentaro Hayashi <kenhys@xdump.org>
Date: Mon, 17 Mar 2025 20:32:46 +0900
Subject: Always disable update check on startup

Forwarded: not-needed
Author: HAYASHI Kentaro <kenhys@xdump.org>
---
 src/apps/deskflow-gui/MainWindow.cpp | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/src/apps/deskflow-gui/MainWindow.cpp b/src/apps/deskflow-gui/MainWindow.cpp
index cf9902c..eada360 100644
--- a/src/apps/deskflow-gui/MainWindow.cpp
+++ b/src/apps/deskflow-gui/MainWindow.cpp
@@ -621,6 +621,8 @@ void MainWindow::serverConnectionConfigureClient(const QString &clientName)
 
 void MainWindow::open()
 {
+  // Always disable update check on startup
+  m_appConfig.setEnableUpdateCheck(false);
   if (!m_appConfig.enableUpdateCheck().has_value()) {
     showAndActivate();
     m_appConfig.setEnableUpdateCheck(messages::showUpdateCheckOption(this));
