From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Reilly Brogan <reilly@reillybrogan.com>
Date: Tue, 23 May 2023 10:37:21 -0500
Subject: [PATCH 6/6] Solus tweak to launchfeedback config

---
 kcms/cursortheme/launchfeedbacksettings.kcfg | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/kcms/cursortheme/launchfeedbacksettings.kcfg b/kcms/cursortheme/launchfeedbacksettings.kcfg
index 0d70303de6..fc29d8ba27 100644
--- a/kcms/cursortheme/launchfeedbacksettings.kcfg
+++ b/kcms/cursortheme/launchfeedbacksettings.kcfg
@@ -13,7 +13,7 @@ SPDX-License-Identifier: LGPL-2.0-only
       <default>false</default>
     </entry>
     <entry name="bouncing" key="Bouncing" type="Bool">
-      <default>true</default>
+      <default>false</default>
     </entry>
     <entry name="cursorTimeout" key="Timeout" type="Int">
       <label>Timeout in seconds</label>
