From 8419a3f89bbe08afbae2811f6dad274dd5e7ee7c Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?=C3=89tienne=20Dupuis?= <e.dupuis@ateme.com>
Date: Wed, 5 Aug 2020 11:42:44 +0200
Subject: [PATCH] Out of bound array access in twolame_bits_for_nonoise().

---
 libtwolame/encode.c | 6 ++++--
 1 file changed, 4 insertions(+), 2 deletions(-)

diff --git a/libtwolame/encode.c b/libtwolame/encode.c
index 7e83a8f..f49be5a 100644
--- a/libtwolame/encode.c
+++ b/libtwolame/encode.c
@@ -751,10 +751,12 @@ int twolame_bits_for_nonoise(twolame_options * glopts,
     /* Count the number of bits required to encode the quantization index for both channels in each
        subband. If we're above the jsbound, then pretend we only have one channel */
     for (sb = 0; sb < jsbound; ++sb)
-        bbal += nch * nbal[line[glopts->tablenum][sb]]; // (*alloc)[sb][0].bits;
+        if (line[glopts->tablenum][sb] != -1)
+            bbal += nch * nbal[line[glopts->tablenum][sb]]; // (*alloc)[sb][0].bits;
 
     for (sb = jsbound; sb < sblimit; ++sb)
-        bbal += nbal[line[glopts->tablenum][sb]];   // (*alloc)[sb][0].bits;
+        if (line[glopts->tablenum][sb] != -1)
+            bbal += nbal[line[glopts->tablenum][sb]];   // (*alloc)[sb][0].bits;
     req_bits = banc + bbal + berr;
 
     for (sb = 0; sb < sblimit; ++sb)
