From 6767064a386563c4af07febd6363ff150f356945 Mon Sep 17 00:00:00 2001
From: Fabio Forni <livingsilver94.solus@redaril.me>
Date: Mon, 25 Nov 2019 01:05:16 +0100
Subject: [PATCH] Default to using openjdk-8

---
 src/script/ant | 6 ++----
 1 file changed, 2 insertions(+), 4 deletions(-)

diff --git a/src/script/ant b/src/script/ant
index 81107b5..2379c51 100644
--- a/src/script/ant
+++ b/src/script/ant
@@ -185,10 +185,8 @@ if [ -z "$JAVACMD" ]; then
       JAVACMD="$JAVA_HOME/bin/java"
     fi
   else
-    JAVACMD=`which java 2> /dev/null `
-    if [ -z "$JAVACMD" ]; then
-      JAVACMD=java
-    fi
+    JAVA_HOME=/usr/lib64/openjdk-8
+    JAVACMD="$JAVA_HOME/bin/java"
   fi
 fi
 
-- 
2.24.0

