From e8ab45027a46b305f0c8917a2ac4bf8c5c94fa6c Mon Sep 17 00:00:00 2001
From: Algent Albrahimi <algent@protonmail.com>
Date: Sat, 26 Feb 2022 12:52:29 +0100
Subject: [PATCH] Fix install paths

---
 qelectrotech.pro | 14 +++++++-------
 1 file changed, 7 insertions(+), 7 deletions(-)

diff --git a/qelectrotech.pro b/qelectrotech.pro
index f182f85..974fd52 100644
--- a/qelectrotech.pro
+++ b/qelectrotech.pro
@@ -5,20 +5,20 @@
 # Chemins utilises pour la compilation et l'installation de QET
 unix {
         # Chemins UNIX
-        COMPIL_PREFIX              = '/usr/local/'
-        INSTALL_PREFIX             = '/usr/local/'
+        COMPIL_PREFIX              = '/usr/'
+        INSTALL_PREFIX             = '/usr/'
         QET_BINARY_PATH            = 'bin/'
         QET_COMMON_COLLECTION_PATH = 'share/qelectrotech/elements/'
         QET_COMMON_TBT_PATH        = 'share/qelectrotech/titleblocks/'
         QET_LANG_PATH              = 'share/qelectrotech/lang/'
         QET_EXAMPLES_PATH          = 'share/qelectrotech/examples/'
-        QET_LICENSE_PATH           = 'doc/qelectrotech/'
-        QET_MIME_XML_PATH          = '../share/mime/application/'
-        QET_MIME_DESKTOP_PATH      = '../share/mimelnk/application/'
-        QET_MIME_PACKAGE_PATH      = '../share/mime/packages/'
+	QET_LICENSE_PATH           = 'share/doc/qelectrotech/'
+	QET_MIME_XML_PATH          = 'share/mime/application/'
+	QET_MIME_DESKTOP_PATH      = 'share/mimelnk/application/'
+	QET_MIME_PACKAGE_PATH      = 'share/mime/packages/'
         QET_DESKTOP_PATH           = 'share/applications/'
         QET_ICONS_PATH             = 'share/icons/hicolor/'
-        QET_MAN_PATH               = 'man/'
+        QET_MAN_PATH               = 'share/man/'
         QET_APPDATA_PATH           = 'share/appdata'
 }
 win32 {
-- 
2.35.1

