From 339ba4074c925d235a89a9da8b7d4703e73b8a36 Mon Sep 17 00:00:00 2001
From: Longwu Ou <xulongwu4@gmail.com>
Date: Mon, 11 Oct 2021 23:43:56 -0400
Subject: [PATCH 2/2] Fix cgroups

---
 config/config.toml.centos | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/config/config.toml.centos b/config/config.toml.centos
index e952183..729b882 100644
--- a/config/config.toml.centos
+++ b/config/config.toml.centos
@@ -17,7 +17,7 @@ environment = []
 #debug = "/var/log/nvidia-container-toolkit.log"
 #ldcache = "/etc/ld.so.cache"
 load-kmods = true
-#no-cgroups = false
+no-cgroups = true
 #user = "root:video"
 ldconfig = "@/sbin/ldconfig"
 
-- 
2.32.0

