diff --git a/toolkit/config.toml.centos b/toolkit/config.toml.centos
index 33d8e82..1f50862 100644
--- a/toolkit/config.toml.centos
+++ b/toolkit/config.toml.centos
@@ -1,3 +1,10 @@
+#
+# nvidia-container-toolkit default configuration
+#
+# Do not make changes to this file. nvidia-container-toolkit is implemented in a stateless
+# fashion on Solus, and will load files in a layered mechanism. If you wish to edit
+# this configuration, copy to /etc/nvidia-container-toolkit/.
+
 disable-require = false
 #swarm-resource = "DOCKER_RESOURCE_GPU"
 
diff --git a/toolkit/nvidia-container-toolkit/hook_config.go b/toolkit/nvidia-container-toolkit/hook_config.go
index c8e1038..4b5865d 100644
--- a/toolkit/nvidia-container-toolkit/hook_config.go
+++ b/toolkit/nvidia-container-toolkit/hook_config.go
@@ -9,13 +9,15 @@ import (
 )
 
 const (
-	configPath = "/etc/nvidia-container-runtime/config.toml"
+	configPath = "/etc/nvidia-container-toolkit/config.toml"
 	driverPath = "/run/nvidia/driver"
+	defaultconfigPath = "/usr/share/nvidia-container-toolkit/config.toml"
 )
 
 var defaultPaths = [...]string{
 	path.Join(driverPath, configPath),
 	configPath,
+	defaultconfigPath,
 }
 
 // CLIConfig: options for nvidia-container-cli.
