From c65cde10235352c936c5e8255173e7f726b8923c Mon Sep 17 00:00:00 2001
From: Peter O'Connor <sunnyflunk@users.noreply.github.com>
Date: Thu, 2 Feb 2017 23:00:19 +1100
Subject: [PATCH] Use Solus flags

Don't override my flags!
---
 Makefile | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/Makefile b/Makefile
index 07a5a1e..0e32cef 100644
--- a/Makefile
+++ b/Makefile
@@ -7,7 +7,7 @@
 # to switch between compilation modes.
 
 # (A) Production use (optimized mode)
-OPT ?= -O2 -DNDEBUG
+#OPT ?= -O2 -DNDEBUG
 # (B) Debug mode, w/ full line-level debugging symbols
 # OPT ?= -g2
 # (C) Profiling mode: opt, but w/debugging symbols
@@ -59,8 +59,8 @@ BENCHMARKS = \
 	doc/bench/db_bench_sqlite3 \
 	doc/bench/db_bench_tree_db
 
-CFLAGS += -I. -I./include $(PLATFORM_CCFLAGS) $(OPT)
-CXXFLAGS += -I. -I./include $(PLATFORM_CXXFLAGS) $(OPT)
+CFLAGS += -I. -I./include $(PLATFORM_CCFLAGS) -DNDEBUG
+CXXFLAGS += -I. -I./include $(PLATFORM_CXXFLAGS) -DNDEBUG
 
 LDFLAGS += $(PLATFORM_LDFLAGS)
 LIBS += $(PLATFORM_LIBS)
