From df052bfa3c66d24109f40f18266ee057d1838b9b Mon Sep 17 00:00:00 2001
From: Heiko Becker <mail@heiko-becker.de>
Date: Wed, 6 Mar 2024 01:17:51 +0100
Subject: [PATCH] cmake: Add missing " to unbreak the build

---
 CMakeLists.txt | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 236abcd..e76484c 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -1,5 +1,5 @@
 cmake_minimum_required(VERSION 3.16)
-project(KPipewire VERSION "6.0.1)
+project(KPipewire VERSION "6.0.1")
 
 set(KF6_MIN_VERSION "6.0.0")
 find_package(ECM ${KF6_MIN_VERSION} REQUIRED NO_MODULE)
-- 
GitLab

