From 91c2e2b7100a690f43ba3866f97204305629add8 Mon Sep 17 00:00:00 2001
From: Ikey Doherty <ikey@solus-project.com>
Date: Sat, 4 Mar 2017 16:25:45 +0000
Subject: [PATCH] Force Solus compatibility

Signed-off-by: Ikey Doherty <ikey@solus-project.com>
---
 configure.ac | 13 +++++++++++++
 1 file changed, 13 insertions(+)

diff --git a/configure.ac b/configure.ac
index bdcf14d..0ca224f 100644
--- a/configure.ac
+++ b/configure.ac
@@ -954,6 +954,7 @@ strip_unwanted_llvm_flags() {
         -e 's/-fno-rtti\>/-Fno-rtti/g' \
         -e 's/\s\+-f\S*//g' \
         -e 's/-Fno-rtti\>/-fno-rtti/g' \
+        -e 's/-m64\>//g' \
         -e 's/^ //'
 }
 
@@ -965,6 +966,18 @@ llvm_set_environment_variables() {
         LLVM_INCLUDEDIR=`$LLVM_CONFIG --includedir`
         LLVM_LIBDIR=`$LLVM_CONFIG --libdir`
 
+        case "$host_cpu" in
+            i*86)
+                LLVM_LIBDIR="/usr/lib32"
+                LLVM_LDFLAGS="-L/usr/lib32"
+                ;;
+            *)
+                LLVM_LIBDIR="/usr/lib64"
+                LLVM_LDFLAGS="-L/usr/lib64"
+                ;;
+        esac
+
+
         # We need to respect LLVM_CPPFLAGS when compiling LLVM headers.
         save_CFLAGS="$CFLAGS"
         CFLAGS="$CFLAGS $LLVM_CPPFLAGS"
-- 
2.12.0

