From eeff52e54f581513b02d8e93e488d874ed2635d4 Mon Sep 17 00:00:00 2001
From: Ikey Doherty <ikey@solus-project.com>
Date: Tue, 14 Feb 2017 17:25:28 +0000
Subject: [PATCH] Force Solus compatibility

Signed-off-by: Ikey Doherty <ikey@solus-project.com>
---
 configure.ac | 12 ++++++++++++
 1 file changed, 12 insertions(+)

diff --git a/configure.ac b/configure.ac
index 8448736..5dd7733 100644
--- a/configure.ac
+++ b/configure.ac
@@ -963,6 +963,7 @@ strip_unwanted_llvm_flags() {
         -e 's/-fomit-frame-pointer\>//g' \
         -e 's/-fvisibility-inlines-hidden\>//g' \
         -e 's/-fPIC\>//g' \
+        -e 's/-m64\>//g' \
         -e 's/-fstack-protector-strong\>//g'
 }
 
@@ -974,6 +975,17 @@ llvm_set_environment_variables() {
         LLVM_INCLUDEDIR=`$LLVM_CONFIG --includedir`
         LLVM_LIBDIR=`$LLVM_CONFIG --libdir`
 
+        case "$host_cpu" in
+            i*86)
+                LLVM_LIBDIR="/usr/lib32"
+                LLVM_LDFLAGS="-L/usr/lib32"
+                ;;
+            *)
+                LLVM_LIBDIR="/usr/lib64"
+                LLVM_LDFLAGS="-L/usr/lib64"
+                ;;
+        esac
+
         # We need to respect LLVM_CPPFLAGS when compiling LLVM headers.
         save_CFLAGS="$CFLAGS"
         CFLAGS="$CFLAGS $LLVM_CPPFLAGS"
-- 
2.11.1

