From dfc87e8d5d98769147db9955a84b5b760d162b1a Mon Sep 17 00:00:00 2001
From: Ikey Doherty <ikey@solus-project.com>
Date: Tue, 15 Dec 2015 04:29:09 +0000
Subject: [PATCH] Force Solus compatibility

Signed-off-by: Ikey Doherty <ikey@solus-project.com>
---
 configure.ac | 11 +++++++++++
 1 file changed, 11 insertions(+)

diff --git a/configure.ac b/configure.ac
index 92799c6..dd94980 100644
--- a/configure.ac
+++ b/configure.ac
@@ -1729,6 +1729,7 @@ strip_unwanted_llvm_flags() {
 	-e 's/-fomit-frame-pointer\>//g' \
 	-e 's/-fvisibility-inlines-hidden\>//g' \
 	-e 's/-fPIC\>//g' \
+	-e 's/-m64\>//g' \
 	-e 's/-fstack-protector-strong\>//g'
 }
 
@@ -1765,6 +1766,16 @@ if test "x$enable_gallium_llvm" = xyes; then
         LLVM_INCLUDEDIR=`$LLVM_CONFIG --includedir`
         LLVM_LIBDIR=`$LLVM_CONFIG --libdir`
 
+        case "$host_cpu" in
+            i*86)
+                LLVM_LIBDIR="/usr/lib32"
+                LLVM_LDFLAGS="-L/usr/lib32"
+                ;;
+            *)
+                LLVM_LIBDIR="/usr/lib64"
+                LLVM_LDFLAGS="-L/usr/lib64"
+                ;;
+        esac
         AC_COMPUTE_INT([LLVM_VERSION_MAJOR], [LLVM_VERSION_MAJOR],
             [#include "${LLVM_INCLUDEDIR}/llvm/Config/llvm-config.h"])
         AC_COMPUTE_INT([LLVM_VERSION_MINOR], [LLVM_VERSION_MINOR],
-- 
2.6.4

