diff --git a/configure.ac b/configure.ac
index 0604ad9..c070c36 100644
--- a/configure.ac
+++ b/configure.ac
@@ -824,12 +824,6 @@ cygwin* )
     ;;
 esac
 
-if test "x$pthread_stubs_possible" = xyes; then
-    PKG_CHECK_MODULES(PTHREADSTUBS, pthread-stubs)
-    AC_SUBST(PTHREADSTUBS_CFLAGS)
-    AC_SUBST(PTHREADSTUBS_LIBS)
-fi
-
 dnl SELinux awareness.
 AC_ARG_ENABLE([selinux],
     [AS_HELP_STRING([--enable-selinux],
