diff --git a/src/mesa/drivers/dri/common/Makefile.am b/src/mesa/drivers/dri/common/Makefile.am
index c85c470..01ef825 100644
--- a/src/mesa/drivers/dri/common/Makefile.am
+++ b/src/mesa/drivers/dri/common/Makefile.am
@@ -50,4 +50,5 @@ libxmlconfig_la_LIBADD = $(EXPAT_LIBS) -lm
 
 libmegadriver_stub_la_SOURCES = $(megadriver_stub_FILES)
 
-sysconf_DATA = drirc
+dridir = $(datadir)/defaults/mesa
+dri_DATA = drirc
diff --git a/src/mesa/drivers/dri/common/xmlconfig.c b/src/mesa/drivers/dri/common/xmlconfig.c
index d464937..0ae07b1 100644
--- a/src/mesa/drivers/dri/common/xmlconfig.c
+++ b/src/mesa/drivers/dri/common/xmlconfig.c
@@ -987,7 +987,7 @@ void
 driParseConfigFiles(driOptionCache *cache, const driOptionCache *info,
                     int screenNum, const char *driverName)
 {
-    char *filenames[2] = { SYSCONFDIR "/drirc", NULL};
+    char *filenames[2] = {"/usr/share/defaults/mesa/drirc", NULL};
     char *home;
     uint32_t i;
     struct OptConfData userData;
