diff --git a/src/util/Makefile.am b/src/util/Makefile.am
index b57da3c..9e5f80b 100644
--- a/src/util/Makefile.am
+++ b/src/util/Makefile.am
@@ -64,7 +64,8 @@ libxmlconfig_la_CFLAGS = \
 	$(EXPAT_CFLAGS)
 libxmlconfig_la_LIBADD = $(EXPAT_LIBS) -lm
 
-sysconf_DATA = drirc
+dridir = $(datadir)/defaults/mesa
+dri_DATA = drirc
 
 u_atomic_test_LDADD = libmesautil.la
 roundeven_test_LDADD = -lm
diff --git a/src/util/xmlconfig.c b/src/util/xmlconfig.c
index 60a6331..ad5b56a 100644
--- a/src/util/xmlconfig.c
+++ b/src/util/xmlconfig.c
@@ -988,7 +988,7 @@ void
 driParseConfigFiles(driOptionCache *cache, const driOptionCache *info,
                     int screenNum, const char *driverName)
 {
-    char *filenames[2] = { SYSCONFDIR "/drirc", NULL};
+    char *filenames[2] = {"/usr/share/defaults/mesa/drirc", NULL};
     char *home;
     uint32_t i;
     struct OptConfData userData;
