From 392b47da299a98f6ff50c27a618995617983a5f9 Mon Sep 17 00:00:00 2001
From: Alexander Vorobyev <avorobyev@protonmail.com>
Date: Mon, 21 Dec 2020 10:17:12 +0300
Subject: Fix pkgconfig for GTK3

---
 zbar-gtk.pc.in | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/zbar-gtk.pc.in b/zbar-gtk.pc.in
index fc712a2..2c8a282 100644
--- a/zbar-gtk.pc.in
+++ b/zbar-gtk.pc.in
@@ -7,6 +7,6 @@ Name: zbar-gtk
 Description: bar code scanning and decoding GTK widget
 URL: http://zbar.sourceforge.net
 Version: @VERSION@
-Requires: zbar, gtk+-2.0, gthread-2.0
+Requires: zbar, gtk+-3.0, gthread-2.0
 Libs: -L${libdir} -lzbargtk
 Cflags: -I${includedir}
-- 
2.29.2

