diff --git a/src/base/convert_to.cpp b/src/base/convert_to.cpp
index cb1bf8dd..02fcccae 100644
--- a/src/base/convert_to.cpp
+++ b/src/base/convert_to.cpp
@@ -12,6 +12,7 @@
 #include "base/sha1.h"
 #include <cstdio>
 #include <cstdlib>
+#include <cstdint>
 
 namespace base {
 
diff --git a/src/base/convert_to.h b/src/base/convert_to.h
index ecd2fe88..7485958d 100644
--- a/src/base/convert_to.h
+++ b/src/base/convert_to.h
@@ -8,6 +8,7 @@
 
 #include "base/base.h"
 #include <string>
+#include <cstdint>
 
 namespace base {
 
diff --git a/src/base/sha1.h b/src/base/sha1.h
index 435faae2..36fabe06 100644
--- a/src/base/sha1.h
+++ b/src/base/sha1.h
@@ -7,6 +7,7 @@
 #pragma once
 
 #include <vector>
+#include <cstdint>
 #include <string>
 
 extern "C" struct SHA1Context;
