From 461c22f236f8e2f677ceee85f677de8d6b52a6cd Mon Sep 17 00:00:00 2001
From: Reilly Brogan <reilly@reillybrogan.com>
Date: Thu, 11 May 2023 22:37:44 -0500
Subject: [PATCH] Fix mangohud path on Solus

---
 overlayunit.pas | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/overlayunit.pas b/overlayunit.pas
index 1263f75..14a019d 100644
--- a/overlayunit.pas
+++ b/overlayunit.pas
@@ -5550,7 +5550,7 @@ begin
    //Determine Mangohud dependency status
 
           //locate MangoHud and store result in tmp folder
-          RunCommand('bash -c ''find /usr/share/vulkan/implicit_layer.d/MangoHud.json >> /tmp/goverlay/dependency_mangohud''', s);
+          RunCommand('bash -c ''find /usr/share/vulkan/implicit_layer.d/MangoHud.x86_64.json >> /tmp/goverlay/dependency_mangohud''', s);
 
           // Assign Text file dependency_mangohud to variable mangohudVAR
           AssignFile(mangohudVAR, '/tmp/goverlay/dependency_mangohud');
@@ -5559,7 +5559,7 @@ begin
           CloseFile(mangohudVAR);
 
           // Read String and store value on mangohuddependencyVALUE based on result
-          if mangohudSTR = '/usr/share/vulkan/implicit_layer.d/MangoHud.json' then
+          if mangohudSTR = '/usr/share/vulkan/implicit_layer.d/MangoHud.x86_64.json' then
           mangohuddependencyVALUE := 1
           else
           mangohuddependencyVALUE := 0;
-- 
2.40.1

