--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -1,5 +1,5 @@
 cmake_minimum_required(VERSION 3.1.3)
-project(binaryen LANGUAGES C CXX VERSION 98)
+project(binaryen LANGUAGES C CXX VERSION 99)
 include(GNUInstallDirs)
 
 # The C++ standard whose features are required to build Binaryen.
@@ -304,6 +304,8 @@
 else()
   message(STATUS "Building libbinaryen as shared library.")
   add_library(binaryen SHARED ${binaryen_SOURCES} ${binaryen_objs})
+  set_target_properties(binaryen PROPERTIES VERSION "0.0.${PROJECT_VERSION}")
+  set_target_properties(binaryen PROPERTIES SOVERSION 0)
 endif()
 install(TARGETS binaryen
   RUNTIME DESTINATION ${CMAKE_INSTALL_BINDIR}
