--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -300,6 +300,8 @@
 else()
   message(STATUS "Building libbinaryen as shared library.")
   add_library(binaryen SHARED ${binaryen_SOURCES} ${binaryen_objs})
+  set_target_properties(binaryen PROPERTIES VERSION ${PROJECT_VERSION})
+  set_target_properties(binaryen PROPERTIES SOVERSION 0)
 endif()
 install(TARGETS binaryen
   RUNTIME DESTINATION ${CMAKE_INSTALL_BINDIR}
