From 9aa04686907762659144e7dbc9e9cc15536c60c5 Mon Sep 17 00:00:00 2001
From: Joey Riches <josephriches@gmail.com>
Date: Sun, 6 Jul 2025 17:10:15 +0100
Subject: [PATCH 1/1] Don't require importlib_metadata on python >= 3.10

---
 setup.py               | 4 +---
 spyder/dependencies.py | 8 ++++++--
 2 files changed, 7 insertions(+), 5 deletions(-)

diff --git a/setup.py b/setup.py
index 17f6418b6..07fcfd8fc 100644
--- a/setup.py
+++ b/setup.py
@@ -276,9 +276,7 @@ def run(self):
     'cloudpickle>=0.5.0',
     'cookiecutter>=1.6.0',
     'diff-match-patch>=20181111',
-    # While this is only required for python <3.10, it is safe enough to
-    # install in all cases and helps the tests to pass.
-    'importlib-metadata>=4.6.0',
+    'importlib-metadata>=4.6.0;python_version<"3.10"',
     'intervaltree>=3.0.2',
     'ipython>=8.13.0,<9.0.0,!=8.17.1',
     'ipython_pygments_lexers>=1.0',
diff --git a/spyder/dependencies.py b/spyder/dependencies.py
index ec5e9aada..51d3f0ed9 100644
--- a/spyder/dependencies.py
+++ b/spyder/dependencies.py
@@ -18,6 +18,9 @@
 
 HERE = osp.dirname(osp.abspath(__file__))
 
+# Python 3.10
+PY310 = sys.version_info[:2] < (3, 10)
+
 # =============================================================================
 # Kind of dependency
 # =============================================================================
@@ -39,7 +42,7 @@
 CLOUDPICKLE_REQVER = '>=0.5.0'
 COOKIECUTTER_REQVER = '>=1.6.0'
 DIFF_MATCH_PATCH_REQVER = '>=20181111'
-IMPORTLIB_METADATA_REQVER = '>=4.6.0'
+IMPORTLIB_METADATA_REQVER = '>=4.6.0' if PY310 else None
 INTERVALTREE_REQVER = '>=3.0.2'
 IPYTHON_REQVER = ">=8.13.0,<9.0.0,!=8.17.1"
 IPYTHON_PYGMENTS_LEXERS_REQVER = ">=1.0"
@@ -135,7 +138,8 @@
     {'modname': 'importlib_metadata',
      'package_name': 'importlib-metadata',
      'features': _('Access the metadata for a Python package'),
-     'required_version': IMPORTLIB_METADATA_REQVER},
+     'required_version': IMPORTLIB_METADATA_REQVER,
+     'display': IMPORTLIB_METADATA_REQVER is not None},
     {'modname': "intervaltree",
      'package_name': "intervaltree",
      'features': _("Compute folding range nesting levels"),
-- 
2.49.0

