From: Phil Thompson <phil@riverbankcomputing.com>
Date: Mon, 25 Apr 2022 11:28:24 +0100
Subject: Fixed an attribute mis-naming in the parsing of exceptions.

Origin: upstream, https://riverbankcomputing.com/hg/sip/rev/d43fc89f7ef244b3.
---
 sipbuild/generator/parser/rules.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/sipbuild/generator/parser/rules.py b/sipbuild/generator/parser/rules.py
index d7f0dc7..fd2e01d 100644
--- a/sipbuild/generator/parser/rules.py
+++ b/sipbuild/generator/parser/rules.py
@@ -2250,7 +2250,7 @@ def p_opt_base_exception(p):
         base = p[2]
 
         # See if it is a project-defined exception.
-        for xd in p.parser_manager.spec.exceptions:
+        for xd in p.parser.pm.spec.exceptions:
             if xd.iface_file.fq_cpp_name == base:
                 defined_base = xd
                 break
