diff -up libdmtx-0.7.5/test/multi_test/Makefile.am.std libdmtx-0.7.5/test/multi_test/Makefile.am
--- libdmtx-0.7.5/test/multi_test/Makefile.am.std	2018-04-06 13:14:23.000000000 +0200
+++ libdmtx-0.7.5/test/multi_test/Makefile.am	2018-04-06 13:14:27.000000000 +0200
@@ -1,4 +1,4 @@
-AM_CPPFLAGS = -Wshadow -Wall -pedantic -ansi
+AM_CPPFLAGS = -Wshadow -Wall -pedantic -std=c99
 
 check_PROGRAMS = multi_test
 
diff -up libdmtx-0.7.5/test/rotate_test/Makefile.am.std libdmtx-0.7.5/test/rotate_test/Makefile.am
--- libdmtx-0.7.5/test/rotate_test/Makefile.am.std	2018-04-06 13:14:13.000000000 +0200
+++ libdmtx-0.7.5/test/rotate_test/Makefile.am	2018-04-06 13:14:18.000000000 +0200
@@ -1,4 +1,4 @@
-AM_CPPFLAGS = -Wshadow -Wall -pedantic -ansi
+AM_CPPFLAGS = -Wshadow -Wall -pedantic -std=c99
 
 check_PROGRAMS = rotate_test
 
diff -up libdmtx-0.7.5/test/simple_test/Makefile.am.std libdmtx-0.7.5/test/simple_test/Makefile.am
--- libdmtx-0.7.5/test/simple_test/Makefile.am.std	2018-04-06 13:13:59.000000000 +0200
+++ libdmtx-0.7.5/test/simple_test/Makefile.am	2018-04-06 13:14:04.000000000 +0200
@@ -1,4 +1,4 @@
-AM_CPPFLAGS = -Wshadow -Wall -pedantic -ansi
+AM_CPPFLAGS = -Wshadow -Wall -pedantic -std=c99
 
 check_PROGRAMS = simple_test
 
diff -up libdmtx-0.7.5/test/unit_test/Makefile.am.std libdmtx-0.7.5/test/unit_test/Makefile.am
--- libdmtx-0.7.5/test/unit_test/Makefile.am.std	2018-04-06 13:13:41.000000000 +0200
+++ libdmtx-0.7.5/test/unit_test/Makefile.am	2018-04-06 13:13:53.000000000 +0200
@@ -1,4 +1,4 @@
-AM_CPPFLAGS = -Wshadow -Wall -pedantic -ansi
+AM_CPPFLAGS = -Wshadow -Wall -pedantic -std=c99
 
 check_PROGRAMS = unit_test
 
