diff --git a/lsb_release b/lsb_release
index e7d6024..b99711b 100755
--- a/lsb_release
+++ b/lsb_release
@@ -42,7 +42,7 @@
 SCRIPTVERSION="1.4"
 
 # Defines the data files
-INFO_ROOT="/etc"                              # directory of config files
+INFO_ROOT="/usr/share/defaults/etc"           # directory of config files
 INFO_LSB_FILE="lsb-release"                   # where to get LSB version
 INFO_DISTRIB_SUFFIX="release"                 # <distrib>-<suffix>
 ALTERNATE_DISTRIB_FILE="/etc/debian_version"  # for Debian [based distrib]
