diff -uNrp a/pluma/pluma.c b/pluma/pluma.c
--- a/pluma/pluma.c	2024-02-18 23:43:06.000000000 +0000
+++ b/pluma/pluma.c	2025-09-10 13:19:12.681591749 +0100
@@ -44,7 +44,7 @@
 #include <gdk/gdkx.h>
 
 #ifdef HAVE_INTROSPECTION
-#include <girepository.h>
+#include <girepository/girepository.h>
 #endif
 
 #include "pluma-app.h"
@@ -527,7 +527,7 @@ main (int argc, char *argv[])
 	g_option_context_add_group (context, egg_sm_client_get_option_group ());
 
 #ifdef HAVE_INTROSPECTION
-	g_option_context_add_group (context, g_irepository_get_option_group ());
+	g_option_context_add_group (context, gi_repository_get_option_group ());
 #endif
 
 	if (!g_option_context_parse (context, &argc, &argv, &error))
diff -uNrp a/pluma/pluma-plugins-engine.c b/pluma/pluma-plugins-engine.c
--- a/pluma/pluma-plugins-engine.c	2024-02-18 23:43:06.000000000 +0000
+++ b/pluma/pluma-plugins-engine.c	2025-09-10 13:17:51.680109104 +0100
@@ -36,7 +36,7 @@
 #include <string.h>
 
 #include <glib/gi18n.h>
-#include <girepository.h>
+#include <girepository/girepository.h>
 
 #include "pluma-plugins-engine.h"
 #include "pluma-debug.h"
@@ -68,21 +68,21 @@ pluma_plugins_engine_init (PlumaPluginsE
 	engine->priv->plugin_settings = g_settings_new (PLUMA_SCHEMA_ID);
 
 	/* This should be moved to libpeas */
-	if (!g_irepository_require (g_irepository_get_default (),
+	if (!gi_repository_require (gi_repository_dup_default (),
 	                            "Peas", "1.0", 0, &error))
 	{
 		g_warning ("Could not load Peas repository: %s", error->message);
 		g_clear_error (&error);
 	}
 
-	if (!g_irepository_require (g_irepository_get_default (),
+	if (!gi_repository_require (gi_repository_dup_default (),
 	                            "PeasGtk", "1.0", 0, &error))
 	{
 		g_warning ("Could not load PeasGtk repository: %s", error->message);
 		g_clear_error (&error);
 	}
 
-	if (!g_irepository_require_private (g_irepository_get_default (),
+	if (!gi_repository_require_private (gi_repository_dup_default (),
 	                                    LIBDIR "/girepository-1.0",
 	                                    "Pluma", "1.0", 0, &error))
 	{
