From bca48c6ac35d1c9e827592dcd1df241193c26520 Mon Sep 17 00:00:00 2001
From: Campbell Jones <git@serebit.com>
Date: Fri, 5 Mar 2021 15:31:05 -0500
Subject: [PATCH] re-enable systemd-inhibit

---
 phoronix-test-suite | 10 +++++-----
 1 file changed, 5 insertions(+), 5 deletions(-)

diff --git a/phoronix-test-suite b/phoronix-test-suite
index d981c0a5a..837fb0e58 100755
--- a/phoronix-test-suite
+++ b/phoronix-test-suite
@@ -309,12 +309,12 @@ export PTS_EXT_LAUNCH_SCRIPT_DIR=`mktemp -d 2>/dev/null || mktemp -d -t ptstmp`
 PTS_PREPEND_LAUNCH=""
 
 # check for systemd-inhibit to handle suspending screensaver/idle
-#if which systemd-inhibit >/dev/null 2>&1
-#then
+if which systemd-inhibit >/dev/null 2>&1
+then
 # Disable by default again due to https://github.com/phoronix-test-suite/phoronix-test-suite/pull/491#issuecomment-762899039
-#	PTS_PREPEND_LAUNCH='systemd-inhibit --what=idle:sleep --who="Phoronix Test Suite" --why="Phoronix Test Suite benchmarking" '
-#	export PTS_SYSTEMD_INHIBITED=1 # Setting to 1 will make toggle_screensaver PTS module disable
-#fi
+	PTS_PREPEND_LAUNCH='systemd-inhibit --what=idle:sleep --who="Phoronix Test Suite" --why="Phoronix Test Suite benchmarking" '
+	export PTS_SYSTEMD_INHIBITED=1 # Setting to 1 will make toggle_screensaver PTS module disable
+fi
 
 # Run The Phoronix Test Suite
 PTS_EXIT_STATUS=8
-- 
2.30.0

