From 9b1f6b5b65abd1aa4076f0c85ed93d31e24277e7 Mon Sep 17 00:00:00 2001
From: Campbell Jones <git@serebit.com>
Date: Sun, 3 Jan 2021 14:51:39 -0500
Subject: [PATCH] Re-enable systemd-inhibit

---
 phoronix-test-suite | 9 +++++----
 1 file changed, 5 insertions(+), 4 deletions(-)

diff --git a/phoronix-test-suite b/phoronix-test-suite
index 5d978de..e19f333 100755
--- a/phoronix-test-suite
+++ b/phoronix-test-suite
@@ -294,15 +294,16 @@ PTS_PREPEND_LAUNCH=""
 if which systemd-inhibit >/dev/null 2>&1
 then
 	# At least as of September 2019, some versions of systemd-inhibit are buggy around argv handling
-	# PTS_PREPEND_LAUNCH="systemd-inhibit --what=idle:sleep --who=\"Phoronix Test Suite\" --why=\"Phoronix Test Suite benchmarking\" "
-	export PTS_SYSTEMD_INHIBITED=0 # Setting to 1 will make toggle_screensaver PTS module disable
+	PTS_PREPEND_LAUNCH='systemd-inhibit --what=idle:sleep --who="Phoronix Test Suite" --why="Phoronix Test Suite benchmarking" '
+    export PTS_SYSTEMD_INHIBITED=1 # Setting to 1 will make toggle_screensaver PTS module disable
 fi
 
 # Run The Phoronix Test Suite
 PTS_EXIT_STATUS=8
 while [ $PTS_EXIT_STATUS -eq 8 ]; do
-	$PTS_PREPEND_LAUNCH $PHP_BIN $PTS_DIR/pts-core/phoronix-test-suite.php $@
-	PTS_EXIT_STATUS=$?
+	PTS_LAUNCH_ARGS="$@"
+    sh -c "$PTS_PREPEND_LAUNCH $PHP_BIN $PTS_DIR/pts-core/phoronix-test-suite.php $PTS_LAUNCH_ARGS"
+    PTS_EXIT_STATUS=$?
 done
 
 if [ -f $PTS_EXT_LAUNCH_SCRIPT_DIR/web-server-launcher ]
-- 
2.30.0

