From d157d329dd0167629e210a35827cb9b6316b0d39 Mon Sep 17 00:00:00 2001
From: Even Rouault <even.rouault@spatialys.com>
Date: Sat, 18 Nov 2023 16:33:25 +0100
Subject: [PATCH] Fix build error with libxml2 2.12 (cont'd)

---
 gdal/gcore/gdaljp2metadatagenerator.cpp | 7 ++++++-
 1 file changed, 6 insertions(+), 1 deletion(-)

diff --git a/gdal/gcore/gdaljp2metadatagenerator.cpp b/gdal/gcore/gdaljp2metadatagenerator.cpp
index 26db3afe17..5e168a2b3a 100644
--- a/gdal/gcore/gdaljp2metadatagenerator.cpp
+++ b/gdal/gcore/gdaljp2metadatagenerator.cpp
@@ -345,7 +345,12 @@ static CPLString GDALGMLJP2EvalExpr(const CPLString& osTemplate,
 /************************************************************************/
 
 static void GDALGMLJP2XPathErrorHandler( void * /* userData */,
-                                         xmlErrorPtr error)
+#if LIBXML_VERSION >= 21200
+                                        const xmlError *error
+#else
+                                        xmlErrorPtr error
+#endif
+)
 {
     if( error->domain == XML_FROM_XPATH &&
         error->str1 != nullptr &&
-- 
2.43.0

