--- a/frmts/pdf/CMakeLists.txt
+++ b/frmts/pdf/CMakeLists.txt
@@ -31,6 +31,9 @@
   if (Poppler_VERSION_MAJOR GREATER 21)
     target_compile_features(gdal_PDF PRIVATE cxx_std_17)
   endif ()
+  if (Poppler_VERSION_STRING VERSION_GREATER_EQUAL "24.05")
+    target_compile_features(gdal_PDF PRIVATE cxx_std_20)
+  endif ()
   target_compile_definitions(gdal_PDF PRIVATE -DHAVE_POPPLER -DPOPPLER_MAJOR_VERSION=${Poppler_VERSION_MAJOR}
                                               -DPOPPLER_MINOR_VERSION=${Poppler_VERSION_MINOR})
 endif ()
