diff --git a/Directory.Build.props b/Directory.Build.props
index e6147318..0dd989e5 100644
--- a/Directory.Build.props
+++ b/Directory.Build.props
@@ -1,7 +1,7 @@
 <Project>
   <PropertyGroup>
     <!-- The target framework can be set with an environment variable of the same name -->
-    <TargetFramework Condition=" '$(TargetFramework)' == '' ">net7.0</TargetFramework>
+    <TargetFramework Condition=" '$(TargetFramework)' == '' ">net8.0</TargetFramework>
     <Nullable>enable</Nullable>
 
     <!-- Versioning and assembly info -->
diff --git a/configure.ac b/configure.ac
index 72c2ee0a..25ae2ce5 100644
--- a/configure.ac
+++ b/configure.ac
@@ -24,7 +24,7 @@ AX_COMPARE_VERSION([$DOTNET_VERSION], [ge], [$DOTNET_MINIMUM_VERSION],
 # Select target framework based on the dotnet version.
 DOTNET_TARGET_FRAMEWORK=net6.0
 AX_COMPARE_VERSION([$DOTNET_VERSION], [ge], [7.0],
-                   [DOTNET_TARGET_FRAMEWORK=net7.0], [])
+                   [DOTNET_TARGET_FRAMEWORK=net8.0], [])
 AC_MSG_NOTICE([Using target framework $DOTNET_TARGET_FRAMEWORK])
 
 DOTNET_CMD="env TargetFramework=$DOTNET_TARGET_FRAMEWORK $DOTNET"
