diff --git a/Directory.Build.props b/Directory.Build.props
index e6147318..0dd989e5 100644
--- a/Directory.Build.props
+++ b/Directory.Build.props
@@ -1,7 +1,7 @@
 <Project>
   <PropertyGroup>
     <!-- The target framework can be set with an environment variable of the same name -->
-    <TargetFramework Condition=" '$(TargetFramework)' == '' ">net7.0</TargetFramework>
+    <TargetFramework Condition=" '$(TargetFramework)' == '' ">net8.0</TargetFramework>
     <Nullable>enable</Nullable>
 
     <!-- Versioning and assembly info -->
diff --git a/configure.ac b/configure.ac
index 72c2ee0a..90b8a904 100644
--- a/configure.ac
+++ b/configure.ac
@@ -14,7 +14,7 @@ if test "x$DOTNET" = "xno"; then
 fi
 
 # Check dotnet version
-DOTNET_MINIMUM_VERSION=6.0
+DOTNET_MINIMUM_VERSION=7.0
 DOTNET_VERSION=$($DOTNET --version)
 AX_COMPARE_VERSION([$DOTNET_VERSION], [ge], [$DOTNET_MINIMUM_VERSION],
     AC_MSG_NOTICE([Found dotnet $DOTNET_VERSION]),
@@ -22,9 +22,8 @@ AX_COMPARE_VERSION([$DOTNET_VERSION], [ge], [$DOTNET_MINIMUM_VERSION],
 )
 
 # Select target framework based on the dotnet version.
-DOTNET_TARGET_FRAMEWORK=net6.0
-AX_COMPARE_VERSION([$DOTNET_VERSION], [ge], [7.0],
-                   [DOTNET_TARGET_FRAMEWORK=net7.0], [])
+DOTNET_MAJOR_VERSION=$(echo $DOTNET_VERSION | cut -d '.' -f 1,2)
+DOTNET_TARGET_FRAMEWORK="net$DOTNET_MAJOR_VERSION"
 AC_MSG_NOTICE([Using target framework $DOTNET_TARGET_FRAMEWORK])
 
 DOTNET_CMD="env TargetFramework=$DOTNET_TARGET_FRAMEWORK $DOTNET"
@@ -36,6 +35,7 @@ GETTEXT_PACKAGE=pinta
 AC_SUBST(GETTEXT_PACKAGE)
 AC_DEFINE_UNQUOTED(GETTEXT_PACKAGE, "$GETTEXT_PACKAGE")
 
+# Note: also need to update installer/macos/Info.plist when adding a language.
 ALL_LINGUAS="af ar ast az be bg bn br bs ca cs da de dv el en_AU en_CA en_GB eo es et eu fa fi fil fo fr ga gl he hi hr hu hy id ie it ja ka kab ko la lt lv ms my nb nl nn oc pl pt pt_BR ro ru si sk sl sq sr sv th tr uk vi zh_CN zh_HK zh_TW"
 AM_GLIB_GNU_GETTEXT
 
