Origin: upstream, d5a0650be5253038d2fcf31d1459b4dfd6cd500b
From: =?utf-8?q?Jeremy_B=C3=ADcha?= <jeremy.bicha@canonical.com>
Date: Tue, 23 Jan 2024 15:34:57 -0500
Subject: Build with webkit2gtk 4.1 and libsoup3 (#1139)

---
 INSTALL.md                     | 2 +-
 cmake/XiphosDependencies.cmake | 4 ++--
 2 files changed, 3 insertions(+), 3 deletions(-)

diff --git a/INSTALL.md b/INSTALL.md
index b4a492d..6ca500a 100644
--- a/INSTALL.md
+++ b/INSTALL.md
@@ -275,7 +275,7 @@ Create a build directory as a sibling of the xiphos directory:
 
 ## 5. Install the required dependencies:
 
-    $ sudo apt-get appstream-util cmake g++ desktop-file-utils fp-utils git gsettings-desktop-schemas-dev intltool itstool libdbus-glib-1-dev libenchant-dev libgail-3-dev libglade2-dev libgtk-3-dev libminizip-dev libsword-dev libwebkit2gtk-4.0-dev libxml2-dev libxml2-utils make python-dev swig uuid-dev uuid-runtime yelp-tools xzip
+    $ sudo apt-get appstream-util cmake g++ desktop-file-utils fp-utils git gsettings-desktop-schemas-dev intltool itstool libdbus-glib-1-dev libenchant-dev libgail-3-dev libglade2-dev libgtk-3-dev libminizip-dev libsword-dev libwebkit2gtk-4.1-dev libxml2-dev libxml2-utils make python-dev swig uuid-dev uuid-runtime yelp-tools xzip
 
 ## 6. Configuration:
 
diff --git a/cmake/XiphosDependencies.cmake b/cmake/XiphosDependencies.cmake
index 0a135b4..1b0b8f5 100644
--- a/cmake/XiphosDependencies.cmake
+++ b/cmake/XiphosDependencies.cmake
@@ -84,7 +84,7 @@ pkg_check_modules(Gnome REQUIRED IMPORTED_TARGET
   "gdk-pixbuf-2.0"
   "gio-2.0"
   "gobject-2.0"
-  "libsoup-2.4"
+  "libsoup-3.0"
   "pango"
   "minizip"
   "zlib"
@@ -144,7 +144,7 @@ else (GTK2)
     # Gtk+-3.0 + Webkit2 + GtkHtml-editor
      pkg_check_modules(Gtk REQUIRED IMPORTED_TARGET
       "gtk+-3.0"
-      "webkit2gtk-4.0"
+      "webkit2gtk-4.1"
       )
   endif()
 endif (GTK2)
