From: Bastian Germann <bage@debian.org>
Date: Mon, 27 Oct 2025 08:38:53 +0100
Subject: Replace appstream-util with appstream-cli
---
--- a/desktop/CMakeLists.txt
+++ b/desktop/CMakeLists.txt
@@ -90,7 +90,7 @@ if (UNIX)
   # add translations and validate xiphos.appdata.xml
   add_custom_command(OUTPUT ${CMAKE_CURRENT_BINARY_DIR}/${PROJECT_NAME}.appdata.xml
     COMMAND ${GETTEXT_MSGFMT_EXECUTABLE} --xml --template "${CMAKE_CURRENT_SOURCE_DIR}/${PROJECT_NAME}.appdata.xml" -d "${PROJECT_SOURCE_DIR}/po" -o "${CMAKE_CURRENT_BINARY_DIR}/${PROJECT_NAME}.appdata.xml"
-    COMMAND ${APPSTREAM_UTIL} validate-relax --nonet ${PROJECT_NAME}.appdata.xml > validate.timestamp
+    COMMAND ${APPSTREAM_UTIL} validate --no-net --override cid-desktopapp-is-not-rdns=info ${PROJECT_NAME}.appdata.xml > validate.timestamp
     WORKING_DIRECTORY ${CMAKE_CURRENT_BINARY_DIR}
     COMMENT "Configuring the AppData metadata file"
     )
--- a/cmake/XiphosBuildTools.cmake
+++ b/cmake/XiphosBuildTools.cmake
@@ -53,7 +53,7 @@ if (UNIX)
   # checking xiphos.desktop
   xiphos_find_program (DESKTOP_FILE_VALIDATE desktop-file-validate)
   # checking xiphos.appdata.xml
-  xiphos_find_program (APPSTREAM_UTIL appstream-util)
+  xiphos_find_program (APPSTREAM_UTIL appstreamcli)
 endif (UNIX)
 
 
