Include <wchar.h> for wcwidth prototype.

diff --git a/complete.c b/complete.c
index 35d86e915a7e74c5..df267e2754098ba2 100644
--- a/complete.c
+++ b/complete.c
@@ -82,6 +82,10 @@ typedef int QSFUNC ();
 #  define LSTAT stat
 #endif
 
+/* Ideally, we would build with -D_XOPEN_SOURCE.  */
+#include <wchar.h>
+int wcwidth (wchar_t);
+
 /* Unix version of a hidden file.  Could be different on other systems. */
 #define HIDDEN_FILE(fname)	((fname)[0] == '.')
 
diff --git a/display.c b/display.c
index 10a7b81cff278df9..dbf6db53300ee2ff 100644
--- a/display.c
+++ b/display.c
@@ -63,6 +63,10 @@
 extern char *strchr (), *strrchr ();
 #endif /* !strchr && !__STDC__ */
 
+/* Ideally, we would build with -D_XOPEN_SOURCE.  */
+#include <wchar.h>
+int wcwidth (wchar_t);
+
 static void update_line PARAMS((char *, char *, int, int, int, int));
 static void space_to_eol PARAMS((int));
 static void delete_chars PARAMS((int));
diff --git a/mbutil.c b/mbutil.c
index b036e0c3264fc71b..792c5ed59e5790c4 100644
--- a/mbutil.c
+++ b/mbutil.c
@@ -42,6 +42,10 @@
 #include <stdio.h>
 #include <ctype.h>
 
+/* Ideally, we would build with -D_XOPEN_SOURCE.  */
+#include <wchar.h>
+int wcwidth (wchar_t);
+
 /* System-specific feature definitions and include files. */
 #include "rldefs.h"
 #include "rlmbutil.h"
