diff -Nrup a/aclocal.m4 b/aclocal.m4
--- a/aclocal.m4	2013-10-20 16:37:50.000000000 -0600
+++ b/aclocal.m4	2020-01-14 10:41:28.792393160 -0700
@@ -1307,7 +1307,7 @@ AC_CACHE_VAL(bash_cv_must_reinstall_sigh
 
 typedef RETSIGTYPE sigfunc();
 
-int nsigint;
+volatile int nsigint;
 
 #ifdef HAVE_POSIX_SIGNALS
 sigfunc *
diff -Nrup a/configure b/configure
--- a/configure	2013-03-13 08:14:53.000000000 -0600
+++ b/configure	2020-01-14 11:34:47.043290210 -0700
@@ -5029,7 +5029,7 @@ else
 
 typedef RETSIGTYPE sigfunc();
 
-int nsigint;
+volatile int nsigint;
 
 #ifdef HAVE_POSIX_SIGNALS
 sigfunc *
