From a7b451c3cb200d51c29142ba3762c561a6714dcc Mon Sep 17 00:00:00 2001
From: Algent Albrahimi <algent@protonmail.com>
Date: Mon, 16 Dec 2019 20:32:53 +0100
Subject: [PATCH] Remove Help and Changelog references to docs

---
 src/layout_util.c | 4 ----
 1 file changed, 4 deletions(-)

diff --git a/src/layout_util.c b/src/layout_util.c
index 5fd058c..c264276 100644
--- a/src/layout_util.c
+++ b/src/layout_util.c
@@ -1984,8 +1984,6 @@ static GtkActionEntry menu_entries[] = {
   { "HelpSearch",	NULL,		N_("On-line help search"),			NULL,			N_("On-line help search"),				CB(layout_menu_help_search_cb) },
   { "HelpShortcuts",	NULL,			N_("_Keyboard shortcuts"),		NULL,			N_("Keyboard shortcuts"),		CB(layout_menu_help_keys_cb) },
   { "HelpKbd",		NULL,			N_("_Keyboard map"),			NULL,			N_("Keyboard map"),			CB(layout_menu_kbd_map_cb) },
-  { "HelpNotes",	NULL,			N_("_Release notes"),			NULL,			N_("Release notes"),			CB(layout_menu_notes_cb) },
-  { "HelpChangeLog",	NULL,			N_("_ChangeLog"),			NULL,			N_("ChangeLog notes"),			CB(layout_menu_changelog_cb) },
   { "About",		GTK_STOCK_ABOUT,	N_("_About"),				NULL,			N_("About"),				CB(layout_menu_about_cb) },
   { "LogWindow",	NULL,			N_("_Log Window"),			NULL,			N_("Log Window"),			CB(layout_menu_log_window_cb) },
   { "ExifWin",		PIXBUF_INLINE_ICON_EXIF,	N_("_Exif window"),			"<control>E",		N_("Exif window"),			CB(layout_menu_bar_exif_cb) },
@@ -2299,8 +2297,6 @@ static const gchar *menu_ui_description =
 "      <menuitem action='HelpSearch'/>"
 "      <menuitem action='HelpShortcuts'/>"
 "      <menuitem action='HelpKbd'/>"
-"      <menuitem action='HelpNotes'/>"
-"      <menuitem action='HelpChangeLog'/>"
 "      <placeholder name='HelpSection'/>"
 "      <separator/>"
 "      <menuitem action='About'/>"
-- 
2.24.1

