diff --git a/src/layout-util.cc b/src/layout-util.cc
index d47068a..6cae1d7 100644
--- a/src/layout-util.cc
+++ b/src/layout-util.cc
@@ -2596,8 +2596,6 @@ static GtkActionEntry menu_entries[] = {
   { "HelpSearch",	nullptr,		N_("On-line help search"),			nullptr,			N_("On-line help search"),				CB(layout_menu_help_search_cb) },
   { "HelpShortcuts",	nullptr,			N_("_Keyboard shortcuts"),		nullptr,			N_("Keyboard shortcuts"),		CB(layout_menu_help_keys_cb) },
   { "HelpKbd",		nullptr,			N_("_Keyboard map"),			nullptr,			N_("Keyboard map"),			CB(layout_menu_kbd_map_cb) },
-  { "HelpNotes",	nullptr,			N_("_Readme"),			nullptr,			N_("Readme"),			CB(layout_menu_notes_cb) },
-  { "HelpChangeLog",	nullptr,			N_("_ChangeLog"),			nullptr,			N_("ChangeLog notes"),			CB(layout_menu_changelog_cb) },
   { "SearchAndRunCommand",	GTK_STOCK_FIND,		N_("Search and Run command"),	"slash",	N_("Search commands by keyword and run them"),	CB(layout_menu_search_and_run_cb) },
   { "About",		GTK_STOCK_ABOUT,	N_("_About"),				nullptr,			N_("About"),				CB(layout_menu_about_cb) },
   { "LogWindow",	nullptr,			N_("_Log Window"),			nullptr,			N_("Log Window"),			CB(layout_menu_log_window_cb) },
@@ -2932,8 +2930,6 @@ static const gchar *menu_ui_description =
 "      <menuitem action='HelpSearch'/>"
 "      <menuitem action='HelpShortcuts'/>"
 "      <menuitem action='HelpKbd'/>"
-"      <menuitem action='HelpNotes'/>"
-"      <menuitem action='HelpChangeLog'/>"
 "      <placeholder name='HelpSection'/>"
 "      <separator/>"
 "      <menuitem action='About'/>"
-- 
2.41.0

