From d0f8724409c332fefee5bb8e2ca9661c8c07e182 Mon Sep 17 00:00:00 2001
From: Algent Albrahimi <algent@protonmail.com>
Date: Mon, 8 Aug 2022 18:12:53 +0200
Subject: [PATCH] help meson to find our lua

---
 meson.build | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/meson.build b/meson.build
index e44cf8c..7e9ca63 100644
--- a/meson.build
+++ b/meson.build
@@ -357,12 +357,12 @@ lua_dep = []
 req_version = '>=5.3'
 option = get_option('lua')
 if not option.disabled()
-    lua_dep = dependency('lua5.3',  version : req_version, required : get_option('lua'))
+    lua_dep = dependency('lua-5.3',  version : req_version, required : get_option('lua'))
     if lua_dep.found()
         conf_data.set('HAVE_LUA', 1)
         summary({'lua' : ['lua supported:', true]}, section : 'Configuration', bool_yn : true)
     else
-        summary({'lua' : ['lua5.3 ' + req_version + ' not found - lua supported:', false]}, section : 'Configuration', bool_yn : true)
+        summary({'lua' : ['lua-5.3 ' + req_version + ' not found - lua supported:', false]}, section : 'Configuration', bool_yn : true)
     endif
 else
     summary({'lua' : ['disabled - lua supported:', false]}, section : 'Configuration', bool_yn : true)
-- 
2.35.4

