From 45274ae0b50ce5458545bd5fb79fbce3c1965249 Mon Sep 17 00:00:00 2001
From: Evan Maddock <maddock.evan@vivaldi.net>
Date: Mon, 19 Dec 2022 16:08:45 -0500
Subject: [PATCH] Only compile the desktop library

Signed-off-by: Evan Maddock <maddock.evan@vivaldi.net>
---
 meson.build | 1 -
 1 file changed, 1 deletion(-)

diff --git a/meson.build b/meson.build
index 73fd815..e526b2d 100644
--- a/meson.build
+++ b/meson.build
@@ -110,7 +110,6 @@ configure_file(
 subdir('install-scripts')
 subdir('po')
 subdir('libcinnamon-desktop')
-subdir('libcvc')
 subdir('schemas')
 
 
-- 
2.39.0

