From ed3dcbfb962b5a49b60eb8bf0cc9d9951ef1b468 Mon Sep 17 00:00:00 2001
From: Ikey Doherty <ikey@solus-project.com>
Date: Wed, 3 Feb 2016 01:50:31 +0000
Subject: [PATCH] mtd_probe: Include stdint

Signed-off-by: Ikey Doherty <ikey@solus-project.com>
---
 src/udev/mtd_probe/mtd_probe.h | 1 +
 1 file changed, 1 insertion(+)

diff --git a/src/udev/mtd_probe/mtd_probe.h b/src/udev/mtd_probe/mtd_probe.h
index 2a37ede..49c1918 100644
--- a/src/udev/mtd_probe/mtd_probe.h
+++ b/src/udev/mtd_probe/mtd_probe.h
@@ -18,6 +18,7 @@
  */
 
 #include <mtd/mtd-user.h>
+#include <stdint.h>
 
 /* Full oob structure as written on the flash */
 struct sm_oob {
-- 
2.7.0

