From e81069c797f1fd77ce017614877c6548b30f20a0 Mon Sep 17 00:00:00 2001
From: Reilly Brogan <reilly@reillybrogan.com>
Date: Wed, 13 Dec 2023 20:31:00 -0600
Subject: [PATCH] Solus: Ensure /var/log/journal is created

---
 tmpfiles.d/systemd.conf.in | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/tmpfiles.d/systemd.conf.in b/tmpfiles.d/systemd.conf.in
index fa838d8d06..760a352f8f 100644
--- a/tmpfiles.d/systemd.conf.in
+++ b/tmpfiles.d/systemd.conf.in
@@ -39,7 +39,7 @@ a+ /run/log/journal/%m/*.journal* - - - - group:wheel:r--
 {% endif %}
 {% endif %}
 
-z /var/log/journal 2755 root systemd-journal - -
+d /var/log/journal 2755 root systemd-journal - -
 z /var/log/journal/%m 2755 root systemd-journal - -
 z /var/log/journal/%m/system.journal 0640 root systemd-journal - -
 {% if HAVE_ACL %}
-- 
2.43.0

