From ade0f820f7f5544a4e253e676c8bc6d41357759f Mon Sep 17 00:00:00 2001
From: Ikey Doherty <ikey@solus-project.com>
Date: Tue, 22 May 2018 19:41:23 +0100
Subject: [PATCH 2/2] sysusers: Explicitly set 9 as lp ID

Signed-off-by: Ikey Doherty <ikey@solus-project.com>
---
 sysusers.d/basic.conf.in | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/sysusers.d/basic.conf.in b/sysusers.d/basic.conf.in
index 8e358c0..1d36bdf 100644
--- a/sysusers.d/basic.conf.in
+++ b/sysusers.d/basic.conf.in
@@ -29,7 +29,7 @@ g dialout -     -            -
 g disk    -     -            -
 g input   -     -            -
 g kvm     -     -            -
-g lp      -     -            -
+g lp      9     -            -
 g render  -     -            -
 g tape    -     -            -
 g video   -     -            -
-- 
2.17.0

