From 4a61ccbde88b7989657d41f678de18b1108aba2a Mon Sep 17 00:00:00 2001
From: Joshua Strobl <joshua@streambits.io>
Date: Mon, 22 Jun 2020 23:29:10 +0300
Subject: [PATCH 1/1] Provide a default max retention and size policy.

---
 src/journal/journald.conf | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/journal/journald.conf b/src/journal/journald.conf
index 2e1aacd8c5..de69af9c3f 100644
--- a/src/journal/journald.conf
+++ b/src/journal/journald.conf
@@ -12,6 +12,8 @@
 # See journald.conf(5) for details.
 
 [Journal]
+SystemMaxUse=1G
+MaxRetentionSec=2week
 #Storage=auto
 #Compress=yes
 #Seal=yes
@@ -19,7 +21,6 @@
 #SyncIntervalSec=5m
 #RateLimitIntervalSec=30s
 #RateLimitBurst=10000
-#SystemMaxUse=
 #SystemKeepFree=
 #SystemMaxFileSize=
 #SystemMaxFiles=100
@@ -27,7 +28,6 @@
 #RuntimeKeepFree=
 #RuntimeMaxFileSize=
 #RuntimeMaxFiles=100
-#MaxRetentionSec=
 #MaxFileSec=1month
 #ForwardToSyslog=no
 #ForwardToKMsg=no
-- 
2.27.0

