From f857b703be2ae326dec7d2596c9d9d3740d02e10 Mon Sep 17 00:00:00 2001
From: Ikey Doherty <ikey@solus-project.com>
Date: Wed, 29 Nov 2017 22:44:36 +0000
Subject: [PATCH] sysusers: Explicitly set 9 as lp ID

Signed-off-by: Ikey Doherty <ikey@solus-project.com>
---
 sysusers.d/basic.conf.in | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/sysusers.d/basic.conf.in b/sysusers.d/basic.conf.in
index 7d6021e..cafafb3 100644
--- a/sysusers.d/basic.conf.in
+++ b/sysusers.d/basic.conf.in
@@ -28,7 +28,8 @@ g cdrom   -     -            -
 g dialout -     -            -
 g disk    -     -            -
 g input   -     -            -
-g lp      -     -            -
+g lp      9     -            -
+
 g kvm     -     -            -
 g tape    -     -            -
 g video   -     -            -
-- 
2.15.1

