From 17bbd4e7340bdd8a76bcab2c513caf57f919f9e5 Mon Sep 17 00:00:00 2001
From: Ikey Doherty <ikey@solus-project.com>
Date: Sun, 9 Oct 2016 06:21:41 +0100
Subject: [PATCH] login: Integrate with Solus PAM config

Signed-off-by: Ikey Doherty <ikey@solus-project.com>
---
 src/login/systemd-user | 11 +++++++++--
 1 file changed, 9 insertions(+), 2 deletions(-)

diff --git a/src/login/systemd-user b/src/login/systemd-user
index 8112d74..a0b6d79 100644
--- a/src/login/systemd-user
+++ b/src/login/systemd-user
@@ -2,5 +2,12 @@
 #
 # Used by systemd --user instances.
 
-account  include system-auth
-session  include system-auth
+account  required pam_access.so
+account  include  system-account
+
+session  required pam_env.so
+session  required pam_limits.so
+session  include  system-session
+
+auth     required pam_deny.so
+password required pam_deny.so
-- 
2.10.1

