From c600748f3cbbb4d38ca697d3345e31aacc81a894 Mon Sep 17 00:00:00 2001
From: Joshua Strobl <joshua@streambits.io>
Date: Mon, 13 Jan 2020 13:28:36 +0200
Subject: [PATCH 1/1] Do not try to set nsswitch.conf any more, this belongs to
 Solus

---
 tmpfiles.d/etc.conf.m4 | 1 -
 1 file changed, 1 deletion(-)

diff --git a/tmpfiles.d/etc.conf.m4 b/tmpfiles.d/etc.conf.m4
index f82e0b82ce..8475ce2f7f 100644
--- a/tmpfiles.d/etc.conf.m4
+++ b/tmpfiles.d/etc.conf.m4
@@ -15,7 +15,6 @@ t /etc/mtab - - - - security.SMACK64=_
 m4_ifdef(`ENABLE_RESOLVE',
 L! /etc/resolv.conf - - - - ../run/systemd/resolve/stub-resolv.conf
 )m4_dnl
-C! /etc/nsswitch.conf - - - -
 m4_ifdef(`HAVE_PAM',
 C! /etc/pam.d - - - -
 )m4_dnl
-- 
2.24.1

