From 344aaaa88084c62a186a02aec11426813b71e172 Mon Sep 17 00:00:00 2001
From: Ikey Doherty <ikey@solus-project.com>
Date: Sun, 8 Oct 2017 19:24:22 +0100
Subject: [PATCH 4/4] Do not try to set nsswitch.conf any more, this belongs to
 Solus

Signed-off-by: Ikey Doherty <ikey@solus-project.com>
---
 tmpfiles.d/etc.conf.m4 | 1 -
 1 file changed, 1 deletion(-)

diff --git a/tmpfiles.d/etc.conf.m4 b/tmpfiles.d/etc.conf.m4
index 35e3809..9b133bd 100644
--- a/tmpfiles.d/etc.conf.m4
+++ b/tmpfiles.d/etc.conf.m4
@@ -16,7 +16,6 @@ t /etc/mtab - - - - security.SMACK64=_
 m4_ifdef(`ENABLE_RESOLVE',
 L! /etc/resolv.conf - - - - ../usr/lib/systemd/resolv.conf
 )m4_dnl
-C /etc/nsswitch.conf - - - -
 m4_ifdef(`HAVE_PAM',
 C /etc/pam.d - - - -
 )m4_dnl
-- 
2.14.2

