From 09cd65233e0f5c04e7e3d063decf357f239a65f6 Mon Sep 17 00:00:00 2001
From: Povilas Kanapickas <povilas@radix.lt>
Date: Thu, 25 Mar 2021 22:48:56 +0200
Subject: [PATCH] xfree86: Set appropriate install_mode for suid_wrapper

---
 hw/xfree86/meson.build | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/hw/xfree86/meson.build b/hw/xfree86/meson.build
index 18ff27c83..a6311cb12 100644
--- a/hw/xfree86/meson.build
+++ b/hw/xfree86/meson.build
@@ -146,7 +146,7 @@ if get_option('suid_wrapper')
         c_args: xorg_c_args,
         install: true,
         install_dir: get_option('libexecdir'),
-        # install_mode: ['r-sr-xr-x', 0, 0],
+        install_mode: ['r-sr-xr-x', 0, 0],
     )
 
     # meson gets confused when there are two targets of the same name
-- 
GitLab

