From 07fe3c112f2de34d73270e66587978633f0c6786 Mon Sep 17 00:00:00 2001
From: Fabio <5798057+livingsilver94@users.noreply.github.com>
Date: Wed, 14 Feb 2024 10:13:50 +0100
Subject: [PATCH] sysinfo: Swap arguments of g_pattern_match_simple

---
 src/as-system-info.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/as-system-info.c b/src/as-system-info.c
index f0f0c64f4..be911a4e8 100644
--- a/src/as-system-info.c
+++ b/src/as-system-info.c
@@ -672,7 +672,7 @@ as_system_info_has_device_matching_modalias (AsSystemInfo *sysinfo, const gchar
 		if (g_strcmp0 (modalias, modalias_glob) == 0)
 			return TRUE;
 
-		if (g_pattern_match_simple (modalias, modalias_glob))
+		if (g_pattern_match_simple (modalias_glob, modalias))
 			return TRUE;
 	}
 
