From 120838f058f368b684d5509c47823b3f81a853cb Mon Sep 17 00:00:00 2001
From: Peter O'Connor <peter@solus-project.com>
Date: Sun, 15 Apr 2018 02:16:31 +1000
Subject: [PATCH] Disable building unshipped code

Signed-off-by: Peter O'Connor <peter@solus-project.com>
---
 CMakeLists.txt | 2 --
 1 file changed, 2 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 5abf78f..7b89669 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -70,8 +70,6 @@ ecm_setup_version (
    )
 
 add_subdirectory (src)
-add_subdirectory (autotests)
-add_subdirectory (tests)
 
 set (CMAKECONFIG_INSTALL_DIR "${KDE_INSTALL_CMAKEPACKAGEDIR}/KF5Activities")
 
-- 
2.16.3

