From 54d28c183ce00dd020e19a15ffd167c64bf57aad Mon Sep 17 00:00:00 2001
From: Algent Albrahimi <algent@protonmail.com>
Date: Thu, 20 Oct 2022 18:56:48 +0200
Subject: [PATCH] Fix icon and desktop file installation

Signed-off-by: Algent Albrahimi <algent@protonmail.com>
---
 CMakeLists.txt | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 952f10c..842e5f4 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -1224,12 +1224,12 @@ if(UNIX AND NOT APPLE)
    #----------- Linux -----------
    # Install the icons
    install(FILES ${filesToInstall_icons}
-           DESTINATION "${installSubDir_data}/icons/hicolor/scalable/apps/"
+           DESTINATION "/usr/share/icons/hicolor/scalable/apps/"
            COMPONENT ${DATA_INSTALL_COMPONENT})
 
    # Install the .desktop file
    install(FILES ${filesToInstall_desktop}
-           DESTINATION "${installSubDir_data}/applications"
+           DESTINATION "/usr/share/applications"
            COMPONENT ${DATA_INSTALL_COMPONENT})
 
    # Install friendly-format change log aka release notes
-- 
2.35.5

