From 3007387745cf84138d0855e0f04ff94261fc7175 Mon Sep 17 00:00:00 2001
From: Nicolas Werner <nicolas.werner@hotmail.de>
Date: Tue, 22 Aug 2023 03:55:28 +0200
Subject: [PATCH] Get rid of provide section since it is unnecessary

---
 meson.build           | 2 +-
 subprojects/curl.wrap | 3 ---
 2 files changed, 1 insertion(+), 4 deletions(-)

diff --git a/meson.build b/meson.build
index 7e260e9..2ad5075 100644
--- a/meson.build
+++ b/meson.build
@@ -1,7 +1,7 @@
 project('coeurl', 'cpp',
   version : '0.3.0',
   license : 'MIT',
-  meson_version : '>=0.54',
+  meson_version : '>=0.55',
   default_options : ['warning_level=3', 'cpp_std=c++17']
   )
 
diff --git a/subprojects/curl.wrap b/subprojects/curl.wrap
index ec9710f..edd210c 100644
--- a/subprojects/curl.wrap
+++ b/subprojects/curl.wrap
@@ -4,6 +4,3 @@ directory = curl-7.77.0
 source_url = https://github.com/curl/curl/releases/download/curl-7_77_0/curl-7.77.0.tar.xz
 source_filename = curl-7.77.0.tar.xz
 source_hash = 0f64582c54282f31c0de9f0a1a596b182776bd4df9a4c4a2a41bbeb54f62594b
-
-[provides]
-dependency_names = libcurl
-- 
GitLab

