From 796334b9fd7ca1b2008410a5cf69f1a644915cbe Mon Sep 17 00:00:00 2001
From: Joshua Strobl <joshua@streambits.io>
Date: Mon, 28 Dec 2020 18:06:51 +0200
Subject: [PATCH 1/1] Patch to fall back to fixed stateless dir.

---
 dnscrypt-proxy/config.go | 18 ++++++++++--------
 1 file changed, 10 insertions(+), 8 deletions(-)

diff --git a/dnscrypt-proxy/config.go b/dnscrypt-proxy/config.go
index d74874e..30cee75 100644
--- a/dnscrypt-proxy/config.go
+++ b/dnscrypt-proxy/config.go
@@ -283,10 +283,12 @@ type ConfigFlags struct {
 	ShowCerts               *bool
 }
 
-func findConfigFile(configFile *string) (string, error) {
-	if _, err := os.Stat(*configFile); os.IsNotExist(err) {
-		cdLocal()
-		if _, err := os.Stat(*configFile); err != nil {
+func findConfigFile(configFile string) (string, error) {
+	if _, err := os.Stat(configFile); os.IsNotExist(err) {
+		baseConfigName := filepath.Base(configFile) // Get the filepath
+		configFile = "/usr/share/defaults/dnscrypt-proxy/"+baseConfigName
+
+		if _, err := os.Stat(configFile); os.IsNotExist(err) {
 			return "", err
 		}
 	}
@@ -294,14 +296,14 @@ func findConfigFile(configFile *string) (string, error) {
 	if err != nil {
 		return "", err
 	}
-	if filepath.IsAbs(*configFile) {
-		return *configFile, nil
+	if filepath.IsAbs(configFile) {
+		return configFile, nil
 	}
-	return path.Join(pwd, *configFile), nil
+	return path.Join(pwd, configFile), nil
 }
 
 func ConfigLoad(proxy *Proxy, flags *ConfigFlags) error {
-	foundConfigFile, err := findConfigFile(flags.ConfigFile)
+	foundConfigFile, err := findConfigFile(*flags.ConfigFile)
 	if err != nil {
 		return fmt.Errorf("Unable to load the configuration file [%s] -- Maybe use the -config command-line switch?", *flags.ConfigFile)
 	}
-- 
2.29.2

