--- a/substratevm/mx.substratevm/mx_substratevm.py
+++ b/substratevm/mx.substratevm/mx_substratevm.py
@@ -29,7 +29,7 @@
 from glob import glob
 from contextlib import contextmanager
 from os.path import join, exists, dirname
-import pipes
+import shutil
 from argparse import ArgumentParser
 import fnmatch
 import collections
@@ -633,7 +633,7 @@
         unittest_image = native_image(['-ea', '-esa'] + build_args + extra_image_args + [macro_junit + '=' + unittest_file] + svm_experimental_options(['-H:Path=' + junit_test_dir]), env=custom_env)
         image_pattern_replacement = unittest_image + ".exe" if mx.is_windows() else unittest_image
         run_args = [arg.replace('${unittest.image}', image_pattern_replacement) for arg in run_args]
-        mx.log('Running: ' + ' '.join(map(pipes.quote, [unittest_image] + run_args)))
+        mx.log('Running: ' + ' '.join(map(shutil.quote, [unittest_image] + run_args)))
         mx.run([unittest_image] + run_args)
     finally:
         if not preserve_image:
