diff --git a/CMakeLists.txt b/CMakeLists.txt
index bb5650d..2f6f234 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -133,18 +133,6 @@ install(
     USE_SOURCE_PERMISSIONS
     PATTERN "hipify-perl"
     PATTERN "*.sh")
-install(
-    DIRECTORY ${LLVM_DIR}/../../clang/${LLVM_VERSION_MAJOR}.${LLVM_VERSION_MINOR}.${LLVM_VERSION_PATCH}/
-    DESTINATION ${CMAKE_INSTALL_PREFIX}
-    COMPONENT clang-resource-headers
-    FILES_MATCHING
-    PATTERN "*.h"
-    PATTERN "*.modulemap"
-    PATTERN "algorithm"
-    PATTERN "complex"
-    PATTERN "new"
-    PATTERN "ppc_wrappers" EXCLUDE
-    PATTERN "openmp_wrappers" EXCLUDE)
 
 option(FILE_REORG_BACKWARD_COMPATIBILITY "Enable File Reorg with backward compatibility" ON)
 
