From c477ebda825f0179e0f44bab01e0000fee37ea2e Mon Sep 17 00:00:00 2001
From: Robert Scheck <robert-scheck@users.noreply.github.com>
Date: Wed, 26 Apr 2023 23:12:44 +0200
Subject: [PATCH] Disable man pages for languages without any translation

---
 Makefile.am | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/Makefile.am b/Makefile.am
index e44070c17..1b20b7ddc 100644
--- a/Makefile.am
+++ b/Makefile.am
@@ -899,7 +899,7 @@ stamps/generate-docs.stamp: stamps/netx.stamp
 	LANG_BACKUP=$$LANG ; \
 	echo "<html><head><title>$(PLUGIN_VERSION)</title></head>" > "$$HTML_DOCS_INDEX" ; \
 	echo "<body><h3>$(PLUGIN_VERSION) docs:</h3>"  >> "$$HTML_DOCS_INDEX" ; \
-	for LANG_ID in en_US.UTF-8 cs_CZ.UTF-8 pl_PL.UTF-8 de_DE.UTF-8 ;  do \
+	for LANG_ID in en_US.UTF-8 ;  do \
 	  ID=`echo "$$LANG_ID" | head -c 2` ; \
 	  ENCOD=`echo "$$LANG_ID" | tail -c 6 -` ; \
 	  export LANG=$$LANG_ID; \
