--- a/Makefile.am
+++ b/Makefile.am
@@ -293,7 +293,7 @@
 JAVADOC_OPTS+=--add-exports=java.base/sun.security.util=ALL-UNNAMED
 JAVADOC_OPTS+=--add-exports=java.base/sun.security.validator=ALL-UNNAMED
 JAVADOC_OPTS+=--add-exports=java.base/sun.security.x509=ALL-UNNAMED
-JAVADOC_OPTS+=--add-exports=java.base/sun.net.www.protocol.jar=ALL-UNNAMED
+JAVADOC_OPTS+=--add-exports=java.base/sun.net.www.protocol.jar=ALL-UNNAMED --ignore-source-errors
 JAVADOC_OPTS+=--add-exports=java.desktop/sun.awt=ALL-UNNAMED
 endif
 if JAVADOC_SUPPORTS_J_OPTIONS
