From 3e0b015f7d018010bdd6e375b9d4623ba03251ba Mon Sep 17 00:00:00 2001
From: Joey Riches <josephriches@gmail.com>
Date: Wed, 22 Nov 2017 01:49:31 +0000
Subject: [PATCH 1/1] linux: don't build a deb or rpm please

Signed-off-by: Joey Riches <josephriches@gmail.com>
---
 app/build/Gruntfile.js | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/app/build/Gruntfile.js b/app/build/Gruntfile.js
index 42837a2eb..f3a499a69 100644
--- a/app/build/Gruntfile.js
+++ b/app/build/Gruntfile.js
@@ -58,6 +58,6 @@ module.exports = grunt => {
   } else if (grunt.option('platform') === 'darwin') {
     grunt.registerTask('build-client', ['package', 'create-mac-zip', 'create-mac-dmg']);
   } else if (grunt.option('platform') === 'linux') {
-    grunt.registerTask('build-client', ['package', 'create-deb-installer', 'create-rpm-installer']);
+    grunt.registerTask('build-client', ['package']);
   }
 };
-- 
2.35.4

