diff --git a/app/build/tasks/package-task.js b/app/build/tasks/package-task.js
index db1533dcc..14185cc9f 100644
--- a/app/build/tasks/package-task.js
+++ b/app/build/tasks/package-task.js
@@ -221,7 +221,6 @@ module.exports = grunt => {
       appBundleId: 'com.mailspring.mailspring',
       afterCopy: [
         runCopyPlatformSpecificResources,
-        runWriteCommitHashIntoPackage,
         runCopySymlinkedPackages,
         runTranspilers,
       ],
