diff --git a/app/build/tasks/package-task.js b/app/build/tasks/package-task.js
index db1533d..14185cc 100644
--- a/app/build/tasks/package-task.js
+++ b/app/build/tasks/package-task.js
@@ -221,7 +221,6 @@ module.exports = grunt => {
       appBundleId: 'com.mailspring.mailspring',
       afterCopy: [
         runCopyPlatformSpecificResources,
-        runWriteCommitHashIntoPackage,
         runCopySymlinkedPackages,
         runTranspilers,
       ],
diff --git a/app/src/browser/main.js b/app/src/browser/main.js
index 35c86dc..9bdbd26 100644
--- a/app/src/browser/main.js
+++ b/app/src/browser/main.js
@@ -120,7 +120,7 @@ const declareOptions = argv => {
 
 const parseCommandLine = argv => {
   const pkg = require('../../package.json');
-  const version = `${pkg.version}-${pkg.commitHash}`;
+  const version = `${pkg.version}`;
 
   const options = declareOptions(argv.slice(1));
   const args = options.argv;
