From be8eb413c38f06e35de7ff5a224ea618d37e66fb Mon Sep 17 00:00:00 2001
From: Reilly Brogan <solus@reillybrogan.com>
Date: Wed, 17 Aug 2022 16:15:26 -0500
Subject: [PATCH 1/1] Don't download mailsync binaries

---
 scripts/postinstall.js | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/scripts/postinstall.js b/scripts/postinstall.js
index 17ec022df..7b693faa6 100644
--- a/scripts/postinstall.js
+++ b/scripts/postinstall.js
@@ -188,8 +188,8 @@ async function run() {
   // if the user hasn't cloned the mailsync module, download
   // the binary for their operating system that was shipped to S3.
   if (!fs.existsSync('./mailsync/build.sh')) {
-    console.log(`\n-- Downloading the last released version of Mailspring mailsync --`);
-    downloadMailsync();
+    console.log(`\n-- The packagers had the foresight to download the mailsync binaries ahead of time! Thanks packagers :) --`);
+    // downloadMailsync();
   } else {
     console.log(
       `\n-- You have the Mailspring mailsync submodule. If you'd prefer ` +
-- 
2.35.4

