From 08d0a66e1195142e097748f51f6bb74b3064f80b Mon Sep 17 00:00:00 2001
From: Peter O'Connor <peter@solus-project.com>
Date: Fri, 20 Apr 2018 10:17:02 +1000
Subject: [PATCH] Disable building unshipped code

Signed-off-by: Peter O'Connor <peter@solus-project.com>
---
 CMakeLists.txt | 1 -
 1 file changed, 1 deletion(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index bc96fe7..32b5345 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -148,7 +148,6 @@ add_subdirectory(lib)
 add_subdirectory(app)
 add_subdirectory(importer)
 add_subdirectory(part)
-add_subdirectory(tests)
 add_subdirectory(icons)
 add_subdirectory(images)
 add_subdirectory(cursors)
-- 
2.16.3

