From c8938ce8b68892109e83f3b4d57b8519b8849b99 Mon Sep 17 00:00:00 2001
From: "F. von Gellhorn" <flinux@vongellhorn.ch>
Date: Sun, 13 Jun 2021 10:54:47 +0200
Subject: [PATCH 1/1] Fix ABI creation

---
 src/2geom/CMakeLists.txt | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/src/2geom/CMakeLists.txt b/src/2geom/CMakeLists.txt
index df03fc1..9eb1253 100755
--- a/src/2geom/CMakeLists.txt
+++ b/src/2geom/CMakeLists.txt
@@ -180,7 +180,8 @@ target_link_libraries(2geom
         ${DoubleConversion_LIBRARIES}
     )
 
-set_target_properties(2geom PROPERTIES SOVERSION "${2GEOM_ABI_VERSION}")
+set_target_properties(2geom PROPERTIES SOVERSION "${2GEOM_MAJOR_VERSION}")
+set_target_properties(2geom PROPERTIES VERSION "${2GEOM_ABI_VERSION}")
 
 install(TARGETS 2geom
     EXPORT 2geom_targets
-- 
2.30.2

