From a275861240b6980e61ae1a004b8cfb9a82f025d0 Mon Sep 17 00:00:00 2001
From: Thomas Staudinger <Staudi.Kaos@gmail.com>
Date: Wed, 14 Dec 2022 23:53:35 +0100
Subject: [PATCH] Fix version number

---
 CMakeLists.txt | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 4d1cd2a..d2b06b9 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -3,8 +3,8 @@ cmake_policy(SET CMP0054 NEW)
 
 
 set(2GEOM_MAJOR_VERSION 1)
-set(2GEOM_MINOR_VERSION 1)
-set(2GEOM_PATCH_VERSION 0)
+set(2GEOM_MINOR_VERSION 2)
+set(2GEOM_PATCH_VERSION 2)
 set(2GEOM_VERSION ${2GEOM_MAJOR_VERSION}.${2GEOM_MINOR_VERSION}.${2GEOM_PATCH_VERSION})
 set(2GEOM_ABI_VERSION ${2GEOM_MAJOR_VERSION}.${2GEOM_MINOR_VERSION}.0)
 
-- 
2.35.5

