From 08e354bcb6fe29e29fc86900ac11724174928949 Mon Sep 17 00:00:00 2001
From: David Harder <david@davidjharder.ca>
Date: Wed, 14 Feb 2024 20:03:11 -0600
Subject: [PATCH] dont-need-post-install-steps

---
 data/meson.build | 7 -------
 1 file changed, 7 deletions(-)

diff --git a/data/meson.build b/data/meson.build
index f9dac58..07e30c6 100644
--- a/data/meson.build
+++ b/data/meson.build
@@ -107,10 +107,3 @@ gnome.compile_resources(
   install: true,
   install_dir: join_paths(datadir, meson.project_name())
 )
-
-# Run required post-install steps
-gnome.post_install(
-  gtk_update_icon_cache: true,
-  glib_compile_schemas: true,
-  update_desktop_database: true,
-)
-- 
2.43.1

