diff --git a/src/common/config/buildConfig.ts b/src/common/config/buildConfig.ts
index 1fc051bc..a11965ae 100644
--- a/src/common/config/buildConfig.ts
+++ b/src/common/config/buildConfig.ts
@@ -33,7 +33,7 @@ const buildConfig: BuildConfig = {
     academyLink: DEFAULT_ACADEMY_LINK,
     upgradeLink: DEFAULT_UPGRADE_LINK,
     enableServerManagement: true,
-    enableUpdateNotifications: true,
+    enableUpdateNotifications: false,
     updateNotificationURL: 'https://releases.mattermost.com/desktop',
     macAppStoreUpdateURL: 'macappstore://apps.apple.com/us/app/mattermost-desktop/id1614666244',
     windowsStoreUpdateURL: 'ms-windows-store://pdp/?productid=XP8BR8MH3LPKLT',
