diff -ruNb a/configure.ac b/configure.ac
--- a/configure.ac	2009-05-16 08:27:39.000000000 -0700
+++ b/configure.ac	2013-04-26 13:21:33.978915825 -0700
@@ -225,7 +225,6 @@
          tests/Makefile build-aux/Makefile
 	find/Makefile find/testsuite/Makefile
 	xargs/Makefile xargs/testsuite/Makefile
-	locate/Makefile locate/testsuite/Makefile
 	 po/Makefile.in po/Makefile
 	doc/Makefile lib/Makefile
 	])
diff -ruNb a/Makefile.am b/Makefile.am
--- a/Makefile.am	2009-05-16 08:17:01.000000000 -0700
+++ b/Makefile.am	2013-04-26 13:21:05.075968189 -0700
@@ -8,7 +8,7 @@
 
 
 # "tests" is the gnulib unit test dir.
-SUBDIRS = gnulib tests build-aux lib find xargs locate doc po m4 
+SUBDIRS = gnulib tests build-aux lib find xargs doc po m4 
 
 ACLOCAL_AMFLAGS = -I gnulib/m4 -I m4
 
diff -ruNb a/Makefile.in b/Makefile.in
--- a/Makefile.in	2009-05-16 08:28:13.000000000 -0700
+++ b/Makefile.in	2013-04-26 13:19:41.203049767 -0700
@@ -575,7 +575,7 @@
 DISTCLEANFILES = tool-versions.txt
 
 # "tests" is the gnulib unit test dir.
-SUBDIRS = gnulib tests build-aux lib find xargs locate doc po m4 
+SUBDIRS = gnulib tests build-aux lib find xargs doc po m4 
 ACLOCAL_AMFLAGS = -I gnulib/m4 -I m4
 TESTFILE_SUFFIXES = .exp .xo .xe .xi
 all: config.h
