diff --git a/CMakeLists.txt b/CMakeLists.txt
index f7e78371b0..25f567ef9c 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -220,8 +220,13 @@ if( CMAKE_BUILD_TYPE_TOLOWER MATCHES debug )
     endif()
 endif()
 
+# this needs to be here because also code in shared/ needs config.h. This is also the
+# reason why various checks are above why they belong under if( WITH_PLAYER )
+configure_file( shared/config.h.cmake ${CMAKE_CURRENT_BINARY_DIR}/shared/config.h )
+
 add_subdirectory( data )
 add_subdirectory( images )
+add_subdirectory( shared )
 
 if( WITH_PLAYER )
     find_package(Threads REQUIRED)
@@ -365,12 +370,6 @@ if( WITH_PLAYER )
     kdoctools_install( po )
 endif()
 
-# this needs to be here because also code in shared/ needs config.h. This is also the
-# reason why various checks are above why they belong under if( WITH_PLAYER )
-configure_file( shared/config.h.cmake ${CMAKE_CURRENT_BINARY_DIR}/shared/config.h )
-
-add_subdirectory( shared )
-
 if( WITH_UTILITIES )
     add_subdirectory( utilities )
     ecm_install_po_files_as_qm( poqm )
