diff --git a/cmd/tailscaled/tailscaled.service b/cmd/tailscaled/tailscaled.service
index d5a65a4..2a3825a 100644
--- a/cmd/tailscaled/tailscaled.service
+++ b/cmd/tailscaled/tailscaled.service
@@ -5,7 +5,8 @@ Wants=network-pre.target
 After=network-pre.target NetworkManager.service systemd-resolved.service
 
 [Service]
-EnvironmentFile=/etc/default/tailscaled
+EnvironmentFile=/usr/share/defaults/tailscale/tailscaled
+EnvironmentFile=-/etc/default/tailscaled
 ExecStartPre=/usr/sbin/tailscaled --cleanup
 ExecStart=/usr/sbin/tailscaled --state=/var/lib/tailscale/tailscaled.state --socket=/run/tailscale/tailscaled.sock --port=${PORT} $FLAGS
 ExecStopPost=/usr/sbin/tailscaled --cleanup
