From 760cca1e5eadbc59336f4d7e79634a730fb8bb6e Mon Sep 17 00:00:00 2001
From: Ikey Doherty <ikey@solusos.com>
Date: Sun, 6 Oct 2013 12:11:35 +0100
Subject: [PATCH] Disable man pages due to build failure

---
 Makefile.am | 2 +-
 Makefile.in | 2 +-
 2 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/Makefile.am b/Makefile.am
index c01c54f..dcdf7a2 100644
--- a/Makefile.am
+++ b/Makefile.am
@@ -1,6 +1,6 @@
 NULL =
 
-SUBDIRS=po man
+SUBDIRS=po
 
 INCLUDES =					\
 	-I$(top_srcdir)				\
diff --git a/Makefile.in b/Makefile.in
index a76065b..bdf5306 100644
--- a/Makefile.in
+++ b/Makefile.in
@@ -302,7 +302,7 @@ top_build_prefix = @top_build_prefix@
 top_builddir = @top_builddir@
 top_srcdir = @top_srcdir@
 NULL = 
-SUBDIRS = po man
+SUBDIRS = po
 INCLUDES = \
 	-I$(top_srcdir)				\
 	-I$(top_builddir)			\
-- 
1.8.4

