From 1da17db6049adea06fe39076c3ffd875557e2b48 Mon Sep 17 00:00:00 2001
From: Philipp Trulson <philipp@trulson.de>
Date: Fri, 9 Feb 2018 10:18:14 +0100
Subject: [PATCH] Fix error when starting with symlink

---
 ts3client_runscript.sh | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/ts3client_runscript.sh b/ts3client_runscript.sh
index 723c792..721eb74 100755
--- a/ts3client_runscript.sh
+++ b/ts3client_runscript.sh
@@ -1,6 +1,6 @@
 #!/bin/bash
 
-cd "$(dirname "${BASH_SOURCE[0]}")"
+cd "$(dirname "$(readlink -f "${BASH_SOURCE[0]}")")"
 
 #If the bundled openssl is used, the app needs to find out where the CA certs
 #are located. This needs to be done with the help of openssl installed on the
-- 
2.16.1

