--- pytorch-v2.5.1/third_party/gloo/gloo/types.h.old	2025-06-05 12:18:00.347953998 +0000
+++ pytorch-v2.5.1/third_party/gloo/gloo/types.h	2025-06-05 12:35:17.320482224 +0000
@@ -4,6 +4,7 @@
 
 #pragma once
 
+#include <cstdint>
 #include <iostream>
 
 #ifdef __CUDA_ARCH__
--- pytorch-v2.5.1/third_party/tensorpipe/tensorpipe/common/allocator.h.old	2025-06-05 12:44:21.234924516 +0000
+++ pytorch-v2.5.1/third_party/tensorpipe/tensorpipe/common/allocator.h	2025-06-05 12:44:45.078682582 +0000
@@ -9,6 +9,7 @@
 #pragma once
 
 #include <cstddef>
+#include <cstdint>
 #include <deque>
 #include <functional>
 #include <memory>
--- pytorch-v2.5.1/third_party/tensorpipe/tensorpipe/common/memory.h.old	2025-06-05 13:17:24.141155629 +0000
+++ pytorch-v2.5.1/third_party/tensorpipe/tensorpipe/common/memory.h	2025-06-05 13:17:37.602567825 +0000
@@ -10,6 +10,7 @@
 
 #include <sys/mman.h>
 
+#include <cstdint>
 #include <memory>
 
 #include <tensorpipe/common/defs.h>
