diff --git i/cmake/Dependencies.cmake w/cmake/Dependencies.cmake
index ef5c2fd4e9..3ee1c9e816 100644
--- i/cmake/Dependencies.cmake
+++ w/cmake/Dependencies.cmake
@@ -1054,7 +1054,7 @@ if(USE_ROCM)
     hip_include_directories(${Caffe2_HIP_INCLUDE})
 
     set(Caffe2_PUBLIC_HIP_DEPENDENCY_LIBS
-      hip::amdhip64 MIOpen hiprtc::hiprtc) # libroctx will be linked in with MIOpen
+      hip::amdhip64 MIOpen hiprtc::hiprtc rocm_smi64) # libroctx will be linked in with MIOpen
 
     # Math libraries
     list(APPEND Caffe2_PUBLIC_HIP_DEPENDENCY_LIBS
diff --git i/cmake/public/LoadHIP.cmake w/cmake/public/LoadHIP.cmake
index 018bca837a..6343b6517a 100644
--- i/cmake/public/LoadHIP.cmake
+++ w/cmake/public/LoadHIP.cmake
@@ -82,6 +82,7 @@ find_package_and_print_version(HIP 1.0 MODULE)
 
 if(HIP_FOUND)
   set(PYTORCH_FOUND_HIP TRUE)
+  enable_language(HIP)
   find_package_and_print_version(hip REQUIRED CONFIG)
 
   # The rocm-core package was only introduced in ROCm 6.4, so we make it optional.
@@ -157,6 +158,7 @@ if(HIP_FOUND)
   # Find ROCM components using Config mode
   # These components will be searced for recursively in ${ROCM_PATH}
   message("\n***** Library versions from cmake find_package *****\n")
+  find_package_and_print_version(rocm_smi REQUIRED)
   find_package_and_print_version(amd_comgr REQUIRED)
   find_package_and_print_version(rocrand REQUIRED)
   find_package_and_print_version(hiprand REQUIRED)
diff --git i/cmake/public/utils.cmake w/cmake/public/utils.cmake
index 68e66bb3fc..3440a191c2 100644
--- i/cmake/public/utils.cmake
+++ w/cmake/public/utils.cmake
@@ -337,7 +337,9 @@ endmacro()
 # Usage:
 #   torch_compile_options(lib_name)
 function(torch_compile_options libname)
-  set_property(TARGET ${libname} PROPERTY CXX_STANDARD 17)
+  set_property(TARGET ${libname} PROPERTY
+    CXX_STANDARD 17
+    HIP_STANDARD 17)
 
   # until they can be unified, keep these lists synced with setup.py
   if(MSVC)
