diff '--color=auto' -pruN a/Makefile b/Makefile
--- a/Makefile	2020-07-05 09:53:24.000000000 -0400
+++ b/Makefile	2026-05-08 16:44:01.808683971 -0400
@@ -56,7 +56,7 @@ distclean: clean
 install-common: $(PROG) $(DISPATCH)
 	install -D --mode=755 usb_modeswitch $(SBINDIR)/usb_modeswitch
 	install -D --mode=755 usb_modeswitch.sh $(UDEVDIR)/usb_modeswitch
-	install -D --mode=644 usb_modeswitch.conf $(ETCDIR)/usb_modeswitch.conf
+	install -D --mode=644 usb_modeswitch.conf $(PREFIX)/share/defaults/usb_modeswitch/usb_modeswitch.conf
 	install -D --mode=644 usb_modeswitch.1 $(MANDIR)/usb_modeswitch.1
 	install -D --mode=644 usb_modeswitch_dispatcher.1 $(MANDIR)/usb_modeswitch_dispatcher.1
 	install -D --mode=755 usb_modeswitch_dispatcher $(SBINDIR)/usb_modeswitch_dispatcher
diff '--color=auto' -pruN a/usb_modeswitch_dispatcher.tcl b/usb_modeswitch_dispatcher.tcl
--- a/usb_modeswitch_dispatcher.tcl	2025-02-18 14:42:22.000000000 -0500
+++ b/usb_modeswitch_dispatcher.tcl	2026-05-08 16:44:09.936889369 -0400
@@ -539,7 +539,7 @@ set configFile ""
 if [string length $path] {
 	set places [list $path]
 } else {
-	set places [list /etc/usb_modeswitch.conf /etc/sysconfig/usb_modeswitch /etc/default/usb_modeswitch]
+	set places [list /etc/usb_modeswitch.conf /etc/sysconfig/usb_modeswitch /etc/default/usb_modeswitch /usr/share/defaults/usb_modeswitch/usb_modeswitch.conf]
 }
 foreach cfg $places {
 	if [file exists $cfg] {
