Subject: [PATCH] patch: create to distribute on solus
---
Index: daemon/service/platform/platform_linux_release.go
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/daemon/service/platform/platform_linux_release.go b/daemon/service/platform/platform_linux_release.go
--- a/daemon/service/platform/platform_linux_release.go	(revision 9b8ee5a7c1b2fcf920772342ce0a97917f2ae669)
+++ b/daemon/service/platform/platform_linux_release.go	(date 1773356325477)
@@ -30,7 +30,7 @@
 )
 
 func doOsInitForBuild() (warnings []string, errors []error, logInfo []string) {
-	installDir := "/opt/ivpn"
+	installDir := "/usr/lib64/ivpn"
 
 	// check if we are running in snap environment
 	if envs := GetSnapEnvs(); envs != nil {
@@ -55,10 +55,10 @@
 	v2rayBinaryPath = path.Join(installDir, "v2ray/v2ray")
 	v2rayConfigTmpFile = path.Join(tmpDir, "v2ray.json")
 
-	wgBinaryPath = path.Join(installDir, "wireguard-tools/wg-quick")
-	wgToolBinaryPath = path.Join(installDir, "wireguard-tools/wg")
+	wgBinaryPath = path.Join("/usr/bin", "wg-quick")
+	wgToolBinaryPath = path.Join("/usr/bin","wg")
 
-	dnscryptproxyBinPath = path.Join(installDir, "dnscrypt-proxy/dnscrypt-proxy")
+	dnscryptproxyBinPath = path.Join("/usr/bin", "dnscrypt-proxy")
 	dnscryptproxyConfigTemplate = path.Join(installDir, "etc/dnscrypt-proxy-template.toml")
 	dnscryptproxyConfigDir = tmpDir
 
