From 8e904aa1c92bd0993123dd46d5a10a58f0516721 Mon Sep 17 00:00:00 2001
From: Lars Wendler <polynomial-c@gentoo.org>
Date: Mon, 16 Dec 2019 18:23:16 +0100
Subject: [PATCH] libetpan.pc.in: Don't put user LDFLAGS into Libs (#337)

Signed-off-by: Lars Wendler <polynomial-c@gentoo.org>
---
 libetpan.pc.in | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/libetpan.pc.in b/libetpan.pc.in
index cbc1d2c4..e0dc7742 100644
--- a/libetpan.pc.in
+++ b/libetpan.pc.in
@@ -6,5 +6,5 @@ libdir=@libdir@
 Name: libetpan
 Description: Libetpan C library.
 Version: @VERSION@
-Libs: -L${libdir} -letpan@LIBSUFFIX@ @LDFLAGS@ @SSLLIBS@ @GNUTLSLIB@ @LIBICONV@ @DBLIB@ @LIBS@ @SASLLIBS@
+Libs: -L${libdir} -letpan@LIBSUFFIX@ @SSLLIBS@ @GNUTLSLIB@ @LIBICONV@ @DBLIB@ @LIBS@ @SASLLIBS@
 Cflags: -I${includedir}
