diff -rU4 sdcv-0.5.5--orig/src/stardict_lib.cpp sdcv-0.5.5/src/stardict_lib.cpp
--- sdcv-0.5.5--orig/src/stardict_lib.cpp	2023-04-18 20:47:55.000000000 +0200
+++ sdcv-0.5.5/src/stardict_lib.cpp	2026-02-22 17:47:50.543847450 +0100
@@ -1046,9 +1046,10 @@
             g_free(casestr);
         }
         // Upper the first character and lower others.
         if (!bFound) {
-            gchar *nextchar = g_utf8_next_char(sWord);
+            gchar *tmp = (gchar*)sWord; // explicitly discard const
+            gchar *nextchar = g_utf8_next_char(tmp);
             gchar *firstchar = g_utf8_strup(sWord, nextchar - sWord);
             nextchar = g_utf8_strdown(nextchar, -1);
             casestr = g_strdup_printf("%s%s", firstchar, nextchar);
             g_free(firstchar);
