diff --git a/srcluaconsole/config.mak b/srcluaconsole/config.mak
index 12baea3..2a3f3d8 100644
--- a/srcluaconsole/config.mak
+++ b/srcluaconsole/config.mak
@@ -197,8 +197,8 @@ else
       # Except in Cygwin
     else
       # In UNIX Lua is always statically linked, late binding is used.
-      NO_LUALINK = Yes
-      SLIB += $(LUA_LIB)/liblua$(LUA_SFX).a
+      #NO_LUALINK = Yes
+      #SLIB += $(LUA_LIB)/liblua$(LUA_SFX).a
     endif
   endif
 endif
diff --git a/srcluascripter/config.mak b/srcluascripter/config.mak
index 76300c6..bb47c4f 100644
--- a/srcluascripter/config.mak
+++ b/srcluascripter/config.mak
@@ -184,8 +184,8 @@ else
       # Except in Cygwin
     else
       # In UNIX Lua is always statically linked, late binding is used.
-      NO_LUALINK = Yes
-      SLIB += $(LUA_LIB)/liblua$(LUA_SFX).a
+      #NO_LUALINK = Yes
+      #SLIB += $(LUA_LIB)/liblua$(LUA_SFX).a
     endif
   endif
 endif
diff --git a/srcview/config.mak b/srcview/config.mak
index db72772..41c4019 100644
--- a/srcview/config.mak
+++ b/srcview/config.mak
@@ -14,7 +14,7 @@ USE_IUP3 = Yes
 ifeq "$(TEC_SYSNAME)" "Haiku"
   USE_HAIKU = Yes
 else
-  USE_STATIC = Yes
+#  USE_STATIC = Yes
 ifdef GTK_DEFAULT
   ifdef USE_MOTIF
     # Build Motif version in Linux and BSD
diff --git a/srcvled/config.mak b/srcvled/config.mak
index d072320..6fbaef9 100644
--- a/srcvled/config.mak
+++ b/srcvled/config.mak
@@ -14,7 +14,7 @@ USE_IUP3 = Yes
 ifeq "$(TEC_SYSNAME)" "Haiku"
   USE_HAIKU = Yes
 else
-  USE_STATIC = Yes
+#  USE_STATIC = Yes
 ifdef GTK_DEFAULT
   ifdef USE_MOTIF
     # Build Motif version in Linux and BSD
@@ -109,6 +109,7 @@ ifndef USE_NO_PLOT
       SLIB += $(CD_LIB)/libcdgl.a
       SLIB += $(CD_LIB)/libcdcontextplus.a
     else
+      LIBS += iup_plot
       LIBS += iupglcontrols
       LIBS += cdgl
     endif             
@@ -129,4 +130,6 @@ ifneq ($(findstring MacOS, $(TEC_UNAME)), )
   LIBS += fontconfig
 endif
 
+LIBS += gmodule-2.0
+
 INCLUDES = ../src
diff --git a/srcweb/config.mak b/srcweb/config.mak
index c528f2c..4ff9689 100644
--- a/srcweb/config.mak
+++ b/srcweb/config.mak
@@ -35,15 +35,20 @@ else
     LINK_WEBKIT = Yes
     
     ifdef USE_GTK3
-      ifneq ($(findstring Linux5, $(TEC_UNAME)), )
+      ifneq ($(findstring Linux6, $(TEC_UNAME)), )
         DEFINES += USE_WEBKIT2
         STDINCS += $(GTK)/include/webkitgtk-4.0
       else
-        ifneq ($(findstring Linux4, $(TEC_UNAME)), )
+        ifneq ($(findstring Linux5, $(TEC_UNAME)), )
           DEFINES += USE_WEBKIT2
           STDINCS += $(GTK)/include/webkitgtk-4.0
         else
-          STDINCS += $(GTK)/include/webkitgtk-3.0
+          ifneq ($(findstring Linux4, $(TEC_UNAME)), )
+            DEFINES += USE_WEBKIT2
+            STDINCS += $(GTK)/include/webkitgtk-4.0
+          else
+            STDINCS += $(GTK)/include/webkitgtk-3.0
+          endif
         endif
       endif
     else 
diff --git a/tecmake.mak b/tecmake.mak
index 48991a4..fdda137 100644
--- a/tecmake.mak
+++ b/tecmake.mak
@@ -297,6 +297,9 @@ endif
 
 ifdef GTK_DEFAULT
   ifndef USE_GTK2
+    ifneq ($(findstring Linux6, $(TEC_UNAME)), )
+      USE_GTK3 = Yes
+    endif
     ifneq ($(findstring Linux5, $(TEC_UNAME)), )
       USE_GTK3 = Yes
     endif
@@ -1062,7 +1065,8 @@ ifdef USE_LUA
       LIBS += lualib$(LUA_SFX)
     endif
     ifndef NO_LUALINK
-        LIBS += lua$(LUA_SFX)
+#        LIBS += lua$(LUA_SFX)
+        LIBS += lua
         LDIR += $(LUA_LIB)
     else
       ifneq ($(findstring cygw, $(TEC_UNAME)), )
@@ -1210,6 +1214,9 @@ ifdef USE_CD
   ifneq ($(findstring Linux5, $(TEC_UNAME)), )
     LIBS += fontconfig
   endif
+  ifneq ($(findstring Linux6, $(TEC_UNAME)), )
+    LIBS += fontconfig
+  endif
   ifneq ($(findstring cygw, $(TEC_UNAME)), )
     LIBS += fontconfig
   endif
@@ -1245,20 +1252,24 @@ ifdef USE_IM
 endif
 
 ifdef LINK_WEBKIT
-  ifneq ($(findstring Linux5, $(TEC_UNAME)), )
+  ifneq ($(findstring Linux6, $(TEC_UNAME)), )
     LIBS += webkit2gtk-4.0 gio-2.0
-  else 
-    ifneq ($(findstring Linux4, $(TEC_UNAME)), )
+  else
+    ifneq ($(findstring Linux5, $(TEC_UNAME)), )
       LIBS += webkit2gtk-4.0 gio-2.0
     else 
-      ifneq ($(findstring Linux3, $(TEC_UNAME)), )
-        ifdef USE_GTK3
-          LIBS += webkitgtk-3.0
+      ifneq ($(findstring Linux4, $(TEC_UNAME)), )
+        LIBS += webkit2gtk-4.0 gio-2.0
+      else 
+        ifneq ($(findstring Linux3, $(TEC_UNAME)), )
+          ifdef USE_GTK3
+            LIBS += webkitgtk-3.0
+          else
+            LIBS += webkitgtk-1.0
+          endif
         else
-          LIBS += webkitgtk-1.0
+          LIBS += webkit-1.0
         endif
-      else
-        LIBS += webkit-1.0
       endif
     endif 
   endif
@@ -1479,6 +1490,9 @@ ifdef USE_GTK
     ifneq ($(findstring Linux5, $(TEC_UNAME)), )
       STDINCS += /usr/include/harfbuzz
     endif
+    ifneq ($(findstring Linux6, $(TEC_UNAME)), )
+      STDINCS += /usr/include/harfbuzz
+    endif
   endif
 endif
 
