diff --git a/srclua5/config.mak b/srclua5/config.mak
index 77a8920..ebc4669 100755
--- a/srclua5/config.mak
+++ b/srclua5/config.mak
@@ -49,7 +49,7 @@ endif
 endif
 endif
 
-LIBNAME := $(LIBNAME)$(LUASFX)
+LIBNAME := $(LIBNAME)
 ifdef NO_LUAOBJECT
   DEFINES += IUPLUA_USELH
   USE_LH_SUBDIR = Yes
@@ -57,7 +57,7 @@ ifdef NO_LUAOBJECT
 else
   DEFINES += IUPLUA_USELOH
   USE_LOH_SUBDIR = Yes
-  LOHDIR = loh$(LUASFX)
+  LOHDIR = loh
 endif
 
 CTRLUA = button.lua canvas.lua dialog.lua colordlg.lua clipboard.lua \
diff --git a/srclua5/iup_mglplot.mak b/srclua5/iup_mglplot.mak
index 3c3407c..c20f783 100755
--- a/srclua5/iup_mglplot.mak
+++ b/srclua5/iup_mglplot.mak
@@ -53,7 +53,7 @@ endif
 endif
 endif
 
-LIBNAME := $(LIBNAME)$(LUASFX)
+LIBNAME := $(LIBNAME)
 ifdef NO_LUAOBJECT
   DEFINES += IUPLUA_USELH
   USE_LH_SUBDIR = Yes
@@ -61,7 +61,7 @@ ifdef NO_LUAOBJECT
 else
   DEFINES += IUPLUA_USELOH
   USE_LOH_SUBDIR = Yes
-  LOHDIR = loh$(LUASFX)
+  LOHDIR = loh
 endif
 
 SRCLUA = mglplot.lua mgllabel.lua
diff --git a/srclua5/iup_plot.mak b/srclua5/iup_plot.mak
index b2af19d..21d1311 100755
--- a/srclua5/iup_plot.mak
+++ b/srclua5/iup_plot.mak
@@ -54,7 +54,7 @@ endif
 endif
 endif
 
-LIBNAME := $(LIBNAME)$(LUASFX)
+LIBNAME := $(LIBNAME)
 ifdef NO_LUAOBJECT
   DEFINES += IUPLUA_USELH
   USE_LH_SUBDIR = Yes
@@ -62,7 +62,7 @@ ifdef NO_LUAOBJECT
 else
   DEFINES += IUPLUA_USELOH
   USE_LOH_SUBDIR = Yes
-  LOHDIR = loh$(LUASFX)
+  LOHDIR = loh
 endif
 
 SRCLUA = plot.lua
diff --git a/srclua5/iup_scintilla.mak b/srclua5/iup_scintilla.mak
index 502bf59..f8ac363 100755
--- a/srclua5/iup_scintilla.mak
+++ b/srclua5/iup_scintilla.mak
@@ -53,7 +53,7 @@ endif
 endif
 endif
 
-LIBNAME := $(LIBNAME)$(LUASFX)
+LIBNAME := $(LIBNAME)
 ifdef NO_LUAOBJECT
   DEFINES += IUPLUA_USELH
   USE_LH_SUBDIR = Yes
@@ -61,7 +61,7 @@ ifdef NO_LUAOBJECT
 else
   DEFINES += IUPLUA_USELOH
   USE_LOH_SUBDIR = Yes
-  LOHDIR = loh$(LUASFX)
+  LOHDIR = loh
 endif
 
 SRCLUA = scintilla.lua scintilladlg.lua
diff --git a/srclua5/iupcd.mak b/srclua5/iupcd.mak
index 429cd3e..38f0be0 100755
--- a/srclua5/iupcd.mak
+++ b/srclua5/iupcd.mak
@@ -55,8 +55,8 @@ endif
 endif
 endif
 
-LIBNAME := $(LIBNAME)$(LUASFX)
-LIBS = iuplua$(LUASFX)
+LIBNAME := $(LIBNAME)
+LIBS = iuplua
 
 ifneq ($(findstring MacOS, $(TEC_UNAME)), )
   USE_IUPLUA:=
diff --git a/srclua5/iupcontrols.mak b/srclua5/iupcontrols.mak
index cb00b95..3174b4b 100755
--- a/srclua5/iupcontrols.mak
+++ b/srclua5/iupcontrols.mak
@@ -54,7 +54,7 @@ endif
 endif
 endif
 
-LIBNAME := $(LIBNAME)$(LUASFX)
+LIBNAME := $(LIBNAME)
 ifdef NO_LUAOBJECT
   DEFINES += IUPLUA_USELH
   USE_LH_SUBDIR = Yes
@@ -62,7 +62,7 @@ ifdef NO_LUAOBJECT
 else
   DEFINES += IUPLUA_USELOH
   USE_LOH_SUBDIR = Yes
-  LOHDIR = loh$(LUASFX)
+  LOHDIR = loh
 endif
 
 SRCLUA = matrix.lua matrixlist.lua matrixex.lua cells.lua
diff --git a/srclua5/iupfiledlg.mak b/srclua5/iupfiledlg.mak
index 6c6cf98..dc78e37 100755
--- a/srclua5/iupfiledlg.mak
+++ b/srclua5/iupfiledlg.mak
@@ -51,7 +51,7 @@ endif
 endif
 endif
 
-LIBNAME := $(LIBNAME)$(LUASFX)
+LIBNAME := $(LIBNAME)
 
 SRC	= ctrl/iuplua_filedlg.c
 
diff --git a/srclua5/iupgl.mak b/srclua5/iupgl.mak
index f633c1c..aace8f5 100755
--- a/srclua5/iupgl.mak
+++ b/srclua5/iupgl.mak
@@ -52,7 +52,7 @@ endif
 endif
 endif
 
-LIBNAME := $(LIBNAME)$(LUASFX)
+LIBNAME := $(LIBNAME)
 ifdef NO_LUAOBJECT
   DEFINES += IUPLUA_USELH
   USE_LH_SUBDIR = Yes
@@ -60,11 +60,11 @@ ifdef NO_LUAOBJECT
 else
   DEFINES += IUPLUA_USELOH
   USE_LOH_SUBDIR = Yes
-  LOHDIR = loh$(LUASFX)
+  LOHDIR = loh
 endif
 
 SRCLUA = glcanvas.lua glbackgroundbox.lua
-LIBS = iupgl iuplua$(LUASFX)
+LIBS = iupgl iuplua
 
 GC = $(addsuffix .c, $(basename $(SRCLUA)))
 GC := $(addprefix ctrl/il_, $(GC))
diff --git a/srclua5/iupglcontrols.mak b/srclua5/iupglcontrols.mak
index e6469f9..c28d2a4 100755
--- a/srclua5/iupglcontrols.mak
+++ b/srclua5/iupglcontrols.mak
@@ -16,7 +16,7 @@ USE_IUP3 = Yes
 USE_IUPLUA = Yes
 
 INCLUDES = ../src gl ../srclua5
-LIBS = iupglcontrols iuplua$(LUASFX)
+LIBS = iupglcontrols iuplua
 DEF_FILE = gl/iupluaglcontrols.def
 
 ifdef USE_LUA_VERSION
@@ -53,7 +53,7 @@ endif
 endif
 endif
 
-LIBNAME := $(LIBNAME)$(LUASFX)
+LIBNAME := $(LIBNAME)
 ifdef NO_LUAOBJECT
   DEFINES += IUPLUA_USELH
   USE_LH_SUBDIR = Yes
@@ -61,7 +61,7 @@ ifdef NO_LUAOBJECT
 else
   DEFINES += IUPLUA_USELOH
   USE_LOH_SUBDIR = Yes
-  LOHDIR = loh$(LUASFX)
+  LOHDIR = loh
 endif
 
 SRCLUA = glcanvasbox.lua glsubcanvas.lua gllabel.lua glbutton.lua \
diff --git a/srclua5/iupim.mak b/srclua5/iupim.mak
index d2eb2d4..dd4f94f 100755
--- a/srclua5/iupim.mak
+++ b/srclua5/iupim.mak
@@ -53,7 +53,7 @@ endif
 endif
 endif
 
-LIBNAME := $(LIBNAME)$(LUASFX)
+LIBNAME := $(LIBNAME)
 
 ifneq ($(findstring MacOS, $(TEC_UNAME)), )
   USE_IUPLUA:=
diff --git a/srclua5/iupimglib.mak b/srclua5/iupimglib.mak
index fb56ad6..e039330 100755
--- a/srclua5/iupimglib.mak
+++ b/srclua5/iupimglib.mak
@@ -51,7 +51,7 @@ endif
 endif
 endif
 
-LIBNAME := $(LIBNAME)$(LUASFX)
+LIBNAME := $(LIBNAME)
 
 ifneq ($(findstring MacOS, $(TEC_UNAME)), )
   USE_IUPLUA:=
diff --git a/srclua5/iupluascripterdlg.mak b/srclua5/iupluascripterdlg.mak
index 56ae104..18a3676 100755
--- a/srclua5/iupluascripterdlg.mak
+++ b/srclua5/iupluascripterdlg.mak
@@ -53,7 +53,7 @@ endif
 endif
 endif
 
-LIBNAME := $(LIBNAME)$(LUASFX)
+LIBNAME := $(LIBNAME)
 ifdef NO_LUAOBJECT
   DEFINES += IUPLUA_USELH
   USE_LH_SUBDIR = Yes
@@ -61,7 +61,7 @@ ifdef NO_LUAOBJECT
 else
   DEFINES += IUPLUA_USELOH
   USE_LOH_SUBDIR = Yes
-  LOHDIR = loh$(LUASFX)
+  LOHDIR = loh
 endif
 
 SRCLUA_CTRL = luascripterdlg.lua
diff --git a/srclua5/iupole.mak b/srclua5/iupole.mak
index 0b625b6..acc2c9e 100755
--- a/srclua5/iupole.mak
+++ b/srclua5/iupole.mak
@@ -52,7 +52,7 @@ endif
 endif
 endif
 
-LIBNAME := $(LIBNAME)$(LUASFX)
+LIBNAME := $(LIBNAME)
 ifdef NO_LUAOBJECT
   DEFINES += IUPLUA_USELH
   USE_LH_SUBDIR = Yes
@@ -60,7 +60,7 @@ ifdef NO_LUAOBJECT
 else
   DEFINES += IUPLUA_USELOH
   USE_LOH_SUBDIR = Yes
-  LOHDIR = loh$(LUASFX)
+  LOHDIR = loh
 endif
 
 SRCLUA = olecontrol.lua
diff --git a/srclua5/iuptuio.mak b/srclua5/iuptuio.mak
index 853430c..a46ce42 100755
--- a/srclua5/iuptuio.mak
+++ b/srclua5/iuptuio.mak
@@ -53,7 +53,7 @@ endif
 endif
 endif
 
-LIBNAME := $(LIBNAME)$(LUASFX)
+LIBNAME := $(LIBNAME)
 ifdef NO_LUAOBJECT
   DEFINES += IUPLUA_USELH
   USE_LH_SUBDIR = Yes
@@ -61,7 +61,7 @@ ifdef NO_LUAOBJECT
 else
   DEFINES += IUPLUA_USELOH
   USE_LOH_SUBDIR = Yes
-  LOHDIR = loh$(LUASFX)
+  LOHDIR = loh
 endif
 
 SRCLUA = tuioclient.lua
diff --git a/srclua5/iupweb.mak b/srclua5/iupweb.mak
index 9d7dff2..40be99c 100755
--- a/srclua5/iupweb.mak
+++ b/srclua5/iupweb.mak
@@ -53,7 +53,7 @@ endif
 endif
 endif
 
-LIBNAME := $(LIBNAME)$(LUASFX)
+LIBNAME := $(LIBNAME)
 ifdef NO_LUAOBJECT
   DEFINES += IUPLUA_USELH
   USE_LH_SUBDIR = Yes
@@ -61,7 +61,7 @@ ifdef NO_LUAOBJECT
 else
   DEFINES += IUPLUA_USELOH
   USE_LOH_SUBDIR = Yes
-  LOHDIR = loh$(LUASFX)
+  LOHDIR = loh
 endif
 
 SRCLUA = webbrowser.lua
diff --git a/srcluaconsole/config.mak b/srcluaconsole/config.mak
index 12baea3..213a1e1 100755
--- a/srcluaconsole/config.mak
+++ b/srcluaconsole/config.mak
@@ -57,7 +57,7 @@ endif
 endif
 endif
 
-APPNAME := $(APPNAME)$(LUASFX)
+APPNAME := $(APPNAME)
 SRC = iup_lua.c
 
 ifdef NO_LUAOBJECT
@@ -67,7 +67,7 @@ ifdef NO_LUAOBJECT
 else
   DEFINES += IUPLUA_USELOH
   USE_LOH_SUBDIR = Yes
-  LOHDIR = loh$(LUASFX)
+  LOHDIR = loh
 endif
 
 ifdef GTK_DEFAULT
@@ -128,16 +128,16 @@ ifdef ALL_STATIC
     USE_CDLUA = Yes
     USE_IUPCONTROLS = Yes
     ifneq ($(findstring Win, $(TEC_SYSNAME)), )
-      LIBS += iuplua_plot$(LUASFX) iup_plot cdgl cdcontextplus
+      LIBS += iuplua_plot iup_plot cdgl cdcontextplus
     else
-      SLIB += $(IUP_LIB)/Lua$(LUASFX)/libiuplua_plot$(LUASFX).a $(IUP_LIB)/libiup_plot.a $(CD_LIB)/libcdgl.a $(CD_LIB)/libcdcontextplus.a
+      SLIB += $(IUP_LIB)/Lua/libiuplua_plot.a $(IUP_LIB)/libiup_plot.a $(CD_LIB)/libcdgl.a $(CD_LIB)/libcdcontextplus.a
     endif
       
     ifndef IUPLUA_NO_IM
       ifneq ($(findstring Win, $(TEC_SYSNAME)), )
-        LIBS += cdluaim$(LUASFX) cdim$(LUASFX)
+        LIBS += cdluaim cdim
       else
-        SLIB += $(CD_LIB)/Lua$(LUASFX)/libcdluaim$(LUASFX).a $(CD_LIB)/libcdim.a
+        SLIB += $(CD_LIB)/Lua/libcdluaim.a $(CD_LIB)/libcdim.a
       endif
     endif
     ifneq ($(findstring Win, $(TEC_SYSNAME)), )
@@ -157,9 +157,9 @@ ifdef ALL_STATIC
   ifndef IUPLUA_NO_IM
     USE_IMLUA = Yes
     ifneq ($(findstring Win, $(TEC_SYSNAME)), )
-      LIBS += imlua_process$(LUASFX) iupluaim$(LUASFX) im_process iupim
+      LIBS += imlua_process iupluaim im_process iupim
     else
-      SLIB += $(IM_LIB)/Lua$(LUASFX)/libimlua_process$(LUASFX).a $(IUP_LIB)/Lua$(LUASFX)/libiupluaim$(LUASFX).a $(IM_LIB)/libim_process.a $(IUP_LIB)/libiupim.a
+      SLIB += $(IM_LIB)/Lua/libimlua_process.a $(IUP_LIB)/Lua/libiupluaim.a $(IM_LIB)/libim_process.a $(IUP_LIB)/libiupim.a
     endif
   else
     DEFINES += IUPLUA_NO_IM
@@ -169,9 +169,9 @@ ifdef ALL_STATIC
   ifdef IUPLUA_IMGLIB
     DEFINES += IUPLUA_IMGLIB
     ifneq ($(findstring Win, $(TEC_SYSNAME)), )
-      LIBS += iupluaimglib$(LUASFX) iupimglib
+      LIBS += iupluaimglib iupimglib
     else
-      SLIB += $(IUP_LIB)/Lua$(LUASFX)/libiupluaimglib$(LUASFX).a $(IUP_LIB)/libiupimglib.a
+      SLIB += $(IUP_LIB)/Lua/libiupluaimglib.a $(IUP_LIB)/libiupimglib.a
     endif
   endif
   
@@ -179,10 +179,10 @@ ifdef ALL_STATIC
   ifdef IUPLUA_TUIO
     DEFINES += IUPLUA_TUIO
     ifneq ($(findstring Win, $(TEC_SYSNAME)), )
-      LIBS += iupluatuio$(LUASFX) iuptuio
+      LIBS += iupluatuio iuptuio
       LIBS += ws2_32 winmm
     else
-      SLIB += $(IUP_LIB)/Lua$(LUASFX)/libiupluatuio$(LUASFX).a $(IUP_LIB)/libiuptuio.a
+      SLIB += $(IUP_LIB)/Lua/libiupluatuio.a $(IUP_LIB)/libiuptuio.a
     endif
   endif
 else
@@ -192,7 +192,7 @@ else
     USE_DLL = Yes
     GEN_MANIFEST = No
   else
-    LDIR += $(IUP_LIB)/Lua$(LUASFX)
+    LDIR += $(IUP_LIB)/Lua
     ifneq ($(findstring cygw, $(TEC_UNAME)), )
       # Except in Cygwin
     else
diff --git a/srcluascripter/config.mak b/srcluascripter/config.mak
index 76300c6..a2dcfb9 100755
--- a/srcluascripter/config.mak
+++ b/srcluascripter/config.mak
@@ -54,7 +54,7 @@ endif
 endif
 endif
 
-APPNAME := $(APPNAME)$(LUASFX)
+APPNAME := $(APPNAME)
 SRC = iupluascripter.c
 
 INCLUDES = ../include
@@ -84,9 +84,9 @@ ifdef ALL_STATIC
   endif  
   
   ifneq ($(findstring Win, $(TEC_SYSNAME)), )
-    LIBS += iupluascripterdlg$(LUASFX) iupimglib iup_scintilla imm32
+    LIBS += iupluascripterdlg iupimglib iup_scintilla imm32
   else
-    SLIB += $(IUP_LIB)/Lua$(LUASFX)/libiupluascripterdlg$(LUASFX).a \
+    SLIB += $(IUP_LIB)/Lua/libiupluascripterdlg.a \
             $(IUP_LIB)/libiup_scintilla.a \
             $(IUP_LIB)/libiupimglib.a
   endif
@@ -94,9 +94,9 @@ ifdef ALL_STATIC
   DEFINES += USE_STATIC
 
   ifneq ($(findstring Win, $(TEC_SYSNAME)), )
-    LIBS += iuplua_scintilla$(LUASFX)
+    LIBS += iuplua_scintilla
   else
-    SLIB += $(IUP_LIB)/Lua$(LUASFX)/libiuplua_scintilla$(LUASFX).a
+    SLIB += $(IUP_LIB)/Lua/libiuplua_scintilla.a
   endif
   
   ifeq "$(TEC_UNAME)" "SunOS510x86"
@@ -116,16 +116,16 @@ ifdef ALL_STATIC
     USE_CDLUA = Yes
     USE_IUPCONTROLS = Yes
     ifneq ($(findstring Win, $(TEC_SYSNAME)), )
-      LIBS += iuplua_plot$(LUASFX) iup_plot cdgl cdcontextplus
+      LIBS += iuplua_plot iup_plot cdgl cdcontextplus
     else
-      SLIB += $(IUP_LIB)/Lua$(LUASFX)/libiuplua_plot$(LUASFX).a $(IUP_LIB)/libiup_plot.a $(CD_LIB)/libcdgl.a $(CD_LIB)/libcdcontextplus.a
+      SLIB += $(IUP_LIB)/Lua/libiuplua_plot.a $(IUP_LIB)/libiup_plot.a $(CD_LIB)/libcdgl.a $(CD_LIB)/libcdcontextplus.a
     endif
       
     ifndef IUPLUA_NO_IM
       ifneq ($(findstring Win, $(TEC_SYSNAME)), )
-        LIBS += cdluaim$(LUASFX) cdim$(LUASFX)
+        LIBS += cdluaim cdim
       else
-        SLIB += $(CD_LIB)/Lua$(LUASFX)/libcdluaim$(LUASFX).a $(CD_LIB)/libcdim.a
+        SLIB += $(CD_LIB)/Lua/libcdluaim.a $(CD_LIB)/libcdim.a
       endif
     endif
   else
@@ -136,9 +136,9 @@ ifdef ALL_STATIC
   ifndef IUPLUA_NO_IM
     USE_IMLUA = Yes
     ifneq ($(findstring Win, $(TEC_SYSNAME)), )
-      LIBS += imlua_process$(LUASFX) iupluaim$(LUASFX) im_process iupim
+      LIBS += imlua_process iupluaim im_process iupim
     else
-      SLIB += $(IM_LIB)/Lua$(LUASFX)/libimlua_process$(LUASFX).a $(IUP_LIB)/Lua$(LUASFX)/libiupluaim$(LUASFX).a $(IM_LIB)/libim_process.a $(IUP_LIB)/libiupim.a
+      SLIB += $(IM_LIB)/Lua/libimlua_process.a $(IUP_LIB)/Lua/libiupluaim.a $(IM_LIB)/libim_process.a $(IUP_LIB)/libiupim.a
     endif
   else
     DEFINES += IUPLUA_NO_IM
@@ -148,9 +148,9 @@ ifdef ALL_STATIC
   ifdef IUPLUA_IMGLIB
     DEFINES += IUPLUA_IMGLIB
     ifneq ($(findstring Win, $(TEC_SYSNAME)), )
-      LIBS += iupluaimglib$(LUASFX)
+      LIBS += iupluaimglib
     else
-      SLIB += $(IUP_LIB)/Lua$(LUASFX)/libiupluaimglib$(LUASFX).a
+      SLIB += $(IUP_LIB)/Lua/libiupluaimglib.a
     endif
   endif
   
@@ -164,10 +164,10 @@ ifdef ALL_STATIC
   ifdef IUPLUA_TUIO
     DEFINES += IUPLUA_TUIO
     ifneq ($(findstring Win, $(TEC_SYSNAME)), )
-      LIBS += iupluatuio$(LUASFX) iuptuio
+      LIBS += iupluatuio iuptuio
       LIBS += ws2_32 winmm
     else
-      SLIB += $(IUP_LIB)/Lua$(LUASFX)/libiupluatuio$(LUASFX).a $(IUP_LIB)/libiuptuio.a
+      SLIB += $(IUP_LIB)/Lua/libiupluatuio.a $(IUP_LIB)/libiuptuio.a
     endif
   endif
 else
@@ -176,10 +176,10 @@ else
     # Must call "tecmake dll10" so USE_* will use the correct TEC_UNAME
     USE_DLL = Yes
     GEN_MANIFEST = No
-    LIBS += iupluascripterdlg$(LUASFX) iupimglib iup_scintilla imm32
+    LIBS += iupluascripterdlg iupimglib iup_scintilla imm32
   else
-    LDIR += $(IUP_LIB)/Lua$(LUASFX)
-    LIBS += iupluascripterdlg$(LUASFX) iupimglib iup_scintilla
+    LDIR += $(IUP_LIB)/Lua
+    LIBS += iupluascripterdlg iupimglib iup_scintilla
     ifneq ($(findstring cygw, $(TEC_UNAME)), )
       # Except in Cygwin
     else
diff --git a/tecmake.mak b/tecmake.mak
index 48991a4..03df592 100755
--- a/tecmake.mak
+++ b/tecmake.mak
@@ -478,12 +478,12 @@ endif
 ifdef USE_OLDLIBLUA
   TEC_UNAME_LIBLUA_DIR ?= $(TEC_UNAME_LIB_DIR)
 else
-  TEC_UNAME_LIBLUA_DIR ?= $(TEC_UNAME_LIB_DIR)/Lua$(LIBLUA_SFX)
+  TEC_UNAME_LIBLUA_DIR ?= $(TEC_UNAME_LIB_DIR)/Lua
 endif
 
 # Subfolder for Lua Modules
 ifdef LUAMOD_DIR
-  TEC_UNAME_DIR := $(TEC_UNAME_DIR)/Lua$(LIBLUA_SFX)
+  TEC_UNAME_DIR := $(TEC_UNAME_DIR)/Lua
 endif
 
 OBJDIR := $(OBJROOT)/$(TEC_UNAME_DIR)
@@ -941,9 +941,9 @@ ifdef USE_IUPCONTROLS
   
   ifdef USE_IUPLUA
     ifdef USE_STATIC
-      SLIB += $(IUPLUA_LIB)/libiupluacontrols$(LIBLUA_SFX).a
+      SLIB += $(IUPLUA_LIB)/libiupluacontrols.a
     else
-      LIBS += iupluacontrols$(LIBLUA_SFX)
+      LIBS += iupluacontrols
     endif
     override USE_CDLUA = Yes
   endif
@@ -964,9 +964,9 @@ ifdef USE_IUPGLCONTROLS
   
   ifdef USE_IUPLUA
     ifdef USE_STATIC
-      SLIB += $(IUPLUA_LIB)/libiupluaglcontrols$(LIBLUA_SFX).a
+      SLIB += $(IUPLUA_LIB)/libiupluaglcontrols.a
     else
-      LIBS += iupluaglcontrols$(LIBLUA_SFX)
+      LIBS += iupluaglcontrols
     endif
   endif
   
@@ -983,9 +983,9 @@ ifdef USE_IUPWEB
   
   ifdef USE_IUPLUA
     ifdef USE_STATIC
-      SLIB += $(IUPLUA_LIB)/libiupluaweb$(LIBLUA_SFX).a
+      SLIB += $(IUPLUA_LIB)/libiupluaweb.a
     else
-      LIBS += iupluaweb$(LIBLUA_SFX)
+      LIBS += iupluaweb
     endif
   endif
   
@@ -1000,9 +1000,9 @@ ifdef USE_IMLUA
   override USE_IM = Yes
   IMLUA_LIB ?= $(IM)/lib/$(TEC_UNAME_LIBLUA_DIR)
   ifdef USE_STATIC
-    SLIB += $(IMLUA_LIB)/libimlua$(LIBLUA_SFX).a
+    SLIB += $(IMLUA_LIB)/libimlua.a
   else
-    LIBS += imlua$(LIBLUA_SFX)
+    LIBS += imlua
     LDIR += $(IMLUA_LIB)
   endif
 endif
@@ -1011,9 +1011,9 @@ ifdef USE_CDLUA
   override USE_CD = Yes
   CDLUA_LIB ?= $(CD)/lib/$(TEC_UNAME_LIBLUA_DIR)
   ifdef USE_STATIC
-    SLIB += $(CDLUA_LIB)/libcdlua$(LIBLUA_SFX).a
+    SLIB += $(CDLUA_LIB)/libcdlua.a
   else
-    LIBS += cdlua$(LIBLUA_SFX)
+    LIBS += cdlua
     LDIR += $(CDLUA_LIB)
   endif
 endif
@@ -1025,27 +1025,27 @@ ifdef USE_IUPLUA
   ifdef USE_STATIC
     ifdef USE_CD
       ifeq ($(findstring iupluacd, $(LIBNAME)), )
-        SLIB += $(IUPLUA_LIB)/libiupluacd$(LIBLUA_SFX).a
+        SLIB += $(IUPLUA_LIB)/libiupluacd.a
       endif
     endif
     ifdef USE_OPENGL
       ifeq ($(findstring iupluagl, $(LIBNAME)), )
-        SLIB += $(IUPLUA_LIB)/libiupluagl$(LIBLUA_SFX).a
+        SLIB += $(IUPLUA_LIB)/libiupluagl.a
       endif
     endif
-    SLIB += $(IUPLUA_LIB)/libiuplua$(LIBLUA_SFX).a
+    SLIB += $(IUPLUA_LIB)/libiuplua.a
   else
     ifdef USE_CD
       ifeq ($(findstring iupluacd, $(LIBNAME)), )
-        LIBS += iupluacd$(LIBLUA_SFX)
+        LIBS += iupluacd
       endif
     endif
     ifdef USE_OPENGL
       ifeq ($(findstring iupluagl, $(LIBNAME)), )
-        LIBS += iupluagl$(LIBLUA_SFX)
+        LIBS += iupluagl
       endif
     endif
-    LIBS += iuplua$(LIBLUA_SFX)
+    LIBS += iuplua
     LDIR += $(IUPLUA_LIB)
   endif
 endif
