From 448b236d893170edc8fc4dff029044948516352b Mon Sep 17 00:00:00 2001
From: Rudi Heitbaum <rudi@heitbaum.com>
Date: Sun, 27 Apr 2025 15:03:59 +1000
Subject: [PATCH] Allow build with gcc-15 by setting -std=gnu17

---
 kernel-open/Kbuild | 1 +
 1 file changed, 1 insertion(+)

diff --git a/kernel-open/Kbuild b/kernel-open/Kbuild
index 44f6e55d5..09373cf31 100644
--- a/kernel-open/Kbuild
+++ b/kernel-open/Kbuild
@@ -87,6 +87,7 @@ EXTRA_CFLAGS += -I$(src)
 EXTRA_CFLAGS += -Wall $(DEFINES) $(INCLUDES) -Wno-cast-qual -Wno-format-extra-args
 EXTRA_CFLAGS += -D__KERNEL__ -DMODULE -DNVRM
 EXTRA_CFLAGS += -DNV_VERSION_STRING=\"570.123.19\"
+EXTRA_CFLAGS += -std=gnu17
 
 ifneq ($(SYSSRCHOST1X),)
  EXTRA_CFLAGS += -I$(SYSSRCHOST1X)
