diff --git a/meson.build b/meson.build
index bf51b7d..26ff0d5 100755
--- a/meson.build
+++ b/meson.build
@@ -57,7 +57,6 @@ executable(
         dependency('json-glib-1.0'),
         dependency('vte-2.91', version: '>0.52'),
         dependency('gee-0.8'),
-        dependency('unity'),
         meson.get_compiler('c').find_library('m', required : false)
     ],
     install: true
diff --git a/src/Views/SourceListView.vala b/src/Views/SourceListView.vala
index 5afa8e0..230e91c 100755
--- a/src/Views/SourceListView.vala
+++ b/src/Views/SourceListView.vala
@@ -125,7 +125,6 @@ namespace EasySSH {
                         window.current_terminal = (TerminalWidget)box.term;
                         window.current_terminal.grab_focus();
                         box.set_selected();
-                        box.remove_badge ();
                     }
                 }
                 var all_read = true;
@@ -226,7 +225,6 @@ namespace EasySSH {
             var box = (TerminalBox)tab.page;
             window.current_terminal = t;
             box.set_selected();
-            box.remove_badge ();
         }
         private void on_tab_switched (Granite.Widgets.Tab? old_tab, Granite.Widgets.Tab new_tab) {
             if(Type.from_instance(new_tab.page).name() == "EasySSHTerminalBox") {
@@ -234,7 +232,6 @@ namespace EasySSH {
                 window.current_terminal = t;
                 var box = (TerminalBox)new_tab.page;
                 box.set_selected();
-                box.remove_badge ();
                 box.dataHost.item.icon = null;
                 new_tab.icon = null;
                 var all_read = true;
@@ -468,4 +465,4 @@ namespace EasySSH {
             message_dialog.destroy ();
         }
     }
-}
\ No newline at end of file
+}
diff --git a/src/Widgets/TerminalBox.vala b/src/Widgets/TerminalBox.vala
index 643d889..004c7ad 100755
--- a/src/Widgets/TerminalBox.vala
+++ b/src/Widgets/TerminalBox.vala
@@ -80,34 +80,12 @@ namespace EasySSH {
             term.feed_child(cmd.to_utf8 ());
         }
 
-        public void add_badge (){
-            var entry = Unity.LauncherEntry.get_for_desktop_id ("com.github.muriloventuroso.easyssh.desktop");
-            entry.count_visible = true;
-            entry.count = entry.count + 1;
-            unread_changes = true;
-            dataHost.item.icon = new GLib.ThemedIcon ("mail-mark-important");
-            tab.icon = new GLib.ThemedIcon ("mail-mark-important");
-        }
-        public void remove_badge (){
-            var entry = Unity.LauncherEntry.get_for_desktop_id ("com.github.muriloventuroso.easyssh.desktop");
-            if(entry.count_visible == true){
-                entry.count = entry.count - 1;
-                if(entry.count == 0){
-                    entry.count_visible = false;
-                }
-            }
-            unread_changes = false;
-            dataHost.item.icon = new GLib.ThemedIcon ("mail-mark-important");
-            tab.icon = null;
-        }
-
         public void on_change_terminal (Vte.Terminal terminal) {
             string? res = terminal.get_text(null, null);
             if(res != null) {
                 string[] lines = res.split("\n");
                 string[] ret = {};
                 if(term != window.current_terminal && unread_changes == false) {
-                    add_badge ();
                 }
                 foreach (unowned string str in lines) {
                     if(str != "") {
@@ -232,4 +210,4 @@ namespace EasySSH {
 
 
     }
-}
\ No newline at end of file
+}
