diff --git a/meson.build b/meson.build
index 3d4785e..d1c90dc 100755
--- a/meson.build
+++ b/meson.build
@@ -43,11 +43,6 @@ dependencies = [
     dependency('gee-0.8'),
     meson.get_compiler('c').find_library('m', required : false)
 ]
-if get_option('libunity')
-    unity_dep = dependency('unity')
-    add_project_arguments('--define=UNITY_SUPPORT=1', language : 'vala')
-    dependencies += unity_dep
-endif
 executable(
     meson.project_name(),
     code_resources,
