diff --git a/src/Widgets/TerminalBox.vala b/src/Widgets/TerminalBox.vala
index d1ae394a..8594fdcc 100755
--- a/src/Widgets/TerminalBox.vala
+++ b/src/Widgets/TerminalBox.vala
@@ -100,7 +100,7 @@ namespace EasySSH {
             #if UBUNTU_BIONIC_PATCHED_VTE
                 term.feed_child(cmd, cmd.length);
             #else
-                term.feed_child(cmd.to_utf8 ());
+                term.feed_child((uint8[])cmd.to_utf8 ());
             #endif
         }
 
@@ -198,7 +198,7 @@ namespace EasySSH {
             #if UBUNTU_BIONIC_PATCHED_VTE
                 term.feed_child(cmd, cmd.length);
             #else
-                term.feed_child(cmd.to_utf8 ());
+                term.feed_child((uint8[])cmd.to_utf8 ());
             #endif
         }
 
@@ -207,7 +207,7 @@ namespace EasySSH {
             #if UBUNTU_BIONIC_PATCHED_VTE
                 term.feed_child(n_cmd, cmd.length);
             #else
-                term.feed_child(n_cmd.to_utf8 ());
+                term.feed_child((uint8[])n_cmd.to_utf8 ());
             #endif
         }
 
diff --git a/src/Widgets/TerminalWidget.vala b/src/Widgets/TerminalWidget.vala
index 0efc6945..f8784e08 100755
--- a/src/Widgets/TerminalWidget.vala
+++ b/src/Widgets/TerminalWidget.vala
@@ -216,7 +216,7 @@ namespace EasySSH {
             #if UBUNTU_BIONIC_PATCHED_VTE
                 this.feed_child(cmd, cmd.length);
             #else
-                this.feed_child(cmd.to_utf8 ());
+                this.feed_child((uint8[])cmd.to_utf8 ());
             #endif
         }
 
