From ffac473657e2659d314bc6095ddd175d9023f8d1 Mon Sep 17 00:00:00 2001
From: Gavin Zhao <git@gzgz.dev>
Date: Sun, 24 Dec 2023 12:00:05 -0500
Subject: [PATCH] Remove isystem include paths

Signed-off-by: Gavin Zhao <git@gzgz.dev>
---
 bin/hipcc.pl | 4 ----
 1 file changed, 4 deletions(-)

diff --git a/bin/hipcc.pl b/bin/hipcc.pl
index 513a427..53b6222 100755
--- a/bin/hipcc.pl
+++ b/bin/hipcc.pl
@@ -218,10 +218,6 @@ if ($HIP_PLATFORM eq "amd") {
     exit (-1);
 }
 
-# Add paths to common HIP includes:
-$HIPCXXFLAGS .= " -isystem " . get_normalized_path("$HIP_INCLUDE_PATH");
-$HIPCFLAGS .= " -isystem " . get_normalized_path("$HIP_INCLUDE_PATH");
-
 my $compileOnly = 0;
 my $needCXXFLAGS = 0;  # need to add CXX flags to compile step
 my $needCFLAGS = 0;    # need to add C flags to compile step
-- 
2.43.0

