From 2240d06d7694a1da1f1c16794a81d7f1b1c3b667 Mon Sep 17 00:00:00 2001
From: Antonio Rojas <arojas@archlinux.org>
Date: Sat, 25 Nov 2023 00:04:19 +0100
Subject: [PATCH] Actually use QGpgme if found

---
 isoimagewriter/CMakeLists.txt | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/isoimagewriter/CMakeLists.txt b/isoimagewriter/CMakeLists.txt
index 9e88217..a1f7788 100644
--- a/isoimagewriter/CMakeLists.txt
+++ b/isoimagewriter/CMakeLists.txt
@@ -81,7 +81,7 @@ target_link_libraries(isoimagewriter
     KF6::Archive
     KF6::Solid
 )
-if(QGpgme_FOUND)
+if(QGpgmeQt6_FOUND)
     add_definitions(-D_USE_GPG)
     target_link_libraries(isoimagewriter QGpgmeQt6)
 endif()
-- 
GitLab

